{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# renaming the reactions according to columns and rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'WesternDietMicrobiomes/escherMap.json'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn [225], line 4\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mmath\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m isclose\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mjson\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m load, dump\n\u001b[0;32m----> 4\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mopen\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mWesternDietMicrobiomes/escherMap.json\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m jsonIn:    escherMap \u001b[38;5;241m=\u001b[39m load(jsonIn)\n\u001b[1;32m      6\u001b[0m \u001b[38;5;66;03m# display(escherMap)\u001b[39;00m\n\u001b[1;32m      8\u001b[0m xs, ys \u001b[38;5;241m=\u001b[39m [], []\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'WesternDietMicrobiomes/escherMap.json'"
     ]
    }
   ],
   "source": [
    "from math import isclose\n",
    "from json import load, dump\n",
    "\n",
    "with open(\"WesternDietMicrobiomes/escherMap.json\", \"r\") as jsonIn:    escherMap = load(jsonIn)\n",
    "\n",
    "# display(escherMap)\n",
    "\n",
    "xs, ys = [], []\n",
    "for index, lst in enumerate(escherMap):\n",
    "    for k, content in lst.items():\n",
    "        if k != \"reactions\":  continue\n",
    "        print(content)\n",
    "        for rxn, content2 in content.items():\n",
    "            if \"label_x\" not in content2:  print(content2) ; continue\n",
    "            if not any([isclose(content2[\"label_x\"], x, abs_tol=50) for x in xs]):   xs.append(content2[\"label_x\"])\n",
    "            if not any([isclose(content2[\"label_y\"], y, abs_tol=50) for y in ys]):   ys.append(content2[\"label_y\"])\n",
    "\n",
    "print(xs, ys)\n",
    "\n",
    "for index, lst in enumerate(escherMap):\n",
    "    for k, content in lst.items():\n",
    "        if k != \"reactions\":  continue\n",
    "        for rxn, content2 in content.items():\n",
    "            for x in xs:\n",
    "                if isclose(content2[\"label_x\"], x, abs_tol=50):\n",
    "                    col_index = xs.index(x)\n",
    "                    continue\n",
    "            for y in ys:\n",
    "                if isclose(content2[\"label_y\"], y, abs_tol=50):\n",
    "                    row_index = ys.index(y)\n",
    "                    continue\n",
    "            for attribute, val in content2.items():\n",
    "                if attribute == \"metabolites\":\n",
    "                    metabolite_dics = []\n",
    "                    for metabolite in val:\n",
    "                        metabolite_dic = {}\n",
    "                        for cpd_index, (k2, val2) in enumerate(metabolite.items()):\n",
    "                            if k2 == \"bigg_id\":  metabolite_dic[k2] = f\"{val2}_{col_index}_{row_index}_{cpd_index}\"\n",
    "                            elif k2 == \"coefficient\":  metabolite_dic[k2] = val2\n",
    "                        metabolite_dics.append(metabolite_dic)\n",
    "                    escherMap[index][k][rxn][attribute] = metabolite_dics\n",
    "                    print(escherMap[index][k][rxn][attribute])\n",
    "                if attribute == \"bigg_id\":   escherMap[index][k][rxn][attribute] = f\"{val}_{col_index}_{row_index}\"\n",
    "                # if attribute == \"metabolites\":\n",
    "                    # escherMap[index][k][rxn][attribute] = val+\n",
    "                # print(attribute, val)\n",
    "\n",
    "\n",
    "with open(\"WesternDietMicrobiomes/escherMap_edited.json\", \"w\") as jsonOut:    escherMap = dump(escherMap, jsonOut, indent=3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# duplicating reactions across a specified number of rows and columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3600.0"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "2961.18779884681--638.8122011531899"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\n",
      "800 400\n",
      "rows\n",
      "800 400\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "dict_keys(['name', 'bigg_id', 'reversibility', 'label_x', 'label_y', 'gene_reaction_rule', 'genes', 'metabolites', 'segments'])"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "dict_keys(['name', 'bigg_id', 'reversibility', 'label_x', 'label_y', 'gene_reaction_rule', 'genes', 'metabolites', 'segments'])"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rows\n",
      "800 400\n",
      "dict_keys(['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22'])\n",
      "dict_keys(['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22'])\n"
     ]
    }
   ],
   "source": [
    "from json import load, dump\n",
    "\n",
    "def duplicate_rxn(escherJSON_path, rows=None, columns=None, row_step_size=800, col_step_size=400):\n",
    "    assert rows is not None or columns is not None, \"What are you doing?\"\n",
    "\n",
    "    with open(escherJSON_path, \"r\") as jsonIn:\n",
    "        escherMap = load(jsonIn)\n",
    "\n",
    "    # display(escherMap)\n",
    "    print(len(escherMap))\n",
    "\n",
    "    print(row_step_size, col_step_size)\n",
    "\n",
    "    for i, lst in enumerate(escherMap):\n",
    "        if i == 0:  continue\n",
    "        for k, content in lst.items():\n",
    "            if k == \"reactions\":\n",
    "                # row reaction development\n",
    "                print(\"rows\")\n",
    "                print(row_step_size, col_step_size)\n",
    "                updated_reactions = {} if \"updated_reactions\" not in locals() else updated_reactions\n",
    "                if isinstance(rows, int):\n",
    "                    row_reactions = {k:v for k,v in content.items()}\n",
    "                    for row_index in range(1, rows):\n",
    "                        # print(content.keys())\n",
    "                        for rxnNum, content2 in content.items():\n",
    "                            nextRxn = str(int(rxnNum)+1)\n",
    "                            ## define the content of this reaction\n",
    "                            new_content = {k:v for k,v in content2.items()}\n",
    "                            ## translate the label\n",
    "                            new_content[\"label_y\"] += row_step_size*row_index\n",
    "                            ## rename the reaction according to its new translation\n",
    "                            rxnName_content = new_content[\"bigg_id\"].split(\"_\")\n",
    "                            colNum, rowNum = rxnName_content[1], rxnName_content[2]\n",
    "                            new_content[\"bigg_id\"] = f\"{rxnName_content[0]}_{colNum}_{str(int(rowNum)+1)}\"\n",
    "                            ## rename the metabolites\n",
    "                            for index, metabolite in enumerate(new_content[\"metabolites\"]):\n",
    "                                ### rename the metabolites\n",
    "                                ID_content = metabolite[\"bigg_id\"].split(\"_\")\n",
    "                                metabolite[\"bigg_id\"] = f\"{ID_content[0]}_{colNum}_{rowNum}_{index}\"\n",
    "                            ## update the metabolite nodes\n",
    "                            display(new_content.keys())\n",
    "                            ## update the edge segments\n",
    "                            for num, content3 in new_content[\"segments\"].items():\n",
    "                                dictionary_values = [value for value in content3.values() if isinstance(value, dict)]\n",
    "                                # print(dictionary_values)\n",
    "                                # for obj, value in content3.items():\n",
    "                                #     # print(list(map(len, [new_content, content3])))\n",
    "                                #     if isinstance(value, dict):\n",
    "                                #         print(rxnNum, num, obj, value) ;  new_content[\"segments\"][num][obj][\"y\"] += row_step_size*row_index\n",
    "                                #     if isinstance(value, dict):\n",
    "                                #         print(new_content[\"segments\"][num][obj])\n",
    "                            row_reactions[str(int(rxnNum)+row_index)] = new_content\n",
    "                            # print(row_reactions[str(int(rxn)+row_index)][\"segments\"][\"312\"][\"b2\"])\n",
    "                        updated_reactions.update({k:v for k,v in row_reactions.items()})\n",
    "                escherMap[i][k] = updated_reactions  \n",
    "            elif k == \"nodes\":\n",
    "                # row metabolite development\n",
    "                print(\"rows\")\n",
    "                print(row_step_size, col_step_size)\n",
    "                updated_reactions = {} if \"updated_reactions\" not in locals() else updated_reactions\n",
    "                if isinstance(rows, int):\n",
    "                    row_nodes = {k:v for k,v in content.items()}\n",
    "                    for row_index in range(1, rows):\n",
    "                        print(content.keys())\n",
    "                        # updated_reactions = {} if \"updated_reactions\" not in locals() else updated_reactions\n",
    "                        new_mets = {}\n",
    "                        for metNum, content2 in row_nodes.items():\n",
    "                            ### rename the node for this set of reactions\n",
    "                            # print(metNum)\n",
    "                            new_mets[f\"{nextRxn}{metNum}\"] = content2\n",
    "                            ### translating the node locations\n",
    "                            if new_mets[f\"{nextRxn}{metNum}\"][\"node_type\"] in [\"multimarker\", \"midmarker\"]:\n",
    "                                new_mets[f\"{nextRxn}{metNum}\"][\"y\"] += row_step_size*row_index\n",
    "                            elif new_mets[f\"{nextRxn}{metNum}\"][\"node_type\"] == \"metabolite\":\n",
    "                                new_mets[f\"{nextRxn}{metNum}\"][\"y\"] += row_step_size*row_index\n",
    "                                new_mets[f\"{nextRxn}{metNum}\"][\"label_y\"] += row_step_size*row_index\n",
    "                        updated_reactions.update({k:v for k,v in row_reactions.items()})\n",
    "                escherMap[i][k] = updated_reactions  \n",
    "            # else:   print(k)\n",
    "        # if index == len(escherMap)-1:  break\n",
    "        # print(\"columns\")\n",
    "        # print(row_step_size, col_step_size)\n",
    "        # if isinstance(columns, int):\n",
    "        #     reactions = content if rows is not None else updated_reactions\n",
    "        #     col_reactions = {}\n",
    "        #     for col_index in range(columns):\n",
    "        #         for rxnNum, content2 in reactions.items():\n",
    "        #             new_content = {k:v for k,v in content2.items()}\n",
    "        #             new_content[\"label_x\"] += col_step_size*col_index\n",
    "        #             for num, content3 in new_content[\"segments\"].items():\n",
    "        #                 for obj, value in content3.items():\n",
    "        #                     if isinstance(value, dict):  print(rxnNum, num, obj, value) ; new_content[\"segments\"][num][obj][\"x\"] += col_step_size*col_index\n",
    "        #                     if isinstance(value, dict):  print(new_content[\"segments\"][num][obj])\n",
    "        #             col_reactions[str(int(rxnNum)+col_index)] = new_content\n",
    "        #         updated_reactions.update({k:v for k,v in row_reactions.items()})\n",
    "\n",
    "        #     escherMap[index][k] = updated_reactions    \n",
    "\n",
    "    with open(escherJSON_path.replace(\".json\", \"_edited.json\"), \"w\") as jsonOut:  dump(escherMap, jsonOut, indent=3)\n",
    "\n",
    "duplicate_rxn(\"oneRXN_localized2.json\", rows=3, columns=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1200.0\n",
      "1200.0\n"
     ]
    }
   ],
   "source": [
    "print(1596.7619819523752-396.7619819523753)\n",
    "print(2796.761981952375-1596.7619819523752)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from json import load, dump\n",
    "\n",
    "path = \"oneRXN_localized_double.json\"\n",
    "\n",
    "with open(path, \"r\") as jsonIn:\n",
    "    escherMap = load(jsonIn)\n",
    "\n",
    "with open(path, \"w\") as jsonOut:\n",
    "    dump(escherMap, jsonOut, indent=3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## update the coordinates of an existing two-reaction map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'name': 'diphosphate phosphohydrolase',\n",
       " 'bigg_id': 'rxn00001_0_0',\n",
       " 'reversibility': False,\n",
       " 'label_x': -682.8153489006918,\n",
       " 'label_y': 228.44559706520863,\n",
       " 'gene_reaction_rule': '',\n",
       " 'genes': [],\n",
       " 'metabolites': [{'bigg_id': 'cpd_0_0_0', 'coefficient': -1},\n",
       "  {'bigg_id': 'cpd_0_0_1', 'coefficient': -1},\n",
       "  {'bigg_id': 'cpd_0_0_2', 'coefficient': -1},\n",
       "  {'bigg_id': 'cpd_0_0_3', 'coefficient': -1},\n",
       "  {'bigg_id': 'cpd_0_0_4', 'coefficient': -1},\n",
       "  {'bigg_id': 'cpd_0_0_5', 'coefficient': -1},\n",
       "  {'bigg_id': 'cpd_0_0_6', 'coefficient': -1},\n",
       "  {'bigg_id': 'cpd_0_0_7', 'coefficient': -1},\n",
       "  {'bigg_id': 'cpd_0_0_8', 'coefficient': -1},\n",
       "  {'bigg_id': 'cpd_0_0_9', 'coefficient': -1},\n",
       "  {'bigg_id': 'cpd_0_0_10', 'coefficient': 1},\n",
       "  {'bigg_id': 'cpd_0_0_11', 'coefficient': 1},\n",
       "  {'bigg_id': 'cpd_0_0_12', 'coefficient': 1},\n",
       "  {'bigg_id': 'cpd_0_0_13', 'coefficient': 1},\n",
       "  {'bigg_id': 'cpd_0_0_14', 'coefficient': 1},\n",
       "  {'bigg_id': 'cpd_0_0_15', 'coefficient': 1},\n",
       "  {'bigg_id': 'cpd_0_0_16', 'coefficient': 1},\n",
       "  {'bigg_id': 'cpd_0_0_17', 'coefficient': 1},\n",
       "  {'bigg_id': 'cpd_0_0_18', 'coefficient': 1},\n",
       "  {'bigg_id': 'cpd_0_0_19', 'coefficient': 1}],\n",
       " 'segments': {'0': {'from_node_id': '1',\n",
       "   'to_node_id': '2',\n",
       "   'b1': None,\n",
       "   'b2': None},\n",
       "  '1': {'from_node_id': '3', 'to_node_id': '2', 'b1': None, 'b2': None},\n",
       "  '2': {'from_node_id': '1',\n",
       "   'to_node_id': '0',\n",
       "   'b1': {'x': -638.8122011531899, 'y': 126.74002323747482},\n",
       "   'b2': {'x': -814.3927331595246, 'y': -59.37735004709407}},\n",
       "  '21': {'from_node_id': '1',\n",
       "   'to_node_id': '4',\n",
       "   'b1': {'x': -638.8122011531899, 'y': 126.74002323747482},\n",
       "   'b2': {'x': -851.7261377005402, 'y': -34.044021800023756}},\n",
       "  '22': {'from_node_id': '1',\n",
       "   'to_node_id': '5',\n",
       "   'b1': {'x': -638.8122011531899, 'y': 126.74002323747482},\n",
       "   'b2': {'x': -885.0594201712433, 'y': -96.71068973825618}},\n",
       "  '23': {'from_node_id': '1',\n",
       "   'to_node_id': '6',\n",
       "   'b1': {'x': -638.8122011531899, 'y': 126.74002323747482},\n",
       "   'b2': {'x': -983.7261377005402, 'y': -28.710679724675856}},\n",
       "  '24': {'from_node_id': '1',\n",
       "   'to_node_id': '7',\n",
       "   'b1': {'x': -638.8122011531899, 'y': 126.74002323747482},\n",
       "   'b2': {'x': -1013.0594506888215, 'y': 75.28931431485967}},\n",
       "  '25': {'from_node_id': '1',\n",
       "   'to_node_id': '8',\n",
       "   'b1': {'x': -638.8122011531899, 'y': 126.74002323747482},\n",
       "   'b2': {'x': -1030.3927941946808, 'y': 117.95597724630193}},\n",
       "  '26': {'from_node_id': '1',\n",
       "   'to_node_id': '9',\n",
       "   'b1': {'x': -638.8122011531899, 'y': 126.74002323747482},\n",
       "   'b2': {'x': -977.0594201712433, 'y': 163.28932552053288}},\n",
       "  '27': {'from_node_id': '1',\n",
       "   'to_node_id': '10',\n",
       "   'b1': {'x': -638.8122011531899, 'y': 126.74002323747482},\n",
       "   'b2': {'x': -931.7261377005402, 'y': 229.9559934587653}},\n",
       "  '28': {'from_node_id': '1',\n",
       "   'to_node_id': '11',\n",
       "   'b1': {'x': -638.8122011531899, 'y': 126.74002323747482},\n",
       "   'b2': {'x': -910.3927941946808, 'y': 267.28930644704656}},\n",
       "  '29': {'from_node_id': '1',\n",
       "   'to_node_id': '12',\n",
       "   'b1': {'x': -638.8122011531899, 'y': 126.74002323747482},\n",
       "   'b2': {'x': -851.7261377005402, 'y': 245.95597438527898}},\n",
       "  '33': {'from_node_id': '3',\n",
       "   'to_node_id': '13',\n",
       "   'b1': {'x': -514.8185576833539, 'y': 125.48448388122324},\n",
       "   'b2': {'x': -437.90465928297624, 'y': -83.0648222165321}},\n",
       "  '34': {'from_node_id': '3',\n",
       "   'to_node_id': '14',\n",
       "   'b1': {'x': -514.8185576833539, 'y': 125.48448388122324},\n",
       "   'b2': {'x': -355.2380256770192, 'y': -56.39815523197399}},\n",
       "  '35': {'from_node_id': '3',\n",
       "   'to_node_id': '15',\n",
       "   'b1': {'x': -514.8185576833539, 'y': 125.48448388122324},\n",
       "   'b2': {'x': -292.57136155348405, 'y': 2.2684945864464225}},\n",
       "  '36': {'from_node_id': '3',\n",
       "   'to_node_id': '16',\n",
       "   'b1': {'x': -514.8185576833539, 'y': 125.48448388122324},\n",
       "   'b2': {'x': -207.23802186232194, 'y': 30.268513183095592}},\n",
       "  '37': {'from_node_id': '3',\n",
       "   'to_node_id': '17',\n",
       "   'b1': {'x': -514.8185576833539, 'y': 125.48448388122324},\n",
       "   'b2': {'x': -188.57136155348405, 'y': 104.93519304225697}},\n",
       "  '38': {'from_node_id': '3',\n",
       "   'to_node_id': '18',\n",
       "   'b1': {'x': -514.8185576833539, 'y': 125.48448388122324},\n",
       "   'b2': {'x': -155.23799325209245, 'y': 154.26851365993275}},\n",
       "  '39': {'from_node_id': '3',\n",
       "   'to_node_id': '19',\n",
       "   'b1': {'x': -514.8185576833539, 'y': 125.48448388122324},\n",
       "   'b2': {'x': -177.90463639479265, 'y': 199.6018495363976}},\n",
       "  '310': {'from_node_id': '3',\n",
       "   'to_node_id': '20',\n",
       "   'b1': {'x': -514.8185576833539, 'y': 125.48448388122324},\n",
       "   'b2': {'x': -217.90468980055437, 'y': 255.60188768337025}},\n",
       "  '311': {'from_node_id': '3',\n",
       "   'to_node_id': '21',\n",
       "   'b1': {'x': -514.8185576833539, 'y': 125.48448388122324},\n",
       "   'b2': {'x': -323.23806382399187, 'y': 250.26855943629994}},\n",
       "  '312': {'from_node_id': '3',\n",
       "   'to_node_id': '22',\n",
       "   'b1': {'x': -514.8185576833539, 'y': 125.48448388122324},\n",
       "   'b2': {'x': -403.2380180476247, 'y': 283.60184190700306}}}}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from json import load\n",
    "\n",
    "with open(\"twoRXN_localized.json\", 'r') as jsonIn:\n",
    "    EscherMap = load(jsonIn)\n",
    "\n",
    "row_step_size, col_step_size = 800, 400\n",
    "column = 0\n",
    "\n",
    "new_EscherMap = {}\n",
    "for i, ele in enumerate(EscherMap):\n",
    "    if i == 0:\n",
    "        new_EscherMap[i] = ele\n",
    "    new_EscherMap[i] = {}\n",
    "    for k, content in ele.items():\n",
    "        # print(k)\n",
    "        if k == \"reactions\":\n",
    "            new_EscherMap[i][\"reactions\"] = {}\n",
    "            for rxnNum, content2 in content.items():\n",
    "                if rxnNum == 0:\n",
    "                    new_EscherMap[i][\"reactions\"][rxnNum] = content2\n",
    "                    continue\n",
    "                for k2, val in content2.items():\n",
    "                    if k == \"bigg_id\":\n",
    "                        baseRXN = val.split(\"_\")[0]\n",
    "                        new_EscherMap[i][\"reactions\"][rxnNum][k2] = f\"{baseRXN}_{column}_{rxnNum}\"\n",
    "                    elif k == \"label_x\":\n",
    "                        new_EscherMap[i][\"reactions\"][rxnNum][k2] = val + column*col_step_size\n",
    "                    elif k == \"label_y\":\n",
    "                        new_EscherMap[i][\"reactions\"][rxnNum][k2] = val + rxnNum*row_step_size\n",
    "                    elif k2 == \"metabolites\":\n",
    "                        new_mets = []\n",
    "                        for cpdNum, cpd in enumerate(val):\n",
    "                            cpd[\"bigg_id\"] = f\"cpd_{column}_{rxnNum}_{cpdNum}\"\n",
    "                            new_mets.append(cpd)\n",
    "                        new_EscherMap[i][\"reactions\"][rxnNum][k2] = new_mets\n",
    "                    elif k2 == \"segments\":\n",
    "                        for segNum, content3 in val.items():\n",
    "                            if isinstance(content3[\"b1\"], dict):\n",
    "                                new_EscherMap[i][\"reactions\"][rxnNum][k2][segNum][]\n",
    "                    \n",
    "\n",
    "\n",
    "                display(content2)\n",
    "                break\n",
    "\n",
    "        elif k == \"nodes\":\n",
    "            pass\n",
    "        elif k == \"text_labels\":\n",
    "            pass\n",
    "        elif k == \"canvas\":\n",
    "            pass"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# process the ASVMetaboliteInteractions community interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## parse the stoichiometric relationships"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import read_csv\n",
    "from json import dump\n",
    "\n",
    "interactions = read_csv(\"ASVMetaboliteInteractions.csv\").set_index(\"ASV\")\n",
    "stoich_mapping = {}\n",
    "for index, row in interactions.iterrows():\n",
    "    stoich_mapping[index] = {\"reactants\":{}, \"products\":{}}\n",
    "    for col, ele in row.items():\n",
    "        metID = col.split(\"_\")[0]\n",
    "        if ele > 0:   stoich_mapping[index][\"products\"][metID] = ele\n",
    "        elif ele < 0:   stoich_mapping[index][\"reactants\"][metID] = ele\n",
    "\n",
    "# display(stoich_mapping)\n",
    "with open(\"ASV_metabolic_activity_stoich.json\", 'w') as jsonOut:\n",
    "    dump(stoich_mapping, jsonOut, indent=3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# parse the model for content and interactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Simple_carbs': {'-1.5': {'RC': 23, 'WD': 17},\n",
       "  '1.5': {'RC': 20, 'WD': 4},\n",
       "  '4.0': {'RC': 12, 'WD': 0},\n",
       "  '6.0': {'RC': 8, 'WD': 8},\n",
       "  '9.0': {'RC': 18, 'WD': 13},\n",
       "  '12.5': {'RC': 20, 'WD': 11}},\n",
       " 'Complex_carbs': {'-1.5': {'RC': 17, 'WD': 17},\n",
       "  '1.5': {'RC': 20, 'WD': 4},\n",
       "  '4.0': {'RC': 9, 'WD': 0},\n",
       "  '6.0': {'RC': 12, 'WD': 8},\n",
       "  '9.0': {'RC': 20, 'WD': 12},\n",
       "  '12.5': {'RC': 19, 'WD': 10}},\n",
       " 'SCFAs': {'-1.5': {'RC': 24, 'WD': 17},\n",
       "  '1.5': {'RC': 18, 'WD': 4},\n",
       "  '4.0': {'RC': 11, 'WD': 0},\n",
       "  '6.0': {'RC': 9, 'WD': 8},\n",
       "  '9.0': {'RC': 14, 'WD': 12},\n",
       "  '12.5': {'RC': 16, 'WD': 7}},\n",
       " 'Fatty Acids': {'-1.5': {'RC': 22, 'WD': 17},\n",
       "  '1.5': {'RC': 20, 'WD': 4},\n",
       "  '4.0': {'RC': 12, 'WD': 0},\n",
       "  '6.0': {'RC': 14, 'WD': 8},\n",
       "  '9.0': {'RC': 20, 'WD': 12},\n",
       "  '12.5': {'RC': 21, 'WD': 11}},\n",
       " 'Amino Acids': {'-1.5': {'RC': 25, 'WD': 17},\n",
       "  '1.5': {'RC': 20, 'WD': 4},\n",
       "  '4.0': {'RC': 12, 'WD': 0},\n",
       "  '6.0': {'RC': 14, 'WD': 8},\n",
       "  '9.0': {'RC': 20, 'WD': 13},\n",
       "  '12.5': {'RC': 21, 'WD': 11}},\n",
       " 'Simple_carbon': {'-1.5': {'RC': 24, 'WD': 17},\n",
       "  '1.5': {'RC': 19, 'WD': 4},\n",
       "  '4.0': {'RC': 12, 'WD': 0},\n",
       "  '6.0': {'RC': 10, 'WD': 8},\n",
       "  '9.0': {'RC': 19, 'WD': 13},\n",
       "  '12.5': {'RC': 19, 'WD': 10}},\n",
       " 'Other': {'-1.5': {'RC': 25, 'WD': 17},\n",
       "  '1.5': {'RC': 20, 'WD': 4},\n",
       "  '4.0': {'RC': 12, 'WD': 0},\n",
       "  '6.0': {'RC': 14, 'WD': 8},\n",
       "  '9.0': {'RC': 20, 'WD': 13},\n",
       "  '12.5': {'RC': 21, 'WD': 11}}}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Simple_carbs</th>\n",
       "      <th>Complex_carbs</th>\n",
       "      <th>SCFAs</th>\n",
       "      <th>Fatty Acids</th>\n",
       "      <th>Amino Acids</th>\n",
       "      <th>Simple_carbon</th>\n",
       "      <th>Other</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Interval Day</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>-1.5</th>\n",
       "      <td>{'RC': 23, 'WD': 17}</td>\n",
       "      <td>{'RC': 17, 'WD': 17}</td>\n",
       "      <td>{'RC': 24, 'WD': 17}</td>\n",
       "      <td>{'RC': 22, 'WD': 17}</td>\n",
       "      <td>{'RC': 25, 'WD': 17}</td>\n",
       "      <td>{'RC': 24, 'WD': 17}</td>\n",
       "      <td>{'RC': 25, 'WD': 17}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1.5</th>\n",
       "      <td>{'RC': 20, 'WD': 4}</td>\n",
       "      <td>{'RC': 20, 'WD': 4}</td>\n",
       "      <td>{'RC': 18, 'WD': 4}</td>\n",
       "      <td>{'RC': 20, 'WD': 4}</td>\n",
       "      <td>{'RC': 20, 'WD': 4}</td>\n",
       "      <td>{'RC': 19, 'WD': 4}</td>\n",
       "      <td>{'RC': 20, 'WD': 4}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4.0</th>\n",
       "      <td>{'RC': 12, 'WD': 0}</td>\n",
       "      <td>{'RC': 9, 'WD': 0}</td>\n",
       "      <td>{'RC': 11, 'WD': 0}</td>\n",
       "      <td>{'RC': 12, 'WD': 0}</td>\n",
       "      <td>{'RC': 12, 'WD': 0}</td>\n",
       "      <td>{'RC': 12, 'WD': 0}</td>\n",
       "      <td>{'RC': 12, 'WD': 0}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6.0</th>\n",
       "      <td>{'RC': 8, 'WD': 8}</td>\n",
       "      <td>{'RC': 12, 'WD': 8}</td>\n",
       "      <td>{'RC': 9, 'WD': 8}</td>\n",
       "      <td>{'RC': 14, 'WD': 8}</td>\n",
       "      <td>{'RC': 14, 'WD': 8}</td>\n",
       "      <td>{'RC': 10, 'WD': 8}</td>\n",
       "      <td>{'RC': 14, 'WD': 8}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9.0</th>\n",
       "      <td>{'RC': 18, 'WD': 13}</td>\n",
       "      <td>{'RC': 20, 'WD': 12}</td>\n",
       "      <td>{'RC': 14, 'WD': 12}</td>\n",
       "      <td>{'RC': 20, 'WD': 12}</td>\n",
       "      <td>{'RC': 20, 'WD': 13}</td>\n",
       "      <td>{'RC': 19, 'WD': 13}</td>\n",
       "      <td>{'RC': 20, 'WD': 13}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12.5</th>\n",
       "      <td>{'RC': 20, 'WD': 11}</td>\n",
       "      <td>{'RC': 19, 'WD': 10}</td>\n",
       "      <td>{'RC': 16, 'WD': 7}</td>\n",
       "      <td>{'RC': 21, 'WD': 11}</td>\n",
       "      <td>{'RC': 21, 'WD': 11}</td>\n",
       "      <td>{'RC': 19, 'WD': 10}</td>\n",
       "      <td>{'RC': 21, 'WD': 11}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Simple_carbs         Complex_carbs  \\\n",
       "Interval Day                                               \n",
       "-1.5          {'RC': 23, 'WD': 17}  {'RC': 17, 'WD': 17}   \n",
       "1.5            {'RC': 20, 'WD': 4}   {'RC': 20, 'WD': 4}   \n",
       "4.0            {'RC': 12, 'WD': 0}    {'RC': 9, 'WD': 0}   \n",
       "6.0             {'RC': 8, 'WD': 8}   {'RC': 12, 'WD': 8}   \n",
       "9.0           {'RC': 18, 'WD': 13}  {'RC': 20, 'WD': 12}   \n",
       "12.5          {'RC': 20, 'WD': 11}  {'RC': 19, 'WD': 10}   \n",
       "\n",
       "                             SCFAs           Fatty Acids  \\\n",
       "Interval Day                                               \n",
       "-1.5          {'RC': 24, 'WD': 17}  {'RC': 22, 'WD': 17}   \n",
       "1.5            {'RC': 18, 'WD': 4}   {'RC': 20, 'WD': 4}   \n",
       "4.0            {'RC': 11, 'WD': 0}   {'RC': 12, 'WD': 0}   \n",
       "6.0             {'RC': 9, 'WD': 8}   {'RC': 14, 'WD': 8}   \n",
       "9.0           {'RC': 14, 'WD': 12}  {'RC': 20, 'WD': 12}   \n",
       "12.5           {'RC': 16, 'WD': 7}  {'RC': 21, 'WD': 11}   \n",
       "\n",
       "                       Amino Acids         Simple_carbon                 Other  \n",
       "Interval Day                                                                    \n",
       "-1.5          {'RC': 25, 'WD': 17}  {'RC': 24, 'WD': 17}  {'RC': 25, 'WD': 17}  \n",
       "1.5            {'RC': 20, 'WD': 4}   {'RC': 19, 'WD': 4}   {'RC': 20, 'WD': 4}  \n",
       "4.0            {'RC': 12, 'WD': 0}   {'RC': 12, 'WD': 0}   {'RC': 12, 'WD': 0}  \n",
       "6.0            {'RC': 14, 'WD': 8}   {'RC': 10, 'WD': 8}   {'RC': 14, 'WD': 8}  \n",
       "9.0           {'RC': 20, 'WD': 13}  {'RC': 19, 'WD': 13}  {'RC': 20, 'WD': 13}  \n",
       "12.5          {'RC': 21, 'WD': 11}  {'RC': 19, 'WD': 10}  {'RC': 21, 'WD': 11}  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from cobra.io import load_json_model\n",
    "from optlang import Objective\n",
    "\n",
    "effectiveASV_model = load_json_model(\"ASVInteractionModel.json\")\n",
    "rxnNames = [rxn.id for rxn in effectiveASV_model.reactions]\n",
    "rcNames = {}\n",
    "numRCs = []\n",
    "for name in rxnNames:\n",
    "    if \"RC\" in name:\n",
    "        nameParts = name.split(\"RC\")\n",
    "        numRCs.append(nameParts[0])\n",
    "        rcNames[nameParts[0]] = rcNames.get(nameParts[0], [])\n",
    "        rcNames[nameParts[0]].append(nameParts[1].split(\"_\")[1])\n",
    "# print(set(rcNames.keys()) - set(numRCs))\n",
    "wdNames = {}\n",
    "numWDs = []\n",
    "for name in rxnNames:\n",
    "    if \"WD\" in name:\n",
    "        nameParts = name.split(\"WD\")\n",
    "        numWDs.append(nameParts[0])\n",
    "        wdNames[nameParts[0]] = wdNames.get(nameParts[0], [])\n",
    "        wdNames[nameParts[0]].append(nameParts[1].split(\"_\")[1])\n",
    "# print(set(wdNames.keys()) - set(numWDs))\n",
    "new_rcNames = [f\"{k}RC-ABX_{time}\" for k,v in rcNames.items() for time in v]\n",
    "new_wdNames = [f\"{k}WD-ABX_{time}\" for k,v in wdNames.items() for time in v]\n",
    "\n",
    "missing_keys = set(rxnNames) - set(new_rcNames+new_wdNames)\n",
    "# print(missing_keys)\n",
    "\n",
    "\n",
    "#define the categories of compounds that will be examined\n",
    "compounds = {\"Simple_carbs\": [\"Dulcose\", \"Galactose\", \"D-Fructose\", \"D-Allose\", \"L-Arabinose\", \"D-Glucose\", \"D-Ribose\",\n",
    "                               \"L-Inositol\", \"Sucrose\", \"L-Fucose\", \"Thyminose\", \"L-Rhamnose\"],\n",
    "             \"Complex_carbs\": [\"Melibiose\", \"Melitose\", \"CELB\"],\n",
    "             \"SCFAs\": [\"Acetate\", \"Propionate\", \"Butyrate\"],\n",
    "             \"Fatty Acids\": [\"Decanoate\", \"Myristic acid\", \"Palmitate\"],\n",
    "             \"Amino Acids\": [\"L-Methionine\", \"L-Histidine\", \"L-Tryptophan\", \"L-Alanine\", \"L-Threonine\", \"L-Glutamate\", \"L-Isoleucine\", \"L-Proline\", \"L-Leucine\", \"Glycine\", 'L-Phenylalanine',\n",
    "                             'L-Tyrosine', 'L-Valine', 'L-Glutamine', 'L-Arginine', 'L-Asparagine', 'L-Aspartate', 'L-Lysine', 'L-Serine', 'L-Cysteine', \"D-Serine\", \"5-Oxoproline\"],\n",
    "             \"Simple_carbon\":[\"Succinate\", \"Citrate\", \"L-Malate\", \"Pyruvate\"],\n",
    "             \"Other\": [\"N-Acetyl-D-glucosamine\", \"Ornithine\", \"Fumarate\", \"trans-4-Hydroxy-L-proline\", \"2-Oxoglutarate\", \"Niacin\"]}\n",
    "\n",
    "\n",
    "def search_days(day):\n",
    "    RCasvs = [asv for asv, days in rcNames.items() if day in days]\n",
    "    WDasvs = [asv for asv, days in wdNames.items() if day in days]\n",
    "    display(\"RC\", sorted(RCasvs))\n",
    "    display(\"WD\", sorted(WDasvs))\n",
    "\n",
    "def biggest_engager(day, diet, compound, all=False, limit=1):\n",
    "    print(compound)\n",
    "    if all:\n",
    "        for rxn in effectiveASV_model.reactions:\n",
    "            if day in rxn.id and diet in rxn.id:\n",
    "                for met, coef in rxn.metabolites.items():\n",
    "                    # print(met.name, coef)\n",
    "                    if compound == met.name and coef > limit:\n",
    "                        print(rxn.name)\n",
    "\n",
    "    maxFlux = [None, 0]\n",
    "    minFlux = [None, 0]\n",
    "    for rxn in effectiveASV_model.reactions:\n",
    "        if day in rxn.id and diet in rxn.id:\n",
    "            for met, coef in rxn.metabolites.items():\n",
    "                # print(met.name, coef)\n",
    "                if compound == met.name:\n",
    "                    if coef > maxFlux[1]:    maxFlux = [rxn.name, coef]\n",
    "                    elif coef < minFlux[1]:  minFlux = [rxn.name, coef]\n",
    "    if maxFlux != [None, 0]:  print(\"maxFlux\", maxFlux)\n",
    "    if minFlux != [None, 0]:  print(\"minFlux\", minFlux)\n",
    "\n",
    "def full_assessment(day, diet, compounds, engagers=True, classification=True, skipping_mets=None):\n",
    "    print(f\"metabolite assessment for {day} on the {diet} diet\")\n",
    "    #determine the biggest consumers/producers of each compound\n",
    "    if engagers:\n",
    "        for category, metabolites in compounds.items():\n",
    "            print(category)\n",
    "            biggest_engager(day, diet, metabolites)\n",
    "\n",
    "    #determine the compounds that exclusively consumed/produced or both\n",
    "    if classification:\n",
    "        metabolites = {\"consumed\": set(), \"produced\": set(), \"intermediates\": set()}\n",
    "        for rxn in effectiveASV_model.reactions:\n",
    "            if \"_\"+day in rxn.id and diet in rxn.id:\n",
    "                for met, coef in rxn.metabolites.items():\n",
    "                    if skipping_mets and met.name in skipping_mets:  continue\n",
    "                    if abs(coef) <= 0.05:   continue\n",
    "                    # print(met.name, coef)\n",
    "                    if coef > 0:\n",
    "                        metabolites[\"produced\"].add(met.name)\n",
    "                        if met.name in metabolites[\"consumed\"]:\n",
    "                            # print(\"produced\", met.name, rxn.name)\n",
    "                            metabolites[\"intermediates\"].add(met.name)\n",
    "                    elif coef < 0:\n",
    "                        metabolites[\"consumed\"].add(met.name)\n",
    "                        # print(\"consumed\", met.name, rxn.name)\n",
    "                        if met.name in metabolites[\"produced\"]:\n",
    "                            # print(\"consumed\", met.name, rxn.name)\n",
    "                            metabolites[\"intermediates\"].add(met.name)\n",
    "        metabolites[\"consumed\"] -= metabolites[\"intermediates\"]\n",
    "        metabolites[\"produced\"] -= metabolites[\"intermediates\"]\n",
    "        display(metabolites)\n",
    "\n",
    "\n",
    "\n",
    "    \n",
    "def count_interactions(compounds):\n",
    "    from itertools import product\n",
    "    days, diets = [\"-1.5\", \"1.5\", \"4.0\", \"6.0\", \"9.0\", \"12.5\"], [\"RC\", \"WD\"]\n",
    "    Interactions = {k:dict() for k in compounds}\n",
    "    categories = {ele:k for k,v in compounds.items() for ele in v}\n",
    "    for day, diet in product(days, diets):\n",
    "        for category in Interactions:\n",
    "            if day in Interactions[category]:  Interactions[category][day].update({diet:set()})\n",
    "            else:                    Interactions[category][day] = {diet:set()}\n",
    "        # for category in InteractionCounts:\n",
    "        #     if day in InteractionCounts[category]:  InteractionCounts[category][day].update({diet:set()})\n",
    "        #     else:                    InteractionCounts[category][day] = {diet:set()}\n",
    "    for day, diet in product(days, diets):\n",
    "        produced, consumed = set(), set()\n",
    "        for rxn in effectiveASV_model.reactions:\n",
    "            metNames = [met.name for met in rxn.metabolites]\n",
    "            # print(metNames)\n",
    "            for metName in metNames:\n",
    "                category = categories[metName]\n",
    "                if \"_\"+day in rxn.id and diet in rxn.id:\n",
    "                    for met, coef in rxn.metabolites.items():\n",
    "                        # print(met.name, coef)\n",
    "                        if coef > 0:\n",
    "                            produced.add(met.name)\n",
    "                            if met.name in consumed:\n",
    "                                # print(\"produced\", met.name, rxn.name)\n",
    "                                Interactions[category][day][diet].add(met.name)\n",
    "                        elif coef < 0:\n",
    "                            consumed.add(met.name)\n",
    "                            # print(\"consumed\", met.name, rxn.name)\n",
    "                            if met.name in produced:\n",
    "                                # print(\"consumed\", met.name, rxn.name)\n",
    "                                Interactions[category][day][diet].add(met.name)\n",
    "    # print(aaInteractionsa)\n",
    "    InteractionCounts = {}\n",
    "    for category, content in Interactions.items():\n",
    "        InteractionCounts[category] = {}\n",
    "        for day, dietInfo in content.items():\n",
    "            # print(day, InteractionCounts[category])?\n",
    "            InteractionCounts[category][day] = InteractionCounts[category].get(day, {})\n",
    "            for diet, mets in dietInfo.items():\n",
    "                InteractionCounts[category][day][diet] = InteractionCounts[category][day].get(diet, {diet: 0})\n",
    "                InteractionCounts[category][day][diet] = len(mets)\n",
    "                # print(day, diet, len(mets)), mets)\n",
    "    display(InteractionCounts)\n",
    "\n",
    "    # convert the amino acid counts into a DataFrame\n",
    "    from pandas import DataFrame\n",
    "    df = DataFrame(InteractionCounts)\n",
    "    df.index.name = \"Interval Day\"\n",
    "    display(df)\n",
    "    df.to_csv(\"InteractionCounts.csv\")\n",
    "    return df\n",
    "\n",
    "# full_assessment(\"12.5\", \"WD\", engagers=False, skipping_mets=aa_names+[\"Acetate\", \"Niacin\"])\n",
    "InteractionCounts = count_interactions(compounds)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### develop the diverging bar graph of microbiome interactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib import pyplot\n",
    "\n",
    "# Plotting the diverging bar chart\n",
    "fig, ax = pyplot.subplots(figsize=(10, 6))\n",
    "ax.barh(InteractionCounts['Category'], InteractionCounts['Positive'], color='g', label='Positive')\n",
    "ax.barh(InteractionCounts['Category'], InteractionCounts['Negative'], color='r', label='Negative')\n",
    "\n",
    "# curtail the figure\n",
    "x_max = max(InteractionCounts['Positive'].max(), -InteractionCounts['Negative'].min())\n",
    "ax.set_xlim(-x_max, x_max)\n",
    "ax.set_xlabel('Values')\n",
    "ax.set_ylabel('Categories')\n",
    "ax.set_title('Double-ended Bar Graph')\n",
    "ax.legend()\n",
    "\n",
    "# Add grid lines for better readability\n",
    "pyplot.grid(True, axis='x', linestyle='--', alpha=0.7)\n",
    "pyplot.savefig(\"diverging_barChart.png\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('-1.5', 'RC'), ('-1.5', 'WD'), ('1.5', 'RC'), ('1.5', 'WD'), ('4.0', 'RC'), ('4.0', 'WD'), ('9.0', 'RC'), ('9.0', 'WD'), ('12.5', 'RC'), ('12.5', 'WD')]\n"
     ]
    }
   ],
   "source": [
    "from itertools import product\n",
    "\n",
    "days, diets = [\"-1.5\", \"1.5\", \"4.0\", \"9.0\", \"12.5\"], [\"RC\", \"WD\"]\n",
    "print([x for x in product(days, diets)])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# define new metabolite categories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from modelseedpy.biochem import from_local\n",
    "\n",
    "msdb = from_local(\"../../ModelSEEDDatabase\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# from deepdiff import DeepDiff\n",
    "from numpy import array, load, append, save, concatenate, setdiff1d\n",
    "\n",
    "scfa = load(\"../../CommScores/commscores/data/categories/scfa.npy\")\n",
    "vitamins = load(\"../../CommScores/commscores/data/categories/vitamins.npy\")\n",
    "aminoAcids = load(\"../../CommScores/commscores/data/categories/aminoAcids.npy\")\n",
    "sugars = load(\"../../CommScores/commscores/data/categories/sugars.npy\")\n",
    "# complex_sugars = array([[\"cpd00158\", \"CELB\"], [\"cpd00382\", \"Melitose\"]])\n",
    "# save(\"../../CommScores/commscores/data/categories/complex_sugars.npy\", complex_sugars)\n",
    "# sugars_dict = dict(sugars)\n",
    "# new_sugars_dict = {}\n",
    "# for cpdID, name in sugars_dict.items():\n",
    "#     if name not in [\"CELB\", \"Melitose\"]:\n",
    "#         new_sugars_dict[cpdID] = name\n",
    "# sugars = array(list(zip(new_sugars_dict.keys(), new_sugars_dict.values())))\n",
    "# save(\"../../CommScores/commscores/data/categories/sugars.npy\", sugars)\n",
    "# print(DeepDiff(sugars_dict, new_sugars_dict))\n",
    "complex_sugars = load(\"../../CommScores/commscores/data/categories/complex_sugars.npy\")\n",
    "# new_sugars = setdiff1d(sugars, complex_sugars)\n",
    "# display(new_sugars)\n",
    "# new_sugars = subtract(sugars, array([[\"cpd00158\", \"CELB\"], [\"cpd00382\", \"Melitose\"]]))\n",
    "# display(new_sugars)\n",
    "# save(\"../../CommScores/commscores/data/categories/sugars.npy\", new_sugars)\n",
    "energy_compounds = load(\"../../CommScores/commscores/data/categories/energy_compounds.npy\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from numpy import save, delete\n",
    "fatty_acids = [[\"cpd01113\", \"Hexanoate\"], [\"cpd03846\", \"Octanoate\"], [\"cpd01107\", \"Caprate\"], [\"cpd01741\", \"Laurate\"],\n",
    "               [\"cpd15640\", \"Tridecanoate\"], [\"cpd03847\", \"Myristate\"], [\"cpd16351\", \"Pentadecanoate\"], [\"cpd00214\", \"Palmitate\"],\n",
    "               [\"cpd15609\", \"heptadecanoate\"], [\"cpd01080\", \"Stearate\"], [\"cpd00536\", \"Oleate\"],\n",
    "               [\"cpd01122\", \"Linoelate\"], [\"cpd03850\", \"Linoleate\"], [\"cpd03848\", \"Arachidate\"]]\n",
    "save(\"../../CommScores/commscores/data/categories/fattyAcids.npy\", fatty_acids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([['cpd00035', 'Alanine'],\n",
       "       ['cpd00051', 'L-Arginine'],\n",
       "       ['cpd00132', 'L-Asparagine'],\n",
       "       ['cpd00041', 'L-Aspartate'],\n",
       "       ['cpd00084', 'L-Cysteine'],\n",
       "       ['cpd00023', 'L-Glutamate'],\n",
       "       ['cpd00053', 'L-Glutamine'],\n",
       "       ['cpd00033', 'Glycine'],\n",
       "       ['cpd00119', 'L-Histidine'],\n",
       "       ['cpd00747', 'L-Hydroxyproline'],\n",
       "       ['cpd00322', 'L-Isoleucine'],\n",
       "       ['cpd00107', 'L-Leucine'],\n",
       "       ['cpd00039', 'L-Lysine'],\n",
       "       ['cpd00060', 'L-Methionine'],\n",
       "       ['cpd00066', 'L-Phenylalanine'],\n",
       "       ['cpd00129', 'L-Proline'],\n",
       "       ['cpd00054', 'L-Serine'],\n",
       "       ['cpd00161', 'L-Threonine'],\n",
       "       ['cpd00065', 'L-Tryptophan'],\n",
       "       ['cpd00069', 'L-Tyrosine'],\n",
       "       ['cpd00156', 'L-Valine']], dtype='<U16')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(aminoAcids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# define the sets of amino acids\n",
    "aromaticAA = array([['cpd00066', 'L-Phenylalanine'], ['cpd00069', 'L-Tyrosine'], ['cpd00065', 'L-Tryptophan']])\n",
    "pyruvateAA = array([['cpd00035', 'Alanine'], ['cpd00107', 'L-Leucine'], ['cpd00322', 'L-Isoleucine'], ['cpd00156', 'L-Valine']])\n",
    "alphaKetaGlutarateAA = array([['cpd00023', 'L-Glutamate'], ['cpd00053', 'L-Glutamine'], ['cpd00129', 'L-Proline'], ['cpd00051', 'L-Arginine'],\n",
    "                              ['cpd00747', 'L-Hydroxyproline']])\n",
    "oxaloacetateAA = array([['cpd00132', 'L-Asparagine'], ['cpd00041', 'L-Aspartate'], \n",
    "                        ['cpd00060', 'L-Methionine'], ['cpd00161', 'L-Threonine'], ['cpd00039', 'L-Lysine']])\n",
    "phosphoglycerateAA = array([['cpd00054', 'L-Serine'], ['cpd00033', 'Glycine'], ['cpd00084', 'L-Cysteine'], [\"cpd00550\", \"D-Serine\"]])\n",
    "oxoAAs = array([[\"cpd01293\", \"Oxoproline\"], [\"cpd01292\", \"Oxoproline\"], [\"cpd00024\", \"2-Oxoglutarate\"]])\n",
    "\n",
    "# save the amino acid categories\n",
    "save(\"../../CommScores/commscores/data/categories/aromaticAA.npy\", aromaticAA)\n",
    "save(\"../../CommScores/commscores/data/categories/pyruvateAA.npy\", pyruvateAA)\n",
    "save(\"../../CommScores/commscores/data/categories/alphaKetaGlutarateAA.npy\", alphaKetaGlutarateAA)\n",
    "save(\"../../CommScores/commscores/data/categories/oxaloacetateAA.npy\", oxaloacetateAA)\n",
    "save(\"../../CommScores/commscores/data/categories/phosphoglycerateAA.npy\", phosphoglycerateAA)\n",
    "save(\"../../CommScores/commscores/data/categories/oxoAAs\", oxoAAs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['cpd00119' 'L-Histidine']\n"
     ]
    }
   ],
   "source": [
    "from numpy import concatenate\n",
    "\n",
    "all_specifiedAAs = concatenate([aromaticAA, pyruvateAA, alphaKetaGlutarateAA, oxaloacetateAA, phosphoglycerateAA])\n",
    "for aa in aminoAcids:\n",
    "    if aa not in all_specifiedAAs:\n",
    "        print(aa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([['cpd00029', 'Acetate'],\n",
       "       ['cpd00141', 'Propionate'],\n",
       "       ['cpd00211', 'Butyrate'],\n",
       "       ['cpd01711', 'Isobutyrate'],\n",
       "       ['cpd00597', 'Valerate'],\n",
       "       ['cpd16265', 'M_2_Keto_isovalerate'],\n",
       "       ['cpd35499', '(S)-2-methylbutanoate'],\n",
       "       ['cpd00106', 'Fumarate']], dtype='<U21')"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scfa\n",
    "# new_energy_compounds = []\n",
    "# for cpd, name in energy_compounds:\n",
    "#     # print(cpd, name)\n",
    "#     if name in fatty_acids:\n",
    "#         new_energy_compounds = delete(energy_compounds, name, 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 29614\n"
     ]
    }
   ],
   "source": [
    "print(len(new_energy_compounds), len(energy_compounds))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([['cpd00029', 'Acetate'],\n",
       "       ['cpd00141', 'Propionate'],\n",
       "       ['cpd00211', 'Butyrate'],\n",
       "       ['cpd01711', 'Isobutyrate'],\n",
       "       ['cpd00597', 'Valerate'],\n",
       "       ['cpd16265', 'M_2_Keto_isovalerate'],\n",
       "       ['cpd35499', '(S)-2-methylbutanoate'],\n",
       "       ['cpd00106', 'Fumarate']], dtype='<U21')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "new_scfa = append(scfa, [[\"cpd00106\", \"Fumarate\"]], axis=0)\n",
    "display(new_scfa)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## categorizing compounds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18/63 compounds are not yet exchanged\n",
      "['cpd35610', 'cpd00851', 'cpd07053', 'cpd02475', 'cpd04945', 'cpd00122', 'cpd00526', 'cpd00137', 'cpd02733', 'cpd00119', 'cpd00359', 'cpd00121', 'cpd00130', 'cpd00392', 'cpd00218', 'cpd00064', 'cpd00020', 'cpd00036']\n",
      "[{'id': 'cpd35610', 'name': '3alpha-hydroxy-7-oxo-5beta-cholan-24-oate', 'compartment': ''}, {'id': 'cpd00851', 'name': 'trans-4-Hydroxy-L-proline', 'compartment': ''}, {'id': 'oxoAA', 'name': '5-Oxoproline', 'compartment': ''}, {'id': 'cpd07053', 'name': 'Picolinic acid', 'compartment': ''}, {'id': 'cpd02475', 'name': 'Lithocholate', 'compartment': ''}, {'id': 'cpd04945', 'name': 'Ursodiol', 'compartment': ''}, {'id': 'scFatty Acid', 'name': 'Acetate', 'compartment': ''}, {'id': 'cpd00122', 'name': 'N-Acetyl-D-glucosamine', 'compartment': ''}, {'id': 'oxoAA', 'name': '2-Oxoglutarate', 'compartment': ''}, {'id': 'Pyruvate AA', 'name': 'L-Alanine', 'compartment': ''}, {'id': 'Sugar', 'name': 'D-Allose', 'compartment': ''}, {'id': 'Sugar', 'name': 'L-Arabinose', 'compartment': ''}, {'id': 'Oxaloacetate AA', 'name': 'L-Asparagine', 'compartment': ''}, {'id': 'Oxaloacetate AA', 'name': 'L-Aspartate', 'compartment': ''}, {'id': 'scFatty Acid', 'name': 'Butyrate', 'compartment': ''}, {'id': 'complex Sugar', 'name': 'CELB', 'compartment': ''}, {'id': 'cpd00526', 'name': 'Cholate', 'compartment': ''}, {'id': 'fatty Acid', 'name': 'Oleate', 'compartment': ''}, {'id': 'cpd00137', 'name': 'Citrate', 'compartment': ''}, {'id': 'phosphoGlycerate AA', 'name': 'L-Cysteine', 'compartment': ''}, {'id': 'fatty Acid', 'name': 'Decanoate', 'compartment': ''}, {'id': 'cpd02733', 'name': 'Deoxycholate', 'compartment': ''}, {'id': 'fatty Acid', 'name': 'ddca', 'compartment': ''}, {'id': 'Sugar', 'name': 'Thyminose', 'compartment': ''}, {'id': 'Sugar', 'name': 'D-Fructose', 'compartment': ''}, {'id': 'Sugar', 'name': 'L-Fucose', 'compartment': ''}, {'id': 'scFatty Acid', 'name': 'Fumarate', 'compartment': ''}, {'id': 'Sugar', 'name': 'Galactose', 'compartment': ''}, {'id': 'Sugar', 'name': 'Dulcose', 'compartment': ''}, {'id': 'Sugar', 'name': 'D-Glucose', 'compartment': ''}, {'id': 'alphaKetaGlutarate AA', 'name': 'L-Glutamate', 'compartment': ''}, {'id': 'phosphoGlycerate AA', 'name': 'Glycine', 'compartment': ''}, {'id': 'fatty Acid', 'name': 'Palmitate', 'compartment': ''}, {'id': 'cpd00119', 'name': 'L-Histidine', 'compartment': ''}, {'id': 'fatty Acid', 'name': 'Hexanoate', 'compartment': ''}, {'id': 'Pyruvate AA', 'name': 'L-Isoleucine', 'compartment': ''}, {'id': 'cpd00359', 'name': 'indol', 'compartment': ''}, {'id': 'cpd00121', 'name': 'L-Inositol', 'compartment': ''}, {'id': 'Pyruvate AA', 'name': 'L-Leucine', 'compartment': ''}, {'id': 'fatty Acid', 'name': 'Linoleate', 'compartment': ''}, {'id': 'Oxaloacetate AA', 'name': 'L-Lysine', 'compartment': ''}, {'id': 'cpd00130', 'name': 'L-Malate', 'compartment': ''}, {'id': 'Sugar', 'name': 'Melibiose', 'compartment': ''}, {'id': 'cpd00392', 'name': 'Phycite', 'compartment': ''}, {'id': 'Oxaloacetate AA', 'name': 'L-Methionine', 'compartment': ''}, {'id': 'cpd00218', 'name': 'Niacin', 'compartment': ''}, {'id': 'fatty Acid', 'name': 'octanoate', 'compartment': ''}, {'id': 'cpd00064', 'name': 'Ornithine', 'compartment': ''}, {'id': 'Aromatic AA', 'name': 'L-Phenylalanine', 'compartment': ''}, {'id': 'scFatty Acid', 'name': 'Propionate', 'compartment': ''}, {'id': 'alphaKetaGlutarate AA', 'name': 'L-Proline', 'compartment': ''}, {'id': 'scFatty Acid', 'name': 'Valerate', 'compartment': ''}, {'id': 'cpd00020', 'name': 'Pyruvate', 'compartment': ''}, {'id': 'complex Sugar', 'name': 'Melitose', 'compartment': ''}, {'id': 'Sugar', 'name': 'D-Ribose', 'compartment': ''}, {'id': 'Sugar', 'name': 'L-Rhamnose', 'compartment': ''}, {'id': 'phosphoGlycerate AA', 'name': 'D-Serine', 'compartment': ''}, {'id': 'cpd00036', 'name': 'Succinate', 'compartment': ''}, {'id': 'Sugar', 'name': 'Sucrose', 'compartment': ''}, {'id': 'Oxaloacetate AA', 'name': 'L-Threonine', 'compartment': ''}, {'id': 'fatty Acid', 'name': 'Myristic acid', 'compartment': ''}, {'id': 'Aromatic AA', 'name': 'L-Tyrosine', 'compartment': ''}, {'id': 'Pyruvate AA', 'name': 'L-Valine', 'compartment': ''}]\n"
     ]
    }
   ],
   "source": [
    "from json import load\n",
    "\n",
    "with open(\"ASVInteractionModel.json\", 'r') as jsonIn:\n",
    "    model = load(jsonIn)\n",
    "\n",
    "# check which vitamins are in the model\n",
    "# metabolites = [met[\"id\"] for met in model[\"metabolites\"]]\n",
    "# vitamin_dict = dict(vitamins)\n",
    "# for metID in metabolites:\n",
    "#     if metID in vitamin_dict:\n",
    "#         print(metID)\n",
    "\n",
    "# define dictionaries of all the categories\n",
    "## amino acids\n",
    "aromaticAA_dict, pyruvateAA_dict, alphaKetaGlutarateAA_dict = dict(aromaticAA), dict(pyruvateAA), dict(alphaKetaGlutarateAA)\n",
    "oxaloacetateAA_dict, phosphoglycerateAA_dict, oxoAA_dict = dict(oxaloacetateAA), dict(phosphoglycerateAA), dict(oxoAAs)\n",
    "## fatty acids\n",
    "fatty_acids_dict, scfa_dict, sugars_dict, complex_sugars_dict = dict(fatty_acids), dict(scfa), dict(sugars), dict(complex_sugars)\n",
    "\n",
    "# categorize the model metabolites\n",
    "not_changed = []\n",
    "for met in model[\"metabolites\"]:\n",
    "    ## amino acids\n",
    "    if met[\"id\"] in aromaticAA_dict:   met[\"id\"] = \"Aromatic AA\"\n",
    "    elif met[\"id\"] in pyruvateAA_dict:   met[\"id\"] = \"Pyruvate AA\"\n",
    "    elif met[\"id\"] in alphaKetaGlutarateAA_dict:   met[\"id\"] = \"alphaKetaGlutarate AA\"\n",
    "    elif met[\"id\"] in oxaloacetateAA_dict:   met[\"id\"] = \"Oxaloacetate AA\"\n",
    "    elif met[\"id\"] in phosphoglycerateAA_dict:   met[\"id\"] = \"phosphoGlycerate AA\"\n",
    "    elif met[\"id\"] in oxoAA_dict:   met[\"id\"] = \"oxoAA\"\n",
    "    ## other compounds\n",
    "    elif met[\"id\"] in fatty_acids_dict:   met[\"id\"] = \"fatty Acid\"\n",
    "    elif met[\"id\"] in scfa_dict:   met[\"id\"] = \"scFatty Acid\"\n",
    "    elif met[\"id\"] in sugars_dict:   met[\"id\"] = \"Sugar\"\n",
    "    elif met[\"id\"] in complex_sugars_dict:   met[\"id\"] = \"complex Sugar\"\n",
    "    else:   not_changed.append(met[\"id\"])\n",
    "\n",
    "print(f\"{len(not_changed)}/{len(model['metabolites'])} compounds are not yet exchanged\")\n",
    "print(not_changed)\n",
    "\n",
    "# export the converted model\n",
    "from json import dump\n",
    "print(model[\"metabolites\"])\n",
    "with open(\"ASVInteractionModel_categorizedIDs.json\", 'w') as jsonOut:\n",
    "    dump(model, jsonOut, indent=3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# filtering the Escher map for fewer reactions based on the metabolite fluxes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 357,
   "metadata": {},
   "outputs": [],
   "source": [
    "aa_names = [\"L-Methionine\", \"L-Histidine\", \"L-Tryptophan\", \"L-Alanine\", \"L-Threonine\", \"L-Glutamate\", \"L-Isoleucine\", \"L-Proline\", \"L-Leucine\", \"Glycine\",\n",
    " 'L-Phenylalanine', 'L-Tyrosine', 'L-Valine', 'L-Glutamine', 'L-Arginine', 'L-Asparagine', 'L-Aspartate', 'L-Lysine', 'L-Serine', 'L-Cysteine', \"D-Serine\",\n",
    " \"5-Oxoproline\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "metadata": {},
   "outputs": [],
   "source": [
    "from json import load, dump\n",
    "with open(\"metabolite_focused_map.json\", 'r') as jsonIn:\n",
    "    escherMap = load(jsonIn)\n",
    "# display(escherMap)\n",
    "\n",
    "og_escherMap = [ele for ele in escherMap]\n",
    "\n",
    "# from collections import Counter\n",
    "nodeID = {nodeNum: content[\"bigg_id\"] for nodeNum, content in og_escherMap[1][\"nodes\"].items() if \"bigg_id\" in content}\n",
    "with open(\"Escher_nodeNum_cpdIDs.json\", 'w') as jsonOut:\n",
    "    dump(nodeID, jsonOut, indent=3)\n",
    "\n",
    "idName = {content[\"bigg_id\"]: content[\"name\"] for content in og_escherMap[1][\"nodes\"].values() if \"bigg_id\" in content}\n",
    "with open(\"EscherNodeMapping.json\", 'w') as jsonOut:\n",
    "    dump(idName, jsonOut, indent=3)\n",
    "\n",
    "newEscher = []\n",
    "for index, ele in enumerate(og_escherMap):\n",
    "    og_ele = {k:v for k,v in ele.items()}\n",
    "    if index == 0:  newEscher = [ele]\n",
    "    newEscher.append({})\n",
    "    nodes_to_remove = []\n",
    "    nodeNumCoef = {}\n",
    "    for k, val in ele.items():\n",
    "        if k == \"reactions\":\n",
    "            newEscher[index][k] = val\n",
    "            # newEscher[index][k] = {}\n",
    "            # reactions = {}\n",
    "            for rxnNum, content in val.items():\n",
    "                # display(len(content[\"metabolites\"]), content)\n",
    "                # print(len(content[\"metabolites\"]))\n",
    "                #create a mapping from IDs to their respective coefficients\n",
    "                metCoefs = {met[\"bigg_id\"]: met[\"coefficient\"] for met in content[\"metabolites\"]}\n",
    "                nameID = {}\n",
    "                # print(metCoefs)\n",
    "                for ID in metCoefs.keys():\n",
    "                    if ID not in idName:   continue\n",
    "                    name = idName[ID]\n",
    "                    if name in nameID: nameID[name].append(ID)\n",
    "                    else: nameID[name] = [ID]\n",
    "                #determine the nodes that should be removed\n",
    "                nodeNumIDs = {content2[\"to_node_id\"]: nodeID[content2[\"to_node_id\"]]\n",
    "                              for content2 in content[\"segments\"].values() if content2[\"to_node_id\"] in nodeID.keys()}\n",
    "                nodeNumIDs.update({content2[\"from_node_id\"]: nodeID[content2[\"from_node_id\"]]\n",
    "                                   for content2 in content[\"segments\"].values() if content2[\"from_node_id\"] in nodeID.keys()})\n",
    "                nodeNumCoef.update({num: metCoefs[ID] for num, ID in nodeNumIDs.items()})\n",
    "                # print(nodeNumCoef)\n",
    "                # for nodeNum, coef in nodeNumCoef.items():\n",
    "                #     if coef < 0.01:   nodes_to_remove.append(nodeNum)\n",
    "                #     elif \"AA\" in idName[content[\"bigg_id\"]] and metCoefs[content[\"bigg_id\"]] < 0.1:  continue\n",
    "                #     for k2, v in content.items():\n",
    "                #         if k2 == \"node_is_primary\" and metCoefs[content[\"bigg_id\"]] >= 1:\n",
    "                #             new_nodes[nodeNum][k2] = True\n",
    "                #         else:  new_nodes[nodeNum][k2] = v\n",
    "\n",
    "                # names = Counter([idName[ID]:ID for ID in metCoefs.keys()])\n",
    "                #determine the ID with the highest coefficient, which will be exclusively kept in the final Escher\n",
    "                # new_met_content = []\n",
    "                # for k2,v in nameID.items():\n",
    "                #     if metCoefs[content[\"bigg_id\"]] < 0.01:   continue\n",
    "                #     elif \"AA\" in idName[content[\"bigg_id\"]] and metCoefs[content[\"bigg_id\"]] < 0.1:  continue\n",
    "                #     for k2, v in content.items():\n",
    "                #         if k2 == \"node_is_primary\" and metCoefs[content[\"bigg_id\"]] >= 1:\n",
    "                #             new_nodes[nodeNum][k2] = True\n",
    "                #         else:  new_nodes[nodeNum][k2] = v\n",
    "                #     if len(v) == 1:\n",
    "                #         new_met_content.append({\"bigg_id\": v[0], \"coefficient\": metCoefs[v[0]]})\n",
    "                #     elif len(v) > 1:\n",
    "                #         ReactantWinner, ProductWinner = [None, 0], [None, 0]\n",
    "                #         for ID, coef in metCoefs.items():\n",
    "                #             if ID in v:\n",
    "                #                 if coef > 0 and coef > ProductWinner[1]:\n",
    "                #                     ProductWinner = [ID, coef]\n",
    "                #                 if coef < 0 and coef < ReactantWinner[1]:\n",
    "                #                     ReactantWinner = [ID, coef]\n",
    "                #         for met in content[\"metabolites\"]:\n",
    "                #             if met[\"bigg_id\"] in v and met[\"bigg_id\"] in [ReactantWinner[0], ProductWinner[0]]:\n",
    "                #                 new_met_content.append(met)\n",
    "                # print(len(new_met_content), len(content[\"metabolites\"]))\n",
    "                #update the new Escher Figure\n",
    "                newEscher[index][k][rxnNum] = content\n",
    "                # for key, value in content.items():\n",
    "                #     if key == \"metabolites\":\n",
    "                #         newEscher[index][k][rxnNum][key] = new_met_content\n",
    "                #     else:\n",
    "                #         newEscher[index][k][rxnNum][key] = value\n",
    "                # escherMap[index][\"reactions\"][rxnNum][\"metabolites\"] = new_met_content\n",
    "                # print(len(content[\"metabolites\"]) == len(newEscher[index][k][rxnNum][\"metabolites\"]))\n",
    "                # display(len(new_met_content), escherMap[index][\"reactions\"][rxnNum][\"metabolites\"])\n",
    "                # break\n",
    "    # display(nodeNumCoef)\n",
    "    for k, val in ele.items():\n",
    "        if k == \"nodes\":\n",
    "            new_nodes = {}\n",
    "            # display(val)\n",
    "            for nodeNum, content in val.items():\n",
    "                # display(content)\n",
    "                new_nodes[nodeNum] = {}\n",
    "                if nodeNum in nodeNumCoef:\n",
    "                    # print(nodeID[nodeNum], nodeNumCoef[nodeNum])\n",
    "                    if abs(nodeNumCoef[nodeNum]) < 0.05:   continue\n",
    "                    if idName[content[\"bigg_id\"]] in aa_names:  continue # and abs(nodeNumCoef[nodeNum]) < 0.1:  continue\n",
    "                    if idName[content[\"bigg_id\"]] in [\"Niacin\", \"Acetate\"]:  continue\n",
    "                    # if abs(nodeNumCoef[nodeNum]) >= 1:   primary = True\n",
    "                for k2, v in content.items():\n",
    "                    if k2 == \"node_is_primary\":  new_nodes[nodeNum][k2] = True\n",
    "                    else:  new_nodes[nodeNum][k2] = v\n",
    "            # break\n",
    "            #remove all of the above nodes, which should remove the segments by default\n",
    "            #replace the IDs with their categories so that the nodes can consolidate\n",
    "            newEscher[index][k] = new_nodes\n",
    "            # DeepDiff(val, new_nodes)\n",
    "        elif k != \"reactions\":\n",
    "            newEscher[index][k] = val\n",
    "    # print(DeepDiff(og_ele, newEscher[index]))\n",
    "\n",
    "from deepdiff import DeepDiff\n",
    "# display(DeepDiff(og_escherMap[1], newEscher[1]))\n",
    "\n",
    "from json import dump\n",
    "with open(\"metabolite_focused_map0.json\", 'w') as jsonOut:\n",
    "    dump(newEscher, jsonOut, indent=3)\n",
    "\n",
    "#make each column an interacting web, which can feed into the next column through some type of overlapping metabolites\n",
    "#the fatty acids and the sugars will likely be the differentiating factor between the RC and WD regimes\n",
    "#amino acids and niacin are the insurance policy for syntrophy, and should be used as the structure of the food web \n",
    "#group the members who have the same major products, and group seemingly parasitic organisms separately, by programmatically analyzing the highest fluxes for each organism's reactants and products"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 326,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Counter({'L-Alanine': 2, 'L-Methionine': 1, 'L-Tryptophan': 1, 'L-Threonine': 1, 'L-Glutamate': 1, 'L-Isoleucine': 1, 'L-Proline': 1, 'L-Leucine': 1, 'Glycine': 1, 'L-Phenylalanine': 1, 'L-Tyrosine': 1, 'L-Valine': 1, 'L-Glutamine': 1, 'L-Arginine': 1, 'L-Asparagine': 1, 'L-Aspartate': 1, 'L-Lysine': 1, 'L-Serine': 1, 'L-Cysteine': 1})\n"
     ]
    }
   ],
   "source": [
    "from collections import Counter\n",
    "print(Counter(aa_names))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### manually restore syntrophic compounds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Turicibacter\n",
    "\n",
    "info = {\"name\":\"Turicibacter.1\",\"bigg_id\":\"Turicibacter.1RC-ABX_-1.5\",\"reversibility\":False,\"label_x\":-2050.9305419921875,\"label_y\":8288.802734375,\"gene_reaction_rule\":\"\",\"genes\":[],\"metabolites\":[{\"bigg_id\":\"cpd00023\",\"coefficient\":-0.1040152173141567},{\"bigg_id\":\"cpd00041\",\"coefficient\":-0.1042731995047922},{\"bigg_id\":\"cpd00064\",\"coefficient\":0.0241392199676796},{\"bigg_id\":\"cpd00084\",\"coefficient\":-0.3059013174870795},{\"bigg_id\":\"cpd00108\",\"coefficient\":0.3980953511859826},{\"bigg_id\":\"cpd00122\",\"coefficient\":-0.225680761223981},{\"bigg_id\":\"cpd00129\",\"coefficient\":-0.0200989691680044},{\"bigg_id\":\"cpd00132\",\"coefficient\":-0.0185135542972458},{\"bigg_id\":\"cpd00141\",\"coefficient\":-0.0158535137273317},{\"bigg_id\":\"cpd00161\",\"coefficient\":-0.0233209413247074},{\"bigg_id\":\"cpd00211\",\"coefficient\":-0.0817767218820333},{\"bigg_id\":\"cpd00214\",\"coefficient\":-0.0129732900159796},{\"bigg_id\":\"cpd00218\",\"coefficient\":-0.0016870647208757},{\"bigg_id\":\"cpd00322\",\"coefficient\":-0.0337539940226029}],\"segments\":{\"296\":{\"from_node_id\":\"313\",\"to_node_id\":\"314\",\"b1\":None,\"b2\":None},\"297\":{\"from_node_id\":\"315\",\"to_node_id\":\"314\",\"b1\":None,\"b2\":None},\"298\":{\"from_node_id\":\"313\",\"to_node_id\":\"312\",\"b1\":{\"x\":-1967.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2236.64208984375,\"y\":8412.42578125}},\"299\":{\"from_node_id\":\"313\",\"to_node_id\":\"316\",\"b1\":{\"x\":-1971.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2131.64208984375,\"y\":8322.42578125}},\"300\":{\"from_node_id\":\"315\",\"to_node_id\":\"317\",\"b1\":{\"x\":-1843.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-1789.111572265625,\"y\":8418.39892578125}},\"301\":{\"from_node_id\":\"313\",\"to_node_id\":\"318\",\"b1\":{\"x\":-1971.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2146.64208984375,\"y\":8340.42578125}},\"302\":{\"from_node_id\":\"315\",\"to_node_id\":\"319\",\"b1\":{\"x\":-1839.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-1834.111572265625,\"y\":8436.39892578125}},\"303\":{\"from_node_id\":\"313\",\"to_node_id\":\"320\",\"b1\":{\"x\":-1971.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2161.64208984375,\"y\":8358.42578125}},\"304\":{\"from_node_id\":\"313\",\"to_node_id\":\"321\",\"b1\":{\"x\":-1971.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2176.64208984375,\"y\":8376.42578125}},\"305\":{\"from_node_id\":\"313\",\"to_node_id\":\"322\",\"b1\":{\"x\":-1971.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2191.64208984375,\"y\":8394.42578125}},\"306\":{\"from_node_id\":\"313\",\"to_node_id\":\"323\",\"b1\":{\"x\":-1971.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2191.64208984375,\"y\":8430.42578125}},\"307\":{\"from_node_id\":\"313\",\"to_node_id\":\"324\",\"b1\":{\"x\":-1971.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2176.64208984375,\"y\":8448.42578125}},\"308\":{\"from_node_id\":\"313\",\"to_node_id\":\"325\",\"b1\":{\"x\":-1971.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2161.64208984375,\"y\":8466.42578125}},\"309\":{\"from_node_id\":\"313\",\"to_node_id\":\"326\",\"b1\":{\"x\":-1971.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2146.64208984375,\"y\":8484.42578125}},\"310\":{\"from_node_id\":\"313\",\"to_node_id\":\"327\",\"b1\":{\"x\":-1971.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2131.64208984375,\"y\":8502.42578125}},\"311\":{\"from_node_id\":\"313\",\"to_node_id\":\"328\",\"b1\":{\"x\":-1971.361572265625,\"y\":8418.39892578125},\"b2\":{\"x\":-2116.64208984375,\"y\":8520.42578125}}}}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "('312', 'cpd00023', 'L-Glutamate')\n",
      "None\n",
      "('316', 'cpd00041', 'L-Aspartate')\n",
      "None\n",
      "('317', 'cpd00064', 'Ornithine')\n",
      "None\n",
      "('318', 'cpd00084', 'L-Cysteine')\n",
      "None\n",
      "('319', 'cpd00108', 'Galactose')\n",
      "None\n",
      "('320', 'cpd00122', 'N-Acetyl-D-glucosamine')\n",
      "None\n",
      "('321', 'cpd00129', 'L-Proline')\n",
      "None\n",
      "('322', 'cpd00132', 'L-Asparagine')\n",
      "None\n",
      "('323', 'cpd00141', 'Propionate')\n",
      "None\n",
      "('324', 'cpd00161', 'L-Threonine')\n",
      "None\n",
      "('325', 'cpd00211', 'Butyrate')\n",
      "None\n",
      "('326', 'cpd00214', 'Palmitate')\n",
      "None\n",
      "('327', 'cpd00218', 'Niacin')\n",
      "None\n",
      "('328', 'cpd00322', 'L-Isoleucine')\n"
     ]
    }
   ],
   "source": [
    "for num, content in info[\"segments\"].items():\n",
    "    if \"from_node_id\" not in content:  continue\n",
    "    # print(content[\"from_node_id\"], content[\"to_node_id\"])\n",
    "    print(nodeID[content[\"from_node_id\"]] if content[\"from_node_id\"] in nodeID else \"None\")\n",
    "    print((content[\"to_node_id\"], nodeID[content[\"to_node_id\"]], idName[nodeID[content[\"to_node_id\"]]]) if content[\"to_node_id\"] in nodeID else \"None\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# editing the SVG Escher Map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "modelseedpy 0.2.2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Metabolite identifier</strong></td><td>cpd00027</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>D-Glucose</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x14a6602b0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Formula</strong></td><td>C6H12O6</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartment</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>In 0 reaction(s)</strong></td><td>\n",
       "                    \n",
       "                    </td>\n",
       "            </tr>\n",
       "        </table>"
      ],
      "text/plain": [
       "<ModelSEEDCompound2 cpd00027 at 0x14a6602b0>"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from modelseedpy.biochem import from_local\n",
    "msdb = from_local(\"../../ModelSEEDDatabase\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_annotation', '_bound', '_id', '_model', '_reaction', '_repr_html_', '_set_id_with_model', 'abbr', 'annotation', 'charge', 'compartment', 'constraint', 'copy', 'delta_g', 'delta_g_error', 'elements', 'flags', 'formula', 'formula_weight', 'id', 'inchi', 'inchi_key', 'is_abstract', 'is_cofactor', 'is_core', 'is_obsolete', 'linked_compound', 'mass', 'model', 'name', 'names', 'notes', 'pka', 'pkb', 'reactions', 'remove_from_model', 'seed_id', 'shadow_price', 'smiles', 'source', 'summary', 'to_template_compartment_compound', 'y']\n",
      "{'glc__D'}\n"
     ]
    }
   ],
   "source": [
    "cpd = msdb.compounds.get_by_id(\"cpd00027\")\n",
    "print(dir(cpd))\n",
    "print(cpd.annotation[\"BiGG\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>abbrev</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sucrose</td>\n",
       "      <td>sucr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Propionate</td>\n",
       "      <td>prpa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Butyrate</td>\n",
       "      <td>butr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L-Malate</td>\n",
       "      <td>malat</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Citrate</td>\n",
       "      <td>citr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2-Oxoglutarate</td>\n",
       "      <td>akg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Succinate</td>\n",
       "      <td>succ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Ornithine</td>\n",
       "      <td>ornth</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Myristic acid</td>\n",
       "      <td>myrst</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Fumarate</td>\n",
       "      <td>fumr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Galactose</td>\n",
       "      <td>galct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Melitose</td>\n",
       "      <td>melit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>D-Ribose</td>\n",
       "      <td>ribs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>L-Inositol</td>\n",
       "      <td>inost</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>L-Fucose</td>\n",
       "      <td>fucos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>N-Acetyl-D-glucosamine</td>\n",
       "      <td>acglum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CELB</td>\n",
       "      <td>cellb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Palmitate</td>\n",
       "      <td>palm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Melibiose</td>\n",
       "      <td>melib</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Dulcose</td>\n",
       "      <td>dulco</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Pyruvate</td>\n",
       "      <td>pyr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>L-Arabinose</td>\n",
       "      <td>arbns</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>D-Fructose</td>\n",
       "      <td>fru</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>trans-4-Hydroxy-L-proline</td>\n",
       "      <td>4hpro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Decanoate</td>\n",
       "      <td>decac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>L-Rhamnose</td>\n",
       "      <td>rhmn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>D-Allose</td>\n",
       "      <td>allos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>D-Glucose</td>\n",
       "      <td>glu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Thyminose</td>\n",
       "      <td>2drib</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         name  abbrev\n",
       "0                     Sucrose    sucr\n",
       "1                  Propionate    prpa\n",
       "2                    Butyrate    butr\n",
       "3                    L-Malate   malat\n",
       "4                     Citrate    citr\n",
       "5              2-Oxoglutarate     akg\n",
       "6                   Succinate    succ\n",
       "7                   Ornithine   ornth\n",
       "8               Myristic acid   myrst\n",
       "9                    Fumarate    fumr\n",
       "10                  Galactose   galct\n",
       "11                   Melitose   melit\n",
       "12                   D-Ribose    ribs\n",
       "13                 L-Inositol   inost\n",
       "14                   L-Fucose   fucos\n",
       "15     N-Acetyl-D-glucosamine  acglum\n",
       "16                       CELB   cellb\n",
       "17                  Palmitate    palm\n",
       "18                  Melibiose   melib\n",
       "19                    Dulcose   dulco\n",
       "20                   Pyruvate     pyr\n",
       "21                L-Arabinose   arbns\n",
       "22                 D-Fructose     fru\n",
       "23  trans-4-Hydroxy-L-proline   4hpro\n",
       "24                  Decanoate   decac\n",
       "25                 L-Rhamnose    rhmn\n",
       "26                   D-Allose   allos\n",
       "27                  D-Glucose     glu\n",
       "28                  Thyminose   2drib"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29\n"
     ]
    }
   ],
   "source": [
    "from pandas import DataFrame\n",
    "\n",
    "nameAbbrev = DataFrame(columns=[\"name\", \"abbrev\"])\n",
    "\n",
    "cpdID_abbrv = {\n",
    "    'cpd00076': 'sucr', 'cpd00141': 'prpa', 'cpd00211': 'butr', 'cpd00130': 'malat', 'cpd00137': 'citr', 'cpd00024': 'akg', \n",
    "    'cpd00036': 'succ', 'cpd00064': 'ornth', 'cpd03847': 'myrst', 'cpd00106': 'fumr', 'cpd00108': 'galct', 'cpd00382': 'melit', \n",
    "    'cpd00105': 'ribs', 'cpd00121': 'inost', 'cpd00751': 'fucos', 'cpd00122': 'acglum', 'cpd00158': 'cellb', \n",
    "    'cpd00214': 'palm', 'cpd03198': 'melib', 'cpd01171': 'dulco', 'cpd00020': 'pyr', 'cpd00224': 'arbns', 'cpd00082': 'fru', \n",
    "    'cpd00851': '4hpro', 'cpd01107': 'decac', 'cpd00396': 'rhmn', 'cpd01055': 'allos', 'cpd00027': 'glu', \n",
    "    'cpd01242': '2drib'}\n",
    "\n",
    "for cpdID, abbrev in cpdID_abbrv.items():\n",
    "    cpd = msdb.compounds.get_by_id(cpdID)\n",
    "    nameAbbrev.loc[len(nameAbbrev)] = [cpd.name, abbrev]\n",
    "\n",
    "display(nameAbbrev)\n",
    "print(len(cpdID_abbrv))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "f3c199\n"
     ]
    }
   ],
   "source": [
    "# from shapely.geometry import Point, box\n",
    "import random\n",
    "import re\n",
    "\n",
    "def tint_color(hex_color, tint_factor=0.5):\n",
    "    rgb_color = tuple(int(hex_color.lstrip('#')[i:i+2], 16) for i in (0, 2, 4))\n",
    "    tinted_rgb = [((1 - tint_factor) * component + tint_factor * 255) for component in rgb_color]\n",
    "    return '{:02x}{:02x}{:02x}'.format(int(tinted_rgb[0]), int(tinted_rgb[1]), int(tinted_rgb[2]))\n",
    "\n",
    "# Function to get bounding box of an SVG element\n",
    "def get_text_bbox(text_element, font_size=10):\n",
    "    x = float(text_element['x'])\n",
    "    y = float(text_element['y'])\n",
    "    # Approximate width and height based on font size\n",
    "    width = len(text_element.get_text()) * font_size * 0.6\n",
    "    height = font_size\n",
    "    return box(x, y - height, x + width, y)\n",
    "\n",
    "# Function to approximate the bounding box of a node element\n",
    "def get_node_bbox(node_element):   # TODO need to be updated\n",
    "    cx = float(node_element['cx'])\n",
    "    cy = float(node_element['cy'])\n",
    "    r = float(node_element['r'])\n",
    "    return box(cx - r, cy - r, cx + r, cy + r)\n",
    "\n",
    "# Function to approximate the bounding box of an edge element (assuming it's a line)\n",
    "def get_edge_bbox(edge_element):   # TODO need to be updated\n",
    "    x1 = float(edge_element['x1'])\n",
    "    y1 = float(edge_element['y1'])\n",
    "    x2 = float(edge_element['x2'])\n",
    "    y2 = float(edge_element['y2'])\n",
    "    # A simple approximation of the bounding box of a line\n",
    "    return box(min(x1, x2), min(y1, y2), max(x1, x2), max(y1, y2))\n",
    "\n",
    "\n",
    "# Function to adjust text labels to minimize overlap\n",
    "def adjust_labels(labels, nodes, edges, font_size=10, textToKeep=[\"l27\", \"l13\"]):\n",
    "    label_positions = [get_text_bbox(label, font_size) for label in labels if label[\"id\" not in textToKeep]]\n",
    "    node_positions = [get_node_bbox(node) for node in nodes]\n",
    "    edge_positions = [get_edge_bbox(edge) for edge in edges]\n",
    "    \n",
    "    all_positions = label_positions + node_positions + edge_positions\n",
    "    for i, label in enumerate(labels):\n",
    "        bbox = label_positions[i]\n",
    "        adjusted = False\n",
    "        \n",
    "        while not adjusted:\n",
    "            overlap = False\n",
    "            for j, other_bbox in enumerate(all_positions):\n",
    "                if i != j and bbox.intersects(other_bbox):\n",
    "                    overlap = True\n",
    "                    break\n",
    "            \n",
    "            if overlap:\n",
    "                # Adjust position (for simplicity, moving randomly; a better algorithm can be used)\n",
    "                match = re.search(r'translate\\((-?\\d+\\.?\\d*),\\s*(-?\\d+\\.?\\d*)\\)', label['transform'])\n",
    "                x, y = float(match.group(1)), float(match.group(2))\n",
    "                x = str(x + random.uniform(-1, 1))\n",
    "                y = str(y + random.uniform(-1, 1))\n",
    "                label[\"transform\"] = f'translate({x},{y})'\n",
    "                bbox = get_text_bbox(label, font_size)\n",
    "                label_positions[i] = bbox\n",
    "            else:     adjusted = True\n",
    "\n",
    "\n",
    "def EscherSVG_processing(svg_path=\"metabolite_focused_map.svg\", labels_to_remove=[\"stoichiometry-labels\"],\n",
    "                         node_label_px=10, rxn_label_px=30, node_label_hex=\"e0865b\", rxn_label_hex=\"202078\",\n",
    "                         rxn_edge_hex=\"5aa6df\", rxn_edge_px=10, colorElements=None, largeNodeLabels=None,\n",
    "                         largeEdgeLabels=None, tint_factor=0.5, abbrevIDs=None, msdb=None):\n",
    "    # load the SVG for parsing\n",
    "    from bs4 import BeautifulSoup\n",
    "    soup = BeautifulSoup(open(svg_path, \"r\").read() , 'lxml-xml')\n",
    "\n",
    "    if abbrevIDs:\n",
    "        for nodeLabel in soup.find_all('text', class_=\"node-label label\"):\n",
    "            cpdID = nodeLabel.string\n",
    "            nodeLabel.string = abbrevIDs[cpdID]\n",
    "\n",
    "    # remove all stoichiometric or other undesirable labels\n",
    "    for label in labels_to_remove:\n",
    "        stoich_labels = soup.find_all(class_=label)\n",
    "        for label in stoich_labels:\n",
    "            label.decompose()\n",
    "\n",
    "    # label font-size and color change\n",
    "    metadata = soup.find(\"style\", attrs={'type': \"text/css\"})\n",
    "    ## change reactions\n",
    "    metadata.string = metadata.string.replace(\".reaction-label{font-size:30px;fill:#202078;text-rendering:optimizelegibility}\",\n",
    "                                              \".reaction-label{\"+f\"font-size:{rxn_label_px}px;fill:#{rxn_label_hex};stroke:#000000;stroke-width:{rxn_label_px/20}px;\"+\"text-rendering:optimizelegibility}\")\n",
    "    if largeEdgeLabels is not None:\n",
    "        fadedEdgesColor = tint_color(rxn_edge_hex, tint_factor)\n",
    "        fadedEdgesLabelsColor = tint_color(rxn_label_hex, tint_factor)\n",
    "        metadata.string = metadata.string + f\" .m{fadedEdgesLabelsColor}-label\"+\"{\"+f\"font-size:{rxn_label_px/1.2}px;fill:#{fadedEdgesLabelsColor};stroke:#000000;stroke-width:{rxn_label_px/30}px;\"+\"text-rendering:optimizelegibility}\"\n",
    "        #f\"stroke:#{fadedEdgesColor};\"+f\"stroke-width:{rxn_edge_px/1.2}px;\"+\"fill:none}\"\n",
    "    for reaction in soup.find_all(class_=\"reaction\"):\n",
    "        label = reaction.find(\"text\")\n",
    "        if largeEdgeLabels is not None and reaction[\"id\"] not in largeEdgeLabels:\n",
    "            label['class'] = f\"m{fadedEdgesLabelsColor}-label\"\n",
    "        cleanedStr = label.string.split('_')[0].replace(\"-ABX\", '').replace(\"RC\", '').replace(\"WD\", '')\n",
    "        label.string = cleanedStr.split('.')[0][:6] + \".\" + cleanedStr.split('.')[1]\n",
    "    ## change nodes\n",
    "    metadata.string = metadata.string.replace(\".node-label{font-size:20px}\", \n",
    "                                              \".node-label{\"+f\"font-size:{node_label_px}px;fill:#{node_label_hex};stroke:#000000;stroke-width:{node_label_px/20}px;\"+\"}\")\n",
    "    metadata.string = metadata.string.replace(\".metabolite-circle{stroke:#a24510;fill:#e0865b}\",\n",
    "                                              \".metabolite-circle{stroke:#\"+f\"{node_label_hex};fill:#{node_label_hex}\"+\"}\")\n",
    "    if largeNodeLabels is not None:\n",
    "        fadedNodesColor = tint_color(node_label_hex, tint_factor)\n",
    "        print(fadedNodesColor)\n",
    "        for node in soup.find_all(class_=\"node\"):\n",
    "            # print(node)\n",
    "            if node[\"id\"] not in largeNodeLabels:\n",
    "                # print(node[\"id\"])\n",
    "                circle = node.find(\"circle\")\n",
    "                label = node.find(\"text\")\n",
    "                if any([x is None for x in [circle, label]]):   continue\n",
    "                circle[\"class\"] = f\"m{fadedNodesColor}\"\n",
    "                label['class'] = f\"m{fadedNodesColor}-label\"\n",
    "                # print(node[\"id\"])\n",
    "        metadata.string = metadata.string + f\" .m{fadedNodesColor}\"+\"{\"+f\"stroke:#{fadedNodesColor};fill:#{fadedNodesColor}\"+\"}\"\n",
    "        metadata.string = metadata.string + f\" .m{fadedNodesColor}-label\"+\"{\"+f\"font-size:{node_label_px/1.2}px;fill:#{fadedNodesColor};stroke:#000000;stroke-width:{node_label_px/30}px;\"+\"}\"\n",
    "    ## change segments\n",
    "    metadata.string = metadata.string.replace(\".segment{\"+\"stroke:#334E75;stroke-width:10px;fill:none}\", \n",
    "                                              \".segment{\"+f\"stroke:#{rxn_edge_hex};\"+f\"stroke-width:{rxn_edge_px}px;\"+\"fill:none}\")\n",
    "    # metadata.string = metadata.string.replace(\".direction-arrow{stroke:#000;stroke-width:1px;fill:#fff;opacity:.3}\",\n",
    "    #                                           \".direction-arrow{stroke:#000;\"+f\"stroke-width:{rxn_edge_px*8}px;fill:#fff;opacity:{tint_factor}\"+\"}\")\n",
    "    # metadata.string = metadata.string.replace(\"arrowhead{fill:#334E75}\",\n",
    "    #                                           \"arrowhead{fill:#000000\"+f\";stroke-width:{rxn_edge_px*8}px;\"+\"}\")\n",
    "    ## change miscellaneous text\n",
    "    metadata.string = metadata.string.replace(\".text-label-input{font-size:50px}\", \".text-label-input{font-size:\"+f\"{rxn_label_px*1.8}\"+\"px}\")\n",
    "    # text['font-weight'] = 'bold'\n",
    "\n",
    "    # optimize text locations\n",
    "    # labelSizes = [node_label_px, rxn_edge_px]\n",
    "    # adjust_labels(soup.find_all('text'), sum(labelSizes)/len(labelSizes))\n",
    "\n",
    "    ## re-order the labels such that the text overlays all other elements\n",
    "    # labels = soup.find_all('text')\n",
    "    # edges = soup.find_all('reaction')  # Assuming edges are defined by <path> elements\n",
    "    # nodes = soup.find_all('circle')  # Assuming nodes are defined by <circle> elements\n",
    "\n",
    "    # # Create a new SVG content with reordered elements\n",
    "    # new_svg = BeautifulSoup('<svg xmlns=\"http://www.w3.org/2000/svg\"></svg>', 'lxml-xml')\n",
    "    # new_svg.svg.attrs = soup.svg.attrs  # Preserve original SVG attributes\n",
    "\n",
    "    # # Function to append elements to new SVG\n",
    "    # def append_elements(elements):\n",
    "    #     for elem in elements:\n",
    "    #         # Find the parent <g> element if the element is nested\n",
    "    #         parent = elem.find_parent('g')\n",
    "    #         if parent and parent not in new_svg.svg:\n",
    "    #             new_svg.svg.append(parent)\n",
    "    #         elif elem not in new_svg.svg:\n",
    "    #             new_svg.svg.append(elem)\n",
    "\n",
    "    # # Append edges first\n",
    "    # append_elements(edges)\n",
    "\n",
    "    # # Append nodes second\n",
    "    # append_elements(nodes)\n",
    "\n",
    "    # # Append labels last\n",
    "    # append_elements(labels)\n",
    "\n",
    "    # # Write the modified SVG content to a new file with proper indentation\n",
    "    # with open(\"output.svg\", \"w\", encoding=\"utf-8\") as file:\n",
    "    #     file.write(new_svg.prettify())\n",
    "\n",
    "\n",
    "    # add ASV nodes\n",
    "    for edge in soup.find_all(class_=\"segment\"):\n",
    "        # stroke-dasharray=\"10,5\"   to implement dash or solid edges conditionally, based on the directionality of the exchange\n",
    "        coordinates = list(map(float, re.split(\",| \", edge[\"d\"].replace(\"M\", \"\").replace(\"M\", \"\").replace(\"C\", \"\").replace(\"undefined\", \"100000000\"))))\n",
    "        if abs(coordinates[0]-coordinates[2]) < 30 and abs(coordinates[1]-coordinates[3]) < 3:\n",
    "            # print(coordinates)\n",
    "            rect = soup.new_tag(\"rect\")\n",
    "            rect['x'] = str(coordinates[0]+float(\"10734.555053710938\")/2.03)\n",
    "            rect['y'] = str(coordinates[1]+float(\"5315.0328369140625\")/7.66)\n",
    "            rect['width'] = \"40\"\n",
    "            rect['height'] = \"15\"\n",
    "            rect['fill'] = f\"#{rxn_label_hex}\"\n",
    "            soup.svg.append(rect)\n",
    "            # print(edge)\n",
    "        # break\n",
    "    # change the color of select edges\n",
    "    if colorElements is not None:\n",
    "        # baseline_metadata = metadata.string\n",
    "        edgeColor = {}\n",
    "        for color, lst in colorElements.items():\n",
    "            edgeColor.update({x:color for x in lst})\n",
    "        for edge in soup.find_all(class_=\"segment-group\"):\n",
    "            if edge[\"id\"] in edgeColor:\n",
    "                color = edgeColor[edge[\"id\"]]\n",
    "                for child in edge.find_all(class_=\"segment\"):\n",
    "                    child['class'] = f\"m{color}\"\n",
    "                # for i, child in enumerate(edge.children):\n",
    "                #     if \"segment\" in child:\n",
    "                #         edge.children[i] = child.replace(\"segment\", f\"segment m{color}\")\n",
    "\n",
    "                # display([x for x in edge.children])\n",
    "                # break\n",
    "            elif largeEdgeLabels is not None:\n",
    "                for child in edge.find_all(class_=\"segment\"):\n",
    "                    child['class'] = f\"m{fadedEdgesColor}\"\n",
    "        for color, elements in colorElements.items():\n",
    "            # \".segment{\"+\"stroke:#334E75;stroke-width:10px;fill:none}\"\n",
    "            metadata.string = metadata.string + f\" .m{color}\"+\"{\"+f\"stroke:#{color};\"+f\"stroke-width:{rxn_edge_px*1.2}px;\"+\"fill:none}\"\n",
    "            if any(['r' in x for x in elements]):\n",
    "                for ele in elements:\n",
    "                    if 'r' in ele:\n",
    "                        reaction = soup.find(attrs={\"id\": ele})\n",
    "                        label = reaction.find(\"text\")\n",
    "                        label[\"class\"] = f\"m{color}-label\"\n",
    "                metadata.string = metadata.string + f\" .m{color}-label\"+\"{\"+f\"font-size:{rxn_label_px}px;fill:#{color};stroke:#000000;stroke-width:{rxn_label_px/20}px;\"+\"text-rendering:optimizelegibility}\"\n",
    "            if any(['n' in x for x in elements]):\n",
    "                for ele in elements:\n",
    "                    if 'n' in ele:\n",
    "                        node = soup.find(attrs={\"id\": ele})\n",
    "                        circle = node.find(\"circle\")\n",
    "                        circle[\"class\"] = f\"m{color}-circle\"\n",
    "                        label = node.find(\"text\")\n",
    "                        label[\"class\"] = f\"m{color}-label\"\n",
    "                metadata.string = metadata.string + f\" .m{color}-circle\"+\"{\"+f\"stroke:#{color};fill:#{color}\"+\"}\"\n",
    "                \n",
    "        if largeEdgeLabels is not None:\n",
    "            metadata.string = metadata.string + f\" .m{fadedEdgesColor}\"+\"{\"+f\"stroke:#{fadedEdgesColor};\"+f\"stroke-width:{rxn_edge_px/1.2}px;\"+\"fill:none}\"\n",
    "    \n",
    "    # change the color of select nodes\n",
    "\n",
    "\n",
    "    with open(svg_path.replace(\".svg\", \"_edited.svg\"), \"w\", encoding=\"utf-8\") as file:\n",
    "        file.write(soup.prettify())\n",
    "\n",
    "EscherSVG_processing(\"metabolite_focused_map_IDs.svg\", node_label_px=90, rxn_label_px=90, rxn_edge_px=6,\n",
    "                     node_label_hex=\"E36400\", rxn_label_hex=\"cc0000\", rxn_edge_hex=\"2C79BD\",\n",
    "                     largeNodeLabels=[\"n202\", \"n311\", \"n900\", \"n963\", \"n33\", \"n32\", \"n830\", \"n71\", \"n1322\", \"n1443\", \"n1672\", \"n1673\",\n",
    "                                      \"n2066\", \"n2102\", \"n206\", \"n404\", \"n873\", \"n1090\", \"n1182\", \"n1746\", \"n2154\", \"n2154\", \"n1338\", # RC\n",
    "                                      \"n539\", \"n1063\", \"n1282\", \"n1511\", \"n1576\" # cellibiose\n",
    "                                      \"n1540\", \"n249\",   # Melitose\n",
    "                                      \"n85\", \"n104\"   # WD\n",
    "                                      ],\n",
    "                     largeEdgeLabels=[\"r10\", \"r23\", \"r48\", \"r50\", \"r49\", \"r41\", \"r3\", \"r60\", \"r64\", \"r65\", \"r73\", \"r68\", \"r72\", \"r88\", \"r86\", \"r83\",\n",
    "                                      \"r95\", \"r108\", \"r106\", \"r107\",\n",
    "                                      \"r31\", \"r53\", \"r63\", \"r75\", \"r76\", \"r99\", \"r77\", \"r79\", \"r121\", \"s1435\", \"s1009\", \"s1218\",      # WD\n",
    "                                      ],\n",
    "                     abbrevIDs=cpdID_abbrv,\n",
    "                     colorElements={\"8E0FFF\":[\"s903\", \"s295\", \"s444\", \"s906\", \"s928\", \"s900\", \"s913\", \"s67\", \"s1246\", \"s1150\", \"s1254\", \"s1238\", \"s1390\",\n",
    "                                              \"s1370\", \"s1666\", \"s1637\", \"s1588\", \"s788\", \"s66\", \"s1589\", \"s1787\", \"s1777\", \"s387\", \"s915\",\n",
    "                                              \"s1170\", \"s1186\", \"s1152\", \"s2024\", \"s2072\", \"s2042\", \"s2024\", \"s2085\", \"s1993\"   # RC\n",
    "                                              \"s599\", \"s602\", \"s593\", \"s596\", \"s597\", \"s603\", \"s2195\", \"s1479\", \"s98\"    # WD\n",
    "                                           ],  # purple\n",
    "                                 \"11C911\":[\"s191\", \"s195\", \"r10\", \"r3\", \"r68\", \"r20\", \"s383\", \"s823\", \"s829\", \"n873\", \"r80\", \"r122\"],  # green\n",
    "                                 \"F15757\":[\"s1304\"]  # red\n",
    "                                 },\n",
    "                     msdb=msdb, tint_factor=0.6)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### create a custom ID mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 402,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<xml.etree.ElementTree.ElementTree at 0x162270670>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{http://www.w3.org/2000/svg}svg\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "'attrs' is an invalid keyword argument for find()",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn [402], line 8\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[38;5;66;03m# display(dir(root))\u001b[39;00m\n\u001b[1;32m      7\u001b[0m \u001b[38;5;28mprint\u001b[39m(root\u001b[38;5;241m.\u001b[39mtag)\n\u001b[0;32m----> 8\u001b[0m reactions \u001b[38;5;241m=\u001b[39m root\u001b[38;5;241m.\u001b[39mfind(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mg\u001b[39m\u001b[38;5;124m\"\u001b[39m, attrs\u001b[38;5;241m=\u001b[39m{\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mid\u001b[39m\u001b[38;5;124m'\u001b[39m: \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mreactions\u001b[39m\u001b[38;5;124m'\u001b[39m})\n\u001b[1;32m      9\u001b[0m \u001b[38;5;66;03m# for child in root:\u001b[39;00m\n\u001b[1;32m     10\u001b[0m \u001b[38;5;66;03m#     print(child.tag, child.attrib)\u001b[39;00m\n\u001b[1;32m     11\u001b[0m \u001b[38;5;66;03m#     if child.tag == \"{http://www.w3.org/2000/svg}g\":\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     15\u001b[0m \u001b[38;5;66;03m#                 for child3 in child2:\u001b[39;00m\n\u001b[1;32m     16\u001b[0m \u001b[38;5;66;03m#                     print(child3.tag, child3.attrib)\u001b[39;00m\n\u001b[1;32m     19\u001b[0m content \u001b[38;5;241m=\u001b[39m root\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mescher-svg\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[0;31mTypeError\u001b[0m: 'attrs' is an invalid keyword argument for find()"
     ]
    }
   ],
   "source": [
    "import xml.etree.ElementTree as et\n",
    "\n",
    "svgContent = et.parse(\"metabolite_focused_map.svg\")\n",
    "display(svgContent)\n",
    "root = svgContent.getroot()\n",
    "# display(dir(root))\n",
    "print(root.tag)\n",
    "reactions = root.find(\"g\", attrs={'id': 'reactions'})\n",
    "# for child in root:\n",
    "#     print(child.tag, child.attrib)\n",
    "#     if child.tag == \"{http://www.w3.org/2000/svg}g\":\n",
    "#         for child2 in child:\n",
    "#             print(child2.tag, child2.attrib)\n",
    "#             if child2.id == \"reactions\":\n",
    "#                 for child3 in child2:\n",
    "#                     print(child3.tag, child3.attrib)\n",
    "\n",
    "\n",
    "content = root.get(\"escher-svg\")\n",
    "display(content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
