{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Notable Narratives and content"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# collective repository\n",
    "- GitHub:   https://github.com/ModelSEED/MicrobiomeNotebooks.git\n",
    "\n",
    "# utilized Narratives\n",
    "- MAG genomes: https://narrative.kbase.us/narrative/155805 & https://narrative.kbase.us/narrative/163264\n",
    "- Pangenome-enhanced genomes: https://narrative.kbase.us/narrative/187797\n",
    "- Annotated pangenome-enhanced genomes: https://narrative.kbase.us/narrative/188040\n",
    "- Reconstructed MAG models: https://narrative.kbase.us/narrative/186678\n",
    "\n",
    "# previously-computed files\n",
    "- FAA of representative sequences: /scratch/fliu/data/cliff/mmseqs_ani_prob_rep_genome.faa\n",
    "- Cluster genome mapping: /scratch/fliu/data/cliff/mmseqs_ani_prob.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "genome_ws = [155805, 163264]\n",
    "mag_ws = 188065\n",
    "new_workspace = 187471\n",
    "pangenome_mag_ws = 187040\n",
    "prob_mapping = \"/scratch/fliu/data/cliff/mmseqs_ani_prob_v2.json\"\n",
    "prob_rep_mapping = \"/scratch/fliu/data/cliff/mmseqs_ani_prob_rep_genome_v2.faa\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Loading hit data from json file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "with open(\"/scratch/fliu/data/cliff/mmseqs_ani_prob_v2.json\", 'r') as f:\n",
    "    hit_data = json.load(f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parsing hits for close genome list and establishing thresholds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "mag_ani_data = {}\n",
    "mag_thresholds = {}\n",
    "mag_genomes = {}\n",
    "mag_hit_distribution = {}\n",
    "max_bins = 27\n",
    "min_sim = 74\n",
    "for protein in hit_data:\n",
    "    for mag in hit_data[protein]:\n",
    "        if mag not in mag_ani_data:\n",
    "            mag_ani_data[mag] = {}\n",
    "            mag_hit_distribution[mag] = [0] * max_bins\n",
    "        for hit in hit_data[protein][mag]:\n",
    "            genome = hit.split(\":\")[0]\n",
    "            if hit_data[protein][mag][hit] < 100 and hit_data[protein][mag][hit] >= min_sim:\n",
    "                if genome not in mag_ani_data[mag]:\n",
    "                    mag_hit_distribution[mag][int(hit_data[protein][mag][hit]-min_sim)] += 1\n",
    "                mag_ani_data[mag][genome] = hit_data[protein][mag][hit]\n",
    "util.save(\"mag_ani_data\",mag_ani_data)\n",
    "util.save(\"mag_hit_distribution\",mag_hit_distribution)\n",
    "records = []\n",
    "for mag in mag_hit_distribution:\n",
    "    if mag not in mag_thresholds:\n",
    "        mag_thresholds[mag] = {\"threshold\":0,\"threshold_count\":0}\n",
    "    record = {\"mag\":mag}\n",
    "    threshold = 0\n",
    "    for i in range(max_bins):\n",
    "        record[\"ani_\" + str(min_sim+i)] = mag_hit_distribution[mag][i]\n",
    "    threshold_count = 0\n",
    "    for i in range(max_bins):\n",
    "        index = max_bins-1-i  \n",
    "        if mag_hit_distribution[mag][index] > 0 and (threshold == 0 or threshold == min_sim+index+1) and threshold_count <= 4 and (threshold_count == 0 or mag_hit_distribution[mag][index]/threshold_count < 5):\n",
    "            threshold_count += mag_hit_distribution[mag][index]\n",
    "            threshold = min_sim+index\n",
    "    mag_thresholds[mag][\"threshold\"] = threshold\n",
    "    mag_thresholds[mag][\"threshold_count\"] = threshold_count\n",
    "    if threshold_count == 1:\n",
    "        new_threshold = 0\n",
    "        additional_count = 0\n",
    "        start = max_bins-(threshold-min_sim)\n",
    "        for i in range(start,max_bins,1):\n",
    "            index = max_bins-1-i\n",
    "            if mag_hit_distribution[mag][index] > 0 and (new_threshold == 0 or new_threshold == min_sim+index+1) and additional_count <= 4 and (additional_count == 0 or mag_hit_distribution[mag][index]/additional_count < 5):\n",
    "                additional_count += mag_hit_distribution[mag][index]\n",
    "                new_threshold = min_sim+index\n",
    "        if new_threshold != 0:\n",
    "            record[\"new_threshold\"] = new_threshold\n",
    "            record[\"new_threshold_count\"] = additional_count\n",
    "            if (threshold-new_threshold)<=10 or threshold != 100:\n",
    "                mag_thresholds[mag][\"threshold\"] = new_threshold\n",
    "                mag_thresholds[mag][\"threshold_count\"] = threshold_count+additional_count\n",
    "    record[\"threshold\"] = threshold\n",
    "    record[\"threshold_count\"] = threshold_count\n",
    "    records.append(record)\n",
    "for mag in mag_ani_data:\n",
    "    mag_genomes[mag] = {}\n",
    "    for genome in mag_ani_data[mag]:\n",
    "        if mag_ani_data[mag][genome] >= mag_thresholds[mag][\"threshold\"]:\n",
    "            mag_genomes[mag][genome] = mag_ani_data[mag][genome]\n",
    "df = pd.DataFrame.from_records(records)\n",
    "df.to_csv(util.output_dir+\"/mag_hit_distribution.csv\",index=False)\n",
    "util.save(\"mag_thresholds\",mag_thresholds)\n",
    "util.save(\"mag_genomes\",mag_genomes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Assigning thresholds manually (not needed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "mag_thresholds = util.load(\"mag_thresholds\")\n",
    "mag_ani_data = util.load(\"mag_ani_data\")\n",
    "mag_genomes = {}\n",
    "for mag in mag_ani_data:\n",
    "    mag_genomes[mag] = {}\n",
    "    for genome in mag_ani_data[mag]:\n",
    "        if mag_ani_data[mag][genome] >= mag_thresholds[mag][\"threshold\"]:\n",
    "            mag_genomes[mag][genome] = mag_ani_data[mag][genome]\n",
    "    mag_thresholds[mag][\"threshold_count\"] = len(mag_genomes[mag])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Getting workspace IDs for MAGs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "mag_wsids = {}\n",
    "for ws in genome_ws:\n",
    "    for mag in util.msrecon.kbase_api.list_objects(ws, object_type=\"KBaseGenomes.Genome\", include_metadata=False):\n",
    "        mag_wsids[mag[1]] = mag[7]\n",
    "util.save(\"mag_wsids\",mag_wsids)\n",
    "util.save(\"mag_list\",mag_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Identifying supplemental proteins for MAGs based on thresholds and hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[23], line 62\u001b[0m\n\u001b[1;32m     60\u001b[0m         all_mag_deep_data[magname][protein][\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtotal\u001b[39m\u001b[38;5;124m\"\u001b[39m][\u001b[38;5;241m1\u001b[39m] \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m\n\u001b[1;32m     61\u001b[0m         all_mag_deep_data[magname][protein][\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgenomes\u001b[39m\u001b[38;5;124m\"\u001b[39m][\u001b[38;5;241m1\u001b[39m] \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m\n\u001b[0;32m---> 62\u001b[0m     genomes_found[genome] \u001b[38;5;241m=\u001b[39m hit_data[protein][mag][hit]\n\u001b[1;32m     63\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m genomes_found[genome] \u001b[38;5;241m!=\u001b[39m hit_data[protein][mag][hit]:\n\u001b[1;32m     64\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mWarning: different similarity values for the same genome\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "mag_protein_supplements = {}\n",
    "mag_thresholds = util.load(\"mag_thresholds\")\n",
    "mag_genomes = util.load(\"mag_genomes\")\n",
    "mag_list = util.load(\"mag_list\")\n",
    "mag_wsids = util.load(\"mag_wsids\")\n",
    "magNames = util.load(\"mag_to_name\")\n",
    "all_mag_deep_data = {}\n",
    "for magname in mag_wsids:\n",
    "    all_mag_deep_data[magname] = {}\n",
    "for protein in hit_data:\n",
    "    for mag in hit_data[protein]:\n",
    "        magname = magNames[mag]+\".pg.G.D\"\n",
    "        if mag_thresholds[mag][\"threshold\"] > 0:\n",
    "            nr_mags_found = {}\n",
    "            rejected_mags_found = {}\n",
    "            genomes_found = {}\n",
    "            all_mag_deep_data[magname][protein] = {\n",
    "                \"self\":False,\n",
    "                \"total\":[0,0],\n",
    "                \"nr_mags\":[0,0],\n",
    "                \"rejected_mags\":[0,0],\n",
    "                \"genomes\":[0,0]\n",
    "            }\n",
    "            for hit in hit_data[protein][mag]:\n",
    "                (genome,gene) = hit.split(\":\")\n",
    "                if genome == \"self\":\n",
    "                    all_mag_deep_data[magname][protein][\"self\"] = True\n",
    "                elif genome in mag_list:\n",
    "                    if genome not in nr_mags_found:\n",
    "                        all_mag_deep_data[magname][protein][\"total\"][0] += 1\n",
    "                        all_mag_deep_data[magname][protein][\"nr_mags\"][0] += 1\n",
    "                        if hit_data[protein][mag][hit] >= mag_thresholds[mag][\"threshold\"]:\n",
    "                            if genome not in mag_genomes[mag]:\n",
    "                                print(genome+\":\"+mag+\":\"+str(hit_data[protein][mag][hit])+\"/\"+str(mag_thresholds[mag][\"threshold\"]))\n",
    "                            all_mag_deep_data[magname][protein][\"total\"][1] += 1\n",
    "                            all_mag_deep_data[magname][protein][\"nr_mags\"][1] += 1\n",
    "                        nr_mags_found[genome] = hit_data[protein][mag][hit]\n",
    "                    elif nr_mags_found[genome] != hit_data[protein][mag][hit]:\n",
    "                        print(\"Warning: different similarity values for the same genome\")\n",
    "                elif genome in mag_wsids:\n",
    "                    if genome not in rejected_mags_found:\n",
    "                        all_mag_deep_data[magname][protein][\"total\"][0] += 1\n",
    "                        all_mag_deep_data[magname][protein][\"rejected_mags\"][0] += 1\n",
    "                        if hit_data[protein][mag][hit] >= mag_thresholds[mag][\"threshold\"]:\n",
    "                            if genome not in mag_genomes[mag]:\n",
    "                                print(genome+\":\"+mag+\":\"+str(hit_data[protein][mag][hit])+\"/\"+str(mag_thresholds[mag][\"threshold\"]))\n",
    "                            all_mag_deep_data[magname][protein][\"total\"][1] += 1\n",
    "                            all_mag_deep_data[magname][protein][\"rejected_mags\"][1] += 1\n",
    "                        rejected_mags_found[genome] = hit_data[protein][mag][hit]\n",
    "                    elif rejected_mags_found[genome] != hit_data[protein][mag][hit]:\n",
    "                        print(\"Warning: different similarity values for the same genome\")\n",
    "                else:\n",
    "                    if genome not in genomes_found:\n",
    "                        all_mag_deep_data[magname][protein][\"total\"][0] += 1\n",
    "                        all_mag_deep_data[magname][protein][\"genomes\"][0] += 1\n",
    "                        if hit_data[protein][mag][hit] >= mag_thresholds[mag][\"threshold\"]:\n",
    "                            if genome not in mag_genomes[mag]:\n",
    "                                print(genome+\":\"+mag+\":\"+str(hit_data[protein][mag][hit])+\"/\"+str(mag_thresholds[mag][\"threshold\"]))\n",
    "                            all_mag_deep_data[magname][protein][\"total\"][1] += 1\n",
    "                            all_mag_deep_data[magname][protein][\"genomes\"][1] += 1\n",
    "                        genomes_found[genome] = hit_data[protein][mag][hit]\n",
    "                    elif genomes_found[genome] != hit_data[protein][mag][hit]:\n",
    "                        print(\"Warning: different similarity values for the same genome\")\n",
    "            #We only add supplemental proteins for a family that does not already have a protein in the MAG\n",
    "            if not all_mag_deep_data[magname][protein][\"self\"]:\n",
    "                if mag not in mag_protein_supplements:\n",
    "                    mag_protein_supplements[mag] = {}\n",
    "                mag_protein_supplements[mag][protein] = [all_mag_deep_data[magname][protein][\"total\"][1],all_mag_deep_data[magname][protein][\"nr_mags\"][1],all_mag_deep_data[magname][protein][\"rejected_mags\"][1]]\n",
    "for magname in all_mag_deep_data:\n",
    "    util.save(\"deepdata/\"+mag,all_mag_deep_data[magname])\n",
    "util.save(\"mag_protein_supplements\",mag_protein_supplements)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load functional data (not needed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/chenry/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/kb_sdk_home/run_local/workdir/tmp/\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "function_hit_data = json.load(open('/home/fliu/cliff_mags/data/annotation_ani_prob_lo_85.json'))\n",
    "mag_function_supplements = {}\n",
    "mag_thresholds = util.load(\"mag_thresholds\")\n",
    "mag_list = util.load(\"mag_list\")\n",
    "mag_wsids = util.load(\"mag_wsids\")\n",
    "all_func_mag_deep_data = {}\n",
    "for mag in mag_wsids:\n",
    "    all_func_mag_deep_data[mag] = {}\n",
    "for protein in function_hit_data:\n",
    "    for mag in function_hit_data[protein]:\n",
    "        nr_mags_found = {}\n",
    "        rejected_mags_found = {}\n",
    "        genomes_found = {}\n",
    "        all_func_mag_deep_data[mag][protein] = {\n",
    "            \"self\":False,\n",
    "            \"total\":[0,0],\n",
    "            \"nr_mags\":[0,0],\n",
    "            \"rejected_mags\":[0,0],\n",
    "            \"genomes\":[0,0]\n",
    "        }\n",
    "        for hit in function_hit_data[protein][mag]:\n",
    "            (genome,gene) = hit.split(\":\")\n",
    "            if genome == \"self\":\n",
    "                all_func_mag_deep_data[mag][protein][\"self\"] = True\n",
    "            elif genome in mag_list:\n",
    "                if genome not in nr_mags_found:\n",
    "                    all_func_mag_deep_data[mag][protein][\"total\"][0] += 1\n",
    "                    all_func_mag_deep_data[mag][protein][\"nr_mags\"][0] += 1\n",
    "                    if function_hit_data[protein][mag][hit] >= mag_thresholds[mag][\"threshold\"]/100:\n",
    "                        all_func_mag_deep_data[mag][protein][\"total\"][1] += 1\n",
    "                        all_func_mag_deep_data[mag][protein][\"nr_mags\"][1] += 1\n",
    "                    nr_mags_found[genome] = function_hit_data[protein][mag][hit]\n",
    "                elif nr_mags_found[genome] != function_hit_data[protein][mag][hit]:\n",
    "                    print(\"Warning: different similarity values for the same genome\")\n",
    "            elif genome in mag_wsids:\n",
    "                if genome not in rejected_mags_found:\n",
    "                    all_func_mag_deep_data[mag][protein][\"total\"][0] += 1\n",
    "                    all_func_mag_deep_data[mag][protein][\"rejected_mags\"][0] += 1\n",
    "                    if function_hit_data[protein][mag][hit] >= mag_thresholds[mag][\"threshold\"]/100:\n",
    "                        all_func_mag_deep_data[mag][protein][\"total\"][1] += 1\n",
    "                        all_func_mag_deep_data[mag][protein][\"rejected_mags\"][1] += 1\n",
    "                    rejected_mags_found[genome] = function_hit_data[protein][mag][hit]\n",
    "                elif rejected_mags_found[genome] != function_hit_data[protein][mag][hit]:\n",
    "                    print(\"Warning: different similarity values for the same genome\")\n",
    "            else:\n",
    "                if genome not in genomes_found:\n",
    "                    all_func_mag_deep_data[mag][protein][\"total\"][0] += 1\n",
    "                    all_func_mag_deep_data[mag][protein][\"genomes\"][0] += 1\n",
    "                    if function_hit_data[protein][mag][hit] >= mag_thresholds[mag][\"threshold\"]/100:\n",
    "                        all_func_mag_deep_data[mag][protein][\"total\"][1] += 1\n",
    "                        all_func_mag_deep_data[mag][protein][\"genomes\"][1] += 1\n",
    "                    genomes_found[genome] = function_hit_data[protein][mag][hit]\n",
    "                elif genomes_found[genome] != function_hit_data[protein][mag][hit]:\n",
    "                    print(\"Warning: different similarity values for the same genome\")\n",
    "        #We only add supplemental proteins for a family that does not already have a protein in the MAG\n",
    "        if not all_func_mag_deep_data[mag][protein][\"self\"]:\n",
    "            if mag not in mag_function_supplements:\n",
    "                mag_function_supplements[mag] = {}\n",
    "            mag_function_supplements[mag][protein] = [all_func_mag_deep_data[mag][protein][\"total\"][1],all_func_mag_deep_data[mag][protein][\"nr_mags\"][1],all_func_mag_deep_data[mag][protein][\"rejected_mags\"][1]]\n",
    "for mag in mag_function_supplements:\n",
    "    util.save(\"funcdeepdata/\"+mag,all_func_mag_deep_data[mag])\n",
    "util.save(\"mag_function_supplements\",mag_function_supplements)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computing stats on additional proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8.contigs__.RAST 0.1 14737\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20.contigs__.RAST 0.1 3860\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.4.contigs__.RAST 0.1 4004\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.21.contigs__.RAST 0.1 4192\n",
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.31.contigs__.RAST 0.1 5106\n",
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.29.contigs__.RAST 0.1 3670\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.50.contigs__.RAST 0.1 4271\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.18.contigs__.RAST 0.1 4144\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.15.contigs__.RAST 0.1 1457\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.88.contigs__.RAST 0.1 2330\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.57.contigs__.RAST 0.1 1738\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10.contigs__.RAST 0.1 2713\n",
      "Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44.contigs__.RAST 0.1 3227\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.51.contigs__.RAST 0.1 1178\n",
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_concoct_out.69.contigs__.RAST 0.1 1707\n",
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_metabat.16.contigs__.RAST 0.1 552\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.18.contigs__.RAST 0.1 1601\n",
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_metabat.10.contigs__.RAST 0.1 526\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.25.contigs__.RAST 0.1 4054\n",
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.67.contigs__.RAST 0.1 4079\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.25.contigs__.RAST 0.1 1642\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.50.contigs__.RAST 0.1 3743\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.29.contigs__.RAST 0.1 2511\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.38.contigs__.RAST 0.1 7764\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.80.contigs__.RAST 0.1 1130\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_concoct_out.13.contigs__.RAST 0.1 1058\n",
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45.contigs__.RAST 0.1 4868\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.17.contigs__.RAST 0.15 12879\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4.contigs__.RAST 0.1 11687\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28.contigs__.RAST 0.1 2667\n",
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_concoct_out.62.contigs__.RAST 0.1 3462\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.42.contigs__.RAST 0.1 3490\n",
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14.contigs__.RAST 0.1 12246\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.17.contigs__.RAST 0.1 2856\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91.contigs__.RAST 0.1 1994\n",
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22.contigs__.RAST 0.1 12202\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.34.contigs__.RAST 0.1 6051\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.19.contigs__.RAST 0.1 17763\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.37.contigs__.RAST 0.1 13720\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.1.contigs__.RAST 0.1 770\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16.contigs__.RAST 0.1 7787\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.52.contigs__.RAST 0.1 4103\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.120.contigs__.RAST 0.1 4530\n",
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.27.contigs__.RAST 0.1 19814\n",
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.45.contigs__.RAST 0.1 4509\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.9.contigs__.RAST 0.1 10271\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.46.contigs__.RAST 0.1 8976\n",
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6.contigs__.RAST 0.1 9104\n",
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_concoct_out.67.contigs__.RAST 0.1 2987\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.45.contigs__.RAST 0.1 6886\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15.contigs__.RAST 0.1 3915\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.69.contigs__.RAST 0.1 4127\n",
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_metabat.14.contigs__.RAST 0.1 7188\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.23.contigs__.RAST 0.1 13334\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.48.contigs__.RAST 0.1 3590\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.73.contigs__.RAST 0.1 9701\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.143.contigs__.RAST 0.1 4194\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.78.contigs__.RAST 0.1 765\n",
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_concoct_out.67.contigs__.RAST 0.1 7983\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12.contigs__.RAST 0.1 3828\n",
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.18.contigs__.RAST 0.1 4579\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.40.contigs__.RAST 0.1 5366\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.87.contigs__.RAST 0.1 4720\n",
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.16.contigs__.RAST 0.1 2249\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19.contigs__.RAST 0.1 1531\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_maxbin.009.contigs__.RAST 0.1 4035\n",
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33.contigs__.RAST 0.1 1541\n",
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10.contigs__.RAST 0.1 8126\n",
      "Salt_Pond_MetaG_R2_B_H2O_MG_DASTool_bins_metabat.17.contigs__.RAST 0.1 4179\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.51.contigs__.RAST 0.1 3660\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.30.contigs__.RAST 0.1 6809\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.42.contigs__.RAST 0.1 5624\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.26.contigs__.RAST 0.1 1764\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.68.contigs__.RAST 0.1 17088\n",
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19.contigs__.RAST 0.1 1389\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.44.contigs__.RAST 0.1 5876\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.63.contigs__.RAST 0.1 1080\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.7.contigs__.RAST 0.1 10147\n",
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_concoct_out.59.contigs__.RAST 0.1 8104\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.2.contigs__.RAST 0.1 6308\n",
      "Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42.contigs__.RAST 0.1 11945\n",
      "Salt_Pond_MetaG_R2A_B_D1_MG_DASTool_bins_concoct_out.4.contigs__.RAST 0.1 11102\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.9.contigs__.RAST 0.1 7971\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8.contigs__.RAST 0.1 9888\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.6.contigs__.RAST 0.1 1836\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13.contigs__.RAST 0.1 8654\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.19.contigs__.RAST 0.1 4445\n",
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.19.contigs__.RAST 0.1 4457\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.51.contigs__.RAST 0.1 4795\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.7.contigs__.RAST 0.1 7154\n",
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_metabat.29.contigs__.RAST 0.1 4688\n",
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_concoct_out.34.contigs__.RAST 0.1 1332\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.32.contigs__.RAST 0.1 4894\n",
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_concoct_out.45.contigs__.RAST 0.1 10269\n",
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.100.contigs__.RAST 0.1 2770\n",
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.16.contigs__.RAST 0.1 1746\n",
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_metabat.42.contigs__.RAST 0.1 647\n",
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28.contigs__.RAST 0.1 8488\n",
      "Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13.contigs__.RAST 0.1 2035\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.15.contigs__.RAST 0.1 10326\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11.contigs__.RAST 0.1 10389\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.26.contigs__.RAST 0.1 5937\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.89.contigs__.RAST 0.1 4252\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28.contigs__.RAST 0.1 4318\n",
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13.contigs__.RAST 0.1 4140\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.25.contigs__.RAST 0.1 3750\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.39.contigs__.RAST 0.1 2514\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.89.contigs__.RAST 0.1 10168\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.45.contigs__.RAST 0.1 3140\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_maxbin.001.contigs__.RAST 0.1 2215\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.37.contigs__.RAST 0.1 1956\n",
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_metabat.23.contigs__.RAST 0.1 1715\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.95.contigs__.RAST 0.15 7235\n",
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_concoct_out.80.contigs__.RAST 0.1 2464\n",
      "Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9.contigs__.RAST 0.1 9423\n",
      "Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23.contigs__.RAST 0.1 4575\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.concoct_out.44.contigs__.RAST 0.1 2614\n",
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.49.contigs__.RAST 0.1 491\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.4.contigs__.RAST 0.1 3572\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38.contigs__.RAST 0.1 4211\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.36.contigs__.RAST 0.1 3144\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.58.contigs__.RAST 0.1 840\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_metabat.11.contigs__.RAST 0.1 5495\n",
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.11.contigs__.RAST 0.1 4269\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.8.contigs__.RAST 0.1 10597\n",
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_maxbin.002.contigs__.RAST 0.1 1469\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.119.contigs__.RAST 0.1 1908\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.39.contigs__.RAST 0.1 7301\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_metabat.17.contigs__.RAST 0.1 1611\n",
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.47.contigs__.RAST 0.1 4654\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.21.contigs__.RAST 0.1 11100\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5.contigs__.RAST 0.1 6377\n",
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_concoct_out.78.contigs__.RAST 0.1 8338\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39.contigs__.RAST 0.1 1096\n",
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_maxbin.029.contigs__.RAST 0.1 3647\n",
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23.contigs__.RAST 0.1 887\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45.contigs__.RAST 0.1 7845\n",
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_metabat.5.contigs__.RAST 0.1 556\n",
      "Salt_Pond_MetaGSF2_C_H2O_MG_DASTool_bins_metabat.29.contigs__.RAST 0.1 1097\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.27.contigs__.RAST 0.1 2113\n",
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40.contigs__.RAST 0.1 3981\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.53.contigs__.RAST 0.1 2440\n",
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_metabat.14.contigs__.RAST 0.1 4191\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65.contigs__.RAST 0.1 9238\n",
      "Salt_Pond_MetaG_R2_B_H2O_MG_DASTool_bins_metabat.29.contigs__.RAST 0.1 591\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.11.contigs__.RAST 0.1 2988\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18.contigs__.RAST 0.1 3452\n",
      "Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_metabat.29.contigs__.RAST 0.1 352\n",
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_concoct_out.16.contigs__.RAST 0.1 571\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_maxbin.062.contigs__.RAST 0.1 6108\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.10.contigs__.RAST 0.1 10540\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11.contigs__.RAST 0.1 2899\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.concoct_out.54.contigs__.RAST 0.1 7022\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.50.contigs__.RAST 0.1 3801\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25.contigs__.RAST 0.1 3131\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40.contigs__.RAST 0.1 1793\n",
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_metabat.8.contigs__.RAST 0.1 4899\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.49.contigs__.RAST 0.1 10297\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.20.contigs__.RAST 0.1 752\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_metabat.5.contigs__.RAST 0.1 4011\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25.contigs__.RAST 0.1 4998\n",
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_maxbin.005.contigs__.RAST 0.1 3128\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.98.contigs__.RAST 0.1 675\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.45.contigs__.RAST 0.1 9497\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.22.contigs__.RAST 0.1 8159\n",
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_maxbin.047.contigs__.RAST 0.1 3398\n",
      "Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_concoct_out.60.contigs__.RAST 0.1 3690\n",
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.22.contigs__.RAST 0.1 1781\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_maxbin.022.contigs__.RAST 0.1 5485\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24.contigs__.RAST 0.1 1043\n",
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.37.contigs__.RAST 0.1 10735\n",
      "Salt_Pond_MetaGSF2_C_H2O_MG_DASTool_bins_metabat.16.contigs__.RAST 0.1 1230\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.18.contigs__.RAST 0.1 11145\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.109.contigs__.RAST 0.1 9491\n",
      "Salt_Pond_MetaG_R2_B_H2O_MG_DASTool_bins_metabat.16.contigs__.RAST 0.1 4400\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.19.contigs__.RAST 0.1 10178\n",
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30.contigs__.RAST 0.1 1342\n",
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.40.contigs__.RAST 0.1 7966\n",
      "Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_metabat.20.contigs__.RAST 0.1 4850\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.59.contigs__.RAST 0.1 9419\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.43.contigs__.RAST 0.1 11786\n",
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.58.contigs__.RAST 0.1 250\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.17.contigs__.RAST 0.1 12216\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.concoct_out.9.contigs__.RAST 0.1 5455\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33.contigs__.RAST 0.1 4450\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.34.contigs__.RAST 0.1 1077\n",
      "Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55.contigs__.RAST 0.1 1024\n",
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.11.contigs__.RAST 0.1 1395\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28.contigs__.RAST 0.1 9227\n",
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_metabat.6.contigs__.RAST 0.1 1645\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16.contigs__.RAST 0.1 4132\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_maxbin.028.contigs__.RAST 0.1 3787\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49.contigs__.RAST 0.1 4372\n",
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.9.contigs__.RAST 0.1 6576\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.7.contigs__.RAST 0.15 8394\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13.contigs__.RAST 0.1 2270\n",
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_concoct_out.44.contigs__.RAST 0.1 1294\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.84.contigs__.RAST 0.1 5422\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31.contigs__.RAST 0.1 3853\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.37.contigs__.RAST 0.1 6045\n",
      "Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_concoct_out.35.contigs__.RAST 0.1 9582\n",
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.20.contigs__.RAST 0.1 1421\n",
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34.contigs__.RAST 0.1 1723\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.51.contigs__.RAST 0.1 911\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.3.contigs__.RAST 0.1 2642\n",
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.25.contigs__.RAST 0.1 2766\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.8.contigs__.RAST 0.15 10535\n",
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_metabat.4.contigs__.RAST 0.1 1760\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14.contigs__.RAST 0.1 1393\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.11.contigs__.RAST 0.1 8787\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.51.contigs__.RAST 0.1 1970\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32.contigs__.RAST 0.1 5166\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17.contigs__.RAST 0.1 1544\n",
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85.contigs__.RAST 0.1 1296\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.55.contigs__.RAST 0.1 4325\n",
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_maxbin.035.contigs__.RAST 0.1 311\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14.contigs__.RAST 0.1 7677\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29.contigs__.RAST 0.1 4008\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.17.contigs__.RAST 0.1 9913\n",
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_concoct_out.79.contigs__.RAST 0.1 5133\n",
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.54.contigs__.RAST 0.1 2713\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.21.contigs__.RAST 0.1 12251\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.42.contigs__.RAST 0.1 3277\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.concoct_out.5.contigs__.RAST 0.1 3773\n",
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23.contigs__.RAST 0.1 3415\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13.contigs__.RAST 0.1 2634\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.41.contigs__.RAST 0.1 12087\n",
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_metabat.16.contigs__.RAST 0.1 9561\n",
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_concoct_out.2.contigs__.RAST 0.1 1451\n",
      "Salt_Pond_MetaGSF2_C_H2O_MG_DASTool_bins_metabat.49.contigs__.RAST 0.1 4030\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.27.contigs__.RAST 0.1 1436\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.52.contigs__.RAST 0.1 436\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.124.contigs__.RAST 0.1 2107\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_concoct_out.22.contigs__.RAST 0.1 10477\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41.contigs__.RAST 0.45 19372\n",
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.21.contigs__.RAST 0.1 638\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.38.contigs__.RAST 0.1 5384\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.66.contigs__.RAST 0.1 953\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.52.contigs__.RAST 0.1 1832\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.27.contigs__.RAST 0.1 13032\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16.contigs__.RAST 0.35 17490\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.38.contigs__.RAST 0.1 6478\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28.contigs__.RAST 0.1 1049\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.25.contigs__.RAST 0.1 537\n",
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_metabat.14.contigs__.RAST 0.1 1923\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014.contigs__.RAST 0.1 2998\n",
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36.contigs__.RAST 0.1 2672\n",
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.110.contigs__.RAST 0.1 3013\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47.contigs__.RAST 0.1 543\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50.contigs__.RAST 0.1 9147\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17.contigs__.RAST 0.1 3891\n",
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.10.contigs__.RAST 0.1 10559\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.16.contigs__.RAST 0.1 11613\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.24.contigs__.RAST 0.1 10255\n",
      "Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4.contigs__.RAST 0.1 3705\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55.contigs__.RAST 0.1 5072\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.84.contigs__.RAST 0.1 2519\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.16.contigs__.RAST 0.1 3782\n",
      "Salt_Pond_MetaGSF2_C_H2O_MG_DASTool_bins_metabat.13.contigs__.RAST 0.1 3338\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.18.contigs__.RAST 0.1 869\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.10.contigs__.RAST 0.1 3442\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.35.contigs__.RAST 0.1 9349\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.40.contigs__.RAST 0.1 4159\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_concoct_out.19.contigs__.RAST 0.1 637\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.32.contigs__.RAST 0.1 3215\n",
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_maxbin.017.contigs__.RAST 0.1 4145\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.93.contigs__.RAST 0.1 529\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.15.contigs__.RAST 0.1 1175\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.10.contigs__.RAST 0.1 10805\n",
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.51.contigs__.RAST 0.1 2509\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.8.contigs__.RAST 0.1 4095\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "mag_list = util.load(\"mag_list\")\n",
    "mag_thresholds = util.load(\"mag_thresholds\")\n",
    "#mag_protein_supplements = util.load(\"mag_protein_supplements\")\n",
    "records = []\n",
    "mag_probability_threshold = {}\n",
    "for mag in mag_list:\n",
    "    if mag[1] in mag_protein_supplements:\n",
    "        count = mag_thresholds[mag[1]][\"threshold_count\"]\n",
    "        record = {\"mag\":mag[1]}\n",
    "        records.append(record)\n",
    "        abundance_count = [0] * 400\n",
    "        for protein in mag_protein_supplements[mag[1]]:\n",
    "            fraction = mag_protein_supplements[mag[1]][protein][0]/count\n",
    "            entry = int(fraction*100/5)\n",
    "            abundance_count[entry] += 1\n",
    "        total = 0\n",
    "        final_total = None\n",
    "        mag_probability_threshold[mag[1]] = 0.1\n",
    "        for (i,entry) in enumerate(abundance_count):\n",
    "            index = len(abundance_count)-i-1\n",
    "            lasttotal = total\n",
    "            total += abundance_count[index]\n",
    "            if total >= 20000 and final_total == None:\n",
    "                threshold = 5*(index+1)/100\n",
    "                if threshold >= 0.1:\n",
    "                    mag_probability_threshold[mag[1]] = threshold\n",
    "                    final_total = lasttotal\n",
    "            if index == 2 and final_total == None:\n",
    "                final_total = total\n",
    "        print(mag[1],mag_probability_threshold[mag[1]],final_total)\n",
    "        for (i,entry) in enumerate(abundance_count):\n",
    "            record[i] = entry\n",
    "util.save(\"mag_probability_threshold\",mag_probability_threshold)\n",
    "df = pd.DataFrame.from_records(records)\n",
    "df.to_csv(util.output_dir+\"/protein_count_abundance.csv\",index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Renaming MAGs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/Users/chenry/code/notebooks/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1722057900.422139 ERROR: Requested data mag_list doesn't exist at /Users/chenry/code/notebooks/MicrobiomeNotebooks/Cliff/datacache/mag_list.json\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ModelSEED: /Users/chenry/code//kb_sdk/run_local/workdir/tmp/\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "Requested data mag_list doesn't exist at /Users/chenry/code/notebooks/MicrobiomeNotebooks/Cliff/datacache/mag_list.json",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 4\u001b[0m\n\u001b[1;32m      2\u001b[0m mag_to_name \u001b[38;5;241m=\u001b[39m {}\n\u001b[1;32m      3\u001b[0m name_to_mag \u001b[38;5;241m=\u001b[39m {}\n\u001b[0;32m----> 4\u001b[0m mag_list \u001b[38;5;241m=\u001b[39m \u001b[43mutil\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mload\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmag_list\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m mag \u001b[38;5;129;01min\u001b[39;00m mag_list:\n\u001b[1;32m      6\u001b[0m     original_name \u001b[38;5;241m=\u001b[39m mag[\u001b[38;5;241m1\u001b[39m]\n",
      "File \u001b[0;32m~/code//chenry_utility_module/lib/chenry_utility_module/kbdevutils.py:147\u001b[0m, in \u001b[0;36mKBDevUtils.load\u001b[0;34m(self, name, default)\u001b[0m\n\u001b[1;32m    145\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m default \u001b[38;5;241m==\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    146\u001b[0m         logger\u001b[38;5;241m.\u001b[39merror(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRequested data \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39mname\u001b[38;5;241m+\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m doesn\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mt exist at \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39mfilename)\n\u001b[0;32m--> 147\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m(\u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRequested data \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39mname\u001b[38;5;241m+\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m doesn\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mt exist at \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39mfilename))\n\u001b[1;32m    148\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m default\n\u001b[1;32m    149\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mopen\u001b[39m(filename, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124m'\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m f:\n",
      "\u001b[0;31mValueError\u001b[0m: Requested data mag_list doesn't exist at /Users/chenry/code/notebooks/MicrobiomeNotebooks/Cliff/datacache/mag_list.json"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "mag_to_name = {}\n",
    "name_to_mag = {}\n",
    "mag_list = util.load(\"mag_list\")\n",
    "for mag in mag_list:\n",
    "    original_name = mag[1]\n",
    "    new_name = original_name\n",
    "    new_name = new_name.replace(\"Salt_Pond_MetaG_\", \"\", 1)\n",
    "    new_name = new_name.replace(\"Salt_Pond_MetaG\", \"\", 1)\n",
    "    new_name = new_name.replace(\"_MG_DASTool_bins_\", \"\", 1)\n",
    "    new_name = new_name.replace(\"metabat\", \"\", 1)\n",
    "    new_name = new_name.replace(\"concoct_out\", \"\", 1)\n",
    "    new_name = new_name.replace(\"maxbin\", \"\", 1)\n",
    "    new_name = new_name.replace(\".contigs__.RAST\", \"\", 1)\n",
    "    mag_to_name[mag[1]] = new_name\n",
    "    name_to_mag[new_name] = mag[1]\n",
    "    print(name_to_mag)\n",
    "util.save(\"mag_to_name\",mag_to_name)\n",
    "util.save(\"name_to_mag\",name_to_mag)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Loading protein sequence data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/chenry/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/kb_sdk_home/run_local/workdir/tmp/\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "from Bio import SeqIO\n",
    "protein_hash = {}\n",
    "for record in SeqIO.parse('/scratch/fliu/data/cliff/mmseqs_ani_prob_rep_genome_v2.faa', 'fasta'):\n",
    "    protein_hash[record.id] = record.seq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Building genomes and assemblies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/chenry/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/kb_sdk_home/run_local/workdir/tmp/\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8.contigs__.RAST 14738\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20.contigs__.RAST 3861\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.4.contigs__.RAST 4005\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.46.contigs__.RAST 4005\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.21.contigs__.RAST 4193\n",
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.31.contigs__.RAST 5107\n",
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.29.contigs__.RAST 3671\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.50.contigs__.RAST 4272\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.18.contigs__.RAST 4145\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.32.contigs__.RAST 4145\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.15.contigs__.RAST 1458\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.88.contigs__.RAST 2331\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.57.contigs__.RAST 1739\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10.contigs__.RAST 2714\n",
      "Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44.contigs__.RAST 3228\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.51.contigs__.RAST 1179\n",
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_concoct_out.69.contigs__.RAST 1708\n",
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_metabat.16.contigs__.RAST 553\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.18.contigs__.RAST 1602\n",
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_metabat.10.contigs__.RAST 527\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.25.contigs__.RAST 4055\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.86.contigs__.RAST 4055\n",
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.67.contigs__.RAST 4080\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.25.contigs__.RAST 1643\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.50.contigs__.RAST 3744\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.2.contigs__.RAST 3744\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.29.contigs__.RAST 2512\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.38.contigs__.RAST 7765\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.80.contigs__.RAST 1131\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_concoct_out.13.contigs__.RAST 1059\n",
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45.contigs__.RAST 4869\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.17.contigs__.RAST 12880\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4.contigs__.RAST 11688\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28.contigs__.RAST 2668\n",
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_concoct_out.62.contigs__.RAST 3463\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.42.contigs__.RAST 3491\n",
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14.contigs__.RAST 12247\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.17.contigs__.RAST 2857\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91.contigs__.RAST 1995\n",
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22.contigs__.RAST 12203\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.34.contigs__.RAST 6052\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.19.contigs__.RAST 17764\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.37.contigs__.RAST 13721\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.1.contigs__.RAST 771\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16.contigs__.RAST 7788\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.52.contigs__.RAST 4104\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.120.contigs__.RAST 4531\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.34.contigs__.RAST 1078\n",
      "Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55.contigs__.RAST 1025\n",
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.11.contigs__.RAST 1396\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28.contigs__.RAST 9228\n",
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_metabat.6.contigs__.RAST 1646\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16.contigs__.RAST 4133\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_maxbin.028.contigs__.RAST 3788\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49.contigs__.RAST 4373\n",
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.9.contigs__.RAST 6577\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.7.contigs__.RAST 8395\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13.contigs__.RAST 2271\n",
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_concoct_out.44.contigs__.RAST 1295\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.84.contigs__.RAST 5423\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31.contigs__.RAST 3854\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.37.contigs__.RAST 6046\n",
      "Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_concoct_out.35.contigs__.RAST 9583\n",
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.20.contigs__.RAST 1422\n",
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34.contigs__.RAST 1724\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.51.contigs__.RAST 912\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.3.contigs__.RAST 2643\n",
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.25.contigs__.RAST 2767\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.8.contigs__.RAST 10536\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.16.contigs__.RAST 10536\n",
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_metabat.4.contigs__.RAST 1761\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14.contigs__.RAST 1394\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.11.contigs__.RAST 8788\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.51.contigs__.RAST 1971\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32.contigs__.RAST 5167\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17.contigs__.RAST 1545\n",
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85.contigs__.RAST 1297\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.20.contigs__.RAST 1297\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.55.contigs__.RAST 4326\n",
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_maxbin.035.contigs__.RAST 312\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14.contigs__.RAST 7678\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.37.contigs__.RAST 7678\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29.contigs__.RAST 4009\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.17.contigs__.RAST 9914\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.8.contigs__.RAST 9914\n",
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_concoct_out.79.contigs__.RAST 5134\n",
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.54.contigs__.RAST 2714\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.21.contigs__.RAST 12252\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.42.contigs__.RAST 3278\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.concoct_out.5.contigs__.RAST 3774\n",
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23.contigs__.RAST 3416\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13.contigs__.RAST 2635\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.41.contigs__.RAST 12088\n",
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_metabat.16.contigs__.RAST 9562\n",
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_concoct_out.2.contigs__.RAST 1452\n",
      "Salt_Pond_MetaGSF2_C_H2O_MG_DASTool_bins_metabat.49.contigs__.RAST 4031\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.27.contigs__.RAST 1437\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.52.contigs__.RAST 437\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.24.contigs__.RAST 437\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.124.contigs__.RAST 2108\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_concoct_out.22.contigs__.RAST 10478\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41.contigs__.RAST 19373\n",
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.21.contigs__.RAST 639\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.38.contigs__.RAST 5385\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.6.contigs__.RAST 5385\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.66.contigs__.RAST 954\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.52.contigs__.RAST 1833\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.27.contigs__.RAST 13033\n",
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.7.contigs__.RAST 13033\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16.contigs__.RAST 17491\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.38.contigs__.RAST 6479\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28.contigs__.RAST 1050\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.25.contigs__.RAST 538\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.62.contigs__.RAST 538\n",
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_metabat.14.contigs__.RAST 1924\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014.contigs__.RAST 2999\n",
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36.contigs__.RAST 2673\n",
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.110.contigs__.RAST 3014\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.7.contigs__.RAST 3014\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.12.contigs__.RAST 3014\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47.contigs__.RAST 544\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_maxbin.037.contigs__.RAST 544\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50.contigs__.RAST 9148\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17.contigs__.RAST 3892\n",
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.10.contigs__.RAST 10560\n",
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.47.contigs__.RAST 10560\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.16.contigs__.RAST 11614\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.24.contigs__.RAST 10256\n",
      "Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4.contigs__.RAST 3706\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55.contigs__.RAST 5073\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.36.contigs__.RAST 5073\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.84.contigs__.RAST 2520\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.16.contigs__.RAST 3783\n",
      "Salt_Pond_MetaGSF2_C_H2O_MG_DASTool_bins_metabat.13.contigs__.RAST 3339\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.18.contigs__.RAST 870\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.10.contigs__.RAST 3443\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.35.contigs__.RAST 9350\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.40.contigs__.RAST 4160\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_concoct_out.19.contigs__.RAST 638\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.32.contigs__.RAST 3216\n",
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.104.contigs__.RAST 3216\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.12.contigs__.RAST 3216\n",
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_maxbin.017.contigs__.RAST 4146\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.93.contigs__.RAST 530\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.15.contigs__.RAST 1176\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.10.contigs__.RAST 10806\n",
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.51.contigs__.RAST 2510\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.8.contigs__.RAST 4096\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "unfiltered_list = util.load(\"mag_list\")\n",
    "filter = [\n",
    "  \"Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.17.contigs__.RAST\",\n",
    "  \"Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.52.contigs__.RAST\",\n",
    "  \"Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.38.contigs__.RAST\",\n",
    "  \"Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.27.contigs__.RAST\",\n",
    "  \"Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.25.contigs__.RAST\",\n",
    "  \"Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.110.contigs__.RAST\",\n",
    "  \"Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47.contigs__.RAST\",\n",
    "  \"Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.10.contigs__.RAST\",\n",
    "  \"Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55.contigs__.RAST\",\n",
    "  \"Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.32.contigs__.RAST\"\n",
    "]\n",
    "mag_list = []\n",
    "for item in unfiltered_list:\n",
    "    if item[1] in filter or len(filter) == 0:\n",
    "        mag_list.append(item)\n",
    "name_to_mag = util.load(\"name_to_mag\")\n",
    "mag_thresholds = util.load(\"mag_thresholds\")\n",
    "mag_probability_threshold = util.load(\"mag_probability_threshold\")\n",
    "#mag_protein_supplements = util.load(\"mag_protein_supplements\")\n",
    "for item in mag_list:\n",
    "#for name in problem_list:\n",
    "    full_data = util.get_object(item[1],item[7])\n",
    "    genome_obj = full_data[\"data\"]\n",
    "    genome_obj[\"assembly_ref\"] = str(item[6])+\"/\"+str(item[0])+\"/\"+str(item[4])+\";\"+genome_obj[\"assembly_ref\"]\n",
    "    name = mag_to_name[item[1]]\n",
    "    cds_hash = {}\n",
    "    for i,ftr in enumerate(genome_obj[\"cdss\"]):\n",
    "        cds_hash[ftr[\"id\"]] = ftr\n",
    "    ftr_hash = {}\n",
    "    for i,ftr in enumerate(genome_obj[\"features\"]):\n",
    "        ftr_hash[ftr[\"id\"]] = ftr\n",
    "        ftr[\"id\"] = name+\"_\"+str(i+1)\n",
    "        if \"cdss\" in ftr:\n",
    "            for j,cds in enumerate(ftr[\"cdss\"]):\n",
    "                cds_hash[cds][\"id\"] = ftr[\"id\"]+\".CDS\"\n",
    "                ftr[\"cdss\"][j] = cds_hash[cds][\"id\"]\n",
    "                cds_hash[cds][\"parent_gene\"] = ftr[\"id\"]\n",
    "    firstgene = genome_obj[\"features\"][0]\n",
    "    count = 0\n",
    "    if item[1] in mag_protein_supplements:\n",
    "        count = 1\n",
    "        total_genomes = mag_thresholds[item[1]][\"threshold_count\"]\n",
    "        for protein in mag_protein_supplements[item[1]]:\n",
    "            if mag_protein_supplements[item[1]][protein][0]/total_genomes >= mag_probability_threshold[item[1]]:\n",
    "                ftrid = name+'.pg_'+str(count)\n",
    "                count += 1\n",
    "                protseq = str(protein_hash[protein])\n",
    "                dnaseq = util.translate_protein_to_gene(protseq)\n",
    "                result = hashlib.md5(protseq.encode())\n",
    "                md5 = result.hexdigest()\n",
    "                result = hashlib.md5(dnaseq.encode())\n",
    "                dnamd5 = result.hexdigest()\n",
    "                newftr = {\n",
    "                    \"aliases\": [[\"MMseqMD5\",protein]],\n",
    "                    \"cdss\": [\n",
    "                        ftrid+\".CDS\"\n",
    "                    ],\n",
    "                    \"functions\":[\"Hypothetical protein\"],\n",
    "                    \"dna_sequence\": dnaseq,\n",
    "                    \"dna_sequence_length\": 3*len(protseq),\n",
    "                    \"id\": ftrid,\n",
    "                    \"location\": [\n",
    "                        [\n",
    "                            firstgene[\"location\"][0][0],\n",
    "                            1,\n",
    "                            \"+\",\n",
    "                            3*len(protseq)\n",
    "                        ]\n",
    "                    ],\n",
    "                    \"md5\": dnamd5,\n",
    "                    \"ontology_terms\": {},\n",
    "                    \"protein_md5\": md5,\n",
    "                    \"protein_translation\": protseq,\n",
    "                    \"protein_translation_length\": len(protseq),\n",
    "                    \"warnings\": []\n",
    "                }\n",
    "                cdsftr = newftr.copy()\n",
    "                del cdsftr[\"cdss\"]\n",
    "                cdsftr[\"id\"] = ftrid+\".CDS\"\n",
    "                cdsftr[\"parent_gene\"] = ftrid\n",
    "                genome_obj[\"features\"].append(newftr)\n",
    "                genome_obj[\"cdss\"].append(cdsftr)\n",
    "        print(item[1],count)\n",
    "        #Saving MAG\n",
    "        util.save(\"genome/\"+name,full_data)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Finding and fixing redundant names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run cliffcommutil.py\n",
    "mag_list = util.load(\"mag_list\")\n",
    "mag_to_name = util.load(\"mag_to_name\")\n",
    "name_to_mag = {}\n",
    "for item in mag_list:\n",
    "    if item[1] not in mag_to_name:\n",
    "        print(\"Missing:\",item[1])\n",
    "    elif mag_to_name[item[1]] in name_to_mag:\n",
    "        print(\"Collision:\",mag_to_name[item[1]])\n",
    "    else:\n",
    "        name_to_mag[mag_to_name[item[1]]] = item[1]\n",
    "util.save(\"name_to_mag\",name_to_mag)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Loading genomes to KBase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run cliffcommutil.py\n",
    "unfiltered_list = util.load(\"mag_list\")\n",
    "filter = []\n",
    "mag_list = []\n",
    "for item in unfiltered_list:\n",
    "    if item[1] in filter or len(filter) == 0:\n",
    "        mag_list.append(item)\n",
    "mag_to_name = util.load(\"mag_to_name\")\n",
    "from datetime import datetime\n",
    "now = datetime.now()\n",
    "timestamp = datetime.timestamp(now)\n",
    "workspace = 187797\n",
    "anno = util.anno_client()\n",
    "anno.clients[\"GenomeFileUtil\"] = util.gfu_client()\n",
    "finished = util.load(\"finished_genomes\",[])\n",
    "unconverted = {}\n",
    "for item in mag_list:\n",
    "    name = mag_to_name[item[1]]\n",
    "    if name not in finished:\n",
    "        genome = util.load(\"annotated/\"+name)\n",
    "        count = 0\n",
    "        for ftr in genome[\"features\"]:\n",
    "            end = len(name)+3\n",
    "            if ftr[\"id\"][0:4] == \"Salt\":\n",
    "                unconverted[item[1]] = 1\n",
    "            if ftr[\"id\"][0:end] == name+\".pg\":\n",
    "                count += 1\n",
    "        print(name,count)\n",
    "        util.save_ws_object(name+\".pg\",187797,genome,\"KBaseGenomes.Genome\")\n",
    "        finished.append(name)\n",
    "        util.save(\"finished_genomes\",finished)\n",
    "for item in unconverted:\n",
    "    print(item)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Inspecting genomes before load"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run cliffcommutil.py\n",
    "mags = util.msrecon.kbase_api.list_objects(187875, object_type=\"KBaseGenomes.Genome\", include_metadata=False)\n",
    "mag_hash = {}\n",
    "for item in mags:\n",
    "    mag_hash[item[1]] = item\n",
    "mag_list = util.load(\"mag_list\")\n",
    "mag_to_name = util.load(\"mag_to_name\")\n",
    "for item in mag_list:\n",
    "    name = mag_to_name[item[1]]\n",
    "    name += \".pangenome.GLM4EC\"\n",
    "    if name in mag_hash:\n",
    "        print(\"Getting \"+name)\n",
    "        full_data = util.get_object(name,187875)\n",
    "        genome = full_data[\"data\"]\n",
    "        ftrhash = {}\n",
    "        for ftr in genome[\"features\"]:\n",
    "            ftrhash[ftr[\"id\"]] = ftr\n",
    "        for ftr in genome[\"cdss\"]:\n",
    "            ftrhash[ftr[\"id\"]] = ftr\n",
    "            if \"parent_gene\" in ftr:\n",
    "                if ftr[\"parent_gene\"] not in ftrhash:\n",
    "                    print(item[1],new_name,\"Gene \"+ftr[\"parent_gene\"]+\" not found!\")\n",
    "        for ftr in genome[\"features\"]:\n",
    "            if \"cdss\" in ftr:\n",
    "                for cds in ftr[\"cdss\"]:\n",
    "                    if cds not in ftrhash:\n",
    "                        print(item[1],new_name,\"CDS \"+cds+\" not found!\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotating genomes using RAST API"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/Users/chenry/code/notebooks/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /Users/chenry/code//kb_sdk/run_local/workdir/tmp/\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1722232442.262796 ERROR: Requested data genome/R1_B_D1.20 doesn't exist at /Users/chenry/code/notebooks/MicrobiomeNotebooks/Cliff/datacache/genome/R1_B_D1.20.json\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Response [200]>\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "Requested data genome/R1_B_D1.20 doesn't exist at /Users/chenry/code/notebooks/MicrobiomeNotebooks/Cliff/datacache/genome/R1_B_D1.20.json",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 18\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m name \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m output:\n\u001b[1;32m     10\u001b[0m     rast_genome \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m     11\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mid\u001b[39m\u001b[38;5;124m\"\u001b[39m:item[\u001b[38;5;241m1\u001b[39m],\n\u001b[1;32m     12\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgenetic_code\u001b[39m\u001b[38;5;124m\"\u001b[39m:\u001b[38;5;241m11\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     16\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfeatures\u001b[39m\u001b[38;5;124m\"\u001b[39m:[]\n\u001b[1;32m     17\u001b[0m     }\n\u001b[0;32m---> 18\u001b[0m     data \u001b[38;5;241m=\u001b[39m \u001b[43mutil\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mload\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mgenome/\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43mname\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     19\u001b[0m     genome \u001b[38;5;241m=\u001b[39m data[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdata\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m     20\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m ftr \u001b[38;5;129;01min\u001b[39;00m genome[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfeatures\u001b[39m\u001b[38;5;124m\"\u001b[39m]:\n",
      "File \u001b[0;32m~/code//chenry_utility_module/lib/chenry_utility_module/kbdevutils.py:147\u001b[0m, in \u001b[0;36mKBDevUtils.load\u001b[0;34m(self, name, default)\u001b[0m\n\u001b[1;32m    145\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m default \u001b[38;5;241m==\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    146\u001b[0m         logger\u001b[38;5;241m.\u001b[39merror(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRequested data \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39mname\u001b[38;5;241m+\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m doesn\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mt exist at \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39mfilename)\n\u001b[0;32m--> 147\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m(\u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRequested data \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39mname\u001b[38;5;241m+\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m doesn\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mt exist at \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39mfilename))\n\u001b[1;32m    148\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m default\n\u001b[1;32m    149\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mopen\u001b[39m(filename, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124m'\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m f:\n",
      "\u001b[0;31mValueError\u001b[0m: Requested data genome/R1_B_D1.20 doesn't exist at /Users/chenry/code/notebooks/MicrobiomeNotebooks/Cliff/datacache/genome/R1_B_D1.20.json"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "mag_list = util.load(\"mag_list\")\n",
    "mag_to_name = util.load(\"mag_to_name\")\n",
    "from modelseedpy.core.rpcclient import RPCClient\n",
    "client = RPCClient(\"https://tutorial.theseed.org/services/genome_annotation\")\n",
    "for item in mag_list:\n",
    "    name = mag_to_name[item[1]]\n",
    "    if name not in output:\n",
    "        rast_genome = {\n",
    "            \"id\":item[1],\n",
    "            \"genetic_code\":11,\n",
    "            \"scientific_name\":\"Unknown\",\n",
    "            \"domain\":\"Bacteria\",\n",
    "            \"contigs\": [],\n",
    "            \"features\":[]\n",
    "        }\n",
    "        data = util.load(\"genome/\"+name)\n",
    "        genome = data[\"data\"]\n",
    "        for ftr in genome[\"features\"]:\n",
    "            rast_genome[\"features\"].append(ftr)\n",
    "        workflow = {\n",
    "            \"stages\":[\n",
    "                {\n",
    "                    \"name\": \"annotate_proteins_kmer_v2\", \n",
    "                    \"kmer_v2_parameters\": {}},\n",
    "                {\n",
    "                    \"name\": \"annotate_proteins_kmer_v1\",\n",
    "                    \"kmer_v1_parameters\": {\"annotate_hypothetical_only\": 1},},\n",
    "                {\n",
    "                    \"name\": \"annotate_proteins_similarity\",\n",
    "                    \"similarity_parameters\": {\"annotate_hypothetical_only\": 1},\n",
    "                },\n",
    "            ]\n",
    "        }\n",
    "        params = [{\"features\": rast_genome[\"features\"]}, workflow]\n",
    "        output = client.call(\"GenomeAnnotation.run_pipeline\",params)\n",
    "        util.save(\"rast/\"+name,output)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Adding RAST ontology event"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run cliffcommutil.py\n",
    "mag_list = util.load(\"mag_list\")\n",
    "mag_to_name = util.load(\"mag_to_name\")\n",
    "from datetime import datetime\n",
    "now = datetime.now()\n",
    "timestamp = datetime.timestamp(now)\n",
    "for item in mag_list:\n",
    "    name = mag_to_name[item[1]]\n",
    "    data = util.load(\"genome/\"+name)\n",
    "    rast = util.load(\"rast/\"+name)\n",
    "    annoapi = util.anno_client(native_python_api=True)\n",
    "    events = [{\n",
    "        \"ontology_id\" : \"SSO\",\n",
    "        \"description\" : \"RAST Annotation API\",\n",
    "        \"method_version\" : \"1.0\",\n",
    "        \"method\" : \"annotate_genome\",\n",
    "        \"ontology_terms\": {},\n",
    "        \"timestamp\":\"2024-07-29T19:32:45\"\n",
    "    }]\n",
    "    for ftr in rast[0][\"features\"]:\n",
    "        if \"function\" in ftr:\n",
    "            events[0][\"ontology_terms\"][ftr[\"id\"]] = [{\"term\":ftr[\"function\"]}]\n",
    "    output = annoapi.add_annotation_ontology_events({\n",
    "        \"output_workspace\":187797,\n",
    "        \"events\":events,\n",
    "        \"overwrite_matching\":True,\n",
    "        \"object\":data[\"data\"],\n",
    "        \"type\":\"KBaseGenomes.Genome\",\n",
    "        \"input_ref\":None,\n",
    "        \"input_workspace\":None,\n",
    "        \"output_name\":None,\n",
    "        \"save\":0\n",
    "    })\n",
    "    util.save(\"annotated/\"+name,output[\"object\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Printing feature probabilities"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "mag_list = util.load(\"mag_list\")\n",
    "mag_thresholds = util.load(\"mag_thresholds\")\n",
    "mag_to_name = util.load(\"mag_to_name\")\n",
    "mag_protein_supplements = util.load(\"mag_protein_supplements\")\n",
    "feature_probabilities = {}\n",
    "for item in mag_list:\n",
    "    name = mag_to_name[item[1]]\n",
    "    data = util.load(\"annotated/\"+name)\n",
    "    feature_probabilities[item[1]] = {}\n",
    "    ftrs = data[\"features\"]\n",
    "    total_genomes = mag_thresholds[item[1]][\"threshold_count\"]\n",
    "    for ftr in ftrs:\n",
    "        if ftr[\"id\"].split(\".\")[-1][0:3] == \"pg_\":\n",
    "            feature_probabilities[item[1]][ftr[\"id\"]] = mag_protein_supplements[item[1]][ftr[\"aliases\"][0][1]][0]/total_genomes\n",
    "            if feature_probabilities[item[1]][ftr[\"id\"]] > 1:\n",
    "                feature_probabilities[item[1]][ftr[\"id\"]] = 1\n",
    "        else:\n",
    "            feature_probabilities[item[1]][ftr[\"id\"]] = 1\n",
    "util.save(\"feature_probabilities\",feature_probabilities)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Processing metabolite data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n",
      "set()\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "metabolite_hash = util.load(\"../data/met_NameMSID\")\n",
    "\n",
    "from numpy import array\n",
    "newMets = array([\"2'-Deoxyuridine\", '2-Oxoglutarate', '2-Oxoisocaproate', '3-Hydroxybutyrate', '3-Hydroxyisovalerate', '3-Methyl-2-oxovalerate', '4-Aminobutyrate', 'Acetate', 'Acetone', 'Alanine', 'Arginine', 'Aspartate', 'Benzoate', 'Betaine', 'Dimethylamine', 'Ethanol', 'Formate', 'Fructose', 'Fumarate', 'Glucose', 'Glutamate', 'Glycerol', 'Isobutyrate', 'Isoleucine', 'Isopropanol', 'Isovalerate', 'Lactate', 'Leucine', 'Maltose', 'Methanol', 'Methionine', 'Methylamine', 'Methylguanidine', 'N,N-Dimethylglycine', 'Phenylacetate', 'Phenylalanine', 'Proline', 'Propionate', 'Propylene glycol', 'Pyroglutamate', 'Succinate', 'Sucrose', 'Thymidine', 'Trehalose', 'Trimethylamine', 'Tryptophan', 'Tyrosine', 'Uracil', 'Uridine', 'Valine'])\n",
    "print(set(newMets)-set(list(metabolite_hash.keys())))\n",
    "\n",
    "metabolite_names = {cpdID: metName for metName, cpdID in metabolite_hash.items()}\n",
    "util.save(\"metabolite_names\",metabolite_names)\n",
    "util.save(\"metabolites\",list(metabolite_names.keys()))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computing SMIPPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python version 3.11.1\n",
      "KBBaseModules 0.0.1\n",
      "modelseedpy 0.3.3\n",
      "cobrakbase 0.3.1\n",
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n",
      "ForkPoolWorker-1\n",
      "[3 'SF2_B_H2O.8.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:08:15+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'db630a36d897caba142685e9e010ae30'\n",
      " 9852570 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048932.944119 INFO: metabolites 1176\n",
      "1723048933.4564693 INFO: reactions 1157\n",
      "1723048933.5926237 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723048934.2600315 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/338/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048934.4035861 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/338/1\"\n",
      "}\n",
      "1723048940.870303 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723048941.5282176 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048944.9326315 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-2\n",
      "[8 'R1_C_D1.31.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:09:16+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '8fd6fa4c20eae776aeebbdf1351d6fca'\n",
      " 9994505 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048949.6453383 INFO: metabolites 866\n",
      "1723048949.9517293 INFO: reactions 795\n",
      "1723048950.041479 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723048950.3453913 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/12/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048950.4747922 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/12/1\"\n",
      "}\n",
      "1723048952.1357155 INFO: cpd05178 not found in model!\n",
      "1723048952.1379213 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_H2O.8.pg.G.D GF: 11291 Prob: 4164 Other: 2897\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048957.6805584 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723048958.2907577 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048961.5442111 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723048963.8053186 INFO: Objective with gapfill database:141.46521674397565; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-3\n",
      "[13 'R1_A_D1.15.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:10:15+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '5059e9cae574d193f3ccf2aef427b5a0'\n",
      " 11681252 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048966.833786 INFO: metabolites 1089\n",
      "1723048967.2534394 INFO: reactions 1067\n",
      "1723048967.39206 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723048967.794016 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/390/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048968.1440628 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/390/1\"\n",
      "}\n",
      "1723048969.0357995 INFO: cpd05178 not found in model!\n",
      "1723048969.0383887 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_D1.31.pg.G.D GF: 12399 Prob: 2670 Other: 3291\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048976.6779687 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723048977.3179498 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048979.6893141 INFO: Objective with gapfill database:133.61966926090932; min objective:0.01\n",
      "1723048980.4803543 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-4\n",
      "[17 'SF2_B_D1.44.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:11:03+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '2f5084e0369d3e5b0acf259a60735c7e'\n",
      " 9747696 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048983.9981263 INFO: metabolites 1015\n",
      "1723048984.3923376 INFO: reactions 985\n",
      "1723048984.624594 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723048985.011015 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/434/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048985.4248235 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/434/1\"\n",
      "}\n",
      "1723048988.0643573 INFO: cpd05178 not found in model!\n",
      "1723048988.0665462 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D1.15.pg.G.D GF: 11473 Prob: 3778 Other: 3103\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048994.9071312 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723048995.5902412 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723048995.8723557 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723048999.4493623 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-5\n",
      "[21 'R2A_C_D1.18.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:11:44+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'b1fc0e925f72aa9691e0b2b344751fab'\n",
      " 8149865 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049000.5242617 INFO: metabolites 1120\n",
      "1723049000.930733 INFO: reactions 1033\n",
      "1723049001.0555546 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049001.4483302 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/454/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049001.5939744 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/454/1\"\n",
      "}\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049006.6290722 INFO: cpd05178 not found in model!\n",
      "1723049006.6311176 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049006.724 INFO: Filtered count:0 out of 1510\n",
      "1723049006.7219837 INFO: Expansion time:Etho.O2:2.6698703069996554\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_D1.44.pg.G.D GF: 11642 Prob: 3694 Other: 3018\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049008.5228279 INFO: Expansion time:Glc:1.7974902039995868\n",
      "1723049008.5249841 INFO: Filtered count:0 out of 1510\n",
      "1723049008.5590422 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049009.1880198 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049009.4869447 INFO: Expansion time:Glc.DMSO:0.9603336190002665\n",
      "1723049009.4889615 INFO: Filtered count:0 out of 1510\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049010.700459 INFO: Expansion time:Glc.O2:1.2101448700004767\n",
      "1723049010.7026591 INFO: Filtered count:0 out of 1510\n",
      "1723049011.7947083 INFO: Expansion time:Glc.TMAO:1.0903276389999519\n",
      "1723049011.7967556 INFO: Filtered count:0 out of 1510\n",
      "1723049012.3537211 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049013.1585498 INFO: Expansion time:Pyr:1.3606176180001057\n",
      "1723049013.160585 INFO: Filtered count:0 out of 1510\n",
      "1723049013.7685516 INFO: Expansion time:Pyr.DMSO:0.6065648399999191\n",
      "1723049013.7705998 INFO: Filtered count:0 out of 1510\n",
      "1723049014.380394 INFO: Expansion time:Pyr.NO:0.6084773890006545\n",
      "1723049014.3825245 INFO: Filtered count:0 out of 1510\n",
      "1723049015.0696125 INFO: Expansion time:Pyr.NO2:0.685674199999994\n",
      "1723049015.0716949 INFO: Filtered count:0 out of 1510\n",
      "1723049015.6790545 INFO: Expansion time:Pyr.NO3:0.6060099920005086\n",
      "1723049015.6810622 INFO: Filtered count:0 out of 1510\n",
      "1723049016.0053625 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-6\n",
      "[25 'R2_A_D2.67.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:12:26+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'f8c640ccb4279792127faf214ba1f614'\n",
      " 8004329 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049017.017196 INFO: metabolites 760\n",
      "1723049017.2892447 INFO: reactions 739\n",
      "1723049017.3881397 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049017.4875727 INFO: Expansion time:Pyr.O2:1.805055994000213\n",
      "1723049017.4896352 INFO: Filtered count:0 out of 1510\n",
      "1723049017.6661603 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/198/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049017.80538 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/198/1\"\n",
      "}\n",
      "1723049018.4333155 INFO: Expansion time:Pyr.SO3:0.9422565840004609\n",
      "1723049018.435482 INFO: Filtered count:0 out of 1510\n",
      "1723049019.0550458 INFO: Expansion time:Pyr.SO4:0.6182340660006957\n",
      "1723049019.0570607 INFO: Filtered count:0 out of 1510\n",
      "1723049019.409802 INFO: cpd05178 not found in model!\n",
      "1723049019.4118698 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D1.18.pg.G.D GF: 11639 Prob: 3650 Other: 3065\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049019.6716666 INFO: Expansion time:Pyr.TMAO:0.6132680110004003\n",
      "1723049019.6735716 INFO: Filtered count:0 out of 1510\n",
      "1723049020.2865376 INFO: Expansion time:empty:0.6115507189997516\n",
      "1723049020.288508 INFO: Filtered count:0 out of 1510\n",
      "1723049022.9776416 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049023.6319396 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049024.9404948 INFO: Expansion time:Ac.O2:0.5967038219996539\n",
      "1723049024.9425414 INFO: Filtered count:0 out of 1368\n",
      "1723049025.535507 INFO: Expansion time:Etho.O2:0.5916443670002991\n",
      "1723049025.5374866 INFO: Filtered count:0 out of 1368\n",
      "1723049026.1484091 INFO: Expansion time:For.O2:0.6095286769996164\n",
      "1723049026.1504748 INFO: Filtered count:0 out of 1368\n",
      "1723049026.703997 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049027.406137 INFO: Expansion time:Glyc.O2:1.2544055309999749\n",
      "1723049027.4080966 INFO: Filtered count:0 out of 1368\n",
      "1723049028.0046024 INFO: Expansion time:Pyr:0.5953990680000061\n",
      "1723049028.0066535 INFO: Filtered count:0 out of 1368\n",
      "1723049028.4860122 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049028.598237 INFO: Expansion time:Pyr.DMSO:0.5903132550001828\n",
      "1723049028.600071 INFO: Filtered count:0 out of 1368\n",
      "1723049029.1899226 INFO: Expansion time:Pyr.NO:0.5885695489996579\n",
      "1723049029.1918778 INFO: Filtered count:0 out of 1368\n",
      "1723049029.7844496 INFO: Expansion time:Pyr.NO2:0.5912008060004155\n",
      "1723049029.7864323 INFO: Filtered count:0 out of 1368\n",
      "1723049031.0537148 INFO: Expansion time:Pyr.NO3:1.2659435069999745\n",
      "1723049031.0555182 INFO: Filtered count:0 out of 1368\n",
      "1723049031.6500268 INFO: Expansion time:Pyr.O2:0.5932063860000198\n",
      "1723049031.6520424 INFO: Filtered count:0 out of 1368\n",
      "1723049032.241362 INFO: Expansion time:Pyr.SO3:0.5879963219995261\n",
      "1723049032.2433596 INFO: Filtered count:0 out of 1368\n",
      "1723049032.848194 INFO: Expansion time:Pyr.SO4:0.6034510199997385\n",
      "1723049032.8499978 INFO: Filtered count:0 out of 1368\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-7\n",
      "[29 'R2_A_D1.29.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:13:23+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '31601334fbc3b7646455df26f85ba98b'\n",
      " 11883993 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049033.4487221 INFO: Expansion time:Pyr.TMAO:0.5974682109999776\n",
      "1723049033.450849 INFO: Filtered count:0 out of 1368\n",
      "1723049033.7771919 INFO: metabolites 1125\n",
      "1723049034.0442064 INFO: Expansion time:empty:0.5920777159999489\n",
      "1723049034.046318 INFO: Filtered count:0 out of 1368\n",
      "1723049034.1921575 INFO: reactions 1070\n",
      "1723049034.2348545 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049034.2327025 INFO: cpd05178 not found in model!\n",
      "1723049034.3200238 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D2.67.pg.G.D GF: 12518 Prob: 2586 Other: 3256\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049034.7213728 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/515/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049035.0589364 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/515/1\"\n",
      "}\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049038.127969 INFO: Expansion time:Ac.NO2:0.611537331999898\n",
      "1723049038.1298983 INFO: Filtered count:0 out of 1453\n",
      "1723049039.3288372 INFO: Expansion time:Ac.O2:1.1975865389995306\n",
      "1723049039.330928 INFO: Filtered count:0 out of 1453\n",
      "1723049039.929885 INFO: Expansion time:Etho.O2:0.5977037019993077\n",
      "1723049039.9320605 INFO: Filtered count:0 out of 1453\n",
      "1723049040.532021 INFO: Expansion time:Fum:0.5986747770002694\n",
      "1723049040.534084 INFO: Filtered count:0 out of 1453\n",
      "1723049041.1390233 INFO: Expansion time:Fum.O2:0.603547088000596\n",
      "1723049041.1411145 INFO: Filtered count:0 out of 1453\n",
      "1723049041.7435439 INFO: Expansion time:Pyr.NO2:0.6010316829997464\n",
      "1723049041.745662 INFO: Filtered count:0 out of 1453\n",
      "1723049042.958904 INFO: Expansion time:Pyr.O2:1.2118107039996175\n",
      "1723049042.96118 INFO: Filtered count:0 out of 1453\n",
      "1723049043.050666 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049043.2060375 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049043.5656073 INFO: Expansion time:empty:0.60302142799992\n",
      "1723049043.5677729 INFO: Filtered count:0 out of 1453\n",
      "1723049043.7149556 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049047.2104123 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-8\n",
      "[33 'R2_C_D2.45.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:14:12+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'addaf3cf6f8fcb60b6b3e5fe846a753b'\n",
      " 15369456 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049050.688066 INFO: metabolites 1249\n",
      "1723049051.3500788 INFO: reactions 1213\n",
      "1723049051.7467585 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049052.2529287 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/526/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049052.403464 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/526/1\"\n",
      "}\n",
      "1723049054.2667267 INFO: Expansion time:Pyr:2.36317420999967\n",
      "1723049054.2687438 INFO: Filtered count:0 out of 995\n",
      "1723049054.8576508 INFO: Expansion time:Pyr.DMSO:0.587589391999245\n",
      "1723049054.8594902 INFO: Filtered count:0 out of 995\n",
      "1723049055.035779 INFO: cpd05178 not found in model!\n",
      "1723049055.0378242 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.29.pg.G.D GF: 11485 Prob: 3874 Other: 2995\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049055.436046 INFO: Expansion time:Pyr.SO3:0.5753058709997276\n",
      "1723049055.4379344 INFO: Filtered count:0 out of 995\n",
      "1723049056.588211 INFO: Expansion time:Pyr.TMAO:1.1487758129997019\n",
      "1723049056.5900536 INFO: Filtered count:0 out of 995\n",
      "1723049057.145306 INFO: Expansion time:empty:0.5539419259994247\n",
      "1723049057.147255 INFO: Filtered count:0 out of 995\n",
      "1723049063.759766 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-9\n",
      "[37 'R2A_A_H2O.62.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:15:01+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'a47223bb8bd671b2cb647d30d7f00622'\n",
      " 5174374 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049067.1572351 INFO: metabolites 925\n",
      "1723049067.5263658 INFO: reactions 881\n",
      "1723049067.6330895 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049067.966631 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/550/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049068.1161768 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/550/1\"\n",
      "}\n",
      "1723049068.2085412 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049069.011022 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049072.4809992 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049072.9005678 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049073.4813008 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049076.3021111 INFO: Expansion time:Ac.NO2:3.3795226120000734\n",
      "1723049076.3042066 INFO: Filtered count:0 out of 1502\n",
      "1723049076.5118484 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049078.8136406 INFO: Expansion time:Ac.O2:2.5081265859998894\n",
      "1723049078.8159087 INFO: Filtered count:0 out of 1502\n",
      "1723049080.0575411 INFO: Expansion time:Etho.O2:1.2403326739995464\n",
      "1723049080.0594487 INFO: Filtered count:0 out of 1502\n",
      "1723049080.121844 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049080.1198087 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D2.45.pg.G.D GF: 10888 Prob: 4618 Other: 2846\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049081.6252093 INFO: Expansion time:Pyr.NO2:1.564301840000553\n",
      "1723049081.6272633 INFO: Filtered count:0 out of 1502\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-10\n",
      "[41 'SF2_B_H2O.91.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:15:43+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '7ed3b5c11032085b66c24f1741edf608'\n",
      " 7327892 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049083.358934 INFO: Expansion time:Pyr.O2:1.7300782029997208\n",
      "1723049083.3608406 INFO: Filtered count:0 out of 1502\n",
      "1723049083.6592977 INFO: metabolites 945\n",
      "1723049084.0037887 INFO: cpd05178 not found in model!\n",
      "1723049084.0040736 INFO: reactions 879\n",
      "1723049084.005879 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049084.1080418 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "3225 "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049084.4374933 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_H2O.62.pg.G.D GF: 12341 Prob: 2786 Other:188040/564/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049084.5806139 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/564/1\"\n",
      "}\n",
      "1723049084.61595 INFO: Expansion time:empty:1.2538447719998658\n",
      "1723049084.617658 INFO: Filtered count:0 out of 1502\n",
      "1723049087.9507442 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049089.8798912 INFO: cpd05178 not found in model!\n",
      "1723049089.8818927 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049091.3910038 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049092.0128067 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049095.1310408 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049096.189196 INFO: Objective with gapfill database:214.96392227376788; min objective:0.01\n",
      "1723049097.5350184 INFO: Expansion time:Dlac.O2:1.4365447059999497\n",
      "1723049097.5370371 INFO: Filtered count:0 out of 1723\n",
      "1723049098.1747854 INFO: Expansion time:Etho.O2:0.6364127319993713\n",
      "1723049098.1768632 INFO: Filtered count:0 out of 1723\n",
      "1723049098.8115063 INFO: Expansion time:For.NO2:0.6334904570003346\n",
      "1723049098.8136182 INFO: Filtered count:0 out of 1723\n",
      "1723049099.455867 INFO: Expansion time:For.O2:0.640754865999952\n",
      "1723049099.4581044 INFO: Filtered count:0 out of 1723\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-11\n",
      "[45 'R2_restored_DShore.37.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:16:28+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9a43df8cead9614f3959d69f993ea85e'\n",
      " 10358772 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049099.560619 INFO: cpd05178 not found in model!\n",
      "1723049099.5624819 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049100.173432 INFO: Expansion time:Fum:0.7133201730002838\n",
      "1723049100.1755295 INFO: Filtered count:0 out of 1723\n",
      "1723049100.2257636 INFO: metabolites 867\n",
      "1723049100.5660176 INFO: reactions 763\n",
      "1723049100.6557438 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049100.8188663 INFO: Expansion time:Fum.O2:0.6420004450001215\n",
      "1723049100.8208702 INFO: Filtered count:0 out of 1723\n",
      "1723049100.9732244 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/320/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049101.1127198 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/320/1\"\n",
      "}\n",
      "1723049102.250946 INFO: Expansion time:Pyr:1.428904844000499\n",
      "1723049102.253153 INFO: Filtered count:0 out of 1723\n",
      "1723049102.7744057 INFO: cpd05178 not found in model!\n",
      "1723049102.7765129 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049102.8956833 INFO: Expansion time:Pyr.DMSO:0.6410690090006028\n",
      "1723049102.8976607 INFO: Filtered count:0 out of 1723\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049103.109331 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_H2O.91.pg.G.D GF: 12013 Prob: 3240 Other: 3099"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049103.1074991 INFO: cpd05178 not found in model!\n",
      "1723049103.5379357 INFO: Expansion time:Pyr.NO:0.6389192439992257\n",
      "1723049103.5400112 INFO: Filtered count:0 out of 1723\n",
      "1723049104.1806405 INFO: Expansion time:Pyr.NO2:0.6394696229999681\n",
      "1723049104.1826794 INFO: Filtered count:0 out of 1723\n",
      "1723049104.8274558 INFO: Expansion time:Pyr.NO3:0.6433195669997076\n",
      "1723049104.8294902 INFO: Filtered count:0 out of 1723\n",
      "1723049105.4723787 INFO: Expansion time:Pyr.O2:0.6414927939995323\n",
      "1723049105.4747238 INFO: Filtered count:0 out of 1723\n",
      "1723049106.128938 INFO: Expansion time:Pyr.SO3:0.6530417929998293\n",
      "1723049106.1310322 INFO: Filtered count:0 out of 1723\n",
      "1723049107.5427527 INFO: Expansion time:Pyr.SO4:1.4102570659997582\n",
      "1723049107.5448875 INFO: Filtered count:0 out of 1723\n",
      "1723049108.1895032 INFO: Expansion time:Pyr.TMAO:0.643289191999429\n",
      "1723049108.19151 INFO: Filtered count:0 out of 1723\n",
      "1723049108.8328905 INFO: Expansion time:empty:0.6400893570007611\n",
      "1723049108.8350577 INFO: Filtered count:0 out of 1723\n",
      "1723049111.4396198 INFO: cpd05178 not found in model!\n",
      "1723049111.4417593 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049112.4059925 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049112.6794088 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049114.8381493 INFO: Objective with gapfill database:148.63932337992762; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-12\n",
      "[49 'R1_B_D2.120.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:17:10+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '83e6785c8811205cb3ba8b3ed4469ea2'\n",
      " 7038728 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049116.5576684 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049116.8298533 INFO: metabolites 917\n",
      "1723049117.1558778 INFO: reactions 855\n",
      "1723049117.253813 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049117.574723 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/594/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049117.7125864 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/594/1\"\n",
      "}\n",
      "1723049123.6413796 INFO: cpd05178 not found in model!\n",
      "1723049123.643414 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.37.pg.G.D"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049123.889638 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " GF: 11844 Prob: 3378 Other: 3138\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049124.4871063 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049127.6141338 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-13\n",
      "[53 'R2_C_D1.71.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:18:03+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '6926ca4cd798ac267901f2fa40caccf5'\n",
      " 17450578 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049133.5972166 INFO: metabolites 1186\n",
      "1723049134.0345886 INFO: reactions 1170\n",
      "1723049134.1867266 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049134.849095 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/603/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049135.0042846 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/603/1\"\n",
      "}\n",
      "1723049135.1294603 INFO: cpd05178 not found in model!\n",
      "1723049135.1313965 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049135.1900802 INFO: Objective with gapfill database:122.29514051089092; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " \n",
      "3279 Other:2828R1_B_D2.120.pg.G.D GF: 12243 Prob: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049143.1373212 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049143.7915788 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049147.0850708 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049147.2057881 INFO: cpd05178 not found in model!\n",
      "1723049147.2080207 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049148.4077427 INFO: Objective with gapfill database:149.01073124013476; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-14\n",
      "[57 'R2A_C_D2.31.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:18:59+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '64a5493dcde0e31714561947cea1a988'\n",
      " 17031060 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049150.0660841 INFO: metabolites 1137\n",
      "1723049150.4513505 INFO: reactions 1063\n",
      "1723049150.6226602 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049151.0275543 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/611/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049151.1761925 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/611/1\"\n",
      "}\n",
      "1723049152.8866622 INFO: cpd05178 not found in model!\n",
      "1723049152.888822 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049154.6962292 INFO: cpd05178 not found in model!\n",
      "1723049154.6984248 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D1.71.pg.G.D GF: 11463 Prob: 3888 Other: 3001\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049159.28239 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049159.9652417 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049162.5242589 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049163.390237 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-15\n",
      "[61 'R1_A_D1.45.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:19:51+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '092d39993ee36a676f112e6fe7c6128f'\n",
      " 9281740 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049166.7998064 INFO: metabolites 1009\n",
      "1723049167.2072132 INFO: reactions 942\n",
      "1723049167.3144925 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049167.6808143 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/33/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049167.835187 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/33/1\"\n",
      "}\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049170.6191692 INFO: cpd05178 not found in model!\n",
      "1723049170.6210678 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.31.pg.G.D GF: 11805 Prob: 3448 Other: 3101\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049173.5775626 INFO: Expansion time:Ac.O2:2.95399511100004\n",
      "1723049173.5797937 INFO: Filtered count:0 out of 1675\n",
      "1723049174.301168 INFO: Expansion time:Etho.O2:0.7199711399998705\n",
      "1723049174.3034315 INFO: Filtered count:0 out of 1675\n",
      "1723049175.0390437 INFO: Expansion time:For.O2:0.7343141740002466\n",
      "1723049175.0411687 INFO: Filtered count:0 out of 1675\n",
      "1723049176.1281316 INFO: Expansion time:Fum:1.0856664930006446\n",
      "1723049176.1303294 INFO: Filtered count:0 out of 1675\n",
      "1723049177.8402767 INFO: Expansion time:Fum.O2:1.7086364979995778\n",
      "1723049177.8424656 INFO: Filtered count:0 out of 1675\n",
      "1723049178.869786 INFO: Expansion time:Glyc.O2:1.0259255339997253\n",
      "1723049178.8719594 INFO: Filtered count:0 out of 1675\n",
      "1723049179.6761942 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049179.7268777 INFO: Expansion time:Pyr:0.8534536770002887\n",
      "1723049179.7290502 INFO: Filtered count:0 out of 1675\n",
      "1723049180.354797 INFO: Expansion time:Pyr.DMSO:0.6240433150005629\n",
      "1723049180.3569603 INFO: Filtered count:0 out of 1675\n",
      "1723049180.4242775 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049180.771726 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049180.9930835 INFO: Expansion time:Pyr.NO:0.6347771969994938\n",
      "1723049180.9952683 INFO: Filtered count:0 out of 1675\n",
      "1723049182.2844422 INFO: Expansion time:Pyr.NO2:1.287869609000154\n",
      "1723049182.2868683 INFO: Filtered count:0 out of 1675\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-16\n",
      "[65 'R2_restored_DShore.23.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:20:37+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '90828e392f008d8c3cffd0ec89cca268'\n",
      " 11356876 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049182.9177654 INFO: Expansion time:Pyr.NO3:0.6293016950003221\n",
      "1723049182.9199305 INFO: Filtered count:0 out of 1675\n",
      "1723049183.3897045 INFO: metabolites 1137\n",
      "1723049183.683202 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049183.9211843 INFO: reactions 1107\n",
      "1723049183.9997258 INFO: Expansion time:Pyr.O2:1.078371282000262\n",
      "1723049184.0019274 INFO: Filtered count:0 out of 1675\n",
      "1723049184.1438782 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049184.5364654 INFO: cpd05178 not found in model!\n",
      "1723049184.5387552 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049184.88625 INFO: Expansion time:Pyr.SO3:0.882904571999461\n",
      "1723049184.8884978 INFO: Filtered count:0 out of 1675\n",
      "1723049185.0219083 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/97/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049185.2177515 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/97/1\"\n",
      "}\n",
      "1723049185.7420702 INFO: Expansion time:Pyr.SO4:0.8514034629997695\n",
      "1723049185.7442863 INFO: Filtered count:0 out of 1675\n",
      "1723049187.2543921 INFO: Expansion time:Pyr.TMAO:1.5088172129999293\n",
      "1723049187.2565842 INFO: Filtered count:0 out of 1675\n",
      "1723049187.887837 INFO: Expansion time:empty:0.629786600999978\n",
      "1723049187.8902285 INFO: Filtered count:0 out of 1675\n",
      "1723049189.4040346 INFO: cpd05178 not found in model!\n",
      "1723049189.4060977 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049190.8434694 INFO: cpd05178 not found in model!\n",
      "1723049190.8455532 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D1.45.pg.G.D GF: 11691 Prob: 3696 Other: 2967\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049196.8445983 INFO: Expansion time:Etho.O2:6.207566639000106\n",
      "1723049196.8466215 INFO: Filtered count:0 out of 1513\n",
      "1723049198.062918 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049198.717452 INFO: Expansion time:For.O2:1.8695169200000237\n",
      "1723049198.7195077 INFO: Filtered count:0 out of 1513\n",
      "1723049198.836649 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-17\n",
      "[69 'R1_B_D2.78.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:21:15+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '285e9a2c7209b4e0e57e08053dd1a906'\n",
      " 3051856 None]\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049199.911441 INFO: metabolites 573\n",
      "1723049200.0262508 INFO: reactions 496\n",
      "1723049200.0876255 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049200.2646546 INFO: Default biomass: [bio1]\n",
      "1723049200.3536825 INFO: Expansion time:Fum.O2:1.6328634869996677\n",
      "1723049200.355367 INFO: Filtered count:0 out of 1513\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/110/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049200.407662 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/110/1\"\n",
      "}\n",
      "1723049200.9921746 INFO: Expansion time:empty:0.6355061360000036\n",
      "1723049200.994348 INFO: Filtered count:0 out of 1513\n",
      "1723049202.2027218 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049202.8975987 INFO: Objective with gapfill database:127.7491263666788; min objective:0.01\n",
      "1723049203.0025086 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049203.8752866 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049207.1916523 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049209.6860137 INFO: cpd05178 not found in model!\n",
      "1723049209.6880596 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.23.pg.G.D GF: 11036 Prob: 4442 Other: 2876\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049214.2990036 INFO: cpd05178 not found in model!\n",
      "1723049214.3012419 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049214.3310533 INFO: cpd05178 not found in model!\n",
      "1723049214.3332598 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "3767 Other:R1_B_D2.78.pg.G.D GF: 13597 Prob: 986 ForkPoolWorker-18\n",
      "[73 'R2_restored_DShore.40.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:22:04+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '29f91197ceafd2520663a1629597e9cf'\n",
      " 12016802 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049216.6193576 INFO: metabolites 934\n",
      "1723049217.0029662 INFO: reactions 823\n",
      "1723049217.1025147 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049217.4401112 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/328/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049217.5839808 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/328/1\"\n",
      "}\n",
      "1723049218.5076356 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049227.3458717 INFO: Objective with gapfill database:24.679749221037998; min objective:0.01\n",
      "1723049228.710832 INFO: Expansion time:Ac.NO3:1.6307172500000888\n",
      "1723049228.7129908 INFO: Filtered count:0 out of 1560\n",
      "1723049229.4344337 INFO: Expansion time:Ac.O2:0.7202047029995811\n",
      "1723049229.4365513 INFO: Filtered count:0 out of 1560\n",
      "1723049229.9481115 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049230.6704557 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049230.8949342 INFO: Expansion time:Etho.O2:1.457081334000577\n",
      "1723049230.8969486 INFO: Filtered count:0 out of 1560\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049231.6096773 INFO: Expansion time:For.NO3:0.7116007249996983\n",
      "1723049231.6117263 INFO: Filtered count:0 out of 1560\n",
      "1723049232.3809493 INFO: Expansion time:For.O2:0.767855399999462\n",
      "1723049232.3829224 INFO: Filtered count:0 out of 1560\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-19\n",
      "[77 'R2A_C_D1.009.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:22:54+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '037f554791f7d3d51e536ef3f41fed25'\n",
      " 20614530 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049233.2278104 INFO: Expansion time:Fum:0.8436296849995415\n",
      "1723049233.229822 INFO: Filtered count:0 out of 1560\n",
      "1723049233.4158537 INFO: metabolites 1115\n",
      "1723049233.9003093 INFO: reactions 1092\n",
      "1723049234.045881 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049234.0618927 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049234.0659647 INFO: Expansion time:Fum.O2:0.8347806020001372\n",
      "1723049234.0678697 INFO: Filtered count:0 out of 1560\n",
      "1723049234.721764 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/171/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049234.8759003 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/171/1\"\n",
      "}\n",
      "1723049235.704663 INFO: Expansion time:Glc:1.6351766269999644\n",
      "1723049235.706784 INFO: Filtered count:0 out of 1560\n",
      "1723049236.5630589 INFO: Expansion time:Glc.DMSO:0.8549959049996687\n",
      "1723049236.565042 INFO: Filtered count:0 out of 1560\n",
      "1723049237.4729724 INFO: Expansion time:Glc.O2:0.9067569149992778\n",
      "1723049237.4747336 INFO: Filtered count:0 out of 1560\n",
      "1723049238.341027 INFO: Expansion time:Glc.TMAO:0.864951038000072\n",
      "1723049238.3431246 INFO: Filtered count:0 out of 1560\n",
      "1723049239.0686657 INFO: Expansion time:Pyr:0.724095638000108\n",
      "1723049239.0709245 INFO: Filtered count:0 out of 1560\n",
      "1723049240.8329499 INFO: Expansion time:Pyr.DMSO:1.7604535459995532\n",
      "1723049240.8350244 INFO: Filtered count:0 out of 1560\n",
      "1723049241.1836333 INFO: cpd05178 not found in model!\n",
      "1723049241.1857812 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.40.pg.G.D GF: 11785 Prob: 3554 Other: 3021\n",
      "Failed:>rxn00062_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049245.1235838 INFO: Expansion time:Pyr.NO:4.287292084999535\n",
      "1723049245.1252556 INFO: Filtered count:1 out of 1560\n",
      "1723049245.755808 INFO: Expansion time:Pyr.NO2:0.6292793650000021\n",
      "1723049245.7577932 INFO: Filtered count:1 out of 1560\n",
      "1723049246.2070882 INFO: cpd05178 not found in model!\n",
      "1723049246.2092597 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049246.3800874 INFO: Expansion time:Pyr.NO3:0.6209534170002371\n",
      "1723049246.3819032 INFO: Filtered count:1 out of 1560\n",
      "1723049247.0127683 INFO: Expansion time:Pyr.O2:0.6294448469998315\n",
      "1723049247.0147676 INFO: Filtered count:1 out of 1560\n",
      "1723049247.6456952 INFO: Expansion time:Pyr.SO3:0.6296196780003811\n",
      "1723049247.6476314 INFO: Filtered count:1 out of 1560\n",
      "1723049249.0186286 INFO: Expansion time:Pyr.SO4:1.369963129000098\n",
      "1723049249.0206604 INFO: Filtered count:1 out of 1560\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-20\n",
      "[81 'R1_A_D1.51.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:23:37+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e6b9c4e68454c49a99ae8d9fa1939f5a'\n",
      " 7822392 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049249.65672 INFO: Expansion time:Pyr.TMAO:0.63476285500019\n",
      "1723049249.658893 INFO: Filtered count:1 out of 1560\n",
      "1723049249.9554913 INFO: metabolites 962\n",
      "1723049250.2993042 INFO: Expansion time:empty:0.6391182100005608\n",
      "1723049250.3012059 INFO: Filtered count:1 out of 1560\n",
      "1723049250.318 INFO: reactions 896\n",
      "1723049250.4227092 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049250.7624621 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/156/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049250.9119585 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/156/1\"\n",
      "}\n",
      "1723049251.4252331 INFO: cpd05178 not found in model!\n",
      "1723049251.427478 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049252.044455 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049252.2488973 INFO: Objective with gapfill database:163.72374659620013; min objective:0.01\n",
      "1723049252.8391654 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049255.7410846 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049257.1909912 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049258.3835833 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049260.01503 INFO: cpd05178 not found in model!\n",
      "1723049260.0173268 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049261.8880715 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049263.6153026 INFO: cpd05178 not found in model!\n",
      "1723049263.6171181 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D1.009.pg.G.D GF: 11227 Prob: 4188 Other: 2939\n",
      "ForkPoolWorker-21\n",
      "[85 'R2_restored_C_black.26.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:24:21+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '186f0417978ac3b42e4d5fefabf4deb9'\n",
      " 10769799 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049266.4885368 INFO: cpd05178 not found in model!\n",
      "1723049266.6515207 INFO: metabolites 951\n",
      "1723049266.7121487 INFO: cpd05178 not found in model!\n",
      "1723049266.7144759 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049267.0077288 INFO: reactions 896\n",
      "1723049267.1095266 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049267.5042546 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/184/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049267.6523576 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/184/1\"\n",
      "}\n",
      "1723049269.3863316 INFO: cpd05178 not found in model!\n",
      "1723049269.3885477 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D1.51.pg.G.D GF: 12147 Prob: 2886 Other: 3317\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049270.486247 INFO: cpd05178 not found in model!\n",
      "1723049270.488435 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049271.6254423 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049275.9605627 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049276.5860536 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049279.940963 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049280.3261375 INFO: Objective with gapfill database:124.5137201225383; min objective:0.01\n",
      "1723049280.6249487 INFO: Expansion time:For.NO3:0.6191423260006559\n",
      "1723049280.6269763 INFO: Filtered count:0 out of 1547\n",
      "1723049281.960101 INFO: Expansion time:Glc:1.3318682990002344\n",
      "1723049281.9621494 INFO: Filtered count:0 out of 1547\n",
      "1723049282.5876813 INFO: Expansion time:Glc.DMSO:0.6242330690001836\n",
      "1723049282.5897927 INFO: Filtered count:0 out of 1547\n",
      "1723049283.2145147 INFO: Expansion time:Glc.O2:0.6232421370004886\n",
      "1723049283.2166355 INFO: Filtered count:0 out of 1547\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-22\n",
      "[89 'R1_B_D2.63.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:25:03+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '654699486664965c689f08e8edb1f923'\n",
      " 6354753 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049283.8536072 INFO: Expansion time:Glc.TMAO:0.6352301460001399\n",
      "1723049283.8557608 INFO: Filtered count:0 out of 1547\n",
      "1723049284.0684435 INFO: metabolites 912\n",
      "1723049284.381807 INFO: reactions 855\n",
      "1723049284.4772196 INFO: Expansion time:Pyr:0.6203289789991686\n",
      "1723049284.4791229 INFO: Filtered count:0 out of 1547\n",
      "1723049284.484694 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049284.8140166 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/225/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049284.9529448 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/225/1\"\n",
      "}\n",
      "1723049285.0932825 INFO: Expansion time:Pyr.DMSO:0.6128807430004599\n",
      "1723049285.0953631 INFO: Filtered count:0 out of 1547\n",
      "1723049286.4183044 INFO: Expansion time:Pyr.NO:1.321642153000539\n",
      "1723049286.4206133 INFO: Filtered count:0 out of 1547\n",
      "1723049287.0473824 INFO: Expansion time:Pyr.NO2:0.6251749470002324\n",
      "1723049287.049622 INFO: Filtered count:0 out of 1547\n",
      "1723049287.0999491 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049287.097981 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_C_black.26.pg.G.D GF: 12147 Prob: 3026 Other: 3179\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049287.6757216 INFO: Expansion time:Pyr.NO3:0.6248241490002329\n",
      "1723049287.6776466 INFO: Filtered count:0 out of 1547\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[18 'R2A_A_D1.51.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:11:13+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '72db3c3d3e9b5cf92a0543826f37dee5'\n",
      " 10769668 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049288.3006237 INFO: Expansion time:Pyr.O2:0.6216605250001521\n",
      "1723049288.3026361 INFO: Filtered count:0 out of 1547\n",
      "1723049288.6372237 INFO: metabolites 1109\n",
      "1723049288.9210706 INFO: Expansion time:Pyr.SO3:0.6171063289993981\n",
      "1723049288.9228778 INFO: Filtered count:0 out of 1547\n",
      "1723049289.0439982 INFO: reactions 1051\n",
      "1723049289.1766171 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049289.575472 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049289.5939546 INFO: cpd05178 not found in model!\n",
      "1723049289.5960326 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/443/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049290.2560513 INFO: Expansion time:Pyr.SO4:1.3318460479995338\n",
      "1723049290.2566147 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049290.2580724 INFO: Filtered count:0 out of 1547\n",
      "1723049290.8910513 INFO: Expansion time:Pyr.TMAO:0.6316209980004714\n",
      "1723049290.8930326 INFO: Filtered count:0 out of 1547\n",
      "1723049290.9757593 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049291.5211926 INFO: Expansion time:empty:0.6268783739997161\n",
      "1723049291.5231967 INFO: Filtered count:0 out of 1547\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049294.3566844 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049296.7318974 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049297.041957 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049299.1687303 INFO: Objective with gapfill database:144.08830822708953; min objective:0.01\n",
      "1723049299.6467223 INFO: cpd05178 not found in model!\n",
      "1723049299.6487508 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-23\n",
      "[93 'R2A_B_H2O.59.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:25:50+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'a5bfbc92593d3143dae80640d252acde'\n",
      " 7946346 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049300.7081006 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049301.0364242 INFO: metabolites 943\n",
      "1723049301.4039106 INFO: reactions 892\n",
      "1723049301.5071409 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049301.8489504 INFO: Default biomass: [bio1]\n",
      "1723049301.9321508 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049301.9341247 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/230/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049301.9873228 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/230/1\"\n",
      "}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.63.pg.G.D GF: 12218 Prob: 2826 Other: 3308\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049303.4480004 INFO: cpd05178 not found in model!\n",
      "1723049303.4500773 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049308.5906587 INFO: cpd05178 not found in model!\n",
      "1723049308.5927277 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049308.695498 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1.51.pg.G.D GF: 11744 Prob: 3530 Other: 3080\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049309.319062 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049313.0018904 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049314.8146782 INFO: Objective with gapfill database:141.17262101741315; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-24\n",
      "[97 'R2A_C_D2.9.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:26:43+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '883cfdaa04490734f340f85a575aecfe'\n",
      " 12365460 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049318.0505605 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049318.0895476 INFO: metabolites 1255\n",
      "1723049318.6522667 INFO: reactions 1187\n",
      "1723049318.79762 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049319.371125 INFO: cpd05178 not found in model!\n",
      "1723049319.3733873 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049319.4948277 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/272/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049319.6416316 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/272/1\"\n",
      "}\n",
      "1723049320.5850654 INFO: cpd05178 not found in model!\n",
      "1723049320.587463 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_B_H2O.59.pg.G.D GF: 12180 Prob: 3028 Other: 3144\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049327.9451394 INFO: Expansion time:Ac.O2:0.6367657020000479\n",
      "1723049327.947203 INFO: Filtered count:0 out of 1501\n",
      "1723049328.5690882 INFO: Expansion time:Etho.O2:0.6202656900004513\n",
      "1723049328.5711217 INFO: Filtered count:0 out of 1501\n",
      "1723049329.201636 INFO: Expansion time:For.O2:0.6292849860001297\n",
      "1723049329.2037678 INFO: Filtered count:0 out of 1501\n",
      "1723049329.829467 INFO: Expansion time:Fum:0.6243677550000939\n",
      "1723049329.831488 INFO: Filtered count:0 out of 1501\n",
      "1723049330.2006485 INFO: cpd05178 not found in model!\n",
      "1723049330.2027981 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049330.5190947 INFO: Expansion time:Fum.O2:0.6859703790005369\n",
      "1723049330.5211172 INFO: Filtered count:0 out of 1501\n",
      "1723049330.7686627 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049331.4802625 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049332.2518208 INFO: Filtered count:0 out of 1501\n",
      "1723049332.2499084 INFO: Expansion time:Glc:1.727137161000428\n",
      "1723049332.874541 INFO: Expansion time:Glc.DMSO:0.6215301599995655\n",
      "1723049332.876512 INFO: Filtered count:0 out of 1501\n",
      "1723049333.1730626 INFO: Objective with gapfill database:184.93944381375368; min objective:0.01\n",
      "1723049333.5050492 INFO: Expansion time:Glc.O2:0.6272573630003535\n",
      "1723049333.506973 INFO: Filtered count:0 out of 1501\n",
      "1723049334.1327658 INFO: Expansion time:Glc.TMAO:0.6245248540008106\n",
      "1723049334.1347241 INFO: Filtered count:0 out of 1501\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-25\n",
      "[101 'R2_B_D2.19.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:27:41+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '50cec739f98c1bae173f91a0b781cebd'\n",
      " 13559556 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049334.76312 INFO: Expansion time:Pyr:0.6272758510003769\n",
      "1723049334.7650633 INFO: Filtered count:0 out of 1501\n",
      "1723049334.811861 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049334.9578297 INFO: metabolites 1243\n",
      "1723049335.3834848 INFO: Expansion time:Pyr.DMSO:0.6171251460000349\n",
      "1723049335.3853724 INFO: Filtered count:0 out of 1501\n",
      "1723049335.39811 INFO: cpd05178 not found in model!\n",
      "1723049335.3996563 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049335.6298473 INFO: reactions 1258\n",
      "1723049336.0119326 INFO: Expansion time:Pyr.NO:0.625094314999842\n",
      "1723049336.0139344 INFO: Filtered count:0 out of 1501\n",
      "1723049336.0222847 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049336.5325913 INFO: Default biomass: [bio1]\n",
      "1723049336.6377673 INFO: Expansion time:Pyr.NO2:0.6223438010001701\n",
      "1723049336.6398695 INFO: Filtered count:0 out of 1501\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/352/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049336.7027082 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/352/1\"\n",
      "}\n",
      "1723049338.3618448 INFO: Expansion time:Pyr.NO3:1.720807849999801\n",
      "1723049338.3639545 INFO: Filtered count:0 out of 1501\n",
      "1723049338.996224 INFO: Expansion time:Pyr.O2:0.6306263149999722\n",
      "1723049338.998187 INFO: Filtered count:0 out of 1501\n",
      "1723049339.640723 INFO: Expansion time:Pyr.SO3:0.6412623260002874\n",
      "1723049339.6428924 INFO: Filtered count:0 out of 1501\n",
      "1723049340.2723439 INFO: Expansion time:Pyr.SO4:0.6282481870002812\n",
      "1723049340.2749996 INFO: Filtered count:0 out of 1501\n",
      "1723049340.9130971 INFO: Expansion time:Pyr.TMAO:0.6368763289992785\n",
      "1723049340.91531 INFO: Filtered count:0 out of 1501\n",
      "1723049341.540573 INFO: Expansion time:empty:0.6238686130000133\n",
      "1723049341.5425055 INFO: Filtered count:0 out of 1501\n",
      "1723049342.2647042 INFO: cpd05178 not found in model!\n",
      "1723049342.2668839 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.9.pg.G.D GF: 10903 Prob: 4612 Other: 2837\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049343.7915976 INFO: cpd05178 not found in model!\n",
      "1723049343.7936573 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049349.4289732 INFO: cpd05178 not found in model!\n",
      "1723049349.6572144 INFO: cpd05178 not found in model!\n",
      "1723049349.6595511 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049350.004988 INFO: cpd05178 not found in model!\n",
      "1723049350.0070384 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-26\n",
      "[105 'R1_B_D2.51.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:28:25+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '892772f367e9885b8bc01fcd60c092dc'\n",
      " 9861639 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049351.4092765 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049352.3807368 INFO: metabolites 1133\n",
      "1723049352.8754058 INFO: reactions 1087\n",
      "1723049353.0413032 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049353.8785508 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/361/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049354.0621629 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/361/1\"\n",
      "}\n",
      "1723049358.1982996 INFO: cpd05178 not found in model!\n",
      "1723049358.2004766 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049358.2004125 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049358.4846256 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049359.6488068 INFO: cpd05178 not found in model!\n",
      "1723049359.6509624 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049362.8669508 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049363.983692 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049364.6600301 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049365.1493077 INFO: Expansion time:Ac.NO2:4.703302083999915\n",
      "1723049365.1512494 INFO: Filtered count:0 out of 1707\n",
      "1723049366.5674741 INFO: Expansion time:Ac.NO3:1.415192966000177\n",
      "1723049366.5695238 INFO: Filtered count:0 out of 1707\n",
      "1723049367.9065232 INFO: Expansion time:Ac.O2:1.3357314109998697\n",
      "1723049367.9085453 INFO: Filtered count:0 out of 1707\n",
      "1723049368.036028 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-27\n",
      "[109 'R2_B_D2.32.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:29:09+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'f628bb3530a067cf2bcd477fbdd0a19c'\n",
      " 14950785 None]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049369.1525176 INFO: metabolites 1269\n",
      "1723049369.2838867 INFO: Expansion time:Etho.O2:1.373914211999363\n",
      "1723049369.2858953 INFO: Filtered count:0 out of 1707\n",
      "1723049369.7453015 INFO: reactions 1254\n",
      "1723049370.2730882 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049370.510548 INFO: cpd05178 not found in model!\n",
      "1723049370.5128143 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[14 'R2_C_D1.88.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:10:29+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'cc17896fc3bdc7075f73e4ce3b1cf2a7'\n",
      " 15568922 None]\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049370.8106203 INFO: Filtered count:0 out of 1707\n",
      "1723049370.8086174 INFO: Expansion time:For.NO2:1.521488204999514\n",
      "1723049370.7957845 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.19.pg.G.D GF: 10649 Prob: 4936 Other: 2769188040/381/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049371.0128477 INFO: metabolites 1041\n",
      "1723049371.01723 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/381/1\"\n",
      "}\n",
      "1723049371.6767676 INFO: Expansion time:For.NO3:0.8648990419997062\n",
      "1723049371.6791148 INFO: Filtered count:0 out of 1707\n",
      "1723049372.0500927 INFO: cpd05178 not found in model!\n",
      "1723049372.0520945 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049372.200193 INFO: reactions 998\n",
      "1723049372.3970587 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049372.6503127 INFO: Filtered count:0 out of 1707\n",
      "1723049372.7875082 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/400/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049372.648251 INFO: Expansion time:For.O2:0.9677631510003266\n",
      "1723049374.0399976 INFO: Expansion time:Fum:1.3881503579996206\n",
      "1723049374.0421348 INFO: Filtered count:0 out of 1707\n",
      "1723049375.4041235 INFO: cpd05178 not found in model!\n",
      "1723049375.406441 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.51.pg.G.D GF: 11260 Prob: 4174 Other: 2920\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049375.8516405 INFO: Expansion time:Fum.O2:1.8082566960001714\n",
      "1723049375.853788 INFO: Filtered count:0 out of 1707\n",
      "1723049378.703517 INFO: Expansion time:Glc:2.848317529000269\n",
      "1723049378.7054586 INFO: Filtered count:0 out of 1707\n",
      "1723049379.8446453 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049380.6884334 INFO: Expansion time:Glc.DMSO:1.9815689919996657\n",
      "1723049380.6905684 INFO: Filtered count:0 out of 1707\n",
      "1723049382.4535887 INFO: Expansion time:Glc.O2:1.7614772589995482\n",
      "1723049382.45574 INFO: Filtered count:0 out of 1707\n",
      "1723049383.5947816 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049383.7406259 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049384.6137888 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-28\n",
      "[113 'R2A_B_H2O.42.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:29:55+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '6f763da282e1178cdd87c33cdfb1ff9f'\n",
      " 4026783 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049384.952836 INFO: Expansion time:Glc.TMAO:2.4957616350002354\n",
      "1723049384.9549956 INFO: Filtered count:0 out of 1707\n",
      "1723049384.995797 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049385.0833657 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049385.466122 INFO: metabolites 974\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049385.7850382 INFO: Expansion time:Pyr:0.8287018780001745\n",
      "1723049385.78723 INFO: Filtered count:0 out of 1707\n",
      "1723049385.8610563 INFO: reactions 882\n",
      "1723049385.970033 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049386.3040996 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/394/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049386.443007 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/394/1\"\n",
      "}\n",
      "1723049387.5648098 INFO: Expansion time:Pyr.DMSO:1.7762589929998285\n",
      "1723049387.5669432 INFO: Filtered count:0 out of 1707\n",
      "1723049388.5383723 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049388.551223 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049388.904525 INFO: Expansion time:Pyr.NO:1.3363885329999903\n",
      "1723049388.9065416 INFO: Filtered count:0 out of 1707\n",
      "1723049389.277749 INFO: Expansion time:Ac.NO2:0.6522814170002675\n",
      "1723049389.2798076 INFO: Filtered count:0 out of 1821\n",
      "1723049389.8216827 INFO: cpd05178 not found in model!\n",
      "1723049389.8237736 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049389.9283938 INFO: Expansion time:Ac.O2:0.6472074240000438\n",
      "1723049389.9305434 INFO: Filtered count:0 out of 1821\n",
      "1723049389.9996574 INFO: Expansion time:Pyr.NO2:1.0918510550000065\n",
      "1723049390.001503 INFO: Filtered count:0 out of 1707\n",
      "1723049390.1276472 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049390.5760593 INFO: Expansion time:Etho.O2:0.6442044400000668\n",
      "1723049390.5781727 INFO: Filtered count:0 out of 1821\n",
      "1723049391.1772604 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049391.2218506 INFO: Expansion time:For.NO2:0.6421375930003705\n",
      "1723049391.224101 INFO: Filtered count:0 out of 1821\n",
      "1723049391.3052707 INFO: Filtered count:0 out of 1707\n",
      "1723049391.3034704 INFO: Expansion time:Pyr.NO3:1.3007593610000185\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049391.9210606 INFO: Expansion time:For.O2:0.6956076590004159\n",
      "1723049391.9233892 INFO: Filtered count:0 out of 1821\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049392.4513466 INFO: cpd05178 not found in model!\n",
      "1723049392.453422 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049392.9447956 INFO: Expansion time:Pyr.O2:1.6382414419995257\n",
      "1723049392.9468877 INFO: Filtered count:0 out of 1707\n",
      "1723049393.3512902 INFO: Expansion time:Etho.O2:1.2575715900002251\n",
      "1723049393.3535633 INFO: Filtered count:0 out of 1535\n",
      "1723049393.3851047 INFO: Expansion time:Fum:1.4604211219993886\n",
      "1723049393.3871615 INFO: Filtered count:0 out of 1821\n",
      "1723049394.0447435 INFO: Expansion time:Fum.O2:0.6563681649995488\n",
      "1723049394.0470257 INFO: Filtered count:0 out of 1821\n",
      "1723049394.0724418 INFO: Expansion time:Fum:0.7174698509998052\n",
      "1723049394.0747945 INFO: Filtered count:0 out of 1535\n",
      "1723049394.658376 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049394.707114 INFO: Expansion time:LLac.O2:0.6588476229999287\n",
      "1723049394.7090936 INFO: Filtered count:0 out of 1821\n",
      "1723049394.7130888 INFO: Expansion time:Pyr.SO3:1.7650463030004175\n",
      "1723049394.7150495 INFO: Filtered count:0 out of 1707\n",
      "1723049394.838107 INFO: Expansion time:Fum.O2:0.7617078159992161\n",
      "1723049394.8402247 INFO: Filtered count:0 out of 1535\n",
      "1723049395.2614949 INFO: cpd05178 not found in model!\n",
      "1723049395.2636795 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049395.363062 INFO: Expansion time:Pyr:0.6526170150000326\n",
      "1723049395.3652654 INFO: Filtered count:0 out of 1821\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3045\n",
      "R2_C_D1.88.pg.G.D GF: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049395.5255322 INFO: Expansion time:Pyr:0.6840687610001623\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11519"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049395.5274637 INFO: Filtered count:0 out of 1535\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Prob: 3790 Other: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049396.028855 INFO: Expansion time:Pyr.DMSO:0.6619320830004654\n",
      "1723049396.0312262 INFO: Filtered count:0 out of 1821\n",
      "1723049396.7867045 INFO: Expansion time:Pyr.NO:0.7541092560004472\n",
      "1723049396.788992 INFO: Filtered count:0 out of 1821\n",
      "1723049396.8162775 INFO: Expansion time:Pyr.DMSO:1.2876228200002515\n",
      "1723049396.8184164 INFO: Filtered count:0 out of 1535\n",
      "1723049396.901537 INFO: cpd05178 not found in model!\n",
      "1723049396.903629 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.32.pg.G.D GF: 10745 Prob: 4778 Other: 2831\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049397.4340985 INFO: Expansion time:Pyr.NO2:0.6434270340005241\n",
      "1723049397.4360976 INFO: Filtered count:0 out of 1821\n",
      "1723049397.445832 INFO: Expansion time:Pyr.NO:0.6260484999993423\n",
      "1723049397.4481153 INFO: Filtered count:0 out of 1535\n",
      "1723049398.0772066 INFO: Expansion time:Pyr.NO2:0.6278515160001916\n",
      "1723049398.0795722 INFO: Filtered count:0 out of 1535\n",
      "1723049398.8090742 INFO: Expansion time:Pyr.NO3:0.7281087809997189\n",
      "1723049398.8112266 INFO: Filtered count:0 out of 1535\n",
      "1723049398.8733876 INFO: Expansion time:Pyr.NO3:1.435868507000123\n",
      "1723049398.8753853 INFO: Filtered count:0 out of 1821\n",
      "1723049399.2173357 INFO: cpd05178 not found in model!\n",
      "1723049399.2197084 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049399.466822 INFO: Filtered count:0 out of 1535\n",
      "1723049399.5199897 INFO: Filtered count:0 out of 1821\n",
      "1723049399.51785 INFO: Expansion time:Pyr.O2:0.6412168260003455\n",
      "1723049399.4646072 INFO: Expansion time:Pyr.O2:0.6520748459997776\n",
      "1723049400.1839564 INFO: Expansion time:Pyr.SO3:0.6626018839997414\n",
      "1723049400.1862392 INFO: Filtered count:0 out of 1821\n",
      "1723049400.8191314 INFO: Expansion time:Pyr.SO3:1.350752411999565\n",
      "1723049400.82146 INFO: Filtered count:0 out of 1535\n",
      "1723049400.8350594 INFO: Expansion time:Pyr.SO4:0.6475660759997481\n",
      "1723049400.8370864 INFO: Filtered count:0 out of 1821\n",
      "1723049401.5006936 INFO: Expansion time:Pyr.TMAO:0.6621499459997722\n",
      "1723049401.5030084 INFO: Filtered count:0 out of 1821\n",
      "1723049401.509786 INFO: Expansion time:Pyr.SO4:0.687073459000203\n",
      "1723049401.5117695 INFO: Filtered count:0 out of 1535\n",
      "1723049401.9641736 INFO: cpd05178 not found in model!\n",
      "1723049401.9665182 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049402.158994 INFO: Filtered count:0 out of 1821\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-29\n",
      "[117 'R1_A_D2.15.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:30:41+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ebb6975906a1a06f96c8f4697222c05d'\n",
      " 13020963 None]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049402.156878 INFO: Expansion time:empty:0.6520289469999625\n",
      "1723049402.1804035 INFO: Expansion time:Pyr.TMAO:0.667260389999683\n",
      "1723049402.1826212 INFO: Filtered count:0 out of 1535\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_B_H2O.42.pg.G.D GF: 12230 Prob: 2958 Other: 3166\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049402.3463588 INFO: cpd05178 not found in model!\n",
      "1723049402.5570421 INFO: cpd05178 not found in model!\n",
      "1723049402.5590189 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049402.7590415 INFO: metabolites 1076\n",
      "1723049402.7989163 INFO: Expansion time:empty:0.6146995889994287\n",
      "1723049402.8010085 INFO: Filtered count:0 out of 1535\n",
      "1723049403.258895 INFO: reactions 1029\n",
      "1723049403.3958862 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049404.028926 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/444/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049404.190526 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/444/1\"\n",
      "}\n",
      "1723049404.9524844 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049405.4337342 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00337_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049406.4127772 INFO: Expansion time:Pyr.SO4:11.696437242000684\n",
      "1723049406.4148304 INFO: Filtered count:1 out of 1707\n",
      "1723049407.305075 INFO: Expansion time:Pyr.TMAO:0.8890281490002963\n",
      "1723049407.3073533 INFO: Filtered count:1 out of 1707\n",
      "1723049407.9602547 INFO: Expansion time:empty:0.6516962989999229\n",
      "1723049407.9622474 INFO: Filtered count:1 out of 1707\n",
      "1723049408.3218029 INFO: cpd05178 not found in model!\n",
      "1723049408.3236926 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049413.7624023 INFO: Objective with gapfill database:112.7574804858254; min objective:0.01\n",
      "1723049414.8153882 INFO: Expansion time:Etho.O2:1.4915491069996278\n",
      "1723049414.8173678 INFO: Filtered count:0 out of 1425\n",
      "1723049415.124242 INFO: Expansion time:Etho.O2:1.4125338949997968\n",
      "1723049415.1264322 INFO: Filtered count:0 out of 1816\n",
      "1723049415.4755669 INFO: Expansion time:For.O2:0.6568820949996734\n",
      "1723049415.477748 INFO: Filtered count:0 out of 1425\n",
      "1723049415.7915766 INFO: Expansion time:For.NO2:0.6639898550001817\n",
      "1723049415.793677 INFO: Filtered count:0 out of 1816\n",
      "1723049416.452197 INFO: Expansion time:Pyr:0.9731939620005505\n",
      "1723049416.454245 INFO: Filtered count:0 out of 1425\n",
      "1723049416.5030415 INFO: Expansion time:For.O2:0.7079602160001741\n",
      "1723049416.5049675 INFO: Filtered count:0 out of 1816\n",
      "1723049417.167716 INFO: Expansion time:Glc.O2:0.6616851749995476\n",
      "1723049417.1697092 INFO: Filtered count:0 out of 1816\n",
      "1723049417.1935084 INFO: Expansion time:Pyr.DMSO:0.7380964719995973\n",
      "1723049417.1954765 INFO: Filtered count:0 out of 1425\n",
      "1723049417.8395765 INFO: Expansion time:Pyr:0.6686863160002758\n",
      "1723049417.8415368 INFO: Filtered count:0 out of 1816\n",
      "1723049418.512553 INFO: Expansion time:Pyr.DMSO:0.6698368059996938\n",
      "1723049418.5145688 INFO: Filtered count:0 out of 1816\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-30\n",
      "[121 'R1_B_D2.28.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:31:34+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9bac8e600098d5edbb1bb6c22805e4e0'\n",
      " 11878589 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049419.173817 INFO: Expansion time:Pyr.NO:1.9772082800000135\n",
      "1723049419.1758003 INFO: Filtered count:0 out of 1425\n",
      "1723049419.925049 INFO: Expansion time:Pyr.NO:1.4092182970007343\n",
      "1723049419.927096 INFO: Filtered count:0 out of 1816\n",
      "1723049419.9494486 INFO: metabolites 1201\n",
      "1723049419.9812944 INFO: Expansion time:Pyr.NO2:0.8044220319998203\n",
      "1723049419.9832978 INFO: Filtered count:0 out of 1425\n",
      "1723049420.0250657 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049420.5185878 INFO: reactions 1195\n",
      "1723049420.6022508 INFO: Expansion time:Pyr.NO2:0.6739411790003942\n",
      "1723049420.6041658 INFO: Filtered count:0 out of 1816\n",
      "1723049420.6704655 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049420.7064672 INFO: Filtered count:0 out of 1425\n",
      "1723049420.704469 INFO: Expansion time:Pyr.NO3:0.7199208289994203\n",
      "1723049420.8502858 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049421.2774038 INFO: Expansion time:Pyr.NO3:0.6720670160002555\n",
      "1723049421.2792914 INFO: Filtered count:0 out of 1816\n",
      "1723049421.6439705 INFO: Expansion time:Pyr.O2:0.9364084509998065\n",
      "1723049421.6460845 INFO: Filtered count:0 out of 1425\n",
      "1723049421.8469276 INFO: Default biomass: [bio1]\n",
      "1723049421.938446 INFO: Expansion time:Pyr.O2:0.658007510999596\n",
      "1723049421.9403 INFO: Filtered count:0 out of 1816\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/505/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049422.0217967 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/505/1\"\n",
      "}\n",
      "1723049422.3627806 INFO: Expansion time:Pyr.SO3:0.7153453090004405\n",
      "1723049422.3648198 INFO: Filtered count:0 out of 1425\n",
      "1723049422.597877 INFO: Expansion time:Pyr.SO3:0.6564341069997681\n",
      "1723049422.5998588 INFO: Filtered count:0 out of 1816\n",
      "1723049422.987191 INFO: Expansion time:Pyr.SO4:0.6211178570001721\n",
      "1723049422.9892313 INFO: Filtered count:0 out of 1425\n",
      "1723049423.2573826 INFO: Expansion time:Pyr.SO4:0.6563820769997619\n",
      "1723049423.2595274 INFO: Filtered count:0 out of 1816\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[26 'R2_restored_DShore.25.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:12:38+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '15bccaa1bf24906f0373e680e565b2ad'\n",
      " 11370723 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049423.6560543 INFO: Filtered count:0 out of 1425\n",
      "1723049423.6540139 INFO: Expansion time:Pyr.TMAO:0.6632908689998658\n",
      "1723049423.8817575 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049423.9976957 INFO: metabolites 1141\n",
      "1723049424.2089195 INFO: cpd05178 not found in model!\n",
      "1723049424.211197 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049424.270367 INFO: Filtered count:0 out of 1425\n",
      "1723049424.2684362 INFO: Expansion time:empty:0.6110468480001146\n",
      "1723049424.649095 INFO: Expansion time:Pyr.TMAO:1.38824182999997\n",
      "1723049424.6515791 INFO: Filtered count:0 out of 1816\n",
      "1723049425.2249956 INFO: reactions 1045\n",
      "1723049425.3460193 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049425.4952414 INFO: Expansion time:empty:0.8422268800004531\n",
      "1723049425.4971285 INFO: Filtered count:0 out of 1816\n",
      "1723049425.7506125 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/480/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049428.5815647 INFO: cpd05178 not found in model!\n",
      "1723049428.8190112 INFO: cpd05178 not found in model!\n",
      "1723049428.8207824 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049431.9462507 INFO: cpd05178 not found in model!\n",
      "1723049431.948481 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.15.pg.G.D GF: 11387 Prob: 3776 Other: 3189\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049433.5616531 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049433.8548334 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049436.391019 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-31\n",
      "[125 'R2_restored_DShore.89.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:32:27+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ca649dac2848f800b6fded398980656c'\n",
      " 13015505 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049436.6843846 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049437.3832254 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Getting tests from attributes"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049437.53201 INFO: metabolites 1205\n",
      "1723049438.2232726 INFO: reactions 1173\n",
      "1723049438.367101 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049439.3360744 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/483/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049439.5019808 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/483/1\"\n",
      "}\n",
      "1723049439.8012955 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049440.1658218 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049444.5534163 INFO: cpd05178 not found in model!\n",
      "1723049444.5557833 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049445.4134524 INFO: cpd05178 not found in model!\n",
      "1723049445.4158084 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.25.pg.G.D GF: 11503 Prob: 3892 Other: 2959\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049448.2579215 INFO: cpd05178 not found in model!\n",
      "1723049448.26006 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.28.pg.G.D GF: 10928 Prob: 4606 Other: 2818\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049449.1545022 INFO: cpd05178 not found in model!\n",
      "1723049449.1562827 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[34 'R2_restored_DShore.17.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:14:24+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'c4e6a8a5e2b7eb2cfa59dc893e2ac7c0'\n",
      " 10052572 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049450.4257967 INFO: metabolites 1065\n",
      "1723049450.4873488 INFO: cpd05178 not found in model!\n",
      "1723049450.4893396 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049450.792512 INFO: reactions 989\n",
      "1723049450.9461434 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049451.1715457 INFO: Expansion time:Dlac:2.4983740730003774\n",
      "1723049451.173735 INFO: Filtered count:0 out of 1437\n",
      "1723049451.3252823 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/518/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049453.4152284 INFO: Expansion time:Dlac.O2:2.240109577000112\n",
      "1723049453.4172623 INFO: Filtered count:0 out of 1437\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-32\n",
      "[129 'R1_C_H2O.23.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:33:07+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e90764056776ff7d69d86fe091586bc7'\n",
      " 6416704 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049454.3972323 INFO: Expansion time:Fum:0.9786989729991546\n",
      "1723049454.3994434 INFO: Filtered count:0 out of 1437\n",
      "1723049454.5435648 INFO: metabolites 1094\n",
      "1723049454.968056 INFO: reactions 1009\n",
      "1723049455.1272864 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049455.3616729 INFO: Expansion time:Fum.O2:0.9610103869999875\n",
      "1723049455.363692 INFO: Filtered count:0 out of 1437\n",
      "1723049455.4768057 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049455.5338836 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/523/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049456.0750234 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/523/1\"\n",
      "}\n",
      "1723049456.3039105 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049456.4996915 INFO: Expansion time:Glc:1.1348402399999031\n",
      "1723049456.5017169 INFO: Filtered count:0 out of 1437\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Getting tests from attributes"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049457.3352437 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049457.7022088 INFO: Expansion time:Glc.DMSO:1.1992133289995763\n",
      "1723049457.7043214 INFO: Filtered count:0 out of 1437\n",
      "1723049458.8090334 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049458.8531554 INFO: Filtered count:0 out of 1437\n",
      "1723049458.8508816 INFO: Expansion time:Glc.O2:1.1450137530000575\n",
      "1723049458.9637983 INFO: Objective with gapfill database:164.73810836770008; min objective:0.01\n",
      "1723049460.1829789 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049460.5857174 INFO: Expansion time:Glc.TMAO:1.7311413620000167\n",
      "1723049460.5878172 INFO: Filtered count:0 out of 1437\n",
      "1723049460.8157291 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049461.4139674 INFO: Expansion time:Pyr:0.824970215000576\n",
      "1723049461.4161222 INFO: Filtered count:0 out of 1437\n",
      "1723049462.2813435 INFO: Expansion time:Pyr.DMSO:0.8640187589999186\n",
      "1723049462.2836215 INFO: Filtered count:0 out of 1437\n",
      "1723049462.4830878 INFO: cpd05178 not found in model!\n",
      "1723049462.485055 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049463.174139 INFO: Expansion time:Pyr.NO:0.8888873169998988\n",
      "1723049463.1762283 INFO: Filtered count:0 out of 1437\n",
      "1723049463.5884602 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049463.9583604 INFO: Expansion time:Pyr.NO2:0.7809897339993768\n",
      "1723049463.960673 INFO: Filtered count:0 out of 1437\n",
      "1723049464.1398995 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049464.8714015 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049465.379787 INFO: Expansion time:Pyr.NO3:1.417871099999502\n",
      "1723049465.3818858 INFO: Filtered count:0 out of 1437\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049466.5454786 INFO: Expansion time:Pyr.O2:1.1624472519997653\n",
      "1723049466.5475893 INFO: Filtered count:0 out of 1437\n",
      "1723049467.220645 INFO: Expansion time:Ac.NO2:0.6557971700003691\n",
      "1723049467.223043 INFO: Filtered count:0 out of 1724\n",
      "1723049467.5041063 INFO: Expansion time:Pyr.SO3:0.9553546139995888\n",
      "1723049467.5062382 INFO: Filtered count:0 out of 1437\n",
      "1723049468.227866 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049468.4000063 INFO: Expansion time:Pyr.SO4:0.8925586630002726\n",
      "1723049468.401988 INFO: Filtered count:0 out of 1437\n",
      "1723049468.586646 INFO: Expansion time:Ac.O2:1.3623491890002697\n",
      "1723049468.5885909 INFO: Filtered count:0 out of 1724\n",
      "1723049469.0693822 INFO: cpd05178 not found in model!\n",
      "1723049469.071709 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049469.1467152 INFO: Filtered count:0 out of 1437\n",
      "1723049469.2337945 INFO: Filtered count:0 out of 1724\n",
      "1723049469.1447167 INFO: Expansion time:Pyr.TMAO:0.7415678329998627\n",
      "1723049469.2317214 INFO: Expansion time:Etho.O2:0.6417155529998126\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.89.pg.G.D GF: 10842 Prob: 4690 Other: 2822\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049469.7664373 INFO: Expansion time:empty:0.6185755249998692\n",
      "1723049469.7684927 INFO: Filtered count:0 out of 1437\n",
      "1723049469.8724625 INFO: Expansion time:Fum:0.637376356000459\n",
      "1723049469.8745089 INFO: Filtered count:0 out of 1724\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-33\n",
      "[133 'SF2_B_D1.9.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:33:55+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'cc4c1233a0deb295e239694d7e2145ff'\n",
      " 11592517 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049470.5183647 INFO: Expansion time:Fum.O2:0.642644227999881\n",
      "1723049470.5202386 INFO: Filtered count:0 out of 1724\n",
      "1723049471.1484697 INFO: metabolites 1116\n",
      "1723049471.1501813 INFO: Expansion time:Pyr:0.6288681919995724\n",
      "1723049471.151956 INFO: Filtered count:0 out of 1724\n",
      "1723049471.37115 INFO: cpd05178 not found in model!\n",
      "1723049471.3731847 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.17.pg.G.D GF: 11808 Prob: 3360 Other: 3186\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049471.6520386 INFO: reactions 1046\n",
      "1723049471.7726002 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049471.783344 INFO: Expansion time:Pyr.DMSO:0.6301942949994555\n",
      "1723049471.78507 INFO: Filtered count:0 out of 1724\n",
      "1723049472.416222 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/560/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049472.5674148 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/560/1\"\n",
      "}\n",
      "1723049473.3290994 INFO: Expansion time:Pyr.NO:1.5426558449998993\n",
      "1723049473.330982 INFO: Filtered count:0 out of 1724\n",
      "1723049473.9635859 INFO: Expansion time:Pyr.NO2:0.6312284559999171\n",
      "1723049473.9658005 INFO: Filtered count:0 out of 1724\n",
      "1723049474.6394114 INFO: Expansion time:Pyr.NO3:0.6719330559999435\n",
      "1723049474.6413507 INFO: Filtered count:0 out of 1724\n",
      "1723049475.2834735 INFO: Expansion time:Pyr.O2:0.6404932850000478\n",
      "1723049475.2856057 INFO: Filtered count:0 out of 1724\n",
      "1723049475.3501096 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049475.3479502 INFO: cpd05178 not found in model!\n",
      "1723049475.8808835 INFO: cpd05178 not found in model!\n",
      "1723049475.8829412 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049475.9236186 INFO: Expansion time:Pyr.SO3:0.6367694259997734\n",
      "1723049475.9255862 INFO: Filtered count:0 out of 1724\n",
      "1723049476.0295916 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_H2O.23.pg.G.D GF: 11692 Prob: 3616 Other: 3044\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049476.2600596 INFO: cpd05178 not found in model!\n",
      "1723049476.262289 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049477.2970245 INFO: Expansion time:Pyr.SO4:1.3701134709999678\n",
      "1723049477.29904 INFO: Filtered count:0 out of 1724\n",
      "1723049477.9417114 INFO: Expansion time:Pyr.TMAO:0.6413263550002739\n",
      "1723049477.9436727 INFO: Filtered count:0 out of 1724\n",
      "1723049478.8268218 INFO: Expansion time:empty:0.8817721049999818\n",
      "1723049478.8308074 INFO: Filtered count:0 out of 1724\n",
      "1723049479.5852275 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049479.5841396 INFO: cpd05178 not found in model!\n",
      "1723049479.8100104 INFO: cpd05178 not found in model!\n",
      "1723049479.812105 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049479.8391306 INFO: Objective with gapfill database:227.34906832306484; min objective:0.01\n",
      "1723049483.8448691 INFO: cpd05178 not found in model!\n",
      "1723049483.8472235 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049483.8936906 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049484.044454 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049484.0417018 INFO: cpd05178 not found in model!\n",
      "1723049485.3886294 INFO: cpd05178 not found in model!\n",
      "1723049485.6234114 INFO: cpd05178 not found in model!\n",
      "1723049485.6254125 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049487.4059315 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049488.228511 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-34\n",
      "[137 'R2A_A_H2O.49.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:34:35+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'cd4c5e27b036e36b25f8ba5ef4702221'\n",
      " 3653639 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049488.402478 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049488.6271465 INFO: cpd05178 not found in model!\n",
      "1723049488.6293364 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049488.6796372 INFO: Expansion time:Etho.O2:0.6312328540007002\n",
      "1723049488.681637 INFO: Filtered count:0 out of 1399\n",
      "1723049489.0102754 INFO: metabolites 897\n",
      "1723049489.2830362 INFO: reactions 821\n",
      "1723049489.3874612 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049489.6991262 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/540/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049489.8551393 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/540/1\"\n",
      "}\n",
      "1723049490.4063509 INFO: Expansion time:Pyr:1.7233349550006096\n",
      "1723049490.4084427 INFO: Filtered count:0 out of 1399\n",
      "1723049491.0068784 INFO: Expansion time:Pyr.DMSO:0.5972715710004195\n",
      "1723049491.0090485 INFO: Filtered count:0 out of 1399\n",
      "1723049491.3832676 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049491.6499324 INFO: Expansion time:Pyr.NO:0.6394593300001361\n",
      "1723049491.651972 INFO: Filtered count:0 out of 1399\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049492.2692204 INFO: Expansion time:Pyr.NO2:0.6161398100002771\n",
      "1723049492.2712567 INFO: Filtered count:0 out of 1399\n",
      "1723049492.7658498 INFO: Expansion time:Etho.O2:0.6126442730001145\n",
      "1723049492.768051 INFO: Filtered count:0 out of 1447\n",
      "1723049492.8277242 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049492.8717027 INFO: Filtered count:0 out of 1399\n",
      "1723049492.8697884 INFO: Expansion time:Pyr.NO3:0.5974241820003954\n",
      "1723049493.4179435 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049493.4747314 INFO: Filtered count:0 out of 1399\n",
      "1723049493.4727042 INFO: Expansion time:Pyr.O2:0.599891359999674\n",
      "1723049493.9892728 INFO: Expansion time:Fum:1.2197872719998486\n",
      "1723049493.9914849 INFO: Filtered count:0 out of 1447\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049494.083581 INFO: Expansion time:Pyr.SO3:0.6077475459997004\n",
      "1723049494.0856295 INFO: Filtered count:0 out of 1399\n",
      "1723049494.5918233 INFO: Expansion time:Fum.O2:0.5990833449995989\n",
      "1723049494.5939608 INFO: Filtered count:0 out of 1447\n",
      "1723049495.1939104 INFO: Expansion time:Glc:0.5986537460003092\n",
      "1723049495.1959956 INFO: Filtered count:0 out of 1447\n",
      "1723049495.7742405 INFO: Expansion time:Pyr.SO4:1.6873805499999435\n",
      "1723049495.7763143 INFO: Filtered count:0 out of 1399\n",
      "1723049495.7980492 INFO: Expansion time:Glc.DMSO:0.6009044689999428\n",
      "1723049495.8000727 INFO: Filtered count:0 out of 1447\n",
      "1723049496.3797283 INFO: Expansion time:Pyr.TMAO:0.6023145569997723\n",
      "1723049496.3818777 INFO: Filtered count:0 out of 1399\n",
      "1723049496.7532368 INFO: cpd05178 not found in model!\n",
      "1723049496.7556112 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049496.9673188 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049496.9931827 INFO: Filtered count:0 out of 1399\n",
      "1723049496.9911997 INFO: Expansion time:empty:0.6081526180005312\n",
      "1723049497.5069447 INFO: Expansion time:Glc.O2:1.7054172600001039\n",
      "1723049497.5091262 INFO: Filtered count:0 out of 1447\n",
      "1723049497.675558 INFO: cpd05178 not found in model!\n",
      "1723049497.9120157 INFO: cpd05178 not found in model!\n",
      "1723049497.914068 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049497.9405346 INFO: cpd05178 not found in model!\n",
      "1723049497.9429097 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049498.1172 INFO: Expansion time:Glc.TMAO:0.6066283700001804\n",
      "1723049498.1192923 INFO: Filtered count:0 out of 1447\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[54 'R2_restored_C_black.43.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:18:19+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '63cf840e1416f451f92d5004ee7da064'\n",
      " 15050027 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049498.6867485 INFO: cpd05178 not found in model!\n",
      "1723049498.6888306 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049498.7109027 INFO: metabolites 1094\n",
      "1723049498.8663106 INFO: Filtered count:0 out of 1447\n",
      "1723049498.8645048 INFO: Expansion time:Pyr:0.7438002550006786\n",
      "1723049499.1297112 INFO: reactions 1077\n",
      "1723049499.255917 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049499.3434038 INFO: cpd05178 not found in model!\n",
      "1723049499.3455048 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049499.4699333 INFO: Expansion time:Pyr.DMSO:0.6025661389994639\n",
      "1723049499.4718943 INFO: Filtered count:0 out of 1447\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049499.6808827 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_D1.9.pg.G.D GF: 11477 Prob:3814 Other: 3061\n",
      "188040/608/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049500.0801663 INFO: Expansion time:Pyr.NO:0.606941122000535\n",
      "1723049500.0824888 INFO: Filtered count:0 out of 1447\n",
      "1723049501.3077722 INFO: Expansion time:Pyr.NO2:1.2238645289999113\n",
      "1723049501.309901 INFO: Filtered count:0 out of 1447\n",
      "1723049501.9125125 INFO: Expansion time:Pyr.NO3:0.6012355369994111\n",
      "1723049501.9146488 INFO: Filtered count:0 out of 1447\n",
      "1723049502.5207326 INFO: Expansion time:Pyr.O2:0.6048410889998195\n",
      "1723049502.5228963 INFO: Filtered count:0 out of 1447\n",
      "1723049503.125166 INFO: Expansion time:Pyr.SO3:0.6009256790002837\n",
      "1723049503.1273172 INFO: Filtered count:0 out of 1447\n",
      "1723049503.7362072 INFO: Expansion time:Pyr.SO4:0.607548166000015\n",
      "1723049503.7382984 INFO: Filtered count:0 out of 1447\n",
      "1723049504.4449058 INFO: cpd05178 not found in model!\n",
      "1723049504.4470935 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_H2O.49.pg.G.D GF: 12671 Prob: 2266 Other: 3415\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049504.9803216 INFO: Expansion time:Pyr.TMAO:1.2408849510002256\n",
      "1723049504.9826066 INFO: Filtered count:0 out of 1447\n",
      "1723049505.591137 INFO: Expansion time:empty:0.6073614249999082\n",
      "1723049505.593192 INFO: Filtered count:0 out of 1447\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-35\n",
      "[141 'R1_B_D1.58.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:35:29+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '4832fb5b0dad799145ce38d96963d800'\n",
      " 7252736 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049506.4256802 INFO: cpd05178 not found in model!\n",
      "1723049506.4277933 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049506.676776 INFO: cpd05178 not found in model!\n",
      "1723049506.679021 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049506.7424664 INFO: metabolites 947\n",
      "1723049507.074109 INFO: reactions 886\n",
      "1723049507.177804 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049507.5220919 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/572/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049507.664328 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/572/1\"\n",
      "}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[66 'R1_A_D2.48.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:20:47+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'd7eb2d8e6e040e77c982efe91763917c'\n",
      " 7887435 None]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049508.2017336 INFO: metabolites 988\n",
      "1723049509.2976298 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049509.4013104 INFO: reactions 934\n",
      "1723049509.515863 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049509.5886798 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[22 'R2A_B_H2O.10.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:11:55+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '64a7a1e5b22bda7b9b34f55855c47d59'\n",
      " 9938856 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049509.875615 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/88/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049510.2964141 INFO: metabolites 1080\n",
      "1723049510.7417634 INFO: reactions 1097\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049510.8654268 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049510.969311 INFO: Objective with gapfill database:240.65774532345074; min objective:0.01\n",
      "1723049511.2623644 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/465/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049512.8916676 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049513.300591 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049513.490148 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[9 'R2_A_H2O.29.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:09:27+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '177401c0bcdc33f0ada9432d65ce06fa'\n",
      " 10356559 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049513.8307855 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049514.11855 INFO: metabolites 860\n",
      "1723049514.4441154 INFO: reactions 778\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049514.560901 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049514.8690438 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/239/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049517.226802 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049517.6533813 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049517.8286636 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049518.12868 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049518.3611352 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Getting tests from attributesGetting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049519.81331 INFO: cpd05178 not found in model!\n",
      "1723049519.8155584 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[4 'R1_B_D1.20.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:08:27+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '18975ce8d06d8eeaa4f262ec7f642498'\n",
      " 12725637 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049521.2262585 INFO: metabolites 1228\n",
      "1723049521.26119 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049521.2590833 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_C_black.43.pg.G.D GF: 11762 Prob: 3436 Other: 3154\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049521.7684362 INFO: reactions 1236\n",
      "1723049521.9120343 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049521.9185352 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049521.949447 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049522.408688 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/346/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049522.9057684 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-36\n",
      "[145 'R2_restored_H2O.002.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:36:14+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '052a1ba8e65e7063cdad5718e71ba0df'\n",
      " 8926692 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049523.3957102 INFO: Expansion time:For.O2:0.5932695560004504\n",
      "1723049523.397975 INFO: Filtered count:0 out of 1089\n",
      "1723049523.9397528 INFO: metabolites 1028\n",
      "1723049523.9903197 INFO: Filtered count:0 out of 1089\n",
      "1723049523.9881704 INFO: Expansion time:Fum:0.5888774129998637\n",
      "1723049524.1755402 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049524.4925716 INFO: reactions 946\n",
      "1723049524.564881 INFO: Filtered count:0 out of 1089\n",
      "1723049524.562647 INFO: Expansion time:Fum.O2:0.5711808079995535\n",
      "1723049524.6839128 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049524.9069831 INFO: cpd05178 not found in model!\n",
      "1723049524.9087396 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D1.58.pg.G.D GF: 12089 Prob: 3028 Other: 3233\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049525.2361238 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/588/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049525.443639 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/588/1\"\n",
      "}\n",
      "1723049525.736377 INFO: Expansion time:Pyr:1.1702236159999302\n",
      "1723049525.7385588 INFO: Filtered count:0 out of 1089\n",
      "1723049526.3212583 INFO: Expansion time:Pyr.DMSO:0.5815209089996642\n",
      "1723049526.3233035 INFO: Filtered count:0 out of 1089\n",
      "1723049526.8997808 INFO: Expansion time:Pyr.NO:0.5752993939995577\n",
      "1723049526.9020047 INFO: Filtered count:0 out of 1089\n",
      "1723049527.4720292 INFO: Expansion time:Pyr.NO2:0.5688035820003279\n",
      "1723049527.4745774 INFO: Filtered count:0 out of 1089\n",
      "1723049527.5823245 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049528.0469654 INFO: Expansion time:Pyr.NO3:0.5711429250004585\n",
      "1723049528.0491114 INFO: Filtered count:0 out of 1089\n",
      "1723049529.2185435 INFO: Expansion time:Pyr.O2:1.168251625999801\n",
      "1723049529.2207682 INFO: Filtered count:0 out of 1089\n",
      "1723049529.774661 INFO: cpd05178 not found in model!\n",
      "1723049529.776702 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049529.7911217 INFO: Expansion time:Pyr.SO3:0.5692186090000178\n",
      "1723049529.7932708 INFO: Filtered count:0 out of 1089\n",
      "1723049529.8006048 INFO: cpd05178 not found in model!\n",
      "1723049529.802559 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_B_H2O.10.pg.G.D GF: R1_A_D2.48.pg.G.D11797  GF:Prob:  120833526  Prob:Other:  29083031 \n",
      "Other: 3361\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049530.3651996 INFO: Expansion time:Pyr.SO4:0.5707327649997751\n",
      "1723049530.3674335 INFO: Filtered count:0 out of 1089\n",
      "1723049531.5605762 INFO: Expansion time:Pyr.TMAO:1.1920018920000075\n",
      "1723049531.5629535 INFO: Filtered count:0 out of 1089\n",
      "1723049532.1874228 INFO: Expansion time:empty:0.6231290709993118\n",
      "1723049532.189558 INFO: Filtered count:0 out of 1089\n",
      "1723049532.832691 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049532.9161015 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049532.9141848 INFO: cpd05178 not found in model!\n",
      "1723049533.4389458 INFO: cpd05178 not found in model!\n",
      "1723049533.441293 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049533.4793677 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049533.5710971 INFO: Objective with gapfill database:176.72496329768458; min objective:0.01\n",
      "1723049533.7579098 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049534.055671 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049535.2201989 INFO: cpd05178 not found in model!\n",
      "1723049535.2223222 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "\n",
      "3228R2_A_H2O.29.pg.G.D GF: 12170 Prob: 2962 Other: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049536.6890633 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049537.6332912 INFO: Objective with gapfill database:84.3140747428064; min objective:0.01\n",
      "1723049537.7381215 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049539.0646837 INFO: cpd05178 not found in model!\n",
      "1723049539.0668335 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-37\n",
      "[149 'R2_C_D2.47.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:37:02+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'a22d9474206662700ad9db065ebeb4f9'\n",
      " 14668269 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049540.9033415 INFO: Objective with gapfill database:162.55349415076645; min objective:0.01\n",
      "1723049541.6501067 INFO: Objective with gapfill database:169.50893253519286; min objective:0.01\n",
      "1723049541.779552 INFO: metabolites 1249\n",
      "1723049542.655536 INFO: reactions 1216\n",
      "1723049543.2338526 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049543.9859421 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/16/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049544.2361057 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/16/1\"\n",
      "}\n",
      "1723049544.2913897 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049544.289254 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_H2O.002.pg.G.D GF: 11806 Prob: 3430 Other: 3116\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049545.520369 INFO: Objective with gapfill database:170.4983507693903; min objective:0.01\n",
      "1723049546.1892762 INFO: cpd05178 not found in model!\n",
      "1723049546.1912177 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D1.20.pg.G.D GF: 10912 Prob: 4606 Other: 2834\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049547.7937114 INFO: cpd05178 not found in model!\n",
      "1723049547.7958243 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049553.0716186 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049555.3873985 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049556.1264637 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "ForkPoolWorker-38\n",
      "[153 'SF2_B_H2O.39.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:37:44+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e52b7ee204949b1cdc65dca49d68e3fe'\n",
      " 3915100 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049557.5055404 INFO: Objective with gapfill database:202.24660999734675; min objective:0.01\n",
      "1723049557.6951694 INFO: cpd05178 not found in model!\n",
      "1723049557.697065 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049558.3264084 INFO: metabolites 974\n",
      "1723049558.641861 INFO: reactions 884\n",
      "1723049558.7549243 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049559.0850544 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/20/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049559.2320704 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/20/1\"\n",
      "}\n",
      "1723049559.7375 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049562.7036936 INFO: Expansion time:Etho.O2:1.213316733999818\n",
      "1723049562.7059588 INFO: Filtered count:0 out of 1314\n",
      "1723049563.3024294 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049563.3377967 INFO: Filtered count:0 out of 1314\n",
      "1723049563.3356967 INFO: Expansion time:For.O2:0.6283719210005074\n",
      "1723049563.9350424 INFO: Expansion time:Pyr:0.5960912929995175\n",
      "1723049563.9372663 INFO: Filtered count:0 out of 1314\n",
      "1723049564.3860013 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049564.5315044 INFO: Expansion time:Pyr.DMSO:0.5928992939998352\n",
      "1723049564.533575 INFO: Filtered count:0 out of 1314\n",
      "1723049564.6379108 INFO: cpd05178 not found in model!\n",
      "1723049564.6400928 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049565.12754 INFO: Expansion time:Pyr.NO:0.5926743879999776\n",
      "1723049565.1295714 INFO: Filtered count:0 out of 1314\n",
      "1723049566.3248205 INFO: Expansion time:Pyr.NO2:1.194097270000384\n",
      "1723049566.327214 INFO: Filtered count:0 out of 1314\n",
      "1723049566.930062 INFO: Expansion time:Pyr.NO3:0.6017183080002724\n",
      "1723049566.932273 INFO: Filtered count:0 out of 1314\n",
      "1723049567.5389757 INFO: Expansion time:Pyr.O2:0.6055859919997602\n",
      "1723049567.541149 INFO: Filtered count:0 out of 1314\n",
      "1723049567.666643 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049567.7037325 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049567.7018087 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D2.47.pg.G.D GF: 11159 Prob: 4300 Other: 2893\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049568.143308 INFO: Expansion time:Pyr.SO3:0.6010104810002304\n",
      "1723049568.1453261 INFO: Filtered count:0 out of 1314\n",
      "1723049568.7429814 INFO: Expansion time:Pyr.SO4:0.5962235500001043\n",
      "1723049568.7451153 INFO: Filtered count:0 out of 1314\n",
      "1723049569.9413679 INFO: Expansion time:Pyr.TMAO:1.1949845069993899\n",
      "1723049569.9435837 INFO: Filtered count:0 out of 1314\n",
      "1723049570.538378 INFO: Expansion time:empty:0.5935772089997045\n",
      "1723049570.5406206 INFO: Filtered count:0 out of 1314\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-39\n",
      "[157 'R2A_C_H2O.5.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:38:33+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '48570a765b2afcee734e05b5b2f046ff'\n",
      " 9844889 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049575.2782753 INFO: cpd05178 not found in model!\n",
      "1723049575.2806315 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_H2O.39.pg.G.D GF: 12365 Prob: 2788 Other: 3201\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049575.6006987 INFO: metabolites 1129\n",
      "1723049576.0283618 INFO: reactions 1090\n",
      "1723049576.1770597 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049576.5965867 INFO: Default biomass: [bio1]\n",
      "1723049576.8617926 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/50/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049576.9814694 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/50/1\"\n",
      "}\n",
      "1723049578.6140149 INFO: cpd05178 not found in model!\n",
      "1723049578.6163487 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049579.639015 INFO: cpd05178 not found in model!\n",
      "1723049579.866962 INFO: cpd05178 not found in model!\n",
      "1723049579.8692107 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049583.0531905 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049583.7974463 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049585.1117513 INFO: cpd05178 not found in model!\n",
      "1723049586.1062627 INFO: cpd05178 not found in model!\n",
      "1723049586.1082265 INFO: Media compound: cpd05178 not found in model.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049588.0228906 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049588.4160416 INFO: Objective with gapfill database:86.52683431653445; min objective:0.01\n",
      "1723049588.8513443 INFO: Expansion time:Etho.O2:0.6880357659993024\n",
      "1723049588.8550656 INFO: Filtered count:0 out of 1733\n",
      "1723049589.5110948 INFO: Expansion time:For.NO2:0.6545613210000738\n",
      "1723049589.5134447 INFO: Filtered count:0 out of 1733\n",
      "1723049590.1716776 INFO: Expansion time:For.O2:0.6568732739997358\n",
      "1723049590.1740284 INFO: Filtered count:0 out of 1733\n",
      "1723049590.3573418 INFO: cpd05178 not found in model!\n",
      "1723049590.6026573 INFO: cpd05178 not found in model!\n",
      "1723049590.6048186 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-40\n",
      "[161 'R1_B_D1.53.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:39:15+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ad37db20484194fb952bbeda362a970c'\n",
      " 10138235 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049592.0623307 INFO: Expansion time:Fum:1.8869578379999439\n",
      "1723049592.0645897 INFO: Filtered count:0 out of 1733\n",
      "1723049592.2165232 INFO: metabolites 1115\n",
      "1723049592.7177289 INFO: Expansion time:Fum.O2:0.6519531730000381\n",
      "1723049592.719992 INFO: Filtered count:0 out of 1733\n",
      "1723049592.779818 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049592.8811338 INFO: reactions 1103\n",
      "1723049592.7776837 INFO: cpd05178 not found in model!\n",
      "1723049593.0664108 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049593.365872 INFO: Expansion time:Pyr.NO2:0.6447248710001077\n",
      "1723049593.3680506 INFO: Filtered count:0 out of 1733\n",
      "1723049593.7615457 INFO: Default biomass: [bio1]\n",
      "1723049594.0383399 INFO: Expansion time:Pyr.O2:0.6691629059996558\n",
      "1723049594.040632 INFO: Filtered count:0 out of 1733\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/96/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049594.5227382 INFO: cpd05178 not found in model!\n",
      "1723049594.5100508 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/96/1\"\n",
      "}\n",
      "1723049594.70105 INFO: Filtered count:0 out of 1733\n",
      "1723049594.7745702 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049594.698852 INFO: Expansion time:empty:0.656952770999851\n",
      "1723049594.7726095 INFO: cpd05178 not found in model!\n",
      "1723049597.4227018 INFO: cpd05178 not found in model!\n",
      "1723049597.424787 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_H2O.5.pg.G.D GF: 11421 Prob: 3918 Other: 3013\n",
      "[106 'R2_restored_DShore.7.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:28:35+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '72bfc3306ad76120e555ec0f63b78a9e'\n",
      " 10051275 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049601.3246312 INFO: metabolites 1059\n",
      "1723049601.9055052 INFO: reactions 981\n",
      "1723049602.055237 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049603.586407 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/368/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049604.0696084 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049604.8839188 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049605.890249 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049607.5570436 INFO: cpd05178 not found in model!\n",
      "1723049607.5595002 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[30 'R1_B_D1.38.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:13:34+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '62d508b167d70f32273791501a294d79'\n",
      " 10196511 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049608.4659889 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049608.5066874 INFO: metabolites 1147\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-41\n",
      "[165 'R2_B_H2O.29.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:40:03+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'fcc75d82a502d9bbb6ec0154829f8b85'\n",
      " 4729616 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049609.1643655 INFO: reactions 1128\n",
      "1723049609.3540437 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049609.445572 INFO: metabolites 540\n",
      "1723049609.6197836 INFO: reactions 464\n",
      "1723049609.7490127 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049610.0271938 INFO: Default biomass: [bio1]\n",
      "1723049610.0661263 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/506/1\n",
      "188040/248/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049610.228227 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/248/1\"\n",
      "}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[78 'SF2_A_H2O.33.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:23:03+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'fc68e91115a1e84747132f19214905c5'\n",
      " 6638047 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049612.683743 INFO: metabolites 938\n",
      "1723049612.9932463 INFO: reactions 863\n",
      "1723049613.0914643 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049613.4220884 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/154/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049614.107609 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049615.1034088 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049616.3123722 INFO: Expansion time:Ac.O2:1.6754789179994987\n",
      "1723049616.31447 INFO: Filtered count:0 out of 1595\n",
      "1723049616.4100761 INFO: cpd05178 not found in model!\n",
      "1723049616.4121103 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D1.53.pg.G.D GF: 11659 Prob: 3636 Other: 3057\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049617.860719 INFO: Expansion time:Dlac.O2:1.5448498149999068\n",
      "1723049617.8628044 INFO: Filtered count:0 out of 1595\n",
      "1723049617.864225 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049618.1605887 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049618.6096115 INFO: Expansion time:For.O2:0.745639126000242\n",
      "1723049618.6117427 INFO: Filtered count:0 out of 1595\n",
      "1723049618.8651865 INFO: cpd05178 not found in model!\n",
      "1723049618.8674483 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049619.7424161 INFO: Expansion time:Glc:1.1294598989998121\n",
      "1723049619.7445967 INFO: Filtered count:0 out of 1595\n",
      "1723049619.8896542 INFO: cpd05178 not found in model!\n",
      "1723049619.8918056 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[110 'R1_C_D1.45.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:29:19+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '657696213affec6ae2e63111ab8fd14a'\n",
      " 10144752 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049620.6591947 INFO: metabolites 988\n",
      "1723049621.3573136 INFO: reactions 953\n",
      "1723049621.4757245 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049621.803222 INFO: cpd05178 not found in model!\n",
      "1723049621.8052447 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049621.879564 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/382/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049623.2678146 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049623.436499 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00062_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049623.8035848 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049623.889114 INFO: Expansion time:Glc.DMSO:4.143306785999812\n",
      "1723049623.8910322 INFO: Filtered count:1 out of 1595\n",
      "1723049624.0240226 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049624.2782478 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049624.3227289 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049625.186774 INFO: Expansion time:Glc.O2:1.2944115460004468\n",
      "1723049625.188959 INFO: Filtered count:1 out of 1595\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-42\n",
      "[169 'R1_C_H2O.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:40:50+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9b4c0a25cc45d26ad627c518fee893c1'\n",
      " 6118550 None]\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049625.8332295 INFO: Expansion time:Glc.TMAO:0.6429527549998966\n",
      "1723049625.8352735 INFO: Filtered count:1 out of 1595\n",
      "1723049626.4541912 INFO: metabolites 902\n",
      "1723049626.4748738 INFO: Filtered count:1 out of 1595\n",
      "1723049626.4729602 INFO: Expansion time:Pyr:0.6364995379999527\n",
      "1723049626.850181 INFO: reactions 797\n",
      "1723049626.9779313 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049627.1192126 INFO: Expansion time:Pyr.DMSO:0.6430989069995121\n",
      "1723049627.1212654 INFO: Filtered count:1 out of 1595\n",
      "1723049627.4344335 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/114/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049627.6067607 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/114/1\"\n",
      "}\n",
      "1723049627.8761075 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049627.9958963 INFO: Expansion time:Pyr.NO:0.8733325269995476\n",
      "1723049627.9979627 INFO: Filtered count:1 out of 1595\n",
      "1723049628.3501873 INFO: Objective with gapfill database:216.243732955965; min objective:0.01\n",
      "1723049629.2254038 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049629.2563963 INFO: Filtered count:1 out of 1595\n",
      "1723049629.2544594 INFO: Expansion time:Pyr.NO2:1.2552561149996109\n",
      "1723049629.8884964 INFO: Expansion time:Pyr.NO3:0.6309574859997156\n",
      "1723049629.8905873 INFO: Filtered count:1 out of 1595\n",
      "1723049630.191625 INFO: cpd05178 not found in model!\n",
      "1723049630.1939542 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.7.pg.G.D GF: 11459 Prob: 4002 Other: 2891\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049630.5343802 INFO: Expansion time:Pyr.O2:0.642668378000053\n",
      "1723049630.5363472 INFO: Filtered count:1 out of 1595\n",
      "1723049630.7851987 INFO: cpd05178 not found in model!\n",
      "1723049630.7871656 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_H2O.29.pg.G.D GF: 13591 Prob: 1082 Other: 3687\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049631.1603847 INFO: Expansion time:Pyr.SO3:0.6229514349997771\n",
      "1723049631.162566 INFO: Filtered count:1 out of 1595\n",
      "1723049632.0232844 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049632.4628708 INFO: Expansion time:Pyr.SO4:1.2989626039998257\n",
      "1723049632.4647946 INFO: Filtered count:1 out of 1595\n",
      "1723049632.5713656 INFO: cpd05178 not found in model!\n",
      "1723049632.5732737 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049633.0644205 INFO: cpd05178 not found in model!\n",
      "1723049633.0663161 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049633.095126 INFO: Expansion time:Pyr.TMAO:0.6292080110006282\n",
      "1723049633.0971222 INFO: Filtered count:1 out of 1595\n",
      "1723049633.3261187 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049633.7261727 INFO: Expansion time:empty:0.6278789529997084\n",
      "1723049633.72818 INFO: Filtered count:1 out of 1595\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Getting tests from attributes"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049635.5471814 INFO: cpd05178 not found in model!\n",
      "1723049635.549243 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049635.7278304 INFO: cpd05178 not found in model!\n",
      "1723049635.7301178 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "3032R1_B_D1.38.pg.G.D GF: 11570 Prob: 3752 Other: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049636.7357316 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049637.3043077 INFO: cpd05178 not found in model!\n",
      "1723049637.3064284 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_H2O.33.pg.G.D GF: 12292 Prob: 2828 Other: 3234\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049638.32311 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049638.6130173 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049640.3444147 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049642.2071176 INFO: Objective with gapfill database:177.25498503171292; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-43\n",
      "[173 'SF2_A_D2.11.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:41:37+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'a3874f46acf94ff0e4ffce652eca751b'\n",
      " 5280601 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049643.020624 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049643.503958 INFO: metabolites 921\n",
      "1723049643.97642 INFO: reactions 804\n",
      "1723049644.1170259 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049644.295652 INFO: cpd05178 not found in model!\n",
      "1723049644.2979195 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049644.5868876 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R1_C_H2O.16.pg.G.D GF: 12668 Prob: 2280 Other: 3404188040/118/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049644.7671964 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/118/1\"\n",
      "}\n",
      "1723049646.7563763 INFO: cpd05178 not found in model!\n",
      "1723049646.7585394 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049647.0119932 INFO: Objective with gapfill database:194.15097125384517; min objective:0.01\n",
      "1723049648.6131272 INFO: cpd05178 not found in model!\n",
      "1723049648.6154394 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049648.9181461 INFO: Objective with gapfill database:146.6974816430017; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049649.8650465 INFO: Expansion time:For.NO2:0.5401430040001287\n",
      "1723049649.8670104 INFO: Filtered count:0 out of 584\n",
      "1723049650.397242 INFO: Expansion time:For.O2:0.5290649640000993\n",
      "1723049650.3994775 INFO: Filtered count:0 out of 584\n",
      "1723049650.9452734 INFO: Expansion time:Fum:0.5445703140003388\n",
      "1723049650.947228 INFO: Filtered count:0 out of 584\n",
      "1723049651.4707239 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049651.7972672 INFO: cpd05178 not found in model!\n",
      "1723049651.799246 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049652.03962 INFO: Expansion time:Pyr:1.0912848819998544\n",
      "1723049652.041567 INFO: Filtered count:0 out of 584\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_D1.45.pg.G.D GF: 11425 Prob: 3772 Other: 3155\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049652.3297932 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049652.6283982 INFO: Expansion time:Pyr.DMSO:0.5855100400003721\n",
      "1723049652.6301172 INFO: Filtered count:0 out of 584\n",
      "1723049653.1810346 INFO: Expansion time:Pyr.NO:0.5495787580002798\n",
      "1723049653.182866 INFO: Filtered count:0 out of 584\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049653.513578 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049653.7348495 INFO: Expansion time:Pyr.NO2:0.550784195000233\n",
      "1723049653.7368374 INFO: Filtered count:0 out of 584\n",
      "1723049654.8383796 INFO: Expansion time:Pyr.NO3:1.1004292099996746\n",
      "1723049654.840383 INFO: Filtered count:0 out of 584\n",
      "1723049654.9796503 INFO: cpd05178 not found in model!\n",
      "1723049655.2035043 INFO: cpd05178 not found in model!\n",
      "1723049655.205662 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049655.4488368 INFO: Expansion time:Pyr.O2:0.6072843539996029\n",
      "1723049655.4510548 INFO: Filtered count:0 out of 584\n",
      "1723049656.2162232 INFO: Expansion time:Pyr.SO3:0.7637815790003515\n",
      "1723049656.2187681 INFO: Filtered count:0 out of 584\n",
      "1723049656.986227 INFO: Expansion time:Pyr.SO4:0.7658533220001118\n",
      "1723049656.9886167 INFO: Filtered count:0 out of 584\n",
      "1723049657.1672227 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049658.4469416 INFO: Expansion time:Pyr.TMAO:1.456944573000328\n",
      "1723049658.4493103 INFO: Filtered count:0 out of 584\n",
      "1723049659.2889488 INFO: Expansion time:empty:0.8381614879999688\n",
      "1723049659.291249 INFO: Filtered count:0 out of 584\n",
      "1723049660.2750864 INFO: cpd05178 not found in model!\n",
      "1723049660.277272 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049660.9341557 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-44\n",
      "[177 'SF2_B_H2O.25.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:42:19+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'aebc58a5adc654674d0d6c6bb7ee64c0'\n",
      " 6389113 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049664.0667279 INFO: Expansion time:Pyr:0.6071162890002597\n",
      "1723049664.0689135 INFO: Filtered count:0 out of 1101\n",
      "1723049664.1892784 INFO: metabolites 952\n",
      "1723049664.4007576 INFO: cpd05178 not found in model!\n",
      "1723049664.4029315 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049664.4987848 INFO: reactions 872\n",
      "1723049664.598752 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12228"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049664.6514072 INFO: Expansion time:Pyr.DMSO:0.5810298569995211\n",
      "1723049664.6534274 INFO: Filtered count:0 out of 1101\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_D2.11.pg.G.D GF:  Prob: 2962 Other: 3162\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049664.926125 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/160/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049665.0782704 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/160/1\"\n",
      "}\n",
      "1723049665.2305794 INFO: Expansion time:Pyr.NO:0.5759188140000333\n",
      "1723049665.2326586 INFO: Filtered count:0 out of 1101\n",
      "1723049666.3273807 INFO: cpd05178 not found in model!\n",
      "1723049666.3294656 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049666.44636 INFO: Expansion time:Pyr.NO2:1.2122769210000115\n",
      "1723049666.4490032 INFO: Filtered count:0 out of 1101\n",
      "1723049667.0831823 INFO: Expansion time:Pyr.NO3:0.632681596999646\n",
      "1723049667.085308 INFO: Filtered count:0 out of 1101\n",
      "1723049667.3037708 INFO: cpd05178 not found in model!\n",
      "1723049667.3061001 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049667.6545484 INFO: Expansion time:Pyr.O2:0.5679704069998479\n",
      "1723049667.6567562 INFO: Filtered count:0 out of 1101\n",
      "1723049668.3170578 INFO: Expansion time:Pyr.SO3:0.6586969079999108\n",
      "1723049668.3190374 INFO: Filtered count:0 out of 1101\n",
      "1723049669.5323868 INFO: Expansion time:Pyr.SO4:1.2122780290001174\n",
      "1723049669.5347445 INFO: Filtered count:0 out of 1101\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049670.1191018 INFO: Expansion time:Pyr.TMAO:0.582971806000387\n",
      "1723049670.1214383 INFO: Filtered count:0 out of 1101\n",
      "1723049670.6965504 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049670.757194 INFO: Expansion time:empty:0.6343186590002006\n",
      "1723049670.7593567 INFO: Filtered count:0 out of 1101\n",
      "1723049671.453247 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049671.5930142 INFO: cpd05178 not found in model!\n",
      "1723049671.5952454 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049673.5072298 INFO: cpd05178 not found in model!\n",
      "1723049673.7372274 INFO: cpd05178 not found in model!\n",
      "1723049673.739387 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049674.7269483 INFO: Expansion time:Dlac:4.807001704999493\n",
      "1723049674.7288568 INFO: Filtered count:0 out of 1353\n",
      "1723049674.8315182 INFO: cpd05178 not found in model!\n",
      "1723049674.8338084 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049675.4773748 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049675.9181075 INFO: Expansion time:Dlac.O2:1.188075936000132\n",
      "1723049675.9200978 INFO: Filtered count:0 out of 1353\n",
      "1723049676.8001878 INFO: Objective with gapfill database:151.08009168376492; min objective:0.01\n",
      "1723049677.2471535 INFO: Expansion time:Glc:1.3259085330000744\n",
      "1723049677.249115 INFO: Filtered count:0 out of 1353\n",
      "1723049678.5544043 INFO: Expansion time:Glc.DMSO:1.304098688000522\n",
      "1723049678.5564406 INFO: Filtered count:0 out of 1353\n",
      "1723049680.001718 INFO: Expansion time:Glc.O2:1.4441060139997717\n",
      "1723049680.0037549 INFO: Filtered count:0 out of 1353\n",
      "1723049681.4659953 INFO: Expansion time:Glc.TMAO:1.4610884920002718\n",
      "1723049681.4681878 INFO: Filtered count:0 out of 1353\n",
      "1723049682.561399 INFO: Expansion time:Pyr:1.0920206520004285\n",
      "1723049682.563343 INFO: Filtered count:0 out of 1353\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-45\n",
      "[181 'R2_A_D1.61.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:43:13+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'f2fa60f92f47ce0a52bc0f61feebf8ee'\n",
      " 7275533 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049683.624706 INFO: cpd05178 not found in model!\n",
      "1723049683.6268964 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049683.8405294 INFO: Expansion time:Pyr.DMSO:1.2760520310002903\n",
      "1723049683.8424218 INFO: Filtered count:0 out of 1353\n",
      "1723049683.9097352 INFO: metabolites 1040\n",
      "1723049683.95272 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049683.950899 INFO: cpd05178 not found in model!\n",
      "1723049684.3964484 INFO: reactions 954\n",
      "1723049684.5681458 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049684.9450903 INFO: cpd05178 not found in model!\n",
      "1723049684.947136 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049685.1307678 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_H2O.25.pg.G.D GF: 12443 Prob: 2634 Other: 3275\n",
      "188040/168/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049685.3246858 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/168/1\"\n",
      "}\n",
      "1723049685.65391 INFO: cpd05178 not found in model!\n",
      "1723049685.6558938 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049686.126899 INFO: Expansion time:Pyr.NO:2.283356946000822\n",
      "1723049686.1287246 INFO: Filtered count:0 out of 1353\n",
      "1723049686.9487376 INFO: Expansion time:Pyr.NO2:0.8188637140001447\n",
      "1723049686.950673 INFO: Filtered count:0 out of 1353\n",
      "1723049686.9578621 INFO: cpd05178 not found in model!\n",
      "1723049686.959402 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049687.6526623 INFO: Expansion time:Pyr.NO3:0.7008554490003007\n",
      "1723049687.6547735 INFO: Filtered count:0 out of 1353\n",
      "1723049687.9854128 INFO: cpd05178 not found in model!\n",
      "1723049688.2138324 INFO: cpd05178 not found in model!\n",
      "1723049688.2159677 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049688.8473713 INFO: Expansion time:Pyr.O2:1.1912541499996223\n",
      "1723049688.8493242 INFO: Filtered count:0 out of 1353\n",
      "1723049689.0031507 INFO: cpd05178 not found in model!\n",
      "1723049689.0050387 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049689.6060436 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049690.0320544 INFO: Expansion time:Pyr.SO3:1.181528631999754\n",
      "1723049690.034024 INFO: Filtered count:0 out of 1353\n",
      "1723049690.2727616 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[38 'R2A_C_D2.42.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:15:12+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '3357320d26ee27e710111d74c75ab4f2'\n",
      " 11286719 None]\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049691.4435315 INFO: Expansion time:Pyr.SO4:1.4083059569993566\n",
      "1723049691.4453728 INFO: Filtered count:0 out of 1353\n",
      "1723049691.6899247 INFO: metabolites 1127\n",
      "1723049692.3862855 INFO: reactions 1066\n",
      "1723049692.6067286 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049692.8685977 INFO: Expansion time:Pyr.TMAO:1.422000451999338\n",
      "1723049692.8706264 INFO: Filtered count:0 out of 1353\n",
      "1723049693.2491753 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/577/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049693.5110776 INFO: Expansion time:empty:0.6392316390001724\n",
      "1723049693.512965 INFO: Filtered count:0 out of 1353\n",
      "1723049693.745732 INFO: cpd05178 not found in model!\n",
      "1723049693.7479072 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049694.8288875 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[15 'R2_restored_C_black.57.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:10:42+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '38be805d04b7baea3301014097151648'\n",
      " 13425401 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049696.4791389 INFO: metabolites 1088\n",
      "1723049696.8830695 INFO: Objective with gapfill database:90.46692500620324; min objective:0.01\n",
      "1723049696.977437 INFO: reactions 1014\n",
      "1723049697.0947373 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049697.6239944 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/416/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049700.0596104 INFO: cpd05178 not found in model!\n",
      "1723049700.0620222 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049700.632206 INFO: cpd05178 not found in model!\n",
      "1723049700.6342373 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-46\n",
      "[185 'R2_restored_H2O.005.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:43:59+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'b75529704e85bd2ddff5e50a33dd6817'\n",
      " 16535670 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049703.2901266 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049703.2883346 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.61.pg.G.D GF: 12263 Prob: 2748 Other: 3341\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049704.0840871 INFO: metabolites 1141\n",
      "1723049704.7943444 INFO: reactions 1066\n",
      "1723049705.0470366 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049705.9947371 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/244/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049706.237402 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/244/1\"\n",
      "}\n",
      "1723049707.1179786 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049707.4177032 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049708.253685 INFO: cpd05178 not found in model!\n",
      "1723049708.2559588 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049709.5467556 INFO: cpd05178 not found in model!\n",
      "1723049709.5488575 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[35 'R2A_C_D1.4.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:14:42+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e62e261a1261ae3cb85073f9f5124910'\n",
      " 15656253 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049711.1143022 INFO: metabolites 1254\n",
      "1723049711.562583 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049711.7629592 INFO: reactions 1222\n",
      "1723049711.8162224 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049711.9249158 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049713.0408962 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049713.777184 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/545/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049716.6921718 INFO: Objective with gapfill database:137.55753426230274; min objective:0.01\n",
      "1723049716.7636993 INFO: cpd05178 not found in model!\n",
      "1723049716.7657409 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049717.591749 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049717.7184734 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049718.4811661 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049720.3113372 INFO: cpd05178 not found in model!\n",
      "1723049720.3134584 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_C_black.57.pg.G.D GF: 11910 Prob: 3344 Other: 3098\n",
      "ForkPoolWorker-47\n",
      "[189 'R1_C_D1.047.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:44:43+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'c7e4596f0b89edda874a24a93ac4cf12'\n",
      " 8849304 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049722.1275897 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049722.4952087 INFO: metabolites 881\n",
      "1723049723.0027323 INFO: reactions 814\n",
      "1723049723.14415 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049723.6823223 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/243/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049723.8805475 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/243/1\"\n",
      "}\n",
      "1723049724.5072172 INFO: cpd05178 not found in model!\n",
      "1723049724.50949 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.42.pg.G.D GF: 11295 Prob: 4128 Other: 2931\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049728.0010893 INFO: cpd05178 not found in model!\n",
      "1723049728.0033972 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049729.5099857 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049729.5684545 INFO: cpd05178 not found in model!\n",
      "1723049729.570638 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_H2O.005.pg.G.D GF: 11221 Prob: 4192 Other: 2939\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049731.5345595 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049732.1733398 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049732.3700693 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049732.802548 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Getting tests from attributes"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049734.2742522 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049736.2100008 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049736.3126633 INFO: cpd05178 not found in model!\n",
      "1723049737.25196 INFO: cpd05178 not found in model!\n",
      "1723049737.254112 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049737.652068 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049738.4149983 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-48\n",
      "[193 'SF2_C_D2.24.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:45:43+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '969783f8f9bea7f711e5d78c23b8c664'\n",
      " 16434977 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049739.830365 INFO: metabolites 1223\n",
      "1723049740.4905953 INFO: reactions 1195\n",
      "1723049740.77916 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049741.8728673 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/284/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049742.14884 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/284/1\"\n",
      "}\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049743.274361 INFO: Expansion time:Ac.NO3:0.6562807990003421\n",
      "1723049743.276402 INFO: Filtered count:0 out of 1515\n",
      "1723049743.9106324 INFO: Expansion time:Ac.O2:0.6330944569999701\n",
      "1723049743.9126892 INFO: Filtered count:0 out of 1515\n",
      "1723049743.9925134 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049743.990662 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_D1.047.pg.G.D GF: 12198 Prob: 2830 Other: 3322\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049744.7969594 INFO: Expansion time:Etho.O2:0.8831343239999114\n",
      "1723049744.7995021 INFO: Filtered count:0 out of 1515\n",
      "1723049745.5729096 INFO: Expansion time:Fum:0.7718768079994334\n",
      "1723049745.5750332 INFO: Filtered count:0 out of 1515\n",
      "1723049746.2121348 INFO: Expansion time:Fum.O2:0.6359662380000373\n",
      "1723049746.2143347 INFO: Filtered count:0 out of 1515\n",
      "1723049746.3932157 INFO: Expansion time:Etho.O2:3.778288608999901\n",
      "1723049746.3953507 INFO: Filtered count:0 out of 1441\n",
      "1723049746.5174975 INFO: cpd05178 not found in model!\n",
      "1723049746.5195382 CRITICAL: Metabolite cpd05178 not found in model\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D1.4.pg.G.D GF: 10716 Prob: 4848 Other: 2790\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049747.577368 INFO: Expansion time:Pyr:1.1808542939998006\n",
      "1723049747.5794814 INFO: Filtered count:0 out of 1441\n",
      "1723049747.991386 INFO: Expansion time:Glc:1.775938689999748\n",
      "1723049747.9933624 INFO: Filtered count:0 out of 1515\n",
      "1723049748.1425111 INFO: Expansion time:Ac.O2:1.5167126369997277\n",
      "1723049748.1447077 INFO: Filtered count:0 out of 1535\n",
      "1723049748.6398833 INFO: Expansion time:Glc.DMSO:0.6451236289994995\n",
      "1723049748.641939 INFO: Filtered count:0 out of 1515\n",
      "1723049748.7424269 INFO: Expansion time:Pyr.DMSO:1.1616119459995389\n",
      "1723049748.7443767 INFO: Filtered count:0 out of 1441\n",
      "1723049748.939565 INFO: Expansion time:Etho.O2:0.7936932029997479\n",
      "1723049748.941785 INFO: Filtered count:0 out of 1535\n",
      "1723049749.2900326 INFO: Expansion time:Glc.O2:0.6469155990007494\n",
      "1723049749.2921338 INFO: Filtered count:0 out of 1515\n",
      "1723049749.79726 INFO: Expansion time:Fum:0.8542814840002393\n",
      "1723049749.7994974 INFO: Filtered count:0 out of 1535\n",
      "1723049749.9439745 INFO: Expansion time:Glc.TMAO:0.6507073669999954\n",
      "1723049749.9458148 INFO: Filtered count:0 out of 1515\n",
      "1723049750.578052 INFO: Expansion time:Pyr:0.6309411860002001\n",
      "1723049750.5801487 INFO: Filtered count:0 out of 1515\n",
      "1723049750.662444 INFO: Expansion time:Fum.O2:0.861814100000629\n",
      "1723049750.6645668 INFO: Filtered count:0 out of 1535\n",
      "1723049751.1142516 INFO: Expansion time:Pyr.NO:2.3687649280000187\n",
      "1723049751.1164682 INFO: Filtered count:0 out of 1441\n",
      "1723049751.2314832 INFO: Expansion time:Pyr.DMSO:0.6501458630000343\n",
      "1723049751.2335975 INFO: Filtered count:0 out of 1515\n",
      "1723049751.3671281 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049751.8824916 INFO: Expansion time:Pyr.NO:0.6476650380000137\n",
      "1723049751.8847175 INFO: Filtered count:0 out of 1515\n",
      "1723049752.0629728 INFO: Expansion time:Pyr.NO2:0.945321944999705\n",
      "1723049752.0651107 INFO: Filtered count:0 out of 1441\n",
      "1723049752.0799577 INFO: Expansion time:Pyr:1.4142438960006984\n",
      "1723049752.0821555 INFO: Filtered count:0 out of 1535\n",
      "1723049752.4452772 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049752.5273705 INFO: Expansion time:Pyr.NO2:0.6414585339998666\n",
      "1723049752.5294147 INFO: Filtered count:0 out of 1515\n",
      "1723049753.09674 INFO: Expansion time:Pyr.NO3:1.030434573999628\n",
      "1723049753.0990286 INFO: Filtered count:0 out of 1441\n",
      "1723049753.3203511 INFO: Expansion time:Pyr.DMSO:1.2369882490002055\n",
      "1723049753.3226812 INFO: Filtered count:0 out of 1535\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049754.0844145 INFO: Expansion time:Pyr.NO:0.7603462619999846\n",
      "1723049754.0866785 INFO: Filtered count:0 out of 1535\n",
      "1723049754.377245 INFO: Expansion time:Pyr.NO3:1.8466423250001753\n",
      "1723049754.3794668 INFO: Filtered count:0 out of 1515\n",
      "1723049754.7521517 INFO: Expansion time:Pyr.O2:1.6518822950001777\n",
      "1723049754.754096 INFO: Filtered count:0 out of 1441\n",
      "1723049754.8036194 INFO: Filtered count:0 out of 1535\n",
      "1723049754.8720498 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049754.8013206 INFO: Expansion time:Pyr.NO2:0.7133671849996972\n",
      "1723049755.0285592 INFO: Expansion time:Pyr.O2:0.6479859239998405\n",
      "1723049755.0307014 INFO: Filtered count:0 out of 1515\n",
      "1723049755.4961107 INFO: Expansion time:Pyr.NO3:0.6913179819994184\n",
      "1723049755.4984448 INFO: Filtered count:0 out of 1535\n",
      "1723049755.6760285 INFO: Expansion time:Pyr.SO3:0.6441130529992733\n",
      "1723049755.6780593 INFO: Filtered count:0 out of 1515\n",
      "1723049755.9395976 INFO: Expansion time:Pyr.SO3:1.1843298329995378\n",
      "1723049755.941726 INFO: Filtered count:0 out of 1441\n",
      "1723049756.3268795 INFO: Expansion time:Pyr.SO4:0.6476743849998456\n",
      "1723049756.3290575 INFO: Filtered count:0 out of 1515\n",
      "1723049756.8894207 INFO: Expansion time:Pyr.SO4:0.9464469690001351\n",
      "1723049756.891554 INFO: Filtered count:0 out of 1441\n",
      "1723049756.9750535 INFO: Filtered count:0 out of 1515\n",
      "1723049757.0694523 INFO: Filtered count:0 out of 1535\n",
      "1723049756.9729793 INFO: Expansion time:Pyr.TMAO:0.6427474100000836\n",
      "1723049757.0672452 INFO: Expansion time:Pyr.O2:1.567648721000296\n",
      "1723049757.6264646 INFO: Expansion time:empty:0.6501693239997621\n",
      "1723049757.6286132 INFO: Filtered count:0 out of 1515\n",
      "1723049757.696477 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[130 'R2_restored_C_black.51.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:33:19+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'd5d9a1bc300dcbf648e597bfbbe94f10'\n",
      " 15286331 None]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049757.8483956 INFO: Expansion time:Pyr.SO3:0.7776466400000572\n",
      "1723049757.8507452 INFO: Filtered count:0 out of 1535\n",
      "1723049758.1154196 INFO: Expansion time:Pyr.TMAO:1.222739821999312\n",
      "1723049758.1171966 INFO: Filtered count:0 out of 1441\n",
      "1723049758.2220457 INFO: metabolites 1187\n",
      "1723049758.5290983 INFO: Expansion time:Pyr.SO4:0.6770838890006416\n",
      "1723049758.5315108 INFO: Filtered count:0 out of 1535\n",
      "1723049758.669061 INFO: Objective with gapfill database:134.24793527711688; min objective:0.01\n",
      "1723049758.749077 INFO: Expansion time:empty:0.6307455969999864\n",
      "1723049758.7512019 INFO: Filtered count:0 out of 1441\n",
      "1723049758.8442173 INFO: reactions 1200\n",
      "1723049759.0458605 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049759.198633 INFO: Expansion time:Pyr.TMAO:0.665884277999794\n",
      "1723049759.200962 INFO: Filtered count:0 out of 1535\n",
      "1723049759.8365238 INFO: Expansion time:empty:0.6343547380001837\n",
      "1723049759.8388958 INFO: Filtered count:0 out of 1535\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-49\n",
      "[197 'R2A_C_D2.18.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:46:28+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '5a315514681f51d5500c09a6dba4a2ad'\n",
      " 14232731 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049760.619283 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/534/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049760.9083652 INFO: metabolites 1090\n",
      "1723049761.7347343 INFO: reactions 997\n",
      "1723049761.8978324 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049762.845005 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/334/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049763.0464938 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/334/1\"\n",
      "}\n",
      "1723049763.0774696 INFO: cpd05178 not found in model!\n",
      "1723049763.2986252 INFO: cpd05178 not found in model!\n",
      "1723049763.3004398 INFO: Media compound: cpd05178 not found in model.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049764.9259622 INFO: cpd05178 not found in model!\n",
      "1723049764.928354 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049765.7630112 INFO: cpd05178 not found in model!\n",
      "1723049765.764935 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2.24.pg.G.D GF: 11239 Prob: 4144 Other: 2971\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049766.4005508 INFO: cpd05178 not found in model!\n",
      "1723049766.4030855 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049766.4837162 INFO: Expansion time:Ac.NO2:2.8227150650000112\n",
      "1723049766.4857347 INFO: Filtered count:0 out of 1730\n",
      "1723049768.0190444 INFO: Expansion time:Ac.O2:1.5321227009999348\n",
      "1723049768.0211966 INFO: Filtered count:0 out of 1730\n",
      "1723049768.2824728 INFO: cpd05178 not found in model!\n",
      "1723049768.2846236 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049768.8376532 INFO: Expansion time:Etho.O2:0.8153251260000616\n",
      "1723049768.8397937 INFO: Filtered count:0 out of 1730\n",
      "1723049769.673837 INFO: Expansion time:For.NO2:0.8329416970000239\n",
      "1723049769.6759536 INFO: Filtered count:0 out of 1730\n",
      "1723049769.925244 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049770.3798754 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049770.5616815 INFO: Expansion time:For.O2:0.8846211339996444\n",
      "1723049770.5637715 INFO: Filtered count:0 out of 1730\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049773.4395177 INFO: Expansion time:Fum:2.8745365810000294\n",
      "1723049773.441745 INFO: Filtered count:0 out of 1730\n",
      "1723049774.1480627 INFO: cpd05178 not found in model!\n",
      "1723049774.249262 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049774.3836324 INFO: cpd05178 not found in model!\n",
      "1723049774.3858747 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049774.55195 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049774.586559 INFO: Filtered count:0 out of 1730\n",
      "1723049774.58449 INFO: Expansion time:Fum.O2:1.1415070969997032\n",
      "1723049775.765625 INFO: Expansion time:Glc:1.1779696370003876\n",
      "1723049775.7677393 INFO: Filtered count:0 out of 1730\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-50\n",
      "[201 'R1_C_D2.30.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:47:19+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '459af062fdcad71a3b76502173e25f2d'\n",
      " 10046282 None]\n",
      "Failed:>rxn00062_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049780.04547 INFO: Expansion time:Glc.DMSO:4.276573391000056\n",
      "1723049780.0476127 INFO: Filtered count:1 out of 1730\n",
      "1723049780.4001694 INFO: metabolites 1130\n",
      "1723049780.708508 INFO: Expansion time:Glc.O2:0.6597130949994607\n",
      "1723049780.710592 INFO: Filtered count:1 out of 1730\n",
      "1723049780.9907134 INFO: reactions 1086\n",
      "1723049781.1781368 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049781.376293 INFO: Expansion time:Glc.TMAO:0.6645725629996377\n",
      "1723049781.3784769 INFO: Filtered count:1 out of 1730\n",
      "1723049781.4240477 INFO: cpd05178 not found in model!\n",
      "1723049781.4263325 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049781.534712 INFO: cpd05178 not found in model!\n",
      "1723049781.6128705 INFO: Default biomass: [bio1]\n",
      "1723049781.7699103 INFO: cpd05178 not found in model!\n",
      "1723049781.7720647 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049782.0492082 INFO: Filtered count:1 out of 1730\n",
      "1723049782.0472586 INFO: Expansion time:Pyr:0.6676306270001078\n",
      "1723049781.9960356 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/322/1\"\n",
      "}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/322/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049782.70524 INFO: Expansion time:Pyr.DMSO:0.6547882620006931\n",
      "1723049782.7073572 INFO: Filtered count:1 out of 1730\n",
      "1723049783.235328 INFO: Expansion time:Ac.O2:0.6597932169997875\n",
      "1723049783.237365 INFO: Filtered count:0 out of 1709\n",
      "1723049783.368757 INFO: Expansion time:Pyr.NO:0.6602740520002044\n",
      "1723049783.370863 INFO: Filtered count:1 out of 1730\n",
      "1723049783.9871933 INFO: Expansion time:Etho.O2:0.7484066690003601\n",
      "1723049783.989291 INFO: Filtered count:0 out of 1709\n",
      "1723049784.0310876 INFO: Expansion time:Pyr.NO2:0.6590614229999119\n",
      "1723049784.0330749 INFO: Filtered count:1 out of 1730\n",
      "1723049784.2576656 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049784.2556825 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_C_black.51.pg.G.D GF: 11434 Prob: 3802 Other: 3116\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049784.6755118 INFO: Expansion time:Glc:0.6850288160003402\n",
      "1723049784.6773372 INFO: Filtered count:0 out of 1709\n",
      "1723049784.6917155 INFO: Expansion time:Pyr.NO3:0.6575490239993087\n",
      "1723049784.6938004 INFO: Filtered count:1 out of 1730\n",
      "1723049785.3589664 INFO: Expansion time:Glc.DMSO:0.6802407459999813\n",
      "1723049785.360945 INFO: Filtered count:0 out of 1709\n",
      "1723049785.5297143 INFO: cpd05178 not found in model!\n",
      "1723049785.5319088 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049786.7150114 INFO: Expansion time:Glc.O2:1.3528233400002136\n",
      "1723049786.716962 INFO: Filtered count:0 out of 1709\n",
      "1723049787.401963 INFO: Expansion time:Glc.TMAO:0.6838811440002246\n",
      "1723049787.4040105 INFO: Filtered count:0 out of 1709\n",
      "1723049787.6072843 INFO: Expansion time:Pyr.O2:2.912186763999671\n",
      "1723049787.6095545 INFO: Filtered count:1 out of 1730\n",
      "1723049787.9271514 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049788.071422 INFO: Expansion time:Glyc.O2:0.6662009129995567\n",
      "1723049788.0733993 INFO: Filtered count:0 out of 1709\n",
      "1723049788.3636136 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049788.5447462 INFO: Expansion time:Pyr.SO3:0.9336616800001138\n",
      "1723049788.5471368 INFO: Filtered count:1 out of 1730\n",
      "1723049789.0388298 INFO: Expansion time:Pyr:0.9641552059993046\n",
      "1723049789.0411587 INFO: Filtered count:0 out of 1709\n",
      "1723049789.256873 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049789.4932282 INFO: Expansion time:Pyr.SO4:0.9445127660001162\n",
      "1723049789.495716 INFO: Filtered count:1 out of 1730\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049789.865216 INFO: Expansion time:Pyr.DMSO:0.8223762100005843\n",
      "1723049789.867282 INFO: Filtered count:0 out of 1709\n",
      "1723049789.9159958 INFO: cpd05178 not found in model!\n",
      "1723049790.2596643 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049790.2605426 INFO: cpd05178 not found in model!\n",
      "1723049790.2629104 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049790.5826557 INFO: Expansion time:Pyr.TMAO:1.0853612370001429\n",
      "1723049790.5850496 INFO: Filtered count:1 out of 1730\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049791.244908 INFO: Expansion time:Pyr.NO:1.3764711799994984\n",
      "1723049791.246763 INFO: Filtered count:0 out of 1709\n",
      "1723049791.669438 INFO: Expansion time:empty:1.0828337470002225\n",
      "1723049791.671851 INFO: Filtered count:1 out of 1730\n",
      "1723049791.917639 INFO: Expansion time:Pyr.NO2:0.6697102809994249\n",
      "1723049791.9197247 INFO: Filtered count:0 out of 1709\n",
      "1723049792.5996606 INFO: Expansion time:Pyr.NO3:0.6787349630003519\n",
      "1723049792.6016488 INFO: Filtered count:0 out of 1709\n",
      "1723049793.0712738 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049793.092596 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049793.3394926 INFO: Expansion time:Pyr.O2:0.7365348969997285\n",
      "1723049793.3415844 INFO: Filtered count:0 out of 1709\n",
      "1723049794.0407612 INFO: Expansion time:Pyr.SO3:0.6977630680003131\n",
      "1723049794.042777 INFO: Filtered count:0 out of 1709\n",
      "1723049794.5316072 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049795.2589543 INFO: cpd05178 not found in model!\n",
      "1723049795.2609785 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049795.3798113 INFO: Expansion time:Pyr.SO4:1.3358997029999955\n",
      "1723049795.3816485 INFO: Filtered count:0 out of 1709\n",
      "1723049796.051694 INFO: Expansion time:Pyr.TMAO:0.6685983770003077\n",
      "1723049796.0536392 INFO: Filtered count:0 out of 1709\n",
      "1723049796.7280297 INFO: Expansion time:empty:0.6732002700000521\n",
      "1723049796.7300403 INFO: Filtered count:0 out of 1709\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[102 'R2_A_D2.19.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:27:51+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '16380a3c307878e28277fa5e9277d6b2'\n",
      " 10857622 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049797.2697134 INFO: metabolites 793\n",
      "1723049797.673239 INFO: reactions 720\n",
      "1723049797.7833226 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049798.2150717 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/180/1ForkPoolWorker-51\n",
      "[205 'R1_B_D1.59.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:48:14+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'fd41da503f72a0c431368848de0403f7'\n",
      " 10157256 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049801.3922973 INFO: metabolites 1153\n",
      "1723049801.9009638 INFO: Expansion time:Etho.O2:0.6374775900003442\n",
      "1723049801.9030738 INFO: Filtered count:0 out of 1732\n",
      "1723049802.0667987 INFO: reactions 1078\n",
      "1723049802.2234209 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049802.473054 INFO: cpd05178 not found in model!\n",
      "1723049802.4756083 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049802.5355954 INFO: Filtered count:0 out of 1732\n",
      "1723049802.5335994 INFO: Expansion time:Fum:0.6292848820003201\n",
      "1723049802.9185846 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/350/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049803.1184227 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/350/1\"\n",
      "}\n",
      "1723049803.1690059 INFO: Expansion time:Fum.O2:0.6323091639997074\n",
      "1723049803.171022 INFO: Filtered count:0 out of 1732\n",
      "1723049803.76671 INFO: cpd05178 not found in model!\n",
      "1723049803.76909 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049803.8115852 INFO: Expansion time:Glc:0.639434878999964\n",
      "1723049803.8136377 INFO: Filtered count:0 out of 1732\n",
      "1723049803.8170831 INFO: cpd05178 not found in model!\n",
      "1723049803.8187778 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049803.9944243 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.18.pg.G.D GF: 11196 Prob: 4142 Other: 3014\n",
      "R1_C_D2.30.pg.G.D GF: 11474 Prob: 3814 Other: 3064\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049804.233205 INFO: cpd05178 not found in model!\n",
      "1723049804.235106 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049804.4525273 INFO: Expansion time:Glc.DMSO:0.6377250989999084\n",
      "1723049804.4545107 INFO: Filtered count:0 out of 1732\n",
      "1723049805.0868368 INFO: Expansion time:Glc.O2:0.6311146799998824\n",
      "1723049805.088899 INFO: Filtered count:0 out of 1732\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[19 'R2A_B_H2O.69.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:11:23+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '451ac06b3eb28d61bb36d02ec040cff0'\n",
      " 4395729 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049806.7362869 INFO: metabolites 871\n",
      "1723049806.7424269 INFO: Expansion time:Glc.TMAO:1.6523964799998794\n",
      "1723049806.7442064 INFO: Filtered count:0 out of 1732\n",
      "1723049807.2018435 INFO: reactions 834\n",
      "1723049807.3712292 INFO: Expansion time:Pyr:0.6258464579996144\n",
      "1723049807.3732188 INFO: Filtered count:0 out of 1732\n",
      "1723049807.4876564 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049808.0025017 INFO: Expansion time:Pyr.DMSO:0.6281346729992947\n",
      "1723049808.0045168 INFO: Filtered count:0 out of 1732\n",
      "1723049808.0071855 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/455/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049808.6349516 INFO: Expansion time:Pyr.NO:0.629313987000387\n",
      "1723049808.6371398 INFO: Filtered count:0 out of 1732\n",
      "1723049808.8246922 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049809.2664182 INFO: Expansion time:Pyr.NO2:0.6280734520005353\n",
      "1723049809.2685022 INFO: Filtered count:0 out of 1732\n",
      "1723049809.8975496 INFO: Expansion time:Pyr.NO3:0.6277449450008135\n",
      "1723049809.899776 INFO: Filtered count:0 out of 1732\n",
      "1723049810.4973145 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049810.5328565 INFO: Filtered count:0 out of 1732\n",
      "1723049810.530852 INFO: Expansion time:Pyr.O2:0.6298890709995248\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049811.5898325 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049811.7359338 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049812.107568 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049812.2138467 INFO: Expansion time:Pyr.SO3:1.6798822910004674\n",
      "1723049812.2159605 INFO: Filtered count:0 out of 1732\n",
      "1723049812.7968683 INFO: cpd05178 not found in model!\n",
      "1723049812.7990954 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049812.8475084 INFO: Expansion time:Pyr.SO4:0.630434363999484\n",
      "1723049812.8496041 INFO: Filtered count:0 out of 1732\n",
      "1723049813.3230512 INFO: cpd05178 not found in model!\n",
      "1723049813.3251982 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049813.478958 INFO: Expansion time:Pyr.TMAO:0.628169326000716\n",
      "1723049813.4810293 INFO: Filtered count:0 out of 1732\n",
      "1723049813.8647223 INFO: cpd05178 not found in model!\n",
      "1723049813.866918 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049814.112446 INFO: Filtered count:0 out of 1732\n",
      "1723049814.1104531 INFO: Expansion time:empty:0.628321359000438\n",
      "1723049814.35467 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[118 'SF2_B_H2O.11.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:30:50+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '1aa432d55801097800f129e0a72460cf'\n",
      " 8696277 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049814.8640468 INFO: cpd05178 not found in model!\n",
      "1723049814.8663197 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049814.888121 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049815.2396588 INFO: metabolites 1050\n",
      "1723049815.3562815 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049815.4114223 INFO: Objective with gapfill database:227.72622442851036; min objective:0.01\n",
      "1723049815.8531294 INFO: reactions 979\n",
      "1723049816.023222 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049816.1236053 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049816.6199744 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/470/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049817.77938 INFO: cpd05178 not found in model!\n",
      "1723049817.7815104 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049819.816429 INFO: cpd05178 not found in model!\n",
      "1723049819.8191454 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049820.7627466 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049821.4915154 INFO: Expansion time:Etho.O2:0.6965597609996621\n",
      "1723049821.493564 INFO: Filtered count:0 out of 1561\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[209 'R2A_A_D1.17.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:48:54+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '15220bffeccf995e81486fd0b8d5dbbe'\n",
      " 10340910 None]ForkPoolWorker-52\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049822.6564538 INFO: metabolites 1112\n",
      "1723049822.7969816 INFO: Expansion time:Fum:1.3020675459993072\n",
      "1723049822.7990952 INFO: Filtered count:0 out of 1561\n",
      "1723049822.8793314 INFO: cpd05178 not found in model!\n",
      "1723049822.8813167 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049823.0705364 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049823.068356 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D2.19.pg.G.D GF: 12544 Prob: 2494 Other: 3322\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049823.4648573 INFO: reactions 1083\n",
      "1723049823.4675255 INFO: Expansion time:Fum.O2:0.6671868860003087\n",
      "1723049823.4690511 INFO: Filtered count:0 out of 1561\n",
      "1723049823.7617536 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049824.1345394 INFO: Expansion time:Glc:0.6643646130005436\n",
      "1723049824.1364617 INFO: Filtered count:0 out of 1561\n",
      "1723049824.7201264 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049824.785265 INFO: Expansion time:Glc.DMSO:0.6474532009997347\n",
      "1723049824.7872057 INFO: Filtered count:0 out of 1561\n",
      "1723049824.7933714 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/411/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049825.083831 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/411/1\"\n",
      "}\n",
      "1723049825.4387703 INFO: Expansion time:Glc.O2:0.650460930999543\n",
      "1723049825.4407094 INFO: Filtered count:0 out of 1561\n",
      "1723049826.626458 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049826.73 INFO: Expansion time:Glc.TMAO:1.2881303159992967\n",
      "1723049826.732105 INFO: Filtered count:0 out of 1561\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049827.3793662 INFO: Expansion time:Glyc.O2:0.646038548999968\n",
      "1723049827.3813527 INFO: Filtered count:0 out of 1561\n",
      "1723049827.4318001 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049827.4261625 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[58 'R2A_A_D1.46.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:19:15+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ef95eddf2be8cbd6f13b05e04fbc4548'\n",
      " 15156526 None]R2A_B_H2O.69.pg.G.D GF: 12304 Prob: 2818 Other: 3232\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049828.030896 INFO: Expansion time:Pyr:0.6483178070002396\n",
      "1723049828.032771 INFO: Filtered count:0 out of 1561\n",
      "1723049828.2076883 INFO: metabolites 1124\n",
      "1723049828.6758919 INFO: Expansion time:Pyr.DMSO:0.6419639770001595\n",
      "1723049828.6777563 INFO: Filtered count:0 out of 1561\n",
      "1723049828.6822345 INFO: cpd05178 not found in model!\n",
      "1723049828.6841974 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049828.6983202 INFO: reactions 1097\n",
      "1723049828.9499154 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049829.4074435 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/618/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049829.958644 INFO: Expansion time:Pyr.NO:1.2797361920002004\n",
      "1723049829.9606023 INFO: Filtered count:0 out of 1561\n",
      "1723049830.280045 INFO: cpd05178 not found in model!\n",
      "1723049830.2820492 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049830.428169 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049830.544743 INFO: cpd05178 not found in model!\n",
      "1723049830.5467863 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D1.59.pg.G.D GF: 11626 Prob: 3478 Other: 3248\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049830.701144 INFO: Expansion time:Pyr.NO2:0.7393314070004635\n",
      "1723049830.7033641 INFO: Filtered count:0 out of 1561\n",
      "1723049831.6244724 INFO: Expansion time:Pyr.NO3:0.9195673939993867\n",
      "1723049831.62671 INFO: Filtered count:0 out of 1561\n",
      "1723049832.555628 INFO: Expansion time:Pyr.O2:0.9273584279999341\n",
      "1723049832.5580575 INFO: Filtered count:0 out of 1561\n",
      "1723049833.3176968 INFO: Expansion time:Pyr.SO3:0.7581290380003338\n",
      "1723049833.3196938 INFO: Filtered count:0 out of 1561\n",
      "1723049834.0568686 INFO: cpd05178 not found in model!\n",
      "1723049834.0590127 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049834.6085207 INFO: Expansion time:Pyr.SO4:1.2876533010003186\n",
      "1723049834.610593 INFO: Filtered count:0 out of 1561\n",
      "1723049835.7294118 INFO: Expansion time:Pyr.TMAO:1.1174298949999866\n",
      "1723049835.731973 INFO: Filtered count:0 out of 1561\n",
      "1723049836.5879843 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049836.709267 INFO: Expansion time:empty:0.9754832429998714\n",
      "1723049836.7115045 INFO: Filtered count:0 out of 1561\n",
      "1723049838.9662707 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049839.4368083 INFO: Objective with gapfill database:144.69744828372166; min objective:0.01\n",
      "1723049840.0443528 INFO: cpd05178 not found in model!\n",
      "1723049840.0470567 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_H2O.11.pg.G.D GF: 11711 Prob: 3616 Other: 3025\n",
      "ForkPoolWorker-53\n",
      "[213 'SF2_B_D2.55.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:49:44+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '0b1437c128e79606685194b4f482b6ca'\n",
      " 11956080 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049841.7681625 INFO: metabolites 1108\n",
      "1723049842.4517164 INFO: reactions 1091\n",
      "1723049842.649047 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049843.6500878 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/403/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049843.8573601 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/403/1\"\n",
      "}\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049846.3538663 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049847.762919 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049848.3909976 INFO: Expansion time:Ac.O2:3.0124677500007238\n",
      "1723049848.3931956 INFO: Filtered count:0 out of 1184\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049848.9890492 INFO: Filtered count:0 out of 1184\n",
      "1723049848.9870217 INFO: Expansion time:Etho.O2:0.592577371000516\n",
      "1723049849.574222 INFO: Expansion time:Pyr:0.5838766730003044\n",
      "1723049849.5764227 INFO: Filtered count:0 out of 1184\n",
      "1723049850.1605542 INFO: Expansion time:Pyr.DMSO:0.5829709999998158\n",
      "1723049850.1627595 INFO: Filtered count:0 out of 1184\n",
      "1723049850.364945 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049850.7501104 INFO: Expansion time:Pyr.NO:0.5861963079996713\n",
      "1723049850.7523441 INFO: Filtered count:0 out of 1184\n",
      "1723049851.0311234 INFO: Expansion time:Etho.O2:1.979876301999866\n",
      "1723049851.0332334 INFO: Filtered count:0 out of 1530\n",
      "1723049851.1623216 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049851.3390121 INFO: Expansion time:Pyr.NO2:0.5854309370006376\n",
      "1723049851.341091 INFO: Filtered count:0 out of 1184\n",
      "1723049851.5089571 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049851.9252903 INFO: Expansion time:Pyr.NO3:0.5830155220000961\n",
      "1723049851.9275634 INFO: Filtered count:0 out of 1184\n",
      "1723049852.130837 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049852.1960464 INFO: Expansion time:Fum:1.1615056939999704\n",
      "1723049852.197926 INFO: Filtered count:0 out of 1530\n",
      "1723049852.513647 INFO: Expansion time:Pyr.O2:0.5848114430000351\n",
      "1723049852.515768 INFO: Filtered count:0 out of 1184\n",
      "1723049852.6155374 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049853.1010883 INFO: Expansion time:Pyr.SO3:0.5841986839996025\n",
      "1723049853.1033375 INFO: Filtered count:0 out of 1184\n",
      "1723049853.1162226 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049853.2041728 INFO: Expansion time:Fum.O2:1.0049288809996142\n",
      "1723049853.206093 INFO: Filtered count:0 out of 1530\n",
      "1723049853.690187 INFO: Expansion time:Pyr.SO4:0.5856213170000046\n",
      "1723049853.6925693 INFO: Filtered count:0 out of 1184\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049854.2454264 INFO: Expansion time:Glc:1.038077162999798\n",
      "1723049854.2473717 INFO: Filtered count:0 out of 1530\n",
      "1723049855.7610042 INFO: Expansion time:Glc.DMSO:1.512465341999814\n",
      "1723049855.7629724 INFO: Filtered count:0 out of 1530\n",
      "1723049855.9334848 INFO: cpd05178 not found in model!\n",
      "1723049855.9355676 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049856.0776453 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049856.3678646 INFO: Expansion time:Pyr.TMAO:2.6738928859995212\n",
      "1723049856.3704228 INFO: Filtered count:0 out of 1184\n",
      "1723049856.755174 INFO: cpd05178 not found in model!\n",
      "1723049856.7573917 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049856.9124663 INFO: Expansion time:Glc.O2:1.1482696459997896\n",
      "1723049856.9144952 INFO: Filtered count:0 out of 1530\n",
      "1723049857.2227423 INFO: Expansion time:empty:0.8508119580001221\n",
      "1723049857.2253544 INFO: Filtered count:0 out of 1184\n",
      "1723049857.6319304 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049857.9995835 INFO: cpd05178 not found in model!\n",
      "1723049858.0019524 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049858.0092547 INFO: Expansion time:Glc.TMAO:1.0934471930004293\n",
      "1723049858.0111115 INFO: Filtered count:0 out of 1530\n",
      "1723049858.7971373 INFO: Expansion time:Pyr:0.7848966929996095\n",
      "1723049858.7991638 INFO: Filtered count:0 out of 1530\n",
      "1723049859.6426668 INFO: Expansion time:Pyr.DMSO:0.8423376830005509\n",
      "1723049859.6447172 INFO: Filtered count:0 out of 1530\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[217 'R2_B_D2.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:50:31+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '4b1c83ab8a9fd357a2090b23c710fb80'\n",
      " 14721420 None]ForkPoolWorker-54\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049860.7217917 INFO: metabolites 1256\n",
      "1723049860.9914606 INFO: cpd05178 not found in model!\n",
      "1723049860.9935582 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049861.1387212 INFO: Expansion time:Pyr.NO:1.4927813540007264\n",
      "1723049861.140634 INFO: Filtered count:0 out of 1530\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1.17.pg.G.D GF: 10960 Prob: 4588 Other: 2806\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049861.6463466 INFO: reactions 1235\n",
      "1723049861.8949888 INFO: Expansion time:Pyr.NO2:0.7532158860003619\n",
      "1723049861.8970275 INFO: Filtered count:0 out of 1530\n",
      "1723049861.9115684 INFO: cpd05178 not found in model!\n",
      "1723049861.9135861 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049862.2106745 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049862.9606876 INFO: Expansion time:Etho.O2:2.801048646999334\n",
      "1723049862.9629703 INFO: Filtered count:0 out of 1385\n",
      "1723049863.0191507 INFO: Expansion time:Pyr.NO3:1.1206524749995879\n",
      "1723049863.0214946 INFO: Filtered count:0 out of 1530\n",
      "1723049863.0336535 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/447/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049863.2612536 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/447/1\"\n",
      "}\n",
      "1723049864.2236123 INFO: cpd05178 not found in model!\n",
      "1723049864.2259848 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049864.3113694 INFO: Expansion time:Pyr.O2:1.2884603830007109\n",
      "1723049864.3136346 INFO: Filtered count:0 out of 1530\n",
      "1723049864.9906914 INFO: Expansion time:Fum:2.0264786090001508\n",
      "1723049864.9927561 INFO: Filtered count:0 out of 1385\n",
      "1723049865.4435275 INFO: cpd05178 not found in model!\n",
      "1723049865.4457495 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049865.6643176 INFO: Expansion time:Pyr.SO3:1.3490944980003405\n",
      "1723049865.6666422 INFO: Filtered count:0 out of 1530\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2958R2A_A_D1.46.pg.G.D GF: 11216 Prob: 4178 Other: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049866.2875352 INFO: cpd05178 not found in model!\n",
      "1723049866.2894416 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049866.9657702 INFO: Expansion time:Fum.O2:1.971794779999982\n",
      "1723049866.9680421 INFO: Filtered count:0 out of 1385\n",
      "1723049867.4545975 INFO: cpd05178 not found in model!\n",
      "1723049867.4566762 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049867.6836057 INFO: Expansion time:Pyr.SO4:2.015541468000265\n",
      "1723049867.6858304 INFO: Filtered count:0 out of 1530\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2945SF2_B_D2.55.pg.G.D GF: 11391 Prob: 4018 Other: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049868.8305078 INFO: Expansion time:Pyr.TMAO:1.1432408780001424\n",
      "1723049868.8328967 INFO: Filtered count:0 out of 1530\n",
      "1723049869.4177802 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049869.7860918 INFO: Expansion time:empty:0.9517475559996456\n",
      "1723049869.7886174 INFO: Filtered count:0 out of 1530\n",
      "1723049870.0510716 INFO: Expansion time:LLac.O2:3.081783247000203\n",
      "1723049870.0532331 INFO: Filtered count:0 out of 1385\n",
      "1723049870.1590667 INFO: cpd05178 not found in model!\n",
      "1723049871.1002407 INFO: cpd05178 not found in model!\n",
      "1723049871.102357 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049871.9704945 INFO: Expansion time:Pyr:1.9159815380007785\n",
      "1723049871.9726415 INFO: Filtered count:0 out of 1385\n",
      "1723049872.7082844 INFO: Expansion time:Pyr.DMSO:0.7344391340002403\n",
      "1723049872.7104537 INFO: Filtered count:0 out of 1385\n",
      "1723049873.372129 INFO: Expansion time:Pyr.NO:0.6604225549999683\n",
      "1723049873.374322 INFO: Filtered count:0 out of 1385\n",
      "1723049873.9029264 INFO: cpd05178 not found in model!\n",
      "1723049874.0169642 INFO: Expansion time:Pyr.NO2:0.641457158000776\n",
      "1723049874.0190701 INFO: Filtered count:0 out of 1385\n",
      "1723049874.1229713 INFO: cpd05178 not found in model!\n",
      "1723049874.1251574 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049874.2188814 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049874.654143 INFO: Expansion time:Pyr.NO3:0.6337637289998383\n",
      "1723049874.6563747 INFO: Filtered count:0 out of 1385\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049878.2662148 INFO: Expansion time:Pyr.O2:3.6086193000001003\n",
      "1723049878.2688177 INFO: Filtered count:0 out of 1385\n",
      "1723049878.6856053 INFO: cpd05178 not found in model!\n",
      "1723049878.7618024 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049878.9001486 INFO: cpd05178 not found in model!\n",
      "1723049878.9021595 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049879.1630654 INFO: Objective with gapfill database:186.14313520204138; min objective:0.01\n",
      "1723049879.4014378 INFO: Expansion time:Dlac.O2:1.4942879270001868\n",
      "1723049879.4037402 INFO: Filtered count:0 out of 1543\n",
      "1723049879.8904095 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049880.069504 INFO: Expansion time:Etho.O2:0.6642963639997106\n",
      "1723049880.0716166 INFO: Filtered count:0 out of 1543\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049880.4922771 INFO: cpd05178 not found in model!\n",
      "1723049880.7292042 INFO: Expansion time:For.O2:0.6562772759998552\n",
      "1723049880.731192 INFO: Filtered count:0 out of 1543\n",
      "1723049880.771708 INFO: cpd05178 not found in model!\n",
      "1723049880.773987 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049881.3912857 INFO: Expansion time:Glc:0.6588408280003932\n",
      "1723049881.3932354 INFO: Filtered count:0 out of 1543\n",
      "1723049881.9665766 INFO: Expansion time:Pyr.SO3:3.696246444999815\n",
      "1723049881.9688053 INFO: Filtered count:0 out of 1385\n",
      "1723049882.0526276 INFO: Expansion time:Glc.DMSO:0.6581979620004859\n",
      "1723049882.0545986 INFO: Filtered count:0 out of 1543\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049882.7399702 INFO: Expansion time:Glc.O2:0.6842052230003901\n",
      "1723049882.742033 INFO: Filtered count:0 out of 1543\n",
      "1723049883.3415983 INFO: Expansion time:Etho.O2:0.6648146249999627\n",
      "1723049883.3438652 INFO: Filtered count:0 out of 1513\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-55\n",
      "[221 'R2_C_D1.7b.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:51:16+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '493e492c838abec309975e637e1edfb1'\n",
      " 14711550 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049884.0114977 INFO: Expansion time:For.O2:0.666362785000274\n",
      "1723049884.013783 INFO: Filtered count:0 out of 1513\n",
      "1723049884.2618918 INFO: Expansion time:Glc.TMAO:1.5186403349998727\n",
      "1723049884.2639482 INFO: Filtered count:0 out of 1543\n",
      "1723049884.3709016 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049884.499378 INFO: metabolites 1105\n",
      "1723049884.676527 INFO: Expansion time:Pyr:0.6614881879995664\n",
      "1723049884.678742 INFO: Filtered count:0 out of 1513\n",
      "1723049884.9415286 INFO: Expansion time:LLac.O2:0.6763314769996214\n",
      "1723049884.943468 INFO: Filtered count:0 out of 1543\n",
      "1723049885.2172651 INFO: reactions 1085\n",
      "1723049885.4415054 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049885.6082466 INFO: Expansion time:Pyr:0.6635853039997528\n",
      "1723049885.6103237 INFO: Filtered count:0 out of 1543\n",
      "1723049885.8116815 INFO: cpd05178 not found in model!\n",
      "1723049885.8139842 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049886.2927482 INFO: Expansion time:Pyr.DMSO:0.6811860439993325\n",
      "1723049886.2948153 INFO: Filtered count:0 out of 1543\n",
      "1723049886.3717663 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/503/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049886.5775187 INFO: Expansion time:Pyr.DMSO:1.897558977000699\n",
      "1723049886.5794775 INFO: Filtered count:0 out of 1513\n",
      "1723049886.5952365 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/503/1\"\n",
      "}\n",
      "1723049886.9689744 INFO: Expansion time:Pyr.NO:0.6729790280005545\n",
      "1723049886.9711542 INFO: Filtered count:0 out of 1543\n",
      "1723049887.218267 INFO: Expansion time:Pyr.NO:0.6376156630003607\n",
      "1723049887.2204254 INFO: Filtered count:0 out of 1513\n",
      "1723049887.6284685 INFO: Expansion time:Pyr.NO2:0.6560084669999924\n",
      "1723049887.6305792 INFO: Filtered count:0 out of 1543\n",
      "1723049887.8666334 INFO: Expansion time:Pyr.NO2:0.6450585949996821\n",
      "1723049887.8687623 INFO: Filtered count:0 out of 1513\n",
      "1723049888.5222201 INFO: Expansion time:Pyr.NO3:0.6521355289996791\n",
      "1723049888.5246596 INFO: Filtered count:0 out of 1513\n",
      "1723049889.0969093 INFO: Expansion time:Pyr.NO3:1.4651893749996816\n",
      "1723049889.0989177 INFO: Filtered count:0 out of 1543\n",
      "1723049889.1679218 INFO: Filtered count:0 out of 1513\n",
      "1723049889.1658802 INFO: Expansion time:Pyr.O2:0.6400473499998043\n",
      "1723049889.777434 INFO: Expansion time:Pyr.O2:0.6774027210003624\n",
      "1723049889.7794156 INFO: Filtered count:0 out of 1543\n",
      "1723049889.8226838 INFO: Filtered count:0 out of 1513\n",
      "1723049889.8205838 INFO: Expansion time:Pyr.SO3:0.6515402579998408\n",
      "1723049890.4628408 INFO: Expansion time:Pyr.SO4:0.6389043250001123\n",
      "1723049890.4648836 INFO: Filtered count:0 out of 1513\n",
      "1723049890.5404673 INFO: Filtered count:0 out of 1543\n",
      "1723049890.6615515 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049890.538467 INFO: Expansion time:Pyr.SO3:0.7579377540005225\n",
      "1723049890.6593277 INFO: cpd05178 not found in model!\n",
      "1723049891.1060562 INFO: Expansion time:Pyr.TMAO:0.6400815690003583\n",
      "1723049891.1082184 INFO: Filtered count:0 out of 1513\n",
      "1723049891.2229304 INFO: Expansion time:Pyr.SO4:0.6810944919998292\n",
      "1723049891.224818 INFO: Filtered count:0 out of 1543\n",
      "1723049891.885573 INFO: Expansion time:Pyr.TMAO:0.6595690750000358\n",
      "1723049891.8876529 INFO: Filtered count:0 out of 1543\n",
      "1723049892.5604022 INFO: Expansion time:empty:0.671491225999489\n",
      "1723049892.562503 INFO: Filtered count:0 out of 1543\n",
      "1723049892.9866037 INFO: Expansion time:empty:1.877207546000136\n",
      "1723049892.988704 INFO: Filtered count:0 out of 1513\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00062_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049893.7934759 INFO: Expansion time:Pyr.SO4:11.823399496999627\n",
      "1723049893.7959495 INFO: Filtered count:1 out of 1385\n",
      "1723049894.7679205 INFO: Expansion time:Pyr.TMAO:0.9704612429995905\n",
      "1723049894.7705266 INFO: Filtered count:1 out of 1385\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[146 'R2_A_D1.119.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:36:27+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '1005f0af612c33eccb820352c703ad89'\n",
      " 9738254 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049895.2057083 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049895.203361 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.16.pg.G.D GF: 10861 Prob: 4630 Other: 2861\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049895.599068 INFO: metabolites 949\n",
      "1723049895.7735724 INFO: Expansion time:empty:1.0014605819997087\n",
      "1723049895.7762606 INFO: Filtered count:1 out of 1385\n",
      "1723049895.9494781 INFO: reactions 919\n",
      "1723049896.0573788 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049896.3991745 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/593/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049897.833668 INFO: cpd05178 not found in model!\n",
      "1723049897.8363705 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049901.5305972 INFO: cpd05178 not found in model!\n",
      "1723049901.5329828 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[166 'R2_restored_DShore.11.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:40:19+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '18e48c38f07a57cb023f6bbd312a686a'\n",
      " 16067290 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049902.517191 INFO: metabolites 1225\n",
      "1723049903.25637 INFO: reactions 1214\n",
      "1723049903.4858558 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049904.0389607 INFO: cpd05178 not found in model!\n",
      "1723049904.0411878 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049904.1361032 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049904.2304928 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/107/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049905.2020602 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049905.686172 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-56\n",
      "[225 'R1_A_D2.31.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:52:02+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9bba42226ae76d29dd14516d1167860a'\n",
      " 10250542 None]\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049906.7201355 INFO: metabolites 932\n",
      "1723049907.2721057 INFO: reactions 881\n",
      "1723049907.4281414 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049908.0071466 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/514/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049908.2111175 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/514/1\"\n",
      "}\n",
      "1723049909.8918204 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049909.9413319 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049910.4113052 INFO: cpd05178 not found in model!\n",
      "1723049910.651465 INFO: cpd05178 not found in model!\n",
      "1723049910.6537912 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049911.2339892 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049913.284174 INFO: cpd05178 not found in model!\n",
      "1723049913.533122 INFO: cpd05178 not found in model!\n",
      "1723049913.5352898 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049914.6927843 INFO: Expansion time:Dlac.O2:1.3887487849997342\n",
      "1723049914.6949868 INFO: Filtered count:0 out of 1756\n",
      "1723049915.3346956 INFO: Expansion time:Etho.O2:0.6385372750000897\n",
      "1723049915.3369536 INFO: Filtered count:0 out of 1756\n",
      "1723049915.8669577 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049915.9747748 INFO: Expansion time:For.NO2:0.6365753990003213\n",
      "1723049915.9769514 INFO: Filtered count:0 out of 1756\n",
      "1723049916.6145222 INFO: Expansion time:For.O2:0.6363619929998094\n",
      "1723049916.6168222 INFO: Filtered count:0 out of 1756\n",
      "1723049917.2553189 INFO: Expansion time:Fum:0.6372916059999625\n",
      "1723049917.2575388 INFO: Filtered count:0 out of 1756\n",
      "1723049917.4683497 INFO: cpd05178 not found in model!\n",
      "1723049917.471165 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049917.8808672 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049917.911212 INFO: Filtered count:0 out of 1756\n",
      "1723049917.9091063 INFO: Expansion time:Fum.O2:0.6503598609997425\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[46 'R1_A_D2.1.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:16:39+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'cb3fc3424efe0561d625994c2673e815'\n",
      " 6888295 None]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049918.2428627 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049918.378781 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049919.1812642 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049919.2979858 INFO: Expansion time:Pyr:1.3856284929997855\n",
      "1723049919.300212 INFO: Filtered count:0 out of 1756\n",
      "1723049919.94981 INFO: Expansion time:Pyr.DMSO:0.6484334470005706\n",
      "1723049919.9520285 INFO: Filtered count:0 out of 1756\n",
      "1723049920.0896487 INFO: metabolites 958\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Getting tests from attributes"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049920.56466 INFO: reactions 886\n",
      "1723049920.5921311 INFO: Expansion time:Pyr.NO:0.6389774700000999\n",
      "1723049920.5941486 INFO: Filtered count:0 out of 1756\n",
      "1723049920.7074587 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049920.8145308 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049920.8120666 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.119.pg.G.D GF: 12222 Prob: 2744 Other: 3384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049921.2303345 INFO: Default biomass: [bio1]\n",
      "1723049921.2338285 INFO: Expansion time:Pyr.NO2:0.6385007219996623\n",
      "1723049921.2356634 INFO: Filtered count:0 out of 1756\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/573/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049921.7357934 INFO: cpd05178 not found in model!\n",
      "1723049921.877003 INFO: Expansion time:Pyr.NO3:0.6401465430008102\n",
      "1723049921.8790822 INFO: Filtered count:0 out of 1756\n",
      "1723049921.9690468 INFO: cpd05178 not found in model!\n",
      "1723049921.971021 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049922.5193899 INFO: Expansion time:Pyr.O2:0.6391448839995064\n",
      "1723049922.5214982 INFO: Filtered count:0 out of 1756\n",
      "1723049922.9602497 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049923.5330384 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049923.8693147 INFO: cpd05178 not found in model!\n",
      "1723049923.8714268 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049923.9068334 INFO: Expansion time:Pyr.SO3:1.384154585000033\n",
      "1723049923.9089491 INFO: Filtered count:0 out of 1756\n",
      "1723049924.5398753 INFO: cpd05178 not found in model!\n",
      "1723049924.5420272 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049924.547616 INFO: Expansion time:Pyr.SO4:0.6375041300007069\n",
      "1723049924.5495238 INFO: Filtered count:0 out of 1756\n",
      "1723049925.1864924 INFO: Expansion time:Pyr.TMAO:0.635766276999675\n",
      "1723049925.1886337 INFO: Filtered count:0 out of 1756\n",
      "1723049925.82265 INFO: Expansion time:empty:0.6328262239994729\n",
      "1723049925.8247364 INFO: Filtered count:0 out of 1756\n",
      "1723049926.0542898 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049926.9921975 INFO: cpd05178 not found in model!\n",
      "1723049926.9948344 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-57\n",
      "[229 'SF2_A_H2O.34.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:52:51+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '0d56dfb1ff738c9d0315042d9b175ae0'\n",
      " 5829869 None]\n",
      "R2_C_D1.7b.pg.G.D GF: 11046 Prob: 4348 Other: 2960\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049928.1017892 INFO: metabolites 997\n",
      "1723049928.5537944 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049928.5748343 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049928.5715446 INFO: cpd05178 not found in model!\n",
      "1723049928.6755116 INFO: reactions 935\n",
      "1723049928.8001542 INFO: cpd05178 not found in model!\n",
      "1723049928.8327363 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049929.1242661 INFO: cpd05178 not found in model!\n",
      "1723049929.1271 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049929.4379764 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/511/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049929.63549 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/511/1\"\n",
      "}\n",
      "1723049932.5594575 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[150 'R2_restored_DShore.21.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:37:14+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '40f51482e08474fc024bf596dcdec053'\n",
      " 11525518 None]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049932.561593 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "3138 "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049933.0480669 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.31.pg.G.D GF: 11910 Prob: 3304 Other:"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049933.042343 INFO: metabolites 1064\n",
      "1723049933.6705196 INFO: reactions 984\n",
      "1723049933.8261573 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049934.438908 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/18/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049934.7675223 INFO: cpd05178 not found in model!\n",
      "1723049934.7697852 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.11.pg.G.D GF: 11318 Prob: 3982 Other: 3054\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049935.8551416 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049935.9391775 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049936.788859 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049937.094613 INFO: Objective with gapfill database:188.54381916447088; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[98 'SF2_A_D2.8.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:26:57+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '8cbf2bcd9f557e8e6751fbdbe00687c2'\n",
      " 14876812 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049939.807541 INFO: metabolites 1137\n",
      "1723049939.990692 INFO: cpd05178 not found in model!\n",
      "1723049939.9928098 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049940.4689422 INFO: reactions 1104\n",
      "1723049940.6415362 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049941.2654324 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/304/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049941.424108 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049942.5285468 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049942.7395582 INFO: cpd05178 not found in model!\n",
      "1723049942.7415152 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.1.pg.G.D GF: 12186 Prob: 2828 Other: 3336\n",
      "[158 'SF2_C_H2O.29.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:38:43+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '5d9d56b389726d22b39422cf2fe071a9'\n",
      " 6230933 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049944.7223182 INFO: metabolites 945\n",
      "1723049945.1806333 INFO: reactions 889\n",
      "1723049945.2794204 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049945.564248 INFO: Objective with gapfill database:172.96728084855584; min objective:0.01\n",
      "1723049945.6110685 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/59/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049945.8419187 INFO: Filtered count:0 out of 1563\n",
      "1723049945.8399184 INFO: Expansion time:Etho.O2:1.3868177620006463\n",
      "1723049946.464872 INFO: Expansion time:For.O2:0.621774471000208\n",
      "1723049946.4671376 INFO: Filtered count:0 out of 1563\n",
      "1723049947.0810497 INFO: Expansion time:Glc:0.6126627069997994\n",
      "1723049947.0834026 INFO: Filtered count:0 out of 1563\n",
      "1723049947.7043772 INFO: Expansion time:Glc.DMSO:0.6197435829999449\n",
      "1723049947.7065432 INFO: Filtered count:0 out of 1563\n",
      "1723049947.7416482 INFO: cpd05178 not found in model!\n",
      "1723049947.743746 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049948.3291895 INFO: Expansion time:Glc.O2:0.6213721809999697\n",
      "1723049948.3312907 INFO: Filtered count:0 out of 1563\n",
      "1723049948.5287976 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-58\n",
      "[233 'R2A_C_D1.8.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:53:36+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'c46a19c5024fb45945723afbac4477ae'\n",
      " 10576978 None]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049948.9274962 INFO: cpd05178 not found in model!\n",
      "1723049948.9295182 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049948.9643564 INFO: Expansion time:Glc.TMAO:0.6318959690006523\n",
      "1723049948.9665468 INFO: Filtered count:0 out of 1563\n",
      "1723049949.0399685 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049949.3188407 INFO: cpd05178 not found in model!\n",
      "1723049949.3208916 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049949.405703 INFO: cpd05178 not found in model!\n",
      "1723049949.40762 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049949.4594917 INFO: metabolites 1039\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049950.0588605 INFO: reactions 1005\n",
      "1723049950.2642639 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049950.3625603 INFO: Expansion time:LLac.O2:1.3948294929996337\n",
      "1723049950.364701 INFO: Filtered count:0 out of 1563\n",
      "1723049950.734867 INFO: cpd05178 not found in model!\n",
      "1723049950.7369738 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049950.8311818 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049950.9758575 INFO: Default biomass: [bio1]\n",
      "1723049951.006192 INFO: Filtered count:0 out of 1563\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_H2O.34.pg.G.D GF: 12141 Prob: 3146 Other: 3065\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049951.0043352 INFO: Expansion time:Pyr:0.6384931950005921\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/538/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049951.2177658 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/538/1\"\n",
      "}\n",
      "1723049951.3201387 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049951.6498432 INFO: Expansion time:Pyr.DMSO:0.6424390219999623\n",
      "1723049951.652004 INFO: Filtered count:0 out of 1563\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049952.2846298 INFO: Expansion time:Pyr.NO:0.6313187750001816\n",
      "1723049952.2866206 INFO: Filtered count:0 out of 1563\n",
      "1723049952.9052432 INFO: Expansion time:Pyr.NO2:0.6175070920007784\n",
      "1723049952.9073477 INFO: Filtered count:0 out of 1563\n",
      "1723049953.5262334 INFO: Expansion time:Pyr.NO3:0.6176728500004174\n",
      "1723049953.528319 INFO: Filtered count:0 out of 1563\n",
      "1723049953.7107532 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049954.4753518 INFO: cpd05178 not found in model!\n",
      "1723049954.651328 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049954.7005675 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049954.698411 INFO: cpd05178 not found in model!\n",
      "1723049954.9067347 INFO: Expansion time:Pyr.O2:1.3772827000002508\n",
      "1723049954.9089036 INFO: Filtered count:0 out of 1563\n",
      "1723049955.2433176 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049955.3705246 INFO: Expansion time:Ac.O2:4.335111000999859\n",
      "1723049955.3727705 INFO: Filtered count:0 out of 1728\n",
      "1723049955.544929 INFO: Expansion time:Pyr.SO3:0.634863618000054\n",
      "1723049955.54712 INFO: Filtered count:0 out of 1563\n",
      "1723049955.7746012 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049956.171596 INFO: Expansion time:Pyr.SO4:0.6233345800001189\n",
      "1723049956.1738064 INFO: Filtered count:0 out of 1563\n",
      "1723049956.255467 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049956.2530499 INFO: cpd05178 not found in model!\n",
      "1723049956.3805628 INFO: Expansion time:Etho.O2:1.006578836000699\n",
      "1723049956.3826532 INFO: Filtered count:0 out of 1728\n",
      "1723049956.626023 INFO: cpd05178 not found in model!\n",
      "1723049956.8135917 INFO: Expansion time:Pyr.TMAO:0.6385050360004243\n",
      "1723049956.815771 INFO: Filtered count:0 out of 1563\n",
      "1723049956.8540428 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049956.9543107 INFO: cpd05178 not found in model!\n",
      "1723049956.9568017 INFO: Media compound: cpd05178 not found in model.\n",
      "1723049957.4355211 INFO: Expansion time:empty:0.6185630580002908\n",
      "1723049957.4376516 INFO: Filtered count:0 out of 1563\n",
      "1723049957.6166666 INFO: Expansion time:Fum:1.232851801999459\n",
      "1723049957.618835 INFO: Filtered count:0 out of 1728\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049958.867225 INFO: Expansion time:Fum.O2:1.2471974359996238\n",
      "1723049958.8692925 INFO: Filtered count:0 out of 1728\n",
      "1723049959.9849849 INFO: Expansion time:Glc:1.1144549600003302\n",
      "1723049959.98714 INFO: Filtered count:0 out of 1728\n",
      "1723049960.715206 INFO: cpd05178 not found in model!\n",
      "1723049960.7175684 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049960.8761346 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049961.0509481 INFO: Expansion time:Glc.DMSO:1.0625811789996078\n",
      "1723049961.053179 INFO: Filtered count:0 out of 1728\n",
      "1723049961.792213 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049962.3273933 INFO: Expansion time:Glc.O2:1.2728437810001196\n",
      "1723049962.3294635 INFO: Filtered count:0 out of 1728\n",
      "1723049962.367418 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049963.0830686 INFO: Objective with gapfill database:155.19513731741043; min objective:0.01\n",
      "1723049963.1363826 INFO: cpd05178 not found in model!\n",
      "1723049963.1382768 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049964.7583795 INFO: Expansion time:Glc.TMAO:2.4277402180005083\n",
      "1723049964.760516 INFO: Filtered count:0 out of 1728\n",
      "1723049965.2026923 INFO: cpd05178 not found in model!\n",
      "1723049965.2050517 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.21.pg.G.D GF: 11602 Prob: 3758 Other: 2994\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049965.964563 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049965.980268 INFO: Expansion time:Glyc.O2:1.2185919419998754\n",
      "1723049965.9821498 INFO: Filtered count:0 out of 1728\n",
      "1723049966.155499 INFO: cpd05178 not found in model!\n",
      "1723049966.157708 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049967.2147725 INFO: Expansion time:Pyr:1.2314737650003735\n",
      "1723049967.2169893 INFO: Filtered count:0 out of 1728\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049968.1206508 INFO: cpd05178 not found in model!\n",
      "1723049968.122995 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_H2O.29.pg.G.D GF: 12314 Prob: 2878 Other: 3160\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049968.7418687 INFO: Expansion time:Fum:0.9666531960001521\n",
      "1723049968.7443547 INFO: Filtered count:0 out of 1215\n",
      "1723049969.9334629 INFO: Expansion time:Fum.O2:1.187684874000297\n",
      "1723049969.9361818 INFO: Filtered count:0 out of 1215\n",
      "1723049970.897588 INFO: Expansion time:Pyr:0.9597789210001793\n",
      "1723049970.900181 INFO: Filtered count:0 out of 1215\n",
      "1723049970.9201927 INFO: Expansion time:Pyr.DMSO:3.702046001999406\n",
      "1723049970.922169 INFO: Filtered count:0 out of 1728\n",
      "1723049971.7760813 INFO: Expansion time:Pyr.NO:0.8527435790001618\n",
      "1723049971.7782524 INFO: Filtered count:0 out of 1728\n",
      "1723049971.878176 INFO: Expansion time:Pyr.DMSO:0.97648794899942\n",
      "1723049971.8806717 INFO: Filtered count:0 out of 1215\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-59\n",
      "[237 'SF2_C_D2.11.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:54:21+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'df5838f4661576d9c68665190441096e'\n",
      " 11901243 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049972.683889 INFO: Expansion time:Pyr.NO2:0.9045127769995815\n",
      "1723049972.6860504 INFO: Filtered count:0 out of 1728\n",
      "1723049972.826245 INFO: Expansion time:Pyr.NO:0.9439031160000013\n",
      "1723049972.828419 INFO: Filtered count:0 out of 1215\n",
      "1723049972.9933689 INFO: metabolites 1198\n",
      "1723049973.020767 INFO: cpd05178 not found in model!\n",
      "1723049973.023514 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_D2.8.pg.G.D GF: 11521 Prob: 3798 Other: 3033\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049973.6443386 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049973.7523136 INFO: reactions 1156\n",
      "1723049973.8143198 INFO: Expansion time:Pyr.NO2:0.9843530670004839\n",
      "1723049973.8168314 INFO: Filtered count:0 out of 1215\n",
      "1723049973.9421043 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049974.72382 INFO: Expansion time:Pyr.NO3:2.0365060909998647\n",
      "1723049974.725844 INFO: Filtered count:0 out of 1728\n",
      "1723049974.918874 INFO: Default biomass: [bio1]\n",
      "1723049974.9276173 INFO: cpd05178 not found in model!\n",
      "1723049974.9301786 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/546/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049975.1429744 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/546/1\"\n",
      "}\n",
      "1723049976.3205462 INFO: Expansion time:Pyr.NO3:2.502122331999999\n",
      "1723049976.3232286 INFO: Filtered count:0 out of 1215\n",
      "1723049977.0110357 INFO: cpd05178 not found in model!\n",
      "1723049977.0136967 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723049977.254562 INFO: Expansion time:Pyr.O2:0.92973239299954\n",
      "1723049977.2572682 INFO: Filtered count:0 out of 1215\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D1.8.pg.G.D GF: 11848 Prob: 3302 Other: 3202\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049977.4513516 INFO: Expansion time:Pyr.O2:2.7242752110005313\n",
      "1723049977.453533 INFO: Filtered count:0 out of 1728\n",
      "1723049978.3050702 INFO: Expansion time:Pyr.SO3:1.046128920999763\n",
      "1723049978.3076892 INFO: Filtered count:0 out of 1215\n",
      "1723049978.452233 INFO: Expansion time:Pyr.SO3:0.9972776120002891\n",
      "1723049978.454392 INFO: Filtered count:0 out of 1728\n",
      "1723049979.1936882 INFO: Expansion time:Pyr.SO4:0.7380730709992349\n",
      "1723049979.1958373 INFO: Filtered count:0 out of 1728\n",
      "1723049979.2273123 INFO: cpd05178 not found in model!\n",
      "1723049979.2295158 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049979.2820435 INFO: Expansion time:Pyr.SO4:0.97284309799943\n",
      "1723049979.2844608 INFO: Filtered count:0 out of 1215\n",
      "1723049979.8609457 INFO: Expansion time:Pyr.TMAO:0.6638648019998072\n",
      "1723049979.8630767 INFO: Filtered count:0 out of 1728\n",
      "1723049980.2619805 INFO: Expansion time:Pyr.TMAO:0.9760093169998072\n",
      "1723049980.2645328 INFO: Filtered count:0 out of 1215\n",
      "1723049980.545243 INFO: Expansion time:empty:0.6809297250001691\n",
      "1723049980.5473564 INFO: Filtered count:0 out of 1728\n",
      "1723049981.2609541 INFO: Expansion time:empty:0.9949511839995466\n",
      "1723049981.2637796 INFO: Filtered count:0 out of 1215\n",
      "1723049982.4366567 INFO: Objective with gapfill database:116.18218438586337; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049983.3210692 INFO: cpd05178 not found in model!\n",
      "1723049983.323095 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049985.025012 INFO: Expansion time:Glc.O2:1.7771369040001446\n",
      "1723049985.0274222 INFO: Filtered count:0 out of 1383\n",
      "1723049985.0813916 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049985.6347048 INFO: Expansion time:Pyr.NO2:0.6061363250000795\n",
      "1723049985.636933 INFO: Filtered count:0 out of 1383\n",
      "1723049986.0244691 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723049986.2442641 INFO: Expansion time:Pyr.NO3:0.6061562559998492\n",
      "1723049986.246517 INFO: Filtered count:0 out of 1383\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049986.8672667 INFO: Expansion time:Pyr.O2:0.6196176370003741\n",
      "1723049986.869485 INFO: Filtered count:0 out of 1383\n",
      "1723049987.4772027 INFO: Expansion time:empty:0.6065699520004273\n",
      "1723049987.4796321 INFO: Filtered count:0 out of 1383\n",
      "1723049987.6760528 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049988.7600782 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723049990.2514596 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049990.3152845 INFO: cpd05178 not found in model!\n",
      "1723049990.6654449 INFO: cpd05178 not found in model!\n",
      "1723049990.6682622 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-60\n",
      "[241 'R1_C_D2.85.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:55:10+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '5553478a6bdcce9a64f6ee08003a964f'\n",
      " 8855724 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049994.8855908 INFO: metabolites 1046\n",
      "1723049995.4614253 INFO: reactions 1011\n",
      "1723049995.588193 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723049995.9732466 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/42/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049996.179141 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/42/1\"\n",
      "}\n",
      "1723049998.7293153 INFO: cpd05178 not found in model!\n",
      "1723049998.7314281 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[86 'R2_restored_DShore.68.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:24:32+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'dcf023b9398a5c1727a21a97201f31b7'\n",
      " 11084143 None]\n",
      "SF2_C_D2.11.pg.G.D GF: 11048 Prob: 4486 Other: 2820\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723049999.1826916 INFO: cpd05178 not found in model!\n",
      "1723049999.4223378 INFO: metabolites 1093\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723049999.6672685 INFO: cpd05178 not found in model!\n",
      "1723049999.6694417 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723049999.8216925 INFO: reactions 1029\n",
      "1723049999.951758 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050000.2233768 INFO: cpd05178 not found in model!\n",
      "1723050000.225573 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050000.2777982 INFO: cpd05178 not found in model!\n",
      "1723050000.279739 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050000.3549147 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/202/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050000.7597797 INFO: Expansion time:Etho.O2:0.9654653430006874\n",
      "1723050000.7623482 INFO: Filtered count:0 out of 1412\n",
      "1723050001.417027 INFO: Expansion time:Ac.NO3:0.6618498179996095\n",
      "1723050001.419051 INFO: Filtered count:0 out of 1580\n",
      "1723050002.0751984 INFO: Expansion time:Ac.O2:0.6549983610002528\n",
      "1723050002.0772715 INFO: Filtered count:0 out of 1580\n",
      "1723050002.4955032 INFO: Expansion time:For.NO2:1.731749137999941\n",
      "1723050002.4982753 INFO: Filtered count:0 out of 1412\n",
      "1723050002.7511 INFO: Expansion time:Etho.O2:0.6725540429997636\n",
      "1723050002.7531137 INFO: Filtered count:0 out of 1580\n",
      "1723050003.4237964 INFO: Expansion time:For.NO3:0.6695134209994649\n",
      "1723050003.4259977 INFO: Filtered count:0 out of 1580\n",
      "1723050003.4566693 INFO: Expansion time:For.O2:0.9565927419998843\n",
      "1723050003.4594223 INFO: Filtered count:0 out of 1412\n",
      "1723050004.0923104 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050004.121166 INFO: Filtered count:0 out of 1580\n",
      "1723050004.1190825 INFO: Expansion time:For.O2:0.6917738780002765\n",
      "1723050004.41666 INFO: Expansion time:Glc.O2:0.9555834980001237\n",
      "1723050004.4193182 INFO: Filtered count:0 out of 1412\n",
      "1723050004.7857308 INFO: Expansion time:Glc.O2:0.6632897670006059\n",
      "1723050004.7878375 INFO: Filtered count:0 out of 1580\n",
      "1723050005.065053 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050005.3583024 INFO: Expansion time:Pyr:0.9374282470007529\n",
      "1723050005.3608742 INFO: Filtered count:0 out of 1412\n",
      "1723050005.4050756 INFO: cpd05178 not found in model!\n",
      "1723050005.407137 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050006.2994704 INFO: Expansion time:Pyr.DMSO:0.93708607499957\n",
      "1723050006.3021274 INFO: Filtered count:0 out of 1412\n",
      "1723050006.692587 INFO: Expansion time:Pyr.NO3:1.9036193100000673\n",
      "1723050006.6948643 INFO: Filtered count:0 out of 1580\n",
      "1723050007.3595235 INFO: Expansion time:Pyr.O2:0.6633836310002152\n",
      "1723050007.3617053 INFO: Filtered count:0 out of 1580\n",
      "1723050008.036251 INFO: Expansion time:empty:0.6733418119993075\n",
      "1723050008.0384784 INFO: Filtered count:0 out of 1580\n",
      "1723050008.0481486 INFO: Expansion time:Pyr.NO:1.7444669480000812\n",
      "1723050008.050523 INFO: Filtered count:0 out of 1412\n",
      "1723050008.1377842 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050009.0421865 INFO: Expansion time:Pyr.NO2:0.9901209350000499\n",
      "1723050009.0448656 INFO: Filtered count:0 out of 1412\n",
      "1723050009.2010858 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050009.9934433 INFO: Expansion time:Pyr.NO3:0.9469907949996923\n",
      "1723050009.99603 INFO: Filtered count:0 out of 1412\n",
      "1723050010.074418 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050010.6015472 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050010.9654968 INFO: Expansion time:Pyr.O2:0.9679570870002863\n",
      "1723050010.968154 INFO: Filtered count:0 out of 1412\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050011.4011445 INFO: cpd05178 not found in model!\n",
      "1723050012.328395 INFO: cpd05178 not found in model!\n",
      "1723050012.3307323 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050012.71473 INFO: Expansion time:Pyr.SO3:1.7451131520001582\n",
      "1723050012.7173529 INFO: Filtered count:0 out of 1412\n",
      "1723050013.6826682 INFO: Expansion time:Pyr.SO4:0.9638355229999434\n",
      "1723050013.6851768 INFO: Filtered count:0 out of 1412\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[186 'R2_A_D1.98.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:44:10+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '884ca14dab7006c7651ec55ee214c306'\n",
      " 8040825 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050014.628015 INFO: Expansion time:Pyr.TMAO:0.941258917999221\n",
      "1723050014.630619 INFO: Filtered count:0 out of 1412\n",
      "1723050014.8284905 INFO: metabolites 959\n",
      "1723050015.0772724 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050015.145509 INFO: reactions 913\n",
      "1723050015.2478075 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050015.5655935 INFO: Expansion time:empty:0.9333270750003066\n",
      "1723050015.5681856 INFO: Filtered count:0 out of 1412\n",
      "1723050015.579217 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/214/1\n",
      "ForkPoolWorker-61\n",
      "[245 'R1_B_D2.14.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:55:54+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '8001f3a06b3ce7bea2f43a014e7262ca'\n",
      " 12854664 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050017.5094264 INFO: metabolites 1164\n",
      "1723050018.013376 INFO: reactions 1145\n",
      "1723050018.1431577 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050018.836679 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/54/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050019.0004091 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/54/1\"\n",
      "}\n",
      "1723050019.4315896 INFO: cpd05178 not found in model!\n",
      "1723050019.76455 INFO: cpd05178 not found in model!\n",
      "1723050019.7670844 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050020.1481977 INFO: cpd05178 not found in model!\n",
      "1723050020.1508856 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_D2.85.pg.G.D GF: 11624 Prob: 3432 Other: 3294\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050021.4645216 INFO: Filtered count:0 out of 1666\n",
      "1723050021.4624913 INFO: Expansion time:Ac.NO3:3.9996292910000193\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[122 'R2_B_D1.13.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:31:50+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e31e2ec8abfd12ffb04ac5c6db08c402'\n",
      " 12447562 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050022.4839525 INFO: Expansion time:Ac.O2:1.0182792989999143\n",
      "1723050022.4860659 INFO: Filtered count:0 out of 1666\n",
      "1723050022.5019588 INFO: metabolites 1217\n",
      "1723050023.172654 INFO: cpd05178 not found in model!\n",
      "1723050023.376151 INFO: cpd05178 not found in model!\n",
      "1723050023.378629 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050023.7565396 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050023.8881938 INFO: Expansion time:Etho.O2:1.400936770000044\n",
      "1723050023.890467 INFO: Filtered count:0 out of 1666\n",
      "1723050024.2241743 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050024.4191115 INFO: reactions 1200\n",
      "1723050024.633476 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050024.9398813 INFO: Expansion time:For.NO3:1.0481400480002776\n",
      "1723050024.9421935 INFO: Filtered count:0 out of 1666\n",
      "1723050025.1580453 INFO: cpd05178 not found in model!\n",
      "1723050025.1602132 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050025.3462706 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/473/1\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050025.9189858 INFO: Expansion time:For.O2:0.9754989069997464\n",
      "1723050025.9208782 INFO: Filtered count:0 out of 1666\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[74 'R1_B_D2.87.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:22:16+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '82db39aaf324d6898e9882c744dbe511'\n",
      " 8535591 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050026.7154422 INFO: cpd05178 not found in model!\n",
      "1723050026.71806 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050026.7960224 INFO: metabolites 1013\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.68.pg.G.D GF: 11845 Prob: 3310 Other: 3197\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050027.341452 INFO: reactions 954\n",
      "1723050027.5277302 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050028.160396 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/140/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050028.4347286 INFO: Expansion time:Glc:2.51260872200055\n",
      "1723050028.4368055 INFO: Filtered count:0 out of 1666\n",
      "1723050029.8618765 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050030.1099265 INFO: Expansion time:Glc.DMSO:1.6718894679997902\n",
      "1723050030.1119652 INFO: Filtered count:0 out of 1666\n",
      "1723050031.9756157 INFO: cpd05178 not found in model!\n",
      "1723050031.9776342 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050031.999613 INFO: Expansion time:Glc.O2:1.886502489999657\n",
      "1723050032.0015633 INFO: Filtered count:0 out of 1666\n",
      "1723050033.066346 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050033.9124053 INFO: Expansion time:Glc.TMAO:1.909535503999905\n",
      "1723050033.9145238 INFO: Filtered count:0 out of 1666\n",
      "1723050034.2249641 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050034.3058884 INFO: Objective with gapfill database:113.95161783132262; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[138 'R2A_C_D2.4.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:34:55+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ada8a07357c8bc8f49769373e35041ee'\n",
      " 18598633 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050034.6634007 INFO: Filtered count:0 out of 1666\n",
      "1723050034.6613438 INFO: Expansion time:Pyr:0.7456123060001119\n",
      "1723050034.9712965 INFO: cpd05178 not found in model!\n",
      "1723050034.9733968 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050035.117698 INFO: metabolites 1177\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050035.8447065 INFO: reactions 1144\n",
      "1723050036.0294042 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050036.1004686 INFO: Expansion time:Pyr.DMSO:1.4359600729994781\n",
      "1723050036.1025424 INFO: Filtered count:0 out of 1666\n",
      "1723050036.6006875 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050036.703985 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/575/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050036.9914932 INFO: Expansion time:Pyr.NO:0.8877800320005917\n",
      "1723050036.993605 INFO: Filtered count:0 out of 1666\n",
      "1723050037.803958 INFO: Expansion time:Pyr.NO2:0.8091193360005491\n",
      "1723050037.8060498 INFO: Filtered count:0 out of 1666\n",
      "1723050038.106851 INFO: cpd05178 not found in model!\n",
      "1723050038.1090825 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.98.pg.G.D GF: 12144 Prob: 2942 Other: 3266\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050038.3874197 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050038.8287024 INFO: Expansion time:Pyr.NO3:1.02145465600006\n",
      "1723050038.830766 INFO: Filtered count:0 out of 1666\n",
      "1723050038.8424408 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050040.0929365 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050040.1568558 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050040.244864 INFO: Expansion time:Pyr.O2:1.4129833699998926\n",
      "1723050040.2469249 INFO: Filtered count:0 out of 1666\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-62\n",
      "[249 'R2_restored_DShore.8.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:56:46+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '54575cef3edd3cdade019239ab9afc18'\n",
      " 22587095 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050041.449118 INFO: metabolites 1207\n",
      "1723050041.9230917 INFO: reactions 1201\n",
      "1723050042.0620158 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050042.270653 INFO: Expansion time:Pyr.SO3:2.0225776119996226\n",
      "1723050042.272667 INFO: Filtered count:0 out of 1666\n",
      "1723050042.3829584 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050042.8918676 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050043.0190532 INFO: Expansion time:Pyr.SO4:0.7452991330001169\n",
      "1723050043.0210693 INFO: Filtered count:0 out of 1666\n",
      "1723050043.26099 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/69/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050043.4754615 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/69/1\"\n",
      "}\n",
      "1723050043.727006 INFO: Expansion time:Pyr.TMAO:0.7048225939997792\n",
      "1723050043.7290614 INFO: Filtered count:0 out of 1666\n",
      "1723050044.3902915 INFO: Expansion time:empty:0.6600045889999819\n",
      "1723050044.3923743 INFO: Filtered count:0 out of 1666\n",
      "1723050045.608959 INFO: cpd05178 not found in model!\n",
      "1723050045.8422387 INFO: cpd05178 not found in model!\n",
      "1723050045.8444397 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050047.0049918 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050047.231697 INFO: cpd05178 not found in model!\n",
      "1723050047.2336044 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050047.4958425 INFO: cpd05178 not found in model!\n",
      "1723050047.4978836 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.14.pg.G.D GF: 11410 Prob: 3882 Other: 3060\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050049.6304512 INFO: cpd05178 not found in model!\n",
      "1723050049.8708608 INFO: cpd05178 not found in model!\n",
      "1723050049.873046 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050051.5351753 INFO: Objective with gapfill database:93.17832089326245; min objective:0.01\n",
      "1723050052.2535863 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050053.1625092 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050053.8302805 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050053.8576856 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050054.514339 INFO: cpd05178 not found in model!\n",
      "1723050054.75628 INFO: cpd05178 not found in model!\n",
      "1723050054.7585533 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050055.1011407 INFO: cpd05178 not found in model!\n",
      "1723050055.103904 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.87.pg.G.D GF: 11801 Prob: 3434 Other: 3119\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050055.8307755 INFO: cpd05178 not found in model!\n",
      "1723050055.83295 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D1.13.pg.G.D GF: 10926 Prob: 4614 Other: 2812\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050056.7501278 INFO: Expansion time:Etho.O2:2.3548734639998656\n",
      "1723050056.7523148 INFO: Filtered count:0 out of 1458\n",
      "1723050057.394242 INFO: Expansion time:For.NO2:0.6406754010004079\n",
      "1723050057.3961442 INFO: Filtered count:0 out of 1458\n",
      "1723050057.4640636 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050057.915925 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050057.913856 INFO: cpd05178 not found in model!\n",
      "1723050058.0207648 INFO: Filtered count:0 out of 1458\n",
      "1723050058.0187888 INFO: Expansion time:For.O2:0.6214646690004884\n",
      "1723050058.6182437 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050058.6587071 INFO: Filtered count:0 out of 1458\n",
      "1723050058.6565957 INFO: Expansion time:Glc.O2:0.6347298789996785\n",
      "1723050059.3240387 INFO: Expansion time:Pyr:0.6641605659997367\n",
      "1723050059.3261342 INFO: Filtered count:0 out of 1458\n",
      "1723050059.9955206 INFO: Expansion time:Pyr.DMSO:0.6681902310001533\n",
      "1723050059.997934 INFO: Filtered count:0 out of 1458\n",
      "1723050060.5428383 INFO: Objective with gapfill database:243.21205695765187; min objective:0.01\n",
      "1723050060.685907 INFO: Expansion time:Pyr.NO:0.68642865000038\n",
      "1723050060.688004 INFO: Filtered count:0 out of 1458\n",
      "1723050062.404342 INFO: Expansion time:Pyr.NO2:1.715159295000376\n",
      "1723050062.4063487 INFO: Filtered count:0 out of 1458\n",
      "1723050062.5752025 INFO: cpd05178 not found in model!\n",
      "1723050062.5771236 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050062.577982 INFO: cpd05178 not found in model!\n",
      "1723050062.579748 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-63\n",
      "[253 'R2A_A_D1.42.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:57:37+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'aa03d76fd8624bd616f70a001b3b153f'\n",
      " 10126089 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050063.4467418 INFO: Expansion time:Pyr.NO3:1.0387429500005965\n",
      "1723050063.4493396 INFO: Filtered count:0 out of 1458\n",
      "1723050063.6946933 INFO: metabolites 1012\n",
      "1723050064.191812 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050064.2793138 INFO: reactions 978\n",
      "1723050064.4479947 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050064.5144548 INFO: Expansion time:Pyr.O2:1.0635179880000578\n",
      "1723050064.5168686 INFO: Filtered count:0 out of 1458\n",
      "1723050065.017821 INFO: cpd05178 not found in model!\n",
      "1723050065.019843 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050065.0308402 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GF:\n",
      "188040/76/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050065.2383628 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/76/1\"\n",
      "}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.8.pg.G.D  11304 Prob: 4096 Other: 2954"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050065.5448027 INFO: Expansion time:Pyr.SO3:1.0264954690001105\n",
      "1723050065.5473886 INFO: Filtered count:0 out of 1458\n",
      "1723050066.6255116 INFO: Expansion time:Pyr.SO4:1.0764674580004794\n",
      "1723050066.6279857 INFO: Filtered count:0 out of 1458\n",
      "1723050066.7360094 INFO: cpd05178 not found in model!\n",
      "1723050066.73807 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050067.3378165 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050067.6033428 INFO: Expansion time:Pyr.TMAO:0.9739201219999813\n",
      "1723050067.6057043 INFO: Filtered count:0 out of 1458\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20 'R2A_C_H2O.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:11:34+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '43af92228571366e6563137926d618d1'\n",
      " 7017714 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050068.4651392 INFO: metabolites 942\n",
      "1723050068.5172155 INFO: cpd05178 not found in model!\n",
      "1723050068.5214508 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050068.6871946 INFO: Expansion time:empty:1.0799798400003056\n",
      "1723050068.6895611 INFO: Filtered count:0 out of 1458\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.4.pg.G.D GF: 11349 Prob: 3936 Other: 3067\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050068.908566 INFO: reactions 870\n",
      "1723050069.0487242 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050069.544427 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/458/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050072.8374949 INFO: cpd05178 not found in model!\n",
      "1723050072.8398814 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050073.0915787 INFO: Expansion time:Ac.NO2:0.6556728639998255\n",
      "1723050073.0936644 INFO: Filtered count:0 out of 1740\n",
      "1723050073.4526713 INFO: cpd05178 not found in model!\n",
      "1723050073.4546933 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050073.751574 INFO: Expansion time:Ac.O2:0.6566757800001142\n",
      "1723050073.7536633 INFO: Filtered count:0 out of 1740\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[42 'R2_B_D1.22.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:15:54+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '98853acf9d4027ac821576fc623f12e9'\n",
      " 9981510 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050074.4003358 INFO: Expansion time:Etho.O2:0.6454750259999855\n",
      "1723050074.4025013 INFO: Filtered count:0 out of 1740\n",
      "1723050074.4881508 INFO: metabolites 1143\n",
      "1723050074.8445175 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050075.1745112 INFO: reactions 1075\n",
      "1723050075.3107216 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050075.400672 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050076.0401242 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/574/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050076.3183527 INFO: Expansion time:For.NO2:1.914716541999951\n",
      "1723050076.320423 INFO: Filtered count:0 out of 1740\n",
      "1723050076.4923527 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050076.962516 INFO: Expansion time:For.O2:0.640949623000779\n",
      "1723050076.9645183 INFO: Filtered count:0 out of 1740\n",
      "1723050077.2178864 INFO: Objective with gapfill database:209.8048604479684; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[39 'SF2_A_H2O.14.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:15:22+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '377b0e43782392d9ed44cdd184ccd496'\n",
      " 7796059 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050077.4944475 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050077.7202454 INFO: Expansion time:Fum:0.7545867460003137\n",
      "1723050077.7222693 INFO: Filtered count:0 out of 1740\n",
      "1723050077.751258 INFO: metabolites 973\n",
      "1723050078.2491102 INFO: reactions 903\n",
      "1723050078.3636327 INFO: cpd05178 not found in model!\n",
      "1723050078.373589 INFO: Expansion time:Fum.O2:0.6501038489996063\n",
      "1723050078.3756027 INFO: Filtered count:0 out of 1740\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050078.4185874 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050078.6063972 INFO: cpd05178 not found in model!\n",
      "1723050078.6084943 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050078.9661746 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/547/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050079.0389671 INFO: Expansion time:LLac.O2:0.6621346440006164\n",
      "1723050079.0409098 INFO: Filtered count:0 out of 1740\n",
      "1723050079.6862721 INFO: Expansion time:Dlac.O2:2.2984533110002303\n",
      "1723050079.6880395 INFO: Expansion time:Pyr:0.6459880860002158\n",
      "1723050079.68887 INFO: Filtered count:0 out of 1335\n",
      "1723050079.6897933 INFO: Filtered count:0 out of 1740\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050080.3330286 INFO: Expansion time:Pyr.DMSO:0.6420654720004677\n",
      "1723050080.3350334 INFO: Filtered count:0 out of 1740\n",
      "1723050080.5713787 INFO: cpd05178 not found in model!\n",
      "1723050080.573565 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050080.6056714 INFO: Objective with gapfill database:209.8848831860838; min objective:0.01\n",
      "1723050080.6843224 INFO: Expansion time:Etho.O2:0.993910228000459\n",
      "1723050080.686858 INFO: Filtered count:0 out of 1335\n",
      "1723050080.8674269 INFO: cpd05178 not found in model!\n",
      "1723050080.8694293 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050080.997953 INFO: Expansion time:Pyr.NO:0.6618125269997108\n",
      "1723050081.0001915 INFO: Filtered count:0 out of 1740\n",
      "1723050081.4145641 INFO: Expansion time:Pyr:0.7261055310000302\n",
      "1723050081.416727 INFO: Filtered count:0 out of 1335\n",
      "1723050082.0429647 INFO: Expansion time:Pyr.DMSO:0.6250810480005384\n",
      "1723050082.04515 INFO: Filtered count:0 out of 1335\n",
      "1723050082.1091847 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050082.1265175 INFO: cpd05178 not found in model!\n",
      "1723050082.1286764 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050082.4454722 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050082.654937 INFO: Expansion time:Pyr.NO:0.6084045490006247\n",
      "1723050082.657173 INFO: Filtered count:0 out of 1335\n",
      "1723050083.2631736 INFO: Expansion time:Pyr.NO2:0.6046529440000086\n",
      "1723050083.2654648 INFO: Filtered count:0 out of 1335\n",
      "1723050083.404901 INFO: Expansion time:Pyr.NO2:2.403310781000073\n",
      "1723050083.4072487 INFO: Filtered count:0 out of 1740\n",
      "1723050083.88015 INFO: Expansion time:Pyr.NO3:0.6134011759995701\n",
      "1723050083.88238 INFO: Filtered count:0 out of 1335\n",
      "1723050084.327632 INFO: Expansion time:Pyr.NO3:0.9189575230002447\n",
      "1723050084.3297071 INFO: Filtered count:0 out of 1740\n",
      "1723050085.2541118 INFO: Expansion time:Pyr.O2:0.9229853880005976\n",
      "1723050085.256641 INFO: Filtered count:0 out of 1740\n",
      "1723050085.5932207 INFO: Expansion time:Pyr.O2:1.7096427699998458\n",
      "1723050085.5955596 INFO: Filtered count:0 out of 1335\n",
      "1723050086.1865683 INFO: Expansion time:Pyr.SO3:0.9284968379997736\n",
      "1723050086.1889179 INFO: Filtered count:0 out of 1740\n",
      "1723050086.220098 INFO: Expansion time:Pyr.SO3:0.6231326970000737\n",
      "1723050086.2221177 INFO: Filtered count:0 out of 1335\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-64\n",
      "[257 'R2A_C_D2.41.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:58:34+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9fe6b2022b800b02f17348227cc66788'\n",
      " 10644279 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050086.7531385 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050086.8362746 INFO: Expansion time:Pyr.SO4:0.6129011699995317\n",
      "1723050086.8385334 INFO: Filtered count:0 out of 1335\n",
      "1723050087.1170948 INFO: Expansion time:Pyr.SO4:0.9267153139999209\n",
      "1723050087.1194565 INFO: Filtered count:0 out of 1740\n",
      "1723050087.2114487 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050087.4599519 INFO: metabolites 1113\n",
      "1723050087.4764838 INFO: Filtered count:0 out of 1335\n",
      "1723050087.4743805 INFO: Expansion time:Pyr.TMAO:0.6346864410006674\n",
      "1723050087.9353607 INFO: reactions 1103\n",
      "1723050088.045144 INFO: Expansion time:Pyr.TMAO:0.9242354829993928\n",
      "1723050088.0474744 INFO: Filtered count:0 out of 1740\n",
      "1723050088.0731707 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050088.0875869 INFO: Expansion time:empty:0.6099054770002112\n",
      "1723050088.089606 INFO: Filtered count:0 out of 1335\n",
      "1723050088.5138001 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/80/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050088.9164252 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/80/1\"\n",
      "}\n",
      "1723050088.9829016 INFO: Filtered count:0 out of 1740\n",
      "1723050088.9806905 INFO: Expansion time:empty:0.9318202979993657\n",
      "1723050089.1932359 INFO: cpd05178 not found in model!\n",
      "1723050089.1956875 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050089.5673144 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050089.569667 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1.42.pg.G.D\n",
      " GF: 11394 Prob: 4014 Other: 2946"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050090.7448711 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050091.2707975 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050093.5944426 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050094.3710215 INFO: cpd05178 not found in model!\n",
      "1723050094.373829 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_H2O.16.pg.G.D GF: 12361 Prob: 2766 Other: 3225\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050096.065064 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050097.6415222 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050098.6177137 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[82 'R1_B_D2.30.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:23:48+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '878e43a45b00b24a0ba970ab629fbc73'\n",
      " 11061860 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050099.2906864 INFO: cpd05178 not found in model!\n",
      "1723050099.3673623 INFO: metabolites 1095\n",
      "1723050099.51684 INFO: cpd05178 not found in model!\n",
      "1723050099.5191875 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Getting tests from attributes[194 'R2A_A_D2.37.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:45:55+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '8d86c1ea3dd7c59f54eb080b7932d2fc'\n",
      " 8668841 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050100.0553575 INFO: reactions 1037\n",
      "1723050100.216299 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050100.3832629 INFO: metabolites 961\n",
      "1723050100.814407 INFO: reactions 887\n",
      "1723050100.819756 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050100.924034 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/186/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050101.2691605 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/292/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050102.2014358 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050102.8058 INFO: cpd05178 not found in model!\n",
      "1723050102.8080359 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050102.9744515 INFO: cpd05178 not found in model!\n",
      "1723050103.0899222 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050103.1928463 INFO: cpd05178 not found in model!\n",
      "1723050103.1950648 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[126 'R2_B_D2.45.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:32:37+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'edca6476b1313ca03ee60e359e80f887'\n",
      " 11276252 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050104.7968602 INFO: metabolites 1076\n",
      "1723050105.4346569 INFO: reactions 1030\n",
      "1723050105.6049213 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050105.814942 INFO: cpd05178 not found in model!\n",
      "1723050105.8173254 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_H2O.14.pg.G.D GF: 12162 Prob: 3108 Other: 3084\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050106.2622995 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/495/1\n",
      "ForkPoolWorker-65\n",
      "[261 'R2_restored_DShore.27.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:59:21+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '3aa082ea15340ee8db43cb8b009aae86'\n",
      " 10638760 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050107.330053 INFO: metabolites 1035\n",
      "1723050107.677095 INFO: reactions 930\n",
      "1723050107.7877717 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050108.136464 INFO: cpd05178 not found in model!\n",
      "1723050108.1389787 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050108.157273 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/86/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050108.2965891 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/86/1\"\n",
      "}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " \n",
      "2969 Other: R2_B_D1.22.pg.G.D GF:11241 Prob: 4142"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050113.0484166 INFO: Objective with gapfill database:144.1769671636279; min objective:0.01\n",
      "1723050113.541118 INFO: cpd05178 not found in model!\n",
      "1723050113.5431979 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.41.pg.G.D GF: 11494 Prob: 3800 Other: 3060\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050115.4006104 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050117.0852714 INFO: cpd05178 not found in model!\n",
      "1723050117.087501 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050117.4101202 INFO: Expansion time:Ac.O2:0.6195096720002766\n",
      "1723050117.4123387 INFO: Filtered count:0 out of 1371\n",
      "1723050118.073124 INFO: Expansion time:Etho.O2:0.6594712169999184\n",
      "1723050118.075323 INFO: Filtered count:0 out of 1371\n",
      "1723050118.6871686 INFO: Expansion time:Pyr:0.6105627320002895\n",
      "1723050118.6893806 INFO: Filtered count:0 out of 1371\n",
      "1723050119.3175356 INFO: Expansion time:Pyr.DMSO:0.6269271299997854\n",
      "1723050119.3198752 INFO: Filtered count:0 out of 1371\n",
      "1723050119.5240486 INFO: cpd05178 not found in model!\n",
      "1723050119.526564 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050120.2800293 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050120.6127045 INFO: Expansion time:Pyr.NO:1.2916081549992668\n",
      "1723050120.6149502 INFO: Filtered count:0 out of 1371\n",
      "1723050120.6484249 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050121.1819053 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050121.231816 INFO: Expansion time:Pyr.NO2:0.6157235099999525\n",
      "1723050121.233868 INFO: Filtered count:0 out of 1371\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050121.855686 INFO: Expansion time:Pyr.NO3:0.6205982880001102\n",
      "1723050121.857905 INFO: Filtered count:0 out of 1371\n",
      "1723050122.484642 INFO: Expansion time:Pyr.O2:0.6255327279995981\n",
      "1723050122.4869673 INFO: Filtered count:0 out of 1371\n",
      "1723050122.6224072 INFO: cpd05178 not found in model!\n",
      "1723050122.6245463 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050122.648266 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050122.8069744 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050123.367893 INFO: Expansion time:Pyr.SO3:0.8795613969996339\n",
      "1723050123.3705025 INFO: Filtered count:0 out of 1371\n",
      "1723050123.7474592 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050124.8216486 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050125.0550501 INFO: Expansion time:Pyr.SO4:1.6830712450000647\n",
      "1723050125.0574641 INFO: Filtered count:0 out of 1371\n",
      "1723050125.6695292 INFO: cpd05178 not found in model!\n",
      "1723050125.6717749 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050125.7136729 INFO: Expansion time:Ac.O2:0.6238158559999647\n",
      "1723050125.7157512 INFO: Filtered count:0 out of 1262\n",
      "1723050125.936386 INFO: Expansion time:Pyr.TMAO:0.8773194229997898\n",
      "1723050125.9388413 INFO: Filtered count:0 out of 1371\n",
      "1723050126.0618625 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050126.3456597 INFO: Expansion time:Etho.O2:0.6287488870002562\n",
      "1723050126.3477874 INFO: Filtered count:0 out of 1262\n",
      "1723050126.5859044 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050126.7875676 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050126.995591 INFO: Expansion time:For.O2:0.6466945809997924\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050126.997765 INFO: Filtered count:0 out of 1262\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050127.133649 INFO: Expansion time:empty:1.1932311470000059\n",
      "1723050127.136334 INFO: Filtered count:0 out of 1371\n",
      "1723050127.1402607 INFO: cpd05178 not found in model!\n",
      "1723050127.1420898 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050127.2089078 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050127.6659033 INFO: Expansion time:Fum:0.6668026830002418\n",
      "1723050127.6680627 INFO: Filtered count:0 out of 1262\n",
      "1723050128.090794 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050128.2905078 INFO: cpd05178 not found in model!\n",
      "1723050128.2928822 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[10 'R2_B_D2.50.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:09:38+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '0a01ffbbd6d59ff841b8749d9f8f79e3'\n",
      " 10736936 None]\n",
      "ForkPoolWorker-66\n",
      "[265 'R1_B_H2O.22.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:00:03+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '5a424dd6c40132bb71d00581b2d50eb5'\n",
      " 7029247 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050129.8999832 INFO: Expansion time:Fum.O2:2.230729441999756\n",
      "1723050129.902059 INFO: Filtered count:0 out of 1262\n",
      "1723050130.1171439 INFO: metabolites 1139\n",
      "1723050130.310095 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050130.5401037 INFO: Expansion time:Glc:0.636821166000118\n",
      "1723050130.542331 INFO: Filtered count:0 out of 1262\n",
      "1723050130.6831427 INFO: reactions 1101\n",
      "1723050130.708688 INFO: metabolites 969\n",
      "1723050130.8484254 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050131.1713126 INFO: Expansion time:Glc.DMSO:0.6277662229995258\n",
      "1723050131.1733623 INFO: Filtered count:0 out of 1262\n",
      "1723050131.24294 INFO: reactions 889\n",
      "1723050131.296739 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/378/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050131.4045308 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050131.7520788 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050131.7915518 INFO: Filtered count:0 out of 1262\n",
      "1723050131.7894292 INFO: Expansion time:Glc.O2:0.6148631420001038\n",
      "1723050131.9259725 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/119/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050132.1335442 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/119/1\"\n",
      "}\n",
      "1723050132.422368 INFO: Expansion time:Glc.TMAO:0.6296851600000082\n",
      "1723050132.4246142 INFO: Filtered count:0 out of 1262\n",
      "1723050133.048166 INFO: Expansion time:Pyr:0.6224152869999671\n",
      "1723050133.0502663 INFO: Filtered count:0 out of 1262\n",
      "1723050133.2640562 INFO: Expansion time:Etho.O2:0.9840004699999554\n",
      "1723050133.2661004 INFO: Filtered count:0 out of 1544\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050133.706446 INFO: Expansion time:Pyr.DMSO:0.6550116270000217\n",
      "1723050133.708593 INFO: Filtered count:0 out of 1262\n",
      "1723050134.3511517 INFO: Expansion time:Pyr.NO:0.641345316999832\n",
      "1723050134.353383 INFO: Filtered count:0 out of 1262\n",
      "1723050134.6116931 INFO: Expansion time:Fum:1.3442775469993649\n",
      "1723050134.6137931 INFO: Filtered count:0 out of 1544\n",
      "1723050134.9755998 INFO: Expansion time:Pyr.NO2:0.6210379320000357\n",
      "1723050134.9776967 INFO: Filtered count:0 out of 1262\n",
      "1723050135.297649 INFO: Expansion time:Fum.O2:0.682589562999965\n",
      "1723050135.299653 INFO: Filtered count:0 out of 1544\n",
      "1723050135.5962136 INFO: Expansion time:Pyr.NO3:0.6172554320000927\n",
      "1723050135.5985174 INFO: Filtered count:0 out of 1262\n",
      "1723050135.9612215 INFO: Expansion time:Pyr:0.6602644819995476\n",
      "1723050135.9631462 INFO: Filtered count:0 out of 1544\n",
      "1723050136.093513 INFO: Expansion time:Ac.NO2:2.509234944999662\n",
      "1723050136.095901 INFO: Filtered count:0 out of 1500\n",
      "1723050136.638349 INFO: Expansion time:Pyr.DMSO:0.6739658090000376\n",
      "1723050136.6404123 INFO: Filtered count:0 out of 1544\n",
      "1723050136.8891072 INFO: cpd05178 not found in model!\n",
      "1723050136.8911858 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050137.0848222 INFO: cpd05178 not found in model!\n",
      "1723050137.087097 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050137.1092114 INFO: Expansion time:Ac.O2:1.011808034999376\n",
      "1723050137.111594 INFO: Filtered count:0 out of 1500\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.27.pg.G.D GF: 12080 Prob: 2948 Other: 3324\n",
      "R2A_A_D2.37.pg.G.D GF: 11763 Prob: 3622 Other: 2969\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050137.5104074 INFO: cpd05178 not found in model!\n",
      "1723050137.5127106 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.45.pg.G.D GF: 11772 Prob: 3606 Other: 2976\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050138.0533378 INFO: Expansion time:Pyr.NO:1.4116220780006188\n",
      "1723050138.0552354 INFO: Filtered count:0 out of 1544\n",
      "1723050138.1283584 INFO: Expansion time:Etho.O2:1.0152891919997273\n",
      "1723050138.1307588 INFO: Filtered count:0 out of 1500\n",
      "1723050138.1602726 INFO: cpd05178 not found in model!\n",
      "1723050138.1628742 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050138.4543037 INFO: Expansion time:Pyr.O2:2.854207953000696\n",
      "1723050138.4563892 INFO: Filtered count:0 out of 1262\n",
      "1723050138.7244496 INFO: Expansion time:Pyr.NO2:0.6680175299998155\n",
      "1723050138.726137 INFO: Filtered count:0 out of 1544\n",
      "1723050139.0941098 INFO: Expansion time:Pyr.SO3:0.6364610309992713\n",
      "1723050139.096333 INFO: Filtered count:0 out of 1262\n",
      "1723050139.111944 INFO: Expansion time:For.NO2:0.9796868109997376\n",
      "1723050139.1142876 INFO: Filtered count:0 out of 1500\n",
      "1723050139.4156756 INFO: Expansion time:Pyr.NO3:0.6883488869998473\n",
      "1723050139.4175308 INFO: Filtered count:0 out of 1544\n",
      "1723050139.7274234 INFO: Expansion time:Pyr.SO4:0.6298342649997721\n",
      "1723050139.729562 INFO: Filtered count:0 out of 1262\n",
      "1723050140.0191195 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050140.0721 INFO: Expansion time:For.O2:0.9564169449995461\n",
      "1723050140.0745022 INFO: Filtered count:0 out of 1500\n",
      "1723050140.0925834 INFO: cpd05178 not found in model!\n",
      "1723050140.106535 INFO: Expansion time:Pyr.O2:0.6878095619995293\n",
      "1723050140.1083376 INFO: Filtered count:0 out of 1544\n",
      "1723050140.3382938 INFO: cpd05178 not found in model!\n",
      "1723050140.3402214 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050140.373673 INFO: Expansion time:Pyr.TMAO:0.6427578389993869\n",
      "1723050140.3758054 INFO: Filtered count:0 out of 1262\n",
      "1723050140.6091497 INFO: cpd05178 not found in model!\n",
      "1723050140.6112812 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050140.709059 INFO: cpd05178 not found in model!\n",
      "1723050140.8525798 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.30.pg.G.D GF: 11234 Prob: 4190 Other: 2930\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050140.8763468 INFO: Expansion time:Pyr.SO3:0.7668641690006552\n",
      "1723050140.8783512 INFO: Filtered count:0 out of 1544\n",
      "1723050140.9314032 INFO: cpd05178 not found in model!\n",
      "1723050140.933564 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050141.0080295 INFO: Expansion time:empty:0.6310829679996459\n",
      "1723050141.0101485 INFO: Filtered count:0 out of 1262\n",
      "1723050141.0339077 INFO: Expansion time:Pyr:0.9579046290000406\n",
      "1723050141.0362773 INFO: Filtered count:0 out of 1500\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050142.0485961 INFO: Expansion time:Pyr.DMSO:1.0107953859996996\n",
      "1723050142.0509775 INFO: Filtered count:0 out of 1500\n",
      "1723050142.2410076 INFO: Expansion time:Pyr.SO4:1.3613277800004653\n",
      "1723050142.2430084 INFO: Filtered count:0 out of 1544\n",
      "1723050142.9111924 INFO: Expansion time:Pyr.TMAO:0.6668603979996988\n",
      "1723050142.9130883 INFO: Filtered count:0 out of 1544\n",
      "1723050143.0311012 INFO: Expansion time:Pyr.NO:0.978670688999955\n",
      "1723050143.0335166 INFO: Filtered count:0 out of 1500\n",
      "1723050143.6118224 INFO: Expansion time:empty:0.6975054639997325\n",
      "1723050143.6137385 INFO: Filtered count:0 out of 1544\n",
      "1723050144.2369003 INFO: cpd05178 not found in model!\n",
      "1723050144.239145 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050144.5211556 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050145.654478 INFO: Expansion time:Pyr.NO2:2.6194172759996945\n",
      "1723050145.6568015 INFO: Filtered count:0 out of 1500\n",
      "1723050145.844076 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050146.3032014 INFO: Expansion time:Pyr.NO3:0.6450497229998291\n",
      "1723050146.3053205 INFO: Filtered count:0 out of 1500\n",
      "1723050146.314104 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050146.9355078 INFO: Expansion time:Pyr.O2:0.6289882330002001\n",
      "1723050146.937767 INFO: Filtered count:0 out of 1500\n",
      "1723050147.5836737 INFO: Expansion time:Pyr.SO3:0.6447054509999361\n",
      "1723050147.5858426 INFO: Filtered count:0 out of 1500\n",
      "1723050148.1945262 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050148.2284863 INFO: Filtered count:0 out of 1500\n",
      "1723050148.2265005 INFO: Expansion time:Pyr.SO4:0.6394191050003428\n",
      "1723050148.329975 INFO: cpd05178 not found in model!\n",
      "1723050148.3322365 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050148.8526409 INFO: Expansion time:Pyr.TMAO:0.623009727999488\n",
      "1723050148.8548324 INFO: Filtered count:0 out of 1500\n",
      "1723050149.487494 INFO: Expansion time:empty:0.6314875369998845\n",
      "1723050149.4896414 INFO: Filtered count:0 out of 1500\n",
      "1723050151.0782533 INFO: Objective with gapfill database:234.38073447320266; min objective:0.01\n",
      "1723050151.710267 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050152.6247637 INFO: Objective with gapfill database:240.60730228649268; min objective:0.01\n",
      "1723050152.9985514 INFO: Objective with gapfill database:253.08223143019262; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-67\n",
      "[269 'R2A_A_D1.6.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:00:51+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'adb03c0a62f7edb12858953a93904696'\n",
      " 16227804 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050155.101778 INFO: cpd05178 not found in model!\n",
      "1723050155.1038458 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_H2O.22.pg.G.D GF: 12080 Prob: 3228 Other: 3046\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050155.4222622 INFO: metabolites 1241\n",
      "1723050156.203898 INFO: reactions 1226\n",
      "1723050156.4097402 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050157.351948 INFO: Default biomass: [bio1]\n",
      "1723050157.4328618 INFO: Expansion time:Etho.O2:0.6017633230003412\n",
      "1723050157.4351115 INFO: Filtered count:0 out of 1241\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/128/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050157.5723863 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/128/1\"\n",
      "}\n",
      "1723050158.372204 INFO: Expansion time:For.O2:0.9357326750005086\n",
      "1723050158.374798 INFO: Filtered count:0 out of 1241\n",
      "1723050159.3165448 INFO: Expansion time:Pyr.O2:0.9403928889996678\n",
      "1723050159.319121 INFO: Filtered count:0 out of 1241\n",
      "1723050160.2319033 INFO: Expansion time:empty:0.9113057269996716\n",
      "1723050160.2346952 INFO: Filtered count:0 out of 1241\n",
      "1723050160.3527505 INFO: cpd05178 not found in model!\n",
      "1723050160.3547077 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050162.5193534 INFO: cpd05178 not found in model!\n",
      "1723050162.5222523 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050162.5371654 INFO: cpd05178 not found in model!\n",
      "1723050162.8751001 INFO: cpd05178 not found in model!\n",
      "1723050162.8778226 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.50.pg.G.D GF: 11383 Prob: 3990 Other: 2981\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050163.275161 INFO: cpd05178 not found in model!\n",
      "1723050163.2770698 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[210 'R1_A_D1.9.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:49:06+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '3585b033869eb5e1060d43a71feeba4f'\n",
      " 7002392 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050165.502308 INFO: metabolites 1047\n",
      "1723050165.7750723 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050165.9713397 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050166.128451 INFO: reactions 986\n",
      "1723050166.3110585 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050166.921952 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050166.9236975 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/412/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050167.5016375 INFO: cpd05178 not found in model!\n",
      "1723050167.5038571 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050169.8791642 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[50 'R1_C_D1.27.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:17:21+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '7a59d3f344c6ccd8977ddb1dee1ede29'\n",
      " 10743259 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050170.1191463 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050170.1172342 INFO: cpd05178 not found in model!\n",
      "1723050170.4668798 INFO: metabolites 1024\n",
      "1723050170.8990881 INFO: reactions 979\n",
      "1723050171.0144548 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050171.4031508 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/602/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050173.158351 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050173.2377512 INFO: cpd05178 not found in model!\n",
      "1723050173.2400913 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050173.7796314 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050174.3444457 INFO: cpd05178 not found in model!\n",
      "1723050174.5895894 INFO: cpd05178 not found in model!\n",
      "1723050174.591837 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050174.8321977 INFO: cpd05178 not found in model!\n",
      "1723050174.8343616 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050174.854818 INFO: cpd05178 not found in model!\n",
      "1723050174.8567326 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050176.4076836 INFO: Expansion time:Dlac.O2:0.6819468159992539\n",
      "1723050176.409778 INFO: Filtered count:0 out of 1229\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-68\n",
      "[273 'R2A_A_D2.7.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:01:38+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '2668b0a86dd49f50f1911cecceb364d7'\n",
      " 15021898 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050177.7060232 INFO: Expansion time:Etho.O2:1.2948450309995678\n",
      "1723050177.707958 INFO: Filtered count:0 out of 1229\n",
      "1723050177.8830705 INFO: metabolites 1190\n",
      "1723050178.349011 INFO: Expansion time:Fum:0.6398851549993196\n",
      "1723050178.351032 INFO: Filtered count:0 out of 1229\n",
      "1723050178.4634213 INFO: cpd05178 not found in model!\n",
      "1723050178.5073636 INFO: reactions 1150\n",
      "1723050178.7092764 INFO: cpd05178 not found in model!\n",
      "1723050178.711521 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050178.764923 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050178.9865227 INFO: Expansion time:Fum.O2:0.6341998240004614\n",
      "1723050178.9885132 INFO: Filtered count:0 out of 1229\n",
      "1723050179.1174366 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050179.6380665 INFO: Expansion time:empty:0.6483069910000268\n",
      "1723050179.6401517 INFO: Filtered count:0 out of 1229\n",
      "1723050179.7543077 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/150/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050180.0050368 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/150/1\"\n",
      "}\n",
      "1723050180.0613208 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050180.0591247 INFO: cpd05178 not found in model!\n",
      "1723050181.0670624 INFO: cpd05178 not found in model!\n",
      "1723050181.0698655 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050181.2163417 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050182.670819 INFO: cpd05178 not found in model!\n",
      "1723050182.673337 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050182.8751216 INFO: Expansion time:Ac.O2:0.6563367519993335\n",
      "1723050182.8773966 INFO: Filtered count:0 out of 1572\n",
      "1723050183.5204062 INFO: Expansion time:Etho.O2:0.6418021189992942\n",
      "1723050183.5227299 INFO: Filtered count:0 out of 1572\n",
      "1723050184.165353 INFO: Expansion time:For.O2:0.6414358239999274\n",
      "1723050184.1676877 INFO: Filtered count:0 out of 1572\n",
      "1723050184.200208 INFO: cpd05178 not found in model!\n",
      "1723050184.202574 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050184.6782162 INFO: cpd05178 not found in model!\n",
      "1723050184.8106837 INFO: Expansion time:Pyr:0.6417451580000488\n",
      "1723050184.812996 INFO: Filtered count:0 out of 1572\n",
      "1723050184.899616 INFO: cpd05178 not found in model!\n",
      "1723050184.9018266 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050185.4524333 INFO: Expansion time:Pyr.DMSO:0.6382439369999702\n",
      "1723050185.454754 INFO: Filtered count:0 out of 1572\n",
      "1723050185.7428062 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050185.8509579 INFO: cpd05178 not found in model!\n",
      "1723050185.8533723 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050186.0923567 INFO: Expansion time:Pyr.NO:0.6363473239998712\n",
      "1723050186.0947475 INFO: Filtered count:0 out of 1572\n",
      "1723050186.196041 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1.6.pg.G.D GF: 11280 Prob: 4096 Other: 2976\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050186.3612888 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050186.7349687 INFO: Expansion time:Pyr.NO2:0.6390200159994492\n",
      "1723050186.7372983 INFO: Filtered count:0 out of 1572\n",
      "1723050187.3783903 INFO: Expansion time:Pyr.NO3:0.6398425329998645\n",
      "1723050187.3807607 INFO: Filtered count:0 out of 1572\n",
      "1723050188.4598687 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050189.034208 INFO: cpd05178 not found in model!\n",
      "1723050189.0364587 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050189.4561992 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050189.6376145 INFO: Expansion time:Pyr.O2:2.2556615169996803\n",
      "1723050189.6400273 INFO: Filtered count:0 out of 1572\n",
      "1723050190.2786193 INFO: Expansion time:Pyr.SO3:0.6373177830000714\n",
      "1723050190.2808995 INFO: Filtered count:0 out of 1572\n",
      "1723050190.3794916 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Getting tests from attributes[222 'R2_restored_C_black.13.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:51:28+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '1585949287d09c117a83968f550ad68b'\n",
      " 13008229 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050190.922522 INFO: Expansion time:Pyr.SO4:0.6404594800005725\n",
      "1723050190.9247642 INFO: Filtered count:0 out of 1572\n",
      "1723050191.1876118 INFO: metabolites 1048\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050191.5629683 INFO: Expansion time:Pyr.TMAO:0.6369659679994584\n",
      "1723050191.5652058 INFO: Filtered count:0 out of 1572\n",
      "1723050191.745754 INFO: reactions 999\n",
      "1723050192.2044663 INFO: Expansion time:empty:0.6379820089996429\n",
      "1723050192.2066588 INFO: Filtered count:0 out of 1572\n",
      "1723050192.518988 INFO: cpd05178 not found in model!\n",
      "1723050192.5215178 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050193.00442 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050193.5572085 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/489/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050194.0426686 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[206 'R2_B_D2.43.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:48:25+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '4734b334595f39733eb4c0e4c992312b'\n",
      " 11702113 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050195.57353 INFO: metabolites 1106\n",
      "1723050197.410066 INFO: reactions 1007\n",
      "1723050197.5911179 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050198.080575 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/374/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050198.4712212 INFO: cpd05178 not found in model!\n",
      "1723050198.4733577 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050198.5710242 INFO: cpd05178 not found in model!\n",
      "1723050198.5749364 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-69\n",
      "[277 'R1_A_D2.25.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:02:21+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '94ced4b8a8791e95e52bbf45bef18c41'\n",
      " 7558329 None]\n",
      "R1_C_D1.27.pg.G.D GF: 11806 Prob: 3466 Other: 3080\n",
      "R1_A_D1.9.pg.G.D GF: 11514 Prob: 3608 Other: 3228\n",
      "[36 'SF2_B_H2O.28.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:14:53+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'faf2355a3d7d5f2403dffe44480a1a2c'\n",
      " 5823502 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050199.438877 INFO: metabolites 937\n",
      "1723050199.6470325 INFO: metabolites 912\n",
      "1723050199.946942 INFO: reactions 920\n",
      "1723050200.0888388 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050200.1434476 INFO: reactions 843\n",
      "1723050200.3008375 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050200.6403074 INFO: Default biomass: [bio1]\n",
      "1723050200.7820776 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/212/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050200.8298156 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/212/1\"\n",
      "}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/531/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050200.9327888 INFO: cpd05178 not found in model!\n",
      "1723050201.1643965 INFO: cpd05178 not found in model!\n",
      "1723050201.1663747 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050201.58824 INFO: cpd05178 not found in model!\n",
      "1723050201.5907297 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050201.7354486 INFO: Objective with gapfill database:297.1016892330733; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[59 'R2A_B_D2.6.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:19:29+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ad5ce5c37c39c9ad54903ff20a5e3b1e'\n",
      " 14623121 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050203.3030033 INFO: metabolites 1106\n",
      "1723050203.3890593 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050203.8037326 INFO: reactions 1071\n",
      "1723050203.832217 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050203.9269302 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050204.3532555 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/620/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050204.534738 INFO: cpd05178 not found in model!\n",
      "1723050204.537043 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D2.7.pg.G.D GF: 11470 Prob: 3816 Other: 3068\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050207.977969 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050208.4949129 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050209.0944917 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050209.2385297 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050209.6415508 INFO: cpd05178 not found in model!\n",
      "1723050209.64381 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050210.0782166 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050210.5190265 INFO: cpd05178 not found in model!\n",
      "1723050210.5213826 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050210.9996226 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050212.7235498 INFO: Objective with gapfill database:170.5148914319508; min objective:0.01\n",
      "1723050214.29379 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050214.5615265 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050216.4612646 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050216.8597457 INFO: cpd05178 not found in model!\n",
      "1723050216.8628964 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050217.0193393 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050217.4449782 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[70 'R1_C_H2O.67.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:21:26+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '53766813a6f86644b6524422b4ea322f'\n",
      " 5552828 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050219.687621 INFO: cpd05178 not found in model!\n",
      "1723050219.6898844 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050219.8398714 INFO: metabolites 962\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_C_black.13.pg.G.D GF: 11714 Prob: 3542 Other: 3096\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050220.1967916 INFO: reactions 916\n",
      "1723050220.3079305 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050220.6534317 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/116/1\n",
      "ForkPoolWorker-70\n",
      "[281 'R1_C_D2.36.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:03:07+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ffdc92b816d1ea82520f56a3b85c72e0'\n",
      " 19881298 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050221.5710485 INFO: cpd05178 not found in model!\n",
      "1723050221.5735319 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050221.6188035 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050221.8595638 INFO: metabolites 1152\n",
      "1723050222.4885333 INFO: reactions 1086\n",
      "1723050222.7090847 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050222.9643016 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050223.3732862 INFO: Default biomass: [bio1]\n",
      "1723050223.458972 INFO: cpd05178 not found in model!\n",
      "1723050223.4611986 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050223.466245 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_H2O.28.pg.G.D GF: 12223 Prob: 2892 Other: 3237\n",
      "188040/231/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050223.9476666 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/231/1\"\n",
      "}\n",
      "1723050224.7581024 INFO: cpd05178 not found in model!\n",
      "1723050224.7604678 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.25.pg.G.D GF: 12150 Prob: 2812 Other: 3388\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050225.086078 INFO: cpd05178 not found in model!\n",
      "1723050225.088246 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050226.7842999 INFO: Expansion time:Glc:1.1105215150000731\n",
      "1723050226.786918 INFO: Filtered count:0 out of 1389\n",
      "1723050227.257648 INFO: cpd05178 not found in model!\n",
      "1723050227.2599626 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050227.3013878 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050227.5981755 INFO: Expansion time:Ac.NO2:1.946676452000247\n",
      "1723050227.6003377 INFO: Filtered count:0 out of 1676\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050227.785394 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[131 'R2_C_D1.95.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:33:32+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '36ebf14e58d402e6af7d50cb37fb07e8'\n",
      " 15782243 None]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050227.9476442 INFO: Filtered count:0 out of 1389\n",
      "1723050228.091222 INFO: metabolites 1107\n",
      "1723050228.1820803 INFO: cpd05178 not found in model!\n",
      "1723050227.9453561 INFO: Expansion time:Glc.DMSO:1.156635796000046\n",
      "1723050228.410925 INFO: cpd05178 not found in model!\n",
      "1723050228.4128804 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050228.6901839 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050228.7693024 INFO: reactions 1042\n",
      "1723050228.9341674 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050228.9788785 INFO: Filtered count:0 out of 1676\n",
      "1723050228.9767368 INFO: Expansion time:Ac.NO3:1.3752601750002214\n",
      "1723050229.1091666 INFO: Expansion time:Glc.O2:1.1600408009999228\n",
      "1723050229.1114504 INFO: Filtered count:0 out of 1389\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050229.54927 INFO: Default biomass: [bio1]\n",
      "1723050230.201235 INFO: cpd05178 not found in model!\n",
      "1723050230.4294808 INFO: Expansion time:Ac.O2:1.449224205000064\n",
      "1723050230.4317589 INFO: Filtered count:0 out of 1676\n",
      "1723050230.4404826 INFO: cpd05178 not found in model!\n",
      "1723050230.442537 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/557/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050231.5033088 INFO: Expansion time:Dlac.O2:1.0703384629996435\n",
      "1723050231.5054686 INFO: Filtered count:0 out of 1676\n",
      "1723050231.8451633 INFO: Expansion time:Glc.TMAO:2.7321543339994605\n",
      "1723050231.8475883 INFO: Filtered count:0 out of 1389\n",
      "1723050231.9680207 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050232.2349386 INFO: cpd05178 not found in model!\n",
      "1723050232.2369761 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050232.3425732 INFO: Expansion time:Etho.O2:0.8358051129998785\n",
      "1723050232.3446712 INFO: Filtered count:0 out of 1676\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.43.pg.G.D GF: 11502 Prob: 3778 Other: 3070\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050232.672106 INFO: Objective with gapfill database:167.2722618663236; min objective:0.01\n",
      "1723050232.859761 INFO: Expansion time:empty:1.0105150230001527\n",
      "1723050232.8617756 INFO: Filtered count:0 out of 1389\n",
      "1723050233.1034625 INFO: Expansion time:For.NO2:0.7575036940006612\n",
      "1723050233.1054797 INFO: Filtered count:0 out of 1676\n",
      "1723050233.48331 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050234.4829924 INFO: Expansion time:For.NO3:1.3763124630004313\n",
      "1723050234.4851325 INFO: Filtered count:0 out of 1676\n",
      "1723050235.3490813 INFO: Objective with gapfill database:210.45422325284855; min objective:0.01\n",
      "1723050235.4458723 INFO: Expansion time:For.O2:0.9594887829998697\n",
      "1723050235.447957 INFO: Filtered count:0 out of 1676\n",
      "1723050236.6293476 INFO: Expansion time:Fum:1.1802685039992866\n",
      "1723050236.631689 INFO: Filtered count:0 out of 1676\n",
      "1723050237.3750823 INFO: cpd05178 not found in model!\n",
      "1723050237.3772256 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050237.5694554 INFO: Expansion time:Fum.O2:0.9365886089999549\n",
      "1723050237.5715215 INFO: Filtered count:0 out of 1676\n",
      "1723050239.3253455 INFO: Expansion time:Glyc.O2:1.7526803170003404\n",
      "1723050239.3276525 INFO: Filtered count:0 out of 1676\n",
      "1723050240.3299475 INFO: Expansion time:LLac.O2:1.001141285999438\n",
      "1723050240.332089 INFO: Filtered count:0 out of 1676\n",
      "1723050241.0761185 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050241.216724 INFO: Expansion time:Pyr:0.8835001719999127\n",
      "1723050241.2188544 INFO: Filtered count:0 out of 1676\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050241.966548 INFO: Expansion time:Pyr.DMSO:0.7465833449996353\n",
      "1723050241.968633 INFO: Filtered count:0 out of 1676\n",
      "1723050242.1497335 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050242.1690848 INFO: cpd05178 not found in model!\n",
      "1723050242.1716826 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_B_D2.6.pg.G.D GF: 11560 Prob: 3728 Other: 3062\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050242.7061253 INFO: Expansion time:Pyr.NO:0.7363871019997532\n",
      "1723050242.7083564 INFO: Filtered count:0 out of 1676\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "ForkPoolWorker-71\n",
      "[285 'R1_A_D1.47.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:03:53+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '08589c636dc269102dab725b390e3b89'\n",
      " 8497525 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050244.2178388 INFO: Expansion time:Pyr.NO2:1.50821070100028\n",
      "1723050244.2200704 INFO: Filtered count:0 out of 1676\n",
      "1723050244.2425096 INFO: metabolites 1012\n",
      "1723050244.7319372 INFO: cpd05178 not found in model!\n",
      "1723050244.7340887 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050244.8237255 INFO: reactions 972\n",
      "1723050244.9592104 INFO: cpd05178 not found in model!\n",
      "1723050244.9614305 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050244.9692109 INFO: Expansion time:For.NO2:3.432115395999972\n",
      "1723050244.9716127 INFO: Filtered count:0 out of 1141\n",
      "1723050244.9847252 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050245.157328 INFO: Expansion time:Pyr.NO3:0.9358775820001028\n",
      "1723050245.1594362 INFO: Filtered count:0 out of 1676\n",
      "1723050245.1635733 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_H2O.67.pg.G.D GF: 11913 Prob: 3334 Other: 3105\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050245.5722742 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/233/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050245.7808247 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/233/1\"\n",
      "}\n",
      "1723050246.099997 INFO: Expansion time:Pyr.O2:0.9394607590002124\n",
      "1723050246.1020422 INFO: Filtered count:0 out of 1676\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[62 'SF2_C_D2.15.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:20:02+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '1c5474d9282ec09910dc07e1545584ff'\n",
      " 13240711 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050246.7322662 INFO: Expansion time:For.O2:1.759045480000168\n",
      "1723050246.7344568 INFO: Filtered count:0 out of 1141\n",
      "1723050246.8737342 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050247.1019948 INFO: metabolites 1198\n",
      "1723050247.5895836 INFO: Expansion time:Pyr.SO3:1.486377922000429\n",
      "1723050247.591934 INFO: Filtered count:0 out of 1676\n",
      "1723050247.6352894 INFO: reactions 1179\n",
      "1723050247.7661042 INFO: Filtered count:0 out of 1141\n",
      "1723050247.8128397 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050247.7641797 INFO: Expansion time:Pyr:1.028482824999628\n",
      "1723050248.1249168 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050248.3737774 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050248.3934429 INFO: Expansion time:Pyr.DMSO:0.6261215900003663\n",
      "1723050248.3957725 INFO: Filtered count:0 out of 1141\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/92/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050248.558844 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050249.0356586 INFO: Filtered count:0 out of 1141\n",
      "1723050249.033477 INFO: Expansion time:Pyr.NO:0.6363284079998266\n",
      "1723050249.2220008 INFO: Filtered count:0 out of 1676\n",
      "1723050249.2197003 INFO: Expansion time:Pyr.SO4:1.626499289000094\n",
      "1723050249.6541123 INFO: Expansion time:Pyr.NO3:0.617207444000087\n",
      "1723050249.656248 INFO: Filtered count:0 out of 1141\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050250.8415139 INFO: Expansion time:Pyr.O2:1.1840669459998026\n",
      "1723050250.8436117 INFO: Filtered count:0 out of 1141\n",
      "1723050251.0960262 INFO: Expansion time:Pyr.TMAO:1.8725819709998177\n",
      "1723050251.098526 INFO: Filtered count:0 out of 1676\n",
      "1723050251.5616026 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050251.8727174 INFO: Expansion time:Pyr.SO3:1.0278553379994264\n",
      "1723050251.8748643 INFO: Filtered count:0 out of 1141\n",
      "1723050252.0349734 INFO: Expansion time:empty:0.9350650759997734\n",
      "1723050252.037317 INFO: Filtered count:0 out of 1676\n",
      "1723050252.4754622 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050252.502841 INFO: Filtered count:0 out of 1141\n",
      "1723050252.5007071 INFO: Expansion time:Pyr.SO4:0.6246376489998511\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050253.8210542 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[111 'R2_C_D2.100.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:29:33+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'bdb05f76307e1af592367ae02efacf03'\n",
      " 15423210 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050254.1781564 INFO: cpd05178 not found in model!\n",
      "1723050254.301573 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050254.412585 INFO: cpd05178 not found in model!\n",
      "1723050254.4148583 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050254.4772782 INFO: metabolites 1103\n",
      "1723050254.8902302 INFO: reactions 1064\n",
      "1723050255.0906255 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050255.2086647 INFO: cpd05178 not found in model!\n",
      "1723050255.2111118 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050255.2363877 INFO: Expansion time:Pyr.TMAO:2.7323391350000747\n",
      "1723050255.2384725 INFO: Filtered count:0 out of 1141\n",
      "1723050255.5140033 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/406/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050255.8134542 INFO: cpd05178 not found in model!\n",
      "1723050255.8156893 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050255.8533287 INFO: Expansion time:empty:0.6136575689997699\n",
      "1723050255.8557389 INFO: Filtered count:0 out of 1141\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050256.1674023 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050256.165118 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R1_C_D2.36.pg.G.D GF: 11558Prob: 3734 Other: 3062"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050257.19368 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050257.3470733 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050258.9992409 INFO: Expansion time:Dlac.O2:0.6966584780002449\n",
      "1723050259.0012982 INFO: Filtered count:0 out of 1440\n",
      "1723050259.6823235 INFO: Expansion time:Etho.O2:0.6796790279995548\n",
      "1723050259.6843343 INFO: Filtered count:0 out of 1440\n",
      "1723050260.336888 INFO: cpd05178 not found in model!\n",
      "1723050260.3391907 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050260.3672144 INFO: Expansion time:Glc:0.6817404339999484\n",
      "1723050260.369091 INFO: Filtered count:0 out of 1440\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050261.5762038 INFO: cpd05178 not found in model!\n",
      "1723050261.7770696 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050261.8052976 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050261.8033295 INFO: cpd05178 not found in model!\n",
      "1723050262.235885 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050262.3629348 INFO: Expansion time:Glc.DMSO:1.992607684000177\n",
      "1723050262.3649192 INFO: Filtered count:0 out of 1440\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050263.0728974 INFO: Expansion time:Glc.O2:0.7066964760006158\n",
      "1723050263.0748074 INFO: Filtered count:0 out of 1440\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050263.7619445 INFO: Expansion time:Glc.TMAO:0.6859336819998134\n",
      "1723050263.7638478 INFO: Filtered count:0 out of 1440\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050264.0802362 INFO: cpd05178 not found in model!\n",
      "1723050264.082689 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D1.95.pg.G.D GF: 11402 Prob: 3878"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050264.4418354 INFO: Expansion time:Glyc.O2:0.6765867670001171\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050264.4435735 INFO: Filtered count:0 out of 1440\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Other: 3072\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050264.973506 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050265.121305 INFO: Expansion time:Pyr:0.6765464130003238\n",
      "1723050265.123292 INFO: Filtered count:0 out of 1440\n",
      "1723050265.8288198 INFO: Filtered count:0 out of 1440\n",
      "1723050265.8269227 INFO: Expansion time:Pyr.DMSO:0.7024198010003602\n",
      "1723050266.3164444 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050266.499252 INFO: Expansion time:Pyr.NO:0.6690959719999228\n",
      "1723050266.5013602 INFO: Filtered count:0 out of 1440\n",
      "1723050266.6407146 INFO: cpd05178 not found in model!\n",
      "1723050266.6430352 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050266.7604787 INFO: Expansion time:Dlac.O2:3.1774592229994596\n",
      "1723050266.7627485 INFO: Filtered count:0 out of 1274\n",
      "1723050267.4840958 INFO: Expansion time:Pyr.NO2:0.9813993979996667\n",
      "1723050267.4867373 INFO: Filtered count:0 out of 1440\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-72\n",
      "[289 'R2A_A_D2.10.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:04:49+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '5056d370228951a8fd4f50a680b28dc7'\n",
      " 16694068 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050268.5743446 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050268.6355412 INFO: metabolites 1137\n",
      "1723050268.781746 INFO: Expansion time:Etho.O2:2.017843866000476\n",
      "1723050268.7842047 INFO: Filtered count:0 out of 1274\n",
      "1723050269.0590436 INFO: cpd05178 not found in model!\n",
      "1723050269.0617344 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050269.4432118 INFO: reactions 1098\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R1_A_D1.47.pg.G.D GF: 12109 Prob: 2942 Other: 3299"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050269.6275935 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050269.8898354 INFO: Expansion time:Pyr.NO3:2.401395367999612\n",
      "1723050269.8918447 INFO: Filtered count:0 out of 1440\n",
      "1723050270.5715258 INFO: Default biomass: [bio1]\n",
      "1723050270.6659412 INFO: Expansion time:Pyr.O2:0.7727848009999434\n",
      "1723050270.6679294 INFO: Filtered count:0 out of 1440\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/312/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050270.7703598 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/312/1\"\n",
      "}\n",
      "1723050271.3796973 INFO: Expansion time:Pyr.SO3:0.7105930710004031\n",
      "1723050271.3815382 INFO: Filtered count:0 out of 1440\n",
      "1723050271.6529446 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050272.0475004 INFO: Expansion time:Pyr.SO4:0.6644998020001367\n",
      "1723050272.049316 INFO: Filtered count:0 out of 1440\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050272.2561285 INFO: Expansion time:Fum:3.470665196000482\n",
      "1723050272.2586334 INFO: Filtered count:0 out of 1274\n",
      "1723050272.7238984 INFO: Expansion time:Pyr.TMAO:0.6732721829994261\n",
      "1723050272.7259424 INFO: Filtered count:0 out of 1440\n",
      "1723050273.3280065 INFO: Expansion time:Etho.O2:1.1896289180003805\n",
      "1723050273.330706 INFO: Filtered count:0 out of 1500\n",
      "1723050273.430269 INFO: Expansion time:empty:0.703004714000599\n",
      "1723050273.4321685 INFO: Filtered count:0 out of 1440\n",
      "1723050274.374512 INFO: cpd05178 not found in model!\n",
      "1723050274.3765378 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050274.4678316 INFO: Expansion time:For.NO3:1.1354665779999777\n",
      "1723050274.4704728 INFO: Filtered count:0 out of 1500\n",
      "1723050274.5172467 INFO: cpd05178 not found in model!\n",
      "1723050274.5195663 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2.15.pg.G.D GF: 11198 Prob: 4166 Other: 2990\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050274.846113 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050275.5139596 INFO: Expansion time:Fum.O2:3.253704093000124\n",
      "1723050275.5165856 INFO: Filtered count:0 out of 1274\n",
      "1723050275.5686827 INFO: Expansion time:For.O2:1.096644340000239\n",
      "1723050275.5711958 INFO: Filtered count:0 out of 1500\n",
      "1723050276.6676 INFO: Expansion time:Fum.O2:1.094918382999822\n",
      "1723050276.6701844 INFO: Filtered count:0 out of 1500\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[218 'R1_B_H2O.028.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:50:42+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '746939037df9b00c469d82cb3ac2cde8'\n",
      " 5711472 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050277.1842008 INFO: metabolites 918\n",
      "1723050277.4990666 INFO: reactions 859\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050277.631713 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050277.95759 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/448/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050278.656053 INFO: Expansion time:Ac.O2:0.931996209999852\n",
      "1723050278.658773 INFO: Filtered count:0 out of 1557\n",
      "1723050278.8043396 INFO: Expansion time:LLac.O2:3.2860489000004236\n",
      "1723050278.807016 INFO: Filtered count:0 out of 1274\n",
      "1723050279.5717292 INFO: Expansion time:Etho.O2:0.911406020000868\n",
      "1723050279.5745153 INFO: Filtered count:0 out of 1557\n",
      "1723050280.11551 INFO: Expansion time:empty:3.443794034000348\n",
      "1723050280.1182728 INFO: Filtered count:0 out of 1500\n",
      "1723050280.285309 INFO: cpd05178 not found in model!\n",
      "1723050280.2876203 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050280.4846861 INFO: Expansion time:For.O2:0.9086817280003743\n",
      "1723050280.4873145 INFO: Filtered count:0 out of 1557\n",
      "1723050281.004996 INFO: Expansion time:Pyr:2.1964443940005367\n",
      "1723050281.007708 INFO: Filtered count:0 out of 1274\n",
      "1723050281.4018052 INFO: Expansion time:Fum:0.9129379579999295\n",
      "1723050281.4043937 INFO: Filtered count:0 out of 1557\n",
      "1723050282.0514524 INFO: Expansion time:Pyr.DMSO:1.0422184459994241\n",
      "1723050282.054154 INFO: Filtered count:0 out of 1274\n",
      "1723050282.9128268 INFO: Objective with gapfill database:224.78364471737243; min objective:0.01\n",
      "1723050282.9345376 INFO: Filtered count:0 out of 1274\n",
      "1723050282.931854 INFO: Expansion time:Pyr.NO:0.8762485050001487\n",
      "1723050283.2207856 INFO: Filtered count:0 out of 1557\n",
      "1723050283.2183747 INFO: Expansion time:Fum.O2:1.812519711999812\n",
      "1723050283.4836972 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050283.4838715 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050283.941035 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050283.9876177 INFO: Filtered count:0 out of 1274\n",
      "1723050283.9850345 INFO: Expansion time:Pyr.NO2:1.0489687600002071\n",
      "1723050284.1570947 INFO: Expansion time:Glc.O2:0.9348532740004885\n",
      "1723050284.159847 INFO: Filtered count:0 out of 1557\n",
      "1723050284.8467941 INFO: Expansion time:Pyr.NO3:0.8576717960004316\n",
      "1723050284.849525 INFO: Filtered count:0 out of 1274\n",
      "1723050285.2721758 INFO: Expansion time:Pyr:1.1105789869998262\n",
      "1723050285.274789 INFO: Filtered count:0 out of 1557\n",
      "1723050285.3361237 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050285.3339412 INFO: cpd05178 not found in model!\n",
      "1723050285.3200188 INFO: Objective with gapfill database:178.46079535300035; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050286.2231848 INFO: Expansion time:Pyr.DMSO:0.9468841790003353\n",
      "1723050286.225828 INFO: Filtered count:0 out of 1557\n",
      "1723050287.109471 INFO: cpd05178 not found in model!\n",
      "1723050287.1552722 INFO: Expansion time:Pyr.NO:0.9278168490000098\n",
      "1723050287.1577337 INFO: Filtered count:0 out of 1557\n",
      "1723050287.3314204 INFO: cpd05178 not found in model!\n",
      "1723050287.3336112 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050288.0838833 INFO: Expansion time:Pyr.NO2:0.9246500750004998\n",
      "1723050288.0865119 INFO: Filtered count:0 out of 1557\n",
      "1723050288.1393828 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050288.200483 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050288.1368577 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D2.100.pg.G.D GF: 11519 Prob: 3832 Other: 3003\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050288.757111 INFO: Expansion time:Pyr.O2:3.906064067000443\n",
      "1723050288.759852 INFO: Filtered count:0 out of 1274\n",
      "1723050289.974766 INFO: Expansion time:Pyr.NO3:1.8867452610002147\n",
      "1723050289.9772449 INFO: Filtered count:0 out of 1557\n",
      "1723050290.9421616 INFO: Expansion time:Pyr.O2:0.9633633839994218\n",
      "1723050290.9447613 INFO: Filtered count:0 out of 1557\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[293 'SF2_B_D2.4.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:05:48+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '04016018fe2ccbd8e88019d618a9db2f'\n",
      " 13462768 None]ForkPoolWorker-73\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050291.876296 INFO: Expansion time:Pyr.SO3:0.9300916480005981\n",
      "1723050291.8789768 INFO: Filtered count:0 out of 1557\n",
      "1723050291.937856 INFO: metabolites 1192\n",
      "1723050292.3005292 INFO: Expansion time:Pyr.SO3:3.5391934479994234\n",
      "1723050292.303228 INFO: Filtered count:0 out of 1274\n",
      "1723050292.631213 INFO: reactions 1165\n",
      "1723050292.7950165 INFO: Expansion time:Pyr.SO4:0.914424501999747\n",
      "1723050292.7974944 INFO: Filtered count:0 out of 1557\n",
      "1723050292.858803 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050293.359018 INFO: Expansion time:Pyr.SO4:1.0539370659998895\n",
      "1723050293.3616796 INFO: Filtered count:0 out of 1274\n",
      "1723050293.725665 INFO: Expansion time:Pyr.TMAO:0.9266692610008249\n",
      "1723050293.728296 INFO: Filtered count:0 out of 1557\n",
      "1723050293.841374 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/278/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050294.0926113 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/278/1\"\n",
      "}\n",
      "1723050294.2372458 INFO: Filtered count:0 out of 1274\n",
      "1723050294.2346148 INFO: Expansion time:Pyr.TMAO:0.871391974999824\n",
      "1723050294.5899005 INFO: cpd05178 not found in model!\n",
      "1723050294.9150944 INFO: cpd05178 not found in model!\n",
      "1723050294.9175527 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050295.0958507 INFO: Expansion time:empty:0.8568560369994884\n",
      "1723050295.0985036 INFO: Filtered count:0 out of 1274\n",
      "1723050295.5093443 INFO: Expansion time:empty:1.7795557689996713\n",
      "1723050295.5119262 INFO: Filtered count:0 out of 1557\n",
      "1723050295.883203 INFO: cpd05178 not found in model!\n",
      "1723050295.8854632 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_H2O.028.pg.G.D GF: 12125 Prob: 3088 Other: 3139\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050296.843456 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050297.2819753 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050300.008224 INFO: Expansion time:Fum:1.9155738909994398\n",
      "1723050300.0108168 INFO: Filtered count:0 out of 1356\n",
      "1723050300.9486752 INFO: Expansion time:Fum.O2:0.9362479430001258\n",
      "1723050300.9512284 INFO: Filtered count:0 out of 1356\n",
      "1723050301.268638 INFO: cpd05178 not found in model!\n",
      "1723050301.2709277 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050301.5936437 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050301.9040055 INFO: Expansion time:LLac.O2:0.9511819500003185\n",
      "1723050301.9062817 INFO: Filtered count:0 out of 1356\n",
      "1723050302.0579755 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050302.8906147 INFO: Expansion time:LLac.SO4:0.9829126419999739\n",
      "1723050302.8930504 INFO: Filtered count:0 out of 1356\n",
      "1723050303.850732 INFO: Expansion time:LLac.SO4.H2:0.9560842930004583\n",
      "1723050303.853175 INFO: Filtered count:0 out of 1356\n",
      "1723050304.242352 INFO: cpd05178 not found in model!\n",
      "1723050304.4807105 INFO: cpd05178 not found in model!\n",
      "1723050304.4827569 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050305.0208714 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050305.2688735 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050305.6150188 INFO: Expansion time:Llac:1.7603662849996908\n",
      "1723050305.6175745 INFO: Filtered count:0 out of 1356\n",
      "1723050306.0686204 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050306.6617231 INFO: Expansion time:Pyr:1.0426204949999374\n",
      "1723050306.6643066 INFO: Filtered count:0 out of 1356\n",
      "1723050307.662998 INFO: Expansion time:Pyr.DMSO:0.9971855429994321\n",
      "1723050307.665539 INFO: Filtered count:0 out of 1356\n",
      "1723050308.3764226 INFO: cpd05178 not found in model!\n",
      "1723050308.3787735 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050308.6392934 INFO: Expansion time:Pyr.NO:0.9722142259997781\n",
      "1723050308.641796 INFO: Filtered count:0 out of 1356\n",
      "1723050308.9786706 INFO: cpd05178 not found in model!\n",
      "1723050308.9809563 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[170 'R2_C_D1.062.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:41:02+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'a2e151a865f15d36f269bbaab68ab438'\n",
      " 10610066 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050310.3973603 INFO: Expansion time:Pyr.NO2:1.7541614970004957\n",
      "1723050310.4000018 INFO: Filtered count:0 out of 1356\n",
      "1723050310.539947 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050310.6438117 INFO: metabolites 999\n",
      "1723050311.2561772 INFO: reactions 961\n",
      "1723050311.373605 INFO: Expansion time:Pyr.NO3:0.9721525470004053\n",
      "1723050311.3761613 INFO: Filtered count:0 out of 1356\n",
      "1723050311.4170325 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050311.9848118 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/148/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050312.3786352 INFO: cpd05178 not found in model!\n",
      "1723050312.3808258 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050312.3855631 INFO: Expansion time:Pyr.O2:1.0080069250007\n",
      "1723050312.3880036 INFO: Filtered count:0 out of 1356\n",
      "1723050312.5491846 INFO: cpd05178 not found in model!\n",
      "1723050312.551655 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D2.10.pg.G.D GF: 11248 Prob: 4152 Other: 2954\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050313.332806 INFO: Expansion time:Pyr.SO3:0.9432128120006382\n",
      "1723050313.3354263 INFO: Filtered count:0 out of 1356\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-74\n",
      "[297 'R1_B_D2.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:06:35+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '65a1e57163003b9028d4216e10710037'\n",
      " 12803207 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050314.277642 INFO: Expansion time:Pyr.SO4:0.9407836299997143\n",
      "1723050314.280012 INFO: Filtered count:0 out of 1356\n",
      "1723050314.6818545 INFO: metabolites 1228\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050315.4029136 INFO: reactions 1211\n",
      "1723050315.5413024 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050315.7869747 INFO: Expansion time:Etho.O2:2.170726993000244\n",
      "1723050315.7892342 INFO: Filtered count:0 out of 1199\n",
      "1723050316.0656915 INFO: Expansion time:Pyr.TMAO:1.784171363000496\n",
      "1723050316.0682726 INFO: Filtered count:0 out of 1356\n",
      "1723050316.2269065 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/370/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050316.3999484 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/370/1\"\n",
      "}\n",
      "1723050316.5346816 INFO: Filtered count:0 out of 1199\n",
      "1723050316.532471 INFO: Expansion time:For.O2:0.742029805999664\n",
      "1723050317.0571656 INFO: Expansion time:empty:0.9873364619998029\n",
      "1723050317.0596666 INFO: Filtered count:0 out of 1356\n",
      "1723050317.413029 INFO: cpd05178 not found in model!\n",
      "1723050317.4154902 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050317.430966 INFO: Expansion time:Fum:0.8951248840003245\n",
      "1723050317.4329784 INFO: Filtered count:0 out of 1199\n",
      "1723050318.4166644 INFO: Expansion time:Fum.O2:0.9825203350001175\n",
      "1723050318.4195719 INFO: Filtered count:0 out of 1199\n",
      "1723050318.6025772 INFO: Expansion time:Ac.O2:3.6158791459993154\n",
      "1723050318.604608 INFO: Filtered count:0 out of 1524\n",
      "1723050319.1407695 INFO: Expansion time:Pyr.O2:0.7199860339997031\n",
      "1723050319.1430817 INFO: Filtered count:0 out of 1199\n",
      "1723050319.4644725 INFO: Expansion time:Dlac.O2:0.8585857799998848\n",
      "1723050319.4667492 INFO: Filtered count:0 out of 1524\n",
      "1723050320.4372582 INFO: Expansion time:Etho.O2:0.9689627070001734\n",
      "1723050320.439445 INFO: Filtered count:0 out of 1524\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050320.7999163 INFO: Expansion time:empty:1.6556761029996778\n",
      "1723050320.8021529 INFO: Filtered count:0 out of 1199\n",
      "1723050321.1441274 INFO: Expansion time:For.O2:0.7033620330003032\n",
      "1723050321.146292 INFO: Filtered count:0 out of 1524\n",
      "1723050321.7180192 INFO: cpd05178 not found in model!\n",
      "1723050321.720209 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_D2.4.pg.G.D GF: 11198 Prob: 4188 Other: 2968\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050322.3934402 INFO: Expansion time:Fum:1.2458556350002254\n",
      "1723050322.3959718 INFO: Filtered count:0 out of 1524\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[202 'R2_B_D1.40.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:47:38+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '91ed28ed9dff5a4e5e9328205e29f9f8'\n",
      " 15259004 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050323.7263565 INFO: Expansion time:Fum.O2:1.3288403879996622\n",
      "1723050323.728173 INFO: Filtered count:0 out of 1524\n",
      "1723050324.178926 INFO: metabolites 1139\n",
      "1723050324.6847525 INFO: cpd05178 not found in model!\n",
      "1723050324.6868238 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050324.8260052 INFO: reactions 1109\n",
      "1723050324.9643738 INFO: Expansion time:Pyr:1.2349847750001572\n",
      "1723050324.9668071 INFO: Filtered count:0 out of 1524\n",
      "1723050324.9959035 INFO: cpd05178 not found in model!\n",
      "1723050325.0032902 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050325.224409 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050325.222291 INFO: cpd05178 not found in model!\n",
      "1723050325.6347027 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "188040/324/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050326.7070646 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050327.6189156 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050328.4118574 INFO: Objective with gapfill database:129.4175323453976; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050330.4883513 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn13085_c0\n",
      "Failed:<rxn12008_c0\n",
      "\n",
      "Failed:<rxn03086_c0"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050331.3448784 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Failed:<rxn00738_c0"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050331.6715324 INFO: cpd05178 not found in model!\n",
      "1723050331.674082 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03030_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050331.8019161 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn02988_c0\n",
      "Failed:>rxn02305_c0\n",
      "\n",
      "Failed:>rxn09043_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050332.6759577 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Failed:<rxn00128_c0\n",
      "Failed:<rxn00762_c0Failed:>rxn00470_c0Failed:>rxn05459_c0Failed:<rxn05514_c0\n",
      "Failed:>rxn00600_c0\n",
      "Failed:>rxn08083_c0\n",
      "Getting tests from attributes\n",
      "Failed:>rxn00272_c0\n",
      "Failed:>rxn10213_c0\n",
      "Failed:>rxn05039_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn09180_c0\n",
      "Failed:>rxn08822_c0\n",
      "\n",
      "Failed:>rxn03891_c0\n",
      "Failed:>rxn10204_c0Failed:<rxn09341_c0\n",
      "ForkPoolWorker-75\n",
      "[301 'R2_A_D1.35.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:07:25+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '4c8efa193c6e3ce1162a6c83eaabf481'\n",
      " 12041319 None]\n",
      "Failed:>rxn09128_c0\n",
      "Failed:>rxn03536_c0\n",
      "Failed:>rxn09158_c0\n",
      "Failed:>rxn01505_c0\n",
      "Failed:>rxn08353_c0\n",
      "Failed:<rxn01484_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050336.0066063 INFO: metabolites 1194\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn08226_c0\n",
      "Failed:<rxn00003_c0\n",
      "Failed:<rxn08815_c0\n",
      "Failed:<rxn08219_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050336.802014 INFO: reactions 1194\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn09163_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050337.0823967 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn04704_c0\n",
      "Failed:>rxn01972_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050337.7706213 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00979_c0\n",
      "Failed:<rxn08166_c0\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050338.209063 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn11703_c0"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050338.1103177 INFO: Default biomass: [bio1]\n",
      "1723050338.2069185 INFO: cpd05178 not found in model!\n",
      "1723050338.2806795 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03537_c0\n",
      "188040/360/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050338.354545 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/360/1\"\n",
      "}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn00966_c0\n",
      "Failed:>rxn05119_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050338.778701 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn01627_c0\n",
      "Failed:<rxn12376_c0\n",
      "Failed:>rxn06023_c0\n",
      "Failed:>rxn05012_c0\n",
      "Getting tests from attributes\n",
      "Failed:>rxn09041_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050339.5899565 INFO: Expansion time:Etho.O2:0.6855625730004249\n",
      "1723050339.5920486 INFO: Filtered count:0 out of 1660\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn09039_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050340.291523 INFO: Expansion time:Glc:0.698276612999507\n",
      "1723050340.2936437 INFO: Filtered count:0 out of 1660\n",
      "1723050340.973723 INFO: Expansion time:Glc.DMSO:0.678838795000047\n",
      "1723050340.975957 INFO: Filtered count:0 out of 1660\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03108_c0\n",
      "Failed:>rxn09045_c0\n",
      "Failed:>rxn03146_c0\n",
      "Failed:>rxn03893_c0\n",
      "Failed:>rxn03150_c0\n",
      "Failed:>rxn03538_c0\n",
      "Failed:>rxn09310_c0\n",
      "\n",
      "\n",
      "Failed:>rxn01213_c0"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050342.4062428 INFO: Expansion time:Glc.O2:1.429030942999816\n",
      "1723050342.4082325 INFO: Filtered count:0 out of 1660\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn08810_c0Failed:>rxn03393_c0\n",
      "Failed:<rxn01790_c0\n",
      "Failed:>rxn02897_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050343.2500439 INFO: Expansion time:Glc.TMAO:0.8405635960007203\n",
      "1723050343.252198 INFO: Filtered count:0 out of 1660\n",
      "1723050343.3423536 INFO: cpd05178 not found in model!\n",
      "1723050343.344914 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn05144_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050343.6884453 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.16.pg.G.D GF: 11084 Prob: 4334 Other: 2934\n",
      "Failed:>rxn01466_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050343.9278622 INFO: Expansion time:Pyr:0.6744736250002461\n",
      "1723050343.9299178 INFO: Filtered count:0 out of 1660\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn00023_c0\n",
      "Failed:>rxn11933_c0\n",
      "Failed:>rxn08131_c0\n",
      "Failed:>rxn01513_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050344.6096394 INFO: Expansion time:Pyr.DMSO:0.6785534399996322\n",
      "1723050344.6118772 INFO: Filtered count:0 out of 1660\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03395_c0\n",
      "Failed:>rxn16573_c0\n",
      "Failed:>rxn41452_c0\n",
      "Failed:>rxn08233_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050345.2786767 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn16107_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050345.3314312 INFO: Expansion time:Pyr.NO:0.7182279600001493\n",
      "1723050345.3337805 INFO: Filtered count:0 out of 1660\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn48586_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050345.512868 INFO: cpd05178 not found in model!\n",
      "1723050345.5151095 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050345.5466704 INFO: Expansion time:Pyr.DMSO:20.57834198099954\n",
      "1723050345.5490603 INFO: Filtered count:72 out of 1524\n",
      "1723050346.3194015 INFO: cpd05178 not found in model!\n",
      "1723050346.4601648 INFO: cpd05178 not found in model!\n",
      "1723050346.4622838 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050346.5488338 INFO: cpd05178 not found in model!\n",
      "1723050346.5510309 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050346.8629923 INFO: Expansion time:Pyr.NO2:1.5276463199998034\n",
      "1723050346.864985 INFO: Filtered count:0 out of 1660\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[234 'R2A_C_D1.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:53:46+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '5f319b2c32cc14842d3bac9f0e0290eb'\n",
      " 7954323 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050347.1978233 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050347.2288775 INFO: cpd05178 not found in model!\n",
      "1723050347.2311623 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050347.4249313 INFO: Expansion time:Pyr.NO:1.8740752990006513\n",
      "1723050347.4274118 INFO: Filtered count:72 out of 1524\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D1.062.pg.G.D GF: 11488 Prob: 3758 Other: 3106\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050347.5468419 INFO: Expansion time:Pyr.NO3:0.6806426680004733\n",
      "1723050347.548844 INFO: Filtered count:0 out of 1660\n",
      "1723050347.6352918 INFO: metabolites 1035\n",
      "1723050347.903539 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050348.115385 INFO: reactions 968\n",
      "1723050348.248127 INFO: Expansion time:Pyr.O2:0.6980137569998988\n",
      "1723050348.2478664 INFO: cpd05178 not found in model!\n",
      "1723050348.2500308 INFO: Filtered count:0 out of 1660\n",
      "1723050348.2727957 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050348.2857091 INFO: cpd05178 not found in model!\n",
      "1723050348.2878022 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050348.482751 INFO: cpd05178 not found in model!\n",
      "1723050348.4847705 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050348.813032 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/530/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050348.9842532 INFO: Expansion time:Pyr.SO3:0.7330325820003054\n",
      "1723050348.986565 INFO: Filtered count:0 out of 1660\n",
      "1723050349.6735713 INFO: Expansion time:Pyr.SO4:0.6854904210003951\n",
      "1723050349.675624 INFO: Filtered count:0 out of 1660\n",
      "1723050350.3708496 INFO: Expansion time:Pyr.TMAO:0.6939250800005539\n",
      "1723050350.3729808 INFO: Filtered count:0 out of 1660\n",
      "1723050351.8104186 INFO: cpd05178 not found in model!\n",
      "1723050351.8125145 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050352.1693456 INFO: Expansion time:empty:1.7950209930004348\n",
      "1723050352.1719027 INFO: Filtered count:0 out of 1660\n",
      "1723050352.9033039 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050353.323144 INFO: cpd05178 not found in model!\n",
      "1723050353.6985552 INFO: cpd05178 not found in model!\n",
      "1723050353.7010143 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00062_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050354.4141269 INFO: Expansion time:Pyr.NO2:6.98511096600032\n",
      "1723050354.4165707 INFO: Filtered count:73 out of 1524\n",
      "1723050355.0751708 INFO: cpd05178 not found in model!\n",
      "1723050355.1085644 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[90 'R2_A_D1.7.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:25:18+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '245e6a9ab69333e9fa6d84533a4e22d2'\n",
      " 15146831 None]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050355.660116 INFO: metabolites 1120\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[94 'R2_restored_DShore.2.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:26:01+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'f30eb35a844afd6c06323c1275fdb6fc'\n",
      " 9735163 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050356.0639188 INFO: cpd05178 not found in model!\n",
      "1723050356.0669062 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D1.40.pg.G.D GF:"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050356.418962 INFO: reactions 1081\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " 11678 Prob: 3536 Other: 3140\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050356.4782376 INFO: cpd05178 not found in model!\n",
      "1723050356.480345 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050356.605491 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050356.8693755 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050357.0106814 INFO: metabolites 1005\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050357.26669 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/252/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050357.4039054 INFO: reactions 930\n",
      "1723050357.5152638 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050357.6786335 INFO: Expansion time:Pyr.NO3:3.260584319999907\n",
      "1723050357.6811104 INFO: Filtered count:73 out of 1524\n",
      "1723050357.876229 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/266/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050358.6219954 INFO: Expansion time:Pyr.O2:0.9392386409999745\n",
      "1723050358.624361 INFO: Filtered count:73 out of 1524\n",
      "1723050359.5644379 INFO: Expansion time:Pyr.SO3:0.9386673030003294\n",
      "1723050359.5668893 INFO: Filtered count:73 out of 1524\n",
      "1723050359.8621242 INFO: Objective with gapfill database:227.56851883607686; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-76\n",
      "[305 'R2_C_D2.104.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:08:10+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'c5a48f37e6ee9e41855cdf027eed4b13'\n",
      " 15817332 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050360.4734247 INFO: cpd05178 not found in model!\n",
      "1723050360.4756076 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050360.5329316 INFO: Expansion time:Pyr.SO4:0.9644707269999344\n",
      "1723050360.5351613 INFO: Filtered count:73 out of 1524\n",
      "1723050361.0092547 INFO: Objective with gapfill database:159.32293656831447; min objective:0.01\n",
      "1723050361.1902354 INFO: metabolites 1093\n",
      "1723050361.3813236 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050361.4802 INFO: Expansion time:Pyr.TMAO:0.9435125039999548\n",
      "1723050361.4823139 INFO: Filtered count:73 out of 1524\n",
      "1723050361.7337437 INFO: reactions 1024\n",
      "1723050361.8946273 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050362.4714081 INFO: Expansion time:empty:0.9877060639992123\n",
      "1723050362.4738894 INFO: Filtered count:73 out of 1524\n",
      "1723050362.4763896 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050362.6807208 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/388/1\"\n",
      "}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/388/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050364.3217463 INFO: cpd05178 not found in model!\n",
      "1723050364.3243861 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.35.pg.G.D GF: 11271 Prob: 4116 Other: 2967\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050367.0951943 INFO: cpd05178 not found in model!\n",
      "1723050367.097395 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050372.809959 INFO: Objective with gapfill database:172.66053912808337; min objective:0.01\n",
      "1723050373.276501 INFO: cpd05178 not found in model!\n",
      "1723050373.2793238 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D1.16.pg.G.D GF: 12073 Prob: 3142 Other: 3139\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050375.5894034 INFO: cpd05178 not found in model!\n",
      "1723050375.5915127 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050375.995314 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050376.4943142 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[16 'R2_A_D1.10.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:10:53+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'fdabe74a049f2fb2fb2fbc2e22139138'\n",
      " 13142561 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050376.9735355 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050377.1795034 INFO: cpd05178 not found in model!\n",
      "1723050377.2350488 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050377.418006 INFO: cpd05178 not found in model!\n",
      "1723050377.4202282 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050377.7058406 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050378.2582605 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050378.9290087 INFO: metabolites 1102\n",
      "1723050379.3764162 INFO: reactions 1085\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[167 'R1_A_D1.18.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:40:31+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '35ca6542ffa5a7f93b1e5d03d8c6a4f1'\n",
      " 11835525 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050379.50415 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050379.9146795 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/437/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050380.0231113 INFO: metabolites 1102\n",
      "1723050380.0550175 INFO: cpd05178 not found in model!\n",
      "1723050380.057207 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050380.059938 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050380.653754 INFO: reactions 1092\n",
      "1723050380.9045022 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050381.2042 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050381.5342546 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/104/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050382.629042 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-77\n",
      "[309 'R2A_C_D1.15.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:09:00+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'c35b065bb3ba9db3ef3c305afefcd1c7'\n",
      " 11896234 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050384.1886384 INFO: cpd05178 not found in model!\n",
      "1723050384.1909895 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050384.5359166 INFO: cpd05178 not found in model!\n",
      "1723050384.5386066 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050384.7703152 INFO: metabolites 1134\n",
      "1723050385.3923535 INFO: reactions 1113\n",
      "1723050385.4405694 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050385.5899522 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050386.2520213 INFO: Default biomass: [bio1]\n",
      "1723050386.3096583 INFO: Expansion time:Ac.O2:1.5382519510003476\n",
      "1723050386.3119948 INFO: Filtered count:0 out of 1699\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/423/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050386.8466308 INFO: get_annotation_ontology_events:{\n",
      "    \"input_ref\": \"188040/423/1\"\n",
      "}\n",
      "1723050386.914618 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050387.1628294 INFO: Expansion time:Dlac.O2:0.849670318999415\n",
      "1723050387.1651354 INFO: Filtered count:0 out of 1699\n",
      "1723050387.972656 INFO: Expansion time:Etho.O2:0.8063072349996219\n",
      "1723050387.9749935 INFO: Filtered count:0 out of 1699\n",
      "1723050388.658567 INFO: Expansion time:For.O2:0.6822369730007267\n",
      "1723050388.6607912 INFO: Filtered count:0 out of 1699\n",
      "1723050390.2643807 INFO: Expansion time:Fum:1.60233702000005\n",
      "1723050390.266795 INFO: Filtered count:0 out of 1699\n",
      "1723050390.4089208 INFO: cpd05178 not found in model!\n",
      "1723050390.4113812 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050390.6811788 INFO: cpd05178 not found in model!\n",
      "1723050390.6833315 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050391.1209335 INFO: Expansion time:Fum.O2:0.8527655990001222\n",
      "1723050391.1233335 INFO: Filtered count:0 out of 1699\n",
      "1723050391.9601483 INFO: Expansion time:LLac.O2:0.8355390779997833\n",
      "1723050391.9625306 INFO: Filtered count:0 out of 1699\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[67 'R1_B_D2.73.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:20:57+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'fbb6fb2c55cfc5408d2dc877cbeac46c'\n",
      " 7790750 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050392.2492256 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050392.390741 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050392.4670243 INFO: metabolites 1050\n",
      "1723050392.5025764 INFO: cpd05178 not found in model!\n",
      "1723050392.5049815 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050392.883965 INFO: Expansion time:Pyr:0.9202449800004615\n",
      "1723050392.8862345 INFO: Filtered count:0 out of 1699\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D2.104.pg.G.D GF: 11970 Prob: 3156 Other: 3228\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050392.9444246 INFO: cpd05178 not found in model!\n",
      "1723050393.037003 INFO: reactions 966\n",
      "1723050393.0982523 INFO: cpd05178 not found in model!\n",
      "1723050393.1006048 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050393.1553447 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050393.1761994 INFO: cpd05178 not found in model!\n",
      "1723050393.1783056 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050393.558501 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/90/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050394.0928724 INFO: cpd05178 not found in model!\n",
      "1723050394.0954788 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050394.1719837 INFO: cpd05178 not found in model!\n",
      "1723050394.1743534 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.2.pg.G.D GF: 11633 Prob: 3730 Other: 2991\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050394.4862468 INFO: Expansion time:Pyr.DMSO:1.5987663060004706\n",
      "1723050394.4884522 INFO: Filtered count:0 out of 1699\n",
      "1723050395.248643 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050396.3744807 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050396.424104 INFO: Expansion time:Pyr.NO:1.9344998959995792\n",
      "1723050396.4262886 INFO: Filtered count:0 out of 1699\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050396.9928901 INFO: cpd05178 not found in model!\n",
      "1723050397.1372128 INFO: Expansion time:Pyr.NO2:0.7097374229997513\n",
      "1723050397.1396043 INFO: Filtered count:0 out of 1699\n",
      "1723050397.2270024 INFO: cpd05178 not found in model!\n",
      "1723050397.229112 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050398.3327203 INFO: cpd05178 not found in model!\n",
      "1723050398.3356304 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050398.6461248 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.7.pg.G.D GF: 11326 Prob: 3876 Other: 3148\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050398.9538639 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050399.473681 INFO: Expansion time:Pyr.NO3:2.332857309999781\n",
      "1723050399.4758742 INFO: Filtered count:0 out of 1699\n",
      "1723050399.664459 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050400.0734007 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050400.3853395 INFO: Expansion time:Pyr.O2:0.9082226560003619\n",
      "1723050400.3875613 INFO: Filtered count:0 out of 1699\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050401.8107715 INFO: Expansion time:Pyr.SO3:1.4220619729994723\n",
      "1723050401.8129845 INFO: Filtered count:0 out of 1699\n",
      "1723050402.719749 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050403.1798885 INFO: Expansion time:Pyr.SO4:1.3657825370000864\n",
      "1723050403.1821928 INFO: Filtered count:0 out of 1699\n",
      "1723050403.8849244 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050403.886104 INFO: Expansion time:Pyr.TMAO:0.7026727510001365\n",
      "1723050403.8880289 INFO: Filtered count:0 out of 1699\n",
      "1723050404.5229578 INFO: Expansion time:empty:0.6337807419995443\n",
      "1723050404.5252275 INFO: Filtered count:0 out of 1699\n",
      "1723050404.7549255 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050404.8621607 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050405.1767824 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[114 'R2A_C_D2.49.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:30:07+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'd5e4362f35e23a151bc7dd9fcd65e7fe'\n",
      " 15301975 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050406.6384299 INFO: Expansion time:Ac.O2:5.752866377999453\n",
      "1723050406.6404524 INFO: Filtered count:0 out of 1352\n",
      "1723050407.217082 INFO: metabolites 1234\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050407.5555618 INFO: cpd05178 not found in model!\n",
      "1723050407.5578012 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050407.878423 INFO: reactions 1215\n",
      "1723050408.119964 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050408.4613273 INFO: Expansion time:Etho.O2:1.8197601849997227\n",
      "1723050408.463383 INFO: Filtered count:0 out of 1352\n",
      "1723050408.8027358 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/430/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050410.0090716 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050410.7835767 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050411.5488396 INFO: Expansion time:Glc:3.084255466000286\n",
      "1723050411.5511496 INFO: Filtered count:0 out of 1352\n",
      "1723050411.7815232 INFO: cpd05178 not found in model!\n",
      "1723050411.784119 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D1.18.pg.G.D GF: 11588 Prob: 3706 Other: 3060\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050412.8525794 INFO: cpd05178 not found in model!\n",
      "1723050412.8552408 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050412.9657898 INFO: Expansion time:Glc.DMSO:1.4133831939998345\n",
      "1723050412.9681435 INFO: Filtered count:0 out of 1352\n",
      "1723050413.0475714 INFO: cpd05178 not found in model!\n",
      "1723050413.049564 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.10.pg.G.D GF: 11472 Prob: 3762 Other: 3118\n",
      "R2A_C_D1.15.pg.G.D GF: 11516 Prob: 3796 Other: 3042"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050414.191054 INFO: cpd05178 not found in model!\n",
      "1723050414.2020605 INFO: Expansion time:Dlac:0.6930396530005964\n",
      "1723050414.2038233 INFO: Filtered count:0 out of 1456\n",
      "1723050414.418291 INFO: cpd05178 not found in model!\n",
      "1723050414.420662 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050415.4286928 INFO: Expansion time:Etho.O2:4.061242022999977\n",
      "1723050415.431099 INFO: Filtered count:0 out of 1279\n",
      "1723050415.5169475 INFO: Expansion time:Dlac.O2:1.311940992000018\n",
      "1723050415.5190187 INFO: Filtered count:0 out of 1456\n",
      "1723050415.7026093 INFO: Expansion time:Glc.O2:2.7331304900008035\n",
      "1723050415.7047267 INFO: Filtered count:0 out of 1352\n",
      "1723050416.1904736 INFO: Expansion time:Etho.O2:0.6702834170000642\n",
      "1723050416.1924503 INFO: Filtered count:0 out of 1456\n",
      "1723050416.8593826 INFO: Expansion time:LLac.O2:0.665507771999728\n",
      "1723050416.8613489 INFO: Filtered count:0 out of 1456\n",
      "1723050417.5173507 INFO: Expansion time:LLac.SO4:0.653970959999242\n",
      "1723050417.5194924 INFO: Filtered count:0 out of 1456\n",
      "1723050417.6135874 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050417.6584904 INFO: Expansion time:For.NO3:2.226002354999764\n",
      "1723050417.6607203 INFO: Filtered count:0 out of 1279\n",
      "1723050417.8688822 INFO: Expansion time:Glc.TMAO:2.1629006709999885\n",
      "1723050417.8711069 INFO: Filtered count:0 out of 1352\n",
      "1723050418.7514932 INFO: Expansion time:For.O2:1.0895113789993047\n",
      "1723050418.753324 INFO: Filtered count:0 out of 1279\n",
      "1723050418.9011123 INFO: Expansion time:LLac.SO4.H2:1.3804758289998063\n",
      "1723050418.9034278 INFO: Filtered count:0 out of 1456\n",
      "1723050419.4152918 INFO: Expansion time:Pyr.O2:1.5429431459997431\n",
      "1723050419.4171474 INFO: Filtered count:0 out of 1352\n",
      "1723050419.461439 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050419.6048076 INFO: Expansion time:Llac:0.700053535000734\n",
      "1723050419.6071646 INFO: Filtered count:0 out of 1456\n",
      "1723050419.711557 INFO: cpd05178 not found in model!\n",
      "1723050419.713528 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "R1_B_D2.73.pg.G.D GF: 11721 Prob: 3406 Other: 3225\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050420.085924 INFO: Expansion time:empty:0.6676001830001042\n",
      "1723050420.0880105 INFO: Filtered count:0 out of 1352\n",
      "1723050420.3700287 INFO: Expansion time:Fum.O2:1.615453376000005\n",
      "1723050420.3721735 INFO: Filtered count:0 out of 1279\n",
      "1723050420.7751799 INFO: Expansion time:Pyr:1.1665191920001234\n",
      "1723050420.7775662 INFO: Filtered count:0 out of 1456\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050421.862501 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[43 'R1_B_D2.34.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:16:05+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'b5d241b9dcb0542ff1ef059017e706ac'\n",
      " 11043009 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050421.9510617 INFO: Expansion time:Pyr.DMSO:1.1720360259996596\n",
      "1723050421.9535513 INFO: Filtered count:0 out of 1456\n",
      "1723050422.3437202 INFO: Expansion time:Dlac:0.6807989010003439\n",
      "1723050422.34579 INFO: Filtered count:0 out of 1563\n",
      "1723050422.3825145 INFO: metabolites 1054\n",
      "1723050422.7192268 INFO: Expansion time:Glc.O2:2.3459032219998335\n",
      "1723050422.7213764 INFO: Filtered count:0 out of 1279\n",
      "1723050422.99668 INFO: Expansion time:Dlac.O2:0.6497156100003849\n",
      "1723050422.9988253 INFO: Filtered count:0 out of 1563\n",
      "1723050423.0004272 INFO: reactions 1013\n",
      "1723050423.1036289 INFO: Expansion time:Pyr.NO:1.1485280650003915\n",
      "1723050423.1059706 INFO: Filtered count:0 out of 1456\n",
      "1723050423.1753938 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050423.6573648 INFO: Expansion time:Fum:0.6573562639996453\n",
      "1723050423.659537 INFO: Filtered count:0 out of 1563\n",
      "1723050423.7835207 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/568/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050423.9000783 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050424.109868 INFO: Expansion time:Pyr.NO3:1.3872773449993474\n",
      "1723050424.1119282 INFO: Filtered count:0 out of 1279\n",
      "1723050424.3280945 INFO: Expansion time:Fum.O2:0.6672620239996832\n",
      "1723050424.3303335 INFO: Filtered count:0 out of 1563\n",
      "1723050424.603081 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050425.0122602 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050425.054298 INFO: Expansion time:Pyr.NO2:1.9467599520003205\n",
      "1723050425.0565746 INFO: Filtered count:0 out of 1456\n",
      "1723050425.700094 INFO: Expansion time:Pyr.O2:1.587006996999662\n",
      "1723050425.702164 INFO: Filtered count:0 out of 1279\n",
      "1723050425.7357504 INFO: cpd05178 not found in model!\n",
      "1723050425.7378845 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050426.1261072 INFO: Expansion time:Pyr.NO3:1.0680063239997253\n",
      "1723050426.1286883 INFO: Filtered count:0 out of 1456\n",
      "1723050426.3074472 INFO: Expansion time:Glc:1.9758461759993224\n",
      "1723050426.309678 INFO: Filtered count:0 out of 1563\n",
      "1723050426.3651843 INFO: Expansion time:empty:0.6617629749998741\n",
      "1723050426.367251 INFO: Filtered count:0 out of 1279\n",
      "1723050426.977046 INFO: Expansion time:Glc.DMSO:0.6660271529999591\n",
      "1723050426.979194 INFO: Filtered count:0 out of 1563\n",
      "1723050427.1309412 INFO: cpd05178 not found in model!\n",
      "1723050427.1332045 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050427.2345288 INFO: Expansion time:Pyr.O2:1.104320743000244\n",
      "1723050427.2369277 INFO: Filtered count:0 out of 1456\n",
      "1723050427.7144973 INFO: Expansion time:Glc.O2:0.7339522649999708\n",
      "1723050427.7166483 INFO: Filtered count:0 out of 1563\n",
      "1723050428.343182 INFO: Expansion time:Pyr.SO3:1.1047675980007625\n",
      "1723050428.3456352 INFO: Filtered count:0 out of 1456\n",
      "1723050428.4425187 INFO: Expansion time:Glc.TMAO:0.7246235659995364\n",
      "1723050428.4445915 INFO: Filtered count:0 out of 1563\n",
      "1723050428.9615595 INFO: cpd05178 not found in model!\n",
      "1723050429.1125903 INFO: Expansion time:Pyr:0.6668329739995897\n",
      "1723050429.1149993 INFO: Filtered count:0 out of 1563\n",
      "1723050429.195841 INFO: cpd05178 not found in model!\n",
      "1723050429.1980093 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050429.7855961 INFO: Expansion time:Pyr.DMSO:0.6692576039995402\n",
      "1723050429.7877405 INFO: Filtered count:0 out of 1563\n",
      "1723050430.442248 INFO: Expansion time:Pyr.NO:0.653127463000601\n",
      "1723050430.4444342 INFO: Filtered count:0 out of 1563\n",
      "1723050430.4524846 INFO: Expansion time:Pyr.SO4:2.1052744889993846\n",
      "1723050430.4550474 INFO: Filtered count:0 out of 1456\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050430.9017315 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050431.1210766 INFO: Expansion time:Pyr.NO2:0.6753755770005228\n",
      "1723050431.123242 INFO: Filtered count:0 out of 1563\n",
      "1723050431.4941607 INFO: Expansion time:Pyr.TMAO:1.0376044639997417\n",
      "1723050431.4966311 INFO: Filtered count:0 out of 1456\n",
      "1723050432.5133636 INFO: Expansion time:empty:1.0151819980001164\n",
      "1723050432.5157433 INFO: Filtered count:0 out of 1456\n",
      "1723050433.042185 INFO: Expansion time:Pyr.NO3:1.9178045010003189\n",
      "1723050433.0444267 INFO: Filtered count:0 out of 1563\n",
      "1723050433.2361953 INFO: cpd05178 not found in model!\n",
      "1723050433.2387168 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050433.5063038 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050433.6286473 INFO: Expansion time:Ac.NO2:2.472968676999699\n",
      "1723050433.630826 INFO: Filtered count:0 out of 1611\n",
      "1723050433.8044033 INFO: Expansion time:Pyr.O2:0.7584527489998436\n",
      "1723050433.8068519 INFO: Filtered count:0 out of 1563\n",
      "1723050434.031749 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050434.7509966 INFO: Expansion time:Pyr.SO3:0.9427104000005784\n",
      "1723050434.7536588 INFO: Filtered count:0 out of 1563\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[142 'R1_B_H2O.11.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:35:40+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '639b62e4051c87da14e6f3e691e0b2ab'\n",
      " 8501672 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050435.0309036 INFO: Expansion time:Ac.NO3:1.398862779999945\n",
      "1723050435.0329406 INFO: Filtered count:0 out of 1611\n",
      "1723050435.4355595 INFO: metabolites 998\n",
      "1723050435.6890855 INFO: Expansion time:Pyr.SO4:0.9338537389994599\n",
      "1723050435.691567 INFO: Filtered count:0 out of 1563\n",
      "1723050435.7523234 INFO: cpd05178 not found in model!\n",
      "1723050436.0160253 INFO: reactions 962\n",
      "1723050436.0629082 INFO: cpd05178 not found in model!\n",
      "1723050436.0656755 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.49.pg.G.D GF: 11410 Prob: 3856 Other: 3086\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050436.9423413 INFO: Expansion time:Pyr.TMAO:1.2493480079992878\n",
      "1723050436.9451103 INFO: Filtered count:0 out of 1563\n",
      "1723050437.1421769 INFO: cpd05178 not found in model!\n",
      "1723050437.1448553 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050437.1624155 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050437.7192104 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/592/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050437.8327386 INFO: cpd05178 not found in model!\n",
      "1723050437.8349605 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050438.0925736 INFO: Expansion time:empty:1.145764075000443\n",
      "1723050438.095237 INFO: Filtered count:0 out of 1563\n",
      "1723050438.0995202 INFO: Expansion time:Ac.O2:3.0650221390005754\n",
      "1723050438.1018705 INFO: Filtered count:0 out of 1611\n",
      "1723050438.7274563 INFO: Expansion time:Etho.O2:0.6829894889997377\n",
      "1723050438.7296698 INFO: Filtered count:0 out of 1551\n",
      "1723050439.3923156 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050440.019634 INFO: Expansion time:Dlac.O2:1.9161734090002938\n",
      "1723050440.02218 INFO: Filtered count:0 out of 1611\n",
      "1723050440.9766757 INFO: Expansion time:Glc:2.2458270200004335\n",
      "1723050440.978847 INFO: Filtered count:0 out of 1551\n",
      "1723050441.4063334 INFO: Expansion time:Etho.O2:1.3827715100005662\n",
      "1723050441.4089222 INFO: Filtered count:0 out of 1611\n",
      "1723050441.6297436 INFO: Expansion time:Glc.DMSO:0.6496372700003121\n",
      "1723050441.6319003 INFO: Filtered count:0 out of 1551\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[79 'R2_B_D1.10.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:23:17+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'dfba50b702dc1b5fc41efb175d035b74'\n",
      " 12009129 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050442.0028746 INFO: Expansion time:Glc:3.8231292229993414\n",
      "1723050442.0054646 INFO: Filtered count:0 out of 1527\n",
      "1723050442.272072 INFO: Expansion time:Glc.O2:0.6389888819994667\n",
      "1723050442.274158 INFO: Filtered count:0 out of 1551\n",
      "1723050442.3932455 INFO: metabolites 1164\n",
      "1723050442.920984 INFO: Expansion time:Glc.TMAO:0.6456190359995162\n",
      "1723050442.9230745 INFO: Filtered count:0 out of 1551\n",
      "1723050443.1587298 INFO: reactions 1157\n",
      "1723050443.3844638 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050443.5691159 INFO: Expansion time:For.NO2:2.158581872000468\n",
      "1723050443.571716 INFO: Filtered count:0 out of 1611\n",
      "1723050443.9930608 INFO: Expansion time:Glc.DMSO:1.9860547150001366\n",
      "1723050443.9956527 INFO: Filtered count:0 out of 1527\n",
      "1723050444.0002446 INFO: Expansion time:Glyc.O2:1.0756625519998124\n",
      "1723050444.0026872 INFO: Filtered count:0 out of 1551\n",
      "1723050444.0737324 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/163/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050444.8102643 INFO: Expansion time:For.NO3:1.2369097290002173\n",
      "1723050444.81281 INFO: Filtered count:0 out of 1611\n",
      "1723050445.10307 INFO: Expansion time:Pyr:1.0988286450001397\n",
      "1723050445.1058428 INFO: Filtered count:0 out of 1551\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050446.1161873 INFO: cpd05178 not found in model!\n",
      "1723050446.118529 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050446.1502876 INFO: Expansion time:Pyr.DMSO:1.0428054690000863\n",
      "1723050446.1527727 INFO: Filtered count:0 out of 1551\n",
      "1723050446.172128 INFO: Expansion time:Glc.O2:2.1750124529999084\n",
      "1723050446.174612 INFO: Filtered count:0 out of 1527\n",
      "1723050446.3722801 INFO: Expansion time:For.O2:1.557869293999829\n",
      "1723050446.3742929 INFO: Filtered count:0 out of 1611\n",
      "1723050447.248856 INFO: Expansion time:Pyr.NO:1.0944634229999792\n",
      "1723050447.2513075 INFO: Filtered count:0 out of 1551\n",
      "1723050447.4585462 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050447.7161846 INFO: Expansion time:Fum:1.3406815079997614\n",
      "1723050447.7185328 INFO: Filtered count:0 out of 1611\n",
      "1723050448.2767093 INFO: Expansion time:Glc.TMAO:2.100635491000503\n",
      "1723050448.279214 INFO: Filtered count:0 out of 1527\n",
      "1723050448.283613 INFO: Expansion time:Pyr.NO2:1.0307483990000037\n",
      "1723050448.285812 INFO: Filtered count:0 out of 1551\n",
      "1723050448.932888 INFO: Expansion time:Fum:3.6672539250002956\n",
      "1723050448.9353788 INFO: Filtered count:0 out of 1355\n",
      "1723050449.1442688 INFO: Expansion time:Fum.O2:1.424323070000355\n",
      "1723050449.1464622 INFO: Filtered count:0 out of 1611\n",
      "1723050449.2806108 INFO: Expansion time:Pyr.NO3:0.9932437470006334\n",
      "1723050449.2829926 INFO: Filtered count:0 out of 1551\n",
      "1723050449.6953712 INFO: Expansion time:Glyc:1.4146790069999042\n",
      "1723050449.6979346 INFO: Filtered count:0 out of 1527\n",
      "1723050450.0261397 INFO: Expansion time:Fum.O2:1.089296473000104\n",
      "1723050450.028583 INFO: Filtered count:0 out of 1355\n",
      "1723050450.7630794 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050451.1529644 INFO: cpd05178 not found in model!\n",
      "1723050451.1554573 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050451.529686 INFO: cpd05178 not found in model!\n",
      "1723050451.5322678 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050451.6946626 INFO: Expansion time:Glyc.O2:2.546893223000552\n",
      "1723050451.6968174 INFO: Filtered count:0 out of 1611\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.34.pg.G.D GF: 11838 Prob: 3432 Other: 3082\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050452.6087894 INFO: Expansion time:Pyr.O2:3.324294328999713\n",
      "1723050452.6113 INFO: Filtered count:0 out of 1551\n",
      "1723050452.7363477 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050452.9216363 INFO: Expansion time:Pyr:2.8915036009993855\n",
      "1723050452.9235728 INFO: Filtered count:0 out of 1355\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050453.6038895 INFO: Expansion time:Pyr.DMSO:0.679182967000088\n",
      "1723050453.6057937 INFO: Filtered count:0 out of 1355\n",
      "1723050453.6121125 INFO: Expansion time:Pyr:1.9137706840001556\n",
      "1723050453.6142614 INFO: Filtered count:0 out of 1611\n",
      "1723050453.7423232 INFO: Expansion time:Pyr.SO3:1.1295089569994161\n",
      "1723050453.7447543 INFO: Filtered count:0 out of 1551\n",
      "1723050454.371714 INFO: Expansion time:Pyr.NO:0.764747375999832\n",
      "1723050454.3738575 INFO: Filtered count:0 out of 1355\n",
      "1723050454.5378456 INFO: Expansion time:Glyc.O2:4.838468740999815\n",
      "1723050454.540644 INFO: Filtered count:0 out of 1527\n",
      "1723050454.716375 INFO: Expansion time:Pyr.DMSO:1.1005648979999023\n",
      "1723050454.7190828 INFO: Filtered count:0 out of 1611\n",
      "1723050454.8170385 INFO: Expansion time:Pyr.SO4:1.0705898109999907\n",
      "1723050454.8193865 INFO: Filtered count:0 out of 1551\n",
      "1723050455.1935976 INFO: Expansion time:Pyr.NO2:0.818506630000229\n",
      "1723050455.1959445 INFO: Filtered count:0 out of 1355\n",
      "1723050455.7787588 INFO: Expansion time:LLac.O2:1.2360068419993695\n",
      "1723050455.7815104 INFO: Filtered count:0 out of 1527\n",
      "1723050455.8538713 INFO: Expansion time:Pyr.TMAO:1.0329571130005206\n",
      "1723050455.8563871 INFO: Filtered count:0 out of 1551\n",
      "1723050455.8774238 INFO: Expansion time:Pyr.NO:1.156926526999996\n",
      "1723050455.8798254 INFO: Filtered count:0 out of 1611\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[119 'R1_A_D1.26.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:31:01+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '34ecc12c246979e325fb1abbe548a3a1'\n",
      " 11456451 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050456.5243356 INFO: Expansion time:Pyr.NO3:1.326848596999298\n",
      "1723050456.526954 INFO: Filtered count:0 out of 1355\n",
      "1723050456.5675876 INFO: metabolites 1204\n",
      "1723050456.883897 INFO: Expansion time:empty:1.0259562270002789\n",
      "1723050456.886555 INFO: Filtered count:0 out of 1551\n",
      "1723050457.1041162 INFO: Expansion time:LLac.SO4:1.3209941149998485\n",
      "1723050457.106673 INFO: Filtered count:0 out of 1527\n",
      "1723050457.2977061 INFO: Expansion time:Pyr.NO2:1.416471268999885\n",
      "1723050457.2997627 INFO: Filtered count:0 out of 1611\n",
      "1723050457.351456 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050457.4087586 INFO: reactions 1225\n",
      "1723050457.6269922 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050457.684538 INFO: Expansion time:Pyr.O2:1.156046840000272\n",
      "1723050457.68694 INFO: Filtered count:0 out of 1355\n",
      "1723050457.7664607 INFO: cpd05178 not found in model!\n",
      "1723050458.0076215 INFO: cpd05178 not found in model!\n",
      "1723050458.0098927 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050458.4337113 INFO: Expansion time:LLac.SO4.H2:1.3255141800000274\n",
      "1723050458.4363337 INFO: Filtered count:0 out of 1527\n",
      "1723050458.4851553 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/464/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050458.9681706 INFO: Expansion time:Pyr.SO3:1.2797026950001964\n",
      "1723050458.970604 INFO: Filtered count:0 out of 1355\n",
      "1723050458.9990728 INFO: Expansion time:Pyr.NO3:1.6981215710002289\n",
      "1723050459.0015488 INFO: Filtered count:0 out of 1611\n",
      "1723050459.3413026 INFO: cpd05178 not found in model!\n",
      "1723050459.6687906 INFO: cpd05178 not found in model!\n",
      "1723050459.671019 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050459.6929264 INFO: Expansion time:Llac:1.2551500330000636\n",
      "1723050459.695622 INFO: Filtered count:0 out of 1527\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050460.0100982 INFO: Expansion time:Pyr.SO4:1.037916304000646\n",
      "1723050460.0123727 INFO: Filtered count:0 out of 1355\n",
      "1723050460.3084269 INFO: Expansion time:Pyr.O2:1.3053535680001005\n",
      "1723050460.3105059 INFO: Filtered count:0 out of 1611\n",
      "1723050460.592542 INFO: cpd05178 not found in model!\n",
      "1723050460.5947125 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050460.7882173 INFO: Expansion time:Etho.O2:0.7176167309999073\n",
      "1723050460.7905047 INFO: Filtered count:0 out of 1767\n",
      "1723050461.156892 INFO: Expansion time:Pyr.TMAO:1.1430905639999764\n",
      "1723050461.1593332 INFO: Filtered count:0 out of 1355\n",
      "1723050461.2210972 INFO: Expansion time:Pyr:1.5240114449998146\n",
      "1723050461.2237234 INFO: Filtered count:0 out of 1527\n",
      "1723050461.2637043 INFO: cpd05178 not found in model!\n",
      "1723050461.2659843 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050461.4969108 INFO: Expansion time:Glc:0.7051774339997792\n",
      "1723050461.4990983 INFO: Filtered count:0 out of 1767\n",
      "1723050461.6225395 INFO: Expansion time:Pyr.SO3:1.3108230699999694\n",
      "1723050461.6244142 INFO: Filtered count:0 out of 1611\n",
      "1723050461.7234616 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050462.0705712 INFO: cpd05178 not found in model!\n",
      "1723050462.073288 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050462.0956423 INFO: Expansion time:empty:0.9348333079997246\n",
      "1723050462.0978906 INFO: Filtered count:0 out of 1355\n",
      "1723050462.2489052 INFO: Expansion time:Glc.DMSO:0.7486122339996655\n",
      "1723050462.2514057 INFO: Filtered count:0 out of 1767\n",
      "1723050462.2693503 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050462.5870008 INFO: Expansion time:Pyr.SO4:0.9614108320001833\n",
      "1723050462.589336 INFO: Filtered count:0 out of 1611\n",
      "1723050462.6970246 INFO: Expansion time:Pyr.DMSO:1.471824822000599\n",
      "1723050462.6998384 INFO: Filtered count:0 out of 1527\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050463.1881618 INFO: Expansion time:Glc.O2:0.935614935000558\n",
      "1723050463.1904476 INFO: Filtered count:0 out of 1767\n",
      "1723050463.7066765 INFO: Expansion time:Pyr.TMAO:1.1158313000005364\n",
      "1723050463.7089536 INFO: Filtered count:0 out of 1611\n",
      "1723050463.894283 INFO: Expansion time:Glc.TMAO:0.7025419649999094\n",
      "1723050463.8963664 INFO: Filtered count:0 out of 1767\n",
      "1723050464.3546493 INFO: Expansion time:Pyr.NO:1.6533560999996553\n",
      "1723050464.3572915 INFO: Filtered count:0 out of 1527\n",
      "1723050464.5643294 INFO: Expansion time:Glyc.O2:0.6667603920004694\n",
      "1723050464.5665166 INFO: Filtered count:0 out of 1767\n",
      "1723050465.17566 INFO: Expansion time:empty:1.4652470690007249\n",
      "1723050465.177668 INFO: Filtered count:0 out of 1611\n",
      "1723050465.2475975 INFO: Expansion time:Pyr:0.679888622999897\n",
      "1723050465.2497478 INFO: Filtered count:0 out of 1767\n",
      "1723050465.5596411 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050465.870746 INFO: cpd05178 not found in model!\n",
      "1723050465.8729587 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050466.0383964 INFO: Expansion time:Pyr.NO2:1.6794815279999966\n",
      "1723050466.0410066 INFO: Filtered count:0 out of 1527\n",
      "1723050466.1275592 INFO: cpd05178 not found in model!\n",
      "1723050466.1296458 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_H2O.11.pg.G.D GF: 11665 Prob: 3746 Other: 2943\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050467.160128 INFO: Expansion time:Pyr.DMSO:1.9090807080001468\n",
      "1723050467.1624684 INFO: Filtered count:0 out of 1767\n",
      "1723050467.5152605 INFO: Expansion time:Pyr.NO3:1.4727736390004793\n",
      "1723050467.517921 INFO: Filtered count:0 out of 1527\n",
      "1723050467.8760571 INFO: Expansion time:Pyr.NO:0.7123266770004193\n",
      "1723050467.8781805 INFO: Filtered count:0 out of 1767\n",
      "1723050467.9163764 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050468.391228 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050468.4510531 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050468.5475683 INFO: cpd05178 not found in model!\n",
      "1723050468.5500302 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050468.565417 INFO: Expansion time:Pyr.NO2:0.6860181120000561\n",
      "1723050468.5675921 INFO: Filtered count:0 out of 1767\n",
      "1723050468.6015594 INFO: cpd05178 not found in model!\n",
      "1723050468.6044505 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050469.1169665 INFO: Expansion time:Pyr.O2:1.597395910999694\n",
      "1723050469.1195328 INFO: Filtered count:0 out of 1527\n",
      "1723050469.268159 INFO: Expansion time:Pyr.NO3:0.6993606500000169\n",
      "1723050469.2705266 INFO: Filtered count:0 out of 1767\n",
      "1723050469.960264 INFO: Expansion time:Pyr.O2:0.6884392000001753\n",
      "1723050469.9624538 INFO: Filtered count:0 out of 1767\n",
      "1723050470.5486531 INFO: Expansion time:Pyr.SO3:1.4275257429999328\n",
      "1723050470.551227 INFO: Filtered count:0 out of 1527\n",
      "1723050470.6974287 INFO: Expansion time:Pyr.SO3:0.7337295880006423\n",
      "1723050470.699807 INFO: Filtered count:0 out of 1767\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050471.4342825 INFO: Expansion time:Pyr.SO4:0.733189370999753\n",
      "1723050471.4365342 INFO: Filtered count:0 out of 1767\n",
      "1723050472.2520297 INFO: Expansion time:Pyr.SO4:1.699367574000462\n",
      "1723050472.2547703 INFO: Filtered count:0 out of 1527\n",
      "1723050472.9213998 INFO: cpd05178 not found in model!\n",
      "1723050473.1522808 INFO: cpd05178 not found in model!\n",
      "1723050473.154686 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050473.363459 INFO: Expansion time:Pyr.TMAO:1.9255709810004191\n",
      "1723050473.3656142 INFO: Filtered count:0 out of 1767\n",
      "1723050473.9267488 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050474.1278152 INFO: Expansion time:empty:0.7609514249998028\n",
      "1723050474.1299224 INFO: Filtered count:0 out of 1767\n",
      "1723050475.1434276 INFO: cpd05178 not found in model!\n",
      "1723050475.1459706 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050476.2333522 INFO: cpd05178 not found in model!\n",
      "1723050476.235592 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D1.10.pg.G.D GF: 11288 Prob: 4010 Other: 3056\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050476.5778236 INFO: Expansion time:Pyr.TMAO:4.321520998999404\n",
      "1723050476.5803425 INFO: Filtered count:0 out of 1527\n",
      "1723050477.4850647 INFO: Expansion time:empty:0.9033295859999271\n",
      "1723050477.4876754 INFO: Filtered count:0 out of 1527\n",
      "1723050477.8967674 INFO: Objective with gapfill database:182.19713049412988; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050478.119855 INFO: cpd05178 not found in model!\n",
      "1723050478.3109698 INFO: cpd05178 not found in model!\n",
      "1723050478.313301 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050478.5051782 INFO: cpd05178 not found in model!\n",
      "1723050478.5079706 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050481.1987464 INFO: Expansion time:Ac.O2:3.1237253090002923\n",
      "1723050481.2009475 INFO: Filtered count:0 out of 1431\n",
      "1723050482.572816 INFO: Expansion time:Dlac.O2:1.3706407689996922\n",
      "1723050482.5753024 INFO: Filtered count:0 out of 1431\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[123 'R2_restored_C_black.25.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:32:01+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '8bd6a9da5d0da4e340373b2555dd0909'\n",
      " 13305194 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050483.284918 INFO: metabolites 1081\n",
      "1723050483.449543 INFO: Expansion time:For.O2:0.8729129189996456\n",
      "1723050483.4517257 INFO: Filtered count:0 out of 1431\n",
      "1723050483.9641676 INFO: reactions 1113\n",
      "1723050484.0423708 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050484.1424012 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050484.4976223 INFO: cpd05178 not found in model!\n",
      "1723050484.7983387 INFO: Default biomass: [bio1]\n",
      "1723050484.8093998 INFO: cpd05178 not found in model!\n",
      "1723050484.8119566 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/486/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050486.3873153 INFO: Expansion time:Fum:2.934438869000587\n",
      "1723050486.3895214 INFO: Filtered count:0 out of 1431\n",
      "1723050487.3681087 INFO: cpd05178 not found in model!\n",
      "1723050487.370941 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050487.5059474 INFO: cpd05178 not found in model!\n",
      "1723050487.5083592 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D1.26.pg.G.D GF: 11112 Prob: 4308 Other: 2934\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050488.0579913 INFO: cpd05178 not found in model!\n",
      "1723050488.0605066 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050488.7701807 INFO: Expansion time:Fum.O2:2.3792662500000006\n",
      "1723050488.7727296 INFO: Filtered count:0 out of 1431\n",
      "1723050491.2684498 INFO: Expansion time:Glyc.O2:2.4942469500001607\n",
      "1723050491.271068 INFO: Filtered count:0 out of 1431\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050492.9859009 INFO: Expansion time:Pyr:1.7133043710000493\n",
      "1723050492.9883397 INFO: Filtered count:0 out of 1431\n",
      "1723050493.5281062 INFO: cpd05178 not found in model!\n",
      "1723050493.530635 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050494.9327033 INFO: Expansion time:Glc:2.459798034999949\n",
      "1723050494.9350102 INFO: Filtered count:0 out of 1665\n",
      "1723050495.2995467 INFO: cpd05178 not found in model!\n",
      "1723050495.5302422 INFO: cpd05178 not found in model!\n",
      "1723050495.532453 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050495.6801615 INFO: Expansion time:Glc.DMSO:0.7437587120002718\n",
      "1723050495.6821656 INFO: Filtered count:0 out of 1665\n",
      "1723050496.3471794 INFO: Expansion time:Glc.O2:0.6637591179996889\n",
      "1723050496.3495038 INFO: Filtered count:0 out of 1665\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[75 'R2A_A_D2.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:22:28+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'a93bb955e2f0eb830cbe20adfaeda92c'\n",
      " 6701169 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050496.6443586 INFO: metabolites 674\n",
      "1723050496.9456234 INFO: reactions 605\n",
      "1723050497.0399053 INFO: Expansion time:Glc.TMAO:0.6891671599996698\n",
      "1723050497.0420763 INFO: Filtered count:0 out of 1665\n",
      "1723050497.0445948 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050497.4150329 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/301/1\n",
      "[198 'R1_A_D2.109.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:46:40+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e10dc07b7972a68b71d63a64f61a6589'\n",
      " 13513353 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050497.7099476 INFO: Expansion time:Pyr:0.666565635000552\n",
      "1723050497.7120104 INFO: Filtered count:0 out of 1665\n",
      "1723050498.016465 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050498.187196 INFO: metabolites 1090\n",
      "1723050498.363879 INFO: Expansion time:Pyr.DMSO:0.6506657560003077\n",
      "1723050498.3660104 INFO: Filtered count:0 out of 1665\n",
      "1723050498.5131965 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050498.8698337 INFO: reactions 1064\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050499.0403252 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050499.0461473 INFO: Expansion time:Pyr.NO:0.6789484400005676\n",
      "1723050499.0481231 INFO: Filtered count:0 out of 1665\n",
      "1723050499.7270513 INFO: Expansion time:Pyr.NO2:0.677694306000376\n",
      "1723050499.7290385 INFO: Filtered count:0 out of 1665\n",
      "1723050500.1888576 INFO: cpd05178 not found in model!\n",
      "1723050500.2465718 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050500.3875816 INFO: Expansion time:Pyr.NO3:0.6573831980003888\n",
      "1723050500.3898463 INFO: Filtered count:0 out of 1665\n",
      "1723050500.52898 INFO: cpd05178 not found in model!\n",
      "1723050500.531594 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050500.916464 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/335/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050501.0785384 INFO: Expansion time:Pyr.O2:0.687399457000538\n",
      "1723050501.080605 INFO: Filtered count:0 out of 1665\n",
      "1723050501.769017 INFO: Expansion time:Pyr.SO3:0.6872740930002692\n",
      "1723050501.7712607 INFO: Filtered count:0 out of 1665\n",
      "1723050503.0077128 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050504.1391253 INFO: Expansion time:Pyr.SO4:2.3666625939995356\n",
      "1723050504.1395092 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050504.1415493 INFO: Filtered count:0 out of 1665\n",
      "1723050504.810356 INFO: Expansion time:Pyr.TMAO:0.6671488840001984\n",
      "1723050504.8126004 INFO: Filtered count:0 out of 1665\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00062_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050505.451653 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050505.495171 INFO: Expansion time:empty:0.6813357789997099\n",
      "1723050505.4972608 INFO: Filtered count:0 out of 1665\n",
      "1723050505.528826 INFO: Expansion time:Pyr.DMSO:12.539045825999892\n",
      "1723050505.5312095 INFO: Filtered count:1 out of 1431\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050506.2344031 INFO: Expansion time:Pyr.NO:0.7016399929998443\n",
      "1723050506.2362952 INFO: Filtered count:1 out of 1431\n",
      "1723050506.5711713 INFO: cpd05178 not found in model!\n",
      "1723050506.5731528 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050506.8798134 INFO: Expansion time:Pyr.NO2:0.6423283730000549\n",
      "1723050506.881831 INFO: Filtered count:1 out of 1431\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[11 'R1_B_D2.18.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:09:49+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e92c4b6fb1f60febaf884a0bf77cd5ca'\n",
      " 13184905 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050507.5159116 INFO: Expansion time:Pyr.NO3:0.632786987000145\n",
      "1723050507.5179856 INFO: Filtered count:1 out of 1431\n",
      "1723050507.7528386 INFO: metabolites 1190\n",
      "1723050508.1606364 INFO: Expansion time:Pyr.O2:0.6414694510003756\n",
      "1723050508.1627827 INFO: Filtered count:1 out of 1431\n",
      "1723050508.3212745 INFO: reactions 1157\n",
      "1723050508.478081 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050508.8091362 INFO: Expansion time:Pyr.SO3:0.64516104500035\n",
      "1723050508.8112679 INFO: Filtered count:1 out of 1431\n",
      "1723050508.9435894 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/389/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050511.018865 INFO: Expansion time:Pyr.SO4:2.2063668079999843\n",
      "1723050511.0210924 INFO: Filtered count:1 out of 1431\n",
      "1723050511.0845165 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050511.1192477 INFO: cpd05178 not found in model!\n",
      "1723050511.1213076 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050511.5365994 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050511.6751742 INFO: Expansion time:Pyr.TMAO:0.6527155529993252\n",
      "1723050511.6773517 INFO: Filtered count:1 out of 1431\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050512.3332868 INFO: Expansion time:empty:0.6545116719998987\n",
      "1723050512.335313 INFO: Filtered count:1 out of 1431\n",
      "1723050512.486718 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050515.7671406 INFO: Expansion time:Ac.NO2:1.128082725999775\n",
      "1723050515.769529 INFO: Filtered count:0 out of 1760\n",
      "1723050516.2167482 INFO: cpd05178 not found in model!\n",
      "1723050516.2195253 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_C_black.25.pg.G.D GF: 11422 Prob: 3840 Other: 3090\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050516.749705 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050516.8092823 INFO: Expansion time:Ac.NO3:1.0382401309998386\n",
      "1723050516.811757 INFO: Filtered count:0 out of 1760\n",
      "1723050517.8201072 INFO: Expansion time:Ac.O2:1.0067882880002799\n",
      "1723050517.8224926 INFO: Filtered count:0 out of 1760\n",
      "1723050518.8957255 INFO: Expansion time:Etho.O2:1.0717678460005118\n",
      "1723050518.8982973 INFO: Filtered count:0 out of 1760\n",
      "1723050519.5787263 INFO: cpd05178 not found in model!\n",
      "1723050519.933589 INFO: cpd05178 not found in model!\n",
      "1723050519.936182 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050520.1043444 INFO: Expansion time:For.NO2:1.204541065999365\n",
      "1723050520.106911 INFO: Filtered count:0 out of 1760\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[27 'R1_A_D2.50.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:12:47+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'c771249adb22b08e30dcd54830a16753'\n",
      " 7627573 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050521.1663175 INFO: metabolites 914\n",
      "1723050521.6782303 INFO: reactions 858\n",
      "1723050521.8235636 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050522.331699 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/478/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050522.4440396 INFO: cpd05178 not found in model!\n",
      "1723050522.4467294 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D2.16.pg.G.D GF: 12790 Prob: 2166 Other: 3404\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050523.55196 INFO: Expansion time:For.NO3:3.4434532690002015\n",
      "1723050523.5542305 INFO: Filtered count:0 out of 1760\n",
      "1723050523.7582765 INFO: cpd05178 not found in model!\n",
      "1723050523.7601914 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050524.3734257 INFO: Expansion time:For.O2:0.8178483729998334\n",
      "1723050524.375299 INFO: Filtered count:0 out of 1760\n",
      "1723050525.123935 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050525.1654828 INFO: Expansion time:Fum:0.7890682229999584\n",
      "1723050525.1677728 INFO: Filtered count:0 out of 1760\n",
      "1723050525.3706565 INFO: cpd05178 not found in model!\n",
      "1723050525.3731759 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050525.5922096 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050526.1608853 INFO: Expansion time:Fum.O2:0.9914708509995762\n",
      "1723050526.163193 INFO: Filtered count:0 out of 1760\n",
      "1723050526.249385 INFO: cpd05178 not found in model!\n",
      "1723050526.251595 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.109.pg.G.D GF: 11807 Prob: 3422 Other: 3123\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050527.1474314 INFO: Expansion time:LLac.O2:0.9827731499999572\n",
      "1723050527.149808 INFO: Filtered count:0 out of 1760\n",
      "1723050528.2042375 INFO: Expansion time:Pyr:1.0529698549999011\n",
      "1723050528.2064748 INFO: Filtered count:0 out of 1760\n",
      "1723050529.037494 INFO: Expansion time:Pyr.DMSO:0.8294881550000355\n",
      "1723050529.0393715 INFO: Filtered count:0 out of 1760\n",
      "1723050529.1461446 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050529.7271457 INFO: Expansion time:Pyr.NO:0.6865963079999347\n",
      "1723050529.7292447 INFO: Filtered count:0 out of 1760\n",
      "1723050530.3770444 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050530.8158684 INFO: Expansion time:Pyr.NO2:1.0851512650006043\n",
      "1723050530.8182793 INFO: Filtered count:0 out of 1760\n",
      "1723050530.8997676 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050531.429531 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050531.9091685 INFO: Expansion time:Pyr.NO3:1.0893943409992062\n",
      "1723050531.9115698 INFO: Filtered count:0 out of 1760\n",
      "1723050531.9154754 INFO: cpd05178 not found in model!\n",
      "1723050531.9172225 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050532.574481 INFO: cpd05178 not found in model!\n",
      "1723050532.5767274 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050533.7727938 INFO: cpd05178 not found in model!\n",
      "1723050533.7749445 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050535.3164728 INFO: Expansion time:Pyr.O2:3.403359195000121\n",
      "1723050535.3188756 INFO: Filtered count:0 out of 1760\n",
      "1723050536.1493912 INFO: cpd05178 not found in model!\n",
      "1723050536.1516404 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050536.2354803 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050536.4346948 INFO: Expansion time:Pyr.SO3:1.1142127560005974\n",
      "1723050536.437308 INFO: Filtered count:0 out of 1760\n",
      "1723050536.8171947 INFO: cpd05178 not found in model!\n",
      "1723050537.055954 INFO: cpd05178 not found in model!\n",
      "1723050537.0580964 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050537.6253288 INFO: Expansion time:Pyr.SO4:1.1864077449999968\n",
      "1723050537.6280239 INFO: Filtered count:0 out of 1760\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[55 'R2_restored_DShore.15.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:18:32+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '20706514aa392b432eab6042a633d7c2'\n",
      " 15114985 None]\n",
      "The R2_restored_DShore.15.pg.G.D.pyr object is not in the 188065 worksheet\n",
      "JSONRPCError: -32500. No object with name R2_restored_DShore.15.pg.G.D.pyr exists in workspace 188065 (name jplfaria:narrative_1722386311348)\n",
      "us.kbase.workspace.database.exceptions.NoSuchObjectException: No object with name R2_restored_DShore.15.pg.G.D.pyr exists in workspace 188065 (name jplfaria:narrative_1722386311348)\n",
      "\tat us.kbase.workspace.database.mongo.MongoWorkspaceDB.queryObjects(MongoWorkspaceDB.java:3121)\n",
      "\tat us.kbase.workspace.database.mongo.MongoWorkspaceDB.resolveObjectIDs(MongoWorkspaceDB.java:3066)\n",
      "\tat us.kbase.workspace.database.mongo.MongoWorkspaceDB.getObjects(MongoWorkspaceDB.java:2354)\n",
      "\tat us.kbase.workspace.database.Workspace.getObjects(Workspace.java:1130)\n",
      "\tat us.kbase.workspace.database.Workspace.getObjects(Workspace.java:1069)\n",
      "\tat us.kbase.workspace.kbase.WorkspaceServerMethods.getObjects(WorkspaceServerMethods.java:479)\n",
      "\tat us.kbase.workspace.WorkspaceServer.getObjects2(WorkspaceServer.java:817)\n",
      "\tat jdk.internal.reflect.GeneratedMethodAccessor102.invoke(Unknown Source)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n",
      "\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\n",
      "\tat us.kbase.common.service.JsonServerServlet.processRpcCall(JsonServerServlet.java:681)\n",
      "\tat us.kbase.common.service.JsonServerServlet.doPost(JsonServerServlet.java:553)\n",
      "\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:661)\n",
      "\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:742)\n",
      "\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)\n",
      "\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)\n",
      "\tat org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)\n",
      "\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)\n",
      "\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)\n",
      "\tat org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:200)\n",
      "\tat org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)\n",
      "\tat org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)\n",
      "\tat org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:137)\n",
      "\tat org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)\n",
      "\tat org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:660)\n",
      "\tat org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)\n",
      "\tat org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)\n",
      "\tat org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:798)\n",
      "\tat org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)\n",
      "\tat org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:806)\n",
      "\tat org.apache.tomcat.util.net.Nio2Endpoint$SocketProcessor.doRun(Nio2Endpoint.java:1735)\n",
      "\tat org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)\n",
      "\tat org.apache.tomcat.util.net.AbstractEndpoint.processSocket(AbstractEndpoint.java:1069)\n",
      "\tat org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$3.completed(Nio2Endpoint.java:575)\n",
      "\tat org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$3.completed(Nio2Endpoint.java:553)\n",
      "\tat java.base/sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:127)\n",
      "\tat java.base/sun.nio.ch.Invoker$2.run(Invoker.java:219)\n",
      "\tat java.base/sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:112)\n",
      "\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n",
      "\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:829)\n",
      "\n",
      "188040/608/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050538.7954445 INFO: Expansion time:Pyr.TMAO:1.165774624999358\n",
      "1723050538.7979283 INFO: Filtered count:0 out of 1760\n",
      "1723050539.412455 INFO: cpd05178 not found in model!\n",
      "1723050539.4147482 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050539.852122 INFO: Expansion time:empty:1.0527599870001723\n",
      "1723050539.855248 INFO: Filtered count:0 out of 1760\n",
      "1723050541.8327382 INFO: cpd05178 not found in model!\n",
      "1723050541.8352945 CRITICAL: Metabolite cpd05178 not found in model\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.18.pg.G.D GF: 11202 Prob: 4230 Other: 2922\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050542.5804436 INFO: Objective with gapfill database:188.5803685467611; min objective:0.01\n",
      "1723050543.0512426 INFO: Objective with gapfill database:116.38775092164673; min objective:0.01\n",
      "1723050545.2711535 INFO: cpd05178 not found in model!\n",
      "1723050545.2734528 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050546.674401 INFO: cpd05178 not found in model!\n",
      "1723050546.6771555 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050546.8648539 INFO: cpd05178 not found in model!\n",
      "1723050546.8671346 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050548.3605375 INFO: cpd05178 not found in model!\n",
      "1723050548.3627992 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050548.702653 INFO: Expansion time:Etho.O2:6.594467094999345\n",
      "1723050548.7052395 INFO: Filtered count:0 out of 1584\n",
      "1723050548.9503627 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050549.2359276 INFO: cpd05178 not found in model!\n",
      "1723050549.238778 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.50.pg.G.D GF: 12167 Prob: 2868 Other: 3315\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050550.1229575 INFO: Expansion time:For.O2:1.416242183000577\n",
      "1723050550.1255496 INFO: Filtered count:0 out of 1584\n",
      "1723050551.921399 INFO: cpd05178 not found in model!\n",
      "1723050552.2065127 INFO: cpd05178 not found in model!\n",
      "1723050552.2087176 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050553.139478 INFO: Expansion time:Fum:3.0122888120004063\n",
      "1723050553.1419685 INFO: Filtered count:0 out of 1584\n",
      "1723050553.362839 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050554.4420052 INFO: cpd05178 not found in model!\n",
      "1723050554.44408 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050556.21406 INFO: Expansion time:Fum.O2:3.070595671999399\n",
      "1723050556.2165647 INFO: Filtered count:0 out of 1584\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[87 'R1_C_D1.19.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:24:45+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '77a819ed1534f8b5366825154b5b6321'\n",
      " 10533654 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050556.8922377 INFO: metabolites 1024\n",
      "1723050557.4277859 INFO: reactions 997\n",
      "1723050557.5985732 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050557.8656013 INFO: cpd05178 not found in model!\n",
      "1723050557.8682742 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050558.215947 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/204/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050558.5332096 INFO: Expansion time:Pyr:2.31506098599948\n",
      "1723050558.53577 INFO: Filtered count:0 out of 1584\n",
      "1723050560.0376213 INFO: Expansion time:Pyr.DMSO:1.5002042589994744\n",
      "1723050560.0401149 INFO: Filtered count:0 out of 1584\n",
      "1723050560.525905 INFO: cpd05178 not found in model!\n",
      "1723050560.5281887 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[238 'R1_A_D2.51.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:54:30+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'd04a9813e58e9c2de15a43e379e0b8e7'\n",
      " 8035717 None]\n",
      "R2_restored_DShore.15.pg.G.D GF: 13206 Prob: 0 Other: 5146\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050561.0908964 INFO: cpd05178 not found in model!\n",
      "1723050561.093 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050561.1206415 INFO: metabolites 961\n",
      "1723050561.6501713 INFO: reactions 902\n",
      "1723050561.7996688 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050562.1312523 INFO: cpd05178 not found in model!\n",
      "1723050562.13349 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050562.3808875 INFO: Default biomass: [bio1]\n",
      "1723050562.423984 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/38/1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050562.426099 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050562.4701476 INFO: cpd05178 not found in model!\n",
      "1723050562.4725585 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050563.539782 INFO: Objective with gapfill database:108.31898744956115; min objective:0.01\n",
      "1723050563.5967731 INFO: Expansion time:Pyr.NO:3.5552142500000627\n",
      "1723050563.5992885 INFO: Filtered count:0 out of 1584\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050565.3273404 INFO: Expansion time:Pyr.NO2:1.7263712270005271\n",
      "1723050565.3297746 INFO: Filtered count:0 out of 1584\n",
      "1723050565.8953369 INFO: cpd05178 not found in model!\n",
      "1723050565.8976612 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050566.4487817 INFO: Expansion time:Pyr.NO3:1.1174141689998578\n",
      "1723050566.4511375 INFO: Filtered count:0 out of 1584\n",
      "1723050566.8598888 INFO: cpd05178 not found in model!\n",
      "1723050566.8625648 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050567.5971909 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050567.836388 INFO: Expansion time:Ac.NO2:3.082809301999987\n",
      "1723050567.8390687 INFO: Filtered count:0 out of 1666\n",
      "1723050568.126477 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050568.4874175 INFO: Expansion time:Pyr.O2:2.0346797149995837\n",
      "1723050568.4899065 INFO: Filtered count:0 out of 1584\n",
      "1723050569.0698674 INFO: Expansion time:Ac.O2:1.2293530870001632\n",
      "1723050569.0726073 INFO: Filtered count:0 out of 1666\n",
      "1723050569.587094 INFO: cpd05178 not found in model!\n",
      "1723050569.5893621 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050569.7714853 INFO: cpd05178 not found in model!\n",
      "1723050570.0980442 INFO: cpd05178 not found in model!\n",
      "1723050570.1006505 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050570.2499907 INFO: Expansion time:Etho.O2:1.1758877719994416\n",
      "1723050570.2528172 INFO: Filtered count:0 out of 1666\n",
      "1723050570.257708 INFO: cpd05178 not found in model!\n",
      "1723050570.259486 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050571.1142814 INFO: Expansion time:Pyr.SO3:2.622881950000192\n",
      "1723050571.1167974 INFO: Filtered count:0 out of 1584\n",
      "1723050571.464434 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050571.921277 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050571.9859138 INFO: Expansion time:Fum:1.7316113899996708\n",
      "1723050571.9885156 INFO: Filtered count:0 out of 1666\n",
      "1723050572.8417573 INFO: Expansion time:Pyr.SO4:1.7234432639997976\n",
      "1723050572.844246 INFO: Filtered count:0 out of 1584\n",
      "1723050573.0821152 INFO: Objective with gapfill database:176.72496329768458; min objective:0.01\n",
      "1723050573.1111367 INFO: cpd05178 not found in model!\n",
      "1723050573.3579524 INFO: cpd05178 not found in model!\n",
      "1723050573.3599946 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050573.6487339 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050573.816119 INFO: Expansion time:Fum.O2:1.8261404529994252\n",
      "1723050573.8189445 INFO: Filtered count:0 out of 1666\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[23 'SF2_C_D2.25.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:12:06+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '0930c291193bf02e88e4ffb18bd96a32'\n",
      " 10911951 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050574.922038 INFO: Expansion time:Pyr.TMAO:2.0762146730003224\n",
      "1723050574.9244936 INFO: Filtered count:0 out of 1584\n",
      "1723050575.6465137 INFO: Expansion time:Pyr:1.8260968800004775\n",
      "1723050575.649134 INFO: Filtered count:0 out of 1666\n",
      "1723050575.885682 INFO: Expansion time:empty:0.9596619450003345\n",
      "1723050575.8879943 INFO: Filtered count:0 out of 1584\n",
      "1723050576.4496207 INFO: Expansion time:Pyr.DMSO:0.7988819140000487\n",
      "1723050576.4518943 INFO: Filtered count:0 out of 1666\n",
      "1723050576.7315245 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[5 'R2_B_D2.4.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:08:44+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'd0d1dbef3a80097cc8c21b4ccf56b85f'\n",
      " 19689749 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050577.2160032 INFO: Expansion time:Pyr.NO:0.7629488860002311\n",
      "1723050577.2183099 INFO: Filtered count:0 out of 1666\n",
      "1723050577.2289913 INFO: metabolites 1179\n",
      "1723050577.4673045 INFO: cpd05178 not found in model!\n",
      "1723050577.4654403 INFO: metabolites 918\n",
      "1723050577.469472 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050577.90578 INFO: reactions 1116\n",
      "1723050578.029637 INFO: reactions 830\n",
      "1723050578.082077 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050578.1630125 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050578.5150118 INFO: Default biomass: [bio1]\n",
      "1723050578.5291185 INFO: Expansion time:Pyr.NO2:1.3095896790000552\n",
      "1723050578.5312505 INFO: Filtered count:0 out of 1666\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/347/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050578.7418323 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/369/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050579.2834964 INFO: cpd05178 not found in model!\n",
      "1723050579.2858934 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050579.9881802 INFO: Expansion time:Pyr.NO3:1.4554208889994698\n",
      "1723050579.99078 INFO: Filtered count:0 out of 1666\n",
      "1723050580.2471454 INFO: cpd05178 not found in model!\n",
      "1723050580.2493823 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050581.881757 INFO: Expansion time:Pyr.O2:1.8894328930000484\n",
      "1723050581.8843627 INFO: Filtered count:0 out of 1666\n",
      "1723050585.137793 INFO: cpd05178 not found in model!\n",
      "1723050585.1404033 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050586.6238542 INFO: cpd05178 not found in model!\n",
      "1723050586.6263921 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050586.9709494 INFO: cpd05178 not found in model!\n",
      "1723050586.973582 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_D1.19.pg.G.D GF: 11949 Prob: 3256 Other: 3147\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050587.2039812 INFO: cpd05178 not found in model!\n",
      "1723050587.427229 INFO: cpd05178 not found in model!\n",
      "1723050587.4293976 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050587.785086 INFO: Expansion time:Pyr.SO3:5.899265312000352\n",
      "1723050587.7877202 INFO: Filtered count:0 out of 1666\n",
      "1723050588.8259974 INFO: cpd05178 not found in model!\n",
      "1723050588.828243 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.51.pg.G.D GF: 12124 Prob: 3018 Other: 3210\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050590.4612658 INFO: Expansion time:Pyr.SO4:2.6720027090004805\n",
      "1723050590.46381 INFO: Filtered count:0 out of 1666\n",
      "1723050591.530764 INFO: Expansion time:Pyr.TMAO:1.0654953599996588\n",
      "1723050591.5334315 INFO: Filtered count:0 out of 1666\n",
      "1723050592.4881997 INFO: Expansion time:empty:0.9531099770001674\n",
      "1723050592.4909031 INFO: Filtered count:0 out of 1666\n",
      "1723050594.2132814 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050594.6529367 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050595.1522179 INFO: cpd05178 not found in model!\n",
      "1723050595.154593 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[274 'SF2_C_D2.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:01:50+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'b1fd3d4238478eb56292e9cc351f966e'\n",
      " 10655338 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050596.1394289 INFO: metabolites 1181\n",
      "1723050596.9194255 INFO: cpd05178 not found in model!\n",
      "1723050596.9216847 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050597.0924196 INFO: cpd05178 not found in model!\n",
      "1723050597.0953777 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050597.8666663 INFO: reactions 1181\n",
      "1723050598.0964644 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050598.8492413 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/157/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050598.9827058 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050600.2280126 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050601.446153 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050602.0491564 INFO: cpd05178 not found in model!\n",
      "1723050602.273684 INFO: cpd05178 not found in model!\n",
      "1723050602.275972 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050603.9176764 INFO: Objective with gapfill database:144.32075214825855; min objective:0.01\n",
      "1723050604.6288562 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050607.5671506 INFO: cpd05178 not found in model!\n",
      "1723050607.5692418 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050607.9302313 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050608.3914962 INFO: cpd05178 not found in model!\n",
      "1723050608.393661 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050608.4680471 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050609.3914878 INFO: Objective with gapfill database:159.158156007144; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050612.3703375 INFO: cpd05178 not found in model!\n",
      "1723050612.3732538 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2.25.pg.G.D GF: 11687 Prob: 3726 Other: 2947\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050613.2788205 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050616.5330029 INFO: cpd05178 not found in model!\n",
      "1723050616.535656 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.4.pg.G.D GF: 11485 Prob: 3818 Other: 3051\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050624.2954118 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[219 'R2_A_D1.49.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:50:54+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '34132fa288fc088d521c6ccc87b410da'\n",
      " 12131344 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050624.5359447 INFO: cpd05178 not found in model!\n",
      "1723050624.5381582 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050625.0263138 INFO: metabolites 1202\n",
      "1723050625.4941728 INFO: cpd05178 not found in model!\n",
      "1723050625.496553 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050625.5535512 INFO: cpd05178 not found in model!\n",
      "1723050625.7266123 INFO: Objective with gapfill database:98.08674908691238; min objective:0.01\n",
      "1723050625.8330324 INFO: reactions 1189\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2.16.pg.G.D GF: 11479 Prob: 3814 Other: 3059\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050625.9106195 INFO: cpd05178 not found in model!\n",
      "1723050625.9131973 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050626.0829766 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050626.8015218 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/472/1\n",
      "[134 'R2A_A_D1.10.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:34:06+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'dd90bf1db84745fe8bb5bde9df56b656'\n",
      " 12973140 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050629.1375878 INFO: cpd05178 not found in model!\n",
      "1723050629.44576 INFO: cpd05178 not found in model!\n",
      "1723050629.448029 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050629.6365666 INFO: metabolites 1207\n",
      "1723050629.950093 INFO: cpd05178 not found in model!\n",
      "1723050629.952228 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050630.0850143 INFO: reactions 1196\n",
      "1723050630.2826948 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050630.962895 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/569/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050633.063044 INFO: Objective with gapfill database:177.31331113540628; min objective:0.01\n",
      "1723050635.7382665 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050637.7040834 INFO: cpd05178 not found in model!\n",
      "1723050637.7065592 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050638.6934993 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050639.2239985 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[103 'R1_B_H2O.7.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:28:04+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '004f797895df7e116c62707917d49836'\n",
      " 10414012 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050641.2881587 INFO: metabolites 1115\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050641.8152053 INFO: cpd05178 not found in model!\n",
      "1723050641.8173175 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050641.9514823 INFO: reactions 1121\n",
      "1723050642.140539 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050642.6509495 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050642.8001065 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/341/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050643.2020566 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050644.0763812 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050645.1841388 INFO: Expansion time:Ac.NO2:1.0223739579996618\n",
      "1723050645.186509 INFO: Filtered count:0 out of 1688\n",
      "1723050646.2286801 INFO: Expansion time:Ac.O2:1.0404961349995574\n",
      "1723050646.2310946 INFO: Filtered count:0 out of 1688\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[282 'R2_B_D1.110.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:03:19+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '04576a50a6ef844f76318966a5375bc1'\n",
      " 11805721 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050647.1713052 INFO: Expansion time:Etho.O2:0.9386890910000147\n",
      "1723050647.1736283 INFO: Filtered count:0 out of 1688\n",
      "1723050647.5611026 INFO: metabolites 1170\n",
      "1723050648.1290336 INFO: Expansion time:For.NO2:0.9539220570004545\n",
      "1723050648.1313748 INFO: Filtered count:0 out of 1688\n",
      "1723050648.2794785 INFO: reactions 1176\n",
      "1723050648.4698746 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050648.8513844 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050649.125302 INFO: Expansion time:For.O2:0.9925338869998086\n",
      "1723050649.127738 INFO: Filtered count:0 out of 1688\n",
      "1723050649.9775226 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050650.3472898 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/224/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050650.4660342 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050651.4848928 INFO: Expansion time:Fum:2.3556371900003796\n",
      "1723050651.4872084 INFO: Filtered count:0 out of 1688\n",
      "1723050652.4381514 INFO: cpd05178 not found in model!\n",
      "1723050652.4401312 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050652.4418845 INFO: Expansion time:Fum.O2:0.9531426040002771\n",
      "1723050652.4436097 INFO: Filtered count:0 out of 1688\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1.10.pg.G.D GF: 11356 Prob: 4002 Other: 2994\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050653.3985796 INFO: Expansion time:Glyc.O2:0.9536063110008399\n",
      "1723050653.4007528 INFO: Filtered count:0 out of 1688\n",
      "1723050654.337644 INFO: Expansion time:Pyr:0.9354129880002802\n",
      "1723050654.3399332 INFO: Filtered count:0 out of 1688\n",
      "1723050655.3088434 INFO: Expansion time:Pyr.DMSO:0.9674266399997578\n",
      "1723050655.3112285 INFO: Filtered count:0 out of 1688\n",
      "1723050656.1864145 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050656.3629918 INFO: Expansion time:Pyr.NO:1.0503662269993583\n",
      "1723050656.36531 INFO: Filtered count:0 out of 1688\n",
      "1723050657.1520555 INFO: Expansion time:Pyr.NO2:0.7853362030000426\n",
      "1723050657.153868 INFO: Filtered count:0 out of 1688\n",
      "1723050657.80968 INFO: Expansion time:Pyr.NO3:0.6545301179994567\n",
      "1723050657.8116004 INFO: Filtered count:0 out of 1688\n",
      "1723050658.1025717 INFO: cpd05178 not found in model!\n",
      "1723050658.1639993 INFO: cpd05178 not found in model!\n",
      "1723050658.3382485 INFO: cpd05178 not found in model!\n",
      "1723050658.3405268 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050658.5000713 INFO: cpd05178 not found in model!\n",
      "1723050658.5027578 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050658.5261385 INFO: cpd05178 not found in model!\n",
      "1723050658.528665 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.49.pg.G.D GF: 11015 Prob: 4460 Other: 2879\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050659.5676122 INFO: Expansion time:Pyr.O2:1.7548822589997144\n",
      "1723050659.5696304 INFO: Filtered count:0 out of 1688\n",
      "1723050660.2043948 INFO: cpd05178 not found in model!\n",
      "1723050660.2065852 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050660.2310457 INFO: Expansion time:Pyr.SO3:0.6601880779999192\n",
      "1723050660.2329845 INFO: Filtered count:0 out of 1688\n",
      "1723050660.8842626 INFO: Expansion time:Pyr.SO4:0.6501088009999876\n",
      "1723050660.886134 INFO: Filtered count:0 out of 1688\n",
      "1723050661.5332658 INFO: Expansion time:Pyr.TMAO:0.6459957780007244\n",
      "1723050661.535324 INFO: Filtered count:0 out of 1688\n",
      "1723050662.1894467 INFO: Expansion time:empty:0.6529583319997982\n",
      "1723050662.1912932 INFO: Filtered count:0 out of 1688\n",
      "1723050662.9774947 INFO: cpd05178 not found in model!\n",
      "1723050663.3550134 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050663.8698156 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050664.567359 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050665.8461506 INFO: cpd05178 not found in model!\n",
      "1723050665.8485844 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "R1_B_H2O.7.pg.G.D GF: 11596 Prob: 3726 Other: 3030\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050666.3371322 INFO: cpd05178 not found in model!\n",
      "1723050666.3395402 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[112 'R2A_A_H2O.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:29:45+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '811b958717b0caea71d14f8ce9672600'\n",
      " 9038859 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050666.999395 INFO: cpd05178 not found in model!\n",
      "1723050667.0569751 INFO: metabolites 1130\n",
      "1723050667.2463045 INFO: cpd05178 not found in model!\n",
      "1723050667.2486026 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050667.765941 INFO: reactions 1099\n",
      "1723050667.9529278 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050668.418017 INFO: cpd05178 not found in model!\n",
      "1723050668.598169 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/397/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050668.7438047 INFO: cpd05178 not found in model!\n",
      "1723050668.746042 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050668.9769144 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050671.4750338 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050674.325438 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050675.6269984 INFO: cpd05178 not found in model!\n",
      "1723050675.9790807 INFO: cpd05178 not found in model!\n",
      "1723050675.9815857 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050677.351229 INFO: cpd05178 not found in model!\n",
      "1723050677.3537838 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050678.2066727 INFO: cpd05178 not found in model!\n",
      "1723050678.2093956 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050678.2447517 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050678.7180696 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[302 'R1_B_D1.40.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:07:35+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'dfcf6860d669a336bc1e650f0e37fdca'\n",
      " 10380613 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050680.8066761 INFO: Expansion time:Ac.NO2:0.6493924449996484\n",
      "1723050680.8088067 INFO: Filtered count:0 out of 1689\n",
      "1723050681.1787138 INFO: metabolites 1167\n",
      "1723050681.2285905 INFO: cpd05178 not found in model!\n",
      "1723050681.231165 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050681.450422 INFO: Expansion time:Ac.O2:0.6403949519999514\n",
      "1723050681.452716 INFO: Filtered count:0 out of 1689\n",
      "1723050681.5299547 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D1.110.pg.G.D GF: 11046 Prob: 4494 Other: 2814\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050681.7519443 INFO: cpd05178 not found in model!\n",
      "1723050681.7542658 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050681.9264805 INFO: reactions 1160\n",
      "1723050682.0005689 INFO: cpd05178 not found in model!\n",
      "1723050682.0960195 INFO: Expansion time:Etho.O2:0.641931649000071\n",
      "1723050682.0982225 INFO: Filtered count:0 out of 1689\n",
      "1723050682.147956 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050682.24422 INFO: cpd05178 not found in model!\n",
      "1723050682.2465928 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050682.7414675 INFO: Expansion time:For.NO2:0.6419794110006478\n",
      "1723050682.7438543 INFO: Filtered count:0 out of 1689\n",
      "1723050682.8937776 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/375/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050683.384367 INFO: Expansion time:For.O2:0.639193857000464\n",
      "1723050683.386834 INFO: Filtered count:0 out of 1689\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050684.041512 INFO: Expansion time:Fum:0.653327316999821\n",
      "1723050684.0437143 INFO: Filtered count:0 out of 1689\n",
      "1723050684.4858146 INFO: Expansion time:Ac.O2:5.6897757690003345\n",
      "1723050684.4883173 INFO: Filtered count:0 out of 1693\n",
      "1723050684.6281717 INFO: Expansion time:Ac.O2:0.670268484999724\n",
      "1723050684.6303358 INFO: Filtered count:0 out of 1593\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[154 'R2A_B_H2O.029.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:37:55+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'bc332af5f4191baf4d05b9aded427734'\n",
      " 5498001 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050684.6976342 INFO: Expansion time:Fum.O2:0.6526336169999922\n",
      "1723050684.699854 INFO: Filtered count:0 out of 1689\n",
      "1723050685.0098312 INFO: metabolites 908\n",
      "1723050685.1869545 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050685.2837439 INFO: Expansion time:Dlac.O2:0.6521056500005216\n",
      "1723050685.2859514 INFO: Filtered count:0 out of 1593\n",
      "1723050685.3405828 INFO: Expansion time:Pyr:0.6393684430004214\n",
      "1723050685.3427584 INFO: Filtered count:0 out of 1689\n",
      "1723050685.3912153 INFO: reactions 856\n",
      "1723050685.4980476 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050685.838693 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/29/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050685.931772 INFO: Expansion time:Etho.O2:0.6445428960005302\n",
      "1723050685.9338777 INFO: Filtered count:0 out of 1593\n",
      "1723050685.9401321 INFO: Expansion time:Dlac.O2:1.4502646700002515\n",
      "1723050685.9419985 INFO: Filtered count:0 out of 1693\n",
      "1723050685.9876156 INFO: Expansion time:Pyr.DMSO:0.6435751669996534\n",
      "1723050685.9897306 INFO: Filtered count:0 out of 1689\n",
      "1723050686.5095975 INFO: cpd05178 not found in model!\n",
      "1723050686.5119808 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050686.6307304 INFO: Expansion time:Pyr.NO:0.6397074450005675\n",
      "1723050686.632991 INFO: Filtered count:0 out of 1689\n",
      "1723050686.9290135 INFO: Expansion time:For.O2:0.9935306670004138\n",
      "1723050686.9311633 INFO: Filtered count:0 out of 1593\n",
      "1723050687.0618198 INFO: Expansion time:Etho.O2:1.1185313449996102\n",
      "1723050687.0641055 INFO: Filtered count:0 out of 1693\n",
      "1723050687.9086845 INFO: Expansion time:Fum:0.9762496090006607\n",
      "1723050687.9113324 INFO: Filtered count:0 out of 1593\n",
      "1723050687.9714367 INFO: Expansion time:For.O2:0.9059559159995842\n",
      "1723050687.9736078 INFO: Filtered count:0 out of 1693\n",
      "1723050688.8123615 INFO: Expansion time:Fum.O2:0.8995926879997569\n",
      "1723050688.8143747 INFO: Filtered count:0 out of 1593\n",
      "1723050688.8684676 INFO: Expansion time:Pyr.NO2:2.234154550999847\n",
      "1723050688.8707163 INFO: Filtered count:0 out of 1689\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[294 'SF2_A_D2.55.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:06:01+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '2279fe09139d270b005978b319d67534'\n",
      " 11247217 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050689.4904103 INFO: Expansion time:Fum:1.5155114809995212\n",
      "1723050689.4929378 INFO: Filtered count:0 out of 1693\n",
      "1723050689.5218563 INFO: Expansion time:Pyr.NO3:0.6498452469995755\n",
      "1723050689.5239894 INFO: Filtered count:0 out of 1689\n",
      "1723050689.9331443 INFO: metabolites 1170\n",
      "1723050690.1645184 INFO: Expansion time:Pyr.O2:0.6392838680003479\n",
      "1723050690.1667466 INFO: Filtered count:0 out of 1689\n",
      "1723050690.1993449 INFO: cpd05178 not found in model!\n",
      "1723050690.2014492 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050690.7376971 INFO: reactions 1119\n",
      "1723050690.80461 INFO: Expansion time:Pyr.SO3:0.6366049270000076\n",
      "1723050690.8068058 INFO: Filtered count:0 out of 1689\n",
      "1723050690.9379008 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050691.0694299 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050691.1136642 INFO: Expansion time:LLac.O2:2.2978327849996276\n",
      "1723050691.1156828 INFO: Filtered count:0 out of 1593\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[151 'R2A_C_D1.5.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:37:26+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '0574eb9a47d61d088cc95278b62ed231'\n",
      " 14294833 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050691.4463809 INFO: Expansion time:Pyr.SO4:0.6382988169998498\n",
      "1723050691.4486558 INFO: Filtered count:0 out of 1689\n",
      "1723050691.5368133 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050691.6194928 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/285/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050691.7713532 INFO: Expansion time:Pyr:0.6544516909998492\n",
      "1723050691.7722476 INFO: Expansion time:Fum.O2:2.2776747449997856\n",
      "1723050691.7733228 INFO: Filtered count:0 out of 1593\n",
      "1723050691.775257 INFO: Filtered count:0 out of 1693\n",
      "1723050692.0905082 INFO: Expansion time:Pyr.TMAO:0.6404623180005728\n",
      "1723050692.0926921 INFO: Filtered count:0 out of 1689\n",
      "1723050692.255622 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050692.4257624 INFO: Expansion time:Pyr.DMSO:0.6508567339997171\n",
      "1723050692.4278743 INFO: Filtered count:0 out of 1593\n",
      "1723050692.7357876 INFO: Expansion time:empty:0.6419096760000684\n",
      "1723050692.7358866 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050692.737924 INFO: Filtered count:0 out of 1689\n",
      "1723050693.0826344 INFO: Expansion time:Pyr.NO:0.6533049170002414\n",
      "1723050693.0847723 INFO: Filtered count:0 out of 1593\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "Getting tests from attributes\n",
      "[258 'R2_A_D2.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:58:50+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '3ee6ebb1e99a1d039066fb0112ce803d'\n",
      " 9800357 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050693.2342126 INFO: cpd05178 not found in model!\n",
      "1723050693.5731263 INFO: cpd05178 not found in model!\n",
      "1723050693.575619 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050693.7353754 INFO: Expansion time:Pyr.NO2:0.6494135860002643\n",
      "1723050693.7375102 INFO: Filtered count:0 out of 1593\n",
      "1723050694.380624 INFO: Expansion time:Pyr.NO3:0.6418611889994281\n",
      "1723050694.3827116 INFO: Filtered count:0 out of 1593\n",
      "1723050694.7208786 INFO: Expansion time:Pyr:2.944258778999938\n",
      "1723050694.722859 INFO: Filtered count:0 out of 1693\n",
      "1723050695.0501611 INFO: Expansion time:Pyr.O2:0.6662484859998585\n",
      "1723050695.05243 INFO: Filtered count:0 out of 1593\n",
      "1723050695.322624 INFO: metabolites 848\n",
      "1723050695.4204316 INFO: Expansion time:Pyr.DMSO:0.6961177550001594\n",
      "1723050695.4223921 INFO: Filtered count:0 out of 1693\n",
      "1723050695.5377522 INFO: metabolites 1272\n",
      "1723050695.7190118 INFO: Expansion time:Pyr.SO3:0.6652807390000817\n",
      "1723050695.7212155 INFO: Filtered count:0 out of 1593\n",
      "1723050695.7460773 INFO: reactions 760\n",
      "1723050695.8648982 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050696.111189 INFO: Expansion time:Pyr.NO:0.6875483060002807\n",
      "1723050696.1132302 INFO: Filtered count:0 out of 1693\n",
      "1723050696.298991 INFO: cpd05178 not found in model!\n",
      "1723050696.3016858 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050696.357066 INFO: Default biomass: [bio1]\n",
      "1723050696.3803215 INFO: Expansion time:Pyr.SO4:0.6578079119999529\n",
      "1723050696.382517 INFO: Filtered count:0 out of 1593\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/24/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050696.4321895 INFO: reactions 1220\n",
      "1723050696.6923206 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_H2O.16.pg.G.D GF: 11166 Prob: 4272 Other: 2914\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050696.7896028 INFO: Expansion time:Pyr.NO2:0.6750994020003418\n",
      "1723050696.791594 INFO: Filtered count:0 out of 1693\n",
      "1723050697.03739 INFO: Expansion time:Pyr.TMAO:0.6537004540005\n",
      "1723050697.039437 INFO: Filtered count:0 out of 1593\n",
      "1723050697.4528036 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/32/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050697.6988497 INFO: cpd05178 not found in model!\n",
      "1723050697.7011974 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050697.7182558 INFO: Expansion time:Pyr.NO3:0.9253771979992962\n",
      "1723050697.720466 INFO: Filtered count:0 out of 1693\n",
      "1723050698.3021457 INFO: cpd05178 not found in model!\n",
      "1723050698.5370414 INFO: cpd05178 not found in model!\n",
      "1723050698.539341 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050698.873278 INFO: Objective with gapfill database:169.62587333898819; min objective:0.01\n",
      "1723050698.9371252 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050699.2140086 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050699.263007 INFO: Expansion time:empty:2.2221675740001956\n",
      "1723050699.2649117 INFO: Filtered count:0 out of 1593\n",
      "1723050699.2713854 INFO: Expansion time:Pyr.O2:1.5494940309999947\n",
      "1723050699.2732034 INFO: Filtered count:0 out of 1693\n",
      "1723050699.5324252 INFO: cpd05178 not found in model!\n",
      "1723050699.535044 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050700.5446608 INFO: Expansion time:Pyr.SO3:1.270084198999939\n",
      "1723050700.5468416 INFO: Filtered count:0 out of 1693\n",
      "1723050702.4084454 INFO: Expansion time:Pyr.SO4:1.860347843999989\n",
      "1723050702.410543 INFO: Filtered count:0 out of 1693\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[254 'R1_A_D1.5.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:57:51+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '937a1778d1a07ef66b03e8a1d8703dec'\n",
      " 11602369 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050703.101075 INFO: Expansion time:Pyr.TMAO:0.6892934240004251\n",
      "1723050703.1032557 INFO: Filtered count:0 out of 1693\n",
      "1723050703.3650298 INFO: metabolites 902\n",
      "1723050703.7908278 INFO: Expansion time:empty:0.6863295440007278\n",
      "1723050703.7930121 INFO: Filtered count:0 out of 1693\n",
      "1723050703.8873878 INFO: reactions 832\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[162 'R2_A_H2O.14.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:39:28+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'b6db30246d5a7a59f972a393e325f5d1'\n",
      " 9763988 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050704.0157993 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050704.5010295 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/6/1\n",
      "[99 'R1_A_D1.6.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:27:12+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'af8c901a1b785deb58ed06e8b1840952'\n",
      " 16890840 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050705.7169735 INFO: metabolites 925\n",
      "1723050705.822652 INFO: metabolites 1104\n",
      "1723050706.0846775 INFO: reactions 834\n",
      "1723050706.18106 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050706.43137 INFO: reactions 1043\n",
      "1723050706.5886037 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050706.6288652 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/228/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050707.0046642 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/315/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050708.3497214 INFO: cpd05178 not found in model!\n",
      "1723050708.3516843 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050708.5987697 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_B_H2O.029.pg.G.D GF: 12348 Prob: 2790 Other: 3212\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050708.9452667 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050710.1185954 INFO: cpd05178 not found in model!\n",
      "1723050710.1208835 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D1.40.pg.G.D GF: 11348 Prob: 4052 Other: 2954\n",
      "[195 'R2A_C_D1.10.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:46:05+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ff0d41c41e26c09b2711a2233b3e609e'\n",
      " 9442892 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050711.914142 INFO: cpd05178 not found in model!\n",
      "1723050711.9161532 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050712.4001968 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050712.9169993 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050713.1463296 INFO: metabolites 1117\n",
      "1723050713.219511 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050713.6673095 INFO: reactions 1034\n",
      "1723050713.845362 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050713.8810134 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050714.4467795 INFO: cpd05178 not found in model!\n",
      "1723050714.4488847 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050714.4479306 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/309/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050715.1810224 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050715.1944299 INFO: Objective with gapfill database:100.70545872372954; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050715.711692 INFO: cpd05178 not found in model!\n",
      "1723050715.9554205 INFO: cpd05178 not found in model!\n",
      "1723050715.9575636 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050716.566346 INFO: cpd05178 not found in model!\n",
      "1723050716.797701 INFO: cpd05178 not found in model!\n",
      "1723050716.7998776 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050716.95465 INFO: cpd05178 not found in model!\n",
      "1723050717.1412907 INFO: cpd05178 not found in model!\n",
      "1723050717.143666 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050717.7560184 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050718.108956 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050718.4178052 INFO: cpd05178 not found in model!\n",
      "1723050718.420167 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050718.6090724 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[47 'R1_B_D1.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:16:49+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '3187aef686b5d39a7db60ebf79cdb97d'\n",
      " 12645582 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050718.7676446 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050718.9551258 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050719.1686733 INFO: metabolites 1116\n",
      "1723050719.179281 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050719.428843 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050719.614337 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050719.7512212 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050719.9395778 INFO: reactions 1117\n",
      "1723050720.1255138 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050720.7813382 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/599/1\n",
      "[31 'R2_B_D2.80.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:13:47+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '550f50fb9f550b866573f8f108068549'\n",
      " 12190967 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050721.3018372 INFO: metabolites 1014\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050721.7195964 INFO: cpd05178 not found in model!\n",
      "1723050721.7224 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050721.865484 INFO: reactions 975\n",
      "1723050722.0134802 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_D2.55.pg.G.D GF: 10873 Prob: 4560 Other: 2921\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050722.2474234 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050722.6606872 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050724.0223842 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/498/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050724.0867987 INFO: cpd05178 not found in model!\n",
      "1723050724.0888166 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[266 'R1_B_D1.41.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:00:13+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '3df33fda84b1517983d68ba3ea8a0cad'\n",
      " 10150025 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050724.3741744 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050724.6528928 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050724.805523 INFO: metabolites 1104\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050725.4002414 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050725.498252 INFO: reactions 1100\n",
      "1723050725.6799157 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050725.7603493 INFO: cpd05178 not found in model!\n",
      "1723050725.7624161 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050726.3451607 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/122/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050727.362831 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050728.3304245 INFO: cpd05178 not found in model!\n",
      "1723050728.3323429 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D2.16.pg.G.D GF: 12457 Prob: 2628 Other: 3275\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050729.5044343 INFO: Expansion time:Glc:0.6970321859998876\n",
      "1723050729.5064688 INFO: Filtered count:0 out of 1207\n",
      "1723050729.821914 INFO: Expansion time:Ac.NO2:1.9068272159993285\n",
      "1723050729.8242667 INFO: Filtered count:0 out of 1681\n",
      "1723050730.1453176 INFO: Expansion time:Glc.DMSO:0.6377218859997811\n",
      "1723050730.1475067 INFO: Filtered count:0 out of 1207\n",
      "1723050730.2294848 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050730.4688692 INFO: Expansion time:Ac.O2:0.6433233300003849\n",
      "1723050730.471349 INFO: Filtered count:0 out of 1681\n",
      "1723050730.5406222 INFO: cpd05178 not found in model!\n",
      "1723050730.5431201 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050730.7858276 INFO: Expansion time:Glc.O2:0.6370127749996755\n",
      "1723050730.7877674 INFO: Filtered count:0 out of 1207\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_H2O.14.pg.G.D GF: 12343 Prob: 2760 Other: 3257\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050731.049432 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050731.118915 INFO: Expansion time:Etho.O2:0.6462804060001872\n",
      "1723050731.1211367 INFO: Filtered count:0 out of 1681\n",
      "1723050731.430884 INFO: Expansion time:Glc.TMAO:0.6418523729998924\n",
      "1723050731.4337204 INFO: Filtered count:0 out of 1207\n",
      "1723050731.7704377 INFO: Expansion time:For.NO2:0.6481007410002348\n",
      "1723050731.7726042 INFO: Filtered count:0 out of 1681\n",
      "1723050731.8598046 INFO: cpd05178 not found in model!\n",
      "1723050732.0257523 INFO: cpd05178 not found in model!\n",
      "1723050732.028103 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050732.09074 INFO: cpd05178 not found in model!\n",
      "1723050732.101087 INFO: cpd05178 not found in model!\n",
      "1723050732.1033442 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050732.3320856 INFO: cpd05178 not found in model!\n",
      "1723050732.3342962 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D1.5.pg.G.D GF: 10915 Prob: 4600 Other: 2837\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050732.4217658 INFO: Expansion time:For.O2:0.647828328000287\n",
      "1723050732.4240315 INFO: Filtered count:0 out of 1681\n",
      "1723050732.8927994 INFO: cpd05178 not found in model!\n",
      "1723050732.8949711 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050733.0734806 INFO: Expansion time:Fum:0.6482087020003746\n",
      "1723050733.0757613 INFO: Filtered count:0 out of 1681\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D1.5.pg.G.D GF: 11972 Prob: 3270 Other: 3118\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050733.1964846 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050733.4188259 INFO: Expansion time:Pyr:1.9835664640004325\n",
      "1723050733.421213 INFO: Filtered count:0 out of 1207\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050733.6374357 INFO: cpd05178 not found in model!\n",
      "1723050733.6395457 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050733.722274 INFO: Expansion time:Fum.O2:0.6452025839998896\n",
      "1723050733.7245286 INFO: Filtered count:0 out of 1681\n",
      "1723050733.894495 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050734.342788 INFO: Expansion time:Pyr.DMSO:0.9200527590001002\n",
      "1723050734.3451707 INFO: Filtered count:0 out of 1207\n",
      "1723050734.7276583 INFO: cpd05178 not found in model!\n",
      "1723050734.7298245 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050735.3788462 INFO: Expansion time:Pyr.NO:1.0320004189998144\n",
      "1723050735.3812366 INFO: Filtered count:0 out of 1207\n",
      "1723050735.5359383 INFO: Expansion time:Pyr:1.810146910000185\n",
      "1723050735.53825 INFO: Filtered count:0 out of 1681\n",
      "1723050736.1916118 INFO: Expansion time:Pyr.DMSO:0.6520531039996058\n",
      "1723050736.1938245 INFO: Filtered count:0 out of 1681\n",
      "1723050736.2118456 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050736.369607 INFO: Expansion time:Pyr.NO2:0.9867243730004702\n",
      "1723050736.3719003 INFO: Filtered count:0 out of 1207\n",
      "1723050736.8416398 INFO: Expansion time:Pyr.NO:0.6464815919998728\n",
      "1723050736.8438668 INFO: Filtered count:0 out of 1681\n",
      "1723050737.278976 INFO: Expansion time:Pyr.NO3:0.9057120690004012\n",
      "1723050737.2813666 INFO: Filtered count:0 out of 1207\n",
      "1723050737.49293 INFO: Expansion time:Pyr.NO2:0.6478596270007984\n",
      "1723050737.4950852 INFO: Filtered count:0 out of 1681\n",
      "1723050737.9969454 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050738.1468935 INFO: Expansion time:Pyr.NO3:0.6505476469992573\n",
      "1723050738.149111 INFO: Filtered count:0 out of 1681\n",
      "1723050738.160531 INFO: Expansion time:Pyr.O2:0.8776097740001205\n",
      "1723050738.162878 INFO: Filtered count:0 out of 1207\n",
      "1723050738.2032773 INFO: cpd05178 not found in model!\n",
      "1723050738.2058866 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050738.5434859 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D1.10.pg.G.D GF: 11958 Prob: 3252 Other: 3144\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050738.8167517 INFO: Expansion time:Pyr.O2:0.6664054420007233\n",
      "1723050738.8189147 INFO: Filtered count:0 out of 1681\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050739.1055312 INFO: Expansion time:Pyr.SO3:0.941173248999803\n",
      "1723050739.1080835 INFO: Filtered count:0 out of 1207\n",
      "1723050739.4829113 INFO: Expansion time:Pyr.SO3:0.6626687500001935\n",
      "1723050739.485332 INFO: Filtered count:0 out of 1681\n",
      "1723050739.501257 INFO: cpd05178 not found in model!\n",
      "1723050739.503529 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050739.5385296 INFO: Objective with gapfill database:103.14785336666252; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D1.6.pg.G.D GF: 11334 Prob: 3952 Other: 3068\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050740.1581454 INFO: Expansion time:Pyr.SO4:0.6714344390002225\n",
      "1723050740.1604757 INFO: Filtered count:0 out of 1681\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[187 'R1_A_D2.45.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:44:21+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'b24d228704c3b64afdb9a49ccf23ced3'\n",
      " 8506651 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050740.8485305 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050741.0814817 INFO: metabolites 1029\n",
      "1723050741.1751785 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050741.3821473 INFO: Objective with gapfill database:163.27549115483683; min objective:0.01\n",
      "1723050741.4072146 INFO: Expansion time:Pyr.SO4:2.2976055159997486\n",
      "1723050741.409783 INFO: Filtered count:0 out of 1207\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050741.5122783 INFO: reactions 966\n",
      "1723050741.5681686 INFO: cpd05178 not found in model!\n",
      "1723050741.5700984 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050741.6241934 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050741.9431572 INFO: Expansion time:Pyr.TMAO:1.7814423879999595\n",
      "1723050741.9453425 INFO: Filtered count:0 out of 1681\n",
      "1723050741.955831 INFO: cpd05178 not found in model!\n",
      "1723050742.1648777 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/218/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050742.2984478 INFO: cpd05178 not found in model!\n",
      "1723050742.3009927 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050742.38545 INFO: Expansion time:Pyr.TMAO:0.974021379000078\n",
      "1723050742.3877015 INFO: Filtered count:0 out of 1207\n",
      "1723050742.8790877 INFO: Expansion time:empty:0.9323586919999798\n",
      "1723050742.8815944 INFO: Filtered count:0 out of 1681\n",
      "1723050743.018133 INFO: Expansion time:empty:0.629009958000097\n",
      "1723050743.020313 INFO: Filtered count:0 out of 1207\n",
      "1723050743.268324 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050744.3213208 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050745.4043698 INFO: Objective with gapfill database:114.12678225412454; min objective:0.01\n",
      "1723050745.8410878 INFO: Expansion time:Etho.O2:0.6724662159995205\n",
      "1723050745.8431015 INFO: Filtered count:0 out of 1627\n",
      "1723050746.312925 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050747.1189961 INFO: cpd05178 not found in model!\n",
      "1723050747.1212246 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050747.2711823 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[40 'R2A_C_D1.17.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:15:32+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '37152cbc8c8354069075f29a21e5cd5a'\n",
      " 7698763 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050747.7609203 INFO: Expansion time:For.O2:1.9166467859995464\n",
      "1723050747.7631567 INFO: Filtered count:0 out of 1627\n",
      "1723050747.8453238 INFO: cpd05178 not found in model!\n",
      "1723050747.8477583 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050747.9525843 INFO: metabolites 1033\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.80.pg.G.D GF: 11977 Prob: 3220 Other: 3155\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050748.551868 INFO: reactions 999\n",
      "1723050748.7220724 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050748.7359793 INFO: Expansion time:Fum:0.971447963999708\n",
      "1723050748.737967 INFO: Filtered count:0 out of 1627\n",
      "1723050749.3592498 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/565/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050749.7034256 INFO: Expansion time:Fum.O2:0.9640052969998578\n",
      "1723050749.705846 INFO: Filtered count:0 out of 1627\n",
      "1723050749.748062 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050750.6965768 INFO: Expansion time:Glc:0.9893222159998913\n",
      "1723050750.6989355 INFO: Filtered count:0 out of 1627\n",
      "1723050751.6863463 INFO: Expansion time:Glc.DMSO:0.9859873770001286\n",
      "1723050751.6890883 INFO: Filtered count:0 out of 1627\n",
      "1723050752.2504823 INFO: cpd05178 not found in model!\n",
      "1723050752.252393 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050752.6982887 INFO: Expansion time:Glc.O2:1.007279952999852\n",
      "1723050752.7011757 INFO: Filtered count:0 out of 1627\n",
      "1723050753.7390585 INFO: cpd05178 not found in model!\n",
      "1723050753.7412086 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050753.7737935 INFO: Expansion time:Glc.TMAO:1.0708311369999137\n",
      "1723050753.7766054 INFO: Filtered count:0 out of 1627\n",
      "1723050753.8429954 INFO: cpd05178 not found in model!\n",
      "1723050753.8451793 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050753.934966 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D1.16.pg.G.D GF: 11118 Prob: 4400 Other: 2834\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050754.393614 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050754.825815 INFO: Expansion time:Pyr:1.0473437009995905\n",
      "1723050754.8288014 INFO: Filtered count:0 out of 1627\n",
      "1723050755.5095532 INFO: cpd05178 not found in model!\n",
      "1723050755.5121703 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[211 'SF2_A_D2.33.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:49:18+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '5112b6a6ae925c7745c8d64621ffc926'\n",
      " 13097300 None]\n",
      "R1_B_D1.41.pg.G.D GF: 11483 Prob: 3754 Other: 3115\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050756.3425608 INFO: metabolites 1133\n",
      "1723050757.0591228 INFO: Expansion time:Ac.NO2:2.4146189279999817\n",
      "1723050757.0617414 INFO: Filtered count:0 out of 1755\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050758.0488882 INFO: Expansion time:Pyr.DMSO:3.218065144999855\n",
      "1723050758.0515573 INFO: Filtered count:0 out of 1627\n",
      "1723050758.162897 INFO: Expansion time:Ac.O2:1.0996863310001572\n",
      "1723050758.165472 INFO: Filtered count:0 out of 1755\n",
      "1723050758.2593825 INFO: reactions 1048\n",
      "1723050758.3809636 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050758.466922 INFO: cpd05178 not found in model!\n",
      "1723050759.0868607 INFO: Default biomass: [bio1]\n",
      "1723050759.1181054 INFO: Expansion time:Pyr.NO:1.065037588999985\n",
      "1723050759.1205418 INFO: Filtered count:0 out of 1627\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/408/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050759.3301 INFO: Expansion time:Etho.O2:1.1630528960004085\n",
      "1723050759.3327606 INFO: Filtered count:0 out of 1755\n",
      "1723050759.359081 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050759.529239 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050759.9316852 INFO: cpd05178 not found in model!\n",
      "1723050759.9340222 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050760.0308743 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050760.2101693 INFO: Expansion time:Pyr.NO2:1.0881462990000728\n",
      "1723050760.2123427 INFO: Filtered count:0 out of 1627\n",
      "1723050760.4228313 INFO: Expansion time:For.NO2:1.088604306999514\n",
      "1723050760.4253855 INFO: Filtered count:0 out of 1755\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[207 'R1_C_D2.58.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:48:31+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '1e5b14d37f03bcdef27b5cc981d6a091'\n",
      " 2245529 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050760.907677 INFO: Expansion time:Ac.NO2:3.6667730529998153\n",
      "1723050760.9098234 INFO: Filtered count:0 out of 1494\n",
      "1723050761.0424016 INFO: metabolites 574\n",
      "1723050761.2087452 INFO: reactions 516\n",
      "1723050761.2306643 INFO: Expansion time:Pyr.NO3:1.0169140119996882\n",
      "1723050761.232912 INFO: Filtered count:0 out of 1627\n",
      "1723050761.3273056 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050761.577082 INFO: Expansion time:For.O2:1.1502492930003427\n",
      "1723050761.5796347 INFO: Filtered count:0 out of 1755\n",
      "1723050761.646507 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/354/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050762.336098 INFO: Expansion time:Pyr.O2:1.1017330069998934\n",
      "1723050762.338629 INFO: Filtered count:0 out of 1627\n",
      "1723050762.785882 INFO: Expansion time:Ac.O2:1.8739477239996631\n",
      "1723050762.7880325 INFO: Filtered count:0 out of 1494\n",
      "1723050762.910788 INFO: Expansion time:Fum:1.3296567050001613\n",
      "1723050762.9133766 INFO: Filtered count:0 out of 1755\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050763.4740224 INFO: Expansion time:Pyr.SO3:1.1338849149997259\n",
      "1723050763.4763293 INFO: Filtered count:0 out of 1627\n",
      "1723050763.6300192 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050763.9874651 INFO: cpd05178 not found in model!\n",
      "1723050764.1421468 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050764.1678655 INFO: Expansion time:Fum.O2:1.252985365999848\n",
      "1723050764.1705432 INFO: Filtered count:0 out of 1755\n",
      "1723050764.3156357 INFO: cpd05178 not found in model!\n",
      "1723050764.3183715 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050764.4454653 INFO: Expansion time:Glc:1.656152619999375\n",
      "1723050764.4477365 INFO: Filtered count:0 out of 1494\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050764.6321285 INFO: Expansion time:Pyr.SO4:1.1542170120001174\n",
      "1723050764.6343262 INFO: Filtered count:0 out of 1627\n",
      "1723050765.2528276 INFO: Objective with gapfill database:203.73793873543016; min objective:0.01\n",
      "1723050765.407758 INFO: Expansion time:Glc.DMSO:0.9587309480002659\n",
      "1723050765.4098809 INFO: Filtered count:0 out of 1494\n",
      "1723050765.6073728 INFO: Expansion time:Glc:1.435366688000613\n",
      "1723050765.610065 INFO: Filtered count:0 out of 1755\n",
      "1723050765.7607107 INFO: Expansion time:Etho.O2:2.469663029000003\n",
      "1723050765.7629747 INFO: Filtered count:0 out of 1478\n",
      "1723050766.2738295 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050766.451249 INFO: Expansion time:Glc:0.6868834759998208\n",
      "1723050766.4533203 INFO: Filtered count:0 out of 1478\n",
      "1723050766.755825 INFO: Objective with gapfill database:243.41652719653808; min objective:0.01\n",
      "1723050766.8773117 INFO: Expansion time:Glc.O2:1.4655532219994711\n",
      "1723050766.879411 INFO: Filtered count:0 out of 1494\n",
      "1723050766.8921065 INFO: Expansion time:Glc.DMSO:1.280527001000337\n",
      "1723050766.8946369 INFO: Filtered count:0 out of 1755\n",
      "1723050767.1285677 INFO: cpd05178 not found in model!\n",
      "1723050767.1307113 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050767.1559362 INFO: Expansion time:Glc.DMSO:0.7012656469996728\n",
      "1723050767.158143 INFO: Filtered count:0 out of 1478\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.45.pg.G.D GF: 11626 Prob: 3526 Other: 3200\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050767.4294555 INFO: Expansion time:Pyr.TMAO:2.7937810670000545\n",
      "1723050767.4317837 INFO: Filtered count:0 out of 1627\n",
      "1723050767.8452706 INFO: Expansion time:Glc.O2:0.6857583419996445\n",
      "1723050767.847592 INFO: Filtered count:0 out of 1478\n",
      "1723050768.1642702 INFO: Expansion time:Glc.TMAO:1.2837680039992847\n",
      "1723050768.1663632 INFO: Filtered count:0 out of 1494\n",
      "1723050768.294479 INFO: Expansion time:Glc.O2:1.398297482000089\n",
      "1723050768.2971444 INFO: Filtered count:0 out of 1755\n",
      "1723050768.455598 INFO: Expansion time:empty:1.0223233209999307\n",
      "1723050768.4579046 INFO: Filtered count:0 out of 1627\n",
      "1723050768.6056855 INFO: Expansion time:Glc.TMAO:0.7563251460005631\n",
      "1723050768.6077576 INFO: Filtered count:0 out of 1478\n",
      "1723050769.2387013 INFO: Expansion time:Pyr.NO2:1.07109984500039\n",
      "1723050769.240822 INFO: Filtered count:0 out of 1494\n",
      "1723050769.2714486 INFO: Expansion time:Pyr:0.6624439890001668\n",
      "1723050769.273582 INFO: Filtered count:0 out of 1478\n",
      "1723050769.5439162 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050769.9904163 INFO: Expansion time:Pyr.DMSO:0.7155188250007996\n",
      "1723050769.9923937 INFO: Filtered count:0 out of 1478\n",
      "1723050770.6229863 INFO: Expansion time:Pyr.O2:1.3809504830005608\n",
      "1723050770.6250885 INFO: Filtered count:0 out of 1494\n",
      "1723050770.660742 INFO: Expansion time:Pyr.NO:0.6668206379999901\n",
      "1723050770.6628737 INFO: Filtered count:0 out of 1478\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[107 'R1_C_H2O.29.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:28:47+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'b5ba8fad565fcc299e132ff5ef45d01e'\n",
      " 5462979 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050771.1451504 INFO: metabolites 806\n",
      "1723050771.3531408 INFO: Expansion time:Pyr.NO2:0.6889283069995145\n",
      "1723050771.3552115 INFO: Filtered count:0 out of 1478\n",
      "1723050771.4844122 INFO: reactions 701\n",
      "1723050771.6049094 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050771.6647701 INFO: Expansion time:empty:1.0382785390002027\n",
      "1723050771.6682816 INFO: Filtered count:0 out of 1494\n",
      "1723050771.8110485 INFO: cpd05178 not found in model!\n",
      "1723050771.8132546 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050772.0004375 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/14/1\n",
      "[174 'R1_A_D1.54.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:41:48+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '67301fc33c9823a3b62a5e860e11fcae'\n",
      " 8658343 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050772.216219 INFO: Expansion time:Pyr.NO3:0.8598128899993753\n",
      "1723050772.218601 INFO: Filtered count:0 out of 1478\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[168 'R2_C_H2O.29.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:40:40+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '79673a3f3a536b49f06a4b757c2875c7'\n",
      " 4097459 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050772.4610708 INFO: cpd05178 not found in model!\n",
      "1723050772.4636643 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050772.507896 INFO: metabolites 1024\n",
      "1723050772.7685573 INFO: metabolites 482\n",
      "1723050772.907994 INFO: reactions 439\n",
      "1723050772.9142673 INFO: reactions 952\n",
      "1723050773.025349 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050773.041192 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050773.1932418 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/277/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050773.3270962 INFO: Objective with gapfill database:185.22556492052422; min objective:0.01\n",
      "1723050773.4475763 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/152/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050773.6398003 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050774.1179817 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050775.1226988 INFO: cpd05178 not found in model!\n",
      "1723050775.1246107 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050775.2510307 INFO: Expansion time:Pyr.O2:3.030981593999968\n",
      "1723050775.2532532 INFO: Filtered count:0 out of 1478\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D1.17.pg.G.D GF: 11684 Prob: 3732 Other: 2938\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050775.9164357 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050776.106956 INFO: Expansion time:Pyr.SO3:0.8523045269994327\n",
      "1723050776.1087408 INFO: Filtered count:0 out of 1478\n",
      "1723050776.1977177 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00062_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050776.4238784 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050776.6716378 INFO: Expansion time:Glc.TMAO:8.372836424000525\n",
      "1723050776.6741722 INFO: Filtered count:1 out of 1755\n",
      "1723050776.7248838 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050776.7689087 INFO: Expansion time:Pyr.SO4:0.658929664000425\n",
      "1723050776.7707627 INFO: Filtered count:0 out of 1478\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050776.9800177 INFO: cpd05178 not found in model!\n",
      "1723050777.2792556 INFO: cpd05178 not found in model!\n",
      "1723050777.2814977 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050777.3281946 INFO: cpd05178 not found in model!\n",
      "1723050777.3309054 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050777.4257536 INFO: Expansion time:Pyr.TMAO:0.6538123930004076\n",
      "1723050777.4276376 INFO: Filtered count:0 out of 1478\n",
      "1723050777.6278899 INFO: Expansion time:Pyr:0.9522382140003174\n",
      "1723050777.6304832 INFO: Filtered count:1 out of 1755\n",
      "1723050778.1267602 INFO: Expansion time:empty:0.6980073849999826\n",
      "1723050778.1286287 INFO: Filtered count:0 out of 1478\n",
      "1723050778.5895674 INFO: Expansion time:Pyr.DMSO:0.957556317000126\n",
      "1723050778.5922525 INFO: Filtered count:1 out of 1755\n",
      "1723050779.0786202 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050779.551639 INFO: Expansion time:Pyr.NO:0.9579483690004054\n",
      "1723050779.554314 INFO: Filtered count:1 out of 1755\n",
      "1723050779.6170125 INFO: Objective with gapfill database:157.6326173700082; min objective:0.01\n",
      "1723050780.5054576 INFO: Expansion time:Pyr.NO2:0.9496902020000562\n",
      "1723050780.5080786 INFO: Filtered count:1 out of 1755\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050781.4709504 INFO: Expansion time:Pyr.NO3:0.9614193570005227\n",
      "1723050781.4734955 INFO: Filtered count:1 out of 1755\n",
      "1723050781.8572235 INFO: cpd05178 not found in model!\n",
      "1723050781.859791 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050781.9359055 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050781.9852693 INFO: cpd05178 not found in model!\n",
      "1723050781.9874806 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_D2.58.pg.G.D GF: 13595 Prob: 890 Other: 3867\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050782.4320407 INFO: Expansion time:Pyr.O2:0.9570163469998079\n",
      "1723050782.4347694 INFO: Filtered count:1 out of 1755\n",
      "1723050783.1314096 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050783.3992424 INFO: Expansion time:Pyr.SO3:0.9629930710007102\n",
      "1723050783.401993 INFO: Filtered count:1 out of 1755\n",
      "1723050784.018845 INFO: cpd05178 not found in model!\n",
      "1723050784.3482955 INFO: Expansion time:Pyr.SO4:0.9447618829999556\n",
      "1723050784.3509898 INFO: Filtered count:1 out of 1755\n",
      "1723050784.3695576 INFO: cpd05178 not found in model!\n",
      "1723050784.3719952 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050784.5332537 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050784.9949133 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050785.3133824 INFO: Expansion time:Pyr.TMAO:0.9608115909995831\n",
      "1723050785.3161077 INFO: Filtered count:1 out of 1755\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050786.1948354 INFO: cpd05178 not found in model!\n",
      "1723050786.1970627 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050786.2704191 INFO: Expansion time:empty:0.9529052389998469\n",
      "1723050786.2729406 INFO: Filtered count:1 out of 1755\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_D2.33.pg.G.D GF: 11421 Prob: 3982 Other: 2949\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050789.5490572 INFO: cpd05178 not found in model!\n",
      "1723050789.551064 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050789.7691815 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_H2O.29.pg.G.D GF: 13666 Prob: 938 Other: 3756\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050790.535361 INFO: cpd05178 not found in model!\n",
      "1723050790.537527 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050791.355466 INFO: cpd05178 not found in model!\n",
      "1723050791.3582225 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050791.6929784 INFO: Objective with gapfill database:109.33550288831425; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_H2O.29.pg.G.D GF: 12780 Prob: 2186 Other: 3394\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050793.7655032 INFO: cpd05178 not found in model!\n",
      "1723050793.767827 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050795.1135721 INFO: cpd05178 not found in model!\n",
      "1723050795.115696 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050796.9556792 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050802.0919633 INFO: cpd05178 not found in model!\n",
      "1723050802.094538 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050802.12139 INFO: Objective with gapfill database:241.68077675723063; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[139 'R2_B_D2.38.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:35:06+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '34fdf053e1024275c18ebfbdf80e20c6'\n",
      " 12154094 None]\n",
      "R1_A_D1.54.pg.G.D GF: 11769 Prob: 3510 Other: 3075\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050802.7807317 INFO: metabolites 1184\n",
      "1723050802.9389496 INFO: Objective with gapfill database:71.17206922691776; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[182 'R2_B_D2.20.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:43:24+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '47a7945b957155525f1552e9343a3a57'\n",
      " 12481325 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050804.2090278 INFO: metabolites 1046\n",
      "1723050804.5927927 INFO: reactions 1020\n",
      "1723050804.7070727 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050804.9472995 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050805.100354 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/188/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050805.3703964 INFO: reactions 1178\n",
      "1723050805.5974555 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050806.3384259 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/576/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050812.8646026 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050813.3055627 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050814.6122077 INFO: Expansion time:For.NO2:0.5791826479999145\n",
      "1723050814.6141448 INFO: Filtered count:0 out of 526\n",
      "1723050814.819179 INFO: cpd05178 not found in model!\n",
      "1723050814.821805 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050815.2074156 INFO: Expansion time:For.O2:0.5920944449999297\n",
      "1723050815.2093208 INFO: Filtered count:0 out of 526\n",
      "1723050815.599236 INFO: Expansion time:For.NO2:6.029684345000533\n",
      "1723050815.6019127 INFO: Filtered count:0 out of 582\n",
      "1723050815.7592962 INFO: Expansion time:Fum:0.5487585320006474\n",
      "1723050815.7612808 INFO: Filtered count:0 out of 526\n",
      "1723050816.2941806 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050816.526695 INFO: Expansion time:empty:0.9231772279999859\n",
      "1723050816.529318 INFO: Filtered count:0 out of 582\n",
      "1723050816.6125433 INFO: Expansion time:Pyr:0.850033223000537\n",
      "1723050816.614892 INFO: Filtered count:0 out of 526\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[115 'SF2_A_H2O.28.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:30:18+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'addd8011c6f0710c8e73a7b378b11c86'\n",
      " 5783696 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050817.2020879 INFO: metabolites 927\n",
      "1723050817.436845 INFO: Expansion time:Pyr.DMSO:0.8204490790003547\n",
      "1723050817.439009 INFO: Filtered count:0 out of 526\n",
      "1723050817.6943014 INFO: reactions 871\n",
      "1723050817.8471365 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050818.0530174 INFO: cpd05178 not found in model!\n",
      "1723050818.0559793 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050818.3012934 INFO: Expansion time:Pyr.NO:0.8608604450000712\n",
      "1723050818.3038042 INFO: Filtered count:0 out of 526\n",
      "1723050818.338705 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/422/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050818.6948478 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050819.2000234 INFO: Expansion time:Pyr.NO2:0.894755694000196\n",
      "1723050819.2026548 INFO: Filtered count:0 out of 526\n",
      "1723050819.9980986 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050820.0360005 INFO: Expansion time:Pyr.NO3:0.8315678970002409\n",
      "1723050820.038229 INFO: Filtered count:0 out of 526\n",
      "1723050820.5066595 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050822.2720397 INFO: cpd05178 not found in model!\n",
      "1723050822.4952698 INFO: cpd05178 not found in model!\n",
      "1723050822.4974842 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050823.2370777 INFO: Expansion time:Pyr.O2:3.1974502749999374\n",
      "1723050823.2392895 INFO: Filtered count:0 out of 526\n",
      "1723050824.06477 INFO: Expansion time:Pyr.SO3:0.8240873280001324\n",
      "1723050824.0670142 INFO: Filtered count:0 out of 526\n",
      "1723050824.8769906 INFO: Expansion time:Pyr.SO4:0.8085797739995542\n",
      "1723050824.8791058 INFO: Filtered count:0 out of 526\n",
      "1723050825.438259 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050825.6132042 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050825.7221653 INFO: Expansion time:Pyr.TMAO:0.8416135480001685\n",
      "1723050825.7245488 INFO: Filtered count:0 out of 526\n",
      "1723050826.0754547 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050826.3899932 INFO: cpd05178 not found in model!\n",
      "1723050826.3919573 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050826.560456 INFO: Expansion time:empty:0.8344478720000552\n",
      "1723050826.5628633 INFO: Filtered count:0 out of 526\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[178 'SF2_A_D2.40.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:42:33+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '6bc8bb10891924b264e01e061e1c8999'\n",
      " 13074829 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050829.672044 INFO: cpd05178 not found in model!\n",
      "1723050829.6745555 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050829.8122444 INFO: metabolites 1237\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.20.pg.G.D GF: 11906 Prob: 3248 Other: 3198\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050830.3768864 INFO: reactions 1223\n",
      "1723050830.57236 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050830.611642 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050831.1164088 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/169/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050833.4951963 INFO: cpd05178 not found in model!\n",
      "1723050833.4973018 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050833.7015793 INFO: Expansion time:Dlac.O2:0.943403397000111\n",
      "1723050833.7042177 INFO: Filtered count:0 out of 1338\n",
      "1723050833.7430253 INFO: cpd05178 not found in model!\n",
      "1723050833.7453387 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050834.1198559 INFO: cpd05178 not found in model!\n",
      "1723050834.1222687 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050834.2419455 INFO: cpd05178 not found in model!\n",
      "1723050834.244593 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.38.pg.G.D GF: 11106 Prob: 4314 Other: 2934\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050834.7983527 INFO: Expansion time:empty:1.092554404999646\n",
      "1723050834.8009763 INFO: Filtered count:0 out of 1338\n",
      "1723050838.8676846 INFO: cpd05178 not found in model!\n",
      "1723050838.869979 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_H2O.28.pg.G.D GF: 12171 Prob: 3018 Other: 3163\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050840.0603838 INFO: cpd05178 not found in model!\n",
      "1723050840.0630243 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050842.169475 INFO: Objective with gapfill database:201.35671147741465; min objective:0.01\n",
      "1723050842.7511754 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050842.8646464 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050843.4467216 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050843.8531442 INFO: cpd05178 not found in model!\n",
      "1723050843.8554955 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050844.9423983 INFO: cpd05178 not found in model!\n",
      "1723050846.3866518 INFO: cpd05178 not found in model!\n",
      "1723050846.3891885 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050846.765663 INFO: cpd05178 not found in model!\n",
      "1723050846.768039 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050848.1667368 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050850.8969646 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050852.0238369 INFO: cpd05178 not found in model!\n",
      "1723050852.026348 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050853.3956573 INFO: Expansion time:Ac.NO2:1.904582766999738\n",
      "1723050853.397958 INFO: Filtered count:0 out of 1697\n",
      "1723050853.537093 INFO: cpd05178 not found in model!\n",
      "1723050853.7577667 INFO: cpd05178 not found in model!\n",
      "1723050853.7597463 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050854.204003 INFO: Expansion time:Ac.NO3:0.8047632899997552\n",
      "1723050854.2063818 INFO: Filtered count:0 out of 1697\n",
      "1723050854.998372 INFO: Expansion time:Ac.O2:0.7908210499999768\n",
      "1723050855.000761 INFO: Filtered count:0 out of 1697\n",
      "1723050856.200361 INFO: Expansion time:Etho.O2:1.1981834759999401\n",
      "1723050856.2031474 INFO: Filtered count:0 out of 1697\n",
      "1723050857.3323903 INFO: Expansion time:For.NO2:1.127646038000421\n",
      "1723050857.3350172 INFO: Filtered count:0 out of 1697\n",
      "1723050858.5145383 INFO: Expansion time:For.NO3:1.1780579560008846\n",
      "1723050858.517278 INFO: Filtered count:0 out of 1697\n",
      "1723050858.703317 INFO: cpd05178 not found in model!\n",
      "1723050858.9316308 INFO: cpd05178 not found in model!\n",
      "1723050858.9338193 INFO: Media compound: cpd05178 not found in model.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050860.1455114 INFO: cpd05178 not found in model!\n",
      "1723050860.1482086 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050860.190251 INFO: cpd05178 not found in model!\n",
      "1723050860.1926193 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_D2.40.pg.G.D GF: 11300 Prob: 4068 Other: 2986\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050862.0089316 INFO: Expansion time:For.O2:3.4901396599998407\n",
      "1723050862.011679 INFO: Filtered count:0 out of 1697\n",
      "1723050863.7674918 INFO: Expansion time:Fum:1.7543747529998654\n",
      "1723050863.7700567 INFO: Filtered count:0 out of 1697\n",
      "1723050863.8090475 INFO: Expansion time:For.NO2:4.33001993400012\n",
      "1723050863.811213 INFO: Filtered count:0 out of 1221\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[190 'R2_C_H2O.60.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:44:59+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9d2c8545f98d7658dcdc3e4fb2badb4e'\n",
      " 14077179 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050864.4146733 INFO: cpd05178 not found in model!\n",
      "1723050864.4169497 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050864.6894276 INFO: metabolites 923\n",
      "1723050864.919157 INFO: cpd05178 not found in model!\n",
      "1723050865.1033764 INFO: Expansion time:Fum:1.290980949999721\n",
      "1723050865.1055982 INFO: Filtered count:0 out of 1221\n",
      "1723050865.1782794 INFO: reactions 845\n",
      "1723050865.2808948 INFO: cpd05178 not found in model!\n",
      "1723050865.2831283 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050865.2849293 INFO: cpd05178 not found in model!\n",
      "1723050865.2874851 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050865.311351 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050865.403364 INFO: Expansion time:Fum.O2:1.6318406270002015\n",
      "1723050865.4059167 INFO: Filtered count:0 out of 1697\n",
      "1723050865.566663 INFO: cpd05178 not found in model!\n",
      "1723050865.5691707 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050865.8058243 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/290/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050865.992197 INFO: cpd05178 not found in model!\n",
      "1723050865.9943457 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050866.3465068 INFO: Expansion time:Fum.O2:1.2397256360000029\n",
      "1723050866.3485694 INFO: Filtered count:0 out of 1221\n",
      "1723050866.8900902 INFO: Expansion time:Pyr:1.4826516579996678\n",
      "1723050866.8926935 INFO: Filtered count:0 out of 1697\n",
      "1723050867.5096862 INFO: Expansion time:Glc:1.1599796139998944\n",
      "1723050867.511828 INFO: Filtered count:0 out of 1221\n",
      "1723050868.0976434 INFO: Expansion time:Pyr.DMSO:1.2034676319999562\n",
      "1723050868.100284 INFO: Filtered count:0 out of 1697\n",
      "1723050869.2616415 INFO: Expansion time:Pyr.NO:1.1599393899996358\n",
      "1723050869.2644603 INFO: Filtered count:0 out of 1697\n",
      "1723050870.2413502 INFO: Expansion time:Glc.DMSO:2.728173753000192\n",
      "1723050870.2437506 INFO: Filtered count:0 out of 1221\n",
      "1723050870.2736306 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050870.7557309 INFO: Expansion time:Pyr.NO2:1.489591021000706\n",
      "1723050870.7584302 INFO: Filtered count:0 out of 1697\n",
      "1723050870.9352732 INFO: cpd05178 not found in model!\n",
      "1723050870.9378593 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050871.9972048 INFO: Expansion time:Glc.O2:1.7521256589998302\n",
      "1723050871.9995313 INFO: Filtered count:0 out of 1221\n",
      "1723050872.1679878 INFO: Expansion time:Pyr.NO3:1.4079143230001137\n",
      "1723050872.170641 INFO: Filtered count:0 out of 1697\n",
      "1723050872.549637 INFO: cpd05178 not found in model!\n",
      "1723050872.551751 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050873.1061864 INFO: cpd05178 not found in model!\n",
      "1723050873.108475 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050873.4294548 INFO: cpd05178 not found in model!\n",
      "1723050873.431659 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050873.459283 INFO: cpd05178 not found in model!\n",
      "1723050873.461533 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050873.5461395 INFO: Expansion time:Glc.TMAO:1.5453422760001558\n",
      "1723050873.548284 INFO: Filtered count:0 out of 1221\n",
      "1723050873.8347952 INFO: Expansion time:Pyr.O2:1.6625635210002656\n",
      "1723050873.837394 INFO: Filtered count:0 out of 1697\n",
      "1723050874.3776674 INFO: Expansion time:Pyr:0.828252696999698\n",
      "1723050874.3797417 INFO: Filtered count:0 out of 1221\n",
      "1723050874.8395584 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050875.4414945 INFO: Expansion time:Pyr.DMSO:1.060598265000408\n",
      "1723050875.4436543 INFO: Filtered count:0 out of 1221\n",
      "1723050876.0852337 INFO: cpd05178 not found in model!\n",
      "1723050876.0873826 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050876.756602 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050876.87474 INFO: Expansion time:Pyr.SO3:3.0358400069999334\n",
      "1723050876.877361 INFO: Filtered count:0 out of 1697\n",
      "1723050877.0191798 INFO: Expansion time:Pyr.NO:1.574165480999909\n",
      "1723050877.0216513 INFO: Filtered count:0 out of 1221\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050878.072265 INFO: Expansion time:Pyr.SO4:1.1932868359999702\n",
      "1723050878.07495 INFO: Filtered count:0 out of 1697\n",
      "1723050878.3394191 INFO: Expansion time:Pyr.NO3:1.316264134999983\n",
      "1723050878.3420684 INFO: Filtered count:0 out of 1221\n",
      "1723050879.7211952 INFO: cpd05178 not found in model!\n",
      "1723050879.723298 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050880.1665764 INFO: Expansion time:Pyr.O2:1.822967220999999\n",
      "1723050880.1690807 INFO: Filtered count:0 out of 1221\n",
      "1723050880.446247 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050881.4438741 INFO: Expansion time:Pyr.TMAO:3.367467127999589\n",
      "1723050881.4465976 INFO: Filtered count:0 out of 1697\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050881.9112904 INFO: Expansion time:Pyr.SO3:1.740686927999377\n",
      "1723050881.9136596 INFO: Filtered count:0 out of 1221\n",
      "1723050882.4198017 INFO: Expansion time:Ac.O2:0.6728940110006079\n",
      "1723050882.4219472 INFO: Filtered count:0 out of 1771\n",
      "1723050882.4314446 INFO: Expansion time:empty:0.9832293570007096\n",
      "1723050882.4338868 INFO: Filtered count:0 out of 1697\n",
      "1723050882.9316158 INFO: Expansion time:Pyr.SO4:1.0164216370003487\n",
      "1723050882.9339304 INFO: Filtered count:0 out of 1221\n",
      "1723050883.0850778 INFO: Expansion time:Etho.O2:0.6618421129996932\n",
      "1723050883.0872364 INFO: Filtered count:0 out of 1771\n",
      "1723050883.766534 INFO: Expansion time:Glc:0.6780417479994867\n",
      "1723050883.7690752 INFO: Filtered count:0 out of 1771\n",
      "1723050884.4631736 INFO: Expansion time:Glc.DMSO:0.6927895810003974\n",
      "1723050884.4652984 INFO: Filtered count:0 out of 1771\n",
      "1723050885.1282804 INFO: Expansion time:Glc.O2:0.6613158549998843\n",
      "1723050885.1305058 INFO: Filtered count:0 out of 1771\n",
      "1723050885.314911 INFO: cpd05178 not found in model!\n",
      "1723050885.3171077 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050885.8137405 INFO: Expansion time:Glc.TMAO:0.6820124890000443\n",
      "1723050885.8147614 INFO: Expansion time:Pyr.TMAO:2.8793742210000346\n",
      "1723050885.8159707 INFO: Filtered count:0 out of 1771\n",
      "1723050885.8167427 INFO: Filtered count:0 out of 1221\n",
      "1723050886.6561399 INFO: Expansion time:Glyc.O2:0.8388169290001315\n",
      "1723050886.6585357 INFO: Filtered count:0 out of 1771\n",
      "1723050886.670889 INFO: cpd05178 not found in model!\n",
      "1723050886.672698 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050886.783275 INFO: Expansion time:empty:0.9650760959993931\n",
      "1723050886.7857678 INFO: Filtered count:0 out of 1221\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[71 'R2_A_D1.12.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:21:40+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9cf24b72b6b19c50083c4559c0c7f3db'\n",
      " 13354217 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050887.5715494 INFO: metabolites 1196\n",
      "1723050888.2720163 INFO: reactions 1166\n",
      "1723050888.3742986 INFO: cpd05178 not found in model!\n",
      "1723050888.3764265 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050888.481257 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_H2O.60.pg.G.D GF: 12125 Prob: 3058 Other: 3177\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050889.14701 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/141/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050889.2516077 INFO: Expansion time:Pyr:2.5913626000001386\n",
      "1723050889.2538471 INFO: Filtered count:0 out of 1771\n",
      "1723050890.128508 INFO: cpd05178 not found in model!\n",
      "1723050890.1307333 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050890.3219948 INFO: Expansion time:Pyr.DMSO:1.0667233619997205\n",
      "1723050890.324337 INFO: Filtered count:0 out of 1771\n",
      "1723050891.3572173 INFO: Expansion time:Pyr.NO:1.0313305629997558\n",
      "1723050891.3594806 INFO: Filtered count:0 out of 1771\n",
      "1723050892.405755 INFO: Expansion time:Pyr.NO2:1.0448704800000996\n",
      "1723050892.4081464 INFO: Filtered count:0 out of 1771\n",
      "1723050893.5202208 INFO: Expansion time:Pyr.NO3:1.1106235059996834\n",
      "1723050893.5225441 INFO: Filtered count:0 out of 1771\n",
      "1723050894.63637 INFO: Expansion time:Pyr.O2:1.1122793209997326\n",
      "1723050894.63865 INFO: Filtered count:0 out of 1771\n",
      "1723050895.7266443 INFO: Expansion time:Pyr.SO3:1.0865456390001782\n",
      "1723050895.7290142 INFO: Filtered count:0 out of 1771\n",
      "1723050895.9860373 INFO: cpd05178 not found in model!\n",
      "1723050895.9883647 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050896.897377 INFO: Expansion time:Pyr.SO4:1.166885954999998\n",
      "1723050896.8998466 INFO: Filtered count:0 out of 1771\n",
      "1723050897.5127993 INFO: cpd05178 not found in model!\n",
      "1723050897.5149844 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050898.835178 INFO: Objective with gapfill database:169.3275729196334; min objective:0.01\n",
      "1723050899.0226831 INFO: cpd05178 not found in model!\n",
      "1723050899.3890927 INFO: cpd05178 not found in model!\n",
      "1723050899.391613 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050899.52969 INFO: Expansion time:Pyr.TMAO:2.6283751060000213\n",
      "1723050899.532053 INFO: Filtered count:0 out of 1771\n",
      "1723050900.5688217 INFO: Expansion time:empty:1.0352739660002044\n",
      "1723050900.571219 INFO: Filtered count:0 out of 1771\n",
      "1723050901.012854 INFO: cpd05178 not found in model!\n",
      "1723050901.2596114 INFO: cpd05178 not found in model!\n",
      "1723050901.261794 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050901.3355393 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050901.843033 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050903.4215345 INFO: cpd05178 not found in model!\n",
      "1723050903.667177 INFO: cpd05178 not found in model!\n",
      "1723050903.6693668 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050905.0461488 INFO: cpd05178 not found in model!\n",
      "1723050905.0485296 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050905.5303376 INFO: cpd05178 not found in model!\n",
      "1723050905.5326235 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[83 'R1_B_D1.13.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:23:59+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '6949296bda38d9a7864d98a3faa105cc'\n",
      " 12481903 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050905.9521563 INFO: cpd05178 not found in model!\n",
      "1723050905.954075 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050906.1224947 INFO: metabolites 1175\n",
      "1723050906.4913387 INFO: cpd05178 not found in model!\n",
      "1723050906.63707 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050906.713864 INFO: reactions 1155\n",
      "1723050906.7307749 INFO: cpd05178 not found in model!\n",
      "1723050906.7329242 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050906.8844452 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050907.5121524 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/177/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050907.6431603 INFO: cpd05178 not found in model!\n",
      "1723050907.645315 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050910.0658677 INFO: cpd05178 not found in model!\n",
      "1723050910.0679963 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050911.0525239 INFO: cpd05178 not found in model!\n",
      "1723050911.055158 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050914.5231962 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050914.826561 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050915.2883718 INFO: cpd05178 not found in model!\n",
      "1723050915.2907135 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.12.pg.G.D GF: 11210 Prob: 4196 Other: 2946\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050919.8488343 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050921.9745295 INFO: cpd05178 not found in model!\n",
      "1723050921.9776785 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050922.2873726 INFO: cpd05178 not found in model!\n",
      "1723050922.6830606 INFO: cpd05178 not found in model!\n",
      "1723050922.6856444 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050925.1682332 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050926.6830103 INFO: cpd05178 not found in model!\n",
      "1723050926.685219 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D1.13.pg.G.D GF: 11522 Prob: 3726 Other: 3106\n",
      "[80 'R2_B_H2O.17.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:23:27+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '61867c32428f497cefc8861c590d760f'\n",
      " 9674119 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050928.264695 INFO: metabolites 859\n",
      "1723050928.6867929 INFO: reactions 776\n",
      "1723050928.8114114 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[242 'R2_restored_DShore.20.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:55:20+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'b1114940c80497adcd0d21a383c5b413'\n",
      " 10843565 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050929.264652 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/256/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050929.407339 INFO: metabolites 1151\n",
      "1723050929.9647946 INFO: reactions 1072\n",
      "1723050930.185999 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[310 'R2A_C_D2.10.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:09:15+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '84758238de7c13468eb00a9e1b3467cd'\n",
      " 15622151 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050930.8895383 INFO: metabolites 1143\n",
      "1723050930.8971655 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/43/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050931.6270525 INFO: reactions 1079\n",
      "1723050931.8123913 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050932.4706264 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/142/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050934.1651592 INFO: cpd05178 not found in model!\n",
      "1723050934.4051726 INFO: cpd05178 not found in model!\n",
      "1723050934.4074411 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[214 'R2A_B_D2.11.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:49:54+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '118e12556de9e97452852470bb54c6af'\n",
      " 10564493 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050935.7563367 INFO: metabolites 1077\n",
      "1723050935.9565077 INFO: Expansion time:Ac.O2:2.5886984019998636\n",
      "1723050935.9588377 INFO: Filtered count:0 out of 1661\n",
      "1723050936.3558471 INFO: reactions 1053\n",
      "1723050936.5163453 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050936.845942 INFO: Expansion time:Etho.O2:0.8857950319998054\n",
      "1723050936.8481832 INFO: Filtered count:0 out of 1661\n",
      "1723050937.1150856 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/417/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050937.9852242 INFO: Expansion time:Glc:1.1358480489998328\n",
      "1723050937.9874814 INFO: Filtered count:0 out of 1661\n",
      "1723050938.2088194 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050938.286627 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050938.6857932 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050938.8076034 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050938.9816365 INFO: Expansion time:Glc.DMSO:0.9929947919999904\n",
      "1723050938.9839566 INFO: Filtered count:0 out of 1661\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050940.2527585 INFO: Expansion time:Glc.O2:1.2676545650001572\n",
      "1723050940.2547967 INFO: Filtered count:0 out of 1661\n",
      "1723050940.2863078 INFO: Objective with gapfill database:219.4465597765115; min objective:0.01\n",
      "1723050941.8990974 INFO: Expansion time:Glc.TMAO:1.6431544479992226\n",
      "1723050941.9012687 INFO: Filtered count:0 out of 1661\n",
      "1723050941.987915 INFO: cpd05178 not found in model!\n",
      "1723050941.9901397 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050942.0289068 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050943.223761 INFO: Expansion time:Pyr.O2:1.3212008239997886\n",
      "1723050943.2259603 INFO: Filtered count:0 out of 1661\n",
      "1723050943.8685908 INFO: Expansion time:empty:0.6413195830000404\n",
      "1723050943.8709354 INFO: Filtered count:0 out of 1661\n",
      "1723050944.05741 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050944.2115586 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050944.5027287 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050945.4960804 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050945.7280858 INFO: cpd05178 not found in model!\n",
      "1723050945.730318 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050946.3007348 INFO: cpd05178 not found in model!\n",
      "1723050946.3033502 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[91 'R2_C_D1.98.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:25:30+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '938364cf3e84eae6d3d22c55d651504f'\n",
      " 11593029 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050947.8644347 INFO: metabolites 1080\n",
      "1723050948.3302574 INFO: cpd05178 not found in model!\n",
      "1723050948.3323624 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050948.3458264 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050948.4121904 INFO: reactions 1042\n",
      "1723050948.5917304 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050949.2325723 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/234/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050949.4779215 INFO: cpd05178 not found in model!\n",
      "1723050949.4800093 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050950.8281748 INFO: cpd05178 not found in model!\n",
      "1723050950.8306181 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050951.4127467 INFO: cpd05178 not found in model!\n",
      "1723050951.415441 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_H2O.17.pg.G.D GF: 12247 Prob: 2852 Other: 3261\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050953.720687 INFO: cpd05178 not found in model!\n",
      "1723050953.741858 INFO: cpd05178 not found in model!\n",
      "1723050953.744285 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.20.pg.G.D GF: 11672 Prob: 3654 Other: 3028\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050955.6741335 INFO: cpd05178 not found in model!\n",
      "1723050955.6766286 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050956.5335002 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050957.0161972 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[127 'R2_A_D1.001.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:32:46+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '452fb65b31e4b808a48e133fe7ab6091'\n",
      " 6110648 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050958.1267602 INFO: metabolites 964\n",
      "1723050958.7922673 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050958.8056412 INFO: reactions 880\n",
      "1723050958.9426146 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050959.2944295 INFO: cpd05178 not found in model!\n",
      "1723050959.2970548 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050959.3082914 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050959.4015334 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/492/1\n",
      "R2A_B_D2.11.pg.G.D GF: 11723 Prob: 3522 Other: 3107\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050960.74134 INFO: cpd05178 not found in model!\n",
      "1723050960.7435973 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050961.0487845 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050962.4287422 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050965.1271136 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050966.4586453 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050966.790629 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050966.9895816 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[68 'R2_B_D2.143.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:21:09+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '95f6e01542550fb748b41bb812373291'\n",
      " 12575251 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050969.5339513 INFO: metabolites 1255\n",
      "1723050970.4088755 INFO: reactions 1252\n",
      "1723050970.4734468 INFO: cpd05178 not found in model!\n",
      "1723050970.4756248 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050970.6109967 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050971.3655846 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/112/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050971.9058838 INFO: cpd05178 not found in model!\n",
      "1723050971.9080756 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050972.0100477 INFO: Expansion time:Etho.O2:0.6082843349995528\n",
      "1723050972.0120618 INFO: Filtered count:0 out of 1065\n",
      "1723050972.6367638 INFO: Expansion time:Glyc.O2:0.6235289200003535\n",
      "1723050972.6393318 INFO: Filtered count:0 out of 1065\n",
      "1723050972.6723883 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050973.2574556 INFO: Expansion time:empty:0.6167478110000957\n",
      "1723050973.2596376 INFO: Filtered count:0 out of 1065\n",
      "1723050973.6792881 INFO: cpd05178 not found in model!\n",
      "1723050973.6818695 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D1.98.pg.G.D GF: 11869 Prob: 3264 Other: 3219\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050974.466705 INFO: cpd05178 not found in model!\n",
      "1723050974.7026896 INFO: cpd05178 not found in model!\n",
      "1723050974.7048995 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050976.3799822 INFO: cpd05178 not found in model!\n",
      "1723050976.382473 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.10.pg.G.D GF: 11344 Prob: 4028 Other: 2982\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050977.3707073 INFO: Objective with gapfill database:234.98762331871953; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050980.0721953 INFO: Expansion time:Etho.O2:1.7554104499995447\n",
      "1723050980.0759294 INFO: Filtered count:0 out of 1513\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[220 'R2_A_D2.9.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:51:04+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '3b577588aabe71e847af5d4f60eaaa99'\n",
      " 7690734 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050982.32487 INFO: metabolites 888\n",
      "1723050982.5434504 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050982.6257749 INFO: reactions 812\n",
      "1723050982.7182934 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050982.8802512 INFO: cpd05178 not found in model!\n",
      "1723050983.038001 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/208/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050983.2283335 INFO: cpd05178 not found in model!\n",
      "1723050983.2310233 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050983.2436912 INFO: cpd05178 not found in model!\n",
      "1723050983.246281 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723050983.5504153 INFO: cpd05178 not found in model!\n",
      "1723050983.5530617 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.001.pg.G.D Failed:>rxn00062_c0GF:\n",
      " 12034 Prob: 3158 Other: 3160\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050983.976221 INFO: Expansion time:Fum:3.8988213960001303\n",
      "1723050983.97894 INFO: Filtered count:1 out of 1513\n",
      "1723050984.9208949 INFO: Expansion time:Fum.O2:0.9403630269998757\n",
      "1723050984.9233918 INFO: Filtered count:1 out of 1513\n",
      "1723050985.0587573 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050985.6294434 INFO: Expansion time:Pyr:0.7042998919996535\n",
      "1723050985.6315625 INFO: Filtered count:1 out of 1513\n",
      "1723050988.0739303 INFO: Expansion time:Pyr.DMSO:2.441123130000051\n",
      "1723050988.0763123 INFO: Filtered count:1 out of 1513\n",
      "1723050988.358993 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050988.7423222 INFO: Expansion time:Pyr.NO:0.6647292699999525\n",
      "1723050988.7444265 INFO: Filtered count:1 out of 1513\n",
      "1723050988.9427557 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050989.4050581 INFO: Expansion time:Pyr.NO2:0.6594149549991926\n",
      "1723050989.4071527 INFO: Filtered count:1 out of 1513\n",
      "1723050989.7253056 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723050990.0924008 INFO: Expansion time:Pyr.NO3:0.6840684819999296\n",
      "1723050990.0946512 INFO: Filtered count:1 out of 1513\n",
      "1723050990.634508 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050990.7594702 INFO: Expansion time:Pyr.O2:0.6635654530000465\n",
      "1723050990.7616372 INFO: Filtered count:1 out of 1513\n",
      "1723050991.087503 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723050991.4608657 INFO: Expansion time:Pyr.SO3:0.6979711929998302\n",
      "1723050991.4630597 INFO: Filtered count:1 out of 1513\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723050991.735911 INFO: cpd05178 not found in model!\n",
      "1723050991.737778 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723050992.128807 INFO: Expansion time:Pyr.SO4:0.6645963070004655\n",
      "1723050992.1311574 INFO: Filtered count:1 out of 1513\n",
      "1723050993.8700495 INFO: Expansion time:Pyr.TMAO:1.7375877300000866\n",
      "1723050993.8722477 INFO: Filtered count:1 out of 1513\n",
      "1723050994.5434008 INFO: Expansion time:empty:0.6699560429997291\n",
      "1723050994.5455763 INFO: Filtered count:1 out of 1513\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050996.8220296 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723050999.4488137 INFO: cpd05178 not found in model!\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723050999.6922896 INFO: cpd05178 not found in model!\n",
      "1723050999.6944897 INFO: Media compound: cpd05178 not found in model.\n",
      "1723050999.7167802 INFO: cpd05178 not found in model!\n",
      "1723050999.9368541 INFO: cpd05178 not found in model!\n",
      "1723050999.9391088 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051000.339705 INFO: cpd05178 not found in model!\n",
      "1723051000.3421254 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051000.5525234 INFO: cpd05178 not found in model!\n",
      "1723051000.7800539 INFO: Expansion time:Ac.O2:1.0294032089996108\n",
      "1723051000.782357 INFO: Filtered count:0 out of 1557\n",
      "1723051000.800589 INFO: cpd05178 not found in model!\n",
      "1723051000.8025277 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051000.811929 INFO: cpd05178 not found in model!\n",
      "1723051000.8140647 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.143.pg.G.D GF: 11222 Prob: 4198 Other: 2932\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051001.2859607 INFO: Objective with gapfill database:138.07650917688392; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051001.7785275 INFO: Expansion time:Etho.O2:0.9946889160000865\n",
      "1723051001.780771 INFO: Filtered count:0 out of 1557\n",
      "1723051002.436811 INFO: cpd05178 not found in model!\n",
      "1723051002.4390552 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051002.526322 INFO: Expansion time:Dlac:0.6684322500004782\n",
      "1723051002.5284367 INFO: Filtered count:0 out of 1499\n",
      "1723051002.7967298 INFO: Expansion time:For.O2:1.014697705000799\n",
      "1723051002.799196 INFO: Filtered count:0 out of 1557\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[226 'R1_B_D2.37.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:52:12+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '7883c5e336ada42d62623e7b6bfd6faa'\n",
      " 9492315 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051003.526428 INFO: metabolites 1195\n",
      "1723051003.6498647 INFO: Expansion time:Dlac.O2:1.1200789750000695\n",
      "1723051003.6527095 INFO: Filtered count:0 out of 1499\n",
      "1723051004.2747567 INFO: reactions 1162\n",
      "1723051004.5154908 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051004.8382387 INFO: Expansion time:Fum:1.1834943779995228\n",
      "1723051004.8407872 INFO: Filtered count:0 out of 1499\n",
      "1723051005.1944764 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/521/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051005.515795 INFO: Expansion time:Glc:2.715145193999888\n",
      "1723051005.518725 INFO: Filtered count:0 out of 1557\n",
      "1723051005.9405148 INFO: Expansion time:Fum.O2:1.0982880470000964\n",
      "1723051005.9431126 INFO: Filtered count:0 out of 1499\n",
      "1723051006.5524168 INFO: Expansion time:Glc.DMSO:1.032015579000472\n",
      "1723051006.5549436 INFO: Filtered count:0 out of 1557\n",
      "1723051006.7059426 INFO: cpd05178 not found in model!\n",
      "1723051006.7086499 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051006.9687145 INFO: Expansion time:Pyr:1.0241681970001082\n",
      "1723051006.9712915 INFO: Filtered count:0 out of 1499\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D2.9.pg.G.D GF: 12382 Prob: 2686 Other: 3292\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051007.5269833 INFO: Expansion time:Glc.O2:0.970591553000304\n",
      "1723051007.529369 INFO: Filtered count:0 out of 1557\n",
      "1723051007.9805508 INFO: Expansion time:Pyr.DMSO:1.0076996630004942\n",
      "1723051007.9830124 INFO: Filtered count:0 out of 1499\n",
      "1723051008.4689796 INFO: Expansion time:Glc.TMAO:0.9381041229999028\n",
      "1723051008.4713767 INFO: Filtered count:0 out of 1557\n",
      "1723051008.986561 INFO: Expansion time:Pyr.NO:1.0021118519998709\n",
      "1723051008.9890032 INFO: Filtered count:0 out of 1499\n",
      "1723051009.5001345 INFO: Expansion time:Pyr:1.0271719520005718\n",
      "1723051009.5026488 INFO: Filtered count:0 out of 1557\n",
      "1723051010.5244122 INFO: cpd05178 not found in model!\n",
      "1723051010.6381197 INFO: Expansion time:Pyr.DMSO:1.1339768600000752\n",
      "1723051010.6407175 INFO: Filtered count:0 out of 1557\n",
      "1723051010.7666314 INFO: cpd05178 not found in model!\n",
      "1723051010.7687452 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051011.672738 INFO: Expansion time:Pyr.NO:1.0304310390001774\n",
      "1723051011.6752295 INFO: Filtered count:0 out of 1557\n",
      "1723051012.2403755 INFO: Expansion time:Pyr.NO2:3.2499471989995072\n",
      "1723051012.2428787 INFO: Filtered count:0 out of 1499\n",
      "1723051012.5254002 INFO: Objective with gapfill database:134.11916786634458; min objective:0.01\n",
      "1723051012.849945 INFO: Expansion time:Pyr.NO2:1.173341783000069\n",
      "1723051012.8525457 INFO: Filtered count:0 out of 1557\n",
      "1723051013.2964718 INFO: Expansion time:Pyr.NO3:1.0520203570004014\n",
      "1723051013.2990923 INFO: Filtered count:0 out of 1499\n",
      "1723051013.783462 INFO: cpd05178 not found in model!\n",
      "1723051014.0326977 INFO: cpd05178 not found in model!\n",
      "1723051014.0351434 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051014.393596 INFO: Expansion time:Pyr.O2:1.0930105739998908\n",
      "1723051014.3961017 INFO: Filtered count:0 out of 1499\n",
      "1723051015.4029188 INFO: Expansion time:Pyr.SO3:1.0054636029999529\n",
      "1723051015.4053764 INFO: Filtered count:0 out of 1499\n",
      "1723051015.663641 INFO: Expansion time:Pyr.NO3:2.8094320000000153\n",
      "1723051015.6661236 INFO: Filtered count:0 out of 1557\n",
      "1723051016.3320937 INFO: Expansion time:Pyr.SO4:0.9253037010003027\n",
      "1723051016.334614 INFO: Filtered count:0 out of 1499\n",
      "1723051016.6536536 INFO: Expansion time:Pyr.O2:0.9860457539998606\n",
      "1723051016.6561766 INFO: Filtered count:0 out of 1557\n",
      "1723051017.4391918 INFO: Expansion time:Pyr.TMAO:1.1031344719995104\n",
      "1723051017.441646 INFO: Filtered count:0 out of 1499\n",
      "1723051017.5288968 INFO: Objective with gapfill database:186.2824921811983; min objective:0.01\n",
      "1723051017.7688975 INFO: Expansion time:Pyr.SO3:1.1112037289994987\n",
      "1723051017.7715092 INFO: Filtered count:0 out of 1557\n",
      "1723051018.4930341 INFO: cpd05178 not found in model!\n",
      "1723051018.5762355 INFO: Expansion time:empty:1.1331905100005315\n",
      "1723051018.5790794 INFO: Filtered count:0 out of 1499\n",
      "1723051018.6246257 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[12 'SF2_C_D2.32.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:10:02+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '2e3b569d01abd54ccb8f53aaaf0873f8'\n",
      " 14478138 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051018.8128266 INFO: Expansion time:Pyr.SO4:1.0396588300000076\n",
      "1723051018.8152635 INFO: Filtered count:0 out of 1557\n",
      "1723051018.832735 INFO: cpd05178 not found in model!\n",
      "1723051018.8351653 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051018.9300032 INFO: cpd05178 not found in model!\n",
      "1723051018.9321783 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051019.8071637 INFO: Expansion time:Pyr.TMAO:0.9904307530005099\n",
      "1723051019.8096511 INFO: Filtered count:0 out of 1557\n",
      "1723051020.8008127 INFO: Expansion time:empty:0.989671917999658\n",
      "1723051020.8033924 INFO: Filtered count:0 out of 1557\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[147 'R2_B_D2.39.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:36:40+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '92baebb432bdf0d7ed949a7a5aa9b307'\n",
      " 11983928 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051021.5590444 INFO: metabolites 1201\n",
      "1723051021.5966058 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051022.1891587 INFO: metabolites 1176\n",
      "1723051022.2966201 INFO: cpd05178 not found in model!\n",
      "1723051022.3369575 INFO: reactions 1178\n",
      "1723051022.535182 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051022.6366825 INFO: cpd05178 not found in model!\n",
      "1723051022.6391728 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051022.8316958 INFO: reactions 1112\n",
      "1723051023.0097802 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051023.1981447 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/600/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051023.2656677 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051023.7150054 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/379/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051023.7923765 INFO: cpd05178 not found in model!\n",
      "1723051023.7943954 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051024.5701919 INFO: cpd05178 not found in model!\n",
      "1723051024.5723162 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[275 'R2A_C_D2.38.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:02:00+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '0d2792d0c791bc0ee316e9ac95190b2f'\n",
      " 7069317 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051028.4302423 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[100 'SF2_A_D2.13.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:27:29+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'c2d02fa153eec405f7f22e167680ef60'\n",
      " 17864599 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051029.0547152 INFO: metabolites 1038\n",
      "1723051029.0855272 INFO: metabolites 1156\n",
      "1723051029.6286614 INFO: reactions 962\n",
      "1723051029.784547 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051029.7933316 INFO: reactions 1096\n",
      "1723051029.964056 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051030.339944 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/175/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051030.6107702 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/314/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051030.8926978 INFO: cpd05178 not found in model!\n",
      "1723051030.8949769 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051031.8705463 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051032.337062 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051033.7257617 INFO: cpd05178 not found in model!\n",
      "1723051033.7281864 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051034.0221753 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051034.4850829 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[235 'R2A_C_H2O.4.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:53:57+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '18a85067502b59901c7585df903b69dc'\n",
      " 10861409 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051036.1478257 INFO: metabolites 1107\n",
      "1723051036.5696785 INFO: reactions 1070\n",
      "1723051036.694288 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051036.7401578 INFO: cpd05178 not found in model!\n",
      "1723051036.7428064 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051037.4697232 INFO: cpd05178 not found in model!\n",
      "1723051037.4721088 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.37.pg.G.D GF: 11041 Prob: 4526 Other: 2785\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051038.3371735 INFO: cpd05178 not found in model!\n",
      "1723051038.3961153 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/541/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051038.5466177 INFO: cpd05178 not found in model!\n",
      "1723051038.5858994 INFO: cpd05178 not found in model!\n",
      "1723051038.5880702 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051038.6231093 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051039.1544785 INFO: cpd05178 not found in model!\n",
      "1723051039.1567845 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051039.7217305 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051039.825558 INFO: cpd05178 not found in model!\n",
      "1723051039.828022 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051041.0722964 INFO: cpd05178 not found in model!\n",
      "1723051041.0751722 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051041.1764147 INFO: cpd05178 not found in model!\n",
      "1723051041.17883 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051042.5951893 INFO: cpd05178 not found in model!\n",
      "1723051042.5976636 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051043.9977188 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051044.2340012 INFO: cpd05178 not found in model!\n",
      "1723051044.2362964 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051044.480962 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051045.8668861 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051045.9425774 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[44 'SF2_C_D2.19.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:16:15+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e01965773b462aa5df448c582a5cdf96'\n",
      " 8761802 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051046.7471356 INFO: metabolites 1029\n",
      "1723051047.201013 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051047.352415 INFO: reactions 1020\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[95 'SF2_B_D2.42.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:26:14+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ba727163c3826d54daaa76f32f310b7a'\n",
      " 10314192 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051047.5219905 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051047.716914 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051047.7245698 INFO: metabolites 991\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051048.0922146 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051048.1235244 INFO: cpd05178 not found in model!\n",
      "1723051048.125104 INFO: reactions 887\n",
      "1723051048.1257558 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051048.1420867 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "188040/585/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051048.2285335 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2.32.pg.G.D GF: 11701 Prob: 3508 Other: 3143\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051048.583127 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/306/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051050.007645 INFO: cpd05178 not found in model!\n",
      "1723051050.009937 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.39.pg.G.D GF: 11140 Prob: 4274 Other: 2938\n",
      "[60 'R2A_C_H2O.67.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:19:40+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '6dad8db523beafba1b01b928545ae2b0'\n",
      " 7053031 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051050.6393573 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051051.0650866 INFO: metabolites 991\n",
      "1723051051.6303546 INFO: reactions 923\n",
      "1723051051.7839673 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051052.3367329 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/28/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051053.595639 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051053.6126878 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051056.7313247 INFO: Expansion time:Ac.O2:2.1980372400003034\n",
      "1723051056.733508 INFO: Filtered count:0 out of 1677\n",
      "1723051057.7136133 INFO: Expansion time:Dlac.O2:0.9788061219996962\n",
      "1723051057.715774 INFO: Filtered count:0 out of 1677\n",
      "1723051057.8136265 INFO: cpd05178 not found in model!\n",
      "1723051057.8156743 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051058.1515799 INFO: cpd05178 not found in model!\n",
      "1723051058.1536973 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_D2.13.pg.G.D GF: 11622 Prob: 3624 Other: 3106\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051058.627068 INFO: cpd05178 not found in model!\n",
      "1723051058.6297698 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051059.106114 INFO: Expansion time:Etho.O2:1.3888984619998155\n",
      "1723051059.1085765 INFO: Filtered count:0 out of 1677\n",
      "1723051060.1085906 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051060.4434104 INFO: Objective with gapfill database:236.58105084645763; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051061.8579698 INFO: cpd05178 not found in model!\n",
      "1723051061.860252 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051062.200207 INFO: Expansion time:For.O2:3.0900046100005056\n",
      "1723051062.202773 INFO: Filtered count:0 out of 1677\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_H2O.4.pg.G.D GF: 11556 Prob: 3862 Other: 2936\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051062.5991068 INFO: cpd05178 not found in model!\n",
      "1723051062.6012173 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.38.pg.G.D GF: 11564 Prob: 3846 Other: 2944\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051063.1258414 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051063.2224317 INFO: cpd05178 not found in model!\n",
      "1723051063.5941896 INFO: cpd05178 not found in model!\n",
      "1723051063.5967402 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051063.8608732 INFO: Expansion time:Fum:1.6562772600000244\n",
      "1723051063.863297 INFO: Filtered count:0 out of 1677\n",
      "1723051064.3797421 INFO: cpd05178 not found in model!\n",
      "1723051064.7190096 INFO: cpd05178 not found in model!\n",
      "1723051064.7215874 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051065.126275 INFO: Expansion time:Fum.O2:1.2615190240003358\n",
      "1723051065.1281815 INFO: Filtered count:0 out of 1677\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[120 'R1_B_D1.89.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:31:21+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '874b44f13ae6e0fbbe9a95aaeba8094a'\n",
      " 14819477 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051065.966671 INFO: metabolites 1025\n",
      "1723051066.0820992 INFO: Expansion time:Pyr:0.9526772489998621\n",
      "1723051066.0841117 INFO: Filtered count:0 out of 1677\n",
      "1723051066.4147093 INFO: reactions 1008\n",
      "1723051066.4584148 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051066.5283737 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051066.9866931 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051067.094633 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/462/1\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051067.6203191 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051068.6702192 INFO: Expansion time:Pyr.DMSO:2.5848949709998124\n",
      "1723051068.672169 INFO: Filtered count:0 out of 1677\n",
      "1723051069.4591029 INFO: Expansion time:Pyr.NO:0.785759839000093\n",
      "1723051069.4610193 INFO: Filtered count:0 out of 1677\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051070.6844249 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051071.0134501 INFO: Expansion time:Pyr.NO2:1.5512667039993175\n",
      "1723051071.0154252 INFO: Filtered count:0 out of 1677\n",
      "1723051071.6039495 INFO: cpd05178 not found in model!\n",
      "1723051071.8107557 INFO: Expansion time:Pyr.NO3:0.7941887429997223\n",
      "1723051071.8127594 INFO: Filtered count:0 out of 1677\n",
      "1723051071.8235543 INFO: cpd05178 not found in model!\n",
      "1723051071.8257751 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051072.1201658 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051074.0973067 INFO: Expansion time:Pyr.O2:2.2833299109997824\n",
      "1723051074.0993247 INFO: Filtered count:0 out of 1677\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051074.5452275 INFO: Objective with gapfill database:141.6131984139673; min objective:0.01\n",
      "1723051075.34772 INFO: Expansion time:Ac.NO2:4.852834414999961\n",
      "1723051075.350172 INFO: Filtered count:0 out of 1692\n",
      "1723051075.3981416 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[298 'SF2_C_H2O.13.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:06:46+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9439019003deb7c2ac322ade9c9a9a07'\n",
      " 9728682 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051077.1441598 INFO: Objective with gapfill database:176.0338208056842; min objective:0.01\n",
      "1723051077.1939917 INFO: metabolites 1168\n",
      "1723051077.9129398 INFO: reactions 1139\n",
      "1723051077.9984941 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051078.1099036 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051078.5009117 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn01632_c0\n",
      "Failed:<rxn12008_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051078.779213 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/340/1Failed:<rxn03086_c0\n",
      "\n",
      "Getting tests from attributes\n",
      "Failed:<rxn09231_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051078.9597409 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn08333_c0\n",
      "Failed:<rxn00738_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051079.329243 INFO: Expansion time:Ac.O2:3.977625612999873\n",
      "1723051079.3317194 INFO: Filtered count:0 out of 1692\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn02898_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051079.4671137 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn12215_c0\n",
      "Failed:>rxn03030_c0\n",
      "Getting tests from attributes\n",
      "Failed:<rxn02988_c0\n",
      "Failed:<rxn08773_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051080.2658327 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn01117_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051080.501288 INFO: cpd05178 not found in model!\n",
      "1723051080.5036578 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn09043_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051080.7436037 INFO: cpd05178 not found in model!\n",
      "1723051080.7457376 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00442_c0\n",
      "Failed:<rxn12249_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051081.2174206 INFO: Expansion time:Etho.O2:1.8842968540002403\n",
      "1723051081.2198055 INFO: Filtered count:0 out of 1692\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn00128_c0\n",
      "Failed:<rxn01048_c0\n",
      "Failed:>rxn09360_c0\n",
      "Failed:<rxn02331_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051081.860547 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn00796_c0\n",
      "Failed:<rxn00902_c0\n",
      "Failed:<rxn00068_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051082.6804247 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn05514_c0\n",
      "Failed:>rxn09427_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051083.0597973 INFO: cpd05178 not found in model!\n",
      "1723051083.06256 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00340_c0\n",
      "Failed:>rxn05522_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2.19.pg.G.D GF: 11274 Prob: 4290 Other: 2790\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051083.7367795 INFO: cpd05178 not found in model!\n",
      "1723051083.7389343 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00957_c0\n",
      "R2A_C_H2O.67.pg.G.D GF: 11930 Prob: 3454 Other: 2970\n",
      "Failed:>rxn02159_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051084.1539779 INFO: Expansion time:Fum:2.9326412740001615\n",
      "1723051084.1564934 INFO: Filtered count:0 out of 1692\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn13722_c0\n",
      "Failed:<rxn14207_c0\n",
      "Failed:>rxn10213_c0\n",
      "Failed:>rxn02404_c0\n",
      "Failed:>rxn05039_c0\n",
      "Failed:>rxn09180_c0\n",
      "Failed:>rxn03891_c0\n",
      "Failed:>rxn03052_c0\n",
      "Failed:>rxn10204_c0\n",
      "Failed:>rxn02175_c0\n",
      "Failed:<rxn05211_c0\n",
      "Failed:>rxn04464_c0\n",
      "Failed:>rxn03536_c0\n",
      "Failed:>rxn09562_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051086.5491154 INFO: Expansion time:Ac.O2:3.1280651670003863\n",
      "1723051086.5512624 INFO: Filtered count:0 out of 1367\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn09158_c0\n",
      "Failed:>rxn02830_c0\n",
      "Failed:>rxn01505_c0\n",
      "Failed:>rxn09233_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051087.28142 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn02893_c0\n",
      "Failed:>rxn03910_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051087.5826714 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051087.6299005 INFO: cpd05178 not found in model!\n",
      "1723051087.6324625 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn00003_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051087.7855525 INFO: Expansion time:Etho.O2:1.2331288530003803\n",
      "1723051087.787558 INFO: Filtered count:0 out of 1367\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn05227_c0\n",
      "Failed:<rxn08125_c0\n",
      "Failed:<rxn03617_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051088.718649 INFO: Expansion time:Fum.O2:4.5606948399999965\n",
      "1723051088.721193 INFO: Filtered count:0 out of 1692\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn01186_c0\n",
      "Failed:<rxn09123_c0\n",
      "Failed:>rxn01972_c0\n",
      "Failed:>rxn00979_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051089.4483216 INFO: Expansion time:Glc.O2:1.659627693999937\n",
      "1723051089.4504437 INFO: Filtered count:0 out of 1367\n",
      "1723051089.6073399 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051089.616794 INFO: cpd05178 not found in model!\n",
      "1723051089.619366 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn09359_c0\n",
      "Failed:>rxn01255_c0\n",
      "Getting tests from attributes\n",
      "Failed:>rxn03537_c0\n",
      "Failed:>rxn00863_c0\n",
      "Failed:<rxn02402_c0\n",
      "Failed:>rxn01730_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051090.9604182 INFO: Expansion time:Pyr:2.237734272000125\n",
      "1723051090.9630382 INFO: Filtered count:0 out of 1692\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn01539_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051091.0298336 INFO: Expansion time:Pyr.O2:1.5782034550002209\n",
      "1723051091.0319703 INFO: Filtered count:0 out of 1367\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn10094_c0\n",
      "Failed:<rxn03005_c0\n",
      "Failed:>rxn02312_c0\n",
      "Failed:>rxn01508_c0\n",
      "Failed:>rxn00420_c0\n",
      "Failed:>rxn09045_c0\n",
      "Failed:>rxn00440_c0\n",
      "Failed:>rxn03146_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051092.112929 INFO: Expansion time:Pyr.DMSO:1.1483783570001833\n",
      "1723051092.1154377 INFO: Filtered count:0 out of 1692\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn05035_c0\n",
      "Failed:>rxn12510_c0\n",
      "Failed:>rxn02405_c0\n",
      "Failed:>rxn01631_c0\n",
      "Failed:<rxn03880_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051092.7375262 INFO: Expansion time:Pyr.SO4:1.7043892070005313\n",
      "1723051092.7396905 INFO: Filtered count:0 out of 1367\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03907_c0\n",
      "Failed:>rxn03538_c0\n",
      "Failed:>rxn09310_c0\n",
      "Failed:>rxn02834_c0\n",
      "Failed:>rxn01213_c0\n",
      "Failed:>rxn05287_c0\n",
      "Failed:>rxn09101_c0\n",
      "Failed:>rxn02937_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051093.5745318 INFO: Expansion time:Pyr.NO:1.4576647479998428\n",
      "1723051093.5769997 INFO: Filtered count:0 out of 1692\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03398_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051093.643088 INFO: Expansion time:empty:0.9020172939999611\n",
      "1723051093.6451921 INFO: Filtered count:0 out of 1367\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03841_c0\n",
      "Failed:>rxn12239_c0\n",
      "Failed:<rxn09142_c0\n",
      "Failed:>rxn02213_c0\n",
      "Failed:>rxn03892_c0\n",
      "Failed:>rxn03393_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051094.4939775 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn01265_c0\n",
      "Failed:>rxn09040_c0\n",
      "Failed:>rxn02897_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051095.2925155 INFO: cpd05178 not found in model!\n",
      "1723051095.2952864 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051095.441998 INFO: cpd05178 not found in model!\n",
      "1723051095.4443057 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051095.459723 INFO: cpd05178 not found in model!\n",
      "1723051095.4952965 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn02277_c0\n",
      "Failed:>rxn05144_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051095.6694515 INFO: cpd05178 not found in model!\n",
      "1723051095.6716459 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_D2.42.pg.G.D GF: 11654 Prob: 3612 Other: 3088\n",
      "R1_B_D1.89.pg.G.D GF: 11846 Prob: 3488 Other: 3020\n",
      "Failed:<rxn08775_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051095.9272978 INFO: Expansion time:Pyr.NO2:2.3488447359995916\n",
      "1723051095.9298832 INFO: Filtered count:0 out of 1692\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn02831_c0\n",
      "Failed:<rxn03866_c0\n",
      "Failed:>rxn01258_c0\n",
      "Failed:<rxn02832_c0\n",
      "Failed:>rxn02297_c0\n",
      "Failed:>rxn08131_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051097.1256008 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn01603_c0\n",
      "Failed:>rxn08233_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051097.5360723 INFO: Expansion time:Pyr.SO3:23.435542262000126\n",
      "1723051097.5384958 INFO: Filtered count:105 out of 1677\n",
      "1723051098.9263794 INFO: Expansion time:Pyr.NO3:2.9949886759995934\n",
      "1723051098.928583 INFO: Filtered count:0 out of 1692\n",
      "1723051100.6949632 INFO: Expansion time:Pyr.SO4:3.154457153000294\n",
      "1723051100.6972678 INFO: Filtered count:105 out of 1677\n",
      "1723051100.9576776 INFO: Expansion time:Pyr.O2:2.024174615999982\n",
      "1723051100.9599028 INFO: Filtered count:0 out of 1692\n",
      "1723051102.3007994 INFO: cpd05178 not found in model!\n",
      "1723051102.5012927 INFO: Expansion time:Pyr.TMAO:1.802356124999278\n",
      "1723051102.5032396 INFO: Filtered count:105 out of 1677\n",
      "1723051102.5300856 INFO: cpd05178 not found in model!\n",
      "1723051102.53214 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051102.5801446 INFO: cpd05178 not found in model!\n",
      "1723051102.8392181 INFO: Expansion time:Pyr.SO3:1.8780931259998397\n",
      "1723051102.841309 INFO: Filtered count:0 out of 1692\n",
      "1723051102.9574745 INFO: cpd05178 not found in model!\n",
      "1723051102.9601974 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051103.21515 INFO: Expansion time:empty:0.7106771900007516\n",
      "1723051103.217467 INFO: Filtered count:105 out of 1677\n",
      "1723051103.648864 INFO: Expansion time:Pyr.SO4:0.8062649180001245\n",
      "1723051103.651023 INFO: Filtered count:0 out of 1692\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[159 'R1_A_D2.27.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:38:54+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '946e2e1cca578688e4665596b8a911f4'\n",
      " 10628660 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051104.622119 INFO: Expansion time:Pyr.TMAO:0.9698612250003862\n",
      "1723051104.6243315 INFO: Filtered count:0 out of 1692\n",
      "1723051104.6365745 INFO: metabolites 1076\n",
      "1723051104.9036317 INFO: cpd05178 not found in model!\n",
      "1723051105.2229145 INFO: reactions 1034\n",
      "1723051105.303932 INFO: cpd05178 not found in model!\n",
      "1723051105.306693 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051105.3819764 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051105.5434055 INFO: cpd05178 not found in model!\n",
      "1723051105.5456696 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_H2O.13.pg.G.D GF: 11492 Prob: 3916 Other: 2944\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051105.9721363 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/70/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051106.7931578 INFO: Expansion time:empty:2.167563046000396\n",
      "1723051106.795439 INFO: Filtered count:0 out of 1692\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051107.125965 INFO: Objective with gapfill database:204.52654489345403; min objective:0.01\n",
      "1723051107.7287796 INFO: Expansion time:Ac.O2:0.6152952279999226\n",
      "1723051107.7309873 INFO: Filtered count:0 out of 1428\n",
      "1723051108.3393815 INFO: Expansion time:Etho.O2:0.6071963780004808\n",
      "1723051108.341586 INFO: Filtered count:0 out of 1428\n",
      "1723051108.7258487 INFO: Objective with gapfill database:140.24025549450639; min objective:0.01\n",
      "1723051109.2094212 INFO: Expansion time:Pyr:0.8664088619998438\n",
      "1723051109.211936 INFO: Filtered count:0 out of 1428\n",
      "1723051109.605824 INFO: cpd05178 not found in model!\n",
      "1723051109.608347 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051110.095545 INFO: Expansion time:Pyr.DMSO:0.8822200449994853\n",
      "1723051110.098254 INFO: Filtered count:0 out of 1428\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051110.9993103 INFO: Expansion time:Pyr.NO:0.8996112179993361\n",
      "1723051111.0017874 INFO: Filtered count:0 out of 1428\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[306 'R2_restored_DShore.12.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:08:26+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '4db22f729915a8c1ef9a730db0ef697a'\n",
      " 15775973 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051111.8478794 INFO: cpd05178 not found in model!\n",
      "1723051111.8500307 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051111.8902702 INFO: Expansion time:Pyr.NO2:0.8870988819999184\n",
      "1723051111.8927312 INFO: Filtered count:0 out of 1428\n",
      "1723051112.7672694 INFO: Expansion time:Pyr.NO3:0.8731133000001137\n",
      "1723051112.769691 INFO: Filtered count:0 out of 1428\n",
      "1723051113.2281563 INFO: metabolites 987\n",
      "1723051113.6399148 INFO: Expansion time:Pyr.O2:0.8686940719999257\n",
      "1723051113.6423664 INFO: Filtered count:0 out of 1428\n",
      "1723051113.711785 INFO: reactions 861\n",
      "1723051113.859036 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051114.41232 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/297/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051114.868984 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051114.9746962 INFO: Expansion time:Etho.O2:3.962998067000626\n",
      "1723051114.9773169 INFO: Filtered count:0 out of 1296\n",
      "1723051116.0126472 INFO: Expansion time:Fum.O2:1.0337002969999958\n",
      "1723051116.015236 INFO: Filtered count:0 out of 1296\n",
      "1723051116.867819 INFO: Expansion time:Pyr.SO3:3.2240023029999065\n",
      "1723051116.8699858 INFO: Filtered count:0 out of 1428\n",
      "1723051117.0023499 INFO: Expansion time:empty:0.9855722069996773\n",
      "1723051117.0046575 INFO: Filtered count:0 out of 1296\n",
      "1723051117.2396262 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051117.468354 INFO: Expansion time:Pyr.SO4:0.5970866389998264\n",
      "1723051117.470499 INFO: Filtered count:0 out of 1428\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051118.0696764 INFO: Expansion time:Pyr.TMAO:0.5979605359998459\n",
      "1723051118.071823 INFO: Filtered count:0 out of 1428\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[246 'R2_C_D1.37.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:56:04+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9ccff7baf0bb5deb0add52b17acd6898'\n",
      " 10221301 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051118.6726444 INFO: Expansion time:empty:0.5996533919997091\n",
      "1723051118.6748722 INFO: Filtered count:0 out of 1428\n",
      "1723051119.0001903 INFO: metabolites 1085\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[230 'R2A_C_D2.51.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:53:02+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '70b2cef82e620c7228c098770e95ee7a'\n",
      " 12258411 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051119.53033 INFO: reactions 1015\n",
      "1723051119.6858025 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051119.9492536 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051119.9821022 INFO: metabolites 1094\n",
      "1723051120.191091 INFO: cpd05178 not found in model!\n",
      "1723051120.1935925 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051120.2668028 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/47/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051120.56307 INFO: reactions 1041\n",
      "1723051120.7371907 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051121.3291097 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/519/1\n",
      "[203 'R2_restored_DShore.13.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:47:51+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '906b7ba604417d5e0e9a644b15ae2519'\n",
      " 15583089 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051122.1841798 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051122.2653968 INFO: metabolites 1143\n",
      "1723051122.6848793 INFO: reactions 1134\n",
      "1723051122.8326178 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051123.2596219 INFO: Default biomass: [bio1]\n",
      "1723051123.2981682 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/330/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051123.6054008 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051123.9093285 INFO: cpd05178 not found in model!\n",
      "1723051123.9114425 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051126.2082746 INFO: cpd05178 not found in model!\n",
      "1723051126.210824 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051127.6732008 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051127.9344475 INFO: cpd05178 not found in model!\n",
      "1723051127.9363291 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051128.186343 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051128.3450067 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051129.0788374 INFO: cpd05178 not found in model!\n",
      "1723051129.0809972 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051129.2919438 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.27.pg.G.D GF: 11665 Prob: 3480 Other: 3207\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051130.4528425 INFO: cpd05178 not found in model!\n",
      "1723051130.4552007 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051131.1126323 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[286 'R2_restored_DShore.037.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:04:08+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'feb9d6328d4bfe7d8e125601feb80c0c'\n",
      " 19744662 None]\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051131.6847506 INFO: metabolites 1168\n",
      "1723051132.1807652 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051132.3624728 INFO: reactions 1131\n",
      "1723051132.5053976 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051132.5501318 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051133.1370375 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051133.2706425 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/253/1\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051135.6847 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051137.1686015 INFO: Expansion time:Ac.O2:5.49166832300034\n",
      "1723051137.171098 INFO: Filtered count:0 out of 1643\n",
      "1723051138.2999065 INFO: cpd05178 not found in model!\n",
      "1723051138.3020546 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051138.3967175 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051139.2549589 INFO: Expansion time:Etho.O2:2.0822669610006415\n",
      "1723051139.257666 INFO: Filtered count:0 out of 1643\n",
      "1723051139.320573 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051140.4479063 INFO: cpd05178 not found in model!\n",
      "1723051140.484698 INFO: cpd05178 not found in model!\n",
      "1723051140.4873521 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051140.6901557 INFO: cpd05178 not found in model!\n",
      "1723051140.692158 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.12.pg.G.D GF: 12255 Prob: 2828 Other: 3277\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051140.9985585 INFO: Expansion time:For.O2:1.7393680420000237\n",
      "1723051141.0011559 INFO: Filtered count:0 out of 1643\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051142.323088 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051142.9795237 INFO: cpd05178 not found in model!\n",
      "1723051142.9814632 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051142.9824069 INFO: cpd05178 not found in model!\n",
      "1723051142.984518 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051143.2402403 INFO: Expansion time:Fum:2.2375656960002743\n",
      "1723051143.2426846 INFO: Filtered count:0 out of 1643\n",
      "1723051143.9890106 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[152 'R2A_A_H2O.78.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:37:34+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'cabae3845051c29e40cdad0aff38a11b'\n",
      " 6695084 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051144.9626315 INFO: cpd05178 not found in model!\n",
      "1723051144.9648316 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051145.1513 INFO: metabolites 943\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.51.pg.G.D GF: 11836 Prob: 3464 Other: 3054\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051145.497349 INFO: Expansion time:Fum.O2:2.2531402140002683\n",
      "1723051145.4996438 INFO: Filtered count:0 out of 1643\n",
      "1723051145.5183022 INFO: cpd05178 not found in model!\n",
      "1723051145.520787 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051145.6652231 INFO: reactions 878\n",
      "1723051145.8158948 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D1.37.pg.G.D GF: 12134 Prob: 2970 Other: 3248\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051146.3449147 INFO: Default biomass: [bio1]\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051147.1861098 INFO: cpd05178 not found in model!\n",
      "1723051147.1884582 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051148.0145106 INFO: cpd05178 not found in model!\n",
      "1723051148.0166223 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.13.pg.G.D GF: 11542 Prob: 3666 Other: 3144\n",
      "188040/22/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051148.7054348 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051150.6151576 INFO: cpd05178 not found in model!\n",
      "1723051150.617811 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051150.6513603 INFO: cpd05178 not found in model!\n",
      "1723051150.6540167 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051152.9231746 INFO: cpd05178 not found in model!\n",
      "1723051152.925835 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051152.9679644 INFO: cpd05178 not found in model!\n",
      "1723051152.9701016 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00062_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051154.9996498 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051155.4479408 INFO: Expansion time:Pyr:9.946742570000424\n",
      "1723051155.4500465 INFO: Filtered count:1 out of 1643\n",
      "1723051155.5788121 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051155.603088 INFO: Expansion time:Fum:3.066984910999963\n",
      "1723051155.6050577 INFO: Filtered count:0 out of 1475\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051156.1434002 INFO: Expansion time:Pyr.DMSO:0.6920558579995486\n",
      "1723051156.14531 INFO: Filtered count:1 out of 1643\n",
      "1723051156.2467654 INFO: Expansion time:Fum.O2:0.6405704200005857\n",
      "1723051156.2489073 INFO: Filtered count:0 out of 1475\n",
      "1723051156.266312 INFO: cpd05178 not found in model!\n",
      "1723051156.2686114 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051156.3612435 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051156.8996859 INFO: Expansion time:Glc:0.6495414759992855\n",
      "1723051156.9017577 INFO: Filtered count:0 out of 1475\n",
      "1723051157.5529869 INFO: Expansion time:Glc.DMSO:0.6499288920003892\n",
      "1723051157.5551503 INFO: Filtered count:0 out of 1475\n",
      "1723051157.9801705 INFO: Expansion time:Pyr.NO:1.8336168890000408\n",
      "1723051157.9822676 INFO: Filtered count:1 out of 1643\n",
      "1723051158.0255575 INFO: Objective with gapfill database:102.43563016303362; min objective:0.01\n",
      "1723051158.2063985 INFO: Expansion time:Glc.O2:0.649905091000619\n",
      "1723051158.20844 INFO: Filtered count:0 out of 1475\n",
      "1723051158.7493615 INFO: Expansion time:Pyr.NO2:0.7657471200000145\n",
      "1723051158.7514522 INFO: Filtered count:1 out of 1643\n",
      "1723051158.8261528 INFO: Objective with gapfill database:146.59365564999234; min objective:0.01\n",
      "1723051159.0560932 INFO: Expansion time:Glc.TMAO:0.846381491000102\n",
      "1723051159.0585628 INFO: Filtered count:0 out of 1475\n",
      "1723051159.5265996 INFO: Expansion time:Pyr.NO3:0.771970952000629\n",
      "1723051159.5286236 INFO: Filtered count:1 out of 1643\n",
      "1723051159.7334054 INFO: cpd05178 not found in model!\n",
      "1723051159.7358425 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051159.8623254 INFO: cpd05178 not found in model!\n",
      "1723051159.864708 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051159.9882245 INFO: Expansion time:Pyr:0.9281845460000113\n",
      "1723051159.9907143 INFO: Filtered count:0 out of 1475\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.037.pg.G.D GF: 11634 Prob: 3558 Other: 3160\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051160.2459888 INFO: Expansion time:Pyr.O2:0.7161629010006436\n",
      "1723051160.2481825 INFO: Filtered count:1 out of 1643\n",
      "1723051160.9189098 INFO: Expansion time:Pyr.DMSO:0.9267165129995192\n",
      "1723051160.9214199 INFO: Filtered count:0 out of 1475\n",
      "1723051160.968291 INFO: Expansion time:Pyr.SO3:0.7188898429994879\n",
      "1723051160.970498 INFO: Filtered count:1 out of 1643\n",
      "1723051161.3570929 INFO: cpd05178 not found in model!\n",
      "1723051161.6585937 INFO: Expansion time:Pyr.SO4:0.6867291830003524\n",
      "1723051161.6608047 INFO: Filtered count:1 out of 1643\n",
      "1723051161.698693 INFO: cpd05178 not found in model!\n",
      "1723051161.7015555 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051161.8496883 INFO: Expansion time:Pyr.NO:0.926787977999993\n",
      "1723051161.8522 INFO: Filtered count:0 out of 1475\n",
      "1723051162.3424637 INFO: Expansion time:Pyr.TMAO:0.6804530090003027\n",
      "1723051162.3446045 INFO: Filtered count:1 out of 1643\n",
      "1723051164.152806 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051164.1768188 INFO: Objective with gapfill database:257.18483702703816; min objective:0.01\n",
      "1723051164.2017655 INFO: Expansion time:empty:1.855931669999336\n",
      "1723051164.2038481 INFO: Filtered count:1 out of 1643\n",
      "1723051164.5470648 INFO: Expansion time:Pyr.NO2:2.693457020999631\n",
      "1723051164.5495055 INFO: Filtered count:0 out of 1475\n",
      "1723051164.9256148 INFO: cpd05178 not found in model!\n",
      "1723051164.9281616 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051165.335203 INFO: cpd05178 not found in model!\n",
      "1723051165.3373556 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051165.4927328 INFO: Expansion time:Pyr.NO3:0.9417797220003195\n",
      "1723051165.4952245 INFO: Filtered count:0 out of 1475\n",
      "1723051166.4738963 INFO: Expansion time:Pyr.O2:0.9771862149991648\n",
      "1723051166.4764676 INFO: Filtered count:0 out of 1475\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[250 'R1_C_H2O.79.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:57:04+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '24c77238354657c679d3e44c769a0afd'\n",
      " 10083304 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051167.5078366 INFO: Expansion time:Pyr.SO3:1.0298780339999212\n",
      "1723051167.510381 INFO: Filtered count:0 out of 1475\n",
      "1723051167.8834608 INFO: metabolites 1097\n",
      "1723051168.5577831 INFO: Expansion time:Pyr.SO4:1.045937199999571\n",
      "1723051168.560455 INFO: Filtered count:0 out of 1475\n",
      "1723051168.5756896 INFO: reactions 1077\n",
      "1723051168.786665 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051169.0465114 INFO: Expansion time:Ac.NO2:3.6868160590001935\n",
      "1723051169.0490906 INFO: Filtered count:0 out of 1629\n",
      "1723051169.4490197 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/64/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051169.5823674 INFO: Expansion time:Pyr.TMAO:1.0203503500006263\n",
      "1723051169.5848305 INFO: Filtered count:0 out of 1475\n",
      "1723051170.4500322 INFO: Expansion time:Ac.O2:1.399321597999915\n",
      "1723051170.452547 INFO: Filtered count:0 out of 1629\n",
      "1723051170.5905 INFO: Expansion time:empty:1.0042695850006567\n",
      "1723051170.5929093 INFO: Filtered count:0 out of 1475\n",
      "1723051171.996597 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051173.684372 INFO: Expansion time:Etho.O2:3.2303131550006583\n",
      "1723051173.6866846 INFO: Filtered count:0 out of 1629\n",
      "1723051174.6428244 INFO: cpd05178 not found in model!\n",
      "1723051174.6540964 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_H2O.78.pg.G.D GF: 12076 Prob: 3204 Other: 3074\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051175.168936 INFO: Expansion time:Glc:1.480990682999618\n",
      "1723051175.171088 INFO: Filtered count:0 out of 1629\n",
      "1723051176.2792447 INFO: Expansion time:Glc.DMSO:1.1069486530004724\n",
      "1723051176.2814312 INFO: Filtered count:0 out of 1629\n",
      "1723051177.5004282 INFO: Expansion time:Glc.O2:1.2176963840001918\n",
      "1723051177.5025194 INFO: Filtered count:0 out of 1629\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051179.0512102 INFO: cpd05178 not found in model!\n",
      "1723051179.0532331 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051179.7226067 INFO: Expansion time:Glc.TMAO:2.218841504000011\n",
      "1723051179.725061 INFO: Filtered count:0 out of 1629\n",
      "1723051180.6703863 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051181.0100338 INFO: Expansion time:Pyr:1.2834230930002377\n",
      "1723051181.012505 INFO: Filtered count:0 out of 1629\n",
      "1723051181.2063472 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051182.1820738 INFO: Expansion time:Pyr.DMSO:1.1679901679999602\n",
      "1723051182.1844368 INFO: Filtered count:0 out of 1629\n",
      "1723051183.5008013 INFO: Expansion time:Pyr.NO:1.3150290169996879\n",
      "1723051183.50305 INFO: Filtered count:0 out of 1629\n",
      "1723051183.7824643 INFO: cpd05178 not found in model!\n",
      "1723051183.7853043 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051185.0740535 INFO: Expansion time:Pyr.NO2:1.569581157000357\n",
      "1723051185.0763881 INFO: Filtered count:0 out of 1629\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051186.359755 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051186.6495159 INFO: Expansion time:Pyr.NO3:1.5716125090002606\n",
      "1723051186.6519794 INFO: Filtered count:0 out of 1629\n",
      "1723051186.8660798 INFO: Expansion time:Ac.O2:0.9615458050002417\n",
      "1723051186.8685303 INFO: Filtered count:0 out of 1653\n",
      "1723051187.9299774 INFO: Expansion time:Dlac.O2:1.0599501769993367\n",
      "1723051187.9323635 INFO: Filtered count:0 out of 1653\n",
      "1723051188.39703 INFO: cpd05178 not found in model!\n",
      "1723051188.3993049 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051190.4967377 INFO: Expansion time:Pyr.O2:3.84325567399992\n",
      "1723051190.498973 INFO: Filtered count:0 out of 1629\n",
      "1723051190.6052752 INFO: Expansion time:Etho.O2:2.6713241970001036\n",
      "1723051190.607741 INFO: Filtered count:0 out of 1653\n",
      "1723051191.0110304 INFO: cpd05178 not found in model!\n",
      "1723051191.0134664 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051191.618051 INFO: Expansion time:For.O2:1.0088267770006496\n",
      "1723051191.6203744 INFO: Filtered count:0 out of 1653\n",
      "1723051192.4197664 INFO: Expansion time:Pyr.SO3:1.9193858549997458\n",
      "1723051192.4220254 INFO: Filtered count:0 out of 1629\n",
      "1723051192.640907 INFO: Expansion time:Fum:1.0190822229997138\n",
      "1723051192.6432567 INFO: Filtered count:0 out of 1653\n",
      "1723051193.1508427 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[303 'R1_B_H2O.19.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:07:45+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '7cc8008d8cc293bd0255df2962cc64ed'\n",
      " 5274953 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051193.6436405 INFO: Expansion time:Pyr.SO4:1.2201693059996614\n",
      "1723051193.6459398 INFO: Filtered count:0 out of 1629\n",
      "1723051193.6461394 INFO: Expansion time:Fum.O2:1.0015359299995907\n",
      "1723051193.6479013 INFO: Filtered count:0 out of 1653\n",
      "1723051193.7230287 INFO: metabolites 923\n",
      "1723051194.1955018 INFO: reactions 872\n",
      "1723051194.3312497 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051194.5561173 INFO: Expansion time:Pyr.TMAO:0.9087453590000223\n",
      "1723051194.558044 INFO: Filtered count:0 out of 1629\n",
      "1723051194.71516 INFO: Expansion time:Glc:1.0658414619992982\n",
      "1723051194.717364 INFO: Filtered count:0 out of 1653\n",
      "1723051194.828712 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/383/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051195.213201 INFO: Expansion time:empty:0.6539561360004882\n",
      "1723051195.2152412 INFO: Filtered count:0 out of 1629\n",
      "1723051195.7874484 INFO: Expansion time:Glc.DMSO:1.0685339989995555\n",
      "1723051195.7897038 INFO: Filtered count:0 out of 1653\n",
      "1723051196.3008533 INFO: cpd05178 not found in model!\n",
      "1723051196.3029249 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051196.8295426 INFO: Expansion time:Glc.O2:1.0384547419998853\n",
      "1723051196.8318706 INFO: Filtered count:0 out of 1653\n",
      "1723051198.247823 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051198.5779276 INFO: cpd05178 not found in model!\n",
      "1723051198.5805876 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051198.6876502 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_H2O.79.pg.G.D GF: 11421 Prob: 4012 Other: 2921\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051199.5157568 INFO: Expansion time:Glc.TMAO:2.6824151029995846\n",
      "1723051199.5179799 INFO: Filtered count:0 out of 1653\n",
      "1723051200.5324552 INFO: Expansion time:Pyr:1.0130076710001958\n",
      "1723051200.5348299 INFO: Filtered count:0 out of 1653\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051201.4660106 INFO: cpd05178 not found in model!\n",
      "1723051201.468742 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051201.581768 INFO: Expansion time:Pyr.DMSO:1.0454525260001901\n",
      "1723051201.584299 INFO: Filtered count:0 out of 1653\n",
      "1723051202.669679 INFO: Expansion time:Pyr.NO:1.0838889719998406\n",
      "1723051202.672299 INFO: Filtered count:0 out of 1653\n",
      "1723051203.274963 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051203.7127619 INFO: Expansion time:Pyr.NO2:1.0389839549998214\n",
      "1723051203.7151928 INFO: Filtered count:0 out of 1653\n",
      "1723051204.7543578 INFO: Expansion time:Pyr.NO3:1.037697871999626\n",
      "1723051204.7568521 INFO: Filtered count:0 out of 1653\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051205.7432637 INFO: Expansion time:Pyr.O2:0.9848784929999965\n",
      "1723051205.7457283 INFO: Filtered count:0 out of 1653\n",
      "1723051206.607541 INFO: cpd05178 not found in model!\n",
      "1723051207.0031042 INFO: cpd05178 not found in model!\n",
      "1723051207.00577 INFO: Media compound: cpd05178 not found in model.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051207.8128235 INFO: Expansion time:Pyr.SO3:2.065615835999779\n",
      "1723051207.8148727 INFO: Filtered count:0 out of 1653\n",
      "1723051208.535415 INFO: Expansion time:Ac.O2:0.924268437000137\n",
      "1723051208.5379457 INFO: Filtered count:0 out of 1211\n",
      "1723051208.8923912 INFO: Expansion time:Pyr.SO4:1.0759216000005836\n",
      "1723051208.894957 INFO: Filtered count:0 out of 1653\n",
      "1723051209.4190226 INFO: Expansion time:For.NO2:0.8794372329994076\n",
      "1723051209.4214716 INFO: Filtered count:0 out of 1211\n",
      "1723051209.817788 INFO: cpd05178 not found in model!\n",
      "1723051209.820382 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051209.9106884 INFO: Expansion time:Pyr.TMAO:1.0141857770004208\n",
      "1723051209.9130187 INFO: Filtered count:0 out of 1653\n",
      "1723051210.2979739 INFO: Expansion time:Fum:0.8749173039996094\n",
      "1723051210.3005228 INFO: Filtered count:0 out of 1211\n",
      "1723051210.9001403 INFO: Expansion time:empty:0.9855463240000972\n",
      "1723051210.9025497 INFO: Filtered count:0 out of 1653\n",
      "1723051211.0106165 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051211.1666365 INFO: Expansion time:Fum.O2:0.8646929009992164\n",
      "1723051211.16912 INFO: Filtered count:0 out of 1211\n",
      "1723051211.46832 INFO: cpd05178 not found in model!\n",
      "1723051211.4706347 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_H2O.19.pg.G.D GF: 12211 Prob: 2964 Other: 3179\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051212.0633757 INFO: Expansion time:Glc:0.8927444410001044\n",
      "1723051212.0658443 INFO: Filtered count:0 out of 1211\n",
      "1723051212.9603345 INFO: Expansion time:Glc.DMSO:0.8929324549999365\n",
      "1723051212.9629145 INFO: Filtered count:0 out of 1211\n",
      "1723051213.84985 INFO: Expansion time:Glc.O2:0.8854724510001688\n",
      "1723051213.852356 INFO: Filtered count:0 out of 1211\n",
      "1723051214.7195203 INFO: Expansion time:Glc.TMAO:0.8656756119999045\n",
      "1723051214.7219656 INFO: Filtered count:0 out of 1211\n",
      "1723051215.5885143 INFO: Expansion time:Pyr:0.8650716950005517\n",
      "1723051215.5909402 INFO: Filtered count:0 out of 1211\n",
      "1723051216.4615858 INFO: Expansion time:Pyr.DMSO:0.86915426700034\n",
      "1723051216.4640687 INFO: Filtered count:0 out of 1211\n",
      "1723051216.9109607 INFO: cpd05178 not found in model!\n",
      "1723051217.2957873 INFO: cpd05178 not found in model!\n",
      "1723051217.2984278 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051217.3791173 INFO: Expansion time:Pyr.NO:0.9135520769996219\n",
      "1723051217.3815722 INFO: Filtered count:0 out of 1211\n",
      "1723051220.284451 INFO: cpd05178 not found in model!\n",
      "1723051220.6307642 INFO: cpd05178 not found in model!\n",
      "1723051220.6332424 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051220.7306178 INFO: Expansion time:Pyr.NO3:3.3476184979999744\n",
      "1723051220.733193 INFO: Filtered count:0 out of 1211\n",
      "1723051221.6390638 INFO: Expansion time:Pyr.O2:0.9042729809998491\n",
      "1723051221.6415756 INFO: Filtered count:0 out of 1211\n",
      "1723051222.3732505 INFO: Objective with gapfill database:186.58845695980503; min objective:0.01\n",
      "1723051222.5424743 INFO: Expansion time:Pyr.SO3:0.8993741840004077\n",
      "1723051222.5449705 INFO: Filtered count:0 out of 1211\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051223.4434164 INFO: Expansion time:Pyr.SO4:0.896811462000187\n",
      "1723051223.445847 INFO: Filtered count:0 out of 1211\n",
      "1723051224.349107 INFO: Expansion time:Pyr.TMAO:0.9017668110000159\n",
      "1723051224.351643 INFO: Filtered count:0 out of 1211\n",
      "1723051225.2418742 INFO: Expansion time:empty:0.8887972850006918\n",
      "1723051225.244436 INFO: Filtered count:0 out of 1211\n",
      "1723051229.2953863 INFO: cpd05178 not found in model!\n",
      "1723051229.2974925 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051229.6268508 INFO: Expansion time:Etho.O2:6.419766005999918\n",
      "1723051229.6294923 INFO: Filtered count:0 out of 1538\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[179 'R1_A_H2O.8.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:42:45+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '826f73f5580662505952b958a49ae0b9'\n",
      " 8611699 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051231.4851294 INFO: metabolites 880\n",
      "1723051231.5330667 INFO: Expansion time:Fum:1.9020033840006363\n",
      "1723051231.5357063 INFO: Filtered count:0 out of 1538\n",
      "1723051231.6375368 INFO: cpd05178 not found in model!\n",
      "1723051231.6400578 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051232.0066078 INFO: reactions 798\n",
      "1723051232.1470716 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051232.7357962 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/10/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051233.4658227 INFO: Expansion time:Fum.O2:1.928311055999984\n",
      "1723051233.4684649 INFO: Filtered count:0 out of 1538\n",
      "1723051234.0592144 INFO: cpd05178 not found in model!\n",
      "1723051234.0618505 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051235.131581 INFO: Expansion time:Pyr:1.6616003310000451\n",
      "1723051235.1340094 INFO: Filtered count:0 out of 1538\n",
      "1723051235.2464252 INFO: cpd05178 not found in model!\n",
      "1723051235.635127 INFO: cpd05178 not found in model!\n",
      "1723051235.637849 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051237.5898402 INFO: cpd05178 not found in model!\n",
      "1723051237.8215435 INFO: cpd05178 not found in model!\n",
      "1723051237.8235137 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051238.332453 INFO: Expansion time:Pyr.DMSO:3.1970714290000615\n",
      "1723051238.3350372 INFO: Filtered count:0 out of 1538\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051239.2962053 INFO: cpd05178 not found in model!\n",
      "1723051239.482143 INFO: Expansion time:Pyr.NO:1.1455894529999568\n",
      "1723051239.4847085 INFO: Filtered count:0 out of 1538\n",
      "1723051239.6549456 INFO: cpd05178 not found in model!\n",
      "1723051239.6576207 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051241.8461285 INFO: Expansion time:Pyr.NO2:2.3599242749996847\n",
      "1723051241.8487115 INFO: Filtered count:0 out of 1538\n",
      "1723051243.7538862 INFO: cpd05178 not found in model!\n",
      "1723051243.7560248 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051243.769856 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051243.8917959 INFO: cpd05178 not found in model!\n",
      "1723051243.8936443 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051244.2223988 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[196 'SF2_C_H2O.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:46:16+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'dd878133de0a54bf30a660f92ace40a9'\n",
      " 9033612 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051245.338427 INFO: metabolites 995\n",
      "1723051245.4287868 INFO: Expansion time:Pyr.NO3:3.5784351759994024\n",
      "1723051245.431333 INFO: Filtered count:0 out of 1538\n",
      "1723051245.7154868 INFO: reactions 918\n",
      "1723051245.85528 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051246.2195463 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/293/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051246.5491796 INFO: cpd05178 not found in model!\n",
      "1723051246.5513847 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051247.3800259 INFO: Expansion time:Pyr.O2:1.9470618450004622\n",
      "1723051247.3826337 INFO: Filtered count:0 out of 1538\n",
      "1723051247.7610583 INFO: cpd05178 not found in model!\n",
      "1723051247.763359 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051249.034535 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051249.7243383 INFO: Expansion time:Pyr.SO3:2.340202992999366\n",
      "1723051249.7267044 INFO: Filtered count:0 out of 1538\n",
      "1723051250.9145124 INFO: Expansion time:Pyr.SO4:1.1863367089999883\n",
      "1723051250.9170141 INFO: Filtered count:0 out of 1538\n",
      "1723051252.0652113 INFO: Expansion time:Pyr.TMAO:1.1466747370004668\n",
      "1723051252.067833 INFO: Filtered count:0 out of 1538\n",
      "1723051252.7606294 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051253.0909317 INFO: Expansion time:empty:1.0215567880004528\n",
      "1723051253.093749 INFO: Filtered count:0 out of 1538\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[243 'R2_B_D2.55.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:55:33+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9d022001cf202708d8d77a7af124cc6c'\n",
      " 10704534 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051255.1799583 INFO: metabolites 1181\n",
      "1723051255.540101 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051255.6881998 INFO: reactions 1176\n",
      "1723051255.9130306 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051256.3829167 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/58/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051258.568455 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051258.5918303 INFO: cpd05178 not found in model!\n",
      "1723051258.5946207 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_H2O.8.pg.G.D GF: 12307 Prob: 2846 Other: 3207\n",
      "[278 'R2_B_D2.62.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:02:32+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '766e41a151c2a7b9e5fabb6ef5bd348b'\n",
      " 9677866 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051260.115499 INFO: metabolites 1024\n",
      "1723051260.4356894 INFO: reactions 927\n",
      "1723051260.53945 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051260.8876884 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/194/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051262.3945048 INFO: cpd05178 not found in model!\n",
      "1723051262.396664 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051263.245626 INFO: cpd05178 not found in model!\n",
      "1723051263.591861 INFO: cpd05178 not found in model!\n",
      "1723051263.5943725 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051263.7682748 INFO: cpd05178 not found in model!\n",
      "1723051264.0038047 INFO: cpd05178 not found in model!\n",
      "1723051264.005905 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051265.2551239 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051265.8274672 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051266.419647 INFO: cpd05178 not found in model!\n",
      "1723051266.4217155 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051266.622236 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051267.4327452 INFO: cpd05178 not found in model!\n",
      "1723051267.4349458 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051267.4922543 INFO: cpd05178 not found in model!\n",
      "1723051267.494365 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_H2O.16.pg.G.D GF: 12123 Prob: 3062 Other: 3169\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051268.1111193 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[262 'R2_A_D1.52.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:59:32+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '81d4173551a66dc4daec41452d0ebb30'\n",
      " 9455232 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051270.4149985 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051270.762826 INFO: metabolites 947\n",
      "1723051271.1805673 INFO: reactions 851\n",
      "1723051271.3147633 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051271.4910483 INFO: cpd05178 not found in model!\n",
      "1723051271.493701 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051271.7463214 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051271.8215585 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/102/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051275.6034415 INFO: Objective with gapfill database:161.36200790983685; min objective:0.01\n",
      "1723051278.1284213 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051278.4462469 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051278.8968043 INFO: cpd05178 not found in model!\n",
      "1723051278.8991807 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.55.pg.G.D GF: 11145 Prob: 4318 Other: 2889\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051279.4994504 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[223 'R2_restored_H2O.44.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:51:38+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '88145c168160a6bb1debf981a140dca8'\n",
      " 8103760 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051280.9496906 INFO: metabolites 1045\n",
      "1723051281.3742754 INFO: cpd05178 not found in model!\n",
      "1723051281.5065079 INFO: reactions 956\n",
      "1723051281.5967236 INFO: cpd05178 not found in model!\n",
      "1723051281.5992658 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051281.6561768 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051282.1723988 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/482/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051282.516224 INFO: cpd05178 not found in model!\n",
      "1723051282.5184932 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051282.5582383 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.62.pg.G.D GF: 12389 Prob: 2650 Other: 3315\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051282.7842166 INFO: cpd05178 not found in model!\n",
      "1723051282.7863188 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051284.3901074 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051289.0008006 INFO: cpd05178 not found in model!\n",
      "1723051289.247723 INFO: cpd05178 not found in model!\n",
      "1723051289.24992 INFO: Media compound: cpd05178 not found in model.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051290.3581734 INFO: cpd05178 not found in model!\n",
      "1723051290.4226055 INFO: cpd05178 not found in model!\n",
      "1723051290.4245844 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051290.5959787 INFO: cpd05178 not found in model!\n",
      "1723051290.5983112 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051290.8883615 INFO: Expansion time:Etho.O2:0.61211645900039\n",
      "1723051290.8905518 INFO: Filtered count:0 out of 1264\n",
      "1723051291.089115 INFO: cpd05178 not found in model!\n",
      "1723051291.091343 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051291.4792428 INFO: Expansion time:Pyr:0.587571418999687\n",
      "1723051291.4814236 INFO: Filtered count:0 out of 1264\n",
      "1723051291.9063227 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051291.9666042 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051292.0798793 INFO: Expansion time:Pyr.DMSO:0.5971960039996702\n",
      "1723051292.0820029 INFO: Filtered count:0 out of 1264\n",
      "1723051292.678924 INFO: Expansion time:Pyr.NO:0.5956741499994678\n",
      "1723051292.6810815 INFO: Filtered count:0 out of 1264\n",
      "1723051293.2789016 INFO: Expansion time:Pyr.NO2:0.5965865690004648\n",
      "1723051293.2810562 INFO: Filtered count:0 out of 1264\n",
      "1723051294.397786 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051294.8159707 INFO: cpd05178 not found in model!\n",
      "1723051294.8188014 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051295.0032055 INFO: Objective with gapfill database:255.09049845310884; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.52.pg.G.D GF: 12475 Prob: 2486 Other: 3393\n",
      "[140 'R1_A_D2.36.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:35:16+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '28a59fcca6231cdcec4a30f0b3faa822'\n",
      " 8418983 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051295.6431231 INFO: cpd05178 not found in model!\n",
      "1723051295.6452086 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051295.650896 INFO: metabolites 1037\n",
      "1723051295.802495 INFO: Expansion time:Pyr.NO3:2.520151050000095\n",
      "1723051295.8047798 INFO: Filtered count:0 out of 1264\n",
      "1723051295.9938369 INFO: reactions 955\n",
      "1723051296.1027708 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051296.3970485 INFO: Expansion time:Pyr.O2:0.5909551079994344\n",
      "1723051296.399254 INFO: Filtered count:0 out of 1264\n",
      "1723051296.462261 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/587/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051296.868639 INFO: cpd05178 not found in model!\n",
      "1723051296.8709524 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051296.9888487 INFO: Expansion time:Pyr.SO3:0.5883791940004812\n",
      "1723051296.9909947 INFO: Filtered count:0 out of 1264\n",
      "1723051297.5805502 INFO: Expansion time:Pyr.SO4:0.5883478820005621\n",
      "1723051297.5826817 INFO: Filtered count:0 out of 1264\n",
      "1723051297.702351 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051298.1857722 INFO: Expansion time:Pyr.TMAO:0.6018807130003552\n",
      "1723051298.1880407 INFO: Filtered count:0 out of 1264\n",
      "1723051298.8005664 INFO: Expansion time:empty:0.6113702060001742\n",
      "1723051298.802776 INFO: Filtered count:0 out of 1264\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[199 'R2_B_H2O.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:46:53+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'f1cd34b9917659a510e4d9c583344b21'\n",
      " 9315754 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051300.6136081 INFO: metabolites 752\n",
      "1723051301.0208228 INFO: reactions 697\n",
      "1723051301.0525837 INFO: cpd05178 not found in model!\n",
      "1723051301.055181 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051301.1404068 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051301.5364003 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/258/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051305.0362444 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051306.2856572 INFO: cpd05178 not found in model!\n",
      "1723051306.288189 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051306.7235146 INFO: Expansion time:Ac.O2:1.8377729330004513\n",
      "1723051306.725981 INFO: Filtered count:0 out of 1695\n",
      "1723051307.3503115 INFO: cpd05178 not found in model!\n",
      "1723051307.3526673 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051307.762035 INFO: Expansion time:Etho.O2:1.0346660250006607\n",
      "1723051307.7644997 INFO: Filtered count:0 out of 1695\n",
      "1723051307.7993312 INFO: cpd05178 not found in model!\n",
      "1723051307.837421 INFO: cpd05178 not found in model!\n",
      "1723051308.024174 INFO: cpd05178 not found in model!\n",
      "1723051308.0263233 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051308.0419567 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051308.0766604 INFO: cpd05178 not found in model!\n",
      "1723051308.0788488 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051308.727208 INFO: Expansion time:For.O2:0.9613768290000735\n",
      "1723051308.7296693 INFO: Filtered count:0 out of 1695\n",
      "1723051309.0145197 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[270 'R2_A_D1.66.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:01:02+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ab5597993ffeefb0bdde1bbe9536de06'\n",
      " 7969358 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051309.464316 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051309.7488482 INFO: metabolites 902\n",
      "1723051309.8856025 INFO: Expansion time:Fum:1.1544703689996823\n",
      "1723051309.8881123 INFO: Filtered count:0 out of 1695\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051310.1311848 INFO: cpd05178 not found in model!\n",
      "1723051310.1337793 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051310.1385024 INFO: reactions 821\n",
      "1723051310.2652767 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_H2O.44.pg.G.D GF: 11859 Prob: 3476 Other: 3017\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051310.726771 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/129/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051310.782686 INFO: Expansion time:Fum.O2:0.8931718909998381\n",
      "1723051310.7848155 INFO: Filtered count:0 out of 1695\n",
      "1723051310.9792943 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051311.6285834 INFO: Expansion time:Pyr:0.8426132109998434\n",
      "1723051311.6307275 INFO: Filtered count:0 out of 1695\n",
      "1723051312.6389215 INFO: cpd05178 not found in model!\n",
      "1723051312.640903 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[72 'R2_C_D2.18.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:21:53+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '1f1446d947b42ef65f9393621f3ddb9c'\n",
      " 12152891 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051313.5191228 INFO: cpd05178 not found in model!\n",
      "1723051313.5212371 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[135 'SF2_A_D1.23.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:34:16+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '0de49a9af03953a9b8be01e2a10e1c48'\n",
      " 11250703 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051313.5966706 INFO: Objective with gapfill database:129.37378448499467; min objective:0.01\n",
      "1723051313.9486146 INFO: metabolites 1201\n",
      "1723051313.9706097 INFO: metabolites 1112\n",
      "1723051314.433059 INFO: Expansion time:Pyr.DMSO:2.8011386469997888\n",
      "1723051314.4355402 INFO: Filtered count:0 out of 1695\n",
      "1723051314.5108998 INFO: reactions 1039\n",
      "1723051314.677704 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051314.6830025 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051314.8181617 INFO: reactions 1198\n",
      "1723051315.0575101 INFO: cpd05178 not found in model!\n",
      "1723051315.0942147 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051315.2544942 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/548/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051315.446618 INFO: cpd05178 not found in model!\n",
      "1723051315.449239 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051315.454919 INFO: Expansion time:Pyr.NO:1.0178269640000508\n",
      "1723051315.457337 INFO: Filtered count:0 out of 1695\n",
      "1723051315.810629 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/134/1\n",
      "[255 'SF2_A_H2O.23.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:58:03+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9ce4c2158b67e6b48903ea70580837c6'\n",
      " 7983984 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051318.3893483 INFO: metabolites 1056\n",
      "1723051318.5382407 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051318.951429 INFO: Expansion time:Pyr.NO2:3.492679611999847\n",
      "1723051318.9540215 INFO: Filtered count:0 out of 1695\n",
      "1723051318.9608858 INFO: reactions 952\n",
      "1723051319.1300764 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051319.7138846 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/36/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051320.4411676 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051320.6944041 INFO: Expansion time:Pyr.NO3:1.738764792999973\n",
      "1723051320.6968403 INFO: Filtered count:0 out of 1695\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051321.9509196 INFO: Expansion time:Pyr.O2:1.2525428990002183\n",
      "1723051321.953466 INFO: Filtered count:0 out of 1695\n",
      "1723051322.2077417 INFO: Objective with gapfill database:111.53436949457844; min objective:0.01\n",
      "1723051322.442493 INFO: cpd05178 not found in model!\n",
      "1723051322.6725135 INFO: cpd05178 not found in model!\n",
      "1723051322.6747212 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051323.2314746 INFO: cpd05178 not found in model!\n",
      "1723051323.2341902 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.36.pg.G.D GF: 11825 Prob: 3354 Other: 3173\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051324.3231661 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051324.6886363 INFO: cpd05178 not found in model!\n",
      "1723051324.6910455 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051324.7911756 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051325.2864041 INFO: cpd05178 not found in model!\n",
      "1723051325.2884617 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051325.3983486 INFO: Expansion time:Pyr.SO3:3.4433976609998354\n",
      "1723051325.4008083 INFO: Filtered count:0 out of 1695\n",
      "1723051326.740948 INFO: cpd05178 not found in model!\n",
      "1723051326.7455504 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_H2O.16.pg.G.D GF: 12681 Prob: 2382"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051327.124076 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Other: 3297\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051327.13081 INFO: cpd05178 not found in model!\n",
      "1723051327.4584522 INFO: cpd05178 not found in model!\n",
      "1723051327.4610543 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051329.0412803 INFO: Expansion time:Pyr.SO4:3.639062635999835\n",
      "1723051329.043804 INFO: Filtered count:0 out of 1695\n",
      "1723051329.8804214 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051331.4660058 INFO: Expansion time:Pyr.TMAO:2.4207098880006015\n",
      "1723051331.4685965 INFO: Filtered count:0 out of 1695\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[155 'SF2_A_H2O.23b.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:38:05+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '74208b4acef93a675b7bfa68e229063e'\n",
      " 6676407 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051332.4984298 INFO: Expansion time:empty:1.0283761560003768\n",
      "1723051332.5009453 INFO: Filtered count:0 out of 1695\n",
      "1723051332.6678967 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051332.7133908 INFO: metabolites 1115\n",
      "1723051333.4160945 INFO: reactions 1041\n",
      "1723051333.6029565 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[295 'R2A_C_D2.36.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:06:13+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e48644d88f004196665747fc808a53f8'\n",
      " 9345248 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051333.7825274 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051334.0238554 INFO: metabolites 1047\n",
      "1723051334.2401388 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051334.2909563 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/442/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051334.3551075 INFO: reactions 949\n",
      "1723051334.5063214 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[51 'SF2_C_D2.26.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:17:36+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'f7376f42394f58a3388249770362bb64'\n",
      " 16457147 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051335.1018498 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/282/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051335.1506202 INFO: Objective with gapfill database:171.54028108152016; min objective:0.01\n",
      "1723051335.3988206 INFO: cpd05178 not found in model!\n",
      "1723051335.4033422 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051335.6261265 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.66.pg.G.D GF: 12475 Prob: 2442 Other: 3435\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051336.852847 INFO: metabolites 1177\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051337.4939559 INFO: reactions 1161\n",
      "1723051337.676996 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051338.3129652 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/597/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051338.6834552 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051339.1812615 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051339.6285896 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051339.8380678 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051341.5852346 INFO: cpd05178 not found in model!\n",
      "1723051341.5878665 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_D1.23.pg.G.D GF: 11909 Prob: 3292 Other: 3153\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051342.0561924 INFO: cpd05178 not found in model!\n",
      "1723051342.058791 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051342.058678 INFO: cpd05178 not found in model!\n",
      "1723051342.2026227 INFO: cpd05178 not found in model!\n",
      "1723051342.2047524 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051342.3761628 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051342.3993828 INFO: cpd05178 not found in model!\n",
      "1723051342.4019415 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051342.4756668 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051342.9178514 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051343.036745 INFO: cpd05178 not found in model!\n",
      "1723051343.0391214 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051344.9014082 INFO: cpd05178 not found in model!\n",
      "1723051344.9040015 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051345.3923793 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051347.959659 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051348.1321502 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051348.2755322 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051349.2581744 INFO: Objective with gapfill database:209.84549286927836; min objective:0.01\n",
      "1723051350.4607024 INFO: cpd05178 not found in model!\n",
      "1723051350.4626245 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051350.7233737 INFO: cpd05178 not found in model!\n",
      "1723051350.7255473 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[124 'SF2_C_D2.39.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:32:13+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'df9401324f51993ce4c022ae8e252165'\n",
      " 8848692 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051351.362492 INFO: cpd05178 not found in model!\n",
      "1723051351.3650942 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051351.3995507 INFO: metabolites 1043\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D2.18.pg.G.D GF: 10882 Prob: 4732 Other: 2738\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051352.0177174 INFO: reactions 1025\n",
      "1723051352.1212697 INFO: Expansion time:Etho.O2:3.295838619999813\n",
      "1723051352.1236012 INFO: Filtered count:0 out of 956\n",
      "1723051352.208636 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051352.4319403 INFO: cpd05178 not found in model!\n",
      "1723051352.4347045 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051352.6981528 INFO: Expansion time:empty:0.5733845839995411\n",
      "1723051352.700401 INFO: Filtered count:0 out of 956\n",
      "1723051352.7901773 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_H2O.23.pg.G.D GF: 11659 Prob: 3608 Other: 3087\n",
      "188040/466/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051353.6451778 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051354.1890962 INFO: cpd05178 not found in model!\n",
      "1723051354.5404842 INFO: cpd05178 not found in model!\n",
      "1723051354.5431406 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051354.6035516 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051356.6390405 INFO: cpd05178 not found in model!\n",
      "1723051356.8617475 INFO: cpd05178 not found in model!\n",
      "1723051356.8639164 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051358.4554486 INFO: cpd05178 not found in model!\n",
      "1723051358.4583004 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.36.pg.G.D GF: 12190 Prob: 2784 Other: 3378\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051360.482191 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051361.1371758 INFO: cpd05178 not found in model!\n",
      "1723051361.1397407 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051361.5143378 INFO: cpd05178 not found in model!\n",
      "1723051361.516555 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_H2O.23b.pg.G.D GF: 11801 Prob: 3440 Other: 3111\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051361.754299 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2.26.pg.G.D GF: 11585 Prob: 3698 Other: 3071\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051363.5502496 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051363.9667702 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051364.6377866 INFO: cpd05178 not found in model!\n",
      "1723051364.640085 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051366.491777 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051366.5180674 INFO: cpd05178 not found in model!\n",
      "1723051366.5201364 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051369.0875692 INFO: cpd05178 not found in model!\n",
      "1723051369.188478 INFO: Expansion time:For.O2:4.349059226999998\n",
      "1723051369.1907737 INFO: Filtered count:0 out of 1475\n",
      "1723051369.3172145 INFO: cpd05178 not found in model!\n",
      "1723051369.3193374 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051369.5300748 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051370.4751894 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[104 'R1_A_H2O.022.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:28:13+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'cc65042c67dce5ff567c25b49eb433de'\n",
      " 4866700 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051371.9513383 INFO: metabolites 948\n",
      "1723051372.350819 INFO: reactions 850\n",
      "1723051372.3624961 INFO: Objective with gapfill database:101.90596678017015; min objective:0.01\n",
      "1723051372.494595 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051372.9738028 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/329/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051374.1962852 INFO: cpd05178 not found in model!\n",
      "1723051374.4237425 INFO: cpd05178 not found in model!\n",
      "1723051374.4256425 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051375.6701312 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051375.6777294 INFO: cpd05178 not found in model!\n",
      "1723051375.6795366 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[63 'R2A_C_D2.69.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:20:12+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '083b7c3e2918256b768f847d9d04fb08'\n",
      " 11549113 None]\n",
      "SF2_C_D2.39.pg.G.D GF: 11556 Prob: 3878 Other: 2920\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051375.9754524 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051376.0945213 INFO: metabolites 1096\n",
      "1723051376.544684 INFO: reactions 1046\n",
      "1723051376.6736739 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00062_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051377.010424 INFO: Expansion time:Fum.O2:7.818225687000449\n",
      "1723051377.0123284 INFO: Filtered count:1 out of 1475\n",
      "1723051377.6760576 INFO: Expansion time:empty:0.6625182280004083\n",
      "1723051377.6780815 INFO: Filtered count:1 out of 1475\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051378.5020254 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/93/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051378.80769 INFO: Expansion time:Ac.O2:1.70296330400015\n",
      "1723051378.8102398 INFO: Filtered count:0 out of 1747\n",
      "1723051379.354243 INFO: Expansion time:Etho.O2:6.3476915289993485\n",
      "1723051379.3564577 INFO: Filtered count:0 out of 1311\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051379.8821151 INFO: Expansion time:Etho.O2:1.0703918609997345\n",
      "1723051379.8847828 INFO: Filtered count:0 out of 1747\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[116 'SF2_C_D1.13.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:30:29+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '652d2e12e39bb202ef7d96091135baf7'\n",
      " 8394287 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051380.3492744 INFO: Expansion time:Etho.O2:0.6742851910003083\n",
      "1723051380.3515134 INFO: Filtered count:0 out of 1649\n",
      "1723051380.6263652 INFO: metabolites 1016\n",
      "1723051380.7199585 INFO: Expansion time:For.NO2:1.3621229540003696\n",
      "1723051380.722193 INFO: Filtered count:0 out of 1311\n",
      "1723051380.880957 INFO: Expansion time:For.O2:0.9947517470000093\n",
      "1723051380.8837025 INFO: Filtered count:0 out of 1747\n",
      "1723051381.0227942 INFO: Expansion time:For.NO2:0.6700440059994435\n",
      "1723051381.0250187 INFO: Filtered count:0 out of 1649\n",
      "1723051381.0912807 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051381.2079291 INFO: reactions 925\n",
      "1723051381.3640804 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051381.9169428 INFO: Expansion time:Fum:1.0316483540000263\n",
      "1723051381.919112 INFO: Filtered count:0 out of 1747\n",
      "1723051381.9505982 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/427/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051382.021727 INFO: Expansion time:For.O2:0.9953127449998647\n",
      "1723051382.024061 INFO: Filtered count:0 out of 1649\n",
      "1723051382.1821816 INFO: Expansion time:For.O2:1.4586144639997656\n",
      "1723051382.1844618 INFO: Filtered count:0 out of 1311\n",
      "1723051382.8218288 INFO: Expansion time:Fum.O2:0.9015044370007672\n",
      "1723051382.82409 INFO: Filtered count:0 out of 1747\n",
      "1723051383.0041585 INFO: Expansion time:Fum:0.9786970350005504\n",
      "1723051383.006694 INFO: Filtered count:0 out of 1649\n",
      "1723051383.617193 INFO: cpd05178 not found in model!\n",
      "1723051383.6196105 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051383.6621373 INFO: Expansion time:Pyr:0.8369023020004533\n",
      "1723051383.6643748 INFO: Filtered count:0 out of 1747\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051384.0591664 INFO: Expansion time:Fum.O2:1.0509509710000202\n",
      "1723051384.0613997 INFO: Filtered count:0 out of 1649\n",
      "1723051384.3812902 INFO: Expansion time:Pyr.DMSO:0.7156384520003485\n",
      "1723051384.3836093 INFO: Filtered count:0 out of 1747\n",
      "1723051384.9235477 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051385.1676743 INFO: Expansion time:Glc:1.104644048999944\n",
      "1723051385.1702242 INFO: Filtered count:0 out of 1649\n",
      "1723051385.3529332 INFO: Expansion time:Fum:3.16704277000008\n",
      "1723051385.355485 INFO: Filtered count:0 out of 1311\n",
      "1723051386.2217636 INFO: Expansion time:Glc.DMSO:1.0500779130006777\n",
      "1723051386.2243836 INFO: Filtered count:0 out of 1649\n",
      "1723051386.7541041 INFO: cpd05178 not found in model!\n",
      "1723051386.7563574 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051387.156779 INFO: Expansion time:Pyr.NO:2.771900122000261\n",
      "1723051387.1589916 INFO: Filtered count:0 out of 1747\n",
      "1723051387.1740534 INFO: Expansion time:Glc.O2:0.948160595000445\n",
      "1723051387.1762972 INFO: Filtered count:0 out of 1649\n",
      "1723051387.4290287 INFO: cpd05178 not found in model!\n",
      "1723051387.4310472 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051387.7273145 INFO: Expansion time:Ac.O2:3.6102216510007565\n",
      "1723051387.7297542 INFO: Filtered count:0 out of 1475\n",
      "1723051387.8161533 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051387.8826883 INFO: Expansion time:Fum.O2:2.5256976460004807\n",
      "1723051387.8850017 INFO: Filtered count:0 out of 1311\n",
      "1723051388.1030705 INFO: Expansion time:Glc.TMAO:0.9252674460003618\n",
      "1723051388.105405 INFO: Filtered count:0 out of 1649\n",
      "1723051388.345397 INFO: Expansion time:Pyr.NO2:1.1851202669995473\n",
      "1723051388.3475845 INFO: Filtered count:0 out of 1747\n",
      "1723051389.612853 INFO: Expansion time:Pyr.NO3:1.2640180510006758\n",
      "1723051389.6150544 INFO: Filtered count:0 out of 1747\n",
      "1723051389.717706 INFO: Expansion time:Etho.O2:1.986433584000224\n",
      "1723051389.7202969 INFO: Filtered count:0 out of 1475\n",
      "1723051389.9927723 INFO: cpd05178 not found in model!\n",
      "1723051389.9948738 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051390.079749 INFO: cpd05178 not found in model!\n",
      "1723051390.081984 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_H2O.022.pg.G.D GF: 12409 Prob: 2634 Other: 3307\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051390.5102766 INFO: Expansion time:Pyr.O2:0.8939836239997021\n",
      "1723051390.5125043 INFO: Filtered count:0 out of 1747\n",
      "1723051390.8293433 INFO: Expansion time:For.O2:1.107425126000635\n",
      "1723051390.8318748 INFO: Filtered count:0 out of 1475\n",
      "1723051390.8662543 INFO: Expansion time:Glyc.O2:2.7593139579994386\n",
      "1723051390.8687093 INFO: Filtered count:0 out of 1649\n",
      "1723051391.3721852 INFO: Expansion time:Pyr:3.485700323999481\n",
      "1723051391.374673 INFO: Filtered count:0 out of 1311\n",
      "1723051391.398608 INFO: Expansion time:Pyr.SO3:0.8848509950003063\n",
      "1723051391.4007368 INFO: Filtered count:0 out of 1747\n",
      "1723051391.5299606 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051391.7869086 INFO: Expansion time:Pyr.NO2:0.9167379380005514\n",
      "1723051391.7893872 INFO: Filtered count:0 out of 1649\n",
      "1723051392.0495298 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[175 'R2A_A_D1.50.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:41:58+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '059d29db9c60225a33188a4cdcf45290'\n",
      " 6723396 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051392.5681517 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051392.692432 INFO: Expansion time:Pyr.SO4:1.2904365679996772\n",
      "1723051392.6948183 INFO: Filtered count:0 out of 1747\n",
      "1723051392.7167716 INFO: Expansion time:Pyr.O2:0.9252201640001658\n",
      "1723051392.719211 INFO: Filtered count:0 out of 1649\n",
      "1723051392.7388558 INFO: Expansion time:Pyr.DMSO:1.3626785979995475\n",
      "1723051392.7413747 INFO: Filtered count:0 out of 1311\n",
      "1723051392.9878337 INFO: metabolites 949\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051393.5237436 INFO: reactions 908\n",
      "1723051393.5786738 INFO: Expansion time:Fum:2.745378051999978\n",
      "1723051393.5806804 INFO: Filtered count:0 out of 1475\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051393.6555617 INFO: Expansion time:empty:0.9348145069998282\n",
      "1723051393.6581402 INFO: Filtered count:0 out of 1649\n",
      "1723051393.6725864 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051394.2036088 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/181/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051394.6749115 INFO: Expansion time:Pyr.NO:1.9318951900004322\n",
      "1723051394.6775017 INFO: Filtered count:0 out of 1311\n",
      "1723051395.8909342 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051396.4938385 INFO: Expansion time:Pyr.TMAO:3.797790418999284\n",
      "1723051396.4960911 INFO: Filtered count:0 out of 1747\n",
      "1723051396.5151842 INFO: Expansion time:Fum.O2:2.933323666999968\n",
      "1723051396.5173156 INFO: Filtered count:0 out of 1475\n",
      "1723051396.5637262 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051396.6061327 INFO: Expansion time:Pyr.NO2:1.9270031739997648\n",
      "1723051396.6085951 INFO: Filtered count:0 out of 1311\n",
      "1723051396.8055222 INFO: Expansion time:Ac.O2:3.0593570370001544\n",
      "1723051396.8080978 INFO: Filtered count:0 out of 1432\n",
      "1723051397.1426933 INFO: Expansion time:empty:0.6453692599998249\n",
      "1723051397.1449559 INFO: Filtered count:0 out of 1747\n",
      "1723051398.112541 INFO: Expansion time:Glc:1.5939809139999852\n",
      "1723051398.1146684 INFO: Filtered count:0 out of 1475\n",
      "1723051398.2229977 INFO: Expansion time:Etho.O2:1.413407566999922\n",
      "1723051398.2256336 INFO: Filtered count:0 out of 1432\n",
      "1723051398.8842118 INFO: Expansion time:Pyr.NO3:2.274078737000309\n",
      "1723051398.8867114 INFO: Filtered count:0 out of 1311\n",
      "1723051399.4081945 INFO: Expansion time:Glc.DMSO:1.2922944680003638\n",
      "1723051399.410608 INFO: Filtered count:0 out of 1475\n",
      "1723051399.5426 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051399.719181 INFO: Expansion time:Pyr:1.49197703200025\n",
      "1723051399.7217286 INFO: Filtered count:0 out of 1432\n",
      "1723051400.8110073 INFO: Expansion time:Pyr.DMSO:1.0877741930007687\n",
      "1723051400.8134875 INFO: Filtered count:0 out of 1432\n",
      "1723051400.8482232 INFO: Expansion time:Pyr.O2:1.9599450499999875\n",
      "1723051400.8504884 INFO: Filtered count:0 out of 1311\n",
      "1723051401.1751647 INFO: cpd05178 not found in model!\n",
      "1723051401.1773355 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051401.6098228 INFO: Expansion time:Glc.O2:2.1977287949994206\n",
      "1723051401.6124055 INFO: Filtered count:0 out of 1475\n",
      "1723051401.931684 INFO: Expansion time:Pyr.NO:1.1167540060005194\n",
      "1723051401.9341273 INFO: Filtered count:0 out of 1432\n",
      "1723051402.0268004 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051402.6209378 INFO: Expansion time:Pyr.SO3:1.7690550479992453\n",
      "1723051402.6235726 INFO: Filtered count:0 out of 1311\n",
      "1723051403.185496 INFO: Expansion time:Pyr.NO2:1.2497811220000585\n",
      "1723051403.1881025 INFO: Filtered count:0 out of 1432\n",
      "1723051404.1128309 INFO: cpd05178 not found in model!\n",
      "1723051404.115327 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051404.2566948 INFO: Expansion time:Glc.TMAO:2.6427507079997667\n",
      "1723051404.2592335 INFO: Filtered count:0 out of 1475\n",
      "1723051404.3261545 INFO: Expansion time:Pyr.NO3:1.1365395060001902\n",
      "1723051404.328675 INFO: Filtered count:0 out of 1432\n",
      "1723051404.4525445 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.69.pg.G.D GF: 11583 Prob: 3800 Other: 2971\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051404.5995195 INFO: Expansion time:Pyr.SO4:1.974425465999957\n",
      "1723051404.6020234 INFO: Filtered count:0 out of 1311\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051405.39249 INFO: Expansion time:Pyr:1.1317476210006134\n",
      "1723051405.3945968 INFO: Filtered count:0 out of 1475\n",
      "1723051406.0522199 INFO: Expansion time:Pyr.DMSO:0.6564503050003623\n",
      "1723051406.0544147 INFO: Filtered count:0 out of 1475\n",
      "1723051406.2366707 INFO: Expansion time:Pyr.TMAO:1.633158886000274\n",
      "1723051406.2390702 INFO: Filtered count:0 out of 1311\n",
      "1723051406.706677 INFO: Expansion time:Pyr.NO:0.6511321440002575\n",
      "1723051406.7089038 INFO: Filtered count:0 out of 1475\n",
      "1723051407.1035287 INFO: cpd05178 not found in model!\n",
      "1723051407.1058168 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051407.1448207 INFO: Expansion time:empty:0.9043452100004288\n",
      "1723051407.147151 INFO: Filtered count:0 out of 1311\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[6 'R2_restored_DShore.46.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:08:56+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9a3594112c7f08ed5237d985e9cb3220'\n",
      " 14266628 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051407.3770118 INFO: Expansion time:Pyr.NO2:0.6669413440004064\n",
      "1723051407.3791785 INFO: Filtered count:0 out of 1475\n",
      "1723051407.6485422 INFO: metabolites 1118\n",
      "1723051408.0444162 INFO: Expansion time:Pyr.NO3:0.6640797419995579\n",
      "1723051408.046643 INFO: Filtered count:0 out of 1475\n",
      "1723051408.3461196 INFO: reactions 1118\n",
      "1723051408.533013 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051409.0436985 INFO: Expansion time:Pyr.O2:4.713553915000375\n",
      "1723051409.0460289 INFO: Filtered count:0 out of 1432\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051409.2497923 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/353/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051409.5191483 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051409.949946 INFO: Expansion time:Etho.O2:0.6234323319995383\n",
      "1723051409.9522512 INFO: Filtered count:0 out of 1161\n",
      "1723051410.2593453 INFO: cpd05178 not found in model!\n",
      "1723051410.4525967 INFO: cpd05178 not found in model!\n",
      "1723051410.4554954 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051410.492571 INFO: cpd05178 not found in model!\n",
      "1723051410.4947345 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051410.5491936 INFO: Expansion time:Pyr.SO3:1.5015911549999146\n",
      "1723051410.5514698 INFO: Filtered count:0 out of 1432\n",
      "1723051410.8068278 INFO: Expansion time:Fum.O2:0.8531659179998314\n",
      "1723051410.8093884 INFO: Filtered count:0 out of 1161\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D1.13.pg.G.D GF: 12075 Prob: 3110 Other: 3169\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051411.2679713 INFO: Expansion time:Pyr.O2:3.220111437000014\n",
      "1723051411.2701135 INFO: Filtered count:0 out of 1475\n",
      "1723051411.4853752 INFO: Expansion time:Pyr.SO4:0.9325013569996372\n",
      "1723051411.4877295 INFO: Filtered count:0 out of 1432\n",
      "1723051411.6551056 INFO: Expansion time:empty:0.8441925839997566\n",
      "1723051411.6575572 INFO: Filtered count:0 out of 1161\n",
      "1723051412.397136 INFO: Expansion time:Pyr.TMAO:0.9080567459996018\n",
      "1723051412.3994195 INFO: Filtered count:0 out of 1432\n",
      "1723051412.7764401 INFO: Expansion time:Pyr.SO3:1.5049747830007618\n",
      "1723051412.778506 INFO: Filtered count:0 out of 1475\n",
      "1723051413.3105166 INFO: Expansion time:empty:0.9096682480003437\n",
      "1723051413.312873 INFO: Filtered count:0 out of 1432\n",
      "1723051415.566003 INFO: cpd05178 not found in model!\n",
      "1723051415.5682676 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051415.891454 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051416.0062907 INFO: cpd05178 not found in model!\n",
      "1723051416.023188 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051416.3923266 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051418.027989 INFO: cpd05178 not found in model!\n",
      "1723051418.0383449 INFO: cpd05178 not found in model!\n",
      "1723051418.0406458 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051418.2609038 INFO: cpd05178 not found in model!\n",
      "1723051418.2634063 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051419.4237447 INFO: cpd05178 not found in model!\n",
      "1723051419.426476 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "Failed:>rxn00062_c0\n",
      "R2A_A_D1.50.pg.G.D GF: 11792 Prob: 3430 Other: 3128\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051419.8500533 INFO: Expansion time:Pyr.SO4:7.070330321000256\n",
      "1723051419.8521087 INFO: Filtered count:1 out of 1475\n",
      "1723051420.5091474 INFO: Expansion time:Pyr.TMAO:0.6558748739998919\n",
      "1723051420.5112832 INFO: Filtered count:1 out of 1475\n",
      "1723051421.1693573 INFO: Expansion time:empty:0.6568261299998994\n",
      "1723051421.171763 INFO: Filtered count:1 out of 1475\n",
      "1723051422.3987088 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051424.4803474 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051426.8888168 INFO: Expansion time:Ac.NO3:0.6446594040007767\n",
      "1723051426.8907561 INFO: Filtered count:0 out of 1471\n",
      "1723051427.5209112 INFO: Expansion time:Ac.O2:0.6290480170000592\n",
      "1723051427.522802 INFO: Filtered count:0 out of 1471\n",
      "1723051428.4633884 INFO: Expansion time:Etho.O2:0.9394383449998713\n",
      "1723051428.465766 INFO: Filtered count:0 out of 1471\n",
      "1723051429.4742284 INFO: Expansion time:Fum:1.006912731999364\n",
      "1723051429.4765713 INFO: Filtered count:0 out of 1471\n",
      "1723051430.4251518 INFO: Expansion time:Fum.O2:0.9471851090002019\n",
      "1723051430.4274445 INFO: Filtered count:0 out of 1471\n",
      "1723051431.394636 INFO: Expansion time:Glc:0.9658072379997975\n",
      "1723051431.3969014 INFO: Filtered count:0 out of 1471\n",
      "1723051431.637046 INFO: cpd05178 not found in model!\n",
      "1723051431.6394262 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.46.pg.G.D GF: 11731 Prob: 3400 Other: 3219\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051432.4542747 INFO: Expansion time:Glc.DMSO:1.0559366369998315\n",
      "1723051432.4566758 INFO: Filtered count:0 out of 1471\n",
      "1723051433.5158234 INFO: Expansion time:Glc.O2:1.0576432060006482\n",
      "1723051433.5181794 INFO: Filtered count:0 out of 1471\n",
      "1723051434.5738454 INFO: Expansion time:Glc.TMAO:1.054252668999652\n",
      "1723051434.5762436 INFO: Filtered count:0 out of 1471\n",
      "1723051435.600969 INFO: Expansion time:Glyc.O2:1.0233237720003672\n",
      "1723051435.6032891 INFO: Filtered count:0 out of 1471\n",
      "1723051435.9109137 INFO: cpd05178 not found in model!\n",
      "1723051435.913246 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051435.921774 INFO: cpd05178 not found in model!\n",
      "1723051436.301968 INFO: cpd05178 not found in model!\n",
      "1723051436.3046362 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051438.7008405 INFO: Expansion time:Pyr:3.096115756000472\n",
      "1723051438.703128 INFO: Filtered count:0 out of 1471\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051439.3490207 INFO: Objective with gapfill database:198.67766718466186; min objective:0.01\n",
      "1723051439.3524072 INFO: cpd05178 not found in model!\n",
      "1723051439.6806843 INFO: cpd05178 not found in model!\n",
      "1723051439.683408 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051439.7046342 INFO: Expansion time:Pyr.DMSO:1.0001012339998852\n",
      "1723051439.7068956 INFO: Filtered count:0 out of 1471\n",
      "1723051439.728112 INFO: Expansion time:For.NO3:0.6314620920002199\n",
      "1723051439.7301905 INFO: Filtered count:0 out of 1267\n",
      "1723051440.282771 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051440.384688 INFO: Expansion time:For.O2:0.6531436489995031\n",
      "1723051440.386793 INFO: Filtered count:0 out of 1267\n",
      "1723051440.7224295 INFO: Expansion time:Pyr.NO:1.0140679539999837\n",
      "1723051440.7247396 INFO: Filtered count:0 out of 1471\n",
      "1723051441.019313 INFO: Expansion time:Fum.O2:0.6313566409999112\n",
      "1723051441.0214477 INFO: Filtered count:0 out of 1267\n",
      "1723051441.6694026 INFO: Expansion time:Glc.O2:0.6461098549998496\n",
      "1723051441.671569 INFO: Filtered count:0 out of 1267\n",
      "1723051441.7589767 INFO: Expansion time:Pyr.NO2:1.0325964519997797\n",
      "1723051441.7612917 INFO: Filtered count:0 out of 1471\n",
      "1723051442.3263633 INFO: Expansion time:empty:0.6535616920000393\n",
      "1723051442.3285203 INFO: Filtered count:0 out of 1267\n",
      "1723051442.8684304 INFO: Expansion time:Pyr.NO3:1.105657482999959\n",
      "1723051442.8708615 INFO: Filtered count:0 out of 1471\n",
      "1723051443.9270892 INFO: Expansion time:Pyr.O2:1.0546360200005438\n",
      "1723051443.9293387 INFO: Filtered count:0 out of 1471\n",
      "1723051445.0025506 INFO: Expansion time:Pyr.SO3:1.0717690999999832\n",
      "1723051445.0049255 INFO: Filtered count:0 out of 1471\n",
      "1723051446.0269282 INFO: Expansion time:Pyr.SO4:1.0204728590006198\n",
      "1723051446.0293546 INFO: Filtered count:0 out of 1471\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[108 'R2_restored_H2O.34.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:28:57+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '84915a5a6fd44c46c222d326300e2558'\n",
      " 4756073 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051446.9983144 INFO: Expansion time:Pyr.TMAO:0.9674091369997768\n",
      "1723051447.0006382 INFO: Filtered count:0 out of 1471\n",
      "1723051447.1036313 INFO: cpd05178 not found in model!\n",
      "1723051447.10595 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051447.469301 INFO: metabolites 892\n",
      "1723051447.8414679 INFO: reactions 826\n",
      "1723051448.0304391 INFO: Expansion time:empty:1.0283706519994666\n",
      "1723051448.0327966 INFO: Filtered count:0 out of 1471\n",
      "1723051448.075955 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051448.4608698 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/342/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051450.3458657 INFO: cpd05178 not found in model!\n",
      "1723051450.6012273 INFO: cpd05178 not found in model!\n",
      "1723051450.6035419 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051453.7603128 INFO: Expansion time:Dlac:4.187429390000034\n",
      "1723051453.7626076 INFO: Filtered count:0 out of 1618\n",
      "1723051453.7882764 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051455.7252924 INFO: Expansion time:Dlac.O2:1.9615187790004711\n",
      "1723051455.727492 INFO: Filtered count:0 out of 1618\n",
      "1723051456.38722 INFO: Expansion time:Etho.O2:0.6584913239994421\n",
      "1723051456.3893347 INFO: Filtered count:0 out of 1618\n",
      "1723051456.6340833 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051457.6126347 INFO: Expansion time:Fum:1.2220786509997197\n",
      "1723051457.6148555 INFO: Filtered count:0 out of 1618\n",
      "1723051458.9971058 INFO: Expansion time:Fum.O2:1.3810549970003194\n",
      "1723051458.9994113 INFO: Filtered count:0 out of 1618\n",
      "1723051459.4429176 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051459.855311 INFO: cpd05178 not found in model!\n",
      "1723051459.857589 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051460.7064655 INFO: Expansion time:Glc.O2:1.705871119999756\n",
      "1723051460.7086265 INFO: Filtered count:0 out of 1618\n",
      "1723051461.9265656 INFO: Expansion time:Pyr:1.2167772170005264\n",
      "1723051461.9287984 INFO: Filtered count:0 out of 1618\n",
      "1723051463.0557911 INFO: cpd05178 not found in model!\n",
      "1723051463.0578418 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[76 'SF2_B_H2O.19.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:22:38+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e26639b812883108cb4111fa9d77bd1e'\n",
      " 7840609 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051464.6163883 INFO: metabolites 1161\n",
      "1723051465.2809727 INFO: reactions 1107\n",
      "1723051465.4268339 INFO: Expansion time:Pyr.DMSO:3.4967692479995094\n",
      "1723051465.4289398 INFO: Filtered count:0 out of 1618\n",
      "1723051465.4663618 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051466.113683 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/132/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051466.4226544 INFO: Expansion time:Pyr.NO:0.9924709360002453\n",
      "1723051466.4248533 INFO: Filtered count:0 out of 1618\n",
      "1723051467.3110566 INFO: Expansion time:Pyr.NO2:0.8849731970003631\n",
      "1723051467.3132606 INFO: Filtered count:0 out of 1618\n",
      "1723051468.4073231 INFO: Expansion time:Pyr.NO3:1.0927207000004273\n",
      "1723051468.4096894 INFO: Filtered count:0 out of 1618\n",
      "1723051469.3118255 INFO: Expansion time:Pyr.O2:0.9007119770003555\n",
      "1723051469.3140109 INFO: Filtered count:0 out of 1618\n",
      "1723051470.2680569 INFO: Expansion time:Pyr.SO3:0.9527592549993642\n",
      "1723051470.2702599 INFO: Filtered count:0 out of 1618\n",
      "1723051470.6675973 INFO: cpd05178 not found in model!\n",
      "1723051470.6698859 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_H2O.34.pg.G.D GF: 12113 Prob: 3020 Other: 3219\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051471.6366441 INFO: Expansion time:Pyr.SO4:1.3650374540002304\n",
      "1723051471.638897 INFO: Filtered count:0 out of 1618\n",
      "1723051472.9723492 INFO: Expansion time:Pyr.TMAO:1.3321695599997838\n",
      "1723051472.974663 INFO: Filtered count:0 out of 1618\n",
      "1723051473.456729 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[227 'R2_C_H2O.35.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:52:23+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '837c1752d55617460fd0e09eb7fc9049'\n",
      " 9134628 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051473.6030715 INFO: Expansion time:empty:0.6272154779999255\n",
      "1723051473.6052988 INFO: Filtered count:0 out of 1618\n",
      "1723051473.7360709 INFO: cpd05178 not found in model!\n",
      "1723051473.7387857 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051473.8042517 INFO: metabolites 880\n",
      "1723051474.152858 INFO: reactions 819\n",
      "1723051474.277997 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051474.7596936 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/303/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051477.3821323 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[171 'R2A_A_D2.59.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:41:17+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '2e7ca336daefead8d18f0ba532cd48db'\n",
      " 18648926 None]\n",
      "[267 'R2_C_D2.21.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:00:24+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '69908af1ae9f55ead16b32a59737a7dc'\n",
      " 10982552 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051478.4210072 INFO: metabolites 1271\n",
      "1723051478.6605258 INFO: metabolites 964\n",
      "1723051478.8250096 INFO: cpd05178 not found in model!\n",
      "1723051478.8275166 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051479.112727 INFO: reactions 933\n",
      "1723051479.1373436 INFO: reactions 1233\n",
      "1723051479.2765453 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051479.7995136 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/126/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051479.870035 INFO: cpd05178 not found in model!\n",
      "1723051479.8721828 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051480.5732043 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051482.6955721 INFO: cpd05178 not found in model!\n",
      "1723051482.6981654 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051483.0264173 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051483.5358396 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[290 'R2A_B_D2.47.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:05:01+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ca2430947575851b25b9793ca6c2877a'\n",
      " 11081976 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051483.8291407 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/135/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051484.066654 INFO: metabolites 1109\n",
      "1723051484.1278093 INFO: cpd05178 not found in model!\n",
      "1723051484.13055 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051484.2956161 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051484.6625667 INFO: reactions 1062\n",
      "1723051484.841776 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051485.4761777 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/262/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051486.7125483 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051486.7598321 INFO: cpd05178 not found in model!\n",
      "1723051486.7620447 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051487.336072 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051487.8691506 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051490.2119088 INFO: cpd05178 not found in model!\n",
      "1723051490.2140403 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051490.3109407 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_H2O.19.pg.G.D GF: 11332 Prob: 4072 Other: 2946\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051490.5027149 INFO: cpd05178 not found in model!\n",
      "1723051490.5214112 INFO: cpd05178 not found in model!\n",
      "1723051490.5232854 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[28 'R2_restored_C_black.2.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:13:09+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'da36c827b143f51eedf52b0d495c9f55'\n",
      " 25209613 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051490.7433782 INFO: cpd05178 not found in model!\n",
      "1723051490.7455366 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051491.2663155 INFO: metabolites 1231\n",
      "1723051492.0142262 INFO: reactions 1243\n",
      "1723051492.0314524 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051492.0926137 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051492.241612 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051492.5357707 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051493.0281599 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/499/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051494.2250926 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[132 'R2A_B_H2O.80.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:33:44+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '20d718fd2451bb7dc2601684ed288973'\n",
      " 7396594 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051495.1278071 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051495.1484294 INFO: metabolites 944\n",
      "1723051495.4700558 INFO: reactions 880\n",
      "1723051495.5229325 INFO: cpd05178 not found in model!\n",
      "1723051495.5251987 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051495.5774887 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051495.9070392 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/556/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051496.491595 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051497.6359472 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051498.542137 INFO: Expansion time:Ac.O2:1.0059835279998879\n",
      "1723051498.5447822 INFO: Filtered count:0 out of 1179\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[24 'R1_B_D2.86.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:12:16+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e2b9d830f45175917dfd29c129bf4594'\n",
      " 9936791 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051499.5288033 INFO: Expansion time:Etho.O2:0.9824024649997227\n",
      "1723051499.5313773 INFO: Filtered count:0 out of 1179\n",
      "1723051500.3858025 INFO: cpd05178 not found in model!\n",
      "1723051500.3880403 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051500.4517899 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051500.5383513 INFO: Expansion time:Pyr:1.0054117590007081\n",
      "1723051500.541026 INFO: Filtered count:0 out of 1179\n",
      "1723051500.8940244 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051501.4345293 INFO: Expansion time:Pyr.DMSO:0.8918995490002999\n",
      "1723051501.437062 INFO: Filtered count:0 out of 1179\n",
      "1723051501.7108016 INFO: cpd05178 not found in model!\n",
      "1723051501.7128007 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051501.7206898 INFO: metabolites 1000\n",
      "1723051501.8513668 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_H2O.35.pg.G.D GF: 12266 Prob: 2822 Other: 3272\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051502.0575528 INFO: reactions 953\n",
      "1723051502.1095245 INFO: Objective with gapfill database:170.36152018842688; min objective:0.01\n",
      "1723051502.1644309 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051502.296602 INFO: Expansion time:Pyr.NO:0.858012902000155\n",
      "1723051502.299231 INFO: Filtered count:0 out of 1179\n",
      "1723051502.5221448 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/475/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051503.1880758 INFO: Expansion time:Pyr.NO2:0.887344748999567\n",
      "1723051503.1906862 INFO: Filtered count:0 out of 1179\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051504.358038 INFO: cpd05178 not found in model!\n",
      "1723051504.3601851 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051505.5385714 INFO: cpd05178 not found in model!\n",
      "1723051505.5405924 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D2.21.pg.G.D GF: 12246 Prob: 2830 Other: 3278\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051506.1272907 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051506.321537 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051506.4759953 INFO: Expansion time:Pyr.NO3:3.2838646350001\n",
      "1723051506.4786518 INFO: Filtered count:0 out of 1179\n",
      "1723051506.648298 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051506.7313144 INFO: cpd05178 not found in model!\n",
      "1723051506.7335796 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "R2A_B_D2.47.pg.G.D GF: 11760 Prob: 3384 Other: 3208\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051507.329584 INFO: Expansion time:Pyr.O2:0.8494363670006351\n",
      "1723051507.3321035 INFO: Filtered count:0 out of 1179\n",
      "1723051507.3596723 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051507.839603 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051508.1976173 INFO: Expansion time:Pyr.SO3:0.8640288169999621\n",
      "1723051508.200195 INFO: Filtered count:0 out of 1179\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051509.0437334 INFO: cpd05178 not found in model!\n",
      "1723051509.045907 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051509.1034331 INFO: Expansion time:Pyr.SO4:0.9015903589997833\n",
      "1723051509.1060126 INFO: Filtered count:0 out of 1179\n",
      "1723051509.9890952 INFO: Expansion time:Pyr.TMAO:0.8816476869997132\n",
      "1723051509.9916236 INFO: Filtered count:0 out of 1179\n",
      "1723051510.7253602 INFO: cpd05178 not found in model!\n",
      "1723051510.7276793 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051510.8735833 INFO: Expansion time:empty:0.8804633769996144\n",
      "1723051510.8762767 INFO: Filtered count:0 out of 1179\n",
      "1723051512.3489687 INFO: Objective with gapfill database:178.84632021308386; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[208 'R2_C_D1.118.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:48:43+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '5d2a2db3226daee2559f09b33ff7bc31'\n",
      " 9892414 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051512.8080494 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051513.0465698 INFO: metabolites 1055\n",
      "1723051513.4053316 INFO: reactions 985\n",
      "1723051513.5285065 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051513.5436943 INFO: cpd05178 not found in model!\n",
      "1723051513.5464213 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051513.6110911 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051513.913328 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D2.59.pg.G.D GF: 11418 Prob: 3796 Other: 188040/366/13136\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051514.4359002 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051515.2966435 INFO: cpd05178 not found in model!\n",
      "1723051515.2985349 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051518.0443356 INFO: cpd05178 not found in model!\n",
      "1723051518.0470266 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051518.2093203 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_B_H2O.80.pg.G.D GF: 12255 Prob: 2876 Other: 3221\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051518.4345791 INFO: cpd05178 not found in model!\n",
      "1723051518.4365263 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051518.451176 INFO: cpd05178 not found in model!\n",
      "1723051518.4532712 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051518.9394588 INFO: Objective with gapfill database:163.8518837810911; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[163 'R2_C_D1.65.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:39:42+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '85054bf63ccd58bf2f43cf31200d7fed'\n",
      " 12849829 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051519.5714347 INFO: metabolites 1243\n",
      "1723051520.3399394 INFO: reactions 1251\n",
      "1723051520.5319126 INFO: cpd05178 not found in model!\n",
      "1723051520.5337915 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051520.5568335 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_C_black.2.pg.G.D GF: 11224 Prob: 4106 Other: 3024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051521.0139139 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051521.2823427 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/105/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051521.52272 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051523.54502 INFO: cpd05178 not found in model!\n",
      "1723051523.5476913 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051523.7305262 INFO: Expansion time:Dlac:0.6477152449988353\n",
      "1723051523.7326603 INFO: Filtered count:0 out of 1323\n",
      "1723051523.7573001 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.86.pg.G.D GF: 12273 Prob: 2814 Other: 3265\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051524.3697765 INFO: Expansion time:Dlac.O2:0.6358576380007435\n",
      "1723051524.371778 INFO: Filtered count:0 out of 1323\n",
      "1723051524.8127744 INFO: cpd05178 not found in model!\n",
      "1723051524.8153963 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051524.9953518 INFO: Expansion time:Glc:0.6224398530011968\n",
      "1723051524.997328 INFO: Filtered count:0 out of 1323\n",
      "1723051525.6150799 INFO: Expansion time:Glc.DMSO:0.6163997369985736\n",
      "1723051525.6170418 INFO: Filtered count:0 out of 1323\n",
      "1723051526.616117 INFO: cpd05178 not found in model!\n",
      "1723051526.6185646 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051527.7684703 INFO: Expansion time:Glc.O2:2.150310830998933\n",
      "1723051527.7705357 INFO: Filtered count:0 out of 1323\n",
      "1723051528.4698386 INFO: cpd05178 not found in model!\n",
      "1723051528.7017345 INFO: Expansion time:Glc.TMAO:0.9298985080004059\n",
      "1723051528.7039638 INFO: Filtered count:0 out of 1323\n",
      "1723051528.71123 INFO: cpd05178 not found in model!\n",
      "1723051528.7131722 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051529.121353 INFO: cpd05178 not found in model!\n",
      "1723051529.1239152 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051529.6019852 INFO: Expansion time:Pyr:0.8966080680002051\n",
      "1723051529.6042473 INFO: Filtered count:0 out of 1323\n",
      "1723051529.939831 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051530.5065532 INFO: Expansion time:Pyr.DMSO:0.9008802090011159\n",
      "1723051530.5089011 INFO: Filtered count:0 out of 1323\n",
      "1723051530.945259 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051531.5053406 INFO: Expansion time:Pyr.NO:0.9949882739983877\n",
      "1723051531.5077832 INFO: Filtered count:0 out of 1323\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051532.2979622 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051532.504233 INFO: Expansion time:Pyr.NO2:0.9949533399994834\n",
      "1723051532.5065298 INFO: Filtered count:0 out of 1323\n",
      "1723051533.443442 INFO: Expansion time:Pyr.NO3:0.9354791470013879\n",
      "1723051533.4458468 INFO: Filtered count:0 out of 1323\n",
      "1723051534.3580585 INFO: Expansion time:Ac.O2:2.3267380990000675\n",
      "1723051534.3603792 INFO: Filtered count:0 out of 1777\n",
      "1723051534.3947995 INFO: Expansion time:Pyr.O2:0.9474439379991964\n",
      "1723051534.3971746 INFO: Filtered count:0 out of 1323\n",
      "1723051535.0615227 INFO: Expansion time:Etho.O2:0.6999707859995397\n",
      "1723051535.0638225 INFO: Filtered count:0 out of 1777\n",
      "1723051535.338408 INFO: Expansion time:Pyr.SO3:0.9396783249994769\n",
      "1723051535.3407035 INFO: Filtered count:0 out of 1323\n",
      "1723051535.7557724 INFO: Expansion time:For.O2:0.6907430329993076\n",
      "1723051535.7580037 INFO: Filtered count:0 out of 1777\n",
      "1723051536.4563801 INFO: Expansion time:Fum:0.6971195909991366\n",
      "1723051536.4586878 INFO: Filtered count:0 out of 1777\n",
      "1723051536.4919844 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051537.0026615 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051537.1606688 INFO: Expansion time:Fum.O2:0.7007871190016886\n",
      "1723051537.1630158 INFO: Filtered count:0 out of 1777\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051537.6763687 INFO: Objective with gapfill database:151.6003148186708; min objective:0.01\n",
      "1723051537.836415 INFO: Expansion time:Pyr.SO4:2.4942817660012224\n",
      "1723051537.8388333 INFO: Filtered count:0 out of 1323\n",
      "1723051537.839027 INFO: cpd05178 not found in model!\n",
      "1723051537.8410432 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051537.8567874 INFO: Expansion time:Glc.O2:0.6926203059992986\n",
      "1723051537.8589048 INFO: Filtered count:0 out of 1777\n",
      "1723051538.5341575 INFO: Expansion time:Pyr:0.6741090510004142\n",
      "1723051538.5363948 INFO: Filtered count:0 out of 1777\n",
      "1723051538.7764614 INFO: Expansion time:Pyr.TMAO:0.9362166850005451\n",
      "1723051538.778751 INFO: Filtered count:0 out of 1323\n",
      "1723051539.249968 INFO: Expansion time:Pyr.DMSO:0.7123771189999388\n",
      "1723051539.2521806 INFO: Filtered count:0 out of 1777\n",
      "1723051539.737578 INFO: Expansion time:empty:0.9572859240015532\n",
      "1723051539.7400527 INFO: Filtered count:0 out of 1323\n",
      "1723051539.9277656 INFO: cpd05178 not found in model!\n",
      "1723051539.9301476 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051539.9497411 INFO: Expansion time:Pyr.NO:0.6963820159999159\n",
      "1723051539.9520893 INFO: Filtered count:0 out of 1777\n",
      "1723051540.0220535 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D1.118.pg.G.D GF: 11937 Prob: 3134 Other: 3279\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051540.351264 INFO: cpd05178 not found in model!\n",
      "1723051540.3536558 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051541.1062255 INFO: cpd05178 not found in model!\n",
      "1723051541.1087031 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051542.196772 INFO: Expansion time:Pyr.NO2:2.243316488000346\n",
      "1723051542.1992357 INFO: Filtered count:0 out of 1777\n",
      "1723051542.5286582 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051542.8877306 INFO: Expansion time:Pyr.NO3:0.6872183800005587\n",
      "1723051542.890051 INFO: Filtered count:0 out of 1777\n",
      "1723051543.565571 INFO: Expansion time:Pyr.O2:0.6741693089988985\n",
      "1723051543.567877 INFO: Filtered count:0 out of 1777\n",
      "1723051544.2351007 INFO: Expansion time:Pyr.SO3:0.6658530989989231\n",
      "1723051544.237266 INFO: Filtered count:0 out of 1777\n",
      "1723051544.4770784 INFO: cpd05178 not found in model!\n",
      "1723051544.4792428 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051544.905894 INFO: Expansion time:Pyr.SO4:0.6671463999991829\n",
      "1723051544.9081109 INFO: Filtered count:0 out of 1777\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051545.5754871 INFO: Expansion time:Pyr.TMAO:0.6662729050003691\n",
      "1723051545.5777225 INFO: Filtered count:0 out of 1777\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051546.0308006 INFO: cpd05178 not found in model!\n",
      "1723051546.2582443 INFO: Expansion time:empty:0.6792801239989785\n",
      "1723051546.260463 INFO: Filtered count:0 out of 1777\n",
      "1723051546.2607558 INFO: cpd05178 not found in model!\n",
      "1723051546.262948 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051546.6010764 INFO: cpd05178 not found in model!\n",
      "1723051546.6036747 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051547.1143064 INFO: Expansion time:Ac.O2:1.1019121120007185\n",
      "1723051547.116698 INFO: Filtered count:0 out of 1797\n",
      "1723051548.1862667 INFO: Expansion time:Etho.O2:1.0680905299996084\n",
      "1723051548.188615 INFO: Filtered count:0 out of 1797\n",
      "1723051549.3053155 INFO: Expansion time:For.O2:1.1152197859992157\n",
      "1723051549.3077269 INFO: Filtered count:0 out of 1797\n",
      "1723051550.385395 INFO: Expansion time:Fum:1.0762215190006827\n",
      "1723051550.3877988 INFO: Filtered count:0 out of 1797\n",
      "1723051550.7687728 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051550.9720385 INFO: cpd05178 not found in model!\n",
      "1723051550.974217 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D1.65.pg.G.D GF: 10927 Prob: 4550 Other: 2877\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051551.432219 INFO: Expansion time:Fum.O2:1.0429716079997888\n",
      "1723051551.4346747 INFO: Filtered count:0 out of 1797\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[183 'R1_B_H2O.5.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:43:34+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '0bc075c164207c4f002e1dbfc0db8ebd'\n",
      " 10868560 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051552.4199111 INFO: metabolites 1197\n",
      "1723051552.5670714 INFO: Expansion time:Glc:1.1308078210004169\n",
      "1723051552.5693886 INFO: Filtered count:0 out of 1797\n",
      "1723051553.1653647 INFO: reactions 1207\n",
      "1723051553.3510664 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051553.673927 INFO: Expansion time:Glc.DMSO:1.1030932539997593\n",
      "1723051553.676281 INFO: Filtered count:0 out of 1797\n",
      "1723051553.890522 INFO: Expansion time:For.NO2:8.318136895000862\n",
      "1723051553.8925643 INFO: Filtered count:0 out of 1209\n",
      "1723051554.027289 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/205/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051554.723687 INFO: Expansion time:Glc.O2:1.0460259680003219\n",
      "1723051554.7260015 INFO: Filtered count:0 out of 1797\n",
      "1723051555.1266732 INFO: Expansion time:Fum:1.2330351330001577\n",
      "1723051555.1287677 INFO: Filtered count:0 out of 1209\n",
      "1723051555.7963703 INFO: Expansion time:Glc.TMAO:1.068886327000655\n",
      "1723051555.798917 INFO: Filtered count:0 out of 1797\n",
      "1723051556.0955086 INFO: Expansion time:Pyr:0.9656349480010249\n",
      "1723051556.097604 INFO: Filtered count:0 out of 1209\n",
      "1723051556.6918278 INFO: cpd05178 not found in model!\n",
      "1723051556.694029 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051556.924781 INFO: Expansion time:Glyc.O2:1.124197113000264\n",
      "1723051556.9271026 INFO: Filtered count:0 out of 1797\n",
      "1723051557.2563176 INFO: Expansion time:Pyr.DMSO:1.1574736330003361\n",
      "1723051557.2583725 INFO: Filtered count:0 out of 1209\n",
      "1723051558.3655818 INFO: Expansion time:Pyr.NO:1.106088329999693\n",
      "1723051558.3677022 INFO: Filtered count:0 out of 1209\n",
      "1723051559.1466076 INFO: cpd05178 not found in model!\n",
      "1723051559.238301 INFO: cpd05178 not found in model!\n",
      "1723051559.2405076 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051559.3702672 INFO: cpd05178 not found in model!\n",
      "1723051559.3723674 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051559.4044802 INFO: Expansion time:Pyr.NO3:1.0356136859991238\n",
      "1723051559.406568 INFO: Filtered count:0 out of 1209\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051560.409411 INFO: cpd05178 not found in model!\n",
      "1723051560.6451888 INFO: cpd05178 not found in model!\n",
      "1723051560.6474433 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051560.7911568 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051560.878421 INFO: Expansion time:Pyr.O2:1.4707288859990513\n",
      "1723051560.8804064 INFO: Filtered count:0 out of 1209\n",
      "1723051560.8812306 INFO: Expansion time:LLac.O2:3.9526345599988417\n",
      "1723051560.8833039 INFO: Filtered count:0 out of 1797\n",
      "1723051561.0496724 INFO: Expansion time:Dlac:0.6880795569995826\n",
      "1723051561.0516915 INFO: Filtered count:0 out of 1376\n",
      "1723051561.8586252 INFO: Expansion time:Pyr.SO3:0.9770488659996772\n",
      "1723051561.860706 INFO: Filtered count:0 out of 1209\n",
      "1723051562.063704 INFO: Expansion time:Pyr:1.178865672000029\n",
      "1723051562.0661952 INFO: Filtered count:0 out of 1797\n",
      "1723051563.1467588 INFO: Expansion time:Pyr.DMSO:1.078991139000209\n",
      "1723051563.149068 INFO: Filtered count:0 out of 1797\n",
      "1723051563.8052404 INFO: Expansion time:Dlac.O2:2.7521590559990727\n",
      "1723051563.8073158 INFO: Filtered count:0 out of 1376\n",
      "1723051563.978896 INFO: cpd05178 not found in model!\n",
      "1723051564.2393088 INFO: Expansion time:Pyr.NO:1.088806145999115\n",
      "1723051564.2416992 INFO: Filtered count:0 out of 1797\n",
      "1723051564.3213506 INFO: cpd05178 not found in model!\n",
      "1723051564.324004 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051564.48741 INFO: Expansion time:Fum:0.6788161370004673\n",
      "1723051564.4894526 INFO: Filtered count:0 out of 1376\n",
      "1723051564.852396 INFO: Expansion time:Pyr.SO4:2.99053000800086\n",
      "1723051564.8545434 INFO: Filtered count:0 out of 1209\n",
      "1723051565.1561885 INFO: Expansion time:Fum.O2:0.665393954001047\n",
      "1723051565.1582458 INFO: Filtered count:0 out of 1376\n",
      "1723051565.4064052 INFO: Expansion time:Pyr.NO2:1.1631192629993166\n",
      "1723051565.40873 INFO: Filtered count:0 out of 1797\n",
      "1723051565.4703715 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051565.7653277 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051565.8110342 INFO: Expansion time:Glc:0.6514544779984135\n",
      "1723051565.8129027 INFO: Filtered count:0 out of 1376\n",
      "1723051565.8808699 INFO: Expansion time:Pyr.TMAO:1.0252153930014174\n",
      "1723051565.8829134 INFO: Filtered count:0 out of 1209\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051566.4594727 INFO: Expansion time:empty:0.5754186230005871\n",
      "1723051566.459817 INFO: Expansion time:Glc.DMSO:0.6456292790007865\n",
      "1723051566.4615476 INFO: Filtered count:0 out of 1209\n",
      "1723051566.4618998 INFO: Filtered count:0 out of 1376\n",
      "1723051566.5316079 INFO: Expansion time:Pyr.NO3:1.1214298970007803\n",
      "1723051566.5339534 INFO: Filtered count:0 out of 1797\n",
      "1723051567.0991895 INFO: Expansion time:Glc.O2:0.635935229000097\n",
      "1723051567.1011856 INFO: Filtered count:0 out of 1376\n",
      "1723051567.6244493 INFO: Expansion time:Pyr.O2:1.0890976139999111\n",
      "1723051567.626915 INFO: Filtered count:0 out of 1797\n",
      "1723051567.74038 INFO: Expansion time:Glc.TMAO:0.6379833230002987\n",
      "1723051567.742343 INFO: Filtered count:0 out of 1376\n",
      "1723051567.7884283 INFO: cpd05178 not found in model!\n",
      "1723051568.0269322 INFO: cpd05178 not found in model!\n",
      "1723051568.0290017 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051568.3976016 INFO: Expansion time:Pyr:0.6541389829999389\n",
      "1723051568.39962 INFO: Filtered count:0 out of 1376\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[311 'R2A_A_D2.51.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:09:28+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'bf80f1a0d8c0aae47b243718beba260e'\n",
      " 13641155 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051568.7942882 INFO: Expansion time:Pyr.SO3:1.165829744999428\n",
      "1723051568.7965643 INFO: Filtered count:0 out of 1797\n",
      "1723051569.0394473 INFO: cpd05178 not found in model!\n",
      "1723051569.0722015 INFO: Expansion time:Pyr.DMSO:0.6714290380004968\n",
      "1723051569.0741832 INFO: Filtered count:0 out of 1376\n",
      "1723051569.1962037 INFO: metabolites 1248\n",
      "1723051569.2776203 INFO: cpd05178 not found in model!\n",
      "1723051569.2798336 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051569.4937844 INFO: cpd05178 not found in model!\n",
      "1723051569.5030339 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051569.661378 INFO: cpd05178 not found in model!\n",
      "1723051569.7219586 INFO: Expansion time:Pyr.NO:0.6464232039998024\n",
      "1723051569.7239604 INFO: Filtered count:0 out of 1376\n",
      "1723051569.8980696 INFO: cpd05178 not found in model!\n",
      "1723051569.9004025 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051569.9032447 INFO: Expansion time:Pyr.SO4:1.1052267140003096\n",
      "1723051569.9050956 INFO: Filtered count:0 out of 1797\n",
      "1723051569.9977152 INFO: reactions 1218\n",
      "1723051570.1959672 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051570.321342 INFO: cpd05178 not found in model!\n",
      "1723051570.3237295 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051570.3355331 INFO: Expansion time:Ac.NO2:0.6692376150003838\n",
      "1723051570.3375201 INFO: Filtered count:0 out of 1786\n",
      "1723051570.365014 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051570.3699222 INFO: Expansion time:Pyr.NO2:0.64475121400028\n",
      "1723051570.3718596 INFO: Filtered count:0 out of 1376\n",
      "1723051570.9433298 INFO: Default biomass: [bio1]\n",
      "1723051570.9781005 INFO: Expansion time:Pyr.TMAO:1.0715279369997006\n",
      "1723051570.9804106 INFO: Filtered count:0 out of 1797\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/432/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051571.0064018 INFO: Expansion time:Ac.NO3:0.6675875289984106\n",
      "1723051571.0082912 INFO: Filtered count:0 out of 1786\n",
      "1723051571.0306504 INFO: Expansion time:Pyr.NO3:0.6574463360011578\n",
      "1723051571.0327013 INFO: Filtered count:0 out of 1376\n",
      "1723051571.6817853 INFO: Expansion time:Ac.O2:0.6723366130008799\n",
      "1723051571.683987 INFO: Filtered count:0 out of 1786\n",
      "1723051572.126371 INFO: Expansion time:empty:1.1442873920004786\n",
      "1723051572.12872 INFO: Filtered count:0 out of 1797\n",
      "1723051572.347702 INFO: Expansion time:Etho.O2:0.6625124999991385\n",
      "1723051572.3498545 INFO: Filtered count:0 out of 1786\n",
      "1723051573.010696 INFO: Expansion time:For.NO2:0.6595898590003344\n",
      "1723051573.012884 INFO: Filtered count:0 out of 1786\n",
      "1723051573.693865 INFO: Expansion time:Pyr.O2:2.6599865640000644\n",
      "1723051573.6959903 INFO: Filtered count:0 out of 1376\n",
      "1723051573.720796 INFO: Expansion time:For.NO3:0.7066754369989212\n",
      "1723051573.7231395 INFO: Filtered count:0 out of 1786\n",
      "1723051574.4187918 INFO: Expansion time:For.O2:0.6944048919995112\n",
      "1723051574.4209502 INFO: Filtered count:0 out of 1786\n",
      "1723051574.638345 INFO: Expansion time:Pyr.SO3:0.9408525369999552\n",
      "1723051574.640856 INFO: Filtered count:0 out of 1376\n",
      "1723051574.851213 INFO: cpd05178 not found in model!\n",
      "1723051574.8538296 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051575.0938258 INFO: Expansion time:Fum:0.6716998100000637\n",
      "1723051575.095933 INFO: Filtered count:0 out of 1786\n",
      "1723051575.702538 INFO: Expansion time:Pyr.SO4:1.0601397580012417\n",
      "1723051575.7048879 INFO: Filtered count:0 out of 1376\n",
      "1723051575.7613363 INFO: Expansion time:Fum.O2:0.664256034000573\n",
      "1723051575.7634544 INFO: Filtered count:0 out of 1786\n",
      "1723051576.433727 INFO: Expansion time:Glc:0.6690930089989706\n",
      "1723051576.4360392 INFO: Filtered count:0 out of 1786\n",
      "1723051576.6347144 INFO: Expansion time:Pyr.TMAO:0.9281358320004074\n",
      "1723051576.6369584 INFO: Filtered count:0 out of 1376\n",
      "1723051577.576247 INFO: Expansion time:empty:0.9376948730005097\n",
      "1723051577.5786064 INFO: Filtered count:0 out of 1376\n",
      "1723051578.3261173 INFO: cpd05178 not found in model!\n",
      "1723051578.328194 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051578.6981792 INFO: Expansion time:Glc.DMSO:2.26095821600029\n",
      "1723051578.7003562 INFO: Filtered count:0 out of 1786\n",
      "1723051578.8595128 INFO: cpd05178 not found in model!\n",
      "1723051578.8620753 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051579.395197 INFO: Expansion time:Glc.O2:0.6936547559998871\n",
      "1723051579.3974147 INFO: Filtered count:0 out of 1786\n",
      "1723051580.0990934 INFO: Expansion time:Glc.TMAO:0.7003621480016591\n",
      "1723051580.1014602 INFO: Filtered count:0 out of 1786\n",
      "1723051580.3704028 INFO: cpd05178 not found in model!\n",
      "1723051580.3727074 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051580.6003454 INFO: cpd05178 not found in model!\n",
      "1723051580.6024678 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_H2O.5.pg.G.D GF: 11377 Prob: 4030 Other: 2945\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051580.781268 INFO: Expansion time:Pyr:0.6785180670012778\n",
      "1723051580.7835715 INFO: Filtered count:0 out of 1786\n",
      "1723051581.456573 INFO: Expansion time:Pyr.DMSO:0.671772644000157\n",
      "1723051581.4586039 INFO: Filtered count:0 out of 1786\n",
      "1723051582.470435 INFO: Expansion time:Pyr.NO:1.010440928999742\n",
      "1723051582.4728985 INFO: Filtered count:0 out of 1786\n",
      "1723051582.865744 INFO: cpd05178 not found in model!\n",
      "1723051582.867927 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051583.4588523 INFO: Expansion time:Pyr.NO2:0.984533721999469\n",
      "1723051583.4611864 INFO: Filtered count:0 out of 1786\n",
      "1723051584.5449839 INFO: Expansion time:Pyr.NO3:1.0823457469996356\n",
      "1723051584.547459 INFO: Filtered count:0 out of 1786\n",
      "1723051585.3587139 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051585.6172626 INFO: Expansion time:Pyr.O2:1.0681391810012428\n",
      "1723051585.6196187 INFO: Filtered count:0 out of 1786\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[92 'R2A_A_D1.14.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:25:40+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '38104fcf75852c2f02bccc69972e8aa4'\n",
      " 11674294 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051585.8453383 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051585.9086576 INFO: cpd05178 not found in model!\n",
      "1723051585.910993 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051586.1728816 INFO: metabolites 1075\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051586.7119806 INFO: Expansion time:Pyr.SO3:1.0909602930005349\n",
      "1723051586.7147245 INFO: Filtered count:0 out of 1786\n",
      "1723051586.723188 INFO: reactions 1070\n",
      "1723051586.8771396 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051587.2929282 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/238/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051587.5922358 INFO: cpd05178 not found in model!\n",
      "1723051587.5946376 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051589.5595822 INFO: Expansion time:Pyr.SO4:2.843446041999414\n",
      "1723051589.5617404 INFO: Filtered count:0 out of 1786\n",
      "1723051590.2303884 INFO: Expansion time:Pyr.TMAO:0.66728494400013\n",
      "1723051590.2322454 INFO: Filtered count:0 out of 1786\n",
      "1723051590.8457382 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051591.1701138 INFO: Expansion time:empty:0.9367815269997664\n",
      "1723051591.172232 INFO: Filtered count:0 out of 1786\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[204 'R2_C_H2O.20.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:48:03+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'f8bda4d5cfbeb2bf21e90fd656598910'\n",
      " 7326706 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051591.6417153 INFO: metabolites 727\n",
      "1723051592.0218022 INFO: reactions 651\n",
      "1723051592.1259527 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051592.510675 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/273/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051594.0181487 INFO: cpd05178 not found in model!\n",
      "1723051594.0203054 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051594.5034087 INFO: Objective with gapfill database:248.1535126520485; min objective:0.01\n",
      "1723051595.1017504 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[244 'R2_A_H2O.035.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:55:42+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'a8e352f088064fbe51464893c0b23839'\n",
      " 3386128 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051595.6352932 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051595.8662655 INFO: metabolites 471\n",
      "1723051595.9607608 INFO: reactions 419\n",
      "1723051596.0284634 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051596.1850584 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/196/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051597.3186517 INFO: cpd05178 not found in model!\n",
      "1723051597.5464454 INFO: cpd05178 not found in model!\n",
      "1723051597.5485005 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051598.061411 INFO: cpd05178 not found in model!\n",
      "1723051598.0637665 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D2.51.pg.G.D GF: 11122 Prob: 4310 Other: 2920\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051598.601855 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051599.0901527 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051601.7996447 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051603.803959 INFO: cpd05178 not found in model!\n",
      "1723051603.8066566 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051604.4598892 INFO: cpd05178 not found in model!\n",
      "1723051604.7844248 INFO: cpd05178 not found in model!\n",
      "1723051604.7869346 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051604.874543 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051605.410563 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051605.8464491 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051608.8351417 INFO: cpd05178 not found in model!\n",
      "1723051608.8372653 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[259 'R1_A_H2O.2.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:59:00+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '80e2500ebe6f98b43d6ba11426452c71'\n",
      " 9642373 None]\n",
      "R2A_A_D1.14.pg.G.D GF: 12017 Prob: 3082 Other: 3255\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051609.4481266 INFO: metabolites 983\n",
      "1723051609.491562 INFO: cpd05178 not found in model!\n",
      "1723051609.4940867 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051609.9862375 INFO: reactions 957\n",
      "1723051610.1570883 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051610.4673622 INFO: Objective with gapfill database:167.54945652225405; min objective:0.01\n",
      "1723051610.7932956 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/78/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051612.12587 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051614.8245447 INFO: cpd05178 not found in model!\n",
      "1723051614.827332 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_H2O.035.pg.G.D GF: 13707 Prob: 912 Other: 3741\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051618.416197 INFO: cpd05178 not found in model!\n",
      "1723051618.7443929 INFO: cpd05178 not found in model!\n",
      "1723051618.7470765 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051618.9567707 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[143 'R2_A_H2O.11.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:35:51+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '61cc7e40ce67a3c6628261416991b392'\n",
      " 13482164 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051620.7269328 INFO: metabolites 931\n",
      "1723051621.0563765 INFO: reactions 827\n",
      "1723051621.1507072 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051621.2803686 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051621.4766748 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/219/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051621.8292985 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051622.300569 INFO: cpd05178 not found in model!\n",
      "1723051622.3031492 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_H2O.20.pg.G.D GF: 12328 Prob: 2852 Other: 3180\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051627.1011932 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051627.1826193 INFO: cpd05178 not found in model!\n",
      "1723051627.1847928 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051628.236074 INFO: Expansion time:Etho.O2:0.6586219900000287\n",
      "1723051628.23816 INFO: Filtered count:0 out of 1472\n",
      "1723051628.647826 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051628.8798523 INFO: Expansion time:For.NO3:0.6405099520015938\n",
      "1723051628.8818855 INFO: Filtered count:0 out of 1472\n",
      "1723051629.35403 INFO: cpd05178 not found in model!\n",
      "1723051629.356262 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051629.9080622 INFO: cpd05178 not found in model!\n",
      "1723051629.9106922 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051631.4320478 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051631.6515074 INFO: Expansion time:For.O2:2.7683380510006828\n",
      "1723051631.6536286 INFO: Filtered count:0 out of 1472\n",
      "1723051631.7289875 INFO: cpd05178 not found in model!\n",
      "1723051631.7311206 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051631.8872218 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051632.307832 INFO: Expansion time:Pyr:0.6530108909992123\n",
      "1723051632.309838 INFO: Filtered count:0 out of 1472\n",
      "1723051632.860496 INFO: cpd05178 not found in model!\n",
      "1723051632.8627715 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051632.9741557 INFO: Expansion time:Pyr.DMSO:0.6630207860016526\n",
      "1723051632.9762073 INFO: Filtered count:0 out of 1472\n",
      "1723051633.2519116 INFO: cpd05178 not found in model!\n",
      "1723051633.2543225 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051633.616286 INFO: Expansion time:Pyr.NO:0.6389336570009618\n",
      "1723051633.6183724 INFO: Filtered count:0 out of 1472\n",
      "1723051633.8967276 INFO: cpd05178 not found in model!\n",
      "1723051633.898838 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051634.2676885 INFO: Expansion time:Pyr.NO2:0.6480719949995546\n",
      "1723051634.2697332 INFO: Filtered count:0 out of 1472\n",
      "1723051634.8853924 INFO: cpd05178 not found in model!\n",
      "1723051634.887825 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051634.9215958 INFO: Expansion time:Pyr.NO3:0.6507087539994245\n",
      "1723051634.9237053 INFO: Filtered count:0 out of 1472\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_H2O.2.pg.G.D GF: 11951 Prob: 3298 Other: 3105\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051635.5701382 INFO: Expansion time:Pyr.O2:0.6451905369995075\n",
      "1723051635.5724454 INFO: Filtered count:0 out of 1472\n",
      "1723051635.6799946 INFO: cpd05178 not found in model!\n",
      "1723051636.2338686 INFO: Expansion time:Pyr.SO3:0.6601204140006303\n",
      "1723051636.2359788 INFO: Filtered count:0 out of 1472\n",
      "1723051636.387228 INFO: Objective with gapfill database:125.8324684524614; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[56 'R2_A_D1.9.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:18:42+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'afcdf199f700547ef486c18cc6579342'\n",
      " 14450863 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051636.8976884 INFO: Expansion time:Pyr.SO4:0.6605141189993446\n",
      "1723051636.8998346 INFO: Filtered count:0 out of 1472\n",
      "1723051637.1145396 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051637.2619336 INFO: metabolites 1085\n",
      "1723051637.5649529 INFO: Expansion time:Pyr.TMAO:0.663980219998848\n",
      "1723051637.5670826 INFO: Filtered count:0 out of 1472\n",
      "1723051637.919884 INFO: cpd05178 not found in model!\n",
      "1723051637.922304 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051638.0659747 INFO: reactions 1050\n",
      "1723051638.2330332 INFO: Expansion time:empty:0.6647155830014526\n",
      "1723051638.2340488 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051638.2351925 INFO: Filtered count:0 out of 1472\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051638.9566205 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/614/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051639.1499712 INFO: Expansion time:Glc:0.5410819339995214\n",
      "1723051639.1521575 INFO: Filtered count:0 out of 517\n",
      "1723051639.541186 INFO: cpd05178 not found in model!\n",
      "1723051639.6752534 INFO: Expansion time:Glc.DMSO:0.5219113639996067\n",
      "1723051639.6773655 INFO: Filtered count:0 out of 517\n",
      "1723051639.7864594 INFO: cpd05178 not found in model!\n",
      "1723051639.7884157 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051640.1953661 INFO: Expansion time:Glc.O2:0.5168255950011371\n",
      "1723051640.1975603 INFO: Filtered count:0 out of 517\n",
      "1723051640.7320588 INFO: Expansion time:Glc.TMAO:0.5332497239996883\n",
      "1723051640.7342823 INFO: Filtered count:0 out of 517\n",
      "1723051641.266348 INFO: Expansion time:Pyr:0.5308834209990891\n",
      "1723051641.2684717 INFO: Filtered count:0 out of 517\n",
      "1723051641.7938724 INFO: Expansion time:Pyr.DMSO:0.5242273280000518\n",
      "1723051641.796058 INFO: Filtered count:0 out of 517\n",
      "1723051642.3146086 INFO: Expansion time:Pyr.NO:0.5174095120000857\n",
      "1723051642.31673 INFO: Filtered count:0 out of 517\n",
      "1723051642.8366213 INFO: Expansion time:Pyr.NO2:0.518755750001219\n",
      "1723051642.8387597 INFO: Filtered count:0 out of 517\n",
      "1723051643.4002285 INFO: Expansion time:Pyr.NO3:0.560122457000034\n",
      "1723051643.402526 INFO: Filtered count:0 out of 517\n",
      "1723051645.662294 INFO: cpd05178 not found in model!\n",
      "1723051645.6646245 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051645.8334684 INFO: Expansion time:Pyr.O2:2.4297978979993786\n",
      "1723051645.8356206 INFO: Filtered count:0 out of 517\n",
      "1723051646.359383 INFO: Expansion time:Pyr.SO3:0.5226282669991633\n",
      "1723051646.3616016 INFO: Filtered count:0 out of 517\n",
      "1723051646.883689 INFO: Expansion time:Pyr.SO4:0.5207938889998331\n",
      "1723051646.885891 INFO: Filtered count:0 out of 517\n",
      "1723051647.405958 INFO: Expansion time:Pyr.TMAO:0.5188221619991964\n",
      "1723051647.4081743 INFO: Filtered count:0 out of 517\n",
      "1723051647.4940374 INFO: cpd05178 not found in model!\n",
      "1723051647.496352 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051647.754415 INFO: Objective with gapfill database:169.11279571976087; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_H2O.11.pg.G.D GF: 12364 Prob: 2748 Other: 3248\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051647.9331958 INFO: Expansion time:empty:0.5238827930006664\n",
      "1723051647.9352925 INFO: Filtered count:0 out of 517\n",
      "1723051649.805853 INFO: cpd05178 not found in model!\n",
      "1723051650.0475948 INFO: cpd05178 not found in model!\n",
      "1723051650.0495818 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051653.2897627 INFO: cpd05178 not found in model!\n",
      "1723051653.5127587 INFO: cpd05178 not found in model!\n",
      "1723051653.5149033 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051654.9275227 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051655.7325463 INFO: cpd05178 not found in model!\n",
      "1723051655.734652 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051656.6553695 INFO: cpd05178 not found in model!\n",
      "1723051656.657801 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051657.7884912 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051661.1689262 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051662.2704768 INFO: cpd05178 not found in model!\n",
      "1723051663.1124265 INFO: Objective with gapfill database:185.48583523177714; min objective:0.01\n",
      "1723051663.6070392 INFO: cpd05178 not found in model!\n",
      "1723051663.6093466 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051665.2986739 INFO: cpd05178 not found in model!\n",
      "1723051665.541205 INFO: cpd05178 not found in model!\n",
      "1723051665.543354 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[251 'R2A_B_D2.54.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:57:14+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '11fcc50cb6d8b9d89c56679237c37527'\n",
      " 7645378 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051673.5816238 INFO: metabolites 1088\n",
      "1723051674.240184 INFO: reactions 1023\n",
      "1723051674.4042196 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051674.6283712 INFO: cpd05178 not found in model!\n",
      "1723051674.630715 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051675.0104353 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/65/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051675.5460763 INFO: cpd05178 not found in model!\n",
      "1723051675.5487688 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051675.6322584 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.9.pg.G.D GF: 11407 Prob: 3762 Other: 3181\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051675.995543 INFO: cpd05178 not found in model!\n",
      "1723051675.9981802 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051679.8568728 INFO: cpd05178 not found in model!\n",
      "1723051680.0905564 INFO: cpd05178 not found in model!\n",
      "1723051680.0926986 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051683.232244 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[283 'R2_C_D1.7.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:03:30+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '9c33cba82d764ca011d641f23ee10154'\n",
      " 14460014 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051683.7393305 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051683.9213586 INFO: metabolites 1144\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051684.5674322 INFO: reactions 1124\n",
      "1723051684.7838926 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051685.4412184 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/488/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051688.331442 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051688.6548157 INFO: cpd05178 not found in model!\n",
      "1723051689.0327716 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051689.055613 INFO: cpd05178 not found in model!\n",
      "1723051689.0583632 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051689.6708486 INFO: cpd05178 not found in model!\n",
      "1723051689.6728122 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[48 'R2_B_D2.52.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:17:01+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '5811f351e78fb5943646b2e477c56d5d'\n",
      " 10444146 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051691.334372 INFO: metabolites 1012\n",
      "1723051691.9242537 INFO: reactions 974\n",
      "1723051692.08646 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051692.6782348 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/582/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051693.263786 INFO: cpd05178 not found in model!\n",
      "1723051693.2657862 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051694.07175 INFO: cpd05178 not found in model!\n",
      "1723051694.07402 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051695.4241228 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051695.9040253 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051696.450951 INFO: cpd05178 not found in model!\n",
      "1723051696.4530528 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051699.8380816 INFO: cpd05178 not found in model!\n",
      "1723051699.8406756 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_B_D2.54.pg.G.D GF: 11731 Prob: 3630 Other: 2993\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051700.8523538 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051703.1124249 INFO: Expansion time:Dlac:0.6673878920009884\n",
      "1723051703.1147203 INFO: Filtered count:0 out of 1492\n",
      "1723051704.0529385 INFO: Expansion time:Dlac.O2:0.9369936130005954\n",
      "1723051704.055513 INFO: Filtered count:0 out of 1492\n",
      "1723051704.9970367 INFO: Expansion time:Fum:0.9400140800007648\n",
      "1723051704.9996972 INFO: Filtered count:0 out of 1492\n",
      "1723051705.3768966 INFO: cpd05178 not found in model!\n",
      "1723051705.3796797 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051705.546787 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051705.9287498 INFO: Expansion time:Fum.O2:0.9275670429997263\n",
      "1723051705.931385 INFO: Filtered count:0 out of 1492\n",
      "1723051706.0555465 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[287 'R2_A_D1.50.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:04:20+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '4f15fdf00b18251d1f9f8c4ecdc67e3b'\n",
      " 8855696 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051706.8704176 INFO: Expansion time:Glc:0.9374267279999913\n",
      "1723051706.8729117 INFO: Filtered count:0 out of 1492\n",
      "1723051707.088857 INFO: metabolites 1067\n",
      "1723051707.5238264 INFO: cpd05178 not found in model!\n",
      "1723051707.7538688 INFO: cpd05178 not found in model!\n",
      "1723051707.7561164 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051707.7576602 INFO: reactions 1035\n",
      "1723051707.9238236 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[215 'SF2_A_D2.28.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:50:08+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '6aca5e958be6670d88e44c9cc56c27bf'\n",
      " 15053566 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051708.5632803 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/260/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051708.8169374 INFO: metabolites 1097\n",
      "1723051709.5195177 INFO: reactions 1060\n",
      "1723051709.691419 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051710.1778767 INFO: Expansion time:Glc.DMSO:3.3034898320001957\n",
      "1723051710.1803803 INFO: Filtered count:0 out of 1492\n",
      "1723051710.3008854 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/431/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051710.6989121 INFO: cpd05178 not found in model!\n",
      "1723051710.7012775 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[276 'R1_B_D1.28.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:02:11+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '8b3e1ea493be9f65c2c3607b14d34142'\n",
      " 11412972 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051711.0775752 INFO: Expansion time:Glc.O2:0.8957408320002287\n",
      "1723051711.0800798 INFO: Filtered count:0 out of 1492\n",
      "1723051711.1229818 INFO: metabolites 1002\n",
      "1723051711.5864184 INFO: cpd05178 not found in model!\n",
      "1723051711.5886278 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051711.6187015 INFO: reactions 975\n",
      "1723051711.7424374 INFO: cpd05178 not found in model!\n",
      "1723051711.7447486 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051711.7809253 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051711.9757955 INFO: Expansion time:Glc.TMAO:0.8942876320015785\n",
      "1723051711.9783916 INFO: Filtered count:0 out of 1492\n",
      "1723051712.0089936 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051712.1312563 INFO: cpd05178 not found in model!\n",
      "1723051712.1335628 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051712.3281507 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/174/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051712.7192235 INFO: cpd05178 not found in model!\n",
      "1723051712.7220592 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051712.7316363 INFO: Expansion time:Pyr:0.7516516109990334\n",
      "1723051712.7336721 INFO: Filtered count:0 out of 1492\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D1.7.pg.G.D GF: 11444 Prob: 3938 Other: 2972\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051713.3852642 INFO: Expansion time:Pyr.DMSO:0.6502618760005134\n",
      "1723051713.3874218 INFO: Filtered count:0 out of 1492\n",
      "1723051713.994962 INFO: cpd05178 not found in model!\n",
      "1723051714.0331638 INFO: Expansion time:Pyr.NO:0.6445817759995407\n",
      "1723051714.0354507 INFO: Filtered count:0 out of 1492\n",
      "1723051714.2167811 INFO: cpd05178 not found in model!\n",
      "1723051714.2190795 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051714.6726449 INFO: Expansion time:Pyr.NO2:0.636007752000296\n",
      "1723051714.6748812 INFO: Filtered count:0 out of 1492\n",
      "1723051715.3048165 INFO: Expansion time:Pyr.NO3:0.6287647549997928\n",
      "1723051715.3069668 INFO: Filtered count:0 out of 1492\n",
      "1723051715.9464066 INFO: Expansion time:Pyr.O2:0.6382988490004209\n",
      "1723051715.9485168 INFO: Filtered count:0 out of 1492\n",
      "1723051716.589837 INFO: Expansion time:Pyr.SO3:0.6401394420008728\n",
      "1723051716.5920098 INFO: Filtered count:0 out of 1492\n",
      "1723051717.1300619 INFO: cpd05178 not found in model!\n",
      "1723051717.1326451 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051717.238973 INFO: Expansion time:Pyr.SO4:0.6458094670015271\n",
      "1723051717.2411983 INFO: Filtered count:0 out of 1492\n",
      "1723051717.886744 INFO: Expansion time:Pyr.TMAO:0.6441794599995774\n",
      "1723051717.8889241 INFO: Filtered count:0 out of 1492\n",
      "1723051718.8597956 INFO: Objective with gapfill database:117.4282448552249; min objective:0.01\n",
      "1723051719.0395088 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051719.3431973 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051720.4501843 INFO: Expansion time:empty:2.560124421999717\n",
      "1723051720.452305 INFO: Filtered count:0 out of 1492\n",
      "1723051720.650736 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051721.6640084 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051722.099535 INFO: cpd05178 not found in model!\n",
      "1723051722.102448 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.52.pg.G.D GF: 11915 Prob: 3182 Other: 3253\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051724.366513 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051724.6823483 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051726.7283306 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051727.5550385 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051729.5534456 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051730.035812 INFO: cpd05178 not found in model!\n",
      "1723051730.038058 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051730.2469733 INFO: cpd05178 not found in model!\n",
      "1723051730.2491949 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051730.6350188 INFO: Expansion time:Ac.O2:0.6450093880011991\n",
      "1723051730.6372216 INFO: Filtered count:0 out of 1617\n",
      "1723051731.5615315 INFO: Expansion time:Etho.O2:0.9229563720000442\n",
      "1723051731.5641525 INFO: Filtered count:0 out of 1617\n",
      "1723051732.6571665 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051734.4342437 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051734.4594142 INFO: Expansion time:For.O2:2.8937953040003777\n",
      "1723051734.4621 INFO: Filtered count:0 out of 1617\n",
      "1723051734.9450178 INFO: cpd05178 not found in model!\n",
      "1723051734.947637 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.50.pg.G.D GF: 11659 Prob: 3650 Other: 3045\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051735.3849115 INFO: Expansion time:Glyc.O2:0.9213245970004209\n",
      "1723051735.3874416 INFO: Filtered count:0 out of 1617\n",
      "1723051736.3153408 INFO: Expansion time:Pyr:0.9264231010001822\n",
      "1723051736.3180141 INFO: Filtered count:0 out of 1617\n",
      "1723051737.2409928 INFO: Expansion time:Pyr.DMSO:0.9214645470001415\n",
      "1723051737.2436578 INFO: Filtered count:0 out of 1617\n",
      "1723051737.2625043 INFO: cpd05178 not found in model!\n",
      "1723051737.2645547 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051738.1668103 INFO: Expansion time:Pyr.NO:0.9215468720012723\n",
      "1723051738.1694887 INFO: Filtered count:0 out of 1617\n",
      "1723051739.0923443 INFO: Expansion time:Pyr.NO2:0.921267159999843\n",
      "1723051739.0950513 INFO: Filtered count:0 out of 1617\n",
      "1723051739.9844975 INFO: cpd05178 not found in model!\n",
      "1723051739.9875405 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051740.0134728 INFO: Expansion time:Pyr.NO3:0.9169116119992395\n",
      "1723051740.0160515 INFO: Filtered count:0 out of 1617\n",
      "1723051740.059908 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D1.28.pg.G.D GF: 12059 Prob: 3198 Other: 3095\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051740.464077 INFO: cpd05178 not found in model!\n",
      "1723051740.466675 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051740.9583328 INFO: Expansion time:Pyr.O2:0.9407805089995236\n",
      "1723051740.960909 INFO: Filtered count:0 out of 1617\n",
      "1723051741.8983755 INFO: Expansion time:Pyr.SO3:0.9360254660005012\n",
      "1723051741.9009867 INFO: Filtered count:0 out of 1617\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051744.4212406 INFO: cpd05178 not found in model!\n",
      "1723051744.424007 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051744.6839242 INFO: Expansion time:Fum:0.6253327850008645\n",
      "1723051744.6859944 INFO: Filtered count:0 out of 1380\n",
      "1723051744.7003524 INFO: Expansion time:Pyr.SO4:2.797880604000966\n",
      "1723051744.7030215 INFO: Filtered count:0 out of 1617\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_D2.28.pg.G.D GF: 11538 Prob: 3806 Other: 3008\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051745.147136 INFO: cpd05178 not found in model!\n",
      "1723051745.3023129 INFO: Expansion time:Fum.O2:0.6151805949994014\n",
      "1723051745.304378 INFO: Filtered count:0 out of 1380\n",
      "1723051745.641248 INFO: Expansion time:Pyr.TMAO:0.9366855770003895\n",
      "1723051745.6438906 INFO: Filtered count:0 out of 1617\n",
      "1723051745.9363632 INFO: Expansion time:LLac.O2:0.6308263640003133\n",
      "1723051745.9384482 INFO: Filtered count:0 out of 1380\n",
      "1723051746.5630674 INFO: cpd05178 not found in model!\n",
      "1723051746.5633867 INFO: Expansion time:LLac.SO4:0.6237703770002554\n",
      "1723051746.5651748 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051746.564485 INFO: Expansion time:empty:0.9191914749990246\n",
      "1723051746.5655255 INFO: Filtered count:0 out of 1380\n",
      "1723051746.5671136 INFO: Filtered count:0 out of 1617\n",
      "1723051747.1986544 INFO: Expansion time:LLac.SO4.H2:0.6315748670003813\n",
      "1723051747.2007751 INFO: Filtered count:0 out of 1380\n",
      "1723051747.8385508 INFO: Expansion time:Llac:0.6365030769993609\n",
      "1723051747.840786 INFO: Filtered count:0 out of 1380\n",
      "1723051748.468238 INFO: Expansion time:Pyr:0.6262437719997251\n",
      "1723051748.4702947 INFO: Filtered count:0 out of 1380\n",
      "1723051749.1029863 INFO: Expansion time:Pyr.DMSO:0.631439008999223\n",
      "1723051749.105272 INFO: Filtered count:0 out of 1380\n",
      "1723051749.9175527 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051750.0066621 INFO: Expansion time:Pyr.NO:0.9000159750012244\n",
      "1723051750.008992 INFO: Filtered count:0 out of 1380\n",
      "1723051750.5341785 INFO: cpd05178 not found in model!\n",
      "1723051750.622671 INFO: cpd05178 not found in model!\n",
      "1723051750.856823 INFO: cpd05178 not found in model!\n",
      "1723051750.8590255 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051750.9527092 INFO: Expansion time:Pyr.NO2:0.9421719999991183\n",
      "1723051750.955279 INFO: Filtered count:0 out of 1380\n",
      "1723051752.7851424 INFO: cpd05178 not found in model!\n",
      "1723051752.7875345 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051753.2011116 INFO: cpd05178 not found in model!\n",
      "1723051753.2037256 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051753.2677798 INFO: cpd05178 not found in model!\n",
      "1723051753.5015943 INFO: cpd05178 not found in model!\n",
      "1723051753.503909 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051753.7598522 INFO: cpd05178 not found in model!\n",
      "1723051754.1068542 INFO: cpd05178 not found in model!\n",
      "1723051754.1094344 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051754.6765108 INFO: Expansion time:Pyr.NO3:3.7197644829993806\n",
      "1723051754.678967 INFO: Filtered count:0 out of 1380\n",
      "1723051755.701541 INFO: Expansion time:Pyr.O2:1.0210841469997831\n",
      "1723051755.7041416 INFO: Filtered count:0 out of 1380\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[239 'R1_B_D1.32.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:54:41+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'af6a87a14ac7059350f7f383d9ea2430'\n",
      " 11280789 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051756.7315643 INFO: Expansion time:Pyr.SO3:1.0259310210003605\n",
      "1723051756.7340124 INFO: Filtered count:0 out of 1380\n",
      "1723051757.1860647 INFO: metabolites 1206\n",
      "1723051757.2621565 INFO: Objective with gapfill database:138.715353354611; min objective:0.01\n",
      "1723051757.6357806 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051757.6894825 INFO: Expansion time:Pyr.SO4:0.954091370000242\n",
      "1723051757.6916928 INFO: Filtered count:0 out of 1380\n",
      "1723051758.0875838 INFO: reactions 1188\n",
      "1723051758.3002403 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051758.3236122 INFO: Expansion time:Pyr.TMAO:0.6307661190003273\n",
      "1723051758.3256505 INFO: Filtered count:0 out of 1380\n",
      "1723051758.7697883 INFO: cpd05178 not found in model!\n",
      "1723051758.7720528 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051758.953161 INFO: Expansion time:empty:0.6263059949997114\n",
      "1723051758.9553912 INFO: Filtered count:0 out of 1380\n",
      "1723051759.0509944 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/52/1\n",
      "[188 'R2_restored_C_black.22.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:44:32+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '96a431c016f4c512c1856463cc0b9f1b'\n",
      " 12954028 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051759.8917508 INFO: metabolites 1013\n",
      "1723051760.3050082 INFO: Expansion time:Ac.O2:0.6695972199995595\n",
      "1723051760.306887 INFO: Filtered count:0 out of 1464\n",
      "1723051760.5647352 INFO: reactions 976\n",
      "1723051760.6808617 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051760.960531 INFO: Expansion time:Etho.O2:0.6525007100008224\n",
      "1723051760.962776 INFO: Filtered count:0 out of 1464\n",
      "1723051761.0728478 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/264/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051761.6123955 INFO: Expansion time:For.O2:0.6483410660002846\n",
      "1723051761.6145432 INFO: Filtered count:0 out of 1464\n",
      "1723051761.811606 INFO: cpd05178 not found in model!\n",
      "1723051762.0493445 INFO: cpd05178 not found in model!\n",
      "1723051762.0515738 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051762.2608445 INFO: Expansion time:Pyr:0.6451332999986334\n",
      "1723051762.2629821 INFO: Filtered count:0 out of 1464\n",
      "1723051762.9033716 INFO: Expansion time:Pyr.DMSO:0.6391984130004857\n",
      "1723051762.9054844 INFO: Filtered count:0 out of 1464\n",
      "1723051763.8050156 INFO: Expansion time:Pyr.NO:0.8983331060007913\n",
      "1723051763.807476 INFO: Filtered count:0 out of 1464\n",
      "1723051764.3767776 INFO: cpd05178 not found in model!\n",
      "1723051764.3790355 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051764.7117808 INFO: Expansion time:Pyr.NO2:0.9028662000000622\n",
      "1723051764.713969 INFO: Filtered count:0 out of 1464\n",
      "1723051765.5994635 INFO: Expansion time:Pyr.NO3:0.884139783000137\n",
      "1723051765.6013205 INFO: Filtered count:0 out of 1464\n",
      "1723051767.438268 INFO: cpd05178 not found in model!\n",
      "1723051767.4405415 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051767.8136752 INFO: Expansion time:Pyr.O2:2.210642807000113\n",
      "1723051767.815617 INFO: Filtered count:0 out of 1464\n",
      "1723051768.473497 INFO: Expansion time:Pyr.SO3:0.6567060029992717\n",
      "1723051768.4755852 INFO: Filtered count:0 out of 1464\n",
      "1723051769.1133306 INFO: Expansion time:Pyr.SO4:0.6365181729997857\n",
      "1723051769.1156483 INFO: Filtered count:0 out of 1464\n",
      "1723051769.7449634 INFO: Expansion time:Pyr.TMAO:0.6281182410002657\n",
      "1723051769.7468312 INFO: Filtered count:0 out of 1464\n",
      "1723051770.3739007 INFO: Expansion time:empty:0.6259410329985258\n",
      "1723051770.3758402 INFO: Filtered count:0 out of 1464\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051772.3881197 INFO: Expansion time:Etho.O2:0.8948797770008241\n",
      "1723051772.3907957 INFO: Filtered count:0 out of 1483\n",
      "1723051773.277829 INFO: Expansion time:For.O2:0.8855939450004371\n",
      "1723051773.2805257 INFO: Filtered count:0 out of 1483\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[191 'R2A_A_H2O.22.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:45:10+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'f9641cedee7e4b12ea940b6803e3e301'\n",
      " 7517495 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051774.0227823 INFO: metabolites 1074\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[160 'R1_C_D2.40.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:39:04+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ba43991bc341214c8618ed2619e1d4e7'\n",
      " 7657370 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051774.5270398 INFO: metabolites 952\n",
      "1723051774.652822 INFO: reactions 1025\n",
      "1723051774.8188515 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051775.0021677 INFO: reactions 909\n",
      "1723051775.1660612 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051775.4065037 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/250/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051775.7595706 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/74/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051776.049097 INFO: Expansion time:Fum.O2:2.7671609990011348\n",
      "1723051776.051628 INFO: Filtered count:0 out of 1483\n",
      "1723051776.9335566 INFO: Expansion time:empty:0.8805256579998968\n",
      "1723051776.9360154 INFO: Filtered count:0 out of 1483\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[307 'R2_restored_H2O.017.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:08:38+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '75617bfce2102b4f7840c992a140f2fb'\n",
      " 11793908 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051778.1044989 INFO: metabolites 1006\n",
      "1723051778.7243996 INFO: reactions 906\n",
      "1723051778.8849764 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[299 'R2_restored_DShore.18.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:06:56+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '29287b67bf954a87964af06de99eeb51'\n",
      " 12034451 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051779.2427995 INFO: cpd05178 not found in model!\n",
      "1723051779.4337058 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/410/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051779.5342808 INFO: metabolites 1084\n",
      "1723051779.5772376 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051779.5966992 INFO: cpd05178 not found in model!\n",
      "1723051779.5994902 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051779.9433997 INFO: reactions 1023\n",
      "1723051780.070512 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051780.0870643 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051780.4813893 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/355/1\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051781.5710928 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051781.8692129 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051781.9831398 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051782.1714673 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "Getting tests from attributes\n",
      "[148 'R1_B_H2O.17.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:36:50+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '5e39ce4fe91052f7deeca51e4ff60b9e'\n",
      " 7859638 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051783.5001345 INFO: metabolites 848\n",
      "1723051783.8076057 INFO: reactions 727\n",
      "1723051783.8955173 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051784.4366307 INFO: cpd05178 not found in model!\n",
      "1723051784.4388003 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051784.5446897 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051785.0273974 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051785.6081183 INFO: cpd05178 not found in model!\n",
      "1723051785.610305 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051785.677127 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[88 'R2_B_D2.44.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:24:55+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '7a5b82a46fff1beacbe6499724a584d4'\n",
      " 11391831 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051785.8050432 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/8/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051786.1332655 INFO: metabolites 1081\n",
      "1723051786.6713753 INFO: cpd05178 not found in model!\n",
      "1723051786.6736834 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051786.6903036 INFO: reactions 994\n",
      "1723051786.8052678 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051787.1995034 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/216/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051787.439773 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051788.3492258 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051788.9963417 INFO: cpd05178 not found in model!\n",
      "1723051788.9990876 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051789.588937 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051790.1041958 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051790.7063353 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051792.4833062 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051792.9209177 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051793.3300965 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051793.8219001 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051795.7412348 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051796.064092 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051796.5219038 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051797.5753748 INFO: cpd05178 not found in model!\n",
      "1723051797.5778785 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_D2.40.pg.G.D GF: 12214 Prob: 2798 Other: 3338\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051798.1511292 INFO: cpd05178 not found in model!\n",
      "1723051798.1536906 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051798.1534457 INFO: cpd05178 not found in model!\n",
      "1723051798.1556857 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051798.1876762 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051798.2241664 INFO: cpd05178 not found in model!\n",
      "1723051798.226755 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_C_black.22.pg.G.D GF: 11280 Prob: 4120 Other: 2950\n",
      "R1_B_D1.32.pg.G.D GF: 10777 Prob: 4816 Other: 2759\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051798.577369 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_H2O.22.pg.G.D GF: 11710 Prob: 3678 Other: 2962\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051801.6170995 INFO: cpd05178 not found in model!\n",
      "1723051801.6194205 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051802.1145205 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051803.513541 INFO: cpd05178 not found in model!\n",
      "1723051803.5158327 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051806.567928 INFO: cpd05178 not found in model!\n",
      "1723051806.5700705 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051807.1196122 INFO: cpd05178 not found in model!\n",
      "1723051807.1216402 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051807.176019 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_H2O.017.pg.G.D GF: 11841 Prob: 3472 Other: 3039\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051807.7335532 INFO: cpd05178 not found in model!\n",
      "1723051807.7362165 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_H2O.17.pg.G.D GF: 12521 Prob: 2502 Other: 3337\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051808.8011208 INFO: Objective with gapfill database:214.4908789544612; min objective:0.01\n",
      "1723051808.9472017 INFO: cpd05178 not found in model!\n",
      "1723051808.9500427 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.18.pg.G.D GF: 11708 Prob: 3602 Other: 3044\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051809.739453 INFO: cpd05178 not found in model!\n",
      "1723051809.7417026 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051810.2972617 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051812.055013 INFO: cpd05178 not found in model!\n",
      "1723051812.0575113 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051812.340891 INFO: Objective with gapfill database:155.9172443131437; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.44.pg.G.D GF: 11863 Prob: 3326 Other: 3165\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051815.8170805 INFO: cpd05178 not found in model!\n",
      "1723051815.8192942 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051817.7552958 INFO: cpd05178 not found in model!\n",
      "1723051817.9771245 INFO: cpd05178 not found in model!\n",
      "1723051817.979496 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051818.9898407 INFO: Objective with gapfill database:172.9795640774827; min objective:0.01\n",
      "1723051821.2730966 INFO: Expansion time:Ac.O2:4.68803894300072\n",
      "1723051821.2751591 INFO: Filtered count:0 out of 1344\n",
      "1723051822.841627 INFO: Expansion time:Etho.O2:1.5652665510006045\n",
      "1723051822.843727 INFO: Filtered count:0 out of 1344\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051824.5708199 INFO: Expansion time:LLac.O2:1.7258721070011234\n",
      "1723051824.5729434 INFO: Filtered count:0 out of 1344\n",
      "1723051824.794323 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051825.3849473 INFO: cpd05178 not found in model!\n",
      "1723051825.3871794 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051826.3517866 INFO: cpd05178 not found in model!\n",
      "1723051826.7002964 INFO: cpd05178 not found in model!\n",
      "1723051826.7030807 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051827.850282 INFO: Objective with gapfill database:151.32043862867346; min objective:0.01\n",
      "1723051828.3916218 INFO: Expansion time:Ac.NO2:4.175040300000546\n",
      "1723051828.3940785 INFO: Filtered count:0 out of 1698\n",
      "1723051828.439785 INFO: cpd05178 not found in model!\n",
      "1723051828.6686134 INFO: cpd05178 not found in model!\n",
      "1723051828.6709313 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051828.8634799 INFO: Expansion time:LLac.SO4:4.28933246500128\n",
      "1723051828.8655045 INFO: Filtered count:0 out of 1344\n",
      "1723051829.8839233 INFO: Expansion time:Ac.O2:1.4884058619991265\n",
      "1723051829.8866923 INFO: Filtered count:0 out of 1698\n",
      "1723051830.895547 INFO: Objective with gapfill database:226.65910045768666; min objective:0.01\n",
      "1723051831.1426942 INFO: Expansion time:LLac.SO4.H2:2.2754608939994796\n",
      "1723051831.1446996 INFO: Filtered count:0 out of 1344\n",
      "1723051831.5362146 INFO: Expansion time:Etho.O2:1.6475669739993464\n",
      "1723051831.5387347 INFO: Filtered count:0 out of 1698\n",
      "1723051832.3999574 INFO: cpd05178 not found in model!\n",
      "1723051832.4019313 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051832.8596478 INFO: Expansion time:For.NO2:1.3193768199998885\n",
      "1723051832.862275 INFO: Filtered count:0 out of 1698\n",
      "1723051834.0869887 INFO: Expansion time:For.O2:1.2232905299988488\n",
      "1723051834.0892007 INFO: Filtered count:0 out of 1698\n",
      "1723051835.874004 INFO: Expansion time:Fum:1.7833646839990251\n",
      "1723051835.876074 INFO: Filtered count:0 out of 1698\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn00650_c0\n",
      "Failed:>rxn10094_c0\n",
      "Failed:>rxn03167_c0\n",
      "Failed:<rxn01792_c0\n",
      "Failed:>rxn02312_c0\n",
      "Failed:>rxn07189_c0\n",
      "Failed:>rxn02508_c0\n",
      "Failed:>rxn05005_c0\n",
      "Failed:>rxn00048_c0\n",
      "Failed:<rxn03034_c0\n",
      "Failed:>rxn02475_c0\n",
      "Failed:>rxn10469_c0\n",
      "Failed:>rxn09041_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051837.7587557 INFO: cpd05178 not found in model!\n",
      "1723051837.7614384 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn09039_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051837.818423 INFO: Expansion time:Fum.O2:1.9409973140009242\n",
      "1723051837.8205733 INFO: Filtered count:0 out of 1698\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03108_c0\n",
      "Failed:>rxn09045_c0\n",
      "Failed:>rxn03146_c0\n",
      "Failed:>rxn03893_c0\n",
      "Failed:>rxn10596_c0\n",
      "Failed:>rxn02405_c0\n",
      "Failed:<rxn00791_c0\n",
      "Failed:>rxn03150_c0\n",
      "Failed:>rxn03538_c0\n",
      "Failed:>rxn09310_c0\n",
      "Failed:>rxn01213_c0\n",
      "Failed:>rxn05153_c0\n",
      "Failed:>rxn01361_c0\n",
      "Failed:>rxn05618_c0\n",
      "Failed:>rxn03136_c0\n",
      "Failed:<rxn00502_c0\n",
      "Failed:>rxn03137_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051839.6783147 INFO: cpd05178 not found in model!\n",
      "1723051839.6810932 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn05006_c0\n",
      "Failed:<rxn13726_c0\n",
      "Failed:>rxn00533_c0\n",
      "Failed:>rxn03841_c0\n",
      "Failed:<rxn06865_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn12239_c0\n",
      "Failed:>rxn01257_c0\n",
      "Failed:>rxn02213_c0\n",
      "Failed:>rxn03393_c0\n",
      "Failed:>rxn10611_c0\n",
      "Failed:>rxn00599_c0\n",
      "[304 'R2_A_D1.32.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:07:56+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'd37bfe11a443c65275639e0703e63885'\n",
      " 10599174 None]\n",
      "Failed:>rxn02322_c0\n",
      "Failed:>rxn11702_c0\n",
      "Failed:>rxn03909_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051841.3222632 INFO: Expansion time:Fum:0.9405405420002353\n",
      "1723051841.3247547 INFO: Filtered count:0 out of 995\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03439_c0\n",
      "Failed:>rxn10468_c0\n",
      "Failed:<rxn05470_c0\n",
      "Failed:>rxn02897_c0\n",
      "Failed:>rxn10436_c0\n",
      "Failed:<rxn02056_c0\n",
      "Failed:>rxn02277_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051842.0751314 INFO: cpd05178 not found in model!\n",
      "1723051842.0772314 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051842.0951617 INFO: Expansion time:Pyr:4.2731391719989915\n",
      "1723051842.0972278 INFO: Filtered count:0 out of 1698\n",
      "1723051842.2054188 INFO: Expansion time:Fum.O2:0.8791317710001749\n",
      "1723051842.207911 INFO: Filtered count:0 out of 995\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn02185_c0\n",
      "Failed:>rxn00912_c0\n",
      "Failed:>rxn11858_c0\n",
      "Failed:>rxn01466_c0\n",
      "Failed:>rxn02288_c0\n",
      "Failed:>rxn02831_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051842.8895757 INFO: metabolites 1080\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn01258_c0\n",
      "Failed:>rxn03901_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051843.0596652 INFO: Expansion time:empty:0.8503400919998967\n",
      "1723051843.0619328 INFO: Filtered count:0 out of 995\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn02832_c0\n",
      "Failed:>rxn02297_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051843.3308263 INFO: reactions 988\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00239_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051843.4466038 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051843.501923 INFO: Expansion time:Pyr.DMSO:1.4032457570010592\n",
      "1723051843.5042014 INFO: Filtered count:0 out of 1698\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03130_c0\n",
      "Failed:>rxn02507_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051843.843924 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn01964_c0\n",
      "188040/395/1\n",
      "Failed:>rxn02341_c0\n",
      "Failed:>rxn03182_c0\n",
      "Failed:>rxn00224_c0\n",
      "Failed:<rxn02474_c0\n",
      "Failed:<rxn02929_c0\n",
      "Failed:<rxn00202_c0\n",
      "Failed:<rxn02914_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051844.8282259 INFO: cpd05178 not found in model!\n",
      "1723051844.8304946 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn00686_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051844.8643322 INFO: Expansion time:Pyr.NO:1.3586544829995546\n",
      "1723051844.8665433 INFO: Filtered count:0 out of 1698\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn01420_c0\n",
      "Failed:>rxn13752_c0\n",
      "Failed:>rxn11859_c0\n",
      "Failed:>rxn03395_c0\n",
      "Failed:>rxn00293_c0\n",
      "Failed:>rxn10595_c0\n",
      "Failed:>rxn08233_c0\n",
      "Failed:>rxn39886_c0\n",
      "Failed:<rxn48586_c0\n",
      "Failed:>rxn20643_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051846.0374472 INFO: Expansion time:Llac:14.891532070001631\n",
      "1723051846.0397334 INFO: Filtered count:84 out of 1344\n",
      "1723051846.6730347 INFO: Expansion time:Pyr.NO2:1.8049984270001005\n",
      "1723051846.6753588 INFO: Filtered count:0 out of 1698\n",
      "1723051847.5689356 INFO: Expansion time:Pyr:1.5274937260001025\n",
      "1723051847.5713272 INFO: Filtered count:84 out of 1344\n",
      "1723051848.626959 INFO: Expansion time:Pyr.NO3:1.9501514749990747\n",
      "1723051848.629395 INFO: Filtered count:0 out of 1698\n",
      "1723051849.3096354 INFO: Expansion time:Pyr.DMSO:1.7368667440005083\n",
      "1723051849.3120315 INFO: Filtered count:84 out of 1344\n",
      "1723051850.4864395 INFO: Expansion time:Pyr.NO:1.1730167830010032\n",
      "1723051850.4888294 INFO: Filtered count:84 out of 1344\n",
      "1723051850.617278 INFO: Expansion time:Pyr.O2:1.98633090399926\n",
      "1723051850.619513 INFO: Filtered count:0 out of 1698\n",
      "1723051851.8704617 INFO: Expansion time:Pyr.NO2:1.3802133000008325\n",
      "1723051851.8728788 INFO: Filtered count:84 out of 1344\n",
      "1723051853.365118 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051853.4021525 INFO: Expansion time:Pyr.NO3:1.52787176199854\n",
      "1723051853.4045622 INFO: Filtered count:84 out of 1344\n",
      "1723051855.4014776 INFO: Expansion time:Pyr.O2:1.9954841950002447\n",
      "1723051855.4038818 INFO: Filtered count:84 out of 1344\n",
      "1723051856.070658 INFO: Expansion time:Pyr.SO3:5.449745640000401\n",
      "1723051856.0728772 INFO: Filtered count:0 out of 1698\n",
      "1723051856.2053816 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051857.2766707 INFO: Expansion time:Pyr.SO4:1.2023339699990174\n",
      "1723051857.2789686 INFO: Filtered count:0 out of 1698\n",
      "1723051857.553442 INFO: Expansion time:Pyr.SO3:2.1481865360001393\n",
      "1723051857.5558293 INFO: Filtered count:84 out of 1344\n",
      "1723051858.5961924 INFO: Expansion time:Pyr.TMAO:1.315813368999443\n",
      "1723051858.5981073 INFO: Filtered count:0 out of 1698\n",
      "1723051859.0518882 INFO: Expansion time:Pyr.SO4:1.4946979379983532\n",
      "1723051859.054164 INFO: Filtered count:84 out of 1344\n",
      "1723051859.2103927 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051859.268047 INFO: Expansion time:empty:0.6686968739995791\n",
      "1723051859.2699342 INFO: Filtered count:0 out of 1698\n",
      "1723051859.9293447 INFO: cpd05178 not found in model!\n",
      "1723051860.1474524 INFO: cpd05178 not found in model!\n",
      "1723051860.14973 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051860.3094456 INFO: Expansion time:Pyr.TMAO:1.25393221000013\n",
      "1723051860.3115273 INFO: Filtered count:84 out of 1344\n",
      "1723051860.9115832 INFO: Expansion time:empty:0.5988627560000168\n",
      "1723051860.9135425 INFO: Filtered count:84 out of 1344\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[32 'R1_B_H2O.13.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:13:57+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'acbaf8b2470e0fd9f6f92311640530dc'\n",
      " 7167649 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051861.894595 INFO: metabolites 1004\n",
      "1723051862.3909333 INFO: reactions 932\n",
      "1723051862.5329454 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051862.9480863 INFO: cpd05178 not found in model!\n",
      "1723051862.9502404 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051863.05605 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/500/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051864.5618057 INFO: cpd05178 not found in model!\n",
      "1723051864.564075 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051864.8694031 INFO: cpd05178 not found in model!\n",
      "1723051865.0912566 INFO: cpd05178 not found in model!\n",
      "1723051865.0934281 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051866.6093693 INFO: cpd05178 not found in model!\n",
      "1723051866.6118784 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051866.8033516 INFO: cpd05178 not found in model!\n",
      "1723051867.0305653 INFO: cpd05178 not found in model!\n",
      "1723051867.0328484 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051868.076119 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051868.3634505 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[212 'R2A_C_D2.34.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:49:31+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'f5e2152a50618e214b5a694bd6a618fa'\n",
      " 15751056 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051869.7774382 INFO: cpd05178 not found in model!\n",
      "1723051869.8814688 INFO: metabolites 1197\n",
      "1723051870.017581 INFO: cpd05178 not found in model!\n",
      "1723051870.0196252 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051870.3318024 INFO: reactions 1152\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051870.4648876 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051870.9103568 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/418/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051871.7930148 INFO: cpd05178 not found in model!\n",
      "1723051871.7956975 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.32.pg.G.D GF: 11661 Prob: 3596 Other: 3097\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051873.1426759 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051876.8015766 INFO: cpd05178 not found in model!\n",
      "1723051876.8038235 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051879.8705614 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051880.2026799 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051880.9579654 INFO: cpd05178 not found in model!\n",
      "1723051880.9601657 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[200 'R1_B_D2.19.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:47:04+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '74bde44140213f514ae09c40b27f70d3'\n",
      " 13488457 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051882.0960066 INFO: metabolites 1087\n",
      "1723051882.663254 INFO: reactions 1052\n",
      "1723051882.7833154 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051883.339714 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051884.7230089 INFO: cpd05178 not found in model!\n",
      "1723051884.869903 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/326/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051884.9687724 INFO: cpd05178 not found in model!\n",
      "1723051884.9710252 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051885.2693484 INFO: cpd05178 not found in model!\n",
      "1723051885.271903 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_H2O.13.pg.G.D GF: 12248 Prob: 2844 Other: 3260\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051886.2415464 INFO: cpd05178 not found in model!\n",
      "1723051886.4837835 INFO: cpd05178 not found in model!\n",
      "1723051886.4859068 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051886.8397813 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051887.1660094 INFO: cpd05178 not found in model!\n",
      "1723051887.1682346 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[268 'R2_restored_DShore.38.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:00:37+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'fb6919e95d0b0428d04050e5153be37a'\n",
      " 9066066 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051890.5746348 INFO: metabolites 1107\n",
      "1723051890.6695457 INFO: Expansion time:For.O2:0.6301278260016261\n",
      "1723051890.6718388 INFO: Filtered count:0 out of 1398\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[180 'R1_B_D1.49.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:42:58+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '22403a45bc81e91e6091185e11e2c5be'\n",
      " 9838122 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051891.2907069 INFO: reactions 1045\n",
      "1723051891.3039362 INFO: Expansion time:Glc:0.6307868610001606\n",
      "1723051891.305865 INFO: Filtered count:0 out of 1398\n",
      "1723051891.4662662 INFO: metabolites 985\n",
      "1723051891.4823477 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051891.9280772 INFO: Expansion time:Glc.DMSO:0.6206679949991667\n",
      "1723051891.9302433 INFO: Filtered count:0 out of 1398\n",
      "1723051892.078708 INFO: reactions 940\n",
      "1723051892.2169557 INFO: Default biomass: [bio1]\n",
      "1723051892.2305682 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/136/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051892.574057 INFO: Expansion time:Glc.O2:0.642591989999346\n",
      "1723051892.5762844 INFO: Filtered count:0 out of 1398\n",
      "1723051892.785756 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/189/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051893.2151148 INFO: Expansion time:Glc.TMAO:0.637525699001344\n",
      "1723051893.2173982 INFO: Filtered count:0 out of 1398\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[136 'R1_A_D1.44.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:34:26+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'e9542ef5ff7a8301bc63b37554b718c4'\n",
      " 7586661 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051894.027649 INFO: metabolites 993\n",
      "1723051894.5450957 INFO: reactions 913\n",
      "1723051894.6860485 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051895.193512 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/562/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051895.8300037 INFO: Expansion time:LLac.O2:2.611327683000127\n",
      "1723051895.8322933 INFO: Filtered count:0 out of 1398\n",
      "1723051896.452987 INFO: Expansion time:Pyr:0.6192997289999767\n",
      "1723051896.4552183 INFO: Filtered count:0 out of 1398\n",
      "1723051897.0764022 INFO: Expansion time:Pyr.DMSO:0.619836485999258\n",
      "1723051897.0785868 INFO: Filtered count:0 out of 1398\n",
      "1723051897.3000722 INFO: cpd05178 not found in model!\n",
      "1723051897.551366 INFO: cpd05178 not found in model!\n",
      "1723051897.5534496 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051897.6989481 INFO: Expansion time:Pyr.NO:0.6190437219993328\n",
      "1723051897.7010922 INFO: Filtered count:0 out of 1398\n",
      "1723051897.972892 INFO: cpd05178 not found in model!\n",
      "1723051897.9751782 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051898.3226943 INFO: Expansion time:Pyr.NO2:0.6203721070014581\n",
      "1723051898.3249788 INFO: Filtered count:0 out of 1398\n",
      "1723051898.9562418 INFO: Expansion time:Pyr.NO3:0.6299290759998257\n",
      "1723051898.958607 INFO: Filtered count:0 out of 1398\n",
      "1723051899.5048592 INFO: Objective with gapfill database:160.21165232170236; min objective:0.01\n",
      "1723051899.5948188 INFO: Expansion time:Pyr.O2:0.634897803000058\n",
      "1723051899.596988 INFO: Filtered count:0 out of 1398\n",
      "1723051900.0322833 INFO: cpd05178 not found in model!\n",
      "1723051900.0342352 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051900.2362192 INFO: Expansion time:Pyr.SO3:0.6380416419997346\n",
      "1723051900.2384806 INFO: Filtered count:0 out of 1398\n",
      "1723051900.4108536 INFO: cpd05178 not found in model!\n",
      "1723051900.4134893 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.34.pg.G.D GF: 11534 Prob: 3776 Other: 3044\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051900.8759964 INFO: Expansion time:Pyr.SO4:0.636197161000382\n",
      "1723051900.8782291 INFO: Filtered count:0 out of 1398\n",
      "1723051901.5084908 INFO: Expansion time:Pyr.TMAO:0.6289753040000505\n",
      "1723051901.5106673 INFO: Filtered count:0 out of 1398\n",
      "1723051901.6297133 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051901.9083586 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051902.1360009 INFO: Expansion time:empty:0.6240209879997565\n",
      "1723051902.1381938 INFO: Filtered count:0 out of 1398\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051903.323939 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051904.2743676 INFO: cpd05178 not found in model!\n",
      "1723051904.2766197 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051905.0891895 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051905.8863199 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "[172 'R2A_C_D2.10b.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:41:28+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'fa2d25f1a063a3109f4cb929cc6cdd9f'\n",
      " 9371039 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051908.5387342 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051908.5712721 INFO: metabolites 908\n",
      "1723051908.9879355 INFO: reactions 830\n",
      "1723051909.1762226 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051909.522764 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/459/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051911.4555366 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051912.0159125 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051913.060781 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051913.572151 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051913.8847697 INFO: cpd05178 not found in model!\n",
      "1723051913.886932 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051915.6243799 INFO: Objective with gapfill database:230.19824669164336; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051916.8181155 INFO: cpd05178 not found in model!\n",
      "1723051916.8199337 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051917.6287568 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051917.8286328 INFO: cpd05178 not found in model!\n",
      "1723051918.0690153 INFO: cpd05178 not found in model!\n",
      "1723051918.0711718 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051918.7311745 INFO: cpd05178 not found in model!\n",
      "1723051918.733835 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.19.pg.G.D GF: 11369 Prob: 3806 Other: 3175\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051919.288447 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[128 'R1_B_D2.37b.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:32:57+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '2cd7b06a6b66dfd333ca9130b703ca24'\n",
      " 10711370 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051921.7737896 INFO: cpd05178 not found in model!\n",
      "1723051921.7759166 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051922.0788627 INFO: metabolites 1057\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D1.44.pg.G.D GF: 11836 Prob: 3474 Other: 3044\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051922.6396732 INFO: reactions 975\n",
      "1723051922.807617 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051923.3783944 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/452/1\n",
      "[156 'R2A_C_D2.45.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:38:18+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '721d3987d04a156364363f5d543493c7'\n",
      " 15703745 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051924.1475646 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051924.5468729 INFO: cpd05178 not found in model!\n",
      "1723051924.5492847 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051924.5713398 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051924.6458507 INFO: metabolites 1131\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D1.49.pg.G.D GF: 11582 Prob: 3618 Other: 3150\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051925.400906 INFO: reactions 1085\n",
      "1723051925.620929 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051926.369073 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/40/1\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051928.796862 INFO: cpd05178 not found in model!\n",
      "1723051928.7992623 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_DShore.38.pg.G.D GF: 11345 Prob: 4092 Other: 2917\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051929.6872633 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051931.417334 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051934.180232 INFO: cpd05178 not found in model!\n",
      "1723051934.416582 INFO: cpd05178 not found in model!\n",
      "1723051934.41877 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051934.4185734 INFO: cpd05178 not found in model!\n",
      "1723051934.420531 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051934.6186876 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051934.8114789 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051935.3123636 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051938.6034937 INFO: cpd05178 not found in model!\n",
      "1723051938.6058335 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051939.0977087 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051940.0836635 INFO: cpd05178 not found in model!\n",
      "1723051940.0857084 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.10b.pg.G.D GF: 11861 Prob: 3496 Other: 2997\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051940.4889426 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[52 'R2_A_H2O.45.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:17:48+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '4b4e540a2a2aeec0969c871d67674525'\n",
      " 10450932 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051941.5269165 INFO: metabolites 800\n",
      "1723051941.9681888 INFO: reactions 757\n",
      "1723051942.055424 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051942.316656 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051942.3464928 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/242/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051945.6280065 INFO: Expansion time:Dlac:0.6140718419992481\n",
      "1723051945.6301363 INFO: Filtered count:0 out of 1325\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051946.1627553 INFO: Expansion time:For.NO3:0.60488013199938\n",
      "1723051946.164726 INFO: Filtered count:0 out of 1262\n",
      "1723051946.2343645 INFO: Expansion time:Dlac.O2:0.603053300999818\n",
      "1723051946.2365353 INFO: Filtered count:0 out of 1325\n",
      "1723051946.3285193 INFO: cpd05178 not found in model!\n",
      "1723051946.3307874 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051946.7645245 INFO: Expansion time:For.O2:0.5987064580003789\n",
      "1723051946.7666483 INFO: Filtered count:0 out of 1262\n",
      "1723051946.8423228 INFO: Expansion time:Fum:0.6046405559991399\n",
      "1723051946.844484 INFO: Filtered count:0 out of 1325\n",
      "1723051946.929121 INFO: Expansion time:Dlac:0.9410509620010998\n",
      "1723051946.931778 INFO: Filtered count:0 out of 1494\n",
      "1723051947.3872793 INFO: Expansion time:Fum.O2:0.6194982050001272\n",
      "1723051947.3894134 INFO: Filtered count:0 out of 1262\n",
      "1723051947.4437904 INFO: Expansion time:Fum.O2:0.5981545689992345\n",
      "1723051947.4460669 INFO: Filtered count:0 out of 1325\n",
      "1723051947.8414567 INFO: Expansion time:Dlac.O2:0.9081670030009263\n",
      "1723051947.844127 INFO: Filtered count:0 out of 1494\n",
      "1723051948.042366 INFO: Expansion time:Pyr:0.5949888710001687\n",
      "1723051948.044602 INFO: Filtered count:0 out of 1325\n",
      "1723051948.2869804 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051948.6089358 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051948.636596 INFO: Expansion time:Pyr.DMSO:0.5908470140002464\n",
      "1723051948.6386738 INFO: Filtered count:0 out of 1325\n",
      "1723051948.7544 INFO: Expansion time:Fum:0.9087885320004716\n",
      "1723051948.756975 INFO: Filtered count:0 out of 1494\n",
      "1723051949.1197596 INFO: cpd05178 not found in model!\n",
      "1723051949.1289434 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723051949.2347972 INFO: Expansion time:Pyr.NO:0.5949149260013655\n",
      "1723051949.2370517 INFO: Filtered count:0 out of 1325\n",
      "1723051949.3480453 INFO: cpd05178 not found in model!\n",
      "1723051949.350219 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051949.5043445 INFO: cpd05178 not found in model!\n",
      "1723051949.5356295 INFO: cpd05178 not found in model!\n",
      "1723051949.765541 INFO: cpd05178 not found in model!\n",
      "1723051949.767752 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051949.8411171 INFO: Expansion time:Pyr.NO2:0.6028842989999248\n",
      "1723051949.8434193 INFO: Filtered count:0 out of 1325\n",
      "1723051949.8458793 INFO: cpd05178 not found in model!\n",
      "1723051949.8481421 INFO: Media compound: cpd05178 not found in model.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051950.0082054 INFO: Expansion time:Glc.O2:2.6175500839999586\n",
      "1723051950.0102558 INFO: Filtered count:0 out of 1262\n",
      "1723051950.4526289 INFO: Expansion time:Pyr.NO3:0.6079649919993244\n",
      "1723051950.4549446 INFO: Filtered count:0 out of 1325\n",
      "1723051950.6276696 INFO: Expansion time:empty:0.6162263089991029\n",
      "1723051950.629882 INFO: Filtered count:0 out of 1262\n",
      "1723051950.8131552 INFO: Expansion time:Etho.O2:0.7342739209998399\n",
      "1723051950.8153741 INFO: Filtered count:0 out of 1461\n",
      "1723051951.0591228 INFO: Expansion time:Pyr.O2:0.6029637139999977\n",
      "1723051951.0613189 INFO: Filtered count:0 out of 1325\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051951.5268965 INFO: Expansion time:Fum:0.710195556999679\n",
      "1723051951.5291946 INFO: Filtered count:0 out of 1461\n",
      "1723051951.6626854 INFO: Expansion time:Pyr.SO3:0.6000381109988666\n",
      "1723051951.6649163 INFO: Filtered count:0 out of 1325\n",
      "1723051952.2090974 INFO: Expansion time:Fum.O2:0.6785657250002259\n",
      "1723051952.211171 INFO: Filtered count:0 out of 1461\n",
      "1723051952.3404443 INFO: Expansion time:Fum.O2:3.581979375001538\n",
      "1723051952.3431647 INFO: Filtered count:0 out of 1494\n",
      "1723051952.3731186 INFO: cpd05178 not found in model!\n",
      "1723051952.375403 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051952.6126192 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051952.8887508 INFO: Expansion time:Pyr:0.676302341000337\n",
      "1723051952.8910747 INFO: Filtered count:0 out of 1461\n",
      "1723051952.8984663 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051953.2750118 INFO: Expansion time:Pyr:0.9303888020003797\n",
      "1723051953.2776747 INFO: Filtered count:0 out of 1494\n",
      "1723051953.3104167 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051953.536225 INFO: Expansion time:Pyr.DMSO:0.6438850750000711\n",
      "1723051953.5384197 INFO: Filtered count:0 out of 1461\n",
      "1723051954.1967916 INFO: Expansion time:Pyr.DMSO:0.9176125199992384\n",
      "1723051954.1995225 INFO: Filtered count:0 out of 1494\n",
      "1723051954.2173402 INFO: Expansion time:Pyr.NO:0.6777259990012681\n",
      "1723051954.219623 INFO: Filtered count:0 out of 1461\n",
      "1723051954.245228 INFO: Expansion time:Pyr.SO4:2.5791329679996124\n",
      "1723051954.2475164 INFO: Filtered count:0 out of 1325\n",
      "1723051954.8402307 INFO: Expansion time:Pyr.TMAO:0.5915813070005242\n",
      "1723051954.8425033 INFO: Filtered count:0 out of 1325\n",
      "1723051954.9352741 INFO: Expansion time:Pyr.NO2:0.7142424570010917\n",
      "1723051954.937433 INFO: Filtered count:0 out of 1461\n",
      "1723051955.1110425 INFO: Expansion time:Pyr.NO:0.9100533410000935\n",
      "1723051955.113797 INFO: Filtered count:0 out of 1494\n",
      "1723051955.4399579 INFO: Expansion time:empty:0.5962944969996897\n",
      "1723051955.442271 INFO: Filtered count:0 out of 1325\n",
      "1723051955.598754 INFO: Expansion time:Pyr.NO3:0.6600007629986067\n",
      "1723051955.6009474 INFO: Filtered count:0 out of 1461\n",
      "1723051955.6461592 INFO: cpd05178 not found in model!\n",
      "1723051955.648813 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.37b.pg.G.D "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051956.02488 INFO: Expansion time:Pyr.NO2:0.9094448720006767\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GF:"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051956.0273163 INFO: Filtered count:0 out of 1494\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " 11943 Prob: 3220 Other: 3189\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051956.2636154 INFO: Expansion time:Pyr.O2:0.6614319180007442\n",
      "1723051956.2657547 INFO: Filtered count:0 out of 1461\n",
      "1723051956.9501803 INFO: Expansion time:Pyr.NO3:0.9214459219983837\n",
      "1723051956.9527702 INFO: Filtered count:0 out of 1494\n",
      "1723051957.0299046 INFO: Expansion time:Pyr.SO3:0.7628035509987967\n",
      "1723051957.0321758 INFO: Filtered count:0 out of 1461\n",
      "1723051957.6484687 INFO: cpd05178 not found in model!\n",
      "1723051957.6505246 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051957.8948064 INFO: Expansion time:Pyr.O2:0.9405102589989838\n",
      "1723051957.89752 INFO: Filtered count:0 out of 1494\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051958.5449166 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051958.7271645 INFO: Expansion time:Etho.O2:0.7093934519998584\n",
      "1723051958.7291903 INFO: Filtered count:0 out of 1157\n",
      "1723051958.8362317 INFO: Expansion time:Pyr.SO3:0.9372027819990763\n",
      "1723051958.8389325 INFO: Filtered count:0 out of 1494\n",
      "1723051959.71924 INFO: Expansion time:Pyr:0.98844025300059\n",
      "1723051959.721582 INFO: Filtered count:0 out of 1157\n",
      "1723051959.7323632 INFO: Expansion time:Pyr.SO4:0.8919435940006224\n",
      "1723051959.735005 INFO: Filtered count:0 out of 1494\n",
      "1723051959.7973578 INFO: Expansion time:Pyr.SO4:2.7639112419983576\n",
      "1723051959.7993681 INFO: Filtered count:0 out of 1461\n",
      "1723051960.4536078 INFO: Expansion time:Pyr.TMAO:0.6531111059994146\n",
      "1723051960.4556425 INFO: Filtered count:0 out of 1461\n",
      "1723051960.6175163 INFO: Expansion time:Pyr.TMAO:0.8810826259996247\n",
      "1723051960.6198008 INFO: Filtered count:0 out of 1494\n",
      "1723051960.727539 INFO: Expansion time:Pyr.DMSO:1.0044026670002495\n",
      "1723051960.7300074 INFO: Filtered count:0 out of 1157\n",
      "1723051961.1434965 INFO: Expansion time:empty:0.686597363999681\n",
      "1723051961.145416 INFO: Filtered count:0 out of 1461\n",
      "1723051961.2679389 INFO: Expansion time:empty:0.6468784930002585\n",
      "1723051961.2701063 INFO: Filtered count:0 out of 1494\n",
      "1723051961.759703 INFO: Expansion time:Pyr.NO:1.0280731980001292\n",
      "1723051961.7620945 INFO: Filtered count:0 out of 1157\n",
      "1723051962.3328042 INFO: cpd05178 not found in model!\n",
      "1723051962.3350039 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051962.512515 INFO: cpd05178 not found in model!\n",
      "1723051962.5163186 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723051962.6506867 INFO: Expansion time:Pyr.NO2:0.8868522150005447\n",
      "1723051962.6530962 INFO: Filtered count:0 out of 1157\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.45.pg.G.D GF: 11380 Prob: 3932 Other: 3042\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051963.5336974 INFO: Expansion time:Pyr.NO3:0.8791758759998629\n",
      "1723051963.5359883 INFO: Filtered count:0 out of 1157\n",
      "1723051964.462234 INFO: Expansion time:Pyr.O2:0.924798815000031\n",
      "1723051964.4644766 INFO: Filtered count:0 out of 1157\n",
      "1723051965.362849 INFO: Expansion time:Pyr.SO3:0.8969587530009449\n",
      "1723051965.365152 INFO: Filtered count:0 out of 1157\n",
      "1723051966.673513 INFO: cpd05178 not found in model!\n",
      "1723051966.6761763 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_H2O.45.pg.G.D GF: 12194 Prob: 2988 Other: 3178\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051968.841612 INFO: Expansion time:Pyr.SO4:3.47500589000083\n",
      "1723051968.8439293 INFO: Filtered count:0 out of 1157\n",
      "1723051969.6758053 INFO: cpd05178 not found in model!\n",
      "1723051969.6779356 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051969.7381384 INFO: Expansion time:Pyr.TMAO:0.8928889759990852\n",
      "1723051969.7403536 INFO: Filtered count:0 out of 1157\n",
      "1723051970.567984 INFO: cpd05178 not found in model!\n",
      "1723051970.7315156 INFO: Expansion time:empty:0.9897010019994923\n",
      "1723051970.7338347 INFO: Filtered count:0 out of 1157\n",
      "1723051970.8121345 INFO: cpd05178 not found in model!\n",
      "1723051970.8141522 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051971.9046905 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[256 'SF2_C_D2.13.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:58:22+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '0e5d4423e3220826affa6953d8ba46be'\n",
      " 17614931 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051972.251438 INFO: Objective with gapfill database:140.3899522429062; min objective:0.01\n",
      "1723051972.2807946 INFO: metabolites 1145\n",
      "1723051972.3577871 INFO: cpd05178 not found in model!\n",
      "1723051972.3597898 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051972.7023673 INFO: reactions 1028\n",
      "1723051972.8219903 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051973.2241766 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/82/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051974.1120906 INFO: cpd05178 not found in model!\n",
      "1723051974.1141717 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051974.1542985 INFO: cpd05178 not found in model!\n",
      "1723051974.49149 INFO: cpd05178 not found in model!\n",
      "1723051974.4940577 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[64 'R2_A_D2.14.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:20:25+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '30054c6072c0ad45838235d7679e302e'\n",
      " 9734570 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051976.059954 INFO: metabolites 907\n",
      "1723051976.4124193 INFO: reactions 850\n",
      "1723051976.5108552 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051976.8423545 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/26/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051978.1419594 INFO: Objective with gapfill database:163.7165992095061; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723051986.1353285 INFO: cpd05178 not found in model!\n",
      "1723051986.1383293 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051986.7030911 INFO: Expansion time:Etho.O2:0.9394229549998272\n",
      "1723051986.7054265 INFO: Filtered count:0 out of 1508\n",
      "1723051987.618712 INFO: Expansion time:For.O2:0.9118647809991671\n",
      "1723051987.6209104 INFO: Filtered count:0 out of 1508\n",
      "1723051987.67547 INFO: cpd05178 not found in model!\n",
      "1723051987.9155395 INFO: cpd05178 not found in model!\n",
      "1723051987.9175873 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051988.5363562 INFO: Expansion time:Pyr:0.9140450360009709\n",
      "1723051988.5387154 INFO: Filtered count:0 out of 1508\n",
      "1723051988.8802545 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051989.2017386 INFO: Expansion time:Pyr.DMSO:0.6615627650007809\n",
      "1723051989.2036903 INFO: Filtered count:0 out of 1508\n",
      "1723051989.6783352 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051989.8493843 INFO: Expansion time:Pyr.NO:0.6444618879995687\n",
      "1723051989.8513901 INFO: Filtered count:0 out of 1508\n",
      "1723051990.0583026 INFO: cpd05178 not found in model!\n",
      "1723051990.1394913 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051990.4099224 INFO: cpd05178 not found in model!\n",
      "1723051990.443379 INFO: cpd05178 not found in model!\n",
      "1723051990.4459496 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051990.482174 INFO: Expansion time:Pyr.NO2:0.6296064299986028\n",
      "1723051990.4840856 INFO: Filtered count:0 out of 1508\n",
      "1723051990.6531646 INFO: cpd05178 not found in model!\n",
      "1723051990.655326 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051991.1267948 INFO: Expansion time:Pyr.NO3:0.6415559209999628\n",
      "1723051991.1287272 INFO: Filtered count:0 out of 1508\n",
      "1723051991.7670238 INFO: Expansion time:Pyr.O2:0.6369750829999248\n",
      "1723051991.7689226 INFO: Filtered count:0 out of 1508\n",
      "1723051992.2244656 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723051992.4006195 INFO: Expansion time:Pyr.SO3:0.6305632630010223\n",
      "1723051992.402608 INFO: Filtered count:0 out of 1508\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051992.8304465 INFO: cpd05178 not found in model!\n",
      "1723051993.0574415 INFO: cpd05178 not found in model!\n",
      "1723051993.059709 INFO: Media compound: cpd05178 not found in model.\n",
      "1723051993.3754091 INFO: cpd05178 not found in model!\n",
      "1723051993.3775601 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723051995.1442988 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051995.1945758 INFO: Expansion time:Pyr.SO4:2.7908398769995983\n",
      "1723051995.196486 INFO: Filtered count:0 out of 1508\n",
      "1723051995.5031257 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051995.8368025 INFO: Expansion time:Pyr.TMAO:0.6391363999991881\n",
      "1723051995.8388724 INFO: Filtered count:0 out of 1508\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[224 'R2_A_D1.84.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:51:49+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '59dcaeffba2825e54baa09c1c7e1da27'\n",
      " 12252255 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051996.3996694 INFO: metabolites 1105\n",
      "1723051996.5113575 INFO: Expansion time:empty:0.6712492120004754\n",
      "1723051996.513289 INFO: Filtered count:0 out of 1508\n",
      "1723051997.146479 INFO: reactions 1039\n",
      "1723051997.3216476 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723051997.9419947 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/494/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723051999.8578613 INFO: cpd05178 not found in model!\n",
      "1723051999.8600934 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052000.3887017 INFO: cpd05178 not found in model!\n",
      "1723052000.6301877 INFO: cpd05178 not found in model!\n",
      "1723052000.632352 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052005.6160073 INFO: cpd05178 not found in model!\n",
      "1723052005.618415 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052006.5959432 INFO: cpd05178 not found in model!\n",
      "1723052006.5982447 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D2.14.pg.G.D GF: 12182 Prob: 2940 Other: 3238\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052007.0957162 INFO: cpd05178 not found in model!\n",
      "1723052007.329884 INFO: cpd05178 not found in model!\n",
      "1723052007.33189 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052009.5640278 INFO: cpd05178 not found in model!\n",
      "1723052009.566657 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2.13.pg.G.D GF: 11701 Prob: 3554 Other: 3099\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052012.8952816 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052016.622957 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052017.5785103 INFO: Objective with gapfill database:154.91818224113632; min objective:0.01\n",
      "1723052018.0879667 INFO: cpd05178 not found in model!\n",
      "1723052018.0900464 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[7 'R1_A_D2.21.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:09:06+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '41aaad994b1d91dc18e29cf6bf7e4d30'\n",
      " 8112998 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052021.8173015 INFO: metabolites 751\n",
      "1723052022.1736414 INFO: reactions 679\n",
      "1723052022.2796986 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052022.662056 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/4/1\n",
      "[164 'R1_B_D2.27.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:39:54+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '22f79c800974a7f5f68f8f45ff434347'\n",
      " 11603617 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052023.7431247 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052024.267196 INFO: metabolites 1132\n",
      "1723052024.906121 INFO: reactions 1119\n",
      "1723052025.165785 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052025.8297973 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/98/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052026.3990757 INFO: Objective with gapfill database:253.6185730093687; min objective:0.01\n",
      "1723052028.775935 INFO: cpd05178 not found in model!\n",
      "1723052028.7781277 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052032.2483022 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052033.2038558 INFO: cpd05178 not found in model!\n",
      "1723052033.20619 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052033.9079046 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052033.9503138 INFO: cpd05178 not found in model!\n",
      "1723052033.952422 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.84.pg.G.D GF: 11493 Prob: 3824 Other: 3035\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052034.4828172 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052035.4451358 INFO: cpd05178 not found in model!\n",
      "1723052035.4473696 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052035.5420034 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052037.725954 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052038.2042832 INFO: cpd05178 not found in model!\n",
      "1723052038.350342 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052038.4392197 INFO: cpd05178 not found in model!\n",
      "1723052038.4414284 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052040.0828657 INFO: cpd05178 not found in model!\n",
      "1723052040.0849822 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052042.3749154 INFO: cpd05178 not found in model!\n",
      "1723052042.3770554 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052047.2573297 INFO: cpd05178 not found in model!\n",
      "1723052047.2596145 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.21.pg.G.D GF: 12620 Prob: 2506 Other: 3234\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052047.686168 INFO: Objective with gapfill database:220.37452231336172; min objective:0.01\n",
      "1723052048.2732089 INFO: cpd05178 not found in model!\n",
      "1723052048.2753308 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[263 'R2_A_D1.24.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:59:42+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '7de198c204c516014f779129fb4f20ab'\n",
      " 10832223 None]\n",
      "R1_B_D2.27.pg.G.D GF: 11717 Prob: 3466 Other: 3171\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052048.6750531 INFO: metabolites 994\n",
      "1723052049.1314723 INFO: reactions 934\n",
      "1723052049.2804172 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052049.834965 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/123/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052053.9483252 INFO: cpd05178 not found in model!\n",
      "1723052053.9504595 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052056.4251506 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052057.0510068 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052057.5148714 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052061.114814 INFO: Objective with gapfill database:127.4293709700386; min objective:0.01\n",
      "1723052062.4609823 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052065.80412 INFO: Expansion time:Etho.O2:0.6647572900001251\n",
      "1723052065.8063796 INFO: Filtered count:0 out of 1600\n",
      "1723052066.452169 INFO: Expansion time:Glc:0.6446017210000718\n",
      "1723052066.4541705 INFO: Filtered count:0 out of 1600\n",
      "1723052067.1194956 INFO: Expansion time:Glc.DMSO:0.6640549209987512\n",
      "1723052067.1215384 INFO: Filtered count:0 out of 1600\n",
      "1723052069.8127303 INFO: Expansion time:Glc.O2:2.69000478199996\n",
      "1723052069.8148446 INFO: Filtered count:0 out of 1600\n",
      "1723052070.4694986 INFO: Expansion time:Glc.TMAO:0.6534633089995623\n",
      "1723052070.471693 INFO: Filtered count:0 out of 1600\n",
      "1723052071.351089 INFO: cpd05178 not found in model!\n",
      "1723052071.3531802 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723052071.4152708 INFO: Expansion time:Pyr:0.9421770780008956\n",
      "1723052071.4176419 INFO: Filtered count:0 out of 1600\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.24.pg.G.D GF: 12110 Prob: 2990 Other: 3252\n",
      "[84 'SF2_C_D2.42.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:24:09+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '8d812ca80a117401dcebe6c9af42ca0e'\n",
      " 11365916 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052072.3442485 INFO: Expansion time:Pyr.DMSO:0.9250307459988107\n",
      "1723052072.3462594 INFO: Filtered count:0 out of 1600\n",
      "1723052072.790799 INFO: metabolites 1048\n",
      "1723052073.255037 INFO: Expansion time:Pyr.NO:0.9074243670002033\n",
      "1723052073.257477 INFO: Filtered count:0 out of 1600\n",
      "1723052073.4447894 INFO: reactions 1036\n",
      "1723052073.6223874 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052074.1403716 INFO: cpd05178 not found in model!\n",
      "1723052074.171826 INFO: Expansion time:Pyr.NO2:0.9128121070007182\n",
      "1723052074.1741781 INFO: Filtered count:0 out of 1600\n",
      "1723052074.2165244 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/192/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052074.3829942 INFO: cpd05178 not found in model!\n",
      "1723052074.384971 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052075.0833123 INFO: Expansion time:Pyr.NO3:0.9076024879996112\n",
      "1723052075.0856555 INFO: Filtered count:0 out of 1600\n",
      "1723052076.0017996 INFO: Expansion time:Pyr.O2:0.9147627869988355\n",
      "1723052076.0041409 INFO: Filtered count:0 out of 1600\n",
      "1723052076.9094584 INFO: Expansion time:Pyr.SO3:0.9038880770003743\n",
      "1723052076.9118056 INFO: Filtered count:0 out of 1600\n",
      "1723052077.8419182 INFO: Expansion time:Pyr.SO4:0.9287742349988548\n",
      "1723052077.8444114 INFO: Filtered count:0 out of 1600\n",
      "1723052078.7609067 INFO: Expansion time:Pyr.TMAO:0.9149952940006187\n",
      "1723052078.7633603 INFO: Filtered count:0 out of 1600\n",
      "1723052079.5694487 INFO: Expansion time:empty:0.8047004559994093\n",
      "1723052079.5715566 INFO: Filtered count:0 out of 1600\n",
      "1723052082.724172 INFO: Objective with gapfill database:200.8024041791281; min objective:0.01\n",
      "1723052083.5986006 INFO: cpd05178 not found in model!\n",
      "1723052083.797241 INFO: cpd05178 not found in model!\n",
      "1723052083.7995074 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052086.450856 INFO: cpd05178 not found in model!\n",
      "1723052086.4527524 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052087.8195832 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052088.2016096 INFO: cpd05178 not found in model!\n",
      "1723052088.2043574 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052088.2694523 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052088.4883614 INFO: cpd05178 not found in model!\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052088.7152958 INFO: cpd05178 not found in model!\n",
      "1723052088.7171075 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052093.3186119 INFO: cpd05178 not found in model!\n",
      "1723052093.320885 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052093.5211017 INFO: cpd05178 not found in model!\n",
      "1723052093.751543 INFO: cpd05178 not found in model!\n",
      "1723052093.753708 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052094.026589 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052094.2042618 INFO: cpd05178 not found in model!\n",
      "1723052094.2064805 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[296 'R1_A_D2.84.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:06:23+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '6ab0554d8bd6637e8f45c83d283bb776'\n",
      " 2049382 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052097.7466657 INFO: metabolites 548\n",
      "1723052097.929908 INFO: reactions 477\n",
      "1723052097.951816 INFO: cpd05178 not found in model!\n",
      "1723052097.953676 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052098.102696 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052098.4100392 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/288/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052100.4485629 INFO: cpd05178 not found in model!\n",
      "1723052100.7680511 INFO: cpd05178 not found in model!\n",
      "1723052100.770555 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052102.3015993 INFO: cpd05178 not found in model!\n",
      "1723052102.3043833 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052104.1150181 INFO: cpd05178 not found in model!\n",
      "1723052104.1178052 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723052104.1701448 INFO: cpd05178 not found in model!\n",
      "1723052104.1721675 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2.42.pg.G.D GF: 11530 Prob: 3820 Other: 3004\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052105.2573926 INFO: cpd05178 not found in model!\n",
      "1723052105.600065 INFO: cpd05178 not found in model!\n",
      "1723052105.602675 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052107.0157714 INFO: cpd05178 not found in model!\n",
      "1723052107.0179894 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052107.8144908 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052108.377458 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052110.2642124 INFO: cpd05178 not found in model!\n",
      "1723052110.2662723 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052114.1016858 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052114.1295753 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052117.8020835 INFO: cpd05178 not found in model!\n",
      "1723052117.8044538 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052119.4461198 INFO: cpd05178 not found in model!\n",
      "1723052119.4488266 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052124.1447864 INFO: cpd05178 not found in model!\n",
      "1723052124.1474354 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723052124.2856276 INFO: cpd05178 not found in model!\n",
      "1723052124.2876432 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2.84.pg.G.D GF: 12964 Prob: 1852 Other: 3538\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052129.1075134 INFO: Expansion time:Ac.O2:5.800376110000798\n",
      "1723052129.1096685 INFO: Filtered count:0 out of 1472\n",
      "1723052129.3872886 INFO: cpd05178 not found in model!\n",
      "1723052129.6972525 INFO: cpd05178 not found in model!\n",
      "1723052129.6998653 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052130.1353507 INFO: Expansion time:Etho.O2:1.0244721149992984\n",
      "1723052130.1376557 INFO: Filtered count:0 out of 1472\n",
      "1723052131.0207202 INFO: Expansion time:For.O2:0.8817329609992157\n",
      "1723052131.0229204 INFO: Filtered count:0 out of 1472\n",
      "1723052132.3803859 INFO: Expansion time:Fum:1.3562987530003738\n",
      "1723052132.3826478 INFO: Filtered count:0 out of 1472\n",
      "1723052133.7776487 INFO: cpd05178 not found in model!\n",
      "1723052133.7798061 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052133.90323 INFO: Expansion time:Fum.O2:1.5194114470014028\n",
      "1723052133.905388 INFO: Filtered count:0 out of 1472\n",
      "1723052135.5171976 INFO: cpd05178 not found in model!\n",
      "1723052135.5896444 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052135.7512085 INFO: cpd05178 not found in model!\n",
      "1723052135.7534585 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052135.7774892 INFO: Expansion time:Glc:1.8708982660009497\n",
      "1723052135.7797515 INFO: Filtered count:0 out of 1472\n",
      "1723052137.0769093 INFO: Expansion time:Glc.DMSO:1.295964061999257\n",
      "1723052137.079163 INFO: Filtered count:0 out of 1472\n",
      "1723052138.6839592 INFO: Expansion time:Glc.O2:1.6035708670005988\n",
      "1723052138.6861308 INFO: Filtered count:0 out of 1472\n",
      "1723052141.0930595 INFO: Expansion time:Glc.TMAO:2.405639901999166\n",
      "1723052141.0953176 INFO: Filtered count:0 out of 1472\n",
      "1723052141.768307 INFO: cpd05178 not found in model!\n",
      "1723052141.7705054 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052141.982892 INFO: Expansion time:Pyr:0.8863682270002755\n",
      "1723052141.9852214 INFO: Filtered count:0 out of 1472\n",
      "1723052142.8838685 INFO: Expansion time:Pyr.DMSO:0.8973456620005891\n",
      "1723052142.8861184 INFO: Filtered count:0 out of 1472\n",
      "1723052143.857525 INFO: Expansion time:Pyr.NO:0.9702011890003632\n",
      "1723052143.8598795 INFO: Filtered count:0 out of 1472\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052145.313248 INFO: Expansion time:Pyr:0.5913072189996456\n",
      "1723052145.3154707 INFO: Filtered count:0 out of 575\n",
      "1723052145.6906836 INFO: cpd05178 not found in model!\n",
      "1723052145.6928413 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052145.9113538 INFO: Expansion time:Pyr.DMSO:0.5946793849998357\n",
      "1723052145.913644 INFO: Filtered count:0 out of 575\n",
      "1723052146.4898863 INFO: Expansion time:Pyr.NO:0.5748017900004925\n",
      "1723052146.4921954 INFO: Filtered count:0 out of 575\n",
      "1723052146.6924467 INFO: Expansion time:Pyr.NO2:2.8313297659988166\n",
      "1723052146.6947474 INFO: Filtered count:0 out of 1472\n",
      "1723052147.088526 INFO: Expansion time:Pyr.NO2:0.5950591999990138\n",
      "1723052147.0906878 INFO: Filtered count:0 out of 575\n",
      "1723052147.6858525 INFO: Expansion time:Pyr.NO3:0.5939721600007033\n",
      "1723052147.6880305 INFO: Filtered count:0 out of 575\n",
      "1723052148.031723 INFO: Expansion time:Pyr.NO3:1.335649362001277\n",
      "1723052148.0343132 INFO: Filtered count:0 out of 1472\n",
      "1723052148.2617376 INFO: Expansion time:Pyr.O2:0.572471019000659\n",
      "1723052148.263963 INFO: Filtered count:0 out of 575\n",
      "1723052148.9081259 INFO: Expansion time:Pyr.SO3:0.6428650480011129\n",
      "1723052148.910351 INFO: Filtered count:0 out of 575\n",
      "1723052149.4959009 INFO: cpd05178 not found in model!\n",
      "1723052149.4983282 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052149.5412166 INFO: Expansion time:Pyr.SO4:0.6294918120001967\n",
      "1723052149.5434408 INFO: Filtered count:0 out of 575\n",
      "1723052150.089879 INFO: Expansion time:Pyr.O2:2.0540403579998383\n",
      "1723052150.0924335 INFO: Filtered count:0 out of 1472\n",
      "1723052150.1168318 INFO: Expansion time:Pyr.TMAO:0.5719935779998195\n",
      "1723052150.118994 INFO: Filtered count:0 out of 575\n",
      "1723052151.918847 INFO: Expansion time:Pyr.SO3:1.8248700860003737\n",
      "1723052151.9215188 INFO: Filtered count:0 out of 1472\n",
      "1723052152.7475097 INFO: Expansion time:empty:2.627307105\n",
      "1723052152.7497878 INFO: Filtered count:0 out of 575\n",
      "1723052153.6649594 INFO: Expansion time:Pyr.SO4:1.7418531160001294\n",
      "1723052153.6676207 INFO: Filtered count:0 out of 1472\n",
      "1723052154.8512056 INFO: Expansion time:Pyr.TMAO:1.1821295409990853\n",
      "1723052154.853389 INFO: Filtered count:0 out of 1472\n",
      "1723052155.4869144 INFO: Expansion time:empty:0.6323162650005543\n",
      "1723052155.4891222 INFO: Filtered count:0 out of 1472\n",
      "1723052156.0515149 INFO: cpd05178 not found in model!\n",
      "1723052156.2900388 INFO: cpd05178 not found in model!\n",
      "1723052156.2922268 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[284 'R2_restored_C_black.12.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:03:43+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '46f8dcb282b8890db02f08fb351b1c93'\n",
      " 11590165 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052158.5050323 INFO: metabolites 999\n",
      "1723052158.8962266 INFO: cpd05178 not found in model!\n",
      "1723052158.8982575 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052159.023256 INFO: reactions 948\n",
      "1723052159.1865134 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052159.7855666 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/220/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052160.6553426 INFO: cpd05178 not found in model!\n",
      "1723052160.6576595 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052164.5135221 INFO: cpd05178 not found in model!\n",
      "1723052164.7531393 INFO: cpd05178 not found in model!\n",
      "1723052164.7554264 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052166.2910209 INFO: cpd05178 not found in model!\n",
      "1723052166.2931285 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052167.0872536 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052167.5288017 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052169.4999197 INFO: cpd05178 not found in model!\n",
      "1723052169.502152 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052173.044404 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052173.3748028 INFO: cpd05178 not found in model!\n",
      "1723052173.3769631 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052174.617589 INFO: cpd05178 not found in model!\n",
      "1723052174.8421512 INFO: cpd05178 not found in model!\n",
      "1723052174.8444188 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052177.125504 INFO: cpd05178 not found in model!\n",
      "1723052177.1276429 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052180.006673 INFO: cpd05178 not found in model!\n",
      "1723052180.0088766 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_C_black.12.pg.G.D GF: 12049 Prob: 3256 Other: 3049\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052182.0259926 INFO: cpd05178 not found in model!\n",
      "1723052182.2431445 INFO: cpd05178 not found in model!\n",
      "1723052182.2450511 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052186.4263885 INFO: cpd05178 not found in model!\n",
      "1723052186.4285164 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052191.4153695 INFO: Objective with gapfill database:93.52553062704273; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[96 'R2A_B_D1.4.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:26:30+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '3c00002d12a535acffeb3098a1c5e76c'\n",
      " 16138584 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052199.9530125 INFO: cpd05178 not found in model!\n",
      "1723052199.9550476 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052200.0850856 INFO: metabolites 1077\n",
      "1723052200.58278 INFO: reactions 1035\n",
      "1723052200.753547 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052201.1941788 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/299/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052204.758101 INFO: cpd05178 not found in model!\n",
      "1723052204.984373 INFO: cpd05178 not found in model!\n",
      "1723052204.986624 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[236 'SF2_B_H2O.14.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:54:06+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'b8fb38b9e30e3e293c903d1ecb609a9a'\n",
      " 4253987 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052206.9202623 INFO: metabolites 901\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[176 'R2A_B_D2.37.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:42:09+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '77a5998ef06608c1bd50ab0944cd08e5'\n",
      " 7915846 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052207.2832286 INFO: reactions 839\n",
      "1723052207.388147 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052207.6113114 INFO: metabolites 1052\n",
      "1723052207.7153068 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/528/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052207.9555757 INFO: reactions 969\n",
      "1723052208.1358387 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052208.5261116 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/166/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052209.0878487 INFO: cpd05178 not found in model!\n",
      "1723052209.090007 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052214.1517494 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052214.6640966 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052214.8893108 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052215.2427413 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052216.746819 INFO: cpd05178 not found in model!\n",
      "1723052216.7491784 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052217.2154417 INFO: cpd05178 not found in model!\n",
      "1723052217.217429 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052220.7491887 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052220.8070076 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052225.8661327 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052226.4449844 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052227.601373 INFO: cpd05178 not found in model!\n",
      "1723052227.6035683 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_B_D2.37.pg.G.D GF: 12047 Prob: 3138 Other: 3169\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052231.2246656 INFO: cpd05178 not found in model!\n",
      "1723052231.2271824 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_H2O.14.pg.G.D GF: 12370 Prob: 2782 Other: 3200\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052232.0759366 INFO: cpd05178 not found in model!\n",
      "1723052232.3177958 INFO: cpd05178 not found in model!\n",
      "1723052232.319894 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052233.418491 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052235.9353848 INFO: cpd05178 not found in model!\n",
      "1723052235.9375546 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052237.730086 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052238.1047764 INFO: cpd05178 not found in model!\n",
      "1723052238.1068752 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052238.508813 INFO: cpd05178 not found in model!\n",
      "1723052238.5108445 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052239.13564 INFO: cpd05178 not found in model!\n",
      "1723052239.1377726 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052239.5346105 INFO: cpd05178 not found in model!\n",
      "1723052239.7574422 INFO: cpd05178 not found in model!\n",
      "1723052239.7596724 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052240.7093399 INFO: cpd05178 not found in model!\n",
      "1723052240.7116675 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_B_D1.4.pg.G.D GF: 11237 Prob: 4170 Other: 2947\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052244.7160764 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052247.7004704 INFO: Expansion time:Ac.O2:1.4104381209999701\n",
      "1723052247.702957 INFO: Filtered count:0 out of 1358\n",
      "1723052248.4590085 INFO: Expansion time:Etho.O2:0.7548153980005736\n",
      "1723052248.4612453 INFO: Filtered count:0 out of 1358\n",
      "1723052249.1576664 INFO: Expansion time:For.O2:0.6951648319991364\n",
      "1723052249.159818 INFO: Filtered count:0 out of 1358\n",
      "1723052249.5985823 INFO: cpd05178 not found in model!\n",
      "1723052249.600866 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052249.8821645 INFO: Expansion time:Pyr:0.7210702649990708\n",
      "1723052249.8842273 INFO: Filtered count:0 out of 1358\n",
      "1723052251.2884188 INFO: Expansion time:Pyr.DMSO:1.4030283369993413\n",
      "1723052251.290471 INFO: Filtered count:0 out of 1358\n",
      "1723052251.336203 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052251.9435027 INFO: Expansion time:Pyr.NO:0.6518530319990532\n",
      "1723052251.9456308 INFO: Filtered count:0 out of 1358\n",
      "1723052252.6337206 INFO: Expansion time:Pyr.NO2:0.6867786729999352\n",
      "1723052252.6357276 INFO: Filtered count:0 out of 1358\n",
      "1723052253.6825736 INFO: Expansion time:Pyr.NO3:1.0457021860001987\n",
      "1723052253.6845655 INFO: Filtered count:0 out of 1358\n",
      "1723052253.8064978 INFO: cpd05178 not found in model!\n",
      "1723052253.8085642 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052254.475514 INFO: Expansion time:Pyr.O2:0.7898393190007482\n",
      "1723052254.4776053 INFO: Filtered count:0 out of 1358\n",
      "1723052255.2782352 INFO: cpd05178 not found in model!\n",
      "1723052255.2803702 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[247 'R1_B_D1.29.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:56:18+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '1c6fdac9fd69e7d8c30e21ee8790af7a'\n",
      " 11205298 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052256.4186065 INFO: metabolites 1164\n",
      "1723052256.9848888 INFO: Expansion time:Pyr:0.6512308319997828\n",
      "1723052256.9871392 INFO: Filtered count:0 out of 1192\n",
      "1723052257.0078006 INFO: reactions 1098\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn01486_c0\n",
      "Failed:>rxn00530_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052257.2100008 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052257.247652 INFO: cpd05178 not found in model!\n",
      "1723052257.249915 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn09043_c0\n",
      "Failed:>rxn07191_c0\n",
      "Failed:>rxn05459_c0\n",
      "Failed:>rxn01739_c0\n",
      "Failed:<rxn00068_c0\n",
      "Failed:>rxn03057_c0\n",
      "Failed:>rxn05293_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052257.8413773 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn09202_c0\n",
      "188040/56/1\n",
      "Failed:>rxn02212_c0\n",
      "Failed:>rxn02160_c0\n",
      "Failed:<rxn01147_c0\n",
      "Failed:>rxn00549_c0\n",
      "Failed:>rxn05522_c0\n",
      "Failed:>rxn00600_c0\n",
      "Failed:<rxn13722_c0\n",
      "Failed:>rxn07848_c0\n",
      "Failed:>rxn08518_c0\n",
      "Failed:<rxn11347_c0\n",
      "Failed:>rxn14207_c0\n",
      "Failed:>rxn00907_c0\n",
      "Failed:>rxn10213_c0\n",
      "Failed:>rxn05039_c0\n",
      "Failed:>rxn09180_c0\n",
      "Failed:>rxn05030_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052259.600947 INFO: Expansion time:Pyr.DMSO:2.6125246260016866\n",
      "1723052259.6030812 INFO: Filtered count:0 out of 1192\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03891_c0\n",
      "Failed:>rxn10204_c0\n",
      "Failed:>rxn10615_c0\n",
      "Failed:>rxn00392_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00029_c0\n",
      "Failed:<rxn05211_c0\n",
      "Failed:>rxn03536_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052260.3026092 INFO: Expansion time:Pyr.NO:0.6982673430011346\n",
      "1723052260.304881 INFO: Filtered count:0 out of 1192\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn00003_c0\n",
      "Failed:>rxn00979_c0\n",
      "Failed:>rxn03537_c0\n",
      "Failed:>rxn01627_c0\n",
      "Failed:>rxn09113_c0\n",
      "Failed:>rxn10785_c0\n",
      "Failed:>rxn04675_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052260.940056 INFO: Expansion time:Pyr.NO2:0.6338719110008242\n",
      "1723052260.9420455 INFO: Filtered count:0 out of 1192\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn00970_c0\n",
      "Failed:>rxn10094_c0\n",
      "Failed:<rxn06723_c0\n",
      "Failed:>rxn00313_c0\n",
      "Failed:>rxn09041_c0\n",
      "Failed:<rxn01333_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052261.5759037 INFO: Expansion time:Ac.O2:1.549321577000228\n",
      "1723052261.5778832 INFO: Filtered count:0 out of 1459\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03150_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052261.5978074 INFO: Expansion time:Pyr.NO3:0.6546044249989791\n",
      "1723052261.5997252 INFO: Filtered count:0 out of 1192\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03538_c0\n",
      "Failed:>rxn01213_c0\n",
      "Failed:>rxn05520_c0\n",
      "Failed:>rxn03393_c0\n",
      "Failed:<rxn01790_c0\n",
      "Failed:>rxn02897_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052262.3315496 INFO: Expansion time:Pyr.O2:0.7306475349996617\n",
      "1723052262.3334577 INFO: Filtered count:0 out of 1192\n",
      "1723052262.4179668 INFO: Expansion time:Etho.O2:0.8388747989993135\n",
      "1723052262.419905 INFO: Filtered count:0 out of 1459\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn05144_c0\n",
      "Failed:<rxn05310_c0\n",
      "Failed:>rxn02831_c0\n",
      "Failed:>rxn01258_c0\n",
      "Failed:<rxn02832_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052262.985611 INFO: Expansion time:Pyr.SO3:0.6507786930014845\n",
      "1723052262.9877248 INFO: Filtered count:0 out of 1192\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn03164_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052263.1185043 INFO: Expansion time:For.O2:0.6973093769993284\n",
      "1723052263.1207037 INFO: Filtered count:0 out of 1459\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn16573_c0\n",
      "Failed:>rxn25839_c0\n",
      "Failed:>rxn08233_c0\n",
      "Failed:>rxn16107_c0\n",
      "Failed:>rxn43837_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052263.6280534 INFO: Expansion time:Pyr.SO3:9.149128260998623\n",
      "1723052263.6301296 INFO: Filtered count:64 out of 1358\n",
      "1723052263.6540678 INFO: Expansion time:Pyr.SO4:0.6649605290003819\n",
      "1723052263.6560876 INFO: Filtered count:0 out of 1192\n",
      "1723052263.9321334 INFO: Expansion time:Glc.O2:0.8098781860007875\n",
      "1723052263.934556 INFO: Filtered count:0 out of 1459\n",
      "1723052264.3027189 INFO: Expansion time:Pyr.TMAO:0.6454819020000286\n",
      "1723052264.3048186 INFO: Filtered count:0 out of 1192\n",
      "1723052264.7977068 INFO: Expansion time:Pyr:0.8618777719984791\n",
      "1723052264.7999547 INFO: Filtered count:0 out of 1459\n",
      "1723052264.9574306 INFO: Expansion time:empty:0.6514061800007767\n",
      "1723052264.959751 INFO: Filtered count:0 out of 1192\n",
      "1723052265.50932 INFO: Expansion time:Pyr.DMSO:0.7080424559990206\n",
      "1723052265.5115921 INFO: Filtered count:0 out of 1459\n",
      "1723052266.248569 INFO: Expansion time:Pyr.SO4:2.6169860160007374\n",
      "1723052266.2508898 INFO: Filtered count:64 out of 1358\n",
      "1723052266.2600245 INFO: Expansion time:Pyr.NO:0.747109340998577\n",
      "1723052266.262162 INFO: Filtered count:0 out of 1459\n",
      "1723052266.9427466 INFO: Expansion time:Pyr.TMAO:0.6904166570002417\n",
      "1723052266.9450982 INFO: Filtered count:64 out of 1358\n",
      "1723052266.9861171 INFO: Expansion time:Pyr.NO2:0.7225530420000723\n",
      "1723052266.9882967 INFO: Filtered count:0 out of 1459\n",
      "1723052267.5848732 INFO: Expansion time:empty:0.6383695360000274\n",
      "1723052267.5870385 INFO: Filtered count:64 out of 1358\n",
      "1723052267.6095984 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052267.6654794 INFO: Expansion time:Pyr.NO3:0.6759305909999966\n",
      "1723052267.6677575 INFO: Filtered count:0 out of 1459\n",
      "1723052268.1162386 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052268.5616922 INFO: Expansion time:Pyr.O2:0.8926322489987797\n",
      "1723052268.5636415 INFO: Filtered count:0 out of 1459\n",
      "1723052269.4810038 INFO: Expansion time:Pyr.SO3:0.9160329599999386\n",
      "1723052269.4831102 INFO: Filtered count:0 out of 1459\n",
      "1723052272.1720898 INFO: cpd05178 not found in model!\n",
      "1723052272.1745822 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052272.3006508 INFO: Expansion time:Pyr.SO4:2.8163676249987475\n",
      "1723052272.3028288 INFO: Filtered count:0 out of 1459\n",
      "1723052273.0477464 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052273.4271731 INFO: Expansion time:Pyr.TMAO:1.1229263019995415\n",
      "1723052273.429709 INFO: Filtered count:0 out of 1459\n",
      "1723052274.1850233 INFO: cpd05178 not found in model!\n",
      "1723052274.1870348 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052274.492668 INFO: Expansion time:empty:1.0614547749992198\n",
      "1723052274.4953332 INFO: Filtered count:0 out of 1459\n",
      "1723052281.4255219 INFO: cpd05178 not found in model!\n",
      "1723052281.427972 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D1.29.pg.G.D GF: 11628 Prob: 3686 Other: 3040\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052282.2820048 INFO: cpd05178 not found in model!\n",
      "1723052284.3041468 INFO: cpd05178 not found in model!\n",
      "1723052284.3062959 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052289.538814 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052294.9322515 INFO: cpd05178 not found in model!\n",
      "1723052295.2117517 INFO: cpd05178 not found in model!\n",
      "1723052295.213955 INFO: Media compound: cpd05178 not found in model.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052298.649794 INFO: Expansion time:Ac.NO2:0.6643417020004563\n",
      "1723052298.652248 INFO: Filtered count:0 out of 1528\n",
      "1723052299.3112483 INFO: Expansion time:Ac.O2:0.6576752020009735\n",
      "1723052299.3135524 INFO: Filtered count:0 out of 1528\n",
      "1723052299.960637 INFO: Expansion time:Etho.O2:0.64584443899912\n",
      "1723052299.962907 INFO: Filtered count:0 out of 1528\n",
      "1723052300.6131203 INFO: Expansion time:For.NO2:0.6488843810002436\n",
      "1723052300.615333 INFO: Filtered count:0 out of 1528\n",
      "1723052301.2669702 INFO: Expansion time:For.O2:0.6504158909992839\n",
      "1723052301.2692943 INFO: Filtered count:0 out of 1528\n",
      "1723052303.4876764 INFO: Expansion time:empty:2.217007310000554\n",
      "1723052303.489941 INFO: Filtered count:0 out of 1528\n",
      "1723052308.6375332 INFO: cpd05178 not found in model!\n",
      "1723052308.8581927 INFO: cpd05178 not found in model!\n",
      "1723052308.8603575 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052308.939103 INFO: cpd05178 not found in model!\n",
      "1723052308.941337 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052310.173473 INFO: cpd05178 not found in model!\n",
      "1723052310.4116597 INFO: cpd05178 not found in model!\n",
      "1723052310.413736 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052310.9087465 INFO: cpd05178 not found in model!\n",
      "1723052311.135081 INFO: cpd05178 not found in model!\n",
      "1723052311.1373909 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052311.6831522 INFO: cpd05178 not found in model!\n",
      "1723052311.9200711 INFO: cpd05178 not found in model!\n",
      "1723052311.922106 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052318.451595 INFO: cpd05178 not found in model!\n",
      "1723052318.453543 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[312 'SF2_C_D2.8.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:09:38+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '11ed482db0646284dbdb1615f7bd16b5'\n",
      " 11828048 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052326.2846954 INFO: metabolites 1189\n",
      "1723052327.0335758 INFO: reactions 1144\n",
      "1723052327.2189565 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[231 'SF2_C_D2.3.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:53:13+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '4c2277369d6b8dd006b9083bc72999f5'\n",
      " 14990596 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052327.8735392 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/440/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052328.129468 INFO: metabolites 1249\n",
      "1723052328.6832705 INFO: reactions 1205\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[228 'R2A_A_H2O.20.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:52:35+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '53888727d94e0367e7857a84d887eeea'\n",
      " 7346175 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052328.8322537 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052329.2645278 INFO: metabolites 1061\n",
      "1723052329.3164058 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/535/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052329.6887488 INFO: cpd05178 not found in model!\n",
      "1723052329.690446 INFO: reactions 1012\n",
      "1723052329.8091257 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052329.9053295 INFO: cpd05178 not found in model!\n",
      "1723052329.9073882 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052330.1921465 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/510/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052330.6182413 INFO: cpd05178 not found in model!\n",
      "1723052330.6203082 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[240 'SF2_C_D2.17.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:54:58+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '25f3cad229c0ddfba32b6226949dd848'\n",
      " 16740559 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052332.172604 INFO: metabolites 1170\n",
      "1723052332.8476472 INFO: reactions 1149\n",
      "1723052333.0120637 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[216 'R1_A_H2O.6.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:50:20+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '028238f8706d33408ac8ff1538680be9'\n",
      " 8244300 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052333.4463673 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/46/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052333.6222715 INFO: metabolites 951\n",
      "1723052333.7702546 INFO: cpd05178 not found in model!\n",
      "1723052333.7723634 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052333.9615154 INFO: reactions 911\n",
      "1723052334.0625596 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052334.4016783 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/425/1\n",
      "[288 'R2_C_D1.17.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:04:36+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'c7d017886c65282cacf8a18e3ae641c3'\n",
      " 14382569 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052335.7720141 INFO: metabolites 1271\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[279 'R1_A_H2O.14.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:02:41+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'd0e12fb4b8205186095c61453793aefa'\n",
      " 6513949 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052336.277789 INFO: metabolites 763\n",
      "1723052336.3871126 INFO: cpd05178 not found in model!\n",
      "1723052336.3892648 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052336.4308207 INFO: reactions 1242\n",
      "1723052336.50806 INFO: reactions 667\n",
      "1723052336.5897138 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052336.6034045 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052336.6159232 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052336.8427296 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/2/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052337.1151667 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/269/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052338.4646478 INFO: cpd05178 not found in model!\n",
      "1723052338.7035253 INFO: cpd05178 not found in model!\n",
      "1723052338.705615 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052338.7532454 INFO: cpd05178 not found in model!\n",
      "1723052338.7555141 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052339.2738929 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052339.6852155 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052339.9838526 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052340.8007112 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052341.4782498 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052342.0628452 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052342.1694686 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052342.3685946 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052342.9288845 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052343.9751961 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052345.0295238 INFO: cpd05178 not found in model!\n",
      "1723052345.2602558 INFO: cpd05178 not found in model!\n",
      "1723052345.2624521 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052345.381293 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052345.4369583 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052345.7347994 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052346.434576 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052348.1609182 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052349.632719 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052350.4824939 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052351.543477 INFO: cpd05178 not found in model!\n",
      "1723052351.5460527 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2.8.pg.G.D GF: 11151 Prob: 4394 Other: 2809\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052353.0724165 INFO: cpd05178 not found in model!\n",
      "1723052353.0743666 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_H2O.20.pg.G.D GF: 11653 Prob: 3702 Other: 2997\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052354.5966592 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052355.1283464 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052355.2471042 INFO: cpd05178 not found in model!\n",
      "1723052355.2492175 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2.3.pg.G.D GF: 10816 Prob: 4674 Other: 2864\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052357.294563 INFO: cpd05178 not found in model!\n",
      "1723052357.2972646 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723052357.4308903 INFO: cpd05178 not found in model!\n",
      "1723052357.4329922 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_H2O.14.pg.G.D GF: 12687 Prob: 2262 Other: 3411\n",
      "SF2_C_D2.17.pg.G.D GF: 11557 Prob: 3684 Other: 3111\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052358.469395 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052359.8467293 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052362.174717 INFO: cpd05178 not found in model!\n",
      "1723052362.3962162 INFO: cpd05178 not found in model!\n",
      "1723052362.398856 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723052362.4281962 INFO: cpd05178 not found in model!\n",
      "1723052362.4301374 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_H2O.6.pg.G.D GF: 11967 Prob: 3236 Other: 3151\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052363.864927 INFO: cpd05178 not found in model!\n",
      "1723052363.867189 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052364.023117 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052365.5226192 INFO: Objective with gapfill database:137.57817455088954; min objective:0.01\n",
      "1723052368.14859 INFO: cpd05178 not found in model!\n",
      "1723052368.1508555 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D1.17.pg.G.D GF: 10945 Prob: 4474 Other: 2933\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052368.9504015 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052371.4134386 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052374.1933208 INFO: cpd05178 not found in model!\n",
      "1723052374.4149976 INFO: cpd05178 not found in model!\n",
      "1723052374.4171653 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052374.4193497 INFO: Expansion time:Ac.NO2:5.109005601001627\n",
      "1723052374.4212205 INFO: Filtered count:0 out of 1656\n",
      "1723052375.3246484 INFO: Expansion time:Ac.NO3:2.213860724999904\n",
      "1723052375.3268318 INFO: Filtered count:0 out of 1702\n",
      "1723052375.4669337 INFO: Expansion time:Ac.NO3:1.0445501129997865\n",
      "1723052375.4690652 INFO: Filtered count:0 out of 1656\n",
      "1723052375.9800057 INFO: Expansion time:Ac.O2:0.6519376310006919\n",
      "1723052375.9820228 INFO: Filtered count:0 out of 1702\n",
      "1723052376.5652514 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052376.6344738 INFO: Expansion time:Etho.O2:0.6512559869988763\n",
      "1723052376.6365066 INFO: Filtered count:0 out of 1702\n",
      "1723052377.2939634 INFO: Expansion time:For.NO3:0.6562908879986935\n",
      "1723052377.2960546 INFO: Filtered count:0 out of 1702\n",
      "1723052377.9475214 INFO: Expansion time:For.O2:0.6503219949991035\n",
      "1723052377.949575 INFO: Filtered count:0 out of 1702\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052378.4591572 INFO: Expansion time:Ac.O2:2.988928995999231\n",
      "1723052378.4610684 INFO: Filtered count:0 out of 1656\n",
      "1723052378.5526485 INFO: Objective with gapfill database:202.13516625439092; min objective:0.01\n",
      "1723052378.5951145 INFO: Expansion time:Fum:0.6442256749996886\n",
      "1723052378.5970712 INFO: Filtered count:0 out of 1702\n",
      "1723052378.8477514 INFO: cpd05178 not found in model!\n",
      "1723052378.8498902 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052379.3466017 INFO: Expansion time:Fum.O2:0.7480498349996196\n",
      "1723052379.3485465 INFO: Filtered count:0 out of 1702\n",
      "1723052379.695835 INFO: Expansion time:Etho.O2:1.233438858000227\n",
      "1723052379.697785 INFO: Filtered count:0 out of 1656\n",
      "1723052379.998795 INFO: Expansion time:Glc:0.6489272720009467\n",
      "1723052380.0008087 INFO: Filtered count:0 out of 1702\n",
      "1723052380.6669214 INFO: Expansion time:Glc.DMSO:0.6649661760002346\n",
      "1723052380.6689954 INFO: Filtered count:0 out of 1702\n",
      "1723052380.759913 INFO: Expansion time:For.NO2:1.0610012769993773\n",
      "1723052380.7619677 INFO: Filtered count:0 out of 1656\n",
      "1723052381.3463497 INFO: Expansion time:Glc.O2:0.676182481000069\n",
      "1723052381.348321 INFO: Filtered count:0 out of 1702\n",
      "1723052382.0139205 INFO: Expansion time:Glc.TMAO:0.6644525460014847\n",
      "1723052382.0159802 INFO: Filtered count:0 out of 1702\n",
      "1723052382.1863575 INFO: Expansion time:Ac.O2:3.6477626060004695\n",
      "1723052382.1885588 INFO: Filtered count:0 out of 1653\n",
      "1723052383.020923 INFO: cpd05178 not found in model!\n",
      "1723052383.0285382 INFO: cpd05178 not found in model!\n",
      "1723052383.0313866 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052383.3720782 INFO: cpd05178 not found in model!\n",
      "1723052383.3745654 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052383.613792 INFO: cpd05178 not found in model!\n",
      "1723052383.6160607 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052383.752896 INFO: Expansion time:For.NO3:2.98968824000076\n",
      "1723052383.7552426 INFO: Filtered count:0 out of 1656\n",
      "1723052383.9291534 INFO: Expansion time:Dlac:1.7392577009995875\n",
      "1723052383.9313047 INFO: Filtered count:0 out of 1653\n",
      "1723052384.1776505 INFO: Expansion time:Pyr:2.1605336270004045\n",
      "1723052384.1797307 INFO: Filtered count:0 out of 1702\n",
      "1723052384.831159 INFO: Expansion time:Pyr.DMSO:0.6502392950005742\n",
      "1723052384.8332524 INFO: Filtered count:0 out of 1702\n",
      "1723052385.1438286 INFO: Expansion time:For.O2:1.3873757009987457\n",
      "1723052385.1459446 INFO: Filtered count:0 out of 1656\n",
      "1723052385.4999523 INFO: Expansion time:Pyr.NO:0.6655666419992485\n",
      "1723052385.5019495 INFO: Filtered count:0 out of 1702\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052386.1684391 INFO: Expansion time:Pyr.NO2:0.6652548469992325\n",
      "1723052386.170609 INFO: Filtered count:0 out of 1702\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052386.8465848 INFO: Expansion time:Pyr.NO3:0.6745995519995631\n",
      "1723052386.8487096 INFO: Filtered count:0 out of 1702\n",
      "1723052386.9313455 INFO: Expansion time:Glc:1.7842266449988529\n",
      "1723052386.933429 INFO: Filtered count:0 out of 1656\n",
      "1723052387.516832 INFO: Expansion time:Pyr.O2:0.6668346410006052\n",
      "1723052387.519168 INFO: Filtered count:0 out of 1702\n",
      "1723052388.1709504 INFO: Expansion time:Pyr.SO3:0.6505637620011839\n",
      "1723052388.1729827 INFO: Filtered count:0 out of 1702\n",
      "1723052388.2106638 INFO: cpd05178 not found in model!\n",
      "1723052388.2128985 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052388.7020152 INFO: Expansion time:Dlac.O2:4.769591008000134\n",
      "1723052388.7042353 INFO: Filtered count:0 out of 1653\n",
      "1723052388.7199478 INFO: Expansion time:Ac.NO2:2.868965281999408\n",
      "1723052388.72193 INFO: Filtered count:0 out of 1767\n",
      "1723052388.8287628 INFO: Expansion time:Pyr.SO4:0.6546673480006575\n",
      "1723052388.8308818 INFO: Filtered count:0 out of 1702\n",
      "1723052389.1132138 INFO: Expansion time:Glc.DMSO:2.1785802909998893\n",
      "1723052389.1152942 INFO: Filtered count:0 out of 1656\n",
      "1723052389.1552768 INFO: Expansion time:Fum:2.7881600379987503\n",
      "1723052389.1577835 INFO: Filtered count:0 out of 913\n",
      "1723052389.49113 INFO: Expansion time:Pyr.TMAO:0.6590614340002503\n",
      "1723052389.4932003 INFO: Filtered count:0 out of 1702\n",
      "1723052389.9003901 INFO: Expansion time:Ac.O2:1.1772637529993517\n",
      "1723052389.90248 INFO: Filtered count:0 out of 1767\n",
      "1723052389.9413476 INFO: Expansion time:Fum.O2:0.7821029900005669\n",
      "1723052389.94389 INFO: Filtered count:0 out of 913\n",
      "1723052390.1446745 INFO: Expansion time:empty:0.6503280449996964\n",
      "1723052390.1466928 INFO: Filtered count:0 out of 1702\n",
      "1723052390.163661 INFO: Expansion time:Etho.O2:1.4581619149994367\n",
      "1723052390.1656985 INFO: Filtered count:0 out of 1653\n",
      "1723052390.727214 INFO: Expansion time:empty:0.7818423609987804\n",
      "1723052390.7296097 INFO: Filtered count:0 out of 913\n",
      "1723052391.0234997 INFO: cpd05178 not found in model!\n",
      "1723052391.0256853 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052391.0820937 INFO: Expansion time:Dlac.O2:1.178484969001147\n",
      "1723052391.0841541 INFO: Filtered count:0 out of 1767\n",
      "1723052391.476689 INFO: Expansion time:Glc.O2:2.360215689001052\n",
      "1723052391.4788349 INFO: Filtered count:0 out of 1656\n",
      "1723052392.0558252 INFO: Expansion time:Etho.O2:0.9705431389993464\n",
      "1723052392.0577977 INFO: Filtered count:0 out of 1767\n",
      "1723052392.4629087 INFO: Expansion time:Fum:2.2960183689992846\n",
      "1723052392.4649985 INFO: Filtered count:0 out of 1653\n",
      "1723052392.9284992 INFO: Expansion time:For.NO2:0.8693938970009185\n",
      "1723052392.930587 INFO: Filtered count:0 out of 1767\n",
      "1723052394.045986 INFO: Expansion time:Glc.TMAO:2.565995454999211\n",
      "1723052394.0481808 INFO: Filtered count:0 out of 1656\n",
      "1723052394.819705 INFO: Expansion time:Pyr:0.7703229670005385\n",
      "1723052394.8218577 INFO: Filtered count:0 out of 1656\n",
      "1723052395.70179 INFO: Expansion time:Pyr.DMSO:0.8787999440010026\n",
      "1723052395.70393 INFO: Filtered count:0 out of 1656\n",
      "1723052395.7740104 INFO: Expansion time:For.O2:2.842305756999849\n",
      "1723052395.775909 INFO: Filtered count:0 out of 1767\n",
      "1723052396.0363727 INFO: Expansion time:Fum.O2:3.570253165000395\n",
      "1723052396.0385365 INFO: Filtered count:0 out of 1653\n",
      "1723052396.6815078 INFO: Expansion time:Pyr.NO:0.9764129140003206\n",
      "1723052396.683658 INFO: Filtered count:0 out of 1656\n",
      "1723052397.0924175 INFO: Expansion time:Fum:1.3154323269991437\n",
      "1723052397.0943387 INFO: Filtered count:0 out of 1767\n",
      "1723052397.81141 INFO: Expansion time:Glyc:1.7716960319994541\n",
      "1723052397.8134348 INFO: Filtered count:0 out of 1653\n",
      "1723052398.0988169 INFO: Expansion time:Pyr.NO2:1.413728022000214\n",
      "1723052398.1008132 INFO: Filtered count:0 out of 1656\n",
      "1723052398.2953818 INFO: Expansion time:Fum.O2:1.1997755030006374\n",
      "1723052398.2971632 INFO: Filtered count:0 out of 1767\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[271 'R2A_C_D1.52.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:01:13+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'd216ca5ca4d8e3cbf6f5cde8220e07e2'\n",
      " 7116936 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052399.3290915 INFO: metabolites 689\n",
      "1723052399.5414329 INFO: Expansion time:Pyr:1.2431520379996073\n",
      "1723052399.5432804 INFO: Filtered count:0 out of 1767\n",
      "1723052399.5455515 INFO: reactions 619\n",
      "1723052399.6565006 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052399.7178886 INFO: Expansion time:Pyr.NO3:1.6158020530001522\n",
      "1723052399.7199845 INFO: Filtered count:0 out of 1656\n",
      "1723052399.9009454 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/317/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052399.9790573 INFO: Expansion time:Glyc.O2:2.164380115000313\n",
      "1723052399.9810987 INFO: Filtered count:0 out of 1653\n",
      "1723052400.4198022 INFO: Expansion time:Pyr.DMSO:0.8754233960007696\n",
      "1723052400.4216893 INFO: Filtered count:0 out of 1767\n",
      "1723052401.44301 INFO: Expansion time:Pyr.NO:1.0201593109995883\n",
      "1723052401.4448752 INFO: Filtered count:0 out of 1767\n",
      "1723052401.4557407 INFO: Expansion time:Pyr.O2:1.7345317550007167\n",
      "1723052401.4576697 INFO: Filtered count:0 out of 1656\n",
      "1723052401.6704516 INFO: Expansion time:Pyr:1.6880959959999018\n",
      "1723052401.6725702 INFO: Filtered count:0 out of 1653\n",
      "1723052402.4687116 INFO: Expansion time:Pyr.NO2:1.0227172329996392\n",
      "1723052402.4706206 INFO: Filtered count:0 out of 1767\n",
      "1723052403.1122038 INFO: Expansion time:Pyr.DMSO:1.438360479000039\n",
      "1723052403.1143436 INFO: Filtered count:0 out of 1653\n",
      "1723052403.720293 INFO: Expansion time:Pyr.NO3:1.2484948259989324\n",
      "1723052403.7221496 INFO: Filtered count:0 out of 1767\n",
      "1723052404.9897 INFO: Expansion time:Pyr.O2:1.2664159279993328\n",
      "1723052404.9915595 INFO: Filtered count:0 out of 1767\n",
      "1723052405.0135775 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052405.3369894 INFO: Expansion time:Pyr.SO3:3.878131702000246\n",
      "1723052405.339112 INFO: Filtered count:0 out of 1656\n",
      "1723052406.2018223 INFO: Expansion time:Pyr.SO3:1.208233666000524\n",
      "1723052406.203688 INFO: Filtered count:0 out of 1767\n",
      "1723052406.529058 INFO: Expansion time:Pyr.NO:3.4135612150003\n",
      "1723052406.5311928 INFO: Filtered count:0 out of 1653\n",
      "1723052407.5085292 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052407.7107823 INFO: cpd05178 not found in model!\n",
      "1723052407.7126188 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052407.9769309 INFO: cpd05178 not found in model!\n",
      "1723052407.9788837 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052408.075001 INFO: Expansion time:Pyr.NO2:1.5425413039993146\n",
      "1723052408.0773337 INFO: Filtered count:0 out of 1653\n",
      "1723052409.4434361 INFO: Expansion time:Pyr.NO3:1.3648885829989013\n",
      "1723052409.4454207 INFO: Filtered count:0 out of 1653\n",
      "1723052409.7905083 INFO: Expansion time:Pyr.SO4:3.585734048001541\n",
      "1723052409.792449 INFO: Filtered count:0 out of 1767\n",
      "1723052410.6356514 INFO: Expansion time:Pyr.TMAO:0.8420068329996866\n",
      "1723052410.6375818 INFO: Filtered count:0 out of 1767\n",
      "1723052411.1739666 INFO: cpd05178 not found in model!\n",
      "1723052411.3896747 INFO: Expansion time:empty:0.7505927099991823\n",
      "1723052411.3915079 INFO: Filtered count:0 out of 1767\n",
      "1723052411.4038174 INFO: cpd05178 not found in model!\n",
      "1723052411.4055626 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052412.1640005 INFO: Expansion time:Pyr.O2:2.717379087000154\n",
      "1723052412.1661084 INFO: Filtered count:0 out of 1653\n",
      "1723052412.6079197 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052414.2726214 INFO: Expansion time:Pyr.SO3:2.105311993000214\n",
      "1723052414.2746973 INFO: Filtered count:0 out of 1653\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn00062_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052415.7627604 INFO: Expansion time:Pyr.SO4:1.486844414999723\n",
      "1723052415.7647953 INFO: Filtered count:0 out of 1653\n",
      "1723052415.7872887 INFO: Expansion time:Pyr.SO4:10.447000415999355\n",
      "1723052415.7891576 INFO: Filtered count:1 out of 1656\n",
      "1723052416.4518135 INFO: Expansion time:Pyr.TMAO:0.6615263100011362\n",
      "1723052416.453877 INFO: Filtered count:1 out of 1656\n",
      "1723052417.0191708 INFO: Expansion time:Pyr.TMAO:1.2532849990002433\n",
      "1723052417.0211926 INFO: Filtered count:0 out of 1653\n",
      "1723052417.0584362 INFO: cpd05178 not found in model!\n",
      "1723052417.1548448 INFO: Expansion time:empty:0.6998173120009596\n",
      "1723052417.156805 INFO: Filtered count:1 out of 1656\n",
      "1723052417.2760065 INFO: cpd05178 not found in model!\n",
      "1723052417.2781515 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052417.4605842 INFO: cpd05178 not found in model!\n",
      "1723052417.46288 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052417.8136396 INFO: Expansion time:empty:0.7909899599999335\n",
      "1723052417.8157566 INFO: Filtered count:0 out of 1653\n",
      "1723052417.995891 INFO: cpd05178 not found in model!\n",
      "1723052417.9977183 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052419.2650945 INFO: cpd05178 not found in model!\n",
      "1723052419.2671616 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052420.28766 INFO: cpd05178 not found in model!\n",
      "1723052420.2897968 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052420.9240315 INFO: cpd05178 not found in model!\n",
      "1723052420.9262757 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723052421.0593426 INFO: cpd05178 not found in model!\n",
      "1723052421.0614305 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D1.52.pg.G.D GF: 12856 Prob: 2084 Other: 3420\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052421.3489518 INFO: cpd05178 not found in model!\n",
      "1723052421.5633173 INFO: cpd05178 not found in model!\n",
      "1723052421.5655856 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052431.5069256 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052435.1997695 INFO: cpd05178 not found in model!\n",
      "1723052435.202131 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052438.6923018 INFO: cpd05178 not found in model!\n",
      "1723052438.915086 INFO: cpd05178 not found in model!\n",
      "1723052438.917294 INFO: Media compound: cpd05178 not found in model.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052441.0340598 INFO: Expansion time:For.NO2:0.5898619220006367\n",
      "1723052441.0362413 INFO: Filtered count:0 out of 851\n",
      "1723052441.6465771 INFO: Expansion time:Pyr:0.6091114819992072\n",
      "1723052441.6486866 INFO: Filtered count:0 out of 851\n",
      "1723052441.7473485 INFO: cpd05178 not found in model!\n",
      "1723052441.9784415 INFO: cpd05178 not found in model!\n",
      "1723052441.9805264 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052442.2530391 INFO: Expansion time:Pyr.DMSO:0.6031650449986046\n",
      "1723052442.255276 INFO: Filtered count:0 out of 851\n",
      "1723052442.854068 INFO: Expansion time:Pyr.NO:0.5975017350010603\n",
      "1723052442.8562148 INFO: Filtered count:0 out of 851\n",
      "1723052443.4411697 INFO: Expansion time:Pyr.NO2:0.5837707570008206\n",
      "1723052443.4433699 INFO: Filtered count:0 out of 851\n",
      "1723052444.027657 INFO: Expansion time:Pyr.NO3:0.5830575880008837\n",
      "1723052444.0297213 INFO: Filtered count:0 out of 851\n",
      "1723052444.6129127 INFO: Expansion time:Pyr.O2:0.5819502399990597\n",
      "1723052444.6151347 INFO: Filtered count:0 out of 851\n",
      "1723052446.6622503 INFO: Expansion time:Pyr.SO3:2.045756686000459\n",
      "1723052446.6642947 INFO: Filtered count:0 out of 851\n",
      "1723052447.257331 INFO: Expansion time:Pyr.SO4:0.591831319999983\n",
      "1723052447.2594945 INFO: Filtered count:0 out of 851\n",
      "1723052447.840071 INFO: Expansion time:Pyr.TMAO:0.579330137999932\n",
      "1723052447.842223 INFO: Filtered count:0 out of 851\n",
      "1723052448.4117131 INFO: Expansion time:empty:0.5682179579998774\n",
      "1723052448.413793 INFO: Filtered count:0 out of 851\n",
      "1723052451.2902944 INFO: cpd05178 not found in model!\n",
      "1723052451.534679 INFO: cpd05178 not found in model!\n",
      "1723052451.5367765 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[252 'R2A_C_D2.21.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:57:27+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '102435c989bbe570b590db96ce5faebf'\n",
      " 12227390 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052454.9371586 INFO: metabolites 1204\n",
      "1723052455.4461448 INFO: reactions 1216\n",
      "1723052455.54733 INFO: cpd05178 not found in model!\n",
      "1723052455.5495086 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052455.6065583 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052456.0741603 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/68/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052465.6686096 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052466.146121 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052468.654423 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[144 'R2_restored_C_black.8.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:36:02+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '3f181d64777a3f3e4cd9bed1ef8881cc'\n",
      " 14947509 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052473.1032455 INFO: cpd05178 not found in model!\n",
      "1723052473.105355 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052473.195874 INFO: metabolites 1226\n",
      "1723052474.2911813 INFO: cpd05178 not found in model!\n",
      "1723052474.5318656 INFO: cpd05178 not found in model!\n",
      "1723052474.5340097 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052474.6943166 INFO: cpd05178 not found in model!\n",
      "1723052474.6964514 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052475.4351957 INFO: reactions 1205\n",
      "1723052475.6099586 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052476.1018968 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/616/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052477.8938837 INFO: cpd05178 not found in model!\n",
      "1723052477.8957357 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.21.pg.G.D GF: 11465 Prob: 3832 Other: 3057\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052485.582967 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052491.9181883 INFO: cpd05178 not found in model!\n",
      "1723052492.1549888 INFO: cpd05178 not found in model!\n",
      "1723052492.1571634 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052493.2317648 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052493.5494173 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052494.5642838 INFO: Expansion time:Etho.O2:0.6632122400005755\n",
      "1723052494.5666242 INFO: Filtered count:0 out of 1740\n",
      "1723052495.2241595 INFO: Expansion time:Fum:0.6563258480000513\n",
      "1723052495.2264209 INFO: Filtered count:0 out of 1740\n",
      "1723052495.8847015 INFO: Expansion time:Fum.O2:0.657058851000329\n",
      "1723052495.886872 INFO: Filtered count:0 out of 1740\n",
      "1723052496.382672 INFO: cpd05178 not found in model!\n",
      "1723052496.384518 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052496.4280908 INFO: cpd05178 not found in model!\n",
      "1723052496.4299743 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052497.4036384 INFO: cpd05178 not found in model!\n",
      "1723052497.405698 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052498.2564764 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052498.5668898 INFO: Expansion time:Glyc.O2:2.678749238000819\n",
      "1723052498.568912 INFO: Filtered count:0 out of 1740\n",
      "1723052499.0522294 INFO: cpd05178 not found in model!\n",
      "1723052499.2162898 INFO: Expansion time:Pyr:0.6461370240012911\n",
      "1723052499.2182903 INFO: Filtered count:0 out of 1740\n",
      "1723052499.2840807 INFO: cpd05178 not found in model!\n",
      "1723052499.2861476 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052499.8658633 INFO: Expansion time:Pyr.DMSO:0.6462879930004419\n",
      "1723052499.8679419 INFO: Filtered count:0 out of 1740\n",
      "1723052500.5166128 INFO: Expansion time:Pyr.NO:0.6474965590005013\n",
      "1723052500.51898 INFO: Filtered count:0 out of 1740\n",
      "1723052501.213144 INFO: Expansion time:Pyr.NO2:0.6926908340010414\n",
      "1723052501.2153287 INFO: Filtered count:0 out of 1740\n",
      "1723052501.8592079 INFO: Expansion time:Pyr.NO3:0.642587401000128\n",
      "1723052501.8613048 INFO: Filtered count:0 out of 1740\n",
      "1723052502.5056896 INFO: Expansion time:Pyr.O2:0.6429627649995382\n",
      "1723052502.507982 INFO: Filtered count:0 out of 1740\n",
      "1723052502.5492604 INFO: cpd05178 not found in model!\n",
      "1723052502.7724288 INFO: cpd05178 not found in model!\n",
      "1723052502.7745585 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052503.1550553 INFO: Expansion time:Pyr.SO3:0.6457533979992149\n",
      "1723052503.1569867 INFO: Filtered count:0 out of 1740\n",
      "1723052503.81198 INFO: Expansion time:Pyr.SO4:0.6534107930001483\n",
      "1723052503.814036 INFO: Filtered count:0 out of 1740\n",
      "1723052504.1879907 INFO: cpd05178 not found in model!\n",
      "1723052504.1901596 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052504.469346 INFO: Expansion time:Pyr.TMAO:0.6538946320015384\n",
      "1723052504.4714227 INFO: Filtered count:0 out of 1740\n",
      "1723052505.1266346 INFO: Expansion time:empty:0.6539277819993004\n",
      "1723052505.1287086 INFO: Filtered count:0 out of 1740\n",
      "1723052505.346351 INFO: cpd05178 not found in model!\n",
      "1723052505.3485284 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_C_black.8.pg.G.D GF: 10793 Prob: 4724 Other: 2835\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052513.2946773 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052516.3266695 INFO: cpd05178 not found in model!\n",
      "1723052516.5532093 INFO: cpd05178 not found in model!\n",
      "1723052516.5553339 INFO: Media compound: cpd05178 not found in model.\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052522.3602142 INFO: Expansion time:Ac.O2:0.6678108159994736\n",
      "1723052522.3623405 INFO: Filtered count:0 out of 1746\n",
      "1723052523.0089314 INFO: Expansion time:Etho.O2:0.6453472270004568\n",
      "1723052523.0109448 INFO: Filtered count:0 out of 1746\n",
      "1723052525.0595484 INFO: cpd05178 not found in model!\n",
      "1723052525.0616982 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052525.671457 INFO: Expansion time:Fum:2.659399244001179\n",
      "1723052525.673532 INFO: Filtered count:0 out of 1746\n",
      "1723052526.3404577 INFO: Expansion time:Fum.O2:0.6654020779988059\n",
      "1723052526.3425598 INFO: Filtered count:0 out of 1746\n",
      "1723052527.0051062 INFO: Expansion time:Pyr:0.6613770569983899\n",
      "1723052527.0071707 INFO: Filtered count:0 out of 1746\n",
      "1723052527.6547508 INFO: Expansion time:Pyr.DMSO:0.6463545199985674\n",
      "1723052527.656772 INFO: Filtered count:0 out of 1746\n",
      "1723052528.304577 INFO: Expansion time:Pyr.NO:0.6465754780001589\n",
      "1723052528.3066678 INFO: Filtered count:0 out of 1746\n",
      "1723052528.9768188 INFO: Expansion time:Pyr.NO2:0.6690064240010543\n",
      "1723052528.9789867 INFO: Filtered count:0 out of 1746\n",
      "1723052529.6408143 INFO: Expansion time:Pyr.NO3:0.6606488950001221\n",
      "1723052529.6428485 INFO: Filtered count:0 out of 1746\n",
      "1723052529.8697944 INFO: cpd05178 not found in model!\n",
      "1723052529.8720741 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052530.2975438 INFO: Expansion time:Pyr.O2:0.6535377829986828\n",
      "1723052530.2996073 INFO: Filtered count:0 out of 1746\n",
      "1723052530.9429958 INFO: Expansion time:Pyr.SO3:0.6422151859987935\n",
      "1723052530.9451575 INFO: Filtered count:0 out of 1746\n",
      "1723052531.5888934 INFO: Expansion time:Pyr.SO4:0.6425995870013139\n",
      "1723052531.590959 INFO: Filtered count:0 out of 1746\n",
      "1723052532.2370598 INFO: Expansion time:Pyr.TMAO:0.6449416320010641\n",
      "1723052532.2389936 INFO: Filtered count:0 out of 1746\n",
      "1723052532.8863468 INFO: Expansion time:empty:0.6461633779999829\n",
      "1723052532.8882487 INFO: Filtered count:0 out of 1746\n",
      "1723052540.446527 INFO: cpd05178 not found in model!\n",
      "1723052540.448523 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052541.7778249 INFO: cpd05178 not found in model!\n",
      "1723052541.779899 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052551.2756531 INFO: cpd05178 not found in model!\n",
      "1723052551.4895656 INFO: cpd05178 not found in model!\n",
      "1723052551.4917283 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052551.502163 INFO: cpd05178 not found in model!\n",
      "1723052551.5041962 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052554.4044619 INFO: cpd05178 not found in model!\n",
      "1723052554.623393 INFO: cpd05178 not found in model!\n",
      "1723052554.6255312 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052556.1377819 INFO: cpd05178 not found in model!\n",
      "1723052556.3742332 INFO: cpd05178 not found in model!\n",
      "1723052556.376476 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052559.5236237 INFO: cpd05178 not found in model!\n",
      "1723052559.5257437 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052566.871508 INFO: cpd05178 not found in model!\n",
      "1723052566.873669 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[291 'R2A_C_D2.16.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:05:24+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '96e70df727fd31e27e36772a2e8b4c8e'\n",
      " 9663785 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052568.8485968 INFO: metabolites 1057\n",
      "1723052569.3612285 INFO: reactions 971\n",
      "1723052569.477331 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052569.8900254 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/296/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052571.7371488 INFO: cpd05178 not found in model!\n",
      "1723052571.7393994 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052578.5840168 INFO: cpd05178 not found in model!\n",
      "1723052578.814451 INFO: cpd05178 not found in model!\n",
      "1723052578.8165233 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052580.7002392 INFO: cpd05178 not found in model!\n",
      "1723052580.7021234 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052584.3159723 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052584.6306663 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052585.1971214 INFO: cpd05178 not found in model!\n",
      "1723052585.3467019 INFO: cpd05178 not found in model!\n",
      "1723052585.4340796 INFO: cpd05178 not found in model!\n",
      "1723052585.436469 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052585.574431 INFO: cpd05178 not found in model!\n",
      "1723052585.5766325 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052586.6769693 INFO: cpd05178 not found in model!\n",
      "1723052586.8923492 INFO: cpd05178 not found in model!\n",
      "1723052586.8944845 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052588.9398699 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052595.8643098 INFO: cpd05178 not found in model!\n",
      "1723052595.8662515 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052597.5534956 INFO: cpd05178 not found in model!\n",
      "1723052597.556015 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.16.pg.G.D GF: 11532 Prob: 3790 Other: 3030\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052598.7183762 INFO: cpd05178 not found in model!\n",
      "1723052599.0877135 INFO: cpd05178 not found in model!\n",
      "1723052599.0902317 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052600.9763107 INFO: cpd05178 not found in model!\n",
      "1723052600.9786325 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052609.5894382 INFO: Objective with gapfill database:178.70362201220613; min objective:0.01\n",
      "1723052617.1169224 INFO: cpd05178 not found in model!\n",
      "1723052617.3283517 INFO: cpd05178 not found in model!\n",
      "1723052617.3306754 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052624.234332 INFO: cpd05178 not found in model!\n",
      "1723052624.4603174 INFO: cpd05178 not found in model!\n",
      "1723052624.4623065 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[184 'R2_C_D1.25.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:43:45+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'a6fe0774e0065cf68fd15973c283a309'\n",
      " 12261739 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052643.722181 INFO: cpd05178 not found in model!\n",
      "1723052643.946105 INFO: cpd05178 not found in model!\n",
      "1723052643.9485128 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052644.1466255 INFO: metabolites 1209\n",
      "1723052644.6341712 INFO: reactions 1186\n",
      "1723052644.7696416 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052645.2205102 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/206/1\n",
      "[248 'R2_restored_C_black.17.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:56:31+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '696e976a7802f13e3c0ac97d8009b2e7'\n",
      " 14336498 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052647.3016863 INFO: metabolites 1114\n",
      "1723052647.7667167 INFO: reactions 1070\n",
      "1723052647.9106092 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052648.3242884 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/62/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052650.1714282 INFO: cpd05178 not found in model!\n",
      "1723052650.3979595 INFO: cpd05178 not found in model!\n",
      "1723052650.3998256 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052654.8503919 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052656.5715878 INFO: cpd05178 not found in model!\n",
      "1723052656.573614 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052656.9296725 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052660.994911 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052661.29075 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052661.678429 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[308 'R2_A_D1.93.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:08:47+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '4a8cb93eab0fdd1be1a96a3f8e7ee03d'\n",
      " 3395761 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052662.4651704 INFO: metabolites 627\n",
      "1723052662.602135 INFO: reactions 529\n",
      "1723052662.7266595 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052662.9408398 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/402/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052664.0397575 INFO: cpd05178 not found in model!\n",
      "1723052664.0417886 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052665.655915 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052666.5992758 INFO: cpd05178 not found in model!\n",
      "1723052666.8315117 INFO: cpd05178 not found in model!\n",
      "1723052666.8335626 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052667.289891 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052667.59889 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052670.534107 INFO: cpd05178 not found in model!\n",
      "1723052670.5361881 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D1.25.pg.G.D GF: 11103 Prob: 4316 Other: 2933\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052671.7404141 INFO: cpd05178 not found in model!\n",
      "1723052671.7428608 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052671.9996347 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052672.376449 INFO: cpd05178 not found in model!\n",
      "1723052672.3787906 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_restored_C_black.17.pg.G.D GF: 11514 Prob: 3702 Other: 3138\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052675.1364357 INFO: cpd05178 not found in model!\n",
      "1723052675.3466268 INFO: cpd05178 not found in model!\n",
      "1723052675.3486965 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052678.3678823 INFO: cpd05178 not found in model!\n",
      "1723052678.3700154 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1.93.pg.G.D GF: 13424 Prob: 1160 Other: 3768\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052682.4567664 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052682.620508 INFO: Objective with gapfill database:236.58018843431014; min objective:0.01\n",
      "1723052686.8101285 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052691.3658304 INFO: Expansion time:Ac.O2:0.6156621800000721\n",
      "1723052691.368095 INFO: Filtered count:0 out of 1511\n",
      "1723052691.9766567 INFO: Expansion time:Etho.O2:0.6072095609997632\n",
      "1723052691.9789298 INFO: Filtered count:0 out of 1511\n",
      "1723052692.5863068 INFO: Expansion time:For.O2:0.6060315760005324\n",
      "1723052692.5887198 INFO: Filtered count:0 out of 1511\n",
      "1723052693.1985033 INFO: Expansion time:Fum:0.6084436319997621\n",
      "1723052693.200717 INFO: Filtered count:0 out of 1511\n",
      "1723052693.8086553 INFO: Expansion time:Fum.O2:0.6067290569990291\n",
      "1723052693.810906 INFO: Filtered count:0 out of 1511\n",
      "1723052694.4356377 INFO: Expansion time:Glc:0.6234567840001546\n",
      "1723052694.4379113 INFO: Filtered count:0 out of 1511\n",
      "1723052695.0662599 INFO: Expansion time:Glc.DMSO:0.6270014450001327\n",
      "1723052695.0685782 INFO: Filtered count:0 out of 1511\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052695.7050803 INFO: Expansion time:Glc.O2:0.6351857130011922\n",
      "1723052695.707302 INFO: Filtered count:0 out of 1511\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[272 'R2A_C_D2.27.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:01:24+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '71be6a839b60505873f1a5c46f6c3a7a'\n",
      " 10774843 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052696.3192215 INFO: Expansion time:Glc.TMAO:0.6106706559985469\n",
      "1723052696.321234 INFO: Filtered count:0 out of 1511\n",
      "1723052696.6521316 INFO: metabolites 1107\n",
      "1723052696.9303713 INFO: Expansion time:Pyr:0.6077240689992323\n",
      "1723052696.9325988 INFO: Filtered count:0 out of 1511\n",
      "1723052697.1217892 INFO: reactions 1061\n",
      "1723052697.241371 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052697.278642 INFO: Expansion time:For.NO2:1.5174933670004975\n",
      "1723052697.2807376 INFO: Filtered count:0 out of 637\n",
      "1723052697.5435417 INFO: Expansion time:Pyr.DMSO:0.6096288120006648\n",
      "1723052697.5456815 INFO: Filtered count:0 out of 1511\n",
      "1723052697.659872 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/162/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052699.7523446 INFO: Expansion time:Pyr:2.4703119179994246\n",
      "1723052699.7543724 INFO: Filtered count:0 out of 637\n",
      "1723052699.7608998 INFO: cpd05178 not found in model!\n",
      "1723052699.7627485 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052700.0189545 INFO: cpd05178 not found in model!\n",
      "1723052700.052212 INFO: Expansion time:Pyr.NO:2.505147545\n",
      "1723052700.054499 INFO: Filtered count:0 out of 1511\n",
      "1723052700.2319663 INFO: cpd05178 not found in model!\n",
      "1723052700.2342916 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052700.2850718 INFO: Expansion time:Pyr.DMSO:0.5291910979995009\n",
      "1723052700.287123 INFO: Filtered count:0 out of 637\n",
      "1723052700.666066 INFO: Expansion time:Pyr.NO2:0.6103881510007341\n",
      "1723052700.6681938 INFO: Filtered count:0 out of 1511\n",
      "1723052700.8229733 INFO: Expansion time:Pyr.NO:0.5344751039992843\n",
      "1723052700.8250813 INFO: Filtered count:0 out of 637\n",
      "1723052701.2840877 INFO: Expansion time:Pyr.NO3:0.6146161609995033\n",
      "1723052701.2864835 INFO: Filtered count:0 out of 1511\n",
      "1723052701.5248835 INFO: Expansion time:Pyr.NO2:0.698314778999702\n",
      "1723052701.5269299 INFO: Filtered count:0 out of 637\n",
      "1723052701.899847 INFO: Expansion time:Pyr.O2:0.6120854790005978\n",
      "1723052701.9022765 INFO: Filtered count:0 out of 1511\n",
      "1723052702.2694001 INFO: Expansion time:Pyr.NO3:0.7411101279994909\n",
      "1723052702.271608 INFO: Filtered count:0 out of 637\n",
      "1723052702.5638552 INFO: Expansion time:Pyr.SO3:0.6601477280000836\n",
      "1723052702.5663996 INFO: Filtered count:0 out of 1511\n",
      "1723052702.9415495 INFO: Expansion time:Pyr.O2:0.6683673530005763\n",
      "1723052702.9436884 INFO: Filtered count:0 out of 637\n",
      "1723052703.1863387 INFO: Expansion time:Pyr.SO4:0.6186706649987173\n",
      "1723052703.1884978 INFO: Filtered count:0 out of 1511\n",
      "1723052703.8051524 INFO: Expansion time:Pyr.TMAO:0.6152652300006594\n",
      "1723052703.8072155 INFO: Filtered count:0 out of 1511\n",
      "1723052704.4229867 INFO: Expansion time:empty:0.6144154799985699\n",
      "1723052704.4252145 INFO: Filtered count:0 out of 1511\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:<rxn11537_c0\n",
      "Failed:<rxn00902_c0\n",
      "Failed:<rxn12933_c0\n",
      "Failed:>rxn06672_c0\n",
      "Failed:>rxn01739_c0\n",
      "Failed:>rxn01268_c0\n",
      "Failed:>rxn00688_c0\n",
      "Failed:<rxn00068_c0\n",
      "Failed:>rxn00762_c0\n",
      "Failed:>rxn05104_c0\n",
      "Failed:>rxn03057_c0\n",
      "Failed:>rxn00060_c0\n",
      "Failed:>rxn11209_c0\n",
      "Failed:>rxn00100_c0\n",
      "Failed:<rxn00789_c0\n",
      "Failed:>rxn03436_c0\n",
      "Failed:>rxn02264_c0\n",
      "Failed:>rxn09202_c0\n",
      "Failed:<rxn05514_c0\n",
      "Failed:>rxn02212_c0\n",
      "Failed:<rxn00191_c0\n",
      "Failed:>rxn13022_c0\n",
      "Failed:<rxn02768_c0\n",
      "Failed:<rxn00692_c0\n",
      "Failed:>rxn00802_c0\n",
      "Failed:>rxn01099_c0\n",
      "Failed:>rxn02160_c0\n",
      "Failed:>rxn00774_c0\n",
      "Failed:>rxn09615_c0\n",
      "Failed:<rxn00131_c0\n",
      "Failed:>rxn00178_c0\n",
      "Failed:<rxn01362_c0\n",
      "Failed:>rxn13690_c0\n",
      "Failed:>rxn00549_c0\n",
      "Failed:>rxn01022_c0\n",
      "Failed:>rxn05522_c0\n",
      "Failed:>rxn00727_c0\n",
      "Failed:<rxn02154_c0\n",
      "Failed:<rxn00612_c0\n",
      "Failed:>rxn00600_c0\n",
      "Failed:>rxn11258_c0\n",
      "Failed:>rxn02159_c0\n",
      "Failed:>rxn00300_c0\n",
      "Failed:>rxn08518_c0\n",
      "Failed:>rxn00471_c0\n",
      "Failed:>rxn05229_c0\n",
      "Failed:<rxn11934_c0\n",
      "Failed:>rxn02404_c0\n",
      "Failed:<rxn01742_c0\n",
      "Failed:>rxn05039_c0\n",
      "Failed:>rxn09188_c0\n",
      "Failed:>rxn09180_c0\n",
      "Failed:<rxn08822_c0\n",
      "Failed:>rxn05030_c0\n",
      "Failed:<rxn04677_c0\n",
      "Failed:>rxn03891_c0\n",
      "Failed:<rxn01465_c0\n",
      "Failed:>rxn03052_c0\n",
      "Failed:>rxn05145_c0\n",
      "Failed:<rxn01643_c0\n",
      "Failed:<rxn09341_c0\n",
      "Failed:>rxn01348_c0\n",
      "Failed:<rxn12264_c0\n",
      "Failed:>rxn00392_c0\n",
      "Failed:>rxn00747_c0\n",
      "Failed:>rxn01334_c0\n",
      "Failed:>rxn00029_c0\n",
      "Failed:>rxn06673_c0\n",
      "Failed:>rxn05744_c0\n",
      "Failed:<rxn09128_c0\n",
      "Failed:>rxn02175_c0\n",
      "Failed:<rxn05211_c0\n",
      "Failed:>rxn02835_c0\n",
      "Failed:>rxn02774_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052711.625164 INFO: cpd05178 not found in model!\n",
      "1723052711.6270888 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn09950_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052711.685717 INFO: cpd05178 not found in model!\n",
      "1723052711.687723 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed:>rxn04464_c0\n",
      "Failed:>rxn03536_c0\n",
      "Failed:<rxn01575_c0\n",
      "Failed:<rxn01454_c0\n",
      "Failed:>rxn05150_c0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052712.2412488 INFO: Expansion time:Pyr.SO3:9.296130453998558\n",
      "1723052712.2431307 INFO: Filtered count:80 out of 637\n",
      "1723052712.7733564 INFO: Expansion time:Pyr.SO4:0.5286754040007509\n",
      "1723052712.7753606 INFO: Filtered count:80 out of 637\n",
      "1723052713.3039942 INFO: Expansion time:Pyr.TMAO:0.527310066998325\n",
      "1723052713.3059769 INFO: Filtered count:80 out of 637\n",
      "1723052713.8420403 INFO: Expansion time:empty:0.5342875590013136\n",
      "1723052713.8439953 INFO: Filtered count:80 out of 637\n",
      "1723052715.354193 INFO: cpd05178 not found in model!\n",
      "1723052715.5739474 INFO: cpd05178 not found in model!\n",
      "1723052715.5761726 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052718.2087734 INFO: cpd05178 not found in model!\n",
      "1723052718.4360003 INFO: cpd05178 not found in model!\n",
      "1723052718.4381676 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052718.7497025 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052719.0918965 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052721.5805435 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[260 'SF2_C_H2O.49.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:59:10+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'ffaa74c48ff1486f6fc0be57d1b90c09'\n",
      " 4285715 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052724.6518111 INFO: metabolites 926\n",
      "1723052725.0140197 INFO: reactions 862\n",
      "1723052725.1235826 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052725.4563503 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/84/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052727.799828 INFO: cpd05178 not found in model!\n",
      "1723052728.03083 INFO: cpd05178 not found in model!\n",
      "1723052728.032795 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052729.556855 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052729.9593477 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052730.5582623 INFO: cpd05178 not found in model!\n",
      "1723052730.5605075 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.27.pg.G.D GF: 11135 Prob: 4276 Other: 2941\n",
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052734.2586923 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052737.4336753 INFO: cpd05178 not found in model!\n",
      "1723052737.6739545 INFO: cpd05178 not found in model!\n",
      "1723052737.6758423 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052743.057551 INFO: cpd05178 not found in model!\n",
      "1723052743.059688 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_H2O.49.pg.G.D GF: 12219 Prob: 2876 Other: 3257\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052743.4764566 INFO: Objective with gapfill database:154.87139065233916; min objective:0.01\n",
      "1723052754.525459 INFO: Objective with gapfill database:200.24539386676148; min objective:0.01\n",
      "1723052760.6509366 INFO: cpd05178 not found in model!\n",
      "1723052762.9496346 INFO: cpd05178 not found in model!\n",
      "1723052762.9518414 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052763.496317 INFO: cpd05178 not found in model!\n",
      "1723052763.4983003 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052767.6717818 INFO: cpd05178 not found in model!\n",
      "1723052767.6739676 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052770.8688657 INFO: cpd05178 not found in model!\n",
      "1723052770.8712604 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052787.4764786 INFO: cpd05178 not found in model!\n",
      "1723052787.4786577 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052795.65955 INFO: cpd05178 not found in model!\n",
      "1723052795.9019156 INFO: cpd05178 not found in model!\n",
      "1723052795.9037404 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052799.5449297 INFO: cpd05178 not found in model!\n",
      "1723052799.7851079 INFO: cpd05178 not found in model!\n",
      "1723052799.7870412 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052816.461489 INFO: cpd05178 not found in model!\n",
      "1723052816.6786962 INFO: cpd05178 not found in model!\n",
      "1723052816.6807973 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052827.779037 INFO: cpd05178 not found in model!\n",
      "1723052827.7810686 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052833.40655 INFO: cpd05178 not found in model!\n",
      "1723052833.6407292 INFO: cpd05178 not found in model!\n",
      "1723052833.642866 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052838.806755 INFO: cpd05178 not found in model!\n",
      "1723052838.8088596 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052850.0704975 INFO: cpd05178 not found in model!\n",
      "1723052850.0725913 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052851.5843883 INFO: cpd05178 not found in model!\n",
      "1723052851.5866284 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052853.5787556 INFO: cpd05178 not found in model!\n",
      "1723052853.5810168 WARNING: cpd05178: could not find metabolite for cpd05178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[192 'R2A_C_D2.022.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:45:27+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '1ab300ba2ef77c353e08c7f57a39d5ed'\n",
      " 14174215 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052855.1030133 INFO: cpd05178 not found in model!\n",
      "1723052855.3246586 INFO: cpd05178 not found in model!\n",
      "1723052855.3266358 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052855.4594738 INFO: metabolites 1167\n",
      "1723052856.0155222 INFO: reactions 1079\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[232 'R2A_A_D2.25.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:53:26+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '37a4aa69bacf65ba21e405598d3f5037'\n",
      " 10653687 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052856.1459436 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052856.4398403 INFO: metabolites 1089\n",
      "1723052856.6109283 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/268/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052856.8582373 INFO: reactions 1066\n",
      "1723052856.9829562 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052857.3723006 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/544/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052868.3236556 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052868.6146824 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052871.0112388 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052872.4990654 INFO: cpd05178 not found in model!\n",
      "1723052872.501211 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052872.9423234 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052873.264234 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052875.73607 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052879.658619 INFO: cpd05178 not found in model!\n",
      "1723052879.6607676 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D2.25.pg.G.D GF: 11441 Prob: 3898 Other: 3015\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052884.6445355 INFO: cpd05178 not found in model!\n",
      "1723052884.6466005 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2.022.pg.G.D GF: 11198 Prob: 4192 Other: 2962\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052886.369366 INFO: cpd05178 not found in model!\n",
      "1723052886.3716922 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052890.3948975 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723052895.867009 INFO: Objective with gapfill database:157.7239765021045; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052900.4842663 INFO: Expansion time:Ac.NO3:0.6170629040007043\n",
      "1723052900.4862669 INFO: Filtered count:0 out of 1523\n",
      "1723052901.0921772 INFO: Expansion time:Ac.O2:0.6046603729992057\n",
      "1723052901.0942478 INFO: Filtered count:0 out of 1523\n",
      "1723052901.7078633 INFO: Expansion time:Etho.O2:0.6122427080008492\n",
      "1723052901.7099388 INFO: Filtered count:0 out of 1523\n",
      "1723052902.319066 INFO: Expansion time:For.NO2:0.6077497450005467\n",
      "1723052902.321016 INFO: Filtered count:0 out of 1523\n",
      "1723052902.929983 INFO: Expansion time:For.NO3:0.6076812159990368\n",
      "1723052902.9320643 INFO: Filtered count:0 out of 1523\n",
      "1723052903.5416234 INFO: Expansion time:For.O2:0.6082470679994003\n",
      "1723052903.5438802 INFO: Filtered count:0 out of 1523\n",
      "1723052904.1588275 INFO: Expansion time:Fum:0.613627535998603\n",
      "1723052904.1609778 INFO: Filtered count:0 out of 1523\n",
      "1723052904.7731347 INFO: Expansion time:Fum.O2:0.610470842999348\n",
      "1723052904.7753608 INFO: Filtered count:0 out of 1523\n",
      "1723052905.3849957 INFO: Expansion time:Glyc.O2:0.6081116469995322\n",
      "1723052905.3871367 INFO: Filtered count:0 out of 1523\n",
      "1723052905.990865 INFO: Expansion time:Pyr:0.6023141910009144\n",
      "1723052905.9927976 INFO: Filtered count:0 out of 1523\n",
      "1723052908.4628856 INFO: Expansion time:Pyr.DMSO:2.46872645700023\n",
      "1723052908.46495 INFO: Filtered count:0 out of 1523\n",
      "1723052909.0768497 INFO: Expansion time:Pyr.NO:0.6102850639999815\n",
      "1723052909.0790002 INFO: Filtered count:0 out of 1523\n",
      "1723052909.6830556 INFO: Expansion time:Pyr.NO2:0.6026767760013172\n",
      "1723052909.6849723 INFO: Filtered count:0 out of 1523\n",
      "1723052910.2893794 INFO: Expansion time:Pyr.NO3:0.6030392849988857\n",
      "1723052910.2915459 INFO: Filtered count:0 out of 1523\n",
      "1723052910.347682 INFO: cpd05178 not found in model!\n",
      "1723052910.3499668 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052910.8962462 INFO: Expansion time:Pyr.O2:0.6034141009986342\n",
      "1723052910.898379 INFO: Filtered count:0 out of 1523\n",
      "1723052911.502056 INFO: Expansion time:Pyr.SO3:0.6023447999996279\n",
      "1723052911.5041199 INFO: Filtered count:0 out of 1523\n",
      "1723052911.5241911 INFO: cpd05178 not found in model!\n",
      "1723052911.526165 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723052912.1085205 INFO: Expansion time:Pyr.SO4:0.6032071600002382\n",
      "1723052912.1105027 INFO: Filtered count:0 out of 1523\n",
      "1723052912.7172852 INFO: Expansion time:Pyr.TMAO:0.6055877749986394\n",
      "1723052912.719261 INFO: Filtered count:0 out of 1523\n",
      "1723052913.3226109 INFO: Expansion time:empty:0.6020766560013726\n",
      "1723052913.3248875 INFO: Filtered count:0 out of 1523\n",
      "1723052917.3589678 INFO: cpd05178 not found in model!\n",
      "1723052917.573564 INFO: cpd05178 not found in model!\n",
      "1723052917.5755048 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052920.5408356 INFO: cpd05178 not found in model!\n",
      "1723052920.7460964 INFO: cpd05178 not found in model!\n",
      "1723052920.748216 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[280 'R1_B_D2.014.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:02:53+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '75204cef287737f5605b4d988701aa0f'\n",
      " 12104684 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052941.1562634 INFO: metabolites 1132\n",
      "1723052941.599664 INFO: reactions 1128\n",
      "1723052941.7294025 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052942.1379468 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/199/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052950.1962137 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723052950.4847052 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052956.859557 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723052963.544697 INFO: cpd05178 not found in model!\n",
      "1723052963.54734 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2.014.pg.G.D GF: 11291 Prob: 4164 Other: 2897\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723052970.6203735 INFO: cpd05178 not found in model!\n",
      "1723052970.8428857 INFO: cpd05178 not found in model!\n",
      "1723052970.8452497 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052973.0100691 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723052984.661885 INFO: cpd05178 not found in model!\n",
      "1723052986.2527268 INFO: Expansion time:Ac.O2:3.9389483939994534\n",
      "1723052986.2550242 INFO: Filtered count:0 out of 1653\n",
      "1723052986.8365304 INFO: cpd05178 not found in model!\n",
      "1723052986.8388221 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052989.2367203 INFO: Expansion time:Etho.O2:2.980421854999804\n",
      "1723052989.2389734 INFO: Filtered count:0 out of 1653\n",
      "1723052990.827111 INFO: Expansion time:Fum:1.5868254619999789\n",
      "1723052990.8294451 INFO: Filtered count:0 out of 1653\n",
      "1723052992.5333836 INFO: Expansion time:Fum.O2:1.7025472059995082\n",
      "1723052992.5355935 INFO: Filtered count:0 out of 1653\n",
      "1723052994.013868 INFO: Expansion time:Pyr:1.4769513139999617\n",
      "1723052994.0161564 INFO: Filtered count:0 out of 1653\n",
      "1723052994.7221415 INFO: Expansion time:Pyr.DMSO:0.7045635319991561\n",
      "1723052994.7244744 INFO: Filtered count:0 out of 1653\n",
      "1723052995.4171329 INFO: Expansion time:Pyr.NO:0.6912815520008735\n",
      "1723052995.4194446 INFO: Filtered count:0 out of 1653\n",
      "1723052996.0783472 INFO: Expansion time:Pyr.NO2:0.657515314000193\n",
      "1723052996.0805602 INFO: Filtered count:0 out of 1653\n",
      "1723052996.1062667 INFO: cpd05178 not found in model!\n",
      "1723052996.3399673 INFO: cpd05178 not found in model!\n",
      "1723052996.3420198 INFO: Media compound: cpd05178 not found in model.\n",
      "1723052996.7709665 INFO: Expansion time:Pyr.NO3:0.6890226790001179\n",
      "1723052996.7732391 INFO: Filtered count:0 out of 1653\n",
      "1723052998.469006 INFO: Expansion time:Pyr.O2:1.6943208360007702\n",
      "1723052998.4713075 INFO: Filtered count:0 out of 1653\n",
      "1723053000.308942 INFO: Expansion time:Pyr.SO3:1.8363237650009978\n",
      "1723053000.3111944 INFO: Filtered count:0 out of 1653\n",
      "1723053001.1347098 INFO: Expansion time:Pyr.SO4:0.8221239449994755\n",
      "1723053001.1369374 INFO: Filtered count:0 out of 1653\n",
      "1723053001.7577488 INFO: Expansion time:Pyr.TMAO:0.6193978770006652\n",
      "1723053001.759909 INFO: Filtered count:0 out of 1653\n",
      "1723053004.1561413 INFO: Expansion time:empty:2.3948379939993174\n",
      "1723053004.1583607 INFO: Filtered count:0 out of 1653\n",
      "1723053005.0596845 INFO: cpd05178 not found in model!\n",
      "1723053005.2896357 INFO: cpd05178 not found in model!\n",
      "1723053005.2918994 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[300 'R2_B_D2.10.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:07:14+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' '3434763263c4554afe11aeb43412273b'\n",
      " 16505768 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723053012.837612 INFO: metabolites 1295\n",
      "1723053013.4171517 INFO: reactions 1288\n",
      "1723053013.5752823 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723053014.08021 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/362/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723053024.2104208 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723053026.4352498 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723053028.8278933 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723053033.9506052 INFO: cpd05178 not found in model!\n",
      "1723053033.952778 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053037.8308275 INFO: cpd05178 not found in model!\n",
      "1723053037.8329556 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.10.pg.G.D GF: 10982 Prob: 4466 Other: 2906\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723053046.7110708 INFO: Objective with gapfill database:0.0; min objective:0.01\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[264 'R2_B_D2.124.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T02:59:53+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'c46fabb740e407eb063088681eea116a'\n",
      " 12283003 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723053053.8941286 INFO: metabolites 1131\n",
      "1723053054.3626683 INFO: reactions 1073\n",
      "1723053054.4880338 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723053054.8977764 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/146/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723053060.2981226 INFO: Expansion time:Ac.NO2:4.328969221000079\n",
      "1723053060.3001664 INFO: Filtered count:0 out of 1861\n",
      "1723053061.576352 INFO: Expansion time:Ac.O2:1.2748492200007604\n",
      "1723053061.578578 INFO: Filtered count:0 out of 1861\n",
      "1723053062.720379 INFO: Expansion time:Etho.O2:1.1403155130010418\n",
      "1723053062.722441 INFO: Filtered count:0 out of 1861\n",
      "1723053063.5506825 INFO: Expansion time:For.NO2:0.8268766830005916\n",
      "1723053063.5527024 INFO: Filtered count:0 out of 1861\n",
      "1723053064.466108 INFO: Expansion time:For.O2:0.9119251089996396\n",
      "1723053064.4682183 INFO: Filtered count:0 out of 1861\n",
      "1723053065.7947962 INFO: Expansion time:Fum:1.3252306659996975\n",
      "1723053065.7969093 INFO: Filtered count:0 out of 1861\n",
      "1723053065.9427562 INFO: cpd05178 not found in model!\n",
      "1723053065.944671 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053067.316376 INFO: Expansion time:Fum.O2:1.5180725299996993\n",
      "1723053067.3186245 INFO: Filtered count:0 out of 1861\n",
      "1723053068.6269193 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723053068.9371126 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723053071.1660705 INFO: Expansion time:Pyr:3.8457646560000285\n",
      "1723053071.1682298 INFO: Filtered count:0 out of 1861\n",
      "1723053071.3217063 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723053072.0769384 INFO: Expansion time:Pyr.DMSO:0.9073517990000255\n",
      "1723053072.0791283 INFO: Filtered count:0 out of 1861\n",
      "1723053073.090159 INFO: Expansion time:Pyr.NO:1.0096381980001752\n",
      "1723053073.092249 INFO: Filtered count:0 out of 1861\n",
      "1723053074.463055 INFO: Expansion time:Pyr.NO2:1.36950539999998\n",
      "1723053074.4649637 INFO: Filtered count:0 out of 1861\n",
      "1723053075.8065732 INFO: Expansion time:Pyr.NO3:1.3403065509992302\n",
      "1723053075.8086624 INFO: Filtered count:0 out of 1861\n",
      "1723053077.5580268 INFO: Expansion time:Pyr.O2:1.7480913449999207\n",
      "1723053077.5601268 INFO: Filtered count:0 out of 1861\n",
      "1723053078.938243 INFO: Expansion time:Pyr.SO3:1.3768561859997135\n",
      "1723053078.9403675 INFO: Filtered count:0 out of 1861\n",
      "1723053080.181631 INFO: cpd05178 not found in model!\n",
      "1723053080.1839864 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2.124.pg.G.D GF: 11203 Prob: 4252 Other: 2899\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723053081.8541064 INFO: Expansion time:Pyr.SO4:2.912428515999636\n",
      "1723053081.8561862 INFO: Filtered count:0 out of 1861\n",
      "1723053082.5601 INFO: Expansion time:Pyr.TMAO:0.7024232910007413\n",
      "1723053082.5622766 INFO: Filtered count:0 out of 1861\n",
      "1723053083.2297065 INFO: Expansion time:empty:0.6660754820004513\n",
      "1723053083.2318 INFO: Filtered count:0 out of 1861\n",
      "1723053089.63461 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "1723053092.4740222 INFO: cpd05178 not found in model!\n",
      "1723053092.476214 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723053103.1655204 INFO: Expansion time:Ac.O2:3.911341415998322\n",
      "1723053103.167633 INFO: Filtered count:0 out of 1577\n",
      "1723053104.6202085 INFO: Expansion time:Etho.O2:1.4512748590004776\n",
      "1723053104.622296 INFO: Filtered count:0 out of 1577\n",
      "1723053105.836573 INFO: Expansion time:Pyr:1.2129639260001568\n",
      "1723053105.8387148 INFO: Filtered count:0 out of 1577\n",
      "1723053108.3791022 INFO: Expansion time:Pyr.DMSO:2.5391641189999063\n",
      "1723053108.3811216 INFO: Filtered count:0 out of 1577\n",
      "1723053109.1456265 INFO: Expansion time:Pyr.NO:0.7632419370002026\n",
      "1723053109.1477606 INFO: Filtered count:0 out of 1577\n",
      "1723053110.1240332 INFO: Expansion time:Pyr.NO2:0.9750381079993531\n",
      "1723053110.1261928 INFO: Filtered count:0 out of 1577\n",
      "1723053110.8693058 INFO: Expansion time:Pyr.NO3:0.741907978999734\n",
      "1723053110.8715148 INFO: Filtered count:0 out of 1577\n",
      "1723053112.264189 INFO: Expansion time:Pyr.O2:1.391356084999643\n",
      "1723053112.2663486 INFO: Filtered count:0 out of 1577\n",
      "1723053113.6058884 INFO: Expansion time:Pyr.SO3:1.3381846290012618\n",
      "1723053113.6079822 INFO: Filtered count:0 out of 1577\n",
      "1723053114.3964925 INFO: Expansion time:Pyr.SO4:0.7872216119994846\n",
      "1723053114.3985004 INFO: Filtered count:0 out of 1577\n",
      "1723053115.1076345 INFO: Expansion time:Pyr.TMAO:0.7078788559992972\n",
      "1723053115.1096318 INFO: Filtered count:0 out of 1577\n",
      "1723053115.7358618 INFO: Expansion time:empty:0.6249141610005609\n",
      "1723053115.7378542 INFO: Filtered count:0 out of 1577\n",
      "1723053137.5854554 INFO: cpd05178 not found in model!\n",
      "1723053137.8081543 INFO: cpd05178 not found in model!\n",
      "1723053137.8101082 INFO: Media compound: cpd05178 not found in model.\n",
      "1723053140.6330285 INFO: cpd05178 not found in model!\n",
      "1723053140.6351109 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053148.6056614 INFO: cpd05178 not found in model!\n",
      "1723053148.6079438 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053158.9736588 INFO: cpd05178 not found in model!\n",
      "1723053158.9757648 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053177.7664475 INFO: cpd05178 not found in model!\n",
      "1723053177.9933226 INFO: cpd05178 not found in model!\n",
      "1723053177.9955506 INFO: Media compound: cpd05178 not found in model.\n",
      "1723053197.3359637 INFO: cpd05178 not found in model!\n",
      "1723053197.3380535 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053205.6905482 INFO: cpd05178 not found in model!\n",
      "1723053205.6926033 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053248.4665945 INFO: cpd05178 not found in model!\n",
      "1723053248.468831 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053253.2696164 INFO: cpd05178 not found in model!\n",
      "1723053253.2718394 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053256.3676848 INFO: cpd05178 not found in model!\n",
      "1723053256.591399 INFO: cpd05178 not found in model!\n",
      "1723053256.5935075 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[292 'R2A_A_D1.24.pg.G.D' 'KBaseGenomes.Genome-11.1'\n",
      " '2023-09-02T03:05:37+0000' 1 'cliff_bueno' 155805\n",
      " 'cliff_bueno:narrative_1693612235241' 'a66935e805207fe72b3542bb7b04a3c6'\n",
      " 8694483 None]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723053298.3397586 INFO: metabolites 1037\n",
      "1723053298.7774925 INFO: reactions 1018\n",
      "1723053298.9117908 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723053299.300493 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "188040/276/1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723053310.9152334 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n",
      "1723053311.3434656 WARNING: Ignoring reaction 'rxn00062_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting tests from attributes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723053314.3989382 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723053319.624403 INFO: cpd05178 not found in model!\n",
      "1723053319.9004157 INFO: cpd05178 not found in model!\n",
      "1723053319.9091704 INFO: Media compound: cpd05178 not found in model.\n",
      "1723053327.7071722 INFO: cpd05178 not found in model!\n",
      "1723053327.7139208 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1.24.pg.G.D GF: 11424 Prob: 3954 Other: 2974\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723053338.5670395 INFO: Objective with gapfill database:0.0; min objective:0.01\n",
      "/opt/env/python3_modelseed/lib/python3.11/site-packages/cobra/util/solver.py:554: UserWarning: Solver status is 'infeasible'.\n",
      "  warn(f\"Solver status is '{status}'.\", UserWarning)\n",
      "1723053352.1110282 INFO: Expansion time:Glc:0.8522181410007761\n",
      "1723053352.115936 INFO: Filtered count:0 out of 1456\n",
      "1723053352.9710886 INFO: Expansion time:Glc.DMSO:0.8423251710009936\n",
      "1723053352.9835794 INFO: Filtered count:0 out of 1456\n",
      "1723053353.8428197 INFO: Expansion time:Glc.O2:0.8555071569990105\n",
      "1723053353.8469527 INFO: Filtered count:0 out of 1456\n",
      "1723053357.1709313 INFO: Expansion time:Glc.TMAO:3.3225352079989534\n",
      "1723053357.1739745 INFO: Filtered count:0 out of 1456\n",
      "1723053357.9660146 INFO: Expansion time:Glyc:0.7903525109995826\n",
      "1723053357.9730146 INFO: Filtered count:0 out of 1456\n",
      "1723053358.9634717 INFO: Expansion time:Glyc.O2:0.9857639450001443\n",
      "1723053358.972334 INFO: Filtered count:0 out of 1456\n",
      "1723053359.9550784 INFO: Expansion time:Pyr:0.9793767000010121\n",
      "1723053359.9608731 INFO: Filtered count:0 out of 1456\n",
      "1723053360.7516298 INFO: Expansion time:Pyr.DMSO:0.7834778179985733\n",
      "1723053360.7593277 INFO: Filtered count:0 out of 1456\n",
      "1723053361.6598368 INFO: Expansion time:Pyr.NO:0.8970121410002321\n",
      "1723053361.6690798 INFO: Filtered count:0 out of 1456\n",
      "1723053362.4757338 INFO: Expansion time:Pyr.NO2:0.8015603229996486\n",
      "1723053362.4796627 INFO: Filtered count:0 out of 1456\n",
      "1723053363.3797562 INFO: Expansion time:Pyr.NO3:0.8974015529984172\n",
      "1723053363.3884282 INFO: Filtered count:0 out of 1456\n",
      "1723053364.2473474 INFO: Expansion time:Pyr.O2:0.8568371210003534\n",
      "1723053364.2500885 INFO: Filtered count:0 out of 1456\n",
      "1723053365.021964 INFO: Expansion time:Pyr.SO3:0.7692518469993956\n",
      "1723053365.0261428 INFO: Filtered count:0 out of 1456\n",
      "1723053365.8045466 INFO: Expansion time:Pyr.SO4:0.7757507860005717\n",
      "1723053365.8128161 INFO: Filtered count:0 out of 1456\n",
      "1723053366.6157393 INFO: Expansion time:Pyr.TMAO:0.7950065130007715\n",
      "1723053366.6233149 INFO: Filtered count:0 out of 1456\n",
      "1723053367.5418067 INFO: Expansion time:empty:0.9158094340000389\n",
      "1723053367.546949 INFO: Filtered count:0 out of 1456\n",
      "1723053382.1970513 INFO: cpd05178 not found in model!\n",
      "1723053382.2016377 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053394.982816 INFO: cpd05178 not found in model!\n",
      "1723053395.2797174 INFO: cpd05178 not found in model!\n",
      "1723053395.2874627 INFO: Media compound: cpd05178 not found in model.\n",
      "1723053428.0732872 INFO: cpd05178 not found in model!\n",
      "1723053428.0754187 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053448.9428544 INFO: cpd05178 not found in model!\n",
      "1723053448.9491088 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053465.23525 INFO: cpd05178 not found in model!\n",
      "1723053465.2422674 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053478.7331731 INFO: cpd05178 not found in model!\n",
      "1723053479.0458949 INFO: cpd05178 not found in model!\n",
      "1723053479.051745 INFO: Media compound: cpd05178 not found in model.\n",
      "1723053555.9903867 INFO: cpd05178 not found in model!\n",
      "1723053555.9925957 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053576.2062383 INFO: cpd05178 not found in model!\n",
      "1723053576.4303877 INFO: cpd05178 not found in model!\n",
      "1723053576.4324582 INFO: Media compound: cpd05178 not found in model.\n",
      "1723053588.2531374 INFO: cpd05178 not found in model!\n",
      "1723053588.2596521 WARNING: cpd05178: could not find metabolite for cpd05178\n",
      "1723053602.040426 INFO: cpd05178 not found in model!\n",
      "1723053602.157436 INFO: cpd05178 not found in model!\n",
      "1723053602.3739612 INFO: cpd05178 not found in model!\n",
      "1723053602.3885303 INFO: Media compound: cpd05178 not found in model.\n",
      "1723053602.5001194 INFO: cpd05178 not found in model!\n",
      "1723053602.5124218 INFO: Media compound: cpd05178 not found in model.\n",
      "1723053634.0326192 INFO: cpd05178 not found in model!\n",
      "1723053634.4289715 INFO: cpd05178 not found in model!\n",
      "1723053634.4366431 INFO: Media compound: cpd05178 not found in model.\n",
      "1723053691.1774635 INFO: cpd05178 not found in model!\n",
      "1723053691.5110085 INFO: cpd05178 not found in model!\n",
      "1723053691.5207665 INFO: Media compound: cpd05178 not found in model.\n",
      "1723053744.4136987 INFO: cpd05178 not found in model!\n",
      "1723053744.7309415 INFO: cpd05178 not found in model!\n",
      "1723053744.740591 INFO: Media compound: cpd05178 not found in model.\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "\n",
    "def smipps(arguments):\n",
    "    # unpack the arguments\n",
    "    (mag_list, mags_to_models, model_ws, metabolites, feature_probabilities, reaction_probabilities, problemlist, auxo_media, gmm_base_media, aerobicity\n",
    "    ) = arguments\n",
    "    \n",
    "    # print status\n",
    "    from multiprocess import current_process\n",
    "    pid = current_process().name\n",
    "    print(pid)\n",
    "    \n",
    "    # define the phenotype sets\n",
    "    uptake_phenoset = util.create_phenotypeset_from_compounds(\n",
    "        metabolites,\n",
    "        base_media=auxo_media,\n",
    "        base_uptake=0,\n",
    "        base_excretion=1000,\n",
    "        global_atom_limits={},\n",
    "        type=\"uptake\"\n",
    "    )\n",
    "    excretion_phenoset = util.create_phenotypeset_from_compounds(\n",
    "        metabolites,\n",
    "        base_media=auxo_media,\n",
    "        base_uptake=0,\n",
    "        base_excretion=1000,\n",
    "        global_atom_limits={},\n",
    "        type=\"excretion\"\n",
    "    )\n",
    "    growth_phenoset = util.create_phenotypeset_from_compounds(\n",
    "        metabolites,\n",
    "        base_media=gmm_base_media,\n",
    "        base_uptake=0,\n",
    "        base_excretion=1000,\n",
    "        global_atom_limits={},\n",
    "        type=\"growth\"\n",
    "    )\n",
    "    phenosets = {\"uptake\":uptake_phenoset,\"excretion\":excretion_phenoset,\"growth\":growth_phenoset}\n",
    "    \n",
    "    # compute the SMIPP for each MAG\n",
    "    gf_phenotype_results = {}\n",
    "    for i,mag in enumerate(mag_list):\n",
    "        ## load the mode\n",
    "        print(mag)\n",
    "        name = mag[1]\n",
    "        if name in problemlist:  continue\n",
    "        try:\n",
    "            mdlutl = util.msrecon.get_model(name+model_suffix,model_ws)\n",
    "        except Exception as e:\n",
    "            print(f\"The {name+model_suffix} object is not in the {model_ws} worksheet\")\n",
    "            print(e)\n",
    "            \n",
    "        ## determine the highest probability for each gene associated with a reaction in the model\n",
    "        reaction_probabilities[name] = {}\n",
    "        for rxn in mdlutl.model.reactions:\n",
    "            highest_prob = None\n",
    "            for gene in rxn.genes:\n",
    "                if gene.id in feature_probabilities[mag[1]]:\n",
    "                    if highest_prob == None or feature_probabilities[mag[1]][gene.id] > highest_prob:\n",
    "                        highest_prob = feature_probabilities[mag[1]][gene.id]\n",
    "            if highest_prob != None:\n",
    "                rxn.probability = highest_prob\n",
    "                reaction_probabilities[name][rxn.id] = highest_prob\n",
    "\n",
    "        ## assign the highest possible probability among the annotated genes for each reaction\n",
    "        print(mdlutl.model.genome_ref)\n",
    "        genome = util.msrecon.get_msgenome_from_ontology(mdlutl.model.genome_ref,native_python_api=True,output_ws=None)\n",
    "        reaction_hash = genome.annoont.get_reaction_gene_hash(feature_type=\"gene\")\n",
    "        for rxn in reaction_hash:\n",
    "            highest_prob = None\n",
    "            for gene in reaction_hash[rxn]:\n",
    "                if gene in feature_probabilities[mag[1]]:\n",
    "                    if highest_prob == None or feature_probabilities[mag[1]][gene] > highest_prob:\n",
    "                        highest_prob = feature_probabilities[mag[1]][gene]\n",
    "            if highest_prob != None and (rxn+\"_c0\" not in reaction_probabilities[name] or highest_prob >= reaction_probabilities[name][rxn+\"_c0\"]):\n",
    "                if rxn+\"_c0\" in mdlutl.model.reactions:\n",
    "                    mdlutl.model.reactions.get_by_id(rxn+\"_c0\").probability = highest_prob\n",
    "                reaction_probabilities[name][rxn+\"_c0\"] = highest_prob\n",
    "\n",
    "        ## gapfill the model\n",
    "        filters = mdlutl.get_attributes(\"gf_filter\")\n",
    "        tests = mdlutl.get_atp_tests(core_template=util.msrecon.core_template,atp_media_filename=util.msrecon.module_dir+\"/data/atp_medias.tsv\",recompute=False)\n",
    "        msgapfill = MSGapfill(\n",
    "            mdlutl,\n",
    "            [util.msrecon.get_template(mdlutl.model.template_ref)],\n",
    "            [],\n",
    "            tests,\n",
    "            blacklist=[],\n",
    "            default_target=\"bio1\",\n",
    "            minimum_obj=0.01,\n",
    "            base_media=None,\n",
    "            base_media_target_element=None\n",
    "        )\n",
    "\n",
    "        ### Adding missing transporter for metabolites to gapfilling database\n",
    "        for cpd in metabolites:\n",
    "            if \"EX_\"+cpd+\"_e0\" not in msgapfill.gfmodelutl.model.reactions:\n",
    "                transport = msgapfill.gfmodelutl.add_transport_and_exchange_for_metabolite(cpd,direction=\"=\",prefix=\"trans\",override=False)\n",
    "\n",
    "        ## create flux coefficients from the gapfilling\n",
    "        coefficients = {}\n",
    "        gf_penalties = msgapfill.gfpkgmgr.getpkg(\"GapfillingPkg\").gapfilling_penalties\n",
    "        gfrxn = 0\n",
    "        probrxn = 0\n",
    "        otherrxn = 0\n",
    "        for reaction in msgapfill.gfmodelutl.model.reactions:\n",
    "            if reaction.id in reaction_probabilities[name]:\n",
    "                probrxn += 2\n",
    "                coefficients[\">\"+reaction.id] = 1-reaction_probabilities[name][reaction.id]\n",
    "                coefficients[\"<\"+reaction.id] = 1-reaction_probabilities[name][reaction.id]\n",
    "            elif reaction.id in gf_penalties:\n",
    "                if \"forward\" in gf_penalties[reaction.id]:\n",
    "                    gfrxn += 1\n",
    "                    coefficients[\">\"+reaction.id] = 1+gf_penalties[reaction.id][\"forward\"]\n",
    "                else:\n",
    "                    otherrxn += 1\n",
    "                    coefficients[\">\"+reaction.id] = 0.95\n",
    "                if \"reverse\" in gf_penalties[reaction.id]:\n",
    "                    gfrxn += 1\n",
    "                    coefficients[\"<\"+reaction.id] = 1+gf_penalties[reaction.id][\"reverse\"]\n",
    "                else:\n",
    "                    otherrxn += 1\n",
    "                    coefficients[\"<\"+reaction.id] = 0.95\n",
    "            else:\n",
    "                otherrxn += 2\n",
    "                coefficients[\">\"+reaction.id] = 0.95\n",
    "                coefficients[\"<\"+reaction.id] = 0.95\n",
    "        print(name,\"GF:\",gfrxn,\"Prob:\",probrxn,\"Other:\",otherrxn)\n",
    "\n",
    "        #TODO Create conditional logic for growth phenotypes if errors are raised here \n",
    "        msgapfill.prefilter(test_conditions=tests,growth_conditions=[],use_prior_filtering=True,base_filter_only=True)\n",
    "\n",
    "        ## simulate the SMIPP condition and parse the results\n",
    "        gf_phenotype_results[name] = {}\n",
    "        for phenoid in phenosets:\n",
    "            gf_phenotype_results[name][phenoid] = {}\n",
    "            output = phenosets[phenoid].simulate_phenotypes(\n",
    "                msgapfill.gfmodelutl,\n",
    "                multiplier=2,\n",
    "                add_missing_exchanges=True,\n",
    "                save_fluxes=False,\n",
    "                save_reaction_list=True,\n",
    "                gapfill_negatives=False,\n",
    "                msgapfill=None,\n",
    "                test_conditions=None,\n",
    "                ignore_experimental_data=True,\n",
    "                flux_coefficients=coefficients\n",
    "            )\n",
    "            for index, row in output[\"details\"].iterrows():\n",
    "                if \"reactions\" in output[\"data\"][row[\"Phenotype\"]]:\n",
    "                    output[\"data\"][row[\"Phenotype\"]][\"average_probability\"] = 0\n",
    "                    for rxn in output[\"data\"][row[\"Phenotype\"]][\"reactions\"]:\n",
    "                        direction = rxn[0:1]\n",
    "                        rxnid = rxn[1:]\n",
    "                        if direction == \">\":\n",
    "                            if rxnid not in gf_penalties or \"forward\" not in gf_penalties[rxnid]:\n",
    "                                if rxnid in reaction_probabilities[name]:\n",
    "                                    output[\"data\"][row[\"Phenotype\"]][\"average_probability\"] += reaction_probabilities[name][rxnid]\n",
    "                                else:\n",
    "                                    output[\"data\"][row[\"Phenotype\"]][\"average_probability\"] += 0.05\n",
    "                        elif direction == \"<\":\n",
    "                            if rxnid not in gf_penalties or \"reverse\" not in gf_penalties[rxnid]:\n",
    "                                if rxnid in reaction_probabilities[name]:\n",
    "                                    output[\"data\"][row[\"Phenotype\"]][\"average_probability\"] += reaction_probabilities[name][rxnid]\n",
    "                                else:\n",
    "                                    output[\"data\"][row[\"Phenotype\"]][\"average_probability\"] += 0.05\n",
    "                    output[\"data\"][row[\"Phenotype\"]][\"average_probability\"] = output[\"data\"][row[\"Phenotype\"]][\"average_probability\"]/len(output[\"data\"][row[\"Phenotype\"]][\"reactions\"])\n",
    "                gf_phenotype_results[name][phenoid][row[\"Phenotype\"]] = output[\"data\"][row[\"Phenotype\"]]\n",
    "            \n",
    "    util.save(f\"SMIPPs/new_gf_phenotype_results_{pid}\",gf_phenotype_results)\n",
    "    util.save(f\"SMIPPs/reaction_probabilities_{pid}\",reaction_probabilities)\n",
    "                \n",
    "# define function parameters\n",
    "mag_list = util.load(\"mag_list\")\n",
    "model_ws = 188065\n",
    "model_suffix = \".pyr\"\n",
    "metabolites = util.load(\"metabolites\")\n",
    "feature_probabilities = util.load(\"feature_probabilities\")\n",
    "reaction_probabilities = util.load(\"reaction_probabilities\")\n",
    "problemlist = util.load(\"problemlist\",[])\n",
    "mag_to_name = util.load(\"mag_to_name\")\n",
    "for mag in mag_list:\n",
    "    mag[1] = mag_to_name[mag[1]]+\".pg.G.D\"\n",
    "mag_feature_probabilities = {mag_to_name[k]+\".pg.G.D\":v for k,v in feature_probabilities.items()}\n",
    "\n",
    "auxo_media = util.msrecon.get_media(f\"{model_ws}/AuxoMedia\")\n",
    "gmm_base_media = util.msrecon.get_media(f\"{model_ws}/PyruateMinimalAerobic\")\n",
    "\n",
    "# run the parallelized code\n",
    "parallelize = True\n",
    "if parallelize:\n",
    "    from multiprocess import Pool\n",
    "    from os import cpu_count\n",
    "    from numpy import array_split\n",
    "    \n",
    "    # the MAGs are partitioned into groups of 4, and a thread is created to handle each group \n",
    "    mag_lists = array_split(mag_list, int(len(mag_list)/4))\n",
    "    pool = Pool(min(cpu_count(), len(mag_lists)))\n",
    "    args = [(mag_list, mag_to_name, model_ws, metabolites, mag_feature_probabilities, reaction_probabilities, problemlist, auxo_media, gmm_base_media, \"aerobic\")\n",
    "           for mag_list in mag_lists]\n",
    "    list_of_outputs = pool.map(smipps, args)\n",
    "else:\n",
    "    outputs = smipps((remaining_mags, mag_to_name, model_ws, metabolites, mag_feature_probabilities, reaction_probabilities, problemlist, auxo_media, gmm_base_media, \"aerobic\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combining the outputs from each parallel thread"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from json import load, dump\n",
    "from glob import glob\n",
    "\n",
    "new_gf_phenotype_results = {}\n",
    "for path in glob(r\"datacache/SMIPPs/new_gf_phenotype_results_*.json\"):\n",
    "    with open(path, 'r') as jsonIn:\n",
    "        new_gf_phenotype_results.update(load(jsonIn))\n",
    "        \n",
    "with open(\"datacache/SMIPPs/new_gf_phenotype_results.json\", 'w') as jsonOut:\n",
    "    dump(new_gf_phenotype_results, jsonOut, indent=3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "310\n"
     ]
    }
   ],
   "source": [
    "print(len(new_gf_phenotype_results.keys()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from json import load, dump\n",
    "from glob import glob\n",
    "\n",
    "reaction_probabilities = {}\n",
    "for path in glob(r\"datacache/SMIPPs/reaction_probabilities_*.json\"):\n",
    "    with open(path, 'r') as jsonIn:\n",
    "        reaction_probabilities.update(load(jsonIn))\n",
    "        \n",
    "with open(\"datacache/SMIPPs/reaction_probabilities.json\", 'w') as jsonOut:\n",
    "    dump(reaction_probabilities, jsonOut, indent=3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computing proper gapfill and filtering bad predictions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### not penalize existing gapfilled reactions, and emphasizing the probabilites of newly gapfilled reactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python version 3.11.1\n",
      "KBBaseModules 0.0.1\n",
      "modelseedpy 0.3.3\n",
      "cobrakbase 0.3.1\n",
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n",
      "77 threads will be created\n",
      "ForkPoolWorker-1\n",
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45.pangenome.RAST.glm4ec.pyr\n",
      "ForkPoolWorker-2\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.88.pangenome.RAST.glm4ec.pyr\n",
      "ForkPoolWorker-3\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055219.2942505 INFO: metabolites 1256\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-4\n",
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_concoct_out.69.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055219.406164 INFO: metabolites 1042\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-5\n",
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055219.5392497 INFO: metabolites 1116\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-6\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.10.pangenome.RAST.glm4ec.pyr\n",
      "ForkPoolWorker-7\n",
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055219.8456206 INFO: metabolites 882\n",
      "1723055219.9331672 INFO: metabolites 982\n",
      "1723055219.9405308 INFO: reactions 1001\n",
      "1723055220.0891407 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055220.102574 INFO: reactions 1106\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-8\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055220.148961 INFO: metabolites 1144\n",
      "1723055220.2818356 INFO: reactions 848\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-9\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055220.41304 INFO: metabolites 1173\n",
      "1723055220.4374707 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055220.4883952 INFO: reactions 921\n",
      "1723055220.6013997 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055220.6533544 INFO: metabolites 1210\n",
      "1723055220.6733508 INFO: reactions 1229\n",
      "1723055220.68615 INFO: reactions 1072\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-10\n",
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_metabat.29.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055220.81982 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055220.8248057 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055220.8526137 INFO: metabolites 1209\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-11\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.34.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055220.9968016 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055220.9996612 INFO: reactions 1170\n",
      "1723055221.0598269 INFO: metabolites 815\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-12\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.73.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055221.1924188 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.57.pangenome.RAST.glm4ec.pyr\n",
      "ForkPoolWorker-13\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.25.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055221.3892035 INFO: Default biomass: [bio1]\n",
      "1723055221.4269526 INFO: metabolites 1059\n",
      "1723055221.4281337 INFO: Default biomass: [bio1]\n",
      "1723055221.4352586 INFO: reactions 702\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-14\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.50.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055221.5390553 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_metabat.29.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055221.5494354 INFO: metabolites 1055\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.17.pangenome.RAST.glm4ec.pyr\n",
      "ForkPoolWorker-15\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.21.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055221.6431022 INFO: Default biomass: [bio1]\n",
      "1723055221.6729016 INFO: metabolites 1109\n",
      "1723055221.7502563 INFO: metabolites 1082\n",
      "1723055221.7559311 INFO: metabolites 473\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_metabat.16.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055221.8325458 INFO: Default biomass: [bio1]\n",
      "1723055221.8551779 INFO: Default biomass: [bio1]\n",
      "1723055221.8664792 INFO: reactions 422\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-16\n",
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_concoct_out.45.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055221.925823 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_concoct_out.34.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.17.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055221.9836078 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055222.0371048 INFO: metabolites 1078\n",
      "1723055222.050762 INFO: Default biomass: [bio1]\n",
      "1723055222.0779018 INFO: Default biomass: [bio1]\n",
      "1723055222.0820098 INFO: reactions 977\n",
      "1723055222.1008847 INFO: metabolites 949\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055222.1028903 INFO: metabolites 1083\n",
      "1723055222.1089928 INFO: reactions 1029\n",
      "1723055222.1697524 INFO: metabolites 1147\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_H2O_MG_DASTool_bins_metabat.16.pangenome.RAST.glm4ec.pyr\n",
      "ForkPoolWorker-17"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055222.2055666 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.46.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055222.2469904 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055222.3173294 INFO: reactions 1202\n",
      "1723055222.3473284 INFO: reactions 1033\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-18\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.7.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055222.3560734 INFO: metabolites 1000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055222.3621738 INFO: metabolites 1042\n",
      "1723055222.387912 INFO: reactions 1124\n",
      "1723055222.4813576 INFO: Default biomass: [bio1]\n",
      "1723055222.4952602 INFO: reactions 1176\n",
      "1723055222.507784 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055222.515382 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055222.5289233 INFO: metabolites 1014\n",
      "1723055222.555287 INFO: reactions 1000\n",
      "1723055222.5878217 INFO: metabolites 1017\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_H2O_MG_DASTool_bins_metabat.17.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055222.6483116 INFO: metabolites 1122\n",
      "1723055222.6525095 INFO: reactions 876\n",
      "1723055222.6879032 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055222.7090664 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055222.785454 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055222.8096464 INFO: reactions 997\n",
      "1723055222.8498933 INFO: metabolites 1121\n",
      "1723055222.928851 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055222.931234 INFO: metabolites 864\n",
      "1723055222.9351516 INFO: reactions 965\n",
      "1723055222.9460707 INFO: reactions 1009\n",
      "1723055222.9567006 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055223.0256374 INFO: reactions 989\n",
      "1723055223.046308 INFO: Default biomass: [bio1]\n",
      "1723055223.0490527 INFO: reactions 1114\n",
      "1723055223.058739 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055223.0624793 INFO: reactions 931\n",
      "1723055223.090903 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-19\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.25.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055223.1628096 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055223.1652133 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.9.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055223.1977103 INFO: Default biomass: [bio1]\n",
      "1723055223.2229474 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055223.237366 INFO: metabolites 914\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-20\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.20.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055223.2597992 INFO: reactions 1101\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_concoct_out.62.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.18.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055223.3314166 INFO: reactions 782\n",
      "1723055223.3613 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-21\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.43.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055223.4281073 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055223.4476163 INFO: Default biomass: [bio1]\n",
      "1723055223.4990692 INFO: metabolites 907\n",
      "1723055223.5099788 INFO: reactions 1086\n",
      "1723055223.5374746 INFO: Default biomass: [bio1]\n",
      "1723055223.5480092 INFO: metabolites 1151\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.143.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055223.595132 INFO: Default biomass: [bio1]\n",
      "1723055223.596742 INFO: Default biomass: [bio1]\n",
      "1723055223.6209114 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055223.6399286 INFO: metabolites 927\n",
      "1723055223.6434665 INFO: reactions 858\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_maxbin.009.pangenome.RAST.glm4ec.pyrSalt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.19.pangenome.RAST.glm4ec.pyr\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055223.7616198 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055223.810408 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-22\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.51.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055223.8377733 INFO: metabolites 1141\n",
      "1723055223.8422627 INFO: Default biomass: [bio1]\n",
      "1723055223.8974814 INFO: metabolites 1208\n",
      "1723055223.9130082 INFO: reactions 827\n",
      "1723055223.9281988 INFO: metabolites 1109\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.39.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055224.0092003 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055224.0516508 INFO: metabolites 1262\n",
      "1723055224.061045 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055224.0723743 INFO: Default biomass: [bio1]\n",
      "1723055224.0828629 INFO: metabolites 1262\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-23\n",
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.19.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055224.1345723 INFO: metabolites 1036\n",
      "1723055224.1730702 INFO: metabolites 1137\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055224.1940362 INFO: reactions 1057\n",
      "1723055224.2068086 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055224.2193317 INFO: metabolites 1116\n",
      "1723055224.2214437 INFO: metabolites 1228\n",
      "1723055224.2245715 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055224.3115616 INFO: Default biomass: [bio1]\n",
      "1723055224.3292613 INFO: Default biomass: [bio1]\n",
      "1723055224.335764 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055224.3465471 INFO: metabolites 1261\n",
      "1723055224.3461664 INFO: Default biomass: [bio1]\n",
      "1723055224.3706698 INFO: reactions 1064\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.51.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.78.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055224.3934183 INFO: metabolites 1045\n",
      "1723055224.4298677 INFO: metabolites 1195\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.100.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055224.480175 INFO: metabolites 806\n",
      "1723055224.5039866 INFO: reactions 1013\n",
      "1723055224.554287 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055224.5923407 INFO: metabolites 951\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055224.622472 INFO: reactions 1215\n",
      "1723055224.633532 INFO: metabolites 574\n",
      "1723055224.6451972 INFO: reactions 1026\n",
      "1723055224.6784194 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.18.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055224.7552097 INFO: metabolites 1110\n",
      "1723055224.789358 INFO: metabolites 1272\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055224.8235767 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055224.8267481 INFO: reactions 1263\n",
      "1723055224.841437 INFO: reactions 1236\n",
      "1723055224.8500657 INFO: reactions 732\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-25\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.49.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055224.9441237 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055224.9691722 INFO: reactions 1027\n",
      "1723055224.974717 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055224.9826112 INFO: Default biomass: [bio1]\n",
      "1723055224.995338 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055225.0187004 INFO: metabolites 1100\n",
      "1723055225.0246503 INFO: reactions 887\n",
      "1723055225.0410194 INFO: metabolites 1056\n",
      "1723055225.072296 INFO: reactions 883\n",
      "1723055225.0866978 INFO: reactions 1099\n",
      "1723055225.1124263 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055225.131376 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.98.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055225.1489325 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055225.191322 INFO: metabolites 1115\n",
      "1723055225.2086234 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055225.2250783 INFO: Default biomass: [bio1]\n",
      "1723055225.2656515 INFO: Default biomass: [bio1]\n",
      "1723055225.2676723 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.23.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055225.2824078 INFO: metabolites 1197\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-26\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.87.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055225.3210552 INFO: reactions 1056\n",
      "1723055225.3291025 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055225.3515706 INFO: reactions 1213\n",
      "1723055225.3690937 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_metabat.7.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055225.4503345 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055225.5338764 INFO: Default biomass: [bio1]\n",
      "1723055225.5404322 INFO: reactions 497\n",
      "1723055225.5471516 INFO: Default biomass: [bio1]\n",
      "1723055225.5630496 INFO: metabolites 1232\n",
      "1723055225.5651681 INFO: Default biomass: [bio1]\n",
      "1723055225.573263 INFO: reactions 1066\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.31.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055225.5923655 INFO: metabolites 1083\n",
      "1723055225.593691 INFO: Default biomass: [bio1]\n",
      "1723055225.6050212 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.42.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055225.641727 INFO: reactions 988\n",
      "1723055225.643734 INFO: reactions 1226\n",
      "1723055225.6827664 INFO: reactions 1204\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-27\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_maxbin.062.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055225.739169 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055225.7576556 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.50.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055225.7905881 INFO: Default biomass: [bio1]\n",
      "1723055225.805823 INFO: Default biomass: [bio1]\n",
      "1723055225.8129203 INFO: reactions 1089\n",
      "1723055225.8177989 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055225.818556 INFO: reactions 1107\n",
      "1723055225.8429983 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055225.8582792 INFO: Default biomass: [bio1]\n",
      "1723055225.8879578 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_concoct_out.67.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055225.8992789 INFO: metabolites 1025\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.58.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055225.936792 INFO: Default biomass: [bio1]\n",
      "1723055225.9430027 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_H2O_MG_DASTool_bins_metabat.29.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055226.0102413 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055226.0365832 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-28\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.55.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055226.0513031 INFO: reactions 1166\n",
      "1723055226.0578558 INFO: reactions 1244\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055226.1221576 INFO: Default biomass: [bio1]\n",
      "1723055226.1649446 INFO: metabolites 581\n",
      "1723055226.1728287 INFO: metabolites 920\n",
      "1723055226.1802773 INFO: metabolites 1133\n",
      "1723055226.1947052 INFO: Default biomass: [bio1]\n",
      "1723055226.2074077 INFO: reactions 1222\n",
      "1723055226.2201152 INFO: metabolites 1002\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.16.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055226.267162 INFO: Default biomass: [bio1]\n",
      "1723055226.2680361 INFO: reactions 1050\n",
      "1723055226.2822847 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055226.2993731 INFO: reactions 520\n",
      "1723055226.3242397 INFO: metabolites 963\n",
      "1723055226.3264103 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055226.3268905 INFO: metabolites 547\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.51.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055226.379494 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055226.386125 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055226.443722 INFO: Default biomass: [bio1]\n",
      "1723055226.4589715 INFO: metabolites 1145\n",
      "1723055226.4650204 INFO: metabolites 1140\n",
      "1723055226.5000505 INFO: Default biomass: [bio1]\n",
      "1723055226.5240462 INFO: metabolites 1200\n",
      "1723055226.5316803 INFO: reactions 474\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.37.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055226.5415328 INFO: metabolites 903\n",
      "1723055226.5448775 INFO: reactions 979\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055226.5761173 INFO: Default biomass: [bio1]\n",
      "1723055226.5906987 INFO: metabolites 1190\n",
      "1723055226.598507 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055226.608116 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055226.605827 INFO: metabolites 1122\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.118.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055226.6668127 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055226.6789947 INFO: metabolites 1133\n",
      "1723055226.7368283 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055226.7432926 INFO: metabolites 1149\n",
      "1723055226.7561157 INFO: reactions 971\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-29\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.43.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055226.7754576 INFO: metabolites 1132\n",
      "1723055226.7984085 INFO: reactions 864\n",
      "1723055226.8160262 INFO: Default biomass: [bio1]\n",
      "1723055226.8399014 INFO: metabolites 949\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.14.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055226.874248 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055226.90437 INFO: reactions 1079\n",
      "1723055226.9095957 INFO: metabolites 877\n",
      "1723055226.9198368 INFO: reactions 836\n",
      "1723055226.9218667 INFO: metabolites 972\n",
      "1723055226.9245768 INFO: Default biomass: [bio1]\n",
      "1723055226.9311945 INFO: reactions 916\n",
      "1723055226.9469733 INFO: Default biomass: [bio1]\n",
      "1723055226.9479516 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055226.9721098 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_maxbin.028.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.11.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_concoct_out.67.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.0273597 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.034926 INFO: Default biomass: [bio1]\n",
      "1723055227.0500715 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.058577 INFO: reactions 1110\n",
      "1723055227.0751712 INFO: metabolites 1063\n",
      "1723055227.093231 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.1304023 INFO: reactions 1109\n",
      "1723055227.1345747 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.26.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.212995 INFO: reactions 1202\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.230133 INFO: reactions 1130\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_metabat.14.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.2349596 INFO: reactions 871\n",
      "1723055227.2505884 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.95.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.3035517 INFO: reactions 1082\n",
      "1723055227.3067853 INFO: Default biomass: [bio1]\n",
      "1723055227.3187325 INFO: Default biomass: [bio1]\n",
      "1723055227.3330033 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.337129 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.3369794 INFO: reactions 804\n",
      "1723055227.3402143 INFO: reactions 1190\n",
      "1723055227.3586364 INFO: metabolites 1103\n",
      "1723055227.3624399 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.3634758 INFO: Default biomass: [bio1]\n",
      "1723055227.3771784 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.3831203 INFO: reactions 1092\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_concoct_out.78.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.15.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.2.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.416341 INFO: reactions 892\n",
      "1723055227.4168026 INFO: Default biomass: [bio1]\n",
      "1723055227.43007 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.4305196 INFO: metabolites 928\n",
      "1723055227.447618 INFO: metabolites 990\n",
      "1723055227.500438 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.5074718 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.5269432 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-31\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.1.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.540545 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.5579963 INFO: metabolites 959\n",
      "1723055227.5977929 INFO: reactions 1001\n",
      "1723055227.6383815 INFO: Default biomass: [bio1]\n",
      "1723055227.654003 INFO: metabolites 1243\n",
      "1723055227.6623414 INFO: Default biomass: [bio1]\n",
      "1723055227.7019324 INFO: metabolites 1110\n",
      "1723055227.7184803 INFO: metabolites 1217\n",
      "1723055227.7196617 INFO: metabolites 946\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.32.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.7559626 INFO: Default biomass: [bio1]\n",
      "1723055227.7572558 INFO: Default biomass: [bio1]\n",
      "1723055227.796077 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.29.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.8178694 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.48.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_maxbin.022.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.8799877 INFO: Default biomass: [bio1]\n",
      "1723055227.8849766 INFO: metabolites 1100\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.9148211 INFO: reactions 917\n",
      "1723055227.9172988 INFO: reactions 1107\n",
      "1723055227.9292989 INFO: reactions 869\n",
      "1723055227.9368024 INFO: metabolites 1080\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.9444067 INFO: Default biomass: [bio1]\n",
      "1723055227.946297 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055227.959231 INFO: metabolites 1251\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-32\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.4.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055227.9908748 INFO: Default biomass: [bio1]\n",
      "1723055227.9952106 INFO: metabolites 965\n",
      "1723055228.0031834 INFO: reactions 860\n",
      "1723055228.0303285 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.7.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.0463989 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.0441198 INFO: Default biomass: [bio1]\n",
      "1723055228.0884328 INFO: reactions 1093\n",
      "1723055228.1072502 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.1172416 INFO: metabolites 874\n",
      "1723055228.1402535 INFO: reactions 891\n",
      "1723055228.1495972 INFO: Default biomass: [bio1]\n",
      "1723055228.187449 INFO: metabolites 1181\n",
      "1723055228.2013175 INFO: metabolites 975\n",
      "1723055228.204671 INFO: metabolites 993\n",
      "1723055228.2343497 INFO: metabolites 932\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.59.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.287051 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.3185494 INFO: reactions 1054\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.16.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.3965006 INFO: Default biomass: [bio1]\n",
      "1723055228.4045694 INFO: Default biomass: [bio1]\n",
      "1723055228.415609 INFO: metabolites 1183\n",
      "1723055228.4270334 INFO: reactions 1241\n",
      "1723055228.440952 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.4474208 INFO: reactions 892\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-33Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.63.pangenome.RAST.glm4ec.pyr\n",
      "\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.98.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.4667637 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.4754307 INFO: Default biomass: [bio1]\n",
      "1723055228.5046818 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.6.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.5233715 INFO: Default biomass: [bio1]\n",
      "1723055228.5255919 INFO: reactions 1074\n",
      "1723055228.5261397 INFO: reactions 790\n",
      "1723055228.5410833 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_maxbin.035.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.5554342 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.5886586 INFO: metabolites 685\n",
      "1723055228.6181867 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.625458 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.6318371 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-34\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.645893 INFO: reactions 885\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_maxbin.029.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.6564648 INFO: reactions 936\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.32.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.31.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.7485595 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.750832 INFO: reactions 1078\n",
      "1723055228.761582 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.7882335 INFO: metabolites 466\n",
      "1723055228.795409 INFO: reactions 1122\n",
      "1723055228.829291 INFO: Default biomass: [bio1]\n",
      "1723055228.84651 INFO: metabolites 1276\n",
      "1723055228.8560095 INFO: reactions 621\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-35\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.8672867 INFO: metabolites 917\n",
      "1723055228.889281 INFO: Default biomass: [bio1]\n",
      "1723055228.9068382 INFO: reactions 417\n",
      "1723055228.9135349 INFO: reactions 1229\n",
      "1723055228.9181523 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.9274256 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.9374323 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_metabat.23.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055228.944981 INFO: Default biomass: [bio1]\n",
      "1723055228.9477599 INFO: metabolites 955\n",
      "1723055228.9596732 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055228.9798253 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055229.0431411 INFO: metabolites 1112\n",
      "1723055229.0870576 INFO: metabolites 1151\n",
      "1723055229.121335 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055229.1234336 INFO: Default biomass: [bio1]\n",
      "1723055229.1416378 INFO: reactions 873\n",
      "1723055229.1429317 INFO: reactions 1156\n",
      "1723055229.1424654 INFO: metabolites 909\n",
      "1723055229.1470976 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.1670346 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.59.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.2146695 INFO: metabolites 1217\n",
      "1723055229.2349362 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-36\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.119.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.253829 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.3001556 INFO: Default biomass: [bio1]\n",
      "1723055229.3320029 INFO: metabolites 1295\n",
      "1723055229.3532155 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055229.353163 INFO: metabolites 1191\n",
      "1723055229.3579795 INFO: reactions 858\n",
      "1723055229.3608904 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.3767128 INFO: Default biomass: [bio1]\n",
      "1723055229.4051163 INFO: Default biomass: [bio1]\n",
      "1723055229.4180233 INFO: metabolites 1100\n",
      "1723055229.4218776 INFO: reactions 918\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.67.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.4304774 INFO: reactions 1264\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.47.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.464956 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055229.5301206 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055229.5314608 INFO: metabolites 1063\n",
      "1723055229.5347807 INFO: metabolites 1027\n",
      "1723055229.5955694 INFO: reactions 1240\n",
      "1723055229.5958903 INFO: Default biomass: [bio1]\n",
      "1723055229.5967236 INFO: Default biomass: [bio1]\n",
      "1723055229.5972722 INFO: metabolites 1140\n",
      "1723055229.6068327 INFO: metabolites 972\n",
      "1723055229.6200168 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055229.6242187 INFO: Default biomass: [bio1]\n",
      "1723055229.6384068 INFO: reactions 1049\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.15.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.7050369 INFO: metabolites 772\n",
      "1723055229.7054412 INFO: Default biomass: [bio1]\n",
      "1723055229.7080755 INFO: reactions 858\n",
      "1723055229.7358088 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.765179 INFO: metabolites 1167\n",
      "1723055229.7762268 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.15.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.7951782 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.8025439 INFO: Default biomass: [bio1]\n",
      "1723055229.81572 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_concoct_out.80.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055229.8350766 INFO: metabolites 1264\n",
      "1723055229.9113011 INFO: reactions 1203\n",
      "1723055229.917017 INFO: reactions 1017\n",
      "1723055229.9851618 INFO: Default biomass: [bio1]\n",
      "1723055230.0478516 INFO: reactions 989\n",
      "1723055230.066299 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055230.0774958 INFO: reactions 935\n",
      "1723055230.0781443 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.89.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055230.0852628 INFO: reactions 749\n",
      "1723055230.086231 INFO: metabolites 1046\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055230.0933511 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055230.1282501 INFO: Default biomass: [bio1]\n",
      "1723055230.1668158 INFO: metabolites 1120\n",
      "1723055230.1697593 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055230.1701376 INFO: metabolites 955\n",
      "1723055230.184375 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055230.198064 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055230.2171776 INFO: Default biomass: [bio1]\n",
      "1723055230.229084 INFO: reactions 1071\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055230.2668877 INFO: reactions 1092\n",
      "1723055230.3161795 INFO: Default biomass: [bio1]\n",
      "1723055230.3369803 INFO: metabolites 1180\n",
      "1723055230.368296 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055230.3920631 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-37\n",
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.40.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055230.4521155 INFO: metabolites 1033\n",
      "1723055230.5568922 INFO: Default biomass: [bio1]\n",
      "1723055230.5580437 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-38\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055230.567478 INFO: metabolites 1069\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.concoct_out.54.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055230.582185 INFO: Default biomass: [bio1]\n",
      "1723055230.587143 INFO: reactions 1271\n",
      "1723055230.5929515 INFO: Default biomass: [bio1]\n",
      "1723055230.6026082 INFO: Default biomass: [bio1]\n",
      "1723055230.6270244 INFO: reactions 937\n",
      "1723055230.6418688 INFO: reactions 997\n",
      "1723055230.6729436 INFO: reactions 895\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.45.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055230.7481017 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055230.7479165 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055230.770575 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055230.7828732 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055230.8074458 INFO: reactions 1142\n",
      "1723055230.80021 INFO: Default biomass: [bio1]\n",
      "1723055230.833822 INFO: Default biomass: [bio1]\n",
      "1723055230.8410203 INFO: metabolites 1144\n",
      "1723055230.8452072 INFO: reactions 1126\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-39\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055230.8933866 INFO: Default biomass: [bio1]\n",
      "1723055230.8995342 INFO: Default biomass: [bio1]\n",
      "1723055230.9473374 INFO: reactions 1147\n",
      "1723055230.980183 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.0076425 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055231.036174 INFO: reactions 1019\n",
      "1723055231.0492163 INFO: metabolites 1042\n",
      "1723055231.0746806 INFO: reactions 1274\n",
      "1723055231.0903728 INFO: metabolites 1031\n",
      "1723055231.106968 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.10.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.1411967 INFO: Default biomass: [bio1]\n",
      "1723055231.147673 INFO: Default biomass: [bio1]\n",
      "1723055231.1513355 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.19.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.40.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.2025058 INFO: reactions 1117\n",
      "1723055231.2050097 INFO: Default biomass: [bio1]\n",
      "1723055231.2107496 INFO: metabolites 1251\n",
      "1723055231.2381637 INFO: metabolites 1192\n",
      "1723055231.2617636 INFO: metabolites 1107\n",
      "1723055231.2675464 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-40\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.2.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.2780747 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.27.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.3394332 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055231.3664622 INFO: Default biomass: [bio1]\n",
      "1723055231.40103 INFO: metabolites 1115\n",
      "1723055231.4383266 INFO: reactions 1117\n",
      "1723055231.4460235 INFO: Default biomass: [bio1]\n",
      "1723055231.4601846 INFO: metabolites 1163\n",
      "1723055231.477807 INFO: metabolites 916\n",
      "1723055231.4874344 INFO: Default biomass: [bio1]\n",
      "1723055231.5299463 INFO: metabolites 944\n",
      "1723055231.5339274 INFO: reactions 973\n",
      "1723055231.5468547 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.52.pangenome.RAST.glm4ec.pyr\n",
      "ForkPoolWorker-41\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.5814016 INFO: Default biomass: [bio1]\n",
      "1723055231.5977733 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.36.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.648187 INFO: reactions 969\n",
      "1723055231.6519485 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_metabat.42.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.713508 INFO: metabolites 1021\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.9.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.7328007 INFO: metabolites 1244\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-42\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.40.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.763996 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.39.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.793952 INFO: Default biomass: [bio1]\n",
      "1723055231.7999616 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_concoct_out.16.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055231.877083 INFO: reactions 1107\n",
      "1723055231.9243512 INFO: reactions 1186\n",
      "1723055231.9327364 INFO: reactions 835\n",
      "1723055231.982167 INFO: reactions 1043\n",
      "1723055232.0054457 INFO: metabolites 984\n",
      "1723055232.011227 INFO: reactions 1032\n",
      "1723055232.0228055 INFO: reactions 1223\n",
      "1723055232.0260103 INFO: metabolites 1052\n",
      "1723055232.0259087 INFO: reactions 834\n",
      "1723055232.0285594 INFO: metabolites 1205\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-43\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055232.0562832 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.109.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055232.0977523 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055232.1132102 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055232.129857 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055232.1359904 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055232.1490004 INFO: metabolites 902\n",
      "1723055232.1525648 INFO: reactions 1114\n",
      "1723055232.1748605 INFO: Default biomass: [bio1]\n",
      "1723055232.2060838 INFO: metabolites 1128\n",
      "1723055232.2060616 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055232.213651 INFO: reactions 936\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.22.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055232.2820618 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055232.285084 INFO: metabolites 1206\n",
      "1723055232.3293345 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055232.3349903 INFO: metabolites 1163\n",
      "1723055232.3708687 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055232.3986287 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.9.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055232.4976432 INFO: reactions 799\n",
      "1723055232.502139 INFO: reactions 957\n",
      "1723055232.5241113 INFO: metabolites 1100\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-44\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.30.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055232.5597205 INFO: Default biomass: [bio1]\n",
      "1723055232.5661275 INFO: Default biomass: [bio1]\n",
      "1723055232.5794446 INFO: Default biomass: [bio1]\n",
      "1723055232.592842 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055232.6104937 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055232.685654 INFO: metabolites 1009\n",
      "1723055232.6940885 INFO: reactions 1189\n",
      "1723055232.7137597 INFO: metabolites 1025\n",
      "1723055232.7189944 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055232.7480562 INFO: Default biomass: [bio1]\n",
      "1723055232.7492914 INFO: Default biomass: [bio1]\n",
      "1723055232.7953181 INFO: reactions 1120\n",
      "1723055232.8175743 INFO: Default biomass: [bio1]\n",
      "1723055232.83353 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.71.pangenome.RAST.glm4ec.pyr\n",
      "ForkPoolWorker-45\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.10.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055232.8872607 INFO: metabolites 1268\n",
      "1723055232.9073424 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.50.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055232.9464374 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055232.9460256 INFO: metabolites 1101\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.13.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055232.985607 INFO: reactions 1187\n",
      "1723055233.0050302 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.84.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055233.0823696 INFO: metabolites 1084\n",
      "1723055233.0939264 INFO: metabolites 1138\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-46\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055233.1414924 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055233.1930213 INFO: reactions 975\n",
      "1723055233.205413 INFO: reactions 1078\n",
      "1723055233.2094898 INFO: Default biomass: [bio1]\n",
      "1723055233.2555447 INFO: reactions 894\n",
      "1723055233.2958558 INFO: metabolites 948\n",
      "1723055233.3077204 INFO: reactions 1265\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.21.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055233.3463345 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055233.3627443 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055233.3968825 INFO: metabolites 1208\n",
      "1723055233.425249 INFO: Default biomass: [bio1]\n",
      "1723055233.437217 INFO: Default biomass: [bio1]\n",
      "1723055233.4601135 INFO: metabolites 1217\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.37.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055233.5078423 INFO: reactions 1047\n",
      "1723055233.517962 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.36.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055233.5224743 INFO: metabolites 1156\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-47\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.68.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055233.6616635 INFO: Default biomass: [bio1]\n",
      "1723055233.6714795 INFO: reactions 1054\n",
      "1723055233.7111454 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_metabat.17.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055233.7138214 INFO: Default biomass: [bio1]\n",
      "1723055233.7384317 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055233.7405546 INFO: metabolites 971\n",
      "1723055233.7481656 INFO: reactions 918\n",
      "1723055233.7522852 INFO: metabolites 1237\n",
      "1723055233.7526 INFO: Default biomass: [bio1]\n",
      "1723055233.792799 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055233.7995546 INFO: reactions 1158\n",
      "1723055233.80348 INFO: metabolites 886\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.49.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055233.8520658 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055233.8537755 INFO: metabolites 1040\n",
      "1723055233.8746824 INFO: metabolites 525\n",
      "1723055233.9811275 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055233.9869022 INFO: reactions 996\n",
      "1723055234.0279553 INFO: reactions 460\n",
      "1723055234.1095462 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055234.116185 INFO: reactions 1146\n",
      "1723055234.134712 INFO: metabolites 1109\n",
      "1723055234.1378121 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-48\n",
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_metabat.10.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.1474187 INFO: Default biomass: [bio1]\n",
      "1723055234.1718137 INFO: reactions 943\n",
      "1723055234.2114272 INFO: Default biomass: [bio1]\n",
      "1723055234.2223716 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.37.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.257162 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.2578576 INFO: metabolites 909\n",
      "1723055234.2656577 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.53.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.2713048 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055234.283693 INFO: metabolites 992\n",
      "1723055234.2971487 INFO: reactions 962\n",
      "1723055234.3135448 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055234.3593657 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_maxbin.002.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.4047549 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055234.423127 INFO: reactions 1226\n",
      "1723055234.423152 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.69.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.4819677 INFO: Default biomass: [bio1]\n",
      "1723055234.5254006 INFO: Default biomass: [bio1]\n",
      "1723055234.5551364 INFO: reactions 1104\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39.pangenome.RAST.glm4ec.pyr\n",
      "ForkPoolWorker-49\n",
      "Salt_Pond_MetaGSF2_C_H2O_MG_DASTool_bins_metabat.29.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.6139972 INFO: reactions 825\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_concoct_out.19.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.6225061 INFO: reactions 1201\n",
      "1723055234.6270661 INFO: metabolites 1125\n",
      "1723055234.6282194 INFO: metabolites 1089\n",
      "1723055234.658797 INFO: metabolites 878\n",
      "1723055234.664257 INFO: metabolites 1072\n",
      "1723055234.6934865 INFO: reactions 1046\n",
      "1723055234.6963158 INFO: Default biomass: [bio1]\n",
      "1723055234.7012753 INFO: Default biomass: [bio1]\n",
      "1723055234.7249057 INFO: reactions 890\n",
      "1723055234.725964 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055234.7273414 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.38.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.7745945 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055234.7840903 INFO: Default biomass: [bio1]\n",
      "1723055234.834146 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_H2O_MG_DASTool_bins_metabat.16.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.8460143 INFO: reactions 1193\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.8596005 INFO: metabolites 1107\n",
      "1723055234.8795254 INFO: Default biomass: [bio1]\n",
      "1723055234.8900967 INFO: metabolites 1031\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_maxbin.005.pangenome.RAST.glm4ec.pyr\n",
      "ForkPoolWorker-50\n",
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.45.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055234.9329858 INFO: metabolites 924\n",
      "1723055234.9457881 INFO: metabolites 979\n",
      "1723055234.980097 INFO: metabolites 956\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.0064635 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.012654 INFO: reactions 746\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-51\n",
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.concoct_out.5.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.103417 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.146958 INFO: metabolites 766\n",
      "1723055235.163578 INFO: reactions 1204\n",
      "1723055235.1798062 INFO: Default biomass: [bio1]\n",
      "1723055235.191454 INFO: Default biomass: [bio1]\n",
      "1723055235.2253056 INFO: reactions 1110\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.2462337 INFO: reactions 780\n",
      "1723055235.2845519 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_D1_MG_DASTool_bins_concoct_out.4.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.3102114 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055235.3426852 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055235.3676386 INFO: metabolites 1094\n",
      "1723055235.3907368 INFO: reactions 955\n",
      "1723055235.3973873 INFO: reactions 890\n",
      "1723055235.4143162 INFO: reactions 1059\n",
      "1723055235.4214308 INFO: reactions 895\n",
      "1723055235.4219701 INFO: reactions 874\n",
      "1723055235.424703 INFO: Default biomass: [bio1]\n",
      "1723055235.4275224 INFO: metabolites 1183\n",
      "1723055235.471822 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.489366 INFO: Default biomass: [bio1]\n",
      "1723055235.4995477 INFO: reactions 709\n",
      "1723055235.5078104 INFO: metabolites 1151\n",
      "1723055235.5093734 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055235.5242243 INFO: metabolites 907\n",
      "1723055235.5294666 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-52\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.55859 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055235.5625763 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055235.5655453 INFO: Default biomass: [bio1]\n",
      "1723055235.5796416 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055235.595285 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055235.622608 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_metabat.20.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.6656635 INFO: Default biomass: [bio1]\n",
      "1723055235.696553 INFO: metabolites 1117\n",
      "1723055235.7503483 INFO: metabolites 1112\n",
      "1723055235.8240883 INFO: Default biomass: [bio1]\n",
      "1723055235.833413 INFO: metabolites 1203\n",
      "1723055235.8587842 INFO: Default biomass: [bio1]\n",
      "1723055235.8754587 INFO: Default biomass: [bio1]\n",
      "1723055235.900238 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-53\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.19.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.27.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.9321632 INFO: metabolites 1133\n",
      "1723055235.940988 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.9757159 INFO: reactions 1043\n",
      "1723055235.9783373 INFO: reactions 841\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.32.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055235.9930751 INFO: metabolites 735\n",
      "1723055235.9943614 INFO: Default biomass: [bio1]\n",
      "1723055236.0003548 INFO: Default biomass: [bio1]\n",
      "1723055236.0039814 INFO: Default biomass: [bio1]\n",
      "1723055236.0179312 INFO: reactions 990\n",
      "1723055236.029454 INFO: reactions 1166\n",
      "1723055236.0627282 INFO: reactions 1117\n",
      "1723055236.0680552 INFO: Default biomass: [bio1]\n",
      "1723055236.0826368 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055236.0837543 INFO: Default biomass: [bio1]\n",
      "1723055236.1004221 INFO: metabolites 1181\n",
      "1723055236.1611195 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055236.1644406 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_metabat.8.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055236.1919851 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055236.2322161 INFO: reactions 1054\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-54\n",
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055236.2441075 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055236.3385534 INFO: reactions 658\n",
      "1723055236.352947 INFO: metabolites 1088\n",
      "1723055236.3593013 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055236.373836 INFO: metabolites 1034\n",
      "1723055236.4227192 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055236.4384727 INFO: metabolites 1109\n",
      "1723055236.4462435 INFO: Default biomass: [bio1]\n",
      "1723055236.4704826 INFO: reactions 1093\n",
      "1723055236.5496202 INFO: metabolites 892\n",
      "1723055236.5554485 INFO: metabolites 1023\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-55\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_maxbin.037.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055236.5999043 INFO: Default biomass: [bio1]\n",
      "1723055236.606117 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_metabat.14.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055236.6290183 INFO: Default biomass: [bio1]\n",
      "1723055236.6366723 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_concoct_out.22.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055236.6594431 INFO: metabolites 1058\n",
      "1723055236.7174451 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.42.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055236.759311 INFO: Default biomass: [bio1]\n",
      "1723055236.7797115 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_concoct_out.59.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.27.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055236.8912559 INFO: reactions 992\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055236.904886 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.25.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055236.9868171 INFO: metabolites 934\n",
      "1723055236.9905283 INFO: reactions 810\n",
      "1723055236.993336 INFO: reactions 1076\n",
      "1723055237.0275598 INFO: reactions 999\n",
      "1723055237.0290146 INFO: metabolites 1089\n",
      "1723055237.0349472 INFO: metabolites 984\n",
      "1723055237.051222 INFO: reactions 1065\n",
      "1723055237.0603952 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.072485 INFO: metabolites 1173\n",
      "1723055237.086757 INFO: reactions 1039\n",
      "1723055237.0882506 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-56\n",
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_metabat.11.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055237.1468923 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.1935506 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.199976 INFO: reactions 1014\n",
      "1723055237.2116363 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.24404 INFO: metabolites 948\n",
      "1723055237.2545195 INFO: metabolites 931\n",
      "1723055237.2649906 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.269476 INFO: metabolites 1097\n",
      "1723055237.2761993 INFO: metabolites 933\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-57\n",
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_concoct_out.79.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055237.3190744 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.3399723 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055237.4273248 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_maxbin.001.pangenome.RAST.glm4ec.pyrSalt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.49.pangenome.RAST.glm4ec.pyr\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055237.5345147 INFO: metabolites 1006\n",
      "1723055237.5356987 INFO: Default biomass: [bio1]\n",
      "1723055237.6029923 INFO: reactions 1058\n",
      "1723055237.6350863 INFO: reactions 870\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.44.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055237.6555917 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055237.6620996 INFO: Default biomass: [bio1]\n",
      "1723055237.6629593 INFO: reactions 906\n",
      "1723055237.6992865 INFO: reactions 1140\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.45.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055237.7129278 INFO: Default biomass: [bio1]\n",
      "1723055237.7234433 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-58\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055237.7509704 INFO: reactions 1177\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.16.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055237.771595 INFO: reactions 996\n",
      "1723055237.7807853 INFO: reactions 842\n",
      "1723055237.7811637 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.7827864 INFO: metabolites 1060\n",
      "1723055237.7941332 INFO: metabolites 969\n",
      "1723055237.8085968 INFO: reactions 1048\n",
      "1723055237.822565 INFO: reactions 814\n",
      "1723055237.8276737 INFO: reactions 908\n",
      "1723055237.83301 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.8346758 INFO: reactions 1188\n",
      "1723055237.8578136 INFO: metabolites 1093\n",
      "1723055237.8775206 INFO: metabolites 1053\n",
      "1723055237.881368 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.890409 INFO: Default biomass: [bio1]\n",
      "1723055237.9319732 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.9325962 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.9532452 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.concoct_out.44.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055237.968303 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055237.970585 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055238.0140996 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055238.1128912 INFO: reactions 976\n",
      "1723055238.1416059 INFO: metabolites 1015\n",
      "1723055238.1585824 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-59\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055238.1923444 INFO: metabolites 1045\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.51.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055238.233705 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055238.2358415 INFO: Default biomass: [bio1]\n",
      "1723055238.2428873 INFO: reactions 884\n",
      "1723055238.2558014 INFO: Default biomass: [bio1]\n",
      "1723055238.2590587 INFO: reactions 959\n",
      "1723055238.298225 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.35.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.86.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055238.3450544 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055238.3722932 INFO: Default biomass: [bio1]\n",
      "1723055238.3823793 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055238.3951397 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055238.4364746 INFO: Default biomass: [bio1]\n",
      "1723055238.4856389 INFO: Default biomass: [bio1]\n",
      "1723055238.5005171 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.26.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055238.559579 INFO: Default biomass: [bio1]\n",
      "1723055238.5776215 INFO: metabolites 966\n",
      "1723055238.5996664 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055238.609441 INFO: metabolites 1090\n",
      "1723055238.6140544 INFO: Default biomass: [bio1]\n",
      "1723055238.671758 INFO: reactions 984\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.18.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055238.693963 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_concoct_out.44.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055238.7094588 INFO: reactions 953\n",
      "1723055238.7094166 INFO: reactions 1046\n",
      "1723055238.7184439 INFO: metabolites 1003\n",
      "1723055238.7187726 INFO: reactions 1070\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-60\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.29.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055238.7523863 INFO: metabolites 983\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.38.pangenome.RAST.glm4ec.pyr\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055238.7740886 INFO: Default biomass: [bio1]\n",
      "1723055238.7955663 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.37.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055238.803798 INFO: metabolites 1228\n",
      "1723055238.8054707 INFO: metabolites 1202\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055238.8256216 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055238.8880854 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055238.953109 INFO: metabolites 1189\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-61\n",
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.104.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055239.070041 INFO: Default biomass: [bio1]\n",
      "1723055239.0791585 INFO: Default biomass: [bio1]\n",
      "1723055239.1048238 INFO: metabolites 1096\n",
      "1723055239.151443 INFO: metabolites 995\n",
      "1723055239.1633267 INFO: metabolites 1062\n",
      "1723055239.1710374 INFO: metabolites 1049\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055239.2010772 INFO: metabolites 1158\n",
      "1723055239.2023132 INFO: reactions 958\n",
      "1723055239.2217011 INFO: metabolites 1038\n",
      "1723055239.2400253 INFO: reactions 900\n",
      "1723055239.2439103 INFO: metabolites 1009\n",
      "1723055239.251721 INFO: Default biomass: [bio1]\n",
      "1723055239.2836516 INFO: reactions 1007\n",
      "1723055239.3181694 INFO: metabolites 1139\n",
      "1723055239.3261116 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055239.389584 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055239.4071834 INFO: Default biomass: [bio1]\n",
      "1723055239.411742 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055239.4322548 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9.pangenome.RAST.glm4ec.pyr\n",
      "ForkPoolWorker-62\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.17.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055239.5082862 INFO: metabolites 1093\n",
      "1723055239.5587277 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055239.582784 INFO: reactions 1176\n",
      "1723055239.675155 INFO: metabolites 1067\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.11.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055239.7212646 INFO: Default biomass: [bio1]\n",
      "1723055239.7228425 INFO: reactions 962\n",
      "1723055239.7244334 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055239.7341816 INFO: reactions 970\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.26.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055239.7668653 INFO: reactions 922\n",
      "1723055239.7732887 INFO: reactions 960\n",
      "1723055239.8338768 INFO: Default biomass: [bio1]\n",
      "1723055239.8338208 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055239.8461332 INFO: Default biomass: [bio1]\n",
      "1723055239.8592057 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055239.8677769 INFO: metabolites 1126\n",
      "1723055239.8738744 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055239.8887095 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.18.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055239.911163 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-63\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055239.949304 INFO: metabolites 1123\n",
      "1723055240.015681 INFO: Default biomass: [bio1]\n",
      "1723055240.0248227 INFO: metabolites 940\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_metabat.4.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055240.033285 INFO: reactions 936\n",
      "1723055240.0438943 INFO: reactions 1205\n",
      "1723055240.0638227 INFO: reactions 1028\n",
      "1723055240.095643 INFO: metabolites 959\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-64\n",
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.15.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.54.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055240.1495905 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055240.1629136 INFO: reactions 1081\n",
      "1723055240.1901836 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055240.194571 INFO: Default biomass: [bio1]\n",
      "1723055240.1976721 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055240.2354257 INFO: metabolites 1123\n",
      "1723055240.2375276 INFO: reactions 1040\n",
      "1723055240.2400625 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.45.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055240.2442386 INFO: reactions 1246\n",
      "1723055240.2643368 INFO: Default biomass: [bio1]\n",
      "1723055240.2827723 INFO: Default biomass: [bio1]\n",
      "1723055240.2945273 INFO: reactions 1042\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.84.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055240.3298085 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_metabat.5.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055240.3585377 INFO: reactions 981\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055240.3702679 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055240.402364 INFO: metabolites 1174\n",
      "1723055240.4162781 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055240.4346883 INFO: metabolites 1107\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-65\n",
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.11.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055240.4545388 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055240.4694076 INFO: reactions 838\n",
      "1723055240.4775066 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055240.4787421 INFO: reactions 1009\n",
      "1723055240.518051 INFO: metabolites 1097\n",
      "1723055240.5406997 INFO: reactions 1098\n",
      "1723055240.5497084 INFO: Default biomass: [bio1]\n",
      "1723055240.5591757 INFO: reactions 907\n",
      "1723055240.5682218 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055240.6079967 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.89.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055240.671331 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055240.6961498 INFO: metabolites 958\n",
      "1723055240.7109094 INFO: Default biomass: [bio1]\n",
      "1723055240.7289617 INFO: metabolites 1107\n",
      "1723055240.7616975 INFO: reactions 1046\n",
      "1723055240.7950656 INFO: Default biomass: [bio1]\n",
      "1723055240.8413813 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055240.8513045 INFO: metabolites 1140\n",
      "1723055240.8704824 INFO: metabolites 1133\n",
      "1723055240.8838696 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.7.pangenome.RAST.glm4ec.pyrForkPoolWorker-66\n",
      "\n",
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055240.916458 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.42.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055240.9586077 INFO: metabolites 1212\n",
      "1723055240.9876504 INFO: Default biomass: [bio1]\n",
      "1723055240.992637 INFO: reactions 1060\n",
      "1723055241.0434368 INFO: reactions 1160\n",
      "1723055241.05377 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.8.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055241.081684 INFO: reactions 1034\n",
      "1723055241.1063175 INFO: reactions 878\n",
      "1723055241.1157343 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055241.1631486 INFO: metabolites 1210\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.4.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055241.2048721 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055241.2153206 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055241.2611902 INFO: Default biomass: [bio1]\n",
      "1723055241.2991583 INFO: metabolites 815\n",
      "1723055241.3321447 INFO: metabolites 1186\n",
      "1723055241.361192 INFO: metabolites 1020\n",
      "1723055241.42785 INFO: metabolites 1012\n",
      "1723055241.442444 INFO: Default biomass: [bio1]\n",
      "1723055241.4547966 INFO: reactions 1121\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-67\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055241.4638758 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055241.5030944 INFO: metabolites 1150\n",
      "1723055241.503198 INFO: reactions 1106\n",
      "1723055241.518966 INFO: Default biomass: [bio1]\n",
      "1723055241.5547342 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.12.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055241.571934 INFO: metabolites 1186\n",
      "1723055241.627309 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.38.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.21.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055241.6816595 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055241.703978 INFO: reactions 1068\n",
      "1723055241.7084615 INFO: reactions 768\n",
      "1723055241.8037894 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-68\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.41.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055241.837481 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055241.8422427 INFO: reactions 1180\n",
      "1723055241.866136 INFO: metabolites 939\n",
      "1723055241.8722498 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055241.8875518 INFO: metabolites 1004\n",
      "1723055241.8911974 INFO: reactions 996\n",
      "1723055241.8966436 INFO: metabolites 1240\n",
      "1723055241.9100134 INFO: reactions 985\n",
      "1723055241.9195344 INFO: Default biomass: [bio1]\n",
      "1723055241.9472814 INFO: reactions 1159\n",
      "1723055241.9860272 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055242.0007164 INFO: metabolites 901\n",
      "1723055242.0159264 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055242.0308192 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055242.0902529 INFO: reactions 1090\n",
      "1723055242.1021297 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.concoct_out.9.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055242.115518 INFO: Default biomass: [bio1]\n",
      "1723055242.1308515 INFO: Default biomass: [bio1]\n",
      "1723055242.1337006 INFO: Default biomass: [bio1]\n",
      "1723055242.137761 INFO: metabolites 1155\n",
      "1723055242.200844 INFO: reactions 1126\n",
      "1723055242.2000384 INFO: metabolites 1274\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.7.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055242.229776 INFO: reactions 1049\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-69\n",
      "Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055242.3174036 INFO: reactions 874\n",
      "1723055242.3259604 INFO: metabolites 1170\n",
      "1723055242.331933 INFO: reactions 888\n",
      "1723055242.337143 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055242.3496408 INFO: Default biomass: [bio1]\n",
      "1723055242.3657641 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055242.366875 INFO: Default biomass: [bio1]\n",
      "1723055242.371442 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055242.376383 INFO: Default biomass: [bio1]\n",
      "1723055242.4239278 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055242.4268367 INFO: Default biomass: [bio1]\n",
      "1723055242.4302337 INFO: Default biomass: [bio1]\n",
      "1723055242.4408147 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055242.4621687 INFO: metabolites 1060\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.120.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.37.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055242.7462301 INFO: metabolites 1131\n",
      "1723055242.7567422 INFO: metabolites 1116\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-70\n",
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055242.7652557 INFO: reactions 1171\n",
      "1723055242.773462 INFO: Default biomass: [bio1]\n",
      "1723055242.7840059 INFO: reactions 1138\n",
      "1723055242.799197 INFO: Default biomass: [bio1]\n",
      "1723055242.8136675 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_maxbin.017.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055242.843625 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055242.8732007 INFO: metabolites 918\n",
      "1723055242.9136372 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055242.9347992 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055242.9396167 INFO: metabolites 1103\n",
      "1723055242.9499123 INFO: reactions 1000\n",
      "1723055242.9674602 INFO: reactions 1252\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.10.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055243.0647776 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055243.0904398 INFO: reactions 839\n",
      "1723055243.1442714 INFO: metabolites 1000\n",
      "1723055243.1466744 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055243.1714823 INFO: metabolites 1012\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-71\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.25.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055243.1929567 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055243.2081604 INFO: metabolites 1225\n",
      "1723055243.2161136 INFO: metabolites 1033\n",
      "1723055243.2786121 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055243.27999 INFO: Default biomass: [bio1]\n",
      "1723055243.2837553 INFO: reactions 857\n",
      "1723055243.2959056 INFO: Default biomass: [bio1]\n",
      "1723055243.3194485 INFO: Default biomass: [bio1]\n",
      "1723055243.3261251 INFO: reactions 1104\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.46.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055243.3847911 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055243.3877988 INFO: reactions 1103\n",
      "1723055243.4407365 INFO: metabolites 1142\n",
      "1723055243.4601233 INFO: Default biomass: [bio1]\n",
      "1723055243.4746287 INFO: Default biomass: [bio1]\n",
      "1723055243.480389 INFO: reactions 1034\n",
      "1723055243.506116 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055243.5087667 INFO: Default biomass: [bio1]\n",
      "1723055243.5121326 INFO: reactions 1215\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055243.5380352 INFO: Default biomass: [bio1]\n",
      "1723055243.5419407 INFO: metabolites 939\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8.pangenome.RAST.glm4ec.pyr\n",
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.51.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055243.60159 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055243.6290884 INFO: reactions 938\n",
      "1723055243.6618037 INFO: Default biomass: [bio1]\n",
      "1723055243.6750677 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.37.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055243.7299488 INFO: reactions 1171\n",
      "1723055243.7312992 INFO: reactions 987\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-72\n",
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_maxbin.047.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055243.7410314 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.58.pangenome.RAST.glm4ec.pyr\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055243.779287 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055243.7805364 INFO: Default biomass: [bio1]\n",
      "1723055243.8310955 INFO: metabolites 1132\n",
      "1723055243.8470523 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055243.8570578 INFO: Default biomass: [bio1]\n",
      "1723055243.8803136 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.110.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055243.9380796 INFO: reactions 1235\n",
      "1723055243.9886758 INFO: reactions 924\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.80.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055244.0377712 INFO: metabolites 968\n",
      "1723055244.0422323 INFO: metabolites 1183\n",
      "1723055244.0423832 INFO: Default biomass: [bio1]\n",
      "1723055244.0438435 INFO: reactions 1087\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-73\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.16.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055244.094485 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055244.101352 INFO: metabolites 887\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055244.1668367 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055244.1786742 INFO: metabolites 945\n",
      "1723055244.1822438 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055244.238581 INFO: reactions 912\n",
      "1723055244.2874823 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055244.291394 INFO: metabolites 1193\n",
      "1723055244.3281772 INFO: metabolites 1020\n",
      "1723055244.3353784 INFO: Default biomass: [bio1]\n",
      "1723055244.3531396 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055244.3586042 INFO: Default biomass: [bio1]\n",
      "1723055244.411139 INFO: metabolites 1183\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.8.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055244.4319644 INFO: Default biomass: [bio1]\n",
      "1723055244.4735997 INFO: reactions 912\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_metabat.16.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055244.5078833 INFO: reactions 1135\n",
      "1723055244.5438244 INFO: reactions 824\n",
      "1723055244.5612135 INFO: metabolites 1233\n",
      "1723055244.5817761 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055244.632886 INFO: metabolites 1150\n",
      "1723055244.6443484 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055244.652593 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055244.6944726 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-74\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.61.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055244.735383 INFO: reactions 1169\n",
      "1723055244.7356565 INFO: Default biomass: [bio1]\n",
      "1723055244.7483644 INFO: metabolites 862\n",
      "1723055244.7509608 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.8.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055244.8182402 INFO: Default biomass: [bio1]\n",
      "1723055244.8584187 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.93.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055244.8831074 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055244.8851798 INFO: reactions 985\n",
      "1723055244.9191613 INFO: metabolites 1215\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.11.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055244.9364617 INFO: reactions 1161\n",
      "1723055244.9506907 INFO: Default biomass: [bio1]\n",
      "1723055245.0033257 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055245.0223866 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32.pangenome.RAST.glm4ec.pyrSalt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.51.pangenome.RAST.glm4ec.pyr\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055245.089055 INFO: reactions 1188\n",
      "1723055245.121502 INFO: Default biomass: [bio1]\n",
      "1723055245.1275523 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-75\n",
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.10.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055245.1413217 INFO: Default biomass: [bio1]\n",
      "1723055245.1408012 INFO: metabolites 1043\n",
      "1723055245.146223 INFO: reactions 772\n",
      "1723055245.1610024 INFO: metabolites 1058\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.21.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055245.2210467 INFO: metabolites 622\n",
      "1723055245.2287805 INFO: reactions 1064\n",
      "1723055245.2269127 INFO: Default biomass: [bio1]\n",
      "1723055245.241099 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055245.2408752 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.62.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055245.3185315 INFO: metabolites 1083\n",
      "1723055245.3780887 INFO: Default biomass: [bio1]\n",
      "1723055245.3839686 INFO: reactions 529\n",
      "1723055245.402195 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055245.4044113 INFO: Default biomass: [bio1]\n",
      "1723055245.416746 INFO: metabolites 755\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.51.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055245.4519281 INFO: metabolites 1155\n",
      "1723055245.453443 INFO: metabolites 1095\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_concoct_out.13.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055245.5201147 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-76\n",
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.27.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055245.5680435 INFO: metabolites 1145\n",
      "1723055245.570182 INFO: Default biomass: [bio1]\n",
      "1723055245.6021245 INFO: reactions 945\n",
      "1723055245.622977 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_concoct_out.2.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055245.666922 INFO: metabolites 1026\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_concoct_out.35.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055245.7179081 INFO: reactions 873\n",
      "1723055245.7202477 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055245.7533832 INFO: Default biomass: [bio1]\n",
      "1723055245.8271425 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_concoct_out.60.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055245.8588538 INFO: Default biomass: [bio1]\n",
      "1723055245.8678474 INFO: reactions 1059\n",
      "1723055245.9047532 INFO: metabolites 1004\n",
      "1723055245.9206233 INFO: reactions 1235\n",
      "1723055245.9453344 INFO: metabolites 1039\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.34.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055245.9916482 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055245.9934793 INFO: reactions 1048\n",
      "1723055245.999324 INFO: metabolites 1252\n",
      "1723055246.0544584 INFO: metabolites 993\n",
      "1723055246.0575023 INFO: reactions 1108\n",
      "1723055246.0613735 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-77\n",
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.6.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055246.114084 INFO: reactions 930\n",
      "1723055246.1175125 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055246.1248071 INFO: Default biomass: [bio1]\n",
      "1723055246.1794817 INFO: Default biomass: [bio1]\n",
      "1723055246.1847007 INFO: reactions 1109\n",
      "1723055246.197998 INFO: metabolites 938\n",
      "1723055246.2193646 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055246.2197022 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055246.3821776 INFO: reactions 933\n",
      "1723055246.3915482 INFO: reactions 933\n",
      "1723055246.4186606 INFO: metabolites 1203\n",
      "1723055246.422234 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055246.5023258 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055246.552607 INFO: Default biomass: [bio1]\n",
      "1723055246.5632126 INFO: Default biomass: [bio1]\n",
      "1723055246.5957959 INFO: Default biomass: [bio1]\n",
      "1723055246.6269903 INFO: reactions 1212\n",
      "1723055246.63382 INFO: metabolites 898\n",
      "1723055246.657996 INFO: metabolites 1246\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.3.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055246.6783714 INFO: reactions 859\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_H2O_MG_DASTool_bins_metabat.13.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055246.7184768 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_metabat.14.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055246.7684786 INFO: Default biomass: [bio1]\n",
      "1723055246.769664 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055246.77435 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.17.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055246.7965589 INFO: reactions 1224\n",
      "1723055246.8127663 INFO: reactions 1023\n",
      "1723055246.8460839 INFO: metabolites 1211\n",
      "1723055246.8548362 INFO: Default biomass: [bio1]\n",
      "1723055246.946629 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.20.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055246.949397 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.7.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055247.038271 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055247.0606885 INFO: metabolites 774\n",
      "1723055247.073773 INFO: reactions 835\n",
      "1723055247.077444 INFO: reactions 686\n",
      "1723055247.1096807 INFO: metabolites 1166\n",
      "1723055247.1375475 INFO: Default biomass: [bio1]\n",
      "1723055247.1568046 INFO: reactions 967\n",
      "1723055247.164918 INFO: metabolites 1123\n",
      "1723055247.1653004 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055247.1763458 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.22.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055247.2460775 INFO: Default biomass: [bio1]\n",
      "1723055247.2720187 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055247.335932 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055247.376014 INFO: reactions 673\n",
      "1723055247.3797872 INFO: metabolites 1108\n",
      "1723055247.3831093 INFO: metabolites 1053\n",
      "1723055247.3909924 INFO: Default biomass: [bio1]\n",
      "1723055247.393779 INFO: metabolites 1176\n",
      "1723055247.4463708 INFO: Default biomass: [bio1]\n",
      "1723055247.4555206 INFO: Default biomass: [bio1]\n",
      "1723055247.5093455 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.8.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055247.5127342 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055247.5316224 INFO: reactions 1194\n",
      "1723055247.534195 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.20.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055247.633916 INFO: metabolites 1078\n",
      "1723055247.6550777 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.52.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055247.6943626 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.47.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055247.7067118 INFO: Default biomass: [bio1]\n",
      "1723055247.7635374 INFO: reactions 1142\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_H2O_MG_DASTool_bins_metabat.49.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055247.8790946 INFO: metabolites 1256\n",
      "1723055247.883025 INFO: reactions 1163\n",
      "1723055247.9114225 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055247.913806 INFO: reactions 1031\n",
      "1723055247.9191556 INFO: reactions 1238\n",
      "1723055247.9648933 INFO: reactions 1082\n",
      "1723055247.981372 INFO: metabolites 1197\n",
      "1723055248.0088882 INFO: reactions 1159\n",
      "1723055248.014703 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.0333054 INFO: metabolites 953\n",
      "1723055248.0332434 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.0496726 INFO: metabolites 1061\n",
      "1723055248.061409 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.081413 INFO: metabolites 927\n",
      "1723055248.0960557 INFO: metabolites 1115\n",
      "1723055248.0977535 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.1534233 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.1667664 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.2055366 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055248.4147682 INFO: reactions 857\n",
      "1723055248.4451678 INFO: Default biomass: [bio1]\n",
      "1723055248.4532914 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055248.4819438 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_metabat.5.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055248.5092132 INFO: reactions 864\n",
      "1723055248.5198188 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.534306 INFO: Default biomass: [bio1]\n",
      "1723055248.5748665 INFO: reactions 1015\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_metabat.6.pangenome.RAST.glm4ec.pyr"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055248.580971 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055248.6204386 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.6279836 INFO: Default biomass: [bio1]\n",
      "1723055248.6310842 INFO: reactions 1073\n",
      "1723055248.6321113 INFO: reactions 1212\n",
      "1723055248.633354 INFO: reactions 1030\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.12.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055248.7025363 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.7491753 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.7503235 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.7599096 INFO: metabolites 1197\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.66.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055248.802954 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055248.895409 INFO: Default biomass: [bio1]\n",
      "1723055248.9237747 INFO: reactions 1086\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.24.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055248.9426138 INFO: metabolites 1131\n",
      "1723055248.9550636 INFO: metabolites 959\n",
      "1723055248.9753506 INFO: Default biomass: [bio1]\n",
      "1723055249.0225441 INFO: metabolites 1005\n",
      "1723055249.0514584 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055249.0603647 INFO: metabolites 908\n",
      "1723055249.1090565 INFO: Default biomass: [bio1]\n",
      "1723055249.1139374 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.18.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055249.1756847 INFO: Default biomass: [bio1]\n",
      "1723055249.1771681 INFO: Default biomass: [bio1]\n",
      "1723055249.268614 INFO: metabolites 1016\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.16.pangenome.RAST.glm4ec.pyrSalt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_maxbin.022.pangenome.RAST.glm4ec.pyr\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055249.3272638 INFO: Default biomass: [bio1]\n",
      "1723055249.3781362 INFO: metabolites 1205\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.25.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055249.4072247 INFO: reactions 921\n",
      "1723055249.429385 INFO: reactions 828\n",
      "1723055249.507869 INFO: Default biomass: [bio1]\n",
      "1723055249.5266478 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055249.533707 INFO: metabolites 1090\n",
      "1723055249.5994873 INFO: reactions 1130\n",
      "1723055249.6593897 INFO: reactions 1153\n",
      "1723055249.7362092 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055249.7784755 INFO: metabolites 1171\n",
      "1723055249.7865472 INFO: metabolites 1094\n",
      "1723055249.8110516 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055249.8624573 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.52.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055250.0327427 INFO: reactions 1215\n",
      "1723055250.0855472 INFO: reactions 1030\n",
      "1723055250.0880857 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055250.1723535 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055250.2076066 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055250.21967 INFO: Default biomass: [bio1]\n",
      "1723055250.2492275 INFO: reactions 959\n",
      "1723055250.2868853 INFO: Default biomass: [bio1]\n",
      "1723055250.297358 INFO: reactions 1171\n",
      "1723055250.36344 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055250.4412253 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055250.4467247 INFO: Default biomass: [bio1]\n",
      "1723055250.451677 INFO: reactions 950\n",
      "1723055250.4533763 INFO: metabolites 1059\n",
      "1723055250.4572449 INFO: reactions 1086\n",
      "1723055250.5705512 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055250.5855284 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055250.6301296 INFO: Default biomass: [bio1]\n",
      "1723055250.667239 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.10.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055250.7580485 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055250.926245 INFO: Default biomass: [bio1]\n",
      "1723055250.9561713 INFO: Default biomass: [bio1]\n",
      "1723055250.9878912 INFO: metabolites 706\n",
      "1723055251.0052109 INFO: reactions 974\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.124.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055251.1209784 INFO: reactions 1083\n",
      "1723055251.1242132 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055251.178235 INFO: metabolites 1214\n",
      "1723055251.2530744 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055251.2654195 INFO: metabolites 1301\n",
      "1723055251.2721524 INFO: reactions 639\n",
      "1723055251.4193377 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055251.4996333 INFO: metabolites 1135\n",
      "1723055251.5222888 INFO: Default biomass: [bio1]\n",
      "1723055251.5397208 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.24.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055251.6740706 INFO: Default biomass: [bio1]\n",
      "1723055251.6845047 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.27.pangenome.RAST.glm4ec.pyr\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723055252.0118968 INFO: metabolites 1040\n",
      "1723055252.1369624 INFO: metabolites 1110\n",
      "1723055252.1476576 INFO: reactions 1080\n",
      "1723055252.277363 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055252.6508863 INFO: reactions 1197\n",
      "1723055252.6578014 INFO: reactions 1066\n",
      "1723055252.7261033 INFO: Default biomass: [bio1]\n",
      "1723055252.7763538 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055252.8001833 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055252.8051422 INFO: reactions 1300\n",
      "1723055252.9736795 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055253.1904063 INFO: Default biomass: [bio1]\n",
      "1723055253.2807436 INFO: Default biomass: [bio1]\n",
      "1723055253.5302608 INFO: Default biomass: [bio1]\n",
      "1723055253.587202 INFO: reactions 1021\n",
      "1723055253.73062 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723055254.1475523 INFO: Default biomass: [bio1]\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "\n",
    "\n",
    "def cleanProbabilities(arguments):\n",
    "    # load arguments\n",
    "    (mag_names_list, name_to_mag, studies, consolidated_gf_results, index) = arguments\n",
    "    \n",
    "    # print status\n",
    "    from multiprocess import current_process\n",
    "    pid = current_process().name\n",
    "    print(pid)\n",
    "    \n",
    "    for name in mag_names_list:\n",
    "        modelname = name_to_mag[name.replace(\".pg.G.D\", '')].replace(\".contigs__.RAST\", \".pangenome.RAST.glm4ec.pyr\")\n",
    "        print(modelname)\n",
    "        mdlutl = util.msrecon.get_model(modelname,186678)\n",
    "        for study in studies:\n",
    "            if study not in consolidated_gf_results[name]:\n",
    "                continue\n",
    "            for phenotype in consolidated_gf_results[name][study]:\n",
    "                if \"reactions\" in consolidated_gf_results[name][study][phenotype]:\n",
    "                    consolidated_gf_results[name][study][phenotype][\"new_probability\"] = 0\n",
    "                    consolidated_gf_results[name][study][phenotype][\"probrxn\"] = 0\n",
    "                    consolidated_gf_results[name][study][phenotype][\"otherrxn\"] = 0\n",
    "                    consolidated_gf_results[name][study][phenotype][\"origgfrxn\"] = 0\n",
    "                    consolidated_gf_results[name][study][phenotype][\"new_gf\"] = 0\n",
    "                    consolidated_gf_results[name][study][phenotype][\"new_gf_reactions\"] = []\n",
    "                    found = False\n",
    "                    for rxn in consolidated_gf_results[name][study][phenotype][\"reactions\"]:\n",
    "                        direction = rxn[0:1]\n",
    "                        rxnid = rxn[1:]\n",
    "                        if rxnid[0:3] == \"EX_\" and rxnid[3:11] == phenotype:\n",
    "                            found = True\n",
    "                        if rxnid not in mdlutl.model.reactions:\n",
    "                            consolidated_gf_results[name][study][phenotype][\"new_gf_reactions\"].append(rxn)\n",
    "                            consolidated_gf_results[name][study][phenotype][\"new_gf\"] += 1\n",
    "                        elif direction == \">\":\n",
    "                            if mdlutl.model.reactions.get_by_id(rxnid).upper_bound <= 0:\n",
    "                                consolidated_gf_results[name][study][phenotype][\"new_gf_reactions\"].append(rxn)\n",
    "                                consolidated_gf_results[name][study][phenotype][\"new_gf\"] += 1\n",
    "                            elif rxnid in reaction_probabilities[name]:\n",
    "                                consolidated_gf_results[name][study][phenotype][\"new_probability\"] += reaction_probabilities[name][rxnid]\n",
    "                                consolidated_gf_results[name][study][phenotype][\"probrxn\"] += 1\n",
    "                            elif rxnid[0:3] != \"bio\" and rxnid[0:3] != \"EX_\" and rxnid[0:3] != \"DM_\" and len(mdlutl.model.reactions.get_by_id(rxnid).genes) == 0:\n",
    "                                consolidated_gf_results[name][study][phenotype][\"origgfrxn\"] += 1\n",
    "                            else:\n",
    "                                consolidated_gf_results[name][study][phenotype][\"otherrxn\"] += 1\n",
    "                                consolidated_gf_results[name][study][phenotype][\"new_probability\"] += 0.05  \n",
    "                        elif direction == \"<\":\n",
    "                            if mdlutl.model.reactions.get_by_id(rxnid).lower_bound >= 0:\n",
    "                                consolidated_gf_results[name][study][phenotype][\"new_gf_reactions\"].append(rxn)\n",
    "                                consolidated_gf_results[name][study][phenotype][\"new_gf\"] += 1\n",
    "                            elif rxnid in reaction_probabilities[name]:\n",
    "                                consolidated_gf_results[name][study][phenotype][\"new_probability\"] += reaction_probabilities[name][rxnid]\n",
    "                                consolidated_gf_results[name][study][phenotype][\"probrxn\"] += 1\n",
    "                            elif rxnid[0:3] != \"bio\" and rxnid[0:3] != \"EX_\" and rxnid[0:3] != \"DM_\" and len(mdlutl.model.reactions.get_by_id(rxnid).genes) == 0:\n",
    "                                consolidated_gf_results[name][study][phenotype][\"origgfrxn\"] += 1\n",
    "                            else:\n",
    "                                consolidated_gf_results[name][study][phenotype][\"otherrxn\"] += 1\n",
    "                                consolidated_gf_results[name][study][phenotype][\"new_probability\"] += 0.05  \n",
    "                    totalrxn = consolidated_gf_results[name][study][phenotype][\"probrxn\"]+consolidated_gf_results[name][study][phenotype][\"new_gf\"]+consolidated_gf_results[name][study][phenotype][\"otherrxn\"]\n",
    "                    consolidated_gf_results[name][study][phenotype][\"new_probability\"] = consolidated_gf_results[name][study][phenotype][\"new_probability\"]/totalrxn\n",
    "\n",
    "                    if not found:\n",
    "                        consolidated_gf_results[name][study][phenotype][\"original_objective_value\"] = consolidated_gf_results[name][study][phenotype][\"objective_value\"]\n",
    "                        consolidated_gf_results[name][study][phenotype][\"objective_value\"] = 0\n",
    "                        consolidated_gf_results[name][study][phenotype][\"class\"] = \"N\"\n",
    "                        consolidated_gf_results[name][study][phenotype][\"positive\"] = False\n",
    "                        consolidated_gf_results[name][study][phenotype][\"new_probability\"] = 0\n",
    "    util.save(f\"cleaned_SMIPPs/gf_phenotype_results_{index}\",consolidated_gf_results)\n",
    "                    \n",
    "\n",
    "# define the arguments\n",
    "consolidated_gf_results = util.load(\"SMIPPs/new_gf_phenotype_results\")\n",
    "reaction_probabilities = util.load(\"SMIPPs/reaction_probabilities\")\n",
    "name_to_mag = util.load(\"name_to_mag\")\n",
    "studies = [\"uptake\",\"excretion\",\"growth\"]\n",
    "\n",
    "# run the parallelized code\n",
    "parallelize = True\n",
    "if parallelize:\n",
    "    from multiprocess import Pool\n",
    "    from os import cpu_count\n",
    "    from numpy import array_split\n",
    "    \n",
    "    # the MAGs are partitioned into groups of 4, and a thread is created to handle each group \n",
    "    mag_names_lists = array_split(list(consolidated_gf_results.keys()), int(len(consolidated_gf_results)/4))\n",
    "    pool_count = min(cpu_count(), len(mag_names_lists))\n",
    "    print(f\"{pool_count} threads will be created\")\n",
    "    pool = Pool(pool_count)\n",
    "    args = [(mag_names_list, name_to_mag, studies, {k:v for k,v in consolidated_gf_results.items() if k in mag_names_list}, index)\n",
    "           for index, mag_names_list in enumerate(mag_names_lists)]\n",
    "    list_of_outputs = pool.map(cleanProbabilities, args)\n",
    "else:\n",
    "    outputs = cleanProbabilities([[list(consolidated_gf_results.keys())[0]], name_to_mag, studies, consolidated_gf_results, 0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## combining the parallelized partitions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from json import load, dump\n",
    "from glob import glob\n",
    "\n",
    "gf_phenotype_results = {}\n",
    "for path in glob(r\"datacache/cleaned_SMIPPs/gf_phenotype_results_*.json\"):\n",
    "    with open(path, 'r') as jsonIn:\n",
    "        gf_phenotype_results.update(load(jsonIn))\n",
    "        \n",
    "with open(\"datacache/cleaned_SMIPPs/gf_phenotype_results.json\", 'w') as jsonOut:\n",
    "    dump(gf_phenotype_results, jsonOut, indent=3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computing MMIPPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "consolidated_gf_results = util.load(\"cleaned_SMIPPs/gf_phenotype_results\")\n",
    "metabolites = util.load(\"metabolites\")\n",
    "metabolite_names = util.load(\"metabolite_names\")\n",
    "mag_abundances = util.load(\"mag_abundances\")\n",
    "mag_to_name = util.load(\"mag_to_name\")\n",
    "# total_mag_abundances = util.load(\"total_mag_abundances\")\n",
    "records = {\n",
    "    \"uptake_prob\":[{\"Sample\":\"Name\"}],\n",
    "    \"excretion_prob\":[{\"Sample\":\"Name\"}],\n",
    "    \"growth_prob\":[{\"Sample\":\"Name\"}]\n",
    "}\n",
    "types = [\"uptake\",\"excretion\",\"growth\"]\n",
    "for met in metabolites:\n",
    "    for record in records:\n",
    "        records[record][0][met] = metabolite_names[met]\n",
    "for sample in mag_abundances:\n",
    "    new_record = {\n",
    "        \"uptake_prob\": {\"Sample\":sample},\n",
    "        \"excretion_prob\": {\"Sample\":sample},\n",
    "        \"growth_prob\": {\"Sample\":sample}\n",
    "    }\n",
    "    for metabolite in metabolites:\n",
    "        for record in new_record:\n",
    "            new_record[record][metabolite] = 0\n",
    "    total_abundance = sum(list(mag_abundances[sample].values()))\n",
    "    for OGname in mag_abundances[sample]:\n",
    "        if OGname == \"unbinned_reads\":  continue\n",
    "        # print(OGname)\n",
    "        name = mag_to_name[OGname+\".contigs__.RAST\"]+\".pg.G.D\"\n",
    "        # print(name)\n",
    "        for metabolite in metabolites:\n",
    "            for currtype in types:\n",
    "                if currtype in consolidated_gf_results[name]:\n",
    "                    if metabolite in consolidated_gf_results[name][currtype] and \"new_probability\" in consolidated_gf_results[name][currtype][metabolite]:\n",
    "                        new_record[currtype+\"_prob\"][metabolite] += mag_abundances[sample][OGname]/total_abundance*consolidated_gf_results[name][currtype][metabolite][\"new_probability\"]\n",
    "    for record in records:\n",
    "        records[record].append(new_record[record])\n",
    "#Printing interation matrices\n",
    "for record in records:\n",
    "    df = pd.DataFrame.from_records(records[record])\n",
    "    df.to_csv(util.output_dir+\"/Sample\"+record+\"Interactions.csv\",index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Processing the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "consolidated_gf_results = util.load(\"cleaned_SMIPPs/gf_phenotype_results\")\n",
    "metabolites = util.load(\"metabolites\")\n",
    "metabolite_names = util.load(\"metabolite_names\")\n",
    "mag_list = util.load(\"mag_list\")\n",
    "mag_to_name = util.load(\"mag_to_name\")\n",
    "records = {\n",
    "    \"uptake_prob\":[{\"MAG\":\"Name\"}],\n",
    "    \"excretion_prob\":[{\"MAG\":\"Name\"}],\n",
    "    \"growth_prob\":[{\"MAG\":\"Name\"}],\n",
    "    \"uptake_data\":[{\"MAG\":\"Name\"}],\n",
    "    \"excretion_data\":[{\"MAG\":\"Name\"}],\n",
    "    \"growth_data\":[{\"MAG\":\"Name\"}],\n",
    "}\n",
    "types = [\"uptake\",\"excretion\",\"growth\"]\n",
    "for met in metabolites:\n",
    "    for record in records:\n",
    "        records[record][0][met] = metabolite_names[met]\n",
    "for mag in mag_list:\n",
    "    name = mag_to_name[mag[1]]+\".pg.G.D\"\n",
    "    if name not in consolidated_gf_results:\n",
    "        print(\"No data for \",name)\n",
    "        continue\n",
    "    new_record = {}\n",
    "    for currtype in types:\n",
    "        new_record[currtype+\"_prob\"] = {\"MAG\":name}\n",
    "        new_record[currtype+\"_data\"] = {\"MAG\":name}\n",
    "    for metabolite in metabolites:\n",
    "        for currtype in types:\n",
    "            if currtype in consolidated_gf_results[name]:\n",
    "                if metabolite in consolidated_gf_results[name][currtype] and \"new_probability\" in consolidated_gf_results[name][currtype][metabolite]:\n",
    "                    new_record[currtype+\"_prob\"][metabolite] = consolidated_gf_results[name][currtype][metabolite][\"new_probability\"]\n",
    "                    new_record[currtype+\"_data\"][metabolite] = str(consolidated_gf_results[name][currtype][metabolite][\"probrxn\"])\n",
    "                    new_record[currtype+\"_data\"][metabolite] += \"/\"+str(consolidated_gf_results[name][currtype][metabolite][\"otherrxn\"])\n",
    "                    new_record[currtype+\"_data\"][metabolite] += \"/\"+str(consolidated_gf_results[name][currtype][metabolite][\"origgfrxn\"])\n",
    "                    new_record[currtype+\"_data\"][metabolite] += \"/\"+str(consolidated_gf_results[name][currtype][metabolite][\"new_gf\"])\n",
    "                    new_record[currtype+\"_data\"][metabolite] += \"/\"+str(consolidated_gf_results[name][currtype][metabolite][\"gapfill_count\"])\n",
    "                    new_record[currtype+\"_data\"][metabolite] += \"/\"+str(consolidated_gf_results[name][currtype][metabolite][\"reaction_count\"])\n",
    "    for record in records:\n",
    "        records[record].append(new_record[record])\n",
    "#Printing interation matrices\n",
    "for record in records:\n",
    "    df = pd.DataFrame.from_records(records[record])\n",
    "    df.to_csv(util.output_dir+\"/MAG\"+record+\"Interactions.csv\",index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Gapfilling models based on gapfilling simulation results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n",
      "77 threads will be created\n",
      "ForkPoolWorker-78\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601464.8968296 INFO: metabolites 1117\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-79\n",
      "ForkPoolWorker-80\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601465.3394754 INFO: metabolites 907\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-81\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601465.5681357 INFO: metabolites 927\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-82\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601465.8872838 INFO: metabolites 966\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-83\n",
      "ForkPoolWorker-84\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601466.2707725 INFO: metabolites 1135\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-85\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601466.502306 INFO: metabolites 875\n",
      "1723601466.5568273 INFO: metabolites 1228\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-86\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601466.8348298 INFO: metabolites 1258\n",
      "1723601466.9034708 INFO: metabolites 864\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-87\n",
      "ForkPoolWorker-88\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601467.310634 INFO: metabolites 1033\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-89\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601467.5299253 INFO: metabolites 1110\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-90\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601467.7599406 INFO: metabolites 1055\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-91\n",
      "ForkPoolWorker-92\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601468.266912 INFO: metabolites 1106\n",
      "1723601468.4624505 INFO: metabolites 1217\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-93\n",
      "ForkPoolWorker-94\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601469.180691 INFO: metabolites 1151\n",
      "1723601469.2479944 INFO: metabolites 1123\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-95\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601469.3193498 INFO: metabolites 985\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-96\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601469.5689058 INFO: metabolites 781\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-97\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601469.855563 INFO: metabolites 1120\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-98\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601470.0990386 INFO: metabolites 1084\n",
      "1723601470.512577 INFO: metabolites 1226\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-99\n",
      "ForkPoolWorker-100\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601471.2363613 INFO: metabolites 1149\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-101\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601471.4922569 INFO: metabolites 1143\n",
      "1723601471.9252706 INFO: metabolites 1018\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-102\n",
      "ForkPoolWorker-103\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601472.5028937 INFO: metabolites 1052\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-104\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601472.9447873 INFO: metabolites 1203\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-105\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601473.2313566 INFO: metabolites 1017\n",
      "1723601473.6640148 INFO: metabolites 1138\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-106"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601474.1142752 INFO: metabolites 956\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-107\n",
      "ForkPoolWorker-108\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601474.7055075 INFO: metabolites 955\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-109\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601475.0267162 INFO: metabolites 1178\n",
      "1723601475.312598 INFO: metabolites 958\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-110\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601476.1445992 INFO: metabolites 1098\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-111"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601476.6721861 INFO: metabolites 1181\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-112\n",
      "ForkPoolWorker-113\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601477.0954149 INFO: metabolites 973\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-114"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601477.2357624 INFO: metabolites 1024\n",
      "1723601477.576889 INFO: metabolites 908\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-115\n",
      "ForkPoolWorker-116\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601478.2005146 INFO: metabolites 1142\n",
      "1723601478.641674 INFO: metabolites 1107\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-117\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601479.1261404 INFO: metabolites 1022\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-118\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601479.6190734 INFO: metabolites 1098\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-119\n",
      "ForkPoolWorker-120\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601480.1930635 INFO: metabolites 1203\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-121\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601480.510607 INFO: metabolites 1164\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-122\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601480.9780097 INFO: metabolites 1102\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-123\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601481.4347272 INFO: metabolites 1237\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-124\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601481.8200793 INFO: metabolites 1211\n",
      "1723601482.1714473 INFO: metabolites 1095\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-125\n",
      "ForkPoolWorker-126\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601482.8186274 INFO: metabolites 1099\n",
      "1723601483.1203427 INFO: metabolites 904\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-127\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601486.5788522 INFO: metabolites 1089\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-128\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601487.084894 INFO: metabolites 948\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-129\n",
      "ForkPoolWorker-130\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601487.834413 INFO: metabolites 1185\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-131\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601488.2351227 INFO: metabolites 1036\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-132\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601488.5788069 INFO: metabolites 1051\n",
      "1723601488.8871412 INFO: metabolites 1088\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-133\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601489.3788948 INFO: metabolites 1037\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-134\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601489.9986272 INFO: metabolites 1160\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-135\n",
      "ForkPoolWorker-136\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601490.723524 INFO: metabolites 1002\n",
      "1723601491.2568398 INFO: metabolites 1090\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-137"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601492.010607 INFO: metabolites 1228\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-138\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601492.8172576 INFO: metabolites 1131\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-139\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601493.4916267 INFO: metabolites 1120\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-140\n",
      "ForkPoolWorker-141\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601494.1161513 INFO: metabolites 1165\n",
      "1723601494.5778227 INFO: metabolites 1173\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-142\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601495.2350843 INFO: metabolites 938\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-143\n",
      "ForkPoolWorker-144\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601495.8210802 INFO: metabolites 939\n",
      "1723601496.2263029 INFO: metabolites 1133\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-145\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601496.837148 INFO: metabolites 1116\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-146\n",
      "ForkPoolWorker-147\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601497.4945066 INFO: metabolites 1164\n",
      "1723601497.9656467 INFO: metabolites 1212\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-148\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601498.6577103 INFO: metabolites 887\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-149\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601499.3099434 INFO: metabolites 1000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-150\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601500.0117486 INFO: metabolites 1038\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-151\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601500.9763381 INFO: metabolites 1246\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-152\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601501.9834576 INFO: metabolites 1043\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-153\n",
      "ForkPoolWorker-154\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601503.4422784 INFO: metabolites 1143\n",
      "1723601631.181608 INFO: reactions 718\n",
      "1723601631.2948856 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601631.6990738 INFO: Default biomass: [bio1]\n",
      "1723601651.2316809 INFO: reactions 787\n",
      "1723601651.3559093 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601651.5466013 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601651.766731 INFO: Default biomass: [bio1]\n",
      "1723601662.304645 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601668.7383 INFO: reactions 803\n",
      "1723601668.862316 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601669.3155968 INFO: Default biomass: [bio1]\n",
      "1723601678.022124 INFO: reactions 883\n",
      "1723601678.1606688 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601678.627691 INFO: Default biomass: [bio1]\n",
      "1723601680.2736547 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601682.8321571 INFO: reactions 827\n",
      "1723601682.9624043 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601683.4195843 INFO: Default biomass: [bio1]\n",
      "1723601688.3262284 INFO: reactions 898\n",
      "1723601688.465649 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601688.9703686 INFO: Default biomass: [bio1]\n",
      "1723601690.0257914 INFO: reactions 951\n",
      "1723601690.1732638 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601690.2459676 INFO: reactions 859\n",
      "1723601690.3673117 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601690.6695447 INFO: Default biomass: [bio1]\n",
      "1723601690.7908216 INFO: Default biomass: [bio1]\n",
      "1723601693.953053 INFO: reactions 885\n",
      "1723601694.0880125 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601694.5756881 INFO: Default biomass: [bio1]\n",
      "1723601695.0210323 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601697.6331809 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601698.1065273 INFO: reactions 977\n",
      "1723601698.2470572 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601698.745755 INFO: Default biomass: [bio1]\n",
      "1723601700.307541 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601700.9595113 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601701.0990524 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601701.7613258 INFO: reactions 935\n",
      "1723601701.8984241 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601702.3777456 INFO: Default biomass: [bio1]\n",
      "1723601704.948053 INFO: reactions 937\n",
      "1723601705.0794492 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601705.5767832 INFO: Default biomass: [bio1]\n",
      "1723601706.098299 INFO: reactions 857\n",
      "1723601706.221581 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601706.63513 INFO: Default biomass: [bio1]\n",
      "1723601709.6492858 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601712.7363129 INFO: reactions 886\n",
      "1723601712.853357 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601713.226112 INFO: reactions 916\n",
      "1723601713.2539577 INFO: Default biomass: [bio1]\n",
      "1723601713.3524735 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601713.44958 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601713.79674 INFO: Default biomass: [bio1]\n",
      "1723601713.8016138 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601714.9456553 INFO: reactions 840\n",
      "1723601715.06556 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601715.4114695 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601715.5695722 INFO: Default biomass: [bio1]\n",
      "1723601715.6743426 INFO: reactions 924\n",
      "1723601715.8029778 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601716.2494617 INFO: Default biomass: [bio1]\n",
      "1723601716.6536143 INFO: reactions 973\n",
      "1723601716.7959938 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601717.3340251 INFO: Default biomass: [bio1]\n",
      "1723601718.3866763 INFO: reactions 892\n",
      "1723601718.510022 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601718.9047463 INFO: Default biomass: [bio1]\n",
      "1723601720.4867778 INFO: reactions 1010\n",
      "1723601720.6311991 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601721.1152737 INFO: Default biomass: [bio1]\n",
      "1723601721.4116035 INFO: reactions 821\n",
      "1723601721.5309036 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601721.7224727 INFO: reactions 957\n",
      "1723601721.8518186 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601721.932178 INFO: Default biomass: [bio1]\n",
      "1723601722.3043156 INFO: Default biomass: [bio1]\n",
      "1723601723.6954348 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601723.9852908 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601724.0585113 INFO: reactions 945\n",
      "1723601724.0740921 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601724.1845386 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601724.6080542 INFO: Default biomass: [bio1]\n",
      "1723601724.996462 INFO: reactions 968\n",
      "1723601725.186948 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601725.6539838 INFO: Default biomass: [bio1]\n",
      "1723601726.449919 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601728.0065827 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601729.6424763 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601730.4643161 INFO: reactions 1082\n",
      "1723601730.6038957 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601731.0399153 INFO: reactions 1020\n",
      "1723601731.1048145 INFO: Default biomass: [bio1]\n",
      "1723601731.1824589 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601731.6331894 INFO: Default biomass: [bio1]\n",
      "1723601731.828335 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601731.833997 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601733.806811 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601734.331311 INFO: reactions 972\n",
      "1723601734.3774486 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601734.4891489 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601734.5819426 INFO: reactions 983\n",
      "1723601734.7317622 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601734.930675 INFO: Default biomass: [bio1]\n",
      "1723601735.1884024 INFO: Default biomass: [bio1]\n",
      "1723601735.2318609 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601737.8620968 INFO: reactions 1025\n",
      "1723601738.0044897 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601738.376949 INFO: reactions 996\n",
      "1723601738.5068645 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601738.5132594 INFO: Default biomass: [bio1]\n",
      "1723601738.9629416 INFO: Default biomass: [bio1]\n",
      "1723601739.5878162 INFO: reactions 1106\n",
      "1723601739.6738439 INFO: reactions 978\n",
      "1723601739.75438 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601739.8014235 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601740.2246258 INFO: Default biomass: [bio1]\n",
      "1723601740.2432108 INFO: Default biomass: [bio1]\n",
      "1723601740.2974622 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601740.46761 INFO: reactions 1110\n",
      "1723601740.6018105 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601740.8828013 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601741.006497 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601741.0918794 INFO: Default biomass: [bio1]\n",
      "1723601742.4442825 INFO: reactions 1055\n",
      "1723601742.5113218 INFO: reactions 1037\n",
      "1723601742.576327 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601742.647411 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601743.0414546 INFO: Default biomass: [bio1]\n",
      "1723601743.0885968 INFO: Default biomass: [bio1]\n",
      "1723601743.6716938 INFO: reactions 1106\n",
      "1723601743.93698 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601744.4367044 INFO: Default biomass: [bio1]\n",
      "1723601747.7069547 INFO: reactions 932\n",
      "1723601747.8238254 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601748.1699157 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601748.2554464 INFO: Default biomass: [bio1]\n",
      "1723601748.6219316 INFO: reactions 1044\n",
      "1723601748.7491715 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601749.2042997 INFO: Default biomass: [bio1]\n",
      "1723601749.4638896 INFO: reactions 1067\n",
      "1723601749.5915937 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601749.9765906 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601750.0943437 INFO: Default biomass: [bio1]\n",
      "1723601750.1139736 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601750.2352934 INFO: reactions 1089\n",
      "1723601750.3607066 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601750.640277 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601750.8682916 INFO: Default biomass: [bio1]\n",
      "1723601750.9140134 INFO: reactions 1008\n",
      "1723601751.034315 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601751.466121 INFO: Default biomass: [bio1]\n",
      "1723601751.7550733 INFO: reactions 1181\n",
      "1723601751.9115274 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601751.9439685 INFO: reactions 997\n",
      "1723601752.080469 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601752.0797942 INFO: reactions 1110\n",
      "1723601752.2332332 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601752.4149487 INFO: Default biomass: [bio1]\n",
      "1723601752.450067 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601752.463783 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601752.5111291 INFO: Default biomass: [bio1]\n",
      "1723601752.7094326 INFO: Default biomass: [bio1]\n",
      "1723601753.3189893 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601753.566969 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601753.7048612 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601754.780546 INFO: reactions 1073\n",
      "1723601754.8683894 INFO: reactions 1076\n",
      "1723601754.9985964 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601755.0020468 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601755.4598002 INFO: Default biomass: [bio1]\n",
      "1723601755.474134 INFO: Default biomass: [bio1]\n",
      "1723601756.4807696 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601756.5562236 INFO: reactions 1188\n",
      "1723601756.7039204 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601757.1213832 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601757.2050781 INFO: Default biomass: [bio1]\n",
      "1723601757.7205136 INFO: reactions 1201\n",
      "1723601757.8205569 INFO: reactions 1077\n",
      "1723601757.9395313 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601757.9490707 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601758.1393216 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601758.22114 INFO: reactions 938\n",
      "1723601758.3360078 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601758.4299293 INFO: Default biomass: [bio1]\n",
      "1723601758.70636 INFO: reactions 1121\n",
      "1723601758.7485175 INFO: Default biomass: [bio1]\n",
      "1723601758.8809867 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601759.0636241 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601759.2568579 INFO: reactions 1126\n",
      "1723601759.3286467 INFO: cpd05178 not found in model!\n",
      "1723601759.33104 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601759.3483295 INFO: Default biomass: [bio1]\n",
      "1723601759.3763347 INFO: reactions 1044\n",
      "1723601759.402804 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601759.563041 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601759.9256923 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601759.9439383 INFO: Default biomass: [bio1]\n",
      "1723601760.01313 INFO: Default biomass: [bio1]\n",
      "1723601760.156727 WARNING: Ignoring reaction 'rxn01790_c0' since it already exists.\n",
      "1723601760.3945835 WARNING: Ignoring reaction 'rxn05493_c0' since it already exists.\n",
      "1723601760.3971734 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723601760.5762181 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723601760.7864835 WARNING: Ignoring reaction 'rxn00548_c0' since it already exists.\n",
      "1723601760.789065 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723601761.0296388 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723601761.0323076 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723601761.1343148 WARNING: Ignoring reaction 'rxn01333_c0' since it already exists.\n",
      "1723601761.1379278 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601761.5994282 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601762.2256188 INFO: reactions 1188\n",
      "1723601762.454301 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601762.6153736 INFO: metabolites 1113\n",
      "1723601762.8281124 INFO: reactions 1098\n",
      "1723601762.986258 INFO: Default biomass: [bio1]\n",
      "1723601762.988272 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601763.177078 INFO: reactions 1107\n",
      "1723601763.2018049 INFO: reactions 1123\n",
      "1723601763.3075857 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601763.35227 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601763.4595757 INFO: Default biomass: [bio1]\n",
      "1723601763.5916986 INFO: reactions 1154\n",
      "1723601763.7267165 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601763.773662 INFO: Default biomass: [bio1]\n",
      "1723601763.8602204 INFO: Default biomass: [bio1]\n",
      "1723601764.2320125 INFO: Default biomass: [bio1]\n",
      "1723601764.478421 INFO: reactions 1106\n",
      "1723601764.532492 INFO: reactions 1102\n",
      "1723601764.6067286 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601764.7266886 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601764.8231199 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601765.0414548 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601765.0726118 INFO: reactions 1114\n",
      "1723601765.079308 INFO: Default biomass: [bio1]\n",
      "1723601765.2080014 INFO: Default biomass: [bio1]\n",
      "1723601765.2224474 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601765.7225041 INFO: Default biomass: [bio1]\n",
      "1723601766.243147 INFO: reactions 1211\n",
      "1723601766.3798301 INFO: Default biomass: [bio1]\n",
      "1723601766.4012098 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601766.911825 INFO: Default biomass: [bio1]\n",
      "1723601767.4223976 INFO: reactions 1198\n",
      "1723601767.5688891 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601767.678256 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601767.9515626 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601768.0811727 INFO: Default biomass: [bio1]\n",
      "1723601768.8717036 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601769.2315013 INFO: reactions 1158\n",
      "1723601769.3747077 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601769.4225776 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601769.4465098 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601769.5093207 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601769.8789763 INFO: Default biomass: [bio1]\n",
      "1723601770.3212361 INFO: reactions 1104\n",
      "1723601770.3253808 INFO: reactions 1159\n",
      "1723601770.4465008 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601770.520506 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601770.6119862 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601770.8535843 INFO: reactions 1099\n",
      "1723601770.9499815 INFO: Default biomass: [bio1]\n",
      "1723601771.0398781 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601771.0543818 INFO: Default biomass: [bio1]\n",
      "1723601771.2954805 INFO: reactions 1164\n",
      "1723601771.4928684 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601771.5107262 INFO: Default biomass: [bio1]\n",
      "1723601772.0001686 INFO: Default biomass: [bio1]\n",
      "1723601772.4718275 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601773.1446614 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601773.2687447 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601773.3139355 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601774.2024207 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601774.4506092 INFO: reactions 1033\n",
      "1723601774.5804665 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601774.8116906 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601775.0214224 INFO: Default biomass: [bio1]\n",
      "1723601775.1870005 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601775.5653777 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601776.277139 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601776.2860513 INFO: reactions 1101\n",
      "1723601776.421953 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601776.4275544 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601776.4640896 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601776.9048576 INFO: Default biomass: [bio1]\n",
      "1723601777.5684466 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601777.6163256 INFO: reactions 1219\n",
      "1723601777.7918615 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601778.3577003 INFO: Default biomass: [bio1]\n",
      "1723601779.3766131 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601780.0225985 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601780.2282507 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601780.8019292 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601780.9718165 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601781.3754377 INFO: cpd05178 not found in model!\n",
      "1723601781.3778338 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601781.773813 INFO: reactions 1227\n",
      "1723601781.8400192 WARNING: Ignoring reaction 'rxn00278_c0' since it already exists.\n",
      "1723601781.9283369 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601782.5135195 INFO: Default biomass: [bio1]\n",
      "1723601782.9714007 WARNING: Ignoring reaction 'rxn00745_c0' since it already exists.\n",
      "1723601783.0894082 WARNING: Ignoring reaction 'rxn00427_c0' since it already exists.\n",
      "1723601783.1837869 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601783.2829678 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723601783.4048207 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601783.9389484 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601785.038128 INFO: metabolites 1261\n",
      "1723601785.690535 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601785.7925537 INFO: reactions 1221\n",
      "1723601786.0719893 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601786.615689 INFO: Default biomass: [bio1]\n",
      "1723601786.7194526 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601790.8254 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601791.9857352 INFO: reactions 1246\n",
      "1723601792.1547718 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601792.6892824 INFO: Default biomass: [bio1]\n",
      "1723601792.721558 INFO: reactions 1167\n",
      "1723601792.8644025 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601793.3576531 INFO: Default biomass: [bio1]\n",
      "1723601793.820836 INFO: cpd05178 not found in model!\n",
      "1723601793.8231006 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601794.6262832 WARNING: Ignoring reaction 'rxn02229_c0' since it already exists.\n",
      "1723601794.6618958 INFO: reactions 1238\n",
      "1723601794.750246 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723601794.8111887 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601794.9726849 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723601795.0442622 WARNING: Ignoring reaction 'rxn01799_c0' since it already exists.\n",
      "1723601795.1060984 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601795.1886039 WARNING: Ignoring reaction 'rxn04954_c0' since it already exists.\n",
      "1723601795.1913314 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723601795.2544026 WARNING: Ignoring reaction 'rxn00558_c0' since it already exists.\n",
      "1723601795.360352 INFO: Default biomass: [bio1]\n",
      "1723601795.8230257 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601797.491924 INFO: metabolites 875\n",
      "1723601800.9645767 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601801.5700448 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601803.2312236 INFO: cpd05178 not found in model!\n",
      "1723601803.2334392 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601803.471451 WARNING: Ignoring reaction 'rxn00422_c0' since it already exists.\n",
      "1723601803.5396383 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723601803.7304702 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601804.3178394 INFO: cpd05178 not found in model!\n",
      "1723601804.3203142 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601804.670712 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601804.6732664 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601804.7645462 WARNING: Ignoring reaction 'rxn00806_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_concoct_out.62.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.03004058733972695}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601804.8521116 WARNING: Ignoring reaction 'rxn01575_c0' since it already exists.\n",
      "1723601805.0073206 WARNING: Ignoring reaction 'rxn00912_c0' since it already exists.\n",
      "1723601805.3188756 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723601805.448842 WARNING: Ignoring reaction 'rxn00225_c0' since it already exists.\n",
      "1723601805.5335782 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601805.5363028 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601805.8534276 WARNING: Ignoring reaction 'rxn04954_c0' since it already exists.\n",
      "1723601806.0207474 WARNING: Ignoring reaction 'rxn02229_c0' since it already exists.\n",
      "1723601806.088976 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723601806.2638607 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601806.50428 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601807.7174234 INFO: metabolites 581\n",
      "1723601808.2337112 INFO: cpd05178 not found in model!\n",
      "1723601808.2360713 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601808.2602234 INFO: cpd05178 not found in model!\n",
      "1723601808.2629616 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601808.277994 INFO: metabolites 1125\n",
      "1723601808.5406158 WARNING: Ignoring reaction 'rxn01418_c0' since it already exists.\n",
      "1723601808.716537 WARNING: Ignoring reaction 'rxn00806_c0' since it already exists.\n",
      "1723601808.747997 WARNING: Ignoring reaction 'rxn01575_c0' since it already exists.\n",
      "1723601808.9758546 WARNING: Ignoring reaction 'rxn04954_c0' since it already exists.\n",
      "1723601809.4103606 WARNING: Ignoring reaction 'rxn00272_c0' since it already exists.\n",
      "1723601809.4782457 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601809.4811385 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601809.5555909 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.8082130711236726, '<': 1.788459450264455}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601809.9133315 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723601809.9767957 WARNING: Ignoring reaction 'rxn00830_c0' since it already exists.\n",
      "1723601810.0342538 WARNING: Ignoring reaction 'rxn00548_c0' since it already exists.\n",
      "1723601810.0366864 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723601810.052696 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723601810.3087049 WARNING: Ignoring reaction 'rxn02774_c0' since it already exists.\n",
      "1723601810.3113065 WARNING: Ignoring reaction 'rxn02056_c0' since it already exists.\n",
      "1723601810.3604453 WARNING: Ignoring reaction 'rxn00499_c0' since it already exists.\n",
      "1723601810.4352117 WARNING: Ignoring reaction 'rxn00270_c0' since it already exists.\n",
      "1723601810.446465 WARNING: Ignoring reaction 'rxn09231_c0' since it already exists.\n",
      "1723601810.4854584 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723601810.5128856 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723601810.5152931 WARNING: Ignoring reaction 'rxn00784_c0' since it already exists.\n",
      "1723601810.6392744 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601810.9672732 WARNING: Ignoring reaction 'rxn00467_c0' since it already exists.\n",
      "1723601810.9860764 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723601811.189592 WARNING: Ignoring reaction 'rxn09231_c0' since it already exists.\n",
      "1723601811.339439 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601812.2460659 INFO: metabolites 1264\n",
      "1723601813.0182152 INFO: metabolites 1143\n",
      "1723601814.2062209 INFO: cpd05178 not found in model!\n",
      "1723601814.2085996 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601814.7440279 WARNING: Ignoring reaction 'rxn09180_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.119.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.9479011890574547, '>': 1.358621204581581}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601815.3458543 WARNING: Ignoring reaction 'rxn00102_c0' since it already exists.\n",
      "1723601815.7344244 INFO: cpd05178 not found in model!\n",
      "1723601815.7368417 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601815.7534585 WARNING: Ignoring reaction 'rxn00499_c0' since it already exists.\n",
      "1723601816.0602713 WARNING: Ignoring reaction 'rxn00003_c0' since it already exists.\n",
      "1723601816.0754676 WARNING: Ignoring reaction 'rxn00173_c0' since it already exists.\n",
      "1723601816.2761126 WARNING: Ignoring reaction 'rxn00747_c0' since it already exists.\n",
      "1723601816.3735821 WARNING: Ignoring reaction 'rxn00375_c0' since it already exists.\n",
      "1723601816.5219615 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723601816.5246704 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723601816.99328 WARNING: Ignoring reaction 'rxn10124_c0' since it already exists.\n",
      "1723601817.1168044 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723601817.1192706 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723601817.8300552 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601817.9896793 WARNING: Ignoring reaction 'rxn01423_c0' since it already exists.\n",
      "1723601818.1559317 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601820.2251208 INFO: metabolites 1120\n",
      "1723601822.052874 INFO: cpd05178 not found in model!\n",
      "1723601822.0553493 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601822.345543 WARNING: Ignoring reaction 'rxn00251_c0' since it already exists.\n",
      "1723601822.5284717 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723601823.345453 INFO: cpd05178 not found in model!\n",
      "1723601823.3476906 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601823.6326241 WARNING: Ignoring reaction 'rxn13707_c0' since it already exists.\n",
      "1723601823.6354687 WARNING: Ignoring reaction 'rxn13746_c0' since it already exists.\n",
      "1723601823.6894855 WARNING: Ignoring reaction 'rxn00102_c0' since it already exists.\n",
      "1723601823.8333097 WARNING: Ignoring reaction 'rxn00362_c0' since it already exists.\n",
      "1723601823.8900704 INFO: cpd05178 not found in model!\n",
      "1723601823.892403 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601823.9843774 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601824.0468326 WARNING: Ignoring reaction 'rxn04794_c0' since it already exists.\n",
      "1723601824.4764612 WARNING: Ignoring reaction 'rxn02285_c0' since it already exists.\n",
      "1723601824.6564255 WARNING: Ignoring reaction 'rxn03194_c0' since it already exists.\n",
      "1723601824.803782 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601824.8065062 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': -0.4315466402478201}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601825.1414573 INFO: cpd05178 not found in model!\n",
      "1723601825.1436336 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_maxbin.029.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.0017978058305046}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601825.1929874 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723601825.3324678 WARNING: Ignoring reaction 'rxn00692_c0' since it already exists.\n",
      "1723601825.4506202 INFO: cpd05178 not found in model!\n",
      "1723601825.45275 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601825.4634905 WARNING: Ignoring reaction 'rxn04954_c0' since it already exists.\n",
      "1723601825.594211 INFO: metabolites 1106\n",
      "1723601825.6137471 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723601825.6578343 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.25.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7814113165516492, '<': 1.3314433767371034}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601825.865012 WARNING: Ignoring reaction 'rxn00251_c0' since it already exists.\n",
      "1723601826.1903934 WARNING: Ignoring reaction 'rxn00003_c0' since it already exists.\n",
      "1723601826.628392 WARNING: Ignoring reaction 'rxn01423_c0' since it already exists.\n",
      "1723601826.69457 WARNING: Ignoring reaction 'rxn00510_c0' since it already exists.\n",
      "1723601826.744926 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723601826.812309 WARNING: Ignoring reaction 'rxn01799_c0' since it already exists.\n",
      "1723601826.852407 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n",
      "1723601826.9834871 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601827.2809858 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723601827.283597 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723601827.305301 WARNING: Ignoring reaction 'rxn00499_c0' since it already exists.\n",
      "1723601827.5300062 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601827.6016479 INFO: cpd05178 not found in model!\n",
      "1723601827.603786 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601827.8991513 INFO: cpd05178 not found in model!\n",
      "1723601827.901406 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601828.0000408 WARNING: Ignoring reaction 'rxn00350_c0' since it already exists.\n",
      "1723601828.1292503 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601828.1577265 INFO: metabolites 1123\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_H2O_MG_DASTool_bins_metabat.29.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.7524458906219056}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601828.914596 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723601828.9763412 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601828.9788997 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601829.0066316 WARNING: Ignoring reaction 'rxn00362_c0' since it already exists.\n",
      "1723601829.2025683 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723601829.3783445 INFO: metabolites 1046\n",
      "1723601829.6075873 INFO: metabolites 1193\n",
      "1723601830.1580853 WARNING: Ignoring reaction 'rxn01423_c0' since it already exists.\n",
      "1723601830.223353 WARNING: Ignoring reaction 'rxn01418_c0' since it already exists.\n",
      "1723601830.225822 WARNING: Ignoring reaction 'rxn00317_c0' since it already exists.\n",
      "1723601830.3982852 WARNING: Ignoring reaction 'rxn10124_c0' since it already exists.\n",
      "1723601830.5567763 WARNING: Ignoring reaction 'rxn00350_c0' since it already exists.\n",
      "1723601830.5794065 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601830.9233818 INFO: metabolites 1016\n",
      "1723601831.3379414 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601833.0368962 INFO: metabolites 937\n",
      "1723601833.1042626 INFO: cpd05178 not found in model!\n",
      "1723601833.106589 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601833.5136547 INFO: cpd05178 not found in model!\n",
      "1723601833.5159662 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601833.6483462 INFO: metabolites 1108\n",
      "1723601833.9993947 WARNING: Ignoring reaction 'rxn00616_c0' since it already exists.\n",
      "1723601834.197119 INFO: metabolites 1210\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.73.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': -0.05618703772568184}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601834.4280643 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723601834.4712684 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601834.724183 WARNING: Ignoring reaction 'rxn00467_c0' since it already exists.\n",
      "1723601835.0202742 WARNING: Ignoring reaction 'rxn04464_c0' since it already exists.\n",
      "1723601835.1737585 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601836.1146615 INFO: metabolites 1259\n",
      "1723601836.7163484 INFO: metabolites 993\n",
      "1723601838.6631002 INFO: reactions 502\n",
      "1723601838.725341 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601838.916575 INFO: Default biomass: [bio1]\n",
      "1723601840.0987048 INFO: cpd05178 not found in model!\n",
      "1723601840.1011672 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601841.3016398 WARNING: Ignoring reaction 'rxn02185_c0' since it already exists.\n",
      "1723601841.6925163 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723601841.747297 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601843.196541 INFO: metabolites 1031\n",
      "1723601846.4608366 INFO: cpd05178 not found in model!\n",
      "1723601846.4632683 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601846.623122 INFO: cpd05178 not found in model!\n",
      "1723601846.625441 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601846.8305662 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601846.9439356 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.61.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 2.338970451834317, '>': -0.040347733399241074}\n",
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_maxbin.062.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.0272812328409875}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601847.8767087 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723601848.3754737 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723601848.625668 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723601848.8725276 WARNING: Ignoring reaction 'rxn00949_c0' since it already exists.\n",
      "1723601848.9411616 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723601849.282829 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723601849.3280115 WARNING: Ignoring reaction 'rxn00680_c0' since it already exists.\n",
      "1723601849.4674377 WARNING: Ignoring reaction 'rxn01423_c0' since it already exists.\n",
      "1723601849.5340621 WARNING: Ignoring reaction 'rxn13143_c0' since it already exists.\n",
      "1723601849.8231733 WARNING: Ignoring reaction 'rxn01418_c0' since it already exists.\n",
      "1723601849.8925688 WARNING: Ignoring reaction 'rxn00510_c0' since it already exists.\n",
      "1723601850.0773504 INFO: cpd05178 not found in model!\n",
      "1723601850.079666 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601850.3925564 WARNING: Ignoring reaction 'rxn02229_c0' since it already exists.\n",
      "1723601850.4008992 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601850.7917824 WARNING: Ignoring reaction 'rxn00499_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.concoct_out.54.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.3859744207347736}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601851.1898072 WARNING: Ignoring reaction 'rxn01790_c0' since it already exists.\n",
      "1723601851.3654475 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601851.445534 WARNING: Ignoring reaction 'rxn01333_c0' since it already exists.\n",
      "1723601851.5435646 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723601851.9536417 INFO: metabolites 1271\n",
      "1723601852.0078728 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601852.3794403 INFO: cpd05178 not found in model!\n",
      "1723601852.3817432 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601852.7692592 INFO: cpd05178 not found in model!\n",
      "1723601852.7714756 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601853.1483052 INFO: metabolites 1052\n",
      "1723601853.5447457 INFO: metabolites 945\n",
      "1723601853.79565 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601853.875017 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601853.8777137 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601854.0348032 WARNING: Ignoring reaction 'rxn01301_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.43.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.9324189769639717}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601854.084273 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723601854.1850717 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723601854.3232622 WARNING: Ignoring reaction 'rxn00364_c0' since it already exists.\n",
      "1723601854.4422753 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601854.960429 INFO: cpd05178 not found in model!\n",
      "1723601854.9627957 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601855.3970048 INFO: metabolites 674\n",
      "1723601856.1141455 INFO: metabolites 578\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.19.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.44998958658920696}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601856.4979446 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723601857.1680484 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601858.838179 INFO: metabolites 1133\n",
      "1723601860.726368 INFO: cpd05178 not found in model!\n",
      "1723601860.72872 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.21.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.008834880976016111}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601862.1834939 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723601862.6857426 INFO: cpd05178 not found in model!\n",
      "1723601862.688043 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601862.7362025 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601863.4226828 INFO: cpd05178 not found in model!\n",
      "1723601863.4250686 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601863.959974 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723601863.9636874 WARNING: Ignoring reaction 'rxn04954_c0' since it already exists.\n",
      "1723601864.4023416 INFO: metabolites 1272\n",
      "1723601864.501886 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601864.5047393 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601864.630377 WARNING: Ignoring reaction 'rxn00427_c0' since it already exists.\n",
      "1723601864.7627492 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601865.3682795 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601866.33411 INFO: metabolites 938\n",
      "1723601866.4918337 INFO: cpd05178 not found in model!\n",
      "1723601866.4939523 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601866.9090872 INFO: metabolites 1210\n",
      "1723601867.3407059 INFO: cpd05178 not found in model!\n",
      "1723601867.3429258 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601867.7244475 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n",
      "1723601867.7936435 WARNING: Ignoring reaction 'rxn00003_c0' since it already exists.\n",
      "1723601868.0778613 INFO: cpd05178 not found in model!\n",
      "1723601868.0802882 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601868.2167084 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601868.348887 INFO: cpd05178 not found in model!\n",
      "1723601868.3512042 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601868.4061232 WARNING: Ignoring reaction 'rxn10052_c0' since it already exists.\n",
      "1723601868.7121153 WARNING: Ignoring reaction 'rxn01575_c0' since it already exists.\n",
      "1723601868.714699 WARNING: Ignoring reaction 'rxn00806_c0' since it already exists.\n",
      "1723601868.7371812 WARNING: Ignoring reaction 'rxn00272_c0' since it already exists.\n",
      "1723601869.033888 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723601869.1352186 WARNING: Ignoring reaction 'rxn00903_c0' since it already exists.\n",
      "1723601869.239227 WARNING: Ignoring reaction 'rxn04794_c0' since it already exists.\n",
      "1723601869.2419965 WARNING: Ignoring reaction 'rxn00435_c0' since it already exists.\n",
      "1723601869.285462 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723601869.2879417 WARNING: Ignoring reaction 'rxn00784_c0' since it already exists.\n",
      "1723601869.3001177 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601869.3405862 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723601869.4247963 WARNING: Ignoring reaction 'rxn00003_c0' since it already exists.\n",
      "1723601869.4729424 WARNING: Ignoring reaction 'rxn09180_c0' since it already exists.\n",
      "1723601869.5663216 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723601869.5886228 WARNING: Ignoring reaction 'rxn00762_c0' since it already exists.\n",
      "1723601869.7636373 INFO: metabolites 1022\n",
      "1723601869.8838384 WARNING: Ignoring reaction 'rxn00364_c0' since it already exists.\n",
      "1723601869.9404497 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723601869.974268 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601869.9991047 WARNING: Ignoring reaction 'rxn00018_c0' since it already exists.\n",
      "1723601870.0557585 WARNING: Ignoring reaction 'rxn00251_c0' since it already exists.\n",
      "1723601870.3182662 WARNING: Ignoring reaction 'rxn00290_c0' since it already exists.\n",
      "1723601870.3209171 WARNING: Ignoring reaction 'rxn00991_c0' since it already exists.\n",
      "1723601870.6417568 WARNING: Ignoring reaction 'rxn00467_c0' since it already exists.\n",
      "1723601870.6968253 WARNING: Ignoring reaction 'rxn00191_c0' since it already exists.\n",
      "1723601870.7195961 WARNING: Ignoring reaction 'rxn00260_c0' since it already exists.\n",
      "1723601870.7842412 INFO: metabolites 1000\n",
      "1723601870.9805377 WARNING: Ignoring reaction 'rxn11537_c0' since it already exists.\n",
      "1723601870.9833717 WARNING: Ignoring reaction 'rxn10124_c0' since it already exists.\n",
      "1723601870.9851093 WARNING: Ignoring reaction 'rxn11535_c0' since it already exists.\n",
      "1723601870.9868214 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723601871.2507265 WARNING: Ignoring reaction 'rxn00612_c0' since it already exists.\n",
      "1723601871.3033285 WARNING: Ignoring reaction 'rxn01575_c0' since it already exists.\n",
      "1723601871.3058562 WARNING: Ignoring reaction 'rxn00806_c0' since it already exists.\n",
      "1723601871.329911 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723601871.3483832 INFO: metabolites 1107\n",
      "1723601871.5411558 WARNING: Ignoring reaction 'rxn00527_c0' since it already exists.\n",
      "1723601871.552001 WARNING: Ignoring reaction 'rxn00931_c0' since it already exists.\n",
      "1723601871.561346 WARNING: Ignoring reaction 'rxn00350_c0' since it already exists.\n",
      "1723601871.8537889 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601872.1336124 INFO: cpd05178 not found in model!\n",
      "1723601872.136646 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601872.3819447 WARNING: Ignoring reaction 'rxn02933_c0' since it already exists.\n",
      "1723601872.451999 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723601872.8187237 WARNING: Ignoring reaction 'rxn00692_c0' since it already exists.\n",
      "1723601872.9238744 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723601873.1591284 INFO: cpd05178 not found in model!\n",
      "1723601873.1614685 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601873.303057 INFO: metabolites 959\n",
      "1723601873.4828238 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723601873.7791417 WARNING: Ignoring reaction 'rxn00251_c0' since it already exists.\n",
      "1723601873.896354 WARNING: Ignoring reaction 'rxn00347_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.27.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.4243807169721465}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601874.2605288 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601874.2630827 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601874.329403 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723601874.747508 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723601874.9584627 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601875.3487287 WARNING: Ignoring reaction 'rxn04954_c0' since it already exists.\n",
      "1723601875.5728798 WARNING: Ignoring reaction 'rxn02123_c0' since it already exists.\n",
      "1723601876.0593395 WARNING: Ignoring reaction 'rxn00877_c0' since it already exists.\n",
      "1723601876.0619757 WARNING: Ignoring reaction 'rxn00899_c0' since it already exists.\n",
      "1723601876.0744312 WARNING: Ignoring reaction 'rxn03798_c0' since it already exists.\n",
      "1723601876.0898545 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601877.2930973 INFO: metabolites 1081\n",
      "1723601877.6378021 INFO: metabolites 961\n",
      "1723601881.2298808 INFO: cpd05178 not found in model!\n",
      "1723601881.2328362 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601881.24376 INFO: cpd05178 not found in model!\n",
      "1723601881.2458646 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601881.5523622 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723601881.5646756 INFO: cpd05178 not found in model!\n",
      "1723601881.5660686 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723601881.566873 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601881.6752563 WARNING: Ignoring reaction 'rxn00796_c0' since it already exists.\n",
      "1723601881.7808678 WARNING: Ignoring reaction 'rxn01333_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.27.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7075888179370076}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601882.6755006 WARNING: Ignoring reaction 'rxn00747_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.2858584111080245}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601882.8623917 WARNING: Ignoring reaction 'rxn00625_c0' since it already exists.\n",
      "1723601882.902705 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601882.9052894 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601883.1615348 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601883.164272 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.3366167846423305}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601883.5921052 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601884.1011744 WARNING: Ignoring reaction 'rxn13143_c0' since it already exists.\n",
      "1723601885.0787356 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601885.0861537 INFO: metabolites 1095\n",
      "1723601885.9318123 WARNING: Ignoring reaction 'rxn04464_c0' since it already exists.\n",
      "1723601886.249067 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601886.490938 INFO: cpd05178 not found in model!\n",
      "1723601886.4931056 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601886.9583955 INFO: cpd05178 not found in model!\n",
      "1723601886.9609063 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601887.1485462 WARNING: Ignoring reaction 'rxn01301_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.25.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.8637269894207206}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601887.6772413 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723601888.1055744 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601888.1624658 INFO: metabolites 1049\n",
      "1723601888.8628242 WARNING: Ignoring reaction 'rxn00239_c0' since it already exists.\n",
      "1723601888.989145 INFO: reactions 528\n",
      "1723601889.0734296 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601889.1800933 INFO: cpd05178 not found in model!\n",
      "1723601889.1824627 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601889.2738647 INFO: Default biomass: [bio1]\n",
      "1723601889.3491118 INFO: cpd05178 not found in model!\n",
      "1723601889.3514166 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601889.4031513 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601889.4996204 INFO: metabolites 1183\n",
      "1723601890.2719803 INFO: metabolites 485\n",
      "1723601890.2754111 WARNING: Ignoring reaction 'rxn00501_c0' since it already exists.\n",
      "1723601890.3507507 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601890.8597097 WARNING: Ignoring reaction 'rxn00686_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.6613367038742844}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601891.08078 INFO: reactions 791\n",
      "1723601891.1707454 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601891.3201294 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601891.499561 INFO: Default biomass: [bio1]\n",
      "1723601892.0052767 INFO: metabolites 918\n",
      "1723601892.86122 INFO: metabolites 1105\n",
      "1723601892.9819615 INFO: metabolites 1115\n",
      "1723601896.3006387 INFO: cpd05178 not found in model!\n",
      "1723601896.3031063 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601897.0966158 INFO: cpd05178 not found in model!\n",
      "1723601897.0991826 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601897.334448 INFO: cpd05178 not found in model!\n",
      "1723601897.3366847 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601897.3479755 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601898.172631 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.30.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.3600642120282014}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601898.870532 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601899.3419452 INFO: cpd05178 not found in model!\n",
      "1723601899.3446841 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601899.71301 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601899.7677598 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601899.7750733 INFO: cpd05178 not found in model!\n",
      "1723601899.777198 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601899.7793846 INFO: metabolites 1183\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_concoct_out.79.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.6933986924076045}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601900.481515 INFO: metabolites 1181\n",
      "1723601900.6700487 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723601900.7298992 INFO: cpd05178 not found in model!\n",
      "1723601900.732357 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601900.8226745 WARNING: Ignoring reaction 'rxn00501_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.57.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.44357251908743434}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601900.9294195 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723601901.071404 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "Exception in thread Thread-10 (_handle_results):\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.11/threading.py\", line 1038, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/local/lib/python3.11/threading.py\", line 975, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/opt/env/python3_modelseed/lib/python3.11/site-packages/multiprocess/pool.py\", line 579, in _handle_results\n",
      "    task = get()\n",
      "           ^^^^^\n",
      "  File \"/opt/env/python3_modelseed/lib/python3.11/site-packages/multiprocess/connection.py\", line 254, in recv\n",
      "    return _ForkingPickler.loads(buf.getbuffer())\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/env/python3_modelseed/lib/python3.11/site-packages/dill/_dill.py\", line 303, in loads\n",
      "    return load(file, ignore, **kwds)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/env/python3_modelseed/lib/python3.11/site-packages/dill/_dill.py\", line 289, in load\n",
      "    return Unpickler(file, ignore=ignore, **kwds).load()\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/opt/env/python3_modelseed/lib/python3.11/site-packages/dill/_dill.py\", line 444, in load\n",
      "    obj = StockUnpickler.load(self)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "TypeError: ServerError.__init__() missing 2 required positional arguments: 'code' and 'message'\n",
      "1723601901.1555893 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601901.788207 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723601902.0017283 INFO: cpd05178 not found in model!\n",
      "1723601902.0044224 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.29.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 0.7871850239668896}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601902.7614377 INFO: metabolites 1119\n",
      "1723601902.89126 INFO: metabolites 1094\n",
      "1723601902.994862 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601903.117794 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601904.6329253 INFO: metabolites 1154\n",
      "1723601905.0048397 INFO: cpd05178 not found in model!\n",
      "1723601905.0074203 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601905.0197332 INFO: metabolites 1189\n",
      "1723601905.366975 INFO: cpd05178 not found in model!\n",
      "1723601905.3695986 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601905.7583454 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723601905.8374894 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.45.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.6526012056949768, '>': 1.584286461775102}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601906.2004006 WARNING: Ignoring reaction 'rxn00931_c0' since it already exists.\n",
      "1723601906.287904 INFO: cpd05178 not found in model!\n",
      "1723601906.2910502 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601906.6857142 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723601906.855393 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723601906.9218109 WARNING: Ignoring reaction 'rxn00931_c0' since it already exists.\n",
      "1723601907.413284 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601907.7807276 WARNING: Ignoring reaction 'rxn02187_c0' since it already exists.\n",
      "1723601908.1605406 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601908.1724043 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601908.2753758 INFO: reactions 1123\n",
      "1723601908.4179692 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601908.506905 INFO: cpd05178 not found in model!\n",
      "1723601908.5092666 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601908.9495718 INFO: Default biomass: [bio1]\n",
      "1723601908.9941878 INFO: metabolites 962\n",
      "1723601909.788567 INFO: metabolites 766\n",
      "1723601909.995422 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601910.1790328 INFO: metabolites 1201\n",
      "1723601911.2168078 INFO: cpd05178 not found in model!\n",
      "1723601911.2196267 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601911.588153 INFO: metabolites 1069\n",
      "1723601911.8424728 INFO: reactions 447\n",
      "1723601911.9452384 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601912.142166 INFO: Default biomass: [bio1]\n",
      "1723601912.8862677 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601913.4291427 INFO: cpd05178 not found in model!\n",
      "1723601913.431633 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601913.7316175 INFO: cpd05178 not found in model!\n",
      "1723601913.7338963 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601914.5937848 INFO: metabolites 1149\n",
      "1723601914.7088344 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601914.711755 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601915.1071565 INFO: cpd05178 not found in model!\n",
      "1723601915.110075 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.3937815548162572}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601915.4223301 INFO: cpd05178 not found in model!\n",
      "1723601915.4249053 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.5247919529915921}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601915.760666 INFO: cpd05178 not found in model!\n",
      "1723601915.763163 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601915.7973344 WARNING: Ignoring reaction 'rxn02229_c0' since it already exists.\n",
      "1723601915.9119146 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601916.521901 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723601916.6722 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601916.8447235 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601916.8735206 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601916.8766344 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601917.027154 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723601917.0305796 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723601917.0805898 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723601917.185373 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723601917.2569206 WARNING: Ignoring reaction 'rxn01740_c0' since it already exists.\n",
      "1723601917.3199294 WARNING: Ignoring reaction 'rxn09149_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.10.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7121850955379195}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601917.6419616 WARNING: Ignoring reaction 'rxn01548_c0' since it already exists.\n",
      "1723601917.671363 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601917.8623996 WARNING: Ignoring reaction 'rxn09069_c0' since it already exists.\n",
      "1723601918.4753854 INFO: metabolites 1055\n",
      "1723601918.5106852 INFO: cpd05178 not found in model!\n",
      "1723601918.5128894 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601919.2239304 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723601919.6493223 INFO: metabolites 1107\n",
      "1723601919.6661355 INFO: metabolites 975\n",
      "1723601919.929342 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601920.2042506 INFO: metabolites 1029\n",
      "1723601920.3902872 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601920.7990327 INFO: cpd05178 not found in model!\n",
      "1723601920.801645 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601921.078425 INFO: cpd05178 not found in model!\n",
      "1723601921.081031 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601921.0916107 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601921.277918 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723601921.4100747 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723601921.7080305 WARNING: Ignoring reaction 'rxn00931_c0' since it already exists.\n",
      "1723601921.9275892 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601921.9307983 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601922.055309 WARNING: Ignoring reaction 'rxn02187_c0' since it already exists.\n",
      "1723601922.0834184 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601922.0859618 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601922.108182 INFO: cpd05178 not found in model!\n",
      "1723601922.1109943 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.10.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.29320988327036324}\n",
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7737385631547831}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601922.3634224 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601922.591767 INFO: metabolites 950\n",
      "1723601922.6427734 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601922.672449 INFO: metabolites 1120\n",
      "1723601922.6806972 INFO: cpd05178 not found in model!\n",
      "1723601922.683434 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601922.9704504 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723601923.0039287 INFO: cpd05178 not found in model!\n",
      "1723601923.0065672 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601923.2654335 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601923.3250675 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723601923.4181902 WARNING: Ignoring reaction 'rxn00804_c0' since it already exists.\n",
      "1723601923.553921 WARNING: Ignoring reaction 'rxn00251_c0' since it already exists.\n",
      "1723601923.7133744 WARNING: Ignoring reaction 'rxn00435_c0' since it already exists.\n",
      "1723601923.7403743 WARNING: Ignoring reaction 'rxn01116_c0' since it already exists.\n",
      "1723601923.7604938 INFO: cpd05178 not found in model!\n",
      "1723601923.7633018 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601923.8970623 WARNING: Ignoring reaction 'rxn05625_c0' since it already exists.\n",
      "1723601923.9188094 WARNING: Ignoring reaction 'rxn00931_c0' since it already exists.\n",
      "1723601923.9655654 WARNING: Ignoring reaction 'rxn01354_c0' since it already exists.\n",
      "1723601923.9854872 INFO: metabolites 1177\n",
      "1723601924.2059379 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601924.209049 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601924.2659528 INFO: metabolites 1116\n",
      "1723601924.2684145 WARNING: Ignoring reaction 'rxn05940_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_metabat.10.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.6698548696572034}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601924.3067873 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723601924.3091996 WARNING: Ignoring reaction 'rxn01799_c0' since it already exists.\n",
      "1723601924.315989 WARNING: Ignoring reaction 'rxn01333_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.58.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7071084157382563}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601924.3767693 WARNING: Ignoring reaction 'rxn02187_c0' since it already exists.\n",
      "1723601924.5182426 WARNING: Ignoring reaction 'rxn00762_c0' since it already exists.\n",
      "1723601924.5211535 WARNING: Ignoring reaction 'rxn00745_c0' since it already exists.\n",
      "1723601924.567741 WARNING: Ignoring reaction 'rxn01573_c0' since it already exists.\n",
      "1723601924.6040184 INFO: cpd05178 not found in model!\n",
      "1723601924.6064944 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601924.640398 WARNING: Ignoring reaction 'rxn00692_c0' since it already exists.\n",
      "1723601924.6737509 WARNING: Ignoring reaction 'rxn00677_c0' since it already exists.\n",
      "1723601924.6975317 WARNING: Ignoring reaction 'rxn00350_c0' since it already exists.\n",
      "1723601924.7716205 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723601924.8436103 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601925.1905255 INFO: metabolites 929\n",
      "1723601925.4745555 INFO: cpd05178 not found in model!\n",
      "1723601925.477529 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601925.4930058 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601925.6645103 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601925.6673787 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.41.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.1852148031476042}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601926.349231 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601926.3512158 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601926.4719489 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601926.8615777 INFO: metabolites 1087\n",
      "1723601926.9263546 INFO: cpd05178 not found in model!\n",
      "1723601926.9298398 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601927.113774 INFO: cpd05178 not found in model!\n",
      "1723601927.116604 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601928.105118 WARNING: Ignoring reaction 'rxn01048_c0' since it already exists.\n",
      "1723601928.1194255 WARNING: Ignoring reaction 'rxn02774_c0' since it already exists.\n",
      "1723601928.1902435 WARNING: Ignoring reaction 'rxn02056_c0' since it already exists.\n",
      "1723601928.193643 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723601928.3480165 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.4.contigs__.RAST "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601928.529933 INFO: metabolites 860\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missing reaction EX_cpd00557_c0 {'<': 1.539369592558088, '>': 0.873592546733377}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601928.6662614 INFO: metabolites 932\n",
      "1723601928.812085 INFO: metabolites 963\n",
      "1723601928.8937223 INFO: cpd05178 not found in model!\n",
      "1723601928.8961978 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601929.2458074 WARNING: Ignoring reaction 'rxn00102_c0' since it already exists.\n",
      "1723601929.90145 WARNING: Ignoring reaction 'rxn11732_c0' since it already exists.\n",
      "1723601929.904515 INFO: cpd05178 not found in model!\n",
      "1723601929.907357 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601930.011535 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601930.047739 INFO: metabolites 1038\n",
      "1723601930.160244 WARNING: Ignoring reaction 'rxn09128_c0' since it already exists.\n",
      "1723601930.3396308 WARNING: Ignoring reaction 'rxn02285_c0' since it already exists.\n",
      "1723601930.3761969 INFO: cpd05178 not found in model!\n",
      "1723601930.3787386 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601930.5304463 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.8456496435982999}\n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.014229936531787, '>': 0.40358650391814777}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601931.4431174 WARNING: Ignoring reaction 'rxn10124_c0' since it already exists.\n",
      "1723601931.685972 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723601931.8306026 INFO: metabolites 1188\n",
      "1723601931.8985682 WARNING: Ignoring reaction 'rxn09128_c0' since it already exists.\n",
      "1723601931.9311547 INFO: metabolites 1106\n",
      "1723601931.9560506 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723601931.9595733 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723601932.3246496 WARNING: Ignoring reaction 'rxn00747_c0' since it already exists.\n",
      "1723601933.0214682 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601933.679492 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601934.4306333 INFO: cpd05178 not found in model!\n",
      "1723601934.4332347 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601934.7630584 INFO: metabolites 1081\n",
      "1723601935.416729 INFO: metabolites 1096\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.40.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.3828854286393393}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601935.9742591 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601937.4118998 INFO: metabolites 924\n",
      "1723601938.9083512 INFO: reactions 614\n",
      "1723601938.9949317 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601939.2909486 INFO: Default biomass: [bio1]\n",
      "1723601941.0234516 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601948.569834 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601948.6770296 INFO: cpd05178 not found in model!\n",
      "1723601948.679251 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601950.9546485 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601953.0218618 INFO: metabolites 971\n",
      "1723601954.1950886 INFO: reactions 858\n",
      "1723601954.3043728 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601954.710542 INFO: Default biomass: [bio1]\n",
      "1723601959.098934 INFO: reactions 1244\n",
      "1723601959.3166106 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601959.9443295 INFO: Default biomass: [bio1]\n",
      "1723601963.7979302 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601966.2300725 INFO: cpd05178 not found in model!\n",
      "1723601966.2332826 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.6.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.0514318742489617, '>': 0.3802479368370724}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601969.4395702 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601969.4441988 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601971.0366058 INFO: metabolites 910\n",
      "1723601971.2174096 INFO: reactions 889\n",
      "1723601971.3422906 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601971.637284 INFO: cpd05178 not found in model!\n",
      "1723601971.6398098 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601971.719209 INFO: Default biomass: [bio1]\n",
      "1723601972.589397 WARNING: Ignoring reaction 'rxn00290_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.8169411803075945}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601973.030595 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601973.166628 INFO: cpd05178 not found in model!\n",
      "1723601973.16946 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601973.7874072 INFO: cpd05178 not found in model!\n",
      "1723601973.790161 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723601974.0900476 WARNING: Ignoring reaction 'rxn00084_c0' since it already exists.\n",
      "1723601974.1815543 WARNING: Ignoring reaction 'rxn13753_c0' since it already exists.\n",
      "1723601974.200653 WARNING: Ignoring reaction 'rxn00799_c0' since it already exists.\n",
      "1723601974.2038028 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723601974.2180111 WARNING: Ignoring reaction 'rxn00762_c0' since it already exists.\n",
      "1723601974.2261925 WARNING: Ignoring reaction 'rxn01045_c0' since it already exists.\n",
      "1723601974.2588825 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601974.3678343 WARNING: Ignoring reaction 'rxn00191_c0' since it already exists.\n",
      "1723601974.4183645 WARNING: Ignoring reaction 'rxn01057_c0' since it already exists.\n",
      "1723601974.4438534 WARNING: Ignoring reaction 'rxn12215_c0' since it already exists.\n",
      "1723601974.5608478 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601974.7165904 INFO: metabolites 915\n",
      "1723601976.3911843 INFO: metabolites 1005\n",
      "1723601976.6510038 INFO: metabolites 892\n",
      "1723601977.1140282 INFO: cpd05178 not found in model!\n",
      "1723601977.1162875 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.2526552836953457, '<': 1.2593839497057884}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723601978.0991924 INFO: reactions 1074\n",
      "1723601978.3056579 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601978.7856085 INFO: Default biomass: [bio1]\n",
      "1723601979.9090905 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601980.089402 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723601983.4040518 INFO: reactions 1077\n",
      "1723601983.571584 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601983.6364052 INFO: metabolites 1185\n",
      "1723601984.02026 INFO: Default biomass: [bio1]\n",
      "1723601986.9448764 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601988.6342356 INFO: reactions 949\n",
      "1723601988.7503443 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601989.1595612 INFO: Default biomass: [bio1]\n",
      "1723601992.9106798 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601993.31424 INFO: reactions 1074\n",
      "1723601993.5320623 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601994.087611 INFO: Default biomass: [bio1]\n",
      "1723601997.554954 INFO: reactions 968\n",
      "1723601997.6910102 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723601998.2100935 INFO: Default biomass: [bio1]\n",
      "1723601998.4927258 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602004.7865849 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602005.888603 INFO: reactions 924\n",
      "1723602006.0097065 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602006.4866555 INFO: Default biomass: [bio1]\n",
      "1723602008.0144804 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602010.7268975 INFO: reactions 916\n",
      "1723602010.8445477 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602011.2964199 INFO: Default biomass: [bio1]\n",
      "1723602016.4206893 INFO: reactions 862\n",
      "1723602016.536285 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602016.6530657 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602016.9714878 INFO: Default biomass: [bio1]\n",
      "1723602021.6593738 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602026.3152125 INFO: reactions 1269\n",
      "1723602026.5287564 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602027.1923683 INFO: Default biomass: [bio1]\n",
      "1723602027.4232306 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602031.031049 INFO: cpd05178 not found in model!\n",
      "1723602031.0338025 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602031.0480812 INFO: reactions 1124\n",
      "1723602031.204058 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602031.8008761 INFO: Default biomass: [bio1]\n",
      "1723602032.5462344 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723602032.7452452 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602032.9352756 WARNING: Ignoring reaction 'rxn01057_c0' since it already exists.\n",
      "1723602033.1951315 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602034.9030895 INFO: metabolites 1262\n",
      "1723602037.2208304 INFO: reactions 874\n",
      "1723602037.3335679 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602037.4766412 INFO: reactions 1187\n",
      "1723602037.6318896 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602037.7439516 INFO: Default biomass: [bio1]\n",
      "1723602037.9557116 INFO: reactions 1054\n",
      "1723602038.1092954 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602038.210766 INFO: Default biomass: [bio1]\n",
      "1723602038.6437376 INFO: Default biomass: [bio1]\n",
      "1723602038.6999216 INFO: cpd05178 not found in model!\n",
      "1723602038.703375 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602039.278794 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723602039.4984746 WARNING: Ignoring reaction 'rxn00781_c0' since it already exists.\n",
      "1723602039.6630688 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602039.665977 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602039.6679974 WARNING: Ignoring reaction 'rxn01048_c0' since it already exists.\n",
      "1723602039.7288785 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723602039.731693 WARNING: Ignoring reaction 'rxn00784_c0' since it already exists.\n",
      "1723602039.8110313 WARNING: Ignoring reaction 'rxn01799_c0' since it already exists.\n",
      "1723602039.8595924 WARNING: Ignoring reaction 'rxn00467_c0' since it already exists.\n",
      "1723602040.1389832 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602040.1420774 WARNING: Ignoring reaction 'rxn00409_c0' since it already exists.\n",
      "1723602040.3017251 WARNING: Ignoring reaction 'rxn00949_c0' since it already exists.\n",
      "1723602040.8054242 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602040.8124986 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602042.0604882 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602042.4138403 INFO: metabolites 1166\n",
      "1723602042.5331297 INFO: reactions 837\n",
      "1723602042.6405463 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602043.03782 INFO: Default biomass: [bio1]\n",
      "1723602044.9660673 INFO: reactions 1051\n",
      "1723602045.102524 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602045.6385388 INFO: Default biomass: [bio1]\n",
      "1723602047.5127456 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602047.5965574 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602047.7978375 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602051.267878 INFO: reactions 1030\n",
      "1723602051.3978405 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602051.5286355 INFO: reactions 865\n",
      "1723602051.6372519 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602051.8493993 INFO: Default biomass: [bio1]\n",
      "1723602052.0184512 INFO: Default biomass: [bio1]\n",
      "1723602052.8303819 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602054.6855567 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602060.0546124 INFO: reactions 1159\n",
      "1723602060.264743 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602060.6304758 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602060.673045 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602060.79778 INFO: Default biomass: [bio1]\n",
      "1723602062.378418 INFO: reactions 709\n",
      "1723602062.399048 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602062.4724667 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602062.812343 INFO: Default biomass: [bio1]\n",
      "1723602064.674566 INFO: reactions 1257\n",
      "1723602064.8314333 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602065.4196756 INFO: Default biomass: [bio1]\n",
      "1723602069.9619076 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602071.6356828 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602074.0249243 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602076.0915523 INFO: reactions 1047\n",
      "1723602076.2255683 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602076.6645234 INFO: Default biomass: [bio1]\n",
      "1723602078.0314867 INFO: reactions 997\n",
      "1723602078.148253 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602078.1899707 INFO: reactions 1048\n",
      "1723602078.3107991 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602078.5637047 INFO: Default biomass: [bio1]\n",
      "1723602078.5717907 INFO: reactions 1105\n",
      "1723602078.739329 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602078.7463763 INFO: Default biomass: [bio1]\n",
      "1723602078.8776827 INFO: reactions 909\n",
      "1723602078.9997363 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602079.2185347 INFO: Default biomass: [bio1]\n",
      "1723602079.3806274 INFO: Default biomass: [bio1]\n",
      "1723602079.7274675 INFO: reactions 881\n",
      "1723602079.831037 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602080.2107334 INFO: Default biomass: [bio1]\n",
      "1723602080.5990088 INFO: reactions 960\n",
      "1723602080.7123153 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602081.1195066 INFO: Default biomass: [bio1]\n",
      "1723602084.9006424 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602087.3929834 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602087.6666675 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602087.7296643 INFO: reactions 877\n",
      "1723602087.8291538 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602088.1089923 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602088.1960406 INFO: Default biomass: [bio1]\n",
      "1723602088.3970883 INFO: reactions 1052\n",
      "1723602088.5683305 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602088.6239822 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602089.0194046 INFO: Default biomass: [bio1]\n",
      "1723602089.423434 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602093.87962 INFO: reactions 1220\n",
      "1723602094.0500343 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602094.4846087 INFO: reactions 770\n",
      "1723602094.573545 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602094.590034 INFO: Default biomass: [bio1]\n",
      "1723602094.9302177 INFO: Default biomass: [bio1]\n",
      "1723602097.1160817 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602099.9992132 INFO: reactions 1022\n",
      "1723602100.144698 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602100.5835567 INFO: Default biomass: [bio1]\n",
      "1723602101.3132365 INFO: cpd05178 not found in model!\n",
      "1723602101.3153896 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602101.34142 INFO: reactions 894\n",
      "1723602101.4510052 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602101.829088 INFO: Default biomass: [bio1]\n",
      "1723602102.8356347 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602102.97276 WARNING: Ignoring reaction 'rxn00548_c0' since it already exists.\n",
      "1723602102.9753556 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723602103.0079958 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602103.4167933 WARNING: Ignoring reaction 'rxn00499_c0' since it already exists.\n",
      "1723602103.4357252 WARNING: Ignoring reaction 'rxn00270_c0' since it already exists.\n",
      "1723602103.4729004 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602103.990589 INFO: reactions 852\n",
      "1723602104.1204667 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602104.4711123 INFO: Default biomass: [bio1]\n",
      "1723602105.416892 INFO: reactions 842\n",
      "1723602105.5153 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602105.8806112 INFO: Default biomass: [bio1]\n",
      "1723602106.2933686 INFO: metabolites 1078\n",
      "1723602106.584021 INFO: reactions 1235\n",
      "1723602106.762795 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602107.3754272 INFO: Default biomass: [bio1]\n",
      "1723602107.609293 INFO: reactions 1036\n",
      "1723602107.7402992 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602108.175636 INFO: Default biomass: [bio1]\n",
      "1723602108.6098394 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602109.935999 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602110.1255517 INFO: reactions 1059\n",
      "1723602110.2609234 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602110.544228 INFO: reactions 1085\n",
      "1723602110.679195 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602110.7077363 INFO: Default biomass: [bio1]\n",
      "1723602111.1102972 INFO: reactions 1160\n",
      "1723602111.145445 INFO: Default biomass: [bio1]\n",
      "1723602111.1948977 INFO: reactions 1060\n",
      "1723602111.3134594 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602111.3160517 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602111.5353308 INFO: reactions 1050\n",
      "1723602111.662721 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602111.75054 INFO: reactions 945\n",
      "1723602111.7578294 INFO: Default biomass: [bio1]\n",
      "1723602111.8475108 INFO: Default biomass: [bio1]\n",
      "1723602111.8707995 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602112.1295826 INFO: Default biomass: [bio1]\n",
      "1723602112.2513056 INFO: Default biomass: [bio1]\n",
      "1723602113.1199403 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602113.280924 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602113.699862 INFO: reactions 999\n",
      "1723602113.8121262 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602114.2335615 INFO: Default biomass: [bio1]\n",
      "1723602114.494956 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602114.58175 INFO: reactions 940\n",
      "1723602114.691174 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602115.0895689 INFO: Default biomass: [bio1]\n",
      "1723602115.1982677 INFO: reactions 810\n",
      "1723602115.2961626 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602115.6574616 INFO: Default biomass: [bio1]\n",
      "1723602115.664431 INFO: reactions 1029\n",
      "1723602115.7821906 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602115.7887335 INFO: reactions 1044\n",
      "1723602115.9114819 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602116.0645492 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602116.2097921 INFO: Default biomass: [bio1]\n",
      "1723602116.3515632 INFO: Default biomass: [bio1]\n",
      "1723602116.92842 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602119.0305998 INFO: reactions 1102\n",
      "1723602119.1730864 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602119.636442 INFO: Default biomass: [bio1]\n",
      "1723602119.9291093 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602120.258676 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602120.3721044 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602120.8879268 INFO: reactions 885\n",
      "1723602120.9105234 INFO: reactions 874\n",
      "1723602121.00597 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602121.0124762 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602121.055081 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602121.1360126 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602121.1468077 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602121.1768234 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602121.2707925 INFO: reactions 916\n",
      "1723602121.3737931 INFO: Default biomass: [bio1]\n",
      "1723602121.3749635 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602121.3754778 INFO: Default biomass: [bio1]\n",
      "1723602121.7499042 INFO: Default biomass: [bio1]\n",
      "1723602122.2660465 INFO: cpd05178 not found in model!\n",
      "1723602122.2683847 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602122.561667 INFO: cpd05178 not found in model!\n",
      "1723602122.5638392 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602122.9128463 INFO: reactions 1091\n",
      "1723602123.0366926 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602123.2110388 INFO: cpd05178 not found in model!\n",
      "1723602123.2130668 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.4298350607090864}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602123.383969 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602123.4985597 INFO: Default biomass: [bio1]\n",
      "1723602123.6355984 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_metabat.7.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7020559015769352}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602123.9660995 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602124.1447496 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602124.2715054 WARNING: Ignoring reaction 'rxn01361_c0' since it already exists.\n",
      "1723602124.45352 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.8812321840473076}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602124.548076 WARNING: Ignoring reaction 'rxn00558_c0' since it already exists.\n",
      "1723602124.601409 INFO: reactions 973\n",
      "1723602124.6291642 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602124.7216578 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602124.861454 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602125.0148804 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602125.1245344 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602125.1316547 INFO: Default biomass: [bio1]\n",
      "1723602125.4756134 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602125.703903 INFO: reactions 1233\n",
      "1723602125.8155358 INFO: metabolites 1153\n",
      "1723602125.8893127 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602126.058781 INFO: metabolites 937\n",
      "1723602126.338944 INFO: metabolites 1104\n",
      "1723602126.4174762 INFO: Default biomass: [bio1]\n",
      "1723602128.2456677 INFO: reactions 828\n",
      "1723602128.339757 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602128.5270934 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602128.6738896 INFO: Default biomass: [bio1]\n",
      "1723602129.4451828 INFO: reactions 1169\n",
      "1723602129.5863857 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602130.0737052 INFO: Default biomass: [bio1]\n",
      "1723602130.0907295 INFO: reactions 978\n",
      "1723602130.2169037 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602130.2691534 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602130.4565034 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602130.6140466 INFO: Default biomass: [bio1]\n",
      "1723602130.6256263 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602131.258146 INFO: reactions 1075\n",
      "1723602131.378793 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602131.8186934 INFO: Default biomass: [bio1]\n",
      "1723602132.2737968 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602134.16343 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602134.5485373 INFO: reactions 1058\n",
      "1723602134.6709704 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602134.9505174 INFO: reactions 1163\n",
      "1723602135.063445 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602135.0863838 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602135.1138399 INFO: Default biomass: [bio1]\n",
      "1723602135.5698755 INFO: Default biomass: [bio1]\n",
      "1723602136.3188808 INFO: cpd05178 not found in model!\n",
      "1723602136.3210516 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.62.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.1725211507081073}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602136.8689017 INFO: reactions 1025\n",
      "1723602136.9848218 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602137.0059557 INFO: reactions 1136\n",
      "1723602137.1354287 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602137.1486652 WARNING: Ignoring reaction 'rxn00102_c0' since it already exists.\n",
      "1723602137.2796595 WARNING: Ignoring reaction 'rxn01333_c0' since it already exists.\n",
      "1723602137.352297 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602137.4183307 INFO: Default biomass: [bio1]\n",
      "1723602137.566888 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602137.5694277 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602137.617313 INFO: Default biomass: [bio1]\n",
      "1723602137.885172 INFO: reactions 1182\n",
      "1723602137.9360082 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602138.0637112 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602138.441638 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602138.5624287 INFO: Default biomass: [bio1]\n",
      "1723602138.6155531 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602138.9149346 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602139.9840832 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602140.0435784 INFO: metabolites 792\n",
      "1723602142.1958404 INFO: cpd05178 not found in model!\n",
      "1723602142.2011845 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602142.848953 WARNING: Ignoring reaction 'rxn01348_c0' since it already exists.\n",
      "1723602143.566905 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602143.6865022 INFO: reactions 1205\n",
      "1723602143.807114 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.52.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.057497590672123}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602143.8715978 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602143.8808439 WARNING: Ignoring reaction 'rxn00278_c0' since it already exists.\n",
      "1723602144.2742534 WARNING: Ignoring reaction 'rxn01116_c0' since it already exists.\n",
      "1723602144.377193 WARNING: Ignoring reaction 'rxn00359_c0' since it already exists.\n",
      "1723602144.3909478 INFO: Default biomass: [bio1]\n",
      "1723602144.976677 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602145.2814827 INFO: cpd05178 not found in model!\n",
      "1723602145.2838821 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.3355213201044668}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602145.7692714 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602146.0059204 INFO: reactions 1178\n",
      "1723602146.1445434 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602146.1653836 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602146.241985 WARNING: Ignoring reaction 'rxn01746_c0' since it already exists.\n",
      "1723602146.244595 WARNING: Ignoring reaction 'rxn01748_c0' since it already exists.\n",
      "1723602146.3910954 WARNING: Ignoring reaction 'rxn00422_c0' since it already exists.\n",
      "1723602146.5303316 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602146.5408304 INFO: metabolites 1010\n",
      "1723602146.6383426 INFO: Default biomass: [bio1]\n",
      "1723602146.8715482 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602148.8505862 INFO: metabolites 905\n",
      "1723602150.347495 INFO: cpd05178 not found in model!\n",
      "1723602150.3496425 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602150.518892 INFO: cpd05178 not found in model!\n",
      "1723602150.521196 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602151.4893816 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602151.4947438 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602152.5567434 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602152.7121332 INFO: cpd05178 not found in model!\n",
      "1723602152.7144868 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602152.8671653 INFO: reactions 1183\n",
      "1723602153.0163517 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602153.2157888 INFO: metabolites 805\n",
      "1723602153.217659 INFO: metabolites 1133\n",
      "1723602153.5423636 INFO: Default biomass: [bio1]\n",
      "1723602153.6330068 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602153.7113643 WARNING: Ignoring reaction 'rxn00270_c0' since it already exists.\n",
      "1723602153.877776 WARNING: Ignoring reaction 'rxn00548_c0' since it already exists.\n",
      "1723602153.8804433 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723602154.0125988 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602154.7055142 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602155.3751626 INFO: metabolites 1087\n",
      "1723602155.7186906 INFO: cpd05178 not found in model!\n",
      "1723602155.72079 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602156.4805775 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723602156.9234061 INFO: cpd05178 not found in model!\n",
      "1723602156.9256904 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602157.0978258 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n",
      "1723602157.1673715 WARNING: Ignoring reaction 'rxn02933_c0' since it already exists.\n",
      "1723602157.2356794 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602157.44424 WARNING: Ignoring reaction 'rxn00747_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.50.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.5437956561837938}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602158.1085463 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602158.2831671 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.45.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.9796051856965199}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602158.6118193 WARNING: Ignoring reaction 'rxn00503_c0' since it already exists.\n",
      "1723602158.8411298 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602159.6957455 WARNING: Ignoring reaction 'rxn00145_c0' since it already exists.\n",
      "1723602160.5430303 WARNING: Ignoring reaction 'rxn00350_c0' since it already exists.\n",
      "1723602160.6589603 INFO: metabolites 1075\n",
      "1723602160.8913202 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602161.69578 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602162.5289326 INFO: metabolites 1011\n",
      "1723602170.7399414 INFO: cpd05178 not found in model!\n",
      "1723602170.7423444 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602171.5208225 WARNING: Ignoring reaction 'rxn04954_c0' since it already exists.\n",
      "1723602171.813278 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602171.815851 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602171.8176074 WARNING: Ignoring reaction 'rxn00131_c0' since it already exists.\n",
      "1723602171.879835 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723602171.9551685 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723602172.5746217 WARNING: Ignoring reaction 'rxn00558_c0' since it already exists.\n",
      "1723602172.6010988 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602172.6036103 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602172.9552155 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602174.4694438 INFO: metabolites 1011\n",
      "1723602177.1188426 INFO: cpd05178 not found in model!\n",
      "1723602177.1218731 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602177.939782 WARNING: Ignoring reaction 'rxn02229_c0' since it already exists.\n",
      "1723602178.2666876 INFO: reactions 1125\n",
      "1723602178.314389 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723602178.3931465 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602178.4927046 WARNING: Ignoring reaction 'rxn00270_c0' since it already exists.\n",
      "1723602178.9097762 INFO: Default biomass: [bio1]\n",
      "1723602179.384105 WARNING: Ignoring reaction 'rxn04954_c0' since it already exists.\n",
      "1723602179.3866603 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602179.4360278 WARNING: Ignoring reaction 'rxn00499_c0' since it already exists.\n",
      "1723602179.5479846 WARNING: Ignoring reaction 'rxn09231_c0' since it already exists.\n",
      "1723602179.8519228 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602180.6378953 INFO: cpd05178 not found in model!\n",
      "1723602180.6401336 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602181.6855578 INFO: metabolites 1231\n",
      "1723602182.0157838 WARNING: Ignoring reaction 'rxn00362_c0' since it already exists.\n",
      "1723602182.8669646 INFO: cpd05178 not found in model!\n",
      "1723602182.8697488 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602183.7355943 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602183.8940723 WARNING: Ignoring reaction 'rxn00023_c0' since it already exists.\n",
      "1723602184.0635042 WARNING: Ignoring reaction 'rxn09180_c0' since it already exists.\n",
      "1723602184.4657393 WARNING: Ignoring reaction 'rxn00350_c0' since it already exists.\n",
      "1723602184.6761112 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602185.5488296 INFO: metabolites 1125\n",
      "1723602186.3770056 INFO: metabolites 1244\n",
      "1723602187.213261 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602192.030487 INFO: cpd05178 not found in model!\n",
      "1723602192.0328043 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602192.2036002 INFO: cpd05178 not found in model!\n",
      "1723602192.2057834 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602192.6964977 WARNING: Ignoring reaction 'rxn00278_c0' since it already exists.\n",
      "1723602193.1497955 WARNING: Ignoring reaction 'rxn01333_c0' since it already exists.\n",
      "1723602193.1628006 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723602193.3576362 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602193.3601775 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602193.4357462 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.0840149028727704}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602193.5842426 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723602193.6791055 WARNING: Ignoring reaction 'rxn00499_c0' since it already exists.\n",
      "1723602193.6814826 WARNING: Ignoring reaction 'rxn00747_c0' since it already exists.\n",
      "1723602193.7736678 INFO: cpd05178 not found in model!\n",
      "1723602193.7748213 INFO: cpd05178 not found in model!\n",
      "1723602193.7758613 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602193.7767944 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602193.8253684 WARNING: Ignoring reaction 'rxn00605_c0' since it already exists.\n",
      "1723602194.0425851 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602194.0704315 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723602194.125321 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602194.4205623 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602194.944239 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602194.9473882 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602195.0401654 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602195.4291162 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602196.1144097 INFO: cpd05178 not found in model!\n",
      "1723602196.1169584 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602196.2013206 INFO: metabolites 989\n",
      "1723602196.4105797 INFO: reactions 1116\n",
      "1723602196.5919023 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602197.056341 INFO: Default biomass: [bio1]\n",
      "1723602197.4223235 INFO: metabolites 1096\n",
      "1723602197.4699717 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602197.719272 INFO: cpd05178 not found in model!\n",
      "1723602197.721853 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.39.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.572374108848189}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602198.0267012 INFO: cpd05178 not found in model!\n",
      "1723602198.029152 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602198.2347553 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602198.2697399 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.8579289103126573}"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602199.081133 INFO: metabolites 1075\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602199.1344435 INFO: cpd05178 not found in model!\n",
      "1723602199.1374643 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602199.1915836 INFO: metabolites 1142\n",
      "1723602199.439381 WARNING: Ignoring reaction 'rxn01418_c0' since it already exists.\n",
      "1723602199.7285078 WARNING: Ignoring reaction 'rxn00704_c0' since it already exists.\n",
      "1723602200.0974443 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.20.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.33066767070461417}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602200.9860783 WARNING: Ignoring reaction 'rxn00003_c0' since it already exists.\n",
      "1723602201.2195206 INFO: metabolites 880\n",
      "1723602201.5726779 INFO: metabolites 983\n",
      "1723602201.6432343 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602201.7962263 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602203.4591618 INFO: metabolites 1205\n",
      "1723602203.4858224 INFO: metabolites 1008\n",
      "1723602204.4501834 INFO: cpd05178 not found in model!\n",
      "1723602204.4523284 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602205.1745675 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602205.3724437 INFO: cpd05178 not found in model!\n",
      "1723602205.374573 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.3429509646147397}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602205.6230927 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723602205.925056 WARNING: Ignoring reaction 'rxn01048_c0' since it already exists.\n",
      "1723602205.9583774 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602206.037312 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602207.5680776 INFO: metabolites 1101\n",
      "1723602207.9354987 INFO: cpd05178 not found in model!\n",
      "1723602207.9380786 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602208.1489086 INFO: metabolites 1154\n",
      "1723602208.1786227 WARNING: Ignoring reaction 'rxn00278_c0' since it already exists.\n",
      "1723602208.5800555 INFO: cpd05178 not found in model!\n",
      "1723602208.5823 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602208.7968295 INFO: cpd05178 not found in model!\n",
      "1723602208.7991934 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602209.0233183 WARNING: Ignoring reaction 'rxn00806_c0' since it already exists.\n",
      "1723602209.100207 WARNING: Ignoring reaction 'rxn01575_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.37.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7550632047162401}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602209.3247366 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602209.5193167 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602209.9719296 WARNING: Ignoring reaction 'rxn00616_c0' since it already exists.\n",
      "1723602210.020756 WARNING: Ignoring reaction 'rxn00907_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_maxbin.001.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7002283036233271, '<': 1.0998394658084483}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602210.230107 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n",
      "1723602210.240875 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723602210.3648732 WARNING: Ignoring reaction 'rxn04954_c0' since it already exists.\n",
      "1723602210.368209 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602210.4646847 WARNING: Ignoring reaction 'rxn02774_c0' since it already exists.\n",
      "1723602210.500687 WARNING: Ignoring reaction 'rxn02056_c0' since it already exists.\n",
      "1723602210.5440187 WARNING: Ignoring reaction 'rxn00270_c0' since it already exists.\n",
      "1723602211.1148303 INFO: metabolites 1082\n",
      "1723602211.1412652 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602211.1912022 WARNING: Ignoring reaction 'rxn01057_c0' since it already exists.\n",
      "1723602211.334296 WARNING: Ignoring reaction 'rxn01333_c0' since it already exists.\n",
      "1723602211.336762 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723602211.426111 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602211.6243873 INFO: cpd05178 not found in model!\n",
      "1723602211.626789 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602212.3007143 WARNING: Ignoring reaction 'rxn01348_c0' since it already exists.\n",
      "1723602212.643339 INFO: metabolites 1054\n",
      "1723602212.9795537 INFO: metabolites 1005\n",
      "1723602213.0259442 WARNING: Ignoring reaction 'rxn01048_c0' since it already exists.\n",
      "1723602213.3210988 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602214.8346283 INFO: cpd05178 not found in model!\n",
      "1723602214.8374496 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602214.9801059 INFO: metabolites 1098\n",
      "1723602215.672252 WARNING: Ignoring reaction 'rxn01302_c0' since it already exists.\n",
      "1723602216.1266854 INFO: cpd05178 not found in model!\n",
      "1723602216.129051 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602216.2853158 WARNING: Ignoring reaction 'rxn02185_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33.contigs__.RAST Missing reaction DM_cpd03091_c0 {'>': 0.33688316289671955}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602217.1307664 INFO: reactions 1236\n",
      "1723602217.1895647 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602217.1929445 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602217.3018532 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602217.8477776 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723602217.8540423 INFO: Default biomass: [bio1]\n",
      "1723602218.0346851 INFO: cpd05178 not found in model!\n",
      "1723602218.0368161 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602218.1219957 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602218.2836635 INFO: cpd05178 not found in model!\n",
      "1723602218.28619 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602218.507677 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723602218.7015696 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602218.7040727 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.37.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7918977349137705}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602218.9314759 INFO: cpd05178 not found in model!\n",
      "1723602218.9336193 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602218.9406013 WARNING: Ignoring reaction 'rxn04464_c0' since it already exists.\n",
      "1723602218.9797087 WARNING: Ignoring reaction 'rxn02011_c0' since it already exists.\n",
      "1723602219.070054 INFO: metabolites 969\n",
      "1723602219.084582 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602219.1631932 WARNING: Ignoring reaction 'rxn00912_c0' since it already exists.\n",
      "1723602219.2182887 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602219.221035 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723602219.8091023 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602220.0307271 INFO: metabolites 878\n",
      "1723602220.1710896 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602220.1737382 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602220.3062544 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_concoct_out.44.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.2614580943803229}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602220.6061351 INFO: metabolites 897\n",
      "1723602220.9785001 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602221.6863933 INFO: metabolites 1003\n",
      "1723602222.4785485 INFO: metabolites 1102\n",
      "1723602223.5244653 INFO: cpd05178 not found in model!\n",
      "1723602223.5269647 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602223.7156723 INFO: cpd05178 not found in model!\n",
      "1723602223.717909 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602223.9490852 WARNING: Ignoring reaction 'rxn00692_c0' since it already exists.\n",
      "1723602224.2650611 INFO: cpd05178 not found in model!\n",
      "1723602224.2675674 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602224.6316075 WARNING: Ignoring reaction 'rxn11537_c0' since it already exists.\n",
      "1723602224.6341376 WARNING: Ignoring reaction 'rxn11535_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.3731647933427882}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602225.056239 INFO: reactions 684\n",
      "1723602225.1397278 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.36.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.8662522370972049}\n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.51.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.49223321097616074}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602225.4613335 INFO: reactions 853\n",
      "1723602225.4787529 INFO: Default biomass: [bio1]\n",
      "1723602225.5699127 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602225.6214545 WARNING: Ignoring reaction 'rxn01746_c0' since it already exists.\n",
      "1723602225.6241205 WARNING: Ignoring reaction 'rxn01748_c0' since it already exists.\n",
      "1723602225.7082038 INFO: cpd05178 not found in model!\n",
      "1723602225.7104325 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602225.843144 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602225.8613677 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602225.9310877 INFO: Default biomass: [bio1]\n",
      "1723602225.9648328 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602226.0375772 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602226.1759176 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602226.5366614 WARNING: Ignoring reaction 'rxn00251_c0' since it already exists.\n",
      "1723602226.7411716 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602226.8124218 WARNING: Ignoring reaction 'rxn08130_c0' since it already exists.\n",
      "1723602226.9236507 WARNING: Ignoring reaction 'rxn02751_c0' since it already exists.\n",
      "1723602226.9398756 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602227.1019192 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602227.2209506 INFO: cpd05178 not found in model!\n",
      "1723602227.2231114 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602227.4722958 INFO: metabolites 571\n",
      "1723602227.5251365 INFO: cpd05178 not found in model!\n",
      "1723602227.5278559 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602227.6181016 WARNING: Ignoring reaction 'rxn02774_c0' since it already exists.\n",
      "1723602227.709408 WARNING: Ignoring reaction 'rxn02056_c0' since it already exists.\n",
      "1723602227.8313825 INFO: metabolites 1131\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_concoct_out.51.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.0478087500652005, '>': 0.9937516963065104}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602228.091357 WARNING: Ignoring reaction 'rxn00272_c0' since it already exists.\n",
      "1723602228.1646843 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723602228.3094242 WARNING: Ignoring reaction 'rxn01361_c0' since it already exists.\n",
      "1723602228.3702807 INFO: metabolites 1115\n",
      "1723602228.4052846 INFO: cpd05178 not found in model!\n",
      "1723602228.4075356 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602228.6387897 WARNING: Ignoring reaction 'rxn09180_c0' since it already exists.\n",
      "1723602228.7135735 INFO: metabolites 1258\n",
      "1723602228.7224479 INFO: cpd05178 not found in model!\n",
      "1723602228.7245555 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602228.8496456 WARNING: Ignoring reaction 'rxn01301_c0' since it already exists.\n",
      "1723602228.8761022 INFO: cpd05178 not found in model!\n",
      "1723602228.8781435 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602228.956729 WARNING: Ignoring reaction 'rxn00102_c0' since it already exists.\n",
      "1723602229.0824995 WARNING: Ignoring reaction 'rxn00198_c0' since it already exists.\n",
      "1723602229.1175668 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723602229.1206422 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602229.210141 WARNING: Ignoring reaction 'rxn00931_c0' since it already exists.\n",
      "1723602229.2929435 WARNING: Ignoring reaction 'rxn00677_c0' since it already exists.\n",
      "1723602229.3084028 WARNING: Ignoring reaction 'rxn00520_c0' since it already exists.\n",
      "1723602229.5252209 WARNING: Ignoring reaction 'rxn00762_c0' since it already exists.\n",
      "1723602229.6337478 WARNING: Ignoring reaction 'rxn00747_c0' since it already exists.\n",
      "1723602229.6929848 WARNING: Ignoring reaction 'rxn01348_c0' since it already exists.\n",
      "1723602229.8567219 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602229.860199 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602229.9598062 WARNING: Ignoring reaction 'rxn00102_c0' since it already exists.\n",
      "1723602229.982724 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602229.9850051 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_concoct_out.19.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.0757541652427849}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602230.1621785 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602230.214359 WARNING: Ignoring reaction 'rxn01048_c0' since it already exists.\n",
      "1723602230.216789 WARNING: Ignoring reaction 'rxn01116_c0' since it already exists.\n",
      "1723602230.2625544 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602230.2763252 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602230.5177495 INFO: metabolites 1010\n",
      "1723602230.6487315 WARNING: Ignoring reaction 'rxn00467_c0' since it already exists.\n",
      "1723602230.6584275 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602231.1969984 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602231.7510614 INFO: metabolites 1150\n",
      "1723602231.8279355 INFO: metabolites 1204\n",
      "1723602232.1348846 INFO: metabolites 1088\n",
      "1723602232.770934 INFO: metabolites 706\n",
      "1723602233.3462553 INFO: reactions 1030\n",
      "1723602233.4675472 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602233.8876219 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602233.9546096 INFO: Default biomass: [bio1]\n",
      "1723602234.3148115 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602239.419184 INFO: cpd05178 not found in model!\n",
      "1723602239.4212117 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602239.6479282 INFO: cpd05178 not found in model!\n",
      "1723602239.650259 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602240.261822 INFO: cpd05178 not found in model!\n",
      "1723602240.264098 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_metabat.4.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.209584549768964}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602241.0925264 WARNING: Ignoring reaction 'rxn00278_c0' since it already exists.\n",
      "1723602241.6333687 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.143.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.49402072375926676}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602241.9706063 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.54.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.9750399696360269}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602242.137819 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602242.2150972 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602242.2729902 INFO: reactions 699\n",
      "1723602242.3122768 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602242.3541129 INFO: reactions 495\n",
      "1723602242.3870997 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602242.4645543 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602242.6771014 INFO: Default biomass: [bio1]\n",
      "1723602242.7687297 INFO: Default biomass: [bio1]\n",
      "1723602242.9787533 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602243.6362357 INFO: metabolites 901\n",
      "1723602244.1927354 INFO: metabolites 1203\n",
      "1723602244.2970157 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602244.7290864 INFO: metabolites 1226\n",
      "1723602245.4864666 INFO: cpd05178 not found in model!\n",
      "1723602245.488826 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602246.6401956 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602247.3301811 INFO: cpd05178 not found in model!\n",
      "1723602247.332892 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602248.5556989 INFO: metabolites 985\n",
      "1723602248.6226735 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602249.3797147 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602251.494912 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602251.835947 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602252.5064037 INFO: metabolites 546\n",
      "1723602254.7479494 INFO: reactions 1060\n",
      "1723602254.898546 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602255.3529723 INFO: Default biomass: [bio1]\n",
      "1723602262.006012 INFO: reactions 944\n",
      "1723602262.1586807 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602262.602358 INFO: Default biomass: [bio1]\n",
      "1723602262.93186 INFO: reactions 852\n",
      "1723602263.0448616 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602263.469859 INFO: Default biomass: [bio1]\n",
      "1723602263.7547023 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602266.0621896 INFO: cpd05178 not found in model!\n",
      "1723602266.0645328 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': -0.07331038920124686}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602266.4459226 INFO: cpd05178 not found in model!\n",
      "1723602266.448167 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602267.188479 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602267.3472204 INFO: cpd05178 not found in model!\n",
      "1723602267.3494012 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602267.5575173 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602267.560162 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602267.8498852 WARNING: Ignoring reaction 'rxn02056_c0' since it already exists.\n",
      "1723602268.1199086 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.95.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.3231943939685618, '>': 0.7464529914473524}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602268.2510324 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602268.7619784 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723602270.1588392 INFO: metabolites 946\n",
      "1723602270.21904 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602270.3663313 INFO: metabolites 1272\n",
      "1723602271.024199 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602271.243403 INFO: cpd05178 not found in model!\n",
      "1723602271.2456644 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602271.7040806 INFO: metabolites 958\n",
      "1723602271.8438518 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.69.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.36802568723269513}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602272.8598356 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602274.3620994 INFO: metabolites 933\n",
      "1723602275.3186986 INFO: cpd05178 not found in model!\n",
      "1723602275.320989 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602276.3954592 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.4864250511669643}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602277.2046483 INFO: cpd05178 not found in model!\n",
      "1723602277.2070096 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602277.355184 WARNING: Ignoring reaction 'rxn00427_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': -0.054197156549810235}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602278.1306114 WARNING: Ignoring reaction 'rxn10568_c0' since it already exists.\n",
      "1723602278.175497 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602279.089062 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602279.62633 INFO: reactions 987\n",
      "1723602279.7770722 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602280.2533672 INFO: Default biomass: [bio1]\n",
      "1723602280.9345734 INFO: cpd05178 not found in model!\n",
      "1723602280.936946 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602281.189713 INFO: cpd05178 not found in model!\n",
      "1723602281.192248 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602281.467168 INFO: metabolites 996\n",
      "1723602281.5272048 INFO: metabolites 996\n",
      "1723602281.9314446 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602282.0012972 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602282.0039485 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.18.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7625880607403235}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602282.455186 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602283.5295868 INFO: metabolites 1103\n",
      "1723602283.785923 INFO: reactions 1039\n",
      "1723602283.905434 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602284.0852823 INFO: reactions 991\n",
      "1723602284.2066286 INFO: metabolites 1181\n",
      "1723602284.2485173 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602284.3934352 INFO: Default biomass: [bio1]\n",
      "1723602284.5579689 INFO: cpd05178 not found in model!\n",
      "1723602284.5603583 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602284.6738837 INFO: Default biomass: [bio1]\n",
      "1723602285.9070892 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602287.8308027 INFO: cpd05178 not found in model!\n",
      "1723602287.833083 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602288.3275995 INFO: metabolites 907\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.18.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.0385740872549971}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602288.5000808 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602288.6421728 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602291.0894482 INFO: metabolites 1126\n",
      "1723602291.4797184 INFO: cpd05178 not found in model!\n",
      "1723602291.4822452 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602292.6268585 WARNING: Ignoring reaction 'rxn01790_c0' since it already exists.\n",
      "1723602292.7792523 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.10.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7830667514773885}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602292.9337285 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602292.9581492 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602293.2554426 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602294.5548046 INFO: reactions 1107\n",
      "1723602294.6844933 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602294.7750666 INFO: reactions 910\n",
      "1723602294.8581247 INFO: metabolites 1248\n",
      "1723602294.882632 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602295.203309 INFO: Default biomass: [bio1]\n",
      "1723602295.2957962 INFO: Default biomass: [bio1]\n",
      "1723602296.0206573 INFO: cpd05178 not found in model!\n",
      "1723602296.0230432 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.47.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.5871952055627838}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602297.1465485 INFO: cpd05178 not found in model!\n",
      "1723602297.1495147 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602297.5055783 WARNING: Ignoring reaction 'rxn00364_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.38.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.4558478726418611}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602298.7631786 WARNING: Ignoring reaction 'rxn03436_c0' since it already exists.\n",
      "1723602298.834776 WARNING: Ignoring reaction 'rxn03435_c0' since it already exists.\n",
      "1723602298.8940105 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602298.9957418 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602300.701041 INFO: metabolites 1006\n",
      "1723602300.8192909 INFO: metabolites 1056\n",
      "1723602301.4535012 INFO: cpd05178 not found in model!\n",
      "1723602301.4562192 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602301.8011885 INFO: reactions 743\n",
      "1723602301.8984928 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.37.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.9658203183979919}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602302.2501774 INFO: Default biomass: [bio1]\n",
      "1723602302.2683196 INFO: cpd05178 not found in model!\n",
      "1723602302.2710888 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.11.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.0517277039215087}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602303.662418 WARNING: Ignoring reaction 'rxn00686_c0' since it already exists.\n",
      "1723602303.7482548 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602304.120504 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602304.123816 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602304.25239 INFO: cpd05178 not found in model!\n",
      "1723602304.2549639 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602304.6005023 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602304.626528 WARNING: Ignoring reaction 'rxn00500_c0' since it already exists.\n",
      "1723602304.7906199 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602304.8708603 WARNING: Ignoring reaction 'rxn00935_c0' since it already exists.\n",
      "1723602304.986324 INFO: cpd05178 not found in model!\n",
      "1723602304.9886954 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602305.015277 WARNING: Ignoring reaction 'rxn00251_c0' since it already exists.\n",
      "1723602305.1055655 WARNING: Ignoring reaction 'rxn00611_c0' since it already exists.\n",
      "1723602305.2535465 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602305.3948882 INFO: metabolites 1153\n",
      "1723602306.3789494 INFO: metabolites 1098\n",
      "1723602306.4267824 INFO: cpd05178 not found in model!\n",
      "1723602306.4296834 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602306.6547985 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602306.777794 INFO: metabolites 1027\n",
      "1723602307.5609858 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602307.5644631 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602308.2840981 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602308.5620985 INFO: metabolites 1030\n",
      "1723602310.9899955 INFO: metabolites 814\n",
      "1723602311.5611596 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602313.6986136 INFO: reactions 468\n",
      "1723602313.8043935 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602314.0260217 INFO: Default biomass: [bio1]\n",
      "1723602314.0580056 INFO: cpd05178 not found in model!\n",
      "1723602314.0610359 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602314.465205 WARNING: Ignoring reaction 'rxn00931_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.38.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.1313579376241196}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602315.2824688 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723602315.9497128 INFO: cpd05178 not found in model!\n",
      "1723602315.9527419 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602316.1998682 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602317.2250967 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602317.2285466 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602317.3591733 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602318.754049 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602319.0801373 INFO: metabolites 1019\n",
      "1723602320.5066316 INFO: metabolites 1051\n",
      "1723602323.246778 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602328.38156 INFO: cpd05178 not found in model!\n",
      "1723602328.3846245 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602328.8206632 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.4.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.4487660284406674}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602330.2248178 WARNING: Ignoring reaction 'rxn00616_c0' since it already exists.\n",
      "1723602330.2887518 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602330.291889 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602330.9191148 INFO: cpd05178 not found in model!\n",
      "1723602330.9224944 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602331.0272288 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.55.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.0992664863328652}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602331.835143 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602332.6908982 INFO: metabolites 1133\n",
      "1723602333.4171784 INFO: metabolites 465\n",
      "1723602337.5880814 INFO: reactions 1056\n",
      "1723602337.7298641 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602338.2595813 INFO: Default biomass: [bio1]\n",
      "1723602339.0613234 INFO: cpd05178 not found in model!\n",
      "1723602339.06442 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602340.0586526 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723602340.3130393 WARNING: Ignoring reaction 'rxn01011_c0' since it already exists.\n",
      "1723602340.3424327 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602340.345197 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602340.5658276 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602340.8053741 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602341.067266 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602341.0815725 INFO: cpd05178 not found in model!\n",
      "1723602341.0840201 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602341.6343594 INFO: reactions 639\n",
      "1723602341.8131905 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602342.122407 INFO: Default biomass: [bio1]\n",
      "1723602343.4275193 INFO: metabolites 1143\n",
      "1723602344.2806206 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602345.696555 INFO: cpd05178 not found in model!\n",
      "1723602345.699186 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602346.1681764 INFO: metabolites 1178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_maxbin.022.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': -0.07078039891380723}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602346.761783 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602346.9132612 WARNING: Ignoring reaction 'rxn02168_c0' since it already exists.\n",
      "1723602347.7173235 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602348.134832 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602349.7416708 INFO: metabolites 1061\n",
      "1723602350.1227794 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602351.5139863 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602354.5522776 INFO: reactions 1046\n",
      "1723602354.6917217 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602355.2116094 INFO: Default biomass: [bio1]\n",
      "1723602356.2591898 INFO: reactions 939\n",
      "1723602356.403802 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602356.867358 INFO: Default biomass: [bio1]\n",
      "1723602356.98101 INFO: cpd05178 not found in model!\n",
      "1723602356.983768 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602357.284343 INFO: cpd05178 not found in model!\n",
      "1723602357.2872314 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602357.3102887 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723602357.3989296 WARNING: Ignoring reaction 'rxn01799_c0' since it already exists.\n",
      "1723602357.5930567 WARNING: Ignoring reaction 'rxn00278_c0' since it already exists.\n",
      "1723602357.6883292 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602357.6912656 WARNING: Ignoring reaction 'rxn01792_c0' since it already exists.\n",
      "1723602357.6941135 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602357.7407348 INFO: cpd05178 not found in model!\n",
      "1723602357.7430403 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.5447900979006954}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602358.0415761 WARNING: Ignoring reaction 'rxn00799_c0' since it already exists.\n",
      "1723602358.0441408 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723602358.056645 WARNING: Ignoring reaction 'rxn01045_c0' since it already exists.\n",
      "1723602358.1480336 WARNING: Ignoring reaction 'rxn00097_c0' since it already exists.\n",
      "1723602358.1507642 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602358.2304692 WARNING: Ignoring reaction 'rxn00275_c0' since it already exists.\n",
      "1723602358.2401857 WARNING: Ignoring reaction 'rxn05734_c0' since it already exists.\n",
      "1723602358.3187284 WARNING: Ignoring reaction 'rxn03798_c0' since it already exists.\n",
      "1723602358.3215744 WARNING: Ignoring reaction 'rxn02123_c0' since it already exists.\n",
      "1723602358.3237453 WARNING: Ignoring reaction 'rxn00668_c0' since it already exists.\n",
      "1723602358.4204423 WARNING: Ignoring reaction 'rxn01057_c0' since it already exists.\n",
      "1723602358.4732509 WARNING: Ignoring reaction 'rxn12215_c0' since it already exists.\n",
      "1723602358.636711 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602358.711276 WARNING: Ignoring reaction 'rxn00553_c0' since it already exists.\n",
      "1723602358.7473032 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602359.1442893 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602359.337257 INFO: reactions 961\n",
      "1723602359.4548452 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602359.9018364 INFO: Default biomass: [bio1]\n",
      "1723602360.5514853 INFO: metabolites 1035\n",
      "1723602360.6024425 INFO: metabolites 1238\n",
      "1723602360.7763362 INFO: metabolites 915\n",
      "1723602364.1025805 INFO: reactions 1012\n",
      "1723602364.2272704 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602364.5100968 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602364.695313 INFO: Default biomass: [bio1]\n",
      "1723602366.2120044 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602367.8330872 INFO: reactions 416\n",
      "1723602367.9087892 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602368.0953262 INFO: Default biomass: [bio1]\n",
      "1723602369.3351734 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602374.1269078 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602377.1197138 INFO: reactions 843\n",
      "1723602377.1931036 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602377.2716804 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602377.6707053 INFO: Default biomass: [bio1]\n",
      "1723602377.9716122 INFO: reactions 974\n",
      "1723602378.1119146 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602378.613922 INFO: Default biomass: [bio1]\n",
      "1723602384.4214778 INFO: reactions 834\n",
      "1723602384.5240653 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602384.9180806 INFO: Default biomass: [bio1]\n",
      "1723602386.7945662 INFO: reactions 914\n",
      "1723602386.8117502 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602386.9185512 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602387.3324814 INFO: Default biomass: [bio1]\n",
      "1723602387.3452754 INFO: reactions 958\n",
      "1723602387.4617615 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602387.5275085 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602387.8722217 INFO: Default biomass: [bio1]\n",
      "1723602388.4323878 INFO: reactions 1212\n",
      "1723602388.6612525 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602388.9815862 INFO: reactions 955\n",
      "1723602389.0999162 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602389.2642589 INFO: Default biomass: [bio1]\n",
      "1723602389.5397077 INFO: Default biomass: [bio1]\n",
      "1723602391.2054048 INFO: cpd05178 not found in model!\n",
      "1723602391.2080498 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602391.3964496 INFO: reactions 975\n",
      "1723602391.5112257 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602391.937124 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.608184096766703}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602392.80928 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602394.1774437 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602394.681799 INFO: metabolites 1067\n",
      "1723602395.665527 INFO: reactions 1104\n",
      "1723602395.8064952 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602396.3013465 INFO: Default biomass: [bio1]\n",
      "1723602396.4096458 INFO: cpd05178 not found in model!\n",
      "1723602396.4127042 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602396.650799 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602396.6762252 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602397.7224972 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602398.3112993 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602398.3143742 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602398.3740325 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.253316909936192}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602398.5098672 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602399.1632633 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602401.108842 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602401.196112 INFO: reactions 930\n",
      "1723602401.2713647 INFO: metabolites 902\n",
      "1723602401.3161058 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602401.7257588 INFO: Default biomass: [bio1]\n",
      "1723602401.9876447 INFO: cpd05178 not found in model!\n",
      "1723602401.990074 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602402.062294 INFO: reactions 838\n",
      "1723602402.1949353 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602402.5637054 INFO: Default biomass: [bio1]\n",
      "1723602402.6774302 WARNING: Ignoring reaction 'rxn08168_c0' since it already exists.\n",
      "1723602402.7012994 WARNING: Ignoring reaction 'rxn00003_c0' since it already exists.\n",
      "1723602402.739587 WARNING: Ignoring reaction 'rxn00251_c0' since it already exists.\n",
      "1723602402.7468953 WARNING: Ignoring reaction 'rxn00278_c0' since it already exists.\n",
      "1723602402.7903242 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723602402.8079464 WARNING: Ignoring reaction 'rxn00611_c0' since it already exists.\n",
      "1723602402.9300997 WARNING: Ignoring reaction 'rxn01057_c0' since it already exists.\n",
      "1723602402.932696 WARNING: Ignoring reaction 'rxn02123_c0' since it already exists.\n",
      "1723602402.934814 WARNING: Ignoring reaction 'rxn00668_c0' since it already exists.\n",
      "1723602402.9552267 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723602402.9571586 WARNING: Ignoring reaction 'rxn00784_c0' since it already exists.\n",
      "1723602402.9591787 INFO: reactions 1071\n",
      "1723602403.038633 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723602403.1038415 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602403.227538 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602403.5746295 INFO: Default biomass: [bio1]\n",
      "1723602404.6187248 INFO: metabolites 1130\n",
      "1723602404.6663756 INFO: cpd05178 not found in model!\n",
      "1723602404.6691134 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602405.0186596 WARNING: Ignoring reaction 'rxn01575_c0' since it already exists.\n",
      "1723602405.2284036 INFO: reactions 1122\n",
      "1723602405.348004 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602405.3858821 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602405.5475833 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602405.8196537 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602405.8707554 INFO: reactions 1071\n",
      "1723602405.8750498 INFO: Default biomass: [bio1]\n",
      "1723602406.0022533 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602406.4963298 INFO: Default biomass: [bio1]\n",
      "1723602407.303807 INFO: metabolites 1061\n",
      "1723602407.8273108 INFO: reactions 1257\n",
      "1723602407.9793584 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602408.5290997 INFO: Default biomass: [bio1]\n",
      "1723602411.3044903 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602411.3875113 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602411.5729384 INFO: cpd05178 not found in model!\n",
      "1723602411.575331 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602411.8992052 WARNING: Ignoring reaction 'rxn01013_c0' since it already exists.\n",
      "1723602412.1688583 INFO: cpd05178 not found in model!\n",
      "1723602412.1712031 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602412.2459753 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602412.7267282 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602412.727118 WARNING: Ignoring reaction 'rxn00430_c0' since it already exists.\n",
      "1723602413.4725592 WARNING: Ignoring reaction 'rxn04794_c0' since it already exists.\n",
      "1723602413.5419896 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602413.5447857 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.24.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.8684346445300275}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602413.7732105 WARNING: Ignoring reaction 'rxn02933_c0' since it already exists.\n",
      "1723602413.8034837 INFO: reactions 928\n",
      "1723602413.8455915 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602413.8483012 WARNING: Ignoring reaction 'rxn02934_c0' since it already exists.\n",
      "1723602413.9156036 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602413.9379494 WARNING: Ignoring reaction 'rxn11732_c0' since it already exists.\n",
      "1723602414.0508559 INFO: metabolites 1016\n",
      "1723602414.3274627 INFO: Default biomass: [bio1]\n",
      "1723602415.0006769 WARNING: Ignoring reaction 'rxn00680_c0' since it already exists.\n",
      "1723602415.2402194 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602415.547275 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602417.000516 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602417.7606127 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602417.8767314 INFO: reactions 1043\n",
      "1723602418.01061 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602418.2516668 INFO: cpd05178 not found in model!\n",
      "1723602418.254521 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602418.487247 INFO: Default biomass: [bio1]\n",
      "1723602418.498736 WARNING: Ignoring reaction 'rxn00251_c0' since it already exists.\n",
      "1723602418.5931954 INFO: metabolites 1137\n",
      "1723602418.5954363 WARNING: Ignoring reaction 'rxn00198_c0' since it already exists.\n",
      "1723602418.6546743 WARNING: Ignoring reaction 'rxn00767_c0' since it already exists.\n",
      "1723602418.7589076 INFO: reactions 1211\n",
      "1723602418.902186 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602419.0205977 WARNING: Ignoring reaction 'rxn02380_c0' since it already exists.\n",
      "1723602419.0912561 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723602419.094131 WARNING: Ignoring reaction 'rxn12204_c0' since it already exists.\n",
      "1723602419.2103562 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602419.412984 INFO: Default biomass: [bio1]\n",
      "1723602421.399139 INFO: reactions 1053\n",
      "1723602421.4576359 INFO: metabolites 1104\n",
      "1723602421.5314407 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602421.9785762 INFO: Default biomass: [bio1]\n",
      "1723602423.3409162 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602424.4291317 INFO: reactions 998\n",
      "1723602424.5637524 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602425.052685 INFO: Default biomass: [bio1]\n",
      "1723602427.8156984 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602428.3700566 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602429.1725795 INFO: reactions 893\n",
      "1723602429.2831824 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602429.2900312 INFO: reactions 1112\n",
      "1723602429.4183166 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602429.652872 INFO: Default biomass: [bio1]\n",
      "1723602429.8999538 INFO: Default biomass: [bio1]\n",
      "1723602430.844687 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602431.576055 INFO: reactions 1012\n",
      "1723602431.7213027 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602432.1435525 INFO: Default biomass: [bio1]\n",
      "1723602432.425546 INFO: reactions 1064\n",
      "1723602432.5484374 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602432.6772392 INFO: cpd05178 not found in model!\n",
      "1723602432.6794243 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602432.9921436 WARNING: Ignoring reaction 'rxn00429_c0' since it already exists.\n",
      "1723602433.0016413 INFO: Default biomass: [bio1]\n",
      "1723602433.0623689 WARNING: Ignoring reaction 'rxn02185_c0' since it already exists.\n",
      "1723602433.4154887 WARNING: Ignoring reaction 'rxn00278_c0' since it already exists.\n",
      "1723602433.8351862 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602433.85122 WARNING: Ignoring reaction 'rxn01011_c0' since it already exists.\n",
      "1723602434.0402641 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602434.0426893 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602434.2821176 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602434.354888 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602434.4087708 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602434.5549939 INFO: reactions 871\n",
      "1723602434.6543984 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602434.7471716 WARNING: Ignoring reaction 'rxn00364_c0' since it already exists.\n",
      "1723602434.7497356 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602434.9419074 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602435.0235739 INFO: Default biomass: [bio1]\n",
      "1723602436.4552643 INFO: metabolites 1127\n",
      "1723602436.4594991 INFO: reactions 898\n",
      "1723602436.5689638 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602436.9503565 INFO: Default biomass: [bio1]\n",
      "1723602438.1759799 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602438.1952112 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602438.6775584 INFO: reactions 767\n",
      "1723602438.7690108 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602438.7972777 INFO: cpd05178 not found in model!\n",
      "1723602438.7995887 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602439.117263 INFO: Default biomass: [bio1]\n",
      "1723602439.2131274 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723602440.6712437 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602440.680403 INFO: reactions 1201\n",
      "1723602440.848203 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602440.8828137 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602441.2293608 INFO: reactions 1123\n",
      "1723602441.3622901 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602441.3995872 INFO: Default biomass: [bio1]\n",
      "1723602441.6822636 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602441.8428013 INFO: Default biomass: [bio1]\n",
      "1723602442.0023947 INFO: cpd05178 not found in model!\n",
      "1723602442.0045264 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602442.3206382 INFO: metabolites 962\n",
      "1723602443.676019 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.22.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.5745091515075897, '>': 0.7358950664470107}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602444.7570357 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602445.5770352 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602445.709967 INFO: cpd05178 not found in model!\n",
      "1723602445.712154 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602445.9674296 INFO: cpd05178 not found in model!\n",
      "1723602445.969602 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602446.0865977 WARNING: Ignoring reaction 'rxn09176_c0' since it already exists.\n",
      "1723602446.4021444 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602446.445082 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602447.221065 INFO: metabolites 1169\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_H2O_MG_DASTool_bins_maxbin.017.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.1513242903097707, '<': 1.4118921421625044}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602447.5544715 WARNING: Ignoring reaction 'rxn02285_c0' since it already exists.\n",
      "1723602447.6578116 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602447.9701068 INFO: metabolites 979\n",
      "1723602448.4293475 WARNING: Ignoring reaction 'rxn00692_c0' since it already exists.\n",
      "1723602448.4650717 WARNING: Ignoring reaction 'rxn04464_c0' since it already exists.\n",
      "1723602448.5144513 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602449.661273 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602450.0362577 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602452.2817373 INFO: metabolites 617\n",
      "1723602453.6291265 INFO: cpd05178 not found in model!\n",
      "1723602453.6314514 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602453.9597328 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723602454.2025352 WARNING: Ignoring reaction 'rxn01204_c0' since it already exists.\n",
      "1723602454.6293223 WARNING: Ignoring reaction 'rxn02285_c0' since it already exists.\n",
      "1723602455.2635891 WARNING: Ignoring reaction 'rxn00781_c0' since it already exists.\n",
      "1723602455.2662625 WARNING: Ignoring reaction 'rxn01100_c0' since it already exists.\n",
      "1723602455.2677517 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723602455.2962162 WARNING: Ignoring reaction 'rxn01301_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.46729641515862574, '<': 1.2990593360767801}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602456.6217232 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602457.8594723 INFO: reactions 861\n",
      "1723602458.0234063 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602458.3911238 INFO: Default biomass: [bio1]\n",
      "1723602458.7928371 INFO: metabolites 1185\n",
      "1723602458.8539374 INFO: reactions 923\n",
      "1723602458.9607663 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602459.3537958 INFO: Default biomass: [bio1]\n",
      "1723602464.1606538 INFO: reactions 1010\n",
      "1723602464.3367662 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602464.7744782 INFO: Default biomass: [bio1]\n",
      "1723602466.9352734 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602467.777815 INFO: reactions 828\n",
      "1723602467.9595244 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602467.9781826 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602468.3274226 INFO: Default biomass: [bio1]\n",
      "1723602469.4852138 INFO: reactions 1084\n",
      "1723602469.6221166 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602470.0924344 INFO: Default biomass: [bio1]\n",
      "1723602471.833038 INFO: reactions 1122\n",
      "1723602472.0034683 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602472.5220318 INFO: reactions 798\n",
      "1723602472.5336015 INFO: Default biomass: [bio1]\n",
      "1723602472.6146522 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602472.9872546 INFO: Default biomass: [bio1]\n",
      "1723602473.3961158 INFO: reactions 524\n",
      "1723602473.538083 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602473.764255 INFO: Default biomass: [bio1]\n",
      "1723602475.1651552 INFO: reactions 1051\n",
      "1723602475.2996345 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602475.7377512 INFO: Default biomass: [bio1]\n",
      "1723602477.107262 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602478.980839 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602481.4762044 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602481.54686 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602482.2340708 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602482.4455163 INFO: reactions 1186\n",
      "1723602482.6763525 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602483.1174881 INFO: reactions 922\n",
      "1723602483.2259583 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602483.2559836 INFO: Default biomass: [bio1]\n",
      "1723602483.5306637 INFO: cpd05178 not found in model!\n",
      "1723602483.5330307 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602483.6117826 INFO: Default biomass: [bio1]\n",
      "1723602484.205319 WARNING: Ignoring reaction 'rxn05010_c0' since it already exists.\n",
      "1723602484.3460233 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602484.3706598 INFO: reactions 1250\n",
      "1723602484.382806 WARNING: Ignoring reaction 'rxn00272_c0' since it already exists.\n",
      "1723602484.5345988 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602484.5929384 WARNING: Ignoring reaction 'rxn01790_c0' since it already exists.\n",
      "1723602484.6553411 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602484.6580002 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602485.0499198 WARNING: Ignoring reaction 'rxn05011_c0' since it already exists.\n",
      "1723602485.0913835 INFO: Default biomass: [bio1]\n",
      "1723602485.1661117 WARNING: Ignoring reaction 'rxn00270_c0' since it already exists.\n",
      "1723602485.2578957 WARNING: Ignoring reaction 'rxn00677_c0' since it already exists.\n",
      "1723602485.8262858 INFO: reactions 977\n",
      "1723602485.945629 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602485.951822 WARNING: Ignoring reaction 'rxn00991_c0' since it already exists.\n",
      "1723602486.3649893 INFO: Default biomass: [bio1]\n",
      "1723602486.5496054 WARNING: Ignoring reaction 'rxn12811_c0' since it already exists.\n",
      "1723602487.0860415 WARNING: Ignoring reaction 'rxn00350_c0' since it already exists.\n",
      "1723602487.1495817 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602487.1842766 INFO: reactions 1218\n",
      "1723602487.3391914 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602487.886574 INFO: Default biomass: [bio1]\n",
      "1723602488.9209712 INFO: metabolites 1044\n",
      "1723602492.1504385 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602492.2292938 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602492.399632 INFO: reactions 974\n",
      "1723602492.5144546 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602492.942992 INFO: Default biomass: [bio1]\n",
      "1723602494.0206451 INFO: reactions 951\n",
      "1723602494.0311785 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602494.2193656 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602494.6330993 INFO: Default biomass: [bio1]\n",
      "1723602495.247021 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602496.6268709 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602497.7157602 INFO: cpd05178 not found in model!\n",
      "1723602497.7179422 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602498.3747303 INFO: cpd05178 not found in model!\n",
      "1723602498.3768988 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602498.5126467 INFO: cpd05178 not found in model!\n",
      "1723602498.514875 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602498.7664542 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723602498.8450286 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n",
      "1723602498.9260275 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602498.966748 WARNING: Ignoring reaction 'rxn09180_c0' since it already exists.\n",
      "1723602499.6469119 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602499.6495142 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602499.66829 WARNING: Ignoring reaction 'rxn00375_c0' since it already exists.\n",
      "1723602499.7252321 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602500.0003438 WARNING: Ignoring reaction 'rxn00912_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.6443270561373875, '>': 1.055447690272147}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602500.069457 WARNING: Ignoring reaction 'rxn00102_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.19.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.48390270490100307}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602500.2551441 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602500.3698149 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602500.3723896 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602500.3883243 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602500.565985 INFO: reactions 1014\n",
      "1723602500.6790714 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602500.84626 WARNING: Ignoring reaction 'rxn00362_c0' since it already exists.\n",
      "1723602500.8928688 INFO: cpd05178 not found in model!\n",
      "1723602500.895089 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602501.0803313 INFO: Default biomass: [bio1]\n",
      "1723602501.1606932 INFO: metabolites 926\n",
      "1723602501.3205447 WARNING: Ignoring reaction 'rxn05909_c0' since it already exists.\n",
      "1723602501.4687653 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602501.6835916 WARNING: Ignoring reaction 'rxn00935_c0' since it already exists.\n",
      "1723602501.8566742 WARNING: Ignoring reaction 'rxn01423_c0' since it already exists.\n",
      "1723602501.9262998 WARNING: Ignoring reaction 'rxn00467_c0' since it already exists.\n",
      "1723602501.9398434 WARNING: Ignoring reaction 'rxn00317_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.10.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.5085877436036559}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602502.0233243 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602502.140862 INFO: metabolites 964\n",
      "1723602502.3551984 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602502.5902386 INFO: cpd05178 not found in model!\n",
      "1723602502.5925357 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602502.8985574 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602503.2184818 WARNING: Ignoring reaction 'rxn09128_c0' since it already exists.\n",
      "1723602503.5325418 INFO: metabolites 1195\n",
      "1723602503.9851203 INFO: metabolites 1007\n",
      "1723602504.014847 WARNING: Ignoring reaction 'rxn03004_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.44.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.6439530981792356}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602504.4883292 INFO: reactions 1236\n",
      "1723602504.7139645 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602504.8539724 INFO: reactions 984\n",
      "1723602504.9458008 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602504.9666288 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602504.970453 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602505.1044018 WARNING: Ignoring reaction 'rxn00796_c0' since it already exists.\n",
      "1723602505.192198 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602505.249393 INFO: Default biomass: [bio1]\n",
      "1723602505.5010753 INFO: Default biomass: [bio1]\n",
      "1723602505.6828938 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602505.860542 INFO: reactions 1137\n",
      "1723602505.9914308 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602506.4737957 INFO: Default biomass: [bio1]\n",
      "1723602507.1991775 INFO: cpd05178 not found in model!\n",
      "1723602507.201445 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602507.4260938 INFO: metabolites 1122\n",
      "1723602507.661248 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602508.4456344 INFO: reactions 1076\n",
      "1723602508.4608104 INFO: cpd05178 not found in model!\n",
      "1723602508.463248 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602508.5809655 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602508.6997845 WARNING: Ignoring reaction 'rxn11209_c0' since it already exists.\n",
      "1723602508.7078924 WARNING: Ignoring reaction 'rxn01575_c0' since it already exists.\n",
      "1723602508.7176292 WARNING: Ignoring reaction 'rxn08168_c0' since it already exists.\n",
      "1723602508.827795 WARNING: Ignoring reaction 'rxn00612_c0' since it already exists.\n",
      "1723602508.8584938 WARNING: Ignoring reaction 'rxn09188_c0' since it already exists.\n",
      "1723602508.8610084 WARNING: Ignoring reaction 'rxn00503_c0' since it already exists.\n",
      "1723602508.8736389 WARNING: Ignoring reaction 'rxn00692_c0' since it already exists.\n",
      "1723602508.9189937 WARNING: Ignoring reaction 'rxn00429_c0' since it already exists.\n",
      "1723602509.0126367 WARNING: Ignoring reaction 'rxn12215_c0' since it already exists.\n",
      "1723602509.015067 WARNING: Ignoring reaction 'rxn03052_c0' since it already exists.\n",
      "1723602509.0191152 WARNING: Ignoring reaction 'rxn05625_c0' since it already exists.\n",
      "1723602509.0206544 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723602509.034777 WARNING: Ignoring reaction 'rxn00747_c0' since it already exists.\n",
      "1723602509.0370834 WARNING: Ignoring reaction 'rxn00767_c0' since it already exists.\n",
      "1723602509.0506046 INFO: Default biomass: [bio1]\n",
      "1723602509.0522287 WARNING: Ignoring reaction 'rxn01057_c0' since it already exists.\n",
      "1723602509.149619 WARNING: Ignoring reaction 'rxn11537_c0' since it already exists.\n",
      "1723602509.1522365 WARNING: Ignoring reaction 'rxn11535_c0' since it already exists.\n",
      "1723602509.214999 INFO: metabolites 948\n",
      "1723602509.2339244 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602509.3089628 WARNING: Ignoring reaction 'rxn00364_c0' since it already exists.\n",
      "1723602509.3584716 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602511.4346507 INFO: cpd05178 not found in model!\n",
      "1723602511.4369738 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602512.6546702 INFO: cpd05178 not found in model!\n",
      "1723602512.6570911 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.16.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7117642729090018}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602513.2067544 WARNING: Ignoring reaction 'rxn01011_c0' since it already exists.\n",
      "1723602513.6866693 INFO: reactions 1043\n",
      "1723602513.8049092 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602514.132925 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602514.1395154 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602514.2883074 INFO: Default biomass: [bio1]\n",
      "1723602514.3076413 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602514.3885098 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602514.6096785 WARNING: Ignoring reaction 'rxn00499_c0' since it already exists.\n",
      "1723602514.6174777 WARNING: Ignoring reaction 'rxn00270_c0' since it already exists.\n",
      "1723602514.8386297 WARNING: Ignoring reaction 'rxn01048_c0' since it already exists.\n",
      "1723602514.8411016 WARNING: Ignoring reaction 'rxn01333_c0' since it already exists.\n",
      "1723602514.843001 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723602514.846006 WARNING: Ignoring reaction 'rxn04954_c0' since it already exists.\n",
      "1723602514.8480299 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602514.8828006 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602515.21264 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602515.8571074 INFO: metabolites 1143\n",
      "1723602516.096509 INFO: reactions 1107\n",
      "1723602516.2455354 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602516.5254936 INFO: metabolites 1068\n",
      "1723602516.7570922 INFO: Default biomass: [bio1]\n",
      "1723602516.7607436 INFO: reactions 1219\n",
      "1723602516.9014833 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602517.3941548 INFO: cpd05178 not found in model!\n",
      "1723602517.3964703 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602517.4189014 INFO: Default biomass: [bio1]\n",
      "1723602517.6506665 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602517.802259 INFO: cpd05178 not found in model!\n",
      "1723602517.8045144 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602518.7551432 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602518.8183842 WARNING: Ignoring reaction 'rxn00912_c0' since it already exists.\n",
      "1723602518.8264108 WARNING: Ignoring reaction 'rxn10124_c0' since it already exists.\n",
      "1723602518.9657264 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.49.contigs__.RAST"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602518.968311 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Missing reaction EX_cpd00557_c0 {'>': 0.2402846518884882}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602519.0411289 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602519.1447716 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602519.5827448 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602519.6953824 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602519.961071 INFO: reactions 957\n",
      "1723602520.068814 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602520.4572136 INFO: Default biomass: [bio1]\n",
      "1723602521.0279427 INFO: metabolites 957\n",
      "1723602521.1620634 INFO: metabolites 896\n",
      "1723602521.7664359 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602522.7885258 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602523.4571886 INFO: cpd05178 not found in model!\n",
      "1723602523.4595075 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602523.5558035 INFO: cpd05178 not found in model!\n",
      "1723602523.558295 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602523.940994 WARNING: Ignoring reaction 'rxn01418_c0' since it already exists.\n",
      "1723602524.1048248 WARNING: Ignoring reaction 'rxn01301_c0' since it already exists.\n",
      "1723602524.4211383 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602524.940441 WARNING: Ignoring reaction 'rxn02229_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14.contigs__.RAST Missing reaction EX_cpd00557_c0 "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602524.9754424 WARNING: Ignoring reaction 'rxn02185_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'>': 0.884423318482912}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602525.1290848 WARNING: Ignoring reaction 'rxn00806_c0' since it already exists.\n",
      "1723602525.220206 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.86.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.9919625520864876}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602525.4878838 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602525.5988023 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602525.8585682 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602525.9580402 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602526.8619323 INFO: cpd05178 not found in model!\n",
      "1723602526.8642178 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602527.0466828 INFO: metabolites 1178\n",
      "1723602527.35406 WARNING: Ignoring reaction 'rxn00427_c0' since it already exists.\n",
      "1723602527.7912865 INFO: metabolites 1140\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.1287679755185094}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602528.078822 INFO: cpd05178 not found in model!\n",
      "1723602528.080861 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602528.4151738 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602528.5087788 WARNING: Ignoring reaction 'rxn01418_c0' since it already exists.\n",
      "1723602528.7725465 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602529.017768 WARNING: Ignoring reaction 'rxn00239_c0' since it already exists.\n",
      "1723602529.8178618 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.37.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7079346984592688}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602529.8512652 INFO: metabolites 1041\n",
      "1723602530.793403 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602532.318833 INFO: metabolites 1010\n",
      "1723602533.712104 INFO: reactions 1139\n",
      "1723602533.8449037 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602534.3286467 INFO: Default biomass: [bio1]\n",
      "1723602538.0052948 INFO: reactions 1160\n",
      "1723602538.1479142 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602538.6529083 INFO: Default biomass: [bio1]\n",
      "1723602542.6769295 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602543.2924392 INFO: cpd05178 not found in model!\n",
      "1723602543.294554 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602543.7873294 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602544.163876 INFO: reactions 1222\n",
      "1723602544.175185 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602544.3534997 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602544.634922 INFO: reactions 995\n",
      "1723602544.8039844 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': -0.17307133129176153}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602544.8943915 INFO: Default biomass: [bio1]\n",
      "1723602545.0675666 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723602545.1127863 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602545.2749312 INFO: Default biomass: [bio1]\n",
      "1723602545.7096977 INFO: cpd05178 not found in model!\n",
      "1723602545.711991 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602545.9646297 WARNING: Ignoring reaction 'rxn01575_c0' since it already exists.\n",
      "1723602546.2302904 INFO: cpd05178 not found in model!\n",
      "1723602546.232495 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602546.681545 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602546.8957996 INFO: metabolites 1208\n",
      "1723602546.9969842 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602547.151241 WARNING: Ignoring reaction 'rxn01265_c0' since it already exists.\n",
      "1723602548.1496344 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602548.5476894 INFO: metabolites 1256\n",
      "1723602548.994566 INFO: cpd05178 not found in model!\n",
      "1723602548.997044 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602549.2961447 WARNING: Ignoring reaction 'rxn14415_c0' since it already exists.\n",
      "1723602549.748284 INFO: metabolites 944\n",
      "1723602549.975064 WARNING: Ignoring reaction 'rxn02229_c0' since it already exists.\n",
      "1723602550.1968029 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723602550.213099 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602551.3677413 INFO: cpd05178 not found in model!\n",
      "1723602551.3700285 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602551.4903398 INFO: metabolites 772\n",
      "1723602552.452945 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602552.45617 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602553.086966 INFO: cpd05178 not found in model!\n",
      "1723602553.0892122 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602553.3746324 WARNING: Ignoring reaction 'rxn00606_c0' since it already exists.\n",
      "1723602553.5018444 WARNING: Ignoring reaction 'rxn01116_c0' since it already exists.\n",
      "1723602553.5051506 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602553.5639176 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602553.653687 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602553.6929667 WARNING: Ignoring reaction 'rxn01048_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.8.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.0555661217702847}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602554.2525861 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602554.7379863 INFO: cpd05178 not found in model!\n",
      "1723602554.7402601 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602555.0163555 WARNING: Ignoring reaction 'rxn00692_c0' since it already exists.\n",
      "1723602555.326382 WARNING: Ignoring reaction 'rxn01048_c0' since it already exists.\n",
      "1723602555.4077895 WARNING: Ignoring reaction 'rxn00427_c0' since it already exists.\n",
      "1723602555.4835184 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723602555.977822 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723602555.9924207 INFO: metabolites 1137\n",
      "1723602556.091714 INFO: reactions 950\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_concoct_out.78.contigs__.RAST Missing reaction EX_cpd00557_c0 "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602556.2011802 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'>': 1.31110365264438}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602556.20544 WARNING: Ignoring reaction 'rxn01575_c0' since it already exists.\n",
      "1723602556.239183 WARNING: Ignoring reaction 'rxn00270_c0' since it already exists.\n",
      "1723602556.339423 INFO: cpd05178 not found in model!\n",
      "1723602556.3416204 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602556.4875793 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602556.5955086 INFO: Default biomass: [bio1]\n",
      "1723602556.9862115 INFO: metabolites 1210\n",
      "1723602557.3713963 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602557.5355961 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602557.6023643 INFO: cpd05178 not found in model!\n",
      "1723602557.604696 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602557.9158936 INFO: reactions 1059\n",
      "1723602558.0479429 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602558.2319555 INFO: cpd05178 not found in model!\n",
      "1723602558.234276 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602558.520472 INFO: Default biomass: [bio1]\n",
      "1723602558.6571386 WARNING: Ignoring reaction 'rxn00692_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.84.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.2728644890728444, '>': 1.107311986047168}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602559.0303965 INFO: metabolites 885\n",
      "1723602559.0458803 WARNING: Ignoring reaction 'rxn00427_c0' since it already exists.\n",
      "1723602559.3794816 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n",
      "1723602559.4183438 INFO: metabolites 1280\n",
      "1723602559.4302773 INFO: cpd05178 not found in model!\n",
      "1723602559.4322197 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.32.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.7246774034295753}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602559.7459536 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n",
      "1723602559.9174194 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602560.010538 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602560.2470298 WARNING: Ignoring reaction 'rxn09149_c0' since it already exists.\n",
      "1723602560.5839534 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602560.6733987 WARNING: Ignoring reaction 'rxn01301_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.38.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7524768451698874}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602560.8526804 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n",
      "1723602560.9223793 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602561.3943887 INFO: reactions 1031\n",
      "1723602561.4792023 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602561.511255 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602561.922361 INFO: Default biomass: [bio1]\n",
      "1723602562.032918 INFO: metabolites 1033\n",
      "1723602562.0458343 INFO: cpd05178 not found in model!\n",
      "1723602562.0481067 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602562.4305584 WARNING: Ignoring reaction 'rxn04954_c0' since it already exists.\n",
      "1723602562.433127 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602562.7175927 WARNING: Ignoring reaction 'rxn01790_c0' since it already exists.\n",
      "1723602562.7205808 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602562.7901118 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602563.8167446 INFO: metabolites 947\n",
      "1723602564.0216208 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602564.0851355 INFO: metabolites 1093\n",
      "1723602564.282704 INFO: reactions 877\n",
      "1723602564.3870692 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602564.904041 INFO: Default biomass: [bio1]\n",
      "1723602565.2142296 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602565.6383684 INFO: metabolites 1206\n",
      "1723602566.180876 INFO: cpd05178 not found in model!\n",
      "1723602566.1831558 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602566.881147 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723602567.183489 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602567.5354939 INFO: reactions 1063\n",
      "1723602567.6579888 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602567.870857 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602568.1453195 INFO: Default biomass: [bio1]\n",
      "1723602570.5492446 INFO: metabolites 1244\n",
      "1723602570.948218 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602573.1105804 INFO: cpd05178 not found in model!\n",
      "1723602573.1130064 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602573.322024 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602573.8619077 WARNING: Ignoring reaction 'rxn00427_c0' since it already exists.\n",
      "1723602574.05013 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602576.422924 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602577.4397726 INFO: cpd05178 not found in model!\n",
      "1723602577.4420466 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': -0.25938141580362645}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602578.5221488 WARNING: Ignoring reaction 'rxn00912_c0' since it already exists.\n",
      "1723602578.6172907 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602578.6198556 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602578.7007015 INFO: cpd05178 not found in model!\n",
      "1723602578.7029457 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602578.9051454 INFO: reactions 905\n",
      "1723602579.0195065 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602579.0630972 WARNING: Ignoring reaction 'rxn12215_c0' since it already exists.\n",
      "1723602579.1488333 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602579.2347245 WARNING: Ignoring reaction 'rxn00877_c0' since it already exists.\n",
      "1723602579.237413 WARNING: Ignoring reaction 'rxn00899_c0' since it already exists.\n",
      "1723602579.4370058 INFO: Default biomass: [bio1]\n",
      "1723602580.0845702 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602583.003149 INFO: reactions 1090\n",
      "1723602583.0068996 INFO: metabolites 1220\n",
      "1723602583.1469066 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602583.6109521 INFO: Default biomass: [bio1]\n",
      "1723602587.582618 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602588.0137048 INFO: cpd05178 not found in model!\n",
      "1723602588.016111 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602588.3475661 WARNING: Ignoring reaction 'rxn02774_c0' since it already exists.\n",
      "1723602588.4514127 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723602589.334268 WARNING: Ignoring reaction 'rxn00781_c0' since it already exists.\n",
      "1723602589.3733573 WARNING: Ignoring reaction 'rxn01011_c0' since it already exists.\n",
      "1723602589.3895807 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723602589.4301286 WARNING: Ignoring reaction 'rxn00251_c0' since it already exists.\n",
      "1723602589.731013 WARNING: Ignoring reaction 'rxn01100_c0' since it already exists.\n",
      "1723602589.9875104 WARNING: Ignoring reaction 'rxn00680_c0' since it already exists.\n",
      "1723602589.9952722 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602592.5382617 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602592.7577434 INFO: cpd05178 not found in model!\n",
      "1723602592.7598543 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602593.746782 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.25.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.6667471560626754}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602594.7534795 INFO: reactions 1085\n",
      "1723602594.8937309 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602595.2751768 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602595.424593 INFO: Default biomass: [bio1]\n",
      "/tmp/ipykernel_230925/2585747718.py:32: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  consolidated_gf_results[OGname][currtype][metabolite][\"zscore\"] = (consolidated_gf_results[OGname][currtype][metabolite][\"new_probability\"]-ave)/stdev\n",
      "1723602597.0456798 INFO: metabolites 1037\n",
      "1723602598.8806238 INFO: cpd05178 not found in model!\n",
      "1723602598.8829637 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.40983435541194974}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602600.2469409 INFO: cpd05178 not found in model!\n",
      "1723602600.2490802 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602600.7035406 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602600.7144735 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723602601.220977 INFO: cpd05178 not found in model!\n",
      "1723602601.2233284 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602601.548372 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602602.6363509 INFO: metabolites 1221\n",
      "1723602603.0619822 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602603.6252878 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602604.6746058 INFO: metabolites 1167\n",
      "1723602613.1442318 INFO: cpd05178 not found in model!\n",
      "1723602613.146824 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602613.6646752 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602613.6769419 INFO: reactions 749\n",
      "1723602613.7664165 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602614.072744 INFO: Default biomass: [bio1]\n",
      "1723602615.292256 INFO: metabolites 1063\n",
      "1723602617.2318442 INFO: cpd05178 not found in model!\n",
      "1723602617.2339494 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602617.9531267 INFO: reactions 1154\n",
      "1723602618.0951233 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602618.5670338 INFO: Default biomass: [bio1]\n",
      "1723602618.8612058 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602619.5787463 INFO: reactions 780\n",
      "1723602619.667495 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602619.9953108 INFO: Default biomass: [bio1]\n",
      "1723602620.5423563 INFO: metabolites 1067\n",
      "1723602621.277531 INFO: cpd05178 not found in model!\n",
      "1723602621.2798388 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602621.3450153 INFO: cpd05178 not found in model!\n",
      "1723602621.3472738 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602621.3939288 INFO: reactions 1076\n",
      "1723602621.519653 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_concoct_out.67.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.8655151871853228}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602621.9023385 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602621.9824922 WARNING: Ignoring reaction 'rxn03643_c0' since it already exists.\n",
      "1723602622.019681 INFO: Default biomass: [bio1]\n",
      "1723602622.0490963 WARNING: Ignoring reaction 'rxn00427_c0' since it already exists.\n",
      "1723602622.051686 WARNING: Ignoring reaction 'rxn03644_c0' since it already exists.\n",
      "1723602622.877918 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602623.0319614 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602623.5045516 INFO: reactions 1001\n",
      "1723602623.6249003 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602624.0458999 INFO: Default biomass: [bio1]\n",
      "1723602624.4329393 INFO: metabolites 917\n",
      "1723602624.7852478 INFO: metabolites 1208\n",
      "1723602624.8304706 INFO: cpd05178 not found in model!\n",
      "1723602624.8334336 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602625.141776 WARNING: Ignoring reaction 'rxn01418_c0' since it already exists.\n",
      "1723602625.1727993 INFO: cpd05178 not found in model!\n",
      "1723602625.1749592 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602625.48775 INFO: reactions 864\n",
      "1723602625.5957556 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602625.9530807 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.0515890399312546}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602626.645255 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723602626.6482778 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723602626.6907432 INFO: cpd05178 not found in model!\n",
      "1723602626.7010305 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602627.0023057 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602627.2222674 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602627.3154798 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602627.4199314 WARNING: Ignoring reaction 'rxn01333_c0' since it already exists.\n",
      "1723602627.7726533 INFO: cpd05178 not found in model!\n",
      "1723602627.7748628 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602627.78302 WARNING: Ignoring reaction 'rxn01361_c0' since it already exists.\n",
      "1723602627.8399317 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602628.0948153 WARNING: Ignoring reaction 'rxn14415_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.concoct_out.44.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.2625145147627024}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602628.300177 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602628.4141126 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602628.425143 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.38976024140679}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602628.6171067 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602628.8138921 INFO: metabolites 918\n",
      "1723602629.1476007 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602629.505154 WARNING: Ignoring reaction 'rxn00359_c0' since it already exists.\n",
      "1723602629.6270688 INFO: metabolites 1210\n",
      "1723602630.0945265 INFO: metabolites 936\n",
      "1723602630.131368 INFO: cpd05178 not found in model!\n",
      "1723602630.1331875 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602630.381967 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602630.4517565 INFO: cpd05178 not found in model!\n",
      "1723602630.453842 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602630.901301 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.16.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.826957072015853, '>': 1.0599664397519681}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602631.2268207 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602631.229937 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602631.3186934 INFO: reactions 925\n",
      "1723602631.4992843 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602631.537143 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723602631.6750584 WARNING: Ignoring reaction 'rxn05010_c0' since it already exists.\n",
      "1723602631.7856894 WARNING: Ignoring reaction 'rxn11732_c0' since it already exists.\n",
      "1723602631.8448458 WARNING: Ignoring reaction 'rxn04464_c0' since it already exists.\n",
      "1723602631.9182332 INFO: Default biomass: [bio1]\n",
      "1723602631.926617 WARNING: Ignoring reaction 'rxn00501_c0' since it already exists.\n",
      "1723602631.9429626 INFO: metabolites 1203\n",
      "1723602632.0510612 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602632.3689718 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602632.707961 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602634.0391355 INFO: metabolites 1040\n",
      "1723602634.0760684 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602637.806231 INFO: cpd05178 not found in model!\n",
      "1723602637.8089805 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602639.163702 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602639.1666708 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602639.3581789 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602639.3619835 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.39707804428894156}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602639.9493294 WARNING: Ignoring reaction 'rxn00453_c0' since it already exists.\n",
      "1723602640.521815 WARNING: Ignoring reaction 'rxn00003_c0' since it already exists.\n",
      "1723602640.524726 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n",
      "1723602640.634425 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602640.8718634 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602642.4287298 INFO: cpd05178 not found in model!\n",
      "1723602642.431472 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602643.1027925 INFO: cpd05178 not found in model!\n",
      "1723602643.1050785 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602643.2718194 INFO: reactions 817\n",
      "1723602643.3722155 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7873536388959989}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602643.7434614 INFO: Default biomass: [bio1]\n",
      "1723602644.4810927 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602644.5100148 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602644.810804 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602645.658574 INFO: cpd05178 not found in model!\n",
      "1723602645.6624513 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602645.9191713 INFO: metabolites 969\n",
      "1723602646.9273076 INFO: metabolites 1195\n",
      "1723602647.2150156 INFO: cpd05178 not found in model!\n",
      "1723602647.2175603 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.32.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 0.8504654152851555}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602647.3349416 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723602647.3375642 WARNING: Ignoring reaction 'rxn02185_c0' since it already exists.\n",
      "1723602647.8670201 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602647.8975813 WARNING: Ignoring reaction 'rxn00686_c0' since it already exists.\n",
      "1723602648.1694798 INFO: reactions 876\n",
      "1723602648.2727096 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602648.6496365 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.80.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.8737952778807454, '<': 1.5994982280374126}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602648.8358212 WARNING: Ignoring reaction 'rxn01507_c0' since it already exists.\n",
      "1723602649.1953716 WARNING: Ignoring reaction 'rxn01301_c0' since it already exists.\n",
      "1723602649.6523633 INFO: metabolites 1245\n",
      "1723602649.7539804 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602650.4944417 INFO: reactions 1039\n",
      "1723602650.6267927 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602651.0799599 INFO: Default biomass: [bio1]\n",
      "1723602651.4037786 INFO: cpd05178 not found in model!\n",
      "1723602651.4061863 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602651.5745962 INFO: metabolites 1005\n",
      "1723602652.3684676 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.5888988521143967}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602652.8963597 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602652.9470925 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602654.5004961 INFO: metabolites 945\n",
      "1723602657.184154 INFO: reactions 905\n",
      "1723602657.2930171 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602657.6763935 INFO: Default biomass: [bio1]\n",
      "1723602657.6876233 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602659.5417526 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602660.0037067 INFO: cpd05178 not found in model!\n",
      "1723602660.0061553 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602661.469621 INFO: cpd05178 not found in model!\n",
      "1723602661.471978 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.3.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.21284151448009572}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602662.2136567 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602662.618301 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602662.6209939 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602663.1556027 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602663.6631382 INFO: reactions 901\n",
      "1723602663.7971468 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602664.0561936 INFO: metabolites 1085\n",
      "1723602664.1718678 INFO: Default biomass: [bio1]\n",
      "1723602664.6170344 INFO: metabolites 755\n",
      "1723602665.554658 INFO: cpd05178 not found in model!\n",
      "1723602665.5570393 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602666.0516083 WARNING: Ignoring reaction 'rxn03004_c0' since it already exists.\n",
      "1723602666.1213362 WARNING: Ignoring reaction 'rxn00686_c0' since it already exists.\n",
      "1723602666.2469096 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.36.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.3751004434716353}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602666.8060536 WARNING: Ignoring reaction 'rxn02933_c0' since it already exists.\n",
      "1723602666.8755267 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602667.0490053 WARNING: Ignoring reaction 'rxn00747_c0' since it already exists.\n",
      "1723602667.2699778 WARNING: Ignoring reaction 'rxn04464_c0' since it already exists.\n",
      "1723602668.0903728 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602669.002497 INFO: cpd05178 not found in model!\n",
      "1723602669.0044417 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.8640193234470626}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602669.6144648 INFO: metabolites 1156\n",
      "1723602669.9814987 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602669.9840686 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602670.3729436 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723602670.8199644 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602672.663371 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602673.4483123 INFO: metabolites 954\n",
      "1723602673.585514 INFO: reactions 1096\n",
      "1723602673.716073 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602674.1751618 INFO: Default biomass: [bio1]\n",
      "1723602678.3317595 INFO: cpd05178 not found in model!\n",
      "1723602678.3357184 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602678.623701 WARNING: Ignoring reaction 'rxn00907_c0' since it already exists.\n",
      "1723602679.2631288 WARNING: Ignoring reaction 'rxn02774_c0' since it already exists.\n",
      "1723602679.7781477 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602680.8721833 INFO: reactions 1017\n",
      "1723602680.9931912 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602681.3714378 INFO: metabolites 1122\n",
      "1723602681.429785 INFO: Default biomass: [bio1]\n",
      "1723602682.832078 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602682.8734105 INFO: cpd05178 not found in model!\n",
      "1723602682.876001 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602683.589215 WARNING: Ignoring reaction 'rxn00953_c0' since it already exists.\n",
      "1723602683.8199508 INFO: cpd05178 not found in model!\n",
      "1723602683.8224502 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602684.077854 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602684.16211 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602684.7376235 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602684.7402456 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.6380389257606642}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602685.04313 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602685.7074006 INFO: metabolites 1065\n",
      "1723602686.7045357 INFO: metabolites 1123\n",
      "1723602689.889374 INFO: cpd05178 not found in model!\n",
      "1723602689.891881 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602689.94687 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602690.3870733 WARNING: Ignoring reaction 'rxn01013_c0' since it already exists.\n",
      "1723602690.48016 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602690.6006975 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602692.125955 INFO: cpd05178 not found in model!\n",
      "1723602692.1286213 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602692.5437517 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602693.2123067 INFO: reactions 1170\n",
      "1723602693.3547661 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.51.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': -0.7031540738986283}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602693.7891574 INFO: reactions 834\n",
      "1723602693.860738 INFO: Default biomass: [bio1]\n",
      "1723602693.8940105 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602694.1075196 INFO: reactions 981\n",
      "1723602694.2254946 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602694.261844 INFO: Default biomass: [bio1]\n",
      "1723602694.3992748 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602694.6352742 INFO: Default biomass: [bio1]\n",
      "1723602696.0037985 INFO: metabolites 1191\n",
      "1723602698.8001986 INFO: reactions 1004\n",
      "1723602698.9167497 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602699.332814 INFO: Default biomass: [bio1]\n",
      "1723602702.3218646 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602702.5802114 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602702.8896449 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602704.2392204 INFO: reactions 1107\n",
      "1723602704.3432467 INFO: cpd05178 not found in model!\n",
      "1723602704.3455553 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602704.3993928 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602704.6910837 WARNING: Ignoring reaction 'rxn01333_c0' since it already exists.\n",
      "1723602704.873197 INFO: Default biomass: [bio1]\n",
      "1723602705.8108108 WARNING: Ignoring reaction 'rxn02774_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.45.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.6044552608115471}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602706.8916914 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602707.5256722 INFO: cpd05178 not found in model!\n",
      "1723602707.5279222 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602707.9587655 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602708.4660344 INFO: reactions 1165\n",
      "1723602708.6066906 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602709.096754 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7074801224822644}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602709.5957625 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602713.0818622 INFO: cpd05178 not found in model!\n",
      "1723602713.0844064 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602713.2535822 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602713.5732963 INFO: cpd05178 not found in model!\n",
      "1723602713.575746 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602713.8729315 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602713.8756504 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.34.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.9360109050280674}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602714.5304933 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602714.8075938 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602714.810127 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602714.95126 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723602714.953576 WARNING: Ignoring reaction 'rxn00784_c0' since it already exists.\n",
      "1723602715.1982315 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602715.2020998 WARNING: Ignoring reaction 'rxn00912_c0' since it already exists.\n",
      "1723602715.2155588 WARNING: Ignoring reaction 'rxn00225_c0' since it already exists.\n",
      "1723602715.7174687 INFO: cpd05178 not found in model!\n",
      "1723602715.719826 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602716.3158164 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602716.8209147 WARNING: Ignoring reaction 'rxn01746_c0' since it already exists.\n",
      "1723602716.823611 WARNING: Ignoring reaction 'rxn01748_c0' since it already exists.\n",
      "1723602717.2173028 INFO: cpd05178 not found in model!\n",
      "1723602717.219631 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602717.3033366 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602717.4432147 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602718.4582708 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602718.7519565 INFO: metabolites 1008\n",
      "1723602719.140062 INFO: cpd05178 not found in model!\n",
      "1723602719.1425743 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602719.356996 INFO: reactions 909\n",
      "1723602719.4607153 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_H2O_MG_DASTool_bins_metabat.49.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.5517514695420821}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602719.725145 WARNING: Ignoring reaction 'rxn01334_c0' since it already exists.\n",
      "1723602719.753771 INFO: reactions 986\n",
      "1723602719.83315 INFO: Default biomass: [bio1]\n",
      "1723602719.8586538 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723602719.8682814 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602720.1208427 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602720.1234174 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602720.2846243 INFO: Default biomass: [bio1]\n",
      "1723602720.2944982 INFO: reactions 858\n",
      "1723602720.300191 INFO: cpd05178 not found in model!\n",
      "1723602720.3025715 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602720.3922024 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602720.6237302 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602720.7412708 INFO: Default biomass: [bio1]\n",
      "1723602721.1824422 WARNING: Ignoring reaction 'rxn00503_c0' since it already exists.\n",
      "1723602721.2162068 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602721.2186832 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602721.2934213 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602721.4346828 INFO: cpd05178 not found in model!\n",
      "1723602721.4369082 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602721.680382 INFO: reactions 858\n",
      "1723602721.7687242 WARNING: Ignoring reaction 'rxn01790_c0' since it already exists.\n",
      "1723602721.7779682 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_H2O_MG_DASTool_bins_concoct_out.22.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.29767053428709733}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602722.1144483 INFO: Default biomass: [bio1]\n",
      "1723602723.4237645 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602725.1239643 INFO: reactions 1015\n",
      "1723602725.26288 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602725.7226174 INFO: Default biomass: [bio1]\n",
      "1723602725.7277734 INFO: cpd05178 not found in model!\n",
      "1723602725.729565 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602726.001676 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602726.0743496 WARNING: Ignoring reaction 'rxn00912_c0' since it already exists.\n",
      "1723602726.7035437 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602726.829343 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602726.9100988 WARNING: Ignoring reaction 'rxn04794_c0' since it already exists.\n",
      "1723602727.2797692 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.27.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.6254198245711645}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602727.8228834 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602727.895174 WARNING: Ignoring reaction 'rxn01818_c0' since it already exists.\n",
      "1723602727.8976698 WARNING: Ignoring reaction 'rxn10614_c0' since it already exists.\n",
      "1723602727.8993938 WARNING: Ignoring reaction 'rxn01862_c0' since it already exists.\n",
      "1723602728.307913 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602728.7272537 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602728.9827392 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602729.142189 INFO: reactions 868\n",
      "1723602729.246538 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602729.5916157 INFO: Default biomass: [bio1]\n",
      "1723602730.3369527 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602731.3611279 INFO: reactions 1113\n",
      "1723602731.5566306 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602732.0298486 INFO: Default biomass: [bio1]\n",
      "1723602732.92094 INFO: reactions 1031\n",
      "1723602733.0379214 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602733.4848545 INFO: Default biomass: [bio1]\n",
      "1723602733.8009956 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602734.0284462 INFO: cpd05178 not found in model!\n",
      "1723602734.030856 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602735.080461 INFO: reactions 1191\n",
      "1723602735.3310397 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602735.3729315 WARNING: Ignoring reaction 'rxn02185_c0' since it already exists.\n",
      "1723602735.8343887 INFO: Default biomass: [bio1]\n",
      "1723602735.9519038 INFO: cpd05178 not found in model!\n",
      "1723602736.001688 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602736.2622814 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602737.1760335 INFO: reactions 1263\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.124.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.8308626262627682}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602737.359873 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602737.47572 WARNING: Ignoring reaction 'rxn00003_c0' since it already exists.\n",
      "1723602737.5544844 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602737.6749716 INFO: reactions 1188\n",
      "1723602737.8233113 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602737.895528 INFO: Default biomass: [bio1]\n",
      "1723602738.335237 INFO: Default biomass: [bio1]\n",
      "1723602738.6293476 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602738.8367636 INFO: reactions 1163\n",
      "1723602739.0185907 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602739.4917994 INFO: Default biomass: [bio1]\n",
      "1723602739.9667206 INFO: cpd05178 not found in model!\n",
      "1723602739.9688196 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602740.3034189 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602741.3900576 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602741.879561 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602744.3540878 INFO: reactions 1204\n",
      "1723602744.4254773 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602744.5305789 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602745.0471647 INFO: Default biomass: [bio1]\n",
      "1723602745.506633 INFO: reactions 1208\n",
      "1723602745.6415749 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602746.1378295 INFO: Default biomass: [bio1]\n",
      "1723602746.394994 INFO: reactions 936\n",
      "1723602746.4996672 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602746.5634193 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602746.8340561 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602746.8773785 INFO: Default biomass: [bio1]\n",
      "1723602747.5171428 INFO: cpd05178 not found in model!\n",
      "1723602747.5194128 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602748.3928866 INFO: cpd05178 not found in model!\n",
      "1723602748.3950884 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602748.8526292 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602748.9116726 WARNING: Ignoring reaction 'rxn00278_c0' since it already exists.\n",
      "1723602749.1713004 WARNING: Ignoring reaction 'rxn01011_c0' since it already exists.\n",
      "1723602749.2072518 WARNING: Ignoring reaction 'rxn00912_c0' since it already exists.\n",
      "1723602749.323138 INFO: reactions 1021\n",
      "1723602749.4594653 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602749.688438 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602749.6909745 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602749.857257 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723602749.8598046 WARNING: Ignoring reaction 'rxn00784_c0' since it already exists.\n",
      "1723602749.8819947 INFO: Default biomass: [bio1]\n",
      "1723602749.9704866 WARNING: Ignoring reaction 'rxn02185_c0' since it already exists.\n",
      "1723602749.987957 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602750.0359197 INFO: cpd05178 not found in model!\n",
      "1723602750.0383706 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602750.3000574 WARNING: Ignoring reaction 'rxn04794_c0' since it already exists.\n",
      "1723602750.3025472 WARNING: Ignoring reaction 'rxn01057_c0' since it already exists.\n",
      "1723602750.6549819 WARNING: Ignoring reaction 'rxn01333_c0' since it already exists.\n",
      "1723602750.6579251 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723602750.6638954 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602752.084226 WARNING: Ignoring reaction 'rxn00362_c0' since it already exists.\n",
      "1723602753.379016 WARNING: Ignoring reaction 'rxn10124_c0' since it already exists.\n",
      "1723602753.450002 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602753.4641178 WARNING: Ignoring reaction 'rxn01423_c0' since it already exists.\n",
      "1723602753.4666076 WARNING: Ignoring reaction 'rxn00317_c0' since it already exists.\n",
      "1723602753.58984 WARNING: Ignoring reaction 'rxn00023_c0' since it already exists.\n",
      "1723602754.0143917 WARNING: Ignoring reaction 'rxn00350_c0' since it already exists.\n",
      "1723602754.1302598 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602755.195709 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602757.5142481 INFO: cpd05178 not found in model!\n",
      "1723602757.5164652 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602757.9080234 WARNING: Ignoring reaction 'rxn02774_c0' since it already exists.\n",
      "1723602757.9812224 WARNING: Ignoring reaction 'rxn02056_c0' since it already exists.\n",
      "1723602758.1048677 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_maxbin.022.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.2962824738376544, '>': -0.12915155970958034}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602758.5095134 WARNING: Ignoring reaction 'rxn08131_c0' since it already exists.\n",
      "1723602759.8273556 WARNING: Ignoring reaction 'rxn00747_c0' since it already exists.\n",
      "1723602760.2187965 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602760.4100797 INFO: reactions 1266\n",
      "1723602760.502621 INFO: reactions 920\n",
      "1723602760.5777783 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602760.60569 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602760.985701 INFO: Default biomass: [bio1]\n",
      "1723602761.1411276 INFO: Default biomass: [bio1]\n",
      "1723602763.8061416 INFO: reactions 1163\n",
      "1723602763.935845 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602764.418845 INFO: Default biomass: [bio1]\n",
      "1723602765.1975605 INFO: reactions 1204\n",
      "1723602765.3666584 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602765.8604107 INFO: Default biomass: [bio1]\n",
      "1723602768.0404909 INFO: reactions 989\n",
      "1723602768.15095 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602768.5520413 INFO: Default biomass: [bio1]\n",
      "1723602769.4944708 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602769.6829472 INFO: reactions 1018\n",
      "1723602769.8059034 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602770.140646 INFO: cpd05178 not found in model!\n",
      "1723602770.142891 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602770.2269802 INFO: Default biomass: [bio1]\n",
      "1723602770.3377476 INFO: cpd05178 not found in model!\n",
      "1723602770.3400264 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602770.5230374 WARNING: Ignoring reaction 'rxn01366_c0' since it already exists.\n",
      "1723602771.1647673 WARNING: Ignoring reaction 'rxn05561_c0' since it already exists.\n",
      "1723602771.4322422 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.31.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.0877443226510612}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602772.042397 WARNING: Ignoring reaction 'rxn00272_c0' since it already exists.\n",
      "1723602772.528026 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602774.5493724 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602774.5494618 INFO: reactions 1205\n",
      "1723602774.7168713 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602775.2116432 INFO: Default biomass: [bio1]\n",
      "1723602777.2765672 INFO: cpd05178 not found in model!\n",
      "1723602777.2788832 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602777.7211728 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602777.7576396 WARNING: Ignoring reaction 'rxn00248_c0' since it already exists.\n",
      "1723602777.9553828 WARNING: Ignoring reaction 'rxn00225_c0' since it already exists.\n",
      "1723602777.9581041 WARNING: Ignoring reaction 'rxn00173_c0' since it already exists.\n",
      "1723602778.031381 WARNING: Ignoring reaction 'rxn02284_c0' since it already exists.\n",
      "1723602778.167961 WARNING: Ignoring reaction 'rxn00503_c0' since it already exists.\n",
      "1723602778.2438486 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602778.8299968 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602783.2045465 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602786.8466513 INFO: reactions 686\n",
      "1723602786.9455738 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602787.2307105 INFO: Default biomass: [bio1]\n",
      "1723602792.288422 INFO: cpd05178 not found in model!\n",
      "1723602792.2907653 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602793.3935285 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602795.3150282 INFO: cpd05178 not found in model!\n",
      "1723602795.3173378 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602796.0470862 INFO: reactions 877\n",
      "1723602796.0638254 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723602796.152149 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602796.1950424 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602796.2608304 WARNING: Ignoring reaction 'rxn00616_c0' since it already exists.\n",
      "1723602796.5472078 INFO: Default biomass: [bio1]\n",
      "1723602796.6935992 INFO: reactions 873\n",
      "1723602796.791402 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602796.8199651 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723602796.8224804 WARNING: Ignoring reaction 'rxn00003_c0' since it already exists.\n",
      "1723602797.1673956 INFO: Default biomass: [bio1]\n",
      "1723602797.5801704 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602801.0455232 INFO: cpd05178 not found in model!\n",
      "1723602801.0477307 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602801.2788138 WARNING: Ignoring reaction 'rxn00503_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.18.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.0436253608331834}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602802.704461 WARNING: Ignoring reaction 'rxn03194_c0' since it already exists.\n",
      "1723602802.9424036 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602804.7441735 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602805.1458328 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602809.7485857 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602810.2763832 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602817.002898 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602818.7648842 INFO: reactions 1201\n",
      "1723602818.9386318 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602819.4547694 INFO: Default biomass: [bio1]\n",
      "1723602820.5590134 INFO: reactions 1192\n",
      "1723602820.6893945 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602821.167309 INFO: Default biomass: [bio1]\n",
      "1723602825.0972981 INFO: reactions 1169\n",
      "1723602825.2667484 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602825.7352793 INFO: Default biomass: [bio1]\n",
      "1723602827.5154183 INFO: cpd05178 not found in model!\n",
      "1723602827.5175595 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602827.5537877 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602828.033764 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723602828.0515137 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602828.5513973 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602828.650798 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723602828.6618712 WARNING: Ignoring reaction 'rxn01106_c0' since it already exists.\n",
      "1723602828.6842499 INFO: reactions 1218\n",
      "1723602828.8135953 WARNING: Ignoring reaction 'rxn01057_c0' since it already exists.\n",
      "1723602828.8298047 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602828.9243793 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602829.307784 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602829.3729446 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602829.3894591 INFO: Default biomass: [bio1]\n",
      "1723602831.2405808 INFO: reactions 1072\n",
      "1723602831.4330688 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602831.8710835 INFO: Default biomass: [bio1]\n",
      "1723602833.5646539 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602833.6050508 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602833.7876055 INFO: reactions 1080\n",
      "1723602833.927042 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602834.2003388 INFO: cpd05178 not found in model!\n",
      "1723602834.202371 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602834.370659 INFO: Default biomass: [bio1]\n",
      "1723602835.6123612 WARNING: Ignoring reaction 'rxn00747_c0' since it already exists.\n",
      "1723602836.7544355 WARNING: Ignoring reaction 'rxn00467_c0' since it already exists.\n",
      "1723602837.0403278 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602837.378914 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602838.7021933 INFO: reactions 1028\n",
      "1723602838.8208168 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602839.2311137 INFO: Default biomass: [bio1]\n",
      "1723602840.064913 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602842.4461207 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602844.0915313 INFO: cpd05178 not found in model!\n",
      "1723602844.0943847 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602844.496052 WARNING: Ignoring reaction 'rxn05902_c0' since it already exists.\n",
      "1723602845.0737565 INFO: reactions 1086\n",
      "1723602845.1953928 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602845.4162157 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602845.4188209 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602845.639087 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.42.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.8536082619135388}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602845.893075 INFO: cpd05178 not found in model!\n",
      "1723602845.89539 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602846.0666637 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602846.4635644 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723602846.5528584 WARNING: Ignoring reaction 'rxn00251_c0' since it already exists.\n",
      "1723602846.5553577 WARNING: Ignoring reaction 'rxn03643_c0' since it already exists.\n",
      "1723602846.5569592 WARNING: Ignoring reaction 'rxn03644_c0' since it already exists.\n",
      "1723602846.5955722 WARNING: Ignoring reaction 'rxn00427_c0' since it already exists.\n",
      "1723602846.8821597 INFO: cpd05178 not found in model!\n",
      "1723602846.8843236 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602846.9626207 INFO: cpd05178 not found in model!\n",
      "1723602846.9645643 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602847.2531087 WARNING: Ignoring reaction 'rxn00102_c0' since it already exists.\n",
      "1723602847.3124938 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602847.4712472 WARNING: Ignoring reaction 'rxn33011_c0' since it already exists.\n",
      "1723602847.5569718 WARNING: Ignoring reaction 'rxn01301_c0' since it already exists.\n",
      "1723602847.6585913 WARNING: Ignoring reaction 'rxn31759_c0' since it already exists.\n",
      "1723602847.6918237 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602847.7626767 WARNING: Ignoring reaction 'rxn09318_c0' since it already exists.\n",
      "1723602847.9173672 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723602848.0501783 WARNING: Ignoring reaction 'rxn00467_c0' since it already exists.\n",
      "1723602848.1287818 WARNING: Ignoring reaction 'rxn00704_c0' since it already exists.\n",
      "1723602848.4260507 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.17.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.005257105918994976, '<': 1.9241427175872283}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602848.6736307 WARNING: Ignoring reaction 'rxn02774_c0' since it already exists.\n",
      "1723602848.6876147 WARNING: Ignoring reaction 'rxn02056_c0' since it already exists.\n",
      "1723602849.1763015 WARNING: Ignoring reaction 'rxn02229_c0' since it already exists.\n",
      "1723602849.270804 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602849.8523173 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602850.2882166 INFO: cpd05178 not found in model!\n",
      "1723602850.2904382 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.63.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.811170181211688}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602852.4757473 INFO: cpd05178 not found in model!\n",
      "1723602852.477884 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602852.7315013 WARNING: Ignoring reaction 'rxn01418_c0' since it already exists.\n",
      "1723602853.2566097 WARNING: Ignoring reaction 'rxn08527_c0' since it already exists.\n",
      "1723602853.3949819 WARNING: Ignoring reaction 'rxn00350_c0' since it already exists.\n",
      "1723602853.5882864 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602853.7002845 WARNING: Ignoring reaction 'rxn02774_c0' since it already exists.\n",
      "1723602853.705202 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602853.711892 WARNING: Ignoring reaction 'rxn02056_c0' since it already exists.\n",
      "1723602853.7270887 WARNING: Ignoring reaction 'rxn02287_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_concoct_out.59.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.563283337075508}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602854.5955238 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602854.8805637 INFO: cpd05178 not found in model!\n",
      "1723602854.8829274 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602855.7123275 INFO: reactions 961\n",
      "1723602855.8149276 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602856.1881583 INFO: Default biomass: [bio1]\n",
      "1723602856.229327 INFO: reactions 1066\n",
      "1723602856.348561 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602856.363699 INFO: cpd05178 not found in model!\n",
      "1723602856.365541 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602856.6317778 WARNING: Ignoring reaction 'rxn02774_c0' since it already exists.\n",
      "1723602856.7873886 INFO: Default biomass: [bio1]\n",
      "1723602857.0040977 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723602857.3393843 WARNING: Ignoring reaction 'rxn00131_c0' since it already exists.\n",
      "1723602857.529579 INFO: cpd05178 not found in model!\n",
      "1723602857.531743 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602857.6023772 WARNING: Ignoring reaction 'rxn00910_c0' since it already exists.\n",
      "1723602857.6675203 WARNING: Ignoring reaction 'rxn01301_c0' since it already exists.\n",
      "1723602857.7425766 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602857.857037 INFO: cpd05178 not found in model!\n",
      "1723602857.8591537 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602857.9678986 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602857.9704814 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602858.0740228 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602858.3108416 INFO: reactions 1143\n",
      "1723602858.4438822 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602858.5329604 WARNING: Ignoring reaction 'rxn01575_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_H2O_MG_DASTool_bins_metabat.5.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.5741709556218482}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602858.7722123 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602858.9619348 INFO: Default biomass: [bio1]\n",
      "1723602859.140831 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602862.5853424 INFO: cpd05178 not found in model!\n",
      "1723602862.5872855 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602863.119099 INFO: cpd05178 not found in model!\n",
      "1723602863.1212566 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602863.2705016 INFO: cpd05178 not found in model!\n",
      "1723602863.2733233 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.9528105167477469}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602863.651733 WARNING: Ignoring reaction 'rxn00903_c0' since it already exists.\n",
      "1723602864.1182735 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602864.1575282 WARNING: Ignoring reaction 'rxn01986_c0' since it already exists.\n",
      "1723602864.1962817 WARNING: Ignoring reaction 'rxn02933_c0' since it already exists.\n",
      "1723602864.2621095 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602864.431115 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.07933038371597392}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602864.9440827 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602864.9656875 WARNING: Ignoring reaction 'rxn00877_c0' since it already exists.\n",
      "1723602864.9903703 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602865.3819904 WARNING: Ignoring reaction 'rxn00949_c0' since it already exists.\n",
      "1723602865.3842328 WARNING: Ignoring reaction 'rxn12777_c0' since it already exists.\n",
      "1723602865.4883766 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602865.497189 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602866.269061 INFO: cpd05178 not found in model!\n",
      "1723602866.2713184 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602867.1858425 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1723602867.5960999 WARNING: Ignoring reaction 'rxn00003_c0' since it already exists.\n",
      "1723602867.9561985 WARNING: Ignoring reaction 'rxn00777_c0' since it already exists.\n",
      "1723602868.0109096 WARNING: Ignoring reaction 'rxn00362_c0' since it already exists.\n",
      "1723602868.124778 INFO: cpd05178 not found in model!\n",
      "1723602868.1269598 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602868.6744568 WARNING: Ignoring reaction 'rxn01423_c0' since it already exists.\n",
      "1723602868.741651 WARNING: Ignoring reaction 'rxn01418_c0' since it already exists.\n",
      "1723602869.2672944 WARNING: Ignoring reaction 'rxn00317_c0' since it already exists.\n",
      "1723602869.387857 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_B_H2O_MG_DASTool_bins_metabat.42.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.7746855733817796}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602870.4254036 WARNING: Ignoring reaction 'rxn00131_c0' since it already exists.\n",
      "1723602870.8986495 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602871.2977872 INFO: cpd05178 not found in model!\n",
      "1723602871.299867 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602871.8285134 WARNING: Ignoring reaction 'rxn09180_c0' since it already exists.\n",
      "1723602872.2316086 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602872.233462 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602872.64783 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n",
      "1723602872.6502442 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602873.3144693 WARNING: Ignoring reaction 'rxn00991_c0' since it already exists.\n",
      "1723602873.498508 WARNING: Ignoring reaction 'rxn01057_c0' since it already exists.\n",
      "1723602873.6678982 WARNING: Ignoring reaction 'rxn11537_c0' since it already exists.\n",
      "1723602873.6697516 WARNING: Ignoring reaction 'rxn11535_c0' since it already exists.\n",
      "1723602873.7346394 WARNING: Ignoring reaction 'rxn00359_c0' since it already exists.\n",
      "1723602873.871593 WARNING: Ignoring reaction 'rxn00677_c0' since it already exists.\n",
      "1723602874.0746439 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602878.7102323 INFO: cpd05178 not found in model!\n",
      "1723602878.712351 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602879.7311075 WARNING: Ignoring reaction 'rxn02269_c0' since it already exists.\n",
      "1723602880.0048065 INFO: cpd05178 not found in model!\n",
      "1723602880.0070674 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602880.0200062 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602880.0350907 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602880.1299732 WARNING: Ignoring reaction 'rxn02185_c0' since it already exists.\n",
      "1723602880.2204 WARNING: Ignoring reaction 'rxn00459_c0' since it already exists.\n",
      "1723602880.4342709 WARNING: Ignoring reaction 'rxn04464_c0' since it already exists.\n",
      "1723602880.528755 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602881.502606 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602890.8177602 INFO: cpd05178 not found in model!\n",
      "1723602890.8199644 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602891.1155145 INFO: cpd05178 not found in model!\n",
      "1723602891.1175256 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602891.5605419 WARNING: Ignoring reaction 'rxn02229_c0' since it already exists.\n",
      "1723602891.7087407 INFO: cpd05178 not found in model!\n",
      "1723602891.710927 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602891.7731187 WARNING: Ignoring reaction 'rxn01048_c0' since it already exists.\n",
      "1723602891.834278 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602892.0595384 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602892.816999 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602892.8195682 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602893.769053 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602893.771941 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602895.0455887 INFO: cpd05178 not found in model!\n",
      "1723602895.0477252 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602896.332575 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602896.8262546 INFO: cpd05178 not found in model!\n",
      "1723602896.8285782 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602897.1068056 WARNING: Ignoring reaction 'rxn14415_c0' since it already exists.\n",
      "1723602897.3075454 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602898.389569 WARNING: Ignoring reaction 'rxn00656_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.9.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.6634154147146776}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602899.104795 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602904.6636126 INFO: cpd05178 not found in model!\n",
      "1723602904.6659245 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602905.133018 INFO: cpd05178 not found in model!\n",
      "1723602905.1353092 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602905.4914827 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602907.0650063 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602911.3446376 INFO: cpd05178 not found in model!\n",
      "1723602911.3469443 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602912.120736 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602918.4040973 INFO: cpd05178 not found in model!\n",
      "1723602918.406411 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602918.7661378 INFO: cpd05178 not found in model!\n",
      "1723602918.7682574 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602918.786683 INFO: cpd05178 not found in model!\n",
      "1723602918.78858 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602919.146506 INFO: cpd05178 not found in model!\n",
      "1723602919.1486714 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602919.536522 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602919.7749436 INFO: cpd05178 not found in model!\n",
      "1723602919.77723 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602920.208282 WARNING: Ignoring reaction 'rxn00738_c0' since it already exists.\n",
      "1723602920.210918 WARNING: Ignoring reaction 'rxn00128_c0' since it already exists.\n",
      "1723602920.3091795 WARNING: Ignoring reaction 'rxn03004_c0' since it already exists.\n",
      "1723602920.576419 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602920.8658717 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602921.39702 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602921.836747 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602928.7765415 INFO: cpd05178 not found in model!\n",
      "1723602928.7787125 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602929.9585934 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602931.7031603 INFO: cpd05178 not found in model!\n",
      "1723602931.7054198 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.19.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': -0.6634299977470964}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602932.965933 WARNING: Ignoring reaction 'rxn00693_c0' since it already exists.\n",
      "1723602933.2055838 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602938.8474092 INFO: cpd05178 not found in model!\n",
      "1723602938.8496761 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 0.9067232169277099}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602940.2739813 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602942.1013837 INFO: cpd05178 not found in model!\n",
      "1723602942.1044497 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602942.3866284 WARNING: Ignoring reaction 'rxn00102_c0' since it already exists.\n",
      "1723602942.8622718 WARNING: Ignoring reaction 'rxn01200_c0' since it already exists.\n",
      "1723602943.6363287 WARNING: Ignoring reaction 'rxn02056_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.3999433232997875, '>': 0.6392891779314719}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602943.9412127 WARNING: Ignoring reaction 'rxn02186_c0' since it already exists.\n",
      "1723602944.3710184 WARNING: Ignoring reaction 'rxn00786_c0' since it already exists.\n",
      "1723602945.0821612 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602977.3314183 INFO: cpd05178 not found in model!\n",
      "1723602977.3336952 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602977.9645457 WARNING: Ignoring reaction 'rxn12215_c0' since it already exists.\n",
      "1723602977.9665155 WARNING: Ignoring reaction 'rxn03052_c0' since it already exists.\n",
      "1723602978.1752448 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602978.6055276 INFO: cpd05178 not found in model!\n",
      "1723602978.6075041 CRITICAL: Metabolite cpd05178 not found in model\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.1198348488905607}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602980.2487895 WARNING: Ignoring reaction 'rxn00178_c0' since it already exists.\n",
      "1723602980.8029957 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602981.3316953 INFO: cpd05178 not found in model!\n",
      "1723602981.3339248 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602981.6038096 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723602981.672911 WARNING: Ignoring reaction 'rxn00502_c0' since it already exists.\n",
      "1723602982.1005924 WARNING: Ignoring reaction 'rxn00931_c0' since it already exists.\n",
      "1723602982.5092402 WARNING: Ignoring reaction 'rxn02285_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25.contigs__.RAST Missing reaction EX_cpd00557_c0 {'<': 1.947727235570204}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602983.8832846 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723602984.5058677 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602990.3318837 INFO: cpd05178 not found in model!\n",
      "1723602990.3337936 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602991.3271735 INFO: cpd05178 not found in model!\n",
      "1723602991.329178 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602991.6368425 WARNING: Ignoring reaction 'rxn12215_c0' since it already exists.\n",
      "1723602991.672429 WARNING: Ignoring reaction 'rxn04464_c0' since it already exists.\n",
      "1723602991.7654483 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602992.3959608 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723602993.1181664 INFO: cpd05178 not found in model!\n",
      "1723602993.1201727 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723602993.502509 WARNING: Ignoring reaction 'rxn00931_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.12.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.1293953917264785}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723602993.7184026 WARNING: Ignoring reaction 'rxn02285_c0' since it already exists.\n",
      "1723602994.0001585 WARNING: Ignoring reaction 'rxn00929_c0' since it already exists.\n",
      "1723602995.4736466 WARNING: Ignoring reaction 'rxn00551_c0' since it already exists.\n",
      "1723602995.5448582 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n",
      "1723603010.6784449 INFO: cpd05178 not found in model!\n",
      "1723603010.6804316 CRITICAL: Metabolite cpd05178 not found in model\n",
      "1723603011.2326171 WARNING: Ignoring reaction 'rxn01048_c0' since it already exists.\n",
      "1723603011.326532 WARNING: Ignoring reaction 'rxn00427_c0' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.8.contigs__.RAST Missing reaction EX_cpd00557_c0 {'>': 1.0205293821453087}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1723603012.8832467 INFO: Setting FBAModel computed_attributes to mdlutl attributes\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "\n",
    "def gapfillModels(arguments):\n",
    "    (contents, name_to_mag, finished_models, metabolites, reactions_to_add, index) = arguments\n",
    "\n",
    "    \n",
    "    # print status\n",
    "    from multiprocess import current_process\n",
    "    pid = current_process().name\n",
    "    print(pid)\n",
    "    \n",
    "    for OGname, fluxes in contents:\n",
    "        # print(content[0], content[1].key())\n",
    "        # OGname, fluxes = content\n",
    "        name = name_to_mag[OGname.replace(\".pg.G.D\", '')]\n",
    "        if name not in finished_models:\n",
    "            mdlutl = util.msrecon.get_model(name.replace(\".contigs__.RAST\",\".pangenome.RAST.glm4ec.pyr.O2\"),186678)\n",
    "            types = [\"uptake\",\"excretion\",\"growth\"]\n",
    "            reactions_to_add[name] = {}\n",
    "            for currtype in types:\n",
    "                problist = []\n",
    "                for metabolite in metabolites:\n",
    "                    if metabolite in consolidated_gf_results[OGname][currtype] and \"new_probability\" in consolidated_gf_results[OGname][currtype][metabolite]:\n",
    "                        if consolidated_gf_results[OGname][currtype][metabolite][\"new_probability\"] > 0:\n",
    "                            problist.append(consolidated_gf_results[OGname][currtype][metabolite][\"new_probability\"])\n",
    "                if len(problist) > 0:\n",
    "                    ave = array(problist).mean()\n",
    "                    stdev = array(problist).std()\n",
    "                    for metabolite in metabolites:\n",
    "                        if metabolite in consolidated_gf_results[OGname][currtype] and \"new_probability\" in consolidated_gf_results[OGname][currtype][metabolite]:\n",
    "                            if consolidated_gf_results[OGname][currtype][metabolite][\"new_probability\"] > 0:\n",
    "                                consolidated_gf_results[OGname][currtype][metabolite][\"zscore\"] = (consolidated_gf_results[OGname][currtype][metabolite][\"new_probability\"]-ave)/stdev\n",
    "                                if consolidated_gf_results[OGname][currtype][metabolite][\"zscore\"] >= -1 and \"reactions\" in consolidated_gf_results[OGname][currtype][metabolite]:\n",
    "                                    for rxn in consolidated_gf_results[OGname][currtype][metabolite][\"reactions\"]:\n",
    "                                        direction = rxn[0:1]\n",
    "                                        rxnid = rxn[1:]\n",
    "                                        if rxnid not in mdlutl.model.reactions:\n",
    "                                            if rxnid not in reactions_to_add[name]:\n",
    "                                                reactions_to_add[name][rxnid] = {}\n",
    "                                            reactions_to_add[name][rxnid][direction] = consolidated_gf_results[OGname][currtype][metabolite][\"zscore\"]\n",
    "                                        elif direction == \">\":\n",
    "                                            if mdlutl.model.reactions.get_by_id(rxnid).upper_bound <= 0:\n",
    "                                                if rxnid not in reactions_to_add[name]:\n",
    "                                                    reactions_to_add[name][rxnid] = {}\n",
    "                                                reactions_to_add[name][rxnid][direction] = consolidated_gf_results[OGname][currtype][metabolite][\"zscore\"]\n",
    "                                        elif direction == \"<\":\n",
    "                                            if mdlutl.model.reactions.get_by_id(rxnid).lower_bound >= 0:\n",
    "                                                if rxnid not in reactions_to_add[name]:\n",
    "                                                    reactions_to_add[name][rxnid] = {}\n",
    "                                                reactions_to_add[name][rxnid][direction] = consolidated_gf_results[OGname][currtype][metabolite][\"zscore\"]\n",
    "            util.save(\"cleaned_SMIPPs/gf_phenotype_results\",consolidated_gf_results)\n",
    "            msgapfill = MSGapfill(\n",
    "                mdlutl,\n",
    "                [util.msrecon.get_template(mdlutl.model.template_ref)],\n",
    "                [],\n",
    "                [],\n",
    "                blacklist=[],\n",
    "                default_target=\"bio1\",\n",
    "                minimum_obj=0.01,\n",
    "                base_media=None,\n",
    "                base_media_target_element=None\n",
    "            )\n",
    "            for cpd in metabolites:\n",
    "                if \"EX_\"+cpd+\"_e0\" not in msgapfill.gfmodelutl.model.reactions:\n",
    "                    transport = msgapfill.gfmodelutl.add_transport_and_exchange_for_metabolite(cpd,direction=\"=\",prefix=\"trans\",override=False)\n",
    "            for rxnid in reactions_to_add[name]:\n",
    "                if rxnid in msgapfill.gfmodel.reactions:\n",
    "                    rxn = msgapfill.gfmodel.reactions.get_by_id(rxnid)\n",
    "                    rxn = rxn.copy()\n",
    "                    mdlutl.model.add_reactions([rxn])\n",
    "                    rxn.upper_bound = 0\n",
    "                    rxn.lower_bound = 0\n",
    "                    if \">\" in reactions_to_add[name][rxnid]:\n",
    "                        rxn.upper_bound = 100\n",
    "                    if \"<\" in reactions_to_add[name][rxnid]:\n",
    "                        rxn.lower_bound = -100\n",
    "                else:\n",
    "                    print(name,\"Missing reaction\",rxnid,reactions_to_add[name][rxnid])\n",
    "            util.msrecon.save_model(mdlutl,186678,name.replace(\".contigs__.RAST\",\".pangenome.RAST.glm4ec.pyr.O2.gf\"))\n",
    "            finished_models[name] = 1\n",
    "        util.save(f\"finished_models_{index}\",finished_models)\n",
    "        util.save(f\"reactions_to_add_{index}\",reactions_to_add)\n",
    "\n",
    "\n",
    "# define the parameters\n",
    "consolidated_gf_results = util.load(\"cleaned_SMIPPs/gf_phenotype_results\")\n",
    "name_to_mag = util.load(\"name_to_mag\")\n",
    "finished_models = util.load(\"finished_models\",{})\n",
    "metabolites = util.load(\"metabolites\")\n",
    "reactions_to_add = util.load(\"reactions_to_add\", {})\n",
    "\n",
    "# run the parallelized code\n",
    "parallelize = True\n",
    "if parallelize:\n",
    "    from multiprocess import Pool\n",
    "    from os import cpu_count\n",
    "    from numpy import array_split, array\n",
    "    mag_results_sets = array_split(list(consolidated_gf_results.items()), int(len(consolidated_gf_results)/4))\n",
    "    \n",
    "    # the MAGs are partitioned into groups of 4, and a thread is created to handle each group \n",
    "    pool_count = min(cpu_count(), len(mag_results_sets))\n",
    "    print(f\"{pool_count} threads will be created\")\n",
    "    pool = Pool(pool_count)\n",
    "    args = [[content, name_to_mag, finished_models, metabolites, reactions_to_add, index] for index, content in enumerate(mag_results_sets)]\n",
    "    list_of_outputs = pool.map(gapfillModels, args)\n",
    "else:\n",
    "    list_of_outputs = buildCommModel(list(mag_abundances.items()), mag_results_sets)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### combining the multiple gapfilling iterations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "77\n",
      "77\n"
     ]
    }
   ],
   "source": [
    "from glob import glob\n",
    "from os import remove\n",
    "from json import dump, load\n",
    "\n",
    "print(len(glob(\"datacache/finished_models_*.json\")))\n",
    "\n",
    "finished_models_tot = {}\n",
    "for finished_models in glob(\"datacache/finished_models_*.json\"):\n",
    "    with open(finished_models, 'r') as jsonIn:\n",
    "        finished_models = load(jsonIn)\n",
    "    finished_models_tot.update(finished_models)\n",
    "    \n",
    "with open(\"datacache/finished_models.json\", 'w') as jsonOut:\n",
    "    dump(finished_models_tot, jsonOut, indent=3)\n",
    "    \n",
    "\n",
    "print(len(glob(\"datacache/reactions_to_add_*.json\")))\n",
    "\n",
    "reactions_to_add_tot = {}\n",
    "for reactions_to_add in glob(\"datacache/reactions_to_add_*.json\"):\n",
    "    with open(reactions_to_add, 'r') as jsonIn:\n",
    "        reactions_to_add = load(jsonIn)\n",
    "    reactions_to_add_tot.update(reactions_to_add)\n",
    "    \n",
    "with open(\"datacache/reactions_to_add.json\", 'w') as jsonOut:\n",
    "    dump(reactions_to_add_tot, jsonOut, indent=3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## check that the gapfilling worked"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['__bool__',\n",
       " '__class__',\n",
       " '__delattr__',\n",
       " '__dir__',\n",
       " '__doc__',\n",
       " '__eq__',\n",
       " '__format__',\n",
       " '__ge__',\n",
       " '__getattribute__',\n",
       " '__getstate__',\n",
       " '__gt__',\n",
       " '__hash__',\n",
       " '__init__',\n",
       " '__init_subclass__',\n",
       " '__le__',\n",
       " '__lt__',\n",
       " '__ne__',\n",
       " '__new__',\n",
       " '__reduce__',\n",
       " '__reduce_ex__',\n",
       " '__repr__',\n",
       " '__setattr__',\n",
       " '__sizeof__',\n",
       " '__str__',\n",
       " '__subclasshook__']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "\n",
    "dir(util.msrecon)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n",
      "Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_metabat.20 does not have a gapfilled model\n",
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.9 does not have a gapfilled model\n",
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.53 does not have a gapfilled model\n",
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11 does not have a gapfilled model\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "from numpy import std, mean\n",
    "\n",
    "# acquire the list of gapfilled content\n",
    "salternModels = util.msrecon.kbase_api.list_objects(186678, object_type=\"KBaseFBA.FBAModel\", include_metadata=True)\n",
    "modelRxnCounts = {}\n",
    "for model in salternModels:\n",
    "    if \"O2\" not in model[1]:  continue\n",
    "    if \"gf\" in model[1]:\n",
    "        modelID = model[1].replace(\".pangenome.RAST.glm4ec.pyr.O2.gf\", '')\n",
    "        if modelID not in modelRxnCounts:\n",
    "            modelRxnCounts[modelID] = {\"gapfilled\": model[-1]['Number reactions']}\n",
    "        else:\n",
    "            modelRxnCounts[modelID].update({\"gapfilled\": model[-1]['Number reactions']})\n",
    "    else:\n",
    "        modelID = model[1].replace(\".pangenome.RAST.glm4ec.pyr.O2\", '')\n",
    "        if modelID not in modelRxnCounts:\n",
    "            modelRxnCounts[modelID] = {\"base\": model[-1]['Number reactions']}\n",
    "        else:\n",
    "            modelRxnCounts[modelID].update({\"base\": model[-1]['Number reactions']})\n",
    "    \n",
    "# determine the gapfilling statistics \n",
    "totRxns, reactionStats = {\"gapfilled\": [], \"base\": []}, {\"gapfilled\": {}, \"base\": {}}\n",
    "badModels = []\n",
    "for model, content in modelRxnCounts.items():\n",
    "    totRxns[\"base\"].append(content[\"base\"])\n",
    "    if \"gapfilled\" not in content:\n",
    "        print(f\"{model} does not have a gapfilled model\")\n",
    "        badModels.append(model)\n",
    "        continue\n",
    "    totRxns[\"gapfilled\"].append(content[\"gapfilled\"])\n",
    "\n",
    "for category, rxnNums in totRxns.items():\n",
    "    vals = [int(x) for x in rxnNums]\n",
    "    reactionStats[category] = {\n",
    "        \"num\": len(rxnNums),\n",
    "        \"ave\": mean(vals),\n",
    "        \"std\": std(vals)\n",
    "    }\n",
    "# print(totRxns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_metabat.20',\n",
       " 'Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.9',\n",
       " 'Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.53',\n",
       " 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11']"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "{'gapfilled': {'num': 304,\n",
       "  'ave': 1226.3355263157894,\n",
       "  'std': 153.49742600614664},\n",
       " 'base': {'num': 308, 'ave': 1009.2175324675325, 'std': 157.55117108671203}}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(badModels, reactionStats)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Building and saving community model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### abundaces adjustment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.9999999999999994\n",
      "1.0000000000000002\n",
      "1.000000000000001\n",
      "0.9999999999999969\n",
      "0.9999999999999988\n",
      "0.9999999999999993\n",
      "0.9999999999999999\n",
      "1.0\n",
      "1.0000000000000002\n",
      "0.9999999999999999\n",
      "1.0000000000000004\n",
      "1.0000000000000009\n",
      "1.0000000000000009\n",
      "1.0000000000000007\n",
      "0.9999999999999996\n",
      "1.0000000000000004\n",
      "1.0\n",
      "0.9999999999999996\n",
      "1.0\n",
      "1.0\n",
      "1.0000000000000007\n",
      "0.9999999999999998\n",
      "0.9999999999999998\n",
      "1.0000000000000007\n"
     ]
    }
   ],
   "source": [
    "from json import load, dump\n",
    "\n",
    "abundancesDic = load(open(\"mag_abundances.json\"))\n",
    "refinedAbundances = {}\n",
    "for sample, abundances in abundancesDic.items():\n",
    "    refinedAbundances[sample] = {}\n",
    "    abundances.pop(\"unbinned_reads\")\n",
    "    totAbundance = sum(list(abundances.values()))\n",
    "    for mag, abundance in abundances.items():\n",
    "        refinedAbundances[sample][mag] = abundance/totAbundance\n",
    "    print(sum(list(refinedAbundances[sample].values())))\n",
    "\n",
    "dump(refinedAbundances, open(\"datacache/refined_MAG_abundances.json\", 'w'), indent=3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## create the models"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### APF version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python version 3.11.1\n",
      "KBBaseModules 0.0.1\n",
      "modelseedpy 0.3.3\n",
      "cobrakbase 0.4.0\n",
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n",
      "MainProcess\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724795223.8013704 INFO: metabolites 1261\n",
      "1724795224.3416324 INFO: reactions 1292\n",
      "1724795224.4795008 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724795225.3879337 INFO: Default biomass: [bio1]\n",
      "1724795227.0119321 INFO: metabolites 1196\n",
      "1724795227.613509 INFO: reactions 1264\n",
      "1724795227.761144 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724795228.6444411 INFO: Default biomass: [bio1]\n",
      "1724795229.1647696 INFO: metabolites 1285\n",
      "1724795229.8651652 INFO: reactions 1315\n",
      "1724795230.0185068 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724795230.8617983 INFO: Default biomass: [bio1]\n",
      "1724795231.3667727 INFO: metabolites 1243\n",
      "1724795231.971922 INFO: reactions 1303\n",
      "1724795232.1094666 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724795232.6112962 INFO: Default biomass: [bio1]\n",
      "1724795233.6646695 INFO: metabolites 1140\n",
      "1724795234.1556923 INFO: reactions 1119\n",
      "1724795234.277382 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724795234.706305 INFO: Default biomass: [bio1]\n",
      "1724795235.1842585 INFO: metabolites 1071\n",
      "1724795236.1248348 INFO: reactions 1103\n",
      "1724795236.2491558 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724795236.661702 INFO: Default biomass: [bio1]\n",
      "1724795237.1517704 INFO: metabolites 1068\n",
      "1724795237.656287 INFO: reactions 1082\n",
      "1724795237.7894754 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724795238.199376 INFO: Default biomass: [bio1]\n",
      "1724795238.8256588 INFO: metabolites 1232\n",
      "1724795239.9788117 INFO: reactions 1353\n",
      "1724795240.1314807 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724795240.651454 INFO: Default biomass: [bio1]\n",
      "1724795242.453638 INFO: metabolites 1154\n",
      "1724795242.9724193 INFO: reactions 1156\n",
      "1724795243.098632 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724795243.5398927 INFO: Default biomass: [bio1]\n",
      "1724795244.6687157 INFO: metabolites 1207\n",
      "1724795245.2972856 INFO: reactions 1267\n",
      "1724795245.4354908 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724795245.9286313 INFO: Default biomass: [bio1]\n",
      "1724795247.2698143 INFO: metabolites 1323\n",
      "1724795248.0494442 INFO: reactions 1349\n",
      "1724795248.2091973 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724795248.7625852 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "MSModelUtil.__init__() got an unexpected keyword argument 'copy'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[1], line 51\u001b[0m\n\u001b[1;32m     49\u001b[0m     list_of_outputs \u001b[38;5;241m=\u001b[39m pool\u001b[38;5;241m.\u001b[39mmap(buildCommModel, args)\n\u001b[1;32m     50\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m---> 51\u001b[0m     list_of_outputs \u001b[38;5;241m=\u001b[39m \u001b[43mbuildCommModel\u001b[49m\u001b[43m(\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;28;43mlist\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mmag_abundances\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mitems\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmag_list\u001b[49m\u001b[43m]\u001b[49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[0;32mIn[1], line 24\u001b[0m, in \u001b[0;36mbuildCommModel\u001b[0;34m(arguments)\u001b[0m\n\u001b[1;32m     22\u001b[0m         member_names\u001b[38;5;241m.\u001b[39mappend(mag)\n\u001b[1;32m     23\u001b[0m         member_abundances[mag] \u001b[38;5;241m=\u001b[39m abundance\n\u001b[0;32m---> 24\u001b[0m comm_model \u001b[38;5;241m=\u001b[39m \u001b[43mbuild_from_species_models\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     25\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmember_models\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     26\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmodel_id\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     27\u001b[0m \u001b[43m    \u001b[49m\u001b[43mname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     28\u001b[0m \u001b[43m    \u001b[49m\u001b[43mabundances\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmember_abundances\u001b[49m\n\u001b[1;32m     29\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     30\u001b[0m cobra\u001b[38;5;241m.\u001b[39mio\u001b[38;5;241m.\u001b[39msave_json_model(comm_model, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmodels/\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39msample\u001b[38;5;241m+\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m_APF.json\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[0;32m/scratch/shared/ModelSEEDpy_CSH_APF/modelseedpy/community/commhelper.py:83\u001b[0m, in \u001b[0;36mbuild_from_species_models\u001b[0;34m(org_models, model_id, name, abundances, standardize, MSmodel, commkinetics, copy_models, printing)\u001b[0m\n\u001b[1;32m     81\u001b[0m member_biomasses \u001b[38;5;241m=\u001b[39m {}\n\u001b[1;32m     82\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m model_index, org_model \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28menumerate\u001b[39m(models):\n\u001b[0;32m---> 83\u001b[0m     model_util \u001b[38;5;241m=\u001b[39m \u001b[43mMSModelUtil\u001b[49m\u001b[43m(\u001b[49m\u001b[43morg_model\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy_models\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     84\u001b[0m     model_reaction_ids \u001b[38;5;241m=\u001b[39m [rxn\u001b[38;5;241m.\u001b[39mid \u001b[38;5;28;01mfor\u001b[39;00m rxn \u001b[38;5;129;01min\u001b[39;00m model_util\u001b[38;5;241m.\u001b[39mmodel\u001b[38;5;241m.\u001b[39mreactions]\n\u001b[1;32m     85\u001b[0m     model_index \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m\n",
      "\u001b[0;31mTypeError\u001b[0m: MSModelUtil.__init__() got an unexpected keyword argument 'copy'"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "\n",
    "\n",
    "\n",
    "def buildCommModel(arguments):\n",
    "    (mag_abundances, mag_list) = arguments\n",
    "    \n",
    "    # print status\n",
    "    from multiprocess import current_process\n",
    "    pid = current_process().name\n",
    "    print(pid)\n",
    "    \n",
    "    # build the community model from the members\n",
    "    for info in mag_abundances:\n",
    "        sample, abundances = info\n",
    "        member_models = []\n",
    "        member_names = []\n",
    "        member_abundances = {}\n",
    "        ## create the member abundances dictionary\n",
    "        for mag, abundance in abundances.items():\n",
    "            if abundance < 0.01:  continue\n",
    "            mdlutl = util.msrecon.get_model(mag+\".pangenome.RAST.glm4ec.pyr.O2.gf\",186678)\n",
    "            member_models.append(mdlutl.model)\n",
    "            member_names.append(mag)\n",
    "            member_abundances[mag] = abundance\n",
    "        comm_model = build_from_species_models(\n",
    "            member_models,\n",
    "            model_id=sample,\n",
    "            name=sample,\n",
    "            abundances=member_abundances\n",
    "        )\n",
    "        cobra.io.save_json_model(comm_model, \"models/\"+sample+'_APF.json')\n",
    "    \n",
    "\n",
    "import cobra\n",
    "mag_list = util.load(\"mag_list\")\n",
    "mag_abundances = util.load(\"refined_MAG_abundances\")\n",
    "\n",
    "# run the parallelized code\n",
    "parallelize = False\n",
    "if parallelize:\n",
    "    from multiprocess import Pool\n",
    "    from os import cpu_count\n",
    "    from numpy import array_split\n",
    "    \n",
    "    # the MAGs are partitioned into groups of 4, and a thread is created to handle each group \n",
    "    pool_count = min(cpu_count(), len(mag_abundances))\n",
    "    print(f\"{pool_count} threads will be created\")\n",
    "    pool = Pool(pool_count)\n",
    "    args = [[[abundances], mag_list] for abundances in list(mag_abundances.items())]\n",
    "    list_of_outputs = pool.map(buildCommModel, args)\n",
    "else:\n",
    "    list_of_outputs = buildCommModel([list(mag_abundances.items()), mag_list])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CSH version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python version 3.11.1\n",
      "KBBaseModules 0.0.1\n",
      "modelseedpy 0.3.3\n",
      "cobrakbase 0.4.0\n",
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n",
      "24 threads will be created\n",
      "ForkPoolWorker-1\n",
      "\n",
      "\n",
      "ForkPoolWorker-2ForkPoolWorker-3\n",
      "ForkPoolWorker-4ForkPoolWorker-5\n",
      "ForkPoolWorker-6\n",
      "ForkPoolWorker-7\n",
      "ForkPoolWorker-8\n",
      "ForkPoolWorker-9\n",
      "ForkPoolWorker-10\n",
      "ForkPoolWorker-11\n",
      "ForkPoolWorker-12\n",
      "ForkPoolWorker-13\n",
      "ForkPoolWorker-14\n",
      "ForkPoolWorker-15\n",
      "ForkPoolWorker-16\n",
      "ForkPoolWorker-17\n",
      "ForkPoolWorker-18\n",
      "ForkPoolWorker-19\n",
      "ForkPoolWorker-20\n",
      "ForkPoolWorker-21\n",
      "ForkPoolWorker-22\n",
      "ForkPoolWorker-23\n",
      "ForkPoolWorker-24\n",
      "The Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.53.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\n",
      "The Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.53.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811750.6927388 INFO: metabolites 1261\n",
      "1724811750.828401 INFO: metabolites 1170\n",
      "1724811750.8623137 INFO: metabolites 1261\n",
      "1724811750.8788252 INFO: metabolites 1170\n",
      "1724811750.9430435 INFO: metabolites 1170\n",
      "1724811750.9522645 INFO: metabolites 1170\n",
      "1724811750.9822638 INFO: metabolites 1261\n",
      "1724811750.9978032 INFO: metabolites 1170\n",
      "1724811751.0060813 INFO: metabolites 1152\n",
      "1724811751.0129082 INFO: metabolites 1152\n",
      "1724811751.1252513 INFO: metabolites 1305\n",
      "1724811751.1525526 INFO: metabolites 1140\n",
      "1724811751.1589227 INFO: metabolites 1261\n",
      "1724811751.1681464 INFO: metabolites 1261\n",
      "1724811751.182422 INFO: metabolites 1327\n",
      "1724811751.1877236 INFO: metabolites 1261\n",
      "1724811751.193667 INFO: metabolites 1152\n",
      "1724811751.1979263 INFO: metabolites 1266\n",
      "1724811751.2181718 INFO: metabolites 1177\n",
      "1724811751.2221508 INFO: metabolites 1024\n",
      "1724811751.235486 INFO: metabolites 1318\n",
      "1724811751.2357593 INFO: reactions 1292\n",
      "1724811751.3301678 INFO: metabolites 1261\n",
      "1724811751.3726609 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.4079125 INFO: reactions 1292\n",
      "1724811751.457655 INFO: reactions 1212\n",
      "1724811751.459659 INFO: reactions 1212\n",
      "1724811751.4616766 INFO: metabolites 1138\n",
      "1724811751.511094 INFO: reactions 1212\n",
      "1724811751.5238235 INFO: metabolites 1318\n",
      "1724811751.5313103 INFO: reactions 1212\n",
      "1724811751.5370705 INFO: reactions 1180\n",
      "1724811751.5404541 INFO: reactions 1180\n",
      "1724811751.5494878 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.5838146 INFO: reactions 1212\n",
      "1724811751.586748 INFO: reactions 1292\n",
      "1724811751.612146 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.6146727 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.6596842 INFO: reactions 1051\n",
      "1724811751.6651187 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.6667948 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.66817 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.687093 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.7060144 INFO: reactions 1292\n",
      "1724811751.715247 INFO: reactions 1292\n",
      "1724811751.7331748 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.734045 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.7591026 INFO: reactions 1389\n",
      "1724811751.770831 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.7712896 INFO: reactions 1180\n",
      "1724811751.7928197 INFO: reactions 1119\n",
      "1724811751.8200943 INFO: reactions 1207\n",
      "1724811751.8443768 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.858532 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.882765 INFO: reactions 1292\n",
      "1724811751.8981261 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.9034338 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.9040725 INFO: reactions 1384\n",
      "1724811751.9177685 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811751.9553368 INFO: reactions 1537\n",
      "1724811751.9893725 INFO: reactions 1306\n",
      "1724811752.0090399 INFO: reactions 1242\n",
      "1724811752.0206447 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811752.0241246 INFO: reactions 1375\n",
      "1724811752.07293 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811752.1206744 INFO: Default biomass: [bio1]\n",
      "1724811752.1415083 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811752.1841464 INFO: Default biomass: [bio1]\n",
      "1724811752.2410026 INFO: reactions 1537\n",
      "1724811752.2995646 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811752.3263013 INFO: Default biomass: [bio1]\n",
      "1724811752.3601904 INFO: Default biomass: [bio1]\n",
      "1724811752.4052455 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811752.4232087 INFO: Default biomass: [bio1]\n",
      "1724811752.433629 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811752.4501033 INFO: Default biomass: [bio1]\n",
      "1724811752.4518018 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.53.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811752.5107121 INFO: Default biomass: [bio1]\n",
      "1724811752.5191655 INFO: Default biomass: [bio1]\n",
      "1724811752.5218232 INFO: Default biomass: [bio1]\n",
      "1724811752.5494845 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811752.6001022 INFO: Default biomass: [bio1]\n",
      "1724811752.6484835 INFO: Default biomass: [bio1]\n",
      "1724811752.686598 INFO: Default biomass: [bio1]\n",
      "1724811752.6913562 INFO: Default biomass: [bio1]\n",
      "1724811752.7328773 INFO: metabolites 1196\n",
      "1724811752.7489865 INFO: Default biomass: [bio1]\n",
      "1724811752.756853 INFO: Default biomass: [bio1]\n",
      "1724811752.7751606 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811752.7763119 INFO: Default biomass: [bio1]\n",
      "1724811752.8619378 INFO: Default biomass: [bio1]\n",
      "1724811752.9340322 INFO: metabolites 1134\n",
      "1724811752.9383156 INFO: Default biomass: [bio1]\n",
      "1724811752.9387934 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811752.9419532 INFO: Default biomass: [bio1]\n",
      "1724811752.9809802 INFO: Default biomass: [bio1]\n",
      "1724811752.9850674 INFO: metabolites 1298\n",
      "1724811752.991401 INFO: metabolites 921\n",
      "1724811753.0225034 INFO: metabolites 1266\n",
      "1724811753.0421283 INFO: metabolites 1134\n",
      "1724811753.0528502 INFO: metabolites 1152\n",
      "1724811753.1294456 INFO: metabolites 1132\n",
      "1724811753.1612644 INFO: Default biomass: [bio1]\n",
      "1724811753.164748 INFO: metabolites 1191\n",
      "1724811753.2265143 INFO: metabolites 1224\n",
      "1724811753.2419865 INFO: metabolites 1283\n",
      "1724811753.2765577 INFO: metabolites 1298\n",
      "1724811753.3054879 INFO: metabolites 1243\n",
      "1724811753.3447993 INFO: reactions 1264\n",
      "1724811753.3579662 INFO: reactions 974\n",
      "1724811753.3579788 INFO: metabolites 1266\n",
      "1724811753.3716137 INFO: metabolites 1266\n",
      "1724811753.370757 INFO: Default biomass: [bio1]\n",
      "1724811753.398067 INFO: reactions 1252\n",
      "1724811753.458559 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811753.463253 INFO: metabolites 1266\n",
      "1724811753.521533 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811753.557953 INFO: metabolites 1196\n",
      "1724811753.5669808 INFO: reactions 1180\n",
      "1724811753.5834537 INFO: reactions 1252\n",
      "1724811753.585652 INFO: metabolites 1351\n",
      "1724811753.600945 INFO: reactions 1419\n",
      "1724811753.606164 INFO: metabolites 1205\n",
      "1724811753.6066682 INFO: metabolites 1261\n",
      "1724811753.6372795 INFO: metabolites 1355\n",
      "1724811753.6661456 INFO: reactions 1353\n",
      "1724811753.6787097 INFO: reactions 1151\n",
      "1724811753.6909733 INFO: metabolites 1177\n",
      "1724811753.6953275 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811753.7055404 INFO: reactions 1375\n",
      "1724811753.7101407 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811753.7639756 INFO: reactions 1278\n",
      "1724811753.7648978 INFO: metabolites 1145\n",
      "1724811753.8070467 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811753.8217177 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811753.8255105 INFO: Default biomass: [bio1]\n",
      "1724811753.8486223 INFO: reactions 1330\n",
      "1724811753.895756 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811753.8975315 INFO: reactions 1303\n",
      "1724811753.9178033 INFO: reactions 1419\n",
      "1724811753.967097 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811753.9911582 INFO: metabolites 1351\n",
      "1724811754.0633612 INFO: reactions 1375\n",
      "1724811754.0807152 INFO: reactions 1375\n",
      "1724811754.1428964 INFO: reactions 1375\n",
      "1724811754.1784577 INFO: reactions 1301\n",
      "1724811754.224267 INFO: metabolites 881\n",
      "1724811754.244656 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.247859 INFO: Default biomass: [bio1]\n",
      "1724811754.2915783 INFO: reactions 1207\n",
      "1724811754.3033173 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.33261 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.41868 INFO: reactions 1264\n",
      "1724811754.475521 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.4864864 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.488176 INFO: Default biomass: [bio1]\n",
      "1724811754.5487893 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.5613883 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.5663626 INFO: Default biomass: [bio1]\n",
      "1724811754.5700207 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.5844672 INFO: Default biomass: [bio1]\n",
      "1724811754.6053407 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.6578104 INFO: Default biomass: [bio1]\n",
      "1724811754.715641 INFO: metabolites 1138\n",
      "1724811754.7303376 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.7379239 INFO: reactions 1306\n",
      "1724811754.7762468 INFO: Default biomass: [bio1]\n",
      "1724811754.785077 INFO: Default biomass: [bio1]\n",
      "1724811754.7860935 INFO: reactions 1513\n",
      "1724811754.797892 INFO: reactions 1233\n",
      "1724811754.8004777 INFO: Default biomass: [bio1]\n",
      "1724811754.8008137 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.8164148 INFO: Default biomass: [bio1]\n",
      "1724811754.8985531 INFO: Default biomass: [bio1]\n",
      "1724811754.9248896 INFO: reactions 1494\n",
      "1724811754.928094 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.9294934 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811754.9775374 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811755.0108342 INFO: metabolites 1285\n",
      "1724811755.0112295 INFO: Default biomass: [bio1]\n",
      "1724811755.0252318 INFO: metabolites 1134\n",
      "1724811755.0730827 INFO: Default biomass: [bio1]\n",
      "1724811755.0780509 INFO: Default biomass: [bio1]\n",
      "1724811755.0849645 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811755.1052384 INFO: reactions 921\n",
      "1724811755.1098382 INFO: Default biomass: [bio1]\n",
      "1724811755.119681 INFO: Default biomass: [bio1]\n",
      "1724811755.1246572 INFO: metabolites 1155\n",
      "1724811755.1497242 INFO: reactions 1513\n",
      "1724811755.157174 INFO: metabolites 1145\n",
      "1724811755.2090547 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811755.259274 INFO: reactions 1242\n",
      "1724811755.2609696 INFO: metabolites 1145\n",
      "1724811755.267244 INFO: Default biomass: [bio1]\n",
      "1724811755.294547 INFO: metabolites 1224\n",
      "1724811755.300418 INFO: metabolites 1140\n",
      "1724811755.3411741 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811755.3852007 INFO: Default biomass: [bio1]\n",
      "1724811755.3953831 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811755.3965383 INFO: Default biomass: [bio1]\n",
      "1724811755.4408832 INFO: metabolites 1196\n",
      "1724811755.441673 INFO: metabolites 1351\n",
      "1724811755.454962 INFO: Default biomass: [bio1]\n",
      "1724811755.510842 INFO: reactions 1252\n",
      "1724811755.537209 INFO: Default biomass: [bio1]\n",
      "1724811755.5688376 INFO: Default biomass: [bio1]\n",
      "1724811755.5876791 INFO: Default biomass: [bio1]\n",
      "1724811755.6069622 INFO: reactions 1354\n",
      "1724811755.628858 INFO: metabolites 1196\n",
      "1724811755.637621 INFO: reactions 1233\n",
      "1724811755.6442714 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811755.6517613 INFO: metabolites 1327\n",
      "1724811755.6541378 INFO: metabolites 1220\n",
      "1724811755.6885118 INFO: Default biomass: [bio1]\n",
      "1724811755.7069993 INFO: reactions 1315\n",
      "1724811755.7410195 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811755.7431529 INFO: metabolites 1196\n",
      "1724811755.7515078 INFO: reactions 1233\n",
      "1724811755.7524583 INFO: metabolites 1351\n",
      "1724811755.7615607 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811755.7970374 INFO: reactions 1119\n",
      "1724811755.8047924 INFO: metabolites 1282\n",
      "1724811755.8351035 INFO: reactions 1278\n",
      "1724811755.8597999 INFO: Default biomass: [bio1]\n",
      "1724811755.8789253 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811755.9163275 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811755.973904 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811755.9958777 INFO: Default biomass: [bio1]\n",
      "1724811756.0434766 INFO: reactions 1264\n",
      "1724811756.1134005 INFO: Default biomass: [bio1]\n",
      "1724811756.1888778 INFO: reactions 1199\n",
      "1724811756.2260268 INFO: Default biomass: [bio1]\n",
      "1724811756.247388 INFO: reactions 1264\n",
      "1724811756.334239 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811756.3351252 INFO: reactions 1264\n",
      "1724811756.3382897 INFO: Default biomass: [bio1]\n",
      "1724811756.3771045 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811756.3816125 INFO: metabolites 1033\n",
      "1724811756.48075 INFO: metabolites 1145\n",
      "1724811756.495327 INFO: metabolites 1224\n",
      "1724811756.4984593 INFO: Default biomass: [bio1]\n",
      "1724811756.5278318 INFO: metabolites 1205\n",
      "1724811756.5678887 INFO: metabolites 1196\n",
      "1724811756.5816538 INFO: metabolites 1145\n",
      "1724811756.5937479 INFO: metabolites 1293\n",
      "1724811756.6451156 INFO: Default biomass: [bio1]\n",
      "1724811756.6921659 INFO: metabolites 1189\n",
      "1724811756.7071543 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811756.7719014 INFO: reactions 1513\n",
      "1724811756.7940722 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811756.8077927 INFO: reactions 1062\n",
      "1724811756.9161682 INFO: Default biomass: [bio1]\n",
      "1724811756.9177794 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811756.920187 INFO: reactions 1513\n",
      "1724811756.9631326 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811756.9632294 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811756.9699168 INFO: reactions 1233\n",
      "1724811756.9828193 INFO: Default biomass: [bio1]\n",
      "1724811756.9855144 INFO: metabolites 1224\n",
      "1724811757.0405796 INFO: reactions 1384\n",
      "1724811757.0548947 INFO: reactions 1305\n",
      "1724811757.0840101 INFO: reactions 1230\n",
      "1724811757.1109715 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.1142023 INFO: metabolites 1071\n",
      "1724811757.138574 INFO: metabolites 1298\n",
      "1724811757.171521 INFO: metabolites 1298\n",
      "1724811757.1762211 INFO: reactions 1264\n",
      "1724811757.2028973 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.2031124 INFO: metabolites 1165\n",
      "1724811757.2096336 INFO: Default biomass: [bio1]\n",
      "1724811757.2756152 INFO: reactions 1208\n",
      "1724811757.2887812 INFO: Default biomass: [bio1]\n",
      "1724811757.3039708 INFO: Default biomass: [bio1]\n",
      "1724811757.3463664 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.4183376 INFO: metabolites 1224\n",
      "1724811757.4241104 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.4532888 INFO: Default biomass: [bio1]\n",
      "1724811757.4548483 INFO: metabolites 1134\n",
      "1724811757.4609082 INFO: metabolites 1243\n",
      "1724811757.480212 INFO: reactions 1233\n",
      "1724811757.5529945 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.5545144 INFO: Default biomass: [bio1]\n",
      "1724811757.5748723 INFO: reactions 1365\n",
      "1724811757.5804987 INFO: reactions 1103\n",
      "1724811757.5952399 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.6118386 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.6415594 INFO: reactions 1230\n",
      "1724811757.6744635 INFO: reactions 1221\n",
      "1724811757.7051044 INFO: Default biomass: [bio1]\n",
      "1724811757.7056122 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.7288034 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.732367 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.7366197 INFO: reactions 1301\n",
      "1724811757.7474897 INFO: reactions 1419\n",
      "1724811757.7660434 INFO: Default biomass: [bio1]\n",
      "1724811757.7931633 INFO: reactions 1419\n",
      "1724811757.7993855 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.8084037 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.855755 INFO: Default biomass: [bio1]\n",
      "1724811757.8828073 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.8998127 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.914661 INFO: metabolites 1220\n",
      "1724811757.919515 INFO: Default biomass: [bio1]\n",
      "1724811757.9299219 INFO: reactions 1252\n",
      "1724811757.9436781 INFO: metabolites 1285\n",
      "1724811757.9513636 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811757.9706397 INFO: metabolites 1282\n",
      "1724811757.9953804 INFO: metabolites 1145\n",
      "1724811758.0093846 INFO: reactions 1230\n",
      "1724811758.020723 INFO: Default biomass: [bio1]\n",
      "1724811758.0622952 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811758.067207 INFO: metabolites 1285\n",
      "1724811758.0757651 INFO: Default biomass: [bio1]\n",
      "1724811758.0800614 INFO: reactions 1303\n",
      "1724811758.0836134 INFO: Default biomass: [bio1]\n",
      "1724811758.1241512 INFO: Default biomass: [bio1]\n",
      "1724811758.159319 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811758.2237103 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811758.2348382 INFO: Default biomass: [bio1]\n",
      "1724811758.2650778 INFO: Default biomass: [bio1]\n",
      "1724811758.2692761 INFO: Default biomass: [bio1]\n",
      "1724811758.275124 INFO: Default biomass: [bio1]\n",
      "1724811758.280702 INFO: metabolites 1145\n",
      "1724811758.2818832 INFO: metabolites 1067\n",
      "1724811758.3313413 INFO: metabolites 1239\n",
      "1724811758.3661165 INFO: Default biomass: [bio1]\n",
      "1724811758.3974686 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.53.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811758.4467528 INFO: Default biomass: [bio1]\n",
      "1724811758.4575202 INFO: metabolites 1243\n",
      "1724811758.4837196 INFO: reactions 1233\n",
      "1724811758.4974515 INFO: reactions 1285\n",
      "1724811758.511926 INFO: Default biomass: [bio1]\n",
      "1724811758.5263531 INFO: Default biomass: [bio1]\n",
      "1724811758.5506604 INFO: metabolites 1224\n",
      "1724811758.5620334 INFO: metabolites 1224\n",
      "1724811758.6107814 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811758.6219056 INFO: reactions 1305\n",
      "1724811758.642183 INFO: reactions 1315\n",
      "1724811758.649546 INFO: Default biomass: [bio1]\n",
      "1724811758.676542 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811758.7026508 INFO: metabolites 1285\n",
      "1724811758.745426 INFO: metabolites 1318\n",
      "1724811758.7489128 INFO: reactions 1110\n",
      "1724811758.752191 INFO: metabolites 1265\n",
      "1724811758.7784424 INFO: reactions 1315\n",
      "1724811758.8025043 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811758.8234804 INFO: reactions 1233\n",
      "1724811758.8666263 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811758.8855984 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811758.9238546 INFO: metabolites 1349\n",
      "1724811758.9350634 INFO: metabolites 1298\n",
      "1724811758.9374974 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811758.9575913 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811759.0215368 INFO: metabolites 1355\n",
      "1724811759.0674834 INFO: reactions 1303\n",
      "1724811759.0683591 INFO: metabolites 1220\n",
      "1724811759.113232 INFO: reactions 1255\n",
      "1724811759.1446767 INFO: reactions 1230\n",
      "1724811759.1917627 INFO: reactions 1230\n",
      "1724811759.2132976 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811759.277208 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811759.3008249 INFO: Default biomass: [bio1]\n",
      "1724811759.349179 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811759.396008 INFO: reactions 1358\n",
      "1724811759.4160485 INFO: reactions 1323\n",
      "1724811759.4338872 INFO: metabolites 1351\n",
      "1724811759.508637 INFO: Default biomass: [bio1]\n",
      "1724811759.5630262 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811759.6374462 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811759.6698666 INFO: metabolites 1255\n",
      "1724811759.6830027 INFO: reactions 1285\n",
      "1724811759.720969 INFO: metabolites 1298\n",
      "1724811759.751396 INFO: Default biomass: [bio1]\n",
      "1724811759.7574399 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811759.781602 INFO: Default biomass: [bio1]\n",
      "1724811759.8090715 INFO: reactions 1494\n",
      "1724811759.8157907 INFO: metabolites 1255\n",
      "1724811759.9106455 INFO: Default biomass: [bio1]\n",
      "1724811759.9155443 INFO: reactions 1543\n",
      "1724811759.9719665 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811759.9721618 INFO: Default biomass: [bio1]\n",
      "1724811759.9756274 INFO: metabolites 1318\n",
      "1724811759.9938066 INFO: metabolites 1140\n",
      "1724811760.072645 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811760.109118 INFO: Default biomass: [bio1]\n",
      "1724811760.1389468 INFO: Default biomass: [bio1]\n",
      "1724811760.1681852 INFO: reactions 1513\n",
      "1724811760.2011263 INFO: reactions 1537\n",
      "1724811760.2026181 INFO: reactions 1419\n",
      "1724811760.221026 INFO: Default biomass: [bio1]\n",
      "1724811760.2542543 INFO: Default biomass: [bio1]\n",
      "1724811760.3102894 INFO: reactions 1410\n",
      "1724811760.3450484 INFO: reactions 1419\n",
      "1724811760.355584 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811760.3626835 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811760.3736932 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811760.3920221 INFO: metabolites 1285\n",
      "1724811760.3947046 INFO: Default biomass: [bio1]\n",
      "1724811760.4156156 INFO: metabolites 986\n",
      "1724811760.4161663 INFO: metabolites 771\n",
      "1724811760.4463165 INFO: reactions 1410\n",
      "1724811760.4664764 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811760.4773989 INFO: reactions 1119\n",
      "1724811760.5061543 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811760.5302062 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811760.585787 INFO: Default biomass: [bio1]\n",
      "1724811760.5896919 INFO: Default biomass: [bio1]\n",
      "1724811760.5993907 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811760.5999537 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811760.615509 INFO: metabolites 1224\n",
      "1724811760.6505046 INFO: Default biomass: [bio1]\n",
      "1724811760.6912706 INFO: reactions 1537\n",
      "1724811760.6923404 INFO: metabolites 1140\n",
      "1724811760.7077003 INFO: reactions 734\n",
      "1724811760.7212954 INFO: Default biomass: [bio1]\n",
      "1724811760.7538538 INFO: metabolites 1243\n",
      "1724811760.7835371 INFO: metabolites 1224\n",
      "1724811760.8045375 INFO: metabolites 1309\n",
      "1724811760.8538268 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811760.8599622 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811760.892186 INFO: reactions 1012\n",
      "1724811760.8991864 INFO: Default biomass: [bio1]\n",
      "1724811760.9117806 INFO: Default biomass: [bio1]\n",
      "1724811760.945685 INFO: Default biomass: [bio1]\n",
      "1724811761.0056002 INFO: Default biomass: [bio1]\n",
      "1724811761.0131054 INFO: Default biomass: [bio1]\n",
      "1724811761.0274026 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811761.0354595 INFO: Default biomass: [bio1]\n",
      "1724811761.0472515 INFO: Default biomass: [bio1]\n",
      "1724811761.0709689 INFO: Default biomass: [bio1]\n",
      "1724811761.097602 INFO: reactions 1315\n",
      "1724811761.1384487 INFO: Default biomass: [bio1]\n",
      "1724811761.1591022 INFO: metabolites 1255\n",
      "1724811761.160057 INFO: Default biomass: [bio1]\n",
      "1724811761.1812847 INFO: reactions 1119\n",
      "1724811761.2111402 INFO: reactions 1230\n",
      "1724811761.2351112 INFO: metabolites 1182\n",
      "1724811761.2474008 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811761.3079457 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811761.3595831 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811761.3874826 INFO: reactions 1230\n",
      "1724811761.41379 INFO: reactions 1303\n",
      "1724811761.4178333 INFO: metabolites 1293\n",
      "1724811761.420462 INFO: reactions 1460\n",
      "1724811761.433258 INFO: Default biomass: [bio1]\n",
      "1724811761.4415083 INFO: Default biomass: [bio1]\n",
      "1724811761.4804957 INFO: metabolites 1243\n",
      "1724811761.531135 INFO: metabolites 1071\n",
      "1724811761.546612 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811761.5631828 INFO: metabolites 1309\n",
      "1724811761.5692408 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811761.5789495 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811761.5957155 INFO: metabolites 1205\n",
      "1724811761.60498 INFO: metabolites 1146\n",
      "1724811761.6411514 INFO: metabolites 1145\n",
      "1724811761.6443594 INFO: metabolites 1106\n",
      "1724811761.701684 INFO: metabolites 1145\n",
      "1724811761.7181153 INFO: metabolites 1146\n",
      "1724811761.735928 INFO: metabolites 1273\n",
      "1724811761.7552643 INFO: Default biomass: [bio1]\n",
      "1724811761.7765179 INFO: Default biomass: [bio1]\n",
      "1724811761.7814295 INFO: reactions 1410\n",
      "1724811761.8036582 INFO: metabolites 1309\n",
      "1724811761.8055735 INFO: reactions 1244\n",
      "1724811761.8443363 INFO: Default biomass: [bio1]\n",
      "1724811761.8580465 INFO: metabolites 1145\n",
      "1724811761.8579001 INFO: metabolites 1282\n",
      "1724811761.9218082 INFO: metabolites 1100\n",
      "1724811761.9370446 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.0134647 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.0538802 INFO: Default biomass: [bio1]\n",
      "1724811762.0696223 INFO: reactions 1365\n",
      "1724811762.092289 INFO: Default biomass: [bio1]\n",
      "1724811762.0947502 INFO: reactions 1303\n",
      "1724811762.125075 INFO: reactions 1171\n",
      "1724811762.1284459 INFO: Default biomass: [bio1]\n",
      "1724811762.1624098 INFO: reactions 1301\n",
      "1724811762.20899 INFO: metabolites 1071\n",
      "1724811762.2291873 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.2382379 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.239727 INFO: reactions 1171\n",
      "1724811762.26109 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.302965 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.361063 INFO: reactions 1233\n",
      "1724811762.3719184 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.3754108 INFO: reactions 1383\n",
      "1724811762.3952196 INFO: reactions 1126\n",
      "1724811762.4977167 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.514644 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.5244672 INFO: Default biomass: [bio1]\n",
      "1724811762.5276237 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.53582 INFO: reactions 1305\n",
      "1724811762.624366 INFO: metabolites 1140\n",
      "1724811762.6357932 INFO: reactions 1103\n",
      "1724811762.667478 INFO: metabolites 1351\n",
      "1724811762.680379 INFO: reactions 1460\n",
      "1724811762.7257288 INFO: metabolites 1255\n",
      "1724811762.7271476 INFO: Default biomass: [bio1]\n",
      "1724811762.7483196 INFO: reactions 1128\n",
      "1724811762.7635205 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.7708187 INFO: Default biomass: [bio1]\n",
      "1724811762.7875056 INFO: Default biomass: [bio1]\n",
      "1724811762.8197978 INFO: Default biomass: [bio1]\n",
      "1724811762.832765 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.866653 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.9126005 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811762.9294024 INFO: Default biomass: [bio1]\n",
      "1724811762.9488306 INFO: reactions 1233\n",
      "1724811762.9781833 INFO: reactions 1233\n",
      "1724811762.98553 INFO: Default biomass: [bio1]\n",
      "1724811763.060383 INFO: metabolites 1243\n",
      "1724811763.0702784 INFO: metabolites 1285\n",
      "1724811763.0784101 INFO: Default biomass: [bio1]\n",
      "1724811763.0921564 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811763.1160316 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811763.1605983 INFO: reactions 1103\n",
      "1724811763.1716063 INFO: reactions 1460\n",
      "1724811763.1857498 INFO: Default biomass: [bio1]\n",
      "1724811763.2896817 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811763.2907236 INFO: metabolites 1194\n",
      "1724811763.3056333 INFO: metabolites 1140\n",
      "1724811763.3191638 INFO: metabolites 1100\n",
      "1724811763.350289 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811763.4082713 INFO: reactions 1513\n",
      "1724811763.4221087 INFO: Default biomass: [bio1]\n",
      "1724811763.4370925 INFO: metabolites 1300\n",
      "1724811763.4493473 INFO: Default biomass: [bio1]\n",
      "1724811763.4951828 INFO: Default biomass: [bio1]\n",
      "1724811763.4947453 INFO: metabolites 1088\n",
      "1724811763.536902 INFO: metabolites 1224\n",
      "1724811763.5747623 INFO: metabolites 1309\n",
      "1724811763.586576 INFO: Default biomass: [bio1]\n",
      "1724811763.5997338 INFO: Default biomass: [bio1]\n",
      "1724811763.6017911 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811763.6101239 INFO: Default biomass: [bio1]\n",
      "1724811763.6902492 INFO: reactions 1303\n",
      "1724811763.7124498 INFO: metabolites 1124\n",
      "1724811763.7285345 INFO: Default biomass: [bio1]\n",
      "1724811763.7378278 INFO: metabolites 1068\n",
      "1724811763.7391448 INFO: metabolites 1240\n",
      "1724811763.7416592 INFO: Default biomass: [bio1]\n",
      "1724811763.7481408 INFO: reactions 1358\n",
      "1724811763.8382971 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811763.9511416 INFO: metabolites 1219\n",
      "1724811763.9530003 INFO: Default biomass: [bio1]\n",
      "1724811763.9801667 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.0012083 INFO: reactions 1089\n",
      "1724811764.0220973 INFO: reactions 1119\n",
      "1724811764.0407965 INFO: metabolites 1125\n",
      "1724811764.0856483 INFO: metabolites 1224\n",
      "1724811764.1232302 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.1389942 INFO: reactions 1230\n",
      "1724811764.1417305 INFO: reactions 1410\n",
      "1724811764.1632311 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.1827826 INFO: reactions 1392\n",
      "1724811764.20277 INFO: metabolites 1004\n",
      "1724811764.2214074 INFO: metabolites 1271\n",
      "1724811764.2294793 INFO: metabolites 1224\n",
      "1724811764.2333255 INFO: reactions 1082\n",
      "1724811764.2636201 INFO: Default biomass: [bio1]\n",
      "1724811764.274561 INFO: reactions 1164\n",
      "1724811764.2943907 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.2980788 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.3099062 INFO: reactions 1038\n",
      "1724811764.3132975 INFO: metabolites 1207\n",
      "1724811764.3524535 INFO: reactions 1460\n",
      "1724811764.356952 INFO: Default biomass: [bio1]\n",
      "1724811764.366463 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.406419 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.4161403 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.4391527 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.4470646 INFO: reactions 1119\n",
      "1724811764.5031693 INFO: metabolites 1198\n",
      "1724811764.5238817 INFO: Default biomass: [bio1]\n",
      "1724811764.5576696 INFO: Default biomass: [bio1]\n",
      "1724811764.5708258 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.5866733 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.5886967 INFO: reactions 1202\n",
      "1724811764.6382473 INFO: reactions 1045\n",
      "1724811764.6470382 INFO: metabolites 1285\n",
      "1724811764.6615365 INFO: reactions 1208\n",
      "1724811764.6703098 INFO: reactions 1266\n",
      "1724811764.6808956 INFO: reactions 1230\n",
      "1724811764.729943 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.7496622 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.782633 INFO: Default biomass: [bio1]\n",
      "1724811764.8259234 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.8316994 INFO: metabolites 1145\n",
      "1724811764.8320706 INFO: reactions 1357\n",
      "1724811764.8344843 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.8551733 INFO: Default biomass: [bio1]\n",
      "1724811764.8601394 INFO: Default biomass: [bio1]\n",
      "1724811764.8676984 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811764.8707128 INFO: Default biomass: [bio1]\n",
      "1724811764.8708794 INFO: metabolites 1140\n",
      "1724811764.8793943 INFO: Default biomass: [bio1]\n",
      "1724811764.9212582 INFO: reactions 1230\n",
      "1724811764.9778976 INFO: Default biomass: [bio1]\n",
      "1724811764.9857445 INFO: reactions 1241\n",
      "1724811765.0054212 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811765.0175521 INFO: Default biomass: [bio1]\n",
      "1724811765.027513 INFO: reactions 1308\n",
      "1724811765.0710945 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811765.0779374 INFO: reactions 1273\n",
      "1724811765.1212559 INFO: metabolites 1255\n",
      "1724811765.1517627 INFO: Default biomass: [bio1]\n",
      "1724811765.184978 INFO: metabolites 1342\n",
      "1724811765.1884558 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811765.2288005 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811765.2727008 INFO: Default biomass: [bio1]\n",
      "1724811765.2883043 INFO: Default biomass: [bio1]\n",
      "1724811765.3109465 INFO: metabolites 1232\n",
      "1724811765.3284883 INFO: Default biomass: [bio1]\n",
      "1724811765.3547807 INFO: reactions 1315\n",
      "1724811765.356541 INFO: reactions 1119\n",
      "1724811765.3710337 INFO: metabolites 1248\n",
      "1724811765.3791015 INFO: Default biomass: [bio1]\n",
      "1724811765.386323 INFO: metabolites 1255\n",
      "1724811765.4020753 INFO: metabolites 1071\n",
      "1724811765.454211 INFO: metabolites 1071\n",
      "1724811765.4567454 INFO: metabolites 1157\n",
      "1724811765.487954 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811765.5211053 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811765.5441456 INFO: Default biomass: [bio1]\n",
      "1724811765.5835996 INFO: metabolites 943\n",
      "1724811765.61365 INFO: metabolites 1219\n",
      "1724811765.676287 INFO: Default biomass: [bio1]\n",
      "1724811765.6918228 INFO: Default biomass: [bio1]\n",
      "1724811765.7700427 INFO: Default biomass: [bio1]\n",
      "1724811765.8251991 INFO: Default biomass: [bio1]\n",
      "1724811765.8571372 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811765.900113 INFO: Default biomass: [bio1]\n",
      "1724811765.9259307 INFO: reactions 1103\n",
      "1724811765.93632 INFO: metabolites 1300\n",
      "1724811765.9446638 INFO: reactions 1103\n",
      "1724811765.9498937 INFO: reactions 1341\n",
      "1724811765.9569814 INFO: metabolites 1323\n",
      "1724811765.970875 INFO: reactions 1233\n",
      "1724811766.0292027 INFO: reactions 1054\n",
      "1724811766.0670722 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811766.0728464 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811766.087226 INFO: Default biomass: [bio1]\n",
      "1724811766.0916271 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811766.1413584 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811766.1689143 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811766.2472343 INFO: metabolites 1255\n",
      "1724811766.2798223 INFO: metabolites 1309\n",
      "1724811766.3226888 INFO: reactions 1410\n",
      "1724811766.3842807 INFO: Default biomass: [bio1]\n",
      "1724811766.3843522 INFO: metabolites 1162\n",
      "1724811766.4530213 INFO: reactions 1432\n",
      "1724811766.4709496 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811766.4842043 INFO: metabolites 1192\n",
      "1724811766.4883482 INFO: reactions 1353\n",
      "1724811766.5061417 INFO: Default biomass: [bio1]\n",
      "1724811766.5070877 INFO: Default biomass: [bio1]\n",
      "1724811766.5352407 INFO: reactions 1168\n",
      "1724811766.552932 INFO: metabolites 1177\n",
      "1724811766.5937355 INFO: metabolites 1275\n",
      "1724811766.6089325 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811766.616555 INFO: Default biomass: [bio1]\n",
      "1724811766.63815 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811766.681397 INFO: Default biomass: [bio1]\n",
      "1724811766.698371 INFO: metabolites 1243\n",
      "1724811766.698567 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811766.7781298 INFO: metabolites 1255\n",
      "1724811766.8125134 INFO: Default biomass: [bio1]\n",
      "1724811766.8453803 INFO: reactions 1337\n",
      "1724811766.8887832 INFO: reactions 1410\n",
      "1724811766.9024427 INFO: reactions 1460\n",
      "1724811766.917626 INFO: reactions 1208\n",
      "1724811766.9621515 INFO: metabolites 1192\n",
      "1724811766.9969513 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.0003982 INFO: Default biomass: [bio1]\n",
      "1724811767.0466871 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.0678973 INFO: reactions 1325\n",
      "1724811767.0810382 INFO: metabolites 1285\n",
      "1724811767.0814972 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.1484504 INFO: Default biomass: [bio1]\n",
      "1724811767.1496463 INFO: reactions 1207\n",
      "1724811767.1730604 INFO: metabolites 1224\n",
      "1724811767.182246 INFO: Default biomass: [bio1]\n",
      "1724811767.1893563 INFO: reactions 1392\n",
      "1724811767.1926484 INFO: Default biomass: [bio1]\n",
      "1724811767.2088995 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.2493627 INFO: reactions 1314\n",
      "1724811767.3675272 INFO: metabolites 1071\n",
      "1724811767.4077358 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.4075744 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.4361138 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.4472172 INFO: reactions 1410\n",
      "1724811767.4852288 INFO: Default biomass: [bio1]\n",
      "1724811767.5031912 INFO: reactions 1349\n",
      "1724811767.510128 INFO: metabolites 1248\n",
      "1724811767.518812 INFO: Default biomass: [bio1]\n",
      "1724811767.5216901 INFO: reactions 1325\n",
      "1724811767.5855925 INFO: Default biomass: [bio1]\n",
      "1724811767.6044462 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.611967 INFO: metabolites 1154\n",
      "1724811767.6126719 INFO: Default biomass: [bio1]\n",
      "1724811767.632748 INFO: metabolites 1068\n",
      "1724811767.6737578 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.701318 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.717951 INFO: Default biomass: [bio1]\n",
      "1724811767.7267244 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.7286901 INFO: metabolites 1225\n",
      "1724811767.7532403 INFO: metabolites 1325\n",
      "1724811767.7675734 INFO: reactions 1230\n",
      "1724811767.77924 INFO: reactions 1358\n",
      "1724811767.8087187 INFO: metabolites 1182\n",
      "1724811767.8497837 INFO: reactions 1103\n",
      "1724811767.9218402 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.9320657 INFO: metabolites 1101\n",
      "1724811767.9332514 INFO: reactions 1180\n",
      "1724811767.9389355 INFO: metabolites 1068\n",
      "1724811767.971207 INFO: Default biomass: [bio1]\n",
      "1724811767.974608 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811767.9998848 INFO: Default biomass: [bio1]\n",
      "1724811768.0034428 INFO: reactions 1303\n",
      "1724811768.0110493 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.057544 INFO: metabolites 1309\n",
      "1724811768.0661898 INFO: Default biomass: [bio1]\n",
      "1724811768.068535 INFO: metabolites 1208\n",
      "1724811768.0697 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.1185377 INFO: reactions 1156\n",
      "1724811768.1256313 INFO: metabolites 1248\n",
      "1724811768.127574 INFO: reactions 1082\n",
      "1724811768.1545691 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.1728406 INFO: reactions 1337\n",
      "1724811768.2094147 INFO: Default biomass: [bio1]\n",
      "1724811768.2130876 INFO: metabolites 1208\n",
      "1724811768.2395792 INFO: Default biomass: [bio1]\n",
      "1724811768.2402742 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.2614026 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.2742887 INFO: reactions 1284\n",
      "1724811768.3007846 INFO: Default biomass: [bio1]\n",
      "1724811768.306986 INFO: Default biomass: [bio1]\n",
      "1724811768.3175468 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.3732615 INFO: reactions 1134\n",
      "1724811768.3736756 INFO: reactions 1244\n",
      "1724811768.398658 INFO: Default biomass: [bio1]\n",
      "1724811768.40783 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.4190004 INFO: reactions 1351\n",
      "1724811768.4223752 INFO: Default biomass: [bio1]\n",
      "1724811768.4414055 INFO: reactions 1082\n",
      "1724811768.4915695 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.5061646 INFO: metabolites 1335\n",
      "1724811768.5441575 INFO: Default biomass: [bio1]\n",
      "1724811768.568594 INFO: Default biomass: [bio1]\n",
      "1724811768.5807612 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.5868034 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.5984285 INFO: metabolites 1086\n",
      "1724811768.6034346 INFO: metabolites 1219\n",
      "1724811768.6087549 INFO: reactions 1250\n",
      "1724811768.6239972 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.670502 INFO: reactions 1460\n",
      "1724811768.6811733 INFO: Default biomass: [bio1]\n",
      "1724811768.6850815 INFO: Default biomass: [bio1]\n",
      "1724811768.6848288 INFO: Default biomass: [bio1]\n",
      "1724811768.7061276 INFO: metabolites 1208\n",
      "1724811768.7503014 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.7844417 INFO: reactions 1250\n",
      "1724811768.8286052 INFO: Default biomass: [bio1]\n",
      "1724811768.8293831 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811768.8416502 INFO: metabolites 1324\n",
      "1724811768.9032416 INFO: Default biomass: [bio1]\n",
      "1724811768.9046493 INFO: metabolites 1107\n",
      "1724811768.9405365 INFO: Default biomass: [bio1]\n",
      "1724811769.0202346 INFO: metabolites 1300\n",
      "1724811769.0314288 INFO: Default biomass: [bio1]\n",
      "1724811769.0709112 INFO: reactions 1172\n",
      "1724811769.0742707 INFO: metabolites 1285\n",
      "1724811769.085979 INFO: Default biomass: [bio1]\n",
      "1724811769.108197 INFO: metabolites 1194\n",
      "1724811769.1288507 INFO: metabolites 1140\n",
      "1724811769.1505582 INFO: reactions 1468\n",
      "1724811769.1642761 INFO: Default biomass: [bio1]\n",
      "1724811769.1912742 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811769.191292 INFO: reactions 1208\n",
      "1724811769.1969893 INFO: metabolites 1255\n",
      "1724811769.239038 INFO: Default biomass: [bio1]\n",
      "1724811769.2886395 INFO: metabolites 1271\n",
      "1724811769.3530262 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811769.373278 INFO: Default biomass: [bio1]\n",
      "1724811769.3991756 INFO: metabolites 1207\n",
      "1724811769.4615843 INFO: reactions 1131\n",
      "1724811769.5314536 INFO: metabolites 1291\n",
      "1724811769.5629327 INFO: reactions 1328\n",
      "1724811769.6217957 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811769.6361866 INFO: reactions 1119\n",
      "1724811769.6483588 INFO: Default biomass: [bio1]\n",
      "1724811769.6475537 INFO: reactions 1337\n",
      "1724811769.6481693 INFO: metabolites 1271\n",
      "1724811769.681949 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811769.70155 INFO: metabolites 1342\n",
      "1724811769.7027647 INFO: reactions 1266\n",
      "1724811769.7210546 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811769.758401 INFO: metabolites 1207\n",
      "1724811769.775336 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811769.801483 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811769.8138437 INFO: reactions 1392\n",
      "1724811769.8317158 INFO: reactions 1410\n",
      "1724811769.8412602 INFO: Default biomass: [bio1]\n",
      "1724811769.8503263 INFO: metabolites 1212\n",
      "1724811769.8920698 INFO: reactions 1357\n",
      "1724811769.9011202 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811769.996046 INFO: reactions 1250\n",
      "1724811769.9987628 INFO: metabolites 1300\n",
      "1724811770.0053494 INFO: metabolites 1309\n",
      "1724811770.0332034 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811770.0376177 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811770.08848 INFO: metabolites 832\n",
      "1724811770.1403594 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811770.2039802 INFO: Default biomass: [bio1]\n",
      "1724811770.2202842 INFO: reactions 1267\n",
      "1724811770.2519526 INFO: Default biomass: [bio1]\n",
      "1724811770.2779436 INFO: Default biomass: [bio1]\n",
      "1724811770.280631 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811770.280808 INFO: reactions 1357\n",
      "1724811770.2892423 INFO: metabolites 1068\n",
      "1724811770.3537543 INFO: Default biomass: [bio1]\n",
      "1724811770.3777823 INFO: reactions 892\n",
      "1724811770.380859 INFO: reactions 1267\n",
      "1724811770.4240823 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811770.4743042 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811770.4787433 INFO: reactions 1358\n",
      "1724811770.5195165 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811770.532301 INFO: reactions 1349\n",
      "1724811770.5983822 INFO: Default biomass: [bio1]\n",
      "1724811770.6159499 INFO: Default biomass: [bio1]\n",
      "1724811770.629249 INFO: reactions 1460\n",
      "1724811770.6508057 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811770.6617863 INFO: Default biomass: [bio1]\n",
      "1724811770.7178085 INFO: metabolites 1192\n",
      "1724811770.7262666 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811770.7379265 INFO: metabolites 1071\n",
      "1724811770.7868805 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811770.8002439 INFO: reactions 1082\n",
      "1724811770.8083446 INFO: Default biomass: [bio1]\n",
      "1724811770.8194003 INFO: metabolites 1189\n",
      "1724811770.8358338 INFO: Default biomass: [bio1]\n",
      "1724811770.8544219 INFO: metabolites 981\n",
      "1724811770.9371853 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811770.936959 INFO: metabolites 1352\n",
      "1724811770.9632707 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811771.0147185 INFO: Default biomass: [bio1]\n",
      "1724811771.0453615 INFO: metabolites 1232\n",
      "1724811771.0847826 INFO: metabolites 1192\n",
      "1724811771.1285865 INFO: metabolites 1208\n",
      "1724811771.130132 INFO: Default biomass: [bio1]\n",
      "1724811771.1954181 INFO: Default biomass: [bio1]\n",
      "1724811771.1965702 INFO: metabolites 1183\n",
      "1724811771.207897 INFO: Default biomass: [bio1]\n",
      "1724811771.2321763 INFO: reactions 1342\n",
      "1724811771.2968204 INFO: Default biomass: [bio1]\n",
      "1724811771.337478 INFO: Default biomass: [bio1]\n",
      "1724811771.3824086 INFO: reactions 1208\n",
      "1724811771.3852088 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811771.4123244 INFO: metabolites 1305\n",
      "1724811771.43812 INFO: reactions 1273\n",
      "1724811771.4537718 INFO: metabolites 1239\n",
      "1724811771.4810848 INFO: Default biomass: [bio1]\n",
      "1724811771.511748 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811771.5338542 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811771.5347006 INFO: metabolites 1323\n",
      "1724811771.58721 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811771.5888631 INFO: reactions 1392\n",
      "1724811771.6009183 INFO: reactions 1432\n",
      "1724811771.6450334 INFO: metabolites 1143\n",
      "1724811771.6578195 INFO: reactions 1495\n",
      "1724811771.666749 INFO: reactions 1325\n",
      "1724811771.6807163 INFO: metabolites 1248\n",
      "1724811771.6920023 INFO: reactions 1332\n",
      "1724811771.7631402 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811771.7761528 INFO: metabolites 1248\n",
      "1724811771.807886 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811771.8179562 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811771.8310804 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811771.8387904 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811771.8887823 INFO: Default biomass: [bio1]\n",
      "1724811771.8999894 INFO: Default biomass: [bio1]\n",
      "1724811771.9449606 INFO: metabolites 1262\n",
      "1724811771.9925685 INFO: metabolites 771\n",
      "1724811772.0043812 INFO: metabolites 1240\n",
      "1724811772.0386517 INFO: Default biomass: [bio1]\n",
      "1724811772.05338 INFO: reactions 1389\n",
      "1724811772.0579498 INFO: reactions 1255\n",
      "1724811772.0641017 INFO: Default biomass: [bio1]\n",
      "1724811772.0865762 INFO: Default biomass: [bio1]\n",
      "1724811772.095009 INFO: Default biomass: [bio1]\n",
      "1724811772.1091645 INFO: reactions 1194\n",
      "1724811772.2035244 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.211598 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.2374527 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.275354 INFO: reactions 1103\n",
      "1724811772.2793632 INFO: reactions 734\n",
      "1724811772.303813 INFO: reactions 1349\n",
      "1724811772.3178697 INFO: Default biomass: [bio1]\n",
      "1724811772.3232205 INFO: Default biomass: [bio1]\n",
      "1724811772.3419015 INFO: Default biomass: [bio1]\n",
      "1724811772.3434887 INFO: reactions 1250\n",
      "1724811772.345907 INFO: metabolites 1143\n",
      "1724811772.3938923 INFO: Default biomass: [bio1]\n",
      "1724811772.410425 INFO: reactions 1337\n",
      "1724811772.421194 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.430972 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.4329891 INFO: metabolites 1287\n",
      "1724811772.4342413 INFO: Default biomass: [bio1]\n",
      "1724811772.436475 INFO: reactions 946\n",
      "1724811772.441263 INFO: reactions 1337\n",
      "1724811772.4582903 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.485677 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.5176923 INFO: metabolites 887\n",
      "1724811772.562586 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.58512 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.588277 INFO: metabolites 1355\n",
      "1724811772.6092887 INFO: metabolites 1271\n",
      "1724811772.6217346 INFO: metabolites 1228\n",
      "1724811772.6350749 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.6968455 INFO: reactions 1290\n",
      "1724811772.7211366 INFO: Default biomass: [bio1]\n",
      "1724811772.7312484 INFO: Default biomass: [bio1]\n",
      "1724811772.7506607 INFO: metabolites 1042\n",
      "1724811772.7709985 INFO: reactions 1308\n",
      "1724811772.7740724 INFO: reactions 1246\n",
      "1724811772.7937937 INFO: Default biomass: [bio1]\n",
      "1724811772.8152466 INFO: metabolites 1183\n",
      "1724811772.81837 INFO: Default biomass: [bio1]\n",
      "1724811772.8968756 INFO: Default biomass: [bio1]\n",
      "1724811772.9056232 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.9152923 INFO: metabolites 1325\n",
      "1724811772.9194796 INFO: reactions 903\n",
      "1724811772.942824 INFO: reactions 1194\n",
      "1724811772.9451659 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.968744 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811772.9693108 INFO: Default biomass: [bio1]\n",
      "1724811773.0149298 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.0171213 INFO: metabolites 1162\n",
      "1724811773.0394971 INFO: Default biomass: [bio1]\n",
      "1724811773.0756602 INFO: reactions 1323\n",
      "1724811773.0944822 INFO: Default biomass: [bio1]\n",
      "1724811773.110943 INFO: Default biomass: [bio1]\n",
      "1724811773.1211157 INFO: reactions 1039\n",
      "1724811773.122756 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.2052505 INFO: reactions 1357\n",
      "1724811773.232322 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.2342315 INFO: metabolites 1122\n",
      "1724811773.2357368 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.2542267 INFO: reactions 1290\n",
      "1724811773.281061 INFO: metabolites 1146\n",
      "1724811773.3048694 INFO: metabolites 1232\n",
      "1724811773.3087847 INFO: reactions 1332\n",
      "1724811773.3478339 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.3681924 INFO: Default biomass: [bio1]\n",
      "1724811773.3934386 INFO: reactions 1494\n",
      "1724811773.409351 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.4236224 INFO: metabolites 1267\n",
      "1724811773.4486427 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.4492154 INFO: Default biomass: [bio1]\n",
      "1724811773.466099 INFO: metabolites 1271\n",
      "1724811773.5243194 INFO: reactions 1244\n",
      "1724811773.5493934 INFO: metabolites 1228\n",
      "1724811773.558329 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.5924351 INFO: Default biomass: [bio1]\n",
      "1724811773.601558 INFO: Default biomass: [bio1]\n",
      "1724811773.6032524 INFO: reactions 1351\n",
      "1724811773.629975 INFO: Default biomass: [bio1]\n",
      "1724811773.6361587 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.6611083 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.6680737 INFO: Default biomass: [bio1]\n",
      "1724811773.6690552 INFO: metabolites 1180\n",
      "1724811773.7337575 INFO: metabolites 1300\n",
      "1724811773.7444754 INFO: metabolites 1309\n",
      "1724811773.7857637 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.7994385 INFO: reactions 1171\n",
      "1724811773.8237147 INFO: metabolites 995\n",
      "1724811773.8738313 INFO: metabolites 1300\n",
      "1724811773.8774095 INFO: Default biomass: [bio1]\n",
      "1724811773.9164147 INFO: Default biomass: [bio1]\n",
      "1724811773.933544 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811773.965871 INFO: Default biomass: [bio1]\n",
      "1724811773.9708269 INFO: reactions 1353\n",
      "1724811774.0152736 INFO: metabolites 1323\n",
      "1724811774.101924 INFO: reactions 1357\n",
      "1724811774.1046553 INFO: metabolites 1232\n",
      "1724811774.1455712 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811774.1624455 INFO: reactions 1290\n",
      "1724811774.1667511 INFO: Default biomass: [bio1]\n",
      "1724811774.2402174 INFO: reactions 1024\n",
      "1724811774.2519135 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811774.3217382 INFO: reactions 1238\n",
      "1724811774.3227906 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811774.3347428 INFO: Default biomass: [bio1]\n",
      "1724811774.3516586 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811774.387992 INFO: metabolites 1143\n",
      "1724811774.41509 INFO: Default biomass: [bio1]\n",
      "1724811774.459792 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811774.4987073 INFO: Default biomass: [bio1]\n",
      "1724811774.5672104 INFO: metabolites 1246\n",
      "1724811774.6050754 INFO: reactions 1392\n",
      "1724811774.6505067 INFO: metabolites 1208\n",
      "1724811774.7675066 INFO: metabolites 832\n",
      "1724811774.8062875 INFO: Default biomass: [bio1]\n",
      "1724811774.8085506 INFO: reactions 1353\n",
      "1724811774.8111422 INFO: Default biomass: [bio1]\n",
      "1724811774.8139737 INFO: Default biomass: [bio1]\n",
      "1724811774.8212974 INFO: reactions 1349\n",
      "1724811774.8644223 INFO: Default biomass: [bio1]\n",
      "1724811774.8820577 INFO: Default biomass: [bio1]\n",
      "1724811774.8886843 INFO: reactions 1194\n",
      "1724811774.8963852 INFO: metabolites 1243\n",
      "1724811774.9063148 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811774.93695 INFO: reactions 1117\n",
      "1724811774.9684756 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811774.976675 INFO: metabolites 1194\n",
      "1724811774.980305 INFO: Default biomass: [bio1]\n",
      "1724811774.990314 INFO: Default biomass: [bio1]\n",
      "1724811774.9916568 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811775.01934 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811775.0588338 INFO: metabolites 1255\n",
      "1724811775.1018078 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811775.2349575 INFO: metabolites 1033\n",
      "1724811775.2496443 INFO: reactions 1460\n",
      "1724811775.2506616 INFO: metabolites 1300\n",
      "1724811775.2781687 INFO: metabolites 1287\n",
      "1724811775.3152359 INFO: reactions 1272\n",
      "1724811775.3779008 INFO: metabolites 1143\n",
      "1724811775.4107866 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811775.4509618 INFO: metabolites 1287\n",
      "1724811775.5064046 INFO: reactions 1293\n",
      "1724811775.5062242 INFO: Default biomass: [bio1]\n",
      "1724811775.5082698 INFO: metabolites 1293\n",
      "1724811775.5306325 INFO: Default biomass: [bio1]\n",
      "1724811775.543728 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811775.5640528 INFO: Default biomass: [bio1]\n",
      "1724811775.5855093 INFO: metabolites 1299\n",
      "1724811775.5868976 INFO: Default biomass: [bio1]\n",
      "1724811775.593088 INFO: Default biomass: [bio1]\n",
      "1724811775.65428 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811775.7115686 INFO: reactions 1410\n",
      "1724811775.7437978 INFO: reactions 1392\n",
      "1724811775.8511133 INFO: reactions 892\n",
      "1724811775.8937101 INFO: reactions 1194\n",
      "1724811775.9058845 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811775.920574 INFO: reactions 1323\n",
      "1724811775.9491816 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811775.9844217 INFO: Default biomass: [bio1]\n",
      "1724811775.9944735 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811775.9977052 INFO: reactions 1392\n",
      "1724811776.0305815 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811776.082301 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811776.0818803 INFO: metabolites 1162\n",
      "1724811776.1115034 INFO: Default biomass: [bio1]\n",
      "1724811776.166583 INFO: reactions 1365\n",
      "1724811776.1724975 INFO: Default biomass: [bio1]\n",
      "1724811776.206482 INFO: metabolites 771\n",
      "1724811776.2212396 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811776.2520618 INFO: reactions 1266\n",
      "1724811776.28794 INFO: Default biomass: [bio1]\n",
      "1724811776.3257523 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811776.3899722 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811776.4110796 INFO: metabolites 1255\n",
      "1724811776.4711618 INFO: metabolites 1202\n",
      "1724811776.5115526 INFO: reactions 734\n",
      "1724811776.5127816 INFO: reactions 1250\n",
      "1724811776.5524764 INFO: Default biomass: [bio1]\n",
      "1724811776.585968 INFO: reactions 1244\n",
      "1724811776.6196504 INFO: Default biomass: [bio1]\n",
      "1724811776.6457114 INFO: metabolites 1108\n",
      "1724811776.6512864 INFO: reactions 1287\n",
      "1724811776.6636515 INFO: Default biomass: [bio1]\n",
      "1724811776.6703925 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811776.6715376 INFO: metabolites 1300\n",
      "1724811776.6845975 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811776.7676291 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811776.7833493 INFO: metabolites 1239\n",
      "1724811776.786864 INFO: metabolites 1237\n",
      "1724811776.8251278 INFO: Default biomass: [bio1]\n",
      "1724811776.8730347 INFO: Default biomass: [bio1]\n",
      "1724811776.8781831 INFO: metabolites 1232\n",
      "1724811776.8828802 INFO: reactions 1111\n",
      "1724811776.8891485 INFO: Default biomass: [bio1]\n",
      "1724811776.9142497 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811776.951862 INFO: reactions 1323\n",
      "1724811776.9915748 INFO: Default biomass: [bio1]\n",
      "1724811777.0033963 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.045862 INFO: metabolites 1154\n",
      "1724811777.0980835 INFO: reactions 1249\n",
      "1724811777.1737907 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.1759732 INFO: reactions 1181\n",
      "1724811777.1832404 INFO: reactions 1282\n",
      "1724811777.1925154 INFO: reactions 1357\n",
      "1724811777.193568 INFO: Default biomass: [bio1]\n",
      "1724811777.2058225 INFO: metabolites 1162\n",
      "1724811777.2339523 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.2446864 INFO: Default biomass: [bio1]\n",
      "1724811777.2526422 INFO: metabolites 1178\n",
      "1724811777.3001459 INFO: metabolites 1309\n",
      "1724811777.3463523 INFO: metabolites 1324\n",
      "1724811777.3514156 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.3776548 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.3817062 INFO: metabolites 1162\n",
      "1724811777.3882718 INFO: reactions 1255\n",
      "1724811777.388531 INFO: reactions 1392\n",
      "1724811777.416296 INFO: metabolites 1323\n",
      "1724811777.4359357 INFO: metabolites 1124\n",
      "1724811777.450531 INFO: Default biomass: [bio1]\n",
      "1724811777.4517767 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.468357 INFO: Default biomass: [bio1]\n",
      "1724811777.4782896 INFO: Default biomass: [bio1]\n",
      "1724811777.5094354 INFO: reactions 1353\n",
      "1724811777.5128915 INFO: reactions 1267\n",
      "1724811777.5941052 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.6012464 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.6170833 INFO: reactions 1156\n",
      "1724811777.642845 INFO: metabolites 1207\n",
      "1724811777.6722426 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.7013607 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.7264528 INFO: Default biomass: [bio1]\n",
      "1724811777.7411897 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.7848163 INFO: metabolites 1192\n",
      "1724811777.795359 INFO: reactions 1244\n",
      "1724811777.807284 INFO: metabolites 1183\n",
      "1724811777.815545 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.9.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811777.8438041 INFO: Default biomass: [bio1]\n",
      "1724811777.877328 INFO: metabolites 1069\n",
      "1724811777.9150472 INFO: reactions 1460\n",
      "1724811777.915966 INFO: metabolites 1073\n",
      "1724811777.9248724 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811777.9256115 INFO: Default biomass: [bio1]\n",
      "1724811777.976031 INFO: metabolites 1232\n",
      "1724811777.9770753 INFO: Default biomass: [bio1]\n",
      "1724811777.9973295 INFO: reactions 1164\n",
      "1724811778.0138166 INFO: reactions 1244\n",
      "1724811778.0563538 INFO: reactions 1231\n",
      "1724811778.0713885 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811778.1170545 INFO: reactions 1328\n",
      "1724811778.146309 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811778.1609588 INFO: Default biomass: [bio1]\n",
      "1724811778.168784 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811778.1713073 INFO: reactions 1349\n",
      "1724811778.2125928 INFO: Default biomass: [bio1]\n",
      "1724811778.213936 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811778.231959 INFO: Default biomass: [bio1]\n",
      "1724811778.2328024 INFO: Default biomass: [bio1]\n",
      "1724811778.2429926 INFO: reactions 1241\n",
      "1724811778.2754335 INFO: metabolites 1269\n",
      "1724811778.279967 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811778.307433 INFO: Default biomass: [bio1]\n",
      "1724811778.3279846 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811778.332757 INFO: metabolites 1291\n",
      "1724811778.3784814 INFO: reactions 1117\n",
      "1724811778.3808293 INFO: reactions 1126\n",
      "1724811778.3991585 INFO: Default biomass: [bio1]\n",
      "1724811778.4351048 INFO: metabolites 1178\n",
      "1724811778.4429839 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811778.4595327 INFO: reactions 1332\n",
      "1724811778.4907205 INFO: metabolites 1154\n",
      "1724811778.5002527 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811778.501653 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811778.5089705 INFO: metabolites 1274\n",
      "1724811778.6017811 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811778.6106374 INFO: Default biomass: [bio1]\n",
      "1724811778.630863 INFO: Default biomass: [bio1]\n",
      "1724811778.633176 INFO: metabolites 1162\n",
      "1724811778.7040057 INFO: reactions 1353\n",
      "1724811778.710781 INFO: Default biomass: [bio1]\n",
      "1724811778.7921352 INFO: Default biomass: [bio1]\n",
      "1724811778.8051987 INFO: metabolites 1122\n",
      "1724811778.8153737 INFO: Default biomass: [bio1]\n",
      "1724811778.8531792 INFO: metabolites 1305\n",
      "1724811778.8832748 INFO: reactions 1360\n",
      "1724811778.8947742 INFO: Default biomass: [bio1]\n",
      "1724811778.9115176 INFO: metabolites 1255\n",
      "1724811778.9219716 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811778.938073 INFO: Default biomass: [bio1]\n",
      "1724811778.9516177 INFO: Default biomass: [bio1]\n",
      "1724811778.9703362 INFO: Default biomass: [bio1]\n",
      "1724811778.9763129 INFO: reactions 1346\n",
      "1724811778.9941578 INFO: metabolites 1192\n",
      "1724811779.0419407 INFO: reactions 1231\n",
      "1724811779.0504158 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_metabat.20.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811779.0741992 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811779.0866754 INFO: reactions 1331\n",
      "1724811779.1076815 INFO: Default biomass: [bio1]\n",
      "1724811779.1213288 INFO: reactions 1244\n",
      "1724811779.12323 INFO: reactions 1156\n",
      "1724811779.1367347 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811779.1903872 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811779.2276247 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811779.2486238 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811779.2497132 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811779.2812135 INFO: reactions 1325\n",
      "1724811779.394257 INFO: reactions 1117\n",
      "1724811779.4367998 INFO: metabolites 981\n",
      "1724811779.4798357 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811779.4948175 INFO: Default biomass: [bio1]\n",
      "1724811779.504071 INFO: metabolites 1287\n",
      "1724811779.5151381 INFO: metabolites 1232\n",
      "1724811779.541395 INFO: metabolites 1324\n",
      "1724811779.5481894 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811779.5651035 INFO: reactions 1325\n",
      "1724811779.602865 INFO: reactions 1282\n",
      "1724811779.6069171 INFO: Default biomass: [bio1]\n",
      "1724811779.6414511 INFO: reactions 1389\n",
      "1724811779.6551816 INFO: metabolites 1246\n",
      "1724811779.6828108 INFO: Default biomass: [bio1]\n",
      "1724811779.6878648 INFO: metabolites 1295\n",
      "1724811779.693677 INFO: metabolites 1246\n",
      "1724811779.7148209 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811779.7681968 INFO: Default biomass: [bio1]\n",
      "1724811779.7966137 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811779.8044581 INFO: Default biomass: [bio1]\n",
      "1724811779.816875 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811779.887398 INFO: reactions 946\n",
      "1724811779.9991732 INFO: metabolites 1154\n",
      "1724811780.0023153 INFO: Default biomass: [bio1]\n",
      "1724811780.064018 INFO: metabolites 1300\n",
      "1724811780.0759416 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811780.2414641 INFO: metabolites 1295\n",
      "1724811780.2410786 INFO: Default biomass: [bio1]\n",
      "1724811780.2818131 INFO: metabolites 1330\n",
      "1724811780.3720975 INFO: reactions 1287\n",
      "1724811780.3920722 INFO: metabolites 1207\n",
      "1724811780.4171054 INFO: Default biomass: [bio1]\n",
      "1724811780.4314811 INFO: Default biomass: [bio1]\n",
      "1724811780.4631677 INFO: metabolites 1157\n",
      "1724811780.490334 INFO: metabolites 1208\n",
      "1724811780.5330064 INFO: reactions 1156\n",
      "1724811780.5353208 INFO: Default biomass: [bio1]\n",
      "1724811780.5631003 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811780.6640499 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811780.6747775 INFO: Default biomass: [bio1]\n",
      "1724811780.7673297 INFO: Default biomass: [bio1]\n",
      "1724811780.7888818 INFO: reactions 1392\n",
      "1724811780.9249349 INFO: reactions 1432\n",
      "1724811780.931656 INFO: Default biomass: [bio1]\n",
      "1724811780.9882364 INFO: metabolites 1154\n",
      "1724811781.0008733 INFO: metabolites 1007\n",
      "1724811781.009181 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811781.040054 INFO: reactions 1250\n",
      "1724811781.0894008 INFO: metabolites 1274\n",
      "1724811781.10085 INFO: reactions 1168\n",
      "1724811781.127113 INFO: Default biomass: [bio1]\n",
      "1724811781.151273 INFO: Default biomass: [bio1]\n",
      "1724811781.1798928 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811781.2066479 INFO: metabolites 1267\n",
      "1724811781.2503395 INFO: reactions 1328\n",
      "1724811781.2505932 INFO: reactions 1246\n",
      "1724811781.268697 INFO: metabolites 1192\n",
      "1724811781.2870612 INFO: reactions 1323\n",
      "1724811781.3046503 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811781.3446362 INFO: metabolites 1237\n",
      "1724811781.4002957 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811781.4077892 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811781.4636145 INFO: metabolites 1360\n",
      "1724811781.4750693 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811781.5645216 INFO: metabolites 1220\n",
      "1724811781.5938165 INFO: Default biomass: [bio1]\n",
      "1724811781.6226912 INFO: reactions 1466\n",
      "1724811781.6391754 INFO: reactions 1039\n",
      "1724811781.6540601 INFO: metabolites 1212\n",
      "1724811781.6768172 INFO: Default biomass: [bio1]\n",
      "1724811781.7969968 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811781.8465884 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811781.8517385 INFO: reactions 1287\n",
      "1724811781.9093502 INFO: metabolites 1208\n",
      "1724811781.9174786 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811781.932019 INFO: Default biomass: [bio1]\n",
      "1724811781.9870074 INFO: Default biomass: [bio1]\n",
      "1724811781.9927797 INFO: Default biomass: [bio1]\n",
      "1724811781.9939914 INFO: reactions 1466\n",
      "1724811782.0390089 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811782.0468998 INFO: Default biomass: [bio1]\n",
      "1724811782.049283 INFO: reactions 1325\n",
      "1724811782.0751672 INFO: reactions 1267\n",
      "1724811782.1340022 INFO: reactions 1297\n",
      "1724811782.1508095 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811782.2023988 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811782.2183256 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811782.2477605 INFO: reactions 1285\n",
      "1724811782.2852237 INFO: reactions 1273\n",
      "1724811782.297697 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811782.309802 INFO: Default biomass: [bio1]\n",
      "1724811782.341164 INFO: reactions 1440\n",
      "1724811782.419162 INFO: Default biomass: [bio1]\n",
      "1724811782.4226174 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811782.4373531 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811782.4709666 INFO: Default biomass: [bio1]\n",
      "1724811782.4808762 INFO: reactions 1156\n",
      "1724811782.4943266 INFO: reactions 1250\n",
      "1724811782.4960709 INFO: metabolites 1323\n",
      "1724811782.5596755 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811782.604498 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811782.6199048 INFO: Default biomass: [bio1]\n",
      "1724811782.6342812 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811782.6833699 INFO: reactions 1331\n",
      "1724811782.7303376 INFO: Default biomass: [bio1]\n",
      "1724811782.7327805 INFO: Default biomass: [bio1]\n",
      "1724811782.7353504 INFO: Default biomass: [bio1]\n",
      "1724811782.800028 INFO: Default biomass: [bio1]\n",
      "1724811782.8317146 INFO: metabolites 1219\n",
      "1724811782.8319774 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811782.8408523 INFO: reactions 1272\n",
      "1724811782.8490775 INFO: metabolites 1305\n",
      "1724811782.9243786 INFO: Default biomass: [bio1]\n",
      "1724811782.939252 INFO: Default biomass: [bio1]\n",
      "1724811783.0525482 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811783.107566 INFO: Default biomass: [bio1]\n",
      "1724811783.113246 INFO: Default biomass: [bio1]\n",
      "1724811783.155502 INFO: Default biomass: [bio1]\n",
      "1724811783.1588461 INFO: metabolites 1208\n",
      "1724811783.1631992 INFO: metabolites 1228\n",
      "1724811783.1734838 INFO: metabolites 1162\n",
      "1724811783.2049246 INFO: metabolites 1202\n",
      "1724811783.2249627 INFO: metabolites 1269\n",
      "1724811783.2573056 INFO: metabolites 1183\n",
      "1724811783.2799559 INFO: reactions 1349\n",
      "1724811783.3400717 INFO: Default biomass: [bio1]\n",
      "1724811783.3866327 INFO: metabolites 1182\n",
      "1724811783.4057987 INFO: metabolites 1188\n",
      "1724811783.4167528 INFO: metabolites 1073\n",
      "1724811783.427989 INFO: reactions 1208\n",
      "1724811783.4419355 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811783.4532638 INFO: metabolites 1269\n",
      "1724811783.5266986 INFO: metabolites 1291\n",
      "1724811783.5344677 INFO: metabolites 1299\n",
      "1724811783.5356662 INFO: reactions 1389\n",
      "1724811783.5569267 INFO: metabolites 1207\n",
      "1724811783.583377 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811783.644927 INFO: Default biomass: [bio1]\n",
      "1724811783.6615407 INFO: reactions 1244\n",
      "1724811783.6841543 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811783.69558 INFO: reactions 1250\n",
      "1724811783.752031 INFO: reactions 1332\n",
      "1724811783.7593977 INFO: metabolites 1228\n",
      "1724811783.7740192 INFO: reactions 1290\n",
      "1724811783.7882698 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811783.794691 INFO: metabolites 1183\n",
      "1724811783.8293355 INFO: metabolites 1305\n",
      "1724811783.8320017 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811783.8342373 INFO: reactions 1360\n",
      "1724811783.8364382 INFO: metabolites 1283\n",
      "1724811783.8381326 INFO: reactions 1249\n",
      "1724811783.8946066 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811783.9155424 INFO: reactions 1079\n",
      "1724811783.9434803 INFO: metabolites 1268\n",
      "1724811783.9716892 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811783.9775991 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811783.9854598 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.021837 INFO: Default biomass: [bio1]\n",
      "1724811784.0275178 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.0451171 INFO: reactions 1210\n",
      "1724811784.0510092 INFO: reactions 1360\n",
      "1724811784.0576797 INFO: reactions 1244\n",
      "1724811784.0742307 INFO: Default biomass: [bio1]\n",
      "1724811784.15522 INFO: reactions 1241\n",
      "1724811784.1919417 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.2209954 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.226675 INFO: Default biomass: [bio1]\n",
      "1724811784.240433 INFO: reactions 1357\n",
      "1724811784.2656126 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.315124 INFO: Default biomass: [bio1]\n",
      "1724811784.319256 INFO: metabolites 1207\n",
      "1724811784.3537889 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.3561106 INFO: reactions 1332\n",
      "1724811784.3971436 INFO: Default biomass: [bio1]\n",
      "1724811784.4070117 INFO: Default biomass: [bio1]\n",
      "1724811784.4245973 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.451959 INFO: reactions 1349\n",
      "1724811784.4532776 INFO: Default biomass: [bio1]\n",
      "1724811784.4689133 INFO: Default biomass: [bio1]\n",
      "1724811784.475552 INFO: reactions 1389\n",
      "1724811784.479794 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.4985628 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.5202372 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.9.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811784.5692275 INFO: reactions 1330\n",
      "1724811784.582058 INFO: metabolites 1324\n",
      "1724811784.597096 INFO: Default biomass: [bio1]\n",
      "1724811784.59788 INFO: metabolites 1208\n",
      "1724811784.6246507 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.6445768 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.6700976 INFO: reactions 1356\n",
      "1724811784.7115364 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.711699 INFO: Default biomass: [bio1]\n",
      "1724811784.7134817 INFO: Default biomass: [bio1]\n",
      "1724811784.7692757 INFO: Default biomass: [bio1]\n",
      "1724811784.8400195 INFO: metabolites 1183\n",
      "1724811784.8647065 INFO: Default biomass: [bio1]\n",
      "1724811784.8724375 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811784.8930192 INFO: metabolites 1007\n",
      "1724811784.94249 INFO: reactions 1267\n",
      "1724811784.9752884 INFO: metabolites 1267\n",
      "1724811785.0010438 INFO: Default biomass: [bio1]\n",
      "1724811785.0201313 INFO: Default biomass: [bio1]\n",
      "1724811785.0668995 INFO: metabolites 1237\n",
      "1724811785.0813084 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811785.1960459 INFO: metabolites 1237\n",
      "1724811785.2067301 INFO: Default biomass: [bio1]\n",
      "1724811785.207873 INFO: reactions 1250\n",
      "1724811785.2214642 INFO: metabolites 1287\n",
      "1724811785.224362 INFO: Default biomass: [bio1]\n",
      "1724811785.2295578 INFO: metabolites 1294\n",
      "1724811785.2307801 INFO: metabolites 1237\n",
      "1724811785.2370403 INFO: Default biomass: [bio1]\n",
      "1724811785.2775755 INFO: reactions 1290\n",
      "1724811785.3019142 INFO: reactions 1328\n",
      "1724811785.3298874 INFO: reactions 1039\n",
      "1724811785.3649795 INFO: metabolites 1088\n",
      "1724811785.3665066 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811785.4116757 INFO: Default biomass: [bio1]\n",
      "1724811785.4400496 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811785.4414158 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811785.4633498 INFO: metabolites 1232\n",
      "1724811785.572559 INFO: metabolites 1188\n",
      "1724811785.5850084 INFO: metabolites 1246\n",
      "1724811785.5965693 INFO: Default biomass: [bio1]\n",
      "1724811785.6437905 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811785.682135 INFO: metabolites 1120\n",
      "1724811785.6894214 INFO: reactions 1272\n",
      "1724811785.7293792 INFO: reactions 1297\n",
      "1724811785.8072298 INFO: metabolites 1246\n",
      "1724811785.8259993 INFO: reactions 1297\n",
      "1724811785.850813 INFO: reactions 1297\n",
      "1724811785.8513396 INFO: metabolites 1370\n",
      "1724811785.8708823 INFO: reactions 1323\n",
      "1724811785.884987 INFO: reactions 1089\n",
      "1724811785.8907545 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811785.9024885 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811785.9249194 INFO: reactions 1406\n",
      "1724811785.9738784 INFO: Default biomass: [bio1]\n",
      "1724811785.9776487 INFO: Default biomass: [bio1]\n",
      "1724811785.9832468 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811786.0083606 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811786.008267 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811786.0357597 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811786.0549717 INFO: metabolites 1267\n",
      "1724811786.0778337 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811786.1026304 INFO: metabolites 1182\n",
      "1724811786.2144568 INFO: reactions 1210\n",
      "1724811786.2977288 INFO: reactions 1332\n",
      "1724811786.3448322 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811786.3689036 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811786.4227607 INFO: Default biomass: [bio1]\n",
      "1724811786.4474223 INFO: Default biomass: [bio1]\n",
      "1724811786.4678996 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811786.5096738 INFO: reactions 1287\n",
      "1724811786.5282009 INFO: metabolites 1287\n",
      "1724811786.5670235 INFO: Default biomass: [bio1]\n",
      "1724811786.655189 INFO: Default biomass: [bio1]\n",
      "1724811786.6643083 INFO: reactions 1187\n",
      "1724811786.699951 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811786.8573365 INFO: Default biomass: [bio1]\n",
      "1724811786.8690567 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811786.889969 INFO: reactions 1272\n",
      "1724811786.988619 INFO: Default biomass: [bio1]\n",
      "1724811787.0047936 INFO: Default biomass: [bio1]\n",
      "1724811787.1161036 INFO: metabolites 1291\n",
      "1724811787.1691778 INFO: Default biomass: [bio1]\n",
      "1724811787.1855152 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811787.1897833 INFO: reactions 1323\n",
      "1724811787.2072508 INFO: reactions 1246\n",
      "1724811787.2390625 INFO: metabolites 1268\n",
      "1724811787.2820034 INFO: Default biomass: [bio1]\n",
      "1724811787.3486738 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811787.3513758 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811787.3520176 INFO: Default biomass: [bio1]\n",
      "1724811787.385595 INFO: reactions 1160\n",
      "1724811787.4537737 INFO: metabolites 1073\n",
      "1724811787.5234623 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811787.526201 INFO: Default biomass: [bio1]\n",
      "1724811787.5348597 INFO: Default biomass: [bio1]\n",
      "1724811787.580605 INFO: Default biomass: [bio1]\n",
      "1724811787.616333 INFO: metabolites 1182\n",
      "1724811787.7133172 INFO: metabolites 1192\n",
      "1724811787.7874675 INFO: reactions 1466\n",
      "1724811787.7998638 INFO: metabolites 1073\n",
      "1724811787.8477454 INFO: reactions 1287\n",
      "1724811787.854468 INFO: Default biomass: [bio1]\n",
      "1724811787.9051704 INFO: Default biomass: [bio1]\n",
      "1724811787.9303296 INFO: reactions 1117\n",
      "1724811787.9526353 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811787.9965055 INFO: Default biomass: [bio1]\n",
      "1724811788.0457795 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811788.0470772 INFO: metabolites 1291\n",
      "1724811788.0551786 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811788.0700097 INFO: metabolites 1183\n",
      "1724811788.1303844 INFO: metabolites 1342\n",
      "1724811788.1911137 INFO: reactions 1187\n",
      "1724811788.1983113 INFO: metabolites 1291\n",
      "1724811788.2428434 INFO: metabolites 1291\n",
      "1724811788.2757711 INFO: reactions 1325\n",
      "1724811788.365113 INFO: reactions 1117\n",
      "1724811788.4055295 INFO: metabolites 1294\n",
      "1724811788.4192228 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811788.418593 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811788.4406579 INFO: metabolites 1245\n",
      "1724811788.4808557 INFO: metabolites 1134\n",
      "1724811788.494051 INFO: Default biomass: [bio1]\n",
      "1724811788.501656 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811788.5386283 INFO: metabolites 1323\n",
      "1724811788.5576046 INFO: metabolites 1178\n",
      "1724811788.5676923 INFO: Default biomass: [bio1]\n",
      "1724811788.6214402 INFO: reactions 1332\n",
      "1724811788.734657 INFO: reactions 1346\n",
      "1724811788.7408986 INFO: Default biomass: [bio1]\n",
      "1724811788.747014 INFO: reactions 1349\n",
      "1724811788.758477 INFO: Default biomass: [bio1]\n",
      "1724811788.771403 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811788.836826 INFO: reactions 1356\n",
      "1724811788.8439982 INFO: metabolites 1246\n",
      "1724811788.8895743 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811788.9278977 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811788.9328184 INFO: reactions 1432\n",
      "1724811788.933473 INFO: Default biomass: [bio1]\n",
      "1724811789.0300322 INFO: metabolites 1073\n",
      "1724811789.0400002 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.0571797 INFO: Default biomass: [bio1]\n",
      "1724811789.061073 INFO: reactions 1195\n",
      "1724811789.0621965 INFO: Default biomass: [bio1]\n",
      "1724811789.0759685 INFO: reactions 1349\n",
      "1724811789.088039 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.1221793 INFO: metabolites 1228\n",
      "1724811789.140203 INFO: reactions 1423\n",
      "1724811789.1503758 INFO: reactions 1406\n",
      "1724811789.180571 INFO: reactions 1349\n",
      "1724811789.1894817 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.2400026 INFO: metabolites 1108\n",
      "1724811789.2585938 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.2867115 INFO: reactions 1231\n",
      "1724811789.2869267 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.297857 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.408021 INFO: Default biomass: [bio1]\n",
      "1724811789.4118679 INFO: Default biomass: [bio1]\n",
      "1724811789.4131975 INFO: reactions 1349\n",
      "1724811789.4272172 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.4299374 INFO: metabolites 1320\n",
      "1724811789.450023 INFO: reactions 1283\n",
      "1724811789.454227 INFO: metabolites 1183\n",
      "1724811789.4627712 INFO: Default biomass: [bio1]\n",
      "1724811789.493983 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.4973378 INFO: reactions 1117\n",
      "1724811789.5606341 INFO: metabolites 1228\n",
      "1724811789.5707061 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.580421 INFO: metabolites 1246\n",
      "1724811789.5952692 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.621025 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.653398 INFO: Default biomass: [bio1]\n",
      "1724811789.6572602 INFO: Default biomass: [bio1]\n",
      "1724811789.7505627 INFO: reactions 1290\n",
      "1724811789.762028 INFO: Default biomass: [bio1]\n",
      "1724811789.797427 INFO: Default biomass: [bio1]\n",
      "1724811789.8267057 INFO: Default biomass: [bio1]\n",
      "1724811789.8581445 INFO: reactions 1181\n",
      "1724811789.8700132 INFO: Default biomass: [bio1]\n",
      "1724811789.9035292 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.945535 INFO: reactions 1332\n",
      "1724811789.9771554 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811789.9820573 INFO: Default biomass: [bio1]\n",
      "1724811789.9914796 INFO: Default biomass: [bio1]\n",
      "1724811790.0097888 INFO: metabolites 1246\n",
      "1724811790.051407 INFO: Default biomass: [bio1]\n",
      "1724811790.0871654 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811790.089402 INFO: Default biomass: [bio1]\n",
      "1724811790.0949054 INFO: metabolites 1360\n",
      "1724811790.097515 INFO: metabolites 1190\n",
      "1724811790.1375647 INFO: reactions 1378\n",
      "1724811790.1677845 INFO: reactions 1290\n",
      "1724811790.1731486 INFO: Default biomass: [bio1]\n",
      "1724811790.2104723 INFO: metabolites 1313\n",
      "1724811790.255853 INFO: reactions 1283\n",
      "1724811790.2920258 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811790.3074625 INFO: metabolites 1325\n",
      "1724811790.3200758 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811790.3376079 INFO: metabolites 1202\n",
      "1724811790.4119565 INFO: Default biomass: [bio1]\n",
      "1724811790.4187615 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811790.4368322 INFO: metabolites 1190\n",
      "1724811790.4394672 INFO: Default biomass: [bio1]\n",
      "1724811790.4887693 INFO: metabolites 1182\n",
      "1724811790.4995801 INFO: metabolites 1268\n",
      "1724811790.5832067 INFO: metabolites 1228\n",
      "1724811790.590673 INFO: Default biomass: [bio1]\n",
      "1724811790.6466625 INFO: reactions 1155\n",
      "1724811790.7105858 INFO: reactions 1287\n",
      "1724811790.7132933 INFO: metabolites 1190\n",
      "1724811790.7271128 INFO: metabolites 1237\n",
      "1724811790.7319477 INFO: metabolites 1377\n",
      "1724811790.7767367 INFO: metabolites 1324\n",
      "1724811790.843701 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811790.8543427 INFO: Default biomass: [bio1]\n",
      "1724811790.894108 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811790.9243379 INFO: Default biomass: [bio1]\n",
      "1724811790.9417148 INFO: reactions 1440\n",
      "1724811790.9574304 INFO: metabolites 1134\n",
      "1724811790.9713902 INFO: Default biomass: [bio1]\n",
      "1724811790.97534 INFO: reactions 1351\n",
      "1724811790.991765 INFO: reactions 1155\n",
      "1724811790.9944165 INFO: reactions 1377\n",
      "1724811791.028134 INFO: metabolites 1287\n",
      "1724811791.0547287 INFO: reactions 1187\n",
      "1724811791.112674 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.1471426 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.1593676 INFO: reactions 1356\n",
      "1724811791.1603897 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.1792815 INFO: metabolites 1246\n",
      "1724811791.1906865 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.2086704 INFO: reactions 1290\n",
      "1724811791.2642953 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.3340123 INFO: reactions 1297\n",
      "1724811791.3685126 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.3700888 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.3772683 INFO: reactions 1155\n",
      "1724811791.4351728 INFO: Default biomass: [bio1]\n",
      "1724811791.4377666 INFO: Default biomass: [bio1]\n",
      "1724811791.4402306 INFO: reactions 1410\n",
      "1724811791.4967437 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.5049226 INFO: reactions 1328\n",
      "1724811791.5417626 INFO: reactions 1195\n",
      "1724811791.5414548 INFO: metabolites 1261\n",
      "1724811791.5865655 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.603651 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.6410062 INFO: metabolites 1406\n",
      "1724811791.6638334 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.671869 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811791.6722589 INFO: Default biomass: [bio1]\n",
      "1724811791.6827302 INFO: reactions 1323\n",
      "1724811791.6991627 INFO: Default biomass: [bio1]\n",
      "1724811791.717485 INFO: Default biomass: [bio1]\n",
      "1724811791.756743 INFO: Default biomass: [bio1]\n",
      "1724811791.8775856 INFO: Default biomass: [bio1]\n",
      "1724811791.889396 INFO: reactions 1287\n",
      "1724811791.9226131 INFO: Default biomass: [bio1]\n",
      "1724811791.9551413 INFO: Default biomass: [bio1]\n",
      "1724811791.9730606 INFO: metabolites 1228\n",
      "1724811791.9870477 INFO: reactions 1249\n",
      "1724811792.0069678 INFO: Default biomass: [bio1]\n",
      "1724811792.03494 INFO: metabolites 1370\n",
      "1724811792.069259 INFO: Default biomass: [bio1]\n",
      "1724811792.0781357 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811792.081715 INFO: metabolites 832\n",
      "1724811792.1307313 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811792.1528718 INFO: Default biomass: [bio1]\n",
      "1724811792.1569023 INFO: Default biomass: [bio1]\n",
      "1724811792.225825 INFO: Default biomass: [bio1]\n",
      "1724811792.2424293 INFO: reactions 1388\n",
      "1724811792.3305657 INFO: metabolites 1327\n",
      "1724811792.3487902 INFO: metabolites 1287\n",
      "1724811792.3707213 INFO: metabolites 1246\n",
      "1724811792.3814673 INFO: reactions 892\n",
      "1724811792.4219282 INFO: metabolites 1313\n",
      "1724811792.4430342 INFO: metabolites 1287\n",
      "1724811792.4897523 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811792.508476 INFO: reactions 1505\n",
      "1724811792.5290167 INFO: metabolites 1313\n",
      "1724811792.561397 INFO: metabolites 1246\n",
      "1724811792.5764315 INFO: metabolites 1291\n",
      "1724811792.584469 INFO: reactions 1290\n",
      "1724811792.656769 INFO: Default biomass: [bio1]\n",
      "1724811792.6975641 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811792.7162163 INFO: metabolites 1302\n",
      "1724811792.7336838 INFO: metabolites 1313\n",
      "1724811792.7404673 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.9.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811792.833883 INFO: metabolites 981\n",
      "1724811792.8395643 INFO: reactions 1466\n",
      "1724811792.8398159 INFO: metabolites 1261\n",
      "1724811792.8410642 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811792.993199 INFO: reactions 1283\n",
      "1724811792.9965012 INFO: reactions 1323\n",
      "1724811793.018077 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811793.1446724 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811793.2130227 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811793.2764888 INFO: reactions 1349\n",
      "1724811793.325472 INFO: metabolites 1237\n",
      "1724811793.3836899 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efee3654dd0>: 1, <ModelCompound cpd11416_c1 at 0x7eff40664050>: -0.09090909090909091, <ModelCompound cpd11416_c2 at 0x7eff3b2e4e10>: -0.09090909090909091, <ModelCompound cpd11416_c3 at 0x7eff39f90ed0>: -0.09090909090909091, <ModelCompound cpd11416_c4 at 0x7eff17ddca50>: -0.09090909090909091, <ModelCompound cpd11416_c5 at 0x7eff14d19710>: -0.09090909090909091, <ModelCompound cpd11416_c6 at 0x7eff0b7c80d0>: -0.09090909090909091, <ModelCompound cpd11416_c7 at 0x7eff06764fd0>: -0.09090909090909091, <ModelCompound cpd11416_c8 at 0x7eff016bbe10>: -0.09090909090909091, <ModelCompound cpd11416_c9 at 0x7efefc6c4910>: -0.09090909090909091, <ModelCompound cpd11416_c10 at 0x7efef76735d0>: -0.09090909090909091, <ModelCompound cpd11416_c11 at 0x7efef23e8bd0>: -0.09090909090909091}\n",
      "SK_cpd11416_c0"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811793.3945832 INFO: reactions 1345\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " {<Metabolite cpd11416_c0 at 0x7efee3654dd0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811793.399473 INFO: reactions 1283\n",
      "1724811793.4008348 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n",
      "1724811793.4489956 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811793.4668775 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811793.4758642 INFO: reactions 1377\n",
      "1724811793.4846559 INFO: reactions 1345\n",
      "1724811793.5405738 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811793.5421221 INFO: reactions 1306\n",
      "1724811793.6014235 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811793.6365983 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811793.6624484 INFO: Default biomass: [bio1]\n",
      "1724811793.7600117 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811793.8152761 INFO: Default biomass: [bio1]\n",
      "1724811793.8889008 INFO: Default biomass: [bio1]\n",
      "1724811793.9161048 INFO: Default biomass: [bio1]\n",
      "1724811794.017498 INFO: metabolites 1178\n",
      "1724811794.019324 INFO: Default biomass: [bio1]\n",
      "1724811794.037368 INFO: Default biomass: [bio1]\n",
      "1724811794.0552332 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811794.1252186 INFO: reactions 1297\n",
      "1724811794.1260083 INFO: Default biomass: [bio1]\n",
      "1724811794.2066927 INFO: metabolites 1228\n",
      "1724811794.2676072 INFO: Default biomass: [bio1]\n",
      "1724811794.291805 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811794.407574 INFO: reactions 946\n",
      "1724811794.4727983 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811794.4917345 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n",
      "1724811794.4966486 INFO: metabolites 1178\n",
      "1724811794.5179052 INFO: metabolites 1122\n",
      "1724811794.5853093 INFO: reactions 1384\n",
      "1724811794.6034157 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811794.6527956 INFO: reactions 1345\n",
      "1724811794.6547601 INFO: reactions 1231\n",
      "1724811794.7247312 INFO: Default biomass: [bio1]\n",
      "1724811794.743223 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811794.7434022 INFO: reactions 1323\n",
      "1724811794.7695127 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811794.7998633 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811794.8169792 INFO: Default biomass: [bio1]\n",
      "1724811794.8524702 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811794.9044461 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55\n",
      "1724811794.9111326 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811794.923441 INFO: reactions 1290\n",
      "1724811795.003408 INFO: Default biomass: [bio1]\n",
      "1724811795.0439472 INFO: Default biomass: [bio1]\n",
      "1724811795.079609 INFO: reactions 1231\n",
      "1724811795.096822 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811795.1129315 INFO: Default biomass: [bio1]\n",
      "1724811795.1185808 INFO: reactions 1117\n",
      "1724811795.135157 INFO: metabolites 1313\n",
      "1724811795.2309935 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811795.272018 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811795.296241 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811795.3132873 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811795.316976 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n",
      "1724811795.345147 INFO: Default biomass: [bio1]\n",
      "1724811795.3760397 INFO: metabolites 1320\n",
      "1724811795.3777568 INFO: Default biomass: [bio1]\n",
      "1724811795.4063728 INFO: metabolites 1290\n",
      "1724811795.4183595 INFO: metabolites 1291\n",
      "1724811795.4520319 INFO: Default biomass: [bio1]\n",
      "1724811795.4618711 INFO: Default biomass: [bio1]\n",
      "1724811795.599951 INFO: metabolites 1254\n",
      "1724811795.6651092 INFO: Default biomass: [bio1]\n",
      "1724811795.6782386 INFO: metabolites 1287\n",
      "1724811795.7238407 INFO: metabolites 1182\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811795.7295544 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n",
      "1724811795.740062 INFO: Default biomass: [bio1]\n",
      "1724811795.7595265 INFO: Default biomass: [bio1]\n",
      "1724811795.8700283 INFO: reactions 1345\n",
      "1724811795.8987746 INFO: metabolites 1290\n",
      "1724811795.910293 INFO: metabolites 1327\n",
      "1724811795.929044 INFO: metabolites 1204\n",
      "1724811795.9603846 INFO: metabolites 1261\n",
      "1724811795.959975 INFO: metabolites 1237\n",
      "1724811795.973079 INFO: metabolites 1178\n",
      "1724811795.9907634 INFO: metabolites 1207\n",
      "1724811796.0409338 INFO: reactions 1315\n",
      "1724811796.0444841 INFO: metabolites 1257\n",
      "1724811796.081016 INFO: reactions 1378\n",
      "1724811796.105547 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.1192765 INFO: reactions 1349\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811796.1432 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n",
      "1724811796.1964054 INFO: metabolites 1287\n",
      "1724811796.2005672 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.2343478 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.2472038 INFO: reactions 1281\n",
      "1724811796.29863 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.322238 INFO: reactions 1323\n",
      "1724811796.3412974 INFO: reactions 1187\n",
      "1724811796.3699155 INFO: metabolites 1237\n",
      "1724811796.3927608 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.411134 INFO: metabolites 1323\n",
      "1724811796.459652 INFO: metabolites 1190\n",
      "1724811796.4788952 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.5267026 INFO: reactions 1315\n",
      "1724811796.5417907 INFO: reactions 1231\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811796.5558095 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44\n",
      "1724811796.5591471 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.5854118 INFO: reactions 1297\n",
      "1724811796.614212 INFO: reactions 1267\n",
      "1724811796.6200328 INFO: reactions 1295\n",
      "1724811796.6672554 INFO: reactions 1384\n",
      "1724811796.6765366 INFO: reactions 1306\n",
      "1724811796.6881444 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.7058976 INFO: Default biomass: [bio1]\n",
      "1724811796.7088919 INFO: Default biomass: [bio1]\n",
      "1724811796.7322938 INFO: reactions 1315\n",
      "1724811796.7407966 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.740405 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.752065 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.7548203 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.7742937 INFO: Default biomass: [bio1]\n",
      "1724811796.8232038 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.8347487 INFO: Default biomass: [bio1]\n",
      "1724811796.8589785 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.8824224 INFO: reactions 1323\n",
      "1724811796.8924425 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811796.8996356 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811796.9680882 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55\n",
      "1724811797.0010338 INFO: Default biomass: [bio1]\n",
      "1724811797.0138726 INFO: reactions 1297\n",
      "1724811797.0269394 INFO: reactions 1155\n",
      "1724811797.0379663 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811797.0525293 INFO: Default biomass: [bio1]\n",
      "1724811797.1569963 INFO: reactions 1349\n",
      "1724811797.1608033 INFO: Default biomass: [bio1]\n",
      "1724811797.2031996 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811797.2256973 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811797.2400541 INFO: Default biomass: [bio1]\n",
      "1724811797.2494347 INFO: Default biomass: [bio1]\n",
      "1724811797.2540045 INFO: Default biomass: [bio1]\n",
      "1724811797.273845 INFO: Default biomass: [bio1]\n",
      "1724811797.2999904 INFO: metabolites 1306\n",
      "1724811797.3129857 INFO: metabolites 1300\n",
      "1724811797.314354 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811797.3586824 INFO: metabolites 1251\n",
      "1724811797.3686352 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811797.3818564 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811797.3847952 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13\n",
      "1724811797.4237356 INFO: Default biomass: [bio1]\n",
      "1724811797.441322 INFO: metabolites 1320\n",
      "1724811797.492412 INFO: metabolites 1178\n",
      "1724811797.523881 INFO: metabolites 1181\n",
      "1724811797.5676568 INFO: Default biomass: [bio1]\n",
      "1724811797.6019173 INFO: metabolites 1246\n",
      "1724811797.6515677 INFO: metabolites 1073\n",
      "1724811797.6953368 INFO: Default biomass: [bio1]\n",
      "1724811797.7460375 INFO: metabolites 1237\n",
      "1724811797.747026 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811797.808821 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n",
      "1724811797.8279276 INFO: metabolites 1377\n",
      "1724811797.828649 INFO: metabolites 1257\n",
      "1724811797.8589666 INFO: metabolites 1291\n",
      "1724811797.9121516 INFO: reactions 1390\n",
      "1724811797.928259 INFO: Default biomass: [bio1]\n",
      "1724811797.9494905 INFO: reactions 1360\n",
      "1724811797.9509513 INFO: metabolites 1261\n",
      "1724811797.9700546 INFO: metabolites 1128\n",
      "1724811797.9782166 INFO: metabolites 1162\n",
      "1724811798.0238264 INFO: reactions 1402\n",
      "1724811798.0538003 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.0890548 INFO: reactions 1231\n",
      "1724811798.0934908 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.1051857 INFO: metabolites 1178\n",
      "1724811798.1535022 INFO: reactions 1079\n",
      "1724811798.1636665 INFO: reactions 1201\n",
      "1724811798.1997147 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811798.2337384 INFO: reactions 1283\n",
      "1724811798.2355864 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n",
      "1724811798.2391357 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.3115258 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.3396351 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.340701 INFO: reactions 1297\n",
      "1724811798.3513372 INFO: metabolites 1291\n",
      "1724811798.37892 INFO: metabolites 1313\n",
      "1724811798.382994 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.4803677 INFO: metabolites 981\n",
      "1724811798.4971817 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.518195 INFO: reactions 1315\n",
      "1724811798.523505 INFO: reactions 1244\n",
      "1724811798.5458329 INFO: reactions 1410\n",
      "1724811798.5486526 INFO: reactions 1349\n",
      "1724811798.5756927 INFO: Default biomass: [bio1]\n",
      "1724811798.5948684 INFO: reactions 1306\n",
      "1724811798.6249418 INFO: Default biomass: [bio1]\n",
      "1724811798.6484 INFO: reactions 1188\n",
      "1724811798.6590626 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.6764119 INFO: reactions 1231\n",
      "1724811798.6783586 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.7115843 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.7178004 INFO: Default biomass: [bio1]\n",
      "1724811798.7276795 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.761711 INFO: Default biomass: [bio1]\n",
      "1724811798.7790692 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.82065 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.8318791 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.8365521 INFO: Default biomass: [bio1]\n",
      "1724811798.8849022 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.9026382 INFO: Default biomass: [bio1]\n",
      "1724811798.921478 INFO: Default biomass: [bio1]\n",
      "1724811798.9267018 INFO: reactions 946\n",
      "1724811798.940043 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811798.9976416 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.13.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811799.1080766 INFO: metabolites 1205\n",
      "1724811799.113943 INFO: reactions 1345\n",
      "1724811799.117382 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811799.1391883 INFO: Default biomass: [bio1]\n",
      "1724811799.1597102 INFO: reactions 1378\n",
      "1724811799.2220516 INFO: Default biomass: [bio1]\n",
      "1724811799.2315233 INFO: metabolites 1406\n",
      "1724811799.284162 INFO: metabolites 1237\n",
      "1724811799.289783 INFO: metabolites 1204\n",
      "1724811799.2915497 INFO: Default biomass: [bio1]\n",
      "1724811799.3058157 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811799.3126833 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811799.3303196 INFO: Default biomass: [bio1]\n",
      "1724811799.336757 INFO: Default biomass: [bio1]\n",
      "1724811799.349418 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811799.4003766 INFO: Default biomass: [bio1]\n",
      "1724811799.5171607 INFO: Default biomass: [bio1]\n",
      "1724811799.5874782 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811799.6269088 INFO: metabolites 1207\n",
      "1724811799.629438 INFO: metabolites 1291\n",
      "1724811799.6731374 INFO: metabolites 1331\n",
      "1724811799.809326 INFO: reactions 1222\n",
      "1724811799.8332791 INFO: metabolites 1302\n",
      "1724811799.8339472 INFO: reactions 1295\n",
      "1724811799.8399444 INFO: metabolites 1328\n",
      "1724811799.8516536 INFO: Default biomass: [bio1]\n",
      "1724811799.864353 INFO: metabolites 1162\n",
      "1724811799.8655477 INFO: Default biomass: [bio1]\n",
      "1724811799.8895762 INFO: reactions 1297\n",
      "1724811799.9632206 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811799.9726849 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811800.048888 INFO: metabolites 1328\n",
      "1724811800.0530114 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811800.0960095 INFO: reactions 1349\n",
      "1724811800.1039124 INFO: reactions 1505\n",
      "1724811800.253009 INFO: reactions 1267\n",
      "1724811800.2757962 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811800.288373 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811800.3243175 INFO: reactions 1349\n",
      "1724811800.387228 INFO: metabolites 1257\n",
      "1724811800.3982089 INFO: reactions 1436\n",
      "1724811800.4043343 INFO: reactions 1244\n",
      "1724811800.4695737 INFO: Default biomass: [bio1]\n",
      "1724811800.4764423 INFO: metabolites 1313\n",
      "1724811800.5059721 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811800.5140898 INFO: reactions 1345\n",
      "1724811800.5415938 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811800.5513968 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811800.560823 INFO: Default biomass: [bio1]\n",
      "1724811800.5815892 INFO: reactions 1409\n",
      "1724811800.6958883 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811800.7734463 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811800.8354447 INFO: Default biomass: [bio1]\n",
      "1724811800.8492308 INFO: reactions 1409\n",
      "1724811800.9628983 INFO: metabolites 1190\n",
      "1724811801.023078 INFO: metabolites 1237\n",
      "1724811801.0379794 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811801.0660243 INFO: metabolites 1377\n",
      "1724811801.0680513 INFO: reactions 1315\n",
      "1724811801.0809045 INFO: Default biomass: [bio1]\n",
      "1724811801.1282516 INFO: Default biomass: [bio1]\n",
      "1724811801.1616335 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811801.1734715 INFO: metabolites 1291\n",
      "1724811801.223035 INFO: reactions 1345\n",
      "1724811801.2245064 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811801.28462 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efedbbec650>: 1, <ModelCompound cpd11416_c1 at 0x7eff40828c50>: -0.06666666666666667, <ModelCompound cpd11416_c2 at 0x7eff3b848150>: -0.06666666666666667, <ModelCompound cpd11416_c3 at 0x7eff3ac7a090>: -0.06666666666666667, <ModelCompound cpd11416_c4 at 0x7eff39f31c10>: -0.06666666666666667, <ModelCompound cpd11416_c5 at 0x7eff392653d0>: -0.06666666666666667, <ModelCompound cpd11416_c6 at 0x7eff175bd190>: -0.06666666666666667, <ModelCompound cpd11416_c7 at 0x7eff0df49cd0>: -0.06666666666666667, <ModelCompound cpd11416_c8 at 0x7eff0b2d5490>: -0.06666666666666667, <ModelCompound cpd11416_c9 at 0x7eff06415210>: -0.06666666666666667, <ModelCompound cpd11416_c10 at 0x7eff017d9d10>: -0.06666666666666667, <ModelCompound cpd11416_c11 at 0x7efefc4e9250>: -0.06666666666666667, <ModelCompound cpd11416_c12 at 0x7efefc5bd910>: -0.06666666666666667, <ModelCompound cpd11416_c13 at 0x7efef4974f50>: -0.06666666666666667, <ModelCompound cpd11416_c14 at 0x7efeefafcfd0>: -0.06666666666666667, <ModelCompound cpd11416_c15 at 0x7efeeacfa9d0>: -0.06666666666666667}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efedbbec650>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811801.313297 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_metabat.8\n",
      "1724811801.423568 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811801.4369106 INFO: metabolites 1190\n",
      "1724811801.519134 INFO: reactions 1155\n",
      "1724811801.577843 INFO: metabolites 1207\n",
      "1724811801.609146 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811801.6247633 INFO: reactions 1297\n",
      "1724811801.6453424 INFO: Default biomass: [bio1]\n",
      "1724811801.722433 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811801.7603636 INFO: Default biomass: [bio1]\n",
      "1724811801.785421 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaG_R1_A_H2O_MG_DASTool_bins_metabat.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811801.8327093 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.19\n",
      "1724811802.001665 INFO: reactions 1155\n",
      "1724811802.0054133 INFO: Default biomass: [bio1]\n",
      "1724811802.1060739 INFO: Default biomass: [bio1]\n",
      "1724811802.1289105 INFO: Default biomass: [bio1]\n",
      "1724811802.1780193 INFO: metabolites 1205\n",
      "1724811802.2060142 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811802.2114942 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811802.352655 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.67\n",
      "1724811802.3655515 INFO: Default biomass: [bio1]\n",
      "1724811802.4986162 INFO: Default biomass: [bio1]\n",
      "1724811802.5806105 INFO: metabolites 1073\n",
      "1724811802.6090014 INFO: Default biomass: [bio1]\n",
      "1724811802.6435862 INFO: metabolites 1306\n",
      "1724811802.6978123 INFO: Default biomass: [bio1]\n",
      "1724811802.725719 INFO: metabolites 1313\n",
      "1724811802.80417 INFO: reactions 1214\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.67\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811802.865736 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.9\n",
      "1724811802.8782325 INFO: Default biomass: [bio1]\n",
      "1724811802.9056625 INFO: metabolites 1328\n",
      "1724811802.945502 INFO: metabolites 1291\n",
      "1724811802.95989 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811802.997597 INFO: reactions 1410\n",
      "1724811803.0706003 INFO: metabolites 1254\n",
      "1724811803.0761113 INFO: metabolites 1190\n",
      "1724811803.0859976 INFO: reactions 1079\n",
      "1724811803.163159 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811803.1988466 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811803.3109288 INFO: metabolites 1320\n",
      "1724811803.3440778 INFO: reactions 1349\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_concoct_out.9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811803.3723204 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_metabat.14\n",
      "1724811803.375005 INFO: metabolites 1272\n",
      "1724811803.3754027 INFO: reactions 1402\n",
      "1724811803.4038694 INFO: reactions 1267\n",
      "1724811803.4418657 INFO: Default biomass: [bio1]\n",
      "1724811803.4519916 INFO: reactions 1345\n",
      "1724811803.536463 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811803.5502086 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811803.5985093 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811803.622985 INFO: Default biomass: [bio1]\n",
      "1724811803.6258056 INFO: reactions 1155\n",
      "1724811803.6404681 INFO: reactions 1349\n",
      "1724811803.6459248 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811803.6728902 INFO: metabolites 1288\n",
      "1724811803.696386 INFO: reactions 1409\n",
      "1724811803.7035322 INFO: metabolites 1328\n",
      "1724811803.7279897 INFO: reactions 1281\n",
      "1724811803.7943738 INFO: Default biomass: [bio1]\n",
      "1724811803.821466 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811803.8219664 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811803.866628 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811803.8828738 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811803.9949994 INFO: metabolites 1128\n",
      "1724811804.0806134 INFO: Default biomass: [bio1]\n",
      "1724811804.0939918 INFO: reactions 1378\n",
      "1724811804.1097984 INFO: Default biomass: [bio1]\n",
      "1724811804.1126812 INFO: Default biomass: [bio1]\n",
      "1724811804.119711 INFO: metabolites 1228\n",
      "1724811804.1616652 INFO: reactions 1327\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.13.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811804.197007 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811804.2124226 INFO: Default biomass: [bio1]\n",
      "1724811804.2490542 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811804.2918177 INFO: Default biomass: [bio1]\n",
      "1724811804.3176556 INFO: reactions 1317\n",
      "1724811804.3199356 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811804.3290195 INFO: metabolites 1302\n",
      "1724811804.3639235 INFO: Default biomass: [bio1]\n",
      "1724811804.3751857 INFO: Default biomass: [bio1]\n",
      "1724811804.4339018 INFO: reactions 1409\n",
      "1724811804.469626 INFO: Default biomass: [bio1]\n",
      "1724811804.4966028 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811804.5232518 INFO: reactions 1188\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811804.601068 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_metabat.16\n",
      "1724811804.6183 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811804.6510594 INFO: metabolites 1257\n",
      "1724811804.6519794 INFO: metabolites 1190\n",
      "1724811804.6698732 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811804.7808466 INFO: reactions 1314\n",
      "1724811804.8038368 INFO: Default biomass: [bio1]\n",
      "1724811804.8354058 INFO: metabolites 1306\n",
      "1724811804.8390062 INFO: Default biomass: [bio1]\n",
      "1724811804.8683798 INFO: metabolites 1261\n",
      "1724811804.9198892 INFO: metabolites 1328\n",
      "1724811804.9285936 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811804.9750938 INFO: metabolites 1190\n",
      "1724811805.0031974 INFO: reactions 1345\n",
      "1724811805.0313811 INFO: Default biomass: [bio1]\n",
      "1724811805.0975618 INFO: metabolites 1335\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R2_A_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811805.1148295 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.11\n",
      "1724811805.1507316 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811805.1596968 INFO: Default biomass: [bio1]\n",
      "1724811805.1814244 INFO: Default biomass: [bio1]\n",
      "1724811805.202537 INFO: reactions 1155\n",
      "1724811805.33687 INFO: reactions 1315\n",
      "1724811805.4010346 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811805.4255288 INFO: reactions 1388\n",
      "1724811805.4291322 INFO: metabolites 1313\n",
      "1724811805.4662666 INFO: Default biomass: [bio1]\n",
      "1724811805.482656 INFO: metabolites 1288\n",
      "1724811805.496248 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.13.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811805.5621202 INFO: reactions 1155\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811805.5651195 INFO: reactions 1402\n",
      "1724811805.5757787 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811805.6278665 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.29\n",
      "1724811805.6536064 INFO: reactions 1409\n",
      "1724811805.668042 INFO: metabolites 1335\n",
      "1724811805.6936707 INFO: Default biomass: [bio1]\n",
      "1724811805.7360435 INFO: metabolites 1335\n",
      "1724811805.7489994 INFO: metabolites 1335\n",
      "1724811805.7655165 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811805.7765288 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811805.8377945 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811805.8715386 INFO: Default biomass: [bio1]\n",
      "1724811805.8856213 INFO: reactions 1379\n",
      "1724811806.0568528 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811806.0604076 INFO: metabolites 1282\n",
      "1724811806.097413 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811806.1359568 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.29"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811806.1384826 INFO: reactions 1317\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811806.140828 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.45\n",
      "1724811806.1694496 INFO: reactions 1345\n",
      "1724811806.186002 INFO: metabolites 1254\n",
      "1724811806.2711103 INFO: Default biomass: [bio1]\n",
      "1724811806.325074 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811806.33042 INFO: Default biomass: [bio1]\n",
      "1724811806.378805 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811806.4041567 INFO: Default biomass: [bio1]\n",
      "1724811806.4435627 INFO: reactions 1379\n",
      "1724811806.5067732 INFO: reactions 1379\n",
      "1724811806.6045408 INFO: metabolites 1406\n",
      "1724811806.6193864 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811806.6238525 INFO: reactions 1379\n",
      "1724811806.6474013 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811806.6631312 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_metabat.14\n",
      "1724811806.6790462 INFO: reactions 1339\n",
      "1724811806.681188 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811806.7917821 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811806.7967746 INFO: reactions 1281\n",
      "1724811806.8262837 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811806.8506503 INFO: Default biomass: [bio1]\n",
      "1724811806.9543307 INFO: metabolites 1257\n",
      "1724811806.9863534 INFO: metabolites 1335\n",
      "1724811806.9893284 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811806.995878 INFO: metabolites 1313\n",
      "1724811807.176835 INFO: metabolites 1330\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811807.1847014 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n",
      "1724811807.1892 INFO: Default biomass: [bio1]\n",
      "1724811807.2580392 INFO: Default biomass: [bio1]\n",
      "1724811807.346153 INFO: Default biomass: [bio1]\n",
      "1724811807.3524458 INFO: Default biomass: [bio1]\n",
      "1724811807.3683698 INFO: Default biomass: [bio1]\n",
      "1724811807.4278412 INFO: reactions 1505\n",
      "1724811807.4936173 INFO: Default biomass: [bio1]\n",
      "1724811807.6140337 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811807.6359913 INFO: reactions 1315\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811807.7012827 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_H2O_MG_DASTool_bins_metabat.16\n",
      "1724811807.7509325 INFO: reactions 1379\n",
      "1724811807.7538471 INFO: metabolites 1213\n",
      "1724811807.764991 INFO: reactions 1345\n",
      "1724811807.7968612 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811807.8648183 INFO: metabolites 1330\n",
      "1724811807.8928943 INFO: metabolites 1241\n",
      "1724811807.9184382 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811807.9361267 INFO: metabolites 1330\n",
      "1724811807.9431772 INFO: metabolites 1331\n",
      "1724811808.0180056 INFO: metabolites 1330\n",
      "1724811808.1210246 INFO: Default biomass: [bio1]\n",
      "1724811808.1638374 INFO: metabolites 1220\n",
      "1724811808.2289662 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2_B_H2O_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811808.247651 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_H2O_MG_DASTool_bins_metabat.17\n",
      "1724811808.3915815 INFO: Default biomass: [bio1]\n",
      "1724811808.4771314 INFO: Default biomass: [bio1]\n",
      "1724811808.5031667 INFO: reactions 1300\n",
      "1724811808.521428 INFO: reactions 1311\n",
      "1724811808.5594425 INFO: reactions 1462\n",
      "1724811808.6297348 INFO: metabolites 1241\n",
      "1724811808.630307 INFO: reactions 1462\n",
      "1724811808.646764 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811808.720622 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811808.7254732 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811808.7651346 INFO: reactions 1436\n",
      "1724811808.7852452 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811808.8232968 INFO: metabolites 1313\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2_B_H2O_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811808.831997 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_concoct_out.35\n",
      "1724811808.9055662 INFO: metabolites 1241\n",
      "1724811808.913922 INFO: reactions 1293\n",
      "1724811809.072563 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811809.10586 INFO: metabolites 1330\n",
      "1724811809.1273096 INFO: reactions 1462\n",
      "1724811809.1899908 INFO: Default biomass: [bio1]\n",
      "1724811809.2751796 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811809.2877462 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811809.2886076 INFO: Default biomass: [bio1]\n",
      "1724811809.2884192 INFO: Default biomass: [bio1]\n",
      "1724811809.291453 INFO: reactions 1300\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_concoct_out.35"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811809.349706 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811809.3531737 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_concoct_out.60\n",
      "1724811809.439123 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811809.615585 INFO: Default biomass: [bio1]\n",
      "1724811809.715483 INFO: metabolites 1128\n",
      "1724811809.857041 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2_C_H2O_MG_DASTool_bins_concoct_out.60\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811809.8822868 INFO: Default biomass: [bio1]\n",
      "1724811809.9628394 INFO: Default biomass: [bio1]\n",
      "1724811809.9714987 INFO: metabolites 1345\n",
      "1724811810.0359411 INFO: reactions 1462\n",
      "1724811810.1299388 INFO: metabolites 1261\n",
      "1724811810.1927047 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811810.2108605 INFO: metabolites 1167\n",
      "1724811810.249902 INFO: reactions 1188\n",
      "1724811810.2791026 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811810.2961733 INFO: metabolites 1156\n",
      "1724811810.3958836 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811810.4069726 INFO: metabolites 1201\n",
      "1724811810.531113 INFO: metabolites 1257\n",
      "1724811810.5482974 INFO: metabolites 1218\n",
      "1724811810.7735868 INFO: Default biomass: [bio1]\n",
      "1724811810.7755115 INFO: reactions 1341\n",
      "1724811810.860422 INFO: Default biomass: [bio1]\n",
      "1724811810.8776815 INFO: Default biomass: [bio1]\n",
      "1724811810.8986895 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811810.925995 INFO: reactions 1345\n",
      "1724811810.9630597 INFO: reactions 1322\n",
      "1724811811.0160513 INFO: reactions 1462\n",
      "1724811811.1014612 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811811.1239944 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811811.1724594 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811811.2112348 INFO: reactions 1260\n",
      "1724811811.23254 INFO: reactions 1315\n",
      "1724811811.3177977 INFO: metabolites 1154\n",
      "1724811811.3925889 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811811.4089637 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811811.4288366 INFO: reactions 1300\n",
      "1724811811.4596722 INFO: metabolites 1335\n",
      "1724811811.5702589 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811811.600769 INFO: Default biomass: [bio1]\n",
      "1724811811.6319268 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811811.7323542 INFO: Default biomass: [bio1]\n",
      "1724811811.7399817 INFO: Default biomass: [bio1]\n",
      "1724811811.8761363 INFO: reactions 1173\n",
      "1724811811.944362 INFO: Default biomass: [bio1]\n",
      "1724811811.945278 INFO: Default biomass: [bio1]\n",
      "1724811811.9885912 INFO: reactions 1382\n",
      "1724811811.997518 INFO: reactions 1388\n",
      "1724811812.084348 INFO: Default biomass: [bio1]\n",
      "1724811812.1044745 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811812.142957 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811812.162708 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811812.248248 INFO: reactions 1379\n",
      "1724811812.3205209 INFO: metabolites 1345\n",
      "1724811812.3581026 INFO: metabolites 1257\n",
      "1724811812.4231102 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811812.5399947 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811812.5616693 INFO: metabolites 1328\n",
      "1724811812.5686157 INFO: metabolites 1128\n",
      "1724811812.5772314 INFO: Default biomass: [bio1]\n",
      "1724811812.6109307 INFO: metabolites 1328\n",
      "1724811812.6734362 INFO: Default biomass: [bio1]\n",
      "1724811812.7339818 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efed1f02590>: 1, <ModelCompound cpd11416_c1 at 0x7eff40664390>: -0.06666666666666667, <ModelCompound cpd11416_c2 at 0x7eff3b100790>: -0.06666666666666667, <ModelCompound cpd11416_c3 at 0x7eff39e23c90>: -0.06666666666666667, <ModelCompound cpd11416_c4 at 0x7eff17ca7fd0>: -0.06666666666666667, <ModelCompound cpd11416_c5 at 0x7eff14aecc50>: -0.06666666666666667, <ModelCompound cpd11416_c6 at 0x7eff0b294890>: -0.06666666666666667, <ModelCompound cpd11416_c7 at 0x7eff06337150>: -0.06666666666666667, <ModelCompound cpd11416_c8 at 0x7eff01499050>: -0.06666666666666667, <ModelCompound cpd11416_c9 at 0x7efefc33c150>: -0.06666666666666667, <ModelCompound cpd11416_c10 at 0x7efef746c3d0>: -0.06666666666666667, <ModelCompound cpd11416_c11 at 0x7efef233cfd0>: -0.06666666666666667, <ModelCompound cpd11416_c12 at 0x7efeeac3b210>: -0.06666666666666667, <ModelCompound cpd11416_c13 at 0x7efeed152a10>: -0.06666666666666667, <ModelCompound cpd11416_c14 at 0x7efee0ed61d0>: -0.06666666666666667, <ModelCompound cpd11416_c15 at 0x7efedbec8590>: -0.06666666666666667}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efed1f02590>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811812.7481396 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n",
      "1724811812.7987669 INFO: reactions 1156\n",
      "1724811812.9975226 INFO: Default biomass: [bio1]\n",
      "1724811813.0052726 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811813.0363743 INFO: reactions 1315\n",
      "1724811813.0842884 INFO: reactions 1188\n",
      "1724811813.0848787 INFO: reactions 1382\n",
      "1724811813.1468306 INFO: metabolites 1149\n",
      "1724811813.1600623 INFO: metabolites 1206\n",
      "1724811813.1634057 INFO: Default biomass: [bio1]\n",
      "1724811813.1930969 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811813.2247288 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811813.2551565 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811813.3234031 INFO: metabolites 1269\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811813.334837 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13\n",
      "1724811813.3469996 INFO: reactions 1409\n",
      "1724811813.4380093 INFO: reactions 1409\n",
      "1724811813.4832401 INFO: Default biomass: [bio1]\n",
      "1724811813.5358958 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811813.628157 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811813.6336002 INFO: metabolites 1330\n",
      "1724811813.6408427 INFO: reactions 1160\n",
      "1724811813.685754 INFO: Default biomass: [bio1]\n",
      "1724811813.6858618 INFO: reactions 1226\n",
      "1724811813.7169302 INFO: Default biomass: [bio1]\n",
      "1724811813.7305954 INFO: metabolites 1335\n",
      "1724811813.7706404 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811813.8153055 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811813.832211 INFO: Default biomass: [bio1]\n",
      "1724811813.8732114 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811813.9190636 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n",
      "1724811813.9614072 INFO: reactions 1352\n",
      "1724811814.0465481 INFO: metabolites 1345\n",
      "1724811814.1051226 INFO: Default biomass: [bio1]\n",
      "1724811814.1412864 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811814.2072623 INFO: metabolites 1241\n",
      "1724811814.2143877 INFO: Default biomass: [bio1]\n",
      "1724811814.2305655 INFO: Default biomass: [bio1]\n",
      "1724811814.2506783 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811814.251083 INFO: metabolites 1218\n",
      "1724811814.2802317 INFO: Default biomass: [bio1]\n",
      "1724811814.3834724 INFO: reactions 1462\n",
      "1724811814.4934325 INFO: reactions 1379\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811814.5040903 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55\n",
      "1724811814.5431073 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811814.6628413 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811814.6921084 INFO: metabolites 1335\n",
      "1724811814.7364824 INFO: Default biomass: [bio1]\n",
      "1724811814.782106 INFO: reactions 1300\n",
      "1724811814.798908 INFO: reactions 1382\n",
      "1724811814.9023077 INFO: reactions 1260\n",
      "1724811814.9198585 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811814.9690719 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811814.985243 INFO: metabolites 1335\n",
      "1724811815.0090933 INFO: metabolites 1230\n",
      "1724811815.0889144 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811815.112169 INFO: Default biomass: [bio1]\n",
      "1724811815.2280276 INFO: metabolites 1201\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efeceb148d0>: 1, <ModelCompound cpd11416_c1 at 0x7eff40660c50>: -0.06666666666666667, <ModelCompound cpd11416_c2 at 0x7eff3b100c50>: -0.06666666666666667, <ModelCompound cpd11416_c3 at 0x7eff39e280d0>: -0.06666666666666667, <ModelCompound cpd11416_c4 at 0x7eff17cb0390>: -0.06666666666666667, <ModelCompound cpd11416_c5 at 0x7eff14aecf50>: -0.06666666666666667, <ModelCompound cpd11416_c6 at 0x7eff0b290c50>: -0.06666666666666667, <ModelCompound cpd11416_c7 at 0x7eff06337550>: -0.06666666666666667, <ModelCompound cpd11416_c8 at 0x7eff01499390>: -0.06666666666666667, <ModelCompound cpd11416_c9 at 0x7efefc33c510>: -0.06666666666666667, <ModelCompound cpd11416_c10 at 0x7efef7470690>: -0.06666666666666667, <ModelCompound cpd11416_c11 at 0x7efef233d310>: -0.06666666666666667, <ModelCompound cpd11416_c12 at 0x7efeed3d1c50>: -0.06666666666666667, <ModelCompound cpd11416_c13 at 0x7efee82a5350>: -0.06666666666666667, <ModelCompound cpd11416_c14 at 0x7efee0a19a10>: -0.06666666666666667, <ModelCompound cpd11416_c15 at 0x7efedb8c59d0>: -0.06666666666666667}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efeceb148d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811815.2502697 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n",
      "1724811815.359152 INFO: Default biomass: [bio1]\n",
      "1724811815.4201474 INFO: Default biomass: [bio1]\n",
      "1724811815.4564936 INFO: reactions 1379\n",
      "1724811815.5398383 INFO: Default biomass: [bio1]\n",
      "1724811815.6128051 INFO: Default biomass: [bio1]\n",
      "1724811815.6269007 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811815.7344115 INFO: reactions 1277\n",
      "1724811815.7722673 INFO: reactions 1322\n",
      "1724811815.7934754 INFO: reactions 1379\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811815.8429127 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13\n",
      "1724811815.909316 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811815.931124 INFO: metabolites 1330\n",
      "1724811815.9511757 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811815.9676864 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811815.9825027 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n",
      "1724811816.0128021 INFO: metabolites 1218\n",
      "1724811816.0363688 INFO: metabolites 1201\n",
      "1724811816.1956937 INFO: Default biomass: [bio1]\n",
      "1724811816.2845862 INFO: metabolites 1335\n",
      "1724811816.4016266 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811816.464903 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n",
      "1724811816.4780185 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811816.5754652 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n",
      "1724811816.5887046 INFO: reactions 1322\n",
      "1724811816.6719155 INFO: reactions 1462\n",
      "1724811816.6959987 INFO: reactions 1260\n",
      "1724811816.7180896 INFO: metabolites 1330\n",
      "1724811816.7282379 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811816.8707163 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811816.8944993 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811816.9620998 INFO: metabolites 1048\n",
      "1724811817.0470998 INFO: metabolites 1272\n",
      "1724811817.0829496 INFO: reactions 1379\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811817.1775033 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n",
      "1724811817.2389567 INFO: Default biomass: [bio1]\n",
      "1724811817.2598615 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811817.2726247 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811817.4001882 INFO: Default biomass: [bio1]\n",
      "1724811817.404885 INFO: reactions 1076\n",
      "1724811817.4639661 INFO: Default biomass: [bio1]\n",
      "1724811817.5341659 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811817.7770016 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44\n",
      "1724811817.8508224 INFO: reactions 1313\n",
      "1724811817.860857 INFO: Default biomass: [bio1]\n",
      "1724811817.8784215 INFO: metabolites 1345\n",
      "1724811817.946569 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811817.9987638 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55\n",
      "1724811818.0204196 INFO: metabolites 1328\n",
      "1724811818.063669 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811818.0684302 INFO: metabolites 1156\n",
      "1724811818.0904837 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811818.3661332 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9\n",
      "1724811818.4084663 INFO: metabolites 1330\n",
      "1724811818.4810395 INFO: metabolites 1345\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811818.5900323 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n",
      "1724811818.608696 INFO: metabolites 1330\n",
      "1724811818.6109824 INFO: Default biomass: [bio1]\n",
      "1724811818.6415596 INFO: reactions 1382\n",
      "1724811818.6418867 INFO: reactions 1156\n",
      "1724811818.677067 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811818.7754838 INFO: reactions 1409\n",
      "1724811818.813611 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811818.8353856 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811818.8662612 INFO: reactions 1462\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811818.9565468 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42\n",
      "1724811818.9685125 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811819.0252268 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811819.1951308 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n",
      "1724811819.3236582 INFO: Default biomass: [bio1]\n",
      "1724811819.330694 INFO: reactions 1462\n",
      "1724811819.4058516 INFO: Default biomass: [bio1]\n",
      "1724811819.4918792 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811819.5549142 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55\n",
      "1724811819.6038024 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811819.7990353 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n",
      "1724811819.9114819 INFO: metabolites 1201\n",
      "1724811819.957167 INFO: metabolites 1345\n",
      "1724811820.1389194 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811820.1481714 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4\n",
      "1724811820.1959298 INFO: metabolites 1345\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811820.390329 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44\n",
      "1724811820.466996 INFO: reactions 1322\n",
      "1724811820.6075606 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811820.6543634 INFO: reactions 1462\n",
      "1724811820.7064502 INFO: reactions 1382\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811820.7334259 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13\n",
      "1724811820.7582495 INFO: metabolites 1345\n",
      "1724811820.7727482 INFO: reactions 1382\n",
      "1724811820.8304808 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811820.877459 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811820.8989403 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811820.9976249 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9\n",
      "1724811821.016329 INFO: reactions 1382\n",
      "1724811821.1009357 INFO: Default biomass: [bio1]\n",
      "1724811821.1334994 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811821.1917312 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811821.3264802 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n",
      "1724811821.448305 INFO: Default biomass: [bio1]\n",
      "1724811821.4544659 INFO: Default biomass: [bio1]\n",
      "1724811821.5204854 INFO: metabolites 1335\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811821.636048 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42\n",
      "1724811821.6615984 INFO: reactions 1382\n",
      "1724811821.7717931 INFO: Default biomass: [bio1]\n",
      "1724811821.8162894 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811821.8369083 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811821.9926515 INFO: metabolites 1154\n",
      "1724811822.0476303 INFO: metabolites 1314\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811822.236281 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55\n",
      "1724811822.2954993 INFO: reactions 1379\n",
      "1724811822.4202774 INFO: Default biomass: [bio1]\n",
      "1724811822.4441483 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811822.4654925 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811822.485528 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811822.718956 INFO: reactions 1173\n",
      "1724811822.8347511 INFO: reactions 1363\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811822.864504 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.40\n",
      "1724811822.923783 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811822.9343233 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13\n",
      "1724811822.9950004 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811823.0068114 INFO: Default biomass: [bio1]\n",
      "1724811823.0291975 INFO: Default biomass: [bio1]\n",
      "1724811823.3878605 INFO: Default biomass: [bio1]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.40\n",
      "12 Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811823.5251687 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17\n",
      "1724811823.5469737 INFO: metabolites 1330\n",
      "1724811823.5507846 INFO: Default biomass: [bio1]\n",
      "1724811823.5868876 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811823.8218515 INFO: metabolites 1314\n",
      "1724811823.9560463 INFO: metabolites 1345\n",
      "1724811824.254039 INFO: reactions 1462\n",
      "1724811824.4099147 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811824.5463712 INFO: reactions 1419\n",
      "1724811824.706908 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811824.7607284 INFO: reactions 1382\n",
      "1724811824.9317153 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811825.0008264 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24\n",
      "1724811825.0029514 INFO: Default biomass: [bio1]\n",
      "1724811825.2627926 INFO: Default biomass: [bio1]\n",
      "1724811825.5632198 INFO: Default biomass: [bio1]\n",
      "1724811825.570326 INFO: metabolites 1345\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811825.6629148 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n",
      "1724811825.921514 INFO: metabolites 1272\n",
      "1724811826.11536 INFO: metabolites 1314\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811826.3294063 INFO: reactions 1382\n",
      "1724811826.5010946 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811826.7223425 INFO: reactions 1313\n",
      "1724811826.9189749 INFO: reactions 1363\n",
      "1724811826.9387228 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811827.073463 INFO: Default biomass: [bio1]\n",
      "1724811827.0874953 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811827.11664 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811827.4919007 INFO: Default biomass: [bio1]\n",
      "1724811827.5540502 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1724811827.670617 INFO: Default biomass: [bio1]\n",
      "1724811827.7114089 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efeba06dfd0>: 1, <ModelCompound cpd11416_c1 at 0x7eff3bc9ccd0>: -0.05263157894736842, <ModelCompound cpd11416_c2 at 0x7eff3a62a210>: -0.05263157894736842, <ModelCompound cpd11416_c3 at 0x7eff4054c490>: -0.05263157894736842, <ModelCompound cpd11416_c4 at 0x7eff174b7f50>: -0.05263157894736842, <ModelCompound cpd11416_c5 at 0x7eff0b59c910>: -0.05263157894736842, <ModelCompound cpd11416_c6 at 0x7eff06436950>: -0.05263157894736842, <ModelCompound cpd11416_c7 at 0x7eff016ee490>: -0.05263157894736842, <ModelCompound cpd11416_c8 at 0x7efefc0e1610>: -0.05263157894736842, <ModelCompound cpd11416_c9 at 0x7eff017ef4d0>: -0.05263157894736842, <ModelCompound cpd11416_c10 at 0x7efef2342b10>: -0.05263157894736842, <ModelCompound cpd11416_c11 at 0x7efeed196c10>: -0.05263157894736842, <ModelCompound cpd11416_c12 at 0x7efee823f510>: -0.05263157894736842, <ModelCompound cpd11416_c13 at 0x7efee0d610d0>: -0.05263157894736842, <ModelCompound cpd11416_c14 at 0x7efee324e490>: -0.05263157894736842, <ModelCompound cpd11416_c15 at 0x7efed6e0ce10>: -0.05263157894736842, <ModelCompound cpd11416_c16 at 0x7efed5a396d0>: -0.05263157894736842, <ModelCompound cpd11416_c17 at 0x7efed42396d0>: -0.05263157894736842, <ModelCompound cpd11416_c18 at 0x7efecf0154d0>: -0.05263157894736842, <ModelCompound cpd11416_c19 at 0x7efed46d0ed0>: -0.05263157894736842}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efeba06dfd0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811831.9310145 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.46\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.46\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811832.7342467 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811833.4863055 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811834.2372136 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811834.9815674 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811835.724621 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.37\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.37\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811836.4699478 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811837.2198977 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.59\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.59\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811839.3144708 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efea47c9f90>: 1, <ModelCompound cpd11416_c1 at 0x7eff403f73d0>: -0.047619047619047616, <ModelCompound cpd11416_c2 at 0x7eff3ad6da10>: -0.047619047619047616, <ModelCompound cpd11416_c3 at 0x7eff39e5df10>: -0.047619047619047616, <ModelCompound cpd11416_c4 at 0x7eff17b87a90>: -0.047619047619047616, <ModelCompound cpd11416_c5 at 0x7eff14994d90>: -0.047619047619047616, <ModelCompound cpd11416_c6 at 0x7eff0b5f1a50>: -0.047619047619047616, <ModelCompound cpd11416_c7 at 0x7eff063e6010>: -0.047619047619047616, <ModelCompound cpd11416_c8 at 0x7eff01122b10>: -0.047619047619047616, <ModelCompound cpd11416_c9 at 0x7efefc4aac90>: -0.047619047619047616, <ModelCompound cpd11416_c10 at 0x7efef7579610>: -0.047619047619047616, <ModelCompound cpd11416_c11 at 0x7efef23a3990>: -0.047619047619047616, <ModelCompound cpd11416_c12 at 0x7efeeae5be10>: -0.047619047619047616, <ModelCompound cpd11416_c13 at 0x7efee36fa410>: -0.047619047619047616, <ModelCompound cpd11416_c14 at 0x7efee08daad0>: -0.047619047619047616, <ModelCompound cpd11416_c15 at 0x7efed9507750>: -0.047619047619047616, <ModelCompound cpd11416_c16 at 0x7efed5d88dd0>: -0.047619047619047616, <ModelCompound cpd11416_c17 at 0x7efed62d7350>: -0.047619047619047616, <ModelCompound cpd11416_c18 at 0x7efed1c6e5d0>: -0.047619047619047616, <ModelCompound cpd11416_c19 at 0x7efece97ee50>: -0.047619047619047616, <ModelCompound cpd11416_c20 at 0x7efecd8de2d0>: -0.047619047619047616, <ModelCompound cpd11416_c21 at 0x7efeca4d8790>: -0.047619047619047616}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efea47c9f90>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811839.943701 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe9fe36050>: 1, <ModelCompound cpd11416_c1 at 0x7eff40733310>: -0.047619047619047616, <ModelCompound cpd11416_c2 at 0x7eff3ab0eb50>: -0.047619047619047616, <ModelCompound cpd11416_c3 at 0x7eff39982ad0>: -0.047619047619047616, <ModelCompound cpd11416_c4 at 0x7eff3b063f10>: -0.047619047619047616, <ModelCompound cpd11416_c5 at 0x7eff0df61c50>: -0.047619047619047616, <ModelCompound cpd11416_c6 at 0x7eff08efb890>: -0.047619047619047616, <ModelCompound cpd11416_c7 at 0x7eff03bbc7d0>: -0.047619047619047616, <ModelCompound cpd11416_c8 at 0x7efefc416750>: -0.047619047619047616, <ModelCompound cpd11416_c9 at 0x7efef73db390>: -0.047619047619047616, <ModelCompound cpd11416_c10 at 0x7efeefef8f10>: -0.047619047619047616, <ModelCompound cpd11416_c11 at 0x7efefc643290>: -0.047619047619047616, <ModelCompound cpd11416_c12 at 0x7efee8352510>: -0.047619047619047616, <ModelCompound cpd11416_c13 at 0x7efee31b2610>: -0.047619047619047616, <ModelCompound cpd11416_c14 at 0x7efedbd2bf10>: -0.047619047619047616, <ModelCompound cpd11416_c15 at 0x7efed683bf10>: -0.047619047619047616, <ModelCompound cpd11416_c16 at 0x7efed6d4cf90>: -0.047619047619047616, <ModelCompound cpd11416_c17 at 0x7efed4856250>: -0.047619047619047616, <ModelCompound cpd11416_c18 at 0x7efecf46abd0>: -0.047619047619047616, <ModelCompound cpd11416_c19 at 0x7efecdd2f810>: -0.047619047619047616, <ModelCompound cpd11416_c20 at 0x7efecc835390>: -0.047619047619047616, <ModelCompound cpd11416_c21 at 0x7efeccbffe10>: -0.047619047619047616}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe9fe36050>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811840.0603683 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811840.0724728 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811840.7676792 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811840.8294969 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D1_MG_DASTool_bins_concoct_out.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe9fda4a10>: 1, <ModelCompound cpd11416_c1 at 0x7eff40661f10>: -0.045454545454545456, <ModelCompound cpd11416_c2 at 0x7eff3b255ed0>: -0.045454545454545456, <ModelCompound cpd11416_c3 at 0x7eff3a232550>: -0.045454545454545456, <ModelCompound cpd11416_c4 at 0x7eff394dc890>: -0.045454545454545456, <ModelCompound cpd11416_c5 at 0x7eff1723dc10>: -0.045454545454545456, <ModelCompound cpd11416_c6 at 0x7eff0b778b90>: -0.045454545454545456, <ModelCompound cpd11416_c7 at 0x7eff06774450>: -0.045454545454545456, <ModelCompound cpd11416_c8 at 0x7efefef69290>: -0.045454545454545456, <ModelCompound cpd11416_c9 at 0x7eff01422bd0>: -0.045454545454545456, <ModelCompound cpd11416_c10 at 0x7eff01343cd0>: -0.045454545454545456, <ModelCompound cpd11416_c11 at 0x7efef2021b50>: -0.045454545454545456, <ModelCompound cpd11416_c12 at 0x7efeeac51850>: -0.045454545454545456, <ModelCompound cpd11416_c13 at 0x7efee5a03810>: -0.045454545454545456, <ModelCompound cpd11416_c14 at 0x7efee08a5510>: -0.045454545454545456, <ModelCompound cpd11416_c15 at 0x7efed9116290>: -0.045454545454545456, <ModelCompound cpd11416_c16 at 0x7efed96b9850>: -0.045454545454545456, <ModelCompound cpd11416_c17 at 0x7efed5356690>: -0.045454545454545456, <ModelCompound cpd11416_c18 at 0x7efed1eda450>: -0.045454545454545456, <ModelCompound cpd11416_c19 at 0x7efed5424c50>: -0.045454545454545456, <ModelCompound cpd11416_c20 at 0x7efece16b1d0>: -0.045454545454545456, <ModelCompound cpd11416_c21 at 0x7efecce3d390>: -0.045454545454545456, <ModelCompound cpd11416_c22 at 0x7efec98b0610>: -0.045454545454545456}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe9fda4a10>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811841.4693754 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2A_B_D1_MG_DASTool_bins_concoct_out.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811841.5913801 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.52\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811842.0804815 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.46\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.52\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811842.3524919 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_maxbin.009\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811842.6982915 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.46\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811842.9050152 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_maxbin.009\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811843.119886 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.10\n",
      "1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811843.5061612 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811843.5108495 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811843.7287314 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811843.8928893 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811844.3624353 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811844.4360201 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.52\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811844.5483406 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811844.6475155 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811845.2070267 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.52\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811845.2396083 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811845.3638992 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811845.4059064 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811846.1017635 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811846.1401103 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_maxbin.009\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811846.158812 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811846.1866388 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe9f536590>: 1, <ModelCompound cpd11416_c1 at 0x7eff40664750>: -0.045454545454545456, <ModelCompound cpd11416_c2 at 0x7eff3b2ffdd0>: -0.045454545454545456, <ModelCompound cpd11416_c3 at 0x7eff39e23e90>: -0.045454545454545456, <ModelCompound cpd11416_c4 at 0x7eff17343090>: -0.045454545454545456, <ModelCompound cpd11416_c5 at 0x7eff17b20890>: -0.045454545454545456, <ModelCompound cpd11416_c6 at 0x7eff179bf590>: -0.045454545454545456, <ModelCompound cpd11416_c7 at 0x7eff03a33290>: -0.045454545454545456, <ModelCompound cpd11416_c8 at 0x7efefecad610>: -0.045454545454545456, <ModelCompound cpd11416_c9 at 0x7efef9c4ebd0>: -0.045454545454545456, <ModelCompound cpd11416_c10 at 0x7efef4c9a710>: -0.045454545454545456, <ModelCompound cpd11416_c11 at 0x7efeefc17410>: -0.045454545454545456, <ModelCompound cpd11416_c12 at 0x7efeeacc7dd0>: -0.045454545454545456, <ModelCompound cpd11416_c13 at 0x7efee35d8b50>: -0.045454545454545456, <ModelCompound cpd11416_c14 at 0x7efee5a7d410>: -0.045454545454545456, <ModelCompound cpd11416_c15 at 0x7efed96db3d0>: -0.045454545454545456, <ModelCompound cpd11416_c16 at 0x7efed5e27a10>: -0.045454545454545456, <ModelCompound cpd11416_c17 at 0x7efed64737d0>: -0.045454545454545456, <ModelCompound cpd11416_c18 at 0x7efed1f3ddd0>: -0.045454545454545456, <ModelCompound cpd11416_c19 at 0x7efeceb1a990>: -0.045454545454545456, <ModelCompound cpd11416_c20 at 0x7efecd975590>: -0.045454545454545456, <ModelCompound cpd11416_c21 at 0x7efecda024d0>: -0.045454545454545456, <ModelCompound cpd11416_c22 at 0x7efec9911350>: -0.045454545454545456}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe9f536590>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811846.907621 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_maxbin.009\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811846.9514291 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811846.957873 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811847.0108137 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811847.7623763 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811847.789692 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811847.8041074 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811847.821222 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811848.0733292 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811848.5857916 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811848.6328835 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.59\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811848.6476827 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.46\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811848.673545 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.27\n",
      "10 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811849.3934534 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.59\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811849.4571252 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.46\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811849.5241098 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.50\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811849.6363382 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811850.3788385 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811850.5230443 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811851.3408182 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811851.4215941 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe9578bb90>: 1, <ModelCompound cpd11416_c1 at 0x7eff40664210>: -0.045454545454545456, <ModelCompound cpd11416_c2 at 0x7eff3b2febd0>: -0.045454545454545456, <ModelCompound cpd11416_c3 at 0x7eff39e23b50>: -0.045454545454545456, <ModelCompound cpd11416_c4 at 0x7eff17d6b8d0>: -0.045454545454545456, <ModelCompound cpd11416_c5 at 0x7eff0da2d7d0>: -0.045454545454545456, <ModelCompound cpd11416_c6 at 0x7eff066b2150>: -0.045454545454545456, <ModelCompound cpd11416_c7 at 0x7eff14ad0150>: -0.045454545454545456, <ModelCompound cpd11416_c8 at 0x7efefe9d6ed0>: -0.045454545454545456, <ModelCompound cpd11416_c9 at 0x7efef9a61810>: -0.045454545454545456, <ModelCompound cpd11416_c10 at 0x7efef4a65c50>: -0.045454545454545456, <ModelCompound cpd11416_c11 at 0x7efeefa32f90>: -0.045454545454545456, <ModelCompound cpd11416_c12 at 0x7efeea906f90>: -0.045454545454545456, <ModelCompound cpd11416_c13 at 0x7efee59c17d0>: -0.045454545454545456, <ModelCompound cpd11416_c14 at 0x7efee0871110>: -0.045454545454545456, <ModelCompound cpd11416_c15 at 0x7efed91e6690>: -0.045454545454545456, <ModelCompound cpd11416_c16 at 0x7efed5c023d0>: -0.045454545454545456, <ModelCompound cpd11416_c17 at 0x7efed4ba1190>: -0.045454545454545456, <ModelCompound cpd11416_c18 at 0x7efecf622c90>: -0.045454545454545456, <ModelCompound cpd11416_c19 at 0x7efece2f1790>: -0.045454545454545456, <ModelCompound cpd11416_c20 at 0x7efeccd47b90>: -0.045454545454545456, <ModelCompound cpd11416_c21 at 0x7efec9af9890>: -0.045454545454545456, <ModelCompound cpd11416_c22 at 0x7efec4c0b4d0>: -0.045454545454545456}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe9578bb90>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811851.4369264 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811851.5760179 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811852.1520376 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811852.323806 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811852.4041674 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D1_MG_DASTool_bins_concoct_out.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811852.4192214 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811852.5901148 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811852.9585712 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2A_B_D1_MG_DASTool_bins_concoct_out.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811853.22293 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811853.2324662 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811853.3345752 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_concoct_out.6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811853.4453518 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811853.7651129 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811854.0428927 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_maxbin.009\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811854.1197133 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811854.2573678 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811854.3030167 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811854.5862226 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_maxbin.009\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811854.8648193 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811855.168086 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811855.17972 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811855.3963344 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811855.6888435 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2A_A_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811856.0252664 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811856.1013448 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811856.216234 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811856.27533 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811856.5017867 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811856.867562 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.37\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811857.0021892 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811857.0406382 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_maxbin.022\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811857.1561582 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811857.3118072 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe89148dd0>: 1, <ModelCompound cpd11416_c1 at 0x7eff407130d0>: -0.041666666666666664, <ModelCompound cpd11416_c2 at 0x7eff3b05e290>: -0.041666666666666664, <ModelCompound cpd11416_c3 at 0x7eff39edce90>: -0.041666666666666664, <ModelCompound cpd11416_c4 at 0x7eff17ab8950>: -0.041666666666666664, <ModelCompound cpd11416_c5 at 0x7eff149515d0>: -0.041666666666666664, <ModelCompound cpd11416_c6 at 0x7eff0b10f250>: -0.041666666666666664, <ModelCompound cpd11416_c7 at 0x7eff06124b10>: -0.041666666666666664, <ModelCompound cpd11416_c8 at 0x7eff012e1090>: -0.041666666666666664, <ModelCompound cpd11416_c9 at 0x7efefc171e50>: -0.041666666666666664, <ModelCompound cpd11416_c10 at 0x7efef7145410>: -0.041666666666666664, <ModelCompound cpd11416_c11 at 0x7efef2077e50>: -0.041666666666666664, <ModelCompound cpd11416_c12 at 0x7efef714fa50>: -0.041666666666666664, <ModelCompound cpd11416_c13 at 0x7efee82da6d0>: -0.041666666666666664, <ModelCompound cpd11416_c14 at 0x7efee0ddd850>: -0.041666666666666664, <ModelCompound cpd11416_c15 at 0x7efed9621cd0>: -0.041666666666666664, <ModelCompound cpd11416_c16 at 0x7efed648db10>: -0.041666666666666664, <ModelCompound cpd11416_c17 at 0x7efed555c7d0>: -0.041666666666666664, <ModelCompound cpd11416_c18 at 0x7efed1ee3e50>: -0.041666666666666664, <ModelCompound cpd11416_c19 at 0x7efecea9b1d0>: -0.041666666666666664, <ModelCompound cpd11416_c20 at 0x7efecd935c90>: -0.041666666666666664, <ModelCompound cpd11416_c21 at 0x7efeca6bf090>: -0.041666666666666664, <ModelCompound cpd11416_c22 at 0x7efec72075d0>: -0.041666666666666664, <ModelCompound cpd11416_c23 at 0x7efec41a1750>: -0.041666666666666664, <ModelCompound cpd11416_c24 at 0x7efec2ee6e50>: -0.041666666666666664}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe89148dd0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811857.6034951 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.37\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811857.71959 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_maxbin.022\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811857.8474433 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811858.0494235 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811858.1289263 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811858.565285 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.59\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811858.5733147 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811858.6647317 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811858.9325414 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811859.1569417 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_concoct_out.59\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811859.414944 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.32\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811859.5435 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811859.8257592 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811860.1015513 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811860.1259298 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811860.5205274 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.23\n",
      "14 Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811860.7166739 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45\n",
      "9 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811861.0169594 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811861.5661056 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D1_MG_DASTool_bins_concoct_out.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811861.679981 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811861.9282973 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2A_B_D1_MG_DASTool_bins_concoct_out.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811862.4120402 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811862.5610933 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811862.8336477 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R2A_A_D2_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811862.865348 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811863.2551877 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811863.4380624 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811863.7294943 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811863.8252754 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.37\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.10\n",
      "13 Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811864.6405258 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.37\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811864.79768 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.47\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811865.5573735 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811865.568096 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.47\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811865.7652302 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.54\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe64208290>: 1, <ModelCompound cpd11416_c1 at 0x7eff40716f90>: -0.038461538461538464, <ModelCompound cpd11416_c2 at 0x7eff3afc1a90>: -0.038461538461538464, <ModelCompound cpd11416_c3 at 0x7eff39325c50>: -0.038461538461538464, <ModelCompound cpd11416_c4 at 0x7eff39aab010>: -0.038461538461538464, <ModelCompound cpd11416_c5 at 0x7eff394053d0>: -0.038461538461538464, <ModelCompound cpd11416_c6 at 0x7eff0b0ef9d0>: -0.038461538461538464, <ModelCompound cpd11416_c7 at 0x7eff03f03f50>: -0.038461538461538464, <ModelCompound cpd11416_c8 at 0x7efefefe1d10>: -0.038461538461538464, <ModelCompound cpd11416_c9 at 0x7efef9de9250>: -0.038461538461538464, <ModelCompound cpd11416_c10 at 0x7efef4a05d90>: -0.038461538461538464, <ModelCompound cpd11416_c11 at 0x7efeed371c50>: -0.038461538461538464, <ModelCompound cpd11416_c12 at 0x7efeef912990>: -0.038461538461538464, <ModelCompound cpd11416_c13 at 0x7efee31f4b50>: -0.038461538461538464, <ModelCompound cpd11416_c14 at 0x7efedbe3b3d0>: -0.038461538461538464, <ModelCompound cpd11416_c15 at 0x7efed69f99d0>: -0.038461538461538464, <ModelCompound cpd11416_c16 at 0x7efed5bddfd0>: -0.038461538461538464, <ModelCompound cpd11416_c17 at 0x7efed4b99ed0>: -0.038461538461538464, <ModelCompound cpd11416_c18 at 0x7efed19ad3d0>: -0.038461538461538464, <ModelCompound cpd11416_c19 at 0x7efece59a050>: -0.038461538461538464, <ModelCompound cpd11416_c20 at 0x7efeccdf2210>: -0.038461538461538464, <ModelCompound cpd11416_c21 at 0x7efec9d9bf50>: -0.038461538461538464, <ModelCompound cpd11416_c22 at 0x7efec4e45ad0>: -0.038461538461538464, <ModelCompound cpd11416_c23 at 0x7efec3e74d90>: -0.038461538461538464, <ModelCompound cpd11416_c24 at 0x7efec2e87190>: -0.038461538461538464, <ModelCompound cpd11416_c25 at 0x7efeadcc7bd0>: -0.038461538461538464, <ModelCompound cpd11416_c26 at 0x7efea68ff510>: -0.038461538461538464}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe64208290>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811865.8861578 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.1615\n",
      " "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811866.4559126 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811866.4584153 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.54\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811866.731352 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811866.9404993 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811867.3391576 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811867.748271 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.52\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811868.0039294 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811868.2857769 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811868.577101 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.52\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811868.7257433 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe5f3e5d50>: 1, <ModelCompound cpd11416_c1 at 0x7eff40596410>: -0.037037037037037035, <ModelCompound cpd11416_c2 at 0x7eff3b134d10>: -0.037037037037037035, <ModelCompound cpd11416_c3 at 0x7eff3a357d90>: -0.037037037037037035, <ModelCompound cpd11416_c4 at 0x7eff395fc650>: -0.037037037037037035, <ModelCompound cpd11416_c5 at 0x7eff1746ffd0>: -0.037037037037037035, <ModelCompound cpd11416_c6 at 0x7eff0dcae690>: -0.037037037037037035, <ModelCompound cpd11416_c7 at 0x7eff08ce0f50>: -0.037037037037037035, <ModelCompound cpd11416_c8 at 0x7eff01381790>: -0.037037037037037035, <ModelCompound cpd11416_c9 at 0x7efef9f36850>: -0.037037037037037035, <ModelCompound cpd11416_c10 at 0x7eff01528f50>: -0.037037037037037035, <ModelCompound cpd11416_c11 at 0x7efeefe83c50>: -0.037037037037037035, <ModelCompound cpd11416_c12 at 0x7efee8775f90>: -0.037037037037037035, <ModelCompound cpd11416_c13 at 0x7efeead9bf90>: -0.037037037037037035, <ModelCompound cpd11416_c14 at 0x7efee08cc110>: -0.037037037037037035, <ModelCompound cpd11416_c15 at 0x7efed96fb2d0>: -0.037037037037037035, <ModelCompound cpd11416_c16 at 0x7efed670f9d0>: -0.037037037037037035, <ModelCompound cpd11416_c17 at 0x7efed59be710>: -0.037037037037037035, <ModelCompound cpd11416_c18 at 0x7efed48acb10>: -0.037037037037037035, <ModelCompound cpd11416_c19 at 0x7efecf3e4090>: -0.037037037037037035, <ModelCompound cpd11416_c20 at 0x7efece327710>: -0.037037037037037035, <ModelCompound cpd11416_c21 at 0x7efecd0fd790>: -0.037037037037037035, <ModelCompound cpd11416_c22 at 0x7efec9d58b10>: -0.037037037037037035, <ModelCompound cpd11416_c23 at 0x7efec46182d0>: -0.037037037037037035, <ModelCompound cpd11416_c24 at 0x7efec33b41d0>: -0.037037037037037035, <ModelCompound cpd11416_c25 at 0x7efeba097c10>: -0.037037037037037035, <ModelCompound cpd11416_c26 at 0x7efea6cbf690>: -0.037037037037037035, <ModelCompound cpd11416_c27 at 0x7efe9fb66f10>: -0.037037037037037035}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe5f3e5d50>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811868.8928106 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811869.0619714 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.8\n",
      "17 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811869.4817345 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811869.6852832 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811869.9884496 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811870.107855 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811870.3882382 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe5f17b2d0>: 1, <ModelCompound cpd11416_c1 at 0x7eff405967d0>: -0.037037037037037035, <ModelCompound cpd11416_c2 at 0x7eff3b1a2f10>: -0.037037037037037035, <ModelCompound cpd11416_c3 at 0x7eff3a25e750>: -0.037037037037037035, <ModelCompound cpd11416_c4 at 0x7eff393df9d0>: -0.037037037037037035, <ModelCompound cpd11416_c5 at 0x7eff1729fe10>: -0.037037037037037035, <ModelCompound cpd11416_c6 at 0x7eff0dafc250>: -0.037037037037037035, <ModelCompound cpd11416_c7 at 0x7eff08936a90>: -0.037037037037037035, <ModelCompound cpd11416_c8 at 0x7eff0172b710>: -0.037037037037037035, <ModelCompound cpd11416_c9 at 0x7efefc54ce90>: -0.037037037037037035, <ModelCompound cpd11416_c10 at 0x7efef75dcdd0>: -0.037037037037037035, <ModelCompound cpd11416_c11 at 0x7efeefe42d90>: -0.037037037037037035, <ModelCompound cpd11416_c12 at 0x7efef2452e10>: -0.037037037037037035, <ModelCompound cpd11416_c13 at 0x7efee5fe8e90>: -0.037037037037037035, <ModelCompound cpd11416_c14 at 0x7efee0e06410>: -0.037037037037037035, <ModelCompound cpd11416_c15 at 0x7efedbe1a910>: -0.037037037037037035, <ModelCompound cpd11416_c16 at 0x7efed6ba24d0>: -0.037037037037037035, <ModelCompound cpd11416_c17 at 0x7efed909e410>: -0.037037037037037035, <ModelCompound cpd11416_c18 at 0x7efed4a9ef90>: -0.037037037037037035, <ModelCompound cpd11416_c19 at 0x7efecf7a41d0>: -0.037037037037037035, <ModelCompound cpd11416_c20 at 0x7efece6df810>: -0.037037037037037035, <ModelCompound cpd11416_c21 at 0x7efecd4bd910>: -0.037037037037037035, <ModelCompound cpd11416_c22 at 0x7efec9f28c90>: -0.037037037037037035, <ModelCompound cpd11416_c23 at 0x7efec49c4c10>: -0.037037037037037035, <ModelCompound cpd11416_c24 at 0x7efec35742d0>: -0.037037037037037035, <ModelCompound cpd11416_c25 at 0x7efeba257d50>: -0.037037037037037035, <ModelCompound cpd11416_c26 at 0x7efea6e6f7d0>: -0.037037037037037035, <ModelCompound cpd11416_c27 at 0x7efe9fbc2950>: -0.037037037037037035}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe5f17b2d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811870.6150732 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811871.1048346 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811871.1738214 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811871.3059912 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811871.7271864 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811872.0424418 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811872.201584 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe5c548fd0>: 1, <ModelCompound cpd11416_c1 at 0x7eff405c4b10>: -0.037037037037037035, <ModelCompound cpd11416_c2 at 0x7eff3ab6efd0>: -0.037037037037037035, <ModelCompound cpd11416_c3 at 0x7eff3992d610>: -0.037037037037037035, <ModelCompound cpd11416_c4 at 0x7eff178b0b50>: -0.037037037037037035, <ModelCompound cpd11416_c5 at 0x7eff0dfce590>: -0.037037037037037035, <ModelCompound cpd11416_c6 at 0x7eff0b070990>: -0.037037037037037035, <ModelCompound cpd11416_c7 at 0x7eff03f83550>: -0.037037037037037035, <ModelCompound cpd11416_c8 at 0x7efefea96710>: -0.037037037037037035, <ModelCompound cpd11416_c9 at 0x7efef732b3d0>: -0.037037037037037035, <ModelCompound cpd11416_c10 at 0x7efef246cad0>: -0.037037037037037035, <ModelCompound cpd11416_c11 at 0x7efeed377f50>: -0.037037037037037035, <ModelCompound cpd11416_c12 at 0x7efee5f70110>: -0.037037037037037035, <ModelCompound cpd11416_c13 at 0x7efee30e3210>: -0.037037037037037035, <ModelCompound cpd11416_c14 at 0x7efedbf533d0>: -0.037037037037037035, <ModelCompound cpd11416_c15 at 0x7efed6e13810>: -0.037037037037037035, <ModelCompound cpd11416_c16 at 0x7efed55edd50>: -0.037037037037037035, <ModelCompound cpd11416_c17 at 0x7efed1e3f190>: -0.037037037037037035, <ModelCompound cpd11416_c18 at 0x7efed430fc10>: -0.037037037037037035, <ModelCompound cpd11416_c19 at 0x7efecde7b610>: -0.037037037037037035, <ModelCompound cpd11416_c20 at 0x7efeccd9f110>: -0.037037037037037035, <ModelCompound cpd11416_c21 at 0x7efec9db0950>: -0.037037037037037035, <ModelCompound cpd11416_c22 at 0x7efec4927c10>: -0.037037037037037035, <ModelCompound cpd11416_c23 at 0x7efec3676150>: -0.037037037037037035, <ModelCompound cpd11416_c24 at 0x7efeba463550>: -0.037037037037037035, <ModelCompound cpd11416_c25 at 0x7efea91043d0>: -0.037037037037037035, <ModelCompound cpd11416_c26 at 0x7efe9fe44190>: -0.037037037037037035, <ModelCompound cpd11416_c27 at 0x7efe9af05950>: -0.037037037037037035}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe5c548fd0>: -1}\n",
      "6"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811872.2194486 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811872.2237542 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811872.2331283 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.36\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.51\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811872.8300219 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811873.0049412 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811873.1139395 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811873.3645334 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811873.940862 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811873.990942 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n",
      "2 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811874.5053606 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811874.7937524 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.36\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811874.9422507 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811875.0163143 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_concoct_out.11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811875.599644 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811875.8887095 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811875.995239 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811876.0435119 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811876.450255 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811876.9445715 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811876.9676998 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811877.1480966 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811877.551961 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.41\n",
      "10"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811878.0034227 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811878.007381 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe54f2b4d0>: 1, <ModelCompound cpd11416_c1 at 0x7eff40618c90>: -0.037037037037037035, <ModelCompound cpd11416_c2 at 0x7eff3aa06690>: -0.037037037037037035, <ModelCompound cpd11416_c3 at 0x7eff398dcb90>: -0.037037037037037035, <ModelCompound cpd11416_c4 at 0x7eff179089d0>: -0.037037037037037035, <ModelCompound cpd11416_c5 at 0x7eff0dfba550>: -0.037037037037037035, <ModelCompound cpd11416_c6 at 0x7eff089f1610>: -0.037037037037037035, <ModelCompound cpd11416_c7 at 0x7eff0144c710>: -0.037037037037037035, <ModelCompound cpd11416_c8 at 0x7eff01574250>: -0.037037037037037035, <ModelCompound cpd11416_c9 at 0x7efef764eb90>: -0.037037037037037035, <ModelCompound cpd11416_c10 at 0x7efef24bc0d0>: -0.037037037037037035, <ModelCompound cpd11416_c11 at 0x7efeed1c5b10>: -0.037037037037037035, <ModelCompound cpd11416_c12 at 0x7efee5f346d0>: -0.037037037037037035, <ModelCompound cpd11416_c13 at 0x7efee0ede6d0>: -0.037037037037037035, <ModelCompound cpd11416_c14 at 0x7efedbce6750>: -0.037037037037037035, <ModelCompound cpd11416_c15 at 0x7efed694bf10>: -0.037037037037037035, <ModelCompound cpd11416_c16 at 0x7efed5418e90>: -0.037037037037037035, <ModelCompound cpd11416_c17 at 0x7efed1d0a4d0>: -0.037037037037037035, <ModelCompound cpd11416_c18 at 0x7efece913fd0>: -0.037037037037037035, <ModelCompound cpd11416_c19 at 0x7efecf0ba390>: -0.037037037037037035, <ModelCompound cpd11416_c20 at 0x7efeca6c9a90>: -0.037037037037037035, <ModelCompound cpd11416_c21 at 0x7efec71dca10>: -0.037037037037037035, <ModelCompound cpd11416_c22 at 0x7efec3dbb850>: -0.037037037037037035, <ModelCompound cpd11416_c23 at 0x7efec28fd110>: -0.037037037037037035, <ModelCompound cpd11416_c24 at 0x7efea94f7090>: -0.037037037037037035, <ModelCompound cpd11416_c25 at 0x7efea4023d50>: -0.037037037037037035, <ModelCompound cpd11416_c26 at 0x7efe9ac636d0>: -0.037037037037037035, <ModelCompound cpd11416_c27 at 0x7efe9ad3e9d0>: -0.037037037037037035}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe54f2b4d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811878.2406816 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811878.3006835 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811878.3107562 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811878.655684 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811879.0104785 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811879.0460126 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811879.3589325 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811879.401457 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811879.4320803 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811879.7637432 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811879.9678442 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811880.0810533 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811880.5043979 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811880.5286744 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811880.8724568 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.87\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811881.141137 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811881.6163843 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811881.6286967 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe463fe010>: 1, <ModelCompound cpd11416_c1 at 0x7eff40596f10>: -0.03571428571428571, <ModelCompound cpd11416_c2 at 0x7eff3af22210>: -0.03571428571428571, <ModelCompound cpd11416_c3 at 0x7eff39cfd510>: -0.03571428571428571, <ModelCompound cpd11416_c4 at 0x7eff39d8e890>: -0.03571428571428571, <ModelCompound cpd11416_c5 at 0x7eff14bda650>: -0.03571428571428571, <ModelCompound cpd11416_c6 at 0x7eff0b00a990>: -0.03571428571428571, <ModelCompound cpd11416_c7 at 0x7eff03f60850>: -0.03571428571428571, <ModelCompound cpd11416_c8 at 0x7efefec96010>: -0.03571428571428571, <ModelCompound cpd11416_c9 at 0x7efef7595cd0>: -0.03571428571428571, <ModelCompound cpd11416_c10 at 0x7efef9a5bc50>: -0.03571428571428571, <ModelCompound cpd11416_c11 at 0x7efeed6ead90>: -0.03571428571428571, <ModelCompound cpd11416_c12 at 0x7efee8697650>: -0.03571428571428571, <ModelCompound cpd11416_c13 at 0x7efee32172d0>: -0.03571428571428571, <ModelCompound cpd11416_c14 at 0x7efedbe7d990>: -0.03571428571428571, <ModelCompound cpd11416_c15 at 0x7efed6bee3d0>: -0.03571428571428571, <ModelCompound cpd11416_c16 at 0x7efed5b18710>: -0.03571428571428571, <ModelCompound cpd11416_c17 at 0x7efed4ae3010>: -0.03571428571428571, <ModelCompound cpd11416_c18 at 0x7efecf51ee50>: -0.03571428571428571, <ModelCompound cpd11416_c19 at 0x7efecdfd2310>: -0.03571428571428571, <ModelCompound cpd11416_c20 at 0x7efecca1ea90>: -0.03571428571428571, <ModelCompound cpd11416_c21 at 0x7efecca54990>: -0.03571428571428571, <ModelCompound cpd11416_c22 at 0x7efec44703d0>: -0.03571428571428571, <ModelCompound cpd11416_c23 at 0x7efec2e09190>: -0.03571428571428571, <ModelCompound cpd11416_c24 at 0x7efeada44cd0>: -0.03571428571428571, <ModelCompound cpd11416_c25 at 0x7efea479ed90>: -0.03571428571428571, <ModelCompound cpd11416_c26 at 0x7efe9aef8110>: -0.03571428571428571, <ModelCompound cpd11416_c27 at 0x7efe97fc3310>: -0.03571428571428571, <ModelCompound cpd11416_c28 at 0x7efe8e8d0d50>: -0.03571428571428571}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe463fe010>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811881.6627016 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_concoct_out.87\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811881.9823265 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811882.1520822 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811882.1820626 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811882.3329968 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_maxbin.022\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R2A_C_D1_MG_DASTool_bins_metabat.5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811882.7247088 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811882.7650356 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811882.8351219 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811883.0899785 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.32\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811883.2232156 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.39\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811883.3253372 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_maxbin.022\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811883.3577394 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811883.816721 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811883.9031227 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811883.9946072 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811884.1920695 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811884.3249128 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811884.4498982 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811884.9169004 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811885.1611145 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.27\n",
      "5 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811885.592252 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "16 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.39\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811885.7740448 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811886.0631068 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811886.3161452 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811886.5355418 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811886.7052443 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811886.7127526 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811886.8259563 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811887.1747582 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.34\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811887.4681547 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811887.6260676 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n",
      "7"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811887.8170416 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811887.821414 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811887.8690457 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe41129590>: 1, <ModelCompound cpd11416_c1 at 0x7eff4079b290>: -0.03225806451612903, <ModelCompound cpd11416_c2 at 0x7eff3b3a5190>: -0.03225806451612903, <ModelCompound cpd11416_c3 at 0x7eff3a398650>: -0.03225806451612903, <ModelCompound cpd11416_c4 at 0x7eff39365310>: -0.03225806451612903, <ModelCompound cpd11416_c5 at 0x7eff1713b450>: -0.03225806451612903, <ModelCompound cpd11416_c6 at 0x7eff0b78fd90>: -0.03225806451612903, <ModelCompound cpd11416_c7 at 0x7eff066a9ad0>: -0.03225806451612903, <ModelCompound cpd11416_c8 at 0x7eff016ef650>: -0.03225806451612903, <ModelCompound cpd11416_c9 at 0x7efefc58fa10>: -0.03225806451612903, <ModelCompound cpd11416_c10 at 0x7efef7567dd0>: -0.03225806451612903, <ModelCompound cpd11416_c11 at 0x7efef276e010>: -0.03225806451612903, <ModelCompound cpd11416_c12 at 0x7efeef8613d0>: -0.03225806451612903, <ModelCompound cpd11416_c13 at 0x7efee84e9450>: -0.03225806451612903, <ModelCompound cpd11416_c14 at 0x7efee0d0c650>: -0.03225806451612903, <ModelCompound cpd11416_c15 at 0x7efede416ed0>: -0.03225806451612903, <ModelCompound cpd11416_c16 at 0x7efed6b6af90>: -0.03225806451612903, <ModelCompound cpd11416_c17 at 0x7efed587e550>: -0.03225806451612903, <ModelCompound cpd11416_c18 at 0x7efed47b7990>: -0.03225806451612903, <ModelCompound cpd11416_c19 at 0x7efecf6fde50>: -0.03225806451612903, <ModelCompound cpd11416_c20 at 0x7efece553c90>: -0.03225806451612903, <ModelCompound cpd11416_c21 at 0x7efecd031510>: -0.03225806451612903, <ModelCompound cpd11416_c22 at 0x7efec9d95410>: -0.03225806451612903, <ModelCompound cpd11416_c23 at 0x7efec46f6050>: -0.03225806451612903, <ModelCompound cpd11416_c24 at 0x7efec317a750>: -0.03225806451612903, <ModelCompound cpd11416_c25 at 0x7efeade5dbd0>: -0.03225806451612903, <ModelCompound cpd11416_c26 at 0x7efea6bec190>: -0.03225806451612903, <ModelCompound cpd11416_c27 at 0x7efe9fdcc0d0>: -0.03225806451612903, <ModelCompound cpd11416_c28 at 0x7efe9a8de090>: -0.03225806451612903, <ModelCompound cpd11416_c29 at 0x7efe955450d0>: -0.03225806451612903, <ModelCompound cpd11416_c30 at 0x7efe8e15c2d0>: -0.03225806451612903, <ModelCompound cpd11416_c31 at 0x7efe84daac50>: -0.03225806451612903}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe41129590>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811888.557536 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811888.624632 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811888.7627287 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811888.9167995 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811888.9336376 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811888.9556296 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.84\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe410d6450>: 1, <ModelCompound cpd11416_c1 at 0x7eff40799890>: -0.03333333333333333, <ModelCompound cpd11416_c2 at 0x7eff3b46b050>: -0.03333333333333333, <ModelCompound cpd11416_c3 at 0x7eff3a538e10>: -0.03333333333333333, <ModelCompound cpd11416_c4 at 0x7eff394c0250>: -0.03333333333333333, <ModelCompound cpd11416_c5 at 0x7eff17187610>: -0.03333333333333333, <ModelCompound cpd11416_c6 at 0x7eff0b3ea0d0>: -0.03333333333333333, <ModelCompound cpd11416_c7 at 0x7eff17292790>: -0.03333333333333333, <ModelCompound cpd11416_c8 at 0x7eff01126550>: -0.03333333333333333, <ModelCompound cpd11416_c9 at 0x7efef99da3d0>: -0.03333333333333333, <ModelCompound cpd11416_c10 at 0x7efef2155690>: -0.03333333333333333, <ModelCompound cpd11416_c11 at 0x7efef9e20bd0>: -0.03333333333333333, <ModelCompound cpd11416_c12 at 0x7efee7ff7b50>: -0.03333333333333333, <ModelCompound cpd11416_c13 at 0x7efee0a09e10>: -0.03333333333333333, <ModelCompound cpd11416_c14 at 0x7efee30c4810>: -0.03333333333333333, <ModelCompound cpd11416_c15 at 0x7efee0cbab10>: -0.03333333333333333, <ModelCompound cpd11416_c16 at 0x7efed5e54050>: -0.03333333333333333, <ModelCompound cpd11416_c17 at 0x7efed4cfaa90>: -0.03333333333333333, <ModelCompound cpd11416_c18 at 0x7efed1aa4a50>: -0.03333333333333333, <ModelCompound cpd11416_c19 at 0x7efece4ea8d0>: -0.03333333333333333, <ModelCompound cpd11416_c20 at 0x7efecd489410>: -0.03333333333333333, <ModelCompound cpd11416_c21 at 0x7efeca3caad0>: -0.03333333333333333, <ModelCompound cpd11416_c22 at 0x7efec7171b50>: -0.03333333333333333, <ModelCompound cpd11416_c23 at 0x7efec3beded0>: -0.03333333333333333, <ModelCompound cpd11416_c24 at 0x7efeba6bc090>: -0.03333333333333333, <ModelCompound cpd11416_c25 at 0x7efea92b9bd0>: -0.03333333333333333, <ModelCompound cpd11416_c26 at 0x7efea90e20d0>: -0.03333333333333333, <ModelCompound cpd11416_c27 at 0x7efe9aaf27d0>: -0.03333333333333333, <ModelCompound cpd11416_c28 at 0x7efe9564c350>: -0.03333333333333333, <ModelCompound cpd11416_c29 at 0x7efe8e2afdd0>: -0.03333333333333333, <ModelCompound cpd11416_c30 at 0x7efe84df6f10>: -0.03333333333333333}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe410d6450>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811889.2615392 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811889.7799585 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.34\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811889.798816 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n",
      "1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811889.862407 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.98\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811889.8668199 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811889.977981 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811890.0195615 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.84\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811890.0737116 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811890.4811885 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811890.887838 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.4\n",
      "1724811890.920197 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811890.935741 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.98\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811890.975029 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811891.020643 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811891.1176174 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.98\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811891.1227121 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811891.1864033 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811891.6906843 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.89\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe3c2431d0>: 1, <ModelCompound cpd11416_c1 at 0x7eff407a4090>: -0.03225806451612903, <ModelCompound cpd11416_c2 at 0x7eff3b347110>: -0.03225806451612903, <ModelCompound cpd11416_c3 at 0x7eff3a4b4650>: -0.03225806451612903, <ModelCompound cpd11416_c4 at 0x7eff3931e410>: -0.03225806451612903, <ModelCompound cpd11416_c5 at 0x7eff172ff190>: -0.03225806451612903, <ModelCompound cpd11416_c6 at 0x7eff0b764410>: -0.03225806451612903, <ModelCompound cpd11416_c7 at 0x7eff065f1a50>: -0.03225806451612903, <ModelCompound cpd11416_c8 at 0x7eff01497750>: -0.03225806451612903, <ModelCompound cpd11416_c9 at 0x7efefc464c10>: -0.03225806451612903, <ModelCompound cpd11416_c10 at 0x7efef7449d90>: -0.03225806451612903, <ModelCompound cpd11416_c11 at 0x7efef2221f10>: -0.03225806451612903, <ModelCompound cpd11416_c12 at 0x7efeed14b450>: -0.03225806451612903, <ModelCompound cpd11416_c13 at 0x7efee8055d10>: -0.03225806451612903, <ModelCompound cpd11416_c14 at 0x7efee0cf5e90>: -0.03225806451612903, <ModelCompound cpd11416_c15 at 0x7efedb9c6f10>: -0.03225806451612903, <ModelCompound cpd11416_c16 at 0x7efed658ee90>: -0.03225806451612903, <ModelCompound cpd11416_c17 at 0x7efed52bf290>: -0.03225806451612903, <ModelCompound cpd11416_c18 at 0x7efed1cbf1d0>: -0.03225806451612903, <ModelCompound cpd11416_c19 at 0x7efece3a7050>: -0.03225806451612903, <ModelCompound cpd11416_c20 at 0x7efeccd40990>: -0.03225806451612903, <ModelCompound cpd11416_c21 at 0x7efece3f1850>: -0.03225806451612903, <ModelCompound cpd11416_c22 at 0x7efec4c93750>: -0.03225806451612903, <ModelCompound cpd11416_c23 at 0x7efec3a7dcd0>: -0.03225806451612903, <ModelCompound cpd11416_c24 at 0x7efec28d6f90>: -0.03225806451612903, <ModelCompound cpd11416_c25 at 0x7efea934a110>: -0.03225806451612903, <ModelCompound cpd11416_c26 at 0x7efea404cad0>: -0.03225806451612903, <ModelCompound cpd11416_c27 at 0x7efe9f07abd0>: -0.03225806451612903, <ModelCompound cpd11416_c28 at 0x7efe97bd6d90>: -0.03225806451612903, <ModelCompound cpd11416_c29 at 0x7efe8eaac810>: -0.03225806451612903, <ModelCompound cpd11416_c30 at 0x7efe8d8d9690>: -0.03225806451612903, <ModelCompound cpd11416_c31 at 0x7efe82430fd0>: -0.03225806451612903}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe3c2431d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811891.769138 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811891.9737911 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2A_B_D2_MG_DASTool_bins_metabat.6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811892.058898 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811892.08218 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n",
      "1724811892.168224 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.98\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811892.216756 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811892.382248 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe3c22b3d0>: 1, <ModelCompound cpd11416_c1 at 0x7eff40596dd0>: -0.03225806451612903, <ModelCompound cpd11416_c2 at 0x7eff3b2e5a90>: -0.03225806451612903, <ModelCompound cpd11416_c3 at 0x7eff3a306650>: -0.03225806451612903, <ModelCompound cpd11416_c4 at 0x7eff39215390>: -0.03225806451612903, <ModelCompound cpd11416_c5 at 0x7eff14ed3410>: -0.03225806451612903, <ModelCompound cpd11416_c6 at 0x7eff14f7a3d0>: -0.03225806451612903, <ModelCompound cpd11416_c7 at 0x7eff06431790>: -0.03225806451612903, <ModelCompound cpd11416_c8 at 0x7eff012297d0>: -0.03225806451612903, <ModelCompound cpd11416_c9 at 0x7efefc1a44d0>: -0.03225806451612903, <ModelCompound cpd11416_c10 at 0x7efef4ed4d90>: -0.03225806451612903, <ModelCompound cpd11416_c11 at 0x7efeed6a9590>: -0.03225806451612903, <ModelCompound cpd11416_c12 at 0x7efeefccd310>: -0.03225806451612903, <ModelCompound cpd11416_c13 at 0x7efee366f450>: -0.03225806451612903, <ModelCompound cpd11416_c14 at 0x7efede6b46d0>: -0.03225806451612903, <ModelCompound cpd11416_c15 at 0x7efed9688e90>: -0.03225806451612903, <ModelCompound cpd11416_c16 at 0x7efed6236bd0>: -0.03225806451612903, <ModelCompound cpd11416_c17 at 0x7efed66f08d0>: -0.03225806451612903, <ModelCompound cpd11416_c18 at 0x7efed41414d0>: -0.03225806451612903, <ModelCompound cpd11416_c19 at 0x7efecee23210>: -0.03225806451612903, <ModelCompound cpd11416_c20 at 0x7efecdd89d90>: -0.03225806451612903, <ModelCompound cpd11416_c21 at 0x7efeccb53dd0>: -0.03225806451612903, <ModelCompound cpd11416_c22 at 0x7efec77ab150>: -0.03225806451612903, <ModelCompound cpd11416_c23 at 0x7efec4081290>: -0.03225806451612903, <ModelCompound cpd11416_c24 at 0x7efec2f75610>: -0.03225806451612903, <ModelCompound cpd11416_c25 at 0x7efeadb51050>: -0.03225806451612903, <ModelCompound cpd11416_c26 at 0x7efea4721250>: -0.03225806451612903, <ModelCompound cpd11416_c27 at 0x7efe9f4f9050>: -0.03225806451612903, <ModelCompound cpd11416_c28 at 0x7efe97f71650>: -0.03225806451612903, <ModelCompound cpd11416_c29 at 0x7efe8eb4da50>: -0.03225806451612903, <ModelCompound cpd11416_c30 at 0x7efe8974f650>: -0.03225806451612903, <ModelCompound cpd11416_c31 at 0x7efe822f8590>: -0.03225806451612903}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe3c22b3d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811892.4670093 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.89\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811892.9490592 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.47\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811893.0511281 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811893.0803852 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811893.1168983 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.46\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811893.189134 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811893.3151982 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811893.632718 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811893.7574372 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811893.8645718 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_concoct_out.46\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811894.1664512 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811894.1763556 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811894.2913184 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811894.321268 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811894.4189365 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811894.7931535 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811894.995657 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.110\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811895.0451198 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811895.2805007 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811895.366574 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811895.5960038 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811895.7132022 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811895.903645 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811895.9586985 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.110\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811896.135967 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811896.3878212 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811896.647338 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811896.73173 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811896.8965633 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811897.113703 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811897.1343217 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811897.2643166 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811897.5165372 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.69\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_restored_DShore_MG_DASTool_bins_metabat.8\n",
      "6 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811897.9218252 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811897.9881864 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R1_A_D2_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811898.119121 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811898.2752666 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811898.2919717 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811898.3496912 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811898.439711 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe2f63e010>: 1, <ModelCompound cpd11416_c1 at 0x7eff40597190>: -0.03125, <ModelCompound cpd11416_c2 at 0x7eff3af22550>: -0.03125, <ModelCompound cpd11416_c3 at 0x7eff39cfd550>: -0.03125, <ModelCompound cpd11416_c4 at 0x7eff39d7e890>: -0.03125, <ModelCompound cpd11416_c5 at 0x7eff14bda690>: -0.03125, <ModelCompound cpd11416_c6 at 0x7eff08a87150>: -0.03125, <ModelCompound cpd11416_c7 at 0x7eff016101d0>: -0.03125, <ModelCompound cpd11416_c8 at 0x7eff08c262d0>: -0.03125, <ModelCompound cpd11416_c9 at 0x7efef755b390>: -0.03125, <ModelCompound cpd11416_c10 at 0x7efeefe4b4d0>: -0.03125, <ModelCompound cpd11416_c11 at 0x7efee855ee90>: -0.03125, <ModelCompound cpd11416_c12 at 0x7efee86920d0>: -0.03125, <ModelCompound cpd11416_c13 at 0x7efeea9ea1d0>: -0.03125, <ModelCompound cpd11416_c14 at 0x7efedba5ec10>: -0.03125, <ModelCompound cpd11416_c15 at 0x7efed69a2a90>: -0.03125, <ModelCompound cpd11416_c16 at 0x7efed5ac4c90>: -0.03125, <ModelCompound cpd11416_c17 at 0x7efed46788d0>: -0.03125, <ModelCompound cpd11416_c18 at 0x7efecf2bc8d0>: -0.03125, <ModelCompound cpd11416_c19 at 0x7efece01f0d0>: -0.03125, <ModelCompound cpd11416_c20 at 0x7efeccf789d0>: -0.03125, <ModelCompound cpd11416_c21 at 0x7efec9d429d0>: -0.03125, <ModelCompound cpd11416_c22 at 0x7efec49a5d50>: -0.03125, <ModelCompound cpd11416_c23 at 0x7efec3273e90>: -0.03125, <ModelCompound cpd11416_c24 at 0x7efeadde6050>: -0.03125, <ModelCompound cpd11416_c25 at 0x7efea6828190>: -0.03125, <ModelCompound cpd11416_c26 at 0x7efe9f3fd590>: -0.03125, <ModelCompound cpd11416_c27 at 0x7efe9834cfd0>: -0.03125, <ModelCompound cpd11416_c28 at 0x7efe95052150>: -0.03125, <ModelCompound cpd11416_c29 at 0x7efe8d8dc2d0>: -0.03125, <ModelCompound cpd11416_c30 at 0x7efe826825d0>: -0.03125, <ModelCompound cpd11416_c31 at 0x7efe66770dd0>: -0.03125, <ModelCompound cpd11416_c32 at 0x7efe5f30ee50>: -0.03125}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe2f63e010>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811898.5333037 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.69\n",
      "4"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811898.644482 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811898.6490524 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe39a59590>: 1, <ModelCompound cpd11416_c1 at 0x7eff4079e3d0>: -0.03225806451612903, <ModelCompound cpd11416_c2 at 0x7eff3af50f10>: -0.03225806451612903, <ModelCompound cpd11416_c3 at 0x7eff3b56d110>: -0.03225806451612903, <ModelCompound cpd11416_c4 at 0x7eff17e9ed10>: -0.03225806451612903, <ModelCompound cpd11416_c5 at 0x7eff14c3edd0>: -0.03225806451612903, <ModelCompound cpd11416_c6 at 0x7eff08f6d290>: -0.03225806451612903, <ModelCompound cpd11416_c7 at 0x7eff03ba1fd0>: -0.03225806451612903, <ModelCompound cpd11416_c8 at 0x7efefc61ffd0>: -0.03225806451612903, <ModelCompound cpd11416_c9 at 0x7eff03c61050>: -0.03225806451612903, <ModelCompound cpd11416_c10 at 0x7efef25d3010>: -0.03225806451612903, <ModelCompound cpd11416_c11 at 0x7efeed2e8dd0>: -0.03225806451612903, <ModelCompound cpd11416_c12 at 0x7efee8038490>: -0.03225806451612903, <ModelCompound cpd11416_c13 at 0x7efee0fd6490>: -0.03225806451612903, <ModelCompound cpd11416_c14 at 0x7efedbc12590>: -0.03225806451612903, <ModelCompound cpd11416_c15 at 0x7efed6a43d10>: -0.03225806451612903, <ModelCompound cpd11416_c16 at 0x7efed550cd10>: -0.03225806451612903, <ModelCompound cpd11416_c17 at 0x7efed1dee350>: -0.03225806451612903, <ModelCompound cpd11416_c18 at 0x7efecea07d10>: -0.03225806451612903, <ModelCompound cpd11416_c19 at 0x7efeceb24110>: -0.03225806451612903, <ModelCompound cpd11416_c20 at 0x7efecca5a990>: -0.03225806451612903, <ModelCompound cpd11416_c21 at 0x7efec74c2a10>: -0.03225806451612903, <ModelCompound cpd11416_c22 at 0x7efec40c72d0>: -0.03225806451612903, <ModelCompound cpd11416_c23 at 0x7efec2c13550>: -0.03225806451612903, <ModelCompound cpd11416_c24 at 0x7efead9ff490>: -0.03225806451612903, <ModelCompound cpd11416_c25 at 0x7efea6a90710>: -0.03225806451612903, <ModelCompound cpd11416_c26 at 0x7efe9f7543d0>: -0.03225806451612903, <ModelCompound cpd11416_c27 at 0x7efe9825b2d0>: -0.03225806451612903, <ModelCompound cpd11416_c28 at 0x7efe8ec33fd0>: -0.03225806451612903, <ModelCompound cpd11416_c29 at 0x7efe896ded50>: -0.03225806451612903, <ModelCompound cpd11416_c30 at 0x7efe8240a850>: -0.03225806451612903, <ModelCompound cpd11416_c31 at 0x7efe6425fb10>: -0.03225806451612903}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe39a59590>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811899.0458188 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.89\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811899.0715256 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n",
      "1724811899.1284134 WARNING: Ignoring reaction 'rxn00062_c7' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811899.1845095 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811899.385732 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811899.3953118 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811899.4392762 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811899.552453 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811899.5611105 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811899.9073002 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811900.1621757 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_concoct_out.89\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811900.3407705 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811900.4532864 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811900.5028188 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811900.6078932 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811900.6681392 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811900.6742802 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811900.7788727 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 22Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14 \n",
      "Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_concoct_out.9"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811901.1786265 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811901.1805403 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_maxbin.022\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7efe2ce6e9d0>: 1, <ModelCompound cpd11416_c1 at 0x7eff40618cd0>: -0.03125, <ModelCompound cpd11416_c2 at 0x7eff3af42f10>: -0.03125, <ModelCompound cpd11416_c3 at 0x7eff39f67e50>: -0.03125, <ModelCompound cpd11416_c4 at 0x7eff177e3c10>: -0.03125, <ModelCompound cpd11416_c5 at 0x7eff0db17610>: -0.03125, <ModelCompound cpd11416_c6 at 0x7eff08b05410>: -0.03125, <ModelCompound cpd11416_c7 at 0x7eff01541450>: -0.03125, <ModelCompound cpd11416_c8 at 0x7efefc199190>: -0.03125, <ModelCompound cpd11416_c9 at 0x7efef7245a50>: -0.03125, <ModelCompound cpd11416_c10 at 0x7efef23d53d0>: -0.03125, <ModelCompound cpd11416_c11 at 0x7efeeaf23310>: -0.03125, <ModelCompound cpd11416_c12 at 0x7efeed527550>: -0.03125, <ModelCompound cpd11416_c13 at 0x7efee0e38250>: -0.03125, <ModelCompound cpd11416_c14 at 0x7efedbd72750>: -0.03125, <ModelCompound cpd11416_c15 at 0x7efed6b978d0>: -0.03125, <ModelCompound cpd11416_c16 at 0x7efed562f990>: -0.03125, <ModelCompound cpd11416_c17 at 0x7efed1f1a2d0>: -0.03125, <ModelCompound cpd11416_c18 at 0x7efecee766d0>: -0.03125, <ModelCompound cpd11416_c19 at 0x7efece33c450>: -0.03125, <ModelCompound cpd11416_c20 at 0x7efecc8ee410>: -0.03125, <ModelCompound cpd11416_c21 at 0x7efec74151d0>: -0.03125, <ModelCompound cpd11416_c22 at 0x7efec3b5ded0>: -0.03125, <ModelCompound cpd11416_c23 at 0x7efeba4f3810>: -0.03125, <ModelCompound cpd11416_c24 at 0x7efea93bac50>: -0.03125, <ModelCompound cpd11416_c25 at 0x7efea419b1d0>: -0.03125, <ModelCompound cpd11416_c26 at 0x7efe9f111150>: -0.03125, <ModelCompound cpd11416_c27 at 0x7efe97dbfd90>: -0.03125, <ModelCompound cpd11416_c28 at 0x7efe8e826790>: -0.03125, <ModelCompound cpd11416_c29 at 0x7efe89590c90>: -0.03125, <ModelCompound cpd11416_c30 at 0x7efe8224f610>: -0.03125, <ModelCompound cpd11416_c31 at 0x7efe642dbb10>: -0.03125, <ModelCompound cpd11416_c32 at 0x7efe5eda1a50>: -0.03125}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7efe2ce6e9d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811901.2043297 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811901.250586 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "124  Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65Salt_Pond_MetaG_R1_A_D1_MG_DASTool_bins.metabat.18\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811901.6088948 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n",
      "1724811901.6097767 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811901.6231232 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811901.7096198 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811901.7634335 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811901.7854178 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811901.936799 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811901.9904668 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_maxbin.022\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811902.2708027 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811902.341843 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811902.54306 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811902.721728 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811902.912123 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.52\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811902.9258223 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811902.9790199 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_concoct_out.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811903.2198017 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811903.2302594 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13\n",
      "24 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811903.454862 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811903.8237205 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811903.833548 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R1_C_H2O_MG_DASTool_bins_concoct_out.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811904.243542 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811904.2575202 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811904.4136293 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811904.443618 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811904.4841828 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811904.5158803 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811904.5580444 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811904.7690783 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n",
      "9 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811905.0871906 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.119\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.524\n",
      " "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811905.5885413 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811905.5924745 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811905.6460426 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811905.657466 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811905.666922 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811905.7455697 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811905.7997942 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811906.0644662 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.119\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811906.3621564 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811906.7198982 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.71\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Salt_Pond_MetaG_R2A_C_D2_MG_DASTool_bins_metabat.38\n",
      "19 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811906.7986937 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811906.9138556 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 10Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.29 \n",
      "Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.16"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811907.0893745 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811907.0912452 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811907.2725341 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.35\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811907.357334 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811907.685144 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.71\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811907.8821218 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811907.9674397 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811908.2257872 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811908.3614821 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.32\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811908.4170256 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811908.5747476 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811908.616284 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811908.6308897 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811909.0000765 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811909.1676645 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811909.530413 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Salt_Pond_MetaG_R1_B_D1_MG_DASTool_bins_metabat.41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811909.7596803 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811909.8435717 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.84\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811909.8972723 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811909.9444926 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811910.1383991 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811910.3188539 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811910.7679656 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_maxbin.014\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811910.8375578 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811911.099371 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1515  Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.31Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.84\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811911.1500533 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n",
      "1724811911.1502428 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811911.234816 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811911.2849424 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811911.3953652 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811911.47328 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811912.0274 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811912.1591113 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811912.3896635 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811912.3973873 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811912.4197063 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811912.4316723 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811912.5046046 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811912.6396594 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811912.6640894 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811913.285116 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13\n",
      "17 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811913.626288 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811913.646209 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811913.7577732 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811913.7811663 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811913.7999077 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811913.9350793 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811914.5573893 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.143\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811914.84768 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811914.9059448 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811915.0628595 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.85\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811915.0913916 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811915.2152839 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811915.225253 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.143\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811915.8244467 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.39\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811916.1032176 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811916.169049 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_concoct_out.51\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811916.4822369 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811916.5066926 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.98\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R1_B_D2_MG_DASTool_bins_metabat.28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811916.5432048 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.39\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811917.1009462 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811917.331801 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811917.6242063 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.8\n",
      "1724811917.7309504 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.98\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811917.7912831 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811917.8519545 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811918.1055045 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.110\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811918.2662544 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811918.3708682 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.8\n",
      "18 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811919.0775979 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811919.0990314 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R1_C_D1_MG_DASTool_bins_metabat.31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811919.1595867 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.110\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811919.3883915 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811919.5959117 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.98\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811919.6323252 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811920.1678944 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.1319\n",
      " "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811920.3435197 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.39\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811920.3477032 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_concoct_out.36\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811920.4783235 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_concoct_out.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811920.6634777 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n",
      "1724811920.8605402 WARNING: Ignoring reaction 'rxn00062_c12' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811920.8863153 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.98\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811920.9224002 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811921.4046578 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.39\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811921.5956793 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811921.7959437 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811921.94799 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811922.26272 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811922.653783 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811922.8628483 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R1_C_D2_MG_DASTool_bins_metabat.40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811923.1674805 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811923.2344728 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811923.3937466 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811923.5910082 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811923.8724673 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811923.8820548 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811924.1044776 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_concoct_out.49\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811924.4740021 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811924.5040214 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811924.6966264 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811924.9097328 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811925.105785 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811925.1216345 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.52\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811925.398973 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.44\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811925.7780035 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811925.9605284 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811926.2327557 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811926.3536675 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.52\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811926.3688328 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.44\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811926.6429157 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.52\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811927.083047 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811927.225547 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.39\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811927.4992473 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811927.552061 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.143\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811927.5815725 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811927.6197193 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.52\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811927.8854964 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.39\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811928.4938726 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1928  Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.12Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811928.8228333 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n",
      "1724811928.8238099 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811928.88386 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.100\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811929.1455643 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811929.7958903 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.44\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2_A_D1_MG_DASTool_bins_metabat.50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811930.1356254 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.100\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811930.1553247 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.104\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811930.2193234 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.143\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811930.812425 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.39\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.44\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811931.0832427 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.55\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.104\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811931.4247189 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811931.4404619 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811931.5048962 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811931.7742276 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811932.079235 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.39\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811932.134147 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811932.3571548 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811932.67423 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811932.743597 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811932.7778215 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.71\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n",
      "30 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811933.3226798 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_concoct_out.4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811933.4535334 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811933.6252606 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_B_D1_MG_DASTool_bins_metabat.22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811934.048222 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.71\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811934.0640182 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.95\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.13\n",
      "21 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811934.7898898 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.10\n",
      "24"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811935.3502028 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.95\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811935.3547378 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_maxbin.062\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811935.654407 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811936.1089122 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n",
      "25"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811936.6298137 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_maxbin.062\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811936.635723 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811936.654598 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811936.9033353 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.47\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811937.416702 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811937.9059525 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811937.949709 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.47\n",
      "24 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811938.7325656 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.44\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n",
      "27 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811939.227628 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.38\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811939.6917057 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.44\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.44\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811940.0440466 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.52\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811940.5077689 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.44\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811941.0042233 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.55\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811942.345613 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.52\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811943.1572092 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.55\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811943.546376 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_concoct_out.65\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811943.7008533 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Salt_Pond_MetaG_R2_B_D2_MG_DASTool_bins_metabat.55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811944.470655 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811944.8167467 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811945.013525 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811945.780134 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Salt_Pond_MetaG_R2_restored_C_black_MG_DASTool_bins_metabat.8\n",
      "31 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811946.3263972 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Salt_Pond_MetaG_R2_C_D1_MG_DASTool_bins_metabat.25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811947.0968373 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.100\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n",
      "30 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.100\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811948.4044144 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.45\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1724811949.7143521 INFO: Making atp hydrolysis reaction for species: Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.47\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32 Salt_Pond_MetaG_R2_C_D2_MG_DASTool_bins_concoct_out.47\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "\n",
    "def buildCommModel(arguments):\n",
    "    (mag_abundances, mag_list) = arguments\n",
    "    \n",
    "    # print status\n",
    "    from multiprocess import current_process\n",
    "    pid = current_process().name\n",
    "    print(pid)\n",
    "    \n",
    "    # build the community model from the members\n",
    "    for info in mag_abundances:\n",
    "        sample, abundances = info\n",
    "        member_models = []\n",
    "        member_names = []\n",
    "        member_abundances = {}\n",
    "        for mag, abundance in abundances.items():\n",
    "            if abundance < 0.01:  continue\n",
    "            try:\n",
    "                mdlutl = util.msrecon.get_model(mag+\".pangenome.RAST.glm4ec.pyr.O2.gf\",186678)\n",
    "            except:\n",
    "                print(f\"The {mag}.pangenome.RAST.glm4ec.pyr.O2.gf is not in 186678\")\n",
    "                continue\n",
    "            member_models.append(mdlutl.model)\n",
    "            member_names.append(mag)\n",
    "            member_abundances[mag] = abundance\n",
    "        comm_model = MSCommunity.build_from_species_models(\n",
    "            member_models,\n",
    "            mdlid=sample,\n",
    "            name=sample,\n",
    "            names=member_names,\n",
    "            abundances=member_abundances\n",
    "        )\n",
    "        cobra.io.save_json_model(comm_model.model, \"models/\"+sample+'.json')\n",
    "    \n",
    "\n",
    "import cobra\n",
    "failingModels = [\"SF2_A_D2\", \"SF2_C_D1\", \"SF2_B_D2\", \"R2_A_D1\", \"R2_B_D1\", \"SF2_C_D2\",\n",
    "                 \"R2_C_D1\", \"R2_C_D2\", \"R2_A_D2\", \"R2_B_D2\", \"R1_C_D1\", \"R1_A_D1\"]\n",
    "mag_list = util.load(\"mag_list\")\n",
    "mag_abundances = util.load(\"refined_MAG_abundances\")\n",
    "\n",
    "# run the parallelized code\n",
    "parallelize = True\n",
    "if parallelize:\n",
    "    from multiprocess import Pool\n",
    "    from os import cpu_count\n",
    "    \n",
    "    # the MAGs are partitioned into groups of 4, and a thread is created to handle each group \n",
    "    pool_count = min(cpu_count(), len(mag_abundances))\n",
    "    print(f\"{pool_count} threads will be created\")\n",
    "    pool = Pool(pool_count)\n",
    "    args = [[[abundances], mag_list] for abundances in list(mag_abundances.items())]\n",
    "    list_of_outputs = pool.map(buildCommModel, args)\n",
    "else:\n",
    "    list_of_outputs = buildCommModel([list(mag_abundances.items()), mag_list])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## compute community model data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run agoracommutil.py\n",
    "import cobra\n",
    "sample_abundanes = util.load(\"mag_abundanes\")\n",
    "\n",
    "community_model_data = {}\n",
    "for pair in sample_pairs:\n",
    "    member_models = []\n",
    "    member_names = []\n",
    "    member_abundances = {}\n",
    "    asv_hash = {}\n",
    "    for asvset in sample_asvset_abundances[pair[0]]:\n",
    "        asvname = asvset_names[asvset]\n",
    "        if asvname not in asv_hash:\n",
    "            asv_hash[asvname] = sample_asvset_abundances[pair[0]][asvset]\n",
    "        else:\n",
    "            asv_hash[asvname] += sample_asvset_abundances[pair[0]][asvset]\n",
    "    for asvset in sample_asvset_abundances[pair[1]]:\n",
    "        asvname = asvset_names[asvset]\n",
    "        if asvname not in asv_hash:\n",
    "            asv_hash[asvname] = sample_asvset_abundances[pair[1]][asvset]\n",
    "        else:\n",
    "            asv_hash[asvname] += sample_asvset_abundances[pair[1]][asvset]\n",
    "    total_abundance = 0\n",
    "    for asvname in asv_hash:\n",
    "        asv_hash[asvname] = asv_hash[asvname]/2\n",
    "        total_abundance += asv_hash[asvname]\n",
    "    for asvname in asv_hash:\n",
    "        asv_hash[asvname] = asv_hash[asvname]/total_abundance\n",
    "        if asvname != \"Thermus.1\" and asv_hash[asvname] > 0.01:\n",
    "            member_names.append(asvname)\n",
    "            member_abundances[asvname] = asv_hash[asvname]\n",
    "    community_model_data[pair[2]] = {\n",
    "        \"names\":member_names,\n",
    "        \"abundances\":member_abundances\n",
    "    }\n",
    "    util.save(\"community_model_data\",community_model_data)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## community model statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from json import load\n",
    "from glob import glob\n",
    "\n",
    "modelStats = {}\n",
    "for path in glob(\"models/*.json\"):\n",
    "    modelStats[path] = {}\n",
    "    for rxn in modelStats[\"reactions\"]:\n",
    "        ID, compartment = rxn.split(\"_\")\n",
    "        modelStats[path][compartment] += 1\n",
    "    \n",
    "modelStats"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running community model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'NotebookApp': {'iopub_data_rate_limit': 10000000}}"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from notebook.services.config import ConfigManager\n",
    "cm = ConfigManager()\n",
    "cm.update('notebook', {\"NotebookApp\": {\"iopub_data_rate_limit\": 10000000}})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## determine the top hits for each methane flux regime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "modelseedpy 0.4.0\n"
     ]
    }
   ],
   "source": [
    "from modelseedpy.biochem import from_local\n",
    "msdb = from_local(\"/scratch/shared/code/ModelSEEDDatabase\")\n",
    "# print(msdb.compounds.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ic| redTop: ['L-Fructose',\n",
      "             'D-Glucose',\n",
      "             'Propionate',\n",
      "             '2-Oxoglutarate',\n",
      "             'Fumarate',\n",
      "             'Sucrose',\n",
      "             'Maltose',\n",
      "             'Acetate',\n",
      "             'Succinate']\n",
      "    greenTop: ['L-Fructose',\n",
      "               'Succinate',\n",
      "               'Maltose',\n",
      "               'Acetate',\n",
      "               'D-Glucose',\n",
      "               'Sucrose',\n",
      "               '2-Oxoglutarate',\n",
      "               'Propionate',\n",
      "               'Fumarate']\n",
      "    blueTop: ['D-Glucose',\n",
      "              'Succinate',\n",
      "              'Maltose',\n",
      "              'Acetate',\n",
      "              'Fumarate',\n",
      "              'Propionate',\n",
      "              'Sucrose',\n",
      "              '2-Oxoglutarate',\n",
      "              'L-Fructose']\n",
      "    purpleTop: ['2-Oxoglutarate',\n",
      "                'Succinate',\n",
      "                'Fumarate',\n",
      "                'L-Fructose',\n",
      "                'Sucrose',\n",
      "                'Acetate',\n",
      "                'Propionate',\n",
      "                'D-Glucose',\n",
      "                'Maltose']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(['L-Fructose',\n",
       "  'D-Glucose',\n",
       "  'Propionate',\n",
       "  '2-Oxoglutarate',\n",
       "  'Fumarate',\n",
       "  'Sucrose',\n",
       "  'Maltose',\n",
       "  'Acetate',\n",
       "  'Succinate'],\n",
       " ['L-Fructose',\n",
       "  'Succinate',\n",
       "  'Maltose',\n",
       "  'Acetate',\n",
       "  'D-Glucose',\n",
       "  'Sucrose',\n",
       "  '2-Oxoglutarate',\n",
       "  'Propionate',\n",
       "  'Fumarate'],\n",
       " ['D-Glucose',\n",
       "  'Succinate',\n",
       "  'Maltose',\n",
       "  'Acetate',\n",
       "  'Fumarate',\n",
       "  'Propionate',\n",
       "  'Sucrose',\n",
       "  '2-Oxoglutarate',\n",
       "  'L-Fructose'],\n",
       " ['2-Oxoglutarate',\n",
       "  'Succinate',\n",
       "  'Fumarate',\n",
       "  'L-Fructose',\n",
       "  'Sucrose',\n",
       "  'Acetate',\n",
       "  'Propionate',\n",
       "  'D-Glucose',\n",
       "  'Maltose'])"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sampleRegime = {'R2A_A_D1': 'red',\n",
    " 'R2A_A_D2': 'red',\n",
    " 'R2A_B_D1': 'red',\n",
    " 'R2A_B_D2': 'red',\n",
    " 'R2A_C_D1': 'red',\n",
    " 'R2A_C_D2': 'red',\n",
    " 'SF2_A_D1': 'red',\n",
    " 'SF2_A_D2': 'red',\n",
    " 'SF2_B_D1': 'red',\n",
    " 'SF2_B_D2': 'red',\n",
    " 'SF2_C_D1': 'red',\n",
    " 'SF2_C_D2': 'red',\n",
    " 'R1_A_D1': 'green',\n",
    " 'R1_A_D2': 'green',\n",
    " 'R1_B_D1': 'blue',\n",
    " 'R1_B_D2': 'blue',\n",
    " 'R1_C_D1': 'green',\n",
    " 'R1_C_D2': 'green',\n",
    " 'R2_A_D1': 'blue',\n",
    " 'R2_A_D2': 'blue',\n",
    " 'R2_B_D1': 'purple',\n",
    " 'R2_B_D2': 'purple',\n",
    " 'R2_C_D1': 'purple',\n",
    " 'R2_C_D2': 'purple'}\n",
    "\n",
    "# load and process the MMIPPs correlations\n",
    "from pandas import read_csv\n",
    "redDF = read_csv(\"nboutput/CH4_MMIPPs_correlations_red.csv\").set_index(\"Unnamed: 0\").rename_axis('Metabolites')\n",
    "greenDF = read_csv(\"nboutput/CH4_MMIPPs_correlations_green.csv\").set_index(\"Unnamed: 0\").rename_axis('Metabolites')\n",
    "blueDF = read_csv(\"nboutput/CH4_MMIPPs_correlations_blue.csv\").set_index(\"Unnamed: 0\").rename_axis('Metabolites')\n",
    "purpleDF = read_csv(\"nboutput/CH4_MMIPPs_correlations_purple.csv\").set_index(\"Unnamed: 0\").rename_axis('Metabolites')\n",
    "\n",
    "# redDF[\"totalEX\"] = redDF[\"Sampleexcretion_probInteractions\"] + redDF[\"Sampleuptake_probInteractions\"]\n",
    "# greenDF[\"totalEX\"] = greenDF[\"Sampleexcretion_probInteractions\"] + greenDF[\"Sampleuptake_probInteractions\"]\n",
    "# blueDF[\"totalEX\"] = blueDF[\"Sampleexcretion_probInteractions\"] + blueDF[\"Sampleuptake_probInteractions\"]\n",
    "# purpleDF[\"totalEX\"] = purpleDF[\"Sampleexcretion_probInteractions\"] + purpleDF[\"Sampleuptake_probInteractions\"]\n",
    "\n",
    "colToSort = \"Sampleuptake_probInteractions\"\n",
    "redDF = redDF.sort_values(by=colToSort, ascending=False)\n",
    "greenDF = greenDF.sort_values(by=colToSort, ascending=False)\n",
    "blueDF = blueDF.sort_values(by=colToSort, ascending=False)\n",
    "purpleDF = purpleDF.sort_values(by=colToSort, ascending=False)\n",
    "\n",
    "# parse and convert the top 10 correlating compounds by uptake and excretion\n",
    "# load the compound categories\n",
    "from json import load\n",
    "cpdNames = load(open(\"/scratch/shared/code/CommScores/data/compoundNames.json\", 'r'))\n",
    "compounds = load(open(\"data/metabolomics_categories.json\", 'r'))\n",
    "metCategory = {v:k for k,vals in compounds.items() for v in vals}\n",
    "new_compounds = {}\n",
    "for category, mets in compounds.items():\n",
    "    new_compounds[category] = {cpdNames[met]:met for met in mets}\n",
    "# print(new_compounds[\"Amino Acids\"])\n",
    "    \n",
    "from numpy import load\n",
    "allCpds = {\"Amino Acids\": dict(load(\"/scratch/shared/code/CommScores/data/categories/aminoAcids.npy\")),\n",
    "          # \"Energy\": load(\"/scratch/shared/code/CommScores/data/categories/energy_compounds.npy\"),\n",
    "          \"Minerals\": dict(load(\"/scratch/shared/code/CommScores/data/categories/minerals.npy\")),\n",
    "          \"Sugars\": dict(load(\"/scratch/shared/code/CommScores/data/categories/sugars.npy\")),\n",
    "          \"Vitamins\": dict(load(\"/scratch/shared/code/CommScores/data/categories/vitamins.npy\"))}\n",
    "allCpds.update(new_compounds)\n",
    "allCpdsCategory = {v:k for k,vals in allCpds.items() for v in vals}\n",
    "\n",
    "# print(allCpds[\"Amino Acids\"])\n",
    "redTop, greenTop, blueTop, purpleTop = [], [], [], []\n",
    "tops = [redTop, greenTop, blueTop, purpleTop]\n",
    "missingCpds = []\n",
    "for i, index in enumerate([redDF.index, greenDF.index, blueDF.index, purpleDF.index]):\n",
    "    for cpd in index:\n",
    "        if cpd not in allCpdsCategory:\n",
    "            missingCpds.append(cpd)  ;  continue\n",
    "        category = allCpdsCategory[cpd]\n",
    "        if category in [\"Amino Acids\"]: continue\n",
    "        tops[i].append(allCpds[category][cpd])\n",
    "        if len(tops[i]) > 10: break\n",
    "    \n",
    "# redTop = [msdb.compounds[cpd][\"name\"] for cpd in list(redDF.index[:10])]\n",
    "# greenTop = [msdb.compounds[cpd][\"name\"] for cpd in list(greenDF.index[:10])]\n",
    "# blueTop = [msdb.compounds[cpd][\"name\"] for cpd in list(blueDF.index[:10])]\n",
    "# purpleTop = [msdb.compounds[cpd][\"name\"] for cpd in list(purpleDF.index[:10])]\n",
    "\n",
    "from icecream import ic\n",
    "ic(redTop, greenTop, blueTop, purpleTop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 {'Sucrose', 'Succinate', 'Maltose', 'Fumarate', 'L-Fructose', 'Acetate', 'Propionate', 'D-Glucose', '2-Oxoglutarate'}\n"
     ]
    }
   ],
   "source": [
    "unionCPDs = set(redTop + greenTop + blueTop + purpleTop)\n",
    "print(len(unionCPDs), unionCPDs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## run the model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %run cliffcommutil.py\n",
    "\n",
    "# def runCommModel(arguments):\n",
    "#     from multiprocess import current_process\n",
    "#     from optlang.symbolics import Zero, add\n",
    "    \n",
    "#     print(arguments)\n",
    "#     (model_paths, feature_probabilities, metabolomics_data, metabolites, dipeptide_exchanges, media) = arguments\n",
    "#     output, rxn_record_hash = {}, {}\n",
    "#     modelIDs = []\n",
    "#     records = util.load(\"records\")\n",
    "#     for model_path in model_paths:\n",
    "#         # print status\n",
    "#         pid = current_process().name\n",
    "#         print(pid)\n",
    "\n",
    "#         modelID = model_path.split(\"/\")[-1].replace(\".json\", '')\n",
    "#         modelIDs.append(modelID)\n",
    "#         output[modelID] = {}\n",
    "#         base_model = load_json_model(model_path)\n",
    "#         current_comm_model = MSCommunity(\n",
    "#             model=base_model,\n",
    "#         )\n",
    "#         # print(dir(current_comm_model))\n",
    "#         # assign reaction probabilities\n",
    "#         reaction_probabilities = {}\n",
    "#         for rxn in base_model.reactions:\n",
    "#             highest_prob = None\n",
    "#             for gene in rxn.genes:\n",
    "#                 if gene.id in feature_probabilities:\n",
    "#                     if highest_prob == None or feature_probabilities[gene.id] > highest_prob:\n",
    "#                         highest_prob = feature_probabilities[gene.id]\n",
    "#             if highest_prob != None:\n",
    "#                 rxn.probability = highest_prob\n",
    "#                 reaction_probabilities[rxn.id] = highest_prob\n",
    "#             elif rxn.id[0:3] != \"bio\" and rxn.id[0:3] != \"EX_\" and rxn.id[0:3] != \"DM_\" and len(rxn.genes) == 0:\n",
    "#                 reaction_probabilities[rxn.id] = 0\n",
    "#             else:\n",
    "#                 reaction_probabilities[rxn.id] = 0.05\n",
    "#         min_prob, prob_exp, ex_weight = 0.05, 1, 1\n",
    "#         mdlutl = current_comm_model.mdlutl\n",
    "#         pkgmgr = MSPackageManager.get_pkg_mgr(mdlutl)\n",
    "#         # add media, community kinetics, and elemental uptake constraints\n",
    "#         pkgmgr.getpkg(\"KBaseMediaPkg\").build_package(media)\n",
    "#         pkgmgr.getpkg(\"CommKineticPkg\").build_package(750, current_comm_model)\n",
    "#         cUptake = 500\n",
    "#         pkgmgr.getpkg(\"ElementUptakePkg\").build_package({\"C\": cUptake})\n",
    "#         # update exchange flux bounds \n",
    "#         for rxn in base_model.reactions:\n",
    "#             if \"EX_\" != rxn.id[0:3]:   continue\n",
    "#             currRxn = mdlutl.model.reactions.get_by_id(rxn.id)\n",
    "#             if rxn.id in dipeptide_exchanges:\n",
    "#                 currRxn.lower_bound = 0\n",
    "#                 currRxn.upper_bound = 0\n",
    "#             else:\n",
    "#                 ## the bounds are set to either -1000 or 1000\n",
    "#                 if currRxn.lower_bound < 0:\n",
    "#                     currRxn.lower_bound = -1000\n",
    "#                 if currRxn.upper_bound > 0:\n",
    "#                     currRxn.upper_bound = 1000\n",
    "#         if \"EX_cpd00007_e0\" in [rxn.id for rxn in mdlutl.exchange_list()]:\n",
    "#             mdlutl.model.reactions.get_by_id(\"EX_cpd00007_e0\").lower_bound = -cUptake/3   # oxygen\n",
    "#         else:\n",
    "#             print(f\"The {mdlutl.model.id} does not have Oxygen defined in the model\")\n",
    "\n",
    "#         # maximize biomass production and store the results\n",
    "#         mdlutl.model.objective = mdlutl.model.problem.Objective(Zero, direction=\"max\")\n",
    "#         mdlutl.model.objective.set_linear_coefficients({mdlutl.model.reactions.bio1.forward_variable: 1})\n",
    "#         maxBiomassSol = mdlutl.model.optimize()\n",
    "#         output[modelID][\"max_growth\"] = maxBiomassSol.objective_value\n",
    "#         output[modelID][\"carbon_uptake\"] = pkgmgr.getpkg(\"ElementUptakePkg\").variables[\"elements\"][\"C\"].primal\n",
    "#         records[0][modelID] = output[modelID][\"max_growth\"]\n",
    "#         records[1][modelID] = output[modelID][\"carbon_uptake\"]\n",
    "#         print(modelID+\" growth:\", output[modelID][\"max_growth\"])\n",
    "#         print(modelID+\"carbon uptake:\",output[modelID][\"carbon_uptake\"])\n",
    "#         ## export the LP\n",
    "#         export_path = util.output_dir+\"/LPs/\"+modelID+\"-growth.lp\"\n",
    "#         if not os.path.exists(export_path):\n",
    "#             os.makedirs(export_path)\n",
    "#         with open(export_path, \"w\") as out:\n",
    "#             out.write(str(mdlutl.model.solver))\n",
    "#         if str(output[modelID][\"max_growth\"]) == \"nan\":\n",
    "#             print(\"Skipping condition due to infeasibility\", modelID)\n",
    "# [modelID]output[modelID][\"carbon_uptake\"])\n",
    "#             return\n",
    "\n",
    "#         # simulate the community with the specified community growth\n",
    "#         for percentOptimum in [1, 0.9, 0.8, 0.7, 0.6, 0.5]:\n",
    "#             mdlutl.add_minimal_objective_cons(output[modelID][\"max_growth\"] * percentOptimum)\n",
    "#             # pkgmgr.getpkg(\"ObjConstPkg\").build_package(, None)\n",
    "#             ## defining the minimal probability objective\n",
    "#             output[modelID][percentOptimum] = {}\n",
    "#             coef, flux_fitting_target = {}, {}\n",
    "#             for rxn in base_model.reactions:\n",
    "#                 if \"rxn\" == rxn.id[0:3]:\n",
    "#                     prob = reaction_probabilities[rxn.id]\n",
    "#                     currRxn = mdlutl.model.reactions.get_by_id(rxn.id)\n",
    "#                     coef.update({currRxn.forward_variable: max(min_prob, (1 - float(prob) ** prob_exp))})\n",
    "#                     coef.update({currRxn.reverse_variable: max(min_prob, (1 - float(prob) ** prob_exp))})\n",
    "#                 elif \"EX_\" == rxn.id[0:3]:\n",
    "#                     currRxn = mdlutl.model.reactions.get_by_id(rxn.id)\n",
    "#                     if rxn.id[3:11] in metabolomics_data:\n",
    "#                         flux_fitting_target[rxn.id] = -1*output[modelID][\"max_growth\"]*metabolomics_data[rxn.id[3:11]][modelID]\n",
    "#                     coef.update({currRxn.forward_variable: ex_weight})\n",
    "#                     coef.update({currRxn.reverse_variable: ex_weight})\n",
    "\n",
    "#             ## Adjusting rxn bounds according to the maximum community growth and metabolomics data\n",
    "#             stuck_reactions = {}\n",
    "#             for iteration in range(1,11,1):\n",
    "#                 print(\"Iteration\",iteration)\n",
    "#                 for rxn in flux_fitting_target:\n",
    "#                     if rxn in stuck_reactions:   continue\n",
    "#                     currRxn = mdlutl.model.reactions.get_by_id(rxn)\n",
    "#                     starting_point = maxBiomassSol.fluxes[rxn]\n",
    "#                     distance = abs(flux_fitting_target[rxn] - starting_point)\n",
    "#                     original_upper = currRxn.upper_bound\n",
    "#                     original_lower = currRxn.lower_bound\n",
    "#                     if starting_point > flux_fitting_target[rxn]:\n",
    "#                         if starting_point-iteration*distance/10 < currRxn.upper_bound:\n",
    "#                             currRxn.lower_bound = starting_point-iteration*distance/10\n",
    "#                             currRxn.upper_bound = starting_point-iteration*distance/10\n",
    "#                         else:\n",
    "#                             currRxn.upper_bound = starting_point-iteration*distance/10\n",
    "#                             currRxn.lower_bound = starting_point-iteration*distance/10\n",
    "#                     else:\n",
    "#                         if flux_fitting_target[rxn]-(10-iteration)*distance/10 < currRxn.upper_bound:\n",
    "#                             currRxn.lower_bound = flux_fitting_target[rxn]-(10-iteration)*distance/10\n",
    "#                             currRxn.upper_bound = flux_fitting_target[rxn]-(10-iteration)*distance/10\n",
    "#                         else:\n",
    "#                             currRxn.upper_bound = flux_fitting_target[rxn]-(10-iteration)*distance/10\n",
    "#                             currRxn.lower_bound = flux_fitting_target[rxn]-(10-iteration)*distance/10\n",
    "#                     solution = mdlutl.model.optimize()\n",
    "#                     if solution.status != \"optimal\":\n",
    "#                         print(\"Stuck\",rxn,starting_point,flux_fitting_target[rxn],original_upper)\n",
    "#                         stuck_reactions[rxn] = 1\n",
    "#                         if original_upper >= currRxn.lower_bound:\n",
    "#                             currRxn.upper_bound = original_upper\n",
    "#                             currRxn.lower_bound = original_lower\n",
    "#                         else:\n",
    "#                             currRxn.lower_bound = original_lower\n",
    "#                             currRxn.upper_bound = original_upper\n",
    "#             for rxn in flux_fitting_target:\n",
    "#                 currRxn = mdlutl.model.reactions.get_by_id(rxn)\n",
    "#                 print(currRxn.id,currRxn.lower_bound,currRxn.upper_bound)\n",
    "\n",
    "#             ## maximize methane excretion\n",
    "#             ch4_weight = 100\n",
    "#             if \"EX_cpd01024_e0\" not in [rxn.id for rxn in mdlutl.exchange_list()]:\n",
    "#                 print(f\"The {mdlutl.model.id} does not have Methane defined in the model\")\n",
    "#             else:\n",
    "#                 methane = mdlutl.model.reactions.get_by_id(\"EX_cpd01024_e0\")\n",
    "#                 coef.update({methane.reverse_variable: ch4_weight, methane.forward_variable: -ch4_weight})\n",
    "\n",
    "#             ## assigning the minimum objective function\n",
    "#             mdlutl.model.objective = mdlutl.model.problem.Objective(Zero, direction=\"min\")\n",
    "#             mdlutl.model.objective.set_linear_coefficients(coef)\n",
    "\n",
    "#             ## simulating and checking the solution\n",
    "#             solution = mdlutl.model.optimize()\n",
    "#             if solution.status != \"optimal\":\n",
    "#                 continue\n",
    "#                 # while solution.status != \"optimal\":\n",
    "#                 #     util.save(\"objConstPkg_contents\", mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"])\n",
    "#                 #     print(dir(mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"]),\n",
    "#                 #           mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"])\n",
    "#                 #     print(\"Infeasible: reducing objective constraint\",0.9*mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"].lb)\n",
    "#                 #     mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"].lb = 0.9*mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"].lb\n",
    "#                 #     solution = mdlutl.model.optimize()\n",
    "\n",
    "#             ## storing and exporting successful simulation results\n",
    "#             export_path = util.output_dir+\"/LPs/\"+modelID+\"-final.lp\"\n",
    "#             if not os.path.exists(export_path):\n",
    "#                 os.makedirs(export_path)\n",
    "#             with open(export_path, \"w\") as out:\n",
    "#                 out.write(str(mdlutl.model.solver))\n",
    "\n",
    "#             output[modelID][percentOptimum][\"minimum probability objective\"] = solution.objective_value\n",
    "#             records[3][modelID] = output[modelID][\"minimum probability objective\"]\n",
    "#             output[modelID][percentOptimum][\"solution\"] = {}\n",
    "#             for rxn in mdlutl.model.reactions:\n",
    "#                 if rxn.id not in rxn_record_hash:\n",
    "#                     rxn_record_hash[rxn.id] = {\"id\":rxn.id}\n",
    "#                     for mdlID in modelIDs:\n",
    "#                         rxn_record_hash[rxn.id][mdlID] = None\n",
    "#                     records.append(rxn_record_hash[rxn.id])\n",
    "#                 rxn_record_hash[rxn.id][modelID] = solution.fluxes[rxn.id]\n",
    "#                 output[modelID][percentOptimum][\"solution\"][rxn.id] = solution.fluxes[rxn.id]\n",
    "            \n",
    "#             dirName = util.output_dir+f\"/{modelID}\"\n",
    "#             if not os.path.exists(dirName):\n",
    "#                 os.makedirs(dirName)\n",
    "#             baseName = dirName+f\"/{percentOptimum*100}%\"\n",
    "#             with open(baseName+\".json\", \"w\") as out:\n",
    "#                 dump(output, out, indent=3)\n",
    "#             df = DataFrame.from_records(records)\n",
    "#             df.to_csv(f\"{baseName}.csv\")\n",
    "        \n",
    "#         with open(path.join(util.output_dir, \"outputs\", f\"{modelID}_output.json\"), \"w\") as out:\n",
    "#             dump(output, out, indent=3)\n",
    "#         with open(path.join(util.output_dir, \"outputs\", f\"{modelID}_rxn_record_hash.json\"), \"w\") as out:\n",
    "#             dump(rxn_record_hash, out, indent=3)\n",
    "#         util.save(f\"{modelID}_records\", records)\n",
    "\n",
    "# # import stored data\n",
    "# metabolites = util.load(\"metabolites\")\n",
    "# metabolomics_path = \"/scratch/shared/code/MicrobiomeNotebooks/Cliff/data/Cliff_Sample_Metadata_BGC_NMR.csv\"\n",
    "# metabolomics_data = read_csv(metabolomics_path).drop([\"New_index\", \"Unnamed: 0\"], axis=1).set_index(\"Sample\").to_dict()\n",
    "# feature_probabilities = util.load(\"feature_probabilities\")\n",
    "# complete_media = util.msrecon.get_media(\"KBaseMedia/Complete\")\n",
    "\n",
    "\n",
    "# # R1_B_D2 is relaly slow for some reason\n",
    "\n",
    "# from glob import glob\n",
    "# from cobra.io import load_json_model\n",
    "# from json import load, dump\n",
    "# with open(\"/scratch/shared/code/MicrobiomeNotebooks/Cliff/data/dipeptide_exchanges.json\", 'r') as jsonIn:\n",
    "#     dipeptide_exchanges = load(jsonIn)\n",
    "# parallelize = True\n",
    "# if parallelize:\n",
    "#     from multiprocess import Pool\n",
    "#     from os import cpu_count\n",
    "#     from numpy import array_split\n",
    "    \n",
    "#     # a thread is given for each community model\n",
    "#     pool_count = min(cpu_count(), len(list(glob(\"models/*.json\"))))\n",
    "#     print(f\"{pool_count} threads will be created\")\n",
    "#     pool = Pool(pool_count)\n",
    "#     args = [[[model_path], feature_probabilities, metabolomics_data, metabolites, dipeptide_exchanges, complete_media]\n",
    "#             for model_path in list(glob(\"models/*.json\"))]\n",
    "#     list_of_outputs = pool.map(runCommModel, args)\n",
    "# else:\n",
    "#     outputs = runCommModel([list(glob(\"models/*.json\")), feature_probabilities,\n",
    "#                             metabolomics_data, metabolites, dipeptide_exchanges, complete_media])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python version 3.11.1\n",
      "KBBaseModules 0.0.1\n",
      "modelseedpy 0.3.3\n",
      "cobrakbase 0.4.0\n",
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n",
      "<cobrakbase.core.kbasebiochem.media.Media object at 0x7ff8a4c45e10>\n",
      "174 threads will be created\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-1 models/R2A_A_D1.json"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247678.346249 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-2 models/R1_B_D1.json"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247695.0202005 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9b8a62390>: 1, <Metabolite cpd11416_c1 at 0x7ff9bb456cd0>: -0.016920161836424443, <Metabolite cpd11416_c10 at 0x7ff9bb0a8610>: -0.06940211531249946, <Metabolite cpd11416_c11 at 0x7ff9b8f29c90>: -0.06330829407178278, <Metabolite cpd11416_c12 at 0x7ff9b8faa0d0>: -0.097817690314725, <Metabolite cpd11416_c13 at 0x7ff9b8e218d0>: -0.03730169507377297, <Metabolite cpd11416_c14 at 0x7ff9b8e96750>: -0.07716344705468689, <Metabolite cpd11416_c15 at 0x7ff9b8d1c290>: -0.024880911377561113, <Metabolite cpd11416_c16 at 0x7ff9b8d8ae50>: -0.13360902600480187, <Metabolite cpd11416_c17 at 0x7ff9b8c1c9d0>: -0.05446758820022214, <Metabolite cpd11416_c18 at 0x7ff9b8c9df90>: -0.01499792713403989, <Metabolite cpd11416_c19 at 0x7ff9b8afcd10>: -0.016092364389151636, <Metabolite cpd11416_c2 at 0x7ff9bb4d7110>: -0.01421441977289274, <Metabolite cpd11416_c20 at 0x7ff9b8b70d10>: -0.024475555919268255, <Metabolite cpd11416_c21 at 0x7ff9b89f61d0>: -0.019982023171133838, <Metabolite cpd11416_c22 at 0x7ff9b8a62310>: -0.020221234600764196, <Metabolite cpd11416_c3 at 0x7ff9bb34f610>: -0.038040013057707024, <Metabolite cpd11416_c4 at 0x7ff9bb3c6090>: -0.01724008464127692, <Metabolite cpd11416_c5 at 0x7ff9bb24b250>: -0.014028940985820873, <Metabolite cpd11416_c6 at 0x7ff9bb2c2850>: -0.0792156293194075, <Metabolite cpd11416_c7 at 0x7ff9bb13a950>: -0.0470747409816837, <Metabolite cpd11416_c8 at 0x7ff9bb1bc090>: -0.09792245772657476, <Metabolite cpd11416_c9 at 0x7ff9bb02a850>: -0.021623679053802074}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9b8a62390>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247709.534171 INFO: Making atp hydrolysis reaction for species: Species1\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n",
      "1726247710.367995 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-3 models/R1_A_D1.json\n",
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247710.7669685 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1726247711.2440228 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species2 23 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247712.06831 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247713.672221 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247714.4872568 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247715.3126438 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247716.1616995 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247717.0066981 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247718.5735579 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247719.4046001 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247720.2233853 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247721.0458794 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247721.8633292 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247722.6916692 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247724.3084528 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247725.1415055 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247725.9900556 INFO: Making atp hydrolysis reaction for species: Species18\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-4 models/R1_B_D2.json18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247726.842008 INFO: Making atp hydrolysis reaction for species: Species19\n",
      "1726247727.352819 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247727.6933842 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247728.5329812 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247730.1229632 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n",
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac699850>: 1, <Metabolite cpd11416_c1 at 0x7ff9b15e0450>: -0.030555393141213385, <Metabolite cpd11416_c10 at 0x7ff9b1074510>: -0.02425097014180005, <Metabolite cpd11416_c11 at 0x7ff9aeef3190>: -0.02507783080143885, <Metabolite cpd11416_c12 at 0x7ff9aef73ad0>: -0.022398020562659347, <Metabolite cpd11416_c13 at 0x7ff9aee08ad0>: -0.020517711429274058, <Metabolite cpd11416_c14 at 0x7ff9aee7d0d0>: -0.02804232829384062, <Metabolite cpd11416_c15 at 0x7ff9aece8710>: -0.03893639574422663, <Metabolite cpd11416_c16 at 0x7ff9aed5d190>: -0.023940349411085412, <Metabolite cpd11416_c17 at 0x7ff9aedd1990>: -0.017172240066672822, <Metabolite cpd11416_c18 at 0x7ff9aec59050>: -0.02342986660844035, <Metabolite cpd11416_c19 at 0x7ff9aeccf550>: -0.016830140003275817, <Metabolite cpd11416_c2 at 0x7ff9b145c9d0>: -0.05125569526358057, <Metabolite cpd11416_c20 at 0x7ff9aeb57990>: -0.09937027073863476, <Metabolite cpd11416_c21 at 0x7ff9aebd8b50>: -0.03275260830831503, <Metabolite cpd11416_c22 at 0x7ff9aea53b50>: -0.053958692460494034, <Metabolite cpd11416_c23 at 0x7ff9ae8e8f50>: -0.08753192894311802, <Metabolite cpd11416_c24 at 0x7ff9ae95ead0>: -0.02532755275600765, <Metabolite cpd11416_c25 at 0x7ff9ae9e3e90>: -0.02001140122273962, <Metabolite cpd11416_c26 at 0x7ff9ae86b850>: -0.021290803698525814, <Metabolite cpd11416_c27 at 0x7ff9ac6fcb50>: -0.02103199710432373, <Metabolite cpd11416_c28 at 0x7ff9ac782290>: -0.038875708238769895, <Metabolite cpd11416_c29 at 0x7ff9ac60b9d0>: -0.037810217335742594, <Metabolite cpd11416_c3 at 0x7ff9b14d0fd0>: -0.027396209708764843, <Metabolite cpd11416_c30 at 0x7ff9ac6997d0>: -0.02609546889337511, <Metabolite cpd11416_c4 at 0x7ff9b1359c90>: -0.018287115493287754, <Metabolite cpd11416_c5 at 0x7ff9b13df350>: -0.01598415934626929, <Metabolite cpd11416_c6 at 0x7ff9b126d8d0>: -0.028530997397831742, <Metabolite cpd11416_c7 at 0x7ff9b10ea050>: -0.07329957333356177, <Metabolite cpd11416_c8 at 0x7ff9b116c210>: -0.033116257982004675, <Metabolite cpd11416_c9 at 0x7ff9b0ff0e50>: -0.016922095570725805}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac699850>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247741.7887933 INFO: Making atp hydrolysis reaction for species: Species1\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-5 models/R2A_A_D1.json1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247743.0357215 INFO: Making atp hydrolysis reaction for species: Species2\n",
      "1726247743.0700097 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247744.3092878 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247745.5475569 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247747.8796217 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247749.1057045 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247750.3674006 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247751.5973504 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ffa21f0d750>: 1, <Metabolite cpd11416_c1 at 0x7ffa22c0e050>: -0.019751108060265848, <Metabolite cpd11416_c10 at 0x7ffa2287a850>: -0.0339275251457229, <Metabolite cpd11416_c11 at 0x7ffa22703110>: -0.016447509739534696, <Metabolite cpd11416_c12 at 0x7ffa22790190>: -0.0219134692614876, <Metabolite cpd11416_c13 at 0x7ffa2260c790>: -0.08911515127315858, <Metabolite cpd11416_c14 at 0x7ffa22681210>: -0.047144857116562355, <Metabolite cpd11416_c15 at 0x7ffa224fb290>: -0.028758419749027573, <Metabolite cpd11416_c16 at 0x7ffa2256fb10>: -0.06701785075654187, <Metabolite cpd11416_c17 at 0x7ffa223e5b10>: -0.02160304119681276, <Metabolite cpd11416_c18 at 0x7ffa2245bf90>: -0.04084734220633616, <Metabolite cpd11416_c19 at 0x7ffa222f0410>: -0.09417645813379602, <Metabolite cpd11416_c2 at 0x7ffa22c57310>: -0.02766186934416273, <Metabolite cpd11416_c20 at 0x7ffa22365590>: -0.03355678401003013, <Metabolite cpd11416_c21 at 0x7ffa221ec650>: -0.0726594785116376, <Metabolite cpd11416_c22 at 0x7ffa2226da10>: -0.0409788834749136, <Metabolite cpd11416_c23 at 0x7ffa220eb590>: -0.028346583628364473, <Metabolite cpd11416_c24 at 0x7ffa22178910>: -0.03566480280774384, <Metabolite cpd11416_c25 at 0x7ffa22002050>: -0.02022232433514859, <Metabolite cpd11416_c26 at 0x7ffa22087790>: -0.026874207691255172, <Metabolite cpd11416_c27 at 0x7ffa21f0d6d0>: -0.01955953359570993, <Metabolite cpd11416_c3 at 0x7ffa22af3310>: -0.01493016721274165, <Metabolite cpd11416_c4 at 0x7ffa22b67a10>: -0.02240236430970038, <Metabolite cpd11416_c5 at 0x7ffa229fc190>: -0.031545042225913415, <Metabolite cpd11416_c6 at 0x7ffa22a70a10>: -0.06663987951566139, <Metabolite cpd11416_c7 at 0x7ffa228eae10>: -0.028003969934144054, <Metabolite cpd11416_c8 at 0x7ffa2296fc50>: -0.015692033716172716, <Metabolite cpd11416_c9 at 0x7ffa227f6990>: -0.034559343047453984}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ffa21f0d750>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247752.4132597 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247752.864207 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247753.4978852 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247754.60204 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247755.1744814 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247755.7299998 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247756.550581 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247757.8020601 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247757.8162472 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247758.929242 INFO: Making atp hydrolysis reaction for species: Species6\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247759.1276793 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-6 models/R1_A_D1.json\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247759.5907967 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247760.0430958 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247760.5271704 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247761.2109687 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247762.3261821 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247763.180408 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247764.4182017 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247764.504776 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247765.5619469 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247765.8399253 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247766.7389798 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247767.1610496 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247767.8928382 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247768.4702752 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247769.0021276 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247769.7256866 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247771.1815386 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247772.203696 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247772.3123455 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247773.447531 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247773.4575365 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9b8a5af10>: 1, <Metabolite cpd11416_c1 at 0x7ff9bb453750>: -0.016920161836424443, <Metabolite cpd11416_c10 at 0x7ff9bb0a9090>: -0.06940211531249946, <Metabolite cpd11416_c11 at 0x7ff9b8f26810>: -0.06330829407178278, <Metabolite cpd11416_c12 at 0x7ff9b8faac50>: -0.097817690314725, <Metabolite cpd11416_c13 at 0x7ff9b8e22450>: -0.03730169507377297, <Metabolite cpd11416_c14 at 0x7ff9b8e972d0>: -0.07716344705468689, <Metabolite cpd11416_c15 at 0x7ff9b8d1ce10>: -0.024880911377561113, <Metabolite cpd11416_c16 at 0x7ff9b8d839d0>: -0.13360902600480187, <Metabolite cpd11416_c17 at 0x7ff9b8c19550>: -0.05446758820022214, <Metabolite cpd11416_c18 at 0x7ff9b8c9eb10>: -0.01499792713403989, <Metabolite cpd11416_c19 at 0x7ff9b8af9890>: -0.016092364389151636, <Metabolite cpd11416_c2 at 0x7ff9bb4d7b90>: -0.01421441977289274, <Metabolite cpd11416_c20 at 0x7ff9b8b6d890>: -0.024475555919268255, <Metabolite cpd11416_c21 at 0x7ff9b89f6d50>: -0.019982023171133838, <Metabolite cpd11416_c22 at 0x7ff9b8a5ae90>: -0.020221234600764196, <Metabolite cpd11416_c3 at 0x7ff9bb35c0d0>: -0.038040013057707024, <Metabolite cpd11416_c4 at 0x7ff9bb3c2b10>: -0.01724008464127692, <Metabolite cpd11416_c5 at 0x7ff9bb24bcd0>: -0.014028940985820873, <Metabolite cpd11416_c6 at 0x7ff9bb2bf2d0>: -0.0792156293194075, <Metabolite cpd11416_c7 at 0x7ff9bb1373d0>: -0.0470747409816837, <Metabolite cpd11416_c8 at 0x7ff9bb1b8b10>: -0.09792245772657476, <Metabolite cpd11416_c9 at 0x7ff9bb0272d0>: -0.021623679053802074}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9b8a5af10>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247774.345012 INFO: Making atp hydrolysis reaction for species: Species1\n",
      "1726247774.5179079 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247774.5636098 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247774.7673306 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247775.1690664 INFO: Making atp hydrolysis reaction for species: Species2\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-7 models/R1_B_D1.json\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac2419d0>: 1, <Metabolite cpd11416_c1 at 0x7ff9b10f63d0>: -0.015063437751931306, <Metabolite cpd11416_c10 at 0x7ff9aed8b790>: -0.015472284020236317, <Metabolite cpd11416_c11 at 0x7ff9aec20810>: -0.015351150748895646, <Metabolite cpd11416_c12 at 0x7ff9aec94e10>: -0.057949211998001404, <Metabolite cpd11416_c13 at 0x7ff9aeb0da90>: -0.025703921762683234, <Metabolite cpd11416_c14 at 0x7ff9aeb83b10>: -0.024233570032900033, <Metabolite cpd11416_c15 at 0x7ff9aea083d0>: -0.027255630377544103, <Metabolite cpd11416_c16 at 0x7ff9aea7f9d0>: -0.025754651689439887, <Metabolite cpd11416_c17 at 0x7ff9ae8f98d0>: -0.015230962488891848, <Metabolite cpd11416_c18 at 0x7ff9ae96fd50>: -0.020710181174136946, <Metabolite cpd11416_c19 at 0x7ff9ae8001d0>: -0.09561255291204401, <Metabolite cpd11416_c2 at 0x7ff9b116b450>: -0.0681958840117142, <Metabolite cpd11416_c20 at 0x7ff9ae875350>: -0.04211271201758899, <Metabolite cpd11416_c21 at 0x7ff9ac6fc410>: -0.05863460294747323, <Metabolite cpd11416_c22 at 0x7ff9ac7817d0>: -0.06559821966736985, <Metabolite cpd11416_c23 at 0x7ff9ac5fb2d0>: -0.025900956626385217, <Metabolite cpd11416_c24 at 0x7ff9ac688950>: -0.015482765055061519, <Metabolite cpd11416_c25 at 0x7ff9ac511c90>: -0.015682892683137918, <Metabolite cpd11416_c26 at 0x7ff9ac594550>: -0.024024765482550323, <Metabolite cpd11416_c27 at 0x7ff9ac40f9d0>: -0.01455381891108557, <Metabolite cpd11416_c28 at 0x7ff9ac4a0d50>: -0.02355621167979012, <Metabolite cpd11416_c29 at 0x7ff9ac32a490>: -0.030096892871452993, <Metabolite cpd11416_c3 at 0x7ff9b0ff0390>: -0.015282422651680989, <Metabolite cpd11416_c30 at 0x7ff9ac3afbd0>: -0.051370644442749275, <Metabolite cpd11416_c31 at 0x7ff9ac241950>: -0.02174423835179016, <Metabolite cpd11416_c4 at 0x7ff9b1075350>: -0.02527033337938133, <Metabolite cpd11416_c5 at 0x7ff9aeeffc10>: -0.022867126595093153, <Metabolite cpd11416_c6 at 0x7ff9aef803d0>: -0.054916455843228504, <Metabolite cpd11416_c7 at 0x7ff9aedfa650>: -0.03448135887819264, <Metabolite cpd11416_c8 at 0x7ff9aee7d510>: -0.033081109807568225, <Metabolite cpd11416_c9 at 0x7ff9aed052d0>: -0.018809033140001027}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac2419d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247775.3505952 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247775.6863594 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247775.9871056 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247776.0354838 INFO: Making atp hydrolysis reaction for species: Species24\n",
      "1726247776.1675088 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247776.6754258 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247776.8090694 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247777.3152704 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247777.8180058 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247778.0551212 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247778.4166722 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247778.9372303 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247779.235724 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247779.387941 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247779.7406335 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247780.0405087 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247780.057325 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247780.8924842 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247781.0046682 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247781.1845298 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247781.6954434 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247781.9156904 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247782.255098 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247782.3087733 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247783.184752 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247783.2841136 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247783.542378 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247784.101096 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247784.3915267 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247784.499434 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247784.8159983 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247784.9100153 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247785.5000653 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247785.739445 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247785.801708 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247786.5680609 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247786.6355357 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247787.0953615 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n",
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247787.37995 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n",
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247788.934783 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247789.5133286 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247789.740537 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247790.5481658 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247790.8027773 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247791.3568194 INFO: Making atp hydrolysis reaction for species: Species19\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-8 models/R1_B_D2.json\n",
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247792.0867298 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247792.1805177 INFO: Making atp hydrolysis reaction for species: Species20\n",
      "1726247792.633093 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1726247792.988045 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species20 20"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247793.3402047 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species12 1221 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247794.5144928 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247794.603667 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n",
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247797.028866 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247798.2983904 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247799.5490363 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ffa21f0a390>: 1, <Metabolite cpd11416_c1 at 0x7ffa22c0ebd0>: -0.019751108060265848, <Metabolite cpd11416_c10 at 0x7ffa22877490>: -0.0339275251457229, <Metabolite cpd11416_c11 at 0x7ffa226ffd50>: -0.016447509739534696, <Metabolite cpd11416_c12 at 0x7ffa2278cdd0>: -0.0219134692614876, <Metabolite cpd11416_c13 at 0x7ffa226053d0>: -0.08911515127315858, <Metabolite cpd11416_c14 at 0x7ffa2267de50>: -0.047144857116562355, <Metabolite cpd11416_c15 at 0x7ffa224fbed0>: -0.028758419749027573, <Metabolite cpd11416_c16 at 0x7ffa22574790>: -0.06701785075654187, <Metabolite cpd11416_c17 at 0x7ffa223e6750>: -0.02160304119681276, <Metabolite cpd11416_c18 at 0x7ffa22468c10>: -0.04084734220633616, <Metabolite cpd11416_c19 at 0x7ffa222ed050>: -0.09417645813379602, <Metabolite cpd11416_c2 at 0x7ffa22c57f50>: -0.02766186934416273, <Metabolite cpd11416_c20 at 0x7ffa223621d0>: -0.03355678401003013, <Metabolite cpd11416_c21 at 0x7ffa221e9290>: -0.0726594785116376, <Metabolite cpd11416_c22 at 0x7ffa2226e650>: -0.0409788834749136, <Metabolite cpd11416_c23 at 0x7ffa220f4210>: -0.028346583628364473, <Metabolite cpd11416_c24 at 0x7ffa22179550>: -0.03566480280774384, <Metabolite cpd11416_c25 at 0x7ffa22002c90>: -0.02022232433514859, <Metabolite cpd11416_c26 at 0x7ffa22090410>: -0.026874207691255172, <Metabolite cpd11416_c27 at 0x7ffa21f0a310>: -0.01955953359570993, <Metabolite cpd11416_c3 at 0x7ffa22aeff50>: -0.01493016721274165, <Metabolite cpd11416_c4 at 0x7ffa22b70690>: -0.02240236430970038, <Metabolite cpd11416_c5 at 0x7ffa229f8dd0>: -0.031545042225913415, <Metabolite cpd11416_c6 at 0x7ffa22a6d650>: -0.06663987951566139, <Metabolite cpd11416_c7 at 0x7ffa228eba50>: -0.028003969934144054, <Metabolite cpd11416_c8 at 0x7ffa229788d0>: -0.015692033716172716, <Metabolite cpd11416_c9 at 0x7ffa227f35d0>: -0.034559343047453984}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ffa21f0a390>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247799.8636043 INFO: Making atp hydrolysis reaction for species: Species1\n",
      "1726247800.7494006 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n",
      "1726247800.795081 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247800.937951 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species1 1Species17 172 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247802.0250647 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247802.040615 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247803.0993905 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247804.4091237 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247805.1508627 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247805.6912029 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247806.251341 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247806.981671 INFO: Making atp hydrolysis reaction for species: Species22\n",
      "1726247807.3313556 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species6 \n",
      "6"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-9 models/R2_A_D1.json\n",
      "1 missing R2_A_D1\n",
      "ForkPoolWorker-9 models/R2_A_D1.json22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247808.262887 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247808.4124002 INFO: Making atp hydrolysis reaction for species: Species8\n",
      "1726247808.4701598 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1726247809.5255363 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species23 23"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247809.5073428 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species8 89 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247811.4993472 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247811.9029357 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247812.581525 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247813.1633031 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247813.672551 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247814.429073 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247814.7588944 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247815.7177465 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247815.8393574 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247817.0099459 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247817.8762043 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247818.26999 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247819.003416 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247820.103372 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247820.6448712 INFO: Making atp hydrolysis reaction for species: Species31\n",
      "1726247821.1684265 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247821.2100575 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n",
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247822.3085706 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac69a690>: 1, <Metabolite cpd11416_c1 at 0x7ff9b15e1290>: -0.030555393141213385, <Metabolite cpd11416_c10 at 0x7ff9b1075350>: -0.02425097014180005, <Metabolite cpd11416_c11 at 0x7ff9aeefc050>: -0.02507783080143885, <Metabolite cpd11416_c12 at 0x7ff9aef80950>: -0.022398020562659347, <Metabolite cpd11416_c13 at 0x7ff9aee09910>: -0.020517711429274058, <Metabolite cpd11416_c14 at 0x7ff9aee7df10>: -0.02804232829384062, <Metabolite cpd11416_c15 at 0x7ff9aeee1550>: -0.03893639574422663, <Metabolite cpd11416_c16 at 0x7ff9aed5dfd0>: -0.023940349411085412, <Metabolite cpd11416_c17 at 0x7ff9aedd27d0>: -0.017172240066672822, <Metabolite cpd11416_c18 at 0x7ff9aec59e90>: -0.02342986660844035, <Metabolite cpd11416_c19 at 0x7ff9aecdc3d0>: -0.016830140003275817, <Metabolite cpd11416_c2 at 0x7ff9b1459810>: -0.05125569526358057, <Metabolite cpd11416_c20 at 0x7ff9aeb64810>: -0.09937027073863476, <Metabolite cpd11416_c21 at 0x7ff9aebd9990>: -0.03275260830831503, <Metabolite cpd11416_c22 at 0x7ff9aea609d0>: -0.053958692460494034, <Metabolite cpd11416_c23 at 0x7ff9ae8e9d90>: -0.08753192894311802, <Metabolite cpd11416_c24 at 0x7ff9ae95f910>: -0.02532755275600765, <Metabolite cpd11416_c25 at 0x7ff9ae7ecd10>: -0.02001140122273962, <Metabolite cpd11416_c26 at 0x7ff9ae8706d0>: -0.021290803698525814, <Metabolite cpd11416_c27 at 0x7ff9ac6f9990>: -0.02103199710432373, <Metabolite cpd11416_c28 at 0x7ff9ac77f0d0>: -0.038875708238769895, <Metabolite cpd11416_c29 at 0x7ff9ac614850>: -0.037810217335742594, <Metabolite cpd11416_c3 at 0x7ff9b14cde10>: -0.027396209708764843, <Metabolite cpd11416_c30 at 0x7ff9ac69a610>: -0.02609546889337511, <Metabolite cpd11416_c4 at 0x7ff9b1356ad0>: -0.018287115493287754, <Metabolite cpd11416_c5 at 0x7ff9b11e81d0>: -0.01598415934626929, <Metabolite cpd11416_c6 at 0x7ff9b126e710>: -0.028530997397831742, <Metabolite cpd11416_c7 at 0x7ff9b12e2e90>: -0.07329957333356177, <Metabolite cpd11416_c8 at 0x7ff9b1169050>: -0.033116257982004675, <Metabolite cpd11416_c9 at 0x7ff9b0ff1c90>: -0.016922095570725805}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac69a690>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247823.2291188 INFO: Making atp hydrolysis reaction for species: Species1\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-10 models/R2_C_D2.json\n",
      "1 missing R2_C_D2\n",
      "ForkPoolWorker-10 models/R2_C_D2.json\n",
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247824.29116 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247824.4678848 INFO: Making atp hydrolysis reaction for species: Species2\n",
      "1726247824.8332176 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247825.390038 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247825.6925824 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247826.4835725 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247826.961121 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247827.5814512 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247828.6820033 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247829.2594066 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247830.5116239 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247830.6527448 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247831.7348382 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247831.778989 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247832.8722482 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247832.9946682 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247834.2192934 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species8 89 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247836.4738014 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247837.702432 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247838.9500108 INFO: Making atp hydrolysis reaction for species: Species12\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-11 models/R2A_A_D1.json12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247840.1867328 INFO: Making atp hydrolysis reaction for species: Species13\n",
      "1726247840.4006417 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247841.4091694 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac2425d0>: 1, <Metabolite cpd11416_c1 at 0x7ff9b10f6fd0>: -0.015063437751931306, <Metabolite cpd11416_c10 at 0x7ff9aed943d0>: -0.015472284020236317, <Metabolite cpd11416_c11 at 0x7ff9aec1d410>: -0.015351150748895646, <Metabolite cpd11416_c12 at 0x7ff9aec91a10>: -0.057949211998001404, <Metabolite cpd11416_c13 at 0x7ff9aeb0a690>: -0.025703921762683234, <Metabolite cpd11416_c14 at 0x7ff9aeb8c750>: -0.024233570032900033, <Metabolite cpd11416_c15 at 0x7ff9aea04fd0>: -0.027255630377544103, <Metabolite cpd11416_c16 at 0x7ff9aea88610>: -0.025754651689439887, <Metabolite cpd11416_c17 at 0x7ff9ae8f24d0>: -0.015230962488891848, <Metabolite cpd11416_c18 at 0x7ff9ae974990>: -0.020710181174136946, <Metabolite cpd11416_c19 at 0x7ff9ae7fcdd0>: -0.09561255291204401, <Metabolite cpd11416_c2 at 0x7ff9b1178090>: -0.0681958840117142, <Metabolite cpd11416_c20 at 0x7ff9ae871f50>: -0.04211271201758899, <Metabolite cpd11416_c21 at 0x7ff9ac6f9010>: -0.05863460294747323, <Metabolite cpd11416_c22 at 0x7ff9ac77e3d0>: -0.06559821966736985, <Metabolite cpd11416_c23 at 0x7ff9ac5f7ed0>: -0.025900956626385217, <Metabolite cpd11416_c24 at 0x7ff9ac689550>: -0.015482765055061519, <Metabolite cpd11416_c25 at 0x7ff9ac512890>: -0.015682892683137918, <Metabolite cpd11416_c26 at 0x7ff9ac595150>: -0.024024765482550323, <Metabolite cpd11416_c27 at 0x7ff9ac41c610>: -0.01455381891108557, <Metabolite cpd11416_c28 at 0x7ff9ac4a1950>: -0.02355621167979012, <Metabolite cpd11416_c29 at 0x7ff9ac32b090>: -0.030096892871452993, <Metabolite cpd11416_c3 at 0x7ff9b0ff0f90>: -0.015282422651680989, <Metabolite cpd11416_c30 at 0x7ff9ac3b8810>: -0.051370644442749275, <Metabolite cpd11416_c31 at 0x7ff9ac242550>: -0.02174423835179016, <Metabolite cpd11416_c4 at 0x7ff9b1075f50>: -0.02527033337938133, <Metabolite cpd11416_c5 at 0x7ff9aef0c850>: -0.022867126595093153, <Metabolite cpd11416_c6 at 0x7ff9aef80fd0>: -0.054916455843228504, <Metabolite cpd11416_c7 at 0x7ff9aedfb250>: -0.03448135887819264, <Metabolite cpd11416_c8 at 0x7ff9aee7e110>: -0.033081109807568225, <Metabolite cpd11416_c9 at 0x7ff9aed05ed0>: -0.018809033140001027}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac2425d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247841.9043944 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247843.2113295 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247843.7185028 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247844.4803717 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247844.9401321 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247845.769225 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247846.1954734 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247847.443779 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247848.1615124 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247848.6681387 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247849.4248002 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247849.8945727 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247850.697387 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247851.9763665 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247852.111389 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247853.233901 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247853.3286505 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247854.5715327 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247855.5940435 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247855.8096745 INFO: Making atp hydrolysis reaction for species: Species24\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-12 models/R1_A_D1.json\n",
      "<class 'optlang.glpk_interface.Model'>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247856.7683554 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac4c6290>: 1, <Metabolite cpd11416_c1 at 0x7ff9b13b4f90>: -0.01735215991796166, <Metabolite cpd11416_c10 at 0x7ff9aee138d0>: -0.04869896633006264, <Metabolite cpd11416_c11 at 0x7ff9aee77e90>: -0.027520764789942392, <Metabolite cpd11416_c12 at 0x7ff9aecfec50>: -0.01808870464267229, <Metabolite cpd11416_c13 at 0x7ff9aed82390>: -0.019688643873466943, <Metabolite cpd11416_c14 at 0x7ff9aec09a50>: -0.025362538158787886, <Metabolite cpd11416_c15 at 0x7ff9aec7ff50>: -0.02448675539920119, <Metabolite cpd11416_c16 at 0x7ff9aeb06490>: -0.018646917861873426, <Metabolite cpd11416_c17 at 0x7ff9aeb8a8d0>: -0.05536347580863413, <Metabolite cpd11416_c18 at 0x7ff9aea0ce10>: -0.040318720831968295, <Metabolite cpd11416_c19 at 0x7ff9aea86010>: -0.05056803480461301, <Metabolite cpd11416_c2 at 0x7ff9b122cd10>: -0.017018641428456077, <Metabolite cpd11416_c20 at 0x7ff9ae909150>: -0.026374740069220068, <Metabolite cpd11416_c21 at 0x7ff9ae98e490>: -0.08215199838107094, <Metabolite cpd11416_c22 at 0x7ff9ae817b50>: -0.023870235789263675, <Metabolite cpd11416_c23 at 0x7ff9ae8a8ed0>: -0.025366247216722673, <Metabolite cpd11416_c24 at 0x7ff9ac732110>: -0.04097440809369844, <Metabolite cpd11416_c25 at 0x7ff9ac7b5b50>: -0.022602183062483152, <Metabolite cpd11416_c26 at 0x7ff9ac63c490>: -0.060194152863036934, <Metabolite cpd11416_c27 at 0x7ff9ac6a3650>: -0.07557761901031668, <Metabolite cpd11416_c28 at 0x7ff9ac538dd0>: -0.044568188507973505, <Metabolite cpd11416_c29 at 0x7ff9ac5be510>: -0.020069564123380238, <Metabolite cpd11416_c3 at 0x7ff9b12a5c10>: -0.02865766522814543, <Metabolite cpd11416_c30 at 0x7ff9ac44c290>: -0.03386355058324213, <Metabolite cpd11416_c31 at 0x7ff9ac4c6210>: -0.02172224615800932, <Metabolite cpd11416_c4 at 0x7ff9b111ff10>: -0.023747020884666833, <Metabolite cpd11416_c5 at 0x7ff9b11a2ed0>: -0.018585792587106475, <Metabolite cpd11416_c6 at 0x7ff9b102ba10>: -0.0242969258140937, <Metabolite cpd11416_c7 at 0x7ff9b10ac450>: -0.018477413914248786, <Metabolite cpd11416_c8 at 0x7ff9aef24e50>: -0.022938742979495007, <Metabolite cpd11416_c9 at 0x7ff9aef9aed0>: -0.02284698088618585}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac4c6290>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247856.822403 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247856.856686 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247857.0355022 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247858.0452893 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247858.173751 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247859.1964567 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247859.261889 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247859.4330578 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247860.418357 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247860.4762676 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247860.7039487 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247861.6423194 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247862.8151073 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247862.915276 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247863.0208077 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247864.0584776 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247864.1327782 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247864.2827308 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247865.2739074 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247865.5694702 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247866.4581597 WARNING: Ignoring reaction 'rxn00062_c7' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247866.5003264 INFO: Making atp hydrolysis reaction for species: Species8\n",
      "1726247866.7891936 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247866.8359501 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247867.7251241 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247868.096356 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247869.9697657 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247870.3625917 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247871.1930196 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247871.6179812 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9b8a5be50>: 1, <Metabolite cpd11416_c1 at 0x7ff9bb45c590>: -0.016920161836424443, <Metabolite cpd11416_c10 at 0x7ff9bb0a5e90>: -0.06940211531249946, <Metabolite cpd11416_c11 at 0x7ff9b8f27750>: -0.06330829407178278, <Metabolite cpd11416_c12 at 0x7ff9b8fa7b90>: -0.097817690314725, <Metabolite cpd11416_c13 at 0x7ff9b8e23390>: -0.03730169507377297, <Metabolite cpd11416_c14 at 0x7ff9b8ea4250>: -0.07716344705468689, <Metabolite cpd11416_c15 at 0x7ff9b8d1dd50>: -0.024880911377561113, <Metabolite cpd11416_c16 at 0x7ff9b8d90950>: -0.13360902600480187, <Metabolite cpd11416_c17 at 0x7ff9b8c1a490>: -0.05446758820022214, <Metabolite cpd11416_c18 at 0x7ff9b8c9fa50>: -0.01499792713403989, <Metabolite cpd11416_c19 at 0x7ff9b8af67d0>: -0.016092364389151636, <Metabolite cpd11416_c2 at 0x7ff9bb4e09d0>: -0.01421441977289274, <Metabolite cpd11416_c20 at 0x7ff9b8b6e7d0>: -0.024475555919268255, <Metabolite cpd11416_c21 at 0x7ff9b89f7c90>: -0.019982023171133838, <Metabolite cpd11416_c22 at 0x7ff9b8a5bdd0>: -0.020221234600764196, <Metabolite cpd11416_c3 at 0x7ff9bb358ed0>: -0.038040013057707024, <Metabolite cpd11416_c4 at 0x7ff9bb3c3910>: -0.01724008464127692, <Metabolite cpd11416_c5 at 0x7ff9bb258b10>: -0.014028940985820873, <Metabolite cpd11416_c6 at 0x7ff9bb2cc110>: -0.0792156293194075, <Metabolite cpd11416_c7 at 0x7ff9bb144210>: -0.0470747409816837, <Metabolite cpd11416_c8 at 0x7ff9bb1b9910>: -0.09792245772657476, <Metabolite cpd11416_c9 at 0x7ff9bb030110>: -0.021623679053802074}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9b8a5be50>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247871.7488585 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247872.4219444 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247872.6408231 INFO: Making atp hydrolysis reaction for species: Species2\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-13 models/R1_A_D2.json\n",
      "1 missing R1_A_D2\n",
      "ForkPoolWorker-13 models/R1_A_D2.json\n",
      "0.9 missing R1_A_D2\n",
      "ForkPoolWorker-13 models/R1_A_D2.json\n",
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247872.8805368 INFO: Making atp hydrolysis reaction for species: Species22\n",
      "1726247873.3071005 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247873.505814 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac1195d0>: 1, <Metabolite cpd11416_c1 at 0x7ff9b11c83d0>: -0.029214169811017817, <Metabolite cpd11416_c10 at 0x7ff9aec335d0>: -0.057648903721611314, <Metabolite cpd11416_c11 at 0x7ff9aecb7a10>: -0.05422548863092152, <Metabolite cpd11416_c12 at 0x7ff9aeb3cc50>: -0.024878597209657283, <Metabolite cpd11416_c13 at 0x7ff9aebc0f90>: -0.039713536717322875, <Metabolite cpd11416_c14 at 0x7ff9aea48050>: -0.019229006756659547, <Metabolite cpd11416_c15 at 0x7ff9aeacd410>: -0.07172425749808502, <Metabolite cpd11416_c16 at 0x7ff9ae957c50>: -0.017902529756248158, <Metabolite cpd11416_c17 at 0x7ff9ae7ed250>: -0.029391237544018654, <Metabolite cpd11416_c18 at 0x7ff9ae878ed0>: -0.01786430762187357, <Metabolite cpd11416_c19 at 0x7ff9ac702290>: -0.02044620224028148, <Metabolite cpd11416_c2 at 0x7ff9b1046250>: -0.02079727571209304, <Metabolite cpd11416_c20 at 0x7ff9ac7874d0>: -0.06025550546555421, <Metabolite cpd11416_c21 at 0x7ff9ac60cd10>: -0.02131939851453135, <Metabolite cpd11416_c22 at 0x7ff9ac682e10>: -0.03461045943427855, <Metabolite cpd11416_c23 at 0x7ff9ac4fe110>: -0.022233350987313397, <Metabolite cpd11416_c24 at 0x7ff9ac583850>: -0.023037810771292987, <Metabolite cpd11416_c25 at 0x7ff9ac414fd0>: -0.024845020859571482, <Metabolite cpd11416_c26 at 0x7ff9ac48a7d0>: -0.10903254498597441, <Metabolite cpd11416_c27 at 0x7ff9ac310910>: -0.02076042619580375, <Metabolite cpd11416_c28 at 0x7ff9ac3947d0>: -0.037649752633103876, <Metabolite cpd11416_c29 at 0x7ff9ac202390>: -0.029110167594556546, <Metabolite cpd11416_c3 at 0x7ff9b10bb4d0>: -0.021970547416986185, <Metabolite cpd11416_c30 at 0x7ff9ac290110>: -0.023284987612649778, <Metabolite cpd11416_c31 at 0x7ff9ac119550>: -0.02800753292805188, <Metabolite cpd11416_c4 at 0x7ff9aef3d910>: -0.02416441346128674, <Metabolite cpd11416_c5 at 0x7ff9aefc08d0>: -0.02852574904824768, <Metabolite cpd11416_c6 at 0x7ff9aee49390>: -0.025728247804444455, <Metabolite cpd11416_c7 at 0x7ff9aeecc2d0>: -0.02164544810278841, <Metabolite cpd11416_c8 at 0x7ff9aed44dd0>: -0.023080573103949863, <Metabolite cpd11416_c9 at 0x7ff9aedbadd0>: -0.017702549859824022}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac1195d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247873.560317 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247873.6543348 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247874.1603787 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247874.4008868 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247874.764234 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247874.8730562 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247875.4163792 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247876.0007672 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247876.116892 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247877.0123532 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247877.0974505 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247877.204598 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247877.7983673 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247877.888426 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247878.3058727 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247878.7733781 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247879.0564723 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247879.5536866 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247879.5869257 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247879.6576128 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247880.3077946 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247880.7722332 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247880.797804 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247881.3462524 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247881.567327 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247881.9899712 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247882.024714 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247882.2274008 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247882.835417 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247883.1289968 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247883.2433493 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247884.0047364 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247884.0844872 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247884.309102 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247884.4608417 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247884.875075 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247885.5402029 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247885.7636585 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247886.3305569 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247886.684574 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247886.7857022 INFO: Making atp hydrolysis reaction for species: Species22\n",
      "1726247887.465511 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Species31 31"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247887.9281774 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247888.0292754 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "15 Species1510 Species1022 Species2216 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247888.3674138 INFO: Making atp hydrolysis reaction for species: Species17\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-14 models/R1_B_D1.json\n",
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247889.1573377 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247889.2375062 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247889.2774527 INFO: Making atp hydrolysis reaction for species: Species24\n",
      "1726247890.123856 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247890.26471 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247890.399204 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247891.262181 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247891.5064814 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247891.6303422 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247892.1913607 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247892.7358756 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247893.9482603 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247893.9845598 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247894.9711437 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247895.1805284 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247895.2713332 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n",
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247896.4086113 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247896.5015192 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247897.6677825 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247898.7472827 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ffa21f0b310>: 1, <Metabolite cpd11416_c1 at 0x7ffa22c0ba10>: -0.019751108060265848, <Metabolite cpd11416_c10 at 0x7ffa22880450>: -0.0339275251457229, <Metabolite cpd11416_c11 at 0x7ffa22708d10>: -0.016447509739534696, <Metabolite cpd11416_c12 at 0x7ffa2278dd50>: -0.0219134692614876, <Metabolite cpd11416_c13 at 0x7ffa22606350>: -0.08911515127315858, <Metabolite cpd11416_c14 at 0x7ffa2267add0>: -0.047144857116562355, <Metabolite cpd11416_c15 at 0x7ffa22500e90>: -0.028758419749027573, <Metabolite cpd11416_c16 at 0x7ffa22575710>: -0.06701785075654187, <Metabolite cpd11416_c17 at 0x7ffa225db6d0>: -0.02160304119681276, <Metabolite cpd11416_c18 at 0x7ffa22461b90>: -0.04084734220633616, <Metabolite cpd11416_c19 at 0x7ffa222e9fd0>: -0.09417645813379602, <Metabolite cpd11416_c2 at 0x7ffa22c84f10>: -0.02766186934416273, <Metabolite cpd11416_c20 at 0x7ffa22363150>: -0.03355678401003013, <Metabolite cpd11416_c21 at 0x7ffa221e6210>: -0.0726594785116376, <Metabolite cpd11416_c22 at 0x7ffa2226b5d0>: -0.0409788834749136, <Metabolite cpd11416_c23 at 0x7ffa220f1190>: -0.028346583628364473, <Metabolite cpd11416_c24 at 0x7ffa221764d0>: -0.03566480280774384, <Metabolite cpd11416_c25 at 0x7ffa21fffc10>: -0.02022232433514859, <Metabolite cpd11416_c26 at 0x7ffa22091390>: -0.026874207691255172, <Metabolite cpd11416_c27 at 0x7ffa21f0b290>: -0.01955953359570993, <Metabolite cpd11416_c3 at 0x7ffa22afcf10>: -0.01493016721274165, <Metabolite cpd11416_c4 at 0x7ffa22b71610>: -0.02240236430970038, <Metabolite cpd11416_c5 at 0x7ffa229f9d50>: -0.031545042225913415, <Metabolite cpd11416_c6 at 0x7ffa22a6e5d0>: -0.06663987951566139, <Metabolite cpd11416_c7 at 0x7ffa228f4a10>: -0.028003969934144054, <Metabolite cpd11416_c8 at 0x7ffa22979850>: -0.015692033716172716, <Metabolite cpd11416_c9 at 0x7ffa227fc590>: -0.034559343047453984}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ffa21f0b310>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247898.8876934 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247898.8994749 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247899.9856932 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247900.0285943 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247900.127361 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247901.1596444 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n",
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247902.2803316 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247902.4271867 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247903.6550496 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247904.399157 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247904.8804603 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247905.5398908 INFO: Making atp hydrolysis reaction for species: Species6\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-15 models/R1_B_D2.json\n",
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247906.1127074 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247906.6770558 INFO: Making atp hydrolysis reaction for species: Species7\n",
      "1726247906.805331 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247907.3619537 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247907.8334203 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247908.9579222 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247909.6126807 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247910.8482969 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247911.0999084 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247912.1110892 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247912.231911 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247913.344773 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247913.368462 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247914.5077412 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247914.5867376 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ffa22088a90>: 1, <Metabolite cpd11416_c1 at 0x7ff9ac38ce50>: -0.018960050813967802, <Metabolite cpd11416_c10 at 0x7ffa227eb050>: -0.02408109032663823, <Metabolite cpd11416_c11 at 0x7ffa2287c050>: -0.017934150300304725, <Metabolite cpd11416_c12 at 0x7ffa226f4650>: -0.08968611862887833, <Metabolite cpd11416_c13 at 0x7ffa22769150>: -0.07467038625385981, <Metabolite cpd11416_c14 at 0x7ffa227df0d0>: -0.03292917046019425, <Metabolite cpd11416_c15 at 0x7ffa2265b9d0>: -0.028016188489175364, <Metabolite cpd11416_c16 at 0x7ffa226db010>: -0.01898686533764011, <Metabolite cpd11416_c17 at 0x7ffa22554f90>: -0.020564557079301462, <Metabolite cpd11416_c18 at 0x7ffa225cb410>: -0.03494930184245175, <Metabolite cpd11416_c19 at 0x7ffa22453850>: -0.0858026451057706, <Metabolite cpd11416_c2 at 0x7ffa22c10090>: -0.020747804505196157, <Metabolite cpd11416_c20 at 0x7ffa224d4a10>: -0.028914252320213548, <Metabolite cpd11416_c21 at 0x7ffa2234fa90>: -0.07933798415359648, <Metabolite cpd11416_c22 at 0x7ffa221e4e10>: -0.034278716038648675, <Metabolite cpd11416_c23 at 0x7ffa2225a990>: -0.022127015320803255, <Metabolite cpd11416_c24 at 0x7ffa222dfd50>: -0.045094590236662756, <Metabolite cpd11416_c25 at 0x7ffa2216d4d0>: -0.023817265702531478, <Metabolite cpd11416_c26 at 0x7ffa21ff6c10>: -0.027718778896880913, <Metabolite cpd11416_c27 at 0x7ffa22088a10>: -0.01675150508763258, <Metabolite cpd11416_c3 at 0x7ffa22c86190>: -0.01956609468288482, <Metabolite cpd11416_c4 at 0x7ff89f53aed0>: -0.061632469982877934, <Metabolite cpd11416_c5 at 0x7ffa22b7a2d0>: -0.029564660417665086, <Metabolite cpd11416_c6 at 0x7ffa229f2a50>: -0.07934270564779794, <Metabolite cpd11416_c7 at 0x7ffa22a74e10>: -0.021098531348048802, <Metabolite cpd11416_c8 at 0x7ffa228efc10>: -0.027123336118717888, <Metabolite cpd11416_c9 at 0x7ffa22963290>: -0.016303764901659308}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ffa22088a90>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247915.2464793 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247915.639637 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247916.364547 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247916.9083657 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247917.471532 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247917.6921923 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n",
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247918.6063411 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247918.8251016 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247919.9595833 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247920.7149737 INFO: Making atp hydrolysis reaction for species: Species5\n",
      "1726247921.0585663 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247921.1034708 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247921.8267531 INFO: Making atp hydrolysis reaction for species: Species6\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-16 models/R1_C_D2.json\n",
      "1 missing R1_C_D2\n",
      "ForkPoolWorker-16 models/R1_C_D2.json\n",
      "0.9 missing R1_C_D2\n",
      "ForkPoolWorker-16 models/R1_C_D2.json\n",
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247922.2376137 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247922.9373093 INFO: Making atp hydrolysis reaction for species: Species7\n",
      "1726247923.0898721 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247924.0514107 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247924.359862 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247925.1613061 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247925.4862523 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247926.6160817 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247927.2002888 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247927.7765167 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247928.3096354 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247928.908379 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247929.4298213 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247930.040741 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247930.5439978 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247931.653808 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247932.0972624 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247933.2317865 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247933.7358744 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n",
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247934.8428857 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247935.9845629 INFO: Making atp hydrolysis reaction for species: Species17\n",
      "1726247937.053952 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n",
      "1726247937.0986497 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n",
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac69b990>: 1, <Metabolite cpd11416_c1 at 0x7ff9b15e2590>: -0.030555393141213385, <Metabolite cpd11416_c10 at 0x7ff9b107a650>: -0.02425097014180005, <Metabolite cpd11416_c11 at 0x7ff9aeefd310>: -0.02507783080143885, <Metabolite cpd11416_c12 at 0x7ff9aef81c50>: -0.022398020562659347, <Metabolite cpd11416_c13 at 0x7ff9aee06c10>: -0.020517711429274058, <Metabolite cpd11416_c14 at 0x7ff9aee83210>: -0.02804232829384062, <Metabolite cpd11416_c15 at 0x7ff9aecea850>: -0.03893639574422663, <Metabolite cpd11416_c16 at 0x7ff9aed5f2d0>: -0.023940349411085412, <Metabolite cpd11416_c17 at 0x7ff9aedd3ad0>: -0.017172240066672822, <Metabolite cpd11416_c18 at 0x7ff9aec5b190>: -0.02342986660844035, <Metabolite cpd11416_c19 at 0x7ff9aecdd6d0>: -0.016830140003275817, <Metabolite cpd11416_c2 at 0x7ff9b145ab10>: -0.05125569526358057, <Metabolite cpd11416_c20 at 0x7ff9aeb61b10>: -0.09937027073863476, <Metabolite cpd11416_c21 at 0x7ff9aebdac90>: -0.03275260830831503, <Metabolite cpd11416_c22 at 0x7ff9aea5dcd0>: -0.053958692460494034, <Metabolite cpd11416_c23 at 0x7ff9aeae3090>: -0.08753192894311802, <Metabolite cpd11416_c24 at 0x7ff9ae968c50>: -0.02532755275600765, <Metabolite cpd11416_c25 at 0x7ff9ae7f2010>: -0.02001140122273962, <Metabolite cpd11416_c26 at 0x7ff9ae8759d0>: -0.021290803698525814, <Metabolite cpd11416_c27 at 0x7ff9ac6fec90>: -0.02103199710432373, <Metabolite cpd11416_c28 at 0x7ff9ac790410>: -0.038875708238769895, <Metabolite cpd11416_c29 at 0x7ff9ac615b50>: -0.037810217335742594, <Metabolite cpd11416_c3 at 0x7ff9b14cf110>: -0.027396209708764843, <Metabolite cpd11416_c30 at 0x7ff9ac69b910>: -0.02609546889337511, <Metabolite cpd11416_c4 at 0x7ff9b1357dd0>: -0.018287115493287754, <Metabolite cpd11416_c5 at 0x7ff9b11ed4d0>: -0.01598415934626929, <Metabolite cpd11416_c6 at 0x7ff9b1273a10>: -0.028530997397831742, <Metabolite cpd11416_c7 at 0x7ff9b10f41d0>: -0.07329957333356177, <Metabolite cpd11416_c8 at 0x7ff9b116e350>: -0.033116257982004675, <Metabolite cpd11416_c9 at 0x7ff9b0ff6f90>: -0.016922095570725805}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac69b990>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247937.3390658 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247938.2723594 INFO: Making atp hydrolysis reaction for species: Species19\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-17 models/R2_A_D1.json\n",
      "1 missing R2_A_D1\n",
      "ForkPoolWorker-17 models/R2_A_D1.json\n",
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247938.519395 INFO: Making atp hydrolysis reaction for species: Species2\n",
      "1726247939.3866937 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247939.4329295 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247939.728159 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247940.9165184 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247941.623828 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247942.774599 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247943.205036 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247943.925152 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247944.3916793 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Species5\n",
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247945.0695553 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247945.6097534 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247946.1946213 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247946.823488 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247948.032109 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247948.3966722 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247949.5189788 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247950.241467 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n",
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247951.4551601 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247952.6927366 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247953.8996534 INFO: Making atp hydrolysis reaction for species: Species13\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-18 models/R2_C_D1.json\n",
      "1 missing R2_C_D1\n",
      "ForkPoolWorker-18 models/R2_C_D1.json\n",
      "0.9 missing R2_C_D1\n",
      "ForkPoolWorker-18 models/R2_C_D1.json13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247955.1077595 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac23fa50>: 1, <Metabolite cpd11416_c1 at 0x7ff9b1100490>: -0.015063437751931306, <Metabolite cpd11416_c10 at 0x7ff9aed91850>: -0.015472284020236317, <Metabolite cpd11416_c11 at 0x7ff9aec1a890>: -0.015351150748895646, <Metabolite cpd11416_c12 at 0x7ff9aec8ee90>: -0.057949211998001404, <Metabolite cpd11416_c13 at 0x7ff9aeb07b10>: -0.025703921762683234, <Metabolite cpd11416_c14 at 0x7ff9aeb89bd0>: -0.024233570032900033, <Metabolite cpd11416_c15 at 0x7ff9aea02450>: -0.027255630377544103, <Metabolite cpd11416_c16 at 0x7ff9aea85a90>: -0.025754651689439887, <Metabolite cpd11416_c17 at 0x7ff9ae8f3950>: -0.015230962488891848, <Metabolite cpd11416_c18 at 0x7ff9ae975e10>: -0.020710181174136946, <Metabolite cpd11416_c19 at 0x7ff9ae7fe250>: -0.09561255291204401, <Metabolite cpd11416_c2 at 0x7ff9b1175510>: -0.0681958840117142, <Metabolite cpd11416_c20 at 0x7ff9ae8733d0>: -0.04211271201758899, <Metabolite cpd11416_c21 at 0x7ff9ac6fa490>: -0.05863460294747323, <Metabolite cpd11416_c22 at 0x7ff9ac77f850>: -0.06559821966736985, <Metabolite cpd11416_c23 at 0x7ff9ac605390>: -0.025900956626385217, <Metabolite cpd11416_c24 at 0x7ff9ac68a9d0>: -0.015482765055061519, <Metabolite cpd11416_c25 at 0x7ff9ac513d10>: -0.015682892683137918, <Metabolite cpd11416_c26 at 0x7ff9ac5925d0>: -0.024024765482550323, <Metabolite cpd11416_c27 at 0x7ff9ac415a90>: -0.01455381891108557, <Metabolite cpd11416_c28 at 0x7ff9ac49add0>: -0.02355621167979012, <Metabolite cpd11416_c29 at 0x7ff9ac330550>: -0.030096892871452993, <Metabolite cpd11416_c3 at 0x7ff9b0ff2410>: -0.015282422651680989, <Metabolite cpd11416_c30 at 0x7ff9ac3b5c90>: -0.051370644442749275, <Metabolite cpd11416_c31 at 0x7ff9ac23f9d0>: -0.02174423835179016, <Metabolite cpd11416_c4 at 0x7ff9b10733d0>: -0.02527033337938133, <Metabolite cpd11416_c5 at 0x7ff9aef05cd0>: -0.022867126595093153, <Metabolite cpd11416_c6 at 0x7ff9aef7e450>: -0.054916455843228504, <Metabolite cpd11416_c7 at 0x7ff9aee00710>: -0.03448135887819264, <Metabolite cpd11416_c8 at 0x7ff9aee77590>: -0.033081109807568225, <Metabolite cpd11416_c9 at 0x7ff9aecff350>: -0.018809033140001027}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac23fa50>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247955.4244306 INFO: Making atp hydrolysis reaction for species: Species1\n",
      "1726247955.93426 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247956.696663 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247957.4179718 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247957.9378633 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247958.6309102 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247959.2180066 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247959.8385098 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247961.0527701 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247961.6003695 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247962.277874 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247962.849417 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247963.4926615 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247964.091907 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247965.3454425 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247965.6576416 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247966.6098833 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247966.8811345 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247968.0956588 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247968.977037 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247969.305131 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247970.2597024 INFO: Making atp hydrolysis reaction for species: Species11\n",
      "1726247970.5406425 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species24 24"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-19 models/R2_C_D2.json\n",
      "1 missing R2_C_D2\n",
      "ForkPoolWorker-19 models/R2_C_D2.json\n",
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247971.565509 INFO: Making atp hydrolysis reaction for species: Species12\n",
      "1726247972.2750814 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247972.776312 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247972.8487923 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247973.9821327 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247974.12612 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9a9d99c90>: 1, <Metabolite cpd11416_c1 at 0x7ff9aedba5d0>: -0.016717677232001397, <Metabolite cpd11416_c10 at 0x7ff9ae869f90>: -0.014471649011430405, <Metabolite cpd11416_c11 at 0x7ff9ac6f3050>: -0.02924475467035163, <Metabolite cpd11416_c12 at 0x7ff9ac767750>: -0.07541247182213304, <Metabolite cpd11416_c13 at 0x7ff9ac7cee10>: -0.015541451698603902, <Metabolite cpd11416_c14 at 0x7ff9ac655bd0>: -0.021167382377927188, <Metabolite cpd11416_c15 at 0x7ff9ac6cbcd0>: -0.025738695327665487, <Metabolite cpd11416_c16 at 0x7ff9ac550590>: -0.023048202757754727, <Metabolite cpd11416_c17 at 0x7ff9ac5c7b90>: -0.01808297791242194, <Metabolite cpd11416_c18 at 0x7ff9ac44e050>: -0.025653095735519195, <Metabolite cpd11416_c19 at 0x7ff9ac4d2490>: -0.10127488931479398, <Metabolite cpd11416_c2 at 0x7ff9aec4c450>: -0.016923667268490566, <Metabolite cpd11416_c20 at 0x7ff9ac34b510>: -0.018833512060924894, <Metabolite cpd11416_c21 at 0x7ff9ac3ce5d0>: -0.03953900262763851, <Metabolite cpd11416_c22 at 0x7ff9ac257990>: -0.058825884983493025, <Metabolite cpd11416_c23 at 0x7ff9ac2d5550>: -0.025030704689031548, <Metabolite cpd11416_c24 at 0x7ff9ac16d250>: -0.023190249386338752, <Metabolite cpd11416_c25 at 0x7ff9abff2610>: -0.024699262555669316, <Metabolite cpd11416_c26 at 0x7ff9ac075fd0>: -0.019260613019941298, <Metabolite cpd11416_c27 at 0x7ff9a9ef8810>: -0.024324572125376993, <Metabolite cpd11416_c28 at 0x7ff9a9f6fc10>: -0.020429598468074642, <Metabolite cpd11416_c29 at 0x7ff9a9e01010>: -0.04059913051061003, <Metabolite cpd11416_c3 at 0x7ff9aeccf110>: -0.019537530263396222, <Metabolite cpd11416_c30 at 0x7ff9a9e86750>: -0.036566210803988634, <Metabolite cpd11416_c31 at 0x7ff9a9d0be90>: -0.02758004763207246, <Metabolite cpd11416_c32 at 0x7ff9a9d99c10>: -0.026782805322885066, <Metabolite cpd11416_c4 at 0x7ff9aeb4bb10>: -0.07467840406223898, <Metabolite cpd11416_c5 at 0x7ff9aebcdfd0>: -0.05314401984014028, <Metabolite cpd11416_c6 at 0x7ff9aea52d10>: -0.0163714986419073, <Metabolite cpd11416_c7 at 0x7ff9aead5ed0>: -0.022069830173868942, <Metabolite cpd11416_c8 at 0x7ff9ae95dd10>: -0.02462135154964412, <Metabolite cpd11416_c9 at 0x7ff9ae9e2650>: -0.02063885615366564}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9a9d99c90>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247974.2241366 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247975.2227132 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247975.530093 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247976.4332798 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247976.4702184 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247976.8375084 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247977.6455483 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Species29\n",
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247977.7486289 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247978.1407993 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247979.0293422 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247979.4667914 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247980.2637706 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247980.3115122 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247981.5953426 INFO: Making atp hydrolysis reaction for species: Species19\n",
      "1726247981.9819927 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species5 56 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247983.3031569 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247983.8891482 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247984.6354678 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247985.1696687 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247985.9773521 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247986.458738 INFO: Making atp hydrolysis reaction for species: Species22\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n",
      "1726247987.2984002 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-20 models/R2A_A_D1.json\n",
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247987.59296 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247987.7396634 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1 growth: 0.0\n",
      "R2A_A_D1carbon uptake: 0.0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247989.019529 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species23 2310 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247989.708542 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac4c38d0>: 1, <Metabolite cpd11416_c1 at 0x7ff9b13b65d0>: -0.01735215991796166, <Metabolite cpd11416_c10 at 0x7ff9aee18f50>: -0.04869896633006264, <Metabolite cpd11416_c11 at 0x7ff9aee85510>: -0.027520764789942392, <Metabolite cpd11416_c12 at 0x7ff9aed042d0>: -0.01808870464267229, <Metabolite cpd11416_c13 at 0x7ff9aed7f9d0>: -0.019688643873466943, <Metabolite cpd11416_c14 at 0x7ff9aec07090>: -0.025362538158787886, <Metabolite cpd11416_c15 at 0x7ff9aec895d0>: -0.02448675539920119, <Metabolite cpd11416_c16 at 0x7ff9aeb03ad0>: -0.018646917861873426, <Metabolite cpd11416_c17 at 0x7ff9aeb87f10>: -0.05536347580863413, <Metabolite cpd11416_c18 at 0x7ff9aea0e450>: -0.040318720831968295, <Metabolite cpd11416_c19 at 0x7ff9aea83650>: -0.05056803480461301, <Metabolite cpd11416_c2 at 0x7ff9b122e350>: -0.017018641428456077, <Metabolite cpd11416_c20 at 0x7ff9ae90a790>: -0.026374740069220068, <Metabolite cpd11416_c21 at 0x7ff9ae98fad0>: -0.08215199838107094, <Metabolite cpd11416_c22 at 0x7ff9ae8211d0>: -0.023870235789263675, <Metabolite cpd11416_c23 at 0x7ff9ae8a6510>: -0.025366247216722673, <Metabolite cpd11416_c24 at 0x7ff9ac72f750>: -0.04097440809369844, <Metabolite cpd11416_c25 at 0x7ff9ac7b3190>: -0.022602183062483152, <Metabolite cpd11416_c26 at 0x7ff9ac635ad0>: -0.060194152863036934, <Metabolite cpd11416_c27 at 0x7ff9ac6accd0>: -0.07557761901031668, <Metabolite cpd11416_c28 at 0x7ff9ac532410>: -0.044568188507973505, <Metabolite cpd11416_c29 at 0x7ff9ac5b7b50>: -0.020069564123380238, <Metabolite cpd11416_c3 at 0x7ff9b12a3250>: -0.02865766522814543, <Metabolite cpd11416_c30 at 0x7ff9ac44d8d0>: -0.03386355058324213, <Metabolite cpd11416_c31 at 0x7ff9ac4c3850>: -0.02172224615800932, <Metabolite cpd11416_c4 at 0x7ff9b1129590>: -0.023747020884666833, <Metabolite cpd11416_c5 at 0x7ff9b11a8550>: -0.018585792587106475, <Metabolite cpd11416_c6 at 0x7ff9b1031090>: -0.0242969258140937, <Metabolite cpd11416_c7 at 0x7ff9b10a5a90>: -0.018477413914248786, <Metabolite cpd11416_c8 at 0x7ff9aef1e490>: -0.022938742979495007, <Metabolite cpd11416_c9 at 0x7ff9aefa0550>: -0.02284698088618585}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac4c38d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247990.9815757 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247991.0403347 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247991.3997712 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247992.318104 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247992.3574512 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247992.676464 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247993.6262598 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247993.688434 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247993.9602659 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247994.9515917 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247995.0114245 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247995.2704425 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247996.5487173 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "415  Species4Species15\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247997.4555914 INFO: Making atp hydrolysis reaction for species: Species5\n",
      "1726247997.456216 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247997.8270197 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species516\n",
      " "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247998.768778 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726247998.7718751 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248000.0961044 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248000.105999 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248000.1498344 INFO: Making atp hydrolysis reaction for species: Species31\n",
      "1726248001.3730903 WARNING: Ignoring reaction 'rxn00062_c7' since it already exists.\n",
      "1726248001.4189427 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n",
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248001.427238 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n",
      "19 8Species19 \n",
      "Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248002.7339149 INFO: Making atp hydrolysis reaction for species: Species20\n",
      "1726248002.7353275 INFO: Making atp hydrolysis reaction for species: Species9\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-21 models/R2A_A_D2.json\n",
      "1 missing R2A_A_D2\n",
      "ForkPoolWorker-21 models/R2A_A_D2.json\n",
      "0.9 missing R2A_A_D2\n",
      "ForkPoolWorker-21 models/R2A_A_D2.json\n",
      "0.8 missing R2A_A_D2\n",
      "ForkPoolWorker-21 models/R2A_A_D2.json"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248004.03126 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248005.1387973 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248005.1982272 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248006.4428046 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248006.5073488 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac0b4bd0>: 1, <Metabolite cpd11416_c1 at 0x7ff9aef367d0>: -0.023627780429884167, <Metabolite cpd11416_c10 at 0x7ff9aebd6a50>: -0.017578609797814088, <Metabolite cpd11416_c11 at 0x7ff9aea5ee90>: -0.11584133341144359, <Metabolite cpd11416_c12 at 0x7ff9aeadc050>: -0.030546322343106717, <Metabolite cpd11416_c13 at 0x7ff9ae964b90>: -0.043956923363500766, <Metabolite cpd11416_c14 at 0x7ff9ae7ecfd0>: -0.022687490154176516, <Metabolite cpd11416_c15 at 0x7ff9ae86c050>: -0.05821868149704713, <Metabolite cpd11416_c16 at 0x7ff9ac6f54d0>: -0.070175959843056, <Metabolite cpd11416_c17 at 0x7ff9ac76b0d0>: -0.027087761868226003, <Metabolite cpd11416_c18 at 0x7ff9ac6004d0>: -0.01524945691745358, <Metabolite cpd11416_c19 at 0x7ff9ac675d50>: -0.022709321320077646, <Metabolite cpd11416_c2 at 0x7ff9aefc6250>: -0.019819275408102337, <Metabolite cpd11416_c20 at 0x7ff9ac4fd790>: -0.021719555680226914, <Metabolite cpd11416_c21 at 0x7ff9ac582ed0>: -0.028724905543068616, <Metabolite cpd11416_c22 at 0x7ff9ac40b290>: -0.024284588495117495, <Metabolite cpd11416_c23 at 0x7ff9ac49c190>: -0.036491891821018266, <Metabolite cpd11416_c24 at 0x7ff9ac315b90>: -0.02082803028693195, <Metabolite cpd11416_c25 at 0x7ff9ac389b90>: -0.01832396972236871, <Metabolite cpd11416_c26 at 0x7ff9ac2132d0>: -0.036973469255808536, <Metabolite cpd11416_c27 at 0x7ff9ac2a4a50>: -0.028842199617851262, <Metabolite cpd11416_c28 at 0x7ff9ac11d550>: -0.02750539594727, <Metabolite cpd11416_c29 at 0x7ff9ac19f390>: -0.019767991545009123, <Metabolite cpd11416_c3 at 0x7ff9aee3ec50>: -0.06434167763984194, <Metabolite cpd11416_c30 at 0x7ff9ac034890>: -0.028113374540842992, <Metabolite cpd11416_c31 at 0x7ff9ac0b4b50>: -0.01714773492288333, <Metabolite cpd11416_c4 at 0x7ff9aeec1510>: -0.03208516118214592, <Metabolite cpd11416_c5 at 0x7ff9aed48f50>: -0.02412847628214966, <Metabolite cpd11416_c6 at 0x7ff9aedcc710>: -0.026192361120037044, <Metabolite cpd11416_c7 at 0x7ff9aec552d0>: -0.03282651657331055, <Metabolite cpd11416_c8 at 0x7ff9aecd6410>: -0.02694598846836467, <Metabolite cpd11416_c9 at 0x7ff9aeb5c550>: -0.017257795001864383}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac0b4bd0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248006.9819403 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248007.7463746 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248007.8309586 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248008.3172233 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248009.0759408 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248009.1584013 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248009.6609771 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248010.4016483 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248010.4589038 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248010.9983428 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248012.3438406 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248012.7901578 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248012.9205585 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248014.111261 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248014.2189522 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248014.8567626 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248015.436423 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248015.5100873 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248016.2208202 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248016.7918084 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248016.8174791 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248017.538023 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248018.1273205 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248018.15486 INFO: Making atp hydrolysis reaction for species: Species19\n",
      "1726248018.8870535 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species8 8<class 'optlang.glpk_interface.Model'>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-22 bio1models/R2A_C_D2.json \n",
      "{<Metabolite cpd11416_c0 at 0x7ff9b8a699d0>: 1, <Metabolite cpd11416_c1 at 0x7ff9bb461e90>: -0.016920161836424443, <Metabolite cpd11416_c10 at 0x7ff9bb0ab790>: -0.06940211531249946, <Metabolite cpd11416_c11 at 0x7ff9b8f352d0>: -0.06330829407178278, <Metabolite cpd11416_c12 at 0x7ff9b8fb9710>: -0.097817690314725, <Metabolite cpd11416_c13 at 0x7ff9b8e30f10>: -0.03730169507377297, <Metabolite cpd11416_c14 at 0x7ff9b8ea5d90>: -0.07716344705468689, <Metabolite cpd11416_c15 at 0x7ff9b8d1f890>: -0.024880911377561113, <Metabolite cpd11416_c16 at 0x7ff9b8d92490>: -0.13360902600480187, <Metabolite cpd11416_c17 at 0x7ff9b8c28050>: -0.05446758820022214, <Metabolite cpd11416_c18 at 0x7ff9b8ca95d0>: -0.01499792713403989, <Metabolite cpd11416_c19 at 0x7ff9b8b04350>: -0.016092364389151636, <Metabolite cpd11416_c2 at 0x7ff9bb4e62d0>: -0.01421441977289274, <Metabolite cpd11416_c20 at 0x7ff9b8b7c350>: -0.024475555919268255, <Metabolite cpd11416_c21 at 0x7ff9b8a05810>: -0.019982023171133838, <Metabolite cpd11416_c22 at 0x7ff9b8a69950>: -0.020221234600764196, <Metabolite cpd11416_c3 at 0x7ff9bb35e7d0>: -0.038040013057707024, <Metabolite cpd11416_c4 at 0x7ff9bb3d1250>: -0.01724008464127692, <Metabolite cpd11416_c5 at 0x7ff9bb25a410>: -0.014028940985820873, <Metabolite cpd11416_c6 at 0x7ff9bb2cda10>: -0.0792156293194075, <Metabolite cpd11416_c7 at 0x7ff9bb145b10>: -0.0470747409816837, <Metabolite cpd11416_c8 at 0x7ff9bb1bb210>: -0.09792245772657476, <Metabolite cpd11416_c9 at 0x7ff9bb035a10>: -0.021623679053802074}1 missing R2A_C_D2\n",
      "\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9b8a699d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248020.1811712 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-22 models/R2A_C_D2.json\n",
      "0.9 missing R2A_C_D2\n",
      "ForkPoolWorker-22 models/R2A_C_D2.json\n",
      "0.8 missing R2A_C_D2\n",
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248020.223569 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-22 models/R2A_C_D2.json19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248020.611364 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248020.6554687 INFO: Making atp hydrolysis reaction for species: Species31\n",
      "1726248020.7124176 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248021.2169075 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248021.5691562 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248021.9247432 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248021.9784393 INFO: Making atp hydrolysis reaction for species: Species32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248022.291837 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248023.218064 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248023.2346478 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32 Species32\n",
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248024.0182235 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248024.539853 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248025.3564584 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248025.844733 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248026.0757673 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248026.6980443 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248026.957806 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac116e50>: 1, <Metabolite cpd11416_c1 at 0x7ff9b11c5c50>: -0.029214169811017817, <Metabolite cpd11416_c10 at 0x7ff9aec40e90>: -0.057648903721611314, <Metabolite cpd11416_c11 at 0x7ff9aecc12d0>: -0.05422548863092152, <Metabolite cpd11416_c12 at 0x7ff9aeb3e4d0>: -0.024878597209657283, <Metabolite cpd11416_c13 at 0x7ff9aebc2810>: -0.039713536717322875, <Metabolite cpd11416_c14 at 0x7ff9aea498d0>: -0.019229006756659547, <Metabolite cpd11416_c15 at 0x7ff9aeacac90>: -0.07172425749808502, <Metabolite cpd11416_c16 at 0x7ff9ae95d510>: -0.017902529756248158, <Metabolite cpd11416_c17 at 0x7ff9ae9e2ad0>: -0.029391237544018654, <Metabolite cpd11416_c18 at 0x7ff9ae87a750>: -0.01786430762187357, <Metabolite cpd11416_c19 at 0x7ff9ac703b10>: -0.02044620224028148, <Metabolite cpd11416_c2 at 0x7ff9b103fad0>: -0.02079727571209304, <Metabolite cpd11416_c20 at 0x7ff9ac790d90>: -0.06025550546555421, <Metabolite cpd11416_c21 at 0x7ff9ac60e590>: -0.02131939851453135, <Metabolite cpd11416_c22 at 0x7ff9ac6906d0>: -0.03461045943427855, <Metabolite cpd11416_c23 at 0x7ff9ac4fb990>: -0.022233350987313397, <Metabolite cpd11416_c24 at 0x7ff9ac58d110>: -0.023037810771292987, <Metabolite cpd11416_c25 at 0x7ff9ac416850>: -0.024845020859571482, <Metabolite cpd11416_c26 at 0x7ff9ac498090>: -0.10903254498597441, <Metabolite cpd11416_c27 at 0x7ff9ac312190>: -0.02076042619580375, <Metabolite cpd11416_c28 at 0x7ff9ac392050>: -0.037649752633103876, <Metabolite cpd11416_c29 at 0x7ff9ac1ffc10>: -0.029110167594556546, <Metabolite cpd11416_c3 at 0x7ff9b10c0d90>: -0.021970547416986185, <Metabolite cpd11416_c30 at 0x7ff9ac28d990>: -0.023284987612649778, <Metabolite cpd11416_c31 at 0x7ff9ac116dd0>: -0.02800753292805188, <Metabolite cpd11416_c4 at 0x7ff9aef3b190>: -0.02416441346128674, <Metabolite cpd11416_c5 at 0x7ff9aefbe150>: -0.02852574904824768, <Metabolite cpd11416_c6 at 0x7ff9aee46c10>: -0.025728247804444455, <Metabolite cpd11416_c7 at 0x7ff9aeec9b50>: -0.02164544810278841, <Metabolite cpd11416_c8 at 0x7ff9aed42650>: -0.023080573103949863, <Metabolite cpd11416_c9 at 0x7ff9aedc4690>: -0.017702549859824022}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac116e50>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248027.4927487 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248027.8536394 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248028.03661 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248028.1852634 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248028.7148626 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248028.8603308 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248029.368581 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248029.570777 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248029.6070216 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248030.2045252 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248031.0288944 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248031.442419 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248031.5250401 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248031.748085 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248032.312673 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248032.3728948 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9bb32b2d0>: 1, <Metabolite cpd11416_c1 at 0x7ff9bdc6e810>: -0.03485692084784196, <Metabolite cpd11416_c10 at 0x7ff9bd8ded10>: -0.09525769145381048, <Metabolite cpd11416_c11 at 0x7ff9bb752c90>: -0.030817666098010114, <Metabolite cpd11416_c12 at 0x7ff9bb7a0050>: -0.019086592390922405, <Metabolite cpd11416_c13 at 0x7ff9bb61a110>: -0.02190844745390366, <Metabolite cpd11416_c14 at 0x7ff9bb69cb50>: -0.020464656703512097, <Metabolite cpd11416_c15 at 0x7ff9bb526190>: -0.06292576125820809, <Metabolite cpd11416_c16 at 0x7ff9bb5ab5d0>: -0.021276743357248332, <Metabolite cpd11416_c17 at 0x7ff9bb42d210>: -0.015247438129513265, <Metabolite cpd11416_c18 at 0x7ff9bb4b0550>: -0.02196139375102277, <Metabolite cpd11416_c19 at 0x7ff9bb32b1d0>: -0.014028943002031516, <Metabolite cpd11416_c2 at 0x7ff9bdafdf90>: -0.018178106961743556, <Metabolite cpd11416_c3 at 0x7ff9bdb827d0>: -0.025382819985511125, <Metabolite cpd11416_c4 at 0x7ff9bda10850>: -0.14781985405809803, <Metabolite cpd11416_c5 at 0x7ff9bda864d0>: -0.08223582353766065, <Metabolite cpd11416_c6 at 0x7ff9bd8f9250>: -0.06512321516206733, <Metabolite cpd11416_c7 at 0x7ff9bd97ed90>: -0.08847946998200662, <Metabolite cpd11416_c8 at 0x7ff9bd810490>: -0.13528217090042355, <Metabolite cpd11416_c9 at 0x7ff9bd85b2d0>: -0.07966628496646433}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9bb32b2d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248032.7682686 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248033.109428 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248033.1714962 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248033.4991126 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248033.7674937 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248034.0259316 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248034.0863671 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248034.219427 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248034.444351 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248034.9019837 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248034.9581947 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248035.4153607 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248035.6835096 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248035.778969 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248035.7861211 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248036.3487954 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248036.4134185 INFO: Making atp hydrolysis reaction for species: Species6\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248036.8061218 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-23 models/R1_A_D1.json\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248037.2922957 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248037.4848642 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248037.7034667 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248037.882961 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248038.1394866 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248038.285744 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248038.3676925 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248038.6076038 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n",
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248039.2439938 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248039.3689046 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248039.470768 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248039.6266713 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248040.0957348 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248040.1072874 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248040.8360355 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248040.9617758 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248040.9717064 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248041.570702 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248041.8293736 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248042.0527043 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248042.2924204 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248043.0332346 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248043.3940074 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248043.587749 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248043.6289356 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248043.7758205 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n",
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248044.5133736 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248044.7257707 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248045.23649 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248045.9617834 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248046.0166605 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248046.054146 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248047.349378 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248047.3780248 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248047.38789 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248048.1167321 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248048.6809466 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n",
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248049.965202 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248050.0250893 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1 growth: 0.0\n",
      "R2A_A_D1carbon uptake: 0.0\n",
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248051.2862444 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248051.3553793 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248052.6118588 INFO: Making atp hydrolysis reaction for species: Species17\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-24 models/R1_A_D2.json\n",
      "1 missing R1_A_D2\n",
      "ForkPoolWorker-24 models/R1_A_D2.json\n",
      "0.9 missing R1_A_D2\n",
      "ForkPoolWorker-24 models/R1_A_D2.json"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248053.7812588 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248053.831572 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248053.9426885 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n",
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248055.2782674 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248056.6153703 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248059.0812404 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248060.4354932 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248061.7537193 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248062.9562087 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9b1299d90>: 1, <Metabolite cpd11416_c1 at 0x7ff9b3f8e4d0>: -0.01758191083480434, <Metabolite cpd11416_c10 at 0x7ff9b39ea0d0>: -0.014037767170268584, <Metabolite cpd11416_c11 at 0x7ff9b3a6ae90>: -0.03743967767525372, <Metabolite cpd11416_c12 at 0x7ff9b38ef850>: -0.042843052538532884, <Metabolite cpd11416_c13 at 0x7ff9b3981650>: -0.02537394808053619, <Metabolite cpd11416_c14 at 0x7ff9b37f57d0>: -0.012310969646181369, <Metabolite cpd11416_c15 at 0x7ff9b387ac90>: -0.04027905970674581, <Metabolite cpd11416_c16 at 0x7ff9b16fe250>: -0.024336331757180103, <Metabolite cpd11416_c17 at 0x7ff9b1773dd0>: -0.03382461269220466, <Metabolite cpd11416_c18 at 0x7ff9b15fb190>: -0.03319117706422763, <Metabolite cpd11416_c19 at 0x7ff9b1688210>: -0.028907772237955993, <Metabolite cpd11416_c2 at 0x7ff9b3e13390>: -0.01701919418206395, <Metabolite cpd11416_c20 at 0x7ff9b1503990>: -0.02539580115442916, <Metabolite cpd11416_c21 at 0x7ff9b1585650>: -0.02549616341971477, <Metabolite cpd11416_c22 at 0x7ff9b141c1d0>: -0.03241559528430899, <Metabolite cpd11416_c23 at 0x7ff9b14a0810>: -0.05679472264453006, <Metabolite cpd11416_c24 at 0x7ff9b1328650>: -0.042890704380216, <Metabolite cpd11416_c25 at 0x7ff9b139e4d0>: -0.016970227109081956, <Metabolite cpd11416_c26 at 0x7ff9b1225190>: -0.030507700527498585, <Metabolite cpd11416_c27 at 0x7ff9b1299d10>: -0.014234242492027094, <Metabolite cpd11416_c3 at 0x7ff9b3e88a10>: -0.14342283684667123, <Metabolite cpd11416_c4 at 0x7ff9b3cffe90>: -0.026906394887270972, <Metabolite cpd11416_c5 at 0x7ff9b3d73d90>: -0.08834206292815247, <Metabolite cpd11416_c6 at 0x7ff9b3bf5fd0>: -0.038890074283252044, <Metabolite cpd11416_c7 at 0x7ff9b3c7b810>: -0.07425259705066527, <Metabolite cpd11416_c8 at 0x7ff9b3b0d190>: -0.033444409675124666, <Metabolite cpd11416_c9 at 0x7ff9b3b71350>: -0.02289099373110169}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9b1299d90>: -1}23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248063.0900998 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248064.0762525 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Species1 1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248064.4174933 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species24 242 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248065.1909206 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248066.9054933 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248067.3911257 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248068.2401967 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248068.5272815 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248069.5718858 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248069.6444974 INFO: Making atp hydrolysis reaction for species: Species6\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-25 models/R1_B_D1.json\n",
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248070.7928817 INFO: Making atp hydrolysis reaction for species: Species7\n",
      "1726248070.8355036 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248070.906011 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248071.9191666 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248072.27263 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248073.0517182 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248074.7897627 INFO: Making atp hydrolysis reaction for species: Species31\n",
      "1726248075.1367073 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species9 931 Species31\n",
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248076.2760494 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248077.3979127 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248078.5355227 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ffa21f11210>: 1, <Metabolite cpd11416_c1 at 0x7ffa22c11650>: -0.019751108060265848, <Metabolite cpd11416_c10 at 0x7ffa2287e310>: -0.0339275251457229, <Metabolite cpd11416_c11 at 0x7ffa22706bd0>: -0.016447509739534696, <Metabolite cpd11416_c12 at 0x7ffa2278bc10>: -0.0219134692614876, <Metabolite cpd11416_c13 at 0x7ffa22610250>: -0.08911515127315858, <Metabolite cpd11416_c14 at 0x7ffa22684cd0>: -0.047144857116562355, <Metabolite cpd11416_c15 at 0x7ffa224fed50>: -0.028758419749027573, <Metabolite cpd11416_c16 at 0x7ffa225735d0>: -0.06701785075654187, <Metabolite cpd11416_c17 at 0x7ffa223e95d0>: -0.02160304119681276, <Metabolite cpd11416_c18 at 0x7ffa22463a50>: -0.04084734220633616, <Metabolite cpd11416_c19 at 0x7ffa222ebe90>: -0.09417645813379602, <Metabolite cpd11416_c2 at 0x7ffa22c7edd0>: -0.02766186934416273, <Metabolite cpd11416_c20 at 0x7ffa22369050>: -0.03355678401003013, <Metabolite cpd11416_c21 at 0x7ffa221f0110>: -0.0726594785116376, <Metabolite cpd11416_c22 at 0x7ffa222754d0>: -0.0409788834749136, <Metabolite cpd11416_c23 at 0x7ffa220ef050>: -0.028346583628364473, <Metabolite cpd11416_c24 at 0x7ffa2217c3d0>: -0.03566480280774384, <Metabolite cpd11416_c25 at 0x7ffa22005b10>: -0.02022232433514859, <Metabolite cpd11416_c26 at 0x7ffa2208b250>: -0.026874207691255172, <Metabolite cpd11416_c27 at 0x7ffa21f11190>: -0.01955953359570993, <Metabolite cpd11416_c3 at 0x7ffa22af6dd0>: -0.01493016721274165, <Metabolite cpd11416_c4 at 0x7ffa22b6b4d0>: -0.02240236430970038, <Metabolite cpd11416_c5 at 0x7ffa229f3c10>: -0.031545042225913415, <Metabolite cpd11416_c6 at 0x7ffa22a744d0>: -0.06663987951566139, <Metabolite cpd11416_c7 at 0x7ffa228ee8d0>: -0.028003969934144054, <Metabolite cpd11416_c8 at 0x7ffa22973710>: -0.015692033716172716, <Metabolite cpd11416_c9 at 0x7ffa227fa450>: -0.034559343047453984}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ffa21f11210>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248079.1560984 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248079.6715717 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248080.2435176 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248081.3328116 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248081.7165782 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248082.8454854 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248083.4004283 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248084.0046945 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248084.4885423 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248085.1267188 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248085.5770235 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248086.2557218 INFO: Making atp hydrolysis reaction for species: Species19\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 \n",
      "Species6\n",
      "ForkPoolWorker-26"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248086.6634037 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " models/R1_B_D2.json\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248087.565696 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248087.7387228 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248088.4380887 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248088.855386 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248089.57389 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248090.7122808 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248090.8765824 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248091.849697 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248091.9623394 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248092.9835706 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248093.0444267 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248094.1403165 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248095.0701249 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248095.2228296 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248096.2129624 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248097.2128258 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248097.3484635 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248098.2881846 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n",
      "<class 'optlang.glpk_interface.Model'>\n",
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248099.3726392 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ffa22086750>: 1, <Metabolite cpd11416_c1 at 0x7ff9ac38e850>: -0.018960050813967802, <Metabolite cpd11416_c10 at 0x7ffa227f4d50>: -0.02408109032663823, <Metabolite cpd11416_c11 at 0x7ffa22879d10>: -0.017934150300304725, <Metabolite cpd11416_c12 at 0x7ffa226f2310>: -0.08968611862887833, <Metabolite cpd11416_c13 at 0x7ffa22766e10>: -0.07467038625385981, <Metabolite cpd11416_c14 at 0x7ffa225ecdd0>: -0.03292917046019425, <Metabolite cpd11416_c15 at 0x7ffa226656d0>: -0.028016188489175364, <Metabolite cpd11416_c16 at 0x7ffa224e4d10>: -0.01898686533764011, <Metabolite cpd11416_c17 at 0x7ffa22552c50>: -0.020564557079301462, <Metabolite cpd11416_c18 at 0x7ffa225d5110>: -0.03494930184245175, <Metabolite cpd11416_c19 at 0x7ffa22459550>: -0.0858026451057706, <Metabolite cpd11416_c2 at 0x7ffa22c11a90>: -0.020747804505196157, <Metabolite cpd11416_c20 at 0x7ffa224d26d0>: -0.028914252320213548, <Metabolite cpd11416_c21 at 0x7ffa22355790>: -0.07933798415359648, <Metabolite cpd11416_c22 at 0x7ffa223daad0>: -0.034278716038648675, <Metabolite cpd11416_c23 at 0x7ffa22260690>: -0.022127015320803255, <Metabolite cpd11416_c24 at 0x7ffa220e9a50>: -0.045094590236662756, <Metabolite cpd11416_c25 at 0x7ffa2216f190>: -0.023817265702531478, <Metabolite cpd11416_c26 at 0x7ffa22000910>: -0.027718778896880913, <Metabolite cpd11416_c27 at 0x7ffa220866d0>: -0.01675150508763258, <Metabolite cpd11416_c3 at 0x7ffa22c87b90>: -0.01956609468288482, <Metabolite cpd11416_c4 at 0x7ff89f53b9d0>: -0.061632469982877934, <Metabolite cpd11416_c5 at 0x7ffa22b77f90>: -0.029564660417665086, <Metabolite cpd11416_c6 at 0x7ffa229fc750>: -0.07934270564779794, <Metabolite cpd11416_c7 at 0x7ffa22a72ad0>: -0.021098531348048802, <Metabolite cpd11416_c8 at 0x7ffa228f9910>: -0.027123336118717888, <Metabolite cpd11416_c9 at 0x7ffa2296cf90>: -0.016303764901659308}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ffa22086750>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248099.5463526 INFO: Making atp hydrolysis reaction for species: Species1\n",
      "1726248100.4129226 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248100.4557874 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248100.6764767 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248101.5479143 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248101.8062992 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248102.9630494 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248103.505788 INFO: Making atp hydrolysis reaction for species: Species20\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-27 models/R1_C_D2.json\n",
      "1 missing R1_C_D2\n",
      "ForkPoolWorker-27 models/R1_C_D2.json\n",
      "0.9 missing R1_C_D2\n",
      "ForkPoolWorker-27 models/R1_C_D2.json\n",
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248104.6258936 INFO: Making atp hydrolysis reaction for species: Species21\n",
      "1726248104.7030509 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248105.231347 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248105.7056582 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248106.3958411 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248106.7884746 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248107.51758 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248107.8863325 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248108.6446908 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248109.7919743 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248109.9273582 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248111.0082583 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248111.964064 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248112.1023293 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248113.0807607 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n",
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248114.2059066 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248115.347606 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248116.4841666 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248118.660243 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248119.7775452 INFO: Making atp hydrolysis reaction for species: Species16\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-28 models/R2_A_D1.json\n",
      "1 missing R2_A_D1\n",
      "ForkPoolWorker-28 models/R2_A_D1.json\n",
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248120.9420953 INFO: Making atp hydrolysis reaction for species: Species17\n",
      "1726248121.0359669 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1726248122.0332997 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n",
      "1726248122.081282 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n",
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248123.1983485 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248124.3197775 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac6a16d0>: 1, <Metabolite cpd11416_c1 at 0x7ff9b13f02d0>: -0.030555393141213385, <Metabolite cpd11416_c10 at 0x7ff9b1080390>: -0.02425097014180005, <Metabolite cpd11416_c11 at 0x7ff9aeefb010>: -0.02507783080143885, <Metabolite cpd11416_c12 at 0x7ff9aef7f950>: -0.022398020562659347, <Metabolite cpd11416_c13 at 0x7ff9aee10950>: -0.020517711429274058, <Metabolite cpd11416_c14 at 0x7ff9aee84f50>: -0.02804232829384062, <Metabolite cpd11416_c15 at 0x7ff9aecf0590>: -0.03893639574422663, <Metabolite cpd11416_c16 at 0x7ff9aed65010>: -0.023940349411085412, <Metabolite cpd11416_c17 at 0x7ff9aedd9810>: -0.017172240066672822, <Metabolite cpd11416_c18 at 0x7ff9aec60ed0>: -0.02342986660844035, <Metabolite cpd11416_c19 at 0x7ff9aecd73d0>: -0.016830140003275817, <Metabolite cpd11416_c2 at 0x7ff9b1464850>: -0.05125569526358057, <Metabolite cpd11416_c20 at 0x7ff9aeb5f810>: -0.09937027073863476, <Metabolite cpd11416_c21 at 0x7ff9aebe09d0>: -0.03275260830831503, <Metabolite cpd11416_c22 at 0x7ff9aea5b9d0>: -0.053958692460494034, <Metabolite cpd11416_c23 at 0x7ff9ae8f0dd0>: -0.08753192894311802, <Metabolite cpd11416_c24 at 0x7ff9ae966950>: -0.02532755275600765, <Metabolite cpd11416_c25 at 0x7ff9ae7efd10>: -0.02001140122273962, <Metabolite cpd11416_c26 at 0x7ff9ae8736d0>: -0.021290803698525814, <Metabolite cpd11416_c27 at 0x7ff9ac7089d0>: -0.02103199710432373, <Metabolite cpd11416_c28 at 0x7ff9ac78a110>: -0.038875708238769895, <Metabolite cpd11416_c29 at 0x7ff9ac613850>: -0.037810217335742594, <Metabolite cpd11416_c3 at 0x7ff9b14d8e50>: -0.027396209708764843, <Metabolite cpd11416_c30 at 0x7ff9ac6a1650>: -0.02609546889337511, <Metabolite cpd11416_c4 at 0x7ff9b1361b10>: -0.018287115493287754, <Metabolite cpd11416_c5 at 0x7ff9b11eb1d0>: -0.01598415934626929, <Metabolite cpd11416_c6 at 0x7ff9b127d750>: -0.028530997397831742, <Metabolite cpd11416_c7 at 0x7ff9b10f5ed0>: -0.07329957333356177, <Metabolite cpd11416_c8 at 0x7ff9b1174090>: -0.033116257982004675, <Metabolite cpd11416_c9 at 0x7ff9b0ffccd0>: -0.016922095570725805}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac6a16d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248126.4312086 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248126.456708 INFO: Making atp hydrolysis reaction for species: Species21\n",
      "1726248127.5834801 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species21 211 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248128.290328 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248128.677629 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248129.8045459 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248129.839536 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248130.9122357 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248131.3837001 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1 growth: 0.0\n",
      "R2A_A_D1carbon uptake: 0.0\n",
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248133.051508 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248134.1744723 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248134.8331559 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n",
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac2497d0>: 1, <Metabolite cpd11416_c1 at 0x7ff9b10fa1d0>: -0.015063437751931306, <Metabolite cpd11416_c10 at 0x7ff9aed8f590>: -0.015472284020236317, <Metabolite cpd11416_c11 at 0x7ff9aec24610>: -0.015351150748895646, <Metabolite cpd11416_c12 at 0x7ff9aec98c10>: -0.057949211998001404, <Metabolite cpd11416_c13 at 0x7ff9aeb11890>: -0.025703921762683234, <Metabolite cpd11416_c14 at 0x7ff9aeb8b910>: -0.024233570032900033, <Metabolite cpd11416_c15 at 0x7ff9aea0c1d0>: -0.027255630377544103, <Metabolite cpd11416_c16 at 0x7ff9aea877d0>: -0.025754651689439887, <Metabolite cpd11416_c17 at 0x7ff9ae8fd6d0>: -0.015230962488891848, <Metabolite cpd11416_c18 at 0x7ff9ae973b50>: -0.020710181174136946, <Metabolite cpd11416_c19 at 0x7ff9ae7fbf90>: -0.09561255291204401, <Metabolite cpd11416_c2 at 0x7ff9b1173250>: -0.0681958840117142, <Metabolite cpd11416_c20 at 0x7ff9ae87d150>: -0.04211271201758899, <Metabolite cpd11416_c21 at 0x7ff9ac704210>: -0.05863460294747323, <Metabolite cpd11416_c22 at 0x7ff9ac7855d0>: -0.06559821966736985, <Metabolite cpd11416_c23 at 0x7ff9ac5ff0d0>: -0.025900956626385217, <Metabolite cpd11416_c24 at 0x7ff9ac690750>: -0.015482765055061519, <Metabolite cpd11416_c25 at 0x7ff9ac519a90>: -0.015682892683137918, <Metabolite cpd11416_c26 at 0x7ff9ac598350>: -0.024024765482550323, <Metabolite cpd11416_c27 at 0x7ff9ac4177d0>: -0.01455381891108557, <Metabolite cpd11416_c28 at 0x7ff9ac4a4b50>: -0.02355621167979012, <Metabolite cpd11416_c29 at 0x7ff9ac32e290>: -0.030096892871452993, <Metabolite cpd11416_c3 at 0x7ff9b0ff8190>: -0.015282422651680989, <Metabolite cpd11416_c30 at 0x7ff9ac3b39d0>: -0.051370644442749275, <Metabolite cpd11416_c31 at 0x7ff9ac249750>: -0.02174423835179016, <Metabolite cpd11416_c4 at 0x7ff9b1079150>: -0.02527033337938133, <Metabolite cpd11416_c5 at 0x7ff9aef03a10>: -0.022867126595093153, <Metabolite cpd11416_c6 at 0x7ff9aef841d0>: -0.054916455843228504, <Metabolite cpd11416_c7 at 0x7ff9aedfe450>: -0.03448135887819264, <Metabolite cpd11416_c8 at 0x7ff9aee81310>: -0.033081109807568225, <Metabolite cpd11416_c9 at 0x7ff9aed090d0>: -0.018809033140001027}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac2497d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248135.955217 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248136.2362041 INFO: Making atp hydrolysis reaction for species: Species6\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-29 models/R2_B_D2.json\n",
      "1 missing R2_B_D2\n",
      "ForkPoolWorker-29 models/R2_B_D2.json\n",
      "0.9 missing R2_B_D2\n",
      "ForkPoolWorker-29 models/R2_B_D2.json\n",
      "0.8 missing R2_B_D2\n",
      "ForkPoolWorker-29 models/R2_B_D2.json\n",
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248137.2256186 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248137.560051 INFO: Making atp hydrolysis reaction for species: Species7\n",
      "1726248138.092375 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248138.5096629 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248138.8881004 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248139.7797709 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248140.2166579 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248142.2298114 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248142.9220355 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248143.5054889 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248144.2464511 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248144.8160121 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248145.5559735 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248146.089976 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248146.8614006 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248147.3789825 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248148.1780872 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248149.5029552 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248149.7166646 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248151.001347 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248151.9422603 INFO: Making atp hydrolysis reaction for species: Species16\n",
      "1726248152.2786517 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species11 1116 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248153.2343667 INFO: Making atp hydrolysis reaction for species: Species17\n",
      "1726248153.5641582 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-30 models/R2_C_D1.json\n",
      "1 missing R2_C_D1\n",
      "ForkPoolWorker-30 models/R2_C_D1.json\n",
      "0.9 missing R2_C_D1\n",
      "ForkPoolWorker-30 models/R2_C_D1.json\n",
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248154.5317345 INFO: Making atp hydrolysis reaction for species: Species18\n",
      "1726248154.6425686 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248154.847643 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248155.8333335 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248157.1152616 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248157.1783726 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248158.4648871 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248159.5285237 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248159.7604194 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9a9d97950>: 1, <Metabolite cpd11416_c1 at 0x7ff9aedc02d0>: -0.016717677232001397, <Metabolite cpd11416_c10 at 0x7ff9ae867c50>: -0.014471649011430405, <Metabolite cpd11416_c11 at 0x7ff9ac6f8d50>: -0.02924475467035163, <Metabolite cpd11416_c12 at 0x7ff9ac76d450>: -0.07541247182213304, <Metabolite cpd11416_c13 at 0x7ff9ac7d8b10>: -0.015541451698603902, <Metabolite cpd11416_c14 at 0x7ff9ac653890>: -0.021167382377927188, <Metabolite cpd11416_c15 at 0x7ff9ac6d19d0>: -0.025738695327665487, <Metabolite cpd11416_c16 at 0x7ff9ac54e250>: -0.023048202757754727, <Metabolite cpd11416_c17 at 0x7ff9ac5cd890>: -0.01808297791242194, <Metabolite cpd11416_c18 at 0x7ff9ac447d10>: -0.025653095735519195, <Metabolite cpd11416_c19 at 0x7ff9ac4d8190>: -0.10127488931479398, <Metabolite cpd11416_c2 at 0x7ff9aec4a110>: -0.016923667268490566, <Metabolite cpd11416_c20 at 0x7ff9ac351210>: -0.018833512060924894, <Metabolite cpd11416_c21 at 0x7ff9ac3d42d0>: -0.03953900262763851, <Metabolite cpd11416_c22 at 0x7ff9ac25d690>: -0.058825884983493025, <Metabolite cpd11416_c23 at 0x7ff9ac2d3210>: -0.025030704689031548, <Metabolite cpd11416_c24 at 0x7ff9ac16af10>: -0.023190249386338752, <Metabolite cpd11416_c25 at 0x7ff9abffc310>: -0.024699262555669316, <Metabolite cpd11416_c26 at 0x7ff9ac077c90>: -0.019260613019941298, <Metabolite cpd11416_c27 at 0x7ff9a9efa4d0>: -0.024324572125376993, <Metabolite cpd11416_c28 at 0x7ff9a9f79910>: -0.020429598468074642, <Metabolite cpd11416_c29 at 0x7ff9a9dfecd0>: -0.04059913051061003, <Metabolite cpd11416_c3 at 0x7ff9aecd8e10>: -0.019537530263396222, <Metabolite cpd11416_c30 at 0x7ff9a9e8c450>: -0.036566210803988634, <Metabolite cpd11416_c31 at 0x7ff9a9d11b90>: -0.02758004763207246, <Metabolite cpd11416_c32 at 0x7ff9a9d978d0>: -0.026782805322885066, <Metabolite cpd11416_c4 at 0x7ff9aeb55810>: -0.07467840406223898, <Metabolite cpd11416_c5 at 0x7ff9aebcbc90>: -0.05314401984014028, <Metabolite cpd11416_c6 at 0x7ff9aea5ca10>: -0.0163714986419073, <Metabolite cpd11416_c7 at 0x7ff9aead7b90>: -0.022069830173868942, <Metabolite cpd11416_c8 at 0x7ff9ae95f9d0>: -0.02462135154964412, <Metabolite cpd11416_c9 at 0x7ff9ae7f0350>: -0.02063885615366564}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9a9d97950>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248160.4696612 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248160.7950673 INFO: Making atp hydrolysis reaction for species: Species22\n",
      "1726248161.0443883 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248161.0941617 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248161.8917105 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248162.0914986 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248162.3908634 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248163.3023446 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248163.386289 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248164.811258 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248164.8617587 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248165.8108988 INFO: Making atp hydrolysis reaction for species: Species25\n",
      "1726248166.1670258 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species20 204 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248166.8035345 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248167.0845013 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248167.4735844 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248168.3694494 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248168.7893767 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248169.638151 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248170.0759861 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248170.2209587 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "<class 'optlang.glpk_interface.Model'>23 Species23\n",
      " {<Metabolite cpd11416_c0 at 0x7ff9ac4cd3d0>: 1, <Metabolite cpd11416_c1 at 0x7ff9b13bc0d0>: -0.01735215991796166, <Metabolite cpd11416_c10 at 0x7ff9aee16a10>: -0.04869896633006264, <Metabolite cpd11416_c11 at 0x7ff9aee7efd0>: -0.027520764789942392, <Metabolite cpd11416_c12 at 0x7ff9aed05d90>: -0.01808870464267229, <Metabolite cpd11416_c13 at 0x7ff9aed894d0>: -0.019688643873466943, <Metabolite cpd11416_c14 at 0x7ff9aec10b90>: -0.025362538158787886, <Metabolite cpd11416_c15 at 0x7ff9aec87090>: -0.02448675539920119, <Metabolite cpd11416_c16 at 0x7ff9aeb0d5d0>: -0.018646917861873426, <Metabolite cpd11416_c17 at 0x7ff9aeb91a10>: -0.05536347580863413, <Metabolite cpd11416_c18 at 0x7ff9aea0bf10>: -0.040318720831968295, <Metabolite cpd11416_c19 at 0x7ff9aea8d150>: -0.05056803480461301, <Metabolite cpd11416_c2 at 0x7ff9b1227e10>: -0.017018641428456077, <Metabolite cpd11416_c20 at 0x7ff9ae914290>: -0.026374740069220068, <Metabolite cpd11416_c21 at 0x7ff9ae9995d0>: -0.08215199838107094, <Metabolite cpd11416_c22 at 0x7ff9ae822c90>: -0.023870235789263675, <Metabolite cpd11416_c23 at 0x7ff9ae8ac050>: -0.025366247216722673, <Metabolite cpd11416_c24 at 0x7ff9ac735250>: -0.04097440809369844, <Metabolite cpd11416_c25 at 0x7ff9ac7b8c90>: -0.022602183062483152, <Metabolite cpd11416_c26 at 0x7ff9ac633590>: -0.060194152863036934, <Metabolite cpd11416_c27 at 0x7ff9ac6aa790>: -0.07557761901031668, <Metabolite cpd11416_c28 at 0x7ff9ac52fed0>: -0.044568188507973505, <Metabolite cpd11416_c29 at 0x7ff9ac5c1650>: -0.020069564123380238, <Metabolite cpd11416_c3 at 0x7ff9b12a8d50>: -0.02865766522814543, <Metabolite cpd11416_c30 at 0x7ff9ac44b390>: -0.03386355058324213, <Metabolite cpd11416_c31 at 0x7ff9ac4cd350>: -0.02172224615800932, <Metabolite cpd11416_c4 at 0x7ff9b1123050>: -0.023747020884666833, <Metabolite cpd11416_c5 at 0x7ff9b11a6010>: -0.018585792587106475, <Metabolite cpd11416_c6 at 0x7ff9b102eb50>: -0.0242969258140937, <Metabolite cpd11416_c7 at 0x7ff9b10a3550>: -0.018477413914248786, <Metabolite cpd11416_c8 at 0x7ff9aef1bf50>: -0.022938742979495007, <Metabolite cpd11416_c9 at 0x7ff9aef9e010>: -0.02284698088618585}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac4cd3d0>: -1}5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248170.447845 INFO: Making atp hydrolysis reaction for species: Species6\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-31 models/R2_C_D2.json\n",
      "1 missing R2_C_D2\n",
      "ForkPoolWorker-31 models/R2_C_D2.json\n",
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248170.9387987 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248171.5395298 INFO: Making atp hydrolysis reaction for species: Species2\n",
      "1726248171.6818974 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248172.2338228 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248172.4495854 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248172.5441866 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248172.8187008 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248173.7336314 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248174.1285617 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n",
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248174.7137675 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248175.0291488 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248176.3130455 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species48\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248176.62847 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248176.636153 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248177.6008623 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248177.9672534 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248178.1793072 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248178.8993578 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248179.3023763 INFO: Making atp hydrolysis reaction for species: Species7\n",
      "1726248180.5894244 WARNING: Ignoring reaction 'rxn00062_c7' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248180.6436217 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248181.2410386 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248181.5772688 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248182.0215168 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n",
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248183.0033023 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248184.4133978 INFO: Making atp hydrolysis reaction for species: Species13\n",
      "1726248184.771625 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species9 913 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248185.8133123 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248186.1308346 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248187.2121625 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248187.4638052 INFO: Making atp hydrolysis reaction for species: Species12\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-32 models/R2A_A_D1.json\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248188.3460226 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1726248188.7929096 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species12 1215 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248189.8399596 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248190.1028593 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248191.2595577 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248192.6352832 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248192.6716208 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9a9b31e50>: 1, <Metabolite cpd11416_c1 at 0x7ff9aeb47e50>: -0.029784588616163357, <Metabolite cpd11416_c10 at 0x7ff9ac7d9a90>: -0.024918752563773197, <Metabolite cpd11416_c11 at 0x7ff9ac661150>: -0.025976526897064593, <Metabolite cpd11416_c12 at 0x7ff9ac6cb190>: -0.021796479196695355, <Metabolite cpd11416_c13 at 0x7ff9ac54f5d0>: -0.06726534945952031, <Metabolite cpd11416_c14 at 0x7ff9ac5d0810>: -0.016995590136303074, <Metabolite cpd11416_c15 at 0x7ff9ac44f910>: -0.02223183713592086, <Metabolite cpd11416_c16 at 0x7ff9ac4d8d10>: -0.09218627927623523, <Metabolite cpd11416_c17 at 0x7ff9ac352890>: -0.017227537061361552, <Metabolite cpd11416_c18 at 0x7ff9ac1e9190>: -0.020217600217910595, <Metabolite cpd11416_c19 at 0x7ff9ac26e850>: -0.025149031805727824, <Metabolite cpd11416_c2 at 0x7ff9aebc8d10>: -0.027878241530080748, <Metabolite cpd11416_c20 at 0x7ff9ac0f7910>: -0.016629857966672584, <Metabolite cpd11416_c21 at 0x7ff9ac18b590>: -0.027914143041348934, <Metabolite cpd11416_c22 at 0x7ff9ac01c990>: -0.039796385157898344, <Metabolite cpd11416_c23 at 0x7ff9ac0a1bd0>: -0.04771806520038735, <Metabolite cpd11416_c24 at 0x7ff9a9f21590>: -0.025542651342820574, <Metabolite cpd11416_c25 at 0x7ff9a9f9be10>: -0.04060812990880808, <Metabolite cpd11416_c26 at 0x7ff9a9e13110>: -0.018079584153952874, <Metabolite cpd11416_c27 at 0x7ff9a9e92550>: -0.019855388712611263, <Metabolite cpd11416_c28 at 0x7ff9a9d17c90>: -0.05732906190277559, <Metabolite cpd11416_c29 at 0x7ff9a9da5410>: -0.027469705494112533, <Metabolite cpd11416_c3 at 0x7ff9aea43690>: -0.04247533276647581, <Metabolite cpd11416_c30 at 0x7ff9a9c1ac10>: -0.025581517624051725, <Metabolite cpd11416_c31 at 0x7ff9a9cac990>: -0.04454270392116313, <Metabolite cpd11416_c32 at 0x7ff9a9b31dd0>: -0.0220028086562943, <Metabolite cpd11416_c4 at 0x7ff9aead4790>: -0.017829153741157455, <Metabolite cpd11416_c5 at 0x7ff9ae94f990>: -0.031071437753612357, <Metabolite cpd11416_c6 at 0x7ff9ae9e0510>: -0.03246288768131126, <Metabolite cpd11416_c7 at 0x7ff9ae869650>: -0.021435101532948312, <Metabolite cpd11416_c8 at 0x7ff9ac6f0490>: -0.025088392995082347, <Metabolite cpd11416_c9 at 0x7ff9ac765210>: -0.02493987654975833}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9a9b31e50>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248192.776509 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248193.9389539 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248194.0010436 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248194.3437426 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248195.2302039 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248195.4074895 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248195.8233187 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248196.518934 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248197.279483 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248197.7991033 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248198.0506918 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248198.7660172 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248199.4243298 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248200.3089895 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248200.8219159 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248201.6021352 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248201.6234877 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248202.2146518 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248202.8931444 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248203.1069667 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248203.5913086 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248204.1722648 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248204.5867174 INFO: Making atp hydrolysis reaction for species: Species8\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-33 models/R2A_A_D2.json\n",
      "1 missing R2A_A_D2\n",
      "ForkPoolWorker-33 models/R2A_A_D2.json\n",
      "0.9 missing R2A_A_D2\n",
      "ForkPoolWorker-33 models/R2A_A_D2.json\n",
      "0.8 missing R2A_A_D2\n",
      "ForkPoolWorker-33 models/R2A_A_D2.json23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248205.448024 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "1726248205.4496598 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248206.0904899 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248206.206622 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac0b6910>: 1, <Metabolite cpd11416_c1 at 0x7ff9aef3c550>: -0.023627780429884167, <Metabolite cpd11416_c10 at 0x7ff9aebdc7d0>: -0.017578609797814088, <Metabolite cpd11416_c11 at 0x7ff9aea60c10>: -0.11584133341144359, <Metabolite cpd11416_c12 at 0x7ff9aead9d90>: -0.030546322343106717, <Metabolite cpd11416_c13 at 0x7ff9ae9628d0>: -0.043956923363500766, <Metabolite cpd11416_c14 at 0x7ff9ae7ead10>: -0.022687490154176516, <Metabolite cpd11416_c15 at 0x7ff9ae869d50>: -0.05821868149704713, <Metabolite cpd11416_c16 at 0x7ff9ac6f3210>: -0.070175959843056, <Metabolite cpd11416_c17 at 0x7ff9ac774e50>: -0.027087761868226003, <Metabolite cpd11416_c18 at 0x7ff9ac5fe210>: -0.01524945691745358, <Metabolite cpd11416_c19 at 0x7ff9ac673a90>: -0.022709321320077646, <Metabolite cpd11416_c2 at 0x7ff9aefc3f90>: -0.019819275408102337, <Metabolite cpd11416_c20 at 0x7ff9ac4fb4d0>: -0.021719555680226914, <Metabolite cpd11416_c21 at 0x7ff9ac58cc50>: -0.028724905543068616, <Metabolite cpd11416_c22 at 0x7ff9ac415010>: -0.024284588495117495, <Metabolite cpd11416_c23 at 0x7ff9ac495ed0>: -0.036491891821018266, <Metabolite cpd11416_c24 at 0x7ff9ac30f8d0>: -0.02082803028693195, <Metabolite cpd11416_c25 at 0x7ff9ac3838d0>: -0.01832396972236871, <Metabolite cpd11416_c26 at 0x7ff9ac219050>: -0.036973469255808536, <Metabolite cpd11416_c27 at 0x7ff9ac29e790>: -0.028842199617851262, <Metabolite cpd11416_c28 at 0x7ff9ac117290>: -0.02750539594727, <Metabolite cpd11416_c29 at 0x7ff9ac1a9110>: -0.019767991545009123, <Metabolite cpd11416_c3 at 0x7ff9aee489d0>: -0.06434167763984194, <Metabolite cpd11416_c30 at 0x7ff9ac0325d0>: -0.028113374540842992, <Metabolite cpd11416_c31 at 0x7ff9ac0b6890>: -0.01714773492288333, <Metabolite cpd11416_c4 at 0x7ff9aeebf250>: -0.03208516118214592, <Metabolite cpd11416_c5 at 0x7ff9aed46c90>: -0.02412847628214966, <Metabolite cpd11416_c6 at 0x7ff9aedca450>: -0.026192361120037044, <Metabolite cpd11416_c7 at 0x7ff9aec53010>: -0.03282651657331055, <Metabolite cpd11416_c8 at 0x7ff9aecdc190>: -0.02694598846836467, <Metabolite cpd11416_c9 at 0x7ff9aeb5e290>: -0.017257795001864383}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac0b6910>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248206.3449333 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248207.5634687 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248207.608199 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248207.8841038 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248207.9339838 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248209.0058327 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248209.1418447 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248209.3783016 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248210.3897254 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248210.4099338 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248210.4700222 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248210.7880042 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248211.769607 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248211.822886 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248211.8659203 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248212.168499 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248213.2173946 INFO: Making atp hydrolysis reaction for species: Species29\n",
      "1726248213.2254798 WARNING: Ignoring reaction 'rxn00062_c12' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248213.2989633 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248214.4169476 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248214.735933 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248215.104205 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248215.7762392 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248215.8088202 INFO: Making atp hydrolysis reaction for species: Species32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248216.1695552 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248216.536287 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248217.0593958 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32 Species32\n",
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248218.047432 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n",
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248218.8608992 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248219.4804204 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9b8a67450>: 1, <Metabolite cpd11416_c1 at 0x7ff9bb45f610>: -0.016920161836424443, <Metabolite cpd11416_c10 at 0x7ff9bb0b0f50>: -0.06940211531249946, <Metabolite cpd11416_c11 at 0x7ff9b8f2ed50>: -0.06330829407178278, <Metabolite cpd11416_c12 at 0x7ff9b8fb3190>: -0.097817690314725, <Metabolite cpd11416_c13 at 0x7ff9b8e2a990>: -0.03730169507377297, <Metabolite cpd11416_c14 at 0x7ff9b8e9f810>: -0.07716344705468689, <Metabolite cpd11416_c15 at 0x7ff9b8d25350>: -0.024880911377561113, <Metabolite cpd11416_c16 at 0x7ff9b8d8bf10>: -0.13360902600480187, <Metabolite cpd11416_c17 at 0x7ff9b8c21a90>: -0.05446758820022214, <Metabolite cpd11416_c18 at 0x7ff9b8ca7050>: -0.01499792713403989, <Metabolite cpd11416_c19 at 0x7ff9b8b01dd0>: -0.016092364389151636, <Metabolite cpd11416_c2 at 0x7ff9bb4dfa50>: -0.01421441977289274, <Metabolite cpd11416_c20 at 0x7ff9b8b75dd0>: -0.024475555919268255, <Metabolite cpd11416_c21 at 0x7ff9b89ff290>: -0.019982023171133838, <Metabolite cpd11416_c22 at 0x7ff9b8a673d0>: -0.020221234600764196, <Metabolite cpd11416_c3 at 0x7ff9bb357f50>: -0.038040013057707024, <Metabolite cpd11416_c4 at 0x7ff9bb3ce9d0>: -0.01724008464127692, <Metabolite cpd11416_c5 at 0x7ff9bb253b90>: -0.014028940985820873, <Metabolite cpd11416_c6 at 0x7ff9bb2c7190>: -0.0792156293194075, <Metabolite cpd11416_c7 at 0x7ff9bb143290>: -0.0470747409816837, <Metabolite cpd11416_c8 at 0x7ff9bb1c09d0>: -0.09792245772657476, <Metabolite cpd11416_c9 at 0x7ff9bb033190>: -0.021623679053802074}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9b8a67450>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248219.7529511 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248220.303639 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248220.6154935 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248220.869773 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "2 Species2bio1\n",
      " "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248221.4519987 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{<Metabolite cpd11416_c0 at 0x7ff9ac120b50>: 1, <Metabolite cpd11416_c1 at 0x7ff9b11c3910>: -0.029214169811017817, <Metabolite cpd11416_c10 at 0x7ff9aec3eb50>: -0.057648903721611314, <Metabolite cpd11416_c11 at 0x7ff9aecbef90>: -0.05422548863092152, <Metabolite cpd11416_c12 at 0x7ff9aeb441d0>: -0.024878597209657283, <Metabolite cpd11416_c13 at 0x7ff9aebc8510>: -0.039713536717322875, <Metabolite cpd11416_c14 at 0x7ff9aea43590>: -0.019229006756659547, <Metabolite cpd11416_c15 at 0x7ff9aead0990>: -0.07172425749808502, <Metabolite cpd11416_c16 at 0x7ff9ae95b1d0>: -0.017902529756248158, <Metabolite cpd11416_c17 at 0x7ff9ae7f07d0>: -0.029391237544018654, <Metabolite cpd11416_c18 at 0x7ff9ae884450>: -0.01786430762187357, <Metabolite cpd11416_c19 at 0x7ff9ac70d810>: -0.02044620224028148, <Metabolite cpd11416_c2 at 0x7ff9b10497d0>: -0.02079727571209304, <Metabolite cpd11416_c20 at 0x7ff9ac78ea50>: -0.06025550546555421, <Metabolite cpd11416_c21 at 0x7ff9ac614290>: -0.02131939851453135, <Metabolite cpd11416_c22 at 0x7ff9ac68a390>: -0.03461045943427855, <Metabolite cpd11416_c23 at 0x7ff9ac501690>: -0.022233350987313397, <Metabolite cpd11416_c24 at 0x7ff9ac586dd0>: -0.023037810771292987, <Metabolite cpd11416_c25 at 0x7ff9ac41c550>: -0.024845020859571482, <Metabolite cpd11416_c26 at 0x7ff9ac491d50>: -0.10903254498597441, <Metabolite cpd11416_c27 at 0x7ff9ac30be50>: -0.02076042619580375, <Metabolite cpd11416_c28 at 0x7ff9ac38fd10>: -0.037649752633103876, <Metabolite cpd11416_c29 at 0x7ff9ac205910>: -0.029110167594556546, <Metabolite cpd11416_c3 at 0x7ff9b10bea50>: -0.021970547416986185, <Metabolite cpd11416_c30 at 0x7ff9ac28b650>: -0.023284987612649778, <Metabolite cpd11416_c31 at 0x7ff9ac120ad0>: -0.02800753292805188, <Metabolite cpd11416_c4 at 0x7ff9aef44e90>: -0.02416441346128674, <Metabolite cpd11416_c5 at 0x7ff9aefbfe10>: -0.02852574904824768, <Metabolite cpd11416_c6 at 0x7ff9aee50910>: -0.025728247804444455, <Metabolite cpd11416_c7 at 0x7ff9aeec7810>: -0.02164544810278841, <Metabolite cpd11416_c8 at 0x7ff9aed4c350>: -0.023080573103949863, <Metabolite cpd11416_c9 at 0x7ff9aedc6350>: -0.017702549859824022}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac120b50>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248221.4606683 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248221.6937623 INFO: Making atp hydrolysis reaction for species: Species18\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248222.2542057 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-34 models/R2A_C_D2.json\n",
      "1 missing R2A_C_D2\n",
      "ForkPoolWorker-34 models/R2A_C_D2.json\n",
      "0.9 missing R2A_C_D2\n",
      "ForkPoolWorker-34 models/R2A_C_D2.json\n",
      "0.8 missing R2A_C_D23\n",
      " Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248222.3041441 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-34 models/R2A_C_D2.json\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248222.775855 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248222.8019698 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248223.0942204 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248223.9729009 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248224.1172376 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248224.5138812 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248224.805388 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248224.8970854 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248225.437058 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248225.6439662 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248226.2394075 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248226.4846978 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248227.1981301 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248227.3537865 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248227.6043477 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248227.9078727 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248228.5889246 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248228.963624 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248228.9919395 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248229.2233613 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D2 growth: 0.0\n",
      "R2A_A_D2carbon uptake: 0.0\n",
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248229.8294868 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248230.0033758 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248230.298629 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248230.5429583 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248230.6644013 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248231.4256258 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248231.512128 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248231.8605697 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248232.3382819 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248232.8232453 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248232.9231877 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248233.171549 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248233.1860754 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248234.2725494 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9bb334890>: 1, <Metabolite cpd11416_c1 at 0x7ff9bdc6ba50>: -0.03485692084784196, <Metabolite cpd11416_c10 at 0x7ff9bd8dbf50>: -0.09525769145381048, <Metabolite cpd11416_c11 at 0x7ff9bb74fed0>: -0.030817666098010114, <Metabolite cpd11416_c12 at 0x7ff9bb79d590>: -0.019086592390922405, <Metabolite cpd11416_c13 at 0x7ff9bb617690>: -0.02190844745390366, <Metabolite cpd11416_c14 at 0x7ff9bb6960d0>: -0.020464656703512097, <Metabolite cpd11416_c15 at 0x7ff9bb51f710>: -0.06292576125820809, <Metabolite cpd11416_c16 at 0x7ff9bb5b0b90>: -0.021276743357248332, <Metabolite cpd11416_c17 at 0x7ff9bb42a790>: -0.015247438129513265, <Metabolite cpd11416_c18 at 0x7ff9bb4adad0>: -0.02196139375102277, <Metabolite cpd11416_c19 at 0x7ff9bb334790>: -0.014028943002031516, <Metabolite cpd11416_c2 at 0x7ff9bdafb1d0>: -0.018178106961743556, <Metabolite cpd11416_c3 at 0x7ff9bdb7fa10>: -0.025382819985511125, <Metabolite cpd11416_c4 at 0x7ff9bda11a90>: -0.14781985405809803, <Metabolite cpd11416_c5 at 0x7ff9bda87710>: -0.08223582353766065, <Metabolite cpd11416_c6 at 0x7ff9bd8fa490>: -0.06512321516206733, <Metabolite cpd11416_c7 at 0x7ff9bd98c050>: -0.08847946998200662, <Metabolite cpd11416_c8 at 0x7ff9bd8116d0>: -0.13528217090042355, <Metabolite cpd11416_c9 at 0x7ff9bd864550>: -0.07966628496646433}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9bb334890>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248234.7383568 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248234.7518322 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248235.515765 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248235.5260394 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248235.577755 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248235.6012888 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248235.6258953 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248236.3040395 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248236.4078486 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248236.924733 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248236.9513357 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248236.9803476 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248237.0766063 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248237.2723653 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248237.8487535 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248238.0976777 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248238.271333 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248238.3496842 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248238.6500783 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248238.925628 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248239.585978 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248239.6353734 INFO: Making atp hydrolysis reaction for species: Species13\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-35 models/R1_A_D1.json\n",
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248239.792246 INFO: Making atp hydrolysis reaction for species: Species29\n",
      "1726248240.2026865 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248240.5629249 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248240.931644 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248240.965357 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248241.0623424 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248241.1943004 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n",
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248241.9805532 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248242.291105 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248242.8309677 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248243.4856648 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248243.6196246 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248243.6513708 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248243.8839035 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248244.4355078 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248244.8040535 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248245.0131776 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248245.2717943 INFO: Making atp hydrolysis reaction for species: Species32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248245.3161383 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248246.1325924 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32 Species32\n",
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248247.4527862 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248247.5891855 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248247.657432 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248248.4663823 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248248.7642252 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248248.913321 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248249.2416592 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248250.0453308 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248250.070696 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248250.2353318 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248250.819568 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248251.5754514 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248252.399916 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248252.4524515 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248252.9158309 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248253.2368898 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248253.7214253 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n",
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248255.0474668 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248255.436896 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species303023 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248256.5613303 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-36 models/R1_A_D2.json\n",
      "1 missing R1_A_D2\n",
      "ForkPoolWorker-36 models/R1_A_D2.json\n",
      "0.9 missing R1_A_D2\n",
      "ForkPoolWorker-36 models/R1_A_D2.json\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248257.5467684 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248258.0039365 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248260.7705452 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248262.149065 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248263.5346687 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248264.9401188 INFO: Making atp hydrolysis reaction for species: Species29\n",
      "1726248265.036484 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9b1297610>: 1, <Metabolite cpd11416_c1 at 0x7ff9b3f8b9d0>: -0.01758191083480434, <Metabolite cpd11416_c10 at 0x7ff9b3be7950>: -0.014037767170268584, <Metabolite cpd11416_c11 at 0x7ff9b3a78750>: -0.03743967767525372, <Metabolite cpd11416_c12 at 0x7ff9b38fd110>: -0.042843052538532884, <Metabolite cpd11416_c13 at 0x7ff9b397eed0>: -0.02537394808053619, <Metabolite cpd11416_c14 at 0x7ff9b37f7050>: -0.012310969646181369, <Metabolite cpd11416_c15 at 0x7ff9b3880550>: -0.04027905970674581, <Metabolite cpd11416_c16 at 0x7ff9b16f7ad0>: -0.024336331757180103, <Metabolite cpd11416_c17 at 0x7ff9b1779690>: -0.03382461269220466, <Metabolite cpd11416_c18 at 0x7ff9b1600a50>: -0.03319117706422763, <Metabolite cpd11416_c19 at 0x7ff9b1685a90>: -0.028907772237955993, <Metabolite cpd11416_c2 at 0x7ff9b3e1c8d0>: -0.01701919418206395, <Metabolite cpd11416_c20 at 0x7ff9b150d250>: -0.02539580115442916, <Metabolite cpd11416_c21 at 0x7ff9b1582ed0>: -0.02549616341971477, <Metabolite cpd11416_c22 at 0x7ff9b1419a50>: -0.03241559528430899, <Metabolite cpd11416_c23 at 0x7ff9b149e090>: -0.05679472264453006, <Metabolite cpd11416_c24 at 0x7ff9b1325ed0>: -0.042890704380216, <Metabolite cpd11416_c25 at 0x7ff9b139bd50>: -0.016970227109081956, <Metabolite cpd11416_c26 at 0x7ff9b1222a10>: -0.030507700527498585, <Metabolite cpd11416_c27 at 0x7ff9b1297590>: -0.014234242492027094, <Metabolite cpd11416_c3 at 0x7ff9b3e86290>: -0.14342283684667123, <Metabolite cpd11416_c4 at 0x7ff9b3d09750>: -0.026906394887270972, <Metabolite cpd11416_c5 at 0x7ff9b3d7d650>: -0.08834206292815247, <Metabolite cpd11416_c6 at 0x7ff9b3bf7850>: -0.038890074283252044, <Metabolite cpd11416_c7 at 0x7ff9b3c850d0>: -0.07425259705066527, <Metabolite cpd11416_c8 at 0x7ff9b3b0aa10>: -0.033444409675124666, <Metabolite cpd11416_c9 at 0x7ff9b3b72bd0>: -0.02289099373110169}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9b1297610>: -1}\n",
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248266.1858578 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248266.2822373 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248267.3207939 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248268.8981576 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248269.4648306 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n",
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248270.5980453 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248271.7274468 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248272.875439 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248274.0186374 INFO: Making atp hydrolysis reaction for species: Species8\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-37 models/R1_B_D1.json\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248275.0210137 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248275.1661704 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248277.280562 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248278.4172702 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248279.5810833 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248280.7202466 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ffa21f0eb10>: 1, <Metabolite cpd11416_c1 at 0x7ffa22c12b90>: -0.019751108060265848, <Metabolite cpd11416_c10 at 0x7ffa2287bc10>: -0.0339275251457229, <Metabolite cpd11416_c11 at 0x7ffa22710510>: -0.016447509739534696, <Metabolite cpd11416_c12 at 0x7ffa22791550>: -0.0219134692614876, <Metabolite cpd11416_c13 at 0x7ffa2260db50>: -0.08911515127315858, <Metabolite cpd11416_c14 at 0x7ffa226825d0>: -0.047144857116562355, <Metabolite cpd11416_c15 at 0x7ffa22504690>: -0.028758419749027573, <Metabolite cpd11416_c16 at 0x7ffa2257cf10>: -0.06701785075654187, <Metabolite cpd11416_c17 at 0x7ffa223eaed0>: -0.02160304119681276, <Metabolite cpd11416_c18 at 0x7ffa2246d390>: -0.04084734220633616, <Metabolite cpd11416_c19 at 0x7ffa222f17d0>: -0.09417645813379602, <Metabolite cpd11416_c2 at 0x7ffa22c88710>: -0.02766186934416273, <Metabolite cpd11416_c20 at 0x7ffa2236a950>: -0.03355678401003013, <Metabolite cpd11416_c21 at 0x7ffa221eda10>: -0.0726594785116376, <Metabolite cpd11416_c22 at 0x7ffa22272dd0>: -0.0409788834749136, <Metabolite cpd11416_c23 at 0x7ffa220f8990>: -0.028346583628364473, <Metabolite cpd11416_c24 at 0x7ffa2217dcd0>: -0.03566480280774384, <Metabolite cpd11416_c25 at 0x7ffa22003410>: -0.02022232433514859, <Metabolite cpd11416_c26 at 0x7ffa22094b90>: -0.026874207691255172, <Metabolite cpd11416_c27 at 0x7ffa21f0ea90>: -0.01955953359570993, <Metabolite cpd11416_c3 at 0x7ffa22b00710>: -0.01493016721274165, <Metabolite cpd11416_c4 at 0x7ffa22b74e10>: -0.02240236430970038, <Metabolite cpd11416_c5 at 0x7ffa229fd550>: -0.031545042225913415, <Metabolite cpd11416_c6 at 0x7ffa22a71dd0>: -0.06663987951566139, <Metabolite cpd11416_c7 at 0x7ffa228f8210>: -0.028003969934144054, <Metabolite cpd11416_c8 at 0x7ffa2297d050>: -0.015692033716172716, <Metabolite cpd11416_c9 at 0x7ffa227f7d50>: -0.034559343047453984}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ffa21f0eb10>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248281.793726 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248281.8658078 INFO: Making atp hydrolysis reaction for species: Species14\n",
      "1726248282.926918 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species1 114 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248283.9594362 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248284.0466995 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248285.0954936 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248286.192284 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248286.245228 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248287.3210833 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248287.391704 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248288.4862678 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248288.5251582 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248289.628064 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248290.5727057 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248290.7549324 INFO: Making atp hydrolysis reaction for species: Species8\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-38 models/R1_B_D2.json20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248291.7375562 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248291.910146 INFO: Making atp hydrolysis reaction for species: Species9\n",
      "1726248292.3975732 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1 growth: 0.0\n",
      "R2A_A_D1carbon uptake: 0.0\n",
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248292.8849328 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248293.987003 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248294.0312316 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248295.1298053 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248295.1699998 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248296.2628036 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248297.316465 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248297.4040444 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248298.4577365 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248298.545034 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "<class 'optlang.glpk_interface.Model'>"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248299.4496229 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248299.6088197 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species26 26{<Metabolite cpd11416_c0 at 0x7ffa22087e50>: -1}bio1 {<Metabolite cpd11416_c0 at 0x7ffa22087e50>: 1, <Metabolite cpd11416_c1 at 0x7ff9ac38bb50>: -0.018960050813967802, <Metabolite cpd11416_c10 at 0x7ffa227f2450>: -0.02408109032663823, <Metabolite cpd11416_c11 at 0x7ffa22873410>: -0.017934150300304725, <Metabolite cpd11416_c12 at 0x7ffa226efa10>: -0.08968611862887833, <Metabolite cpd11416_c13 at 0x7ffa22770550>: -0.07467038625385981, <Metabolite cpd11416_c14 at 0x7ffa225ea4d0>: -0.03292917046019425, <Metabolite cpd11416_c15 at 0x7ffa2265edd0>: -0.028016188489175364, <Metabolite cpd11416_c16 at 0x7ffa224e6410>: -0.01898686533764011, <Metabolite cpd11416_c17 at 0x7ffa22558390>: -0.020564557079301462, <Metabolite cpd11416_c18 at 0x7ffa225ce810>: -0.03494930184245175, <Metabolite cpd11416_c19 at 0x7ffa22456c50>: -0.0858026451057706, <Metabolite cpd11416_c2 at 0x7ffa22c0ed90>: -0.020747804505196157, <Metabolite cpd11416_c20 at 0x7ffa224cbdd0>: -0.028914252320213548, <Metabolite cpd11416_c21 at 0x7ffa22352e90>: -0.07933798415359648, <Metabolite cpd11416_c22 at 0x7ffa221e8210>: -0.034278716038648675, <Metabolite cpd11416_c23 at 0x7ffa2225dd90>: -0.022127015320803255, <Metabolite cpd11416_c24 at 0x7ffa220e7150>: -0.045094590236662756, <Metabolite cpd11416_c25 at 0x7ffa221788d0>: -0.023817265702531478, <Metabolite cpd11416_c26 at 0x7ffa22002010>: -0.027718778896880913, <Metabolite cpd11416_c27 at 0x7ffa22087dd0>: -0.01675150508763258, <Metabolite cpd11416_c3 at 0x7ffa22c90ed0>: -0.01956609468288482, <Metabolite cpd11416_c4 at 0x7ff89f531050>: -0.061632469982877934, <Metabolite cpd11416_c5 at 0x7ffa22b7d6d0>: -0.029564660417665086, <Metabolite cpd11416_c6 at 0x7ffa229f9e50>: -0.07934270564779794, <Metabolite cpd11416_c7 at 0x7ffa22a78210>: -0.021098531348048802, <Metabolite cpd11416_c8 at 0x7ffa228f3010>: -0.027123336118717888, <Metabolite cpd11416_c9 at 0x7ffa2296a690>: -0.016303764901659308}\n",
      "SK_cpd11416_c0 "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248300.6136985 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species1 \n",
      "Species27 27"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248300.6088324 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "Species14 14"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248301.7739704 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248301.749891 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species153 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248302.8799934 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248302.907699 INFO: Making atp hydrolysis reaction for species: Species17\n",
      "1726248304.0104403 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n",
      "1726248304.057473 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n",
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248305.0490937 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248305.1951802 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248306.1896234 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248307.3461838 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248307.3697417 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248308.4917352 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248308.5186403 INFO: Making atp hydrolysis reaction for species: Species21\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-39 models/R1_C_D1.json\n",
      "1 missing R1_C_D1\n",
      "ForkPoolWorker-39 models/R1_C_D1.json\n",
      "0.9 missing R1_C_D1\n",
      "ForkPoolWorker-39 models/R1_C_D1.json\n",
      "0.8 missing R1_C_D1\n",
      "ForkPoolWorker-39 models/R1_C_D1.json\n",
      "0.7 missing R1_C_D1\n",
      "ForkPoolWorker-39 models/R1_C_D1.json\n",
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248309.6524544 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248309.6796658 INFO: Making atp hydrolysis reaction for species: Species22\n",
      "1726248309.7315414 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248310.8322992 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248311.7399728 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248311.9762487 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248312.8865912 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248314.0279055 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248314.0458882 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248315.189005 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248315.2122853 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248316.3377588 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248316.3530204 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n",
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248318.5433466 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248319.680981 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248320.8228755 INFO: Making atp hydrolysis reaction for species: Species17\n",
      "1726248321.9308054 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n",
      "1726248321.9765606 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n",
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248323.1219623 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248324.264808 INFO: Making atp hydrolysis reaction for species: Species20\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "ForkPoolWorker-40 models/R1_C_D2.json\n",
      "1 missing R1_C_D2\n",
      "ForkPoolWorker-40 models/R1_C_D2.json\n",
      "0.9 missing R1_C_D2\n",
      "ForkPoolWorker-40 models/R1_C_D2.json20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248326.3654914 INFO: Making atp hydrolysis reaction for species: Species21\n",
      "1726248327.0408006 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248327.533289 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248328.6823225 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac6a2f90>: 1, <Metabolite cpd11416_c1 at 0x7ff9b13edb90>: -0.030555393141213385, <Metabolite cpd11416_c10 at 0x7ff9b107dc50>: -0.02425097014180005, <Metabolite cpd11416_c11 at 0x7ff9aef04910>: -0.02507783080143885, <Metabolite cpd11416_c12 at 0x7ff9aef89250>: -0.022398020562659347, <Metabolite cpd11416_c13 at 0x7ff9aee12210>: -0.020517711429274058, <Metabolite cpd11416_c14 at 0x7ff9aee86810>: -0.02804232829384062, <Metabolite cpd11416_c15 at 0x7ff9aecede50>: -0.03893639574422663, <Metabolite cpd11416_c16 at 0x7ff9aed628d0>: -0.023940349411085412, <Metabolite cpd11416_c17 at 0x7ff9aeddb0d0>: -0.017172240066672822, <Metabolite cpd11416_c18 at 0x7ff9aec5e790>: -0.02342986660844035, <Metabolite cpd11416_c19 at 0x7ff9aece0cd0>: -0.016830140003275817, <Metabolite cpd11416_c2 at 0x7ff9b1462110>: -0.05125569526358057, <Metabolite cpd11416_c20 at 0x7ff9aeb69110>: -0.09937027073863476, <Metabolite cpd11416_c21 at 0x7ff9aebe2290>: -0.03275260830831503, <Metabolite cpd11416_c22 at 0x7ff9aea652d0>: -0.053958692460494034, <Metabolite cpd11416_c23 at 0x7ff9ae8ee690>: -0.08753192894311802, <Metabolite cpd11416_c24 at 0x7ff9ae970250>: -0.02532755275600765, <Metabolite cpd11416_c25 at 0x7ff9ae7f5610>: -0.02001140122273962, <Metabolite cpd11416_c26 at 0x7ff9ae878fd0>: -0.021290803698525814, <Metabolite cpd11416_c27 at 0x7ff9ac702290>: -0.02103199710432373, <Metabolite cpd11416_c28 at 0x7ff9ac7879d0>: -0.038875708238769895, <Metabolite cpd11416_c29 at 0x7ff9ac61d150>: -0.037810217335742594, <Metabolite cpd11416_c3 at 0x7ff9b14d6710>: -0.027396209708764843, <Metabolite cpd11416_c30 at 0x7ff9ac6a2f10>: -0.02609546889337511, <Metabolite cpd11416_c4 at 0x7ff9b135f3d0>: -0.018287115493287754, <Metabolite cpd11416_c5 at 0x7ff9b11f0ad0>: -0.01598415934626929, <Metabolite cpd11416_c6 at 0x7ff9b1277010>: -0.028530997397831742, <Metabolite cpd11416_c7 at 0x7ff9b10ef790>: -0.07329957333356177, <Metabolite cpd11416_c8 at 0x7ff9b1171950>: -0.033116257982004675, <Metabolite cpd11416_c9 at 0x7ff9b0ffa590>: -0.016922095570725805}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac6a2f90>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248329.4745107 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248329.8325577 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248330.987067 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248331.0482693 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248332.7352161 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248333.1246119 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248334.1285684 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248334.2661343 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n",
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248337.403834 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248338.7031713 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248340.0325673 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248341.3262417 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac242f10>: 1, <Metabolite cpd11416_c1 at 0x7ff9b10fb910>: -0.015063437751931306, <Metabolite cpd11416_c10 at 0x7ff9aed98d10>: -0.015472284020236317, <Metabolite cpd11416_c11 at 0x7ff9aec1dd50>: -0.015351150748895646, <Metabolite cpd11416_c12 at 0x7ff9aec96350>: -0.057949211998001404, <Metabolite cpd11416_c13 at 0x7ff9aeb0efd0>: -0.025703921762683234, <Metabolite cpd11416_c14 at 0x7ff9aeb91090>: -0.024233570032900033, <Metabolite cpd11416_c15 at 0x7ff9aea09910>: -0.027255630377544103, <Metabolite cpd11416_c16 at 0x7ff9aea8cf50>: -0.025754651689439887, <Metabolite cpd11416_c17 at 0x7ff9ae8fae10>: -0.015230962488891848, <Metabolite cpd11416_c18 at 0x7ff9ae9792d0>: -0.020710181174136946, <Metabolite cpd11416_c19 at 0x7ff9ae801710>: -0.09561255291204401, <Metabolite cpd11416_c2 at 0x7ff9b117c9d0>: -0.0681958840117142, <Metabolite cpd11416_c20 at 0x7ff9ae876890>: -0.04211271201758899, <Metabolite cpd11416_c21 at 0x7ff9ac6fd950>: -0.05863460294747323, <Metabolite cpd11416_c22 at 0x7ff9ac782d10>: -0.06559821966736985, <Metabolite cpd11416_c23 at 0x7ff9ac608850>: -0.025900956626385217, <Metabolite cpd11416_c24 at 0x7ff9ac68de90>: -0.015482765055061519, <Metabolite cpd11416_c25 at 0x7ff9ac5171d0>: -0.015682892683137918, <Metabolite cpd11416_c26 at 0x7ff9ac599a90>: -0.024024765482550323, <Metabolite cpd11416_c27 at 0x7ff9ac420f50>: -0.01455381891108557, <Metabolite cpd11416_c28 at 0x7ff9ac4a6290>: -0.02355621167979012, <Metabolite cpd11416_c29 at 0x7ff9ac32f9d0>: -0.030096892871452993, <Metabolite cpd11416_c3 at 0x7ff9b0ff58d0>: -0.015282422651680989, <Metabolite cpd11416_c30 at 0x7ff9ac3b9150>: -0.051370644442749275, <Metabolite cpd11416_c31 at 0x7ff9ac242e90>: -0.02174423835179016, <Metabolite cpd11416_c4 at 0x7ff9b107a890>: -0.02527033337938133, <Metabolite cpd11416_c5 at 0x7ff9aef11190>: -0.022867126595093153, <Metabolite cpd11416_c6 at 0x7ff9aef85910>: -0.054916455843228504, <Metabolite cpd11416_c7 at 0x7ff9aedffb90>: -0.03448135887819264, <Metabolite cpd11416_c8 at 0x7ff9aee82a50>: -0.033081109807568225, <Metabolite cpd11416_c9 at 0x7ff9aed0a810>: -0.018809033140001027}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac242f10>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248342.0804467 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248342.653561 INFO: Making atp hydrolysis reaction for species: Species9\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-41 models/R2_A_D1.json\n",
      "1 missing R2_A_D1\n",
      "ForkPoolWorker-41 models/R2_A_D1.json\n",
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248343.4046297 INFO: Making atp hydrolysis reaction for species: Species2\n",
      "1726248344.1925068 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248344.7361054 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248345.2480721 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248346.071385 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248346.545473 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248347.826139 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248348.5203252 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248349.1074347 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248349.8834534 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248350.4243155 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248351.2166073 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248351.7034817 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248352.5795374 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248353.9138393 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248354.1256022 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248355.425958 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248356.4386914 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248356.7205172 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248357.7736115 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248358.0017228 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9aec6c6d0>: 1, <Metabolite cpd11416_c1 at 0x7ff9b3a83010>: -0.021782143531949547, <Metabolite cpd11416_c10 at 0x7ff9b151bc90>: -0.07455270637011172, <Metabolite cpd11416_c11 at 0x7ff9b159dd50>: -0.03212236634147578, <Metabolite cpd11416_c12 at 0x7ff9b1416650>: -0.025143792047339197, <Metabolite cpd11416_c13 at 0x7ff9b1499c90>: -0.018241800932210013, <Metabolite cpd11416_c14 at 0x7ff9b1320150>: -0.03343751949798929, <Metabolite cpd11416_c15 at 0x7ff9b13a4590>: -0.13288804197790222, <Metabolite cpd11416_c16 at 0x7ff9b121d710>: -0.030324494742815945, <Metabolite cpd11416_c17 at 0x7ff9b12a0850>: -0.06979368640939841, <Metabolite cpd11416_c18 at 0x7ff9b1129c10>: -0.057603450262314525, <Metabolite cpd11416_c19 at 0x7ff9b119f790>: -0.029046275318022126, <Metabolite cpd11416_c2 at 0x7ff9b3914e90>: -0.031464008994993076, <Metabolite cpd11416_c20 at 0x7ff9b1034b90>: -0.01501974019699651, <Metabolite cpd11416_c21 at 0x7ff9b10b85d0>: -0.01700594667027133, <Metabolite cpd11416_c22 at 0x7ff9aef41a10>: -0.043968317647473366, <Metabolite cpd11416_c23 at 0x7ff9aefc3150>: -0.02411779470588885, <Metabolite cpd11416_c24 at 0x7ff9aee548d0>: -0.03019149257289139, <Metabolite cpd11416_c25 at 0x7ff9aeeda690>: -0.017400879606693558, <Metabolite cpd11416_c26 at 0x7ff9aed622d0>: -0.01781169947918669, <Metabolite cpd11416_c27 at 0x7ff9aebe8390>: -0.016585146543067024, <Metabolite cpd11416_c28 at 0x7ff9aec6c650>: -0.01867305656434946, <Metabolite cpd11416_c3 at 0x7ff9b399bb50>: -0.02268050224172141, <Metabolite cpd11416_c4 at 0x7ff9b381c590>: -0.06602882202510517, <Metabolite cpd11416_c5 at 0x7ff9b3892a10>: -0.04048283250219998, <Metabolite cpd11416_c6 at 0x7ff9b1715d50>: -0.02601920296056429, <Metabolite cpd11416_c7 at 0x7ff9b1795b90>: -0.03119325215290066, <Metabolite cpd11416_c8 at 0x7ff9b161e4d0>: -0.03569448281335653, <Metabolite cpd11416_c9 at 0x7ff9b16a3590>: -0.02072654489081206}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9aec6c6d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248358.3414211 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248359.115353 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248359.2766266 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248360.4615376 INFO: Making atp hydrolysis reaction for species: Species13\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-42 models/R2_B_D2.json\n",
      "1 missing R2_B_D2\n",
      "ForkPoolWorker-42 models/R2_B_D2.json\n",
      "0.9 missing R2_B_D2\n",
      "ForkPoolWorker-42 models/R2_B_D2.json\n",
      "0.8 missing R2_B_D2\n",
      "ForkPoolWorker-42 models/R2_B_D2.json\n",
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248361.7315948 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248361.802191 INFO: Making atp hydrolysis reaction for species: Species14\n",
      "1726248362.0312505 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248362.0854561 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248363.0242846 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248363.6308722 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248364.2217693 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248364.3076434 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248364.9868565 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248365.5826473 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248365.6399002 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248366.401577 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248367.0503597 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248367.7467048 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248368.0307717 INFO: Making atp hydrolysis reaction for species: Species25\n",
      "1726248368.4755733 WARNING: Ignoring reaction 'rxn00062_c17' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248368.5621746 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248369.3336096 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248369.9960854 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248370.6362195 INFO: Making atp hydrolysis reaction for species: Species27\n",
      "1726248370.9698093 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "155.36637899228947 6 Species6R2A_A_D1 methane flux:27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248371.9037962 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248372.197102 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248372.8418736 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248373.1842163 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248373.4305685 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248374.1932042 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248374.4596262 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248374.6569285 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-1 models/R2A_A_D1.json\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248375.4032679 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248375.5818276 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248375.8681242 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n",
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248377.0494828 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248378.316921 INFO: Making atp hydrolysis reaction for species: Species12\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-43 models/R2_C_D1.json\n",
      "1 missing R2_C_D1\n",
      "ForkPoolWorker-43 models/R2_C_D1.json\n",
      "0.9 missing R2_C_D1\n",
      "ForkPoolWorker-43 models/R2_C_D1.json\n",
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248378.665113 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248379.5323427 INFO: Making atp hydrolysis reaction for species: Species13\n",
      "1726248379.5956204 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248380.7871742 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248382.0916476 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248382.0111685 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9a9da1250>: 1, <Metabolite cpd11416_c1 at 0x7ff9aedbdb90>: -0.016717677232001397, <Metabolite cpd11416_c10 at 0x7ff9ae871550>: -0.014471649011430405, <Metabolite cpd11416_c11 at 0x7ff9ac6f6610>: -0.02924475467035163, <Metabolite cpd11416_c12 at 0x7ff9ac76ed10>: -0.07541247182213304, <Metabolite cpd11416_c13 at 0x7ff9ac7d23d0>: -0.015541451698603902, <Metabolite cpd11416_c14 at 0x7ff9ac659190>: -0.021167382377927188, <Metabolite cpd11416_c15 at 0x7ff9ac6cf290>: -0.025738695327665487, <Metabolite cpd11416_c16 at 0x7ff9ac547b10>: -0.023048202757754727, <Metabolite cpd11416_c17 at 0x7ff9ac5cb150>: -0.01808297791242194, <Metabolite cpd11416_c18 at 0x7ff9ac451610>: -0.025653095735519195, <Metabolite cpd11416_c19 at 0x7ff9ac4d5a50>: -0.10127488931479398, <Metabolite cpd11416_c2 at 0x7ff9aec479d0>: -0.016923667268490566, <Metabolite cpd11416_c20 at 0x7ff9ac34ead0>: -0.018833512060924894, <Metabolite cpd11416_c21 at 0x7ff9ac3d1b90>: -0.03953900262763851, <Metabolite cpd11416_c22 at 0x7ff9ac25af50>: -0.058825884983493025, <Metabolite cpd11416_c23 at 0x7ff9ac2dcb10>: -0.025030704689031548, <Metabolite cpd11416_c24 at 0x7ff9ac174810>: -0.023190249386338752, <Metabolite cpd11416_c25 at 0x7ff9abff9bd0>: -0.024699262555669316, <Metabolite cpd11416_c26 at 0x7ff9ac07d590>: -0.019260613019941298, <Metabolite cpd11416_c27 at 0x7ff9a9ef3d90>: -0.024324572125376993, <Metabolite cpd11416_c28 at 0x7ff9a9f771d0>: -0.020429598468074642, <Metabolite cpd11416_c29 at 0x7ff9a9e045d0>: -0.04059913051061003, <Metabolite cpd11416_c3 at 0x7ff9aecda6d0>: -0.019537530263396222, <Metabolite cpd11416_c30 at 0x7ff9a9e89d10>: -0.036566210803988634, <Metabolite cpd11416_c31 at 0x7ff9a9d0f450>: -0.02758004763207246, <Metabolite cpd11416_c32 at 0x7ff9a9da11d0>: -0.026782805322885066, <Metabolite cpd11416_c4 at 0x7ff9aeb530d0>: -0.07467840406223898, <Metabolite cpd11416_c5 at 0x7ff9aebd5590>: -0.05314401984014028, <Metabolite cpd11416_c6 at 0x7ff9aea5e2d0>: -0.0163714986419073, <Metabolite cpd11416_c7 at 0x7ff9aeadd490>: -0.022069830173868942, <Metabolite cpd11416_c8 at 0x7ff9ae9652d0>: -0.02462135154964412, <Metabolite cpd11416_c9 at 0x7ff9ae7edc10>: -0.02063885615366564}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9a9da1250>: -1}\n",
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248382.4797344 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248383.2848387 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248383.4913146 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248383.8639812 INFO: Making atp hydrolysis reaction for species: Species26\n",
      "1726248384.9171717 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species2 "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248385.2543037 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species26 262"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248385.7059968 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species16 163 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248386.3271582 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248386.6190746 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248386.9382632 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248387.7150183 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248387.985246 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248388.1946115 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248389.3578932 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248389.3994281 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248390.4046013 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248390.6601334 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248391.8060646 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248391.8682942 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248392.5317037 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248393.2026463 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac4cedd0>: 1, <Metabolite cpd11416_c1 at 0x7ff9b13bdad0>: -0.01735215991796166, <Metabolite cpd11416_c10 at 0x7ff9aee24450>: -0.04869896633006264, <Metabolite cpd11416_c11 at 0x7ff9aee8ca10>: -0.027520764789942392, <Metabolite cpd11416_c12 at 0x7ff9aed07790>: -0.01808870464267229, <Metabolite cpd11416_c13 at 0x7ff9aed8aed0>: -0.019688643873466943, <Metabolite cpd11416_c14 at 0x7ff9aec12590>: -0.025362538158787886, <Metabolite cpd11416_c15 at 0x7ff9aec94ad0>: -0.02448675539920119, <Metabolite cpd11416_c16 at 0x7ff9aeb0efd0>: -0.018646917861873426, <Metabolite cpd11416_c17 at 0x7ff9aeb93410>: -0.05536347580863413, <Metabolite cpd11416_c18 at 0x7ff9aea19950>: -0.040318720831968295, <Metabolite cpd11416_c19 at 0x7ff9aea8eb50>: -0.05056803480461301, <Metabolite cpd11416_c2 at 0x7ff9b1235850>: -0.017018641428456077, <Metabolite cpd11416_c20 at 0x7ff9ae911c90>: -0.026374740069220068, <Metabolite cpd11416_c21 at 0x7ff9ae996fd0>: -0.08215199838107094, <Metabolite cpd11416_c22 at 0x7ff9ae82c6d0>: -0.023870235789263675, <Metabolite cpd11416_c23 at 0x7ff9ae8b1a10>: -0.025366247216722673, <Metabolite cpd11416_c24 at 0x7ff9ac73ac50>: -0.04097440809369844, <Metabolite cpd11416_c25 at 0x7ff9ac7ba690>: -0.022602183062483152, <Metabolite cpd11416_c26 at 0x7ff9ac640fd0>: -0.060194152863036934, <Metabolite cpd11416_c27 at 0x7ff9ac6b81d0>: -0.07557761901031668, <Metabolite cpd11416_c28 at 0x7ff9ac53d910>: -0.044568188507973505, <Metabolite cpd11416_c29 at 0x7ff9ac5c3050>: -0.020069564123380238, <Metabolite cpd11416_c3 at 0x7ff9b12aa750>: -0.02865766522814543, <Metabolite cpd11416_c30 at 0x7ff9ac454dd0>: -0.03386355058324213, <Metabolite cpd11416_c31 at 0x7ff9ac4ced50>: -0.02172224615800932, <Metabolite cpd11416_c4 at 0x7ff9b1130a90>: -0.023747020884666833, <Metabolite cpd11416_c5 at 0x7ff9b11a7a10>: -0.018585792587106475, <Metabolite cpd11416_c6 at 0x7ff9b103c590>: -0.0242969258140937, <Metabolite cpd11416_c7 at 0x7ff9b10b0f90>: -0.018477413914248786, <Metabolite cpd11416_c8 at 0x7ff9aef29990>: -0.022938742979495007, <Metabolite cpd11416_c9 at 0x7ff9aefa3a10>: -0.02284698088618585}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac4cedd0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248393.8706422 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n",
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248394.2228553 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248394.623051 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248395.2104313 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248395.4369771 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248396.0207334 INFO: Making atp hydrolysis reaction for species: Species10\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-44 models/R2_C_D2.json\n",
      "1 missing R2_C_D2\n",
      "ForkPoolWorker-44 models/R2_C_D2.json\n",
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248396.5841608 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248396.64775 INFO: Making atp hydrolysis reaction for species: Species25\n",
      "1726248397.2151728 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248397.891914 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248397.942212 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248398.757394 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248400.1709461 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248400.2075756 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248400.492079 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248401.4096997 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248401.56257 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248401.8339512 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n",
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248402.9626102 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248403.1999855 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248404.3689358 INFO: Making atp hydrolysis reaction for species: Species15\n",
      "1726248404.532666 WARNING: Ignoring reaction 'rxn00062_c7' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248404.583262 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248405.939074 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248406.932952 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248408.3309348 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248408.4792953 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff97b207c90>: 1, <Metabolite cpd11416_c1 at 0x7ff9ac6aa010>: -0.016920161836424443, <Metabolite cpd11416_c10 at 0x7ff9aecbfd90>: -0.06940211531249946, <Metabolite cpd11416_c11 at 0x7ff9aec11c50>: -0.06330829407178278, <Metabolite cpd11416_c12 at 0x7ff9aed6ea90>: -0.097817690314725, <Metabolite cpd11416_c13 at 0x7ff97b4a7990>: -0.03730169507377297, <Metabolite cpd11416_c14 at 0x7ff9aee52890>: -0.07716344705468689, <Metabolite cpd11416_c15 at 0x7ff9aefa4690>: -0.024880911377561113, <Metabolite cpd11416_c16 at 0x7ff9aef07c90>: -0.13360902600480187, <Metabolite cpd11416_c17 at 0x7ff9b107e010>: -0.05446758820022214, <Metabolite cpd11416_c18 at 0x7ff97b3a8850>: -0.01499792713403989, <Metabolite cpd11416_c19 at 0x7ff9b11ad290>: -0.016092364389151636, <Metabolite cpd11416_c2 at 0x7ff9ac605750>: -0.01421441977289274, <Metabolite cpd11416_c20 at 0x7ff9b11500d0>: -0.024475555919268255, <Metabolite cpd11416_c21 at 0x7ff9b12af4d0>: -0.019982023171133838, <Metabolite cpd11416_c22 at 0x7ff97b207490>: -0.020221234600764196, <Metabolite cpd11416_c3 at 0x7ff9ac730990>: -0.038040013057707024, <Metabolite cpd11416_c4 at 0x7ff9ae8a8390>: -0.01724008464127692, <Metabolite cpd11416_c5 at 0x7ff9ae9d70d0>: -0.014028940985820873, <Metabolite cpd11416_c6 at 0x7ff9ae93fa50>: -0.0792156293194075, <Metabolite cpd11416_c7 at 0x7ff9aea9c9d0>: -0.0470747409816837, <Metabolite cpd11416_c8 at 0x7ff9aebdd0d0>: -0.09792245772657476, <Metabolite cpd11416_c9 at 0x7ff9aeb543d0>: -0.021623679053802074}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff97b207c90>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248408.4962564 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248409.3760445 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248409.7477765 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248409.8264334 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248410.2497013 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248411.1270182 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248411.1557982 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248411.1932042 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248412.0013468 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species1219\n",
      " "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248412.562769 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248412.5661678 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248412.8742344 INFO: Making atp hydrolysis reaction for species: Species6\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248413.766648 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-45 models/R2A_A_D1.json\n",
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248413.9237049 INFO: Making atp hydrolysis reaction for species: Species14\n",
      "1726248414.1372292 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248414.6411028 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248415.3376355 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248416.4009085 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9a9b336d0>: 1, <Metabolite cpd11416_c1 at 0x7ff9aeb55710>: -0.029784588616163357, <Metabolite cpd11416_c10 at 0x7ff9ac7db310>: -0.024918752563773197, <Metabolite cpd11416_c11 at 0x7ff9ac65e9d0>: -0.025976526897064593, <Metabolite cpd11416_c12 at 0x7ff9ac6d4a50>: -0.021796479196695355, <Metabolite cpd11416_c13 at 0x7ff9ac55ce90>: -0.06726534945952031, <Metabolite cpd11416_c14 at 0x7ff9ac5d2090>: -0.016995590136303074, <Metabolite cpd11416_c15 at 0x7ff9ac4591d0>: -0.02223183713592086, <Metabolite cpd11416_c16 at 0x7ff9ac4de590>: -0.09218627927623523, <Metabolite cpd11416_c17 at 0x7ff9ac364150>: -0.017227537061361552, <Metabolite cpd11416_c18 at 0x7ff9ac1eaa10>: -0.020217600217910595, <Metabolite cpd11416_c19 at 0x7ff9ac27c110>: -0.025149031805727824, <Metabolite cpd11416_c2 at 0x7ff9aebce590>: -0.027878241530080748, <Metabolite cpd11416_c20 at 0x7ff9ac1051d0>: -0.016629857966672584, <Metabolite cpd11416_c21 at 0x7ff9ac198e50>: -0.027914143041348934, <Metabolite cpd11416_c22 at 0x7ff9ac01e210>: -0.039796385157898344, <Metabolite cpd11416_c23 at 0x7ff9ac0a3450>: -0.04771806520038735, <Metabolite cpd11416_c24 at 0x7ff9a9f26e10>: -0.025542651342820574, <Metabolite cpd11416_c25 at 0x7ff9a9fa96d0>: -0.04060812990880808, <Metabolite cpd11416_c26 at 0x7ff9a9e1c9d0>: -0.018079584153952874, <Metabolite cpd11416_c27 at 0x7ff9a9e93dd0>: -0.019855388712611263, <Metabolite cpd11416_c28 at 0x7ff9a9d25550>: -0.05732906190277559, <Metabolite cpd11416_c29 at 0x7ff9a9daac90>: -0.027469705494112533, <Metabolite cpd11416_c3 at 0x7ff9aea50f50>: -0.04247533276647581, <Metabolite cpd11416_c30 at 0x7ff9a9c284d0>: -0.025581517624051725, <Metabolite cpd11416_c31 at 0x7ff9a9cae210>: -0.04454270392116313, <Metabolite cpd11416_c32 at 0x7ff9a9b33650>: -0.0220028086562943, <Metabolite cpd11416_c4 at 0x7ff9aead6010>: -0.017829153741157455, <Metabolite cpd11416_c5 at 0x7ff9ae95d250>: -0.031071437753612357, <Metabolite cpd11416_c6 at 0x7ff9ae9e1d90>: -0.03246288768131126, <Metabolite cpd11416_c7 at 0x7ff9ae866ed0>: -0.021435101532948312, <Metabolite cpd11416_c8 at 0x7ff9ac6edd10>: -0.025088392995082347, <Metabolite cpd11416_c9 at 0x7ff9ac762a90>: -0.02493987654975833}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9a9b336d0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248416.579715 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248416.7496927 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248417.1243541 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248417.75777 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248417.9636815 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248417.99648 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248418.151343 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248418.8720868 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248419.1069286 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248419.335356 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248419.5419624 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248419.7484477 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248420.4676704 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248420.6212275 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248420.7324808 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248420.9370108 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248421.4950445 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248421.815553 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248422.1062448 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248422.3692925 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248423.2488778 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248423.6106915 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248424.1244462 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248424.4088295 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1 methane flux: 155.36637899228947\n",
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248424.7551398 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248424.9908795 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248425.0221517 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248425.791402 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248426.131648 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248426.4205327 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248427.1437054 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248427.4468412 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248427.5492342 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248427.825855 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248428.324813 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-5 models/R2A_A_D1.json\n",
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248428.5023344 INFO: Making atp hydrolysis reaction for species: Species23\n",
      "1726248428.8051813 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248428.9534476 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248429.191641 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248429.2199996 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248429.8864017 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248430.062033 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248430.3554833 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-46 models/R2A_A_D2.json\n",
      "1 missing R2A_A_D2\n",
      "ForkPoolWorker-46 models/R2A_A_D2.json\n",
      "0.9 missing R2A_A_D2\n",
      "ForkPoolWorker-46 models/R2A_A_D2.json\n",
      "0.8 missing R2A_A_D2\n",
      "ForkPoolWorker-46 models/R2A_A_D2.json\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248431.8066964 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248431.9934502 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248432.5301254 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac0bc190>: 1, <Metabolite cpd11416_c1 at 0x7ff9aef39d90>: -0.023627780429884167, <Metabolite cpd11416_c10 at 0x7ff9aebda010>: -0.017578609797814088, <Metabolite cpd11416_c11 at 0x7ff9aea62450>: -0.11584133341144359, <Metabolite cpd11416_c12 at 0x7ff9aead75d0>: -0.030546322343106717, <Metabolite cpd11416_c13 at 0x7ff9ae96c150>: -0.043956923363500766, <Metabolite cpd11416_c14 at 0x7ff9ae7f4590>: -0.022687490154176516, <Metabolite cpd11416_c15 at 0x7ff9ae86b590>: -0.05821868149704713, <Metabolite cpd11416_c16 at 0x7ff9ac6fca90>: -0.070175959843056, <Metabolite cpd11416_c17 at 0x7ff9ac776690>: -0.027087761868226003, <Metabolite cpd11416_c18 at 0x7ff9ac5ffa50>: -0.01524945691745358, <Metabolite cpd11416_c19 at 0x7ff9ac681310>: -0.022709321320077646, <Metabolite cpd11416_c2 at 0x7ff9aefcd810>: -0.019819275408102337, <Metabolite cpd11416_c20 at 0x7ff9ac500d50>: -0.021719555680226914, <Metabolite cpd11416_c21 at 0x7ff9ac586490>: -0.028724905543068616, <Metabolite cpd11416_c22 at 0x7ff9ac40e850>: -0.024284588495117495, <Metabolite cpd11416_c23 at 0x7ff9ac493710>: -0.036491891821018266, <Metabolite cpd11416_c24 at 0x7ff9ac319150>: -0.02082803028693195, <Metabolite cpd11416_c25 at 0x7ff9ac38d150>: -0.01832396972236871, <Metabolite cpd11416_c26 at 0x7ff9ac216890>: -0.036973469255808536, <Metabolite cpd11416_c27 at 0x7ff9ac2a8050>: -0.028842199617851262, <Metabolite cpd11416_c28 at 0x7ff9ac120b10>: -0.02750539594727, <Metabolite cpd11416_c29 at 0x7ff9ac1a6950>: -0.019767991545009123, <Metabolite cpd11416_c3 at 0x7ff9aee46210>: -0.06434167763984194, <Metabolite cpd11416_c30 at 0x7ff9ac02fe10>: -0.028113374540842992, <Metabolite cpd11416_c31 at 0x7ff9ac0bc110>: -0.01714773492288333, <Metabolite cpd11416_c4 at 0x7ff9aeec8ad0>: -0.03208516118214592, <Metabolite cpd11416_c5 at 0x7ff9aed4c510>: -0.02412847628214966, <Metabolite cpd11416_c6 at 0x7ff9aedc7c90>: -0.026192361120037044, <Metabolite cpd11416_c7 at 0x7ff9aec58890>: -0.03282651657331055, <Metabolite cpd11416_c8 at 0x7ff9aecdd9d0>: -0.02694598846836467, <Metabolite cpd11416_c9 at 0x7ff9aeb57ad0>: -0.017257795001864383}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac0bc190>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248432.7073903 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248432.8799126 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248433.4313023 INFO: Making atp hydrolysis reaction for species: Species32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248433.881179 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248434.1160014 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248434.276717 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32 Species32\n",
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248435.2443004 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248435.6221423 INFO: Making atp hydrolysis reaction for species: Species3\n",
      "1726248435.6231637 WARNING: Ignoring reaction 'rxn00062_c12' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248435.6800902 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248436.6087563 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n",
      "3"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248437.0845013 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248437.0886965 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248437.9638991 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248438.4953241 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248438.525722 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248440.3815978 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248441.0074546 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248441.337132 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248441.7401133 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248442.4172416 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248442.7241492 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Species31\n",
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248443.8485286 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248444.1348865 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D2 growth: 0.0\n",
      "R2A_A_D2carbon uptake: 0.0\n",
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248445.244988 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248445.5047102 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248446.6462307 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248446.8727732 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9b8a74dd0>: 1, <Metabolite cpd11416_c1 at 0x7ff9bb46c910>: -0.016920161836424443, <Metabolite cpd11416_c10 at 0x7ff89f552050>: -0.06940211531249946, <Metabolite cpd11416_c11 at 0x7ff9b8f406d0>: -0.06330829407178278, <Metabolite cpd11416_c12 at 0x7ff9b8fc4b10>: -0.097817690314725, <Metabolite cpd11416_c13 at 0x7ff9b8e3c310>: -0.03730169507377297, <Metabolite cpd11416_c14 at 0x7ff9b8eb1190>: -0.07716344705468689, <Metabolite cpd11416_c15 at 0x7ff9b8d2ac90>: -0.024880911377561113, <Metabolite cpd11416_c16 at 0x7ff9b8da1890>: -0.13360902600480187, <Metabolite cpd11416_c17 at 0x7ff9b8c273d0>: -0.05446758820022214, <Metabolite cpd11416_c18 at 0x7ff9b8cb89d0>: -0.01499792713403989, <Metabolite cpd11416_c19 at 0x7ff9b8b07710>: -0.016092364389151636, <Metabolite cpd11416_c2 at 0x7ff9bb2f4d50>: -0.01421441977289274, <Metabolite cpd11416_c20 at 0x7ff9b8b7b710>: -0.024475555919268255, <Metabolite cpd11416_c21 at 0x7ff9b8a10c10>: -0.019982023171133838, <Metabolite cpd11416_c22 at 0x7ff9b8a74d50>: -0.020221234600764196, <Metabolite cpd11416_c3 at 0x7ff9bb369250>: -0.038040013057707024, <Metabolite cpd11416_c4 at 0x7ff9bb3d3c90>: -0.01724008464127692, <Metabolite cpd11416_c5 at 0x7ff9bb264e90>: -0.014028940985820873, <Metabolite cpd11416_c6 at 0x7ff9bb2dc490>: -0.0792156293194075, <Metabolite cpd11416_c7 at 0x7ff9bb154590>: -0.0470747409816837, <Metabolite cpd11416_c8 at 0x7ff9bb1c9c90>: -0.09792245772657476, <Metabolite cpd11416_c9 at 0x7ff9bb040490>: -0.021623679053802074}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9b8a74dd0>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248447.4470956 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248448.280251 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248448.3236895 INFO: Making atp hydrolysis reaction for species: Species2\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-47 models/R2A_C_D2.json\n",
      "1 missing R2A_C_D2\n",
      "ForkPoolWorker-47 models/R2A_C_D2.json\n",
      "0.9 missing R2A_C_D2\n",
      "ForkPoolWorker-47 models/R2A_C_D2.json\n",
      "0.8 missing R2A_C_D2\n",
      "ForkPoolWorker-47 models/R2A_C_D2.json\n",
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248449.2315547 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248449.2971394 INFO: Making atp hydrolysis reaction for species: Species21\n",
      "1726248449.4497685 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248450.1372693 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248450.6899776 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248450.9586456 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248451.8854232 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248452.1003044 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248452.3645172 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248452.7697866 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9ac11e350>: 1, <Metabolite cpd11416_c1 at 0x7ff9b11cd150>: -0.029214169811017817, <Metabolite cpd11416_c10 at 0x7ff9aec44390>: -0.057648903721611314, <Metabolite cpd11416_c11 at 0x7ff9aecc87d0>: -0.05422548863092152, <Metabolite cpd11416_c12 at 0x7ff9aeb419d0>: -0.024878597209657283, <Metabolite cpd11416_c13 at 0x7ff9aebc5d10>: -0.039713536717322875, <Metabolite cpd11416_c14 at 0x7ff9aea4cdd0>: -0.019229006756659547, <Metabolite cpd11416_c15 at 0x7ff9aead2190>: -0.07172425749808502, <Metabolite cpd11416_c16 at 0x7ff9ae964a10>: -0.017902529756248158, <Metabolite cpd11416_c17 at 0x7ff9ae7edfd0>: -0.029391237544018654, <Metabolite cpd11416_c18 at 0x7ff9ae87dc50>: -0.01786430762187357, <Metabolite cpd11416_c19 at 0x7ff9ac707010>: -0.02044620224028148, <Metabolite cpd11416_c2 at 0x7ff9b104afd0>: -0.02079727571209304, <Metabolite cpd11416_c20 at 0x7ff9ac798290>: -0.06025550546555421, <Metabolite cpd11416_c21 at 0x7ff9ac611a90>: -0.02131939851453135, <Metabolite cpd11416_c22 at 0x7ff9ac687b90>: -0.03461045943427855, <Metabolite cpd11416_c23 at 0x7ff9ac4fee90>: -0.022233350987313397, <Metabolite cpd11416_c24 at 0x7ff9ac590610>: -0.023037810771292987, <Metabolite cpd11416_c25 at 0x7ff9ac419d50>: -0.024845020859571482, <Metabolite cpd11416_c26 at 0x7ff9ac48f550>: -0.10903254498597441, <Metabolite cpd11416_c27 at 0x7ff9ac315690>: -0.02076042619580375, <Metabolite cpd11416_c28 at 0x7ff9ac399550>: -0.037649752633103876, <Metabolite cpd11416_c29 at 0x7ff9ac203110>: -0.029110167594556546, <Metabolite cpd11416_c3 at 0x7ff9b10c8290>: -0.021970547416986185, <Metabolite cpd11416_c30 at 0x7ff9ac294e90>: -0.023284987612649778, <Metabolite cpd11416_c31 at 0x7ff9ac11e2d0>: -0.02800753292805188, <Metabolite cpd11416_c4 at 0x7ff9aef42690>: -0.02416441346128674, <Metabolite cpd11416_c5 at 0x7ff9aefc5650>: -0.02852574904824768, <Metabolite cpd11416_c6 at 0x7ff9aee4e110>: -0.025728247804444455, <Metabolite cpd11416_c7 at 0x7ff9aeed1050>: -0.02164544810278841, <Metabolite cpd11416_c8 at 0x7ff9aed49b50>: -0.023080573103949863, <Metabolite cpd11416_c9 at 0x7ff9aedbfb50>: -0.017702549859824022}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9ac11e350>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248452.8917618 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248453.5020008 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248453.6807928 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248453.7661242 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248454.2519517 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248454.5773585 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248454.9149 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248455.1447935 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248455.4809961 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248455.6015873 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248456.5186844 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248456.9557517 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248457.281309 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248457.4530828 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248458.1955147 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248458.8655672 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248459.090033 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248459.2307532 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248459.5571465 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248459.9828138 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248460.267478 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248460.6303875 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248460.8683836 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248460.9256794 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248461.6818538 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248461.7918582 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9b1274150>: 1, <Metabolite cpd11416_c1 at 0x7ff9ac6138d0>: -0.016920161836424443, <Metabolite cpd11416_c10 at 0x7ff9aec3bd90>: -0.06940211531249946, <Metabolite cpd11416_c11 at 0x7ff9aed78810>: -0.06330829407178278, <Metabolite cpd11416_c12 at 0x7ff9aeebf250>: -0.097817690314725, <Metabolite cpd11416_c13 at 0x7ff9aee3e910>: -0.03730169507377297, <Metabolite cpd11416_c14 at 0x7ff9aef8bdd0>: -0.07716344705468689, <Metabolite cpd11416_c15 at 0x7ff97b43c250>: -0.024880911377561113, <Metabolite cpd11416_c16 at 0x7ff9b1092250>: -0.13360902600480187, <Metabolite cpd11416_c17 at 0x7ff9b103c610>: -0.05446758820022214, <Metabolite cpd11416_c18 at 0x7ff9b11d6290>: -0.01499792713403989, <Metabolite cpd11416_c19 at 0x7ff9b118d250>: -0.016092364389151636, <Metabolite cpd11416_c2 at 0x7ff9ac729e10>: -0.01421441977289274, <Metabolite cpd11416_c20 at 0x7ff97b2e5ed0>: -0.024475555919268255, <Metabolite cpd11416_c21 at 0x7ff9b12ce490>: -0.019982023171133838, <Metabolite cpd11416_c22 at 0x7ff9b12762d0>: -0.020221234600764196, <Metabolite cpd11416_c3 at 0x7ff9ae891950>: -0.038040013057707024, <Metabolite cpd11416_c4 at 0x7ff9ae7ee990>: -0.01724008464127692, <Metabolite cpd11416_c5 at 0x7ff9ae9467d0>: -0.014028940985820873, <Metabolite cpd11416_c6 at 0x7ff9aeaa4c10>: -0.0792156293194075, <Metabolite cpd11416_c7 at 0x7ff9aebe2bd0>: -0.0470747409816837, <Metabolite cpd11416_c8 at 0x7ff9aeb56810>: -0.09792245772657476, <Metabolite cpd11416_c9 at 0x7ff9aeccde50>: -0.021623679053802074}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9b1274150>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248461.9941647 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248462.078118 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248462.3039868 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248462.8606021 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248463.0816505 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'optlang.glpk_interface.Model'>\n",
      "bio1 {<Metabolite cpd11416_c0 at 0x7ff9bb332190>: 1, <Metabolite cpd11416_c1 at 0x7ff9bdc70d10>: -0.03485692084784196, <Metabolite cpd11416_c10 at 0x7ff9bd8e9210>: -0.09525769145381048, <Metabolite cpd11416_c11 at 0x7ffa204004d0>: -0.030817666098010114, <Metabolite cpd11416_c12 at 0x7ff9bb79ae90>: -0.019086592390922405, <Metabolite cpd11416_c13 at 0x7ff9bb61cfd0>: -0.02190844745390366, <Metabolite cpd11416_c14 at 0x7ff9bb6979d0>: -0.020464656703512097, <Metabolite cpd11416_c15 at 0x7ff9bb529050>: -0.06292576125820809, <Metabolite cpd11416_c16 at 0x7ff9bb5ae490>: -0.021276743357248332, <Metabolite cpd11416_c17 at 0x7ff9bb4340d0>: -0.015247438129513265, <Metabolite cpd11416_c18 at 0x7ff9bb4ab3d0>: -0.02196139375102277, <Metabolite cpd11416_c19 at 0x7ff9bb332090>: -0.014028943002031516, <Metabolite cpd11416_c2 at 0x7ff9bdb04490>: -0.018178106961743556, <Metabolite cpd11416_c3 at 0x7ff9bdb88cd0>: -0.025382819985511125, <Metabolite cpd11416_c4 at 0x7ff9bda0ed10>: -0.14781985405809803, <Metabolite cpd11416_c5 at 0x7ff9bda909d0>: -0.08223582353766065, <Metabolite cpd11416_c6 at 0x7ff9bd8f7710>: -0.06512321516206733, <Metabolite cpd11416_c7 at 0x7ff9bd989290>: -0.08847946998200662, <Metabolite cpd11416_c8 at 0x7ff9bd80e950>: -0.13528217090042355, <Metabolite cpd11416_c9 at 0x7ff9bd8657d0>: -0.07966628496646433}\n",
      "SK_cpd11416_c0 {<Metabolite cpd11416_c0 at 0x7ff9bb332190>: -1}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248463.3880289 INFO: Making atp hydrolysis reaction for species: Species1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248463.5590441 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248463.6497006 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248463.7194705 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248463.7947571 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Species1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248464.1744645 INFO: Making atp hydrolysis reaction for species: Species2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248464.5060322 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248464.58403 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Species2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248464.9645534 INFO: Making atp hydrolysis reaction for species: Species3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248464.9956534 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1 methane flux: 155.36637899228947\n",
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248465.4459262 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248465.6202853 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248465.663258 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Species3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248465.74857 INFO: Making atp hydrolysis reaction for species: Species4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248466.3088446 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 Species4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248466.5270643 INFO: Making atp hydrolysis reaction for species: Species5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248466.5838034 INFO: Making atp hydrolysis reaction for species: Species19\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-48 models/R1_A_D1.json\n",
      "31 Species31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248467.0929985 INFO: Making atp hydrolysis reaction for species: Species32\n",
      "1726248467.1266973 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248467.171811 INFO: Making atp hydrolysis reaction for species: Species7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 Species5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248467.33135 INFO: Making atp hydrolysis reaction for species: Species6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248467.409658 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248467.581729 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248467.5908425 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248468.0355082 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248468.4950786 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32 Species32\n",
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248468.7999349 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ForkPoolWorker-11 models/R2A_A_D1.json\n",
      "10 Species10\n",
      "6"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248468.9308631 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Species6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248468.934915 INFO: Making atp hydrolysis reaction for species: Species7\n",
      "1726248469.2869291 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 Species7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248469.7141693 INFO: Making atp hydrolysis reaction for species: Species8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248470.1900194 INFO: Making atp hydrolysis reaction for species: Species23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248470.283057 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248470.3355563 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248470.4996037 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 Species8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248470.5605505 INFO: Making atp hydrolysis reaction for species: Species9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 Species22\n",
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248471.2819831 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 Species9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248471.427345 INFO: Making atp hydrolysis reaction for species: Species10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 Species23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248471.5700324 INFO: Making atp hydrolysis reaction for species: Species24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248471.6442618 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248472.0636296 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 Species10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248472.2881348 INFO: Making atp hydrolysis reaction for species: Species11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248472.8519928 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24 Species24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248472.9755993 INFO: Making atp hydrolysis reaction for species: Species25\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248472.9905856 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 Species11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248473.1544077 INFO: Making atp hydrolysis reaction for species: Species12\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248474.0150836 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12 Species12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248474.3851252 INFO: Making atp hydrolysis reaction for species: Species13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248474.8802936 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 Species13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248475.1677496 INFO: Making atp hydrolysis reaction for species: Species14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248475.5695603 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 Species25\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248475.6069617 INFO: Making atp hydrolysis reaction for species: Species26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248475.746956 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14 Species14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248475.9490407 INFO: Making atp hydrolysis reaction for species: Species15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248476.612948 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248476.732605 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Species15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248476.9139717 INFO: Making atp hydrolysis reaction for species: Species16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26 Species26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248477.005687 INFO: Making atp hydrolysis reaction for species: Species27\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248477.4783092 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248477.5512779 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1 growth: 0.0\n",
      "R2A_A_D1carbon uptake: 0.0\n",
      "16 Species16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248478.2539415 INFO: Making atp hydrolysis reaction for species: Species17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248478.343976 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 Species27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248478.3607166 INFO: Making atp hydrolysis reaction for species: Species28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248479.160398 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Species17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248479.6046474 INFO: Making atp hydrolysis reaction for species: Species18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28 Species28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248479.7173114 INFO: Making atp hydrolysis reaction for species: Species29\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248479.9452713 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n",
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248480.8695016 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18 Species18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248480.9540958 INFO: Making atp hydrolysis reaction for species: Species19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29 Species29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248481.0986304 INFO: Making atp hydrolysis reaction for species: Species30\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248481.7358375 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 Species19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248482.300717 INFO: Making atp hydrolysis reaction for species: Species20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248482.6056054 INFO: Making atp hydrolysis reaction for species: Species21\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21 Species21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248483.4753923 INFO: Making atp hydrolysis reaction for species: Species22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30 Species30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248483.7235684 INFO: Making atp hydrolysis reaction for species: Species31\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Species22 2220 Species20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1726248484.8380296 INFO: Making atp hydrolysis reaction for species: Species21\n",
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ForkPoolWorker-49 models/R1_A_D2.json\n",
      "1 missing R1_A_D2\n",
      "31 Species31\n",
      "ForkPoolWorker-49 models/R1_A_D2.json\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "\n",
    "def runCommModel(arguments):\n",
    "    from multiprocess import current_process\n",
    "    from optlang.symbolics import Zero, add\n",
    "    \n",
    "    print(arguments)\n",
    "    (model_paths, feature_probabilities, metabolomics_data, metabolites, dipeptide_exchanges, media,\n",
    "     commKin, cUptake, missing) = arguments\n",
    "    output, rxn_record_hash = {}, {}\n",
    "    modelIDs = []\n",
    "    records = [{\"id\":\"max_growth\"}, {\"id\":\"carbon_uptake\"}, {\"id\":\"flux fitting objective\"}, {\"id\":\"minimum probability objective\"}]\n",
    "    \n",
    "    # simulate the community with the specified community growth\n",
    "    percentOptimums = [1, 0.9, 0.8, 0.7, 0.6, 0.5]\n",
    "    for percentOptimum in percentOptimums:\n",
    "        optStr = str(int(percentOptimum*100))\n",
    "        if missing is not None and optStr not in missing:  continue\n",
    "        output[percentOptimum] = {}\n",
    "        dirName = util.output_dir+f\"/{int(percentOptimum*100)}\"\n",
    "        if not os.path.exists(dirName):\n",
    "            os.makedirs(dirName)\n",
    "        for model_path in model_paths:\n",
    "            # print status\n",
    "            pid = current_process().name\n",
    "            print(pid, model_path)\n",
    "\n",
    "            modelID = model_path.split(\"/\")[-1].replace(\".json\", '')\n",
    "            if missing is not None and modelID not in missing[optStr]:\n",
    "                print(f\"{percentOptimum} missing {modelID}\")\n",
    "                continue\n",
    "            modelIDs.append(modelID)\n",
    "            output[percentOptimum][modelID] = {}\n",
    "            base_model = load_json_model(model_path)\n",
    "            print(type(base_model.solver))\n",
    "            current_comm_model = MSCommunity(model=base_model)\n",
    "            # print(dir(current_comm_model))\n",
    "            # assign reaction probabilities\n",
    "            reaction_probabilities = {}\n",
    "            for rxn in base_model.reactions:\n",
    "                # highest_prob = max([feature_probabilities.get(gene.id) for gene in rxn.genes])\n",
    "                highest_prob = None\n",
    "                for gene in rxn.genes:\n",
    "                    if gene.id in feature_probabilities:\n",
    "                        if highest_prob == None or feature_probabilities[gene.id] > highest_prob:\n",
    "                            highest_prob = feature_probabilities[gene.id]\n",
    "                if highest_prob != None:\n",
    "                    rxn.probability = highest_prob\n",
    "                    reaction_probabilities[rxn.id] = highest_prob\n",
    "                elif not any([rxn.id[0:3] in [\"bio\", \"EX_\", \"DM_\"]]) and len(rxn.genes) == 0:\n",
    "                    reaction_probabilities[rxn.id] = 0\n",
    "                else:\n",
    "                    reaction_probabilities[rxn.id] = 0.05\n",
    "            min_prob, prob_exp, ex_weight = 0.05, 1, 1\n",
    "            mdlutl = current_comm_model.mdlutl\n",
    "            pkgmgr = MSPackageManager.get_pkg_mgr(mdlutl)\n",
    "            # add media, community kinetics, and elemental uptake constraints\n",
    "            pkgmgr.getpkg(\"KBaseMediaPkg\").build_package(media)\n",
    "            pkgmgr.getpkg(\"CommKineticPkg\").build_package(commKin, current_comm_model)\n",
    "            pkgmgr.getpkg(\"ElementUptakePkg\").build_package({\"C\": cUptake})\n",
    "            # update exchange flux bounds \n",
    "            for rxn in base_model.reactions:\n",
    "                if \"EX_\" != rxn.id[0:3]:   continue\n",
    "                currRxn = mdlutl.model.reactions.get_by_id(rxn.id)\n",
    "                if rxn.id in dipeptide_exchanges:\n",
    "                    currRxn.lower_bound = 0\n",
    "                    currRxn.upper_bound = 0\n",
    "                else:\n",
    "                    ## the bounds are set to either -1000 or 1000\n",
    "                    if currRxn.lower_bound < 0:\n",
    "                        currRxn.lower_bound = -1000\n",
    "                    if currRxn.upper_bound > 0:\n",
    "                        currRxn.upper_bound = 1000\n",
    "            if \"EX_cpd00007_e0\" in [rxn.id for rxn in mdlutl.exchange_list()]:\n",
    "                mdlutl.model.reactions.get_by_id(\"EX_cpd00007_e0\").lower_bound = -cUptake/3   # oxygen\n",
    "            else:\n",
    "                print(f\"The {mdlutl.model.id} does not have Oxygen defined in the model\")\n",
    "\n",
    "            # optimizations\n",
    "            ## maximize biomass production and store the results\n",
    "            mdlutl.add_objective(Zero, \"max\", {mdlutl.model.reactions.bio1.forward_variable: 1})\n",
    "            maxBiomassSol = mdlutl.model.optimize()\n",
    "            output[percentOptimum][modelID][\"max_growth\"] = maxBiomassSol.objective_value\n",
    "            output[percentOptimum][modelID][\"carbon_uptake\"] = pkgmgr.getpkg(\"ElementUptakePkg\").variables[\"elements\"][\"C\"].primal\n",
    "            records[0][modelID] = output[percentOptimum][modelID][\"max_growth\"]\n",
    "            records[1][modelID] = output[percentOptimum][modelID][\"carbon_uptake\"]\n",
    "            print(modelID+\" growth:\", output[percentOptimum][modelID][\"max_growth\"])\n",
    "            print(modelID+\"carbon uptake:\", output[percentOptimum][modelID][\"carbon_uptake\"])\n",
    "            ## export the LP\n",
    "            export_path = util.output_dir+\"/LPs/\"+modelID+\"-growth.lp\"\n",
    "            if not os.path.exists(export_path):\n",
    "                os.makedirs(export_path)\n",
    "            with open(export_path, \"w\") as out:\n",
    "                out.write(str(mdlutl.model.solver))\n",
    "            if str(output[percentOptimum][modelID][\"max_growth\"]) == \"nan\":\n",
    "                print(\"Skipping condition due to infeasibility\", modelID)\n",
    "                return\n",
    "            mdlutl.add_minimal_objective_cons(output[percentOptimum][modelID][\"max_growth\"] * percentOptimum, name=\"minGrowth\")\n",
    "            # pkgmgr.getpkg(\"ObjConstPkg\").build_package(, None)\n",
    "            \n",
    "            ## maximize methane excretion\n",
    "            mdlutl.add_objective(Zero, \"max\", {mdlutl.model.reactions.get_by_id(\"EX_cpd01024_e0\").forward_variable: 1})\n",
    "            maxMethaneSol = mdlutl.model.optimize()\n",
    "            output[percentOptimum][modelID][\"max_methane\"] = maxMethaneSol.objective_value\n",
    "            print(modelID, \"methane flux:\", output[percentOptimum][modelID][\"max_methane\"])\n",
    "            mdlutl.add_minimal_objective_cons(output[percentOptimum][modelID][\"max_methane\"], name=\"minCH4\")\n",
    "            # ch4_weight = 100\n",
    "            # if \"EX_cpd01024_e0\" not in [rxn.id for rxn in mdlutl.exchange_list()]:\n",
    "            #     print(f\"The {mdlutl.model.id} does not have Methane defined in the model\")\n",
    "            # else:\n",
    "            #     methane = mdlutl.model.reactions.get_by_id(\"EX_cpd01024_e0\")\n",
    "            #     coef.update({methane.reverse_variable: ch4_weight, methane.forward_variable: -ch4_weight})\n",
    "            \n",
    "                \n",
    "            ## defining the minimal probability objective\n",
    "            output[percentOptimum][modelID] = {}\n",
    "            coef, flux_fitting_target = {}, {}\n",
    "            for rxn in base_model.reactions:\n",
    "                if \"rxn\" == rxn.id[0:3]:\n",
    "                    prob = reaction_probabilities[rxn.id]\n",
    "                    currRxn = mdlutl.model.reactions.get_by_id(rxn.id)\n",
    "                    coef.update({currRxn.forward_variable: max(min_prob, (1 - float(prob) ** prob_exp))})\n",
    "                    coef.update({currRxn.reverse_variable: max(min_prob, (1 - float(prob) ** prob_exp))})\n",
    "                elif \"EX_\" == rxn.id[0:3]:\n",
    "                    currRxn = mdlutl.model.reactions.get_by_id(rxn.id)\n",
    "                    if rxn.id[3:11] in metabolomics_data:\n",
    "                        flux_fitting_target[rxn.id] = -1*output[modelID][\"max_growth\"]*metabolomics_data[rxn.id[3:11]][modelID]\n",
    "                    coef.update({currRxn.forward_variable: ex_weight})\n",
    "                    coef.update({currRxn.reverse_variable: ex_weight})\n",
    "\n",
    "            ## Adjusting rxn bounds according to the maximum community growth and metabolomics data\n",
    "            ## This works around Linear Constraint problems and Quadratic Constraint infeasiblity\n",
    "            ### TODO replace constraining all metabolomics metabolites with only top (+)-correlating compounds, and their compount categories\n",
    "            stuck_reactions = {}\n",
    "\n",
    "            ## assigning the minimum objective function\n",
    "            mdlutl.model.objective = mdlutl.model.problem.Objective(Zero, direction=\"min\")\n",
    "            mdlutl.model.objective.set_linear_coefficients(coef)\n",
    "\n",
    "            ## simulating and checking the solution\n",
    "            solution = mdlutl.model.optimize()\n",
    "            if solution.status != \"optimal\":\n",
    "                print(f\"{model_path} is infeasible at {percentOptimum} of optimum growth\")\n",
    "                continue\n",
    "                # while solution.status != \"optimal\":\n",
    "                #     util.save(\"objConstPkg_contents\", mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"])\n",
    "                #     print(dir(mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"]),\n",
    "                #           mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"])\n",
    "                #     print(\"Infeasible: reducing objective constraint\",0.9*mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"].lb)\n",
    "                #     mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"].lb = 0.9*mdlutl.pkgmgr.getpkg(\"ObjConstPkg\").constraints[\"objc\"].lb\n",
    "                #     solution = mdlutl.model.optimize()\n",
    "\n",
    "            ## storing and exporting successful simulation results\n",
    "            export_path = util.output_dir+\"/LPs/\"+modelID+\"-final.lp\"\n",
    "            if not os.path.exists(export_path):\n",
    "                os.makedirs(export_path)\n",
    "            with open(export_path, \"w\") as out:\n",
    "                out.write(str(mdlutl.model.solver))\n",
    "\n",
    "            output[percentOptimum][modelID][\"minimum probability objective\"] = solution.objective_value\n",
    "            records[3][modelID] = output[percentOptimum][modelID][\"minimum probability objective\"]\n",
    "            output[percentOptimum][modelID][\"solution\"] = {}\n",
    "            for rxn in mdlutl.model.reactions:\n",
    "                if rxn.id not in rxn_record_hash:\n",
    "                    rxn_record_hash[rxn.id] = {\"id\":rxn.id}\n",
    "                    for mdlID in modelIDs:\n",
    "                        rxn_record_hash[rxn.id][mdlID] = None\n",
    "                    records.append(rxn_record_hash[rxn.id])\n",
    "                rxn_record_hash[rxn.id][modelID] = solution.fluxes[rxn.id]\n",
    "                output[percentOptimum][modelID][\"solution\"][rxn.id] = solution.fluxes[rxn.id]\n",
    "            \n",
    "            baseName = dirName+f\"/{modelID}\"\n",
    "            with open(baseName+\".json\", \"w\") as out:\n",
    "                dump(output, out, indent=3)\n",
    "            df = DataFrame.from_records(records)\n",
    "            df.to_csv(f\"{baseName}.csv\")\n",
    "        \n",
    "        with open(dirName+f\"/{modelID}_output.json\", \"w\") as out:\n",
    "            dump(output[percentOptimum], out, indent=3)\n",
    "        with open(dirName+f\"/{modelID}_rxn_record_hash.json\", \"w\") as out:\n",
    "            dump(rxn_record_hash, out, indent=3)\n",
    "        util.save(f\"{percentOptimum}_{modelID}_records\", records)\n",
    "\n",
    "# import stored data\n",
    "metabolites = util.load(\"metabolites\")\n",
    "metabolomics_path = \"/scratch/shared/code/MicrobiomeNotebooks/Cliff/data/Cliff_Sample_Metadata_BGC_NMR.csv\"\n",
    "metabolomics_data = read_csv(metabolomics_path).drop([\"New_index\", \"Unnamed: 0\"], axis=1).set_index(\"Sample\").to_dict()\n",
    "feature_probabilities = util.load(\"feature_probabilities\")\n",
    "# media = util.msrecon.get_media(\"KBaseMedia/Complete\")\n",
    "media = util.msrecon.get_media(\"188065/1875/1\")\n",
    "print(media)\n",
    "\n",
    "\n",
    "# R1_B_D2 / R1_B_D1 / R1_C_D2 are really slow for some reason\n",
    "\n",
    "from glob import glob\n",
    "from cobra.io import load_json_model\n",
    "from json import load, dump\n",
    "with open(\"/scratch/shared/code/MicrobiomeNotebooks/Cliff/data/dipeptide_exchanges.json\", 'r') as jsonIn:\n",
    "    dipeptide_exchanges = load(jsonIn)\n",
    "    \n",
    "missing_models = {\"100\": [\"R2A_A_D1\", \"R1_B_D1\", \"R1_A_D1\", \"R1_B_D2\"],\n",
    "                  '90': ['R2A_A_D1', 'R1_A_D1', 'R1_B_D1', 'R1_B_D2', 'R2_A_D1', 'R2_C_D2'],\n",
    "                  '80': ['R2A_A_D1', 'R1_A_D1', 'R1_A_D2', 'R1_B_D1', 'R1_B_D2', 'R1_C_D2', 'R2_A_D1', 'R2_C_D1', 'R2_C_D2'],\n",
    "                  '70': ['R2A_A_D1', 'R2A_A_D2', 'R2A_C_D2', 'R1_A_D1', 'R1_A_D2', 'R1_B_D1', 'R1_B_D2', 'R1_C_D2', 'R2_A_D1', 'R2_B_D2', 'R2_C_D1', 'R2_C_D2'],\n",
    "                  '60': ['R2A_A_D1', 'R2A_A_D2', 'R2A_C_D2', 'R1_A_D1', 'R1_A_D2', 'R1_B_D1', 'R1_B_D2', 'R1_C_D1', 'R1_C_D2', 'R2_A_D1', 'R2_B_D2', 'R2_C_D1', 'R2_C_D2'],\n",
    "                  '50': ['R2A_A_D1', 'R2A_A_D2', 'R2A_C_D2', 'R1_A_D1', 'R1_A_D2', 'R1_B_D1', 'R1_B_D2', 'R1_C_D1', 'R1_C_D2', 'R2_A_D1', 'R2_B_D1', 'R2_B_D2', 'R2_C_D1', 'R2_C_D2']}\n",
    "# models = list(glob(\"models/*.json\")) # [\"models/R2A_B_D2.json\"] # [\"models/R2A_A_D1.json\", \"models/R1_B_D2.json\"] #  # \n",
    "from itertools import chain\n",
    "models = [f\"models/{m}.json\" for m in list(chain.from_iterable(list(missing_models.values())))]\n",
    "parallelize = True\n",
    "commKin = 750\n",
    "cUptake = 500\n",
    "if parallelize:\n",
    "    from multiprocess import Pool\n",
    "    from os import cpu_count\n",
    "    from numpy import array_split\n",
    "    \n",
    "    # a thread is given for each community model\n",
    "    pool_count = min(cpu_count(), len(models)*3)\n",
    "    print(f\"{pool_count} threads will be created\")\n",
    "    pool = Pool(pool_count)\n",
    "    args = [[[model_path], feature_probabilities, metabolomics_data, metabolites,\n",
    "             dipeptide_exchanges, media, commKin, cUptake, missing_models]\n",
    "            for model_path in models]\n",
    "    list_of_outputs = pool.map(runCommModel, args)\n",
    "else:\n",
    "    outputs = runCommModel([models, feature_probabilities, metabolomics_data, metabolites,\n",
    "                            dipeptide_exchanges, media, commKin, cUptake, missing_models])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## combine the various outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0.1</th>\n",
       "      <th>SF2_A_D1</th>\n",
       "      <th>SF2_A_D2</th>\n",
       "      <th>SF2_B_D1</th>\n",
       "      <th>SF2_B_D2</th>\n",
       "      <th>SF2_C_D1</th>\n",
       "      <th>R2A_A_D2</th>\n",
       "      <th>SF2_C_D2</th>\n",
       "      <th>R2A_B_D1</th>\n",
       "      <th>R2A_B_D2</th>\n",
       "      <th>...</th>\n",
       "      <th>R2_C_D2</th>\n",
       "      <th>R2_A_D2</th>\n",
       "      <th>R2_C_D1</th>\n",
       "      <th>R2_B_D2</th>\n",
       "      <th>R1_C_D2</th>\n",
       "      <th>R1_B_D1</th>\n",
       "      <th>R1_C_D1</th>\n",
       "      <th>R1_A_D2</th>\n",
       "      <th>R1_B_D2</th>\n",
       "      <th>R1_A_D1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>EX_cpd00001_e0</th>\n",
       "      <td>12619818</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>55380.981518</td>\n",
       "      <td>54237.515140</td>\n",
       "      <td>42741.332130</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14765.727163</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5254.596150</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_cpd00007_e0</th>\n",
       "      <td>12591144</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-32742.123623</td>\n",
       "      <td>-32039.059057</td>\n",
       "      <td>-25623.848739</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-10593.837628</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-2430.128847</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_cpd00009_e0</th>\n",
       "      <td>12575424</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-6932.646960</td>\n",
       "      <td>-6783.988908</td>\n",
       "      <td>-10612.005906</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1056.885681</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-97.456634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_cpd00011_e0</th>\n",
       "      <td>12585912</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23290.527786</td>\n",
       "      <td>22783.077453</td>\n",
       "      <td>15577.560878</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10695.010054</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>818.023950</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_cpd00012_e0</th>\n",
       "      <td>12757674</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2785.271916</td>\n",
       "      <td>2725.546807</td>\n",
       "      <td>4727.964647</td>\n",
       "      <td>0.0</td>\n",
       "      <td>284.749576</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul00010_c3</th>\n",
       "      <td>536346</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul00010_c4</th>\n",
       "      <td>522888</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul00010_c5</th>\n",
       "      <td>1067892</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul00010_c6</th>\n",
       "      <td>501012</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul00010_c7</th>\n",
       "      <td>1568658</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.861548</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30531 rows  25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Unnamed: 0.1  SF2_A_D1  SF2_A_D2      SF2_B_D1      SF2_B_D2  \\\n",
       "id                                                                             \n",
       "EX_cpd00001_e0      12619818       0.0       0.0  55380.981518  54237.515140   \n",
       "EX_cpd00007_e0      12591144       0.0       0.0 -32742.123623 -32039.059057   \n",
       "EX_cpd00009_e0      12575424       0.0       0.0  -6932.646960  -6783.988908   \n",
       "EX_cpd00011_e0      12585912       0.0       0.0  23290.527786  22783.077453   \n",
       "EX_cpd00012_e0      12757674       0.0       0.0   2785.271916   2725.546807   \n",
       "...                      ...       ...       ...           ...           ...   \n",
       "sul00010_c3           536346       0.0       0.0      0.000000      0.000000   \n",
       "sul00010_c4           522888       0.0       0.0      0.000000      0.000000   \n",
       "sul00010_c5          1067892       0.0       0.0      0.000000      0.000000   \n",
       "sul00010_c6           501012       0.0       0.0      0.000000      0.000000   \n",
       "sul00010_c7          1568658       0.0       0.0      0.000000      0.000000   \n",
       "\n",
       "                    SF2_C_D1  R2A_A_D2      SF2_C_D2  R2A_B_D1  R2A_B_D2  ...  \\\n",
       "id                                                                        ...   \n",
       "EX_cpd00001_e0  42741.332130       0.0  14765.727163       0.0       0.0  ...   \n",
       "EX_cpd00007_e0 -25623.848739       0.0 -10593.837628       0.0       0.0  ...   \n",
       "EX_cpd00009_e0 -10612.005906       0.0  -1056.885681       0.0       0.0  ...   \n",
       "EX_cpd00011_e0  15577.560878       0.0  10695.010054       0.0       0.0  ...   \n",
       "EX_cpd00012_e0   4727.964647       0.0    284.749576       0.0       0.0  ...   \n",
       "...                      ...       ...           ...       ...       ...  ...   \n",
       "sul00010_c3         0.000000       0.0      0.000000       0.0       0.0  ...   \n",
       "sul00010_c4         0.000000       0.0      0.000000       0.0       0.0  ...   \n",
       "sul00010_c5         0.000000       0.0      0.000000       0.0       0.0  ...   \n",
       "sul00010_c6         0.000000       0.0      0.000000       0.0       0.0  ...   \n",
       "sul00010_c7         0.000000       0.0      1.861548       0.0       0.0  ...   \n",
       "\n",
       "                R2_C_D2      R2_A_D2  R2_C_D1  R2_B_D2  R1_C_D2  R1_B_D1  \\\n",
       "id                                                                         \n",
       "EX_cpd00001_e0      0.0  5254.596150      0.0      0.0      0.0      0.0   \n",
       "EX_cpd00007_e0      0.0 -2430.128847      0.0      0.0      0.0      0.0   \n",
       "EX_cpd00009_e0      0.0   -97.456634      0.0      0.0      0.0      0.0   \n",
       "EX_cpd00011_e0      0.0   818.023950      0.0      0.0      0.0      0.0   \n",
       "EX_cpd00012_e0      0.0     0.000000      0.0      0.0      0.0      0.0   \n",
       "...                 ...          ...      ...      ...      ...      ...   \n",
       "sul00010_c3         0.0     0.000000      0.0      0.0      0.0      0.0   \n",
       "sul00010_c4         0.0     0.000000      0.0      0.0      0.0      0.0   \n",
       "sul00010_c5         0.0     0.000000      0.0      0.0      0.0      0.0   \n",
       "sul00010_c6         0.0     0.000000      0.0      0.0      0.0      0.0   \n",
       "sul00010_c7         0.0     0.000000      0.0      0.0      0.0      0.0   \n",
       "\n",
       "                R1_C_D1  R1_A_D2  R1_B_D2  R1_A_D1  \n",
       "id                                                  \n",
       "EX_cpd00001_e0      0.0      0.0      0.0      0.0  \n",
       "EX_cpd00007_e0      0.0      0.0      0.0      0.0  \n",
       "EX_cpd00009_e0      0.0      0.0      0.0      0.0  \n",
       "EX_cpd00011_e0      0.0      0.0      0.0      0.0  \n",
       "EX_cpd00012_e0      0.0      0.0      0.0      0.0  \n",
       "...                 ...      ...      ...      ...  \n",
       "sul00010_c3         0.0      0.0      0.0      0.0  \n",
       "sul00010_c4         0.0      0.0      0.0      0.0  \n",
       "sul00010_c5         0.0      0.0      0.0      0.0  \n",
       "sul00010_c6         0.0      0.0      0.0      0.0  \n",
       "sul00010_c7         0.0      0.0      0.0      0.0  \n",
       "\n",
       "[30531 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# %run cliffcommutil.py\n",
    "\n",
    "# from pandas import read_csv, concat\n",
    "# from glob import glob\n",
    "# from os import remove\n",
    "\n",
    "# CSVs = []\n",
    "\n",
    "#     for csv in glob(util.output_dir+\"/*%.csv\"):\n",
    "#         df = read_csv(csv)\n",
    "#         # print(df[\"id\"])\n",
    "#         remove(csv)\n",
    "#         CSVs.append(df)\n",
    "\n",
    "# totCSV = concat(CSVs).drop(\"Unnamed: 0\", axis=1).set_index(\"id\").groupby(level=0).sum()\n",
    "# display(totCSV)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## combine the simulation outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "\n",
    "from json import load, dump\n",
    "from glob import glob\n",
    "from os import remove\n",
    "\n",
    "# combine outputs\n",
    "outputs = {}\n",
    "maxGrowthPercent = 100\n",
    "baseDir = util.output_dir+f\"/{maxGrowthPercent}\"\n",
    "for output in glob(baseDir+\"/*_output.json\"):\n",
    "    mag = output.replace(baseDir+\"/\", '').replace(\"_output.json\", '')\n",
    "    with open(output, 'r') as jsonIn:\n",
    "        outputs.update(load(jsonIn))\n",
    "    # remove(output)\n",
    "        \n",
    "with open(baseDir+\"/MAGfluxes.json\", 'w') as jsonOut:\n",
    "    dump(outputs, jsonOut, indent=3)\n",
    "    \n",
    "# combine rxn_record_hashes\n",
    "outputs = {}\n",
    "for output in glob(baseDir+\"/*_rxn_record_hash.json\"):\n",
    "    mag = output.replace(baseDir+\"/\", '').replace(\"_rxn_record_hash\", '')\n",
    "    with open(output, 'r') as jsonIn:\n",
    "        outputs.update({mag: load(jsonIn)})\n",
    "    # remove(output)\n",
    "        \n",
    "with open(baseDir+\"/MAG_rxn_record_hash.json\", 'w') as jsonOut:\n",
    "    dump(outputs, jsonOut, indent=3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "print(len(set([i for i in totCSV.index])) == len([i for i in totCSV.index]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Python 3.11.1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import subprocess\n",
    "\n",
    "# Run the command and capture the output\n",
    "result = subprocess.run([\"python\", \"-V\"], capture_output=True, text=True)\n",
    "\n",
    "# Print the output\n",
    "print(result.stdout)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## check the outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "22294 2661\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "\n",
    "from json import load, dump\n",
    "from glob import glob\n",
    "from os import remove\n",
    "\n",
    "output = util.output_dir+\"/R1_A_D1/0.7%.json\"\n",
    "with open(output, 'r') as jsonIn:\n",
    "    data = load(jsonIn)\n",
    "    \n",
    "print(len(data[\"R1_A_D1\"][\"solution\"]), len([v for k,v in data[\"R1_A_D1\"][\"solution\"].items() if v != 0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from glob import glob\n",
    "from itertools import product\n",
    "\n",
    "sites = [\"R2A\", \"SF2\", \"R1\", \"R2\"]\n",
    "cores = [\"A\", \"B\", \"C\"]\n",
    "depths = [\"D1\", \"D2\"]\n",
    "\n",
    "sampleParts = product(sites, cores, depths)\n",
    "print(list(sampleParts))\n",
    "samples = [f\"{site}_{core}_{depth}\" for (site, core, depth) in product(sites, cores, depths)]\n",
    "\n",
    "for x in [\"100\", \"90.0\", \"80.0\", \"70.0\", \"60.0\", \"50.0\"]:\n",
    "    models = glob(f\"/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput/{x}/*_output.json\")\n",
    "    sampleModels = [model.split(\"/\")[-1].replace(\"_output.json\", '') for model in models]\n",
    "    if set(samples) - set(sampleModels) != set():\n",
    "        print(x, len(models))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# computing the community interactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n",
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556544.6722653 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_D1 2362 reactions  11 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556561.6754322 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_D1 2516 reactions  15 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556582.6287735 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D1 2484 reactions  15 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556602.7817354 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D2 2156 reactions  15 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556622.3747659 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_A_D2 2678 reactions  22 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556653.5490954 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_B_D2 2671 reactions  22 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556687.8185422 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SF2_C_D2 2743 reactions  26 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556726.0416288 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_B_D1 2688 reactions  21 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556757.0573468 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D2 2732 reactions  19 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556786.5018601 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_A_D1 2786 reactions  22 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556819.558982 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D1 2692 reactions  21 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556845.5904927 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_C_D2 2814 reactions  27 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556888.6159916 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D1 2777 reactions  27 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556932.2527647 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_A_D2 2822 reactions  27 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725556977.8175907 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2A_B_D2 2792 reactions  24 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725557008.730818 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D1 2806 reactions  27 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725557057.6311133 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D1 2915 reactions  30 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725557097.9016235 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_D1 2749 reactions  28 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725557150.162412 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_A_D1 2870 reactions  31 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725557207.182311 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D1 2736 reactions  31 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725557249.7971487 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_C_D2 2870 reactions  31 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725557309.6287737 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_B_D2 2864 reactions  31 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725557351.0636017 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R2_B_D2 2924 reactions  32 members\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725557416.155094 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R1_C_D2 2884 reactions  32 members\n",
      "[{'id': 'rxn02201', 'equation': 'rxn02201_c1: cpd00443_c1 + cpd02920_c1 --> cpd00012_c1 + cpd00067_c1 + cpd00683_c1', 'name': '2-amino-4-hydroxy-6-hydroxymethyl-7,8-dihydropteridine-diphosphate:4-aminobenzoate 2-amino-4-hydroxydihydropteridine-6-methenyltransferase [c0]', 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_A_D1': 2.7693604168155426e-38, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_A_D2': 2.8178604185778387e-35, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_B_D1': -4.179035040155979e-44, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_B_D1': 1.4841547707082385e-45, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_A_D1': 1.8033580354630776e-28, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_A_D1': 2.987619276021446e-23, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_C_D1': -9.350088149353903e-21, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_B_D2': -3.650190026622896e-23, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_B_D2': 1.42170651042098e-18, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_B_D2': 1.2493418102894373e-34, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_C_D2': -4.714942693892743e-46, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24_R1_C_D2': 0.0}, {'id': 'rxn00836', 'equation': 'rxn00836_c1: cpd00012_c1 + cpd00067_c1 + cpd00114_c1 <-- cpd00103_c1 + cpd00226_c1', 'name': 'IMP:diphosphate phospho-D-ribosyltransferase [c0]', 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_A_D1': -0.00862952967907399, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_C_D1': 2.2111737969009604e-31, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_SF2_C_D1': -2.786891926193039e-20, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_A_D2': 3.952294625487586e-18, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_A_D2': -4.071300214517333e-05, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_A_D2': -1.2899960418282062e-11, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_A_D2': 4.656796820335137e-13, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_A_D2': -4.6662285589140805e-15, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_C_D2': -7.66585225339083e-27, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_A_D1': 5.240757127220802e-15, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_A_D1': 9.318376154999698e-18, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_B_D1': 1.402940376081219e-39, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_B_D1': 5.588325383272071e-39, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_B_D1': -3.026312706659013e-41, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_B_D1': 4.4400634450265364e-42, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_B_D1': 2.4034086279259274e-39, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_B_D1': 3.1938198017565845e-41, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_B_D1': -1.3208749086748603e-44, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_B_D1': 4.034335068393577e-58, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_B_D1': -9.282807941294431e-39, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_C_D1': -2.072638418660379e-10, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2_A_D1': -5.189463507528162e-25, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_A_D1': 1.849455244418427e-22, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_A_D1': -3.979612224884045e-23, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_A_D1': 3.0082085832694125e-28, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_A_D1': -4.2479478284975245e-22, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_A_D1': -6.910510993500888e-22, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_C_D1': -5.425140540633442e-20, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_C_D1': 1.446621760168521e-17, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R2_C_D1': -5.2251537143357155e-21, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_B_D2': -7.503891346327268e-08, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_B_D2': 1.0113564014062143e-23, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_B_D2': 4.140783968573519e-21, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_B_D2': -1.8206600453501482e-20, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_B_D2': 1.6176220521177647e-31, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24_R2_B_D2': -2.4252079351412126e-24, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_C_D2': 7.827232441920741e-45, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_C_D2': -4.891424484190857e-45, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R1_C_D2': 0.0}, {'id': 'rxn00423', 'equation': 'rxn00423_c1: cpd00022_c1 + cpd00054_c1 --> cpd00010_c1 + cpd00722_c1', 'name': 'acetyl-CoA:L-serine O-acetyltransferase [c0]', 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_A_D1': 0.014301391575940058, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_A_D1': 0.0018050035402757303, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_A_D1': 0.002725133410271756, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_SF2_B_D1': 1.3799888405990515e-05, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_B_D1': 2.910262971788036e-20, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_SF2_C_D1': 9.748180404103002e-06, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_SF2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_C_D1': -1.1920634829661588e-48, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_A_D2': 3.520977299406007e-10, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_A_D2': 0.009371661827866487, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_SF2_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_SF2_A_D2': 5.355138387936635e-13, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_SF2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_SF2_C_D2': 2.673661779084417e-19, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_SF2_C_D2': 2.0167883879679614e-19, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_SF2_C_D2': 1.1333985798916995e-16, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_SF2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_B_D1': 8.58297081032899e-15, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_B_D1': 3.632274975548871e-14, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_B_D1': -3.164480496554907e-17, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_A_D2': 1.6671300146953717e-07, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_A_D2': 2.291877003906566e-07, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_A_D2': 6.618798720142964e-08, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_A_D1': 5.328754773868731e-08, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2A_A_D1': 4.461403906444876e-07, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_A_D1': 4.983366797043879e-09, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2A_A_D1': 8.914176852360985e-22, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_C_D1': 8.292654109683635e-08, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_C_D1': 2.8721450731737178e-11, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_C_D2': -5.797017257199348e-36, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D2': -4.234861581129308e-42, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_C_D2': -1.6646421153039683e-25, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_C_D2': 3.664593998012818e-07, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2A_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_A_D1': 2.1046032148509295e-18, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_A_D1': 4.691241697586647e-37, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_A_D2': -3.768438791459147e-22, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_A_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2A_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_B_D1': 4.8407853244426625e-08, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_B_D1': 3.850592109653727e-42, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_B_D1': 2.333679293748476e-40, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_B_D1': -2.4357578857195868e-21, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_B_D1': -1.9117005335651394e-17, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_B_D1': 1.675430111319456e-11, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R1_B_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_C_D1': 2.6233685423430967e-07, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_C_D1': 3.268965602496392e-10, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R1_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2_A_D1': 4.5247303307005425e-25, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R2_A_D1': -1.15465273559647e-24, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R2_A_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_C_D1': 6.100247721713115e-08, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_C_D1': -4.5322830416095744e-23, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_C_D1': 1.443989742195174e-09, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R2_C_D1': -1.1451109792964048e-22, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R2_C_D1': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R2_C_D1': 1.7083128132175908e-21, 'Salt_Pond_MetaGSF2_A_D1_MG_DASTool_bins_concoct_out.23_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_C_D2': 1.0396102933893395e-06, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_metabat.4_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R2_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R1_B_D2': 2.4392447942654677e-06, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R1_B_D2': 9.823768938555427e-07, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R1_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.11_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.33_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.23_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.14_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R2_B_D2': 9.793995920562619e-24, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R2_B_D2': 2.219172473980033e-09, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.17_R2_B_D2': 1.0029539475546719e-17, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24_R2_B_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.13_R1_C_D2': 2.7104662941131415e-07, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.28_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.40_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.55_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_D2_MG_DASTool_bins_concoct_out.8_R1_C_D2': 3.742936047109266e-43, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_concoct_out.33_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.14_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.23_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.28_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_A_H2O_MG_DASTool_bins_metabat.34_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_concoct_out.44_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D1_MG_DASTool_bins_metabat.9_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.42_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_D2_MG_DASTool_bins_concoct_out.55_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_concoct_out.91_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.11_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.19_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.25_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.28_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.39_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_B_H2O_MG_DASTool_bins_metabat.8_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D1_MG_DASTool_bins_concoct_out.13_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.13_R1_C_D2': 2.5076644569949033e-09, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.15_R1_C_D2': 0.0, 'Salt_Pond_MetaGSF2_C_D2_MG_DASTool_bins_concoct_out.24_R1_C_D2': 0.0}]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "\n",
    "from json import load\n",
    "maxGrowthPercent = 100\n",
    "baseDir = util.output_dir+f\"/{maxGrowthPercent}\"\n",
    "with open(f\"/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput/{maxGrowthPercent}/MAGfluxes.json\", 'r') as jsonIn:\n",
    "    CommunityFluxSolution = load(jsonIn)\n",
    "# CommunityFluxSolution = {k2:v2 for k,v in CommunityFluxSolution.items() for k2, v2 in v.items()}   # temporary patch of combining bug\n",
    "\n",
    "from cobra.io import load_json_model\n",
    "from pandas import read_csv\n",
    "metabolomics_path = \"/scratch/shared/code/MicrobiomeNotebooks/Cliff/data/Cliff_Sample_Metadata_BGC_NMR.csv\"\n",
    "metabolomics_data = read_csv(metabolomics_path)\n",
    "mag_abundances = util.load(\"refined_MAG_abundances\")\n",
    "metabolites = util.load(\"metabolites\")\n",
    "metabolite_names = util.load(\"metabolite_names\")\n",
    "mag_list = util.load(\"mag_list\")\n",
    "\n",
    "sample_interval_exchanges = {}#[Sample interval][metabolite] = Flux\n",
    "mag_sample_flux_profiles = {}#[Sample][MAG][Reaction] = Flux\n",
    "rxn_data_hash = {}\n",
    "mag_biomass = {}\n",
    "mag_rescaled_biomass = {}\n",
    "biomasses = {}\n",
    "for sample, content in CommunityFluxSolution.items():\n",
    "    sample_interval_exchanges[sample] = {}\n",
    "    mag_sample_flux_profiles[sample] = {}\n",
    "    rxn_data_hash[sample], mag_biomass[sample] = {}, {}\n",
    "    base_model = load_json_model(\"models/\"+sample+'.json')\n",
    "    for rxnobj in base_model.reactions:\n",
    "        if rxnobj.id[0:3] == \"rxn\":\n",
    "            (baseid, compartment) = rxnobj.id.split(\"_\")\n",
    "            if baseid not in rxn_data_hash[sample]:\n",
    "                rxn_data_hash[sample][baseid] = {\n",
    "                    \"id\":baseid,\n",
    "                    \"equation\":str(rxnobj),\n",
    "                    \"name\":rxnobj.name,\n",
    "                    \"obj\":rxnobj\n",
    "                }\n",
    "    print(sample, f\"{len(rxn_data_hash[sample])} reactions\", end=\"  \")\n",
    "    biomasses[sample] = content[\"solution\"][\"bio1\"]\n",
    "    mag_biomass[sample] = {}\n",
    "    mag_rescaled_biomass[sample] = {}\n",
    "    for mag in mag_list:\n",
    "        magname = mag[1].replace(\".contigs__.RAST\", '')\n",
    "        # print(list(map(type, [sample, magname])))\n",
    "        mag_biomass[sample][magname] = mag_abundances[sample][magname]*biomasses[sample]\n",
    "        mag_rescaled_biomass[sample][magname] = 0 if magname not in mag_abundances[sample] else mag_abundances[sample][magname] \n",
    "    sample_mags = list(mag_abundances[sample].keys())\n",
    "    mag_sample_flux_profiles[sample] = {}\n",
    "    for rxn, flux in content[\"solution\"].items():\n",
    "        if rxn[0:3] == \"EX_\":\n",
    "            sample_interval_exchanges[sample][rxn[3:11]] = content[\"solution\"][rxn]\n",
    "        elif rxn[0:3] == \"rxn\":\n",
    "            (baseid, compartment) = rxn.split(\"_\")\n",
    "            if compartment[0:1] == \"c\":\n",
    "                index = int(rxn.split(\"_\")[1][1:])\n",
    "                magname = sample_mags[index-1].replace(\".contigs__.RAST\", '') \n",
    "                ## store the flux profile for a mag in a given sample\n",
    "                if magname not in mag_sample_flux_profiles[sample]:\n",
    "                    mag_sample_flux_profiles[sample][magname] = {}\n",
    "                total_flux = 0 if magname not in mag_abundances[sample] else flux*mag_abundances[sample][magname]\n",
    "                mag_sample_flux_profiles[sample][magname][baseid] = total_flux\n",
    "            # print(f\"{len(mag_sample_flux_profiles[sample][magname])} fluxes\")\n",
    "    print(f\"{len(mag_sample_flux_profiles[sample])} members\")\n",
    "\n",
    "#Computing metabolite interactions and printing dataframe of scaled mag flux profiles per sample inteval\n",
    "record_hash = {}\n",
    "record_dic = {}\n",
    "records = []\n",
    "indices = []\n",
    "zero_biomass = {}\n",
    "mag_metabolite_interactions = {}\n",
    "for sample, modelFluxes in mag_sample_flux_profiles.items():\n",
    "    mag_metabolite_interactions[sample] = {}\n",
    "    zero_biomass[sample] = []\n",
    "    record_dic[sample] = {}\n",
    "    for magname, fluxes in modelFluxes.items():\n",
    "        mag_metabolite_interactions[sample][magname] = {}\n",
    "        for rxn, flux in fluxes.items():\n",
    "            if rxn not in record_hash:\n",
    "                record_hash[rxn] = {\n",
    "                    \"id\":rxn,\n",
    "                    \"equation\":rxn_data_hash[sample][rxn][\"equation\"],\n",
    "                    \"name\":rxn_data_hash[sample][rxn][\"name\"]\n",
    "                }\n",
    "                records.append(record_hash[rxn])\n",
    "            for met in rxn_data_hash[sample][rxn][\"obj\"].metabolites:\n",
    "                if met.id.split(\"_\")[1] == \"e0\" and met.id.split(\"_\")[0] in metabolites:\n",
    "                    if met.id not in mag_metabolite_interactions[sample][magname]:\n",
    "                        mag_metabolite_interactions[sample][magname][met.id.split(\"_\")[0]] = 0\n",
    "                    if rxn_data_hash[sample][rxn][\"obj\"].metabolites[met] < 0:\n",
    "                        #This transport is a consumption reaction; consumption should be negative\n",
    "                        mag_metabolite_interactions[sample][magname][met.id.split(\"_\")[0]] += -1*flux\n",
    "                    elif rxn_data_hash[sample][rxn][\"obj\"].metabolites[met] > 0:\n",
    "                        #This transport is a excretion reaction; excretion should be positive\n",
    "                        mag_metabolite_interactions[sample][magname][met.id.split(\"_\")[0]] += flux       \n",
    "            if mag_biomass[sample][magname] == 0:\n",
    "                zero_biomass[sample].append(magname)\n",
    "                # continue\n",
    "            record_hash[rxn][magname+\"_\"+sample] = flux #/mag_biomass[sample][magname]\n",
    "            record_dic[sample][rxn] = flux #/mag_biomass[sample][magname]\n",
    "            if rxn not in indices:\n",
    "                indices.append(rxn)\n",
    "\n",
    "print(records[:3])\n",
    "MAGdf = pd.DataFrame.from_records(records).T #, columns=list(mag_sample_flux_profiles.keys())).T\n",
    "# MAGdf = pd.DataFrame(record_dic) #, index=indices)\n",
    "# df.\n",
    "MAGdf.to_csv(util.output_dir+\"/magFluxes.csv\")\n",
    "\n",
    "print(f\"The non-growing biomasses\", zero_biomass)\n",
    "    \n",
    "#Printing mag metabolite interation matrix\n",
    "record_hash = {}\n",
    "records = []\n",
    "for magname, modelMets in mag_metabolite_interactions.items():\n",
    "    new_record = {\"MAG\":magname}\n",
    "    for sample, metInteractions in modelMets.items():\n",
    "        new_record.update({met+\"_\"+sample: flux for met, flux in metInteractions.items()})\n",
    "    for met in metabolites:\n",
    "        total = count = 0\n",
    "        for sample, metInteractions in modelMets.items():\n",
    "            if met in metInteractions:\n",
    "                total += metInteractions[met]\n",
    "                count += 1\n",
    "        if count > 0:\n",
    "            new_record[met+\"_ave\"] = total/count\n",
    "    records.append(new_record)\n",
    "df = pd.DataFrame.from_records(records)\n",
    "df.to_csv(baseDir+\"/magMetaboliteInteractions.csv\",index=False)\n",
    "\n",
    "#Printing dataframe of Sample metabolite interactions per interval and per sample\n",
    "# record_hash = {}\n",
    "# records = []\n",
    "# for sample in sample_interval_exchanges:\n",
    "#     new_record[sample] = {}\n",
    "#     for metabolite in sample_interval_exchanges[sample]:\n",
    "#         new_record[metabolite] = sample_interval_exchanges[sample][metabolite]\n",
    "#     records.append(new_record)\n",
    "# df = pd.DataFrame(records)\n",
    "# df.to_csv(util.output_dir+\"/SampleIntervalMetaboliteInterations.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['SF2_A_D1', 'SF2_A_D2', 'SF2_B_D1', 'SF2_B_D2', 'SF2_C_D1', 'R2A_A_D2',\n",
       "       'R2A_B_D1', 'SF2_C_D2', 'R2A_B_D2', 'R2A_A_D1', 'R2A_C_D1', 'R2A_C_D2',\n",
       "       'R2_A_D1', 'R2_B_D1', 'R2_C_D2', 'R2_A_D2', 'R2_C_D1', 'R2_B_D2',\n",
       "       'R1_C_D2', 'R1_B_D1', 'R1_C_D1', 'R1_A_D2', 'R1_B_D2', 'R1_A_D1'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MAGdf.index"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# creating a COBRApy Interaction Model of each community and member fluxes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from modelseedpy.biochem import from_local\n",
    "\n",
    "msdb = from_local(\"../../ModelSEEDDatabase\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files printed to:/scratch/shared/code/MicrobiomeNotebooks/Cliff/nboutput when using KBDevUtils.output_dir\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1725557487.1637022 INFO: The current solver interface glpk doesn't support setting the optimality tolerance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ModelSEED: /scratch/shared//sdkmount/run_local/workdir/tmp/\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'msdb' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[4], line 34\u001b[0m\n\u001b[1;32m     32\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m mag \u001b[38;5;129;01min\u001b[39;00m mags:\n\u001b[1;32m     33\u001b[0m         magMets[sample][mag] \u001b[38;5;241m=\u001b[39m magMets[sample]\u001b[38;5;241m.\u001b[39mget(mag, {\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mrxn\u001b[39m\u001b[38;5;124m\"\u001b[39m: Reaction(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00msample\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m__\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mmag\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, name\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m), \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmets\u001b[39m\u001b[38;5;124m\"\u001b[39m: {}})\n\u001b[0;32m---> 34\u001b[0m         metObj \u001b[38;5;241m=\u001b[39m Metabolite(met, name\u001b[38;5;241m=\u001b[39m\u001b[43mmsdb\u001b[49m\u001b[38;5;241m.\u001b[39mcompounds\u001b[38;5;241m.\u001b[39mget_by_id(met)\u001b[38;5;241m.\u001b[39mname) \u001b[38;5;66;03m#, compartment='c')   msdb.compounds.get(met)[\"name\"]\u001b[39;00m\n\u001b[1;32m     35\u001b[0m         magMets[sample][mag][\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmets\u001b[39m\u001b[38;5;124m\"\u001b[39m][metObj] \u001b[38;5;241m=\u001b[39m flux\n\u001b[1;32m     36\u001b[0m \u001b[38;5;66;03m# for mag, metFluxes in magMets.items():\u001b[39;00m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'msdb' is not defined"
     ]
    }
   ],
   "source": [
    "%run cliffcommutil.py\n",
    "from cobra import Model, Reaction, Metabolite, io\n",
    "from pandas import read_csv\n",
    "\n",
    "\n",
    "def isNumber(val):\n",
    "    import re\n",
    "    remainder = re.sub(\"[0-9.-]+\", \"\", str(val))\n",
    "    if remainder != \"\": \n",
    "        # print(remainder)\n",
    "        return False\n",
    "    return True\n",
    "\n",
    "# construct the interaction models for each sample community\n",
    "sampleMAGs = util.load(\"refined_MAG_abundances\")\n",
    "maxGrowthPercent = 100\n",
    "baseDir = util.output_dir+f\"/{maxGrowthPercent}\"\n",
    "# display(read_csv(open(util.output_dir+\"/magMetaboliteInteractions.csv\")))\n",
    "MAGdf = read_csv(open(baseDir+\"/magMetaboliteInteractions.csv\")).set_index(\"MAG\")\n",
    "# display(MAGdf)\n",
    "interactionModel = Model()\n",
    "magMets = {}\n",
    "for sample, row in MAGdf.iterrows():\n",
    "    ## add all of the interactions to the sample model\n",
    "    mags = list(sampleMAGs[sample].keys())\n",
    "    magMets[sample] = {}\n",
    "    for metMAG, flux in row.items():\n",
    "        met, mag = metMAG.split(\"_\", 1)\n",
    "        if not isNumber(flux):\n",
    "            # print(mag, met, flux)\n",
    "            continue\n",
    "        if mag in mags:\n",
    "            magMets[sample][mag] = magMets[sample].get(mag, {\"rxn\": Reaction(f\"{sample}__{mag}\", name=\"\"), \"mets\": {}})\n",
    "            metObj = Metabolite(met, name=msdb.compounds.get_by_id(met).name) #, compartment='c')   msdb.compounds.get(met)[\"name\"]\n",
    "            magMets[sample][mag][\"mets\"][metObj] = flux\n",
    "    # for mag, metFluxes in magMets.items():\n",
    "    for mag, content in magMets[sample].items():\n",
    "        content[\"rxn\"].add_metabolites(content[\"mets\"])\n",
    "\n",
    "    ## add reactions to the Interaction model\n",
    "    interactionModel.add_reactions([content[\"rxn\"] for content in magMets[sample].values()])\n",
    "\n",
    "## write the model for visualizing interactions\n",
    "io.save_json_model(interactionModel, baseDir+\"/MAGInteractionModel.json\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Correlate methane flux and methane abundance of all communities\n",
    "\n",
    "## there should be variability among the communities"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Test whether the models can produce methane given methane sources (glycine betaine, MPP, formate)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python3_ModelSEED",
   "language": "python",
   "name": "python3_modelseed"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
